{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\18103939.py:1: DtypeWarning: Columns (2,618) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  raw_methy_df = pd.read_csv('./data/raw_data/CCLE_RRBS_TSS1kb_20181022.txt', sep='\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>locus_id</th>\n",
       "      <th>CpG_sites_hg19</th>\n",
       "      <th>avg_coverage</th>\n",
       "      <th>DMS53_LUNG</th>\n",
       "      <th>SW1116_LARGE_INTESTINE</th>\n",
       "      <th>P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>UMUC3_URINARY_TRACT</th>\n",
       "      <th>HOS_BONE</th>\n",
       "      <th>AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>...</th>\n",
       "      <th>JHUEM7_ENDOMETRIUM</th>\n",
       "      <th>OE21_OESOPHAGUS</th>\n",
       "      <th>MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HOP62_LUNG</th>\n",
       "      <th>EKVX_LUNG</th>\n",
       "      <th>UO31_KIDNEY</th>\n",
       "      <th>SF268_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SF539_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SNB75_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HOP92_LUNG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SGIP1_1_66998638_66999638</td>\n",
       "      <td>1:66998970;1:66998973;1:66998993;1:66999404;1:...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.75000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.95239</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.03448</td>\n",
       "      <td>0.77660</td>\n",
       "      <td>0.06665</td>\n",
       "      <td>0.66667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SGIP1_1_66998251_66999251</td>\n",
       "      <td>1:66998970;1:66998973;1:66998993</td>\n",
       "      <td>8.27</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.94443</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.66667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AZIN2_1_33545713_33546713</td>\n",
       "      <td>1:33546151;1:33546209;1:33546210;1:33546385;1:...</td>\n",
       "      <td>326.58</td>\n",
       "      <td>0.00729</td>\n",
       "      <td>0.03232</td>\n",
       "      <td>0.33160</td>\n",
       "      <td>0.22042</td>\n",
       "      <td>0.01291</td>\n",
       "      <td>0.03444</td>\n",
       "      <td>0.14411</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.04821</td>\n",
       "      <td>0.06721</td>\n",
       "      <td>0.11539</td>\n",
       "      <td>0.00046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AZIN2_1_33546778_33547778</td>\n",
       "      <td>1:33546783;1:33546788;1:33546795;1:33546797;1:...</td>\n",
       "      <td>480.54</td>\n",
       "      <td>0.22276</td>\n",
       "      <td>0.07784</td>\n",
       "      <td>0.26034</td>\n",
       "      <td>0.32106</td>\n",
       "      <td>0.27485</td>\n",
       "      <td>0.15940</td>\n",
       "      <td>0.34742</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.15045</td>\n",
       "      <td>0.34739</td>\n",
       "      <td>0.30208</td>\n",
       "      <td>0.05159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AGBL4_1_50489626_50490626</td>\n",
       "      <td>1:50489632;1:50489641;1:50489671;1:50489677;1:...</td>\n",
       "      <td>263.36</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01216</td>\n",
       "      <td>0.98113</td>\n",
       "      <td>0.56622</td>\n",
       "      <td>0.33960</td>\n",
       "      <td>0.43177</td>\n",
       "      <td>0.93363</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.01874</td>\n",
       "      <td>0.19714</td>\n",
       "      <td>0.05052</td>\n",
       "      <td>0.54553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21333</th>\n",
       "      <td>SYCE3_22_51001381_51002381</td>\n",
       "      <td>22:51001382;22:51001388;22:51001398;22:5100139...</td>\n",
       "      <td>359.68</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.36057</td>\n",
       "      <td>0.19895</td>\n",
       "      <td>0.10039</td>\n",
       "      <td>0.11989</td>\n",
       "      <td>0.02138</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.10934</td>\n",
       "      <td>0.01969</td>\n",
       "      <td>0.02364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21334</th>\n",
       "      <td>RABL2B_22_51222110_51223110</td>\n",
       "      <td>22:51222730</td>\n",
       "      <td>6.38</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.09090</td>\n",
       "      <td>0.57140</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21335</th>\n",
       "      <td>RPL23AP82_22_51194513_51195513</td>\n",
       "      <td>22:51195083;22:51195404;22:51195430</td>\n",
       "      <td>62.10</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.07277</td>\n",
       "      <td>0.00263</td>\n",
       "      <td>0.26433</td>\n",
       "      <td>0.04521</td>\n",
       "      <td>0.21740</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.08333</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.25713</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21336</th>\n",
       "      <td>RPL23AP82_22_51221156_51222156</td>\n",
       "      <td>22:51222010;22:51222012;22:51222018;22:5122202...</td>\n",
       "      <td>179.18</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21337</th>\n",
       "      <td>NA_NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21338 rows × 846 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             locus_id  \\\n",
       "0           SGIP1_1_66998638_66999638   \n",
       "1           SGIP1_1_66998251_66999251   \n",
       "2           AZIN2_1_33545713_33546713   \n",
       "3           AZIN2_1_33546778_33547778   \n",
       "4           AGBL4_1_50489626_50490626   \n",
       "...                               ...   \n",
       "21333      SYCE3_22_51001381_51002381   \n",
       "21334     RABL2B_22_51222110_51223110   \n",
       "21335  RPL23AP82_22_51194513_51195513   \n",
       "21336  RPL23AP82_22_51221156_51222156   \n",
       "21337                           NA_NA   \n",
       "\n",
       "                                          CpG_sites_hg19 avg_coverage  \\\n",
       "0      1:66998970;1:66998973;1:66998993;1:66999404;1:...         25.0   \n",
       "1                       1:66998970;1:66998973;1:66998993         8.27   \n",
       "2      1:33546151;1:33546209;1:33546210;1:33546385;1:...       326.58   \n",
       "3      1:33546783;1:33546788;1:33546795;1:33546797;1:...       480.54   \n",
       "4      1:50489632;1:50489641;1:50489671;1:50489677;1:...       263.36   \n",
       "...                                                  ...          ...   \n",
       "21333  22:51001382;22:51001388;22:51001398;22:5100139...       359.68   \n",
       "21334                                        22:51222730         6.38   \n",
       "21335                22:51195083;22:51195404;22:51195430        62.10   \n",
       "21336  22:51222010;22:51222012;22:51222018;22:5122202...       179.18   \n",
       "21337                                                NaN           NA   \n",
       "\n",
       "      DMS53_LUNG SW1116_LARGE_INTESTINE  \\\n",
       "0        0.00000                0.00000   \n",
       "1        0.00000                    NaN   \n",
       "2        0.00729                0.03232   \n",
       "3        0.22276                0.07784   \n",
       "4        0.00000                0.01216   \n",
       "...          ...                    ...   \n",
       "21333    0.00000                0.00000   \n",
       "21334    0.00000                0.00000   \n",
       "21335    0.00000                0.00000   \n",
       "21336    0.00000                0.00000   \n",
       "21337         NA                     NA   \n",
       "\n",
       "      P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0                                      0.75000   \n",
       "1                                          NaN   \n",
       "2                                      0.33160   \n",
       "3                                      0.26034   \n",
       "4                                      0.98113   \n",
       "...                                        ...   \n",
       "21333                                  0.36057   \n",
       "21334                                  0.00000   \n",
       "21335                                  0.07277   \n",
       "21336                                  0.00000   \n",
       "21337                                       NA   \n",
       "\n",
       "      HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE UMUC3_URINARY_TRACT HOS_BONE  \\\n",
       "0                                      0.00000             0.95239      NaN   \n",
       "1                                      0.00000             0.94443      NaN   \n",
       "2                                      0.22042             0.01291  0.03444   \n",
       "3                                      0.32106             0.27485  0.15940   \n",
       "4                                      0.56622             0.33960  0.43177   \n",
       "...                                        ...                 ...      ...   \n",
       "21333                                  0.19895             0.10039  0.11989   \n",
       "21334                                  0.09090             0.57140  0.00000   \n",
       "21335                                  0.00263             0.26433  0.04521   \n",
       "21336                                  0.00000             0.00000  0.00274   \n",
       "21337                                       NA                  NA       NA   \n",
       "\n",
       "      AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ... JHUEM7_ENDOMETRIUM  \\\n",
       "0                                       1.00000  ...                 NA   \n",
       "1                                           NaN  ...                 NA   \n",
       "2                                       0.14411  ...                 NA   \n",
       "3                                       0.34742  ...                 NA   \n",
       "4                                       0.93363  ...                 NA   \n",
       "...                                         ...  ...                ...   \n",
       "21333                                   0.02138  ...                 NA   \n",
       "21334                                   0.00000  ...                 NA   \n",
       "21335                                   0.21740  ...                 NA   \n",
       "21336                                   0.00000  ...                 NA   \n",
       "21337                                        NA  ...                 NA   \n",
       "\n",
       "      OE21_OESOPHAGUS MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE HOP62_LUNG  \\\n",
       "0                  NA                                       NA         NA   \n",
       "1                  NA                                       NA         NA   \n",
       "2                  NA                                       NA         NA   \n",
       "3                  NA                                       NA         NA   \n",
       "4                  NA                                       NA         NA   \n",
       "...               ...                                      ...        ...   \n",
       "21333              NA                                       NA         NA   \n",
       "21334              NA                                       NA         NA   \n",
       "21335              NA                                       NA         NA   \n",
       "21336              NA                                       NA         NA   \n",
       "21337              NA                                       NA         NA   \n",
       "\n",
       "      EKVX_LUNG UO31_KIDNEY SF268_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "0            NA          NA                      0.03448   \n",
       "1            NA          NA                      0.00000   \n",
       "2            NA          NA                      0.04821   \n",
       "3            NA          NA                      0.15045   \n",
       "4            NA          NA                      0.01874   \n",
       "...         ...         ...                          ...   \n",
       "21333        NA          NA                      0.00000   \n",
       "21334        NA          NA                      0.00000   \n",
       "21335        NA          NA                      0.08333   \n",
       "21336        NA          NA                      0.00000   \n",
       "21337        NA          NA                           NA   \n",
       "\n",
       "      SF539_CENTRAL_NERVOUS_SYSTEM SNB75_CENTRAL_NERVOUS_SYSTEM HOP92_LUNG  \n",
       "0                          0.77660                      0.06665    0.66667  \n",
       "1                              NaN                          NaN    0.66667  \n",
       "2                          0.06721                      0.11539    0.00046  \n",
       "3                          0.34739                      0.30208    0.05159  \n",
       "4                          0.19714                      0.05052    0.54553  \n",
       "...                            ...                          ...        ...  \n",
       "21333                      0.10934                      0.01969    0.02364  \n",
       "21334                      0.00000                      0.00000    0.00000  \n",
       "21335                      0.00000                      0.25713    0.00000  \n",
       "21336                          NaN                      0.00000    0.00000  \n",
       "21337                           NA                           NA         NA  \n",
       "\n",
       "[21338 rows x 846 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "raw_methy_df = pd.read_csv('./data/raw_data/CCLE_RRBS_TSS1kb_20181022.txt', sep='\\t')\n",
    "display(raw_methy_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>locus_id</th>\n",
       "      <th>DMS53_LUNG</th>\n",
       "      <th>SW1116_LARGE_INTESTINE</th>\n",
       "      <th>P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>UMUC3_URINARY_TRACT</th>\n",
       "      <th>HOS_BONE</th>\n",
       "      <th>AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>RVH421_SKIN</th>\n",
       "      <th>NCIH1184_LUNG</th>\n",
       "      <th>...</th>\n",
       "      <th>JHUEM7_ENDOMETRIUM</th>\n",
       "      <th>OE21_OESOPHAGUS</th>\n",
       "      <th>MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HOP62_LUNG</th>\n",
       "      <th>EKVX_LUNG</th>\n",
       "      <th>UO31_KIDNEY</th>\n",
       "      <th>SF268_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SF539_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SNB75_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HOP92_LUNG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SGIP1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.75000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.95239</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.11864</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.03448</td>\n",
       "      <td>0.77660</td>\n",
       "      <td>0.06665</td>\n",
       "      <td>0.66667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SGIP1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.94443</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22579</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.66667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AZIN2</td>\n",
       "      <td>0.00729</td>\n",
       "      <td>0.03232</td>\n",
       "      <td>0.33160</td>\n",
       "      <td>0.22042</td>\n",
       "      <td>0.01291</td>\n",
       "      <td>0.03444</td>\n",
       "      <td>0.14411</td>\n",
       "      <td>0.02525</td>\n",
       "      <td>0.22553</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.04821</td>\n",
       "      <td>0.06721</td>\n",
       "      <td>0.11539</td>\n",
       "      <td>0.00046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AZIN2</td>\n",
       "      <td>0.22276</td>\n",
       "      <td>0.07784</td>\n",
       "      <td>0.26034</td>\n",
       "      <td>0.32106</td>\n",
       "      <td>0.27485</td>\n",
       "      <td>0.15940</td>\n",
       "      <td>0.34742</td>\n",
       "      <td>0.28963</td>\n",
       "      <td>0.10735</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.15045</td>\n",
       "      <td>0.34739</td>\n",
       "      <td>0.30208</td>\n",
       "      <td>0.05159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AGBL4</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01216</td>\n",
       "      <td>0.98113</td>\n",
       "      <td>0.56622</td>\n",
       "      <td>0.33960</td>\n",
       "      <td>0.43177</td>\n",
       "      <td>0.93363</td>\n",
       "      <td>0.13089</td>\n",
       "      <td>0.07346</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.01874</td>\n",
       "      <td>0.19714</td>\n",
       "      <td>0.05052</td>\n",
       "      <td>0.54553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21332</th>\n",
       "      <td>CHKB</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00951</td>\n",
       "      <td>0.00552</td>\n",
       "      <td>0.00079</td>\n",
       "      <td>0.04100</td>\n",
       "      <td>0.01047</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00663</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21333</th>\n",
       "      <td>SYCE3</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.36057</td>\n",
       "      <td>0.19895</td>\n",
       "      <td>0.10039</td>\n",
       "      <td>0.11989</td>\n",
       "      <td>0.02138</td>\n",
       "      <td>0.04334</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.10934</td>\n",
       "      <td>0.01969</td>\n",
       "      <td>0.02364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21334</th>\n",
       "      <td>RABL2B</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.09090</td>\n",
       "      <td>0.57140</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21335</th>\n",
       "      <td>RPL23AP82</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.07277</td>\n",
       "      <td>0.00263</td>\n",
       "      <td>0.26433</td>\n",
       "      <td>0.04521</td>\n",
       "      <td>0.21740</td>\n",
       "      <td>0.06030</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.08333</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.25713</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21336</th>\n",
       "      <td>RPL23AP82</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00599</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21337 rows × 844 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        locus_id DMS53_LUNG SW1116_LARGE_INTESTINE  \\\n",
       "0          SGIP1    0.00000                0.00000   \n",
       "1          SGIP1    0.00000                    NaN   \n",
       "2          AZIN2    0.00729                0.03232   \n",
       "3          AZIN2    0.22276                0.07784   \n",
       "4          AGBL4    0.00000                0.01216   \n",
       "...          ...        ...                    ...   \n",
       "21332       CHKB    0.00000                0.00951   \n",
       "21333      SYCE3    0.00000                0.00000   \n",
       "21334     RABL2B    0.00000                0.00000   \n",
       "21335  RPL23AP82    0.00000                0.00000   \n",
       "21336  RPL23AP82    0.00000                0.00000   \n",
       "\n",
       "      P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0                                      0.75000   \n",
       "1                                          NaN   \n",
       "2                                      0.33160   \n",
       "3                                      0.26034   \n",
       "4                                      0.98113   \n",
       "...                                        ...   \n",
       "21332                                  0.00552   \n",
       "21333                                  0.36057   \n",
       "21334                                  0.00000   \n",
       "21335                                  0.07277   \n",
       "21336                                  0.00000   \n",
       "\n",
       "      HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE UMUC3_URINARY_TRACT HOS_BONE  \\\n",
       "0                                      0.00000             0.95239      NaN   \n",
       "1                                      0.00000             0.94443      NaN   \n",
       "2                                      0.22042             0.01291  0.03444   \n",
       "3                                      0.32106             0.27485  0.15940   \n",
       "4                                      0.56622             0.33960  0.43177   \n",
       "...                                        ...                 ...      ...   \n",
       "21332                                  0.00079             0.04100  0.01047   \n",
       "21333                                  0.19895             0.10039  0.11989   \n",
       "21334                                  0.09090             0.57140  0.00000   \n",
       "21335                                  0.00263             0.26433  0.04521   \n",
       "21336                                  0.00000             0.00000  0.00274   \n",
       "\n",
       "      AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE RVH421_SKIN NCIH1184_LUNG  \\\n",
       "0                                       1.00000         NaN       0.11864   \n",
       "1                                           NaN         NaN       0.22579   \n",
       "2                                       0.14411     0.02525       0.22553   \n",
       "3                                       0.34742     0.28963       0.10735   \n",
       "4                                       0.93363     0.13089       0.07346   \n",
       "...                                         ...         ...           ...   \n",
       "21332                                   0.00000     0.00663       0.00000   \n",
       "21333                                   0.02138     0.04334       0.00000   \n",
       "21334                                   0.00000     0.00000       0.00000   \n",
       "21335                                   0.21740     0.06030       0.00000   \n",
       "21336                                   0.00000     0.00599       0.00000   \n",
       "\n",
       "       ... JHUEM7_ENDOMETRIUM OE21_OESOPHAGUS  \\\n",
       "0      ...                 NA              NA   \n",
       "1      ...                 NA              NA   \n",
       "2      ...                 NA              NA   \n",
       "3      ...                 NA              NA   \n",
       "4      ...                 NA              NA   \n",
       "...    ...                ...             ...   \n",
       "21332  ...                 NA              NA   \n",
       "21333  ...                 NA              NA   \n",
       "21334  ...                 NA              NA   \n",
       "21335  ...                 NA              NA   \n",
       "21336  ...                 NA              NA   \n",
       "\n",
       "      MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE HOP62_LUNG EKVX_LUNG  \\\n",
       "0                                           NA         NA        NA   \n",
       "1                                           NA         NA        NA   \n",
       "2                                           NA         NA        NA   \n",
       "3                                           NA         NA        NA   \n",
       "4                                           NA         NA        NA   \n",
       "...                                        ...        ...       ...   \n",
       "21332                                       NA         NA        NA   \n",
       "21333                                       NA         NA        NA   \n",
       "21334                                       NA         NA        NA   \n",
       "21335                                       NA         NA        NA   \n",
       "21336                                       NA         NA        NA   \n",
       "\n",
       "      UO31_KIDNEY SF268_CENTRAL_NERVOUS_SYSTEM SF539_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "0              NA                      0.03448                      0.77660   \n",
       "1              NA                      0.00000                          NaN   \n",
       "2              NA                      0.04821                      0.06721   \n",
       "3              NA                      0.15045                      0.34739   \n",
       "4              NA                      0.01874                      0.19714   \n",
       "...           ...                          ...                          ...   \n",
       "21332          NA                      0.00000                      0.00000   \n",
       "21333          NA                      0.00000                      0.10934   \n",
       "21334          NA                      0.00000                      0.00000   \n",
       "21335          NA                      0.08333                      0.00000   \n",
       "21336          NA                      0.00000                          NaN   \n",
       "\n",
       "      SNB75_CENTRAL_NERVOUS_SYSTEM HOP92_LUNG  \n",
       "0                          0.06665    0.66667  \n",
       "1                              NaN    0.66667  \n",
       "2                          0.11539    0.00046  \n",
       "3                          0.30208    0.05159  \n",
       "4                          0.05052    0.54553  \n",
       "...                            ...        ...  \n",
       "21332                      0.00000    0.00000  \n",
       "21333                      0.01969    0.02364  \n",
       "21334                      0.00000    0.00000  \n",
       "21335                      0.25713    0.00000  \n",
       "21336                      0.00000    0.00000  \n",
       "\n",
       "[21337 rows x 844 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# column locus_id should be separate the content after the  '_'\n",
    "raw_methy_df['locus_id'] = raw_methy_df['locus_id'].apply(lambda x: x.split('_')[0])\n",
    "# drop na in the column locus_id\n",
    "raw_methy_df = raw_methy_df.dropna(subset=['CpG_sites_hg19'])\n",
    "# remove the column CpG_sites_hg19 and avg_coverage\n",
    "raw_methy_df = raw_methy_df.drop(columns=['CpG_sites_hg19', 'avg_coverage'])\n",
    "display(raw_methy_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>locus_id</th>\n",
       "      <th>DMS53_LUNG</th>\n",
       "      <th>SW1116_LARGE_INTESTINE</th>\n",
       "      <th>P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>UMUC3_URINARY_TRACT</th>\n",
       "      <th>HOS_BONE</th>\n",
       "      <th>AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>RVH421_SKIN</th>\n",
       "      <th>NCIH1184_LUNG</th>\n",
       "      <th>...</th>\n",
       "      <th>JHUEM7_ENDOMETRIUM</th>\n",
       "      <th>OE21_OESOPHAGUS</th>\n",
       "      <th>MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HOP62_LUNG</th>\n",
       "      <th>EKVX_LUNG</th>\n",
       "      <th>UO31_KIDNEY</th>\n",
       "      <th>SF268_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SF539_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SNB75_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HOP92_LUNG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SGIP1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.75000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.95239</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.11864</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.03448</td>\n",
       "      <td>0.77660</td>\n",
       "      <td>0.06665</td>\n",
       "      <td>0.66667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SGIP1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.94443</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.22579</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.66667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AZIN2</td>\n",
       "      <td>0.00729</td>\n",
       "      <td>0.03232</td>\n",
       "      <td>0.33160</td>\n",
       "      <td>0.22042</td>\n",
       "      <td>0.01291</td>\n",
       "      <td>0.03444</td>\n",
       "      <td>0.14411</td>\n",
       "      <td>0.02525</td>\n",
       "      <td>0.22553</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.04821</td>\n",
       "      <td>0.06721</td>\n",
       "      <td>0.11539</td>\n",
       "      <td>0.00046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AZIN2</td>\n",
       "      <td>0.22276</td>\n",
       "      <td>0.07784</td>\n",
       "      <td>0.26034</td>\n",
       "      <td>0.32106</td>\n",
       "      <td>0.27485</td>\n",
       "      <td>0.15940</td>\n",
       "      <td>0.34742</td>\n",
       "      <td>0.28963</td>\n",
       "      <td>0.10735</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.15045</td>\n",
       "      <td>0.34739</td>\n",
       "      <td>0.30208</td>\n",
       "      <td>0.05159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AGBL4</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01216</td>\n",
       "      <td>0.98113</td>\n",
       "      <td>0.56622</td>\n",
       "      <td>0.33960</td>\n",
       "      <td>0.43177</td>\n",
       "      <td>0.93363</td>\n",
       "      <td>0.13089</td>\n",
       "      <td>0.07346</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01874</td>\n",
       "      <td>0.19714</td>\n",
       "      <td>0.05052</td>\n",
       "      <td>0.54553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21332</th>\n",
       "      <td>CHKB</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00951</td>\n",
       "      <td>0.00552</td>\n",
       "      <td>0.00079</td>\n",
       "      <td>0.04100</td>\n",
       "      <td>0.01047</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00663</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21333</th>\n",
       "      <td>SYCE3</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.36057</td>\n",
       "      <td>0.19895</td>\n",
       "      <td>0.10039</td>\n",
       "      <td>0.11989</td>\n",
       "      <td>0.02138</td>\n",
       "      <td>0.04334</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.10934</td>\n",
       "      <td>0.01969</td>\n",
       "      <td>0.02364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21334</th>\n",
       "      <td>RABL2B</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.09090</td>\n",
       "      <td>0.57140</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21335</th>\n",
       "      <td>RPL23AP82</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.07277</td>\n",
       "      <td>0.00263</td>\n",
       "      <td>0.26433</td>\n",
       "      <td>0.04521</td>\n",
       "      <td>0.21740</td>\n",
       "      <td>0.06030</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.08333</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.25713</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21336</th>\n",
       "      <td>RPL23AP82</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00599</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21337 rows × 844 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        locus_id  DMS53_LUNG  SW1116_LARGE_INTESTINE  \\\n",
       "0          SGIP1     0.00000                 0.00000   \n",
       "1          SGIP1     0.00000                 0.00000   \n",
       "2          AZIN2     0.00729                 0.03232   \n",
       "3          AZIN2     0.22276                 0.07784   \n",
       "4          AGBL4     0.00000                 0.01216   \n",
       "...          ...         ...                     ...   \n",
       "21332       CHKB     0.00000                 0.00951   \n",
       "21333      SYCE3     0.00000                 0.00000   \n",
       "21334     RABL2B     0.00000                 0.00000   \n",
       "21335  RPL23AP82     0.00000                 0.00000   \n",
       "21336  RPL23AP82     0.00000                 0.00000   \n",
       "\n",
       "       P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0                                       0.75000   \n",
       "1                                       0.00000   \n",
       "2                                       0.33160   \n",
       "3                                       0.26034   \n",
       "4                                       0.98113   \n",
       "...                                         ...   \n",
       "21332                                   0.00552   \n",
       "21333                                   0.36057   \n",
       "21334                                   0.00000   \n",
       "21335                                   0.07277   \n",
       "21336                                   0.00000   \n",
       "\n",
       "       HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  UMUC3_URINARY_TRACT  \\\n",
       "0                                       0.00000              0.95239   \n",
       "1                                       0.00000              0.94443   \n",
       "2                                       0.22042              0.01291   \n",
       "3                                       0.32106              0.27485   \n",
       "4                                       0.56622              0.33960   \n",
       "...                                         ...                  ...   \n",
       "21332                                   0.00079              0.04100   \n",
       "21333                                   0.19895              0.10039   \n",
       "21334                                   0.09090              0.57140   \n",
       "21335                                   0.00263              0.26433   \n",
       "21336                                   0.00000              0.00000   \n",
       "\n",
       "       HOS_BONE  AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  RVH421_SKIN  \\\n",
       "0       0.00000                                    1.00000      0.00000   \n",
       "1       0.00000                                    0.00000      0.00000   \n",
       "2       0.03444                                    0.14411      0.02525   \n",
       "3       0.15940                                    0.34742      0.28963   \n",
       "4       0.43177                                    0.93363      0.13089   \n",
       "...         ...                                        ...          ...   \n",
       "21332   0.01047                                    0.00000      0.00663   \n",
       "21333   0.11989                                    0.02138      0.04334   \n",
       "21334   0.00000                                    0.00000      0.00000   \n",
       "21335   0.04521                                    0.21740      0.06030   \n",
       "21336   0.00274                                    0.00000      0.00599   \n",
       "\n",
       "       NCIH1184_LUNG  ...  JHUEM7_ENDOMETRIUM  OE21_OESOPHAGUS  \\\n",
       "0            0.11864  ...                 0.0              0.0   \n",
       "1            0.22579  ...                 0.0              0.0   \n",
       "2            0.22553  ...                 0.0              0.0   \n",
       "3            0.10735  ...                 0.0              0.0   \n",
       "4            0.07346  ...                 0.0              0.0   \n",
       "...              ...  ...                 ...              ...   \n",
       "21332        0.00000  ...                 0.0              0.0   \n",
       "21333        0.00000  ...                 0.0              0.0   \n",
       "21334        0.00000  ...                 0.0              0.0   \n",
       "21335        0.00000  ...                 0.0              0.0   \n",
       "21336        0.00000  ...                 0.0              0.0   \n",
       "\n",
       "       MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  HOP62_LUNG  EKVX_LUNG  \\\n",
       "0                                           0.0         0.0        0.0   \n",
       "1                                           0.0         0.0        0.0   \n",
       "2                                           0.0         0.0        0.0   \n",
       "3                                           0.0         0.0        0.0   \n",
       "4                                           0.0         0.0        0.0   \n",
       "...                                         ...         ...        ...   \n",
       "21332                                       0.0         0.0        0.0   \n",
       "21333                                       0.0         0.0        0.0   \n",
       "21334                                       0.0         0.0        0.0   \n",
       "21335                                       0.0         0.0        0.0   \n",
       "21336                                       0.0         0.0        0.0   \n",
       "\n",
       "       UO31_KIDNEY  SF268_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "0              0.0                       0.03448   \n",
       "1              0.0                       0.00000   \n",
       "2              0.0                       0.04821   \n",
       "3              0.0                       0.15045   \n",
       "4              0.0                       0.01874   \n",
       "...            ...                           ...   \n",
       "21332          0.0                       0.00000   \n",
       "21333          0.0                       0.00000   \n",
       "21334          0.0                       0.00000   \n",
       "21335          0.0                       0.08333   \n",
       "21336          0.0                       0.00000   \n",
       "\n",
       "       SF539_CENTRAL_NERVOUS_SYSTEM  SNB75_CENTRAL_NERVOUS_SYSTEM  HOP92_LUNG  \n",
       "0                           0.77660                       0.06665     0.66667  \n",
       "1                           0.00000                       0.00000     0.66667  \n",
       "2                           0.06721                       0.11539     0.00046  \n",
       "3                           0.34739                       0.30208     0.05159  \n",
       "4                           0.19714                       0.05052     0.54553  \n",
       "...                             ...                           ...         ...  \n",
       "21332                       0.00000                       0.00000     0.00000  \n",
       "21333                       0.10934                       0.01969     0.02364  \n",
       "21334                       0.00000                       0.00000     0.00000  \n",
       "21335                       0.00000                       0.25713     0.00000  \n",
       "21336                       0.00000                       0.00000     0.00000  \n",
       "\n",
       "[21337 rows x 844 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Identify columns to convert (all except 'locus_id')\n",
    "cols_to_convert = raw_methy_df.columns.difference(['locus_id'])\n",
    "# Convert these columns to numeric (using pd.to_numeric for safety, which can handle errors)\n",
    "raw_methy_df[cols_to_convert] = raw_methy_df[cols_to_convert].apply(pd.to_numeric, errors='coerce')\n",
    "raw_methy_df = raw_methy_df.fillna(0.0)\n",
    "display(raw_methy_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3346799322.py:2: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  methy_df = raw_methy_df.groupby('locus_id', as_index=False).mean()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>locus_id</th>\n",
       "      <th>DMS53_LUNG</th>\n",
       "      <th>SW1116_LARGE_INTESTINE</th>\n",
       "      <th>P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>UMUC3_URINARY_TRACT</th>\n",
       "      <th>HOS_BONE</th>\n",
       "      <th>AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>RVH421_SKIN</th>\n",
       "      <th>NCIH1184_LUNG</th>\n",
       "      <th>...</th>\n",
       "      <th>JHUEM7_ENDOMETRIUM</th>\n",
       "      <th>OE21_OESOPHAGUS</th>\n",
       "      <th>MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HOP62_LUNG</th>\n",
       "      <th>EKVX_LUNG</th>\n",
       "      <th>UO31_KIDNEY</th>\n",
       "      <th>SF268_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SF539_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SNB75_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HOP92_LUNG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.51504</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.98288</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.55295</td>\n",
       "      <td>0.87010</td>\n",
       "      <td>0.90770</td>\n",
       "      <td>0.86540</td>\n",
       "      <td>0.71970</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.89135</td>\n",
       "      <td>0.28806</td>\n",
       "      <td>0.50752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>0.59476</td>\n",
       "      <td>0.44440</td>\n",
       "      <td>0.96342</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.77274</td>\n",
       "      <td>0.88889</td>\n",
       "      <td>0.97580</td>\n",
       "      <td>0.94286</td>\n",
       "      <td>0.88976</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.17722</td>\n",
       "      <td>0.92126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M-AS1</td>\n",
       "      <td>0.00442</td>\n",
       "      <td>0.00645</td>\n",
       "      <td>0.95503</td>\n",
       "      <td>0.92654</td>\n",
       "      <td>0.53876</td>\n",
       "      <td>0.07932</td>\n",
       "      <td>0.95646</td>\n",
       "      <td>0.22245</td>\n",
       "      <td>0.13635</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.4363</td>\n",
       "      <td>0.00985</td>\n",
       "      <td>0.02073</td>\n",
       "      <td>0.49632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.16665</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.25580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01401</td>\n",
       "      <td>0.00806</td>\n",
       "      <td>0.22223</td>\n",
       "      <td>0.00504</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.76363</td>\n",
       "      <td>0.16626</td>\n",
       "      <td>0.03705</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.7493</td>\n",
       "      <td>0.00106</td>\n",
       "      <td>0.08999</td>\n",
       "      <td>0.02828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17175</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.52940</td>\n",
       "      <td>0.83335</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01725</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.98000</td>\n",
       "      <td>0.04285</td>\n",
       "      <td>0.03330</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.32245</td>\n",
       "      <td>0.44000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17176</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.02290</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01404</td>\n",
       "      <td>0.00548</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.02240</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17177</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.00203</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.07789</td>\n",
       "      <td>0.00331</td>\n",
       "      <td>0.00101</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01156</td>\n",
       "      <td>0.00031</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17178</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.01010</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.03783</td>\n",
       "      <td>0.10369</td>\n",
       "      <td>0.04651</td>\n",
       "      <td>0.12195</td>\n",
       "      <td>0.01603</td>\n",
       "      <td>0.02255</td>\n",
       "      <td>0.02312</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.05190</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17179</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01905</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00379</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17180 rows × 844 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       locus_id  DMS53_LUNG  SW1116_LARGE_INTESTINE  \\\n",
       "0          A1BG     0.51504                 0.50000   \n",
       "1      A1BG-AS1     0.59476                 0.44440   \n",
       "2       A2M-AS1     0.00442                 0.00645   \n",
       "3       A3GALT2     0.00000                 0.16665   \n",
       "4        A4GALT     0.00000                 0.01401   \n",
       "...         ...         ...                     ...   \n",
       "17175    ZYG11A     0.00000                 0.52940   \n",
       "17176    ZYG11B     0.02290                 0.00000   \n",
       "17177       ZYX     0.00203                 0.00000   \n",
       "17178     ZZEF1     0.01010                 0.00000   \n",
       "17179      ZZZ3     0.00000                 0.00000   \n",
       "\n",
       "       P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0                                       0.98288   \n",
       "1                                       0.96342   \n",
       "2                                       0.95503   \n",
       "3                                       0.00000   \n",
       "4                                       0.00806   \n",
       "...                                         ...   \n",
       "17175                                   0.83335   \n",
       "17176                                   0.01404   \n",
       "17177                                   0.07789   \n",
       "17178                                   0.03783   \n",
       "17179                                   0.01905   \n",
       "\n",
       "       HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  UMUC3_URINARY_TRACT  \\\n",
       "0                                       1.00000              0.55295   \n",
       "1                                       1.00000              0.77274   \n",
       "2                                       0.92654              0.53876   \n",
       "3                                       0.00000              0.00000   \n",
       "4                                       0.22223              0.00504   \n",
       "...                                         ...                  ...   \n",
       "17175                                   0.00000              0.01725   \n",
       "17176                                   0.00548              0.00000   \n",
       "17177                                   0.00331              0.00101   \n",
       "17178                                   0.10369              0.04651   \n",
       "17179                                   0.00000              0.00379   \n",
       "\n",
       "       HOS_BONE  AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  RVH421_SKIN  \\\n",
       "0       0.87010                                    0.90770      0.86540   \n",
       "1       0.88889                                    0.97580      0.94286   \n",
       "2       0.07932                                    0.95646      0.22245   \n",
       "3       0.00000                                    1.00000      0.00000   \n",
       "4       0.00000                                    0.76363      0.16626   \n",
       "...         ...                                        ...          ...   \n",
       "17175   0.00000                                    0.98000      0.04285   \n",
       "17176   0.02240                                    0.00000      0.00000   \n",
       "17177   0.00000                                    0.00000      0.01156   \n",
       "17178   0.12195                                    0.01603      0.02255   \n",
       "17179   0.00000                                    0.00000      0.00000   \n",
       "\n",
       "       NCIH1184_LUNG  ...  JHUEM7_ENDOMETRIUM  OE21_OESOPHAGUS  \\\n",
       "0            0.71970  ...                 0.0              0.0   \n",
       "1            0.88976  ...                 0.0              0.0   \n",
       "2            0.13635  ...                 0.0              0.0   \n",
       "3            0.00000  ...                 0.0              0.0   \n",
       "4            0.03705  ...                 0.0              0.0   \n",
       "...              ...  ...                 ...              ...   \n",
       "17175        0.03330  ...                 0.0              0.0   \n",
       "17176        0.00000  ...                 0.0              0.0   \n",
       "17177        0.00031  ...                 0.0              0.0   \n",
       "17178        0.02312  ...                 0.0              0.0   \n",
       "17179        0.00000  ...                 0.0              0.0   \n",
       "\n",
       "       MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  HOP62_LUNG  EKVX_LUNG  \\\n",
       "0                                           0.0         0.0        0.0   \n",
       "1                                           0.0         0.0        0.0   \n",
       "2                                           0.0         0.0        0.0   \n",
       "3                                           0.0         0.0        0.0   \n",
       "4                                           0.0         0.0        0.0   \n",
       "...                                         ...         ...        ...   \n",
       "17175                                       0.0         0.0        0.0   \n",
       "17176                                       0.0         0.0        0.0   \n",
       "17177                                       0.0         0.0        0.0   \n",
       "17178                                       0.0         0.0        0.0   \n",
       "17179                                       0.0         0.0        0.0   \n",
       "\n",
       "       UO31_KIDNEY  SF268_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "0              0.0                        0.5000   \n",
       "1              0.0                        1.0000   \n",
       "2              0.0                        0.4363   \n",
       "3              0.0                        1.0000   \n",
       "4              0.0                        0.7493   \n",
       "...            ...                           ...   \n",
       "17175          0.0                        1.0000   \n",
       "17176          0.0                        0.0000   \n",
       "17177          0.0                        0.0000   \n",
       "17178          0.0                        0.0000   \n",
       "17179          0.0                        0.0000   \n",
       "\n",
       "       SF539_CENTRAL_NERVOUS_SYSTEM  SNB75_CENTRAL_NERVOUS_SYSTEM  HOP92_LUNG  \n",
       "0                           0.89135                       0.28806     0.50752  \n",
       "1                           1.00000                       0.17722     0.92126  \n",
       "2                           0.00985                       0.02073     0.49632  \n",
       "3                           1.00000                       0.00000     0.25580  \n",
       "4                           0.00106                       0.08999     0.02828  \n",
       "...                             ...                           ...         ...  \n",
       "17175                       0.32245                       0.44000     0.00000  \n",
       "17176                       0.00000                       0.00000     0.00000  \n",
       "17177                       0.00000                       0.00000     0.00273  \n",
       "17178                       0.05190                       0.00000     0.00552  \n",
       "17179                       0.00000                       0.00000     0.00000  \n",
       "\n",
       "[17180 rows x 844 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Now perform the groupby mean operation\n",
    "methy_df = raw_methy_df.groupby('locus_id', as_index=False).mean()\n",
    "display(methy_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\811266666.py:1: DtypeWarning: Columns (4,5,6,11,12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmg_promoter_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Promoter/BioMedGraphica_Conn_Promoter.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>HGNC_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>NAT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>ADA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>CDH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>AKT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_PM86234</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_PM86235</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_PM86236</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID HGNC_Symbol\n",
       "0               BMGC_PM00001        A1BG\n",
       "1               BMGC_PM00002        NAT2\n",
       "2               BMGC_PM00003         ADA\n",
       "3               BMGC_PM00004        CDH2\n",
       "4               BMGC_PM00005        AKT3\n",
       "...                      ...         ...\n",
       "86233           BMGC_PM86234         NaN\n",
       "86234           BMGC_PM86235         NaN\n",
       "86235           BMGC_PM86236         NaN\n",
       "86236           BMGC_PM86237         NaN\n",
       "86237           BMGC_PM86238         NaN\n",
       "\n",
       "[86238 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bmg_promoter_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Promoter/BioMedGraphica_Conn_Promoter.csv')\n",
    "# keep BioMedGraphica_ID and HGNC_Symbol\n",
    "bmg_promoter_df = bmg_promoter_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']]\n",
    "display(bmg_promoter_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>DMS53_LUNG</th>\n",
       "      <th>SW1116_LARGE_INTESTINE</th>\n",
       "      <th>P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>UMUC3_URINARY_TRACT</th>\n",
       "      <th>HOS_BONE</th>\n",
       "      <th>AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>RVH421_SKIN</th>\n",
       "      <th>NCIH1184_LUNG</th>\n",
       "      <th>...</th>\n",
       "      <th>JHUEM7_ENDOMETRIUM</th>\n",
       "      <th>OE21_OESOPHAGUS</th>\n",
       "      <th>MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HOP62_LUNG</th>\n",
       "      <th>EKVX_LUNG</th>\n",
       "      <th>UO31_KIDNEY</th>\n",
       "      <th>SF268_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SF539_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SNB75_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HOP92_LUNG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.51504</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.982880</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.552950</td>\n",
       "      <td>0.87010</td>\n",
       "      <td>0.90770</td>\n",
       "      <td>0.86540</td>\n",
       "      <td>0.71970</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.89135</td>\n",
       "      <td>0.28806</td>\n",
       "      <td>0.50752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.01768</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.099330</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.045510</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00385</td>\n",
       "      <td>0.00499</td>\n",
       "      <td>0.01187</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00306</td>\n",
       "      <td>0.00160</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.00481</td>\n",
       "      <td>0.85522</td>\n",
       "      <td>0.962970</td>\n",
       "      <td>0.68518</td>\n",
       "      <td>0.111570</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.03094</td>\n",
       "      <td>0.00935</td>\n",
       "      <td>0.00457</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00704</td>\n",
       "      <td>0.00862</td>\n",
       "      <td>0.91237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00008</td>\n",
       "      <td>0.04435</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.512840</td>\n",
       "      <td>0.45453</td>\n",
       "      <td>0.541650</td>\n",
       "      <td>0.96265</td>\n",
       "      <td>0.82994</td>\n",
       "      <td>0.42857</td>\n",
       "      <td>0.38245</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.59091</td>\n",
       "      <td>0.38090</td>\n",
       "      <td>0.63461</td>\n",
       "      <td>0.92726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00012</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.037610</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.01279</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17431</th>\n",
       "      <td>BMGC_PM69840</td>\n",
       "      <td>0.93452</td>\n",
       "      <td>0.39385</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.67802</td>\n",
       "      <td>0.986100</td>\n",
       "      <td>0.96078</td>\n",
       "      <td>0.96775</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.98182</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.90909</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.97222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17432</th>\n",
       "      <td>BMGC_PM69841</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.350365</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.001283</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.22449</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.25000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17433</th>\n",
       "      <td>BMGC_PM70228</td>\n",
       "      <td>0.25279</td>\n",
       "      <td>0.51015</td>\n",
       "      <td>0.308140</td>\n",
       "      <td>0.26662</td>\n",
       "      <td>0.820340</td>\n",
       "      <td>0.83620</td>\n",
       "      <td>0.85999</td>\n",
       "      <td>0.65949</td>\n",
       "      <td>0.32076</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.38639</td>\n",
       "      <td>0.84845</td>\n",
       "      <td>0.62212</td>\n",
       "      <td>0.75415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17434</th>\n",
       "      <td>BMGC_PM70273</td>\n",
       "      <td>0.03293</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.017440</td>\n",
       "      <td>0.01087</td>\n",
       "      <td>0.047620</td>\n",
       "      <td>0.05430</td>\n",
       "      <td>0.09606</td>\n",
       "      <td>0.07547</td>\n",
       "      <td>0.05774</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.06818</td>\n",
       "      <td>0.06719</td>\n",
       "      <td>0.11995</td>\n",
       "      <td>0.05687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17435</th>\n",
       "      <td>BMGC_PM72370</td>\n",
       "      <td>0.26677</td>\n",
       "      <td>0.03508</td>\n",
       "      <td>0.765950</td>\n",
       "      <td>0.58843</td>\n",
       "      <td>0.103900</td>\n",
       "      <td>0.27093</td>\n",
       "      <td>0.40293</td>\n",
       "      <td>0.03574</td>\n",
       "      <td>0.53906</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02657</td>\n",
       "      <td>0.04233</td>\n",
       "      <td>0.22917</td>\n",
       "      <td>0.10684</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17436 rows × 844 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  DMS53_LUNG  SW1116_LARGE_INTESTINE  \\\n",
       "0               BMGC_PM00001     0.51504                 0.50000   \n",
       "1               BMGC_PM00003     0.01768                 0.00000   \n",
       "2               BMGC_PM00004     0.00481                 0.85522   \n",
       "3               BMGC_PM00008     0.04435                 1.00000   \n",
       "4               BMGC_PM00012     0.00000                 0.00000   \n",
       "...                      ...         ...                     ...   \n",
       "17431           BMGC_PM69840     0.93452                 0.39385   \n",
       "17432           BMGC_PM69841     0.00000                 0.00000   \n",
       "17433           BMGC_PM70228     0.25279                 0.51015   \n",
       "17434           BMGC_PM70273     0.03293                 0.00000   \n",
       "17435           BMGC_PM72370     0.26677                 0.03508   \n",
       "\n",
       "       P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0                                      0.982880   \n",
       "1                                      0.099330   \n",
       "2                                      0.962970   \n",
       "3                                      0.512840   \n",
       "4                                      0.037610   \n",
       "...                                         ...   \n",
       "17431                                  1.000000   \n",
       "17432                                  0.350365   \n",
       "17433                                  0.308140   \n",
       "17434                                  0.017440   \n",
       "17435                                  0.765950   \n",
       "\n",
       "       HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  UMUC3_URINARY_TRACT  \\\n",
       "0                                       1.00000             0.552950   \n",
       "1                                       0.00000             0.045510   \n",
       "2                                       0.68518             0.111570   \n",
       "3                                       0.45453             0.541650   \n",
       "4                                       0.00000             0.000000   \n",
       "...                                         ...                  ...   \n",
       "17431                                   0.67802             0.986100   \n",
       "17432                                   0.00000             0.001283   \n",
       "17433                                   0.26662             0.820340   \n",
       "17434                                   0.01087             0.047620   \n",
       "17435                                   0.58843             0.103900   \n",
       "\n",
       "       HOS_BONE  AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  RVH421_SKIN  \\\n",
       "0       0.87010                                    0.90770      0.86540   \n",
       "1       0.00000                                    0.00385      0.00499   \n",
       "2       0.00000                                    0.03094      0.00935   \n",
       "3       0.96265                                    0.82994      0.42857   \n",
       "4       0.01279                                    0.00000      0.00000   \n",
       "...         ...                                        ...          ...   \n",
       "17431   0.96078                                    0.96775      1.00000   \n",
       "17432   0.00000                                    0.22449      0.00000   \n",
       "17433   0.83620                                    0.85999      0.65949   \n",
       "17434   0.05430                                    0.09606      0.07547   \n",
       "17435   0.27093                                    0.40293      0.03574   \n",
       "\n",
       "       NCIH1184_LUNG  ...  JHUEM7_ENDOMETRIUM  OE21_OESOPHAGUS  \\\n",
       "0            0.71970  ...                 0.0              0.0   \n",
       "1            0.01187  ...                 0.0              0.0   \n",
       "2            0.00457  ...                 0.0              0.0   \n",
       "3            0.38245  ...                 0.0              0.0   \n",
       "4            0.00000  ...                 0.0              0.0   \n",
       "...              ...  ...                 ...              ...   \n",
       "17431        0.98182  ...                 0.0              0.0   \n",
       "17432        0.25000  ...                 0.0              0.0   \n",
       "17433        0.32076  ...                 0.0              0.0   \n",
       "17434        0.05774  ...                 0.0              0.0   \n",
       "17435        0.53906  ...                 0.0              0.0   \n",
       "\n",
       "       MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  HOP62_LUNG  EKVX_LUNG  \\\n",
       "0                                           0.0         0.0        0.0   \n",
       "1                                           0.0         0.0        0.0   \n",
       "2                                           0.0         0.0        0.0   \n",
       "3                                           0.0         0.0        0.0   \n",
       "4                                           0.0         0.0        0.0   \n",
       "...                                         ...         ...        ...   \n",
       "17431                                       0.0         0.0        0.0   \n",
       "17432                                       0.0         0.0        0.0   \n",
       "17433                                       0.0         0.0        0.0   \n",
       "17434                                       0.0         0.0        0.0   \n",
       "17435                                       0.0         0.0        0.0   \n",
       "\n",
       "       UO31_KIDNEY  SF268_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "0              0.0                       0.50000   \n",
       "1              0.0                       0.00306   \n",
       "2              0.0                       0.00000   \n",
       "3              0.0                       0.59091   \n",
       "4              0.0                       0.00000   \n",
       "...            ...                           ...   \n",
       "17431          0.0                       1.00000   \n",
       "17432          0.0                       0.00000   \n",
       "17433          0.0                       0.38639   \n",
       "17434          0.0                       0.06818   \n",
       "17435          0.0                       0.02657   \n",
       "\n",
       "       SF539_CENTRAL_NERVOUS_SYSTEM  SNB75_CENTRAL_NERVOUS_SYSTEM  HOP92_LUNG  \n",
       "0                           0.89135                       0.28806     0.50752  \n",
       "1                           0.00160                       0.00000     0.00375  \n",
       "2                           0.00704                       0.00862     0.91237  \n",
       "3                           0.38090                       0.63461     0.92726  \n",
       "4                           0.00000                       0.00000     0.00000  \n",
       "...                             ...                           ...         ...  \n",
       "17431                       0.90909                       1.00000     0.97222  \n",
       "17432                       0.00000                       0.00000     0.00000  \n",
       "17433                       0.84845                       0.62212     0.75415  \n",
       "17434                       0.06719                       0.11995     0.05687  \n",
       "17435                       0.04233                       0.22917     0.10684  \n",
       "\n",
       "[17436 rows x 844 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# merge the biomedgraphica_id with the raw_methy_df\n",
    "merged_methy_df = pd.merge(bmg_promoter_df, methy_df, left_on='HGNC_Symbol', right_on='locus_id', how='inner')\n",
    "merged_methy_df.drop(columns=['HGNC_Symbol', 'locus_id'], inplace=True)\n",
    "display(merged_methy_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>DMS53_LUNG</th>\n",
       "      <th>SW1116_LARGE_INTESTINE</th>\n",
       "      <th>P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>UMUC3_URINARY_TRACT</th>\n",
       "      <th>HOS_BONE</th>\n",
       "      <th>AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>RVH421_SKIN</th>\n",
       "      <th>NCIH1184_LUNG</th>\n",
       "      <th>...</th>\n",
       "      <th>JHUEM7_ENDOMETRIUM</th>\n",
       "      <th>OE21_OESOPHAGUS</th>\n",
       "      <th>MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>HOP62_LUNG</th>\n",
       "      <th>EKVX_LUNG</th>\n",
       "      <th>UO31_KIDNEY</th>\n",
       "      <th>SF268_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SF539_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>SNB75_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>HOP92_LUNG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.51504</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.98288</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.55295</td>\n",
       "      <td>0.8701</td>\n",
       "      <td>0.90770</td>\n",
       "      <td>0.86540</td>\n",
       "      <td>0.71970</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.89135</td>\n",
       "      <td>0.28806</td>\n",
       "      <td>0.50752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.01768</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.09933</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04551</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00385</td>\n",
       "      <td>0.00499</td>\n",
       "      <td>0.01187</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00306</td>\n",
       "      <td>0.00160</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.00481</td>\n",
       "      <td>0.85522</td>\n",
       "      <td>0.96297</td>\n",
       "      <td>0.68518</td>\n",
       "      <td>0.11157</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.03094</td>\n",
       "      <td>0.00935</td>\n",
       "      <td>0.00457</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00704</td>\n",
       "      <td>0.00862</td>\n",
       "      <td>0.91237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_PM86234</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_PM86235</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_PM86236</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 844 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  DMS53_LUNG  SW1116_LARGE_INTESTINE  \\\n",
       "0               BMGC_PM00001     0.51504                 0.50000   \n",
       "1               BMGC_PM00002     0.00000                 0.00000   \n",
       "2               BMGC_PM00003     0.01768                 0.00000   \n",
       "3               BMGC_PM00004     0.00481                 0.85522   \n",
       "4               BMGC_PM00005     0.00000                 0.00000   \n",
       "...                      ...         ...                     ...   \n",
       "86233           BMGC_PM86234     0.00000                 0.00000   \n",
       "86234           BMGC_PM86235     0.00000                 0.00000   \n",
       "86235           BMGC_PM86236     0.00000                 0.00000   \n",
       "86236           BMGC_PM86237     0.00000                 0.00000   \n",
       "86237           BMGC_PM86238     0.00000                 0.00000   \n",
       "\n",
       "       P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0                                       0.98288   \n",
       "1                                       0.00000   \n",
       "2                                       0.09933   \n",
       "3                                       0.96297   \n",
       "4                                       0.00000   \n",
       "...                                         ...   \n",
       "86233                                   0.00000   \n",
       "86234                                   0.00000   \n",
       "86235                                   0.00000   \n",
       "86236                                   0.00000   \n",
       "86237                                   0.00000   \n",
       "\n",
       "       HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  UMUC3_URINARY_TRACT  \\\n",
       "0                                       1.00000              0.55295   \n",
       "1                                       0.00000              0.00000   \n",
       "2                                       0.00000              0.04551   \n",
       "3                                       0.68518              0.11157   \n",
       "4                                       0.00000              0.00000   \n",
       "...                                         ...                  ...   \n",
       "86233                                   0.00000              0.00000   \n",
       "86234                                   0.00000              0.00000   \n",
       "86235                                   0.00000              0.00000   \n",
       "86236                                   0.00000              0.00000   \n",
       "86237                                   0.00000              0.00000   \n",
       "\n",
       "       HOS_BONE  AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  RVH421_SKIN  \\\n",
       "0        0.8701                                    0.90770      0.86540   \n",
       "1        0.0000                                    0.00000      0.00000   \n",
       "2        0.0000                                    0.00385      0.00499   \n",
       "3        0.0000                                    0.03094      0.00935   \n",
       "4        0.0000                                    0.00000      0.00000   \n",
       "...         ...                                        ...          ...   \n",
       "86233    0.0000                                    0.00000      0.00000   \n",
       "86234    0.0000                                    0.00000      0.00000   \n",
       "86235    0.0000                                    0.00000      0.00000   \n",
       "86236    0.0000                                    0.00000      0.00000   \n",
       "86237    0.0000                                    0.00000      0.00000   \n",
       "\n",
       "       NCIH1184_LUNG  ...  JHUEM7_ENDOMETRIUM  OE21_OESOPHAGUS  \\\n",
       "0            0.71970  ...                 0.0              0.0   \n",
       "1            0.00000  ...                 0.0              0.0   \n",
       "2            0.01187  ...                 0.0              0.0   \n",
       "3            0.00457  ...                 0.0              0.0   \n",
       "4            0.00000  ...                 0.0              0.0   \n",
       "...              ...  ...                 ...              ...   \n",
       "86233        0.00000  ...                 0.0              0.0   \n",
       "86234        0.00000  ...                 0.0              0.0   \n",
       "86235        0.00000  ...                 0.0              0.0   \n",
       "86236        0.00000  ...                 0.0              0.0   \n",
       "86237        0.00000  ...                 0.0              0.0   \n",
       "\n",
       "       MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  HOP62_LUNG  EKVX_LUNG  \\\n",
       "0                                           0.0         0.0        0.0   \n",
       "1                                           0.0         0.0        0.0   \n",
       "2                                           0.0         0.0        0.0   \n",
       "3                                           0.0         0.0        0.0   \n",
       "4                                           0.0         0.0        0.0   \n",
       "...                                         ...         ...        ...   \n",
       "86233                                       0.0         0.0        0.0   \n",
       "86234                                       0.0         0.0        0.0   \n",
       "86235                                       0.0         0.0        0.0   \n",
       "86236                                       0.0         0.0        0.0   \n",
       "86237                                       0.0         0.0        0.0   \n",
       "\n",
       "       UO31_KIDNEY  SF268_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "0              0.0                       0.50000   \n",
       "1              0.0                       0.00000   \n",
       "2              0.0                       0.00306   \n",
       "3              0.0                       0.00000   \n",
       "4              0.0                       0.00000   \n",
       "...            ...                           ...   \n",
       "86233          0.0                       0.00000   \n",
       "86234          0.0                       0.00000   \n",
       "86235          0.0                       0.00000   \n",
       "86236          0.0                       0.00000   \n",
       "86237          0.0                       0.00000   \n",
       "\n",
       "       SF539_CENTRAL_NERVOUS_SYSTEM  SNB75_CENTRAL_NERVOUS_SYSTEM  HOP92_LUNG  \n",
       "0                           0.89135                       0.28806     0.50752  \n",
       "1                           0.00000                       0.00000     0.00000  \n",
       "2                           0.00160                       0.00000     0.00375  \n",
       "3                           0.00704                       0.00862     0.91237  \n",
       "4                           0.00000                       0.00000     0.00000  \n",
       "...                             ...                           ...         ...  \n",
       "86233                       0.00000                       0.00000     0.00000  \n",
       "86234                       0.00000                       0.00000     0.00000  \n",
       "86235                       0.00000                       0.00000     0.00000  \n",
       "86236                       0.00000                       0.00000     0.00000  \n",
       "86237                       0.00000                       0.00000     0.00000  \n",
       "\n",
       "[86238 rows x 844 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "final_merged_methy_df = pd.merge(bmg_promoter_df, methy_df, left_on='HGNC_Symbol', right_on='locus_id', how='left')\n",
    "# fill the NaN with 0.0\n",
    "final_merged_methy_df = final_merged_methy_df.fillna(0.0)\n",
    "final_merged_methy_df.drop(columns=['HGNC_Symbol', 'locus_id'], inplace=True)\n",
    "display(final_merged_methy_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>RHEB (6009)</th>\n",
       "      <th>TIPIN (54962)</th>\n",
       "      <th>OR4A47 (403253)</th>\n",
       "      <th>NUP133 (55746)</th>\n",
       "      <th>CPNE8-AS1 (107984504)</th>\n",
       "      <th>PDIA4 (9601)</th>\n",
       "      <th>RNU6-899P (106479964)</th>\n",
       "      <th>NRXN1-DT (730100)</th>\n",
       "      <th>SMIM41 (113523638)</th>\n",
       "      <th>...</th>\n",
       "      <th>RN7SKP98 (106480863)</th>\n",
       "      <th>RPSAP15 (220885)</th>\n",
       "      <th>FAM149B1P1 (388181)</th>\n",
       "      <th>RNU6-1020P (106481503)</th>\n",
       "      <th>MIR1285-1 (100302218)</th>\n",
       "      <th>FXYD5 (53827)</th>\n",
       "      <th>RNA5SP154 (100873420)</th>\n",
       "      <th>TMEM87B (84910)</th>\n",
       "      <th>RNU6-1313P (106480139)</th>\n",
       "      <th>RNU6-1158P (106480076)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000628</td>\n",
       "      <td>0.733588</td>\n",
       "      <td>1.273030</td>\n",
       "      <td>1.190365</td>\n",
       "      <td>1.409367</td>\n",
       "      <td>0.734211</td>\n",
       "      <td>0.733588</td>\n",
       "      <td>1.083607</td>\n",
       "      <td>1.043100</td>\n",
       "      <td>0.671598</td>\n",
       "      <td>...</td>\n",
       "      <td>1.409367</td>\n",
       "      <td>0.773267</td>\n",
       "      <td>1.273030</td>\n",
       "      <td>1.059823</td>\n",
       "      <td>0.766085</td>\n",
       "      <td>0.962284</td>\n",
       "      <td>1.009244</td>\n",
       "      <td>1.060017</td>\n",
       "      <td>0.698082</td>\n",
       "      <td>0.698082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000974</td>\n",
       "      <td>0.950736</td>\n",
       "      <td>1.000040</td>\n",
       "      <td>1.048747</td>\n",
       "      <td>1.163154</td>\n",
       "      <td>1.041892</td>\n",
       "      <td>0.950736</td>\n",
       "      <td>1.032830</td>\n",
       "      <td>1.016793</td>\n",
       "      <td>1.041892</td>\n",
       "      <td>...</td>\n",
       "      <td>1.163154</td>\n",
       "      <td>1.070037</td>\n",
       "      <td>1.033814</td>\n",
       "      <td>1.019035</td>\n",
       "      <td>0.950736</td>\n",
       "      <td>0.998635</td>\n",
       "      <td>1.089437</td>\n",
       "      <td>1.020576</td>\n",
       "      <td>0.989170</td>\n",
       "      <td>0.989170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-003071</td>\n",
       "      <td>1.125149</td>\n",
       "      <td>0.738668</td>\n",
       "      <td>1.170261</td>\n",
       "      <td>0.779756</td>\n",
       "      <td>1.420911</td>\n",
       "      <td>1.177429</td>\n",
       "      <td>1.132756</td>\n",
       "      <td>0.791989</td>\n",
       "      <td>1.120336</td>\n",
       "      <td>...</td>\n",
       "      <td>0.788941</td>\n",
       "      <td>0.763276</td>\n",
       "      <td>0.714417</td>\n",
       "      <td>1.229201</td>\n",
       "      <td>1.194755</td>\n",
       "      <td>1.434221</td>\n",
       "      <td>0.878627</td>\n",
       "      <td>0.764194</td>\n",
       "      <td>1.047771</td>\n",
       "      <td>1.047771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-001338</td>\n",
       "      <td>1.491525</td>\n",
       "      <td>0.988869</td>\n",
       "      <td>0.996618</td>\n",
       "      <td>0.999843</td>\n",
       "      <td>0.998373</td>\n",
       "      <td>1.491525</td>\n",
       "      <td>0.996618</td>\n",
       "      <td>1.005134</td>\n",
       "      <td>0.998373</td>\n",
       "      <td>...</td>\n",
       "      <td>1.542706</td>\n",
       "      <td>0.464602</td>\n",
       "      <td>0.988869</td>\n",
       "      <td>0.996042</td>\n",
       "      <td>1.035373</td>\n",
       "      <td>0.924157</td>\n",
       "      <td>0.946639</td>\n",
       "      <td>0.935294</td>\n",
       "      <td>2.055489</td>\n",
       "      <td>1.946219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-001608</td>\n",
       "      <td>1.268255</td>\n",
       "      <td>0.799244</td>\n",
       "      <td>1.378248</td>\n",
       "      <td>0.781063</td>\n",
       "      <td>0.879289</td>\n",
       "      <td>1.268255</td>\n",
       "      <td>1.355953</td>\n",
       "      <td>1.140606</td>\n",
       "      <td>0.790556</td>\n",
       "      <td>...</td>\n",
       "      <td>0.813189</td>\n",
       "      <td>0.472556</td>\n",
       "      <td>0.875332</td>\n",
       "      <td>0.816484</td>\n",
       "      <td>1.279431</td>\n",
       "      <td>0.811873</td>\n",
       "      <td>0.872874</td>\n",
       "      <td>1.147173</td>\n",
       "      <td>1.238517</td>\n",
       "      <td>1.394133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1924</th>\n",
       "      <td>ACH-001389</td>\n",
       "      <td>1.786448</td>\n",
       "      <td>1.942519</td>\n",
       "      <td>1.124677</td>\n",
       "      <td>0.736300</td>\n",
       "      <td>1.119763</td>\n",
       "      <td>1.786448</td>\n",
       "      <td>0.817688</td>\n",
       "      <td>0.772048</td>\n",
       "      <td>0.775908</td>\n",
       "      <td>...</td>\n",
       "      <td>1.096212</td>\n",
       "      <td>0.367679</td>\n",
       "      <td>0.927629</td>\n",
       "      <td>0.811307</td>\n",
       "      <td>0.798644</td>\n",
       "      <td>0.773894</td>\n",
       "      <td>0.827995</td>\n",
       "      <td>0.785493</td>\n",
       "      <td>1.484038</td>\n",
       "      <td>2.095260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1925</th>\n",
       "      <td>ACH-001835</td>\n",
       "      <td>0.713098</td>\n",
       "      <td>1.058476</td>\n",
       "      <td>1.442339</td>\n",
       "      <td>1.337263</td>\n",
       "      <td>1.029053</td>\n",
       "      <td>0.713098</td>\n",
       "      <td>1.442339</td>\n",
       "      <td>1.521287</td>\n",
       "      <td>1.029053</td>\n",
       "      <td>...</td>\n",
       "      <td>1.368689</td>\n",
       "      <td>0.620956</td>\n",
       "      <td>1.058476</td>\n",
       "      <td>0.684878</td>\n",
       "      <td>1.734274</td>\n",
       "      <td>1.639451</td>\n",
       "      <td>0.685640</td>\n",
       "      <td>1.565325</td>\n",
       "      <td>1.343979</td>\n",
       "      <td>1.343979</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1926</th>\n",
       "      <td>ACH-000296</td>\n",
       "      <td>1.156063</td>\n",
       "      <td>0.851183</td>\n",
       "      <td>1.128537</td>\n",
       "      <td>1.610137</td>\n",
       "      <td>0.969262</td>\n",
       "      <td>1.190981</td>\n",
       "      <td>1.128537</td>\n",
       "      <td>1.100498</td>\n",
       "      <td>1.146175</td>\n",
       "      <td>...</td>\n",
       "      <td>1.444669</td>\n",
       "      <td>0.333722</td>\n",
       "      <td>0.851183</td>\n",
       "      <td>1.130857</td>\n",
       "      <td>0.854886</td>\n",
       "      <td>1.053836</td>\n",
       "      <td>0.761691</td>\n",
       "      <td>1.089399</td>\n",
       "      <td>0.775674</td>\n",
       "      <td>0.771612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1927</th>\n",
       "      <td>ACH-001971</td>\n",
       "      <td>2.152319</td>\n",
       "      <td>1.180349</td>\n",
       "      <td>1.208479</td>\n",
       "      <td>1.049469</td>\n",
       "      <td>1.127606</td>\n",
       "      <td>2.152319</td>\n",
       "      <td>0.870848</td>\n",
       "      <td>0.984712</td>\n",
       "      <td>1.127606</td>\n",
       "      <td>...</td>\n",
       "      <td>1.114345</td>\n",
       "      <td>0.956893</td>\n",
       "      <td>1.180349</td>\n",
       "      <td>0.614354</td>\n",
       "      <td>2.044447</td>\n",
       "      <td>0.927493</td>\n",
       "      <td>0.890273</td>\n",
       "      <td>0.984712</td>\n",
       "      <td>1.097954</td>\n",
       "      <td>1.097954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1928</th>\n",
       "      <td>ACH-002276</td>\n",
       "      <td>0.995697</td>\n",
       "      <td>0.964208</td>\n",
       "      <td>0.988156</td>\n",
       "      <td>1.026052</td>\n",
       "      <td>1.005615</td>\n",
       "      <td>0.995697</td>\n",
       "      <td>0.988156</td>\n",
       "      <td>0.519251</td>\n",
       "      <td>1.005615</td>\n",
       "      <td>...</td>\n",
       "      <td>1.026052</td>\n",
       "      <td>0.837707</td>\n",
       "      <td>0.964208</td>\n",
       "      <td>0.965424</td>\n",
       "      <td>0.997949</td>\n",
       "      <td>1.002959</td>\n",
       "      <td>1.003016</td>\n",
       "      <td>1.018937</td>\n",
       "      <td>0.989859</td>\n",
       "      <td>0.989859</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1929 rows × 38591 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0  RHEB (6009)  TIPIN (54962)  OR4A47 (403253)  NUP133 (55746)  \\\n",
       "0     ACH-000628     0.733588       1.273030         1.190365        1.409367   \n",
       "1     ACH-000974     0.950736       1.000040         1.048747        1.163154   \n",
       "2     ACH-003071     1.125149       0.738668         1.170261        0.779756   \n",
       "3     ACH-001338     1.491525       0.988869         0.996618        0.999843   \n",
       "4     ACH-001608     1.268255       0.799244         1.378248        0.781063   \n",
       "...          ...          ...            ...              ...             ...   \n",
       "1924  ACH-001389     1.786448       1.942519         1.124677        0.736300   \n",
       "1925  ACH-001835     0.713098       1.058476         1.442339        1.337263   \n",
       "1926  ACH-000296     1.156063       0.851183         1.128537        1.610137   \n",
       "1927  ACH-001971     2.152319       1.180349         1.208479        1.049469   \n",
       "1928  ACH-002276     0.995697       0.964208         0.988156        1.026052   \n",
       "\n",
       "      CPNE8-AS1 (107984504)  PDIA4 (9601)  RNU6-899P (106479964)  \\\n",
       "0                  0.734211      0.733588               1.083607   \n",
       "1                  1.041892      0.950736               1.032830   \n",
       "2                  1.420911      1.177429               1.132756   \n",
       "3                  0.998373      1.491525               0.996618   \n",
       "4                  0.879289      1.268255               1.355953   \n",
       "...                     ...           ...                    ...   \n",
       "1924               1.119763      1.786448               0.817688   \n",
       "1925               1.029053      0.713098               1.442339   \n",
       "1926               0.969262      1.190981               1.128537   \n",
       "1927               1.127606      2.152319               0.870848   \n",
       "1928               1.005615      0.995697               0.988156   \n",
       "\n",
       "      NRXN1-DT (730100)  SMIM41 (113523638)  ...  RN7SKP98 (106480863)  \\\n",
       "0              1.043100            0.671598  ...              1.409367   \n",
       "1              1.016793            1.041892  ...              1.163154   \n",
       "2              0.791989            1.120336  ...              0.788941   \n",
       "3              1.005134            0.998373  ...              1.542706   \n",
       "4              1.140606            0.790556  ...              0.813189   \n",
       "...                 ...                 ...  ...                   ...   \n",
       "1924           0.772048            0.775908  ...              1.096212   \n",
       "1925           1.521287            1.029053  ...              1.368689   \n",
       "1926           1.100498            1.146175  ...              1.444669   \n",
       "1927           0.984712            1.127606  ...              1.114345   \n",
       "1928           0.519251            1.005615  ...              1.026052   \n",
       "\n",
       "      RPSAP15 (220885)  FAM149B1P1 (388181)  RNU6-1020P (106481503)  \\\n",
       "0             0.773267             1.273030                1.059823   \n",
       "1             1.070037             1.033814                1.019035   \n",
       "2             0.763276             0.714417                1.229201   \n",
       "3             0.464602             0.988869                0.996042   \n",
       "4             0.472556             0.875332                0.816484   \n",
       "...                ...                  ...                     ...   \n",
       "1924          0.367679             0.927629                0.811307   \n",
       "1925          0.620956             1.058476                0.684878   \n",
       "1926          0.333722             0.851183                1.130857   \n",
       "1927          0.956893             1.180349                0.614354   \n",
       "1928          0.837707             0.964208                0.965424   \n",
       "\n",
       "      MIR1285-1 (100302218)  FXYD5 (53827)  RNA5SP154 (100873420)  \\\n",
       "0                  0.766085       0.962284               1.009244   \n",
       "1                  0.950736       0.998635               1.089437   \n",
       "2                  1.194755       1.434221               0.878627   \n",
       "3                  1.035373       0.924157               0.946639   \n",
       "4                  1.279431       0.811873               0.872874   \n",
       "...                     ...            ...                    ...   \n",
       "1924               0.798644       0.773894               0.827995   \n",
       "1925               1.734274       1.639451               0.685640   \n",
       "1926               0.854886       1.053836               0.761691   \n",
       "1927               2.044447       0.927493               0.890273   \n",
       "1928               0.997949       1.002959               1.003016   \n",
       "\n",
       "      TMEM87B (84910)  RNU6-1313P (106480139)  RNU6-1158P (106480076)  \n",
       "0            1.060017                0.698082                0.698082  \n",
       "1            1.020576                0.989170                0.989170  \n",
       "2            0.764194                1.047771                1.047771  \n",
       "3            0.935294                2.055489                1.946219  \n",
       "4            1.147173                1.238517                1.394133  \n",
       "...               ...                     ...                     ...  \n",
       "1924         0.785493                1.484038                2.095260  \n",
       "1925         1.565325                1.343979                1.343979  \n",
       "1926         1.089399                0.775674                0.771612  \n",
       "1927         0.984712                1.097954                1.097954  \n",
       "1928         1.018937                0.989859                0.989859  \n",
       "\n",
       "[1929 rows x 38591 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "raw_gene_df = pd.read_csv('./data/raw_data/OmicsCNGene.csv')\n",
    "display(raw_gene_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000628</th>\n",
       "      <th>ACH-000974</th>\n",
       "      <th>ACH-003071</th>\n",
       "      <th>ACH-001338</th>\n",
       "      <th>ACH-001608</th>\n",
       "      <th>ACH-000278</th>\n",
       "      <th>ACH-001350</th>\n",
       "      <th>ACH-000150</th>\n",
       "      <th>ACH-001132</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000236</th>\n",
       "      <th>ACH-002177</th>\n",
       "      <th>ACH-002163</th>\n",
       "      <th>ACH-002316</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001389</th>\n",
       "      <th>ACH-001835</th>\n",
       "      <th>ACH-000296</th>\n",
       "      <th>ACH-001971</th>\n",
       "      <th>ACH-002276</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RHEB</td>\n",
       "      <td>0.733588</td>\n",
       "      <td>0.950736</td>\n",
       "      <td>1.125149</td>\n",
       "      <td>1.491525</td>\n",
       "      <td>1.268255</td>\n",
       "      <td>1.593313</td>\n",
       "      <td>0.768985</td>\n",
       "      <td>1.678589</td>\n",
       "      <td>1.533440</td>\n",
       "      <td>...</td>\n",
       "      <td>1.732311</td>\n",
       "      <td>0.891360</td>\n",
       "      <td>1.335196</td>\n",
       "      <td>0.988235</td>\n",
       "      <td>0.962098</td>\n",
       "      <td>1.786448</td>\n",
       "      <td>0.713098</td>\n",
       "      <td>1.156063</td>\n",
       "      <td>2.152319</td>\n",
       "      <td>0.995697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TIPIN</td>\n",
       "      <td>1.273030</td>\n",
       "      <td>1.000040</td>\n",
       "      <td>0.738668</td>\n",
       "      <td>0.988869</td>\n",
       "      <td>0.799244</td>\n",
       "      <td>1.114384</td>\n",
       "      <td>0.909451</td>\n",
       "      <td>1.142067</td>\n",
       "      <td>0.736635</td>\n",
       "      <td>...</td>\n",
       "      <td>0.983290</td>\n",
       "      <td>0.899519</td>\n",
       "      <td>1.022967</td>\n",
       "      <td>0.989033</td>\n",
       "      <td>1.215076</td>\n",
       "      <td>1.942519</td>\n",
       "      <td>1.058476</td>\n",
       "      <td>0.851183</td>\n",
       "      <td>1.180349</td>\n",
       "      <td>0.964208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4A47</td>\n",
       "      <td>1.190365</td>\n",
       "      <td>1.048747</td>\n",
       "      <td>1.170261</td>\n",
       "      <td>0.996618</td>\n",
       "      <td>1.378248</td>\n",
       "      <td>1.189679</td>\n",
       "      <td>1.371610</td>\n",
       "      <td>0.971120</td>\n",
       "      <td>1.056310</td>\n",
       "      <td>...</td>\n",
       "      <td>1.224336</td>\n",
       "      <td>0.940104</td>\n",
       "      <td>0.741997</td>\n",
       "      <td>0.995188</td>\n",
       "      <td>0.935269</td>\n",
       "      <td>1.124677</td>\n",
       "      <td>1.442339</td>\n",
       "      <td>1.128537</td>\n",
       "      <td>1.208479</td>\n",
       "      <td>0.988156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NUP133</td>\n",
       "      <td>1.409367</td>\n",
       "      <td>1.163154</td>\n",
       "      <td>0.779756</td>\n",
       "      <td>0.999843</td>\n",
       "      <td>0.781063</td>\n",
       "      <td>1.080624</td>\n",
       "      <td>0.946615</td>\n",
       "      <td>1.150672</td>\n",
       "      <td>1.972796</td>\n",
       "      <td>...</td>\n",
       "      <td>0.783025</td>\n",
       "      <td>1.234318</td>\n",
       "      <td>1.035702</td>\n",
       "      <td>0.986356</td>\n",
       "      <td>1.254152</td>\n",
       "      <td>0.736300</td>\n",
       "      <td>1.337263</td>\n",
       "      <td>1.610137</td>\n",
       "      <td>1.049469</td>\n",
       "      <td>1.026052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CPNE8-AS1</td>\n",
       "      <td>0.734211</td>\n",
       "      <td>1.041892</td>\n",
       "      <td>1.420911</td>\n",
       "      <td>0.998373</td>\n",
       "      <td>0.879289</td>\n",
       "      <td>1.140088</td>\n",
       "      <td>1.009862</td>\n",
       "      <td>0.809097</td>\n",
       "      <td>1.380724</td>\n",
       "      <td>...</td>\n",
       "      <td>1.124372</td>\n",
       "      <td>0.863252</td>\n",
       "      <td>0.979797</td>\n",
       "      <td>0.976575</td>\n",
       "      <td>0.937251</td>\n",
       "      <td>1.119763</td>\n",
       "      <td>1.029053</td>\n",
       "      <td>0.969262</td>\n",
       "      <td>1.127606</td>\n",
       "      <td>1.005615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38585</th>\n",
       "      <td>FXYD5</td>\n",
       "      <td>0.962284</td>\n",
       "      <td>0.998635</td>\n",
       "      <td>1.434221</td>\n",
       "      <td>0.924157</td>\n",
       "      <td>0.811873</td>\n",
       "      <td>1.030370</td>\n",
       "      <td>0.874268</td>\n",
       "      <td>1.010786</td>\n",
       "      <td>0.962975</td>\n",
       "      <td>...</td>\n",
       "      <td>0.768815</td>\n",
       "      <td>1.123477</td>\n",
       "      <td>0.848746</td>\n",
       "      <td>0.977653</td>\n",
       "      <td>0.965869</td>\n",
       "      <td>0.773894</td>\n",
       "      <td>1.639451</td>\n",
       "      <td>1.053836</td>\n",
       "      <td>0.927493</td>\n",
       "      <td>1.002959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38586</th>\n",
       "      <td>RNA5SP154</td>\n",
       "      <td>1.009244</td>\n",
       "      <td>1.089437</td>\n",
       "      <td>0.878627</td>\n",
       "      <td>0.946639</td>\n",
       "      <td>0.872874</td>\n",
       "      <td>1.074350</td>\n",
       "      <td>1.028733</td>\n",
       "      <td>0.648140</td>\n",
       "      <td>1.073053</td>\n",
       "      <td>...</td>\n",
       "      <td>0.706832</td>\n",
       "      <td>1.158749</td>\n",
       "      <td>0.520117</td>\n",
       "      <td>0.957798</td>\n",
       "      <td>0.947472</td>\n",
       "      <td>0.827995</td>\n",
       "      <td>0.685640</td>\n",
       "      <td>0.761691</td>\n",
       "      <td>0.890273</td>\n",
       "      <td>1.003016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38587</th>\n",
       "      <td>TMEM87B</td>\n",
       "      <td>1.060017</td>\n",
       "      <td>1.020576</td>\n",
       "      <td>0.764194</td>\n",
       "      <td>0.935294</td>\n",
       "      <td>1.147173</td>\n",
       "      <td>1.239277</td>\n",
       "      <td>0.702147</td>\n",
       "      <td>1.166930</td>\n",
       "      <td>1.049231</td>\n",
       "      <td>...</td>\n",
       "      <td>1.071300</td>\n",
       "      <td>0.895312</td>\n",
       "      <td>0.971115</td>\n",
       "      <td>0.972381</td>\n",
       "      <td>0.961541</td>\n",
       "      <td>0.785493</td>\n",
       "      <td>1.565325</td>\n",
       "      <td>1.089399</td>\n",
       "      <td>0.984712</td>\n",
       "      <td>1.018937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38588</th>\n",
       "      <td>RNU6-1313P</td>\n",
       "      <td>0.698082</td>\n",
       "      <td>0.989170</td>\n",
       "      <td>1.047771</td>\n",
       "      <td>2.055489</td>\n",
       "      <td>1.238517</td>\n",
       "      <td>1.097696</td>\n",
       "      <td>0.962941</td>\n",
       "      <td>1.273156</td>\n",
       "      <td>1.217933</td>\n",
       "      <td>...</td>\n",
       "      <td>1.036596</td>\n",
       "      <td>0.909479</td>\n",
       "      <td>0.828142</td>\n",
       "      <td>1.008336</td>\n",
       "      <td>1.267007</td>\n",
       "      <td>1.484038</td>\n",
       "      <td>1.343979</td>\n",
       "      <td>0.775674</td>\n",
       "      <td>1.097954</td>\n",
       "      <td>0.989859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38589</th>\n",
       "      <td>RNU6-1158P</td>\n",
       "      <td>0.698082</td>\n",
       "      <td>0.989170</td>\n",
       "      <td>1.047771</td>\n",
       "      <td>1.946219</td>\n",
       "      <td>1.394133</td>\n",
       "      <td>1.097696</td>\n",
       "      <td>1.234954</td>\n",
       "      <td>1.352245</td>\n",
       "      <td>1.217933</td>\n",
       "      <td>...</td>\n",
       "      <td>1.036596</td>\n",
       "      <td>0.909479</td>\n",
       "      <td>0.828142</td>\n",
       "      <td>1.008336</td>\n",
       "      <td>1.267007</td>\n",
       "      <td>2.095260</td>\n",
       "      <td>1.343979</td>\n",
       "      <td>0.771612</td>\n",
       "      <td>1.097954</td>\n",
       "      <td>0.989859</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38590 rows × 1930 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0   gene_name  ACH-000628  ACH-000974  ACH-003071  ACH-001338  \\\n",
       "0                 RHEB    0.733588    0.950736    1.125149    1.491525   \n",
       "1                TIPIN    1.273030    1.000040    0.738668    0.988869   \n",
       "2               OR4A47    1.190365    1.048747    1.170261    0.996618   \n",
       "3               NUP133    1.409367    1.163154    0.779756    0.999843   \n",
       "4            CPNE8-AS1    0.734211    1.041892    1.420911    0.998373   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "38585            FXYD5    0.962284    0.998635    1.434221    0.924157   \n",
       "38586        RNA5SP154    1.009244    1.089437    0.878627    0.946639   \n",
       "38587          TMEM87B    1.060017    1.020576    0.764194    0.935294   \n",
       "38588       RNU6-1313P    0.698082    0.989170    1.047771    2.055489   \n",
       "38589       RNU6-1158P    0.698082    0.989170    1.047771    1.946219   \n",
       "\n",
       "Unnamed: 0  ACH-001608  ACH-000278  ACH-001350  ACH-000150  ACH-001132  ...  \\\n",
       "0             1.268255    1.593313    0.768985    1.678589    1.533440  ...   \n",
       "1             0.799244    1.114384    0.909451    1.142067    0.736635  ...   \n",
       "2             1.378248    1.189679    1.371610    0.971120    1.056310  ...   \n",
       "3             0.781063    1.080624    0.946615    1.150672    1.972796  ...   \n",
       "4             0.879289    1.140088    1.009862    0.809097    1.380724  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "38585         0.811873    1.030370    0.874268    1.010786    0.962975  ...   \n",
       "38586         0.872874    1.074350    1.028733    0.648140    1.073053  ...   \n",
       "38587         1.147173    1.239277    0.702147    1.166930    1.049231  ...   \n",
       "38588         1.238517    1.097696    0.962941    1.273156    1.217933  ...   \n",
       "38589         1.394133    1.097696    1.234954    1.352245    1.217933  ...   \n",
       "\n",
       "Unnamed: 0  ACH-000236  ACH-002177  ACH-002163  ACH-002316  ACH-001039  \\\n",
       "0             1.732311    0.891360    1.335196    0.988235    0.962098   \n",
       "1             0.983290    0.899519    1.022967    0.989033    1.215076   \n",
       "2             1.224336    0.940104    0.741997    0.995188    0.935269   \n",
       "3             0.783025    1.234318    1.035702    0.986356    1.254152   \n",
       "4             1.124372    0.863252    0.979797    0.976575    0.937251   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "38585         0.768815    1.123477    0.848746    0.977653    0.965869   \n",
       "38586         0.706832    1.158749    0.520117    0.957798    0.947472   \n",
       "38587         1.071300    0.895312    0.971115    0.972381    0.961541   \n",
       "38588         1.036596    0.909479    0.828142    1.008336    1.267007   \n",
       "38589         1.036596    0.909479    0.828142    1.008336    1.267007   \n",
       "\n",
       "Unnamed: 0  ACH-001389  ACH-001835  ACH-000296  ACH-001971  ACH-002276  \n",
       "0             1.786448    0.713098    1.156063    2.152319    0.995697  \n",
       "1             1.942519    1.058476    0.851183    1.180349    0.964208  \n",
       "2             1.124677    1.442339    1.128537    1.208479    0.988156  \n",
       "3             0.736300    1.337263    1.610137    1.049469    1.026052  \n",
       "4             1.119763    1.029053    0.969262    1.127606    1.005615  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "38585         0.773894    1.639451    1.053836    0.927493    1.002959  \n",
       "38586         0.827995    0.685640    0.761691    0.890273    1.003016  \n",
       "38587         0.785493    1.565325    1.089399    0.984712    1.018937  \n",
       "38588         1.484038    1.343979    0.775674    1.097954    0.989859  \n",
       "38589         2.095260    1.343979    0.771612    1.097954    0.989859  \n",
       "\n",
       "[38590 rows x 1930 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# First, set the \"Unnamed: 0\" column as the index:\n",
    "raw_gene_t_df = raw_gene_df.set_index(\"Unnamed: 0\")\n",
    "# Then transpose the DataFrame:\n",
    "raw_gene_t_df = raw_gene_t_df.transpose()\n",
    "# move the index to a column and rename it as the gene_name\n",
    "raw_gene_t_df.reset_index(inplace=True)\n",
    "raw_gene_t_df.rename(columns={'index': 'gene_name'}, inplace=True)\n",
    "# column gene_name should be kept with the content by removing \"()\" and remove and \" \" content\n",
    "raw_gene_t_df['gene_name'] = raw_gene_t_df['gene_name'].apply(lambda x: x.split('(')[0].strip())\n",
    "# Optionally, if you want to view the result:\n",
    "display(raw_gene_t_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000628</th>\n",
       "      <th>ACH-000974</th>\n",
       "      <th>ACH-003071</th>\n",
       "      <th>ACH-001338</th>\n",
       "      <th>ACH-001608</th>\n",
       "      <th>ACH-000278</th>\n",
       "      <th>ACH-001350</th>\n",
       "      <th>ACH-000150</th>\n",
       "      <th>ACH-001132</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000236</th>\n",
       "      <th>ACH-002177</th>\n",
       "      <th>ACH-002163</th>\n",
       "      <th>ACH-002316</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001389</th>\n",
       "      <th>ACH-001835</th>\n",
       "      <th>ACH-000296</th>\n",
       "      <th>ACH-001971</th>\n",
       "      <th>ACH-002276</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RHEB</td>\n",
       "      <td>0.733588</td>\n",
       "      <td>0.950736</td>\n",
       "      <td>1.125149</td>\n",
       "      <td>1.491525</td>\n",
       "      <td>1.268255</td>\n",
       "      <td>1.593313</td>\n",
       "      <td>0.768985</td>\n",
       "      <td>1.678589</td>\n",
       "      <td>1.533440</td>\n",
       "      <td>...</td>\n",
       "      <td>1.732311</td>\n",
       "      <td>0.891360</td>\n",
       "      <td>1.335196</td>\n",
       "      <td>0.988235</td>\n",
       "      <td>0.962098</td>\n",
       "      <td>1.786448</td>\n",
       "      <td>0.713098</td>\n",
       "      <td>1.156063</td>\n",
       "      <td>2.152319</td>\n",
       "      <td>0.995697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TIPIN</td>\n",
       "      <td>1.273030</td>\n",
       "      <td>1.000040</td>\n",
       "      <td>0.738668</td>\n",
       "      <td>0.988869</td>\n",
       "      <td>0.799244</td>\n",
       "      <td>1.114384</td>\n",
       "      <td>0.909451</td>\n",
       "      <td>1.142067</td>\n",
       "      <td>0.736635</td>\n",
       "      <td>...</td>\n",
       "      <td>0.983290</td>\n",
       "      <td>0.899519</td>\n",
       "      <td>1.022967</td>\n",
       "      <td>0.989033</td>\n",
       "      <td>1.215076</td>\n",
       "      <td>1.942519</td>\n",
       "      <td>1.058476</td>\n",
       "      <td>0.851183</td>\n",
       "      <td>1.180349</td>\n",
       "      <td>0.964208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4A47</td>\n",
       "      <td>1.190365</td>\n",
       "      <td>1.048747</td>\n",
       "      <td>1.170261</td>\n",
       "      <td>0.996618</td>\n",
       "      <td>1.378248</td>\n",
       "      <td>1.189679</td>\n",
       "      <td>1.371610</td>\n",
       "      <td>0.971120</td>\n",
       "      <td>1.056310</td>\n",
       "      <td>...</td>\n",
       "      <td>1.224336</td>\n",
       "      <td>0.940104</td>\n",
       "      <td>0.741997</td>\n",
       "      <td>0.995188</td>\n",
       "      <td>0.935269</td>\n",
       "      <td>1.124677</td>\n",
       "      <td>1.442339</td>\n",
       "      <td>1.128537</td>\n",
       "      <td>1.208479</td>\n",
       "      <td>0.988156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NUP133</td>\n",
       "      <td>1.409367</td>\n",
       "      <td>1.163154</td>\n",
       "      <td>0.779756</td>\n",
       "      <td>0.999843</td>\n",
       "      <td>0.781063</td>\n",
       "      <td>1.080624</td>\n",
       "      <td>0.946615</td>\n",
       "      <td>1.150672</td>\n",
       "      <td>1.972796</td>\n",
       "      <td>...</td>\n",
       "      <td>0.783025</td>\n",
       "      <td>1.234318</td>\n",
       "      <td>1.035702</td>\n",
       "      <td>0.986356</td>\n",
       "      <td>1.254152</td>\n",
       "      <td>0.736300</td>\n",
       "      <td>1.337263</td>\n",
       "      <td>1.610137</td>\n",
       "      <td>1.049469</td>\n",
       "      <td>1.026052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CPNE8-AS1</td>\n",
       "      <td>0.734211</td>\n",
       "      <td>1.041892</td>\n",
       "      <td>1.420911</td>\n",
       "      <td>0.998373</td>\n",
       "      <td>0.879289</td>\n",
       "      <td>1.140088</td>\n",
       "      <td>1.009862</td>\n",
       "      <td>0.809097</td>\n",
       "      <td>1.380724</td>\n",
       "      <td>...</td>\n",
       "      <td>1.124372</td>\n",
       "      <td>0.863252</td>\n",
       "      <td>0.979797</td>\n",
       "      <td>0.976575</td>\n",
       "      <td>0.937251</td>\n",
       "      <td>1.119763</td>\n",
       "      <td>1.029053</td>\n",
       "      <td>0.969262</td>\n",
       "      <td>1.127606</td>\n",
       "      <td>1.005615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38585</th>\n",
       "      <td>FXYD5</td>\n",
       "      <td>0.962284</td>\n",
       "      <td>0.998635</td>\n",
       "      <td>1.434221</td>\n",
       "      <td>0.924157</td>\n",
       "      <td>0.811873</td>\n",
       "      <td>1.030370</td>\n",
       "      <td>0.874268</td>\n",
       "      <td>1.010786</td>\n",
       "      <td>0.962975</td>\n",
       "      <td>...</td>\n",
       "      <td>0.768815</td>\n",
       "      <td>1.123477</td>\n",
       "      <td>0.848746</td>\n",
       "      <td>0.977653</td>\n",
       "      <td>0.965869</td>\n",
       "      <td>0.773894</td>\n",
       "      <td>1.639451</td>\n",
       "      <td>1.053836</td>\n",
       "      <td>0.927493</td>\n",
       "      <td>1.002959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38586</th>\n",
       "      <td>RNA5SP154</td>\n",
       "      <td>1.009244</td>\n",
       "      <td>1.089437</td>\n",
       "      <td>0.878627</td>\n",
       "      <td>0.946639</td>\n",
       "      <td>0.872874</td>\n",
       "      <td>1.074350</td>\n",
       "      <td>1.028733</td>\n",
       "      <td>0.648140</td>\n",
       "      <td>1.073053</td>\n",
       "      <td>...</td>\n",
       "      <td>0.706832</td>\n",
       "      <td>1.158749</td>\n",
       "      <td>0.520117</td>\n",
       "      <td>0.957798</td>\n",
       "      <td>0.947472</td>\n",
       "      <td>0.827995</td>\n",
       "      <td>0.685640</td>\n",
       "      <td>0.761691</td>\n",
       "      <td>0.890273</td>\n",
       "      <td>1.003016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38587</th>\n",
       "      <td>TMEM87B</td>\n",
       "      <td>1.060017</td>\n",
       "      <td>1.020576</td>\n",
       "      <td>0.764194</td>\n",
       "      <td>0.935294</td>\n",
       "      <td>1.147173</td>\n",
       "      <td>1.239277</td>\n",
       "      <td>0.702147</td>\n",
       "      <td>1.166930</td>\n",
       "      <td>1.049231</td>\n",
       "      <td>...</td>\n",
       "      <td>1.071300</td>\n",
       "      <td>0.895312</td>\n",
       "      <td>0.971115</td>\n",
       "      <td>0.972381</td>\n",
       "      <td>0.961541</td>\n",
       "      <td>0.785493</td>\n",
       "      <td>1.565325</td>\n",
       "      <td>1.089399</td>\n",
       "      <td>0.984712</td>\n",
       "      <td>1.018937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38588</th>\n",
       "      <td>RNU6-1313P</td>\n",
       "      <td>0.698082</td>\n",
       "      <td>0.989170</td>\n",
       "      <td>1.047771</td>\n",
       "      <td>2.055489</td>\n",
       "      <td>1.238517</td>\n",
       "      <td>1.097696</td>\n",
       "      <td>0.962941</td>\n",
       "      <td>1.273156</td>\n",
       "      <td>1.217933</td>\n",
       "      <td>...</td>\n",
       "      <td>1.036596</td>\n",
       "      <td>0.909479</td>\n",
       "      <td>0.828142</td>\n",
       "      <td>1.008336</td>\n",
       "      <td>1.267007</td>\n",
       "      <td>1.484038</td>\n",
       "      <td>1.343979</td>\n",
       "      <td>0.775674</td>\n",
       "      <td>1.097954</td>\n",
       "      <td>0.989859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38589</th>\n",
       "      <td>RNU6-1158P</td>\n",
       "      <td>0.698082</td>\n",
       "      <td>0.989170</td>\n",
       "      <td>1.047771</td>\n",
       "      <td>1.946219</td>\n",
       "      <td>1.394133</td>\n",
       "      <td>1.097696</td>\n",
       "      <td>1.234954</td>\n",
       "      <td>1.352245</td>\n",
       "      <td>1.217933</td>\n",
       "      <td>...</td>\n",
       "      <td>1.036596</td>\n",
       "      <td>0.909479</td>\n",
       "      <td>0.828142</td>\n",
       "      <td>1.008336</td>\n",
       "      <td>1.267007</td>\n",
       "      <td>2.095260</td>\n",
       "      <td>1.343979</td>\n",
       "      <td>0.771612</td>\n",
       "      <td>1.097954</td>\n",
       "      <td>0.989859</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38590 rows × 1930 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0   gene_name  ACH-000628  ACH-000974  ACH-003071  ACH-001338  \\\n",
       "0                 RHEB    0.733588    0.950736    1.125149    1.491525   \n",
       "1                TIPIN    1.273030    1.000040    0.738668    0.988869   \n",
       "2               OR4A47    1.190365    1.048747    1.170261    0.996618   \n",
       "3               NUP133    1.409367    1.163154    0.779756    0.999843   \n",
       "4            CPNE8-AS1    0.734211    1.041892    1.420911    0.998373   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "38585            FXYD5    0.962284    0.998635    1.434221    0.924157   \n",
       "38586        RNA5SP154    1.009244    1.089437    0.878627    0.946639   \n",
       "38587          TMEM87B    1.060017    1.020576    0.764194    0.935294   \n",
       "38588       RNU6-1313P    0.698082    0.989170    1.047771    2.055489   \n",
       "38589       RNU6-1158P    0.698082    0.989170    1.047771    1.946219   \n",
       "\n",
       "Unnamed: 0  ACH-001608  ACH-000278  ACH-001350  ACH-000150  ACH-001132  ...  \\\n",
       "0             1.268255    1.593313    0.768985    1.678589    1.533440  ...   \n",
       "1             0.799244    1.114384    0.909451    1.142067    0.736635  ...   \n",
       "2             1.378248    1.189679    1.371610    0.971120    1.056310  ...   \n",
       "3             0.781063    1.080624    0.946615    1.150672    1.972796  ...   \n",
       "4             0.879289    1.140088    1.009862    0.809097    1.380724  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "38585         0.811873    1.030370    0.874268    1.010786    0.962975  ...   \n",
       "38586         0.872874    1.074350    1.028733    0.648140    1.073053  ...   \n",
       "38587         1.147173    1.239277    0.702147    1.166930    1.049231  ...   \n",
       "38588         1.238517    1.097696    0.962941    1.273156    1.217933  ...   \n",
       "38589         1.394133    1.097696    1.234954    1.352245    1.217933  ...   \n",
       "\n",
       "Unnamed: 0  ACH-000236  ACH-002177  ACH-002163  ACH-002316  ACH-001039  \\\n",
       "0             1.732311    0.891360    1.335196    0.988235    0.962098   \n",
       "1             0.983290    0.899519    1.022967    0.989033    1.215076   \n",
       "2             1.224336    0.940104    0.741997    0.995188    0.935269   \n",
       "3             0.783025    1.234318    1.035702    0.986356    1.254152   \n",
       "4             1.124372    0.863252    0.979797    0.976575    0.937251   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "38585         0.768815    1.123477    0.848746    0.977653    0.965869   \n",
       "38586         0.706832    1.158749    0.520117    0.957798    0.947472   \n",
       "38587         1.071300    0.895312    0.971115    0.972381    0.961541   \n",
       "38588         1.036596    0.909479    0.828142    1.008336    1.267007   \n",
       "38589         1.036596    0.909479    0.828142    1.008336    1.267007   \n",
       "\n",
       "Unnamed: 0  ACH-001389  ACH-001835  ACH-000296  ACH-001971  ACH-002276  \n",
       "0             1.786448    0.713098    1.156063    2.152319    0.995697  \n",
       "1             1.942519    1.058476    0.851183    1.180349    0.964208  \n",
       "2             1.124677    1.442339    1.128537    1.208479    0.988156  \n",
       "3             0.736300    1.337263    1.610137    1.049469    1.026052  \n",
       "4             1.119763    1.029053    0.969262    1.127606    1.005615  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "38585         0.773894    1.639451    1.053836    0.927493    1.002959  \n",
       "38586         0.827995    0.685640    0.761691    0.890273    1.003016  \n",
       "38587         0.785493    1.565325    1.089399    0.984712    1.018937  \n",
       "38588         1.484038    1.343979    0.775674    1.097954    0.989859  \n",
       "38589         2.095260    1.343979    0.771612    1.097954    0.989859  \n",
       "\n",
       "[38590 rows x 1930 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Identify columns to convert (all except 'Unnamed: 0' and 'gene_name')\n",
    "cols_to_convert = raw_gene_t_df.columns.difference(['Unnamed: 0', 'gene_name'])\n",
    "# Convert these columns to numeric (using pd.to_numeric for safety, which can handle errors)\n",
    "raw_gene_t_df[cols_to_convert] = raw_gene_t_df[cols_to_convert].apply(pd.to_numeric, errors='coerce')\n",
    "raw_gene_t_df = raw_gene_t_df.fillna(0.0)\n",
    "display(raw_gene_t_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4156208938.py:2: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  gene_df = raw_gene_t_df.groupby('gene_name', as_index=False).mean()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000628</th>\n",
       "      <th>ACH-000974</th>\n",
       "      <th>ACH-003071</th>\n",
       "      <th>ACH-001338</th>\n",
       "      <th>ACH-001608</th>\n",
       "      <th>ACH-000278</th>\n",
       "      <th>ACH-001350</th>\n",
       "      <th>ACH-000150</th>\n",
       "      <th>ACH-001132</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000236</th>\n",
       "      <th>ACH-002177</th>\n",
       "      <th>ACH-002163</th>\n",
       "      <th>ACH-002316</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001389</th>\n",
       "      <th>ACH-001835</th>\n",
       "      <th>ACH-000296</th>\n",
       "      <th>ACH-001971</th>\n",
       "      <th>ACH-002276</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>1.268945</td>\n",
       "      <td>0.998635</td>\n",
       "      <td>1.449249</td>\n",
       "      <td>0.974584</td>\n",
       "      <td>0.766355</td>\n",
       "      <td>1.149191</td>\n",
       "      <td>0.814400</td>\n",
       "      <td>1.010786</td>\n",
       "      <td>1.384487</td>\n",
       "      <td>...</td>\n",
       "      <td>1.462479</td>\n",
       "      <td>1.137881</td>\n",
       "      <td>0.825004</td>\n",
       "      <td>0.977653</td>\n",
       "      <td>0.993997</td>\n",
       "      <td>0.732935</td>\n",
       "      <td>1.641815</td>\n",
       "      <td>1.170731</td>\n",
       "      <td>0.926029</td>\n",
       "      <td>1.028622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>1.268945</td>\n",
       "      <td>0.998635</td>\n",
       "      <td>1.449249</td>\n",
       "      <td>0.974584</td>\n",
       "      <td>0.766355</td>\n",
       "      <td>1.149191</td>\n",
       "      <td>0.814400</td>\n",
       "      <td>1.010786</td>\n",
       "      <td>1.384487</td>\n",
       "      <td>...</td>\n",
       "      <td>1.462479</td>\n",
       "      <td>1.137881</td>\n",
       "      <td>0.825004</td>\n",
       "      <td>0.977653</td>\n",
       "      <td>0.993997</td>\n",
       "      <td>0.732935</td>\n",
       "      <td>1.641815</td>\n",
       "      <td>1.170731</td>\n",
       "      <td>0.926029</td>\n",
       "      <td>1.028622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.731812</td>\n",
       "      <td>1.030577</td>\n",
       "      <td>0.765131</td>\n",
       "      <td>0.996227</td>\n",
       "      <td>1.399586</td>\n",
       "      <td>1.174703</td>\n",
       "      <td>1.061718</td>\n",
       "      <td>0.923442</td>\n",
       "      <td>1.053757</td>\n",
       "      <td>...</td>\n",
       "      <td>1.041980</td>\n",
       "      <td>0.611262</td>\n",
       "      <td>0.532489</td>\n",
       "      <td>0.984514</td>\n",
       "      <td>0.949537</td>\n",
       "      <td>0.824844</td>\n",
       "      <td>1.977986</td>\n",
       "      <td>1.060355</td>\n",
       "      <td>0.935096</td>\n",
       "      <td>1.004427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.700969</td>\n",
       "      <td>1.041507</td>\n",
       "      <td>0.779162</td>\n",
       "      <td>0.834302</td>\n",
       "      <td>0.863143</td>\n",
       "      <td>1.154409</td>\n",
       "      <td>1.352536</td>\n",
       "      <td>1.010928</td>\n",
       "      <td>2.104212</td>\n",
       "      <td>...</td>\n",
       "      <td>1.123299</td>\n",
       "      <td>0.863252</td>\n",
       "      <td>1.034677</td>\n",
       "      <td>0.976575</td>\n",
       "      <td>0.937251</td>\n",
       "      <td>1.122751</td>\n",
       "      <td>1.029053</td>\n",
       "      <td>1.122235</td>\n",
       "      <td>1.127606</td>\n",
       "      <td>1.005615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M-AS1</td>\n",
       "      <td>0.700969</td>\n",
       "      <td>1.041507</td>\n",
       "      <td>0.779162</td>\n",
       "      <td>0.834302</td>\n",
       "      <td>0.863143</td>\n",
       "      <td>1.154409</td>\n",
       "      <td>1.352536</td>\n",
       "      <td>1.010928</td>\n",
       "      <td>2.104212</td>\n",
       "      <td>...</td>\n",
       "      <td>1.123299</td>\n",
       "      <td>0.863252</td>\n",
       "      <td>1.034677</td>\n",
       "      <td>0.976575</td>\n",
       "      <td>0.937251</td>\n",
       "      <td>1.122751</td>\n",
       "      <td>1.029053</td>\n",
       "      <td>1.122235</td>\n",
       "      <td>1.127606</td>\n",
       "      <td>1.005615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38585</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.999906</td>\n",
       "      <td>1.021629</td>\n",
       "      <td>1.179644</td>\n",
       "      <td>1.007182</td>\n",
       "      <td>1.615836</td>\n",
       "      <td>1.114022</td>\n",
       "      <td>1.250759</td>\n",
       "      <td>0.662718</td>\n",
       "      <td>0.974934</td>\n",
       "      <td>...</td>\n",
       "      <td>0.703032</td>\n",
       "      <td>1.199221</td>\n",
       "      <td>1.027847</td>\n",
       "      <td>0.968364</td>\n",
       "      <td>0.945624</td>\n",
       "      <td>1.534429</td>\n",
       "      <td>1.300574</td>\n",
       "      <td>1.150561</td>\n",
       "      <td>0.865839</td>\n",
       "      <td>0.996423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38586</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.733588</td>\n",
       "      <td>0.950736</td>\n",
       "      <td>1.038878</td>\n",
       "      <td>1.451015</td>\n",
       "      <td>1.268255</td>\n",
       "      <td>1.529940</td>\n",
       "      <td>0.764897</td>\n",
       "      <td>1.831970</td>\n",
       "      <td>1.509656</td>\n",
       "      <td>...</td>\n",
       "      <td>1.732311</td>\n",
       "      <td>0.639230</td>\n",
       "      <td>1.335196</td>\n",
       "      <td>0.988235</td>\n",
       "      <td>0.962098</td>\n",
       "      <td>1.786448</td>\n",
       "      <td>0.713098</td>\n",
       "      <td>0.871164</td>\n",
       "      <td>2.152319</td>\n",
       "      <td>0.995697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38587</th>\n",
       "      <td>ZYXP1</td>\n",
       "      <td>1.338678</td>\n",
       "      <td>1.289464</td>\n",
       "      <td>1.258723</td>\n",
       "      <td>1.002211</td>\n",
       "      <td>0.759560</td>\n",
       "      <td>2.222632</td>\n",
       "      <td>2.029633</td>\n",
       "      <td>0.492568</td>\n",
       "      <td>1.045696</td>\n",
       "      <td>...</td>\n",
       "      <td>1.396531</td>\n",
       "      <td>1.167229</td>\n",
       "      <td>0.939060</td>\n",
       "      <td>0.965529</td>\n",
       "      <td>0.916855</td>\n",
       "      <td>0.836631</td>\n",
       "      <td>1.088331</td>\n",
       "      <td>1.269616</td>\n",
       "      <td>1.108575</td>\n",
       "      <td>0.998080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38588</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.991158</td>\n",
       "      <td>0.981948</td>\n",
       "      <td>1.442267</td>\n",
       "      <td>0.935950</td>\n",
       "      <td>0.813832</td>\n",
       "      <td>0.572681</td>\n",
       "      <td>0.645097</td>\n",
       "      <td>1.035650</td>\n",
       "      <td>0.482337</td>\n",
       "      <td>...</td>\n",
       "      <td>0.786005</td>\n",
       "      <td>1.545586</td>\n",
       "      <td>1.254272</td>\n",
       "      <td>0.997750</td>\n",
       "      <td>1.051698</td>\n",
       "      <td>0.781627</td>\n",
       "      <td>0.662658</td>\n",
       "      <td>0.360974</td>\n",
       "      <td>0.852137</td>\n",
       "      <td>1.008135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38589</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>1.084448</td>\n",
       "      <td>1.021629</td>\n",
       "      <td>1.152662</td>\n",
       "      <td>1.024286</td>\n",
       "      <td>1.652248</td>\n",
       "      <td>1.196307</td>\n",
       "      <td>1.375039</td>\n",
       "      <td>0.719510</td>\n",
       "      <td>1.050575</td>\n",
       "      <td>...</td>\n",
       "      <td>0.617262</td>\n",
       "      <td>1.185440</td>\n",
       "      <td>1.027847</td>\n",
       "      <td>0.968364</td>\n",
       "      <td>0.624576</td>\n",
       "      <td>1.534429</td>\n",
       "      <td>1.300574</td>\n",
       "      <td>1.150561</td>\n",
       "      <td>0.865839</td>\n",
       "      <td>0.996423</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38590 rows × 1930 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 gene_name  ACH-000628  ACH-000974  ACH-003071  ACH-001338  \\\n",
       "0               A1BG    1.268945    0.998635    1.449249    0.974584   \n",
       "1           A1BG-AS1    1.268945    0.998635    1.449249    0.974584   \n",
       "2               A1CF    0.731812    1.030577    0.765131    0.996227   \n",
       "3                A2M    0.700969    1.041507    0.779162    0.834302   \n",
       "4            A2M-AS1    0.700969    1.041507    0.779162    0.834302   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "38585         ZYG11B    0.999906    1.021629    1.179644    1.007182   \n",
       "38586            ZYX    0.733588    0.950736    1.038878    1.451015   \n",
       "38587          ZYXP1    1.338678    1.289464    1.258723    1.002211   \n",
       "38588          ZZEF1    0.991158    0.981948    1.442267    0.935950   \n",
       "38589           ZZZ3    1.084448    1.021629    1.152662    1.024286   \n",
       "\n",
       "Unnamed: 0  ACH-001608  ACH-000278  ACH-001350  ACH-000150  ACH-001132  ...  \\\n",
       "0             0.766355    1.149191    0.814400    1.010786    1.384487  ...   \n",
       "1             0.766355    1.149191    0.814400    1.010786    1.384487  ...   \n",
       "2             1.399586    1.174703    1.061718    0.923442    1.053757  ...   \n",
       "3             0.863143    1.154409    1.352536    1.010928    2.104212  ...   \n",
       "4             0.863143    1.154409    1.352536    1.010928    2.104212  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "38585         1.615836    1.114022    1.250759    0.662718    0.974934  ...   \n",
       "38586         1.268255    1.529940    0.764897    1.831970    1.509656  ...   \n",
       "38587         0.759560    2.222632    2.029633    0.492568    1.045696  ...   \n",
       "38588         0.813832    0.572681    0.645097    1.035650    0.482337  ...   \n",
       "38589         1.652248    1.196307    1.375039    0.719510    1.050575  ...   \n",
       "\n",
       "Unnamed: 0  ACH-000236  ACH-002177  ACH-002163  ACH-002316  ACH-001039  \\\n",
       "0             1.462479    1.137881    0.825004    0.977653    0.993997   \n",
       "1             1.462479    1.137881    0.825004    0.977653    0.993997   \n",
       "2             1.041980    0.611262    0.532489    0.984514    0.949537   \n",
       "3             1.123299    0.863252    1.034677    0.976575    0.937251   \n",
       "4             1.123299    0.863252    1.034677    0.976575    0.937251   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "38585         0.703032    1.199221    1.027847    0.968364    0.945624   \n",
       "38586         1.732311    0.639230    1.335196    0.988235    0.962098   \n",
       "38587         1.396531    1.167229    0.939060    0.965529    0.916855   \n",
       "38588         0.786005    1.545586    1.254272    0.997750    1.051698   \n",
       "38589         0.617262    1.185440    1.027847    0.968364    0.624576   \n",
       "\n",
       "Unnamed: 0  ACH-001389  ACH-001835  ACH-000296  ACH-001971  ACH-002276  \n",
       "0             0.732935    1.641815    1.170731    0.926029    1.028622  \n",
       "1             0.732935    1.641815    1.170731    0.926029    1.028622  \n",
       "2             0.824844    1.977986    1.060355    0.935096    1.004427  \n",
       "3             1.122751    1.029053    1.122235    1.127606    1.005615  \n",
       "4             1.122751    1.029053    1.122235    1.127606    1.005615  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "38585         1.534429    1.300574    1.150561    0.865839    0.996423  \n",
       "38586         1.786448    0.713098    0.871164    2.152319    0.995697  \n",
       "38587         0.836631    1.088331    1.269616    1.108575    0.998080  \n",
       "38588         0.781627    0.662658    0.360974    0.852137    1.008135  \n",
       "38589         1.534429    1.300574    1.150561    0.865839    0.996423  \n",
       "\n",
       "[38590 rows x 1930 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# groupby the gene_name and calculate the mean\n",
    "gene_df = raw_gene_t_df.groupby('gene_name', as_index=False).mean()\n",
    "display(gene_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\2858376915.py:1: DtypeWarning: Columns (4,5,6,11,12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmg_gene_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>HGNC_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>NAT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>ADA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>CDH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>AKT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_GN86234</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_GN86235</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_GN86236</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID HGNC_Symbol\n",
       "0               BMGC_GN00001        A1BG\n",
       "1               BMGC_GN00002        NAT2\n",
       "2               BMGC_GN00003         ADA\n",
       "3               BMGC_GN00004        CDH2\n",
       "4               BMGC_GN00005        AKT3\n",
       "...                      ...         ...\n",
       "86233           BMGC_GN86234         NaN\n",
       "86234           BMGC_GN86235         NaN\n",
       "86235           BMGC_GN86236         NaN\n",
       "86236           BMGC_GN86237         NaN\n",
       "86237           BMGC_GN86238         NaN\n",
       "\n",
       "[86238 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bmg_gene_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene.csv')\n",
    "# keep BioMedGraphica_ID and HGNC_Symbol\n",
    "bmg_gene_df = bmg_gene_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']]\n",
    "display(bmg_gene_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000628</th>\n",
       "      <th>ACH-000974</th>\n",
       "      <th>ACH-003071</th>\n",
       "      <th>ACH-001338</th>\n",
       "      <th>ACH-001608</th>\n",
       "      <th>ACH-000278</th>\n",
       "      <th>ACH-001350</th>\n",
       "      <th>ACH-000150</th>\n",
       "      <th>ACH-001132</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000236</th>\n",
       "      <th>ACH-002177</th>\n",
       "      <th>ACH-002163</th>\n",
       "      <th>ACH-002316</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001389</th>\n",
       "      <th>ACH-001835</th>\n",
       "      <th>ACH-000296</th>\n",
       "      <th>ACH-001971</th>\n",
       "      <th>ACH-002276</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>1.268945</td>\n",
       "      <td>0.998635</td>\n",
       "      <td>1.449249</td>\n",
       "      <td>0.974584</td>\n",
       "      <td>0.766355</td>\n",
       "      <td>1.149191</td>\n",
       "      <td>0.814400</td>\n",
       "      <td>1.010786</td>\n",
       "      <td>1.384487</td>\n",
       "      <td>...</td>\n",
       "      <td>1.462479</td>\n",
       "      <td>1.137881</td>\n",
       "      <td>0.825004</td>\n",
       "      <td>0.977653</td>\n",
       "      <td>0.993997</td>\n",
       "      <td>0.732935</td>\n",
       "      <td>1.641815</td>\n",
       "      <td>1.170731</td>\n",
       "      <td>0.926029</td>\n",
       "      <td>1.028622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>0.731485</td>\n",
       "      <td>0.799281</td>\n",
       "      <td>1.156498</td>\n",
       "      <td>1.005014</td>\n",
       "      <td>0.748238</td>\n",
       "      <td>0.611852</td>\n",
       "      <td>0.696108</td>\n",
       "      <td>0.473682</td>\n",
       "      <td>1.032457</td>\n",
       "      <td>...</td>\n",
       "      <td>0.349017</td>\n",
       "      <td>0.890035</td>\n",
       "      <td>0.474526</td>\n",
       "      <td>0.965529</td>\n",
       "      <td>0.589237</td>\n",
       "      <td>1.145237</td>\n",
       "      <td>1.088331</td>\n",
       "      <td>0.498268</td>\n",
       "      <td>1.108575</td>\n",
       "      <td>0.998080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>1.882274</td>\n",
       "      <td>1.056340</td>\n",
       "      <td>1.052670</td>\n",
       "      <td>1.430607</td>\n",
       "      <td>1.535008</td>\n",
       "      <td>1.188952</td>\n",
       "      <td>1.832935</td>\n",
       "      <td>1.380597</td>\n",
       "      <td>0.773227</td>\n",
       "      <td>...</td>\n",
       "      <td>2.058987</td>\n",
       "      <td>0.898108</td>\n",
       "      <td>0.977277</td>\n",
       "      <td>1.017972</td>\n",
       "      <td>1.457635</td>\n",
       "      <td>0.726367</td>\n",
       "      <td>1.019664</td>\n",
       "      <td>1.282703</td>\n",
       "      <td>1.600703</td>\n",
       "      <td>0.989567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>0.946750</td>\n",
       "      <td>1.056789</td>\n",
       "      <td>1.431942</td>\n",
       "      <td>0.978040</td>\n",
       "      <td>0.932105</td>\n",
       "      <td>1.107280</td>\n",
       "      <td>1.360372</td>\n",
       "      <td>0.912048</td>\n",
       "      <td>1.064972</td>\n",
       "      <td>...</td>\n",
       "      <td>0.731571</td>\n",
       "      <td>0.919491</td>\n",
       "      <td>0.774345</td>\n",
       "      <td>0.972422</td>\n",
       "      <td>0.630750</td>\n",
       "      <td>0.394350</td>\n",
       "      <td>1.315620</td>\n",
       "      <td>0.338280</td>\n",
       "      <td>0.860810</td>\n",
       "      <td>1.010605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>1.409367</td>\n",
       "      <td>1.163154</td>\n",
       "      <td>0.803250</td>\n",
       "      <td>0.999843</td>\n",
       "      <td>1.134695</td>\n",
       "      <td>1.080624</td>\n",
       "      <td>0.946615</td>\n",
       "      <td>1.150672</td>\n",
       "      <td>2.083959</td>\n",
       "      <td>...</td>\n",
       "      <td>0.783025</td>\n",
       "      <td>1.234318</td>\n",
       "      <td>1.035702</td>\n",
       "      <td>0.986356</td>\n",
       "      <td>1.254152</td>\n",
       "      <td>0.736300</td>\n",
       "      <td>1.594780</td>\n",
       "      <td>1.164666</td>\n",
       "      <td>1.049469</td>\n",
       "      <td>1.026052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42798</th>\n",
       "      <td>BMGC_GN70228</td>\n",
       "      <td>0.945141</td>\n",
       "      <td>1.048747</td>\n",
       "      <td>1.134761</td>\n",
       "      <td>0.996618</td>\n",
       "      <td>0.722325</td>\n",
       "      <td>0.531759</td>\n",
       "      <td>1.256330</td>\n",
       "      <td>0.835083</td>\n",
       "      <td>0.989560</td>\n",
       "      <td>...</td>\n",
       "      <td>1.224336</td>\n",
       "      <td>0.449821</td>\n",
       "      <td>1.013743</td>\n",
       "      <td>0.995188</td>\n",
       "      <td>0.935269</td>\n",
       "      <td>1.069099</td>\n",
       "      <td>0.816622</td>\n",
       "      <td>1.020159</td>\n",
       "      <td>0.912347</td>\n",
       "      <td>0.988156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42799</th>\n",
       "      <td>BMGC_GN70238</td>\n",
       "      <td>4.304596</td>\n",
       "      <td>12.632883</td>\n",
       "      <td>12.244713</td>\n",
       "      <td>1.533437</td>\n",
       "      <td>4.385596</td>\n",
       "      <td>2.726301</td>\n",
       "      <td>8.075323</td>\n",
       "      <td>1.273156</td>\n",
       "      <td>18.004317</td>\n",
       "      <td>...</td>\n",
       "      <td>1.036596</td>\n",
       "      <td>0.909479</td>\n",
       "      <td>2.239729</td>\n",
       "      <td>0.997750</td>\n",
       "      <td>1.258731</td>\n",
       "      <td>0.781627</td>\n",
       "      <td>4.005427</td>\n",
       "      <td>2.065520</td>\n",
       "      <td>1.526971</td>\n",
       "      <td>3.329526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42800</th>\n",
       "      <td>BMGC_GN70239</td>\n",
       "      <td>4.304596</td>\n",
       "      <td>12.632883</td>\n",
       "      <td>12.244713</td>\n",
       "      <td>1.533437</td>\n",
       "      <td>4.385596</td>\n",
       "      <td>2.726301</td>\n",
       "      <td>8.075323</td>\n",
       "      <td>1.273156</td>\n",
       "      <td>18.004317</td>\n",
       "      <td>...</td>\n",
       "      <td>1.036596</td>\n",
       "      <td>0.909479</td>\n",
       "      <td>2.239729</td>\n",
       "      <td>0.997750</td>\n",
       "      <td>1.258731</td>\n",
       "      <td>0.781627</td>\n",
       "      <td>4.005427</td>\n",
       "      <td>2.065520</td>\n",
       "      <td>1.526971</td>\n",
       "      <td>3.329526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42801</th>\n",
       "      <td>BMGC_GN70273</td>\n",
       "      <td>0.717960</td>\n",
       "      <td>1.041892</td>\n",
       "      <td>1.190471</td>\n",
       "      <td>0.998373</td>\n",
       "      <td>1.032063</td>\n",
       "      <td>0.619187</td>\n",
       "      <td>1.240514</td>\n",
       "      <td>0.632949</td>\n",
       "      <td>1.250405</td>\n",
       "      <td>...</td>\n",
       "      <td>1.093958</td>\n",
       "      <td>1.178416</td>\n",
       "      <td>1.052442</td>\n",
       "      <td>0.976575</td>\n",
       "      <td>1.221468</td>\n",
       "      <td>0.734510</td>\n",
       "      <td>1.029053</td>\n",
       "      <td>1.119342</td>\n",
       "      <td>1.127606</td>\n",
       "      <td>1.005615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42802</th>\n",
       "      <td>BMGC_GN86221</td>\n",
       "      <td>0.733141</td>\n",
       "      <td>1.065895</td>\n",
       "      <td>0.718076</td>\n",
       "      <td>0.457805</td>\n",
       "      <td>0.584762</td>\n",
       "      <td>0.638456</td>\n",
       "      <td>0.690094</td>\n",
       "      <td>0.569980</td>\n",
       "      <td>1.051230</td>\n",
       "      <td>...</td>\n",
       "      <td>0.684748</td>\n",
       "      <td>0.389129</td>\n",
       "      <td>0.684839</td>\n",
       "      <td>0.485461</td>\n",
       "      <td>0.547492</td>\n",
       "      <td>0.753132</td>\n",
       "      <td>0.647551</td>\n",
       "      <td>0.333722</td>\n",
       "      <td>0.967685</td>\n",
       "      <td>0.837707</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42803 rows × 1930 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000628  ACH-000974  ACH-003071  ACH-001338  \\\n",
       "0               BMGC_GN00001    1.268945    0.998635    1.449249    0.974584   \n",
       "1               BMGC_GN00002    0.731485    0.799281    1.156498    1.005014   \n",
       "2               BMGC_GN00003    1.882274    1.056340    1.052670    1.430607   \n",
       "3               BMGC_GN00004    0.946750    1.056789    1.431942    0.978040   \n",
       "4               BMGC_GN00005    1.409367    1.163154    0.803250    0.999843   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "42798           BMGC_GN70228    0.945141    1.048747    1.134761    0.996618   \n",
       "42799           BMGC_GN70238    4.304596   12.632883   12.244713    1.533437   \n",
       "42800           BMGC_GN70239    4.304596   12.632883   12.244713    1.533437   \n",
       "42801           BMGC_GN70273    0.717960    1.041892    1.190471    0.998373   \n",
       "42802           BMGC_GN86221    0.733141    1.065895    0.718076    0.457805   \n",
       "\n",
       "       ACH-001608  ACH-000278  ACH-001350  ACH-000150  ACH-001132  ...  \\\n",
       "0        0.766355    1.149191    0.814400    1.010786    1.384487  ...   \n",
       "1        0.748238    0.611852    0.696108    0.473682    1.032457  ...   \n",
       "2        1.535008    1.188952    1.832935    1.380597    0.773227  ...   \n",
       "3        0.932105    1.107280    1.360372    0.912048    1.064972  ...   \n",
       "4        1.134695    1.080624    0.946615    1.150672    2.083959  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "42798    0.722325    0.531759    1.256330    0.835083    0.989560  ...   \n",
       "42799    4.385596    2.726301    8.075323    1.273156   18.004317  ...   \n",
       "42800    4.385596    2.726301    8.075323    1.273156   18.004317  ...   \n",
       "42801    1.032063    0.619187    1.240514    0.632949    1.250405  ...   \n",
       "42802    0.584762    0.638456    0.690094    0.569980    1.051230  ...   \n",
       "\n",
       "       ACH-000236  ACH-002177  ACH-002163  ACH-002316  ACH-001039  ACH-001389  \\\n",
       "0        1.462479    1.137881    0.825004    0.977653    0.993997    0.732935   \n",
       "1        0.349017    0.890035    0.474526    0.965529    0.589237    1.145237   \n",
       "2        2.058987    0.898108    0.977277    1.017972    1.457635    0.726367   \n",
       "3        0.731571    0.919491    0.774345    0.972422    0.630750    0.394350   \n",
       "4        0.783025    1.234318    1.035702    0.986356    1.254152    0.736300   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "42798    1.224336    0.449821    1.013743    0.995188    0.935269    1.069099   \n",
       "42799    1.036596    0.909479    2.239729    0.997750    1.258731    0.781627   \n",
       "42800    1.036596    0.909479    2.239729    0.997750    1.258731    0.781627   \n",
       "42801    1.093958    1.178416    1.052442    0.976575    1.221468    0.734510   \n",
       "42802    0.684748    0.389129    0.684839    0.485461    0.547492    0.753132   \n",
       "\n",
       "       ACH-001835  ACH-000296  ACH-001971  ACH-002276  \n",
       "0        1.641815    1.170731    0.926029    1.028622  \n",
       "1        1.088331    0.498268    1.108575    0.998080  \n",
       "2        1.019664    1.282703    1.600703    0.989567  \n",
       "3        1.315620    0.338280    0.860810    1.010605  \n",
       "4        1.594780    1.164666    1.049469    1.026052  \n",
       "...           ...         ...         ...         ...  \n",
       "42798    0.816622    1.020159    0.912347    0.988156  \n",
       "42799    4.005427    2.065520    1.526971    3.329526  \n",
       "42800    4.005427    2.065520    1.526971    3.329526  \n",
       "42801    1.029053    1.119342    1.127606    1.005615  \n",
       "42802    0.647551    0.333722    0.967685    0.837707  \n",
       "\n",
       "[42803 rows x 1930 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# merge the biomedgraphica_id with the raw_gene_t_df\n",
    "merged_gene_df = pd.merge(bmg_gene_df, gene_df, left_on='HGNC_Symbol', right_on='gene_name', how='inner')\n",
    "merged_gene_df.drop(columns=['HGNC_Symbol', 'gene_name'], inplace=True)\n",
    "display(merged_gene_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000628</th>\n",
       "      <th>ACH-000974</th>\n",
       "      <th>ACH-003071</th>\n",
       "      <th>ACH-001338</th>\n",
       "      <th>ACH-001608</th>\n",
       "      <th>ACH-000278</th>\n",
       "      <th>ACH-001350</th>\n",
       "      <th>ACH-000150</th>\n",
       "      <th>ACH-001132</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000236</th>\n",
       "      <th>ACH-002177</th>\n",
       "      <th>ACH-002163</th>\n",
       "      <th>ACH-002316</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001389</th>\n",
       "      <th>ACH-001835</th>\n",
       "      <th>ACH-000296</th>\n",
       "      <th>ACH-001971</th>\n",
       "      <th>ACH-002276</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>1.268945</td>\n",
       "      <td>0.998635</td>\n",
       "      <td>1.449249</td>\n",
       "      <td>0.974584</td>\n",
       "      <td>0.766355</td>\n",
       "      <td>1.149191</td>\n",
       "      <td>0.814400</td>\n",
       "      <td>1.010786</td>\n",
       "      <td>1.384487</td>\n",
       "      <td>...</td>\n",
       "      <td>1.462479</td>\n",
       "      <td>1.137881</td>\n",
       "      <td>0.825004</td>\n",
       "      <td>0.977653</td>\n",
       "      <td>0.993997</td>\n",
       "      <td>0.732935</td>\n",
       "      <td>1.641815</td>\n",
       "      <td>1.170731</td>\n",
       "      <td>0.926029</td>\n",
       "      <td>1.028622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>0.731485</td>\n",
       "      <td>0.799281</td>\n",
       "      <td>1.156498</td>\n",
       "      <td>1.005014</td>\n",
       "      <td>0.748238</td>\n",
       "      <td>0.611852</td>\n",
       "      <td>0.696108</td>\n",
       "      <td>0.473682</td>\n",
       "      <td>1.032457</td>\n",
       "      <td>...</td>\n",
       "      <td>0.349017</td>\n",
       "      <td>0.890035</td>\n",
       "      <td>0.474526</td>\n",
       "      <td>0.965529</td>\n",
       "      <td>0.589237</td>\n",
       "      <td>1.145237</td>\n",
       "      <td>1.088331</td>\n",
       "      <td>0.498268</td>\n",
       "      <td>1.108575</td>\n",
       "      <td>0.998080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>1.882274</td>\n",
       "      <td>1.056340</td>\n",
       "      <td>1.052670</td>\n",
       "      <td>1.430607</td>\n",
       "      <td>1.535008</td>\n",
       "      <td>1.188952</td>\n",
       "      <td>1.832935</td>\n",
       "      <td>1.380597</td>\n",
       "      <td>0.773227</td>\n",
       "      <td>...</td>\n",
       "      <td>2.058987</td>\n",
       "      <td>0.898108</td>\n",
       "      <td>0.977277</td>\n",
       "      <td>1.017972</td>\n",
       "      <td>1.457635</td>\n",
       "      <td>0.726367</td>\n",
       "      <td>1.019664</td>\n",
       "      <td>1.282703</td>\n",
       "      <td>1.600703</td>\n",
       "      <td>0.989567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>0.946750</td>\n",
       "      <td>1.056789</td>\n",
       "      <td>1.431942</td>\n",
       "      <td>0.978040</td>\n",
       "      <td>0.932105</td>\n",
       "      <td>1.107280</td>\n",
       "      <td>1.360372</td>\n",
       "      <td>0.912048</td>\n",
       "      <td>1.064972</td>\n",
       "      <td>...</td>\n",
       "      <td>0.731571</td>\n",
       "      <td>0.919491</td>\n",
       "      <td>0.774345</td>\n",
       "      <td>0.972422</td>\n",
       "      <td>0.630750</td>\n",
       "      <td>0.394350</td>\n",
       "      <td>1.315620</td>\n",
       "      <td>0.338280</td>\n",
       "      <td>0.860810</td>\n",
       "      <td>1.010605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>1.409367</td>\n",
       "      <td>1.163154</td>\n",
       "      <td>0.803250</td>\n",
       "      <td>0.999843</td>\n",
       "      <td>1.134695</td>\n",
       "      <td>1.080624</td>\n",
       "      <td>0.946615</td>\n",
       "      <td>1.150672</td>\n",
       "      <td>2.083959</td>\n",
       "      <td>...</td>\n",
       "      <td>0.783025</td>\n",
       "      <td>1.234318</td>\n",
       "      <td>1.035702</td>\n",
       "      <td>0.986356</td>\n",
       "      <td>1.254152</td>\n",
       "      <td>0.736300</td>\n",
       "      <td>1.594780</td>\n",
       "      <td>1.164666</td>\n",
       "      <td>1.049469</td>\n",
       "      <td>1.026052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_GN86234</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_GN86235</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_GN86236</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 1930 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000628  ACH-000974  ACH-003071  ACH-001338  \\\n",
       "0               BMGC_GN00001    1.268945    0.998635    1.449249    0.974584   \n",
       "1               BMGC_GN00002    0.731485    0.799281    1.156498    1.005014   \n",
       "2               BMGC_GN00003    1.882274    1.056340    1.052670    1.430607   \n",
       "3               BMGC_GN00004    0.946750    1.056789    1.431942    0.978040   \n",
       "4               BMGC_GN00005    1.409367    1.163154    0.803250    0.999843   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_GN86234    0.000000    0.000000    0.000000    0.000000   \n",
       "86234           BMGC_GN86235    0.000000    0.000000    0.000000    0.000000   \n",
       "86235           BMGC_GN86236    0.000000    0.000000    0.000000    0.000000   \n",
       "86236           BMGC_GN86237    0.000000    0.000000    0.000000    0.000000   \n",
       "86237           BMGC_GN86238    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-001608  ACH-000278  ACH-001350  ACH-000150  ACH-001132  ...  \\\n",
       "0        0.766355    1.149191    0.814400    1.010786    1.384487  ...   \n",
       "1        0.748238    0.611852    0.696108    0.473682    1.032457  ...   \n",
       "2        1.535008    1.188952    1.832935    1.380597    0.773227  ...   \n",
       "3        0.932105    1.107280    1.360372    0.912048    1.064972  ...   \n",
       "4        1.134695    1.080624    0.946615    1.150672    2.083959  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "       ACH-000236  ACH-002177  ACH-002163  ACH-002316  ACH-001039  ACH-001389  \\\n",
       "0        1.462479    1.137881    0.825004    0.977653    0.993997    0.732935   \n",
       "1        0.349017    0.890035    0.474526    0.965529    0.589237    1.145237   \n",
       "2        2.058987    0.898108    0.977277    1.017972    1.457635    0.726367   \n",
       "3        0.731571    0.919491    0.774345    0.972422    0.630750    0.394350   \n",
       "4        0.783025    1.234318    1.035702    0.986356    1.254152    0.736300   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-001835  ACH-000296  ACH-001971  ACH-002276  \n",
       "0        1.641815    1.170731    0.926029    1.028622  \n",
       "1        1.088331    0.498268    1.108575    0.998080  \n",
       "2        1.019664    1.282703    1.600703    0.989567  \n",
       "3        1.315620    0.338280    0.860810    1.010605  \n",
       "4        1.594780    1.164666    1.049469    1.026052  \n",
       "...           ...         ...         ...         ...  \n",
       "86233    0.000000    0.000000    0.000000    0.000000  \n",
       "86234    0.000000    0.000000    0.000000    0.000000  \n",
       "86235    0.000000    0.000000    0.000000    0.000000  \n",
       "86236    0.000000    0.000000    0.000000    0.000000  \n",
       "86237    0.000000    0.000000    0.000000    0.000000  \n",
       "\n",
       "[86238 rows x 1930 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "final_merged_gene_df = pd.merge(bmg_gene_df, gene_df, left_on='HGNC_Symbol', right_on='gene_name', how='left')\n",
    "# fill the NaN with -1.0\n",
    "final_merged_gene_df = final_merged_gene_df.fillna(0.0)\n",
    "final_merged_gene_df.drop(columns=['HGNC_Symbol', 'gene_name'], inplace=True)\n",
    "display(final_merged_gene_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Transcript"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>TFE3 (7030)</th>\n",
       "      <th>MMADHC (27249)</th>\n",
       "      <th>PABPC1L (80336)</th>\n",
       "      <th>RELL2 (285613)</th>\n",
       "      <th>GPT (2875)</th>\n",
       "      <th>GUSB (2990)</th>\n",
       "      <th>H4-16 (121504)</th>\n",
       "      <th>DDB2 (1643)</th>\n",
       "      <th>RPS4X (6191)</th>\n",
       "      <th>...</th>\n",
       "      <th>OR2A42 (402317)</th>\n",
       "      <th>HOXB7 (3217)</th>\n",
       "      <th>NSA2 (10412)</th>\n",
       "      <th>UGT2B11 (10720)</th>\n",
       "      <th>ZNF747 (65988)</th>\n",
       "      <th>NAA60 (79903)</th>\n",
       "      <th>MYH4 (4622)</th>\n",
       "      <th>MTAP (4507)</th>\n",
       "      <th>C14orf93 (60686)</th>\n",
       "      <th>NR3C2 (4306)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000873</td>\n",
       "      <td>3.607691</td>\n",
       "      <td>3.531071</td>\n",
       "      <td>4.982285</td>\n",
       "      <td>2.326016</td>\n",
       "      <td>0.607679</td>\n",
       "      <td>5.118549</td>\n",
       "      <td>1.282028</td>\n",
       "      <td>4.512297</td>\n",
       "      <td>10.075438</td>\n",
       "      <td>...</td>\n",
       "      <td>0.042564</td>\n",
       "      <td>3.870581</td>\n",
       "      <td>5.178480</td>\n",
       "      <td>-0.007723</td>\n",
       "      <td>2.005498</td>\n",
       "      <td>4.213717</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>3.904800</td>\n",
       "      <td>2.190931</td>\n",
       "      <td>0.203910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000860</td>\n",
       "      <td>4.192566</td>\n",
       "      <td>6.865682</td>\n",
       "      <td>4.853288</td>\n",
       "      <td>2.725583</td>\n",
       "      <td>0.197016</td>\n",
       "      <td>5.666426</td>\n",
       "      <td>1.154470</td>\n",
       "      <td>4.306808</td>\n",
       "      <td>10.215240</td>\n",
       "      <td>...</td>\n",
       "      <td>0.031621</td>\n",
       "      <td>-0.005897</td>\n",
       "      <td>6.749713</td>\n",
       "      <td>-0.007723</td>\n",
       "      <td>1.393644</td>\n",
       "      <td>4.334535</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>4.812949</td>\n",
       "      <td>3.366939</td>\n",
       "      <td>0.370726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000439</td>\n",
       "      <td>3.265517</td>\n",
       "      <td>5.804615</td>\n",
       "      <td>1.777636</td>\n",
       "      <td>4.185835</td>\n",
       "      <td>0.389025</td>\n",
       "      <td>6.593342</td>\n",
       "      <td>0.410244</td>\n",
       "      <td>4.411381</td>\n",
       "      <td>11.644890</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.013226</td>\n",
       "      <td>-0.033868</td>\n",
       "      <td>6.877322</td>\n",
       "      <td>1.596383</td>\n",
       "      <td>3.143557</td>\n",
       "      <td>4.624226</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>4.122929</td>\n",
       "      <td>2.197136</td>\n",
       "      <td>0.015412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000318</td>\n",
       "      <td>3.705500</td>\n",
       "      <td>5.813170</td>\n",
       "      <td>4.536005</td>\n",
       "      <td>2.290409</td>\n",
       "      <td>0.471169</td>\n",
       "      <td>5.160089</td>\n",
       "      <td>2.325673</td>\n",
       "      <td>5.017968</td>\n",
       "      <td>9.163992</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.013226</td>\n",
       "      <td>3.051092</td>\n",
       "      <td>5.610027</td>\n",
       "      <td>-0.007723</td>\n",
       "      <td>2.334419</td>\n",
       "      <td>3.881219</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>0.394778</td>\n",
       "      <td>3.217293</td>\n",
       "      <td>0.277760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-001142</td>\n",
       "      <td>6.105227</td>\n",
       "      <td>7.051513</td>\n",
       "      <td>4.457555</td>\n",
       "      <td>2.543055</td>\n",
       "      <td>0.022363</td>\n",
       "      <td>4.589966</td>\n",
       "      <td>2.369987</td>\n",
       "      <td>4.105131</td>\n",
       "      <td>10.179188</td>\n",
       "      <td>...</td>\n",
       "      <td>0.053402</td>\n",
       "      <td>3.582671</td>\n",
       "      <td>7.416571</td>\n",
       "      <td>-0.007723</td>\n",
       "      <td>2.294077</td>\n",
       "      <td>4.889203</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>4.379944</td>\n",
       "      <td>2.355443</td>\n",
       "      <td>0.520338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1668</th>\n",
       "      <td>ACH-001843</td>\n",
       "      <td>5.343844</td>\n",
       "      <td>6.553367</td>\n",
       "      <td>5.674036</td>\n",
       "      <td>3.390159</td>\n",
       "      <td>0.644076</td>\n",
       "      <td>5.555928</td>\n",
       "      <td>1.037326</td>\n",
       "      <td>4.275462</td>\n",
       "      <td>11.191982</td>\n",
       "      <td>...</td>\n",
       "      <td>0.026960</td>\n",
       "      <td>5.222260</td>\n",
       "      <td>7.411799</td>\n",
       "      <td>0.010848</td>\n",
       "      <td>0.563822</td>\n",
       "      <td>4.885868</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>4.898450</td>\n",
       "      <td>2.716015</td>\n",
       "      <td>0.067286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1669</th>\n",
       "      <td>ACH-002074</td>\n",
       "      <td>5.252185</td>\n",
       "      <td>6.258400</td>\n",
       "      <td>2.872415</td>\n",
       "      <td>1.886574</td>\n",
       "      <td>0.796319</td>\n",
       "      <td>6.534374</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>4.587816</td>\n",
       "      <td>9.946776</td>\n",
       "      <td>...</td>\n",
       "      <td>0.104563</td>\n",
       "      <td>4.270369</td>\n",
       "      <td>6.188434</td>\n",
       "      <td>0.010848</td>\n",
       "      <td>3.447185</td>\n",
       "      <td>5.964889</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>4.181277</td>\n",
       "      <td>3.931034</td>\n",
       "      <td>0.695929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1670</th>\n",
       "      <td>ACH-001164</td>\n",
       "      <td>4.050893</td>\n",
       "      <td>6.248805</td>\n",
       "      <td>2.790849</td>\n",
       "      <td>1.599247</td>\n",
       "      <td>0.663686</td>\n",
       "      <td>5.533158</td>\n",
       "      <td>0.646986</td>\n",
       "      <td>5.091849</td>\n",
       "      <td>10.865960</td>\n",
       "      <td>...</td>\n",
       "      <td>0.026960</td>\n",
       "      <td>0.470043</td>\n",
       "      <td>6.898124</td>\n",
       "      <td>1.385374</td>\n",
       "      <td>2.641063</td>\n",
       "      <td>4.824132</td>\n",
       "      <td>0.022977</td>\n",
       "      <td>3.384695</td>\n",
       "      <td>2.413815</td>\n",
       "      <td>0.221968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1671</th>\n",
       "      <td>ACH-003180</td>\n",
       "      <td>5.161680</td>\n",
       "      <td>7.678250</td>\n",
       "      <td>1.932748</td>\n",
       "      <td>1.463134</td>\n",
       "      <td>0.394206</td>\n",
       "      <td>6.464239</td>\n",
       "      <td>0.360318</td>\n",
       "      <td>4.761168</td>\n",
       "      <td>9.531740</td>\n",
       "      <td>...</td>\n",
       "      <td>0.104563</td>\n",
       "      <td>4.741295</td>\n",
       "      <td>6.250017</td>\n",
       "      <td>0.010848</td>\n",
       "      <td>2.652496</td>\n",
       "      <td>5.139537</td>\n",
       "      <td>0.044994</td>\n",
       "      <td>4.773658</td>\n",
       "      <td>2.686730</td>\n",
       "      <td>0.011768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1672</th>\n",
       "      <td>ACH-001979</td>\n",
       "      <td>4.986577</td>\n",
       "      <td>7.612689</td>\n",
       "      <td>3.058782</td>\n",
       "      <td>3.065330</td>\n",
       "      <td>0.394206</td>\n",
       "      <td>6.845823</td>\n",
       "      <td>0.739760</td>\n",
       "      <td>5.612840</td>\n",
       "      <td>11.378735</td>\n",
       "      <td>...</td>\n",
       "      <td>0.026960</td>\n",
       "      <td>4.768690</td>\n",
       "      <td>7.245916</td>\n",
       "      <td>0.010848</td>\n",
       "      <td>2.319096</td>\n",
       "      <td>5.181626</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>4.881492</td>\n",
       "      <td>2.779229</td>\n",
       "      <td>0.528664</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1673 rows × 19139 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0  TFE3 (7030)  MMADHC (27249)  PABPC1L (80336)  \\\n",
       "0     ACH-000873     3.607691        3.531071         4.982285   \n",
       "1     ACH-000860     4.192566        6.865682         4.853288   \n",
       "2     ACH-000439     3.265517        5.804615         1.777636   \n",
       "3     ACH-000318     3.705500        5.813170         4.536005   \n",
       "4     ACH-001142     6.105227        7.051513         4.457555   \n",
       "...          ...          ...             ...              ...   \n",
       "1668  ACH-001843     5.343844        6.553367         5.674036   \n",
       "1669  ACH-002074     5.252185        6.258400         2.872415   \n",
       "1670  ACH-001164     4.050893        6.248805         2.790849   \n",
       "1671  ACH-003180     5.161680        7.678250         1.932748   \n",
       "1672  ACH-001979     4.986577        7.612689         3.058782   \n",
       "\n",
       "      RELL2 (285613)  GPT (2875)  GUSB (2990)  H4-16 (121504)  DDB2 (1643)  \\\n",
       "0           2.326016    0.607679     5.118549        1.282028     4.512297   \n",
       "1           2.725583    0.197016     5.666426        1.154470     4.306808   \n",
       "2           4.185835    0.389025     6.593342        0.410244     4.411381   \n",
       "3           2.290409    0.471169     5.160089        2.325673     5.017968   \n",
       "4           2.543055    0.022363     4.589966        2.369987     4.105131   \n",
       "...              ...         ...          ...             ...          ...   \n",
       "1668        3.390159    0.644076     5.555928        1.037326     4.275462   \n",
       "1669        1.886574    0.796319     6.534374        3.269016     4.587816   \n",
       "1670        1.599247    0.663686     5.533158        0.646986     5.091849   \n",
       "1671        1.463134    0.394206     6.464239        0.360318     4.761168   \n",
       "1672        3.065330    0.394206     6.845823        0.739760     5.612840   \n",
       "\n",
       "      RPS4X (6191)  ...  OR2A42 (402317)  HOXB7 (3217)  NSA2 (10412)  \\\n",
       "0        10.075438  ...         0.042564      3.870581      5.178480   \n",
       "1        10.215240  ...         0.031621     -0.005897      6.749713   \n",
       "2        11.644890  ...        -0.013226     -0.033868      6.877322   \n",
       "3         9.163992  ...        -0.013226      3.051092      5.610027   \n",
       "4        10.179188  ...         0.053402      3.582671      7.416571   \n",
       "...            ...  ...              ...           ...           ...   \n",
       "1668     11.191982  ...         0.026960      5.222260      7.411799   \n",
       "1669      9.946776  ...         0.104563      4.270369      6.188434   \n",
       "1670     10.865960  ...         0.026960      0.470043      6.898124   \n",
       "1671      9.531740  ...         0.104563      4.741295      6.250017   \n",
       "1672     11.378735  ...         0.026960      4.768690      7.245916   \n",
       "\n",
       "      UGT2B11 (10720)  ZNF747 (65988)  NAA60 (79903)  MYH4 (4622)  \\\n",
       "0           -0.007723        2.005498       4.213717    -0.001724   \n",
       "1           -0.007723        1.393644       4.334535    -0.001724   \n",
       "2            1.596383        3.143557       4.624226    -0.001724   \n",
       "3           -0.007723        2.334419       3.881219    -0.001724   \n",
       "4           -0.007723        2.294077       4.889203    -0.001724   \n",
       "...               ...             ...            ...          ...   \n",
       "1668         0.010848        0.563822       4.885868     0.000741   \n",
       "1669         0.010848        3.447185       5.964889     0.000741   \n",
       "1670         1.385374        2.641063       4.824132     0.022977   \n",
       "1671         0.010848        2.652496       5.139537     0.044994   \n",
       "1672         0.010848        2.319096       5.181626     0.000741   \n",
       "\n",
       "      MTAP (4507)  C14orf93 (60686)  NR3C2 (4306)  \n",
       "0        3.904800          2.190931      0.203910  \n",
       "1        4.812949          3.366939      0.370726  \n",
       "2        4.122929          2.197136      0.015412  \n",
       "3        0.394778          3.217293      0.277760  \n",
       "4        4.379944          2.355443      0.520338  \n",
       "...           ...               ...           ...  \n",
       "1668     4.898450          2.716015      0.067286  \n",
       "1669     4.181277          3.931034      0.695929  \n",
       "1670     3.384695          2.413815      0.221968  \n",
       "1671     4.773658          2.686730      0.011768  \n",
       "1672     4.881492          2.779229      0.528664  \n",
       "\n",
       "[1673 rows x 19139 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "raw_transcript_df = pd.read_csv('./data/raw_data/OmicsExpressionProteinCodingGenesTPMLogp1BatchCorrected.csv')\n",
    "display(raw_transcript_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000873</th>\n",
       "      <th>ACH-000860</th>\n",
       "      <th>ACH-000439</th>\n",
       "      <th>ACH-000318</th>\n",
       "      <th>ACH-001142</th>\n",
       "      <th>ACH-000789</th>\n",
       "      <th>ACH-000544</th>\n",
       "      <th>ACH-000531</th>\n",
       "      <th>ACH-000695</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-002996</th>\n",
       "      <th>ACH-003061</th>\n",
       "      <th>ACH-001384</th>\n",
       "      <th>ACH-001700</th>\n",
       "      <th>ACH-002470</th>\n",
       "      <th>ACH-001843</th>\n",
       "      <th>ACH-002074</th>\n",
       "      <th>ACH-001164</th>\n",
       "      <th>ACH-003180</th>\n",
       "      <th>ACH-001979</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TFE3</td>\n",
       "      <td>3.607691</td>\n",
       "      <td>4.192566</td>\n",
       "      <td>3.265517</td>\n",
       "      <td>3.705500</td>\n",
       "      <td>6.105227</td>\n",
       "      <td>3.610912</td>\n",
       "      <td>2.446179</td>\n",
       "      <td>5.681865</td>\n",
       "      <td>4.364434</td>\n",
       "      <td>...</td>\n",
       "      <td>5.586557</td>\n",
       "      <td>4.076275</td>\n",
       "      <td>5.469260</td>\n",
       "      <td>4.482530</td>\n",
       "      <td>3.543934</td>\n",
       "      <td>5.343844</td>\n",
       "      <td>5.252185</td>\n",
       "      <td>4.050893</td>\n",
       "      <td>5.161680</td>\n",
       "      <td>4.986577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MMADHC</td>\n",
       "      <td>3.531071</td>\n",
       "      <td>6.865682</td>\n",
       "      <td>5.804615</td>\n",
       "      <td>5.813170</td>\n",
       "      <td>7.051513</td>\n",
       "      <td>6.427401</td>\n",
       "      <td>5.885347</td>\n",
       "      <td>6.781293</td>\n",
       "      <td>6.502812</td>\n",
       "      <td>...</td>\n",
       "      <td>7.616900</td>\n",
       "      <td>6.957816</td>\n",
       "      <td>6.276709</td>\n",
       "      <td>6.061848</td>\n",
       "      <td>6.524889</td>\n",
       "      <td>6.553367</td>\n",
       "      <td>6.258400</td>\n",
       "      <td>6.248805</td>\n",
       "      <td>7.678250</td>\n",
       "      <td>7.612689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PABPC1L</td>\n",
       "      <td>4.982285</td>\n",
       "      <td>4.853288</td>\n",
       "      <td>1.777636</td>\n",
       "      <td>4.536005</td>\n",
       "      <td>4.457555</td>\n",
       "      <td>4.756976</td>\n",
       "      <td>2.603382</td>\n",
       "      <td>1.889260</td>\n",
       "      <td>3.187749</td>\n",
       "      <td>...</td>\n",
       "      <td>2.992110</td>\n",
       "      <td>5.160160</td>\n",
       "      <td>4.805626</td>\n",
       "      <td>3.787419</td>\n",
       "      <td>3.395479</td>\n",
       "      <td>5.674036</td>\n",
       "      <td>2.872415</td>\n",
       "      <td>2.790849</td>\n",
       "      <td>1.932748</td>\n",
       "      <td>3.058782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RELL2</td>\n",
       "      <td>2.326016</td>\n",
       "      <td>2.725583</td>\n",
       "      <td>4.185835</td>\n",
       "      <td>2.290409</td>\n",
       "      <td>2.543055</td>\n",
       "      <td>4.364494</td>\n",
       "      <td>2.392491</td>\n",
       "      <td>2.070912</td>\n",
       "      <td>3.921761</td>\n",
       "      <td>...</td>\n",
       "      <td>2.219140</td>\n",
       "      <td>3.081301</td>\n",
       "      <td>2.934213</td>\n",
       "      <td>2.397418</td>\n",
       "      <td>0.801529</td>\n",
       "      <td>3.390159</td>\n",
       "      <td>1.886574</td>\n",
       "      <td>1.599247</td>\n",
       "      <td>1.463134</td>\n",
       "      <td>3.065330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GPT</td>\n",
       "      <td>0.607679</td>\n",
       "      <td>0.197016</td>\n",
       "      <td>0.389025</td>\n",
       "      <td>0.471169</td>\n",
       "      <td>0.022363</td>\n",
       "      <td>0.929382</td>\n",
       "      <td>0.112811</td>\n",
       "      <td>-0.009340</td>\n",
       "      <td>0.633240</td>\n",
       "      <td>...</td>\n",
       "      <td>0.394206</td>\n",
       "      <td>0.814637</td>\n",
       "      <td>0.437848</td>\n",
       "      <td>0.663686</td>\n",
       "      <td>0.372059</td>\n",
       "      <td>0.644076</td>\n",
       "      <td>0.796319</td>\n",
       "      <td>0.663686</td>\n",
       "      <td>0.394206</td>\n",
       "      <td>0.394206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19133</th>\n",
       "      <td>NAA60</td>\n",
       "      <td>4.213717</td>\n",
       "      <td>4.334535</td>\n",
       "      <td>4.624226</td>\n",
       "      <td>3.881219</td>\n",
       "      <td>4.889203</td>\n",
       "      <td>4.739785</td>\n",
       "      <td>3.552251</td>\n",
       "      <td>4.360793</td>\n",
       "      <td>5.393965</td>\n",
       "      <td>...</td>\n",
       "      <td>3.665560</td>\n",
       "      <td>4.485363</td>\n",
       "      <td>4.352813</td>\n",
       "      <td>5.027732</td>\n",
       "      <td>4.295766</td>\n",
       "      <td>4.885868</td>\n",
       "      <td>5.964889</td>\n",
       "      <td>4.824132</td>\n",
       "      <td>5.139537</td>\n",
       "      <td>5.181626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19134</th>\n",
       "      <td>MYH4</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.066796</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.022977</td>\n",
       "      <td>0.044994</td>\n",
       "      <td>0.000741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19135</th>\n",
       "      <td>MTAP</td>\n",
       "      <td>3.904800</td>\n",
       "      <td>4.812949</td>\n",
       "      <td>4.122929</td>\n",
       "      <td>0.394778</td>\n",
       "      <td>4.379944</td>\n",
       "      <td>4.911818</td>\n",
       "      <td>3.890611</td>\n",
       "      <td>3.643557</td>\n",
       "      <td>4.974483</td>\n",
       "      <td>...</td>\n",
       "      <td>6.052774</td>\n",
       "      <td>5.392289</td>\n",
       "      <td>3.797140</td>\n",
       "      <td>4.540203</td>\n",
       "      <td>4.702090</td>\n",
       "      <td>4.898450</td>\n",
       "      <td>4.181277</td>\n",
       "      <td>3.384695</td>\n",
       "      <td>4.773658</td>\n",
       "      <td>4.881492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19136</th>\n",
       "      <td>C14orf93</td>\n",
       "      <td>2.190931</td>\n",
       "      <td>3.366939</td>\n",
       "      <td>2.197136</td>\n",
       "      <td>3.217293</td>\n",
       "      <td>2.355443</td>\n",
       "      <td>3.855860</td>\n",
       "      <td>2.318863</td>\n",
       "      <td>1.962670</td>\n",
       "      <td>4.300053</td>\n",
       "      <td>...</td>\n",
       "      <td>2.645150</td>\n",
       "      <td>3.394596</td>\n",
       "      <td>2.325457</td>\n",
       "      <td>3.070859</td>\n",
       "      <td>2.465353</td>\n",
       "      <td>2.716015</td>\n",
       "      <td>3.931034</td>\n",
       "      <td>2.413815</td>\n",
       "      <td>2.686730</td>\n",
       "      <td>2.779229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19137</th>\n",
       "      <td>NR3C2</td>\n",
       "      <td>0.203910</td>\n",
       "      <td>0.370726</td>\n",
       "      <td>0.015412</td>\n",
       "      <td>0.277760</td>\n",
       "      <td>0.520338</td>\n",
       "      <td>1.852901</td>\n",
       "      <td>0.057915</td>\n",
       "      <td>2.178547</td>\n",
       "      <td>0.846391</td>\n",
       "      <td>...</td>\n",
       "      <td>0.039796</td>\n",
       "      <td>1.248954</td>\n",
       "      <td>1.357860</td>\n",
       "      <td>0.011768</td>\n",
       "      <td>0.339199</td>\n",
       "      <td>0.067286</td>\n",
       "      <td>0.695929</td>\n",
       "      <td>0.221968</td>\n",
       "      <td>0.011768</td>\n",
       "      <td>0.528664</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19138 rows × 1674 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 gene_name  ACH-000873  ACH-000860  ACH-000439  ACH-000318  \\\n",
       "0               TFE3    3.607691    4.192566    3.265517    3.705500   \n",
       "1             MMADHC    3.531071    6.865682    5.804615    5.813170   \n",
       "2            PABPC1L    4.982285    4.853288    1.777636    4.536005   \n",
       "3              RELL2    2.326016    2.725583    4.185835    2.290409   \n",
       "4                GPT    0.607679    0.197016    0.389025    0.471169   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "19133          NAA60    4.213717    4.334535    4.624226    3.881219   \n",
       "19134           MYH4   -0.001724   -0.001724   -0.001724   -0.001724   \n",
       "19135           MTAP    3.904800    4.812949    4.122929    0.394778   \n",
       "19136       C14orf93    2.190931    3.366939    2.197136    3.217293   \n",
       "19137          NR3C2    0.203910    0.370726    0.015412    0.277760   \n",
       "\n",
       "Unnamed: 0  ACH-001142  ACH-000789  ACH-000544  ACH-000531  ACH-000695  ...  \\\n",
       "0             6.105227    3.610912    2.446179    5.681865    4.364434  ...   \n",
       "1             7.051513    6.427401    5.885347    6.781293    6.502812  ...   \n",
       "2             4.457555    4.756976    2.603382    1.889260    3.187749  ...   \n",
       "3             2.543055    4.364494    2.392491    2.070912    3.921761  ...   \n",
       "4             0.022363    0.929382    0.112811   -0.009340    0.633240  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "19133         4.889203    4.739785    3.552251    4.360793    5.393965  ...   \n",
       "19134        -0.001724   -0.001724   -0.001724   -0.001724   -0.001724  ...   \n",
       "19135         4.379944    4.911818    3.890611    3.643557    4.974483  ...   \n",
       "19136         2.355443    3.855860    2.318863    1.962670    4.300053  ...   \n",
       "19137         0.520338    1.852901    0.057915    2.178547    0.846391  ...   \n",
       "\n",
       "Unnamed: 0  ACH-002996  ACH-003061  ACH-001384  ACH-001700  ACH-002470  \\\n",
       "0             5.586557    4.076275    5.469260    4.482530    3.543934   \n",
       "1             7.616900    6.957816    6.276709    6.061848    6.524889   \n",
       "2             2.992110    5.160160    4.805626    3.787419    3.395479   \n",
       "3             2.219140    3.081301    2.934213    2.397418    0.801529   \n",
       "4             0.394206    0.814637    0.437848    0.663686    0.372059   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "19133         3.665560    4.485363    4.352813    5.027732    4.295766   \n",
       "19134         0.000741    0.066796    0.000741    0.000741    0.000741   \n",
       "19135         6.052774    5.392289    3.797140    4.540203    4.702090   \n",
       "19136         2.645150    3.394596    2.325457    3.070859    2.465353   \n",
       "19137         0.039796    1.248954    1.357860    0.011768    0.339199   \n",
       "\n",
       "Unnamed: 0  ACH-001843  ACH-002074  ACH-001164  ACH-003180  ACH-001979  \n",
       "0             5.343844    5.252185    4.050893    5.161680    4.986577  \n",
       "1             6.553367    6.258400    6.248805    7.678250    7.612689  \n",
       "2             5.674036    2.872415    2.790849    1.932748    3.058782  \n",
       "3             3.390159    1.886574    1.599247    1.463134    3.065330  \n",
       "4             0.644076    0.796319    0.663686    0.394206    0.394206  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "19133         4.885868    5.964889    4.824132    5.139537    5.181626  \n",
       "19134         0.000741    0.000741    0.022977    0.044994    0.000741  \n",
       "19135         4.898450    4.181277    3.384695    4.773658    4.881492  \n",
       "19136         2.716015    3.931034    2.413815    2.686730    2.779229  \n",
       "19137         0.067286    0.695929    0.221968    0.011768    0.528664  \n",
       "\n",
       "[19138 rows x 1674 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# First, set the \"Unnamed: 0\" column as the index:\n",
    "raw_transcript_df = raw_transcript_df.set_index(\"Unnamed: 0\")\n",
    "# Then transpose the DataFrame:\n",
    "raw_transcript_df = raw_transcript_df.transpose()\n",
    "# move the index to a column and rename it as the gene_name\n",
    "raw_transcript_df.reset_index(inplace=True)\n",
    "raw_transcript_df.rename(columns={'index': 'gene_name'}, inplace=True)\n",
    "# column gene_name should be kept with the content by removing \"()\" and remove and\n",
    "raw_transcript_df['gene_name'] = raw_transcript_df['gene_name'].apply(lambda x: x.split('(')[0].strip())\n",
    "# Optionally, if you want to view the result:\n",
    "display(raw_transcript_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000873</th>\n",
       "      <th>ACH-000860</th>\n",
       "      <th>ACH-000439</th>\n",
       "      <th>ACH-000318</th>\n",
       "      <th>ACH-001142</th>\n",
       "      <th>ACH-000789</th>\n",
       "      <th>ACH-000544</th>\n",
       "      <th>ACH-000531</th>\n",
       "      <th>ACH-000695</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-002996</th>\n",
       "      <th>ACH-003061</th>\n",
       "      <th>ACH-001384</th>\n",
       "      <th>ACH-001700</th>\n",
       "      <th>ACH-002470</th>\n",
       "      <th>ACH-001843</th>\n",
       "      <th>ACH-002074</th>\n",
       "      <th>ACH-001164</th>\n",
       "      <th>ACH-003180</th>\n",
       "      <th>ACH-001979</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TFE3</td>\n",
       "      <td>3.607691</td>\n",
       "      <td>4.192566</td>\n",
       "      <td>3.265517</td>\n",
       "      <td>3.705500</td>\n",
       "      <td>6.105227</td>\n",
       "      <td>3.610912</td>\n",
       "      <td>2.446179</td>\n",
       "      <td>5.681865</td>\n",
       "      <td>4.364434</td>\n",
       "      <td>...</td>\n",
       "      <td>5.586557</td>\n",
       "      <td>4.076275</td>\n",
       "      <td>5.469260</td>\n",
       "      <td>4.482530</td>\n",
       "      <td>3.543934</td>\n",
       "      <td>5.343844</td>\n",
       "      <td>5.252185</td>\n",
       "      <td>4.050893</td>\n",
       "      <td>5.161680</td>\n",
       "      <td>4.986577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MMADHC</td>\n",
       "      <td>3.531071</td>\n",
       "      <td>6.865682</td>\n",
       "      <td>5.804615</td>\n",
       "      <td>5.813170</td>\n",
       "      <td>7.051513</td>\n",
       "      <td>6.427401</td>\n",
       "      <td>5.885347</td>\n",
       "      <td>6.781293</td>\n",
       "      <td>6.502812</td>\n",
       "      <td>...</td>\n",
       "      <td>7.616900</td>\n",
       "      <td>6.957816</td>\n",
       "      <td>6.276709</td>\n",
       "      <td>6.061848</td>\n",
       "      <td>6.524889</td>\n",
       "      <td>6.553367</td>\n",
       "      <td>6.258400</td>\n",
       "      <td>6.248805</td>\n",
       "      <td>7.678250</td>\n",
       "      <td>7.612689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PABPC1L</td>\n",
       "      <td>4.982285</td>\n",
       "      <td>4.853288</td>\n",
       "      <td>1.777636</td>\n",
       "      <td>4.536005</td>\n",
       "      <td>4.457555</td>\n",
       "      <td>4.756976</td>\n",
       "      <td>2.603382</td>\n",
       "      <td>1.889260</td>\n",
       "      <td>3.187749</td>\n",
       "      <td>...</td>\n",
       "      <td>2.992110</td>\n",
       "      <td>5.160160</td>\n",
       "      <td>4.805626</td>\n",
       "      <td>3.787419</td>\n",
       "      <td>3.395479</td>\n",
       "      <td>5.674036</td>\n",
       "      <td>2.872415</td>\n",
       "      <td>2.790849</td>\n",
       "      <td>1.932748</td>\n",
       "      <td>3.058782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RELL2</td>\n",
       "      <td>2.326016</td>\n",
       "      <td>2.725583</td>\n",
       "      <td>4.185835</td>\n",
       "      <td>2.290409</td>\n",
       "      <td>2.543055</td>\n",
       "      <td>4.364494</td>\n",
       "      <td>2.392491</td>\n",
       "      <td>2.070912</td>\n",
       "      <td>3.921761</td>\n",
       "      <td>...</td>\n",
       "      <td>2.219140</td>\n",
       "      <td>3.081301</td>\n",
       "      <td>2.934213</td>\n",
       "      <td>2.397418</td>\n",
       "      <td>0.801529</td>\n",
       "      <td>3.390159</td>\n",
       "      <td>1.886574</td>\n",
       "      <td>1.599247</td>\n",
       "      <td>1.463134</td>\n",
       "      <td>3.065330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GPT</td>\n",
       "      <td>0.607679</td>\n",
       "      <td>0.197016</td>\n",
       "      <td>0.389025</td>\n",
       "      <td>0.471169</td>\n",
       "      <td>0.022363</td>\n",
       "      <td>0.929382</td>\n",
       "      <td>0.112811</td>\n",
       "      <td>-0.009340</td>\n",
       "      <td>0.633240</td>\n",
       "      <td>...</td>\n",
       "      <td>0.394206</td>\n",
       "      <td>0.814637</td>\n",
       "      <td>0.437848</td>\n",
       "      <td>0.663686</td>\n",
       "      <td>0.372059</td>\n",
       "      <td>0.644076</td>\n",
       "      <td>0.796319</td>\n",
       "      <td>0.663686</td>\n",
       "      <td>0.394206</td>\n",
       "      <td>0.394206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19133</th>\n",
       "      <td>NAA60</td>\n",
       "      <td>4.213717</td>\n",
       "      <td>4.334535</td>\n",
       "      <td>4.624226</td>\n",
       "      <td>3.881219</td>\n",
       "      <td>4.889203</td>\n",
       "      <td>4.739785</td>\n",
       "      <td>3.552251</td>\n",
       "      <td>4.360793</td>\n",
       "      <td>5.393965</td>\n",
       "      <td>...</td>\n",
       "      <td>3.665560</td>\n",
       "      <td>4.485363</td>\n",
       "      <td>4.352813</td>\n",
       "      <td>5.027732</td>\n",
       "      <td>4.295766</td>\n",
       "      <td>4.885868</td>\n",
       "      <td>5.964889</td>\n",
       "      <td>4.824132</td>\n",
       "      <td>5.139537</td>\n",
       "      <td>5.181626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19134</th>\n",
       "      <td>MYH4</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.066796</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.022977</td>\n",
       "      <td>0.044994</td>\n",
       "      <td>0.000741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19135</th>\n",
       "      <td>MTAP</td>\n",
       "      <td>3.904800</td>\n",
       "      <td>4.812949</td>\n",
       "      <td>4.122929</td>\n",
       "      <td>0.394778</td>\n",
       "      <td>4.379944</td>\n",
       "      <td>4.911818</td>\n",
       "      <td>3.890611</td>\n",
       "      <td>3.643557</td>\n",
       "      <td>4.974483</td>\n",
       "      <td>...</td>\n",
       "      <td>6.052774</td>\n",
       "      <td>5.392289</td>\n",
       "      <td>3.797140</td>\n",
       "      <td>4.540203</td>\n",
       "      <td>4.702090</td>\n",
       "      <td>4.898450</td>\n",
       "      <td>4.181277</td>\n",
       "      <td>3.384695</td>\n",
       "      <td>4.773658</td>\n",
       "      <td>4.881492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19136</th>\n",
       "      <td>C14orf93</td>\n",
       "      <td>2.190931</td>\n",
       "      <td>3.366939</td>\n",
       "      <td>2.197136</td>\n",
       "      <td>3.217293</td>\n",
       "      <td>2.355443</td>\n",
       "      <td>3.855860</td>\n",
       "      <td>2.318863</td>\n",
       "      <td>1.962670</td>\n",
       "      <td>4.300053</td>\n",
       "      <td>...</td>\n",
       "      <td>2.645150</td>\n",
       "      <td>3.394596</td>\n",
       "      <td>2.325457</td>\n",
       "      <td>3.070859</td>\n",
       "      <td>2.465353</td>\n",
       "      <td>2.716015</td>\n",
       "      <td>3.931034</td>\n",
       "      <td>2.413815</td>\n",
       "      <td>2.686730</td>\n",
       "      <td>2.779229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19137</th>\n",
       "      <td>NR3C2</td>\n",
       "      <td>0.203910</td>\n",
       "      <td>0.370726</td>\n",
       "      <td>0.015412</td>\n",
       "      <td>0.277760</td>\n",
       "      <td>0.520338</td>\n",
       "      <td>1.852901</td>\n",
       "      <td>0.057915</td>\n",
       "      <td>2.178547</td>\n",
       "      <td>0.846391</td>\n",
       "      <td>...</td>\n",
       "      <td>0.039796</td>\n",
       "      <td>1.248954</td>\n",
       "      <td>1.357860</td>\n",
       "      <td>0.011768</td>\n",
       "      <td>0.339199</td>\n",
       "      <td>0.067286</td>\n",
       "      <td>0.695929</td>\n",
       "      <td>0.221968</td>\n",
       "      <td>0.011768</td>\n",
       "      <td>0.528664</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19138 rows × 1674 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 gene_name  ACH-000873  ACH-000860  ACH-000439  ACH-000318  \\\n",
       "0               TFE3    3.607691    4.192566    3.265517    3.705500   \n",
       "1             MMADHC    3.531071    6.865682    5.804615    5.813170   \n",
       "2            PABPC1L    4.982285    4.853288    1.777636    4.536005   \n",
       "3              RELL2    2.326016    2.725583    4.185835    2.290409   \n",
       "4                GPT    0.607679    0.197016    0.389025    0.471169   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "19133          NAA60    4.213717    4.334535    4.624226    3.881219   \n",
       "19134           MYH4   -0.001724   -0.001724   -0.001724   -0.001724   \n",
       "19135           MTAP    3.904800    4.812949    4.122929    0.394778   \n",
       "19136       C14orf93    2.190931    3.366939    2.197136    3.217293   \n",
       "19137          NR3C2    0.203910    0.370726    0.015412    0.277760   \n",
       "\n",
       "Unnamed: 0  ACH-001142  ACH-000789  ACH-000544  ACH-000531  ACH-000695  ...  \\\n",
       "0             6.105227    3.610912    2.446179    5.681865    4.364434  ...   \n",
       "1             7.051513    6.427401    5.885347    6.781293    6.502812  ...   \n",
       "2             4.457555    4.756976    2.603382    1.889260    3.187749  ...   \n",
       "3             2.543055    4.364494    2.392491    2.070912    3.921761  ...   \n",
       "4             0.022363    0.929382    0.112811   -0.009340    0.633240  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "19133         4.889203    4.739785    3.552251    4.360793    5.393965  ...   \n",
       "19134        -0.001724   -0.001724   -0.001724   -0.001724   -0.001724  ...   \n",
       "19135         4.379944    4.911818    3.890611    3.643557    4.974483  ...   \n",
       "19136         2.355443    3.855860    2.318863    1.962670    4.300053  ...   \n",
       "19137         0.520338    1.852901    0.057915    2.178547    0.846391  ...   \n",
       "\n",
       "Unnamed: 0  ACH-002996  ACH-003061  ACH-001384  ACH-001700  ACH-002470  \\\n",
       "0             5.586557    4.076275    5.469260    4.482530    3.543934   \n",
       "1             7.616900    6.957816    6.276709    6.061848    6.524889   \n",
       "2             2.992110    5.160160    4.805626    3.787419    3.395479   \n",
       "3             2.219140    3.081301    2.934213    2.397418    0.801529   \n",
       "4             0.394206    0.814637    0.437848    0.663686    0.372059   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "19133         3.665560    4.485363    4.352813    5.027732    4.295766   \n",
       "19134         0.000741    0.066796    0.000741    0.000741    0.000741   \n",
       "19135         6.052774    5.392289    3.797140    4.540203    4.702090   \n",
       "19136         2.645150    3.394596    2.325457    3.070859    2.465353   \n",
       "19137         0.039796    1.248954    1.357860    0.011768    0.339199   \n",
       "\n",
       "Unnamed: 0  ACH-001843  ACH-002074  ACH-001164  ACH-003180  ACH-001979  \n",
       "0             5.343844    5.252185    4.050893    5.161680    4.986577  \n",
       "1             6.553367    6.258400    6.248805    7.678250    7.612689  \n",
       "2             5.674036    2.872415    2.790849    1.932748    3.058782  \n",
       "3             3.390159    1.886574    1.599247    1.463134    3.065330  \n",
       "4             0.644076    0.796319    0.663686    0.394206    0.394206  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "19133         4.885868    5.964889    4.824132    5.139537    5.181626  \n",
       "19134         0.000741    0.000741    0.022977    0.044994    0.000741  \n",
       "19135         4.898450    4.181277    3.384695    4.773658    4.881492  \n",
       "19136         2.716015    3.931034    2.413815    2.686730    2.779229  \n",
       "19137         0.067286    0.695929    0.221968    0.011768    0.528664  \n",
       "\n",
       "[19138 rows x 1674 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Identify columns to convert (all except 'Unnamed: 0' and 'gene_name')\n",
    "cols_to_convert = raw_transcript_df.columns.difference(['Unnamed: 0', 'gene_name'])\n",
    "# Convert these columns to numeric (using pd.to_numeric for safety, which can handle errors)\n",
    "raw_transcript_df[cols_to_convert] = raw_transcript_df[cols_to_convert].apply(pd.to_numeric, errors='coerce')\n",
    "raw_transcript_df = raw_transcript_df.fillna(0.0)\n",
    "display(raw_transcript_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\594098286.py:1: DtypeWarning: Columns (14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmg_transcript_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Transcript/BioMedGraphica_Conn_Transcript.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>HGNC_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>ARF5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>M6PR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>ESRRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>FKBP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>CYP26B1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412034</th>\n",
       "      <td>BMGC_TS412035</td>\n",
       "      <td>LINC02968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412035</th>\n",
       "      <td>BMGC_TS412036</td>\n",
       "      <td>LINC03112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412036</th>\n",
       "      <td>BMGC_TS412037</td>\n",
       "      <td>LINC03112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412037</th>\n",
       "      <td>BMGC_TS412038</td>\n",
       "      <td>LINC03112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412038</th>\n",
       "      <td>BMGC_TS412039</td>\n",
       "      <td>CD99P1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412039 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID HGNC_Symbol\n",
       "0               BMGC_TS000001        ARF5\n",
       "1               BMGC_TS000002        M6PR\n",
       "2               BMGC_TS000003       ESRRA\n",
       "3               BMGC_TS000004       FKBP4\n",
       "4               BMGC_TS000005     CYP26B1\n",
       "...                       ...         ...\n",
       "412034          BMGC_TS412035   LINC02968\n",
       "412035          BMGC_TS412036   LINC03112\n",
       "412036          BMGC_TS412037   LINC03112\n",
       "412037          BMGC_TS412038   LINC03112\n",
       "412038          BMGC_TS412039      CD99P1\n",
       "\n",
       "[412039 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bmg_transcript_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Transcript/BioMedGraphica_Conn_Transcript.csv')\n",
    "# keep BioMedGraphica_ID and HGNC_Symbol\n",
    "bmg_transcript_df = bmg_transcript_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']]\n",
    "display(bmg_transcript_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000873</th>\n",
       "      <th>ACH-000860</th>\n",
       "      <th>ACH-000439</th>\n",
       "      <th>ACH-000318</th>\n",
       "      <th>ACH-001142</th>\n",
       "      <th>ACH-000789</th>\n",
       "      <th>ACH-000544</th>\n",
       "      <th>ACH-000531</th>\n",
       "      <th>ACH-000695</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-002996</th>\n",
       "      <th>ACH-003061</th>\n",
       "      <th>ACH-001384</th>\n",
       "      <th>ACH-001700</th>\n",
       "      <th>ACH-002470</th>\n",
       "      <th>ACH-001843</th>\n",
       "      <th>ACH-002074</th>\n",
       "      <th>ACH-001164</th>\n",
       "      <th>ACH-003180</th>\n",
       "      <th>ACH-001979</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>6.010975</td>\n",
       "      <td>6.650175</td>\n",
       "      <td>5.885916</td>\n",
       "      <td>7.142466</td>\n",
       "      <td>6.720866</td>\n",
       "      <td>8.317689</td>\n",
       "      <td>5.401430</td>\n",
       "      <td>6.562183</td>\n",
       "      <td>7.781027</td>\n",
       "      <td>...</td>\n",
       "      <td>6.265994</td>\n",
       "      <td>7.902718</td>\n",
       "      <td>7.112875</td>\n",
       "      <td>8.403703</td>\n",
       "      <td>6.377124</td>\n",
       "      <td>7.161867</td>\n",
       "      <td>8.339458</td>\n",
       "      <td>7.267401</td>\n",
       "      <td>6.349134</td>\n",
       "      <td>7.053324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>7.038597</td>\n",
       "      <td>5.851489</td>\n",
       "      <td>7.032545</td>\n",
       "      <td>6.123710</td>\n",
       "      <td>6.438191</td>\n",
       "      <td>6.674373</td>\n",
       "      <td>5.617127</td>\n",
       "      <td>5.864259</td>\n",
       "      <td>4.408654</td>\n",
       "      <td>...</td>\n",
       "      <td>6.463395</td>\n",
       "      <td>7.781410</td>\n",
       "      <td>6.244496</td>\n",
       "      <td>5.552777</td>\n",
       "      <td>6.305295</td>\n",
       "      <td>6.782487</td>\n",
       "      <td>6.672697</td>\n",
       "      <td>5.816018</td>\n",
       "      <td>5.844664</td>\n",
       "      <td>6.555430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>5.463087</td>\n",
       "      <td>4.154380</td>\n",
       "      <td>4.578244</td>\n",
       "      <td>5.522401</td>\n",
       "      <td>3.750699</td>\n",
       "      <td>4.963768</td>\n",
       "      <td>3.523931</td>\n",
       "      <td>3.163117</td>\n",
       "      <td>4.831130</td>\n",
       "      <td>...</td>\n",
       "      <td>3.906757</td>\n",
       "      <td>4.659487</td>\n",
       "      <td>4.521520</td>\n",
       "      <td>4.750653</td>\n",
       "      <td>3.971007</td>\n",
       "      <td>5.337452</td>\n",
       "      <td>3.530706</td>\n",
       "      <td>4.306837</td>\n",
       "      <td>3.879305</td>\n",
       "      <td>3.721688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>6.376783</td>\n",
       "      <td>5.864786</td>\n",
       "      <td>6.265494</td>\n",
       "      <td>6.496076</td>\n",
       "      <td>4.836182</td>\n",
       "      <td>8.226768</td>\n",
       "      <td>6.183044</td>\n",
       "      <td>4.384002</td>\n",
       "      <td>6.179894</td>\n",
       "      <td>...</td>\n",
       "      <td>6.368665</td>\n",
       "      <td>7.368284</td>\n",
       "      <td>6.055639</td>\n",
       "      <td>5.348946</td>\n",
       "      <td>6.446293</td>\n",
       "      <td>5.623277</td>\n",
       "      <td>7.118444</td>\n",
       "      <td>5.467079</td>\n",
       "      <td>5.451382</td>\n",
       "      <td>5.297473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>2.813257</td>\n",
       "      <td>2.289688</td>\n",
       "      <td>-0.046550</td>\n",
       "      <td>0.735355</td>\n",
       "      <td>0.043941</td>\n",
       "      <td>1.202660</td>\n",
       "      <td>0.641833</td>\n",
       "      <td>0.757766</td>\n",
       "      <td>0.423583</td>\n",
       "      <td>...</td>\n",
       "      <td>1.344332</td>\n",
       "      <td>3.388847</td>\n",
       "      <td>1.874635</td>\n",
       "      <td>0.602110</td>\n",
       "      <td>0.235919</td>\n",
       "      <td>0.221142</td>\n",
       "      <td>0.714818</td>\n",
       "      <td>0.578576</td>\n",
       "      <td>1.966306</td>\n",
       "      <td>0.235919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184492</th>\n",
       "      <td>BMGC_TS411885</td>\n",
       "      <td>0.754822</td>\n",
       "      <td>1.026857</td>\n",
       "      <td>0.998199</td>\n",
       "      <td>1.338890</td>\n",
       "      <td>-0.027065</td>\n",
       "      <td>-0.041679</td>\n",
       "      <td>-0.012595</td>\n",
       "      <td>-0.027065</td>\n",
       "      <td>0.001734</td>\n",
       "      <td>...</td>\n",
       "      <td>2.572651</td>\n",
       "      <td>0.288850</td>\n",
       "      <td>0.055933</td>\n",
       "      <td>0.151145</td>\n",
       "      <td>0.055933</td>\n",
       "      <td>0.083800</td>\n",
       "      <td>0.324180</td>\n",
       "      <td>0.802649</td>\n",
       "      <td>0.055933</td>\n",
       "      <td>1.945652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184493</th>\n",
       "      <td>BMGC_TS411886</td>\n",
       "      <td>0.754822</td>\n",
       "      <td>1.026857</td>\n",
       "      <td>0.998199</td>\n",
       "      <td>1.338890</td>\n",
       "      <td>-0.027065</td>\n",
       "      <td>-0.041679</td>\n",
       "      <td>-0.012595</td>\n",
       "      <td>-0.027065</td>\n",
       "      <td>0.001734</td>\n",
       "      <td>...</td>\n",
       "      <td>2.572651</td>\n",
       "      <td>0.288850</td>\n",
       "      <td>0.055933</td>\n",
       "      <td>0.151145</td>\n",
       "      <td>0.055933</td>\n",
       "      <td>0.083800</td>\n",
       "      <td>0.324180</td>\n",
       "      <td>0.802649</td>\n",
       "      <td>0.055933</td>\n",
       "      <td>1.945652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184494</th>\n",
       "      <td>BMGC_TS411887</td>\n",
       "      <td>0.754822</td>\n",
       "      <td>1.026857</td>\n",
       "      <td>0.998199</td>\n",
       "      <td>1.338890</td>\n",
       "      <td>-0.027065</td>\n",
       "      <td>-0.041679</td>\n",
       "      <td>-0.012595</td>\n",
       "      <td>-0.027065</td>\n",
       "      <td>0.001734</td>\n",
       "      <td>...</td>\n",
       "      <td>2.572651</td>\n",
       "      <td>0.288850</td>\n",
       "      <td>0.055933</td>\n",
       "      <td>0.151145</td>\n",
       "      <td>0.055933</td>\n",
       "      <td>0.083800</td>\n",
       "      <td>0.324180</td>\n",
       "      <td>0.802649</td>\n",
       "      <td>0.055933</td>\n",
       "      <td>1.945652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184495</th>\n",
       "      <td>BMGC_TS412018</td>\n",
       "      <td>3.665326</td>\n",
       "      <td>3.764924</td>\n",
       "      <td>4.628350</td>\n",
       "      <td>3.606514</td>\n",
       "      <td>4.904823</td>\n",
       "      <td>6.962571</td>\n",
       "      <td>2.996201</td>\n",
       "      <td>3.097047</td>\n",
       "      <td>5.627501</td>\n",
       "      <td>...</td>\n",
       "      <td>4.407802</td>\n",
       "      <td>3.795327</td>\n",
       "      <td>3.903598</td>\n",
       "      <td>3.948776</td>\n",
       "      <td>4.658968</td>\n",
       "      <td>5.434939</td>\n",
       "      <td>4.204795</td>\n",
       "      <td>5.267714</td>\n",
       "      <td>3.429109</td>\n",
       "      <td>3.956163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184496</th>\n",
       "      <td>BMGC_TS412019</td>\n",
       "      <td>0.955486</td>\n",
       "      <td>0.600762</td>\n",
       "      <td>5.926539</td>\n",
       "      <td>0.656376</td>\n",
       "      <td>-0.053776</td>\n",
       "      <td>1.108757</td>\n",
       "      <td>0.240539</td>\n",
       "      <td>0.322649</td>\n",
       "      <td>0.332571</td>\n",
       "      <td>...</td>\n",
       "      <td>0.172218</td>\n",
       "      <td>0.156200</td>\n",
       "      <td>0.156200</td>\n",
       "      <td>0.324194</td>\n",
       "      <td>0.436222</td>\n",
       "      <td>0.280088</td>\n",
       "      <td>0.203782</td>\n",
       "      <td>0.234740</td>\n",
       "      <td>0.172218</td>\n",
       "      <td>0.203782</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>184497 rows × 1674 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000873  ACH-000860  ACH-000439  ACH-000318  \\\n",
       "0               BMGC_TS000001    6.010975    6.650175    5.885916    7.142466   \n",
       "1               BMGC_TS000002    7.038597    5.851489    7.032545    6.123710   \n",
       "2               BMGC_TS000003    5.463087    4.154380    4.578244    5.522401   \n",
       "3               BMGC_TS000004    6.376783    5.864786    6.265494    6.496076   \n",
       "4               BMGC_TS000005    2.813257    2.289688   -0.046550    0.735355   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "184492          BMGC_TS411885    0.754822    1.026857    0.998199    1.338890   \n",
       "184493          BMGC_TS411886    0.754822    1.026857    0.998199    1.338890   \n",
       "184494          BMGC_TS411887    0.754822    1.026857    0.998199    1.338890   \n",
       "184495          BMGC_TS412018    3.665326    3.764924    4.628350    3.606514   \n",
       "184496          BMGC_TS412019    0.955486    0.600762    5.926539    0.656376   \n",
       "\n",
       "        ACH-001142  ACH-000789  ACH-000544  ACH-000531  ACH-000695  ...  \\\n",
       "0         6.720866    8.317689    5.401430    6.562183    7.781027  ...   \n",
       "1         6.438191    6.674373    5.617127    5.864259    4.408654  ...   \n",
       "2         3.750699    4.963768    3.523931    3.163117    4.831130  ...   \n",
       "3         4.836182    8.226768    6.183044    4.384002    6.179894  ...   \n",
       "4         0.043941    1.202660    0.641833    0.757766    0.423583  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "184492   -0.027065   -0.041679   -0.012595   -0.027065    0.001734  ...   \n",
       "184493   -0.027065   -0.041679   -0.012595   -0.027065    0.001734  ...   \n",
       "184494   -0.027065   -0.041679   -0.012595   -0.027065    0.001734  ...   \n",
       "184495    4.904823    6.962571    2.996201    3.097047    5.627501  ...   \n",
       "184496   -0.053776    1.108757    0.240539    0.322649    0.332571  ...   \n",
       "\n",
       "        ACH-002996  ACH-003061  ACH-001384  ACH-001700  ACH-002470  \\\n",
       "0         6.265994    7.902718    7.112875    8.403703    6.377124   \n",
       "1         6.463395    7.781410    6.244496    5.552777    6.305295   \n",
       "2         3.906757    4.659487    4.521520    4.750653    3.971007   \n",
       "3         6.368665    7.368284    6.055639    5.348946    6.446293   \n",
       "4         1.344332    3.388847    1.874635    0.602110    0.235919   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "184492    2.572651    0.288850    0.055933    0.151145    0.055933   \n",
       "184493    2.572651    0.288850    0.055933    0.151145    0.055933   \n",
       "184494    2.572651    0.288850    0.055933    0.151145    0.055933   \n",
       "184495    4.407802    3.795327    3.903598    3.948776    4.658968   \n",
       "184496    0.172218    0.156200    0.156200    0.324194    0.436222   \n",
       "\n",
       "        ACH-001843  ACH-002074  ACH-001164  ACH-003180  ACH-001979  \n",
       "0         7.161867    8.339458    7.267401    6.349134    7.053324  \n",
       "1         6.782487    6.672697    5.816018    5.844664    6.555430  \n",
       "2         5.337452    3.530706    4.306837    3.879305    3.721688  \n",
       "3         5.623277    7.118444    5.467079    5.451382    5.297473  \n",
       "4         0.221142    0.714818    0.578576    1.966306    0.235919  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "184492    0.083800    0.324180    0.802649    0.055933    1.945652  \n",
       "184493    0.083800    0.324180    0.802649    0.055933    1.945652  \n",
       "184494    0.083800    0.324180    0.802649    0.055933    1.945652  \n",
       "184495    5.434939    4.204795    5.267714    3.429109    3.956163  \n",
       "184496    0.280088    0.203782    0.234740    0.172218    0.203782  \n",
       "\n",
       "[184497 rows x 1674 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# merge the biomedgraphica_id with the raw_transcript_df\n",
    "merge_transcript_df = pd.merge(bmg_transcript_df, raw_transcript_df, left_on='HGNC_Symbol', right_on='gene_name', how='inner')\n",
    "merge_transcript_df.drop(columns=['HGNC_Symbol', 'gene_name'], inplace=True)\n",
    "display(merge_transcript_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000873</th>\n",
       "      <th>ACH-000860</th>\n",
       "      <th>ACH-000439</th>\n",
       "      <th>ACH-000318</th>\n",
       "      <th>ACH-001142</th>\n",
       "      <th>ACH-000789</th>\n",
       "      <th>ACH-000544</th>\n",
       "      <th>ACH-000531</th>\n",
       "      <th>ACH-000695</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-002996</th>\n",
       "      <th>ACH-003061</th>\n",
       "      <th>ACH-001384</th>\n",
       "      <th>ACH-001700</th>\n",
       "      <th>ACH-002470</th>\n",
       "      <th>ACH-001843</th>\n",
       "      <th>ACH-002074</th>\n",
       "      <th>ACH-001164</th>\n",
       "      <th>ACH-003180</th>\n",
       "      <th>ACH-001979</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>6.010975</td>\n",
       "      <td>6.650175</td>\n",
       "      <td>5.885916</td>\n",
       "      <td>7.142466</td>\n",
       "      <td>6.720866</td>\n",
       "      <td>8.317689</td>\n",
       "      <td>5.401430</td>\n",
       "      <td>6.562183</td>\n",
       "      <td>7.781027</td>\n",
       "      <td>...</td>\n",
       "      <td>6.265994</td>\n",
       "      <td>7.902718</td>\n",
       "      <td>7.112875</td>\n",
       "      <td>8.403703</td>\n",
       "      <td>6.377124</td>\n",
       "      <td>7.161867</td>\n",
       "      <td>8.339458</td>\n",
       "      <td>7.267401</td>\n",
       "      <td>6.349134</td>\n",
       "      <td>7.053324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>7.038597</td>\n",
       "      <td>5.851489</td>\n",
       "      <td>7.032545</td>\n",
       "      <td>6.123710</td>\n",
       "      <td>6.438191</td>\n",
       "      <td>6.674373</td>\n",
       "      <td>5.617127</td>\n",
       "      <td>5.864259</td>\n",
       "      <td>4.408654</td>\n",
       "      <td>...</td>\n",
       "      <td>6.463395</td>\n",
       "      <td>7.781410</td>\n",
       "      <td>6.244496</td>\n",
       "      <td>5.552777</td>\n",
       "      <td>6.305295</td>\n",
       "      <td>6.782487</td>\n",
       "      <td>6.672697</td>\n",
       "      <td>5.816018</td>\n",
       "      <td>5.844664</td>\n",
       "      <td>6.555430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>5.463087</td>\n",
       "      <td>4.154380</td>\n",
       "      <td>4.578244</td>\n",
       "      <td>5.522401</td>\n",
       "      <td>3.750699</td>\n",
       "      <td>4.963768</td>\n",
       "      <td>3.523931</td>\n",
       "      <td>3.163117</td>\n",
       "      <td>4.831130</td>\n",
       "      <td>...</td>\n",
       "      <td>3.906757</td>\n",
       "      <td>4.659487</td>\n",
       "      <td>4.521520</td>\n",
       "      <td>4.750653</td>\n",
       "      <td>3.971007</td>\n",
       "      <td>5.337452</td>\n",
       "      <td>3.530706</td>\n",
       "      <td>4.306837</td>\n",
       "      <td>3.879305</td>\n",
       "      <td>3.721688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>6.376783</td>\n",
       "      <td>5.864786</td>\n",
       "      <td>6.265494</td>\n",
       "      <td>6.496076</td>\n",
       "      <td>4.836182</td>\n",
       "      <td>8.226768</td>\n",
       "      <td>6.183044</td>\n",
       "      <td>4.384002</td>\n",
       "      <td>6.179894</td>\n",
       "      <td>...</td>\n",
       "      <td>6.368665</td>\n",
       "      <td>7.368284</td>\n",
       "      <td>6.055639</td>\n",
       "      <td>5.348946</td>\n",
       "      <td>6.446293</td>\n",
       "      <td>5.623277</td>\n",
       "      <td>7.118444</td>\n",
       "      <td>5.467079</td>\n",
       "      <td>5.451382</td>\n",
       "      <td>5.297473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>2.813257</td>\n",
       "      <td>2.289688</td>\n",
       "      <td>-0.046550</td>\n",
       "      <td>0.735355</td>\n",
       "      <td>0.043941</td>\n",
       "      <td>1.202660</td>\n",
       "      <td>0.641833</td>\n",
       "      <td>0.757766</td>\n",
       "      <td>0.423583</td>\n",
       "      <td>...</td>\n",
       "      <td>1.344332</td>\n",
       "      <td>3.388847</td>\n",
       "      <td>1.874635</td>\n",
       "      <td>0.602110</td>\n",
       "      <td>0.235919</td>\n",
       "      <td>0.221142</td>\n",
       "      <td>0.714818</td>\n",
       "      <td>0.578576</td>\n",
       "      <td>1.966306</td>\n",
       "      <td>0.235919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412034</th>\n",
       "      <td>BMGC_TS412035</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412035</th>\n",
       "      <td>BMGC_TS412036</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412036</th>\n",
       "      <td>BMGC_TS412037</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412037</th>\n",
       "      <td>BMGC_TS412038</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412038</th>\n",
       "      <td>BMGC_TS412039</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412039 rows × 1674 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000873  ACH-000860  ACH-000439  ACH-000318  \\\n",
       "0               BMGC_TS000001    6.010975    6.650175    5.885916    7.142466   \n",
       "1               BMGC_TS000002    7.038597    5.851489    7.032545    6.123710   \n",
       "2               BMGC_TS000003    5.463087    4.154380    4.578244    5.522401   \n",
       "3               BMGC_TS000004    6.376783    5.864786    6.265494    6.496076   \n",
       "4               BMGC_TS000005    2.813257    2.289688   -0.046550    0.735355   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "412034          BMGC_TS412035    0.000000    0.000000    0.000000    0.000000   \n",
       "412035          BMGC_TS412036    0.000000    0.000000    0.000000    0.000000   \n",
       "412036          BMGC_TS412037    0.000000    0.000000    0.000000    0.000000   \n",
       "412037          BMGC_TS412038    0.000000    0.000000    0.000000    0.000000   \n",
       "412038          BMGC_TS412039    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "        ACH-001142  ACH-000789  ACH-000544  ACH-000531  ACH-000695  ...  \\\n",
       "0         6.720866    8.317689    5.401430    6.562183    7.781027  ...   \n",
       "1         6.438191    6.674373    5.617127    5.864259    4.408654  ...   \n",
       "2         3.750699    4.963768    3.523931    3.163117    4.831130  ...   \n",
       "3         4.836182    8.226768    6.183044    4.384002    6.179894  ...   \n",
       "4         0.043941    1.202660    0.641833    0.757766    0.423583  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "412034    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412035    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412036    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412037    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412038    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "        ACH-002996  ACH-003061  ACH-001384  ACH-001700  ACH-002470  \\\n",
       "0         6.265994    7.902718    7.112875    8.403703    6.377124   \n",
       "1         6.463395    7.781410    6.244496    5.552777    6.305295   \n",
       "2         3.906757    4.659487    4.521520    4.750653    3.971007   \n",
       "3         6.368665    7.368284    6.055639    5.348946    6.446293   \n",
       "4         1.344332    3.388847    1.874635    0.602110    0.235919   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "412034    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "412035    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "412036    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "412037    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "412038    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "        ACH-001843  ACH-002074  ACH-001164  ACH-003180  ACH-001979  \n",
       "0         7.161867    8.339458    7.267401    6.349134    7.053324  \n",
       "1         6.782487    6.672697    5.816018    5.844664    6.555430  \n",
       "2         5.337452    3.530706    4.306837    3.879305    3.721688  \n",
       "3         5.623277    7.118444    5.467079    5.451382    5.297473  \n",
       "4         0.221142    0.714818    0.578576    1.966306    0.235919  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "412034    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "412035    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "412036    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "412037    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "412038    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "\n",
       "[412039 rows x 1674 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# merge the biomedgraphica_id with the raw_transcript_df\n",
    "final_merged_transcript_df = pd.merge(bmg_transcript_df, raw_transcript_df, left_on='HGNC_Symbol', right_on='gene_name', how='left')\n",
    "# fill the NaN with -1.0\n",
    "final_merged_transcript_df = final_merged_transcript_df.fillna(0.0)\n",
    "final_merged_transcript_df.drop(columns=['HGNC_Symbol', 'gene_name'], inplace=True)\n",
    "display(final_merged_transcript_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uniprot_Acc</th>\n",
       "      <th>MDAMB468_BREAST_TenPx01</th>\n",
       "      <th>SH4_SKIN_TenPx01</th>\n",
       "      <th>AU565_BREAST_TenPx01</th>\n",
       "      <th>KMRC1_KIDNEY_TenPx01</th>\n",
       "      <th>CAL51_BREAST_TenPx01</th>\n",
       "      <th>RPMI7951_SKIN_TenPx01</th>\n",
       "      <th>RERFLCMS_LUNG_TenPx01</th>\n",
       "      <th>IGR37_SKIN_TenPx01</th>\n",
       "      <th>VMRCRCW_KIDNEY_TenPx01</th>\n",
       "      <th>...</th>\n",
       "      <th>NCIH2030_LUNG_TenPx18</th>\n",
       "      <th>22RV1_PROSTATE_TenPx24</th>\n",
       "      <th>A172_CENTRAL_NERVOUS_SYSTEM_TenPx24</th>\n",
       "      <th>BT20_BREAST_TenPx24</th>\n",
       "      <th>CALU6_LUNG_TenPx24</th>\n",
       "      <th>FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24</th>\n",
       "      <th>KP4_PANCREAS_TenPx24</th>\n",
       "      <th>MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "      <th>OVCAR8_OVARY_TenPx24</th>\n",
       "      <th>THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P55011</td>\n",
       "      <td>2.111348</td>\n",
       "      <td>0.070468</td>\n",
       "      <td>-0.463928</td>\n",
       "      <td>-0.883645</td>\n",
       "      <td>0.788565</td>\n",
       "      <td>-0.912352</td>\n",
       "      <td>0.729167</td>\n",
       "      <td>-0.658213</td>\n",
       "      <td>-1.150653</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.345024</td>\n",
       "      <td>1.811205</td>\n",
       "      <td>-0.921176</td>\n",
       "      <td>-1.045727</td>\n",
       "      <td>0.845574</td>\n",
       "      <td>-0.071063</td>\n",
       "      <td>0.702306</td>\n",
       "      <td>-1.395878</td>\n",
       "      <td>-1.165032</td>\n",
       "      <td>-0.547536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P35453</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O60341</td>\n",
       "      <td>0.379683</td>\n",
       "      <td>-0.283538</td>\n",
       "      <td>0.191211</td>\n",
       "      <td>-0.595606</td>\n",
       "      <td>0.391243</td>\n",
       "      <td>-0.381329</td>\n",
       "      <td>0.410168</td>\n",
       "      <td>0.066076</td>\n",
       "      <td>-0.324100</td>\n",
       "      <td>...</td>\n",
       "      <td>0.161196</td>\n",
       "      <td>-0.337994</td>\n",
       "      <td>-0.455601</td>\n",
       "      <td>0.392393</td>\n",
       "      <td>0.258380</td>\n",
       "      <td>-0.817245</td>\n",
       "      <td>-0.790010</td>\n",
       "      <td>0.929522</td>\n",
       "      <td>-0.828736</td>\n",
       "      <td>0.735431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O00570</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.040198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P48431</td>\n",
       "      <td>-0.246367</td>\n",
       "      <td>-0.572753</td>\n",
       "      <td>-0.341305</td>\n",
       "      <td>-0.335211</td>\n",
       "      <td>-0.124474</td>\n",
       "      <td>0.043893</td>\n",
       "      <td>1.882953</td>\n",
       "      <td>-0.616307</td>\n",
       "      <td>-0.533335</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.627606</td>\n",
       "      <td>-1.261703</td>\n",
       "      <td>1.136851</td>\n",
       "      <td>-3.179027</td>\n",
       "      <td>1.424570</td>\n",
       "      <td>2.549687</td>\n",
       "      <td>4.424095</td>\n",
       "      <td>-1.083698</td>\n",
       "      <td>-1.640990</td>\n",
       "      <td>-2.905547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12750</th>\n",
       "      <td>Q9Y258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12751</th>\n",
       "      <td>P20292</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.690046</td>\n",
       "      <td>-1.600231</td>\n",
       "      <td>-1.110583</td>\n",
       "      <td>-1.047633</td>\n",
       "      <td>0.023850</td>\n",
       "      <td>-1.393552</td>\n",
       "      <td>4.556818</td>\n",
       "      <td>-0.531699</td>\n",
       "      <td>0.640295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12752</th>\n",
       "      <td>Q9H1C7</td>\n",
       "      <td>-0.313512</td>\n",
       "      <td>0.105086</td>\n",
       "      <td>-0.586815</td>\n",
       "      <td>-0.132940</td>\n",
       "      <td>0.452423</td>\n",
       "      <td>-0.546389</td>\n",
       "      <td>0.298349</td>\n",
       "      <td>0.461098</td>\n",
       "      <td>0.309062</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.393738</td>\n",
       "      <td>0.604400</td>\n",
       "      <td>1.643414</td>\n",
       "      <td>-1.481004</td>\n",
       "      <td>0.408420</td>\n",
       "      <td>-0.177391</td>\n",
       "      <td>0.445977</td>\n",
       "      <td>-2.497558</td>\n",
       "      <td>0.964551</td>\n",
       "      <td>-2.004067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12753</th>\n",
       "      <td>Q99735</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12754</th>\n",
       "      <td>Q9P003</td>\n",
       "      <td>0.138191</td>\n",
       "      <td>0.355856</td>\n",
       "      <td>0.001459</td>\n",
       "      <td>0.172501</td>\n",
       "      <td>0.200756</td>\n",
       "      <td>-0.262573</td>\n",
       "      <td>0.120958</td>\n",
       "      <td>-0.035599</td>\n",
       "      <td>-0.352327</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.488230</td>\n",
       "      <td>0.139008</td>\n",
       "      <td>0.933261</td>\n",
       "      <td>0.642768</td>\n",
       "      <td>-0.495987</td>\n",
       "      <td>-1.291515</td>\n",
       "      <td>-0.944284</td>\n",
       "      <td>-1.022551</td>\n",
       "      <td>0.220199</td>\n",
       "      <td>0.094595</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12755 rows × 379 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Uniprot_Acc  MDAMB468_BREAST_TenPx01  SH4_SKIN_TenPx01  \\\n",
       "0          P55011                 2.111348          0.070468   \n",
       "1          P35453                      NaN               NaN   \n",
       "2          O60341                 0.379683         -0.283538   \n",
       "3          O00570                      NaN               NaN   \n",
       "4          P48431                -0.246367         -0.572753   \n",
       "...           ...                      ...               ...   \n",
       "12750      Q9Y258                      NaN               NaN   \n",
       "12751      P20292                      NaN               NaN   \n",
       "12752      Q9H1C7                -0.313512          0.105086   \n",
       "12753      Q99735                      NaN               NaN   \n",
       "12754      Q9P003                 0.138191          0.355856   \n",
       "\n",
       "       AU565_BREAST_TenPx01  KMRC1_KIDNEY_TenPx01  CAL51_BREAST_TenPx01  \\\n",
       "0                 -0.463928             -0.883645              0.788565   \n",
       "1                       NaN                   NaN                   NaN   \n",
       "2                  0.191211             -0.595606              0.391243   \n",
       "3                       NaN                   NaN                   NaN   \n",
       "4                 -0.341305             -0.335211             -0.124474   \n",
       "...                     ...                   ...                   ...   \n",
       "12750                   NaN                   NaN                   NaN   \n",
       "12751                   NaN                   NaN                   NaN   \n",
       "12752             -0.586815             -0.132940              0.452423   \n",
       "12753                   NaN                   NaN                   NaN   \n",
       "12754              0.001459              0.172501              0.200756   \n",
       "\n",
       "       RPMI7951_SKIN_TenPx01  RERFLCMS_LUNG_TenPx01  IGR37_SKIN_TenPx01  \\\n",
       "0                  -0.912352               0.729167           -0.658213   \n",
       "1                        NaN                    NaN                 NaN   \n",
       "2                  -0.381329               0.410168            0.066076   \n",
       "3                        NaN                    NaN                 NaN   \n",
       "4                   0.043893               1.882953           -0.616307   \n",
       "...                      ...                    ...                 ...   \n",
       "12750                    NaN                    NaN                 NaN   \n",
       "12751                    NaN                    NaN                 NaN   \n",
       "12752              -0.546389               0.298349            0.461098   \n",
       "12753                    NaN                    NaN                 NaN   \n",
       "12754              -0.262573               0.120958           -0.035599   \n",
       "\n",
       "       VMRCRCW_KIDNEY_TenPx01  ...  NCIH2030_LUNG_TenPx18  \\\n",
       "0                   -1.150653  ...              -1.345024   \n",
       "1                         NaN  ...                    NaN   \n",
       "2                   -0.324100  ...               0.161196   \n",
       "3                         NaN  ...              -0.040198   \n",
       "4                   -0.533335  ...              -4.627606   \n",
       "...                       ...  ...                    ...   \n",
       "12750                     NaN  ...                    NaN   \n",
       "12751                     NaN  ...                    NaN   \n",
       "12752                0.309062  ...              -0.393738   \n",
       "12753                     NaN  ...                    NaN   \n",
       "12754               -0.352327  ...              -0.488230   \n",
       "\n",
       "       22RV1_PROSTATE_TenPx24  A172_CENTRAL_NERVOUS_SYSTEM_TenPx24  \\\n",
       "0                    1.811205                            -0.921176   \n",
       "1                         NaN                                  NaN   \n",
       "2                   -0.337994                            -0.455601   \n",
       "3                         NaN                                  NaN   \n",
       "4                   -1.261703                             1.136851   \n",
       "...                       ...                                  ...   \n",
       "12750                     NaN                                  NaN   \n",
       "12751               -0.690046                            -1.600231   \n",
       "12752                0.604400                             1.643414   \n",
       "12753                     NaN                                  NaN   \n",
       "12754                0.139008                             0.933261   \n",
       "\n",
       "       BT20_BREAST_TenPx24  CALU6_LUNG_TenPx24  \\\n",
       "0                -1.045727            0.845574   \n",
       "1                      NaN                 NaN   \n",
       "2                 0.392393            0.258380   \n",
       "3                      NaN                 NaN   \n",
       "4                -3.179027            1.424570   \n",
       "...                    ...                 ...   \n",
       "12750                  NaN                 NaN   \n",
       "12751            -1.110583           -1.047633   \n",
       "12752            -1.481004            0.408420   \n",
       "12753                  NaN                 NaN   \n",
       "12754             0.642768           -0.495987   \n",
       "\n",
       "       FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24  KP4_PANCREAS_TenPx24  \\\n",
       "0                                   -0.071063              0.702306   \n",
       "1                                         NaN                   NaN   \n",
       "2                                   -0.817245             -0.790010   \n",
       "3                                         NaN                   NaN   \n",
       "4                                    2.549687              4.424095   \n",
       "...                                       ...                   ...   \n",
       "12750                                     NaN                   NaN   \n",
       "12751                                0.023850             -1.393552   \n",
       "12752                               -0.177391              0.445977   \n",
       "12753                                     NaN                   NaN   \n",
       "12754                               -1.291515             -0.944284   \n",
       "\n",
       "       MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \\\n",
       "0                                              -1.395878     \n",
       "1                                                    NaN     \n",
       "2                                               0.929522     \n",
       "3                                                    NaN     \n",
       "4                                              -1.083698     \n",
       "...                                                  ...     \n",
       "12750                                                NaN     \n",
       "12751                                           4.556818     \n",
       "12752                                          -2.497558     \n",
       "12753                                                NaN     \n",
       "12754                                          -1.022551     \n",
       "\n",
       "       OVCAR8_OVARY_TenPx24  THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \n",
       "0                 -1.165032                                        -0.547536  \n",
       "1                       NaN                                              NaN  \n",
       "2                 -0.828736                                         0.735431  \n",
       "3                       NaN                                              NaN  \n",
       "4                 -1.640990                                        -2.905547  \n",
       "...                     ...                                              ...  \n",
       "12750                   NaN                                              NaN  \n",
       "12751             -0.531699                                         0.640295  \n",
       "12752              0.964551                                        -2.004067  \n",
       "12753                   NaN                                              NaN  \n",
       "12754              0.220199                                         0.094595  \n",
       "\n",
       "[12755 rows x 379 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "raw_protein_df = pd.read_csv('./data/raw_data/protein_quant_current_normalized.csv')\n",
    "raw_protein_df = raw_protein_df.drop(columns=['Protein_Id', 'Gene_Symbol', 'Description', 'Group_ID', 'Uniprot'])\n",
    "# Also drop columns names contain Peptides\n",
    "raw_protein_df = raw_protein_df[raw_protein_df.columns.drop(list(raw_protein_df.filter(regex='Peptides')))]\n",
    "display(raw_protein_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uniprot_Acc</th>\n",
       "      <th>MDAMB468_BREAST_TenPx01</th>\n",
       "      <th>SH4_SKIN_TenPx01</th>\n",
       "      <th>AU565_BREAST_TenPx01</th>\n",
       "      <th>KMRC1_KIDNEY_TenPx01</th>\n",
       "      <th>CAL51_BREAST_TenPx01</th>\n",
       "      <th>RPMI7951_SKIN_TenPx01</th>\n",
       "      <th>RERFLCMS_LUNG_TenPx01</th>\n",
       "      <th>IGR37_SKIN_TenPx01</th>\n",
       "      <th>VMRCRCW_KIDNEY_TenPx01</th>\n",
       "      <th>...</th>\n",
       "      <th>NCIH2030_LUNG_TenPx18</th>\n",
       "      <th>22RV1_PROSTATE_TenPx24</th>\n",
       "      <th>A172_CENTRAL_NERVOUS_SYSTEM_TenPx24</th>\n",
       "      <th>BT20_BREAST_TenPx24</th>\n",
       "      <th>CALU6_LUNG_TenPx24</th>\n",
       "      <th>FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24</th>\n",
       "      <th>KP4_PANCREAS_TenPx24</th>\n",
       "      <th>MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "      <th>OVCAR8_OVARY_TenPx24</th>\n",
       "      <th>THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P55011</td>\n",
       "      <td>2.111348</td>\n",
       "      <td>0.070468</td>\n",
       "      <td>-0.463928</td>\n",
       "      <td>-0.883645</td>\n",
       "      <td>0.788565</td>\n",
       "      <td>-0.912352</td>\n",
       "      <td>0.729167</td>\n",
       "      <td>-0.658213</td>\n",
       "      <td>-1.150653</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.345024</td>\n",
       "      <td>1.811205</td>\n",
       "      <td>-0.921176</td>\n",
       "      <td>-1.045727</td>\n",
       "      <td>0.845574</td>\n",
       "      <td>-0.071063</td>\n",
       "      <td>0.702306</td>\n",
       "      <td>-1.395878</td>\n",
       "      <td>-1.165032</td>\n",
       "      <td>-0.547536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P35453</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O60341</td>\n",
       "      <td>0.379683</td>\n",
       "      <td>-0.283538</td>\n",
       "      <td>0.191211</td>\n",
       "      <td>-0.595606</td>\n",
       "      <td>0.391243</td>\n",
       "      <td>-0.381329</td>\n",
       "      <td>0.410168</td>\n",
       "      <td>0.066076</td>\n",
       "      <td>-0.324100</td>\n",
       "      <td>...</td>\n",
       "      <td>0.161196</td>\n",
       "      <td>-0.337994</td>\n",
       "      <td>-0.455601</td>\n",
       "      <td>0.392393</td>\n",
       "      <td>0.258380</td>\n",
       "      <td>-0.817245</td>\n",
       "      <td>-0.790010</td>\n",
       "      <td>0.929522</td>\n",
       "      <td>-0.828736</td>\n",
       "      <td>0.735431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O00570</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.040198</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P48431</td>\n",
       "      <td>-0.246367</td>\n",
       "      <td>-0.572753</td>\n",
       "      <td>-0.341305</td>\n",
       "      <td>-0.335211</td>\n",
       "      <td>-0.124474</td>\n",
       "      <td>0.043893</td>\n",
       "      <td>1.882953</td>\n",
       "      <td>-0.616307</td>\n",
       "      <td>-0.533335</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.627606</td>\n",
       "      <td>-1.261703</td>\n",
       "      <td>1.136851</td>\n",
       "      <td>-3.179027</td>\n",
       "      <td>1.424570</td>\n",
       "      <td>2.549687</td>\n",
       "      <td>4.424095</td>\n",
       "      <td>-1.083698</td>\n",
       "      <td>-1.640990</td>\n",
       "      <td>-2.905547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12750</th>\n",
       "      <td>Q9Y258</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12751</th>\n",
       "      <td>P20292</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.690046</td>\n",
       "      <td>-1.600231</td>\n",
       "      <td>-1.110583</td>\n",
       "      <td>-1.047633</td>\n",
       "      <td>0.023850</td>\n",
       "      <td>-1.393552</td>\n",
       "      <td>4.556818</td>\n",
       "      <td>-0.531699</td>\n",
       "      <td>0.640295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12752</th>\n",
       "      <td>Q9H1C7</td>\n",
       "      <td>-0.313512</td>\n",
       "      <td>0.105086</td>\n",
       "      <td>-0.586815</td>\n",
       "      <td>-0.132940</td>\n",
       "      <td>0.452423</td>\n",
       "      <td>-0.546389</td>\n",
       "      <td>0.298349</td>\n",
       "      <td>0.461098</td>\n",
       "      <td>0.309062</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.393738</td>\n",
       "      <td>0.604400</td>\n",
       "      <td>1.643414</td>\n",
       "      <td>-1.481004</td>\n",
       "      <td>0.408420</td>\n",
       "      <td>-0.177391</td>\n",
       "      <td>0.445977</td>\n",
       "      <td>-2.497558</td>\n",
       "      <td>0.964551</td>\n",
       "      <td>-2.004067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12753</th>\n",
       "      <td>Q99735</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12754</th>\n",
       "      <td>Q9P003</td>\n",
       "      <td>0.138191</td>\n",
       "      <td>0.355856</td>\n",
       "      <td>0.001459</td>\n",
       "      <td>0.172501</td>\n",
       "      <td>0.200756</td>\n",
       "      <td>-0.262573</td>\n",
       "      <td>0.120958</td>\n",
       "      <td>-0.035599</td>\n",
       "      <td>-0.352327</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.488230</td>\n",
       "      <td>0.139008</td>\n",
       "      <td>0.933261</td>\n",
       "      <td>0.642768</td>\n",
       "      <td>-0.495987</td>\n",
       "      <td>-1.291515</td>\n",
       "      <td>-0.944284</td>\n",
       "      <td>-1.022551</td>\n",
       "      <td>0.220199</td>\n",
       "      <td>0.094595</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12755 rows × 379 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Uniprot_Acc  MDAMB468_BREAST_TenPx01  SH4_SKIN_TenPx01  \\\n",
       "0          P55011                 2.111348          0.070468   \n",
       "1          P35453                 0.000000          0.000000   \n",
       "2          O60341                 0.379683         -0.283538   \n",
       "3          O00570                 0.000000          0.000000   \n",
       "4          P48431                -0.246367         -0.572753   \n",
       "...           ...                      ...               ...   \n",
       "12750      Q9Y258                 0.000000          0.000000   \n",
       "12751      P20292                 0.000000          0.000000   \n",
       "12752      Q9H1C7                -0.313512          0.105086   \n",
       "12753      Q99735                 0.000000          0.000000   \n",
       "12754      Q9P003                 0.138191          0.355856   \n",
       "\n",
       "       AU565_BREAST_TenPx01  KMRC1_KIDNEY_TenPx01  CAL51_BREAST_TenPx01  \\\n",
       "0                 -0.463928             -0.883645              0.788565   \n",
       "1                  0.000000              0.000000              0.000000   \n",
       "2                  0.191211             -0.595606              0.391243   \n",
       "3                  0.000000              0.000000              0.000000   \n",
       "4                 -0.341305             -0.335211             -0.124474   \n",
       "...                     ...                   ...                   ...   \n",
       "12750              0.000000              0.000000              0.000000   \n",
       "12751              0.000000              0.000000              0.000000   \n",
       "12752             -0.586815             -0.132940              0.452423   \n",
       "12753              0.000000              0.000000              0.000000   \n",
       "12754              0.001459              0.172501              0.200756   \n",
       "\n",
       "       RPMI7951_SKIN_TenPx01  RERFLCMS_LUNG_TenPx01  IGR37_SKIN_TenPx01  \\\n",
       "0                  -0.912352               0.729167           -0.658213   \n",
       "1                   0.000000               0.000000            0.000000   \n",
       "2                  -0.381329               0.410168            0.066076   \n",
       "3                   0.000000               0.000000            0.000000   \n",
       "4                   0.043893               1.882953           -0.616307   \n",
       "...                      ...                    ...                 ...   \n",
       "12750               0.000000               0.000000            0.000000   \n",
       "12751               0.000000               0.000000            0.000000   \n",
       "12752              -0.546389               0.298349            0.461098   \n",
       "12753               0.000000               0.000000            0.000000   \n",
       "12754              -0.262573               0.120958           -0.035599   \n",
       "\n",
       "       VMRCRCW_KIDNEY_TenPx01  ...  NCIH2030_LUNG_TenPx18  \\\n",
       "0                   -1.150653  ...              -1.345024   \n",
       "1                    0.000000  ...               0.000000   \n",
       "2                   -0.324100  ...               0.161196   \n",
       "3                    0.000000  ...              -0.040198   \n",
       "4                   -0.533335  ...              -4.627606   \n",
       "...                       ...  ...                    ...   \n",
       "12750                0.000000  ...               0.000000   \n",
       "12751                0.000000  ...               0.000000   \n",
       "12752                0.309062  ...              -0.393738   \n",
       "12753                0.000000  ...               0.000000   \n",
       "12754               -0.352327  ...              -0.488230   \n",
       "\n",
       "       22RV1_PROSTATE_TenPx24  A172_CENTRAL_NERVOUS_SYSTEM_TenPx24  \\\n",
       "0                    1.811205                            -0.921176   \n",
       "1                    0.000000                             0.000000   \n",
       "2                   -0.337994                            -0.455601   \n",
       "3                    0.000000                             0.000000   \n",
       "4                   -1.261703                             1.136851   \n",
       "...                       ...                                  ...   \n",
       "12750                0.000000                             0.000000   \n",
       "12751               -0.690046                            -1.600231   \n",
       "12752                0.604400                             1.643414   \n",
       "12753                0.000000                             0.000000   \n",
       "12754                0.139008                             0.933261   \n",
       "\n",
       "       BT20_BREAST_TenPx24  CALU6_LUNG_TenPx24  \\\n",
       "0                -1.045727            0.845574   \n",
       "1                 0.000000            0.000000   \n",
       "2                 0.392393            0.258380   \n",
       "3                 0.000000            0.000000   \n",
       "4                -3.179027            1.424570   \n",
       "...                    ...                 ...   \n",
       "12750             0.000000            0.000000   \n",
       "12751            -1.110583           -1.047633   \n",
       "12752            -1.481004            0.408420   \n",
       "12753             0.000000            0.000000   \n",
       "12754             0.642768           -0.495987   \n",
       "\n",
       "       FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24  KP4_PANCREAS_TenPx24  \\\n",
       "0                                   -0.071063              0.702306   \n",
       "1                                    0.000000              0.000000   \n",
       "2                                   -0.817245             -0.790010   \n",
       "3                                    0.000000              0.000000   \n",
       "4                                    2.549687              4.424095   \n",
       "...                                       ...                   ...   \n",
       "12750                                0.000000              0.000000   \n",
       "12751                                0.023850             -1.393552   \n",
       "12752                               -0.177391              0.445977   \n",
       "12753                                0.000000              0.000000   \n",
       "12754                               -1.291515             -0.944284   \n",
       "\n",
       "       MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \\\n",
       "0                                              -1.395878     \n",
       "1                                               0.000000     \n",
       "2                                               0.929522     \n",
       "3                                               0.000000     \n",
       "4                                              -1.083698     \n",
       "...                                                  ...     \n",
       "12750                                           0.000000     \n",
       "12751                                           4.556818     \n",
       "12752                                          -2.497558     \n",
       "12753                                           0.000000     \n",
       "12754                                          -1.022551     \n",
       "\n",
       "       OVCAR8_OVARY_TenPx24  THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \n",
       "0                 -1.165032                                        -0.547536  \n",
       "1                  0.000000                                         0.000000  \n",
       "2                 -0.828736                                         0.735431  \n",
       "3                  0.000000                                         0.000000  \n",
       "4                 -1.640990                                        -2.905547  \n",
       "...                     ...                                              ...  \n",
       "12750              0.000000                                         0.000000  \n",
       "12751             -0.531699                                         0.640295  \n",
       "12752              0.964551                                        -2.004067  \n",
       "12753              0.000000                                         0.000000  \n",
       "12754              0.220199                                         0.094595  \n",
       "\n",
       "[12755 rows x 379 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Identify columns to convert (all except 'Unnamed: 0' and 'protein_name')\n",
    "cols_to_convert = raw_protein_df.columns.difference(['Uniprot_Acc'])\n",
    "# Convert these columns to numeric (using pd.to_numeric for safety, which can handle errors)\n",
    "raw_protein_df[cols_to_convert] = raw_protein_df[cols_to_convert].apply(pd.to_numeric, errors='coerce')\n",
    "raw_protein_df = raw_protein_df.fillna(0.0)\n",
    "display(raw_protein_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3695680667.py:2: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_df = raw_protein_df.groupby('Uniprot_Acc', as_index=False).mean()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uniprot_Acc</th>\n",
       "      <th>MDAMB468_BREAST_TenPx01</th>\n",
       "      <th>SH4_SKIN_TenPx01</th>\n",
       "      <th>AU565_BREAST_TenPx01</th>\n",
       "      <th>KMRC1_KIDNEY_TenPx01</th>\n",
       "      <th>CAL51_BREAST_TenPx01</th>\n",
       "      <th>RPMI7951_SKIN_TenPx01</th>\n",
       "      <th>RERFLCMS_LUNG_TenPx01</th>\n",
       "      <th>IGR37_SKIN_TenPx01</th>\n",
       "      <th>VMRCRCW_KIDNEY_TenPx01</th>\n",
       "      <th>...</th>\n",
       "      <th>NCIH2030_LUNG_TenPx18</th>\n",
       "      <th>22RV1_PROSTATE_TenPx24</th>\n",
       "      <th>A172_CENTRAL_NERVOUS_SYSTEM_TenPx24</th>\n",
       "      <th>BT20_BREAST_TenPx24</th>\n",
       "      <th>CALU6_LUNG_TenPx24</th>\n",
       "      <th>FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24</th>\n",
       "      <th>KP4_PANCREAS_TenPx24</th>\n",
       "      <th>MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "      <th>OVCAR8_OVARY_TenPx24</th>\n",
       "      <th>THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0AUZ9-2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.035406</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0AV02-4</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0AV96</td>\n",
       "      <td>0.358567</td>\n",
       "      <td>-1.112410</td>\n",
       "      <td>0.855575</td>\n",
       "      <td>0.061377</td>\n",
       "      <td>0.284258</td>\n",
       "      <td>-0.485070</td>\n",
       "      <td>0.092390</td>\n",
       "      <td>-0.824035</td>\n",
       "      <td>0.926286</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.347445</td>\n",
       "      <td>2.090889</td>\n",
       "      <td>-1.348085</td>\n",
       "      <td>2.449369</td>\n",
       "      <td>-1.362958</td>\n",
       "      <td>0.348655</td>\n",
       "      <td>-1.442707</td>\n",
       "      <td>0.436273</td>\n",
       "      <td>-1.585523</td>\n",
       "      <td>0.077705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0AVF1</td>\n",
       "      <td>-0.172066</td>\n",
       "      <td>0.339446</td>\n",
       "      <td>-0.181171</td>\n",
       "      <td>-0.341233</td>\n",
       "      <td>-0.059558</td>\n",
       "      <td>0.175596</td>\n",
       "      <td>-0.103205</td>\n",
       "      <td>0.359589</td>\n",
       "      <td>-0.042908</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.650165</td>\n",
       "      <td>0.909498</td>\n",
       "      <td>0.761474</td>\n",
       "      <td>-0.181335</td>\n",
       "      <td>-1.032205</td>\n",
       "      <td>-0.684201</td>\n",
       "      <td>-0.770205</td>\n",
       "      <td>0.289067</td>\n",
       "      <td>-1.031203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0AVI2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.458639</td>\n",
       "      <td>0.578876</td>\n",
       "      <td>-0.487621</td>\n",
       "      <td>-2.524540</td>\n",
       "      <td>-0.532270</td>\n",
       "      <td>2.834559</td>\n",
       "      <td>-0.101096</td>\n",
       "      <td>0.621485</td>\n",
       "      <td>3.827484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12750</th>\n",
       "      <td>U3KPW6</td>\n",
       "      <td>-0.421581</td>\n",
       "      <td>0.845537</td>\n",
       "      <td>0.446649</td>\n",
       "      <td>0.085607</td>\n",
       "      <td>0.079110</td>\n",
       "      <td>-1.501110</td>\n",
       "      <td>0.125189</td>\n",
       "      <td>1.085682</td>\n",
       "      <td>-0.523255</td>\n",
       "      <td>...</td>\n",
       "      <td>0.637635</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12751</th>\n",
       "      <td>U3KQ28</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054038</td>\n",
       "      <td>0.568362</td>\n",
       "      <td>-0.300467</td>\n",
       "      <td>-0.144897</td>\n",
       "      <td>-0.023671</td>\n",
       "      <td>0.088995</td>\n",
       "      <td>0.287703</td>\n",
       "      <td>0.124844</td>\n",
       "      <td>0.354875</td>\n",
       "      <td>0.052102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12752</th>\n",
       "      <td>U3KQ54</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.062826</td>\n",
       "      <td>-0.367516</td>\n",
       "      <td>-0.323326</td>\n",
       "      <td>0.422455</td>\n",
       "      <td>0.617692</td>\n",
       "      <td>0.697067</td>\n",
       "      <td>-0.435225</td>\n",
       "      <td>-0.362489</td>\n",
       "      <td>-0.360115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12753</th>\n",
       "      <td>U3KQB3</td>\n",
       "      <td>-1.372372</td>\n",
       "      <td>-0.648092</td>\n",
       "      <td>0.154645</td>\n",
       "      <td>0.593048</td>\n",
       "      <td>0.268130</td>\n",
       "      <td>0.025445</td>\n",
       "      <td>0.584467</td>\n",
       "      <td>0.035815</td>\n",
       "      <td>0.101186</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12754</th>\n",
       "      <td>U3KQT9</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>4.319799</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12755 rows × 379 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Uniprot_Acc  MDAMB468_BREAST_TenPx01  SH4_SKIN_TenPx01  \\\n",
       "0        A0AUZ9-2                 0.000000          0.000000   \n",
       "1        A0AV02-4                 0.000000          0.000000   \n",
       "2          A0AV96                 0.358567         -1.112410   \n",
       "3          A0AVF1                -0.172066          0.339446   \n",
       "4          A0AVI2                 0.000000          0.000000   \n",
       "...           ...                      ...               ...   \n",
       "12750      U3KPW6                -0.421581          0.845537   \n",
       "12751      U3KQ28                 0.000000          0.000000   \n",
       "12752      U3KQ54                 0.000000          0.000000   \n",
       "12753      U3KQB3                -1.372372         -0.648092   \n",
       "12754      U3KQT9                 0.000000          0.000000   \n",
       "\n",
       "       AU565_BREAST_TenPx01  KMRC1_KIDNEY_TenPx01  CAL51_BREAST_TenPx01  \\\n",
       "0                  0.000000              0.000000              0.000000   \n",
       "1                  0.000000              0.000000              0.000000   \n",
       "2                  0.855575              0.061377              0.284258   \n",
       "3                 -0.181171             -0.341233             -0.059558   \n",
       "4                  0.000000              0.000000              0.000000   \n",
       "...                     ...                   ...                   ...   \n",
       "12750              0.446649              0.085607              0.079110   \n",
       "12751              0.000000              0.000000              0.000000   \n",
       "12752              0.000000              0.000000              0.000000   \n",
       "12753              0.154645              0.593048              0.268130   \n",
       "12754              0.000000              0.000000              0.000000   \n",
       "\n",
       "       RPMI7951_SKIN_TenPx01  RERFLCMS_LUNG_TenPx01  IGR37_SKIN_TenPx01  \\\n",
       "0                   0.000000               0.000000            0.000000   \n",
       "1                   0.000000               0.000000            0.000000   \n",
       "2                  -0.485070               0.092390           -0.824035   \n",
       "3                   0.175596              -0.103205            0.359589   \n",
       "4                   0.000000               0.000000            0.000000   \n",
       "...                      ...                    ...                 ...   \n",
       "12750              -1.501110               0.125189            1.085682   \n",
       "12751               0.000000               0.000000            0.000000   \n",
       "12752               0.000000               0.000000            0.000000   \n",
       "12753               0.025445               0.584467            0.035815   \n",
       "12754               0.000000               0.000000            0.000000   \n",
       "\n",
       "       VMRCRCW_KIDNEY_TenPx01  ...  NCIH2030_LUNG_TenPx18  \\\n",
       "0                    0.000000  ...               0.035406   \n",
       "1                    0.000000  ...               0.000000   \n",
       "2                    0.926286  ...              -1.347445   \n",
       "3                   -0.042908  ...               0.000000   \n",
       "4                    0.000000  ...               0.000000   \n",
       "...                       ...  ...                    ...   \n",
       "12750               -0.523255  ...               0.637635   \n",
       "12751                0.000000  ...               0.054038   \n",
       "12752                0.000000  ...               0.000000   \n",
       "12753                0.101186  ...               0.000000   \n",
       "12754                0.000000  ...               4.319799   \n",
       "\n",
       "       22RV1_PROSTATE_TenPx24  A172_CENTRAL_NERVOUS_SYSTEM_TenPx24  \\\n",
       "0                    0.000000                             0.000000   \n",
       "1                    0.000000                             0.000000   \n",
       "2                    2.090889                            -1.348085   \n",
       "3                   -0.650165                             0.909498   \n",
       "4                   -0.458639                             0.578876   \n",
       "...                       ...                                  ...   \n",
       "12750                0.000000                             0.000000   \n",
       "12751                0.568362                            -0.300467   \n",
       "12752                1.062826                            -0.367516   \n",
       "12753                0.000000                             0.000000   \n",
       "12754                0.000000                             0.000000   \n",
       "\n",
       "       BT20_BREAST_TenPx24  CALU6_LUNG_TenPx24  \\\n",
       "0                 0.000000            0.000000   \n",
       "1                 0.000000            0.000000   \n",
       "2                 2.449369           -1.362958   \n",
       "3                 0.761474           -0.181335   \n",
       "4                -0.487621           -2.524540   \n",
       "...                    ...                 ...   \n",
       "12750             0.000000            0.000000   \n",
       "12751            -0.144897           -0.023671   \n",
       "12752            -0.323326            0.422455   \n",
       "12753             0.000000            0.000000   \n",
       "12754             0.000000            0.000000   \n",
       "\n",
       "       FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24  KP4_PANCREAS_TenPx24  \\\n",
       "0                                    0.000000              0.000000   \n",
       "1                                    0.000000              0.000000   \n",
       "2                                    0.348655             -1.442707   \n",
       "3                                   -1.032205             -0.684201   \n",
       "4                                   -0.532270              2.834559   \n",
       "...                                       ...                   ...   \n",
       "12750                                0.000000              0.000000   \n",
       "12751                                0.088995              0.287703   \n",
       "12752                                0.617692              0.697067   \n",
       "12753                                0.000000              0.000000   \n",
       "12754                                0.000000              0.000000   \n",
       "\n",
       "       MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \\\n",
       "0                                               0.000000     \n",
       "1                                               0.000000     \n",
       "2                                               0.436273     \n",
       "3                                              -0.770205     \n",
       "4                                              -0.101096     \n",
       "...                                                  ...     \n",
       "12750                                           0.000000     \n",
       "12751                                           0.124844     \n",
       "12752                                          -0.435225     \n",
       "12753                                           0.000000     \n",
       "12754                                           0.000000     \n",
       "\n",
       "       OVCAR8_OVARY_TenPx24  THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \n",
       "0                  0.000000                                         0.000000  \n",
       "1                  0.000000                                         0.000000  \n",
       "2                 -1.585523                                         0.077705  \n",
       "3                  0.289067                                        -1.031203  \n",
       "4                  0.621485                                         3.827484  \n",
       "...                     ...                                              ...  \n",
       "12750              0.000000                                         0.000000  \n",
       "12751              0.354875                                         0.052102  \n",
       "12752             -0.362489                                        -0.360115  \n",
       "12753              0.000000                                         0.000000  \n",
       "12754              0.000000                                         0.000000  \n",
       "\n",
       "[12755 rows x 379 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# groupby the protein_name and calculate the mean\n",
    "protein_df = raw_protein_df.groupby('Uniprot_Acc', as_index=False).mean()\n",
    "display(protein_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\1778311738.py:2: DtypeWarning: Columns (2,7) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmg_protein_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>A0A023T787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>A0A024QYX0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>A0A024QYX5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>A0A024QZ37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>A0A024QZ45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  Uniprot_ID\n",
       "0               BMGC_PT000001  A0A023T787\n",
       "1               BMGC_PT000002  A0A024QYX0\n",
       "2               BMGC_PT000003  A0A024QYX5\n",
       "3               BMGC_PT000004  A0A024QZ37\n",
       "4               BMGC_PT000005  A0A024QZ45\n",
       "...                       ...         ...\n",
       "121414          BMGC_PT121415         NaN\n",
       "121415          BMGC_PT121416         NaN\n",
       "121416          BMGC_PT121417         NaN\n",
       "121417          BMGC_PT121418         NaN\n",
       "121418          BMGC_PT121419         NaN\n",
       "\n",
       "[121419 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# merge the biomedgraphica_id with the raw_protein_df\n",
    "bmg_protein_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein.csv')\n",
    "bmg_protein_df = bmg_protein_df[['BioMedGraphica_Conn_ID', 'Uniprot_ID']]\n",
    "display(bmg_protein_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>MDAMB468_BREAST_TenPx01</th>\n",
       "      <th>SH4_SKIN_TenPx01</th>\n",
       "      <th>AU565_BREAST_TenPx01</th>\n",
       "      <th>KMRC1_KIDNEY_TenPx01</th>\n",
       "      <th>CAL51_BREAST_TenPx01</th>\n",
       "      <th>RPMI7951_SKIN_TenPx01</th>\n",
       "      <th>RERFLCMS_LUNG_TenPx01</th>\n",
       "      <th>IGR37_SKIN_TenPx01</th>\n",
       "      <th>VMRCRCW_KIDNEY_TenPx01</th>\n",
       "      <th>...</th>\n",
       "      <th>NCIH2030_LUNG_TenPx18</th>\n",
       "      <th>22RV1_PROSTATE_TenPx24</th>\n",
       "      <th>A172_CENTRAL_NERVOUS_SYSTEM_TenPx24</th>\n",
       "      <th>BT20_BREAST_TenPx24</th>\n",
       "      <th>CALU6_LUNG_TenPx24</th>\n",
       "      <th>FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24</th>\n",
       "      <th>KP4_PANCREAS_TenPx24</th>\n",
       "      <th>MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "      <th>OVCAR8_OVARY_TenPx24</th>\n",
       "      <th>THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT001337</td>\n",
       "      <td>0.358567</td>\n",
       "      <td>-1.112410</td>\n",
       "      <td>0.855575</td>\n",
       "      <td>0.061377</td>\n",
       "      <td>0.284258</td>\n",
       "      <td>-0.485070</td>\n",
       "      <td>0.092390</td>\n",
       "      <td>-0.824035</td>\n",
       "      <td>0.926286</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.347445</td>\n",
       "      <td>2.090889</td>\n",
       "      <td>-1.348085</td>\n",
       "      <td>2.449369</td>\n",
       "      <td>-1.362958</td>\n",
       "      <td>0.348655</td>\n",
       "      <td>-1.442707</td>\n",
       "      <td>0.436273</td>\n",
       "      <td>-1.585523</td>\n",
       "      <td>0.077705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT001347</td>\n",
       "      <td>-0.172066</td>\n",
       "      <td>0.339446</td>\n",
       "      <td>-0.181171</td>\n",
       "      <td>-0.341233</td>\n",
       "      <td>-0.059558</td>\n",
       "      <td>0.175596</td>\n",
       "      <td>-0.103205</td>\n",
       "      <td>0.359589</td>\n",
       "      <td>-0.042908</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.650165</td>\n",
       "      <td>0.909498</td>\n",
       "      <td>0.761474</td>\n",
       "      <td>-0.181335</td>\n",
       "      <td>-1.032205</td>\n",
       "      <td>-0.684201</td>\n",
       "      <td>-0.770205</td>\n",
       "      <td>0.289067</td>\n",
       "      <td>-1.031203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT001355</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.458639</td>\n",
       "      <td>0.578876</td>\n",
       "      <td>-0.487621</td>\n",
       "      <td>-2.524540</td>\n",
       "      <td>-0.532270</td>\n",
       "      <td>2.834559</td>\n",
       "      <td>-0.101096</td>\n",
       "      <td>0.621485</td>\n",
       "      <td>3.827484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT001356</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT001359</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10611</th>\n",
       "      <td>BMGC_PT048268</td>\n",
       "      <td>0.531404</td>\n",
       "      <td>-0.228307</td>\n",
       "      <td>0.270226</td>\n",
       "      <td>0.041147</td>\n",
       "      <td>0.044493</td>\n",
       "      <td>-0.320486</td>\n",
       "      <td>-0.084089</td>\n",
       "      <td>-0.439713</td>\n",
       "      <td>0.108860</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.114846</td>\n",
       "      <td>0.438270</td>\n",
       "      <td>-0.314925</td>\n",
       "      <td>-0.064443</td>\n",
       "      <td>-0.077137</td>\n",
       "      <td>-0.614710</td>\n",
       "      <td>-0.511348</td>\n",
       "      <td>-0.163492</td>\n",
       "      <td>-0.016835</td>\n",
       "      <td>0.113500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10612</th>\n",
       "      <td>BMGC_PT048269</td>\n",
       "      <td>0.140324</td>\n",
       "      <td>0.629608</td>\n",
       "      <td>-0.387707</td>\n",
       "      <td>-0.218811</td>\n",
       "      <td>0.205440</td>\n",
       "      <td>-0.163033</td>\n",
       "      <td>-0.363271</td>\n",
       "      <td>0.510087</td>\n",
       "      <td>-0.220798</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.187114</td>\n",
       "      <td>-0.394658</td>\n",
       "      <td>0.907714</td>\n",
       "      <td>-0.270577</td>\n",
       "      <td>0.003400</td>\n",
       "      <td>-1.732360</td>\n",
       "      <td>-0.685070</td>\n",
       "      <td>1.224937</td>\n",
       "      <td>0.328577</td>\n",
       "      <td>-0.276705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10613</th>\n",
       "      <td>BMGC_PT048274</td>\n",
       "      <td>0.007468</td>\n",
       "      <td>-0.316567</td>\n",
       "      <td>0.084812</td>\n",
       "      <td>0.161356</td>\n",
       "      <td>0.072855</td>\n",
       "      <td>0.166777</td>\n",
       "      <td>-0.161962</td>\n",
       "      <td>-0.391137</td>\n",
       "      <td>0.215005</td>\n",
       "      <td>...</td>\n",
       "      <td>0.088471</td>\n",
       "      <td>-0.371191</td>\n",
       "      <td>0.785289</td>\n",
       "      <td>-0.158731</td>\n",
       "      <td>0.087618</td>\n",
       "      <td>-0.428187</td>\n",
       "      <td>-0.291988</td>\n",
       "      <td>0.452430</td>\n",
       "      <td>-0.251963</td>\n",
       "      <td>-0.546281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10614</th>\n",
       "      <td>BMGC_PT048275</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.019504</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10615</th>\n",
       "      <td>BMGC_PT048276</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10616 rows × 379 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  MDAMB468_BREAST_TenPx01  SH4_SKIN_TenPx01  \\\n",
       "0              BMGC_PT001337                 0.358567         -1.112410   \n",
       "1              BMGC_PT001347                -0.172066          0.339446   \n",
       "2              BMGC_PT001355                 0.000000          0.000000   \n",
       "3              BMGC_PT001356                 0.000000          0.000000   \n",
       "4              BMGC_PT001359                 0.000000          0.000000   \n",
       "...                      ...                      ...               ...   \n",
       "10611          BMGC_PT048268                 0.531404         -0.228307   \n",
       "10612          BMGC_PT048269                 0.140324          0.629608   \n",
       "10613          BMGC_PT048274                 0.007468         -0.316567   \n",
       "10614          BMGC_PT048275                 0.000000          0.000000   \n",
       "10615          BMGC_PT048276                 0.000000          0.000000   \n",
       "\n",
       "       AU565_BREAST_TenPx01  KMRC1_KIDNEY_TenPx01  CAL51_BREAST_TenPx01  \\\n",
       "0                  0.855575              0.061377              0.284258   \n",
       "1                 -0.181171             -0.341233             -0.059558   \n",
       "2                  0.000000              0.000000              0.000000   \n",
       "3                  0.000000              0.000000              0.000000   \n",
       "4                  0.000000              0.000000              0.000000   \n",
       "...                     ...                   ...                   ...   \n",
       "10611              0.270226              0.041147              0.044493   \n",
       "10612             -0.387707             -0.218811              0.205440   \n",
       "10613              0.084812              0.161356              0.072855   \n",
       "10614              0.000000              0.000000              0.000000   \n",
       "10615              0.000000              0.000000              0.000000   \n",
       "\n",
       "       RPMI7951_SKIN_TenPx01  RERFLCMS_LUNG_TenPx01  IGR37_SKIN_TenPx01  \\\n",
       "0                  -0.485070               0.092390           -0.824035   \n",
       "1                   0.175596              -0.103205            0.359589   \n",
       "2                   0.000000               0.000000            0.000000   \n",
       "3                   0.000000               0.000000            0.000000   \n",
       "4                   0.000000               0.000000            0.000000   \n",
       "...                      ...                    ...                 ...   \n",
       "10611              -0.320486              -0.084089           -0.439713   \n",
       "10612              -0.163033              -0.363271            0.510087   \n",
       "10613               0.166777              -0.161962           -0.391137   \n",
       "10614               0.000000               0.000000            0.000000   \n",
       "10615               0.000000               0.000000            0.000000   \n",
       "\n",
       "       VMRCRCW_KIDNEY_TenPx01  ...  NCIH2030_LUNG_TenPx18  \\\n",
       "0                    0.926286  ...              -1.347445   \n",
       "1                   -0.042908  ...               0.000000   \n",
       "2                    0.000000  ...               0.000000   \n",
       "3                    0.000000  ...               0.000000   \n",
       "4                    0.000000  ...               0.000000   \n",
       "...                       ...  ...                    ...   \n",
       "10611                0.108860  ...              -0.114846   \n",
       "10612               -0.220798  ...              -0.187114   \n",
       "10613                0.215005  ...               0.088471   \n",
       "10614                0.000000  ...              -1.019504   \n",
       "10615                0.000000  ...               0.000000   \n",
       "\n",
       "       22RV1_PROSTATE_TenPx24  A172_CENTRAL_NERVOUS_SYSTEM_TenPx24  \\\n",
       "0                    2.090889                            -1.348085   \n",
       "1                   -0.650165                             0.909498   \n",
       "2                   -0.458639                             0.578876   \n",
       "3                    0.000000                             0.000000   \n",
       "4                    0.000000                             0.000000   \n",
       "...                       ...                                  ...   \n",
       "10611                0.438270                            -0.314925   \n",
       "10612               -0.394658                             0.907714   \n",
       "10613               -0.371191                             0.785289   \n",
       "10614                0.000000                             0.000000   \n",
       "10615                0.000000                             0.000000   \n",
       "\n",
       "       BT20_BREAST_TenPx24  CALU6_LUNG_TenPx24  \\\n",
       "0                 2.449369           -1.362958   \n",
       "1                 0.761474           -0.181335   \n",
       "2                -0.487621           -2.524540   \n",
       "3                 0.000000            0.000000   \n",
       "4                 0.000000            0.000000   \n",
       "...                    ...                 ...   \n",
       "10611            -0.064443           -0.077137   \n",
       "10612            -0.270577            0.003400   \n",
       "10613            -0.158731            0.087618   \n",
       "10614             0.000000            0.000000   \n",
       "10615             0.000000            0.000000   \n",
       "\n",
       "       FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24  KP4_PANCREAS_TenPx24  \\\n",
       "0                                    0.348655             -1.442707   \n",
       "1                                   -1.032205             -0.684201   \n",
       "2                                   -0.532270              2.834559   \n",
       "3                                    0.000000              0.000000   \n",
       "4                                    0.000000              0.000000   \n",
       "...                                       ...                   ...   \n",
       "10611                               -0.614710             -0.511348   \n",
       "10612                               -1.732360             -0.685070   \n",
       "10613                               -0.428187             -0.291988   \n",
       "10614                                0.000000              0.000000   \n",
       "10615                                0.000000              0.000000   \n",
       "\n",
       "       MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \\\n",
       "0                                               0.436273     \n",
       "1                                              -0.770205     \n",
       "2                                              -0.101096     \n",
       "3                                               0.000000     \n",
       "4                                               0.000000     \n",
       "...                                                  ...     \n",
       "10611                                          -0.163492     \n",
       "10612                                           1.224937     \n",
       "10613                                           0.452430     \n",
       "10614                                           0.000000     \n",
       "10615                                           0.000000     \n",
       "\n",
       "       OVCAR8_OVARY_TenPx24  THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \n",
       "0                 -1.585523                                         0.077705  \n",
       "1                  0.289067                                        -1.031203  \n",
       "2                  0.621485                                         3.827484  \n",
       "3                  0.000000                                         0.000000  \n",
       "4                  0.000000                                         0.000000  \n",
       "...                     ...                                              ...  \n",
       "10611             -0.016835                                         0.113500  \n",
       "10612              0.328577                                        -0.276705  \n",
       "10613             -0.251963                                        -0.546281  \n",
       "10614              0.000000                                         0.000000  \n",
       "10615              0.000000                                         0.000000  \n",
       "\n",
       "[10616 rows x 379 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# merge the biomedgraphica_id with the raw_protein_df\n",
    "merged_protein_df = pd.merge(bmg_protein_df, protein_df, left_on='Uniprot_ID', right_on='Uniprot_Acc', how='inner')\n",
    "merged_protein_df.drop(columns=['Uniprot_ID', 'Uniprot_Acc'], inplace=True)\n",
    "display(merged_protein_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>MDAMB468_BREAST_TenPx01</th>\n",
       "      <th>SH4_SKIN_TenPx01</th>\n",
       "      <th>AU565_BREAST_TenPx01</th>\n",
       "      <th>KMRC1_KIDNEY_TenPx01</th>\n",
       "      <th>CAL51_BREAST_TenPx01</th>\n",
       "      <th>RPMI7951_SKIN_TenPx01</th>\n",
       "      <th>RERFLCMS_LUNG_TenPx01</th>\n",
       "      <th>IGR37_SKIN_TenPx01</th>\n",
       "      <th>VMRCRCW_KIDNEY_TenPx01</th>\n",
       "      <th>...</th>\n",
       "      <th>NCIH2030_LUNG_TenPx18</th>\n",
       "      <th>22RV1_PROSTATE_TenPx24</th>\n",
       "      <th>A172_CENTRAL_NERVOUS_SYSTEM_TenPx24</th>\n",
       "      <th>BT20_BREAST_TenPx24</th>\n",
       "      <th>CALU6_LUNG_TenPx24</th>\n",
       "      <th>FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24</th>\n",
       "      <th>KP4_PANCREAS_TenPx24</th>\n",
       "      <th>MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "      <th>OVCAR8_OVARY_TenPx24</th>\n",
       "      <th>THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 379 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  MDAMB468_BREAST_TenPx01  SH4_SKIN_TenPx01  \\\n",
       "0               BMGC_PT000001                      0.0               0.0   \n",
       "1               BMGC_PT000002                      0.0               0.0   \n",
       "2               BMGC_PT000003                      0.0               0.0   \n",
       "3               BMGC_PT000004                      0.0               0.0   \n",
       "4               BMGC_PT000005                      0.0               0.0   \n",
       "...                       ...                      ...               ...   \n",
       "121414          BMGC_PT121415                      0.0               0.0   \n",
       "121415          BMGC_PT121416                      0.0               0.0   \n",
       "121416          BMGC_PT121417                      0.0               0.0   \n",
       "121417          BMGC_PT121418                      0.0               0.0   \n",
       "121418          BMGC_PT121419                      0.0               0.0   \n",
       "\n",
       "        AU565_BREAST_TenPx01  KMRC1_KIDNEY_TenPx01  CAL51_BREAST_TenPx01  \\\n",
       "0                        0.0                   0.0                   0.0   \n",
       "1                        0.0                   0.0                   0.0   \n",
       "2                        0.0                   0.0                   0.0   \n",
       "3                        0.0                   0.0                   0.0   \n",
       "4                        0.0                   0.0                   0.0   \n",
       "...                      ...                   ...                   ...   \n",
       "121414                   0.0                   0.0                   0.0   \n",
       "121415                   0.0                   0.0                   0.0   \n",
       "121416                   0.0                   0.0                   0.0   \n",
       "121417                   0.0                   0.0                   0.0   \n",
       "121418                   0.0                   0.0                   0.0   \n",
       "\n",
       "        RPMI7951_SKIN_TenPx01  RERFLCMS_LUNG_TenPx01  IGR37_SKIN_TenPx01  \\\n",
       "0                         0.0                    0.0                 0.0   \n",
       "1                         0.0                    0.0                 0.0   \n",
       "2                         0.0                    0.0                 0.0   \n",
       "3                         0.0                    0.0                 0.0   \n",
       "4                         0.0                    0.0                 0.0   \n",
       "...                       ...                    ...                 ...   \n",
       "121414                    0.0                    0.0                 0.0   \n",
       "121415                    0.0                    0.0                 0.0   \n",
       "121416                    0.0                    0.0                 0.0   \n",
       "121417                    0.0                    0.0                 0.0   \n",
       "121418                    0.0                    0.0                 0.0   \n",
       "\n",
       "        VMRCRCW_KIDNEY_TenPx01  ...  NCIH2030_LUNG_TenPx18  \\\n",
       "0                          0.0  ...                    0.0   \n",
       "1                          0.0  ...                    0.0   \n",
       "2                          0.0  ...                    0.0   \n",
       "3                          0.0  ...                    0.0   \n",
       "4                          0.0  ...                    0.0   \n",
       "...                        ...  ...                    ...   \n",
       "121414                     0.0  ...                    0.0   \n",
       "121415                     0.0  ...                    0.0   \n",
       "121416                     0.0  ...                    0.0   \n",
       "121417                     0.0  ...                    0.0   \n",
       "121418                     0.0  ...                    0.0   \n",
       "\n",
       "        22RV1_PROSTATE_TenPx24  A172_CENTRAL_NERVOUS_SYSTEM_TenPx24  \\\n",
       "0                          0.0                                  0.0   \n",
       "1                          0.0                                  0.0   \n",
       "2                          0.0                                  0.0   \n",
       "3                          0.0                                  0.0   \n",
       "4                          0.0                                  0.0   \n",
       "...                        ...                                  ...   \n",
       "121414                     0.0                                  0.0   \n",
       "121415                     0.0                                  0.0   \n",
       "121416                     0.0                                  0.0   \n",
       "121417                     0.0                                  0.0   \n",
       "121418                     0.0                                  0.0   \n",
       "\n",
       "        BT20_BREAST_TenPx24  CALU6_LUNG_TenPx24  \\\n",
       "0                       0.0                 0.0   \n",
       "1                       0.0                 0.0   \n",
       "2                       0.0                 0.0   \n",
       "3                       0.0                 0.0   \n",
       "4                       0.0                 0.0   \n",
       "...                     ...                 ...   \n",
       "121414                  0.0                 0.0   \n",
       "121415                  0.0                 0.0   \n",
       "121416                  0.0                 0.0   \n",
       "121417                  0.0                 0.0   \n",
       "121418                  0.0                 0.0   \n",
       "\n",
       "        FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24  KP4_PANCREAS_TenPx24  \\\n",
       "0                                          0.0                   0.0   \n",
       "1                                          0.0                   0.0   \n",
       "2                                          0.0                   0.0   \n",
       "3                                          0.0                   0.0   \n",
       "4                                          0.0                   0.0   \n",
       "...                                        ...                   ...   \n",
       "121414                                     0.0                   0.0   \n",
       "121415                                     0.0                   0.0   \n",
       "121416                                     0.0                   0.0   \n",
       "121417                                     0.0                   0.0   \n",
       "121418                                     0.0                   0.0   \n",
       "\n",
       "        MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \\\n",
       "0                                                     0.0     \n",
       "1                                                     0.0     \n",
       "2                                                     0.0     \n",
       "3                                                     0.0     \n",
       "4                                                     0.0     \n",
       "...                                                   ...     \n",
       "121414                                                0.0     \n",
       "121415                                                0.0     \n",
       "121416                                                0.0     \n",
       "121417                                                0.0     \n",
       "121418                                                0.0     \n",
       "\n",
       "        OVCAR8_OVARY_TenPx24  THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \n",
       "0                        0.0                                              0.0  \n",
       "1                        0.0                                              0.0  \n",
       "2                        0.0                                              0.0  \n",
       "3                        0.0                                              0.0  \n",
       "4                        0.0                                              0.0  \n",
       "...                      ...                                              ...  \n",
       "121414                   0.0                                              0.0  \n",
       "121415                   0.0                                              0.0  \n",
       "121416                   0.0                                              0.0  \n",
       "121417                   0.0                                              0.0  \n",
       "121418                   0.0                                              0.0  \n",
       "\n",
       "[121419 rows x 379 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# merge the biomedgraphica_id with the raw_protein_df\n",
    "final_merged_protein_df = pd.merge(bmg_protein_df, protein_df, left_on='Uniprot_ID', right_on='Uniprot_Acc', how='left')\n",
    "# fill the NaN with 0.0\n",
    "final_merged_protein_df = final_merged_protein_df.fillna(0.0)\n",
    "final_merged_protein_df.drop(columns=['Uniprot_ID', 'Uniprot_Acc'], inplace=True)\n",
    "display(final_merged_protein_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Drug"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DATASET</th>\n",
       "      <th>COSMIC_ID</th>\n",
       "      <th>DRUG_ID</th>\n",
       "      <th>MIN_CONC</th>\n",
       "      <th>MAX_CONC</th>\n",
       "      <th>RMSE_PUBLISHED</th>\n",
       "      <th>Z_SCORE_PUBLISHED</th>\n",
       "      <th>IC50_PUBLISHED</th>\n",
       "      <th>AUC_PUBLISHED</th>\n",
       "      <th>ARXSPAN_ID</th>\n",
       "      <th>DRUG_NAME</th>\n",
       "      <th>BROAD_ID</th>\n",
       "      <th>upper_limit</th>\n",
       "      <th>ec50</th>\n",
       "      <th>slope</th>\n",
       "      <th>lower_limit</th>\n",
       "      <th>auc</th>\n",
       "      <th>log2.ic50</th>\n",
       "      <th>mse</th>\n",
       "      <th>R2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GDSC1</td>\n",
       "      <td>683665</td>\n",
       "      <td>1</td>\n",
       "      <td>0.007813</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.022518</td>\n",
       "      <td>-0.192056</td>\n",
       "      <td>10.977393</td>\n",
       "      <td>0.982116</td>\n",
       "      <td>ACH-002270</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BRD-K70401845</td>\n",
       "      <td>0.992788</td>\n",
       "      <td>2.839376e+00</td>\n",
       "      <td>-5.670993</td>\n",
       "      <td>0.514389</td>\n",
       "      <td>0.990834</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0.904675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GDSC1</td>\n",
       "      <td>684055</td>\n",
       "      <td>1</td>\n",
       "      <td>0.007813</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.031831</td>\n",
       "      <td>0.505823</td>\n",
       "      <td>23.133991</td>\n",
       "      <td>0.984820</td>\n",
       "      <td>ACH-002104</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BRD-K70401845</td>\n",
       "      <td>1.006405</td>\n",
       "      <td>2.864875e-02</td>\n",
       "      <td>-0.186377</td>\n",
       "      <td>0.990054</td>\n",
       "      <td>0.997138</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000057</td>\n",
       "      <td>0.028903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GDSC1</td>\n",
       "      <td>684057</td>\n",
       "      <td>1</td>\n",
       "      <td>0.007813</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.026047</td>\n",
       "      <td>1.280750</td>\n",
       "      <td>52.935278</td>\n",
       "      <td>0.985696</td>\n",
       "      <td>ACH-002106</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BRD-K70401845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GDSC1</td>\n",
       "      <td>684059</td>\n",
       "      <td>1</td>\n",
       "      <td>0.007813</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.110056</td>\n",
       "      <td>0.086028</td>\n",
       "      <td>14.774223</td>\n",
       "      <td>0.972701</td>\n",
       "      <td>ACH-002108</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BRD-K70401845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDSC1</td>\n",
       "      <td>684062</td>\n",
       "      <td>1</td>\n",
       "      <td>0.007813</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.087010</td>\n",
       "      <td>-0.114395</td>\n",
       "      <td>11.926884</td>\n",
       "      <td>0.944463</td>\n",
       "      <td>ACH-002111</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BRD-K70401845</td>\n",
       "      <td>0.989580</td>\n",
       "      <td>7.580375e-02</td>\n",
       "      <td>-12.222777</td>\n",
       "      <td>0.894027</td>\n",
       "      <td>0.933185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000623</td>\n",
       "      <td>0.777093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387621</th>\n",
       "      <td>GDSC2</td>\n",
       "      <td>905951</td>\n",
       "      <td>2172</td>\n",
       "      <td>0.009766</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.143863</td>\n",
       "      <td>0.042524</td>\n",
       "      <td>25.410793</td>\n",
       "      <td>0.877741</td>\n",
       "      <td>ACH-000288</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BRD-K54606188, NA</td>\n",
       "      <td>3.929045</td>\n",
       "      <td>3.287745e+11</td>\n",
       "      <td>-0.012190</td>\n",
       "      <td>-3.449953</td>\n",
       "      <td>0.856099</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006178</td>\n",
       "      <td>0.292447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387622</th>\n",
       "      <td>GDSC2</td>\n",
       "      <td>906862</td>\n",
       "      <td>2172</td>\n",
       "      <td>0.009766</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.088631</td>\n",
       "      <td>-2.223819</td>\n",
       "      <td>0.339325</td>\n",
       "      <td>0.510590</td>\n",
       "      <td>ACH-001065</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BRD-K54606188, NA</td>\n",
       "      <td>0.935866</td>\n",
       "      <td>4.096968e-01</td>\n",
       "      <td>-2.014115</td>\n",
       "      <td>0.176515</td>\n",
       "      <td>0.585800</td>\n",
       "      <td>-1.073816</td>\n",
       "      <td>0.003198</td>\n",
       "      <td>0.971991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387623</th>\n",
       "      <td>GDSC2</td>\n",
       "      <td>907046</td>\n",
       "      <td>2172</td>\n",
       "      <td>0.009766</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.114748</td>\n",
       "      <td>-0.578942</td>\n",
       "      <td>7.780877</td>\n",
       "      <td>0.843211</td>\n",
       "      <td>ACH-000930</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BRD-K54606188, NA</td>\n",
       "      <td>0.960799</td>\n",
       "      <td>3.695946e-01</td>\n",
       "      <td>-1.161533</td>\n",
       "      <td>0.687792</td>\n",
       "      <td>0.830671</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002792</td>\n",
       "      <td>0.795935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387624</th>\n",
       "      <td>GDSC2</td>\n",
       "      <td>749709</td>\n",
       "      <td>2172</td>\n",
       "      <td>0.009766</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.047625</td>\n",
       "      <td>1.642265</td>\n",
       "      <td>534.688321</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>ACH-000859</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BRD-K54606188, NA</td>\n",
       "      <td>0.977260</td>\n",
       "      <td>1.161541e+30</td>\n",
       "      <td>-0.031889</td>\n",
       "      <td>0.900234</td>\n",
       "      <td>0.969870</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000879</td>\n",
       "      <td>0.000540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387625</th>\n",
       "      <td>GDSC2</td>\n",
       "      <td>906801</td>\n",
       "      <td>2172</td>\n",
       "      <td>0.009766</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.085767</td>\n",
       "      <td>0.858430</td>\n",
       "      <td>120.177282</td>\n",
       "      <td>0.952633</td>\n",
       "      <td>ACH-000536</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BRD-K54606188, NA</td>\n",
       "      <td>0.961719</td>\n",
       "      <td>7.840383e+01</td>\n",
       "      <td>-0.554315</td>\n",
       "      <td>0.664756</td>\n",
       "      <td>0.940827</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003325</td>\n",
       "      <td>0.159037</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>387626 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       DATASET  COSMIC_ID  DRUG_ID  MIN_CONC  MAX_CONC  RMSE_PUBLISHED  \\\n",
       "0        GDSC1     683665        1  0.007813       2.0        0.022518   \n",
       "1        GDSC1     684055        1  0.007813       2.0        0.031831   \n",
       "2        GDSC1     684057        1  0.007813       2.0        0.026047   \n",
       "3        GDSC1     684059        1  0.007813       2.0        0.110056   \n",
       "4        GDSC1     684062        1  0.007813       2.0        0.087010   \n",
       "...        ...        ...      ...       ...       ...             ...   \n",
       "387621   GDSC2     905951     2172  0.009766      10.0        0.143863   \n",
       "387622   GDSC2     906862     2172  0.009766      10.0        0.088631   \n",
       "387623   GDSC2     907046     2172  0.009766      10.0        0.114748   \n",
       "387624   GDSC2     749709     2172  0.009766      10.0        0.047625   \n",
       "387625   GDSC2     906801     2172  0.009766      10.0        0.085767   \n",
       "\n",
       "        Z_SCORE_PUBLISHED  IC50_PUBLISHED  AUC_PUBLISHED  ARXSPAN_ID  \\\n",
       "0               -0.192056       10.977393       0.982116  ACH-002270   \n",
       "1                0.505823       23.133991       0.984820  ACH-002104   \n",
       "2                1.280750       52.935278       0.985696  ACH-002106   \n",
       "3                0.086028       14.774223       0.972701  ACH-002108   \n",
       "4               -0.114395       11.926884       0.944463  ACH-002111   \n",
       "...                   ...             ...            ...         ...   \n",
       "387621           0.042524       25.410793       0.877741  ACH-000288   \n",
       "387622          -2.223819        0.339325       0.510590  ACH-001065   \n",
       "387623          -0.578942        7.780877       0.843211  ACH-000930   \n",
       "387624           1.642265      534.688321       0.983634  ACH-000859   \n",
       "387625           0.858430      120.177282       0.952633  ACH-000536   \n",
       "\n",
       "        DRUG_NAME           BROAD_ID  upper_limit          ec50      slope  \\\n",
       "0       ERLOTINIB      BRD-K70401845     0.992788  2.839376e+00  -5.670993   \n",
       "1       ERLOTINIB      BRD-K70401845     1.006405  2.864875e-02  -0.186377   \n",
       "2       ERLOTINIB      BRD-K70401845          NaN           NaN        NaN   \n",
       "3       ERLOTINIB      BRD-K70401845          NaN           NaN        NaN   \n",
       "4       ERLOTINIB      BRD-K70401845     0.989580  7.580375e-02 -12.222777   \n",
       "...           ...                ...          ...           ...        ...   \n",
       "387621        JQ1  BRD-K54606188, NA     3.929045  3.287745e+11  -0.012190   \n",
       "387622        JQ1  BRD-K54606188, NA     0.935866  4.096968e-01  -2.014115   \n",
       "387623        JQ1  BRD-K54606188, NA     0.960799  3.695946e-01  -1.161533   \n",
       "387624        JQ1  BRD-K54606188, NA     0.977260  1.161541e+30  -0.031889   \n",
       "387625        JQ1  BRD-K54606188, NA     0.961719  7.840383e+01  -0.554315   \n",
       "\n",
       "        lower_limit       auc  log2.ic50       mse        R2  \n",
       "0          0.514389  0.990834        NaN  0.000034  0.904675  \n",
       "1          0.990054  0.997138        NaN  0.000057  0.028903  \n",
       "2               NaN       NaN        NaN       NaN       NaN  \n",
       "3               NaN       NaN        NaN       NaN       NaN  \n",
       "4          0.894027  0.933185        NaN  0.000623  0.777093  \n",
       "...             ...       ...        ...       ...       ...  \n",
       "387621    -3.449953  0.856099        NaN  0.006178  0.292447  \n",
       "387622     0.176515  0.585800  -1.073816  0.003198  0.971991  \n",
       "387623     0.687792  0.830671        NaN  0.002792  0.795935  \n",
       "387624     0.900234  0.969870        NaN  0.000879  0.000540  \n",
       "387625     0.664756  0.940827        NaN  0.003325  0.159037  \n",
       "\n",
       "[387626 rows x 20 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "raw_drug_df = pd.read_csv('./data/raw_data/sanger-dose-response.csv')\n",
    "display(raw_drug_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ARXSPAN_ID        8621\n",
      "DRUG_NAME           74\n",
      "IC50_PUBLISHED       0\n",
      "AUC_PUBLISHED        0\n",
      "dtype: int64\n",
      "ARXSPAN_ID        0\n",
      "DRUG_NAME         0\n",
      "IC50_PUBLISHED    0\n",
      "AUC_PUBLISHED     0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ARXSPAN_ID</th>\n",
       "      <th>DRUG_NAME</th>\n",
       "      <th>IC50_PUBLISHED</th>\n",
       "      <th>AUC_PUBLISHED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-002270</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>10.977393</td>\n",
       "      <td>0.982116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-002104</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>23.133991</td>\n",
       "      <td>0.984820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-002106</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>52.935278</td>\n",
       "      <td>0.985696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-002108</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>14.774223</td>\n",
       "      <td>0.972701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-002111</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>11.926884</td>\n",
       "      <td>0.944463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379000</th>\n",
       "      <td>ACH-000288</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>25.410793</td>\n",
       "      <td>0.877741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379001</th>\n",
       "      <td>ACH-001065</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>0.339325</td>\n",
       "      <td>0.510590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379002</th>\n",
       "      <td>ACH-000930</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>7.780877</td>\n",
       "      <td>0.843211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379003</th>\n",
       "      <td>ACH-000859</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>534.688321</td>\n",
       "      <td>0.983634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379004</th>\n",
       "      <td>ACH-000536</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>120.177282</td>\n",
       "      <td>0.952633</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>379005 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ARXSPAN_ID  DRUG_NAME  IC50_PUBLISHED  AUC_PUBLISHED\n",
       "0       ACH-002270  ERLOTINIB       10.977393       0.982116\n",
       "1       ACH-002104  ERLOTINIB       23.133991       0.984820\n",
       "2       ACH-002106  ERLOTINIB       52.935278       0.985696\n",
       "3       ACH-002108  ERLOTINIB       14.774223       0.972701\n",
       "4       ACH-002111  ERLOTINIB       11.926884       0.944463\n",
       "...            ...        ...             ...            ...\n",
       "379000  ACH-000288        JQ1       25.410793       0.877741\n",
       "379001  ACH-001065        JQ1        0.339325       0.510590\n",
       "379002  ACH-000930        JQ1        7.780877       0.843211\n",
       "379003  ACH-000859        JQ1      534.688321       0.983634\n",
       "379004  ACH-000536        JQ1      120.177282       0.952633\n",
       "\n",
       "[379005 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# keep columns ['ARXSPAN_ID', 'DRUG_NAME', 'IC50_PUBLISHED']\n",
    "raw_drug_df = raw_drug_df[['ARXSPAN_ID', 'DRUG_NAME', 'IC50_PUBLISHED', 'AUC_PUBLISHED']]\n",
    "# check if there is nan in all of the dataframe\n",
    "print(raw_drug_df.isnull().sum())\n",
    "# drop the nan in this dataframe\n",
    "raw_drug_df = raw_drug_df.dropna().reset_index(drop=True)\n",
    "# check if there is nan in all of the dataframe again\n",
    "print(raw_drug_df.isnull().sum())\n",
    "display(raw_drug_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DRUG_NAME</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERLOTINIB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SIROLIMUS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SUNITINIB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PHA-665752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MG-132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>LJI308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>AZ6102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>GSK591, EPZ015866, GSK3203591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>VE-821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>AT13148</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>397 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         DRUG_NAME\n",
       "0                        ERLOTINIB\n",
       "1                        SIROLIMUS\n",
       "2                        SUNITINIB\n",
       "3                       PHA-665752\n",
       "4                           MG-132\n",
       "..                             ...\n",
       "392                         LJI308\n",
       "393                         AZ6102\n",
       "394  GSK591, EPZ015866, GSK3203591\n",
       "395                         VE-821\n",
       "396                        AT13148\n",
       "\n",
       "[397 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# fetch the drug_name as a independent dataframe\n",
    "drug_name_df = raw_drug_df[['DRUG_NAME']]\n",
    "# remove the duplicate in the drug_name_df\n",
    "drug_name_df = drug_name_df.drop_duplicates().reset_index(drop=True)\n",
    "display(drug_name_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>BioMedGraphica_ID</th>\n",
       "      <th>PubChem_CID</th>\n",
       "      <th>PubChem_SID</th>\n",
       "      <th>PubChem_Name</th>\n",
       "      <th>CAS_RN</th>\n",
       "      <th>IUPAC_Name</th>\n",
       "      <th>UNII</th>\n",
       "      <th>UNII_Name</th>\n",
       "      <th>NDC</th>\n",
       "      <th>DrugBank_ID</th>\n",
       "      <th>DrugBank_Name</th>\n",
       "      <th>PubChem_Canonical_SMILES</th>\n",
       "      <th>UNII_SMILES</th>\n",
       "      <th>InChI</th>\n",
       "      <th>InChIKEY</th>\n",
       "      <th>PubChem_Synonym</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_DG00001</td>\n",
       "      <td>BMG_DG000018</td>\n",
       "      <td>10001648.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12,13-Dihydroxyfumitremorgin C</td>\n",
       "      <td>111427-99-7</td>\n",
       "      <td>(1R,2S,12S,15S)-1,2-dihydroxy-7-methoxy-12-(2-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CC(=C[C@H]1C2=C([C@@H]([C@@]3(N1C(=O)[C@@H]4CC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>InChI=1S/C22H25N3O5/c1-11(2)9-16-18-17(13-7-6-...</td>\n",
       "      <td>CPHRCQUGNAGVIB-FWJAZRMLSA-N</td>\n",
       "      <td>12,13-dihydroxyfumitremorgin C|111427-99-7|12a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_DG00002</td>\n",
       "      <td>BMG_DG000025</td>\n",
       "      <td>100028.0</td>\n",
       "      <td>474493534</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3280-71-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TN5YFF4JQQ</td>\n",
       "      <td>2-AMINO-4-SEC-BUTYLPHENOL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C(CC)(C)c1cc(N)c(O)cc1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>XKJRYOAGNVYYIU-UHFFFAOYSA-N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_DG00003</td>\n",
       "      <td>BMG_DG000029</td>\n",
       "      <td>10002952.0</td>\n",
       "      <td>198966102</td>\n",
       "      <td>NaN</td>\n",
       "      <td>179044-94-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>U90K6D923K</td>\n",
       "      <td>AMG-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c12c([C@]3([C@](C(O1)(C)C)(CC=C(C3)C)[H])[H])c...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JECXXFXYJAQVAH-WOJBJXKFSA-N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_DG00004</td>\n",
       "      <td>BMG_DG000037</td>\n",
       "      <td>10004545.0</td>\n",
       "      <td>198937626</td>\n",
       "      <td>NaN</td>\n",
       "      <td>693823-79-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HX4T7JEV3D</td>\n",
       "      <td>AMG-076 FREE BASE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C1(CCCCC1)(C(=O)O)CCN2C[C@@]3([C@H](c4c5cc(ccc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WAOZNBXSEVIAGX-MBOZVWFJSA-N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_DG00005</td>\n",
       "      <td>BMG_DG000045</td>\n",
       "      <td>10006.0</td>\n",
       "      <td>198983861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>461-72-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>I6208298TA</td>\n",
       "      <td>HYDANTOIN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C1(NC(=O)NC1)=O</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WJRBRSLFGCUECM-UHFFFAOYSA-N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20913</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>BMG_DG269064</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DB19372</td>\n",
       "      <td>Zoxazolamine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20914</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>BMG_DG269070</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DB19378</td>\n",
       "      <td>Megestrol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20915</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>BMG_DG269073</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DB19381</td>\n",
       "      <td>AZACYCLONOL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20916</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>BMG_DG269082</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DB19397</td>\n",
       "      <td>AZD8630</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20917</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>BMG_DG269087</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DB19413</td>\n",
       "      <td>Influenza A Virus A/Thailand/8/2022 IVR-237 (H...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20918 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID BioMedGraphica_ID  PubChem_CID PubChem_SID  \\\n",
       "0               BMGC_DG00001      BMG_DG000018   10001648.0         NaN   \n",
       "1               BMGC_DG00002      BMG_DG000025     100028.0   474493534   \n",
       "2               BMGC_DG00003      BMG_DG000029   10002952.0   198966102   \n",
       "3               BMGC_DG00004      BMG_DG000037   10004545.0   198937626   \n",
       "4               BMGC_DG00005      BMG_DG000045      10006.0   198983861   \n",
       "...                      ...               ...          ...         ...   \n",
       "20913           BMGC_DG20914      BMG_DG269064          NaN         NaN   \n",
       "20914           BMGC_DG20915      BMG_DG269070          NaN         NaN   \n",
       "20915           BMGC_DG20916      BMG_DG269073          NaN         NaN   \n",
       "20916           BMGC_DG20917      BMG_DG269082          NaN         NaN   \n",
       "20917           BMGC_DG20918      BMG_DG269087          NaN         NaN   \n",
       "\n",
       "                         PubChem_Name       CAS_RN  \\\n",
       "0      12,13-Dihydroxyfumitremorgin C  111427-99-7   \n",
       "1                                 NaN    3280-71-5   \n",
       "2                                 NaN  179044-94-1   \n",
       "3                                 NaN  693823-79-9   \n",
       "4                                 NaN     461-72-3   \n",
       "...                               ...          ...   \n",
       "20913                             NaN          NaN   \n",
       "20914                             NaN          NaN   \n",
       "20915                             NaN          NaN   \n",
       "20916                             NaN          NaN   \n",
       "20917                             NaN          NaN   \n",
       "\n",
       "                                              IUPAC_Name        UNII  \\\n",
       "0      (1R,2S,12S,15S)-1,2-dihydroxy-7-methoxy-12-(2-...         NaN   \n",
       "1                                                    NaN  TN5YFF4JQQ   \n",
       "2                                                    NaN  U90K6D923K   \n",
       "3                                                    NaN  HX4T7JEV3D   \n",
       "4                                                    NaN  I6208298TA   \n",
       "...                                                  ...         ...   \n",
       "20913                                                NaN         NaN   \n",
       "20914                                                NaN         NaN   \n",
       "20915                                                NaN         NaN   \n",
       "20916                                                NaN         NaN   \n",
       "20917                                                NaN         NaN   \n",
       "\n",
       "                       UNII_Name  NDC DrugBank_ID  \\\n",
       "0                            NaN  NaN         NaN   \n",
       "1      2-AMINO-4-SEC-BUTYLPHENOL  NaN         NaN   \n",
       "2                          AMG-3  NaN         NaN   \n",
       "3              AMG-076 FREE BASE  NaN         NaN   \n",
       "4                      HYDANTOIN  NaN         NaN   \n",
       "...                          ...  ...         ...   \n",
       "20913                        NaN  NaN     DB19372   \n",
       "20914                        NaN  NaN     DB19378   \n",
       "20915                        NaN  NaN     DB19381   \n",
       "20916                        NaN  NaN     DB19397   \n",
       "20917                        NaN  NaN     DB19413   \n",
       "\n",
       "                                           DrugBank_Name  \\\n",
       "0                                                    NaN   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3                                                    NaN   \n",
       "4                                                    NaN   \n",
       "...                                                  ...   \n",
       "20913                                       Zoxazolamine   \n",
       "20914                                          Megestrol   \n",
       "20915                                        AZACYCLONOL   \n",
       "20916                                            AZD8630   \n",
       "20917  Influenza A Virus A/Thailand/8/2022 IVR-237 (H...   \n",
       "\n",
       "                                PubChem_Canonical_SMILES  \\\n",
       "0      CC(=C[C@H]1C2=C([C@@H]([C@@]3(N1C(=O)[C@@H]4CC...   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3                                                    NaN   \n",
       "4                                                    NaN   \n",
       "...                                                  ...   \n",
       "20913                                                NaN   \n",
       "20914                                                NaN   \n",
       "20915                                                NaN   \n",
       "20916                                                NaN   \n",
       "20917                                                NaN   \n",
       "\n",
       "                                             UNII_SMILES  \\\n",
       "0                                                    NaN   \n",
       "1                                 C(CC)(C)c1cc(N)c(O)cc1   \n",
       "2      c12c([C@]3([C@](C(O1)(C)C)(CC=C(C3)C)[H])[H])c...   \n",
       "3      C1(CCCCC1)(C(=O)O)CCN2C[C@@]3([C@H](c4c5cc(ccc...   \n",
       "4                                        C1(NC(=O)NC1)=O   \n",
       "...                                                  ...   \n",
       "20913                                                NaN   \n",
       "20914                                                NaN   \n",
       "20915                                                NaN   \n",
       "20916                                                NaN   \n",
       "20917                                                NaN   \n",
       "\n",
       "                                                   InChI  \\\n",
       "0      InChI=1S/C22H25N3O5/c1-11(2)9-16-18-17(13-7-6-...   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3                                                    NaN   \n",
       "4                                                    NaN   \n",
       "...                                                  ...   \n",
       "20913                                                NaN   \n",
       "20914                                                NaN   \n",
       "20915                                                NaN   \n",
       "20916                                                NaN   \n",
       "20917                                                NaN   \n",
       "\n",
       "                          InChIKEY  \\\n",
       "0      CPHRCQUGNAGVIB-FWJAZRMLSA-N   \n",
       "1      XKJRYOAGNVYYIU-UHFFFAOYSA-N   \n",
       "2      JECXXFXYJAQVAH-WOJBJXKFSA-N   \n",
       "3      WAOZNBXSEVIAGX-MBOZVWFJSA-N   \n",
       "4      WJRBRSLFGCUECM-UHFFFAOYSA-N   \n",
       "...                            ...   \n",
       "20913                          NaN   \n",
       "20914                          NaN   \n",
       "20915                          NaN   \n",
       "20916                          NaN   \n",
       "20917                          NaN   \n",
       "\n",
       "                                         PubChem_Synonym  \n",
       "0      12,13-dihydroxyfumitremorgin C|111427-99-7|12a...  \n",
       "1                                                    NaN  \n",
       "2                                                    NaN  \n",
       "3                                                    NaN  \n",
       "4                                                    NaN  \n",
       "...                                                  ...  \n",
       "20913                                                NaN  \n",
       "20914                                                NaN  \n",
       "20915                                                NaN  \n",
       "20916                                                NaN  \n",
       "20917                                                NaN  \n",
       "\n",
       "[20918 rows x 17 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>PubChem_Name</th>\n",
       "      <th>IUPAC_Name</th>\n",
       "      <th>UNII_Name</th>\n",
       "      <th>DrugBank_Name</th>\n",
       "      <th>PubChem_Synonym</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_DG00001</td>\n",
       "      <td>12,13-Dihydroxyfumitremorgin C</td>\n",
       "      <td>(1R,2S,12S,15S)-1,2-dihydroxy-7-methoxy-12-(2-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12,13-dihydroxyfumitremorgin C|111427-99-7|12a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_DG00002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2-AMINO-4-SEC-BUTYLPHENOL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_DG00003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMG-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_DG00004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMG-076 FREE BASE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_DG00005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HYDANTOIN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20913</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Zoxazolamine</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20914</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Megestrol</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20915</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AZACYCLONOL</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20916</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AZD8630</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20917</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Influenza A Virus A/Thailand/8/2022 IVR-237 (H...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20918 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID                    PubChem_Name  \\\n",
       "0               BMGC_DG00001  12,13-Dihydroxyfumitremorgin C   \n",
       "1               BMGC_DG00002                             NaN   \n",
       "2               BMGC_DG00003                             NaN   \n",
       "3               BMGC_DG00004                             NaN   \n",
       "4               BMGC_DG00005                             NaN   \n",
       "...                      ...                             ...   \n",
       "20913           BMGC_DG20914                             NaN   \n",
       "20914           BMGC_DG20915                             NaN   \n",
       "20915           BMGC_DG20916                             NaN   \n",
       "20916           BMGC_DG20917                             NaN   \n",
       "20917           BMGC_DG20918                             NaN   \n",
       "\n",
       "                                              IUPAC_Name  \\\n",
       "0      (1R,2S,12S,15S)-1,2-dihydroxy-7-methoxy-12-(2-...   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3                                                    NaN   \n",
       "4                                                    NaN   \n",
       "...                                                  ...   \n",
       "20913                                                NaN   \n",
       "20914                                                NaN   \n",
       "20915                                                NaN   \n",
       "20916                                                NaN   \n",
       "20917                                                NaN   \n",
       "\n",
       "                       UNII_Name  \\\n",
       "0                            NaN   \n",
       "1      2-AMINO-4-SEC-BUTYLPHENOL   \n",
       "2                          AMG-3   \n",
       "3              AMG-076 FREE BASE   \n",
       "4                      HYDANTOIN   \n",
       "...                          ...   \n",
       "20913                        NaN   \n",
       "20914                        NaN   \n",
       "20915                        NaN   \n",
       "20916                        NaN   \n",
       "20917                        NaN   \n",
       "\n",
       "                                           DrugBank_Name  \\\n",
       "0                                                    NaN   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3                                                    NaN   \n",
       "4                                                    NaN   \n",
       "...                                                  ...   \n",
       "20913                                       Zoxazolamine   \n",
       "20914                                          Megestrol   \n",
       "20915                                        AZACYCLONOL   \n",
       "20916                                            AZD8630   \n",
       "20917  Influenza A Virus A/Thailand/8/2022 IVR-237 (H...   \n",
       "\n",
       "                                         PubChem_Synonym  \n",
       "0      12,13-dihydroxyfumitremorgin C|111427-99-7|12a...  \n",
       "1                                                    NaN  \n",
       "2                                                    NaN  \n",
       "3                                                    NaN  \n",
       "4                                                    NaN  \n",
       "...                                                  ...  \n",
       "20913                                                NaN  \n",
       "20914                                                NaN  \n",
       "20915                                                NaN  \n",
       "20916                                                NaN  \n",
       "20917                                                NaN  \n",
       "\n",
       "[20918 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bmg_drug_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Drug/BioMedGraphica_Conn_Drug.csv')\n",
    "display(bmg_drug_df)\n",
    "bmg_drug_name_df = bmg_drug_df[['BioMedGraphica_Conn_ID', 'PubChem_Name', 'IUPAC_Name', 'UNII_Name', 'DrugBank_Name', 'PubChem_Synonym']]\n",
    "# keep BioMedGraphica_Conn_ID and Drug_Name\n",
    "display(bmg_drug_name_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\185313899.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  bmg_drug_name_df['Drug_Name'] = bmg_drug_name_df[['PubChem_Name', 'IUPAC_Name', 'UNII_Name', 'DrugBank_Name', 'PubChem_Synonym']].apply(lambda x: ';'.join(x.dropna().astype(str)), axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Drug_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_DG00001</td>\n",
       "      <td>12,13-Dihydroxyfumitremorgin C;(1R,2S,12S,15S)...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_DG00002</td>\n",
       "      <td>2-AMINO-4-SEC-BUTYLPHENOL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_DG00003</td>\n",
       "      <td>AMG-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_DG00004</td>\n",
       "      <td>AMG-076 FREE BASE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_DG00005</td>\n",
       "      <td>HYDANTOIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20913</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>Zoxazolamine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20914</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>Megestrol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20915</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>AZACYCLONOL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20916</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>AZD8630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20917</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>Influenza A Virus A/Thailand/8/2022 IVR-237 (H...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20918 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  \\\n",
       "0               BMGC_DG00001   \n",
       "1               BMGC_DG00002   \n",
       "2               BMGC_DG00003   \n",
       "3               BMGC_DG00004   \n",
       "4               BMGC_DG00005   \n",
       "...                      ...   \n",
       "20913           BMGC_DG20914   \n",
       "20914           BMGC_DG20915   \n",
       "20915           BMGC_DG20916   \n",
       "20916           BMGC_DG20917   \n",
       "20917           BMGC_DG20918   \n",
       "\n",
       "                                               Drug_Name  \n",
       "0      12,13-Dihydroxyfumitremorgin C;(1R,2S,12S,15S)...  \n",
       "1                              2-AMINO-4-SEC-BUTYLPHENOL  \n",
       "2                                                  AMG-3  \n",
       "3                                      AMG-076 FREE BASE  \n",
       "4                                              HYDANTOIN  \n",
       "...                                                  ...  \n",
       "20913                                       Zoxazolamine  \n",
       "20914                                          Megestrol  \n",
       "20915                                        AZACYCLONOL  \n",
       "20916                                            AZD8630  \n",
       "20917  Influenza A Virus A/Thailand/8/2022 IVR-237 (H...  \n",
       "\n",
       "[20918 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Combine 'PubChem_Name', 'IUPAC_Name', 'UNII_Name', 'DrugBank_Name', 'PubChem_Synonym' into one column called 'Drug_Name'\n",
    "bmg_drug_name_df['Drug_Name'] = bmg_drug_name_df[['PubChem_Name', 'IUPAC_Name', 'UNII_Name', 'DrugBank_Name', 'PubChem_Synonym']].apply(lambda x: ';'.join(x.dropna().astype(str)), axis=1)\n",
    "bmg_drug_name_df = bmg_drug_name_df[['BioMedGraphica_Conn_ID', 'Drug_Name']]\n",
    "display(bmg_drug_name_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of NaN values in Drug_Name column: 0\n"
     ]
    }
   ],
   "source": [
    "# since in the column drug name, there are lots of names split by \";\", if this space contain the name in the drug_name_df, then we should match this 2 rows and merge the drug_name_df with the bmg_drug_df\n",
    "# Fix the drug name processing\n",
    "# First, check if there are any NaN values in Drug_Name column\n",
    "print(f\"Number of NaN values in Drug_Name column: {bmg_drug_name_df['Drug_Name'].isna().sum()}\")\n",
    "\n",
    "# Fill NaN values with empty string to avoid errors\n",
    "bmg_drug_name_df['Drug_Name'] = bmg_drug_name_df['Drug_Name'].fillna('')\n",
    "\n",
    "# Convert to string to ensure split() works on all entries\n",
    "bmg_drug_name_df['drug_name_list'] = bmg_drug_name_df['Drug_Name'].astype(str).apply(lambda x: [name.strip() for name in x.split(';')] if x else [])\n",
    "\n",
    "# Similarly for drug_name_df\n",
    "drug_name_df['drug_name_list'] = drug_name_df['DRUG_NAME'].astype(str).apply(lambda x: [name.strip() for name in x.split(';')] if x else [])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Drug_Name</th>\n",
       "      <th>drug_name_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_DG00001</td>\n",
       "      <td>12,13-Dihydroxyfumitremorgin C;(1R,2S,12S,15S)...</td>\n",
       "      <td>[12,13-Dihydroxyfumitremorgin C, (1R,2S,12S,15...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_DG00002</td>\n",
       "      <td>2-AMINO-4-SEC-BUTYLPHENOL</td>\n",
       "      <td>[2-AMINO-4-SEC-BUTYLPHENOL]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_DG00003</td>\n",
       "      <td>AMG-3</td>\n",
       "      <td>[AMG-3]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_DG00004</td>\n",
       "      <td>AMG-076 FREE BASE</td>\n",
       "      <td>[AMG-076 FREE BASE]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_DG00005</td>\n",
       "      <td>HYDANTOIN</td>\n",
       "      <td>[HYDANTOIN]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20913</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>Zoxazolamine</td>\n",
       "      <td>[Zoxazolamine]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20914</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>Megestrol</td>\n",
       "      <td>[Megestrol]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20915</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>AZACYCLONOL</td>\n",
       "      <td>[AZACYCLONOL]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20916</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>AZD8630</td>\n",
       "      <td>[AZD8630]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20917</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>Influenza A Virus A/Thailand/8/2022 IVR-237 (H...</td>\n",
       "      <td>[Influenza A Virus A/Thailand/8/2022 IVR-237 (...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20918 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  \\\n",
       "0               BMGC_DG00001   \n",
       "1               BMGC_DG00002   \n",
       "2               BMGC_DG00003   \n",
       "3               BMGC_DG00004   \n",
       "4               BMGC_DG00005   \n",
       "...                      ...   \n",
       "20913           BMGC_DG20914   \n",
       "20914           BMGC_DG20915   \n",
       "20915           BMGC_DG20916   \n",
       "20916           BMGC_DG20917   \n",
       "20917           BMGC_DG20918   \n",
       "\n",
       "                                               Drug_Name  \\\n",
       "0      12,13-Dihydroxyfumitremorgin C;(1R,2S,12S,15S)...   \n",
       "1                              2-AMINO-4-SEC-BUTYLPHENOL   \n",
       "2                                                  AMG-3   \n",
       "3                                      AMG-076 FREE BASE   \n",
       "4                                              HYDANTOIN   \n",
       "...                                                  ...   \n",
       "20913                                       Zoxazolamine   \n",
       "20914                                          Megestrol   \n",
       "20915                                        AZACYCLONOL   \n",
       "20916                                            AZD8630   \n",
       "20917  Influenza A Virus A/Thailand/8/2022 IVR-237 (H...   \n",
       "\n",
       "                                          drug_name_list  \n",
       "0      [12,13-Dihydroxyfumitremorgin C, (1R,2S,12S,15...  \n",
       "1                            [2-AMINO-4-SEC-BUTYLPHENOL]  \n",
       "2                                                [AMG-3]  \n",
       "3                                    [AMG-076 FREE BASE]  \n",
       "4                                            [HYDANTOIN]  \n",
       "...                                                  ...  \n",
       "20913                                     [Zoxazolamine]  \n",
       "20914                                        [Megestrol]  \n",
       "20915                                      [AZACYCLONOL]  \n",
       "20916                                          [AZD8630]  \n",
       "20917  [Influenza A Virus A/Thailand/8/2022 IVR-237 (...  \n",
       "\n",
       "[20918 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DRUG_NAME</th>\n",
       "      <th>drug_name_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>[ERLOTINIB]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SIROLIMUS</td>\n",
       "      <td>[SIROLIMUS]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SUNITINIB</td>\n",
       "      <td>[SUNITINIB]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PHA-665752</td>\n",
       "      <td>[PHA-665752]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MG-132</td>\n",
       "      <td>[MG-132]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>LJI308</td>\n",
       "      <td>[LJI308]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>AZ6102</td>\n",
       "      <td>[AZ6102]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>GSK591, EPZ015866, GSK3203591</td>\n",
       "      <td>[GSK591, EPZ015866, GSK3203591]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>VE-821</td>\n",
       "      <td>[VE-821]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>AT13148</td>\n",
       "      <td>[AT13148]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>397 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         DRUG_NAME                   drug_name_list\n",
       "0                        ERLOTINIB                      [ERLOTINIB]\n",
       "1                        SIROLIMUS                      [SIROLIMUS]\n",
       "2                        SUNITINIB                      [SUNITINIB]\n",
       "3                       PHA-665752                     [PHA-665752]\n",
       "4                           MG-132                         [MG-132]\n",
       "..                             ...                              ...\n",
       "392                         LJI308                         [LJI308]\n",
       "393                         AZ6102                         [AZ6102]\n",
       "394  GSK591, EPZ015866, GSK3203591  [GSK591, EPZ015866, GSK3203591]\n",
       "395                         VE-821                         [VE-821]\n",
       "396                        AT13148                        [AT13148]\n",
       "\n",
       "[397 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(bmg_drug_name_df)\n",
    "display(drug_name_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Successfully matched 172 out of 397 drugs\n",
      "\n",
      "Sample of drug name mappings:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DRUG_NAME</th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SIROLIMUS</td>\n",
       "      <td>BMGC_DG11078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SUNITINIB</td>\n",
       "      <td>BMGC_DG11707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PHA-665752</td>\n",
       "      <td>BMGC_DG00627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MG-132</td>\n",
       "      <td>BMGC_DG09927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>OSIMERTINIB</td>\n",
       "      <td>BMGC_DG15052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>CEDIRANIB</td>\n",
       "      <td>BMGC_DG17637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>GDC-0810</td>\n",
       "      <td>BMGC_DG12527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>SAVOLITINIB</td>\n",
       "      <td>BMGC_DG14332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>VE-821</td>\n",
       "      <td>BMGC_DG10537</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>172 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       DRUG_NAME BioMedGraphica_Conn_ID\n",
       "0      ERLOTINIB           BMGC_DG04654\n",
       "1      SIROLIMUS           BMGC_DG11078\n",
       "2      SUNITINIB           BMGC_DG11707\n",
       "3     PHA-665752           BMGC_DG00627\n",
       "4         MG-132           BMGC_DG09927\n",
       "..           ...                    ...\n",
       "167  OSIMERTINIB           BMGC_DG15052\n",
       "168    CEDIRANIB           BMGC_DG17637\n",
       "169     GDC-0810           BMGC_DG12527\n",
       "170  SAVOLITINIB           BMGC_DG14332\n",
       "171       VE-821           BMGC_DG10537\n",
       "\n",
       "[172 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Create a flattened mapping of drug names to their BMG IDs\n",
    "drug_name_to_bmg = {}\n",
    "for idx, row in bmg_drug_name_df.iterrows():\n",
    "    bmg_id = row['BioMedGraphica_Conn_ID']\n",
    "    for drug_name in row['drug_name_list']:\n",
    "        if drug_name:  # Avoid empty strings\n",
    "            drug_name_to_bmg[drug_name.upper()] = bmg_id\n",
    "\n",
    "# Create a new mapping from ARXSPAN_ID to BMG_ID based on drug name matches\n",
    "arxspan_to_bmg = {}\n",
    "for idx, row in drug_name_df.iterrows():\n",
    "    arxspan_id = row['DRUG_NAME']\n",
    "    for drug_name in row['drug_name_list']:\n",
    "        if drug_name.upper() in drug_name_to_bmg:\n",
    "            arxspan_to_bmg[arxspan_id] = drug_name_to_bmg[drug_name.upper()]\n",
    "            break\n",
    "\n",
    "# Create a mapping dataframe\n",
    "mapping_df = pd.DataFrame(list(arxspan_to_bmg.items()), columns=['DRUG_NAME', 'BioMedGraphica_Conn_ID'])\n",
    "\n",
    "# Display how many drug names were successfully matched\n",
    "print(f\"Successfully matched {len(mapping_df)} out of {len(drug_name_df)} drugs\")\n",
    "\n",
    "# Display the first few rows of the mapping\n",
    "print(\"\\nSample of drug name mappings:\")\n",
    "display(mapping_df)\n",
    "\n",
    "# Now you can use this mapping to merge with your drug_response_df\n",
    "# Example:\n",
    "# merged_df = pd.merge(drug_response_df, mapping_df, on='DRUG_NAME', how='left')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DRUG_NAME</th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AR-12</td>\n",
       "      <td>BMGC_DG00030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PAZOPANIB</td>\n",
       "      <td>BMGC_DG00141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SELUMETINIB</td>\n",
       "      <td>BMGC_DG00151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AFATINIB</td>\n",
       "      <td>BMGC_DG00254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SARACATINIB</td>\n",
       "      <td>BMGC_DG00420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>ABT-737</td>\n",
       "      <td>BMGC_DG20690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>DACINOSTAT</td>\n",
       "      <td>BMGC_DG20696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>PI-103</td>\n",
       "      <td>BMGC_DG20703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>SNX-2112</td>\n",
       "      <td>BMGC_DG20709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>TW-37</td>\n",
       "      <td>BMGC_DG20711</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>172 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       DRUG_NAME BioMedGraphica_Conn_ID\n",
       "0          AR-12           BMGC_DG00030\n",
       "1      PAZOPANIB           BMGC_DG00141\n",
       "2    SELUMETINIB           BMGC_DG00151\n",
       "3       AFATINIB           BMGC_DG00254\n",
       "4    SARACATINIB           BMGC_DG00420\n",
       "..           ...                    ...\n",
       "167      ABT-737           BMGC_DG20690\n",
       "168   DACINOSTAT           BMGC_DG20696\n",
       "169       PI-103           BMGC_DG20703\n",
       "170     SNX-2112           BMGC_DG20709\n",
       "171        TW-37           BMGC_DG20711\n",
       "\n",
       "[172 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "final_merged_drug_df = mapping_df.copy()\n",
    "# sort values by the column BioMedGraphica_ID\n",
    "final_merged_drug_df = final_merged_drug_df.sort_values(by='BioMedGraphica_Conn_ID').reset_index(drop=True)\n",
    "display(final_merged_drug_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ARXSPAN_ID</th>\n",
       "      <th>DRUG_NAME</th>\n",
       "      <th>IC50_PUBLISHED</th>\n",
       "      <th>AUC_PUBLISHED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-002270</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>10.977393</td>\n",
       "      <td>0.982116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-002104</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>23.133991</td>\n",
       "      <td>0.984820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-002106</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>52.935278</td>\n",
       "      <td>0.985696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-002108</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>14.774223</td>\n",
       "      <td>0.972701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-002111</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>11.926884</td>\n",
       "      <td>0.944463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175318</th>\n",
       "      <td>ACH-000288</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>25.410793</td>\n",
       "      <td>0.877741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175319</th>\n",
       "      <td>ACH-001065</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>0.339325</td>\n",
       "      <td>0.510590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175320</th>\n",
       "      <td>ACH-000930</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>7.780877</td>\n",
       "      <td>0.843211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175321</th>\n",
       "      <td>ACH-000859</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>534.688321</td>\n",
       "      <td>0.983634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175322</th>\n",
       "      <td>ACH-000536</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>120.177282</td>\n",
       "      <td>0.952633</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>175323 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ARXSPAN_ID  DRUG_NAME  IC50_PUBLISHED  AUC_PUBLISHED\n",
       "0       ACH-002270  ERLOTINIB       10.977393       0.982116\n",
       "1       ACH-002104  ERLOTINIB       23.133991       0.984820\n",
       "2       ACH-002106  ERLOTINIB       52.935278       0.985696\n",
       "3       ACH-002108  ERLOTINIB       14.774223       0.972701\n",
       "4       ACH-002111  ERLOTINIB       11.926884       0.944463\n",
       "...            ...        ...             ...            ...\n",
       "175318  ACH-000288        JQ1       25.410793       0.877741\n",
       "175319  ACH-001065        JQ1        0.339325       0.510590\n",
       "175320  ACH-000930        JQ1        7.780877       0.843211\n",
       "175321  ACH-000859        JQ1      534.688321       0.983634\n",
       "175322  ACH-000536        JQ1      120.177282       0.952633\n",
       "\n",
       "[175323 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# filter the final drug score dataframe by filtering the 'DRUG_NAME' in the final_merged_drug_df DRUG_NAME\n",
    "final_drug_df = raw_drug_df[raw_drug_df['DRUG_NAME'].isin(final_merged_drug_df['DRUG_NAME'])].reset_index(drop=True)\n",
    "display(final_drug_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. CRISPR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BioMedGraphica_Conn_ID    0\n",
      "BioMedGraphica_ID         0\n",
      "Type                      0\n",
      "dtype: int64\n",
      "BioMedGraphica_Conn_ID    0\n",
      "BioMedGraphica_ID         0\n",
      "Type                      0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>BioMedGraphica_ID</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMG_PM000001</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMG_PM000002</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>BMG_PM000003</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>BMG_PM000004</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>BMG_PM000005</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705929</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>BMG_PT173974</td>\n",
       "      <td>Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705930</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>BMG_PT173975</td>\n",
       "      <td>Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705931</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>BMG_PT173976</td>\n",
       "      <td>Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705932</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>BMG_PT173977</td>\n",
       "      <td>Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705933</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>Protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>705934 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID BioMedGraphica_ID      Type\n",
       "0                BMGC_PM00001      BMG_PM000001  Promoter\n",
       "1                BMGC_PM00002      BMG_PM000002  Promoter\n",
       "2                BMGC_PM00003      BMG_PM000003  Promoter\n",
       "3                BMGC_PM00004      BMG_PM000004  Promoter\n",
       "4                BMGC_PM00005      BMG_PM000005  Promoter\n",
       "...                       ...               ...       ...\n",
       "705929          BMGC_PT121415      BMG_PT173974   Protein\n",
       "705930          BMGC_PT121416      BMG_PT173975   Protein\n",
       "705931          BMGC_PT121417      BMG_PT173976   Protein\n",
       "705932          BMGC_PT121418      BMG_PT173977   Protein\n",
       "705933          BMGC_PT121419      BMG_PT173978   Protein\n",
       "\n",
       "[705934 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# load entity type ['Promoter', 'Gene', 'Transcript', 'Protein'] for bmgc_entity_df\n",
    "bmgc_entity_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/BioMedGraphica_Conn_Entity.csv')\n",
    "# check if there is any null values in the bmgc_entity_df\n",
    "print(bmgc_entity_df.isnull().sum())\n",
    "# filter out the rows with Type in the list ['promoter', 'gene', 'transcript', 'protein']\n",
    "bmgc_omics_df = bmgc_entity_df[bmgc_entity_df['Type'].isin(['Promoter', 'Gene', 'Transcript', 'Protein'])].reset_index(drop=True)\n",
    "# check if there is any null values in the bmgc_omics_df\n",
    "print(bmgc_omics_df.isnull().sum())\n",
    "display(bmgc_omics_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\1745844678.py:2: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmgc_relation_df = pd.read_csv('./data/BioMedGraphica-Conn/Relation/BioMedGraphica_Conn_Relation.csv')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BioMedGraphica_Conn_ID        0\n",
      "BioMedGraphica_ID             0\n",
      "BMGC_From_ID                  0\n",
      "BMGC_To_ID                    0\n",
      "BMG_From_ID                   0\n",
      "BMG_To_ID                     0\n",
      "Source                    86238\n",
      "Type                          0\n",
      "dtype: int64\n",
      "BioMedGraphica_Conn_ID        0\n",
      "BioMedGraphica_ID             0\n",
      "BMGC_From_ID                  0\n",
      "BMGC_To_ID                    0\n",
      "BMG_From_ID                   0\n",
      "BMG_To_ID                     0\n",
      "Source                    86238\n",
      "Type                          0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>BioMedGraphica_ID</th>\n",
       "      <th>BMGC_From_ID</th>\n",
       "      <th>BMGC_To_ID</th>\n",
       "      <th>BMG_From_ID</th>\n",
       "      <th>BMG_To_ID</th>\n",
       "      <th>Source</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_ED_PMGN00001</td>\n",
       "      <td>BMG_ED_PMGN00001</td>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMG_PM000001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_ED_PMGN00002</td>\n",
       "      <td>BMG_ED_PMGN00002</td>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMG_PM000002</td>\n",
       "      <td>BMG_GN000002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_ED_PMGN00003</td>\n",
       "      <td>BMG_ED_PMGN00003</td>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>BMG_PM000003</td>\n",
       "      <td>BMG_GN000003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_ED_PMGN00004</td>\n",
       "      <td>BMG_ED_PMGN00004</td>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>BMG_PM000004</td>\n",
       "      <td>BMG_GN000004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_ED_PMGN00005</td>\n",
       "      <td>BMG_ED_PMGN00005</td>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>BMG_PM000005</td>\n",
       "      <td>BMG_GN000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151448</th>\n",
       "      <td>BMGC_ED_PTPT16484816</td>\n",
       "      <td>BMG_ED_PTPT16484816</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT117260</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT169819</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151449</th>\n",
       "      <td>BMGC_ED_PTPT16484817</td>\n",
       "      <td>BMG_ED_PTPT16484817</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120142</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT172701</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151450</th>\n",
       "      <td>BMGC_ED_PTPT16484818</td>\n",
       "      <td>BMG_ED_PTPT16484818</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120825</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT173384</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151451</th>\n",
       "      <td>BMGC_ED_PTPT16484819</td>\n",
       "      <td>BMG_ED_PTPT16484819</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120826</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT173385</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151452</th>\n",
       "      <td>BMGC_ED_PTPT16484820</td>\n",
       "      <td>BMG_ED_PTPT16484820</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120827</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT173386</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17151453 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         BioMedGraphica_Conn_ID    BioMedGraphica_ID   BMGC_From_ID  \\\n",
       "0             BMGC_ED_PMGN00001     BMG_ED_PMGN00001   BMGC_PM00001   \n",
       "1             BMGC_ED_PMGN00002     BMG_ED_PMGN00002   BMGC_PM00002   \n",
       "2             BMGC_ED_PMGN00003     BMG_ED_PMGN00003   BMGC_PM00003   \n",
       "3             BMGC_ED_PMGN00004     BMG_ED_PMGN00004   BMGC_PM00004   \n",
       "4             BMGC_ED_PMGN00005     BMG_ED_PMGN00005   BMGC_PM00005   \n",
       "...                         ...                  ...            ...   \n",
       "17151448   BMGC_ED_PTPT16484816  BMG_ED_PTPT16484816  BMGC_PT121419   \n",
       "17151449   BMGC_ED_PTPT16484817  BMG_ED_PTPT16484817  BMGC_PT121419   \n",
       "17151450   BMGC_ED_PTPT16484818  BMG_ED_PTPT16484818  BMGC_PT121419   \n",
       "17151451   BMGC_ED_PTPT16484819  BMG_ED_PTPT16484819  BMGC_PT121419   \n",
       "17151452   BMGC_ED_PTPT16484820  BMG_ED_PTPT16484820  BMGC_PT121419   \n",
       "\n",
       "             BMGC_To_ID   BMG_From_ID     BMG_To_ID Source             Type  \n",
       "0          BMGC_GN00001  BMG_PM000001  BMG_GN000001    NaN    Promoter-Gene  \n",
       "1          BMGC_GN00002  BMG_PM000002  BMG_GN000002    NaN    Promoter-Gene  \n",
       "2          BMGC_GN00003  BMG_PM000003  BMG_GN000003    NaN    Promoter-Gene  \n",
       "3          BMGC_GN00004  BMG_PM000004  BMG_GN000004    NaN    Promoter-Gene  \n",
       "4          BMGC_GN00005  BMG_PM000005  BMG_GN000005    NaN    Promoter-Gene  \n",
       "...                 ...           ...           ...    ...              ...  \n",
       "17151448  BMGC_PT117260  BMG_PT173978  BMG_PT169819   KEGG  Protein-Protein  \n",
       "17151449  BMGC_PT120142  BMG_PT173978  BMG_PT172701   KEGG  Protein-Protein  \n",
       "17151450  BMGC_PT120825  BMG_PT173978  BMG_PT173384   KEGG  Protein-Protein  \n",
       "17151451  BMGC_PT120826  BMG_PT173978  BMG_PT173385   KEGG  Protein-Protein  \n",
       "17151452  BMGC_PT120827  BMG_PT173978  BMG_PT173386   KEGG  Protein-Protein  \n",
       "\n",
       "[17151453 rows x 8 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# load the relation\n",
    "bmgc_relation_df = pd.read_csv('./data/BioMedGraphica-Conn/Relation/BioMedGraphica_Conn_Relation.csv')\n",
    "# check if there is any null values in the bmgc_relation_df\n",
    "print(bmgc_relation_df.isnull().sum())\n",
    "\n",
    "# filter our the rows with Relation type in the list ['Promoter-Gene', 'Gene-Transcript', 'Transcript-Protein', 'Protein-Protein']\n",
    "bmgc_omics_relation_df = bmgc_relation_df[bmgc_relation_df['Type'].isin(['Promoter-Gene', 'Gene-Transcript', 'Transcript-Protein', 'Protein-Protein'])].reset_index(drop=True)\n",
    "# check if there is any null values in the bmgc_omics_relation_df\n",
    "print(bmgc_omics_relation_df.isnull().sum())\n",
    "display(bmgc_omics_relation_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>BioMedGraphica_ID</th>\n",
       "      <th>BMGC_From_ID</th>\n",
       "      <th>BMGC_To_ID</th>\n",
       "      <th>BMG_From_ID</th>\n",
       "      <th>BMG_To_ID</th>\n",
       "      <th>Source</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_ED_PMGN00001</td>\n",
       "      <td>BMG_ED_PMGN00001</td>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMG_PM000001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_ED_PMGN00002</td>\n",
       "      <td>BMG_ED_PMGN00002</td>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMG_PM000002</td>\n",
       "      <td>BMG_GN000002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_ED_PMGN00003</td>\n",
       "      <td>BMG_ED_PMGN00003</td>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>BMG_PM000003</td>\n",
       "      <td>BMG_GN000003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_ED_PMGN00004</td>\n",
       "      <td>BMG_ED_PMGN00004</td>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>BMG_PM000004</td>\n",
       "      <td>BMG_GN000004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_ED_PMGN00005</td>\n",
       "      <td>BMG_ED_PMGN00005</td>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>BMG_PM000005</td>\n",
       "      <td>BMG_GN000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151448</th>\n",
       "      <td>BMGC_ED_PTPT16484816</td>\n",
       "      <td>BMG_ED_PTPT16484816</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT117260</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT169819</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151449</th>\n",
       "      <td>BMGC_ED_PTPT16484817</td>\n",
       "      <td>BMG_ED_PTPT16484817</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120142</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT172701</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151450</th>\n",
       "      <td>BMGC_ED_PTPT16484818</td>\n",
       "      <td>BMG_ED_PTPT16484818</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120825</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT173384</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151451</th>\n",
       "      <td>BMGC_ED_PTPT16484819</td>\n",
       "      <td>BMG_ED_PTPT16484819</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120826</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT173385</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151452</th>\n",
       "      <td>BMGC_ED_PTPT16484820</td>\n",
       "      <td>BMG_ED_PTPT16484820</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120827</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT173386</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17151453 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         BioMedGraphica_Conn_ID    BioMedGraphica_ID   BMGC_From_ID  \\\n",
       "0             BMGC_ED_PMGN00001     BMG_ED_PMGN00001   BMGC_PM00001   \n",
       "1             BMGC_ED_PMGN00002     BMG_ED_PMGN00002   BMGC_PM00002   \n",
       "2             BMGC_ED_PMGN00003     BMG_ED_PMGN00003   BMGC_PM00003   \n",
       "3             BMGC_ED_PMGN00004     BMG_ED_PMGN00004   BMGC_PM00004   \n",
       "4             BMGC_ED_PMGN00005     BMG_ED_PMGN00005   BMGC_PM00005   \n",
       "...                         ...                  ...            ...   \n",
       "17151448   BMGC_ED_PTPT16484816  BMG_ED_PTPT16484816  BMGC_PT121419   \n",
       "17151449   BMGC_ED_PTPT16484817  BMG_ED_PTPT16484817  BMGC_PT121419   \n",
       "17151450   BMGC_ED_PTPT16484818  BMG_ED_PTPT16484818  BMGC_PT121419   \n",
       "17151451   BMGC_ED_PTPT16484819  BMG_ED_PTPT16484819  BMGC_PT121419   \n",
       "17151452   BMGC_ED_PTPT16484820  BMG_ED_PTPT16484820  BMGC_PT121419   \n",
       "\n",
       "             BMGC_To_ID   BMG_From_ID     BMG_To_ID Source             Type  \n",
       "0          BMGC_GN00001  BMG_PM000001  BMG_GN000001    NaN    Promoter-Gene  \n",
       "1          BMGC_GN00002  BMG_PM000002  BMG_GN000002    NaN    Promoter-Gene  \n",
       "2          BMGC_GN00003  BMG_PM000003  BMG_GN000003    NaN    Promoter-Gene  \n",
       "3          BMGC_GN00004  BMG_PM000004  BMG_GN000004    NaN    Promoter-Gene  \n",
       "4          BMGC_GN00005  BMG_PM000005  BMG_GN000005    NaN    Promoter-Gene  \n",
       "...                 ...           ...           ...    ...              ...  \n",
       "17151448  BMGC_PT117260  BMG_PT173978  BMG_PT169819   KEGG  Protein-Protein  \n",
       "17151449  BMGC_PT120142  BMG_PT173978  BMG_PT172701   KEGG  Protein-Protein  \n",
       "17151450  BMGC_PT120825  BMG_PT173978  BMG_PT173384   KEGG  Protein-Protein  \n",
       "17151451  BMGC_PT120826  BMG_PT173978  BMG_PT173385   KEGG  Protein-Protein  \n",
       "17151452  BMGC_PT120827  BMG_PT173978  BMG_PT173386   KEGG  Protein-Protein  \n",
       "\n",
       "[17151453 rows x 8 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Null values in bmgc_omics_relation_df:\n",
      "BioMedGraphica_Conn_ID        0\n",
      "BioMedGraphica_ID             0\n",
      "BMGC_From_ID                  0\n",
      "BMGC_To_ID                    0\n",
      "BMG_From_ID                   0\n",
      "BMG_To_ID                     0\n",
      "Source                    86238\n",
      "Type                          0\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# translation chain converging to the same node\n",
    "# fetch the promoter, gene, transcript and protein entity alone\n",
    "promoter_entity_df = bmgc_omics_df[bmgc_omics_df['Type'] == 'Promoter'].copy()\n",
    "gene_entity_df = bmgc_omics_df[bmgc_omics_df['Type'] == 'Gene'].copy()\n",
    "transcript_entity_df = bmgc_omics_df[bmgc_omics_df['Type'] == 'Transcript'].copy()\n",
    "protein_entity_df = bmgc_omics_df[bmgc_omics_df['Type'] == 'Protein'].copy()\n",
    "\n",
    "display(bmgc_omics_relation_df)\n",
    "# recheck the null values in bmgc_omics_relation_df\n",
    "print(\"Null values in bmgc_omics_relation_df:\")\n",
    "print(bmgc_omics_relation_df.isnull().sum())\n",
    "\n",
    "# fetch the Promoter-Gene, Gene-Transcript, Transcript-Protein relation alone\n",
    "promoter_gene_relation_df = bmgc_omics_relation_df[bmgc_omics_relation_df['Type'] == 'Promoter-Gene'].copy()\n",
    "gene_transcript_relation_df = bmgc_omics_relation_df[bmgc_omics_relation_df['Type'] == 'Gene-Transcript'].copy()\n",
    "transcript_protein_relation_df = bmgc_omics_relation_df[bmgc_omics_relation_df['Type'] == 'Transcript-Protein'].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_GN_ID</th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_PT013542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_PT091702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT070222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT041299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT038889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126009</th>\n",
       "      <td>BMGC_GN70323</td>\n",
       "      <td>BMGC_PT121177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126010</th>\n",
       "      <td>BMGC_GN70327</td>\n",
       "      <td>BMGC_PT121280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126011</th>\n",
       "      <td>BMGC_GN70329</td>\n",
       "      <td>BMGC_PT121282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126012</th>\n",
       "      <td>BMGC_GN76774</td>\n",
       "      <td>BMGC_PT007612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126013</th>\n",
       "      <td>BMGC_GN86213</td>\n",
       "      <td>BMGC_PT082937</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>126014 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BMGC_GN_ID     BMGC_PT_ID\n",
       "0       BMGC_GN00001  BMGC_PT013542\n",
       "1       BMGC_GN00001  BMGC_PT091702\n",
       "2       BMGC_GN00002  BMGC_PT070222\n",
       "3       BMGC_GN00002  BMGC_PT041299\n",
       "4       BMGC_GN00002  BMGC_PT038889\n",
       "...              ...            ...\n",
       "126009  BMGC_GN70323  BMGC_PT121177\n",
       "126010  BMGC_GN70327  BMGC_PT121280\n",
       "126011  BMGC_GN70329  BMGC_PT121282\n",
       "126012  BMGC_GN76774  BMGC_PT007612\n",
       "126013  BMGC_GN86213  BMGC_PT082937\n",
       "\n",
       "[126014 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_transcript_entity_df = pd.merge(gene_entity_df, gene_transcript_relation_df[['BMGC_From_ID', 'BMGC_To_ID']], left_on='BioMedGraphica_Conn_ID', right_on='BMGC_From_ID', how='outer')\n",
    "gene_transcript_protein_entity_df = pd.merge(gene_transcript_entity_df, transcript_protein_relation_df[['BMGC_From_ID', 'BMGC_To_ID']], left_on='BMGC_To_ID', right_on='BMGC_From_ID', how='outer')\n",
    "# drop NaN values in BMGC_From_ID_x\tBMGC_To_ID_x BMGC_From_ID_y\tBMGC_To_ID_y\n",
    "gene_transcript_protein_entity_df = gene_transcript_protein_entity_df.dropna(subset=['BMGC_From_ID_x', 'BMGC_To_ID_x', 'BMGC_From_ID_y', 'BMGC_To_ID_y']).reset_index(drop=True)\n",
    "# keep the columns ['BioMedGraphica_Conn_ID', 'BMGC_To_ID_x', 'BMGC_To_ID_y'] and rename the columns to ['BMGC_GN_ID', 'BMGC_TS_ID', 'BMGC_PT_ID']\n",
    "gene_transcript_protein_entity_df = gene_transcript_protein_entity_df[['BioMedGraphica_Conn_ID', 'BMGC_To_ID_y']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_GN_ID', 'BMGC_To_ID_y': 'BMGC_PT_ID'}).sort_values(by='BMGC_GN_ID').reset_index(drop=True)\n",
    "# drop duplicates rows in gene_transcript_protein_entity_df\n",
    "gene_transcript_protein_entity_df = gene_transcript_protein_entity_df.drop_duplicates().reset_index(drop=True)\n",
    "display(gene_transcript_protein_entity_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\390326110.py:2: DtypeWarning: Columns (4,5,6,11,12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmgc_promoter_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Promoter/BioMedGraphica_Conn_Promoter.csv')\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\390326110.py:3: DtypeWarning: Columns (4,5,6,11,12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmgc_gene_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_PM_ID</th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_PT013542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_PT091702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMGC_PT070222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMGC_PT041299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMGC_PT038889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126009</th>\n",
       "      <td>BMGC_PM70323</td>\n",
       "      <td>BMGC_PT121177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126010</th>\n",
       "      <td>BMGC_PM70327</td>\n",
       "      <td>BMGC_PT121280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126011</th>\n",
       "      <td>BMGC_PM70329</td>\n",
       "      <td>BMGC_PT121282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126012</th>\n",
       "      <td>BMGC_PM76774</td>\n",
       "      <td>BMGC_PT007612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126013</th>\n",
       "      <td>BMGC_PM86213</td>\n",
       "      <td>BMGC_PT082937</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>126014 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BMGC_PM_ID     BMGC_PT_ID\n",
       "0       BMGC_PM00001  BMGC_PT013542\n",
       "1       BMGC_PM00001  BMGC_PT091702\n",
       "2       BMGC_PM00002  BMGC_PT070222\n",
       "3       BMGC_PM00002  BMGC_PT041299\n",
       "4       BMGC_PM00002  BMGC_PT038889\n",
       "...              ...            ...\n",
       "126009  BMGC_PM70323  BMGC_PT121177\n",
       "126010  BMGC_PM70327  BMGC_PT121280\n",
       "126011  BMGC_PM70329  BMGC_PT121282\n",
       "126012  BMGC_PM76774  BMGC_PT007612\n",
       "126013  BMGC_PM86213  BMGC_PT082937\n",
       "\n",
       "[126014 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# just copy gene_transcript_protein_entity_df as promoter_gene_transcript_protein_entity_df\n",
    "bmgc_promoter_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Promoter/BioMedGraphica_Conn_Promoter.csv')\n",
    "bmgc_gene_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene.csv')\n",
    "promoter_gene_transcript_protein_entity_df = gene_transcript_protein_entity_df.copy()\n",
    "promoter_gene_df = pd.concat([bmgc_promoter_df[['BioMedGraphica_Conn_ID']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_PM_ID'}), bmgc_gene_df[['BioMedGraphica_Conn_ID']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_GN_ID'})], axis=1)\n",
    "promoter_protein_entity_df = pd.merge(promoter_gene_transcript_protein_entity_df, promoter_gene_df, left_on='BMGC_GN_ID', right_on='BMGC_GN_ID', how='left').drop(columns=['BMGC_GN_ID'])\n",
    "promoter_protein_entity_df = promoter_protein_entity_df[['BMGC_PM_ID', 'BMGC_PT_ID']]\n",
    "display(promoter_protein_entity_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_TS_ID</th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>BMGC_PT017571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>BMGC_PT014978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>BMGC_PT014403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>BMGC_PT017807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>BMGC_PT043262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152580</th>\n",
       "      <td>BMGC_TS411885</td>\n",
       "      <td>BMGC_PT121415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152581</th>\n",
       "      <td>BMGC_TS411886</td>\n",
       "      <td>BMGC_PT121416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152582</th>\n",
       "      <td>BMGC_TS411887</td>\n",
       "      <td>BMGC_PT121417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152583</th>\n",
       "      <td>BMGC_TS412018</td>\n",
       "      <td>BMGC_PT121418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152584</th>\n",
       "      <td>BMGC_TS412019</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>152585 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           BMGC_TS_ID     BMGC_PT_ID\n",
       "0       BMGC_TS000001  BMGC_PT017571\n",
       "1       BMGC_TS000002  BMGC_PT014978\n",
       "2       BMGC_TS000003  BMGC_PT014403\n",
       "3       BMGC_TS000004  BMGC_PT017807\n",
       "4       BMGC_TS000005  BMGC_PT043262\n",
       "...               ...            ...\n",
       "152580  BMGC_TS411885  BMGC_PT121415\n",
       "152581  BMGC_TS411886  BMGC_PT121416\n",
       "152582  BMGC_TS411887  BMGC_PT121417\n",
       "152583  BMGC_TS412018  BMGC_PT121418\n",
       "152584  BMGC_TS412019  BMGC_PT121419\n",
       "\n",
       "[152585 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "transcript_protein_entity_df = pd.merge(transcript_entity_df, transcript_protein_relation_df[['BMGC_From_ID', 'BMGC_To_ID']], left_on='BioMedGraphica_Conn_ID', right_on='BMGC_From_ID', how='outer')\n",
    "# drop NaN rows in the BMGC_From_ID and\tBMGC_To_ID\n",
    "transcript_protein_entity_df = transcript_protein_entity_df.dropna(subset=['BMGC_From_ID', 'BMGC_To_ID']).reset_index(drop=True)\n",
    "# keep the columns ['BioMedGraphica_Conn_ID', 'BMGC_To_ID'] and rename the columns to ['BMGC_TS_ID', 'BMGC_PT_ID']\n",
    "transcript_protein_entity_df = transcript_protein_entity_df[['BioMedGraphica_Conn_ID', 'BMGC_To_ID']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_TS_ID', 'BMGC_To_ID': 'BMGC_PT_ID'}).sort_values(by='BMGC_TS_ID').reset_index(drop=True)\n",
    "# drop duplicates rows in transcript_protein_entity_df\n",
    "transcript_protein_entity_df = transcript_protein_entity_df.drop_duplicates().reset_index(drop=True)\n",
    "display(transcript_protein_entity_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           BMGC_PT_ID\n",
       "0       BMGC_PT000001\n",
       "1       BMGC_PT000002\n",
       "2       BMGC_PT000003\n",
       "3       BMGC_PT000004\n",
       "4       BMGC_PT000005\n",
       "...               ...\n",
       "121414  BMGC_PT121415\n",
       "121415  BMGC_PT121416\n",
       "121416  BMGC_PT121417\n",
       "121417  BMGC_PT121418\n",
       "121418  BMGC_PT121419\n",
       "\n",
       "[121419 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# keep the columns ['BioMedGraphica_Conn_ID'] and rename the columns to ['BMGC_PT_ID']\n",
    "only_protein_entity_df = protein_entity_df[['BioMedGraphica_Conn_ID']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_PT_ID'}).sort_values(by='BMGC_PT_ID').reset_index(drop=True)\n",
    "display(only_protein_entity_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 6.1 CRISPR top 100 gene entities"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>A1BG (1)</th>\n",
       "      <th>A1CF (29974)</th>\n",
       "      <th>A2M (2)</th>\n",
       "      <th>A2ML1 (144568)</th>\n",
       "      <th>A3GALT2 (127550)</th>\n",
       "      <th>A4GALT (53947)</th>\n",
       "      <th>A4GNT (51146)</th>\n",
       "      <th>AAAS (8086)</th>\n",
       "      <th>AACS (65985)</th>\n",
       "      <th>...</th>\n",
       "      <th>ZWILCH (55055)</th>\n",
       "      <th>ZWINT (11130)</th>\n",
       "      <th>ZXDA (7789)</th>\n",
       "      <th>ZXDB (158586)</th>\n",
       "      <th>ZXDC (79364)</th>\n",
       "      <th>ZYG11A (440590)</th>\n",
       "      <th>ZYG11B (79699)</th>\n",
       "      <th>ZYX (7791)</th>\n",
       "      <th>ZZEF1 (23140)</th>\n",
       "      <th>ZZZ3 (26009)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>0.042647</td>\n",
       "      <td>0.026102</td>\n",
       "      <td>-0.147614</td>\n",
       "      <td>-0.020425</td>\n",
       "      <td>0.184014</td>\n",
       "      <td>-0.076020</td>\n",
       "      <td>-0.097068</td>\n",
       "      <td>0.182759</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.120536</td>\n",
       "      <td>0.107286</td>\n",
       "      <td>0.166390</td>\n",
       "      <td>0.262275</td>\n",
       "      <td>0.163876</td>\n",
       "      <td>-0.249508</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>0.067986</td>\n",
       "      <td>0.136285</td>\n",
       "      <td>-0.438536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.088500</td>\n",
       "      <td>-0.088198</td>\n",
       "      <td>-0.041225</td>\n",
       "      <td>-0.028074</td>\n",
       "      <td>-0.157206</td>\n",
       "      <td>0.280101</td>\n",
       "      <td>-0.106904</td>\n",
       "      <td>0.178125</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.192527</td>\n",
       "      <td>-0.324059</td>\n",
       "      <td>0.230377</td>\n",
       "      <td>0.087609</td>\n",
       "      <td>0.074897</td>\n",
       "      <td>0.054335</td>\n",
       "      <td>-0.330343</td>\n",
       "      <td>0.099067</td>\n",
       "      <td>0.274566</td>\n",
       "      <td>0.001871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000005</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>0.003300</td>\n",
       "      <td>0.160881</td>\n",
       "      <td>0.086224</td>\n",
       "      <td>-0.149315</td>\n",
       "      <td>-0.253837</td>\n",
       "      <td>0.167011</td>\n",
       "      <td>-0.101209</td>\n",
       "      <td>-0.129827</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.312827</td>\n",
       "      <td>-0.338023</td>\n",
       "      <td>-0.039700</td>\n",
       "      <td>-0.055349</td>\n",
       "      <td>-0.000367</td>\n",
       "      <td>-0.205605</td>\n",
       "      <td>-0.066032</td>\n",
       "      <td>-0.054518</td>\n",
       "      <td>0.035579</td>\n",
       "      <td>-0.150486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000007</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.113717</td>\n",
       "      <td>0.082872</td>\n",
       "      <td>0.099633</td>\n",
       "      <td>-0.008378</td>\n",
       "      <td>-0.022310</td>\n",
       "      <td>0.014416</td>\n",
       "      <td>-0.184977</td>\n",
       "      <td>-0.173739</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.334843</td>\n",
       "      <td>-0.355499</td>\n",
       "      <td>-0.014183</td>\n",
       "      <td>0.230944</td>\n",
       "      <td>0.044628</td>\n",
       "      <td>-0.081863</td>\n",
       "      <td>-0.390748</td>\n",
       "      <td>-0.036547</td>\n",
       "      <td>-0.273129</td>\n",
       "      <td>-0.382723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000009</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>-0.134669</td>\n",
       "      <td>0.060323</td>\n",
       "      <td>0.076647</td>\n",
       "      <td>0.078922</td>\n",
       "      <td>-0.100243</td>\n",
       "      <td>0.047559</td>\n",
       "      <td>-0.136988</td>\n",
       "      <td>0.037759</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.299593</td>\n",
       "      <td>-0.194427</td>\n",
       "      <td>-0.027365</td>\n",
       "      <td>0.236591</td>\n",
       "      <td>-0.084224</td>\n",
       "      <td>-0.098271</td>\n",
       "      <td>-0.510495</td>\n",
       "      <td>0.052938</td>\n",
       "      <td>0.018623</td>\n",
       "      <td>-0.258353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1173</th>\n",
       "      <td>ACH-002926</td>\n",
       "      <td>-0.107883</td>\n",
       "      <td>-0.101501</td>\n",
       "      <td>0.046677</td>\n",
       "      <td>0.096868</td>\n",
       "      <td>-0.118650</td>\n",
       "      <td>-0.065258</td>\n",
       "      <td>-0.064944</td>\n",
       "      <td>-0.175912</td>\n",
       "      <td>-0.044707</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.409541</td>\n",
       "      <td>-0.579376</td>\n",
       "      <td>0.038706</td>\n",
       "      <td>0.338610</td>\n",
       "      <td>-0.174379</td>\n",
       "      <td>-0.040264</td>\n",
       "      <td>-0.239763</td>\n",
       "      <td>0.065168</td>\n",
       "      <td>-0.144088</td>\n",
       "      <td>-0.334424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1174</th>\n",
       "      <td>ACH-002928</td>\n",
       "      <td>-0.287606</td>\n",
       "      <td>-0.179101</td>\n",
       "      <td>-0.135824</td>\n",
       "      <td>-0.046579</td>\n",
       "      <td>-0.249888</td>\n",
       "      <td>-0.097376</td>\n",
       "      <td>0.070385</td>\n",
       "      <td>0.049444</td>\n",
       "      <td>-0.244937</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.095328</td>\n",
       "      <td>-0.440452</td>\n",
       "      <td>0.303500</td>\n",
       "      <td>0.097690</td>\n",
       "      <td>0.151790</td>\n",
       "      <td>0.039994</td>\n",
       "      <td>0.098240</td>\n",
       "      <td>-0.266654</td>\n",
       "      <td>-0.115108</td>\n",
       "      <td>-0.080421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1175</th>\n",
       "      <td>ACH-003012</td>\n",
       "      <td>-0.145878</td>\n",
       "      <td>-0.004000</td>\n",
       "      <td>0.091933</td>\n",
       "      <td>0.191332</td>\n",
       "      <td>-0.065385</td>\n",
       "      <td>-0.134152</td>\n",
       "      <td>0.012353</td>\n",
       "      <td>-0.321012</td>\n",
       "      <td>0.001571</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.280119</td>\n",
       "      <td>-0.662673</td>\n",
       "      <td>-0.180936</td>\n",
       "      <td>0.169997</td>\n",
       "      <td>-0.036326</td>\n",
       "      <td>0.158702</td>\n",
       "      <td>-0.213281</td>\n",
       "      <td>-0.017884</td>\n",
       "      <td>-0.195395</td>\n",
       "      <td>-0.309021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>ACH-003177</td>\n",
       "      <td>-0.038435</td>\n",
       "      <td>-0.055660</td>\n",
       "      <td>0.018989</td>\n",
       "      <td>0.117182</td>\n",
       "      <td>0.063397</td>\n",
       "      <td>0.049993</td>\n",
       "      <td>0.175520</td>\n",
       "      <td>-0.085752</td>\n",
       "      <td>0.078435</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.080700</td>\n",
       "      <td>-0.476797</td>\n",
       "      <td>-0.158375</td>\n",
       "      <td>-0.008999</td>\n",
       "      <td>-0.030058</td>\n",
       "      <td>-0.103513</td>\n",
       "      <td>-0.119234</td>\n",
       "      <td>-0.153744</td>\n",
       "      <td>0.032584</td>\n",
       "      <td>-0.536071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1177</th>\n",
       "      <td>ACH-003180</td>\n",
       "      <td>-0.007314</td>\n",
       "      <td>-0.212478</td>\n",
       "      <td>0.129900</td>\n",
       "      <td>-0.023112</td>\n",
       "      <td>-0.068613</td>\n",
       "      <td>0.067406</td>\n",
       "      <td>0.003678</td>\n",
       "      <td>-0.312161</td>\n",
       "      <td>0.009587</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.038622</td>\n",
       "      <td>-0.667881</td>\n",
       "      <td>-0.090500</td>\n",
       "      <td>0.077756</td>\n",
       "      <td>0.055249</td>\n",
       "      <td>0.040616</td>\n",
       "      <td>-0.378505</td>\n",
       "      <td>-0.143920</td>\n",
       "      <td>-0.039985</td>\n",
       "      <td>-0.439905</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1178 rows × 17917 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0  A1BG (1)  A1CF (29974)   A2M (2)  A2ML1 (144568)  \\\n",
       "0     ACH-000001 -0.121964      0.042647  0.026102       -0.147614   \n",
       "1     ACH-000004 -0.016484     -0.088500 -0.088198       -0.041225   \n",
       "2     ACH-000005 -0.184847      0.003300  0.160881        0.086224   \n",
       "3     ACH-000007 -0.071921     -0.113717  0.082872        0.099633   \n",
       "4     ACH-000009 -0.019163     -0.134669  0.060323        0.076647   \n",
       "...          ...       ...           ...       ...             ...   \n",
       "1173  ACH-002926 -0.107883     -0.101501  0.046677        0.096868   \n",
       "1174  ACH-002928 -0.287606     -0.179101 -0.135824       -0.046579   \n",
       "1175  ACH-003012 -0.145878     -0.004000  0.091933        0.191332   \n",
       "1176  ACH-003177 -0.038435     -0.055660  0.018989        0.117182   \n",
       "1177  ACH-003180 -0.007314     -0.212478  0.129900       -0.023112   \n",
       "\n",
       "      A3GALT2 (127550)  A4GALT (53947)  A4GNT (51146)  AAAS (8086)  \\\n",
       "0            -0.020425        0.184014      -0.076020    -0.097068   \n",
       "1            -0.028074       -0.157206       0.280101    -0.106904   \n",
       "2            -0.149315       -0.253837       0.167011    -0.101209   \n",
       "3            -0.008378       -0.022310       0.014416    -0.184977   \n",
       "4             0.078922       -0.100243       0.047559    -0.136988   \n",
       "...                ...             ...            ...          ...   \n",
       "1173         -0.118650       -0.065258      -0.064944    -0.175912   \n",
       "1174         -0.249888       -0.097376       0.070385     0.049444   \n",
       "1175         -0.065385       -0.134152       0.012353    -0.321012   \n",
       "1176          0.063397        0.049993       0.175520    -0.085752   \n",
       "1177         -0.068613        0.067406       0.003678    -0.312161   \n",
       "\n",
       "      AACS (65985)  ...  ZWILCH (55055)  ZWINT (11130)  ZXDA (7789)  \\\n",
       "0         0.182759  ...       -0.120536       0.107286     0.166390   \n",
       "1         0.178125  ...       -0.192527      -0.324059     0.230377   \n",
       "2        -0.129827  ...       -0.312827      -0.338023    -0.039700   \n",
       "3        -0.173739  ...       -0.334843      -0.355499    -0.014183   \n",
       "4         0.037759  ...       -0.299593      -0.194427    -0.027365   \n",
       "...            ...  ...             ...            ...          ...   \n",
       "1173     -0.044707  ...       -0.409541      -0.579376     0.038706   \n",
       "1174     -0.244937  ...       -0.095328      -0.440452     0.303500   \n",
       "1175      0.001571  ...       -0.280119      -0.662673    -0.180936   \n",
       "1176      0.078435  ...       -0.080700      -0.476797    -0.158375   \n",
       "1177      0.009587  ...       -0.038622      -0.667881    -0.090500   \n",
       "\n",
       "      ZXDB (158586)  ZXDC (79364)  ZYG11A (440590)  ZYG11B (79699)  \\\n",
       "0          0.262275      0.163876        -0.249508        0.053725   \n",
       "1          0.087609      0.074897         0.054335       -0.330343   \n",
       "2         -0.055349     -0.000367        -0.205605       -0.066032   \n",
       "3          0.230944      0.044628        -0.081863       -0.390748   \n",
       "4          0.236591     -0.084224        -0.098271       -0.510495   \n",
       "...             ...           ...              ...             ...   \n",
       "1173       0.338610     -0.174379        -0.040264       -0.239763   \n",
       "1174       0.097690      0.151790         0.039994        0.098240   \n",
       "1175       0.169997     -0.036326         0.158702       -0.213281   \n",
       "1176      -0.008999     -0.030058        -0.103513       -0.119234   \n",
       "1177       0.077756      0.055249         0.040616       -0.378505   \n",
       "\n",
       "      ZYX (7791)  ZZEF1 (23140)  ZZZ3 (26009)  \n",
       "0       0.067986       0.136285     -0.438536  \n",
       "1       0.099067       0.274566      0.001871  \n",
       "2      -0.054518       0.035579     -0.150486  \n",
       "3      -0.036547      -0.273129     -0.382723  \n",
       "4       0.052938       0.018623     -0.258353  \n",
       "...          ...            ...           ...  \n",
       "1173    0.065168      -0.144088     -0.334424  \n",
       "1174   -0.266654      -0.115108     -0.080421  \n",
       "1175   -0.017884      -0.195395     -0.309021  \n",
       "1176   -0.153744       0.032584     -0.536071  \n",
       "1177   -0.143920      -0.039985     -0.439905  \n",
       "\n",
       "[1178 rows x 17917 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "raw_crispr_df = pd.read_csv('./data/raw_data/CRISPRGeneEffect.csv')\n",
    "display(raw_crispr_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['A1BG (1)', 'A1CF (29974)', 'A2M (2)', 'A2ML1 (144568)', 'A3GALT2 (127550)', 'A4GALT (53947)', 'A4GNT (51146)', 'AAAS (8086)', 'AACS (65985)', 'AADAC (13)', 'AADACL2 (344752)', 'AADACL3 (126767)', 'AADACL4 (343066)', 'AADAT (51166)', 'AAGAB (79719)', 'AAK1 (22848)', 'AAMDC (28971)', 'AAMP (14)', 'AANAT (15)', 'AAR2 (25980)', 'AARD (441376)', 'AARS1 (16)', 'AARS2 (57505)', 'AARSD1 (80755)', 'AASDH (132949)', 'AASDHPPT (60496)', 'AASS (10157)', 'AATF (26574)', 'AATK (9625)', 'ABAT (18)', 'ABCA1 (19)', 'ABCA10 (10349)', 'ABCA12 (26154)', 'ABCA13 (154664)', 'ABCA2 (20)', 'ABCA3 (21)', 'ABCA4 (24)', 'ABCA5 (23461)', 'ABCA6 (23460)', 'ABCA7 (10347)', 'ABCA8 (10351)', 'ABCA9 (10350)', 'ABCB1 (5243)', 'ABCB10 (23456)', 'ABCB11 (8647)', 'ABCB4 (5244)', 'ABCB5 (340273)', 'ABCB6 (10058)', 'ABCB7 (22)', 'ABCB8 (11194)', 'ABCB9 (23457)', 'ABCC1 (4363)', 'ABCC10 (89845)', 'ABCC11 (85320)', 'ABCC12 (94160)', 'ABCC2 (1244)', 'ABCC3 (8714)', 'ABCC4 (10257)', 'ABCC5 (10057)', 'ABCC6 (368)', 'ABCC8 (6833)', 'ABCC9 (10060)', 'ABCD1 (215)', 'ABCD2 (225)', 'ABCD3 (5825)', 'ABCD4 (5826)', 'ABCE1 (6059)', 'ABCF1 (23)', 'ABCF3 (55324)', 'ABCG1 (9619)', 'ABCG2 (9429)', 'ABCG4 (64137)', 'ABCG5 (64240)', 'ABCG8 (64241)', 'ABHD1 (84696)', 'ABHD10 (55347)', 'ABHD11 (83451)', 'ABHD12 (26090)', 'ABHD12B (145447)', 'ABHD13 (84945)', 'ABHD14A (25864)', 'ABHD14B (84836)', 'ABHD15 (116236)', 'ABHD16A (7920)', 'ABHD16B (140701)', 'ABHD17A (81926)', 'ABHD17B (51104)', 'ABHD17C (58489)', 'ABHD18 (80167)', 'ABHD2 (11057)', 'ABHD3 (171586)', 'ABHD4 (63874)', 'ABHD5 (51099)', 'ABHD6 (57406)', 'ABHD8 (79575)', 'ABI1 (10006)', 'ABI2 (10152)', 'ABI3 (51225)', 'ABI3BP (25890)', 'ABITRAM (54942)', 'ABL1 (25)', 'ABL2 (27)', 'ABLIM1 (3983)', 'ABLIM2 (84448)', 'ABLIM3 (22885)', 'ABR (29)', 'ABRA (137735)', 'ABRACL (58527)', 'ABRAXAS1 (84142)', 'ABRAXAS2 (23172)', 'ABT1 (29777)', 'ABTB1 (80325)', 'ABTB2 (25841)', 'ABTB3 (121551)', 'ACAA1 (30)', 'ACAA2 (10449)', 'ACACA (31)', 'ACACB (32)', 'ACAD10 (80724)', 'ACAD11 (84129)', 'ACAD8 (27034)', 'ACAD9 (28976)', 'ACADL (33)', 'ACADM (34)', 'ACADS (35)', 'ACADSB (36)', 'ACADVL (37)', 'ACAN (176)', 'ACAP1 (9744)', 'ACAP2 (23527)', 'ACAP3 (116983)', 'ACAT1 (38)', 'ACAT2 (39)', 'ACBD3 (64746)', 'ACBD4 (79777)', 'ACBD5 (91452)', 'ACBD6 (84320)', 'ACBD7 (414149)', 'ACCS (84680)', 'ACCSL (390110)', 'ACD (65057)', 'ACE (1636)', 'ACE2 (59272)', 'ACER1 (125981)', 'ACER2 (340485)', 'ACER3 (55331)', 'ACHE (43)', 'ACIN1 (22985)', 'ACKR1 (2532)', 'ACKR2 (1238)', 'ACKR3 (57007)', 'ACKR4 (51554)', 'ACLY (47)', 'ACMSD (130013)', 'ACO1 (48)', 'ACO2 (50)', 'ACOD1 (730249)', 'ACOT11 (26027)', 'ACOT12 (134526)', 'ACOT13 (55856)', 'ACOT2 (10965)', 'ACOT4 (122970)', 'ACOT6 (641372)', 'ACOT7 (11332)', 'ACOT8 (10005)', 'ACOT9 (23597)', 'ACOX1 (51)', 'ACOX2 (8309)', 'ACOX3 (8310)', 'ACOXL (55289)', 'ACP1 (52)', 'ACP2 (53)', 'ACP3 (55)', 'ACP4 (93650)', 'ACP5 (54)', 'ACP6 (51205)', 'ACP7 (390928)', 'ACR (49)', 'ACRBP (84519)', 'ACRV1 (56)', 'ACSBG1 (23205)', 'ACSBG2 (81616)', 'ACSF2 (80221)', 'ACSF3 (197322)', 'ACSL1 (2180)', 'ACSL3 (2181)', 'ACSL4 (2182)', 'ACSL5 (51703)', 'ACSL6 (23305)', 'ACSM1 (116285)', 'ACSM2A (123876)', 'ACSM2B (348158)', 'ACSM3 (6296)', 'ACSM4 (341392)', 'ACSM5 (54988)', 'ACSM6 (142827)', 'ACSS1 (84532)', 'ACSS2 (55902)', 'ACSS3 (79611)', 'ACTA1 (58)', 'ACTA2 (59)', 'ACTB (60)', 'ACTBL2 (345651)', 'ACTC1 (70)', 'ACTG1 (71)', 'ACTG2 (72)', 'ACTL10 (170487)', 'ACTL6A (86)', 'ACTL6B (51412)', 'ACTL7A (10881)', 'ACTL7B (10880)', 'ACTL8 (81569)', 'ACTL9 (284382)', 'ACTMAP (284325)', 'ACTN1 (87)', 'ACTN2 (88)', 'ACTN4 (81)', 'ACTR10 (55860)', 'ACTR1A (10121)', 'ACTR1B (10120)', 'ACTR2 (10097)', 'ACTR3 (10096)', 'ACTR3B (57180)', 'ACTR3C (653857)', 'ACTR5 (79913)', 'ACTR6 (64431)', 'ACTR8 (93973)', 'ACTRT1 (139741)', 'ACTRT2 (140625)', 'ACTRT3 (84517)', 'ACVR1 (90)', 'ACVR1B (91)', 'ACVR1C (130399)', 'ACVR2A (92)', 'ACVR2B (93)', 'ACVRL1 (94)', 'ACY1 (95)', 'ACY3 (91703)', 'ACYP1 (97)', 'ACYP2 (98)', 'ADA (100)', 'ADA2 (51816)', 'ADAD1 (132612)', 'ADAD2 (161931)', 'ADAL (161823)', 'ADAM10 (102)', 'ADAM11 (4185)', 'ADAM12 (8038)', 'ADAM15 (8751)', 'ADAM17 (6868)', 'ADAM18 (8749)', 'ADAM19 (8728)', 'ADAM2 (2515)', 'ADAM20 (8748)', 'ADAM21 (8747)', 'ADAM22 (53616)', 'ADAM23 (8745)', 'ADAM28 (10863)', 'ADAM29 (11086)', 'ADAM30 (11085)', 'ADAM32 (203102)', 'ADAM33 (80332)', 'ADAM7 (8756)', 'ADAM8 (101)', 'ADAM9 (8754)', 'ADAMDEC1 (27299)', 'ADAMTS1 (9510)', 'ADAMTS10 (81794)', 'ADAMTS12 (81792)', 'ADAMTS13 (11093)', 'ADAMTS14 (140766)', 'ADAMTS15 (170689)', 'ADAMTS16 (170690)', 'ADAMTS17 (170691)', 'ADAMTS18 (170692)', 'ADAMTS19 (171019)', 'ADAMTS2 (9509)', 'ADAMTS20 (80070)', 'ADAMTS3 (9508)', 'ADAMTS4 (9507)', 'ADAMTS5 (11096)', 'ADAMTS6 (11174)', 'ADAMTS7 (11173)', 'ADAMTS8 (11095)', 'ADAMTS9 (56999)', 'ADAMTSL1 (92949)', 'ADAMTSL2 (9719)', 'ADAMTSL3 (57188)', 'ADAMTSL4 (54507)', 'ADAMTSL5 (339366)', 'ADAP1 (11033)', 'ADAP2 (55803)', 'ADAR (103)', 'ADARB1 (104)', 'ADARB2 (105)', 'ADAT1 (23536)', 'ADAT2 (134637)', 'ADAT3 (113179)', 'ADCK1 (57143)', 'ADCK2 (90956)', 'ADCK5 (203054)', 'ADCY1 (107)', 'ADCY10 (55811)', 'ADCY2 (108)', 'ADCY3 (109)', 'ADCY4 (196883)', 'ADCY5 (111)', 'ADCY6 (112)', 'ADCY7 (113)', 'ADCY8 (114)', 'ADCY9 (115)', 'ADCYAP1 (116)', 'ADCYAP1R1 (117)', 'ADD1 (118)', 'ADD2 (119)', 'ADD3 (120)', 'ADGRA1 (84435)', 'ADGRA2 (25960)', 'ADGRA3 (166647)', 'ADGRB1 (575)', 'ADGRB2 (576)', 'ADGRB3 (577)', 'ADGRD1 (283383)', 'ADGRE1 (2015)', 'ADGRE2 (30817)', 'ADGRE3 (84658)', 'ADGRE5 (976)', 'ADGRF1 (266977)', 'ADGRF3 (165082)', 'ADGRF4 (221393)', 'ADGRF5 (221395)', 'ADGRG1 (9289)', 'ADGRG2 (10149)', 'ADGRG3 (222487)', 'ADGRG4 (139378)', 'ADGRG5 (221188)', 'ADGRG6 (57211)', 'ADGRG7 (84873)', 'ADGRL1 (22859)', 'ADGRL2 (23266)', 'ADGRL3 (23284)', 'ADGRL4 (64123)', 'ADGRV1 (84059)', 'ADH1A (124)', 'ADH1B (125)', 'ADH4 (127)', 'ADH5 (128)', 'ADH6 (130)', 'ADH7 (131)', 'ADHFE1 (137872)', 'ADI1 (55256)', 'ADIG (149685)', 'ADIPOQ (9370)', 'ADIPOR1 (51094)', 'ADIPOR2 (79602)', 'ADIRF (10974)', 'ADISSP (54976)', 'ADK (132)', 'ADM (133)', 'ADM2 (79924)', 'ADM5 (199800)', 'ADNP (23394)', 'ADNP2 (22850)', 'ADO (84890)', 'ADORA1 (134)', 'ADORA2A (135)', 'ADORA2B (136)', 'ADORA3 (140)', 'ADPGK (83440)', 'ADPRH (141)', 'ADPRHL1 (113622)', 'ADPRM (56985)', 'ADPRS (54936)', 'ADRA1A (148)', 'ADRA1B (147)', 'ADRA1D (146)', 'ADRA2A (150)', 'ADRA2B (151)', 'ADRA2C (152)', 'ADRB1 (153)', 'ADRB2 (154)', 'ADRB3 (155)', 'ADRM1 (11047)', 'ADSL (158)', 'ADSS1 (122622)', 'ADSS2 (159)', 'ADTRP (84830)', 'AEBP1 (165)', 'AEBP2 (121536)', 'AEN (64782)', 'AFAP1 (60312)', 'AFAP1L1 (134265)', 'AFAP1L2 (84632)', 'AFDN (4301)', 'AFF1 (4299)', 'AFF2 (2334)', 'AFF3 (3899)', 'AFF4 (27125)', 'AFG1L (246269)', 'AFG2A (166378)', 'AFG2B (79029)', 'AFG3L2 (10939)', 'AFM (173)', 'AFMID (125061)', 'AFP (174)', 'AFTPH (54812)', 'AGA (175)', 'AGAP1 (116987)', 'AGAP2 (116986)', 'AGAP3 (116988)', 'AGAP5 (729092)', 'AGAP6 (414189)', 'AGAP9 (642517)', 'AGBL1 (123624)', 'AGBL2 (79841)', 'AGBL3 (340351)', 'AGBL4 (84871)', 'AGBL5 (60509)', 'AGER (177)', 'AGFG1 (3267)', 'AGFG2 (3268)', 'AGGF1 (55109)', 'AGK (55750)', 'AGL (178)', 'AGMAT (79814)', 'AGMO (392636)', 'AGO1 (26523)', 'AGO2 (27161)', 'AGO3 (192669)', 'AGO4 (192670)', 'AGPAT1 (10554)', 'AGPAT2 (10555)', 'AGPAT3 (56894)', 'AGPAT4 (56895)', 'AGPAT5 (55326)', 'AGPS (8540)', 'AGR2 (10551)', 'AGR3 (155465)', 'AGRN (375790)', 'AGRP (181)', 'AGT (183)', 'AGTPBP1 (23287)', 'AGTR1 (185)', 'AGTR2 (186)', 'AGTRAP (57085)', 'AGXT (189)', 'AGXT2 (64902)', 'AHCTF1 (25909)', 'AHCY (191)', 'AHCYL1 (10768)', 'AHCYL2 (23382)', 'AHDC1 (27245)', 'AHI1 (54806)', 'AHNAK (79026)', 'AHNAK2 (113146)', 'AHR (196)', 'AHRR (57491)', 'AHSA1 (10598)', 'AHSG (197)', 'AHSP (51327)', 'AICDA (57379)', 'AIDA (64853)', 'AIF1 (199)', 'AIF1L (83543)', 'AIFM1 (9131)', 'AIFM2 (84883)', 'AIFM3 (150209)', 'AIG1 (51390)', 'AIM2 (9447)', 'AIMP1 (9255)', 'AIMP2 (7965)', 'AIP (9049)', 'AIPL1 (23746)', 'AIRE (326)', 'AIRIM (54955)', 'AJAP1 (55966)', 'AJM1 (389813)', 'AJUBA (84962)', 'AK1 (203)', 'AK2 (204)', 'AK3 (50808)', 'AK4 (205)', 'AK5 (26289)', 'AK6 (102157402)', 'AK7 (122481)', 'AK8 (158067)', 'AK9 (221264)', 'AKAIN1 (642597)', 'AKAP1 (8165)', 'AKAP10 (11216)', 'AKAP11 (11215)', 'AKAP12 (9590)', 'AKAP13 (11214)', 'AKAP14 (158798)', 'AKAP3 (10566)', 'AKAP4 (8852)', 'AKAP5 (9495)', 'AKAP6 (9472)', 'AKAP7 (9465)', 'AKAP8 (10270)', 'AKAP8L (26993)', 'AKAP9 (10142)', 'AKIP1 (56672)', 'AKIRIN1 (79647)', 'AKIRIN2 (55122)', 'AKNA (80709)', 'AKNAD1 (254268)', 'AKR1A1 (10327)', 'AKR1B1 (231)', 'AKR1B10 (57016)', 'AKR1B15 (441282)', 'AKR1C2 (1646)', 'AKR1C3 (8644)', 'AKR1C4 (1109)', 'AKR1D1 (6718)', 'AKR1E2 (83592)', 'AKR7A2 (8574)', 'AKR7A3 (22977)', 'AKT1 (207)', 'AKT1S1 (84335)', 'AKT2 (208)', 'AKT3 (10000)', 'AKTIP (64400)', 'ALAD (210)', 'ALAS1 (211)', 'ALAS2 (212)', 'ALB (213)', 'ALCAM (214)', 'ALDH16A1 (126133)', 'ALDH18A1 (5832)', 'ALDH1A1 (216)', 'ALDH1A2 (8854)', 'ALDH1A3 (220)', 'ALDH1B1 (219)', 'ALDH1L1 (10840)', 'ALDH1L2 (160428)', 'ALDH2 (217)', 'ALDH3A1 (218)', 'ALDH3A2 (224)', 'ALDH3B2 (222)', 'ALDH4A1 (8659)', 'ALDH5A1 (7915)', 'ALDH6A1 (4329)', 'ALDH7A1 (501)', 'ALDH8A1 (64577)', 'ALDH9A1 (223)', 'ALDOA (226)', 'ALDOB (229)', 'ALDOC (230)', 'ALG1 (56052)', 'ALG10 (84920)', 'ALG10B (144245)', 'ALG11 (440138)', 'ALG12 (79087)', 'ALG13 (79868)', 'ALG14 (199857)', 'ALG2 (85365)', 'ALG3 (10195)', 'ALG5 (29880)', 'ALG6 (29929)', 'ALG8 (79053)', 'ALG9 (79796)', 'ALK (238)', 'ALKAL1 (389658)', 'ALKAL2 (285016)', 'ALKBH1 (8846)', 'ALKBH2 (121642)', 'ALKBH3 (221120)', 'ALKBH4 (54784)', 'ALKBH5 (54890)', 'ALKBH6 (84964)', 'ALKBH7 (84266)', 'ALKBH8 (91801)', 'ALLC (55821)', 'ALMS1 (7840)', 'ALOX12 (239)', 'ALOX12B (242)', 'ALOX15 (246)', 'ALOX15B (247)', 'ALOX5 (240)', 'ALOX5AP (241)', 'ALOXE3 (59344)', 'ALPG (251)', 'ALPI (248)', 'ALPK1 (80216)', 'ALPK2 (115701)', 'ALPK3 (57538)', 'ALPL (249)', 'ALPP (250)', 'ALS2 (57679)', 'ALS2CL (259173)', 'ALX1 (8092)', 'ALX3 (257)', 'ALX4 (60529)', 'ALYREF (10189)', 'AMACR (23600)', 'AMBN (258)', 'AMBP (259)', 'AMBRA1 (55626)', 'AMD1 (262)', 'AMDHD1 (144193)', 'AMDHD2 (51005)', 'AMELX (265)', 'AMELY (266)', 'AMER1 (139285)', 'AMER2 (219287)', 'AMER3 (205147)', 'AMFR (267)', 'AMH (268)', 'AMHR2 (269)', 'AMIGO1 (57463)', 'AMIGO2 (347902)', 'AMIGO3 (386724)', 'AMMECR1 (9949)', 'AMMECR1L (83607)', 'AMN (81693)', 'AMN1 (196394)', 'AMOT (154796)', 'AMOTL1 (154810)', 'AMOTL2 (51421)', 'AMPD1 (270)', 'AMPD2 (271)', 'AMPD3 (272)', 'AMPH (273)', 'AMT (275)', 'AMTN (401138)', 'AMY2A (279)', 'AMZ1 (155185)', 'AMZ2 (51321)', 'ANAPC1 (64682)', 'ANAPC10 (10393)', 'ANAPC11 (51529)', 'ANAPC13 (25847)', 'ANAPC15 (25906)', 'ANAPC16 (119504)', 'ANAPC2 (29882)', 'ANAPC4 (29945)', 'ANAPC5 (51433)', 'ANAPC7 (51434)', 'ANG (283)', 'ANGEL1 (23357)', 'ANGEL2 (90806)', 'ANGPT1 (284)', 'ANGPT2 (285)', 'ANGPT4 (51378)', 'ANGPTL1 (9068)', 'ANGPTL2 (23452)', 'ANGPTL3 (27329)', 'ANGPTL4 (51129)', 'ANGPTL5 (253935)', 'ANGPTL6 (83854)', 'ANGPTL7 (10218)', 'ANGPTL8 (55908)', 'ANHX (647589)', 'ANK1 (286)', 'ANK2 (287)', 'ANK3 (288)', 'ANKAR (150709)', 'ANKDD1A (348094)', 'ANKDD1B (728780)', 'ANKEF1 (63926)', 'ANKFN1 (162282)', 'ANKFY1 (51479)', 'ANKH (56172)', 'ANKHD1 (54882)', 'ANKIB1 (54467)', 'ANKK1 (255239)', 'ANKLE1 (126549)', 'ANKLE2 (23141)', 'ANKMY1 (51281)', 'ANKMY2 (57037)', 'ANKRA2 (57763)', 'ANKRD1 (27063)', 'ANKRD10 (55608)', 'ANKRD11 (29123)', 'ANKRD12 (23253)', 'ANKRD13A (88455)', 'ANKRD13B (124930)', 'ANKRD13C (81573)', 'ANKRD13D (338692)', 'ANKRD16 (54522)', 'ANKRD17 (26057)', 'ANKRD18A (253650)', 'ANKRD2 (26287)', 'ANKRD22 (118932)', 'ANKRD23 (200539)', 'ANKRD24 (170961)', 'ANKRD26 (22852)', 'ANKRD27 (84079)', 'ANKRD28 (23243)', 'ANKRD29 (147463)', 'ANKRD30A (91074)', 'ANKRD30B (374860)', 'ANKRD31 (256006)', 'ANKRD33 (341405)', 'ANKRD33B (651746)', 'ANKRD34A (284615)', 'ANKRD34B (340120)', 'ANKRD34C (390616)', 'ANKRD35 (148741)', 'ANKRD36 (375248)', 'ANKRD37 (353322)', 'ANKRD39 (51239)', 'ANKRD40 (91369)', 'ANKRD42 (338699)', 'ANKRD44 (91526)', 'ANKRD45 (339416)', 'ANKRD46 (157567)', 'ANKRD49 (54851)', 'ANKRD50 (57182)', 'ANKRD52 (283373)', 'ANKRD53 (79998)', 'ANKRD54 (129138)', 'ANKRD55 (79722)', 'ANKRD6 (22881)', 'ANKRD61 (100310846)', 'ANKRD62 (342850)', 'ANKRD65 (441869)', 'ANKRD66 (100287718)', 'ANKRD7 (56311)', 'ANKRD9 (122416)', 'ANKS1A (23294)', 'ANKS1B (56899)', 'ANKS3 (124401)', 'ANKS4B (257629)', 'ANKS6 (203286)', 'ANKZF1 (55139)', 'ANLN (54443)', 'ANO1 (55107)', 'ANO10 (55129)', 'ANO2 (57101)', 'ANO3 (63982)', 'ANO4 (121601)', 'ANO5 (203859)', 'ANO6 (196527)', 'ANO7 (50636)', 'ANO8 (57719)', 'ANO9 (338440)', 'ANOS1 (3730)', 'ANP32A (8125)', 'ANP32B (10541)', 'ANP32D (23519)', 'ANP32E (81611)', 'ANPEP (290)', 'ANTKMT (65990)', 'ANTXR1 (84168)', 'ANTXR2 (118429)', 'ANTXRL (195977)', 'ANXA1 (301)', 'ANXA10 (11199)', 'ANXA11 (311)', 'ANXA13 (312)', 'ANXA2 (302)', 'ANXA2R (389289)', 'ANXA3 (306)', 'ANXA4 (307)', 'ANXA5 (308)', 'ANXA6 (309)', 'ANXA7 (310)', 'ANXA8 (653145)', 'ANXA8L1 (728113)', 'ANXA9 (8416)', 'AOAH (313)', 'AOC1 (26)', 'AOC2 (314)', 'AOC3 (8639)', 'AOPEP (84909)', 'AOX1 (316)', 'AP1AR (55435)', 'AP1B1 (162)', 'AP1G1 (164)', 'AP1G2 (8906)', 'AP1M1 (8907)', 'AP1M2 (10053)', 'AP1S1 (1174)', 'AP1S2 (8905)', 'AP1S3 (130340)', 'AP2A1 (160)', 'AP2A2 (161)', 'AP2B1 (163)', 'AP2M1 (1173)', 'AP2S1 (1175)', 'AP3B1 (8546)', 'AP3B2 (8120)', 'AP3D1 (8943)', 'AP3M1 (26985)', 'AP3M2 (10947)', 'AP3S1 (1176)', 'AP3S2 (10239)', 'AP4B1 (10717)', 'AP4E1 (23431)', 'AP4M1 (9179)', 'AP4S1 (11154)', 'AP5B1 (91056)', 'AP5M1 (55745)', 'AP5S1 (55317)', 'AP5Z1 (9907)', 'APAF1 (317)', 'APBA1 (320)', 'APBA2 (321)', 'APBA3 (9546)', 'APBB1 (322)', 'APBB1IP (54518)', 'APBB2 (323)', 'APBB3 (10307)', 'APC (324)', 'APC2 (10297)', 'APCDD1 (147495)', 'APCDD1L (164284)', 'APCS (325)', 'APEH (327)', 'APEX1 (328)', 'APEX2 (27301)', 'APH1A (51107)', 'APH1B (83464)', 'API5 (8539)', 'APIP (51074)', 'APLF (200558)', 'APLN (8862)', 'APLNR (187)', 'APLP1 (333)', 'APLP2 (334)', 'APMAP (57136)', 'APOA1 (335)', 'APOA2 (336)', 'APOA4 (337)', 'APOA5 (116519)', 'APOB (338)', 'APOBEC1 (339)', 'APOBEC2 (10930)', 'APOBEC3A (200315)', 'APOBEC3B (9582)', 'APOBEC3C (27350)', 'APOBEC3D (140564)', 'APOBEC3F (200316)', 'APOBEC3G (60489)', 'APOBEC3H (164668)', 'APOBEC4 (403314)', 'APOBR (55911)', 'APOC1 (341)', 'APOC2 (344)', 'APOC3 (345)', 'APOC4 (346)', 'APOD (347)', 'APOE (348)', 'APOF (319)', 'APOH (350)', 'APOL1 (8542)', 'APOL2 (23780)', 'APOL3 (80833)', 'APOL5 (80831)', 'APOL6 (80830)', 'APOLD1 (81575)', 'APOM (55937)', 'APOO (79135)', 'APOOL (139322)', 'APP (351)', 'APPBP2 (10513)', 'APPL1 (26060)', 'APPL2 (55198)', 'APRT (353)', 'APTX (54840)', 'AQP1 (358)', 'AQP10 (89872)', 'AQP11 (282679)', 'AQP12B (653437)', 'AQP2 (359)', 'AQP3 (360)', 'AQP4 (361)', 'AQP5 (362)', 'AQP6 (363)', 'AQP7 (364)', 'AQP8 (343)', 'AQP9 (366)', 'AQR (9716)', 'AR (367)', 'ARAF (369)', 'ARAP1 (116985)', 'ARAP2 (116984)', 'ARAP3 (64411)', 'ARB2A (83989)', 'ARC (23237)', 'ARCN1 (372)', 'AREG (374)', 'AREL1 (9870)', 'ARF1 (375)', 'ARF3 (377)', 'ARF4 (378)', 'ARF5 (381)', 'ARF6 (382)', 'ARFGAP1 (55738)', 'ARFGAP2 (84364)', 'ARFGAP3 (26286)', 'ARFGEF1 (10565)', 'ARFGEF2 (10564)', 'ARFGEF3 (57221)', 'ARFIP1 (27236)', 'ARFIP2 (23647)', 'ARFRP1 (10139)', 'ARG1 (383)', 'ARG2 (384)', 'ARGFX (503582)', 'ARGLU1 (55082)', 'ARHGAP1 (392)', 'ARHGAP10 (79658)', 'ARHGAP11A (9824)', 'ARHGAP12 (94134)', 'ARHGAP15 (55843)', 'ARHGAP17 (55114)', 'ARHGAP18 (93663)', 'ARHGAP19 (84986)', 'ARHGAP20 (57569)', 'ARHGAP21 (57584)', 'ARHGAP22 (58504)', 'ARHGAP23 (57636)', 'ARHGAP24 (83478)', 'ARHGAP25 (9938)', 'ARHGAP26 (23092)', 'ARHGAP27 (201176)', 'ARHGAP28 (79822)', 'ARHGAP29 (9411)', 'ARHGAP30 (257106)', 'ARHGAP31 (57514)', 'ARHGAP32 (9743)', 'ARHGAP33 (115703)', 'ARHGAP35 (2909)', 'ARHGAP36 (158763)', 'ARHGAP39 (80728)', 'ARHGAP4 (393)', 'ARHGAP44 (9912)', 'ARHGAP45 (23526)', 'ARHGAP5 (394)', 'ARHGAP6 (395)', 'ARHGAP8 (23779)', 'ARHGAP9 (64333)', 'ARHGDIA (396)', 'ARHGDIB (397)', 'ARHGDIG (398)', 'ARHGEF1 (9138)', 'ARHGEF10 (9639)', 'ARHGEF10L (55160)', 'ARHGEF11 (9826)', 'ARHGEF12 (23365)', 'ARHGEF15 (22899)', 'ARHGEF16 (27237)', 'ARHGEF17 (9828)', 'ARHGEF18 (23370)', 'ARHGEF19 (128272)', 'ARHGEF2 (9181)', 'ARHGEF25 (115557)', 'ARHGEF26 (26084)', 'ARHGEF28 (64283)', 'ARHGEF3 (50650)', 'ARHGEF33 (100271715)', 'ARHGEF35 (445328)', 'ARHGEF37 (389337)', 'ARHGEF38 (54848)', 'ARHGEF39 (84904)', 'ARHGEF4 (50649)', 'ARHGEF40 (55701)', 'ARHGEF5 (7984)', 'ARHGEF6 (9459)', 'ARHGEF7 (8874)', 'ARHGEF9 (23229)', 'ARID1A (8289)', 'ARID1B (57492)', 'ARID2 (196528)', 'ARID3A (1820)', 'ARID3B (10620)', 'ARID3C (138715)', 'ARID4A (5926)', 'ARID4B (51742)', 'ARID5A (10865)', 'ARID5B (84159)', 'ARIH1 (25820)', 'ARIH2 (10425)', 'ARK2C (494470)', 'ARK2N (147339)', 'ARL1 (400)', 'ARL10 (285598)', 'ARL11 (115761)', 'ARL13A (392509)', 'ARL13B (200894)', 'ARL14 (80117)', 'ARL14EP (120534)', 'ARL14EPL (644100)', 'ARL15 (54622)', 'ARL16 (339231)', 'ARL2 (402)', 'ARL2BP (23568)', 'ARL3 (403)', 'ARL4A (10124)', 'ARL4C (10123)', 'ARL4D (379)', 'ARL5A (26225)', 'ARL5B (221079)', 'ARL5C (390790)', 'ARL6 (84100)', 'ARL6IP1 (23204)', 'ARL6IP5 (10550)', 'ARL6IP6 (151188)', 'ARL8A (127829)', 'ARL8B (55207)', 'ARL9 (132946)', 'ARMC1 (55156)', 'ARMC10 (83787)', 'ARMC12 (221481)', 'ARMC2 (84071)', 'ARMC3 (219681)', 'ARMC5 (79798)', 'ARMC6 (93436)', 'ARMC7 (79637)', 'ARMC8 (25852)', 'ARMC9 (80210)', 'ARMCX1 (51309)', 'ARMCX2 (9823)', 'ARMCX3 (51566)', 'ARMCX4 (100131755)', 'ARMCX5 (64860)', 'ARMCX6 (54470)', 'ARMH1 (339541)', 'ARMH2 (101928603)', 'ARMH3 (79591)', 'ARMH4 (145407)', 'ARMS2 (387715)', 'ARMT1 (79624)', 'ARNT (405)', 'ARNT2 (9915)', 'ARPC1A (10552)', 'ARPC1B (10095)', 'ARPC2 (10109)', 'ARPC3 (10094)', 'ARPC4 (10093)', 'ARPC5 (10092)', 'ARPC5L (81873)', 'ARPIN (348110)', 'ARPP19 (10776)', 'ARPP21 (10777)', 'ARR3 (407)', 'ARRB1 (408)', 'ARRB2 (409)', 'ARRDC1 (92714)', 'ARRDC2 (27106)', 'ARRDC3 (57561)', 'ARRDC4 (91947)', 'ARRDC5 (645432)', 'ARSA (410)', 'ARSB (411)', 'ARSD (414)', 'ARSF (416)', 'ARSG (22901)', 'ARSH (347527)', 'ARSI (340075)', 'ARSJ (79642)', 'ARSK (153642)', 'ARSL (415)', 'ART1 (417)', 'ART3 (419)', 'ART4 (420)', 'ART5 (116969)', 'ARTN (9048)', 'ARV1 (64801)', 'ARVCF (421)', 'ARX (170302)', 'AS3MT (57412)', 'ASAH1 (427)', 'ASAH2 (56624)', 'ASAP1 (50807)', 'ASAP2 (8853)', 'ASAP3 (55616)', 'ASB1 (51665)', 'ASB10 (136371)', 'ASB11 (140456)', 'ASB12 (142689)', 'ASB13 (79754)', 'ASB14 (142686)', 'ASB15 (142685)', 'ASB16 (92591)', 'ASB17 (127247)', 'ASB18 (401036)', 'ASB2 (51676)', 'ASB3 (51130)', 'ASB4 (51666)', 'ASB5 (140458)', 'ASB6 (140459)', 'ASB7 (140460)', 'ASB8 (140461)', 'ASB9 (140462)', 'ASCC1 (51008)', 'ASCC2 (84164)', 'ASCC3 (10973)', 'ASCL1 (429)', 'ASCL2 (430)', 'ASCL3 (56676)', 'ASCL4 (121549)', 'ASCL5 (647219)', 'ASF1A (25842)', 'ASF1B (55723)', 'ASGR1 (432)', 'ASGR2 (433)', 'ASH1L (55870)', 'ASH2L (9070)', 'ASIC1 (41)', 'ASIC2 (40)', 'ASIC3 (9311)', 'ASIC4 (55515)', 'ASIC5 (51802)', 'ASIP (434)', 'ASL (435)', 'ASNS (440)', 'ASNSD1 (54529)', 'ASPA (443)', 'ASPDH (554235)', 'ASPG (374569)', 'ASPH (444)', 'ASPHD1 (253982)', 'ASPHD2 (57168)', 'ASPM (259266)', 'ASPRV1 (151516)', 'ASPSCR1 (79058)', 'ASRGL1 (80150)', 'ASS1 (445)', 'ASTE1 (28990)', 'ASTL (431705)', 'ASTN1 (460)', 'ASTN2 (23245)', 'ASXL1 (171023)', 'ASXL3 (80816)', 'ASZ1 (136991)', 'ATAD1 (84896)', 'ATAD2 (29028)', 'ATAD2B (54454)', 'ATAD3A (55210)', 'ATAD3B (83858)', 'ATAD3C (219293)', 'ATAD5 (79915)', 'ATAT1 (79969)', 'ATCAY (85300)', 'ATE1 (11101)', 'ATF1 (466)', 'ATF2 (1386)', 'ATF3 (467)', 'ATF4 (468)', 'ATF5 (22809)', 'ATF6 (22926)', 'ATF6B (1388)', 'ATF7 (11016)', 'ATF7IP (55729)', 'ATF7IP2 (80063)', 'ATG10 (83734)', 'ATG101 (60673)', 'ATG12 (9140)', 'ATG13 (9776)', 'ATG14 (22863)', 'ATG16L1 (55054)', 'ATG16L2 (89849)', 'ATG2A (23130)', 'ATG2B (55102)', 'ATG3 (64422)', 'ATG4A (115201)', 'ATG4B (23192)', 'ATG4C (84938)', 'ATG4D (84971)', 'ATG5 (9474)', 'ATG7 (10533)', 'ATG9A (79065)', 'ATIC (471)', 'ATL1 (51062)', 'ATL2 (64225)', 'ATL3 (25923)', 'ATM (472)', 'ATMIN (23300)', 'ATN1 (1822)', 'ATOH1 (474)', 'ATOH7 (220202)', 'ATOH8 (84913)', 'ATOSA (56204)', 'ATOSB (80256)', 'ATOX1 (475)', 'ATP10A (57194)', 'ATP10B (23120)', 'ATP10D (57205)', 'ATP11A (23250)', 'ATP11B (23200)', 'ATP11C (286410)', 'ATP12A (479)', 'ATP13A1 (57130)', 'ATP13A2 (23400)', 'ATP13A3 (79572)', 'ATP13A4 (84239)', 'ATP13A5 (344905)', 'ATP1A1 (476)', 'ATP1A2 (477)', 'ATP1A3 (478)', 'ATP1A4 (480)', 'ATP1B1 (481)', 'ATP1B2 (482)', 'ATP1B3 (483)', 'ATP1B4 (23439)', 'ATP23 (91419)', 'ATP2A1 (487)', 'ATP2A2 (488)', 'ATP2A3 (489)', 'ATP2B1 (490)', 'ATP2B2 (491)', 'ATP2B3 (492)', 'ATP2B4 (493)', 'ATP2C1 (27032)', 'ATP2C2 (9914)', 'ATP4A (495)', 'ATP4B (496)', 'ATP5F1A (498)', 'ATP5F1B (506)', 'ATP5F1C (509)', 'ATP5F1D (513)', 'ATP5F1E (514)', 'ATP5IF1 (93974)', 'ATP5MC1 (516)', 'ATP5MC2 (517)', 'ATP5MC3 (518)', 'ATP5ME (521)', 'ATP5MF (9551)', 'ATP5MG (10632)', 'ATP5MGL (267020)', 'ATP5MJ (9556)', 'ATP5MK (84833)', 'ATP5PB (515)', 'ATP5PD (10476)', 'ATP5PF (522)', 'ATP5PO (539)', 'ATP6AP1 (537)', 'ATP6AP2 (10159)', 'ATP6V0A1 (535)', 'ATP6V0A2 (23545)', 'ATP6V0A4 (50617)', 'ATP6V0B (533)', 'ATP6V0C (527)', 'ATP6V0D1 (9114)', 'ATP6V0D2 (245972)', 'ATP6V0E1 (8992)', 'ATP6V0E2 (155066)', 'ATP6V1A (523)', 'ATP6V1B1 (525)', 'ATP6V1B2 (526)', 'ATP6V1C1 (528)', 'ATP6V1C2 (245973)', 'ATP6V1D (51382)', 'ATP6V1E1 (529)', 'ATP6V1E2 (90423)', 'ATP6V1F (9296)', 'ATP6V1G1 (9550)', 'ATP6V1G2 (534)', 'ATP6V1G3 (127124)', 'ATP6V1H (51606)', 'ATP7A (538)', 'ATP7B (540)', 'ATP8A1 (10396)', 'ATP8A2 (51761)', 'ATP8B1 (5205)', 'ATP8B2 (57198)', 'ATP8B3 (148229)', 'ATP8B4 (79895)', 'ATP9A (10079)', 'ATP9B (374868)', 'ATPAF1 (64756)', 'ATPAF2 (91647)', 'ATPSCKMT (134145)', 'ATR (545)', 'ATRAID (51374)', 'ATRIP (84126)', 'ATRN (8455)', 'ATRNL1 (26033)', 'ATRX (546)', 'ATXN1 (6310)', 'ATXN10 (25814)', 'ATXN1L (342371)', 'ATXN2 (6311)', 'ATXN2L (11273)', 'ATXN3 (4287)', 'ATXN3L (92552)', 'ATXN7 (6314)', 'ATXN7L1 (222255)', 'ATXN7L2 (127002)', 'ATXN7L3 (56970)', 'ATXN7L3B (552889)', 'AUH (549)', 'AUNIP (79000)', 'AUP1 (550)', 'AURKA (6790)', 'AURKAIP1 (54998)', 'AURKB (9212)', 'AURKC (6795)', 'AUTS2 (26053)', 'AVEN (57099)', 'AVIL (10677)', 'AVL9 (23080)', 'AVP (551)', 'AVPI1 (60370)', 'AVPR1A (552)', 'AVPR1B (553)', 'AVPR2 (554)', 'AWAT1 (158833)', 'AWAT2 (158835)', 'AXDND1 (126859)', 'AXIN1 (8312)', 'AXIN2 (8313)', 'AXL (558)', 'AZGP1 (563)', 'AZI2 (64343)', 'AZIN1 (51582)', 'AZIN2 (113451)', 'AZU1 (566)', 'B2M (567)', 'B3GALNT1 (8706)', 'B3GALNT2 (148789)', 'B3GALT1 (8708)', 'B3GALT2 (8707)', 'B3GALT4 (8705)', 'B3GALT5 (10317)', 'B3GALT6 (126792)', 'B3GAT1 (27087)', 'B3GAT2 (135152)', 'B3GAT3 (26229)', 'B3GLCT (145173)', 'B3GNT2 (10678)', 'B3GNT3 (10331)', 'B3GNT4 (79369)', 'B3GNT5 (84002)', 'B3GNT6 (192134)', 'B3GNT7 (93010)', 'B3GNT8 (374907)', 'B3GNT9 (84752)', 'B3GNTL1 (146712)', 'B4GALNT1 (2583)', 'B4GALNT2 (124872)', 'B4GALNT3 (283358)', 'B4GALNT4 (338707)', 'B4GALT1 (2683)', 'B4GALT2 (8704)', 'B4GALT3 (8703)', 'B4GALT4 (8702)', 'B4GALT5 (9334)', 'B4GALT6 (9331)', 'B4GALT7 (11285)', 'B4GAT1 (11041)', 'B9D1 (27077)', 'B9D2 (80776)', 'BAALC (79870)', 'BAAT (570)', 'BABAM1 (29086)', 'BABAM2 (9577)', 'BACE1 (23621)', 'BACE2 (25825)', 'BACH1 (571)', 'BACH2 (60468)', 'BAD (572)', 'BAG1 (573)', 'BAG2 (9532)', 'BAG3 (9531)', 'BAG4 (9530)', 'BAG5 (9529)', 'BAG6 (7917)', 'BAHD1 (22893)', 'BAIAP2 (10458)', 'BAIAP2L1 (55971)', 'BAIAP2L2 (80115)', 'BAIAP3 (8938)', 'BAK1 (578)', 'BAMBI (25805)', 'BANF1 (8815)', 'BANF2 (140836)', 'BANK1 (55024)', 'BANP (54971)', 'BAP1 (8314)', 'BARD1 (580)', 'BARHL1 (56751)', 'BARHL2 (343472)', 'BARX1 (56033)', 'BARX2 (8538)', 'BASP1 (10409)', 'BATF (10538)', 'BATF2 (116071)', 'BATF3 (55509)', 'BAX (581)', 'BAZ1A (11177)', 'BAZ1B (9031)', 'BAZ2A (11176)', 'BAZ2B (29994)', 'BBC3 (27113)', 'BBIP1 (92482)', 'BBLN (79095)', 'BBOF1 (80127)', 'BBOX1 (8424)', 'BBS1 (582)', 'BBS10 (79738)', 'BBS12 (166379)', 'BBS2 (583)', 'BBS4 (585)', 'BBS5 (129880)', 'BBS7 (55212)', 'BBS9 (27241)', 'BBX (56987)', 'BCAM (4059)', 'BCAN (63827)', 'BCAP29 (55973)', 'BCAP31 (10134)', 'BCAR1 (9564)', 'BCAR3 (8412)', 'BCAS1 (8537)', 'BCAS2 (10286)', 'BCAS3 (54828)', 'BCAS4 (55653)', 'BCAT1 (586)', 'BCAT2 (587)', 'BCCIP (56647)', 'BCDIN3D (144233)', 'BCHE (590)', 'BCKDHA (593)', 'BCKDHB (594)', 'BCKDK (10295)', 'BCL10 (8915)', 'BCL11A (53335)', 'BCL11B (64919)', 'BCL2 (596)', 'BCL2A1 (597)', 'BCL2L1 (598)', 'BCL2L10 (10017)', 'BCL2L11 (10018)', 'BCL2L12 (83596)', 'BCL2L13 (23786)', 'BCL2L14 (79370)', 'BCL2L15 (440603)', 'BCL2L2 (599)', 'BCL3 (602)', 'BCL6 (604)', 'BCL6B (255877)', 'BCL7A (605)', 'BCL7B (9275)', 'BCL7C (9274)', 'BCL9 (607)', 'BCL9L (283149)', 'BCLAF1 (9774)', 'BCLAF3 (256643)', 'BCO1 (53630)', 'BCO2 (83875)', 'BCOR (54880)', 'BCORL1 (63035)', 'BCR (613)', 'BCS1L (617)', 'BDH1 (622)', 'BDH2 (56898)', 'BDKRB1 (623)', 'BDKRB2 (624)', 'BDNF (627)', 'BDP1 (55814)', 'BEAN1 (146227)', 'BECN1 (8678)', 'BEGAIN (57596)', 'BEND2 (139105)', 'BEND3 (57673)', 'BEND4 (389206)', 'BEND5 (79656)', 'BEND6 (221336)', 'BEND7 (222389)', 'BEST1 (7439)', 'BEST2 (54831)', 'BEST3 (144453)', 'BEST4 (266675)', 'BET1 (10282)', 'BET1L (51272)', 'BEX1 (55859)', 'BEX2 (84707)', 'BEX3 (27018)', 'BEX4 (56271)', 'BEX5 (340542)', 'BFAR (51283)', 'BFSP1 (631)', 'BFSP2 (8419)', 'BGLAP (632)', 'BGN (633)', 'BHLHA15 (168620)', 'BHLHA9 (727857)', 'BHLHE22 (27319)', 'BHLHE23 (128408)', 'BHLHE40 (8553)', 'BHLHE41 (79365)', 'BHMT (635)', 'BHMT2 (23743)', 'BICC1 (80114)', 'BICD1 (636)', 'BICD2 (23299)', 'BICDL1 (92558)', 'BICDL2 (146439)', 'BICRA (29998)', 'BICRAL (23506)', 'BID (637)', 'BIK (638)', 'BIN1 (274)', 'BIN2 (51411)', 'BIN3 (55909)', 'BIRC2 (329)', 'BIRC3 (330)', 'BIRC5 (332)', 'BIRC6 (57448)', 'BIRC7 (79444)', 'BIVM (54841)', 'BLCAP (10904)', 'BLID (414899)', 'BLK (640)', 'BLM (641)', 'BLMH (642)', 'BLNK (29760)', 'BLOC1S1 (2647)', 'BLOC1S2 (282991)', 'BLOC1S3 (388552)', 'BLOC1S4 (55330)', 'BLOC1S5 (63915)', 'BLOC1S6 (26258)', 'BLTP1 (84162)', 'BLTP2 (9703)', 'BLTP3A (54887)', 'BLTP3B (23074)', 'BLVRA (644)', 'BLVRB (645)', 'BLZF1 (8548)', 'BMAL1 (406)', 'BMAL2 (56938)', 'BMERB1 (89927)', 'BMF (90427)', 'BMI1 (648)', 'BMP1 (649)', 'BMP10 (27302)', 'BMP15 (9210)', 'BMP2 (650)', 'BMP2K (55589)', 'BMP3 (651)', 'BMP4 (652)', 'BMP5 (653)', 'BMP6 (654)', 'BMP7 (655)', 'BMP8A (353500)', 'BMP8B (656)', 'BMPER (168667)', 'BMPR1A (657)', 'BMPR1B (658)', 'BMPR2 (659)', 'BMS1 (9790)', 'BMT2 (154743)', 'BMX (660)', 'BNC1 (646)', 'BNC2 (54796)', 'BNIP1 (662)', 'BNIP2 (663)', 'BNIP3 (664)', 'BNIP3L (665)', 'BNIP5 (389384)', 'BNIPL (149428)', 'BOC (91653)', 'BOD1 (91272)', 'BOD1L1 (259282)', 'BOD1L2 (284257)', 'BOK (666)', 'BOLA1 (51027)', 'BOLA3 (388962)', 'BOLL (66037)', 'BOP1 (23246)', 'BORA (79866)', 'BORCS5 (118426)', 'BORCS6 (54785)', 'BORCS7 (119032)', 'BORCS8 (729991)', 'BPGM (669)', 'BPHL (670)', 'BPI (671)', 'BPIFA1 (51297)', 'BPIFA2 (140683)', 'BPIFA3 (128861)', 'BPIFB1 (92747)', 'BPIFB2 (80341)', 'BPIFB3 (359710)', 'BPIFB4 (149954)', 'BPIFB6 (128859)', 'BPIFC (254240)', 'BPNT1 (10380)', 'BPNT2 (54928)', 'BPTF (2186)', 'BRAF (673)', 'BRAP (8315)', 'BRAT1 (221927)', 'BRCA1 (672)', 'BRCA2 (675)', 'BRCC3 (79184)', 'BRD1 (23774)', 'BRD2 (6046)', 'BRD3 (8019)', 'BRD4 (23476)', 'BRD7 (29117)', 'BRD8 (10902)', 'BRD9 (65980)', 'BRDT (676)', 'BRF1 (2972)', 'BRF2 (55290)', 'BRI3 (25798)', 'BRI3BP (140707)', 'BRICD5 (283870)', 'BRINP1 (1620)', 'BRINP2 (57795)', 'BRINP3 (339479)', 'BRIP1 (83990)', 'BRIX1 (55299)', 'BRK1 (55845)', 'BRME1 (79173)', 'BRMS1 (25855)', 'BRMS1L (84312)', 'BROX (148362)', 'BRPF1 (7862)', 'BRPF3 (27154)', 'BRS3 (680)', 'BRSK1 (84446)', 'BRSK2 (9024)', 'BRWD1 (54014)', 'BRWD3 (254065)', 'BSCL2 (26580)', 'BSDC1 (55108)', 'BSG (682)', 'BSN (8927)', 'BSND (7809)', 'BSPH1 (100131137)', 'BSPRY (54836)', 'BST1 (683)', 'BST2 (684)', 'BSX (390259)', 'BTAF1 (9044)', 'BTBD1 (53339)', 'BTBD10 (84280)', 'BTBD16 (118663)', 'BTBD17 (388419)', 'BTBD18 (643376)', 'BTBD2 (55643)', 'BTBD3 (22903)', 'BTBD6 (90135)', 'BTBD7 (55727)', 'BTBD8 (284697)', 'BTBD9 (114781)', 'BTC (685)', 'BTD (686)', 'BTF3 (689)', 'BTF3L4 (91408)', 'BTG1 (694)', 'BTG2 (7832)', 'BTG3 (10950)', 'BTG4 (54766)', 'BTK (695)', 'BTLA (151888)', 'BTN1A1 (696)', 'BTN2A1 (11120)', 'BTN2A2 (10385)', 'BTN3A1 (11119)', 'BTN3A2 (11118)', 'BTN3A3 (10384)', 'BTNL3 (10917)', 'BTNL8 (79908)', 'BTNL9 (153579)', 'BTRC (8945)', 'BUB1 (699)', 'BUB1B (701)', 'BUB3 (9184)', 'BUD13 (84811)', 'BUD23 (114049)', 'BUD31 (8896)', 'BVES (11149)', 'BYSL (705)', 'BZW1 (9689)', 'BZW2 (28969)', 'C10orf105 (414152)', 'C10orf120 (399814)', 'C10orf53 (282966)', 'C10orf62 (414157)', 'C10orf67 (256815)', 'C10orf71 (118461)', 'C10orf88 (80007)', 'C10orf90 (118611)', 'C10orf95 (79946)', 'C11orf16 (56673)', 'C11orf21 (29125)', 'C11orf24 (53838)', 'C11orf42 (160298)', 'C11orf52 (91894)', 'C11orf54 (28970)', 'C11orf58 (10944)', 'C11orf65 (160140)', 'C11orf68 (83638)', 'C11orf71 (54494)', 'C11orf86 (254439)', 'C11orf87 (399947)', 'C11orf91 (100131378)', 'C12orf42 (374470)', 'C12orf43 (64897)', 'C12orf50 (160419)', 'C12orf54 (121273)', 'C12orf56 (115749)', 'C12orf57 (113246)', 'C12orf60 (144608)', 'C12orf71 (728858)', 'C12orf75 (387882)', 'C14orf119 (55017)', 'C14orf180 (400258)', 'C14orf28 (122525)', 'C14orf39 (317761)', 'C14orf93 (60686)', 'C15orf39 (56905)', 'C15orf40 (123207)', 'C15orf48 (84419)', 'C15orf61 (145853)', 'C15orf62 (643338)', 'C16orf46 (123775)', 'C16orf54 (283897)', 'C16orf74 (404550)', 'C16orf78 (123970)', 'C16orf86 (388284)', 'C16orf87 (388272)', 'C16orf89 (146556)', 'C16orf90 (646174)', 'C16orf92 (146378)', 'C16orf95 (100506581)', 'C16orf96 (342346)', 'C17orf107 (100130311)', 'C17orf49 (124944)', 'C17orf50 (146853)', 'C17orf58 (284018)', 'C17orf67 (339210)', 'C17orf75 (64149)', 'C17orf78 (284099)', 'C17orf99 (100141515)', 'C18orf21 (83608)', 'C18orf32 (497661)', 'C18orf54 (162681)', 'C18orf63 (644041)', 'C19orf12 (83636)', 'C19orf18 (147685)', 'C19orf25 (148223)', 'C19orf33 (64073)', 'C19orf38 (255809)', 'C19orf44 (84167)', 'C19orf47 (126526)', 'C19orf53 (28974)', 'C19orf67 (646457)', 'C19orf73 (55150)', 'C19orf81 (342918)', 'C19orf84 (147646)', 'C1D (10438)', 'C1GALT1 (56913)', 'C1GALT1C1 (29071)', 'C1QA (712)', 'C1QB (713)', 'C1QBP (708)', 'C1QC (714)', 'C1QL1 (10882)', 'C1QL2 (165257)', 'C1QL3 (389941)', 'C1QL4 (338761)', 'C1QTNF1 (114897)', 'C1QTNF12 (388581)', 'C1QTNF2 (114898)', 'C1QTNF3 (114899)', 'C1QTNF4 (114900)', 'C1QTNF5 (114902)', 'C1QTNF6 (114904)', 'C1QTNF7 (114905)', 'C1QTNF8 (390664)', 'C1QTNF9 (338872)', 'C1QTNF9B (387911)', 'C1RL (51279)', 'C1S (716)', 'C1orf105 (92346)', 'C1orf115 (79762)', 'C1orf116 (79098)', 'C1orf122 (127687)', 'C1orf127 (148345)', 'C1orf131 (128061)', 'C1orf141 (400757)', 'C1orf146 (388649)', 'C1orf159 (54991)', 'C1orf162 (128346)', 'C1orf174 (339448)', 'C1orf185 (284546)', 'C1orf198 (84886)', 'C1orf21 (81563)', 'C1orf210 (149466)', 'C1orf216 (127703)', 'C1orf226 (400793)', 'C1orf35 (79169)', 'C1orf43 (25912)', 'C1orf50 (79078)', 'C1orf52 (148423)', 'C1orf53 (388722)', 'C1orf54 (79630)', 'C1orf56 (54964)', 'C1orf74 (148304)', 'C1orf87 (127795)', 'C1orf94 (84970)', 'C2 (717)', 'C20orf141 (128653)', 'C20orf144 (128864)', 'C20orf173 (140873)', 'C20orf202 (400831)', 'C20orf96 (140680)', 'C21orf140 (101928147)', 'C21orf58 (54058)', 'C21orf91 (54149)', 'C22orf15 (150248)', 'C22orf23 (84645)', 'C22orf31 (25770)', 'C22orf39 (128977)', 'C22orf42 (150297)', 'C2CD2 (25966)', 'C2CD2L (9854)', 'C2CD3 (26005)', 'C2CD4A (145741)', 'C2CD4B (388125)', 'C2CD4C (126567)', 'C2CD4D (100191040)', 'C2CD5 (9847)', 'C2CD6 (151254)', 'C2orf15 (150590)', 'C2orf42 (54980)', 'C2orf49 (79074)', 'C2orf50 (130813)', 'C2orf66 (401027)', 'C2orf68 (388969)', 'C2orf69 (205327)', 'C2orf72 (257407)', 'C2orf73 (129852)', 'C2orf76 (130355)', 'C2orf78 (388960)', 'C2orf80 (389073)', 'C2orf88 (84281)', 'C3 (718)', 'C3AR1 (719)', 'C3orf18 (51161)', 'C3orf20 (84077)', 'C3orf22 (152065)', 'C3orf33 (285315)', 'C3orf38 (285237)', 'C3orf52 (79669)', 'C3orf62 (375341)', 'C3orf70 (285382)', 'C3orf80 (401097)', 'C3orf84 (646498)', 'C4BPA (722)', 'C4BPB (725)', 'C4orf17 (84103)', 'C4orf19 (55286)', 'C4orf3 (401152)', 'C4orf33 (132321)', 'C4orf36 (132989)', 'C4orf46 (201725)', 'C4orf51 (646603)', 'C5 (727)', 'C5AR1 (728)', 'C5AR2 (27202)', 'C5orf15 (56951)', 'C5orf22 (55322)', 'C5orf24 (134553)', 'C5orf34 (375444)', 'C5orf46 (389336)', 'C5orf47 (133491)', 'C5orf52 (100190949)', 'C5orf58 (133874)', 'C5orf63 (401207)', 'C6 (729)', 'C6orf118 (168090)', 'C6orf120 (387263)', 'C6orf132 (647024)', 'C6orf136 (221545)', 'C6orf141 (135398)', 'C6orf15 (29113)', 'C6orf163 (206412)', 'C6orf226 (441150)', 'C6orf47 (57827)', 'C6orf52 (347744)', 'C6orf58 (352999)', 'C6orf62 (81688)', 'C6orf89 (221477)', 'C7 (730)', 'C7orf25 (79020)', 'C7orf33 (202865)', 'C7orf50 (84310)', 'C7orf57 (136288)', 'C8A (731)', 'C8B (732)', 'C8G (733)', 'C8orf33 (65265)', 'C8orf34 (116328)', 'C8orf48 (157773)', 'C8orf58 (541565)', 'C8orf74 (203076)', 'C8orf76 (84933)', 'C8orf82 (414919)', 'C8orf88 (100127983)', 'C9 (735)', 'C9orf152 (401546)', 'C9orf153 (389766)', 'C9orf40 (55071)', 'C9orf43 (257169)', 'C9orf50 (375759)', 'C9orf57 (138240)', 'C9orf72 (203228)', 'C9orf78 (51759)', 'C9orf85 (138241)', 'CA1 (759)', 'CA10 (56934)', 'CA11 (770)', 'CA12 (771)', 'CA13 (377677)', 'CA14 (23632)', 'CA2 (760)', 'CA3 (761)', 'CA4 (762)', 'CA5A (763)', 'CA5B (11238)', 'CA6 (765)', 'CA7 (766)', 'CA8 (767)', 'CA9 (768)', 'CAAP1 (79886)', 'CAB39 (51719)', 'CAB39L (81617)', 'CABCOCO1 (219621)', 'CABIN1 (23523)', 'CABLES1 (91768)', 'CABLES2 (81928)', 'CABP1 (9478)', 'CABP2 (51475)', 'CABP4 (57010)', 'CABP5 (56344)', 'CABP7 (164633)', 'CABS1 (85438)', 'CABYR (26256)', 'CACFD1 (11094)', 'CACHD1 (57685)', 'CACNA1A (773)', 'CACNA1B (774)', 'CACNA1C (775)', 'CACNA1D (776)', 'CACNA1E (777)', 'CACNA1F (778)', 'CACNA1G (8913)', 'CACNA1H (8912)', 'CACNA1I (8911)', 'CACNA1S (779)', 'CACNA2D1 (781)', 'CACNA2D2 (9254)', 'CACNA2D3 (55799)', 'CACNA2D4 (93589)', 'CACNB1 (782)', 'CACNB2 (783)', 'CACNB3 (784)', 'CACNB4 (785)', 'CACNG1 (786)', 'CACNG2 (10369)', 'CACNG3 (10368)', 'CACNG4 (27092)', 'CACNG5 (27091)', 'CACNG6 (59285)', 'CACNG7 (59284)', 'CACNG8 (59283)', 'CACTIN (58509)', 'CACUL1 (143384)', 'CACYBP (27101)', 'CAD (790)', 'CADM1 (23705)', 'CADM2 (253559)', 'CADM3 (57863)', 'CADM4 (199731)', 'CADPS (8618)', 'CADPS2 (93664)', 'CAGE1 (285782)', 'CALB1 (793)', 'CALB2 (794)', 'CALCA (796)', 'CALCB (797)', 'CALCOCO1 (57658)', 'CALCOCO2 (10241)', 'CALCR (799)', 'CALCRL (10203)', 'CALD1 (800)', 'CALHM1 (255022)', 'CALHM2 (51063)', 'CALHM3 (119395)', 'CALHM4 (221301)', 'CALHM5 (254228)', 'CALHM6 (441168)', 'CALM1 (801)', 'CALM2 (805)', 'CALM3 (808)', 'CALML3 (810)', 'CALML4 (91860)', 'CALML5 (51806)', 'CALML6 (163688)', 'CALN1 (83698)', 'CALR (811)', 'CALR3 (125972)', 'CALU (813)', 'CALY (50632)', 'CAMK1 (8536)', 'CAMK1D (57118)', 'CAMK1G (57172)', 'CAMK2A (815)', 'CAMK2B (816)', 'CAMK2D (817)', 'CAMK2G (818)', 'CAMK2N1 (55450)', 'CAMK2N2 (94032)', 'CAMK4 (814)', 'CAMKK1 (84254)', 'CAMKK2 (10645)', 'CAMKMT (79823)', 'CAMKV (79012)', 'CAMLG (819)', 'CAMP (820)', 'CAMSAP1 (157922)', 'CAMSAP2 (23271)', 'CAMSAP3 (57662)', 'CAMTA1 (23261)', 'CAMTA2 (23125)', 'CAND1 (55832)', 'CAND2 (23066)', 'CANT1 (124583)', 'CANX (821)', 'CAP1 (10487)', 'CAP2 (10486)', 'CAPG (822)', 'CAPN1 (823)', 'CAPN10 (11132)', 'CAPN11 (11131)', 'CAPN12 (147968)', 'CAPN13 (92291)', 'CAPN14 (440854)', 'CAPN15 (6650)', 'CAPN2 (824)', 'CAPN3 (825)', 'CAPN5 (726)', 'CAPN6 (827)', 'CAPN7 (23473)', 'CAPN9 (10753)', 'CAPNS1 (826)', 'CAPNS2 (84290)', 'CAPRIN1 (4076)', 'CAPRIN2 (65981)', 'CAPS (828)', 'CAPS2 (84698)', 'CAPSL (133690)', 'CAPZA1 (829)', 'CAPZA2 (830)', 'CAPZA3 (93661)', 'CAPZB (832)', 'CARD10 (29775)', 'CARD11 (84433)', 'CARD14 (79092)', 'CARD16 (114769)', 'CARD18 (59082)', 'CARD19 (84270)', 'CARD6 (84674)', 'CARD8 (22900)', 'CARD9 (64170)', 'CARF (79800)', 'CARHSP1 (23589)', 'CARM1 (10498)', 'CARMIL1 (55604)', 'CARMIL2 (146206)', 'CARMIL3 (90668)', 'CARNMT1 (138199)', 'CARNS1 (57571)', 'CARS1 (833)', 'CARS2 (79587)', 'CARTPT (9607)', 'CASC3 (22794)', 'CASD1 (64921)', 'CASK (8573)', 'CASKIN1 (57524)', 'CASKIN2 (57513)', 'CASP1 (834)', 'CASP10 (843)', 'CASP14 (23581)', 'CASP2 (835)', 'CASP3 (836)', 'CASP4 (837)', 'CASP5 (838)', 'CASP6 (839)', 'CASP7 (840)', 'CASP8 (841)', 'CASP9 (842)', 'CASQ1 (844)', 'CASQ2 (845)', 'CASR (846)', 'CASS4 (57091)', 'CAST (831)', 'CASTOR1 (652968)', 'CASTOR2 (729438)', 'CASZ1 (54897)', 'CAT (847)', 'CATIP (375307)', 'CATSPER1 (117144)', 'CATSPER2 (117155)', 'CATSPER3 (347732)', 'CATSPER4 (378807)', 'CATSPERB (79820)', 'CATSPERD (257062)', 'CATSPERE (257044)', 'CATSPERG (57828)', 'CAV1 (857)', 'CAV2 (858)', 'CAV3 (859)', 'CAVIN1 (284119)', 'CAVIN2 (8436)', 'CAVIN3 (112464)', 'CAVIN4 (347273)', 'CBARP (255057)', 'CBFA2T2 (9139)', 'CBFA2T3 (863)', 'CBFB (865)', 'CBL (867)', 'CBLB (868)', 'CBLC (23624)', 'CBLIF (2694)', 'CBLL1 (79872)', 'CBLL2 (158506)', 'CBLN1 (869)', 'CBLN2 (147381)', 'CBLN3 (643866)', 'CBLN4 (140689)', 'CBR1 (873)', 'CBR3 (874)', 'CBR4 (84869)', 'CBS (875)', 'CBX1 (10951)', 'CBX2 (84733)', 'CBX3 (11335)', 'CBX4 (8535)', 'CBX5 (23468)', 'CBX6 (23466)', 'CBX7 (23492)', 'CBX8 (57332)', 'CBY1 (25776)', 'CBY2 (220082)', 'CBY3 (646019)', 'CC2D1A (54862)', 'CC2D1B (200014)', 'CC2D2A (57545)', 'CC2D2B (387707)', 'CCAR1 (55749)', 'CCAR2 (57805)', 'CCBE1 (147372)', 'CCDC102A (92922)', 'CCDC102B (79839)', 'CCDC103 (388389)', 'CCDC106 (29903)', 'CCDC107 (203260)', 'CCDC110 (256309)', 'CCDC112 (153733)', 'CCDC115 (84317)', 'CCDC116 (164592)', 'CCDC117 (150275)', 'CCDC12 (151903)', 'CCDC120 (90060)', 'CCDC121 (79635)', 'CCDC122 (160857)', 'CCDC124 (115098)', 'CCDC125 (202243)', 'CCDC126 (90693)', 'CCDC127 (133957)', 'CCDC13 (152206)', 'CCDC134 (79879)', 'CCDC136 (64753)', 'CCDC137 (339230)', 'CCDC138 (165055)', 'CCDC14 (64770)', 'CCDC142 (84865)', 'CCDC144A (9720)', 'CCDC146 (57639)', 'CCDC148 (130940)', 'CCDC149 (91050)', 'CCDC15 (80071)', 'CCDC150 (284992)', 'CCDC152 (100129792)', 'CCDC157 (550631)', 'CCDC158 (339965)', 'CCDC159 (126075)', 'CCDC160 (347475)', 'CCDC166 (100130274)', 'CCDC167 (154467)', 'CCDC169 (728591)', 'CCDC17 (149483)', 'CCDC170 (80129)', 'CCDC171 (203238)', 'CCDC172 (374355)', 'CCDC174 (51244)', 'CCDC175 (729665)', 'CCDC177 (56936)', 'CCDC178 (374864)', 'CCDC179 (100500938)', 'CCDC180 (100499483)', 'CCDC181 (57821)', 'CCDC182 (101927581)', 'CCDC183 (84960)', 'CCDC184 (387856)', 'CCDC185 (164127)', 'CCDC186 (55088)', 'CCDC190 (339512)', 'CCDC191 (57577)', 'CCDC198 (55195)', 'CCDC22 (28952)', 'CCDC24 (149473)', 'CCDC25 (55246)', 'CCDC27 (148870)', 'CCDC28A (25901)', 'CCDC28B (79140)', 'CCDC3 (83643)', 'CCDC30 (728621)', 'CCDC32 (90416)', 'CCDC33 (80125)', 'CCDC34 (91057)', 'CCDC38 (120935)', 'CCDC39 (339829)', 'CCDC40 (55036)', 'CCDC42 (146849)', 'CCDC43 (124808)', 'CCDC47 (57003)', 'CCDC50 (152137)', 'CCDC51 (79714)', 'CCDC54 (84692)', 'CCDC59 (29080)', 'CCDC6 (8030)', 'CCDC60 (160777)', 'CCDC61 (729440)', 'CCDC62 (84660)', 'CCDC63 (160762)', 'CCDC65 (85478)', 'CCDC66 (285331)', 'CCDC68 (80323)', 'CCDC69 (26112)', 'CCDC7 (79741)', 'CCDC70 (83446)', 'CCDC71 (64925)', 'CCDC71L (168455)', 'CCDC73 (493860)', 'CCDC74A (90557)', 'CCDC74B (91409)', 'CCDC77 (84318)', 'CCDC78 (124093)', 'CCDC8 (83987)', 'CCDC80 (151887)', 'CCDC81 (60494)', 'CCDC82 (79780)', 'CCDC83 (220047)', 'CCDC85A (114800)', 'CCDC85B (11007)', 'CCDC85C (317762)', 'CCDC86 (79080)', 'CCDC87 (55231)', 'CCDC88A (55704)', 'CCDC88B (283234)', 'CCDC88C (440193)', 'CCDC89 (220388)', 'CCDC9 (26093)', 'CCDC90B (60492)', 'CCDC91 (55297)', 'CCDC92 (80212)', 'CCDC93 (54520)', 'CCDC97 (90324)', 'CCDC9B (388115)', 'CCER1 (196477)', 'CCER2 (643669)', 'CCHCR1 (54535)', 'CCIN (881)', 'CCK (885)', 'CCKAR (886)', 'CCKBR (887)', 'CCL1 (6346)', 'CCL11 (6356)', 'CCL13 (6357)', 'CCL14 (6358)', 'CCL15 (6359)', 'CCL16 (6360)', 'CCL17 (6361)', 'CCL18 (6362)', 'CCL19 (6363)', 'CCL2 (6347)', 'CCL20 (6364)', 'CCL21 (6366)', 'CCL22 (6367)', 'CCL23 (6368)', 'CCL24 (6369)', 'CCL25 (6370)', 'CCL26 (10344)', 'CCL27 (10850)', 'CCL28 (56477)', 'CCL3 (6348)', 'CCL3L3 (414062)', 'CCL4 (6351)', 'CCL5 (6352)', 'CCL7 (6354)', 'CCL8 (6355)', 'CCM2 (83605)', 'CCM2L (140706)', 'CCN1 (3491)', 'CCN2 (1490)', 'CCN3 (4856)', 'CCN4 (8840)', 'CCN5 (8839)', 'CCN6 (8838)', 'CCNA1 (8900)', 'CCNA2 (890)', 'CCNB1 (891)', 'CCNB1IP1 (57820)', 'CCNB2 (9133)', 'CCNB3 (85417)', 'CCNC (892)', 'CCND1 (595)', 'CCND2 (894)', 'CCND3 (896)', 'CCNDBP1 (23582)', 'CCNE1 (898)', 'CCNE2 (9134)', 'CCNF (899)', 'CCNG1 (900)', 'CCNG2 (901)', 'CCNH (902)', 'CCNI (10983)', 'CCNI2 (645121)', 'CCNJ (54619)', 'CCNJL (79616)', 'CCNK (8812)', 'CCNL1 (57018)', 'CCNL2 (81669)', 'CCNO (10309)', 'CCNP (79935)', 'CCNT1 (904)', 'CCNT2 (905)', 'CCNY (219771)', 'CCNYL1 (151195)', 'CCP110 (9738)', 'CCPG1 (9236)', 'CCR1 (1230)', 'CCR10 (2826)', 'CCR2 (729230)', 'CCR3 (1232)', 'CCR4 (1233)', 'CCR5 (1234)', 'CCR6 (1235)', 'CCR7 (1236)', 'CCR8 (1237)', 'CCR9 (10803)', 'CCRL2 (9034)', 'CCS (9973)', 'CCSAP (126731)', 'CCSER1 (401145)', 'CCSER2 (54462)', 'CCT2 (10576)', 'CCT3 (7203)', 'CCT4 (10575)', 'CCT5 (22948)', 'CCT6A (908)', 'CCT6B (10693)', 'CCT7 (10574)', 'CCT8 (10694)', 'CCT8L2 (150160)', 'CD101 (9398)', 'CD109 (135228)', 'CD14 (929)', 'CD151 (977)', 'CD160 (11126)', 'CD163 (9332)', 'CD163L1 (283316)', 'CD164 (8763)', 'CD164L2 (388611)', 'CD177 (57126)', 'CD180 (4064)', 'CD19 (930)', 'CD1A (909)', 'CD1B (910)', 'CD1C (911)', 'CD1D (912)', 'CD1E (913)', 'CD2 (914)', 'CD200 (4345)', 'CD200R1 (131450)', 'CD200R1L (344807)', 'CD209 (30835)', 'CD22 (933)', 'CD226 (10666)', 'CD244 (51744)', 'CD247 (919)', 'CD248 (57124)', 'CD27 (939)', 'CD274 (29126)', 'CD276 (80381)', 'CD28 (940)', 'CD2AP (23607)', 'CD2BP2 (10421)', 'CD300A (11314)', 'CD300C (10871)', 'CD300E (342510)', 'CD300LB (124599)', 'CD300LD (100131439)', 'CD300LF (146722)', 'CD300LG (146894)', 'CD302 (9936)', 'CD320 (51293)', 'CD33 (945)', 'CD34 (947)', 'CD36 (948)', 'CD37 (951)', 'CD38 (952)', 'CD3D (915)', 'CD3E (916)', 'CD3G (917)', 'CD4 (920)', 'CD40 (958)', 'CD40LG (959)', 'CD44 (960)', 'CD46 (4179)', 'CD47 (961)', 'CD48 (962)', 'CD5 (921)', 'CD52 (1043)', 'CD53 (963)', 'CD55 (1604)', 'CD58 (965)', 'CD59 (966)', 'CD5L (922)', 'CD6 (923)', 'CD63 (967)', 'CD68 (968)', 'CD69 (969)', 'CD7 (924)', 'CD70 (970)', 'CD72 (971)', 'CD74 (972)', 'CD79A (973)', 'CD79B (974)', 'CD80 (941)', 'CD81 (975)', 'CD82 (3732)', 'CD83 (9308)', 'CD84 (8832)', 'CD86 (942)', 'CD8A (925)', 'CD9 (928)', 'CD93 (22918)', 'CD96 (10225)', 'CD99L2 (83692)', 'CDA (978)', 'CDADC1 (81602)', 'CDAN1 (146059)', 'CDC123 (8872)', 'CDC14A (8556)', 'CDC14B (8555)', 'CDC16 (8881)', 'CDC20 (991)', 'CDC20B (166979)', 'CDC23 (8697)', 'CDC25A (993)', 'CDC25B (994)', 'CDC25C (995)', 'CDC26 (246184)', 'CDC27 (996)', 'CDC34 (997)', 'CDC37 (11140)', 'CDC37L1 (55664)', 'CDC40 (51362)', 'CDC42 (998)', 'CDC42BPA (8476)', 'CDC42BPB (9578)', 'CDC42BPG (55561)', 'CDC42EP1 (11135)', 'CDC42EP2 (10435)', 'CDC42EP3 (10602)', 'CDC42EP4 (23580)', 'CDC42EP5 (148170)', 'CDC42SE1 (56882)', 'CDC42SE2 (56990)', 'CDC45 (8318)', 'CDC5L (988)', 'CDC6 (990)', 'CDC7 (8317)', 'CDC73 (79577)', 'CDCA2 (157313)', 'CDCA3 (83461)', 'CDCA4 (55038)', 'CDCA5 (113130)', 'CDCA7 (83879)', 'CDCA7L (55536)', 'CDCA8 (55143)', 'CDCP1 (64866)', 'CDCP2 (200008)', 'CDH1 (999)', 'CDH10 (1008)', 'CDH11 (1009)', 'CDH12 (1010)', 'CDH13 (1012)', 'CDH15 (1013)', 'CDH16 (1014)', 'CDH17 (1015)', 'CDH18 (1016)', 'CDH19 (28513)', 'CDH2 (1000)', 'CDH20 (28316)', 'CDH22 (64405)', 'CDH23 (64072)', 'CDH24 (64403)', 'CDH26 (60437)', 'CDH3 (1001)', 'CDH4 (1002)', 'CDH5 (1003)', 'CDH6 (1004)', 'CDH7 (1005)', 'CDH8 (1006)', 'CDH9 (1007)', 'CDHR1 (92211)', 'CDHR2 (54825)', 'CDHR3 (222256)', 'CDHR4 (389118)', 'CDHR5 (53841)', 'CDIN1 (84529)', 'CDIP1 (29965)', 'CDIPT (10423)', 'CDK1 (983)', 'CDK10 (8558)', 'CDK12 (51755)', 'CDK13 (8621)', 'CDK14 (5218)', 'CDK15 (65061)', 'CDK16 (5127)', 'CDK17 (5128)', 'CDK18 (5129)', 'CDK19 (23097)', 'CDK2 (1017)', 'CDK20 (23552)', 'CDK2AP1 (8099)', 'CDK2AP2 (10263)', 'CDK3 (1018)', 'CDK4 (1019)', 'CDK5 (1020)', 'CDK5R1 (8851)', 'CDK5R2 (8941)', 'CDK5RAP1 (51654)', 'CDK5RAP2 (55755)', 'CDK5RAP3 (80279)', 'CDK6 (1021)', 'CDK7 (1022)', 'CDK8 (1024)', 'CDK9 (1025)', 'CDKAL1 (54901)', 'CDKL1 (8814)', 'CDKL2 (8999)', 'CDKL3 (51265)', 'CDKL4 (344387)', 'CDKL5 (6792)', 'CDKN1A (1026)', 'CDKN1B (1027)', 'CDKN1C (1028)', 'CDKN2A (1029)', 'CDKN2AIP (55602)', 'CDKN2AIPNL (91368)', 'CDKN2B (1030)', 'CDKN2C (1031)', 'CDKN2D (1032)', 'CDKN3 (1033)', 'CDNF (441549)', 'CDO1 (1036)', 'CDON (50937)', 'CDPF1 (150383)', 'CDR2 (1039)', 'CDR2L (30850)', 'CDRT15 (146822)', 'CDRT15L2 (256223)', 'CDS1 (1040)', 'CDS2 (8760)', 'CDSN (1041)', 'CDT1 (81620)', 'CDV3 (55573)', 'CDX1 (1044)', 'CDX2 (1045)', 'CDX4 (1046)', 'CDYL (9425)', 'CDYL2 (124359)', 'CEACAM1 (634)', 'CEACAM16 (388551)', 'CEACAM19 (56971)', 'CEACAM21 (90273)', 'CEACAM3 (1084)', 'CEACAM4 (1089)', 'CEACAM5 (1048)', 'CEACAM6 (4680)', 'CEACAM7 (1087)', 'CEACAM8 (1088)', 'CEBPA (1050)', 'CEBPB (1051)', 'CEBPD (1052)', 'CEBPE (1053)', 'CEBPG (1054)', 'CEBPZ (10153)', 'CEL (1056)', 'CELA1 (1990)', 'CELA2A (63036)', 'CELA2B (51032)', 'CELA3A (10136)', 'CELA3B (23436)', 'CELF1 (10658)', 'CELF2 (10659)', 'CELF3 (11189)', 'CELF4 (56853)', 'CELF5 (60680)', 'CELF6 (60677)', 'CELSR1 (9620)', 'CELSR2 (1952)', 'CELSR3 (1951)', 'CEMIP (57214)', 'CEMIP2 (23670)', 'CEMP1 (752014)', 'CENATAC (338657)', 'CEND1 (51286)', 'CENPA (1058)', 'CENPB (1059)', 'CENPC (1060)', 'CENPE (1062)', 'CENPF (1063)', 'CENPH (64946)', 'CENPI (2491)', 'CENPJ (55835)', 'CENPK (64105)', 'CENPL (91687)', 'CENPM (79019)', 'CENPN (55839)', 'CENPO (79172)', 'CENPP (401541)', 'CENPQ (55166)', 'CENPT (80152)', 'CENPU (79682)', 'CENPV (201161)', 'CENPW (387103)', 'CENPX (201254)', 'CEP104 (9731)', 'CEP112 (201134)', 'CEP120 (153241)', 'CEP126 (57562)', 'CEP128 (145508)', 'CEP131 (22994)', 'CEP135 (9662)', 'CEP15 (57415)', 'CEP152 (22995)', 'CEP162 (22832)', 'CEP164 (22897)', 'CEP170 (9859)', 'CEP170B (283638)', 'CEP19 (84984)', 'CEP192 (55125)', 'CEP20 (123811)', 'CEP250 (11190)', 'CEP290 (80184)', 'CEP295 (85459)', 'CEP295NL (100653515)', 'CEP350 (9857)', 'CEP41 (95681)', 'CEP43 (11116)', 'CEP44 (80817)', 'CEP55 (55165)', 'CEP57 (9702)', 'CEP57L1 (285753)', 'CEP63 (80254)', 'CEP68 (23177)', 'CEP70 (80321)', 'CEP72 (55722)', 'CEP76 (79959)', 'CEP78 (84131)', 'CEP83 (51134)', 'CEP85 (64793)', 'CEP85L (387119)', 'CEP89 (84902)', 'CEP95 (90799)', 'CEP97 (79598)', 'CEPT1 (10390)', 'CER1 (9350)', 'CERCAM (51148)', 'CERK (64781)', 'CERKL (375298)', 'CERS1 (10715)', 'CERS2 (29956)', 'CERS3 (204219)', 'CERS4 (79603)', 'CERS5 (91012)', 'CERS6 (253782)', 'CERT1 (10087)', 'CES1 (1066)', 'CES2 (8824)', 'CES3 (23491)', 'CES4A (283848)', 'CES5A (221223)', 'CETN1 (1068)', 'CETN2 (1069)', 'CETN3 (1070)', 'CETP (1071)', 'CFAP100 (348807)', 'CFAP107 (93190)', 'CFAP119 (90835)', 'CFAP126 (257177)', 'CFAP141 (388701)', 'CFAP144 (440585)', 'CFAP157 (286207)', 'CFAP161 (161502)', 'CFAP184 (257236)', 'CFAP20 (29105)', 'CFAP206 (154313)', 'CFAP20DC (200844)', 'CFAP210 (129881)', 'CFAP221 (200373)', 'CFAP251 (144406)', 'CFAP263 (29070)', 'CFAP276 (127003)', 'CFAP298 (56683)', 'CFAP299 (255119)', 'CFAP300 (85016)', 'CFAP36 (112942)', 'CFAP410 (755)', 'CFAP418 (157657)', 'CFAP43 (80217)', 'CFAP44 (55779)', 'CFAP45 (25790)', 'CFAP46 (54777)', 'CFAP47 (286464)', 'CFAP52 (146845)', 'CFAP53 (220136)', 'CFAP57 (149465)', 'CFAP58 (159686)', 'CFAP61 (26074)', 'CFAP65 (255101)', 'CFAP68 (64776)', 'CFAP69 (79846)', 'CFAP70 (118491)', 'CFAP73 (387885)', 'CFAP77 (389799)', 'CFAP90 (134121)', 'CFAP91 (89876)', 'CFAP92 (57501)', 'CFAP95 (138255)', 'CFAP96 (441054)', 'CFAP97 (57587)', 'CFAP97D1 (284067)', 'CFB (629)', 'CFC1 (55997)', 'CFC1B (653275)', 'CFD (1675)', 'CFDP1 (10428)', 'CFH (3075)', 'CFHR1 (3078)', 'CFHR2 (3080)', 'CFHR3 (10878)', 'CFHR4 (10877)', 'CFHR5 (81494)', 'CFI (3426)', 'CFL1 (1072)', 'CFL2 (1073)', 'CFLAR (8837)', 'CFP (5199)', 'CFTR (1080)', 'CGA (1081)', 'CGAS (115004)', 'CGB7 (94027)', 'CGGBP1 (8545)', 'CGN (57530)', 'CGNL1 (84952)', 'CGREF1 (10669)', 'CGRRF1 (10668)', 'CH25H (9023)', 'CHAC1 (79094)', 'CHAC2 (494143)', 'CHAD (1101)', 'CHADL (150356)', 'CHAF1A (10036)', 'CHAF1B (8208)', 'CHAMP1 (283489)', 'CHAT (1103)', 'CHCHD1 (118487)', 'CHCHD10 (400916)', 'CHCHD2 (51142)', 'CHCHD3 (54927)', 'CHCHD4 (131474)', 'CHCHD5 (84269)', 'CHCHD6 (84303)', 'CHCHD7 (79145)', 'CHCT1 (124773)', 'CHD1 (1105)', 'CHD1L (9557)', 'CHD2 (1106)', 'CHD3 (1107)', 'CHD4 (1108)', 'CHD5 (26038)', 'CHD6 (84181)', 'CHD7 (55636)', 'CHD8 (57680)', 'CHD9 (80205)', 'CHDH (55349)', 'CHEK1 (1111)', 'CHEK2 (11200)', 'CHERP (10523)', 'CHFR (55743)', 'CHGA (1113)', 'CHGB (1114)', 'CHI3L1 (1116)', 'CHI3L2 (1117)', 'CHIA (27159)', 'CHIC1 (53344)', 'CHIC2 (26511)', 'CHID1 (66005)', 'CHIT1 (1118)', 'CHKA (1119)', 'CHKB (1120)', 'CHL1 (10752)', 'CHM (1121)', 'CHML (1122)', 'CHMP1A (5119)', 'CHMP1B (57132)', 'CHMP2A (27243)', 'CHMP2B (25978)', 'CHMP3 (51652)', 'CHMP4A (29082)', 'CHMP4B (128866)', 'CHMP4C (92421)', 'CHMP5 (51510)', 'CHMP6 (79643)', 'CHMP7 (91782)', 'CHN1 (1123)', 'CHN2 (1124)', 'CHODL (140578)', 'CHORDC1 (26973)', 'CHP1 (11261)', 'CHP2 (63928)', 'CHPF (79586)', 'CHPF2 (54480)', 'CHPT1 (56994)', 'CHRAC1 (54108)', 'CHRD (8646)', 'CHRDL1 (91851)', 'CHRDL2 (25884)', 'CHRFAM7A (89832)', 'CHRM1 (1128)', 'CHRM2 (1129)', 'CHRM3 (1131)', 'CHRM4 (1132)', 'CHRM5 (1133)', 'CHRNA1 (1134)', 'CHRNA10 (57053)', 'CHRNA2 (1135)', 'CHRNA3 (1136)', 'CHRNA4 (1137)', 'CHRNA5 (1138)', 'CHRNA6 (8973)', 'CHRNA7 (1139)', 'CHRNA9 (55584)', 'CHRNB1 (1140)', 'CHRNB2 (1141)', 'CHRNB3 (1142)', 'CHRNB4 (1143)', 'CHRND (1144)', 'CHRNE (1145)', 'CHRNG (1146)', 'CHST1 (8534)', 'CHST10 (9486)', 'CHST11 (50515)', 'CHST12 (55501)', 'CHST13 (166012)', 'CHST14 (113189)', 'CHST15 (51363)', 'CHST2 (9435)', 'CHST3 (9469)', 'CHST4 (10164)', 'CHST5 (23563)', 'CHST6 (4166)', 'CHST7 (56548)', 'CHST8 (64377)', 'CHST9 (83539)', 'CHSY1 (22856)', 'CHSY3 (337876)', 'CHTF18 (63922)', 'CHTF8 (54921)', 'CHTOP (26097)', 'CHUK (1147)', 'CHURC1 (91612)', 'CIAO1 (9391)', 'CIAO2A (84191)', 'CIAO2B (51647)', 'CIAO3 (64428)', 'CIAPIN1 (57019)', 'CIART (148523)', 'CIB1 (10519)', 'CIB2 (10518)', 'CIB3 (117286)', 'CIB4 (130106)', 'CIBAR1 (137392)', 'CIBAR2 (339145)', 'CIC (23152)', 'CIDEA (1149)', 'CIDEB (27141)', 'CIDEC (63924)', 'CIITA (4261)', 'CILK1 (22858)', 'CILP (8483)', 'CILP2 (148113)', 'CIMAP1A (113746)', 'CIMAP1B (440836)', 'CIMAP1C (161753)', 'CIMAP1D (284451)', 'CIMAP2 (163747)', 'CIMAP3 (128344)', 'CIMIP1 (128602)', 'CIMIP2A (401565)', 'CIMIP2B (730112)', 'CIMIP2C (339778)', 'CIMIP4 (339669)', 'CINP (51550)', 'CIP2A (57650)', 'CIPC (85457)', 'CIR1 (9541)', 'CIRBP (1153)', 'CISD1 (55847)', 'CISD2 (493856)', 'CISD3 (284106)', 'CISH (1154)', 'CIT (11113)', 'CITED1 (4435)', 'CITED2 (10370)', 'CITED4 (163732)', 'CIZ1 (25792)', 'CKAP2 (26586)', 'CKAP2L (150468)', 'CKAP4 (10970)', 'CKAP5 (9793)', 'CKB (1152)', 'CKLF (51192)', 'CKM (1158)', 'CKMT2 (1160)', 'CKS1B (1163)', 'CKS2 (1164)', 'CLASRP (11129)', 'CLBA1 (122616)', 'CLC (1178)', 'CLCA1 (1179)', 'CLCA2 (9635)', 'CLCA4 (22802)', 'CLCC1 (23155)', 'CLCF1 (23529)', 'CLCN1 (1180)', 'CLCN2 (1181)', 'CLCN3 (1182)', 'CLCN4 (1183)', 'CLCN5 (1184)', 'CLCN6 (1185)', 'CLCN7 (1186)', 'CLCNKA (1187)', 'CLCNKB (1188)', 'CLDN1 (9076)', 'CLDN10 (9071)', 'CLDN11 (5010)', 'CLDN12 (9069)', 'CLDN14 (23562)', 'CLDN15 (24146)', 'CLDN16 (10686)', 'CLDN17 (26285)', 'CLDN18 (51208)', 'CLDN19 (149461)', 'CLDN2 (9075)', 'CLDN20 (49861)', 'CLDN22 (53842)', 'CLDN23 (137075)', 'CLDN25 (644672)', 'CLDN3 (1365)', 'CLDN4 (1364)', 'CLDN5 (7122)', 'CLDN6 (9074)', 'CLDN7 (1366)', 'CLDN8 (9073)', 'CLDN9 (9080)', 'CLDND1 (56650)', 'CLDND2 (125875)', 'CLEC10A (10462)', 'CLEC11A (6320)', 'CLEC12A (160364)', 'CLEC12B (387837)', 'CLEC14A (161198)', 'CLEC16A (23274)', 'CLEC17A (388512)', 'CLEC18A (348174)', 'CLEC18B (497190)', 'CLEC1A (51267)', 'CLEC1B (51266)', 'CLEC2A (387836)', 'CLEC2B (9976)', 'CLEC2D (29121)', 'CLEC2L (154790)', 'CLEC3A (10143)', 'CLEC3B (7123)', 'CLEC4A (50856)', 'CLEC4C (170482)', 'CLEC4D (338339)', 'CLEC4E (26253)', 'CLEC4F (165530)', 'CLEC4G (339390)', 'CLEC4M (10332)', 'CLEC5A (23601)', 'CLEC6A (93978)', 'CLEC7A (64581)', 'CLEC9A (283420)', 'CLGN (1047)', 'CLHC1 (130162)', 'CLIC1 (1192)', 'CLIC2 (1193)', 'CLIC3 (9022)', 'CLIC4 (25932)', 'CLIC5 (53405)', 'CLIC6 (54102)', 'CLINT1 (9685)', 'CLIP1 (6249)', 'CLIP2 (7461)', 'CLIP3 (25999)', 'CLIP4 (79745)', 'CLK1 (1195)', 'CLK2 (1196)', 'CLK3 (1198)', 'CLK4 (57396)', 'CLMN (79789)', 'CLMP (79827)', 'CLN3 (1201)', 'CLN5 (1203)', 'CLN6 (54982)', 'CLN8 (2055)', 'CLNK (116449)', 'CLNS1A (1207)', 'CLOCK (9575)', 'CLP1 (10978)', 'CLPB (81570)', 'CLPP (8192)', 'CLPS (1208)', 'CLPSL1 (340204)', 'CLPSL2 (389383)', 'CLPTM1 (1209)', 'CLPTM1L (81037)', 'CLPX (10845)', 'CLRN1 (7401)', 'CLRN2 (645104)', 'CLRN3 (119467)', 'CLSPN (63967)', 'CLSTN1 (22883)', 'CLSTN2 (64084)', 'CLSTN3 (9746)', 'CLTA (1211)', 'CLTB (1212)', 'CLTC (1213)', 'CLTCL1 (8218)', 'CLTRN (57393)', 'CLU (1191)', 'CLUAP1 (23059)', 'CLUH (23277)', 'CLUL1 (27098)', 'CLVS1 (157807)', 'CLVS2 (134829)', 'CLXN (79645)', 'CLYBL (171425)', 'CMA1 (1215)', 'CMAS (55907)', 'CMBL (134147)', 'CMC1 (152100)', 'CMC2 (56942)', 'CMC4 (100272147)', 'CMIP (80790)', 'CMKLR1 (1240)', 'CMKLR2 (2825)', 'CMPK1 (51727)', 'CMPK2 (129607)', 'CMSS1 (84319)', 'CMTM1 (113540)', 'CMTM2 (146225)', 'CMTM3 (123920)', 'CMTM4 (146223)', 'CMTM5 (116173)', 'CMTM6 (54918)', 'CMTM7 (112616)', 'CMTM8 (152189)', 'CMTR1 (23070)', 'CMTR2 (55783)', 'CMYA5 (202333)', 'CNBD1 (168975)', 'CNBD2 (140894)', 'CNBP (7555)', 'CNDP1 (84735)', 'CNDP2 (55748)', 'CNEP1R1 (255919)', 'CNFN (84518)', 'CNGA1 (1259)', 'CNGA2 (1260)', 'CNGA3 (1261)', 'CNGA4 (1262)', 'CNGB1 (1258)', 'CNGB3 (54714)', 'CNIH1 (10175)', 'CNIH2 (254263)', 'CNIH3 (149111)', 'CNIH4 (29097)', 'CNKSR1 (10256)', 'CNKSR2 (22866)', 'CNKSR3 (154043)', 'CNMD (11061)', 'CNN1 (1264)', 'CNN2 (1265)', 'CNN3 (1266)', 'CNNM1 (26507)', 'CNNM2 (54805)', 'CNNM3 (26505)', 'CNNM4 (26504)', 'CNOT1 (23019)', 'CNOT10 (25904)', 'CNOT11 (55571)', 'CNOT2 (4848)', 'CNOT3 (4849)', 'CNOT4 (4850)', 'CNOT6 (57472)', 'CNOT6L (246175)', 'CNOT7 (29883)', 'CNOT8 (9337)', 'CNOT9 (9125)', 'CNP (1267)', 'CNPPD1 (27013)', 'CNPY1 (285888)', 'CNPY2 (10330)', 'CNPY3 (10695)', 'CNPY4 (245812)', 'CNR1 (1268)', 'CNR2 (1269)', 'CNRIP1 (25927)', 'CNST (163882)', 'CNTD1 (124817)', 'CNTF (1270)', 'CNTFR (1271)', 'CNTLN (54875)', 'CNTN1 (1272)', 'CNTN2 (6900)', 'CNTN3 (5067)', 'CNTN4 (152330)', 'CNTN5 (53942)', 'CNTN6 (27255)', 'CNTNAP1 (8506)', 'CNTNAP2 (26047)', 'CNTNAP3 (79937)', 'CNTNAP3B (728577)', 'CNTNAP4 (85445)', 'CNTNAP5 (129684)', 'CNTRL (11064)', 'CNTROB (116840)', 'COA1 (55744)', 'COA3 (28958)', 'COA4 (51287)', 'COA5 (493753)', 'COA6 (388753)', 'COA7 (65260)', 'COA8 (84334)', 'COASY (80347)', 'COBL (23242)', 'COBLL1 (22837)', 'COCH (1690)', 'COG1 (9382)', 'COG2 (22796)', 'COG3 (83548)', 'COG4 (25839)', 'COG5 (10466)', 'COG6 (57511)', 'COG7 (91949)', 'COG8 (84342)', 'COIL (8161)', 'COL10A1 (1300)', 'COL11A1 (1301)', 'COL11A2 (1302)', 'COL12A1 (1303)', 'COL13A1 (1305)', 'COL14A1 (7373)', 'COL15A1 (1306)', 'COL16A1 (1307)', 'COL17A1 (1308)', 'COL18A1 (80781)', 'COL19A1 (1310)', 'COL1A1 (1277)', 'COL1A2 (1278)', 'COL20A1 (57642)', 'COL21A1 (81578)', 'COL22A1 (169044)', 'COL23A1 (91522)', 'COL24A1 (255631)', 'COL25A1 (84570)', 'COL26A1 (136227)', 'COL27A1 (85301)', 'COL28A1 (340267)', 'COL2A1 (1280)', 'COL3A1 (1281)', 'COL4A1 (1282)', 'COL4A2 (1284)', 'COL4A3 (1285)', 'COL4A4 (1286)', 'COL4A5 (1287)', 'COL4A6 (1288)', 'COL5A1 (1289)', 'COL5A2 (1290)', 'COL5A3 (50509)', 'COL6A1 (1291)', 'COL6A2 (1292)', 'COL6A3 (1293)', 'COL6A6 (131873)', 'COL7A1 (1294)', 'COL8A1 (1295)', 'COL8A2 (1296)', 'COL9A1 (1297)', 'COL9A2 (1298)', 'COL9A3 (1299)', 'COLEC10 (10584)', 'COLEC11 (78989)', 'COLEC12 (81035)', 'COLGALT1 (79709)', 'COLGALT2 (23127)', 'COLQ (8292)', 'COMMD1 (150684)', 'COMMD10 (51397)', 'COMMD2 (51122)', 'COMMD3 (23412)', 'COMMD4 (54939)', 'COMMD5 (28991)', 'COMMD6 (170622)', 'COMMD7 (149951)', 'COMMD8 (54951)', 'COMMD9 (29099)', 'COMP (1311)', 'COMT (1312)', 'COMTD1 (118881)', 'COP1 (64326)', 'COPA (1314)', 'COPB1 (1315)', 'COPB2 (9276)', 'COPE (11316)', 'COPG1 (22820)', 'COPRS (55352)', 'COPS2 (9318)', 'COPS3 (8533)', 'COPS4 (51138)', 'COPS5 (10987)', 'COPS6 (10980)', 'COPS7A (50813)', 'COPS7B (64708)', 'COPS8 (10920)', 'COPS9 (150678)', 'COPZ1 (22818)', 'COQ10A (93058)', 'COQ10B (80219)', 'COQ2 (27235)', 'COQ3 (51805)', 'COQ4 (51117)', 'COQ5 (84274)', 'COQ6 (51004)', 'COQ7 (10229)', 'COQ8A (56997)', 'COQ8B (79934)', 'COQ9 (57017)', 'CORIN (10699)', 'CORO1A (11151)', 'CORO1B (57175)', 'CORO1C (23603)', 'CORO2A (7464)', 'CORO2B (10391)', 'CORO6 (84940)', 'CORO7 (79585)', 'CORT (1325)', 'COTL1 (23406)', 'COX10 (1352)', 'COX11 (1353)', 'COX14 (84987)', 'COX15 (1355)', 'COX16 (51241)', 'COX17 (10063)', 'COX18 (285521)', 'COX19 (90639)', 'COX20 (116228)', 'COX4I1 (1327)', 'COX4I2 (84701)', 'COX5A (9377)', 'COX5B (1329)', 'COX6A1 (1337)', 'COX6A2 (1339)', 'COX6B1 (1340)', 'COX6B2 (125965)', 'COX6C (1345)', 'COX7A1 (1346)', 'COX7A2 (1347)', 'COX7A2L (9167)', 'COX7B (1349)', 'COX7B2 (170712)', 'COX7C (1350)', 'COX8A (1351)', 'COX8C (341947)', 'CP (1356)', 'CPA1 (1357)', 'CPA2 (1358)', 'CPA3 (1359)', 'CPA4 (51200)', 'CPA5 (93979)', 'CPA6 (57094)', 'CPAMD8 (27151)', 'CPB1 (1360)', 'CPB2 (1361)', 'CPD (1362)', 'CPE (1363)', 'CPEB1 (64506)', 'CPEB2 (132864)', 'CPEB3 (22849)', 'CPEB4 (80315)', 'CPED1 (79974)', 'CPLANE1 (65250)', 'CPLANE2 (79363)', 'CPLX1 (10815)', 'CPLX2 (10814)', 'CPLX3 (594855)', 'CPLX4 (339302)', 'CPM (1368)', 'CPN1 (1369)', 'CPN2 (1370)', 'CPNE1 (8904)', 'CPNE2 (221184)', 'CPNE3 (8895)', 'CPNE4 (131034)', 'CPNE5 (57699)', 'CPNE6 (9362)', 'CPNE7 (27132)', 'CPNE8 (144402)', 'CPNE9 (151835)', 'CPO (130749)', 'CPOX (1371)', 'CPPED1 (55313)', 'CPQ (10404)', 'CPS1 (1373)', 'CPSF1 (29894)', 'CPSF2 (53981)', 'CPSF3 (51692)', 'CPSF4 (10898)', 'CPSF4L (642843)', 'CPSF6 (11052)', 'CPSF7 (79869)', 'CPT1A (1374)', 'CPT1B (1375)', 'CPT1C (126129)', 'CPT2 (1376)', 'CPTP (80772)', 'CPVL (54504)', 'CPXCR1 (53336)', 'CPXM1 (56265)', 'CPXM2 (119587)', 'CPZ (8532)', 'CR1 (1378)', 'CR1L (1379)', 'CR2 (1380)', 'CRABP1 (1381)', 'CRABP2 (1382)', 'CRACD (57482)', 'CRACDL (343990)', 'CRACR2A (84766)', 'CRACR2B (283229)', 'CRADD (8738)', 'CRAMP1 (57585)', 'CRAT (1384)', 'CRB1 (23418)', 'CRB2 (286204)', 'CRB3 (92359)', 'CRBN (51185)', 'CRCP (27297)', 'CRCT1 (54544)', 'CREB1 (1385)', 'CREB3 (10488)', 'CREB3L1 (90993)', 'CREB3L2 (64764)', 'CREB3L3 (84699)', 'CREB3L4 (148327)', 'CREB5 (9586)', 'CREBBP (1387)', 'CREBL2 (1389)', 'CREBRF (153222)', 'CREBZF (58487)', 'CREG1 (8804)', 'CREG2 (200407)', 'CRELD1 (78987)', 'CRELD2 (79174)', 'CREM (1390)', 'CRH (1392)', 'CRHBP (1393)', 'CRHR1 (1394)', 'CRHR2 (1395)', 'CRIM1 (51232)', 'CRIP1 (1396)', 'CRIP2 (1397)', 'CRIP3 (401262)', 'CRIPT (9419)', 'CRIPTO (6997)', 'CRISP1 (167)', 'CRISP2 (7180)', 'CRISP3 (10321)', 'CRISPLD1 (83690)', 'CRISPLD2 (83716)', 'CRK (1398)', 'CRKL (1399)', 'CRLF1 (9244)', 'CRLF3 (51379)', 'CRLS1 (54675)', 'CRMP1 (1400)', 'CRNKL1 (51340)', 'CRNN (49860)', 'CROCC (9696)', 'CROT (54677)', 'CRP (1401)', 'CRTAC1 (55118)', 'CRTAM (56253)', 'CRTAP (10491)', 'CRTC1 (23373)', 'CRTC2 (200186)', 'CRTC3 (64784)', 'CRX (1406)', 'CRY1 (1407)', 'CRY2 (1408)', 'CRYAA (1409)', 'CRYAB (1410)', 'CRYBA1 (1411)', 'CRYBA2 (1412)', 'CRYBA4 (1413)', 'CRYBB1 (1414)', 'CRYBB2 (1415)', 'CRYBB3 (1417)', 'CRYBG1 (202)', 'CRYGA (1418)', 'CRYGB (1419)', 'CRYGC (1420)', 'CRYGD (1421)', 'CRYGN (155051)', 'CRYGS (1427)', 'CRYL1 (51084)', 'CRYM (1428)', 'CRYZ (1429)', 'CRYZL1 (9946)', 'CS (1431)', 'CSAD (51380)', 'CSAG1 (158511)', 'CSDC2 (27254)', 'CSDE1 (7812)', 'CSE1L (1434)', 'CSF1 (1435)', 'CSF1R (1436)', 'CSF2 (1437)', 'CSF2RB (1439)', 'CSF3 (1440)', 'CSF3R (1441)', 'CSGALNACT1 (55790)', 'CSGALNACT2 (55454)', 'CSH2 (1443)', 'CSHL1 (1444)', 'CSK (1445)', 'CSKMT (751071)', 'CSMD1 (64478)', 'CSMD2 (114784)', 'CSMD3 (114788)', 'CSN1S1 (1446)', 'CSN2 (1447)', 'CSN3 (1448)', 'CSNK1A1 (1452)', 'CSNK1A1L (122011)', 'CSNK1D (1453)', 'CSNK1G1 (53944)', 'CSNK1G2 (1455)', 'CSNK1G3 (1456)', 'CSNK2A1 (1457)', 'CSNK2A2 (1459)', 'CSNK2A3 (283106)', 'CSNK2B (1460)', 'CSPG4 (1464)', 'CSPG5 (10675)', 'CSPP1 (79848)', 'CSRNP1 (64651)', 'CSRNP2 (81566)', 'CSRNP3 (80034)', 'CSRP1 (1465)', 'CSRP2 (1466)', 'CSRP3 (8048)', 'CST1 (1469)', 'CST11 (140880)', 'CST2 (1470)', 'CST3 (1471)', 'CST4 (1472)', 'CST5 (1473)', 'CST6 (1474)', 'CST7 (8530)', 'CST8 (10047)', 'CST9 (128822)', 'CST9L (128821)', 'CSTA (1475)', 'CSTB (1476)', 'CSTF1 (1477)', 'CSTF2 (1478)', 'CSTF2T (23283)', 'CSTF3 (1479)', 'CSTL1 (128817)', 'CSTPP1 (79096)', 'CT47B1 (643311)', 'CT55 (54967)', 'CT83 (203413)', 'CTAG2 (30848)', 'CTAGE1 (64693)', 'CTAGE9 (643854)', 'CTBP1 (1487)', 'CTBP2 (1488)', 'CTBS (1486)', 'CTC1 (80169)', 'CTCF (10664)', 'CTCFL (140690)', 'CTDNEP1 (23399)', 'CTDP1 (9150)', 'CTDSP1 (58190)', 'CTDSP2 (10106)', 'CTDSPL (10217)', 'CTDSPL2 (51496)', 'CTF1 (1489)', 'CTH (1491)', 'CTHRC1 (115908)', 'CTIF (9811)', 'CTLA4 (1493)', 'CTNNA1 (1495)', 'CTNNA2 (1496)', 'CTNNA3 (29119)', 'CTNNAL1 (8727)', 'CTNNB1 (1499)', 'CTNNBIP1 (56998)', 'CTNNBL1 (56259)', 'CTNND1 (1500)', 'CTNND2 (1501)', 'CTNS (1497)', 'CTPS1 (1503)', 'CTPS2 (56474)', 'CTR9 (9646)', 'CTRB1 (1504)', 'CTRB2 (440387)', 'CTRC (11330)', 'CTRL (1506)', 'CTSA (5476)', 'CTSB (1508)', 'CTSC (1075)', 'CTSD (1509)', 'CTSE (1510)', 'CTSF (8722)', 'CTSG (1511)', 'CTSH (1512)', 'CTSK (1513)', 'CTSL (1514)', 'CTSO (1519)', 'CTSS (1520)', 'CTSV (1515)', 'CTSW (1521)', 'CTSZ (1522)', 'CTTN (2017)', 'CTTNBP2 (83992)', 'CTTNBP2NL (55917)', 'CTU1 (90353)', 'CTU2 (348180)', 'CTXN1 (404217)', 'CTXN2 (399697)', 'CTXN3 (613212)', 'CUBN (8029)', 'CUEDC1 (404093)', 'CUEDC2 (79004)', 'CUL1 (8454)', 'CUL2 (8453)', 'CUL3 (8452)', 'CUL4A (8451)', 'CUL4B (8450)', 'CUL5 (8065)', 'CUL7 (9820)', 'CUL9 (23113)', 'CUTA (51596)', 'CUTC (51076)', 'CUX1 (1523)', 'CUX2 (23316)', 'CUZD1 (50624)', 'CWC22 (57703)', 'CWC25 (54883)', 'CWC27 (10283)', 'CWF19L1 (55280)', 'CWF19L2 (143884)', 'CWH43 (80157)', 'CX3CL1 (6376)', 'CX3CR1 (1524)', 'CXADR (1525)', 'CXCL1 (2919)', 'CXCL10 (3627)', 'CXCL11 (6373)', 'CXCL12 (6387)', 'CXCL13 (10563)', 'CXCL14 (9547)', 'CXCL16 (58191)', 'CXCL17 (284340)', 'CXCL2 (2920)', 'CXCL3 (2921)', 'CXCL5 (6374)', 'CXCL6 (6372)', 'CXCL8 (3576)', 'CXCL9 (4283)', 'CXCR1 (3577)', 'CXCR2 (3579)', 'CXCR3 (2833)', 'CXCR4 (7852)', 'CXCR5 (643)', 'CXCR6 (10663)', 'CXXC1 (30827)', 'CXXC4 (80319)', 'CXXC5 (51523)', 'CXorf38 (159013)', 'CXorf58 (254158)', 'CXorf65 (158830)', 'CXorf66 (347487)', 'CYB561 (1534)', 'CYB561A3 (220002)', 'CYB561D1 (284613)', 'CYB561D2 (11068)', 'CYB5A (1528)', 'CYB5B (80777)', 'CYB5D1 (124637)', 'CYB5D2 (124936)', 'CYB5R1 (51706)', 'CYB5R2 (51700)', 'CYB5R3 (1727)', 'CYB5R4 (51167)', 'CYB5RL (606495)', 'CYBA (1535)', 'CYBB (1536)', 'CYBC1 (79415)', 'CYBRD1 (79901)', 'CYC1 (1537)', 'CYCS (54205)', 'CYFIP1 (23191)', 'CYGB (114757)', 'CYLC1 (1538)', 'CYLC2 (1539)', 'CYLD (1540)', 'CYP11A1 (1583)', 'CYP11B1 (1584)', 'CYP11B2 (1585)', 'CYP17A1 (1586)', 'CYP19A1 (1588)', 'CYP1A1 (1543)', 'CYP1A2 (1544)', 'CYP1B1 (1545)', 'CYP20A1 (57404)', 'CYP21A2 (1589)', 'CYP24A1 (1591)', 'CYP26A1 (1592)', 'CYP26B1 (56603)', 'CYP26C1 (340665)', 'CYP27A1 (1593)', 'CYP27B1 (1594)', 'CYP27C1 (339761)', 'CYP2A13 (1553)', 'CYP2A6 (1548)', 'CYP2A7 (1549)', 'CYP2B6 (1555)', 'CYP2C18 (1562)', 'CYP2C19 (1557)', 'CYP2C8 (1558)', 'CYP2C9 (1559)', 'CYP2D6 (1565)', 'CYP2E1 (1571)', 'CYP2F1 (1572)', 'CYP2J2 (1573)', 'CYP2R1 (120227)', 'CYP2S1 (29785)', 'CYP2U1 (113612)', 'CYP2W1 (54905)', 'CYP39A1 (51302)', 'CYP3A4 (1576)', 'CYP3A43 (64816)', 'CYP3A5 (1577)', 'CYP3A7 (1551)', 'CYP46A1 (10858)', 'CYP4A11 (1579)', 'CYP4A22 (284541)', 'CYP4B1 (1580)', 'CYP4F11 (57834)', 'CYP4F12 (66002)', 'CYP4F2 (8529)', 'CYP4F22 (126410)', 'CYP4F3 (4051)', 'CYP4V2 (285440)', 'CYP4X1 (260293)', 'CYP4Z1 (199974)', 'CYP51A1 (1595)', 'CYP7A1 (1581)', 'CYP7B1 (9420)', 'CYP8B1 (1582)', 'CYREN (78996)', 'CYRIA (81553)', 'CYRIB (51571)', 'CYS1 (192668)', 'CYSLTR1 (10800)', 'CYSLTR2 (57105)', 'CYSRT1 (375791)', 'CYSTM1 (84418)', 'CYTH1 (9267)', 'CYTH2 (9266)', 'CYTH3 (9265)', 'CYTH4 (27128)', 'CYTIP (9595)', 'CYTL1 (54360)', 'CYYR1 (116159)', 'CZIB (54987)', 'D2HGDH (728294)', 'DAAM1 (23002)', 'DAAM2 (23500)', 'DAB1 (1600)', 'DAB2 (1601)', 'DAB2IP (153090)', 'DACH1 (1602)', 'DACH2 (117154)', 'DACT1 (51339)', 'DACT2 (168002)', 'DACT3 (147906)', 'DAD1 (1603)', 'DAG1 (1605)', 'DAGLA (747)', 'DAGLB (221955)', 'DALRD3 (55152)', 'DAND5 (199699)', 'DAO (1610)', 'DAOA (267012)', 'DAP (1611)', 'DAP3 (7818)', 'DAPK1 (1612)', 'DAPK2 (23604)', 'DAPK3 (1613)', 'DAPL1 (92196)', 'DAPP1 (27071)', 'DARS1 (1615)', 'DARS2 (55157)', 'DAW1 (164781)', 'DAXX (1616)', 'DAZAP1 (26528)', 'DAZAP2 (9802)', 'DAZL (1618)', 'DBF4 (10926)', 'DBF4B (80174)', 'DBH (1621)', 'DBI (1622)', 'DBN1 (1627)', 'DBNDD1 (79007)', 'DBNDD2 (55861)', 'DBNL (28988)', 'DBP (1628)', 'DBR1 (51163)', 'DBT (1629)', 'DBX2 (440097)', 'DCAF10 (79269)', 'DCAF11 (80344)', 'DCAF12 (25853)', 'DCAF12L1 (139170)', 'DCAF12L2 (340578)', 'DCAF15 (90379)', 'DCAF16 (54876)', 'DCAF17 (80067)', 'DCAF4 (26094)', 'DCAF4L1 (285429)', 'DCAF4L2 (138009)', 'DCAF5 (8816)', 'DCAF6 (55827)', 'DCAF8 (50717)', 'DCAF8L1 (139425)', 'DCAF8L2 (347442)', 'DCAKD (79877)', 'DCANP1 (140947)', 'DCBLD1 (285761)', 'DCBLD2 (131566)', 'DCC (1630)', 'DCD (117159)', 'DCDC1 (341019)', 'DCDC2 (51473)', 'DCDC2B (149069)', 'DCHS1 (8642)', 'DCHS2 (54798)', 'DCK (1633)', 'DCLK1 (9201)', 'DCLK2 (166614)', 'DCLK3 (85443)', 'DCLRE1A (9937)', 'DCLRE1B (64858)', 'DCLRE1C (64421)', 'DCN (1634)', 'DCP1B (196513)', 'DCP2 (167227)', 'DCPS (28960)', 'DCST1 (149095)', 'DCST2 (127579)', 'DCSTAMP (81501)', 'DCT (1638)', 'DCTD (1635)', 'DCTN1 (1639)', 'DCTN2 (10540)', 'DCTN3 (11258)', 'DCTN4 (51164)', 'DCTN5 (84516)', 'DCTN6 (10671)', 'DCTPP1 (79077)', 'DCUN1D1 (54165)', 'DCUN1D2 (55208)', 'DCUN1D3 (123879)', 'DCUN1D4 (23142)', 'DCUN1D5 (84259)', 'DCX (1641)', 'DCXR (51181)', 'DDA1 (79016)', 'DDAH1 (23576)', 'DDAH2 (23564)', 'DDB1 (1642)', 'DDB2 (1643)', 'DDC (1644)', 'DDHD1 (80821)', 'DDHD2 (23259)', 'DDI1 (414301)', 'DDI2 (84301)', 'DDIAS (220042)', 'DDIT3 (1649)', 'DDIT4 (54541)', 'DDIT4L (115265)', 'DDN (23109)', 'DDO (8528)', 'DDOST (1650)', 'DDR1 (780)', 'DDR2 (4921)', 'DDRGK1 (65992)', 'DDX1 (1653)', 'DDX10 (1662)', 'DDX11 (1663)', 'DDX17 (10521)', 'DDX18 (8886)', 'DDX19A (55308)', 'DDX19B (11269)', 'DDX20 (11218)', 'DDX21 (9188)', 'DDX23 (9416)', 'DDX24 (57062)', 'DDX25 (29118)', 'DDX27 (55661)', 'DDX28 (55794)', 'DDX31 (64794)', 'DDX39A (10212)', 'DDX39B (7919)', 'DDX3X (1654)', 'DDX3Y (8653)', 'DDX4 (54514)', 'DDX41 (51428)', 'DDX42 (11325)', 'DDX43 (55510)', 'DDX46 (9879)', 'DDX47 (51202)', 'DDX49 (54555)', 'DDX5 (1655)', 'DDX50 (79009)', 'DDX51 (317781)', 'DDX52 (11056)', 'DDX53 (168400)', 'DDX54 (79039)', 'DDX55 (57696)', 'DDX56 (54606)', 'DDX59 (83479)', 'DDX6 (1656)', 'DDX60 (55601)', 'DDX60L (91351)', 'DEAF1 (10522)', 'DECR1 (1666)', 'DECR2 (26063)', 'DEDD (9191)', 'DEDD2 (162989)', 'DEF6 (50619)', 'DEF8 (54849)', 'DEFA4 (1669)', 'DEFA5 (1670)', 'DEFA6 (1671)', 'DEFB1 (1672)', 'DEFB110 (245913)', 'DEFB112 (245915)', 'DEFB113 (245927)', 'DEFB114 (245928)', 'DEFB115 (245929)', 'DEFB116 (245930)', 'DEFB118 (117285)', 'DEFB119 (245932)', 'DEFB121 (245934)', 'DEFB123 (245936)', 'DEFB124 (245937)', 'DEFB125 (245938)', 'DEFB126 (81623)', 'DEFB127 (140850)', 'DEFB128 (245939)', 'DEFB129 (140881)', 'DEFB131A (644414)', 'DEFB132 (400830)', 'DEFB134 (613211)', 'DEFB135 (613209)', 'DEFB136 (613210)', 'DEGS1 (8560)', 'DEGS2 (123099)', 'DEK (7913)', 'DELE1 (9812)', 'DENND10 (404636)', 'DENND11 (57189)', 'DENND1A (57706)', 'DENND1B (163486)', 'DENND1C (79958)', 'DENND2A (27147)', 'DENND2B (6764)', 'DENND2C (163259)', 'DENND2D (79961)', 'DENND3 (22898)', 'DENND4A (10260)', 'DENND4B (9909)', 'DENND4C (55667)', 'DENND5A (23258)', 'DENND5B (160518)', 'DENND6A (201627)', 'DENND6B (414918)', 'DENR (8562)', 'DEPDC1 (55635)', 'DEPDC1B (55789)', 'DEPDC4 (120863)', 'DEPDC5 (9681)', 'DEPDC7 (91614)', 'DEPP1 (11067)', 'DEPTOR (64798)', 'DERA (51071)', 'DERL1 (79139)', 'DERL2 (51009)', 'DERL3 (91319)', 'DES (1674)', 'DESI1 (27351)', 'DESI2 (51029)', 'DET1 (55070)', 'DEUP1 (159989)', 'DFFA (1676)', 'DFFB (1677)', 'DGAT1 (8694)', 'DGAT2 (84649)', 'DGAT2L6 (347516)', 'DGCR2 (9993)', 'DGCR6 (8214)', 'DGCR6L (85359)', 'DGCR8 (54487)', 'DGKA (1606)', 'DGKB (1607)', 'DGKD (8527)', 'DGKE (8526)', 'DGKG (1608)', 'DGKH (160851)', 'DGKI (9162)', 'DGKQ (1609)', 'DGKZ (8525)', 'DGLUCY (80017)', 'DGUOK (1716)', 'DHCR24 (1718)', 'DHCR7 (1717)', 'DHDDS (79947)', 'DHDH (27294)', 'DHFR (1719)', 'DHFR2 (200895)', 'DHH (50846)', 'DHODH (1723)', 'DHPS (1725)', 'DHRS1 (115817)', 'DHRS11 (79154)', 'DHRS12 (79758)', 'DHRS13 (147015)', 'DHRS2 (10202)', 'DHRS3 (9249)', 'DHRS4 (10901)', 'DHRS4L2 (317749)', 'DHRS7 (51635)', 'DHRS7B (25979)', 'DHRS7C (201140)', 'DHRS9 (10170)', 'DHTKD1 (55526)', 'DHX15 (1665)', 'DHX16 (8449)', 'DHX29 (54505)', 'DHX30 (22907)', 'DHX32 (55760)', 'DHX33 (56919)', 'DHX34 (9704)', 'DHX35 (60625)', 'DHX36 (170506)', 'DHX37 (57647)', 'DHX38 (9785)', 'DHX40 (79665)', 'DHX57 (90957)', 'DHX58 (79132)', 'DHX8 (1659)', 'DHX9 (1660)', 'DIABLO (56616)', 'DIAPH1 (1729)', 'DIAPH2 (1730)', 'DIAPH3 (81624)', 'DICER1 (23405)', 'DIDO1 (11083)', 'DIMT1 (27292)', 'DIO1 (1733)', 'DIO2 (1734)', 'DIO3 (1735)', 'DIP2A (23181)', 'DIP2B (57609)', 'DIP2C (22982)', 'DIPK1A (388650)', 'DIPK1B (138311)', 'DIPK1C (125704)', 'DIPK2A (205428)', 'DIPK2B (79742)', 'DIRAS1 (148252)', 'DIRAS2 (54769)', 'DIRAS3 (9077)', 'DIS3 (22894)', 'DIS3L (115752)', 'DIS3L2 (129563)', 'DISC1 (27185)', 'DISP1 (84976)', 'DISP2 (85455)', 'DISP3 (57540)', 'DIXDC1 (85458)', 'DKC1 (1736)', 'DKK1 (22943)', 'DKK2 (27123)', 'DKK3 (27122)', 'DKK4 (27121)', 'DKKL1 (27120)', 'DLAT (1737)', 'DLC1 (10395)', 'DLD (1738)', 'DLEC1 (9940)', 'DLEU7 (220107)', 'DLG1 (1739)', 'DLG2 (1740)', 'DLG3 (1741)', 'DLG4 (1742)', 'DLG5 (9231)', 'DLGAP1 (9229)', 'DLGAP2 (9228)', 'DLGAP3 (58512)', 'DLGAP4 (22839)', 'DLGAP5 (9787)', 'DLK1 (8788)', 'DLK2 (65989)', 'DLL1 (28514)', 'DLL3 (10683)', 'DLL4 (54567)', 'DLST (1743)', 'DLX1 (1745)', 'DLX2 (1746)', 'DLX3 (1747)', 'DLX4 (1748)', 'DLX5 (1749)', 'DLX6 (1750)', 'DMAC1 (90871)', 'DMAC2 (55101)', 'DMAC2L (27109)', 'DMAP1 (55929)', 'DMBT1 (1755)', 'DMBX1 (127343)', 'DMC1 (11144)', 'DMD (1756)', 'DMGDH (29958)', 'DMKN (93099)', 'DMP1 (1758)', 'DMPK (1760)', 'DMRT1 (1761)', 'DMRT2 (10655)', 'DMRT3 (58524)', 'DMRTA1 (63951)', 'DMRTA2 (63950)', 'DMRTB1 (63948)', 'DMRTC2 (63946)', 'DMTF1 (9988)', 'DMTN (2039)', 'DMWD (1762)', 'DMXL1 (1657)', 'DMXL2 (23312)', 'DNAAF1 (123872)', 'DNAAF10 (116143)', 'DNAAF11 (23639)', 'DNAAF2 (55172)', 'DNAAF3 (352909)', 'DNAAF4 (161582)', 'DNAAF5 (54919)', 'DNAAF6 (139212)', 'DNAAF8 (146562)', 'DNAAF9 (25943)', 'DNAH1 (25981)', 'DNAH10 (196385)', 'DNAH11 (8701)', 'DNAH12 (201625)', 'DNAH14 (127602)', 'DNAH2 (146754)', 'DNAH3 (55567)', 'DNAH5 (1767)', 'DNAH6 (1768)', 'DNAH7 (56171)', 'DNAH9 (1770)', 'DNAI1 (27019)', 'DNAI2 (64446)', 'DNAI3 (126820)', 'DNAI4 (79819)', 'DNAI7 (55259)', 'DNAJA1 (3301)', 'DNAJA2 (10294)', 'DNAJA3 (9093)', 'DNAJA4 (55466)', 'DNAJB1 (3337)', 'DNAJB11 (51726)', 'DNAJB12 (54788)', 'DNAJB13 (374407)', 'DNAJB14 (79982)', 'DNAJB2 (3300)', 'DNAJB4 (11080)', 'DNAJB5 (25822)', 'DNAJB6 (10049)', 'DNAJB7 (150353)', 'DNAJB8 (165721)', 'DNAJB9 (4189)', 'DNAJC1 (64215)', 'DNAJC10 (54431)', 'DNAJC11 (55735)', 'DNAJC12 (56521)', 'DNAJC13 (23317)', 'DNAJC14 (85406)', 'DNAJC15 (29103)', 'DNAJC16 (23341)', 'DNAJC17 (55192)', 'DNAJC18 (202052)', 'DNAJC19 (131118)', 'DNAJC2 (27000)', 'DNAJC21 (134218)', 'DNAJC22 (79962)', 'DNAJC24 (120526)', 'DNAJC25 (548645)', 'DNAJC27 (51277)', 'DNAJC28 (54943)', 'DNAJC3 (5611)', 'DNAJC30 (84277)', 'DNAJC4 (3338)', 'DNAJC5 (80331)', 'DNAJC5B (85479)', 'DNAJC5G (285126)', 'DNAJC6 (9829)', 'DNAJC7 (7266)', 'DNAJC8 (22826)', 'DNAJC9 (23234)', 'DNAL1 (83544)', 'DNAL4 (10126)', 'DNALI1 (7802)', 'DNASE1 (1773)', 'DNASE1L1 (1774)', 'DNASE1L2 (1775)', 'DNASE1L3 (1776)', 'DNASE2 (1777)', 'DNASE2B (58511)', 'DND1 (373863)', 'DNER (92737)', 'DNHD1 (144132)', 'DNLZ (728489)', 'DNM1 (1759)', 'DNM1L (10059)', 'DNM2 (1785)', 'DNM3 (26052)', 'DNMBP (23268)', 'DNMT1 (1786)', 'DNMT3A (1788)', 'DNMT3B (1789)', 'DNMT3L (29947)', 'DNPEP (23549)', 'DNPH1 (10591)', 'DNTT (1791)', 'DNTTIP1 (116092)', 'DNTTIP2 (30836)', 'DOC2A (8448)', 'DOCK10 (55619)', 'DOCK11 (139818)', 'DOCK2 (1794)', 'DOCK3 (1795)', 'DOCK4 (9732)', 'DOCK5 (80005)', 'DOCK6 (57572)', 'DOCK7 (85440)', 'DOCK8 (81704)', 'DOCK9 (23348)', 'DOHH (83475)', 'DOK1 (1796)', 'DOK2 (9046)', 'DOK3 (79930)', 'DOK4 (55715)', 'DOK5 (55816)', 'DOK6 (220164)', 'DOK7 (285489)', 'DOLK (22845)', 'DOLPP1 (57171)', 'DONSON (29980)', 'DOP1A (23033)', 'DOP1B (9980)', 'DOT1L (84444)', 'DPAGT1 (1798)', 'DPCD (25911)', 'DPEP1 (1800)', 'DPEP2 (64174)', 'DPEP3 (64180)', 'DPF1 (8193)', 'DPF2 (5977)', 'DPF3 (8110)', 'DPH1 (1801)', 'DPH2 (1802)', 'DPH3 (285381)', 'DPH5 (51611)', 'DPH6 (89978)', 'DPH7 (92715)', 'DPM1 (8813)', 'DPM2 (8818)', 'DPM3 (54344)', 'DPP10 (57628)', 'DPP3 (10072)', 'DPP4 (1803)', 'DPP7 (29952)', 'DPP8 (54878)', 'DPP9 (91039)', 'DPPA2 (151871)', 'DPPA3 (359787)', 'DPPA4 (55211)', 'DPPA5 (340168)', 'DPRX (503834)', 'DPT (1805)', 'DPY19L1 (23333)', 'DPY19L2 (283417)', 'DPY19L3 (147991)', 'DPY19L4 (286148)', 'DPY30 (84661)', 'DPYD (1806)', 'DPYS (1807)', 'DPYSL2 (1808)', 'DPYSL3 (1809)', 'DPYSL4 (10570)', 'DPYSL5 (56896)', 'DQX1 (165545)', 'DR1 (1810)', 'DRAM1 (55332)', 'DRAM2 (128338)', 'DRAP1 (10589)', 'DRAXIN (374946)', 'DRC1 (92749)', 'DRC12 (283152)', 'DRC3 (83450)', 'DRC7 (84229)', 'DRD1 (1812)', 'DRD2 (1813)', 'DRD3 (1814)', 'DRD4 (1815)', 'DRD5 (1816)', 'DRG1 (4733)', 'DRG2 (1819)', 'DRGX (644168)', 'DRICH1 (51233)', 'DROSHA (29102)', 'DRP2 (1821)', 'DSC1 (1823)', 'DSC2 (1824)', 'DSC3 (1825)', 'DSCAM (1826)', 'DSCAML1 (57453)', 'DSCC1 (79075)', 'DSE (29940)', 'DSEL (92126)', 'DSG1 (1828)', 'DSG2 (1829)', 'DSG3 (1830)', 'DSG4 (147409)', 'DSN1 (79980)', 'DSP (1832)', 'DSPP (1834)', 'DST (667)', 'DSTN (11034)', 'DSTYK (25778)', 'DTD1 (92675)', 'DTD2 (112487)', 'DTHD1 (401124)', 'DTL (51514)', 'DTNA (1837)', 'DTNB (1838)', 'DTNBP1 (84062)', 'DTWD1 (56986)', 'DTWD2 (285605)', 'DTX1 (1840)', 'DTX2 (113878)', 'DTX3 (196403)', 'DTX3L (151636)', 'DTX4 (23220)', 'DTYMK (1841)', 'DUOX1 (53905)', 'DUOX2 (50506)', 'DUOXA1 (90527)', 'DUOXA2 (405753)', 'DUS1L (64118)', 'DUS2 (54920)', 'DUS3L (56931)', 'DUS4L (11062)', 'DUSP1 (1843)', 'DUSP10 (11221)', 'DUSP12 (11266)', 'DUSP13B (51207)', 'DUSP14 (11072)', 'DUSP15 (128853)', 'DUSP16 (80824)', 'DUSP18 (150290)', 'DUSP19 (142679)', 'DUSP2 (1844)', 'DUSP21 (63904)', 'DUSP22 (56940)', 'DUSP23 (54935)', 'DUSP26 (78986)', 'DUSP28 (285193)', 'DUSP29 (338599)', 'DUSP3 (1845)', 'DUSP4 (1846)', 'DUSP5 (1847)', 'DUSP6 (1848)', 'DUSP7 (1849)', 'DUSP8 (1850)', 'DUSP9 (1852)', 'DUT (1854)', 'DUXA (503835)', 'DVL1 (1855)', 'DVL2 (1856)', 'DVL3 (1857)', 'DXO (1797)', 'DYDC1 (143241)', 'DYDC2 (84332)', 'DYM (54808)', 'DYNAP (284254)', 'DYNC1H1 (1778)', 'DYNC1I1 (1780)', 'DYNC1I2 (1781)', 'DYNC1LI1 (51143)', 'DYNC1LI2 (1783)', 'DYNC2H1 (79659)', 'DYNC2I1 (55112)', 'DYNC2I2 (89891)', 'DYNC2LI1 (51626)', 'DYNLL1 (8655)', 'DYNLL2 (140735)', 'DYNLRB1 (83658)', 'DYNLRB2 (83657)', 'DYNLT1 (6993)', 'DYNLT2 (6991)', 'DYNLT2B (255758)', 'DYNLT3 (6990)', 'DYNLT4 (343521)', 'DYNLT5 (200132)', 'DYRK1A (1859)', 'DYRK1B (9149)', 'DYRK2 (8445)', 'DYRK3 (8444)', 'DYRK4 (8798)', 'DYSF (8291)', 'DYTN (391475)', 'DZANK1 (55184)', 'DZIP1 (22873)', 'DZIP1L (199221)', 'DZIP3 (9666)', 'E2F1 (1869)', 'E2F2 (1870)', 'E2F3 (1871)', 'E2F4 (1874)', 'E2F5 (1875)', 'E2F6 (1876)', 'E2F7 (144455)', 'E2F8 (79733)', 'E4F1 (1877)', 'EAF1 (85403)', 'EAF2 (55840)', 'EAPP (55837)', 'EARS2 (124454)', 'EBAG9 (9166)', 'EBF1 (1879)', 'EBF2 (64641)', 'EBF3 (253738)', 'EBF4 (57593)', 'EBI3 (10148)', 'EBLN1 (340900)', 'EBLN2 (55096)', 'EBNA1BP2 (10969)', 'EBP (10682)', 'EBPL (84650)', 'ECD (11319)', 'ECE1 (1889)', 'ECE2 (9718)', 'ECEL1 (9427)', 'ECH1 (1891)', 'ECHDC1 (55862)', 'ECHDC2 (55268)', 'ECHDC3 (79746)', 'ECHS1 (1892)', 'ECI1 (1632)', 'ECI2 (10455)', 'ECM1 (1893)', 'ECM2 (1842)', 'ECPAS (23392)', 'ECRG4 (84417)', 'ECSIT (51295)', 'ECT2 (1894)', 'ECT2L (345930)', 'EDA (1896)', 'EDA2R (60401)', 'EDAR (10913)', 'EDARADD (128178)', 'EDC3 (80153)', 'EDC4 (23644)', 'EDDM3A (10876)', 'EDDM3B (64184)', 'EDEM1 (9695)', 'EDEM2 (55741)', 'EDEM3 (80267)', 'EDF1 (8721)', 'EDIL3 (10085)', 'EDN1 (1906)', 'EDN2 (1907)', 'EDN3 (1908)', 'EDNRA (1909)', 'EDNRB (1910)', 'EDRF1 (26098)', 'EEA1 (8411)', 'EED (8726)', 'EEF1A1 (1915)', 'EEF1A2 (1917)', 'EEF1AKMT1 (221143)', 'EEF1AKMT2 (399818)', 'EEF1AKMT3 (25895)', 'EEF1AKMT4 (110599564)', 'EEF1B2 (1933)', 'EEF1D (1936)', 'EEF1E1 (9521)', 'EEF1G (1937)', 'EEF2 (1938)', 'EEF2K (29904)', 'EEF2KMT (196483)', 'EEFSEC (60678)', 'EEIG1 (399665)', 'EEIG2 (284611)', 'EEPD1 (80820)', 'EFCAB11 (90141)', 'EFCAB12 (90288)', 'EFCAB13 (124989)', 'EFCAB14 (9813)', 'EFCAB2 (84288)', 'EFCAB3 (146779)', 'EFCAB5 (374786)', 'EFCAB6 (64800)', 'EFCAB7 (84455)', 'EFCAB9 (285588)', 'EFCC1 (79825)', 'EFEMP1 (2202)', 'EFEMP2 (30008)', 'EFHB (151651)', 'EFHC1 (114327)', 'EFHC2 (80258)', 'EFHD1 (80303)', 'EFHD2 (79180)', 'EFL1 (79631)', 'EFNA1 (1942)', 'EFNA2 (1943)', 'EFNA3 (1944)', 'EFNA4 (1945)', 'EFNA5 (1946)', 'EFNB1 (1947)', 'EFNB2 (1948)', 'EFNB3 (1949)', 'EFR3A (23167)', 'EFR3B (22979)', 'EFS (10278)', 'EFTUD2 (9343)', 'EGF (1950)', 'EGFL6 (25975)', 'EGFL7 (51162)', 'EGFL8 (80864)', 'EGFLAM (133584)', 'EGFR (1956)', 'EGLN1 (54583)', 'EGLN2 (112398)', 'EGLN3 (112399)', 'EGR1 (1958)', 'EGR2 (1959)', 'EGR3 (1960)', 'EGR4 (1961)', 'EHBP1 (23301)', 'EHBP1L1 (254102)', 'EHD1 (10938)', 'EHD2 (30846)', 'EHD3 (30845)', 'EHD4 (30844)', 'EHF (26298)', 'EHHADH (1962)', 'EHMT1 (79813)', 'EHMT2 (10919)', 'EID1 (23741)', 'EID2 (163126)', 'EID2B (126272)', 'EID3 (493861)', 'EIF1 (10209)', 'EIF1AD (84285)', 'EIF1AX (1964)', 'EIF1AY (9086)', 'EIF1B (10289)', 'EIF2A (83939)', 'EIF2AK1 (27102)', 'EIF2AK2 (5610)', 'EIF2AK3 (9451)', 'EIF2AK4 (440275)', 'EIF2B1 (1967)', 'EIF2B2 (8892)', 'EIF2B3 (8891)', 'EIF2B4 (8890)', 'EIF2B5 (8893)', 'EIF2D (1939)', 'EIF2S1 (1965)', 'EIF2S2 (8894)', 'EIF2S3 (1968)', 'EIF3A (8661)', 'EIF3B (8662)', 'EIF3D (8664)', 'EIF3E (3646)', 'EIF3F (8665)', 'EIF3G (8666)', 'EIF3H (8667)', 'EIF3I (8668)', 'EIF3J (8669)', 'EIF3K (27335)', 'EIF3L (51386)', 'EIF3M (10480)', 'EIF4A1 (1973)', 'EIF4A2 (1974)', 'EIF4A3 (9775)', 'EIF4B (1975)', 'EIF4E (1977)', 'EIF4E1B (253314)', 'EIF4E2 (9470)', 'EIF4E3 (317649)', 'EIF4EBP1 (1978)', 'EIF4EBP2 (1979)', 'EIF4EBP3 (8637)', 'EIF4ENIF1 (56478)', 'EIF4G1 (1981)', 'EIF4G2 (1982)', 'EIF4G3 (8672)', 'EIF4H (7458)', 'EIF5 (1983)', 'EIF5A (1984)', 'EIF5A2 (56648)', 'EIF5AL1 (143244)', 'EIF5B (9669)', 'EIF6 (3692)', 'EIPR1 (7260)', 'ELAC1 (55520)', 'ELAC2 (60528)', 'ELANE (1991)', 'ELAPOR1 (57535)', 'ELAPOR2 (222223)', 'ELAVL1 (1994)', 'ELAVL2 (1993)', 'ELAVL3 (1995)', 'ELAVL4 (1996)', 'ELF1 (1997)', 'ELF2 (1998)', 'ELF3 (1999)', 'ELF4 (2000)', 'ELF5 (2001)', 'ELFN1 (392617)', 'ELFN2 (114794)', 'ELK1 (2002)', 'ELK3 (2004)', 'ELK4 (2005)', 'ELL (8178)', 'ELL2 (22936)', 'ELL3 (80237)', 'ELMO1 (9844)', 'ELMO2 (63916)', 'ELMO3 (79767)', 'ELMOD1 (55531)', 'ELMOD2 (255520)', 'ELMOD3 (84173)', 'ELN (2006)', 'ELOA (6924)', 'ELOA2 (51224)', 'ELOB (6923)', 'ELOC (6921)', 'ELOF1 (84337)', 'ELOVL1 (64834)', 'ELOVL2 (54898)', 'ELOVL3 (83401)', 'ELOVL4 (6785)', 'ELOVL5 (60481)', 'ELOVL6 (79071)', 'ELOVL7 (79993)', 'ELP1 (8518)', 'ELP2 (55250)', 'ELP3 (55140)', 'ELP4 (26610)', 'ELP5 (23587)', 'ELP6 (54859)', 'ELSPBP1 (64100)', 'EMB (133418)', 'EMC1 (23065)', 'EMC10 (284361)', 'EMC2 (9694)', 'EMC3 (55831)', 'EMC4 (51234)', 'EMC6 (83460)', 'EMC7 (56851)', 'EMC8 (10328)', 'EMC9 (51016)', 'EMCN (51705)', 'EMD (2010)', 'EME1 (146956)', 'EME2 (197342)', 'EMID1 (129080)', 'EMILIN1 (11117)', 'EMILIN2 (84034)', 'EMILIN3 (90187)', 'EML1 (2009)', 'EML2 (24139)', 'EML3 (256364)', 'EML4 (27436)', 'EML5 (161436)', 'EML6 (400954)', 'EMP1 (2012)', 'EMP2 (2013)', 'EMP3 (2014)', 'EMSY (56946)', 'EMX1 (2016)', 'EMX2 (2018)', 'EN1 (2019)', 'EN2 (2020)', 'ENAH (55740)', 'ENAM (10117)', 'ENC1 (8507)', 'ENDOD1 (23052)', 'ENDOG (2021)', 'ENDOU (8909)', 'ENDOV (284131)', 'ENG (2022)', 'ENGASE (64772)', 'ENHO (375704)', 'ENKD1 (84080)', 'ENKUR (219670)', 'ENO1 (2023)', 'ENO2 (2026)', 'ENO3 (2027)', 'ENOPH1 (58478)', 'ENOSF1 (55556)', 'ENOX1 (55068)', 'ENOX2 (10495)', 'ENPEP (2028)', 'ENPP1 (5167)', 'ENPP2 (5168)', 'ENPP3 (5169)', 'ENPP4 (22875)', 'ENPP5 (59084)', 'ENPP6 (133121)', 'ENPP7 (339221)', 'ENSA (2029)', 'ENTHD1 (150350)', 'ENTPD1 (953)', 'ENTPD2 (954)', 'ENTPD3 (956)', 'ENTPD4 (9583)', 'ENTPD5 (957)', 'ENTPD6 (955)', 'ENTPD7 (57089)', 'ENTPD8 (377841)', 'ENTR1 (10807)', 'ENTREP1 (9413)', 'ENTREP2 (23359)', 'ENTREP3 (10712)', 'ENY2 (56943)', 'EOGT (285203)', 'EOLA1 (91966)', 'EOLA2 (541578)', 'EOMES (8320)', 'EP300 (2033)', 'EP400 (57634)', 'EPAS1 (2034)', 'EPB41 (2035)', 'EPB41L1 (2036)', 'EPB41L2 (2037)', 'EPB41L3 (23136)', 'EPB41L4A (64097)', 'EPB41L4B (54566)', 'EPB41L5 (57669)', 'EPB42 (2038)', 'EPC1 (80314)', 'EPC2 (26122)', 'EPCAM (4072)', 'EPCIP (56245)', 'EPDR1 (54749)', 'EPG5 (57724)', 'EPGN (255324)', 'EPHA1 (2041)', 'EPHA10 (284656)', 'EPHA2 (1969)', 'EPHA3 (2042)', 'EPHA4 (2043)', 'EPHA5 (2044)', 'EPHA6 (285220)', 'EPHA7 (2045)', 'EPHA8 (2046)', 'EPHB1 (2047)', 'EPHB2 (2048)', 'EPHB3 (2049)', 'EPHB4 (2050)', 'EPHB6 (2051)', 'EPHX1 (2052)', 'EPHX2 (2053)', 'EPHX3 (79852)', 'EPHX4 (253152)', 'EPM2A (7957)', 'EPM2AIP1 (9852)', 'EPN1 (29924)', 'EPN2 (22905)', 'EPN3 (55040)', 'EPO (2056)', 'EPOP (100170841)', 'EPOR (2057)', 'EPPIN (57119)', 'EPRS1 (2058)', 'EPS15 (2060)', 'EPS15L1 (58513)', 'EPS8 (2059)', 'EPS8L1 (54869)', 'EPS8L2 (64787)', 'EPS8L3 (79574)', 'EPSTI1 (94240)', 'EPX (8288)', 'EPYC (1833)', 'EQTN (54586)', 'ERAL1 (26284)', 'ERAP1 (51752)', 'ERAP2 (64167)', 'ERAS (3266)', 'ERBB2 (2064)', 'ERBB3 (2065)', 'ERBB4 (2066)', 'ERBIN (55914)', 'ERC1 (23085)', 'ERC2 (26059)', 'ERCC1 (2067)', 'ERCC2 (2068)', 'ERCC3 (2071)', 'ERCC4 (2072)', 'ERCC5 (2073)', 'ERCC6 (2074)', 'ERCC6L (54821)', 'ERCC6L2 (375748)', 'ERCC8 (1161)', 'EREG (2069)', 'ERF (2077)', 'ERG (2078)', 'ERG28 (11161)', 'ERGIC1 (57222)', 'ERGIC2 (51290)', 'ERGIC3 (51614)', 'ERH (2079)', 'ERI1 (90459)', 'ERI2 (112479)', 'ERI3 (79033)', 'ERICH1 (157697)', 'ERICH3 (127254)', 'ERICH4 (100170765)', 'ERICH5 (203111)', 'ERICH6 (131831)', 'ERICH6B (220081)', 'ERLEC1 (27248)', 'ERLIN1 (10613)', 'ERLIN2 (11160)', 'ERMAP (114625)', 'ERMARD (55780)', 'ERMN (57471)', 'ERMP1 (79956)', 'ERN1 (2081)', 'ERN2 (10595)', 'ERO1A (30001)', 'ERO1B (56605)', 'ERP27 (121506)', 'ERP29 (10961)', 'ERP44 (23071)', 'ERRFI1 (54206)', 'ERV3-1 (2086)', 'ERVFRD-1 (405754)', 'ERVW-1 (30816)', 'ESAM (90952)', 'ESCO1 (114799)', 'ESCO2 (157570)', 'ESD (2098)', 'ESF1 (51575)', 'ESM1 (11082)', 'ESPL1 (9700)', 'ESPN (83715)', 'ESPNL (339768)', 'ESR1 (2099)', 'ESR2 (2100)', 'ESRP1 (54845)', 'ESRP2 (80004)', 'ESRRA (2101)', 'ESRRB (2103)', 'ESRRG (2104)', 'ESS2 (8220)', 'ESX1 (80712)', 'ESYT1 (23344)', 'ESYT2 (57488)', 'ESYT3 (83850)', 'ETAA1 (54465)', 'ETF1 (2107)', 'ETFA (2108)', 'ETFB (2109)', 'ETFBKMT (254013)', 'ETFDH (2110)', 'ETFRF1 (144363)', 'ETHE1 (23474)', 'ETNK1 (55500)', 'ETNK2 (55224)', 'ETNPPL (64850)', 'ETS1 (2113)', 'ETS2 (2114)', 'ETV1 (2115)', 'ETV2 (2116)', 'ETV3 (2117)', 'ETV3L (440695)', 'ETV4 (2118)', 'ETV5 (2119)', 'ETV6 (2120)', 'ETV7 (51513)', 'EVA1A (84141)', 'EVA1B (55194)', 'EVA1C (59271)', 'EVC (2121)', 'EVC2 (132884)', 'EVI2A (2123)', 'EVI2B (2124)', 'EVI5 (7813)', 'EVI5L (115704)', 'EVL (51466)', 'EVPL (2125)', 'EVPLL (645027)', 'EVX1 (2128)', 'EVX2 (344191)', 'EWSR1 (2130)', 'EXD1 (161829)', 'EXD2 (55218)', 'EXD3 (54932)', 'EXO1 (9156)', 'EXO5 (64789)', 'EXOC1 (55763)', 'EXOC2 (55770)', 'EXOC3 (11336)', 'EXOC3L1 (283849)', 'EXOC3L2 (90332)', 'EXOC3L4 (91828)', 'EXOC4 (60412)', 'EXOC5 (10640)', 'EXOC6 (54536)', 'EXOC6B (23233)', 'EXOC7 (23265)', 'EXOC8 (149371)', 'EXOG (9941)', 'EXOSC1 (51013)', 'EXOSC10 (5394)', 'EXOSC2 (23404)', 'EXOSC3 (51010)', 'EXOSC4 (54512)', 'EXOSC5 (56915)', 'EXOSC6 (118460)', 'EXOSC7 (23016)', 'EXOSC8 (11340)', 'EXOSC9 (5393)', 'EXPH5 (23086)', 'EXT1 (2131)', 'EXT2 (2132)', 'EXTL1 (2134)', 'EXTL2 (2135)', 'EXTL3 (2137)', 'EYA1 (2138)', 'EYA2 (2139)', 'EYA3 (2140)', 'EYA4 (2070)', 'EYS (346007)', 'EZH1 (2145)', 'EZH2 (2146)', 'EZR (7430)', 'F10 (2159)', 'F11 (2160)', 'F11R (50848)', 'F12 (2161)', 'F13A1 (2162)', 'F13B (2165)', 'F2 (2147)', 'F2R (2149)', 'F2RL1 (2150)', 'F2RL2 (2151)', 'F2RL3 (9002)', 'F3 (2152)', 'F5 (2153)', 'F7 (2155)', 'F8 (2157)', 'F9 (2158)', 'FA2H (79152)', 'FAAH (2166)', 'FAAH2 (158584)', 'FAAP100 (80233)', 'FAAP20 (199990)', 'FAAP24 (91442)', 'FABP1 (2168)', 'FABP12 (646486)', 'FABP2 (2169)', 'FABP3 (2170)', 'FABP4 (2167)', 'FABP6 (2172)', 'FABP7 (2173)', 'FADD (8772)', 'FADS1 (3992)', 'FADS2 (9415)', 'FADS3 (3995)', 'FADS6 (283985)', 'FAF1 (11124)', 'FAF2 (23197)', 'FAH (2184)', 'FAHD1 (81889)', 'FAHD2A (51011)', 'FAHD2B (151313)', 'FAIM (55179)', 'FAIM2 (23017)', 'FAM107A (11170)', 'FAM107B (83641)', 'FAM110A (83541)', 'FAM110B (90362)', 'FAM110C (642273)', 'FAM110D (79927)', 'FAM111A (63901)', 'FAM111B (374393)', 'FAM114A1 (92689)', 'FAM114A2 (10827)', 'FAM117A (81558)', 'FAM117B (150864)', 'FAM118A (55007)', 'FAM118B (79607)', 'FAM120A (23196)', 'FAM120AOS (158293)', 'FAM120B (84498)', 'FAM120C (54954)', 'FAM124A (220108)', 'FAM124B (79843)', 'FAM131A (131408)', 'FAM131B (9715)', 'FAM131C (348487)', 'FAM133A (286499)', 'FAM133B (257415)', 'FAM135A (57579)', 'FAM135B (51059)', 'FAM136A (84908)', 'FAM13A (10144)', 'FAM13B (51306)', 'FAM13C (220965)', 'FAM149A (25854)', 'FAM149B1 (317662)', 'FAM151A (338094)', 'FAM151B (167555)', 'FAM153A (285596)', 'FAM161A (84140)', 'FAM161B (145483)', 'FAM162A (26355)', 'FAM162B (221303)', 'FAM163A (148753)', 'FAM163B (642968)', 'FAM167A (83648)', 'FAM167B (84734)', 'FAM168A (23201)', 'FAM168B (130074)', 'FAM169A (26049)', 'FAM170A (340069)', 'FAM170B (170370)', 'FAM171A1 (221061)', 'FAM171A2 (284069)', 'FAM171B (165215)', 'FAM174A (345757)', 'FAM174B (400451)', 'FAM174C (55009)', 'FAM177A1 (283635)', 'FAM177B (400823)', 'FAM178B (51252)', 'FAM180A (389558)', 'FAM181A (90050)', 'FAM181B (220382)', 'FAM184A (79632)', 'FAM184B (27146)', 'FAM185A (222234)', 'FAM186A (121006)', 'FAM186B (84070)', 'FAM187B (148109)', 'FAM193A (8603)', 'FAM193B (54540)', 'FAM199X (139231)', 'FAM200A (221786)', 'FAM200B (285550)', 'FAM200C (63920)', 'FAM204A (63877)', 'FAM209A (200232)', 'FAM209B (388799)', 'FAM20A (54757)', 'FAM20B (9917)', 'FAM20C (56975)', 'FAM210A (125228)', 'FAM210B (116151)', 'FAM216A (29902)', 'FAM216B (144809)', 'FAM217A (222826)', 'FAM217B (63939)', 'FAM219A (203259)', 'FAM219B (57184)', 'FAM220A (84792)', 'FAM221A (340277)', 'FAM221B (392307)', 'FAM222A (84915)', 'FAM222B (55731)', 'FAM227B (196951)', 'FAM228A (653140)', 'FAM229A (100128071)', 'FAM229B (619208)', 'FAM234A (83986)', 'FAM234B (57613)', 'FAM241A (132720)', 'FAM241B (219738)', 'FAM24A (118670)', 'FAM24B (196792)', 'FAM25A (643161)', 'FAM32A (26017)', 'FAM3A (60343)', 'FAM3B (54097)', 'FAM3C (10447)', 'FAM3D (131177)', 'FAM43A (131583)', 'FAM43B (163933)', 'FAM47A (158724)', 'FAM47B (170062)', 'FAM47C (442444)', 'FAM47E (100129583)', 'FAM50A (9130)', 'FAM50B (26240)', 'FAM53A (152877)', 'FAM53B (9679)', 'FAM53C (51307)', 'FAM76A (199870)', 'FAM76B (143684)', 'FAM78A (286336)', 'FAM78B (149297)', 'FAM81A (145773)', 'FAM81B (153643)', 'FAM83A (84985)', 'FAM83B (222584)', 'FAM83C (128876)', 'FAM83D (81610)', 'FAM83E (54854)', 'FAM83F (113828)', 'FAM83G (644815)', 'FAM83H (286077)', 'FAM86B2 (653333)', 'FAM89A (375061)', 'FAM89B (23625)', 'FAM8A1 (51439)', 'FAM90A1 (55138)', 'FAM91A1 (157769)', 'FAM98A (25940)', 'FAM98B (283742)', 'FAM98C (147965)', 'FAM9A (171482)', 'FAM9B (171483)', 'FAM9C (171484)', 'FAN1 (22909)', 'FANCA (2175)', 'FANCB (2187)', 'FANCC (2176)', 'FANCD2 (2177)', 'FANCD2OS (115795)', 'FANCE (2178)', 'FANCF (2188)', 'FANCG (2189)', 'FANCI (55215)', 'FANCL (55120)', 'FANCM (57697)', 'FANK1 (92565)', 'FAP (2191)', 'FAR1 (84188)', 'FAR2 (55711)', 'FARP1 (10160)', 'FARP2 (9855)', 'FARS2 (10667)', 'FARSA (2193)', 'FARSB (10056)', 'FAS (355)', 'FASLG (356)', 'FASN (2194)', 'FASTK (10922)', 'FASTKD1 (79675)', 'FASTKD2 (22868)', 'FASTKD3 (79072)', 'FASTKD5 (60493)', 'FAT1 (2195)', 'FAT2 (2196)', 'FAT4 (79633)', 'FATE1 (89885)', 'FAU (2197)', 'FAXC (84553)', 'FAXDC2 (10826)', 'FBF1 (85302)', 'FBH1 (84893)', 'FBL (2091)', 'FBLIM1 (54751)', 'FBLN1 (2192)', 'FBLN2 (2199)', 'FBLN5 (10516)', 'FBLN7 (129804)', 'FBN1 (2200)', 'FBN2 (2201)', 'FBN3 (84467)', 'FBP1 (2203)', 'FBP2 (8789)', 'FBRSL1 (57666)', 'FBXL12 (54850)', 'FBXL13 (222235)', 'FBXL14 (144699)', 'FBXL15 (79176)', 'FBXL16 (146330)', 'FBXL17 (64839)', 'FBXL18 (80028)', 'FBXL19 (54620)', 'FBXL2 (25827)', 'FBXL20 (84961)', 'FBXL22 (283807)', 'FBXL3 (26224)', 'FBXL4 (26235)', 'FBXL5 (26234)', 'FBXL6 (26233)', 'FBXL7 (23194)', 'FBXL8 (55336)', 'FBXO10 (26267)', 'FBXO11 (80204)', 'FBXO15 (201456)', 'FBXO16 (157574)', 'FBXO17 (115290)', 'FBXO2 (26232)', 'FBXO21 (23014)', 'FBXO22 (26263)', 'FBXO24 (26261)', 'FBXO25 (26260)', 'FBXO27 (126433)', 'FBXO28 (23219)', 'FBXO3 (26273)', 'FBXO30 (84085)', 'FBXO31 (79791)', 'FBXO32 (114907)', 'FBXO33 (254170)', 'FBXO34 (55030)', 'FBXO36 (130888)', 'FBXO38 (81545)', 'FBXO39 (162517)', 'FBXO4 (26272)', 'FBXO40 (51725)', 'FBXO41 (150726)', 'FBXO42 (54455)', 'FBXO43 (286151)', 'FBXO44 (93611)', 'FBXO45 (200933)', 'FBXO46 (23403)', 'FBXO47 (494188)', 'FBXO48 (554251)', 'FBXO5 (26271)', 'FBXO6 (26270)', 'FBXO7 (25793)', 'FBXO8 (26269)', 'FBXO9 (26268)', 'FBXW10 (10517)', 'FBXW10B (374286)', 'FBXW11 (23291)', 'FBXW12 (285231)', 'FBXW2 (26190)', 'FBXW4 (6468)', 'FBXW5 (54461)', 'FBXW7 (55294)', 'FBXW8 (26259)', 'FBXW9 (84261)', 'FCAMR (83953)', 'FCAR (2204)', 'FCER1A (2205)', 'FCER1G (2207)', 'FCER2 (2208)', 'FCF1 (51077)', 'FCGR1A (2209)', 'FCGR2A (2212)', 'FCGR2B (2213)', 'FCGR3A (2214)', 'FCGR3B (2215)', 'FCGRT (2217)', 'FCHO1 (23149)', 'FCHO2 (115548)', 'FCHSD1 (89848)', 'FCHSD2 (9873)', 'FCMR (9214)', 'FCN1 (2219)', 'FCN2 (2220)', 'FCN3 (8547)', 'FCRL1 (115350)', 'FCRL2 (79368)', 'FCRL3 (115352)', 'FCRL4 (83417)', 'FCRL5 (83416)', 'FCRL6 (343413)', 'FCRLA (84824)', 'FCRLB (127943)', 'FCSK (197258)', 'FDCSP (260436)', 'FDFT1 (2222)', 'FDPS (2224)', 'FDX1 (2230)', 'FDX2 (112812)', 'FDXACB1 (91893)', 'FDXR (2232)', 'FECH (2235)', 'FEM1A (55527)', 'FEM1B (10116)', 'FEM1C (56929)', 'FEN1 (2237)', 'FER (2241)', 'FER1L6 (654463)', 'FERD3L (222894)', 'FERMT1 (55612)', 'FERMT2 (10979)', 'FERMT3 (83706)', 'FERRY3 (57102)', 'FES (2242)', 'FETUB (26998)', 'FEV (54738)', 'FEZ1 (9638)', 'FEZ2 (9637)', 'FEZF1 (389549)', 'FEZF2 (55079)', 'FFAR1 (2864)', 'FFAR2 (2867)', 'FFAR4 (338557)', 'FGA (2243)', 'FGB (2244)', 'FGD1 (2245)', 'FGD2 (221472)', 'FGD3 (89846)', 'FGD4 (121512)', 'FGD5 (152273)', 'FGD6 (55785)', 'FGF1 (2246)', 'FGF10 (2255)', 'FGF11 (2256)', 'FGF12 (2257)', 'FGF13 (2258)', 'FGF14 (2259)', 'FGF17 (8822)', 'FGF18 (8817)', 'FGF19 (9965)', 'FGF2 (2247)', 'FGF20 (26281)', 'FGF21 (26291)', 'FGF22 (27006)', 'FGF23 (8074)', 'FGF3 (2248)', 'FGF4 (2249)', 'FGF5 (2250)', 'FGF6 (2251)', 'FGF7 (2252)', 'FGF8 (2253)', 'FGF9 (2254)', 'FGFBP1 (9982)', 'FGFBP2 (83888)', 'FGFBP3 (143282)', 'FGFR1 (2260)', 'FGFR1OP2 (26127)', 'FGFR2 (2263)', 'FGFR3 (2261)', 'FGFR4 (2264)', 'FGFRL1 (53834)', 'FGG (2266)', 'FGGY (55277)', 'FGL1 (2267)', 'FGL2 (10875)', 'FGR (2268)', 'FH (2271)', 'FHDC1 (85462)', 'FHIP1A (729830)', 'FHIP1B (84067)', 'FHIP2A (57700)', 'FHIP2B (64760)', 'FHIT (2272)', 'FHL1 (2273)', 'FHL2 (2274)', 'FHL3 (2275)', 'FHL5 (9457)', 'FHOD1 (29109)', 'FHOD3 (80206)', 'FIBCD1 (84929)', 'FIBIN (387758)', 'FIBP (9158)', 'FICD (11153)', 'FIG4 (9896)', 'FIGLA (344018)', 'FIGN (55137)', 'FIGNL1 (63979)', 'FILIP1 (27145)', 'FILIP1L (11259)', 'FIP1L1 (81608)', 'FIRRM (55732)', 'FIS1 (51024)', 'FITM1 (161247)', 'FITM2 (128486)', 'FIZ1 (84922)', 'FJX1 (24147)', 'FKBP10 (60681)', 'FKBP11 (51303)', 'FKBP14 (55033)', 'FKBP15 (23307)', 'FKBP1A (2280)', 'FKBP1B (2281)', 'FKBP2 (2286)', 'FKBP3 (2287)', 'FKBP4 (2288)', 'FKBP5 (2289)', 'FKBP6 (8468)', 'FKBP7 (51661)', 'FKBP8 (23770)', 'FKBP9 (11328)', 'FKBPL (63943)', 'FKRP (79147)', 'FKTN (2218)', 'FLACC1 (130540)', 'FLAD1 (80308)', 'FLCN (201163)', 'FLG (2312)', 'FLG2 (388698)', 'FLI1 (2313)', 'FLII (2314)', 'FLNA (2316)', 'FLNB (2317)', 'FLNC (2318)', 'FLOT1 (10211)', 'FLOT2 (2319)', 'FLRT1 (23769)', 'FLRT2 (23768)', 'FLRT3 (23767)', 'FLT1 (2321)', 'FLT3 (2322)', 'FLT3LG (2323)', 'FLT4 (2324)', 'FLVCR1 (28982)', 'FLVCR2 (55640)', 'FLYWCH1 (84256)', 'FLYWCH2 (114984)', 'FMC1 (154791)', 'FMN1 (342184)', 'FMN2 (56776)', 'FMNL1 (752)', 'FMNL2 (114793)', 'FMNL3 (91010)', 'FMO1 (2326)', 'FMO2 (2327)', 'FMO3 (2328)', 'FMO4 (2329)', 'FMO5 (2330)', 'FMOD (2331)', 'FMR1 (2332)', 'FMR1NB (158521)', 'FN1 (2335)', 'FN3K (64122)', 'FN3KRP (79672)', 'FNBP1 (23048)', 'FNBP1L (54874)', 'FNBP4 (23360)', 'FNDC1 (84624)', 'FNDC10 (643988)', 'FNDC11 (79025)', 'FNDC3A (22862)', 'FNDC3B (64778)', 'FNDC4 (64838)', 'FNDC5 (252995)', 'FNDC7 (163479)', 'FNDC8 (54752)', 'FNDC9 (408263)', 'FNIP1 (96459)', 'FNIP2 (57600)', 'FNTA (2339)', 'FNTB (2342)', 'FOCAD (54914)', 'FOLH1 (2346)', 'FOLR1 (2348)', 'FOLR2 (2350)', 'FOS (2353)', 'FOSB (2354)', 'FOSL1 (8061)', 'FOSL2 (2355)', 'FOXA1 (3169)', 'FOXA2 (3170)', 'FOXA3 (3171)', 'FOXB1 (27023)', 'FOXB2 (442425)', 'FOXC1 (2296)', 'FOXC2 (2303)', 'FOXD2 (2306)', 'FOXD3 (27022)', 'FOXD4 (2298)', 'FOXD4L5 (653427)', 'FOXE1 (2304)', 'FOXE3 (2301)', 'FOXF1 (2294)', 'FOXF2 (2295)', 'FOXG1 (2290)', 'FOXH1 (8928)', 'FOXI1 (2299)', 'FOXI2 (399823)', 'FOXJ1 (2302)', 'FOXJ2 (55810)', 'FOXJ3 (22887)', 'FOXK1 (221937)', 'FOXK2 (3607)', 'FOXL1 (2300)', 'FOXL2 (668)', 'FOXL2NB (401089)', 'FOXM1 (2305)', 'FOXN1 (8456)', 'FOXN2 (3344)', 'FOXN3 (1112)', 'FOXN4 (121643)', 'FOXO1 (2308)', 'FOXO3 (2309)', 'FOXO3B (2310)', 'FOXO4 (4303)', 'FOXP1 (27086)', 'FOXP2 (93986)', 'FOXP3 (50943)', 'FOXP4 (116113)', 'FOXQ1 (94234)', 'FOXR1 (283150)', 'FOXR2 (139628)', 'FOXRED1 (55572)', 'FOXRED2 (80020)', 'FOXS1 (2307)', 'FPGS (2356)', 'FPGT (8790)', 'FPR1 (2357)', 'FPR2 (2358)', 'FPR3 (2359)', 'FRA10AC1 (118924)', 'FRAS1 (80144)', 'FRAT1 (10023)', 'FRAT2 (23401)', 'FREM1 (158326)', 'FREM2 (341640)', 'FREM3 (166752)', 'FREY1 (143678)', 'FRG1 (2483)', 'FRG2B (441581)', 'FRG2C (100288801)', 'FRK (2444)', 'FRMD1 (79981)', 'FRMD3 (257019)', 'FRMD4A (55691)', 'FRMD4B (23150)', 'FRMD5 (84978)', 'FRMD6 (122786)', 'FRMD7 (90167)', 'FRMD8 (83786)', 'FRMPD1 (22844)', 'FRMPD2 (143162)', 'FRMPD4 (9758)', 'FRRS1 (391059)', 'FRRS1L (23732)', 'FRS2 (10818)', 'FRS3 (10817)', 'FRY (10129)', 'FRYL (285527)', 'FRZB (2487)', 'FSBP (100861412)', 'FSCB (84075)', 'FSCN1 (6624)', 'FSCN2 (25794)', 'FSCN3 (29999)', 'FSD1 (79187)', 'FSD1L (83856)', 'FSD2 (123722)', 'FSHB (2488)', 'FSHR (2492)', 'FSIP1 (161835)', 'FSIP2 (401024)', 'FST (10468)', 'FSTL1 (11167)', 'FSTL3 (10272)', 'FSTL4 (23105)', 'FSTL5 (56884)', 'FTCD (10841)', 'FTH1 (2495)', 'FTHL17 (53940)', 'FTL (2512)', 'FTMT (94033)', 'FTO (79068)', 'FTSJ1 (24140)', 'FTSJ3 (117246)', 'FUBP1 (8880)', 'FUBP3 (8939)', 'FUCA1 (2517)', 'FUCA2 (2519)', 'FUNDC1 (139341)', 'FUNDC2 (65991)', 'FUOM (282969)', 'FURIN (5045)', 'FUS (2521)', 'FUT1 (2523)', 'FUT10 (84750)', 'FUT11 (170384)', 'FUT2 (2524)', 'FUT3 (2525)', 'FUT4 (2526)', 'FUT5 (2527)', 'FUT6 (2528)', 'FUT7 (2529)', 'FUT8 (2530)', 'FUT9 (10690)', 'FUZ (80199)', 'FXN (2395)', 'FXR1 (8087)', 'FXR2 (9513)', 'FXYD1 (5348)', 'FXYD2 (486)', 'FXYD3 (5349)', 'FXYD4 (53828)', 'FXYD5 (53827)', 'FXYD6 (53826)', 'FXYD7 (53822)', 'FYB1 (2533)', 'FYB2 (199920)', 'FYCO1 (79443)', 'FYN (2534)', 'FYTTD1 (84248)', 'FZD1 (8321)', 'FZD10 (11211)', 'FZD2 (2535)', 'FZD3 (7976)', 'FZD4 (8322)', 'FZD5 (7855)', 'FZD6 (8323)', 'FZD7 (8324)', 'FZD8 (8325)', 'FZD9 (8326)', 'FZR1 (51343)', 'G0S2 (50486)', 'G2E3 (55632)', 'G3BP1 (10146)', 'G3BP2 (9908)', 'G6PC1 (2538)', 'G6PC2 (57818)', 'G6PC3 (92579)', 'G6PD (2539)', 'GAA (2548)', 'GAB1 (2549)', 'GAB2 (9846)', 'GAB3 (139716)', 'GAB4 (128954)', 'GABARAP (11337)', 'GABARAPL1 (23710)', 'GABARAPL2 (11345)', 'GABBR1 (2550)', 'GABBR2 (9568)', 'GABPA (2551)', 'GABPB1 (2553)', 'GABPB2 (126626)', 'GABRA1 (2554)', 'GABRA2 (2555)', 'GABRA3 (2556)', 'GABRA4 (2557)', 'GABRA5 (2558)', 'GABRA6 (2559)', 'GABRB1 (2560)', 'GABRB2 (2561)', 'GABRB3 (2562)', 'GABRD (2563)', 'GABRE (2564)', 'GABRG1 (2565)', 'GABRG2 (2566)', 'GABRG3 (2567)', 'GABRP (2568)', 'GABRQ (55879)', 'GABRR1 (2569)', 'GABRR2 (2570)', 'GAD1 (2571)', 'GAD2 (2572)', 'GADD45A (1647)', 'GADD45B (4616)', 'GADD45G (10912)', 'GADD45GIP1 (90480)', 'GADL1 (339896)', 'GAGE12H (729442)', 'GAK (2580)', 'GAL (51083)', 'GAL3ST1 (9514)', 'GAL3ST2 (64090)', 'GAL3ST3 (89792)', 'GAL3ST4 (79690)', 'GALC (2581)', 'GALE (2582)', 'GALK1 (2584)', 'GALK2 (2585)', 'GALM (130589)', 'GALNS (2588)', 'GALNT1 (2589)', 'GALNT10 (55568)', 'GALNT11 (63917)', 'GALNT12 (79695)', 'GALNT13 (114805)', 'GALNT14 (79623)', 'GALNT15 (117248)', 'GALNT16 (57452)', 'GALNT17 (64409)', 'GALNT18 (374378)', 'GALNT2 (2590)', 'GALNT3 (2591)', 'GALNT4 (8693)', 'GALNT5 (11227)', 'GALNT6 (11226)', 'GALNT7 (51809)', 'GALNT8 (26290)', 'GALNT9 (50614)', 'GALNTL5 (168391)', 'GALNTL6 (442117)', 'GALP (85569)', 'GALR1 (2587)', 'GALR2 (8811)', 'GALR3 (8484)', 'GALT (2592)', 'GAMT (2593)', 'GAN (8139)', 'GANAB (23193)', 'GANC (2595)', 'GAP43 (2596)', 'GAPDH (2597)', 'GAPDHS (26330)', 'GAPT (202309)', 'GAPVD1 (26130)', 'GAR1 (54433)', 'GAREM1 (64762)', 'GAREM2 (150946)', 'GARIN1A (346653)', 'GARIN1B (84691)', 'GARIN2 (161142)', 'GARIN3 (153745)', 'GARIN4 (149647)', 'GARIN5A (112703)', 'GARIN6 (196472)', 'GARNL3 (84253)', 'GARRE1 (9710)', 'GARS1 (2617)', 'GART (2618)', 'GAS1 (2619)', 'GAS2 (2620)', 'GAS2L1 (10634)', 'GAS2L2 (246176)', 'GAS2L3 (283431)', 'GAS6 (2621)', 'GAS7 (8522)', 'GAS8 (2622)', 'GASK1A (729085)', 'GASK1B (51313)', 'GAST (2520)', 'GATA1 (2623)', 'GATA2 (2624)', 'GATA3 (2625)', 'GATA4 (2626)', 'GATA5 (140628)', 'GATA6 (2627)', 'GATAD1 (57798)', 'GATAD2A (54815)', 'GATAD2B (57459)', 'GATB (5188)', 'GATC (283459)', 'GATD1 (347862)', 'GATD3 (8209)', 'GATM (2628)', 'GBA1 (2629)', 'GBA2 (57704)', 'GBE1 (2632)', 'GBF1 (8729)', 'GBGT1 (26301)', 'GBP1 (2633)', 'GBP2 (2634)', 'GBP3 (2635)', 'GBP4 (115361)', 'GBP5 (115362)', 'GBP6 (163351)', 'GBP7 (388646)', 'GBX1 (2636)', 'GBX2 (2637)', 'GC (2638)', 'GCA (25801)', 'GCAT (23464)', 'GCC1 (79571)', 'GCC2 (9648)', 'GCDH (2639)', 'GCFC2 (6936)', 'GCG (2641)', 'GCGR (2642)', 'GCH1 (2643)', 'GCHFR (2644)', 'GCK (2645)', 'GCKR (2646)', 'GCLC (2729)', 'GCLM (2730)', 'GCM1 (8521)', 'GCM2 (9247)', 'GCN1 (10985)', 'GCNA (93953)', 'GCNT1 (2650)', 'GCNT2 (2651)', 'GCNT3 (9245)', 'GCNT4 (51301)', 'GCSAM (257144)', 'GCSAML (148823)', 'GCSH (2653)', 'GDA (9615)', 'GDAP1 (54332)', 'GDAP1L1 (78997)', 'GDAP2 (54834)', 'GDE1 (51573)', 'GDF1 (2657)', 'GDF10 (2662)', 'GDF11 (10220)', 'GDF15 (9518)', 'GDF2 (2658)', 'GDF3 (9573)', 'GDF5 (8200)', 'GDF6 (392255)', 'GDF7 (151449)', 'GDF9 (2661)', 'GDI1 (2664)', 'GDI2 (2665)', 'GDNF (2668)', 'GDPD1 (284161)', 'GDPD2 (54857)', 'GDPD3 (79153)', 'GDPD4 (220032)', 'GDPD5 (81544)', 'GDPGP1 (390637)', 'GEM (2669)', 'GEMIN2 (8487)', 'GEMIN4 (50628)', 'GEMIN5 (25929)', 'GEMIN6 (79833)', 'GEMIN7 (79760)', 'GEMIN8 (54960)', 'GEN1 (348654)', 'GET1 (7485)', 'GET3 (439)', 'GET4 (51608)', 'GFAP (2670)', 'GFER (2671)', 'GFI1 (2672)', 'GFI1B (8328)', 'GFM1 (85476)', 'GFM2 (84340)', 'GFOD1 (54438)', 'GFOD2 (81577)', 'GFPT1 (2673)', 'GFPT2 (9945)', 'GFRA1 (2674)', 'GFRA2 (2675)', 'GFRA3 (2676)', 'GFRA4 (64096)', 'GFRAL (389400)', 'GFUS (7264)', 'GFY (100507003)', 'GGA1 (26088)', 'GGA2 (23062)', 'GGA3 (23163)', 'GGACT (87769)', 'GGCT (79017)', 'GGCX (2677)', 'GGH (8836)', 'GGN (199720)', 'GGNBP2 (79893)', 'GGPS1 (9453)', 'GGT1 (2678)', 'GGT5 (2687)', 'GGT6 (124975)', 'GGT7 (2686)', 'GGTLC1 (92086)', 'GGTLC2 (91227)', 'GH1 (2688)', 'GH2 (2689)', 'GHDC (84514)', 'GHITM (27069)', 'GHR (2690)', 'GHRH (2691)', 'GHRHR (2692)', 'GHRL (51738)', 'GHSR (2693)', 'GID4 (79018)', 'GID8 (54994)', 'GIGYF1 (64599)', 'GIGYF2 (26058)', 'GIMAP1 (170575)', 'GIMAP2 (26157)', 'GIMAP4 (55303)', 'GIMAP5 (55340)', 'GIMAP6 (474344)', 'GIMAP7 (168537)', 'GIMAP8 (155038)', 'GIN1 (54826)', 'GINM1 (116254)', 'GINS1 (9837)', 'GINS2 (51659)', 'GINS3 (64785)', 'GINS4 (84296)', 'GIP (2695)', 'GIPC1 (10755)', 'GIPC2 (54810)', 'GIPC3 (126326)', 'GIPR (2696)', 'GIT1 (28964)', 'GIT2 (9815)', 'GJA1 (2697)', 'GJA10 (84694)', 'GJA3 (2700)', 'GJA4 (2701)', 'GJA5 (2702)', 'GJA8 (2703)', 'GJA9 (81025)', 'GJB1 (2705)', 'GJB2 (2706)', 'GJB3 (2707)', 'GJB4 (127534)', 'GJB5 (2709)', 'GJB6 (10804)', 'GJB7 (375519)', 'GJC1 (10052)', 'GJC2 (57165)', 'GJC3 (349149)', 'GJD2 (57369)', 'GJD3 (125111)', 'GJD4 (219770)', 'GK (2710)', 'GK2 (2712)', 'GK5 (256356)', 'GKAP1 (80318)', 'GKN1 (56287)', 'GKN2 (200504)', 'GLA (2717)', 'GLB1 (2720)', 'GLB1L (79411)', 'GLB1L2 (89944)', 'GLB1L3 (112937)', 'GLCCI1 (113263)', 'GLCE (26035)', 'GLDC (2731)', 'GLDN (342035)', 'GLE1 (2733)', 'GLG1 (2734)', 'GLI1 (2735)', 'GLI2 (2736)', 'GLI3 (2737)', 'GLI4 (2738)', 'GLIPR1 (11010)', 'GLIPR1L1 (256710)', 'GLIPR1L2 (144321)', 'GLIPR2 (152007)', 'GLIS1 (148979)', 'GLIS2 (84662)', 'GLIS3 (169792)', 'GLMN (11146)', 'GLMP (112770)', 'GLO1 (2739)', 'GLOD4 (51031)', 'GLOD5 (392465)', 'GLP1R (2740)', 'GLP2R (9340)', 'GLRA1 (2741)', 'GLRA2 (2742)', 'GLRA3 (8001)', 'GLRB (2743)', 'GLRX (2745)', 'GLRX2 (51022)', 'GLRX3 (10539)', 'GLRX5 (51218)', 'GLS (2744)', 'GLS2 (27165)', 'GLT1D1 (144423)', 'GLT6D1 (360203)', 'GLT8D1 (55830)', 'GLT8D2 (83468)', 'GLTP (51228)', 'GLTPD2 (388323)', 'GLUD1 (2746)', 'GLUD2 (2747)', 'GLUL (2752)', 'GLYAT (10249)', 'GLYATL1 (92292)', 'GLYATL2 (219970)', 'GLYATL3 (389396)', 'GLYCTK (132158)', 'GLYR1 (84656)', 'GM2A (2760)', 'GMCL1 (64395)', 'GMDS (2762)', 'GMEB1 (10691)', 'GMEB2 (26205)', 'GMFB (2764)', 'GMFG (9535)', 'GMIP (51291)', 'GML (2765)', 'GMNC (647309)', 'GMNN (51053)', 'GMPPA (29926)', 'GMPPB (29925)', 'GMPR (2766)', 'GMPR2 (51292)', 'GMPS (8833)', 'GNA11 (2767)', 'GNA12 (2768)', 'GNA13 (10672)', 'GNA14 (9630)', 'GNA15 (2769)', 'GNAI1 (2770)', 'GNAI2 (2771)', 'GNAI3 (2773)', 'GNAL (2774)', 'GNAO1 (2775)', 'GNAQ (2776)', 'GNAS (2778)', 'GNAT1 (2779)', 'GNAT2 (2780)', 'GNAT3 (346562)', 'GNAZ (2781)', 'GNB1 (2782)', 'GNB1L (54584)', 'GNB2 (2783)', 'GNB3 (2784)', 'GNB4 (59345)', 'GNB5 (10681)', 'GNE (10020)', 'GNG10 (2790)', 'GNG11 (2791)', 'GNG12 (55970)', 'GNG13 (51764)', 'GNG2 (54331)', 'GNG3 (2785)', 'GNG4 (2786)', 'GNG5 (2787)', 'GNG7 (2788)', 'GNG8 (94235)', 'GNGT1 (2792)', 'GNGT2 (2793)', 'GNL1 (2794)', 'GNL2 (29889)', 'GNL3 (26354)', 'GNL3L (54552)', 'GNLY (10578)', 'GNMT (27232)', 'GNPAT (8443)', 'GNPDA1 (10007)', 'GNPDA2 (132789)', 'GNPNAT1 (64841)', 'GNPTAB (79158)', 'GNPTG (84572)', 'GNRH1 (2796)', 'GNRH2 (2797)', 'GNRHR (2798)', 'GNS (2799)', 'GOLGA1 (2800)', 'GOLGA2 (2801)', 'GOLGA3 (2802)', 'GOLGA4 (2803)', 'GOLGA5 (9950)', 'GOLGA6B (55889)', 'GOLGA6D (653643)', 'GOLGA6L6 (727832)', 'GOLGA7 (51125)', 'GOLGA7B (401647)', 'GOLGA8M (653720)', 'GOLGA8N (643699)', 'GOLGA8R (101059918)', 'GOLGB1 (2804)', 'GOLIM4 (27333)', 'GOLM1 (51280)', 'GOLM2 (113201)', 'GOLPH3 (64083)', 'GOLPH3L (55204)', 'GOLT1A (127845)', 'GOLT1B (51026)', 'GON4L (54856)', 'GON7 (84520)', 'GOPC (57120)', 'GORAB (92344)', 'GORASP1 (64689)', 'GORASP2 (26003)', 'GOSR1 (9527)', 'GOSR2 (9570)', 'GOT1 (2805)', 'GOT1L1 (137362)', 'GOT2 (2806)', 'GP1BA (2811)', 'GP1BB (2812)', 'GP2 (2813)', 'GP5 (2814)', 'GP6 (51206)', 'GP9 (2815)', 'GPA33 (10223)', 'GPAA1 (8733)', 'GPALPP1 (55425)', 'GPAM (57678)', 'GPANK1 (7918)', 'GPAT2 (150763)', 'GPAT3 (84803)', 'GPAT4 (137964)', 'GPATCH1 (55094)', 'GPATCH11 (253635)', 'GPATCH2 (55105)', 'GPATCH2L (55668)', 'GPATCH3 (63906)', 'GPATCH8 (23131)', 'GPBAR1 (151306)', 'GPBP1 (65056)', 'GPBP1L1 (60313)', 'GPC1 (2817)', 'GPC2 (221914)', 'GPC3 (2719)', 'GPC4 (2239)', 'GPC5 (2262)', 'GPC6 (10082)', 'GPCPD1 (56261)', 'GPD1 (2819)', 'GPD1L (23171)', 'GPD2 (2820)', 'GPER1 (2852)', 'GPHA2 (170589)', 'GPHN (10243)', 'GPI (2821)', 'GPIHBP1 (338328)', 'GPKOW (27238)', 'GPLD1 (2822)', 'GPM6A (2823)', 'GPM6B (2824)', 'GPN1 (11321)', 'GPN2 (54707)', 'GPN3 (51184)', 'GPNMB (10457)', 'GPR101 (83550)', 'GPR107 (57720)', 'GPR108 (56927)', 'GPR119 (139760)', 'GPR12 (2835)', 'GPR132 (29933)', 'GPR135 (64582)', 'GPR137 (56834)', 'GPR137B (7107)', 'GPR137C (283554)', 'GPR139 (124274)', 'GPR141 (353345)', 'GPR142 (350383)', 'GPR143 (4935)', 'GPR146 (115330)', 'GPR148 (344561)', 'GPR149 (344758)', 'GPR15 (2838)', 'GPR150 (285601)', 'GPR151 (134391)', 'GPR152 (390212)', 'GPR153 (387509)', 'GPR155 (151556)', 'GPR156 (165829)', 'GPR157 (80045)', 'GPR158 (57512)', 'GPR15LG (387695)', 'GPR160 (26996)', 'GPR161 (23432)', 'GPR162 (27239)', 'GPR17 (2840)', 'GPR171 (29909)', 'GPR173 (54328)', 'GPR174 (84636)', 'GPR176 (11245)', 'GPR179 (440435)', 'GPR18 (2841)', 'GPR180 (160897)', 'GPR182 (11318)', 'GPR183 (1880)', 'GPR19 (2842)', 'GPR20 (2843)', 'GPR21 (2844)', 'GPR22 (2845)', 'GPR25 (2848)', 'GPR26 (2849)', 'GPR27 (2850)', 'GPR3 (2827)', 'GPR31 (2853)', 'GPR32 (2854)', 'GPR34 (2857)', 'GPR35 (2859)', 'GPR37 (2861)', 'GPR37L1 (9283)', 'GPR39 (2863)', 'GPR4 (2828)', 'GPR45 (11250)', 'GPR50 (9248)', 'GPR52 (9293)', 'GPR55 (9290)', 'GPR6 (2830)', 'GPR61 (83873)', 'GPR62 (118442)', 'GPR63 (81491)', 'GPR65 (8477)', 'GPR68 (8111)', 'GPR75 (10936)', 'GPR78 (27201)', 'GPR82 (27197)', 'GPR83 (10888)', 'GPR84 (53831)', 'GPR85 (54329)', 'GPR87 (53836)', 'GPR88 (54112)', 'GPRASP1 (9737)', 'GPRC5A (9052)', 'GPRC5B (51704)', 'GPRC5C (55890)', 'GPRC5D (55507)', 'GPRC6A (222545)', 'GPRIN1 (114787)', 'GPRIN2 (9721)', 'GPRIN3 (285513)', 'GPS1 (2873)', 'GPS2 (2874)', 'GPSM1 (26086)', 'GPSM2 (29899)', 'GPSM3 (63940)', 'GPT (2875)', 'GPT2 (84706)', 'GPX1 (2876)', 'GPX2 (2877)', 'GPX3 (2878)', 'GPX4 (2879)', 'GPX5 (2880)', 'GPX6 (257202)', 'GPX7 (2882)', 'GPX8 (493869)', 'GRAMD1A (57655)', 'GRAMD1B (57476)', 'GRAMD1C (54762)', 'GRAMD2A (196996)', 'GRAMD2B (65983)', 'GRAMD4 (23151)', 'GRAP2 (9402)', 'GRB10 (2887)', 'GRB14 (2888)', 'GRB2 (2885)', 'GRB7 (2886)', 'GREB1 (9687)', 'GREB1L (80000)', 'GREM1 (26585)', 'GREM2 (64388)', 'GRHL1 (29841)', 'GRHL2 (79977)', 'GRHL3 (57822)', 'GRHPR (9380)', 'GRIA1 (2890)', 'GRIA2 (2891)', 'GRIA3 (2892)', 'GRIA4 (2893)', 'GRID1 (2894)', 'GRID2 (2895)', 'GRID2IP (392862)', 'GRIK1 (2897)', 'GRIK2 (2898)', 'GRIK3 (2899)', 'GRIK4 (2900)', 'GRIK5 (2901)', 'GRIN1 (2902)', 'GRIN2A (2903)', 'GRIN2B (2904)', 'GRIN2C (2905)', 'GRIN2D (2906)', 'GRIN3A (116443)', 'GRIN3B (116444)', 'GRINA (2907)', 'GRIP1 (23426)', 'GRIPAP1 (56850)', 'GRK2 (156)', 'GRK3 (157)', 'GRK4 (2868)', 'GRK5 (2869)', 'GRK6 (2870)', 'GRK7 (131890)', 'GRM1 (2911)', 'GRM2 (2912)', 'GRM3 (2913)', 'GRM4 (2914)', 'GRM5 (2915)', 'GRM6 (2916)', 'GRM7 (2917)', 'GRM8 (2918)', 'GRN (2896)', 'GRP (2922)', 'GRPEL1 (80273)', 'GRPEL2 (134266)', 'GRPR (2925)', 'GRSF1 (2926)', 'GRTP1 (79774)', 'GRWD1 (83743)', 'GRXCR1 (389207)', 'GRXCR2 (643226)', 'GSAP (54103)', 'GSC (145258)', 'GSC2 (2928)', 'GSDMA (284110)', 'GSDMB (55876)', 'GSDMC (56169)', 'GSDMD (79792)', 'GSDME (1687)', 'GSE1 (23199)', 'GSG1 (83445)', 'GSG1L (146395)', 'GSK3A (2931)', 'GSK3B (2932)', 'GSKIP (51527)', 'GSN (2934)', 'GSPT1 (2935)', 'GSPT2 (23708)', 'GSR (2936)', 'GSS (2937)', 'GSTA1 (2938)', 'GSTA2 (2939)', 'GSTA3 (2940)', 'GSTA4 (2941)', 'GSTA5 (221357)', 'GSTCD (79807)', 'GSTK1 (373156)', 'GSTM1 (2944)', 'GSTM2 (2946)', 'GSTM3 (2947)', 'GSTM4 (2948)', 'GSTM5 (2949)', 'GSTO1 (9446)', 'GSTO2 (119391)', 'GSTP1 (2950)', 'GSTT2B (653689)', 'GSTZ1 (2954)', 'GSX1 (219409)', 'GSX2 (170825)', 'GTDC1 (79712)', 'GTF2A1 (2957)', 'GTF2A1L (11036)', 'GTF2A2 (2958)', 'GTF2B (2959)', 'GTF2E1 (2960)', 'GTF2E2 (2961)', 'GTF2F1 (2962)', 'GTF2F2 (2963)', 'GTF2H1 (2965)', 'GTF2H3 (2967)', 'GTF2H4 (2968)', 'GTF2H5 (404672)', 'GTF2I (2969)', 'GTF2IRD1 (9569)', 'GTF3A (2971)', 'GTF3C1 (2975)', 'GTF3C2 (2976)', 'GTF3C3 (9330)', 'GTF3C4 (9329)', 'GTF3C5 (9328)', 'GTF3C6 (112495)', 'GTPBP1 (9567)', 'GTPBP10 (85865)', 'GTPBP2 (54676)', 'GTPBP3 (84705)', 'GTPBP4 (23560)', 'GTPBP8 (29083)', 'GTSE1 (51512)', 'GTSF1 (121355)', 'GTSF1L (149699)', 'GUCA1ANB-GUCA1A (118142757)', 'GUCA1B (2979)', 'GUCA1C (9626)', 'GUCA2A (2980)', 'GUCA2B (2981)', 'GUCD1 (83606)', 'GUCY1A1 (2982)', 'GUCY1A2 (2977)', 'GUCY1B1 (2983)', 'GUCY2C (2984)', 'GUCY2D (3000)', 'GUCY2F (2986)', 'GUF1 (60558)', 'GUK1 (2987)', 'GULP1 (51454)', 'GUSB (2990)', 'GXYLT1 (283464)', 'GXYLT2 (727936)', 'GYG1 (2992)', 'GYG2 (8908)', 'GYPA (2993)', 'GYPC (2995)', 'GYPE (2996)', 'GYS1 (2997)', 'GYS2 (2998)', 'GZF1 (64412)', 'GZMA (3001)', 'GZMB (3002)', 'GZMH (2999)', 'GZMK (3003)', 'GZMM (3004)', 'H1-0 (3005)', 'H1-1 (3024)', 'H1-10 (8971)', 'H1-2 (3006)', 'H1-3 (3007)', 'H1-4 (3008)', 'H1-5 (3009)', 'H1-6 (3010)', 'H1-7 (341567)', 'H1-8 (132243)', 'H2AC1 (221613)', 'H2AC11 (8969)', 'H2AC12 (85235)', 'H2AC15 (8330)', 'H2AC16 (8332)', 'H2AC17 (8336)', 'H2AC20 (8338)', 'H2AC21 (317772)', 'H2AC25 (92815)', 'H2AC4 (8335)', 'H2AC6 (8334)', 'H2AC7 (3013)', 'H2AC8 (3012)', 'H2AJ (55766)', 'H2AP (25763)', 'H2AX (3014)', 'H2AZ1 (3015)', 'H2AZ2 (94239)', 'H2BC1 (255626)', 'H2BC10 (8346)', 'H2BC11 (8970)', 'H2BC13 (8340)', 'H2BC14 (8342)', 'H2BC15 (8341)', 'H2BC17 (8348)', 'H2BC18 (440689)', 'H2BC21 (8349)', 'H2BC26 (128312)', 'H2BC3 (3018)', 'H2BC4 (8347)', 'H2BC5 (3017)', 'H2BC6 (8344)', 'H2BC7 (8343)', 'H2BC8 (8339)', 'H2BC9 (8345)', 'H2BW1 (158983)', 'H2BW2 (286436)', 'H3-3A (3020)', 'H3-3B (3021)', 'H3-4 (8290)', 'H3-5 (440093)', 'H3C1 (8350)', 'H3C10 (8357)', 'H3C11 (8354)', 'H3C12 (8356)', 'H3C2 (8358)', 'H3C3 (8352)', 'H3C4 (8351)', 'H3C6 (8353)', 'H3C7 (8968)', 'H3C8 (8355)', 'H4C1 (8359)', 'H4C13 (8368)', 'H4C16 (121504)', 'H4C2 (8366)', 'H4C3 (8364)', 'H4C4 (8360)', 'H4C5 (8367)', 'H4C6 (8361)', 'H4C7 (8369)', 'H4C8 (8365)', 'H4C9 (8294)', 'H6PD (9563)', 'HAAO (23498)', 'HABP2 (3026)', 'HABP4 (22927)', 'HACD1 (9200)', 'HACD2 (201562)', 'HACD3 (51495)', 'HACD4 (401494)', 'HACE1 (57531)', 'HACL1 (26061)', 'HADH (3033)', 'HADHA (3030)', 'HADHB (3032)', 'HAGH (3029)', 'HAGHL (84264)', 'HAL (3034)', 'HAMP (57817)', 'HAND1 (9421)', 'HAND2 (9464)', 'HAO1 (54363)', 'HAO2 (51179)', 'HAP1 (9001)', 'HAPLN1 (1404)', 'HAPLN2 (60484)', 'HAPLN3 (145864)', 'HAPLN4 (404037)', 'HAPSTR1 (29035)', 'HARBI1 (283254)', 'HARS1 (3035)', 'HARS2 (23438)', 'HAS1 (3036)', 'HAS2 (3037)', 'HAS3 (3038)', 'HASPIN (83903)', 'HAT1 (8520)', 'HAUS1 (115106)', 'HAUS2 (55142)', 'HAUS3 (79441)', 'HAUS4 (54930)', 'HAUS5 (23354)', 'HAUS6 (54801)', 'HAUS7 (55559)', 'HAUS8 (93323)', 'HAVCR1 (26762)', 'HAVCR2 (84868)', 'HAX1 (10456)', 'HBB (3043)', 'HBD (3045)', 'HBE1 (3046)', 'HBEGF (1839)', 'HBM (3042)', 'HBP1 (26959)', 'HBQ1 (3049)', 'HBS1L (10767)', 'HBZ (3050)', 'HCAR1 (27198)', 'HCAR2 (338442)', 'HCAR3 (8843)', 'HCCS (3052)', 'HCFC1 (3054)', 'HCFC1R1 (54985)', 'HCFC2 (29915)', 'HCK (3055)', 'HCLS1 (3059)', 'HCN1 (348980)', 'HCN2 (610)', 'HCN3 (57657)', 'HCN4 (10021)', 'HCRT (3060)', 'HCRTR1 (3061)', 'HCRTR2 (3062)', 'HCST (10870)', 'HDAC1 (3065)', 'HDAC10 (83933)', 'HDAC11 (79885)', 'HDAC2 (3066)', 'HDAC3 (8841)', 'HDAC4 (9759)', 'HDAC5 (10014)', 'HDAC6 (10013)', 'HDAC7 (51564)', 'HDAC8 (55869)', 'HDAC9 (9734)', 'HDC (3067)', 'HDDC2 (51020)', 'HDDC3 (374659)', 'HDGF (3068)', 'HDGFL1 (154150)', 'HDGFL2 (84717)', 'HDGFL3 (50810)', 'HDHD2 (84064)', 'HDHD3 (81932)', 'HDHD5 (27440)', 'HDLBP (3069)', 'HDX (139324)', 'HEATR1 (55127)', 'HEATR3 (55027)', 'HEATR4 (399671)', 'HEATR5B (54497)', 'HEATR6 (63897)', 'HEATR9 (256957)', 'HEBP1 (50865)', 'HEBP2 (23593)', 'HECA (51696)', 'HECTD1 (25831)', 'HECTD2 (143279)', 'HECTD3 (79654)', 'HECW1 (23072)', 'HECW2 (57520)', 'HEG1 (57493)', 'HELB (92797)', 'HELLS (3070)', 'HELQ (113510)', 'HELT (391723)', 'HELZ (9931)', 'HELZ2 (85441)', 'HEMGN (55363)', 'HEMK1 (51409)', 'HENMT1 (113802)', 'HEPACAM (220296)', 'HEPACAM2 (253012)', 'HEPH (9843)', 'HEPHL1 (341208)', 'HERC1 (8925)', 'HERC2 (8924)', 'HERC3 (8916)', 'HERC4 (26091)', 'HERC5 (51191)', 'HERC6 (55008)', 'HERPUD1 (9709)', 'HERPUD2 (64224)', 'HES1 (3280)', 'HES2 (54626)', 'HES3 (390992)', 'HES4 (57801)', 'HES5 (388585)', 'HES6 (55502)', 'HES7 (84667)', 'HESX1 (8820)', 'HEXA (3073)', 'HEXB (3074)', 'HEXD (284004)', 'HEXIM1 (10614)', 'HEXIM2 (124790)', 'HEY1 (23462)', 'HEY2 (23493)', 'HEYL (26508)', 'HFE (3077)', 'HFM1 (164045)', 'HGD (3081)', 'HGF (3082)', 'HGFAC (3083)', 'HGH1 (51236)', 'HGS (9146)', 'HGSNAT (138050)', 'HHAT (55733)', 'HHATL (57467)', 'HHEX (3087)', 'HHIP (64399)', 'HHIPL1 (84439)', 'HHIPL2 (79802)', 'HHLA2 (11148)', 'HIBADH (11112)', 'HIBCH (26275)', 'HIC1 (3090)', 'HIC2 (23119)', 'HID1 (283987)', 'HIF1A (3091)', 'HIF1AN (55662)', 'HIF3A (64344)', 'HIGD1A (25994)', 'HIGD1B (51751)', 'HIGD1C (613227)', 'HIGD2A (192286)', 'HIKESHI (51501)', 'HILPDA (29923)', 'HINFP (25988)', 'HINT1 (3094)', 'HINT2 (84681)', 'HINT3 (135114)', 'HIP1 (3092)', 'HIP1R (9026)', 'HIPK1 (204851)', 'HIPK3 (10114)', 'HIPK4 (147746)', 'HIRA (7290)', 'HIRIP3 (8479)', 'HIVEP1 (3096)', 'HIVEP2 (3097)', 'HIVEP3 (59269)', 'HJURP (55355)', 'HJV (148738)', 'HK1 (3098)', 'HK2 (3099)', 'HK3 (3101)', 'HKDC1 (80201)', 'HLA-A (3105)', 'HLA-B (3106)', 'HLA-C (3107)', 'HLA-DMA (3108)', 'HLA-DMB (3109)', 'HLA-DOA (3111)', 'HLA-DOB (3112)', 'HLA-DPA1 (3113)', 'HLA-DPB1 (3115)', 'HLA-DQA1 (3117)', 'HLA-DQA2 (3118)', 'HLA-DQB1 (3119)', 'HLA-DQB2 (3120)', 'HLA-DRA (3122)', 'HLA-DRB1 (3123)', 'HLA-DRB5 (3127)', 'HLA-E (3133)', 'HLA-F (3134)', 'HLA-G (3135)', 'HLCS (3141)', 'HLF (3131)', 'HLTF (6596)', 'HLX (3142)', 'HM13 (81502)', 'HMBOX1 (79618)', 'HMBS (3145)', 'HMCES (56941)', 'HMCN1 (83872)', 'HMG20A (10363)', 'HMG20B (10362)', 'HMGA1 (3159)', 'HMGA2 (8091)', 'HMGB1 (3146)', 'HMGB2 (3148)', 'HMGB3 (3149)', 'HMGB4 (127540)', 'HMGCL (3155)', 'HMGCLL1 (54511)', 'HMGCR (3156)', 'HMGCS1 (3157)', 'HMGCS2 (3158)', 'HMGN1 (3150)', 'HMGN2 (3151)', 'HMGN3 (9324)', 'HMGN4 (10473)', 'HMGN5 (79366)', 'HMGXB3 (22993)', 'HMGXB4 (10042)', 'HMHB1 (57824)', 'HMMR (3161)', 'HMOX1 (3162)', 'HMOX2 (3163)', 'HMSD (284293)', 'HMX1 (3166)', 'HMX2 (3167)', 'HMX3 (340784)', 'HNF1A (6927)', 'HNF1B (6928)', 'HNF4A (3172)', 'HNF4G (3174)', 'HNMT (3176)', 'HNRNPA0 (10949)', 'HNRNPA1 (3178)', 'HNRNPA1L2 (144983)', 'HNRNPA2B1 (3181)', 'HNRNPA3 (220988)', 'HNRNPAB (3182)', 'HNRNPC (3183)', 'HNRNPD (3184)', 'HNRNPDL (9987)', 'HNRNPF (3185)', 'HNRNPH1 (3187)', 'HNRNPH2 (3188)', 'HNRNPH3 (3189)', 'HNRNPK (3190)', 'HNRNPL (3191)', 'HNRNPLL (92906)', 'HNRNPM (4670)', 'HNRNPR (10236)', 'HNRNPU (3192)', 'HNRNPUL1 (11100)', 'HNRNPUL2 (221092)', 'HOATZ (399949)', 'HOGA1 (112817)', 'HOMER1 (9456)', 'HOMER2 (9455)', 'HOMER3 (9454)', 'HOMEZ (57594)', 'HOOK1 (51361)', 'HOOK2 (29911)', 'HOOK3 (84376)', 'HOPX (84525)', 'HORMAD1 (84072)', 'HORMAD2 (150280)', 'HOXA1 (3198)', 'HOXA10 (3206)', 'HOXA11 (3207)', 'HOXA13 (3209)', 'HOXA2 (3199)', 'HOXA3 (3200)', 'HOXA4 (3201)', 'HOXA5 (3202)', 'HOXA6 (3203)', 'HOXA7 (3204)', 'HOXA9 (3205)', 'HOXB1 (3211)', 'HOXB13 (10481)', 'HOXB2 (3212)', 'HOXB3 (3213)', 'HOXB4 (3214)', 'HOXB5 (3215)', 'HOXB6 (3216)', 'HOXB7 (3217)', 'HOXB8 (3218)', 'HOXB9 (3219)', 'HOXC10 (3226)', 'HOXC11 (3227)', 'HOXC12 (3228)', 'HOXC13 (3229)', 'HOXC4 (3221)', 'HOXC5 (3222)', 'HOXC6 (3223)', 'HOXC8 (3224)', 'HOXC9 (3225)', 'HOXD1 (3231)', 'HOXD10 (3236)', 'HOXD11 (3237)', 'HOXD12 (3238)', 'HOXD13 (3239)', 'HOXD3 (3232)', 'HOXD4 (3233)', 'HOXD8 (3234)', 'HOXD9 (3235)', 'HP (3240)', 'HP1BP3 (50809)', 'HPCA (3208)', 'HPCAL1 (3241)', 'HPCAL4 (51440)', 'HPD (3242)', 'HPDL (84842)', 'HPF1 (54969)', 'HPGD (3248)', 'HPGDS (27306)', 'HPN (3249)', 'HPR (3250)', 'HPRT1 (3251)', 'HPS1 (3257)', 'HPS3 (84343)', 'HPS4 (89781)', 'HPS5 (11234)', 'HPS6 (79803)', 'HPSE (10855)', 'HPSE2 (60495)', 'HPX (3263)', 'HR (55806)', 'HRAS (3265)', 'HRC (3270)', 'HRCT1 (646962)', 'HRG (3273)', 'HRH1 (3269)', 'HRH2 (3274)', 'HRH3 (11255)', 'HRH4 (59340)', 'HRK (8739)', 'HRNR (388697)', 'HROB (78995)', 'HS1BP3 (64342)', 'HS2ST1 (9653)', 'HS3ST1 (9957)', 'HS3ST2 (9956)', 'HS3ST3A1 (9955)', 'HS3ST3B1 (9953)', 'HS3ST4 (9951)', 'HS3ST5 (222537)', 'HS3ST6 (64711)', 'HS6ST1 (9394)', 'HS6ST2 (90161)', 'HS6ST3 (266722)', 'HSBP1 (3281)', 'HSBP1L1 (440498)', 'HSCB (150274)', 'HSD11B1 (3290)', 'HSD11B1L (374875)', 'HSD11B2 (3291)', 'HSD17B1 (3292)', 'HSD17B10 (3028)', 'HSD17B11 (51170)', 'HSD17B12 (51144)', 'HSD17B13 (345275)', 'HSD17B14 (51171)', 'HSD17B2 (3294)', 'HSD17B3 (3293)', 'HSD17B4 (3295)', 'HSD17B6 (8630)', 'HSD17B7 (51478)', 'HSD17B8 (7923)', 'HSD3B1 (3283)', 'HSD3B2 (3284)', 'HSD3B7 (80270)', 'HSDL1 (83693)', 'HSDL2 (84263)', 'HSF1 (3297)', 'HSF2 (3298)', 'HSF2BP (11077)', 'HSF4 (3299)', 'HSF5 (124535)', 'HSP90AA1 (3320)', 'HSP90AB1 (3326)', 'HSP90B1 (7184)', 'HSPA12A (259217)', 'HSPA12B (116835)', 'HSPA13 (6782)', 'HSPA14 (51182)', 'HSPA1L (3305)', 'HSPA2 (3306)', 'HSPA4 (3308)', 'HSPA4L (22824)', 'HSPA5 (3309)', 'HSPA6 (3310)', 'HSPA8 (3312)', 'HSPA9 (3313)', 'HSPB1 (3315)', 'HSPB2 (3316)', 'HSPB3 (8988)', 'HSPB6 (126393)', 'HSPB7 (27129)', 'HSPB8 (26353)', 'HSPB9 (94086)', 'HSPBAP1 (79663)', 'HSPBP1 (23640)', 'HSPD1 (3329)', 'HSPE1 (3336)', 'HSPG2 (3339)', 'HSPH1 (10808)', 'HTATIP2 (10553)', 'HTATSF1 (27336)', 'HTN1 (3346)', 'HTN3 (3347)', 'HTR1A (3350)', 'HTR1B (3351)', 'HTR1D (3352)', 'HTR1E (3354)', 'HTR1F (3355)', 'HTR2A (3356)', 'HTR2B (3357)', 'HTR2C (3358)', 'HTR3A (3359)', 'HTR3B (9177)', 'HTR3C (170572)', 'HTR3D (200909)', 'HTR3E (285242)', 'HTR4 (3360)', 'HTR5A (3361)', 'HTR6 (3362)', 'HTR7 (3363)', 'HTRA1 (5654)', 'HTRA2 (27429)', 'HTRA3 (94031)', 'HTRA4 (203100)', 'HTT (3064)', 'HUNK (30811)', 'HUS1 (3364)', 'HUS1B (135458)', 'HUWE1 (10075)', 'HVCN1 (84329)', 'HYAL1 (3373)', 'HYAL2 (8692)', 'HYAL3 (8372)', 'HYAL4 (23553)', 'HYCC1 (84668)', 'HYCC2 (285172)', 'HYDIN (54768)', 'HYI (81888)', 'HYKK (123688)', 'HYLS1 (219844)', 'HYOU1 (10525)', 'HYPK (25764)', 'IAH1 (285148)', 'IAPP (3375)', 'IARS1 (3376)', 'IARS2 (55699)', 'IBA57 (200205)', 'IBSP (3381)', 'IBTK (25998)', 'ICA1 (3382)', 'ICA1L (130026)', 'ICAM1 (3383)', 'ICAM2 (3384)', 'ICAM3 (3385)', 'ICAM4 (3386)', 'ICAM5 (7087)', 'ICE1 (23379)', 'ICE2 (79664)', 'ICMT (23463)', 'ICOS (29851)', 'ICOSLG (23308)', 'ID1 (3397)', 'ID2 (3398)', 'ID3 (3399)', 'ID4 (3400)', 'IDE (3416)', 'IDH1 (3417)', 'IDH2 (3418)', 'IDH3A (3419)', 'IDH3B (3420)', 'IDH3G (3421)', 'IDI1 (3422)', 'IDI2 (91734)', 'IDNK (414328)', 'IDO1 (3620)', 'IDS (3423)', 'IDUA (3425)', 'IER2 (9592)', 'IER3 (8870)', 'IER3IP1 (51124)', 'IER5 (51278)', 'IER5L (389792)', 'IFFO1 (25900)', 'IFFO2 (126917)', 'IFI16 (3428)', 'IFI27 (3429)', 'IFI27L1 (122509)', 'IFI27L2 (83982)', 'IFI30 (10437)', 'IFI35 (3430)', 'IFI44 (10561)', 'IFI44L (10964)', 'IFI6 (2537)', 'IFIH1 (64135)', 'IFIT1 (3434)', 'IFIT1B (439996)', 'IFIT2 (3433)', 'IFIT3 (3437)', 'IFIT5 (24138)', 'IFITM1 (8519)', 'IFITM10 (402778)', 'IFITM2 (10581)', 'IFITM3 (10410)', 'IFITM5 (387733)', 'IFNA1 (3439)', 'IFNA10 (3446)', 'IFNA14 (3448)', 'IFNA16 (3449)', 'IFNA17 (3451)', 'IFNA2 (3440)', 'IFNA21 (3452)', 'IFNA4 (3441)', 'IFNA5 (3442)', 'IFNA6 (3443)', 'IFNA7 (3444)', 'IFNA8 (3445)', 'IFNAR1 (3454)', 'IFNAR2 (3455)', 'IFNB1 (3456)', 'IFNE (338376)', 'IFNG (3458)', 'IFNGR1 (3459)', 'IFNGR2 (3460)', 'IFNK (56832)', 'IFNL1 (282618)', 'IFNL2 (282616)', 'IFNL3 (282617)', 'IFNLR1 (163702)', 'IFNW1 (3467)', 'IFRD1 (3475)', 'IFRD2 (7866)', 'IFT122 (55764)', 'IFT140 (9742)', 'IFT172 (26160)', 'IFT20 (90410)', 'IFT22 (64792)', 'IFT25 (51668)', 'IFT27 (11020)', 'IFT43 (112752)', 'IFT46 (56912)', 'IFT52 (51098)', 'IFT56 (79989)', 'IFT57 (55081)', 'IFT70A (92104)', 'IFT70B (150737)', 'IFT74 (80173)', 'IFT80 (57560)', 'IFT81 (28981)', 'IFT88 (8100)', 'IFTAP (119710)', 'IGBP1 (3476)', 'IGDCC3 (9543)', 'IGDCC4 (57722)', 'IGF1 (3479)', 'IGF1R (3480)', 'IGF2 (3481)', 'IGF2BP1 (10642)', 'IGF2BP2 (10644)', 'IGF2BP3 (10643)', 'IGF2R (3482)', 'IGFALS (3483)', 'IGFBP1 (3484)', 'IGFBP2 (3485)', 'IGFBP3 (3486)', 'IGFBP4 (3487)', 'IGFBP5 (3488)', 'IGFBP6 (3489)', 'IGFBP7 (3490)', 'IGFBPL1 (347252)', 'IGFL1 (374918)', 'IGFL2 (147920)', 'IGFL3 (388555)', 'IGFL4 (444882)', 'IGFLR1 (79713)', 'IGFN1 (91156)', 'IGHMBP2 (3508)', 'IGIP (492311)', 'IGLL1 (3543)', 'IGLL5 (100423062)', 'IGLON5 (402665)', 'IGSF1 (3547)', 'IGSF10 (285313)', 'IGSF11 (152404)', 'IGSF21 (84966)', 'IGSF22 (283284)', 'IGSF23 (147710)', 'IGSF3 (3321)', 'IGSF5 (150084)', 'IGSF6 (10261)', 'IGSF8 (93185)', 'IGSF9 (57549)', 'IGSF9B (22997)', 'IHH (3549)', 'IHO1 (339834)', 'IK (3550)', 'IKBIP (121457)', 'IKBKB (3551)', 'IKBKE (9641)', 'IKBKG (8517)', 'IKZF1 (10320)', 'IKZF2 (22807)', 'IKZF3 (22806)', 'IKZF4 (64375)', 'IKZF5 (64376)', 'IL10 (3586)', 'IL10RA (3587)', 'IL10RB (3588)', 'IL11 (3589)', 'IL11RA (3590)', 'IL12A (3592)', 'IL12B (3593)', 'IL12RB1 (3594)', 'IL12RB2 (3595)', 'IL13 (3596)', 'IL13RA1 (3597)', 'IL13RA2 (3598)', 'IL15 (3600)', 'IL15RA (3601)', 'IL16 (3603)', 'IL17A (3605)', 'IL17B (27190)', 'IL17C (27189)', 'IL17D (53342)', 'IL17F (112744)', 'IL17RA (23765)', 'IL17RB (55540)', 'IL17RC (84818)', 'IL17RD (54756)', 'IL17RE (132014)', 'IL17REL (400935)', 'IL18 (3606)', 'IL18BP (10068)', 'IL18R1 (8809)', 'IL18RAP (8807)', 'IL19 (29949)', 'IL1A (3552)', 'IL1B (3553)', 'IL1F10 (84639)', 'IL1R1 (3554)', 'IL1R2 (7850)', 'IL1RAP (3556)', 'IL1RAPL1 (11141)', 'IL1RAPL2 (26280)', 'IL1RL1 (9173)', 'IL1RL2 (8808)', 'IL1RN (3557)', 'IL2 (3558)', 'IL20 (50604)', 'IL20RA (53832)', 'IL20RB (53833)', 'IL21 (59067)', 'IL21R (50615)', 'IL22 (50616)', 'IL22RA1 (58985)', 'IL22RA2 (116379)', 'IL23A (51561)', 'IL23R (149233)', 'IL24 (11009)', 'IL25 (64806)', 'IL26 (55801)', 'IL27 (246778)', 'IL27RA (9466)', 'IL2RA (3559)', 'IL2RB (3560)', 'IL2RG (3561)', 'IL3 (3562)', 'IL31 (386653)', 'IL31RA (133396)', 'IL32 (9235)', 'IL33 (90865)', 'IL34 (146433)', 'IL36A (27179)', 'IL36B (27177)', 'IL36G (56300)', 'IL36RN (26525)', 'IL37 (27178)', 'IL4 (3565)', 'IL4I1 (259307)', 'IL4R (3566)', 'IL5 (3567)', 'IL5RA (3568)', 'IL6 (3569)', 'IL6R (3570)', 'IL6ST (3572)', 'IL7 (3574)', 'IL7R (3575)', 'IL9 (3578)', 'IL9R (3581)', 'ILDR1 (286676)', 'ILDR2 (387597)', 'ILF2 (3608)', 'ILF3 (3609)', 'ILK (3611)', 'ILKAP (80895)', 'ILRUN (64771)', 'ILVBL (10994)', 'IMMP1L (196294)', 'IMMP2L (83943)', 'IMMT (10989)', 'IMP3 (55272)', 'IMP4 (92856)', 'IMPA1 (3612)', 'IMPA2 (3613)', 'IMPACT (55364)', 'IMPDH1 (3614)', 'IMPDH2 (3615)', 'IMPG1 (3617)', 'IMPG2 (50939)', 'INA (9118)', 'INAFM1 (255783)', 'INAVA (55765)', 'INCA1 (388324)', 'INCENP (3619)', 'INF2 (64423)', 'ING1 (3621)', 'ING2 (3622)', 'ING3 (54556)', 'ING4 (51147)', 'ING5 (84289)', 'INHA (3623)', 'INHBA (3624)', 'INHBB (3625)', 'INHBC (3626)', 'INHBE (83729)', 'INIP (58493)', 'INKA1 (389119)', 'INKA2 (55924)', 'INMT (11185)', 'INO80 (54617)', 'INO80B (83444)', 'INO80C (125476)', 'INO80D (54891)', 'INO80E (283899)', 'INPP1 (3628)', 'INPP4A (3631)', 'INPP4B (8821)', 'INPP5A (3632)', 'INPP5B (3633)', 'INPP5E (56623)', 'INPP5F (22876)', 'INPP5J (27124)', 'INPP5K (51763)', 'INPPL1 (3636)', 'INS (3630)', 'INSC (387755)', 'INSIG1 (3638)', 'INSIG2 (51141)', 'INSL3 (3640)', 'INSL4 (3641)', 'INSL5 (10022)', 'INSL6 (11172)', 'INSM1 (3642)', 'INSM2 (84684)', 'INSR (3643)', 'INSRR (3645)', 'INSYN1 (388135)', 'INSYN2A (642938)', 'INSYN2B (100131897)', 'INTS1 (26173)', 'INTS10 (55174)', 'INTS11 (54973)', 'INTS12 (57117)', 'INTS13 (55726)', 'INTS14 (81556)', 'INTS15 (79034)', 'INTS2 (57508)', 'INTS3 (65123)', 'INTS4 (92105)', 'INTS5 (80789)', 'INTS6 (26512)', 'INTS6L (203522)', 'INTS7 (25896)', 'INTS8 (55656)', 'INTS9 (55756)', 'INTU (27152)', 'INVS (27130)', 'IP6K1 (9807)', 'IP6K2 (51447)', 'IP6K3 (117283)', 'IPCEF1 (26034)', 'IPMK (253430)', 'IPO11 (51194)', 'IPO13 (9670)', 'IPO4 (79711)', 'IPO5 (3843)', 'IPO7 (10527)', 'IPO8 (10526)', 'IPO9 (55705)', 'IPP (3652)', 'IPPK (64768)', 'IQCA1 (79781)', 'IQCB1 (9657)', 'IQCC (55721)', 'IQCD (115811)', 'IQCE (23288)', 'IQCF1 (132141)', 'IQCF2 (389123)', 'IQCF3 (401067)', 'IQCF5 (389124)', 'IQCF6 (440956)', 'IQCG (84223)', 'IQCH (64799)', 'IQCJ (654502)', 'IQCK (124152)', 'IQCN (80726)', 'IQGAP1 (8826)', 'IQGAP2 (10788)', 'IQGAP3 (128239)', 'IQSEC1 (9922)', 'IQSEC2 (23096)', 'IQSEC3 (440073)', 'IQUB (154865)', 'IRAG1 (10335)', 'IRAG2 (4033)', 'IRAK1 (3654)', 'IRAK1BP1 (134728)', 'IRAK2 (3656)', 'IRAK3 (11213)', 'IRAK4 (51135)', 'IREB2 (3658)', 'IRF1 (3659)', 'IRF2 (3660)', 'IRF2BP1 (26145)', 'IRF2BP2 (359948)', 'IRF2BPL (64207)', 'IRF3 (3661)', 'IRF4 (3662)', 'IRF5 (3663)', 'IRF6 (3664)', 'IRF7 (3665)', 'IRF8 (3394)', 'IRF9 (10379)', 'IRGC (56269)', 'IRGM (345611)', 'IRGQ (126298)', 'IRS1 (3667)', 'IRS2 (8660)', 'IRS4 (8471)', 'IRX1 (79192)', 'IRX2 (153572)', 'IRX3 (79191)', 'IRX4 (50805)', 'IRX5 (10265)', 'IRX6 (79190)', 'ISCA1 (81689)', 'ISCA2 (122961)', 'ISCU (23479)', 'ISG15 (9636)', 'ISG20 (3669)', 'ISG20L2 (81875)', 'ISL1 (3670)', 'ISL2 (64843)', 'ISLR (3671)', 'ISLR2 (57611)', 'ISM1 (140862)', 'ISM2 (145501)', 'ISOC1 (51015)', 'ISOC2 (79763)', 'IST1 (9798)', 'ISX (91464)', 'ISY1 (57461)', 'ISYNA1 (51477)', 'ITCH (83737)', 'ITFG1 (81533)', 'ITFG2 (55846)', 'ITGA1 (3672)', 'ITGA10 (8515)', 'ITGA11 (22801)', 'ITGA2 (3673)', 'ITGA2B (3674)', 'ITGA3 (3675)', 'ITGA4 (3676)', 'ITGA5 (3678)', 'ITGA6 (3655)', 'ITGA7 (3679)', 'ITGA8 (8516)', 'ITGA9 (3680)', 'ITGAD (3681)', 'ITGAE (3682)', 'ITGAL (3683)', 'ITGAM (3684)', 'ITGAV (3685)', 'ITGAX (3687)', 'ITGB1 (3688)', 'ITGB1BP1 (9270)', 'ITGB1BP2 (26548)', 'ITGB2 (3689)', 'ITGB3 (3690)', 'ITGB3BP (23421)', 'ITGB4 (3691)', 'ITGB5 (3693)', 'ITGB6 (3694)', 'ITGB7 (3695)', 'ITGB8 (3696)', 'ITGBL1 (9358)', 'ITIH1 (3697)', 'ITIH2 (3698)', 'ITIH3 (3699)', 'ITIH4 (3700)', 'ITIH6 (347365)', 'ITK (3702)', 'ITLN1 (55600)', 'ITLN2 (142683)', 'ITM2A (9452)', 'ITM2B (9445)', 'ITM2C (81618)', 'ITPA (3704)', 'ITPK1 (3705)', 'ITPKA (3706)', 'ITPKB (3707)', 'ITPKC (80271)', 'ITPR1 (3708)', 'ITPR2 (3709)', 'ITPR3 (3710)', 'ITPRID1 (223075)', 'ITPRID2 (6744)', 'ITPRIP (85450)', 'ITPRIPL1 (150771)', 'ITPRIPL2 (162073)', 'ITSN1 (6453)', 'ITSN2 (50618)', 'IVD (3712)', 'IVL (3713)', 'IVNS1ABP (10625)', 'IWS1 (55677)', 'IYD (389434)', 'IZUMO1 (284359)', 'IZUMO2 (126123)', 'IZUMO3 (100129669)', 'IZUMO4 (113177)', 'JADE1 (79960)', 'JADE2 (23338)', 'JADE3 (9767)', 'JAG1 (182)', 'JAG2 (3714)', 'JAGN1 (84522)', 'JAK1 (3716)', 'JAK2 (3717)', 'JAK3 (3718)', 'JAKMIP1 (152789)', 'JAKMIP2 (9832)', 'JAKMIP3 (282973)', 'JAM2 (58494)', 'JAM3 (83700)', 'JAML (120425)', 'JARID2 (3720)', 'JAZF1 (221895)', 'JCAD (57608)', 'JCHAIN (3512)', 'JDP2 (122953)', 'JHY (79864)', 'JKAMP (51528)', 'JMJD1C (221037)', 'JMJD4 (65094)', 'JMJD6 (23210)', 'JMJD7 (100137047)', 'JMJD8 (339123)', 'JMY (133746)', 'JOSD1 (9929)', 'JOSD2 (126119)', 'JPH1 (56704)', 'JPH2 (57158)', 'JPH3 (57338)', 'JPH4 (84502)', 'JPT1 (51155)', 'JPT2 (90861)', 'JRKL (8690)', 'JSRP1 (126306)', 'JTB (10899)', 'JUN (3725)', 'JUNB (3726)', 'JUND (3727)', 'JUP (3728)', 'KALRN (8997)', 'KANK1 (23189)', 'KANK2 (25959)', 'KANK3 (256949)', 'KANK4 (163782)', 'KANSL1 (284058)', 'KANSL1L (151050)', 'KANSL2 (54934)', 'KANSL3 (55683)', 'KARS1 (3735)', 'KASH5 (147872)', 'KAT14 (57325)', 'KAT2A (2648)', 'KAT2B (8850)', 'KAT5 (10524)', 'KAT6A (7994)', 'KAT6B (23522)', 'KAT7 (11143)', 'KAT8 (84148)', 'KATNA1 (11104)', 'KATNAL1 (84056)', 'KATNAL2 (83473)', 'KATNB1 (10300)', 'KATNBL1 (79768)', 'KATNIP (23247)', 'KAZALD1 (81621)', 'KAZN (23254)', 'KBTBD11 (9920)', 'KBTBD12 (166348)', 'KBTBD13 (390594)', 'KBTBD2 (25948)', 'KBTBD3 (143879)', 'KBTBD4 (55709)', 'KBTBD6 (89890)', 'KBTBD7 (84078)', 'KBTBD8 (84541)', 'KCMF1 (56888)', 'KCNA1 (3736)', 'KCNA10 (3744)', 'KCNA2 (3737)', 'KCNA3 (3738)', 'KCNA4 (3739)', 'KCNA5 (3741)', 'KCNA6 (3742)', 'KCNA7 (3743)', 'KCNAB1 (7881)', 'KCNAB2 (8514)', 'KCNAB3 (9196)', 'KCNB1 (3745)', 'KCNB2 (9312)', 'KCNC1 (3746)', 'KCNC2 (3747)', 'KCNC3 (3748)', 'KCNC4 (3749)', 'KCND1 (3750)', 'KCND2 (3751)', 'KCND3 (3752)', 'KCNE1 (3753)', 'KCNE2 (9992)', 'KCNE3 (10008)', 'KCNE4 (23704)', 'KCNE5 (23630)', 'KCNF1 (3754)', 'KCNG1 (3755)', 'KCNG2 (26251)', 'KCNG3 (170850)', 'KCNG4 (93107)', 'KCNH1 (3756)', 'KCNH2 (3757)', 'KCNH3 (23416)', 'KCNH4 (23415)', 'KCNH5 (27133)', 'KCNH6 (81033)', 'KCNH7 (90134)', 'KCNH8 (131096)', 'KCNIP1 (30820)', 'KCNIP2 (30819)', 'KCNIP3 (30818)', 'KCNIP4 (80333)', 'KCNJ1 (3758)', 'KCNJ10 (3766)', 'KCNJ11 (3767)', 'KCNJ13 (3769)', 'KCNJ14 (3770)', 'KCNJ15 (3772)', 'KCNJ16 (3773)', 'KCNJ2 (3759)', 'KCNJ3 (3760)', 'KCNJ4 (3761)', 'KCNJ5 (3762)', 'KCNJ6 (3763)', 'KCNJ8 (3764)', 'KCNJ9 (3765)', 'KCNK1 (3775)', 'KCNK10 (54207)', 'KCNK12 (56660)', 'KCNK13 (56659)', 'KCNK15 (60598)', 'KCNK16 (83795)', 'KCNK17 (89822)', 'KCNK18 (338567)', 'KCNK2 (3776)', 'KCNK3 (3777)', 'KCNK4 (50801)', 'KCNK5 (8645)', 'KCNK6 (9424)', 'KCNK7 (10089)', 'KCNK9 (51305)', 'KCNMA1 (3778)', 'KCNMB1 (3779)', 'KCNMB2 (10242)', 'KCNMB3 (27094)', 'KCNMB4 (27345)', 'KCNN1 (3780)', 'KCNN2 (3781)', 'KCNN3 (3782)', 'KCNN4 (3783)', 'KCNQ1 (3784)', 'KCNQ2 (3785)', 'KCNQ3 (3786)', 'KCNQ4 (9132)', 'KCNQ5 (56479)', 'KCNRG (283518)', 'KCNS1 (3787)', 'KCNS2 (3788)', 'KCNS3 (3790)', 'KCNT1 (57582)', 'KCNT2 (343450)', 'KCNU1 (157855)', 'KCNV1 (27012)', 'KCNV2 (169522)', 'KCTD1 (284252)', 'KCTD10 (83892)', 'KCTD11 (147040)', 'KCTD12 (115207)', 'KCTD13 (253980)', 'KCTD14 (65987)', 'KCTD15 (79047)', 'KCTD16 (57528)', 'KCTD17 (79734)', 'KCTD18 (130535)', 'KCTD19 (146212)', 'KCTD2 (23510)', 'KCTD20 (222658)', 'KCTD21 (283219)', 'KCTD3 (51133)', 'KCTD4 (386618)', 'KCTD5 (54442)', 'KCTD6 (200845)', 'KCTD7 (154881)', 'KCTD8 (386617)', 'KCTD9 (54793)', 'KDELR1 (10945)', 'KDELR2 (11014)', 'KDELR3 (11015)', 'KDF1 (126695)', 'KDM1A (23028)', 'KDM1B (221656)', 'KDM2A (22992)', 'KDM2B (84678)', 'KDM3A (55818)', 'KDM3B (51780)', 'KDM4A (9682)', 'KDM4B (23030)', 'KDM4C (23081)', 'KDM4D (55693)', 'KDM4E (390245)', 'KDM5A (5927)', 'KDM5B (10765)', 'KDM5C (8242)', 'KDM5D (8284)', 'KDM6A (7403)', 'KDM6B (23135)', 'KDM7A (80853)', 'KDM8 (79831)', 'KDR (3791)', 'KDSR (2531)', 'KEAP1 (9817)', 'KEL (3792)', 'KERA (11081)', 'KGD4 (92259)', 'KHDC1 (80759)', 'KHDC1L (100129128)', 'KHDC3L (154288)', 'KHDC4 (22889)', 'KHDRBS1 (10657)', 'KHDRBS2 (202559)', 'KHDRBS3 (10656)', 'KHK (3795)', 'KHNYN (23351)', 'KHSRP (8570)', 'KIAA0232 (9778)', 'KIAA0319 (9856)', 'KIAA0319L (79932)', 'KIAA0408 (9729)', 'KIAA0513 (9764)', 'KIAA0586 (9786)', 'KIAA0753 (9851)', 'KIAA0825 (285600)', 'KIAA0930 (23313)', 'KIAA1143 (57456)', 'KIAA1191 (57179)', 'KIAA1210 (57481)', 'KIAA1217 (56243)', 'KIAA1328 (57536)', 'KIAA1549 (57670)', 'KIAA1549L (25758)', 'KIAA1586 (57691)', 'KIAA1614 (57710)', 'KIAA1671 (85379)', 'KIAA1755 (85449)', 'KIAA1958 (158405)', 'KIAA2013 (90231)', 'KICS2 (144577)', 'KIDINS220 (57498)', 'KIF11 (3832)', 'KIF12 (113220)', 'KIF13A (63971)', 'KIF13B (23303)', 'KIF14 (9928)', 'KIF15 (56992)', 'KIF16B (55614)', 'KIF17 (57576)', 'KIF18A (81930)', 'KIF18B (146909)', 'KIF19 (124602)', 'KIF1A (547)', 'KIF1B (23095)', 'KIF1C (10749)', 'KIF20A (10112)', 'KIF20B (9585)', 'KIF21A (55605)', 'KIF21B (23046)', 'KIF22 (3835)', 'KIF23 (9493)', 'KIF24 (347240)', 'KIF25 (3834)', 'KIF26A (26153)', 'KIF26B (55083)', 'KIF27 (55582)', 'KIF2A (3796)', 'KIF2B (84643)', 'KIF2C (11004)', 'KIF3A (11127)', 'KIF3B (9371)', 'KIF3C (3797)', 'KIF4A (24137)', 'KIF4B (285643)', 'KIF5A (3798)', 'KIF5B (3799)', 'KIF6 (221458)', 'KIF7 (374654)', 'KIF9 (64147)', 'KIFAP3 (22920)', 'KIFBP (26128)', 'KIFC1 (3833)', 'KIFC2 (90990)', 'KIFC3 (3801)', 'KIN (22944)', 'KIR2DL1 (3802)', 'KIR2DL3 (3804)', 'KIR2DL4 (3805)', 'KIR3DL1 (3811)', 'KIR3DL2 (3812)', 'KIR3DL3 (115653)', 'KIRREL1 (55243)', 'KIRREL2 (84063)', 'KIRREL3 (84623)', 'KISS1 (3814)', 'KISS1R (84634)', 'KIT (3815)', 'KITLG (4254)', 'KL (9365)', 'KLB (152831)', 'KLC1 (3831)', 'KLC2 (64837)', 'KLC3 (147700)', 'KLC4 (89953)', 'KLF1 (10661)', 'KLF10 (7071)', 'KLF11 (8462)', 'KLF12 (11278)', 'KLF13 (51621)', 'KLF14 (136259)', 'KLF15 (28999)', 'KLF16 (83855)', 'KLF17 (128209)', 'KLF2 (10365)', 'KLF3 (51274)', 'KLF4 (9314)', 'KLF5 (688)', 'KLF6 (1316)', 'KLF7 (8609)', 'KLF8 (11279)', 'KLF9 (687)', 'KLHDC1 (122773)', 'KLHDC10 (23008)', 'KLHDC2 (23588)', 'KLHDC3 (116138)', 'KLHDC4 (54758)', 'KLHDC7A (127707)', 'KLHDC7B (113730)', 'KLHDC8A (55220)', 'KLHDC8B (200942)', 'KLHDC9 (126823)', 'KLHL1 (57626)', 'KLHL10 (317719)', 'KLHL11 (55175)', 'KLHL12 (59349)', 'KLHL13 (90293)', 'KLHL14 (57565)', 'KLHL15 (80311)', 'KLHL17 (339451)', 'KLHL18 (23276)', 'KLHL2 (11275)', 'KLHL20 (27252)', 'KLHL21 (9903)', 'KLHL22 (84861)', 'KLHL24 (54800)', 'KLHL25 (64410)', 'KLHL26 (55295)', 'KLHL28 (54813)', 'KLHL29 (114818)', 'KLHL3 (26249)', 'KLHL30 (377007)', 'KLHL31 (401265)', 'KLHL32 (114792)', 'KLHL33 (123103)', 'KLHL34 (257240)', 'KLHL35 (283212)', 'KLHL36 (79786)', 'KLHL38 (340359)', 'KLHL4 (56062)', 'KLHL40 (131377)', 'KLHL41 (10324)', 'KLHL42 (57542)', 'KLHL5 (51088)', 'KLHL6 (89857)', 'KLHL7 (55975)', 'KLHL8 (57563)', 'KLHL9 (55958)', 'KLK1 (3816)', 'KLK10 (5655)', 'KLK11 (11012)', 'KLK12 (43849)', 'KLK13 (26085)', 'KLK14 (43847)', 'KLK15 (55554)', 'KLK2 (3817)', 'KLK3 (354)', 'KLK4 (9622)', 'KLK5 (25818)', 'KLK6 (5653)', 'KLK7 (5650)', 'KLK8 (11202)', 'KLK9 (284366)', 'KLKB1 (3818)', 'KLLN (100144748)', 'KLRB1 (3820)', 'KLRC1 (3821)', 'KLRC2 (3822)', 'KLRC3 (3823)', 'KLRC4 (8302)', 'KLRD1 (3824)', 'KLRF1 (51348)', 'KLRF2 (100431172)', 'KLRG1 (10219)', 'KLRG2 (346689)', 'KMO (8564)', 'KMT2A (4297)', 'KMT2B (9757)', 'KMT2C (58508)', 'KMT2D (8085)', 'KMT2E (55904)', 'KMT5A (387893)', 'KMT5B (51111)', 'KMT5C (84787)', 'KNCN (148930)', 'KNDC1 (85442)', 'KNG1 (3827)', 'KNL1 (57082)', 'KNOP1 (400506)', 'KNSTRN (90417)', 'KNTC1 (9735)', 'KPLCE (100129271)', 'KPNA1 (3836)', 'KPNA2 (3838)', 'KPNA3 (3839)', 'KPNA4 (3840)', 'KPNA5 (3841)', 'KPNA6 (23633)', 'KPNA7 (402569)', 'KPNB1 (3837)', 'KPRP (448834)', 'KPTN (11133)', 'KRAS (3845)', 'KRBA2 (124751)', 'KRBOX1 (100506243)', 'KRBOX5 (124411)', 'KRCC1 (51315)', 'KREMEN1 (83999)', 'KREMEN2 (79412)', 'KRI1 (65095)', 'KRIT1 (889)', 'KRR1 (11103)', 'KRT1 (3848)', 'KRT10 (3858)', 'KRT12 (3859)', 'KRT13 (3860)', 'KRT14 (3861)', 'KRT15 (3866)', 'KRT16 (3868)', 'KRT18 (3875)', 'KRT19 (3880)', 'KRT2 (3849)', 'KRT20 (54474)', 'KRT222 (125113)', 'KRT23 (25984)', 'KRT24 (192666)', 'KRT25 (147183)', 'KRT26 (353288)', 'KRT27 (342574)', 'KRT28 (162605)', 'KRT3 (3850)', 'KRT31 (3881)', 'KRT32 (3882)', 'KRT33A (3883)', 'KRT33B (3884)', 'KRT34 (3885)', 'KRT35 (3886)', 'KRT36 (8689)', 'KRT37 (8688)', 'KRT38 (8687)', 'KRT39 (390792)', 'KRT4 (3851)', 'KRT40 (125115)', 'KRT5 (3852)', 'KRT6A (3853)', 'KRT6B (3854)', 'KRT6C (286887)', 'KRT7 (3855)', 'KRT71 (112802)', 'KRT72 (140807)', 'KRT73 (319101)', 'KRT74 (121391)', 'KRT75 (9119)', 'KRT76 (51350)', 'KRT77 (374454)', 'KRT78 (196374)', 'KRT79 (338785)', 'KRT8 (3856)', 'KRT80 (144501)', 'KRT81 (3887)', 'KRT82 (3888)', 'KRT83 (3889)', 'KRT84 (3890)', 'KRT85 (3891)', 'KRT86 (3892)', 'KRT9 (3857)', 'KRTAP1-1 (81851)', 'KRTAP1-4 (728255)', 'KRTAP1-5 (83895)', 'KRTAP10-1 (386677)', 'KRTAP10-10 (353333)', 'KRTAP10-11 (386678)', 'KRTAP10-2 (386679)', 'KRTAP10-3 (386682)', 'KRTAP10-5 (386680)', 'KRTAP10-8 (386681)', 'KRTAP10-9 (386676)', 'KRTAP11-1 (337880)', 'KRTAP12-1 (353332)', 'KRTAP12-2 (353323)', 'KRTAP12-3 (386683)', 'KRTAP12-4 (386684)', 'KRTAP13-1 (140258)', 'KRTAP13-2 (337959)', 'KRTAP13-3 (337960)', 'KRTAP13-4 (284827)', 'KRTAP15-1 (254950)', 'KRTAP16-1 (100505753)', 'KRTAP17-1 (83902)', 'KRTAP19-1 (337882)', 'KRTAP19-2 (337969)', 'KRTAP19-3 (337970)', 'KRTAP19-4 (337971)', 'KRTAP19-5 (337972)', 'KRTAP19-6 (337973)', 'KRTAP19-7 (337974)', 'KRTAP19-8 (728299)', 'KRTAP20-1 (337975)', 'KRTAP20-2 (337976)', 'KRTAP20-3 (337985)', 'KRTAP21-1 (337977)', 'KRTAP21-2 (337978)', 'KRTAP21-3 (100288323)', 'KRTAP22-1 (337979)', 'KRTAP22-2 (100288287)', 'KRTAP23-1 (337963)', 'KRTAP24-1 (643803)', 'KRTAP25-1 (100131902)', 'KRTAP26-1 (388818)', 'KRTAP27-1 (643812)', 'KRTAP29-1 (100533177)', 'KRTAP3-1 (83896)', 'KRTAP3-2 (83897)', 'KRTAP3-3 (85293)', 'KRTAP4-11 (653240)', 'KRTAP4-2 (85291)', 'KRTAP4-3 (85290)', 'KRTAP4-4 (84616)', 'KRTAP4-5 (85289)', 'KRTAP4-6 (81871)', 'KRTAP4-8 (728224)', 'KRTAP4-9 (100132386)', 'KRTAP5-1 (387264)', 'KRTAP5-10 (387273)', 'KRTAP5-11 (440051)', 'KRTAP5-2 (440021)', 'KRTAP5-3 (387266)', 'KRTAP5-5 (439915)', 'KRTAP5-6 (440023)', 'KRTAP5-7 (440050)', 'KRTAP5-9 (3846)', 'KRTAP6-1 (337966)', 'KRTAP6-2 (337967)', 'KRTAP8-1 (337879)', 'KRTAP9-1 (728318)', 'KRTAP9-6 (100507608)', 'KRTAP9-8 (83901)', 'KRTAP9-9 (81870)', 'KRTCAP2 (200185)', 'KRTCAP3 (200634)', 'KRTDAP (388533)', 'KSR1 (8844)', 'KSR2 (283455)', 'KTI12 (112970)', 'KTN1 (3895)', 'KXD1 (79036)', 'KY (339855)', 'KYAT1 (883)', 'KYAT3 (56267)', 'KYNU (8942)', 'L1CAM (3897)', 'L1TD1 (54596)', 'L2HGDH (79944)', 'L3HYPDH (112849)', 'L3MBTL1 (26013)', 'L3MBTL2 (83746)', 'L3MBTL3 (84456)', 'L3MBTL4 (91133)', 'LACC1 (144811)', 'LACRT (90070)', 'LACTB (114294)', 'LACTB2 (51110)', 'LAD1 (3898)', 'LAG3 (3902)', 'LAGE3 (8270)', 'LAIR1 (3903)', 'LAIR2 (3904)', 'LALBA (3906)', 'LAMA1 (284217)', 'LAMA2 (3908)', 'LAMA3 (3909)', 'LAMA4 (3910)', 'LAMA5 (3911)', 'LAMB1 (3912)', 'LAMB2 (3913)', 'LAMB3 (3914)', 'LAMB4 (22798)', 'LAMC1 (3915)', 'LAMC2 (3918)', 'LAMC3 (10319)', 'LAMP1 (3916)', 'LAMP2 (3920)', 'LAMP3 (27074)', 'LAMP5 (24141)', 'LAMTOR1 (55004)', 'LAMTOR2 (28956)', 'LAMTOR3 (8649)', 'LAMTOR4 (389541)', 'LAMTOR5 (10542)', 'LANCL1 (10314)', 'LANCL2 (55915)', 'LANCL3 (347404)', 'LAP3 (51056)', 'LAPTM4A (9741)', 'LAPTM4B (55353)', 'LAPTM5 (7805)', 'LARGE1 (9215)', 'LARGE2 (120071)', 'LARP1 (23367)', 'LARP1B (55132)', 'LARP4 (113251)', 'LARP4B (23185)', 'LARP6 (55323)', 'LARP7 (51574)', 'LARS1 (51520)', 'LARS2 (23395)', 'LAS1L (81887)', 'LASP1 (3927)', 'LAT (27040)', 'LAT2 (7462)', 'LATS1 (9113)', 'LATS2 (26524)', 'LAX1 (54900)', 'LAYN (143903)', 'LBH (81606)', 'LBHD1 (79081)', 'LBP (3929)', 'LBR (3930)', 'LBX1 (10660)', 'LBX2 (85474)', 'LCA5 (167691)', 'LCA5L (150082)', 'LCAT (3931)', 'LCE1B (353132)', 'LCE1D (353134)', 'LCE1F (353137)', 'LCE2A (353139)', 'LCE2B (26239)', 'LCE2D (353141)', 'LCE3A (353142)', 'LCE3B (353143)', 'LCE3C (353144)', 'LCE3D (84648)', 'LCE3E (353145)', 'LCE4A (199834)', 'LCE5A (254910)', 'LCE6A (448835)', 'LCK (3932)', 'LCLAT1 (253558)', 'LCMT1 (51451)', 'LCMT2 (9836)', 'LCN1 (3933)', 'LCN10 (414332)', 'LCN12 (286256)', 'LCN15 (389812)', 'LCN2 (3934)', 'LCN6 (158062)', 'LCN8 (138307)', 'LCN9 (392399)', 'LCNL1 (401562)', 'LCOR (84458)', 'LCORL (254251)', 'LCP1 (3936)', 'LCP2 (3937)', 'LCT (3938)', 'LCTL (197021)', 'LDAF1 (57146)', 'LDAH (60526)', 'LDB1 (8861)', 'LDB2 (9079)', 'LDB3 (11155)', 'LDHA (3939)', 'LDHAL6A (160287)', 'LDHAL6B (92483)', 'LDHB (3945)', 'LDHC (3948)', 'LDHD (197257)', 'LDLR (3949)', 'LDLRAD1 (388633)', 'LDLRAD2 (401944)', 'LDLRAD3 (143458)', 'LDLRAD4 (753)', 'LDLRAP1 (26119)', 'LDOC1 (23641)', 'LEAP2 (116842)', 'LECT2 (3950)', 'LEF1 (51176)', 'LEFTY1 (10637)', 'LEFTY2 (7044)', 'LEKR1 (389170)', 'LELP1 (149018)', 'LEMD1 (93273)', 'LEMD2 (221496)', 'LEMD3 (23592)', 'LENEP (55891)', 'LENG1 (79165)', 'LENG8 (114823)', 'LENG9 (94059)', 'LEO1 (123169)', 'LEP (3952)', 'LEPR (3953)', 'LEPROT (54741)', 'LEPROTL1 (23484)', 'LETM1 (3954)', 'LETM2 (137994)', 'LETMD1 (25875)', 'LFNG (3955)', 'LGALS1 (3956)', 'LGALS12 (85329)', 'LGALS13 (29124)', 'LGALS14 (56891)', 'LGALS16 (148003)', 'LGALS2 (3957)', 'LGALS3 (3958)', 'LGALS3BP (3959)', 'LGALS4 (3960)', 'LGALS7 (3963)', 'LGALS7B (653499)', 'LGALS8 (3964)', 'LGALS9 (3965)', 'LGALS9B (284194)', 'LGALS9C (654346)', 'LGALSL (29094)', 'LGI1 (9211)', 'LGI2 (55203)', 'LGI3 (203190)', 'LGI4 (163175)', 'LGMN (5641)', 'LGR4 (55366)', 'LGR5 (8549)', 'LGR6 (59352)', 'LGSN (51557)', 'LHB (3972)', 'LHCGR (3973)', 'LHFPL1 (340596)', 'LHFPL2 (10184)', 'LHFPL4 (375323)', 'LHFPL5 (222662)', 'LHFPL6 (10186)', 'LHFPL7 (255349)', 'LHPP (64077)', 'LHX1 (3975)', 'LHX2 (9355)', 'LHX3 (8022)', 'LHX4 (89884)', 'LHX5 (64211)', 'LHX6 (26468)', 'LHX8 (431707)', 'LHX9 (56956)', 'LIAS (11019)', 'LIAT1 (400566)', 'LIF (3976)', 'LIFR (3977)', 'LIG1 (3978)', 'LIG3 (3980)', 'LIG4 (3981)', 'LILRA1 (11024)', 'LILRA2 (11027)', 'LILRA4 (23547)', 'LILRA5 (353514)', 'LILRB1 (10859)', 'LILRB2 (10288)', 'LILRB3 (11025)', 'LILRB4 (11006)', 'LILRB5 (10990)', 'LIM2 (3982)', 'LIMA1 (51474)', 'LIMCH1 (22998)', 'LIMD1 (8994)', 'LIMD2 (80774)', 'LIME1 (54923)', 'LIMK1 (3984)', 'LIMK2 (3985)', 'LIMS2 (55679)', 'LIN28A (79727)', 'LIN28B (389421)', 'LIN37 (55957)', 'LIN52 (91750)', 'LIN54 (132660)', 'LIN7A (8825)', 'LIN7B (64130)', 'LIN7C (55327)', 'LIN9 (286826)', 'LINGO1 (84894)', 'LINGO2 (158038)', 'LINGO3 (645191)', 'LINGO4 (339398)', 'LINS1 (55180)', 'LIPA (3988)', 'LIPC (3990)', 'LIPE (3991)', 'LIPF (8513)', 'LIPG (9388)', 'LIPH (200879)', 'LIPI (149998)', 'LIPJ (142910)', 'LIPK (643414)', 'LIPM (340654)', 'LIPN (643418)', 'LIPT1 (51601)', 'LIPT2 (387787)', 'LITAF (9516)', 'LIX1 (167410)', 'LIX1L (128077)', 'LKAAEAR1 (198437)', 'LLCFC1 (135927)', 'LLGL1 (3996)', 'LLGL2 (3993)', 'LLPH (84298)', 'LMAN1 (3998)', 'LMAN1L (79748)', 'LMAN2 (10960)', 'LMAN2L (81562)', 'LMBR1 (64327)', 'LMBR1L (55716)', 'LMBRD1 (55788)', 'LMBRD2 (92255)', 'LMCD1 (29995)', 'LMF1 (64788)', 'LMF2 (91289)', 'LMNA (4000)', 'LMNB1 (4001)', 'LMNB2 (84823)', 'LMNTD1 (160492)', 'LMNTD2 (256329)', 'LMO1 (4004)', 'LMO2 (4005)', 'LMO3 (55885)', 'LMO4 (8543)', 'LMO7 (4008)', 'LMOD1 (25802)', 'LMOD2 (442721)', 'LMOD3 (56203)', 'LMTK2 (22853)', 'LMTK3 (114783)', 'LMX1A (4009)', 'LMX1B (4010)', 'LNP1 (348801)', 'LNPEP (4012)', 'LNPK (80856)', 'LNX1 (84708)', 'LNX2 (222484)', 'LONP1 (9361)', 'LONP2 (83752)', 'LONRF1 (91694)', 'LONRF2 (164832)', 'LONRF3 (79836)', 'LORICRIN (4014)', 'LOX (4015)', 'LOXHD1 (125336)', 'LOXL1 (4016)', 'LOXL2 (4017)', 'LOXL3 (84695)', 'LOXL4 (84171)', 'LPA (4018)', 'LPAR1 (1902)', 'LPAR3 (23566)', 'LPAR4 (2846)', 'LPAR5 (57121)', 'LPAR6 (10161)', 'LPCAT1 (79888)', 'LPCAT2 (54947)', 'LPCAT3 (10162)', 'LPCAT4 (254531)', 'LPGAT1 (9926)', 'LPIN1 (23175)', 'LPIN2 (9663)', 'LPIN3 (64900)', 'LPL (4023)', 'LPO (4025)', 'LPP (4026)', 'LPXN (9404)', 'LRAT (9227)', 'LRATD1 (151354)', 'LRATD2 (157638)', 'LRBA (987)', 'LRCH1 (23143)', 'LRCH2 (57631)', 'LRCH3 (84859)', 'LRCH4 (4034)', 'LRFN1 (57622)', 'LRFN2 (57497)', 'LRFN3 (79414)', 'LRFN4 (78999)', 'LRFN5 (145581)', 'LRG1 (116844)', 'LRGUK (136332)', 'LRIF1 (55791)', 'LRIG1 (26018)', 'LRIG2 (9860)', 'LRIG3 (121227)', 'LRIT1 (26103)', 'LRIT2 (340745)', 'LRIT3 (345193)', 'LRMDA (83938)', 'LRP1 (4035)', 'LRP10 (26020)', 'LRP11 (84918)', 'LRP12 (29967)', 'LRP1B (53353)', 'LRP2 (4036)', 'LRP2BP (55805)', 'LRP3 (4037)', 'LRP4 (4038)', 'LRP5 (4041)', 'LRP6 (4040)', 'LRP8 (7804)', 'LRPAP1 (4043)', 'LRPPRC (10128)', 'LRR1 (122769)', 'LRRC1 (55227)', 'LRRC10 (376132)', 'LRRC10B (390205)', 'LRRC14 (9684)', 'LRRC14B (389257)', 'LRRC15 (131578)', 'LRRC17 (10234)', 'LRRC18 (474354)', 'LRRC19 (64922)', 'LRRC2 (79442)', 'LRRC20 (55222)', 'LRRC23 (10233)', 'LRRC24 (441381)', 'LRRC25 (126364)', 'LRRC26 (389816)', 'LRRC27 (80313)', 'LRRC28 (123355)', 'LRRC3 (81543)', 'LRRC30 (339291)', 'LRRC31 (79782)', 'LRRC32 (2615)', 'LRRC34 (151827)', 'LRRC36 (55282)', 'LRRC37A2 (474170)', 'LRRC37A3 (374819)', 'LRRC37B (114659)', 'LRRC38 (126755)', 'LRRC39 (127495)', 'LRRC3B (116135)', 'LRRC3C (100505591)', 'LRRC4 (64101)', 'LRRC40 (55631)', 'LRRC41 (10489)', 'LRRC42 (115353)', 'LRRC43 (254050)', 'LRRC45 (201255)', 'LRRC46 (90506)', 'LRRC47 (57470)', 'LRRC49 (54839)', 'LRRC4B (94030)', 'LRRC4C (57689)', 'LRRC51 (120356739)', 'LRRC52 (440699)', 'LRRC55 (219527)', 'LRRC56 (115399)', 'LRRC57 (255252)', 'LRRC58 (116064)', 'LRRC59 (55379)', 'LRRC61 (65999)', 'LRRC63 (220416)', 'LRRC66 (339977)', 'LRRC7 (57554)', 'LRRC70 (100130733)', 'LRRC71 (149499)', 'LRRC72 (100506049)', 'LRRC73 (221424)', 'LRRC74A (145497)', 'LRRC75A (388341)', 'LRRC75B (388886)', 'LRRC8A (56262)', 'LRRC8B (23507)', 'LRRC8C (84230)', 'LRRC8D (55144)', 'LRRC8E (80131)', 'LRRCC1 (85444)', 'LRRD1 (401387)', 'LRRFIP1 (9208)', 'LRRFIP2 (9209)', 'LRRIQ1 (84125)', 'LRRIQ3 (127255)', 'LRRIQ4 (344657)', 'LRRK1 (79705)', 'LRRK2 (120892)', 'LRRN1 (57633)', 'LRRN2 (10446)', 'LRRN3 (54674)', 'LRRN4 (164312)', 'LRRN4CL (221091)', 'LRRTM1 (347730)', 'LRRTM2 (26045)', 'LRRTM3 (347731)', 'LRRTM4 (80059)', 'LRSAM1 (90678)', 'LRTM1 (57408)', 'LRTM2 (654429)', 'LRWD1 (222229)', 'LSAMP (4045)', 'LSG1 (55341)', 'LSM1 (27257)', 'LSM10 (84967)', 'LSM11 (134353)', 'LSM12 (124801)', 'LSM14A (26065)', 'LSM14B (149986)', 'LSM2 (57819)', 'LSM3 (27258)', 'LSM4 (25804)', 'LSM5 (23658)', 'LSM6 (11157)', 'LSM7 (51690)', 'LSM8 (51691)', 'LSMEM1 (286006)', 'LSMEM2 (132228)', 'LSP1 (4046)', 'LSR (51599)', 'LSS (4047)', 'LST1 (7940)', 'LTA (4049)', 'LTA4H (4048)', 'LTB (4050)', 'LTB4R (1241)', 'LTB4R2 (56413)', 'LTBP1 (4052)', 'LTBP2 (4053)', 'LTBP3 (4054)', 'LTBR (4055)', 'LTC4S (4056)', 'LTF (4057)', 'LTK (4058)', 'LTN1 (26046)', 'LTO1 (220064)', 'LTV1 (84946)', 'LUC7L (55692)', 'LUC7L2 (51631)', 'LUC7L3 (51747)', 'LUM (4060)', 'LURAP1 (541468)', 'LURAP1L (286343)', 'LUZP1 (7798)', 'LUZP2 (338645)', 'LUZP4 (51213)', 'LVRN (206338)', 'LXN (56925)', 'LY6D (8581)', 'LY6E (4061)', 'LY6G5B (58496)', 'LY6G5C (80741)', 'LY6G6C (80740)', 'LY6H (4062)', 'LY6K (54742)', 'LY75 (4065)', 'LY86 (9450)', 'LY9 (4063)', 'LY96 (23643)', 'LYAR (55646)', 'LYG1 (129530)', 'LYG2 (254773)', 'LYL1 (4066)', 'LYN (4067)', 'LYNX1 (66004)', 'LYPD1 (116372)', 'LYPD2 (137797)', 'LYPD3 (27076)', 'LYPD4 (147719)', 'LYPD5 (284348)', 'LYPD6 (130574)', 'LYPD6B (130576)', 'LYPLA1 (10434)', 'LYPLA2 (11313)', 'LYPLAL1 (127018)', 'LYRM1 (57149)', 'LYRM2 (57226)', 'LYRM4 (57128)', 'LYRM7 (90624)', 'LYRM9 (201229)', 'LYSET (26175)', 'LYSMD1 (388695)', 'LYSMD2 (256586)', 'LYSMD3 (116068)', 'LYSMD4 (145748)', 'LYST (1130)', 'LYVE1 (10894)', 'LYZ (4069)', 'LYZL4 (131375)', 'LYZL6 (57151)', 'LZIC (84328)', 'LZTFL1 (54585)', 'LZTR1 (8216)', 'LZTS1 (11178)', 'LZTS2 (84445)', 'LZTS3 (9762)', 'M1AP (130951)', 'M6PR (4074)', 'MAB21L1 (4081)', 'MAB21L2 (10586)', 'MAB21L3 (126868)', 'MAB21L4 (79919)', 'MACC1 (346389)', 'MACF1 (23499)', 'MACIR (90355)', 'MACO1 (55219)', 'MACROD1 (28992)', 'MACROD2 (140733)', 'MACROH2A1 (9555)', 'MACROH2A2 (55506)', 'MAD1L1 (8379)', 'MAD2L1 (4085)', 'MAD2L1BP (9587)', 'MAD2L2 (10459)', 'MADCAM1 (8174)', 'MADD (8567)', 'MAEA (10296)', 'MAEL (84944)', 'MAF (4094)', 'MAF1 (84232)', 'MAFA (389692)', 'MAFB (9935)', 'MAFF (23764)', 'MAFG (4097)', 'MAFK (7975)', 'MAG (4099)', 'MAGEA1 (4100)', 'MAGEA10 (4109)', 'MAGEA11 (4110)', 'MAGEA12 (4111)', 'MAGEA4 (4103)', 'MAGEA6 (4105)', 'MAGEA8 (4107)', 'MAGEB1 (4112)', 'MAGEB10 (139422)', 'MAGEB16 (139604)', 'MAGEB17 (645864)', 'MAGEB18 (286514)', 'MAGEB2 (4113)', 'MAGEB3 (4114)', 'MAGEB4 (4115)', 'MAGEB5 (347541)', 'MAGEB6 (158809)', 'MAGEC1 (9947)', 'MAGEC2 (51438)', 'MAGEC3 (139081)', 'MAGED1 (9500)', 'MAGED2 (10916)', 'MAGEE1 (57692)', 'MAGEE2 (139599)', 'MAGEF1 (64110)', 'MAGEH1 (28986)', 'MAGI1 (9223)', 'MAGI2 (9863)', 'MAGI3 (260425)', 'MAGIX (79917)', 'MAGOH (4116)', 'MAGOHB (55110)', 'MAGT1 (84061)', 'MAIP1 (79568)', 'MAJIN (283129)', 'MAK (4117)', 'MAK16 (84549)', 'MAL (4118)', 'MALL (7851)', 'MALSU1 (115416)', 'MALT1 (10892)', 'MAMDC2 (256691)', 'MAMDC4 (158056)', 'MAML1 (9794)', 'MAML2 (84441)', 'MAML3 (55534)', 'MAMLD1 (10046)', 'MAMSTR (284358)', 'MAN1A1 (4121)', 'MAN1A2 (10905)', 'MAN1B1 (11253)', 'MAN1C1 (57134)', 'MAN2A1 (4124)', 'MAN2A2 (4122)', 'MAN2B1 (4125)', 'MAN2B2 (23324)', 'MAN2C1 (4123)', 'MANBA (4126)', 'MANBAL (63905)', 'MANEA (79694)', 'MANEAL (149175)', 'MANF (7873)', 'MANSC1 (54682)', 'MANSC4 (100287284)', 'MAOA (4128)', 'MAOB (4129)', 'MAP10 (54627)', 'MAP1A (4130)', 'MAP1B (4131)', 'MAP1LC3A (84557)', 'MAP1LC3B (81631)', 'MAP1LC3B2 (643246)', 'MAP1LC3C (440738)', 'MAP1S (55201)', 'MAP2 (4133)', 'MAP2K1 (5604)', 'MAP2K2 (5605)', 'MAP2K3 (5606)', 'MAP2K4 (6416)', 'MAP2K5 (5607)', 'MAP2K6 (5608)', 'MAP2K7 (5609)', 'MAP3K1 (4214)', 'MAP3K10 (4294)', 'MAP3K11 (4296)', 'MAP3K12 (7786)', 'MAP3K13 (9175)', 'MAP3K19 (80122)', 'MAP3K2 (10746)', 'MAP3K20 (51776)', 'MAP3K21 (84451)', 'MAP3K3 (4215)', 'MAP3K4 (4216)', 'MAP3K5 (4217)', 'MAP3K6 (9064)', 'MAP3K7 (6885)', 'MAP3K7CL (56911)', 'MAP3K8 (1326)', 'MAP3K9 (4293)', 'MAP4 (4134)', 'MAP4K1 (11184)', 'MAP4K2 (5871)', 'MAP4K3 (8491)', 'MAP4K4 (9448)', 'MAP6 (4135)', 'MAP6D1 (79929)', 'MAP7 (9053)', 'MAP7D1 (55700)', 'MAP7D2 (256714)', 'MAP7D3 (79649)', 'MAP9 (79884)', 'MAPK1 (5594)', 'MAPK10 (5602)', 'MAPK11 (5600)', 'MAPK12 (6300)', 'MAPK13 (5603)', 'MAPK14 (1432)', 'MAPK15 (225689)', 'MAPK1IP1L (93487)', 'MAPK3 (5595)', 'MAPK4 (5596)', 'MAPK6 (5597)', 'MAPK7 (5598)', 'MAPK8 (5599)', 'MAPK8IP1 (9479)', 'MAPK8IP3 (23162)', 'MAPK9 (5601)', 'MAPKAP1 (79109)', 'MAPKAPK2 (9261)', 'MAPKAPK3 (7867)', 'MAPKAPK5 (8550)', 'MAPKBP1 (23005)', 'MAPRE1 (22919)', 'MAPRE2 (10982)', 'MAPRE3 (22924)', 'MAPT (4137)', 'MARCHF1 (55016)', 'MARCHF10 (162333)', 'MARCHF11 (441061)', 'MARCHF2 (51257)', 'MARCHF3 (115123)', 'MARCHF4 (57574)', 'MARCHF5 (54708)', 'MARCHF6 (10299)', 'MARCHF7 (64844)', 'MARCHF8 (220972)', 'MARCHF9 (92979)', 'MARCKS (4082)', 'MARCKSL1 (65108)', 'MARCO (8685)', 'MARF1 (9665)', 'MARK1 (4139)', 'MARK2 (2011)', 'MARK3 (4140)', 'MARK4 (57787)', 'MARS1 (4141)', 'MARS2 (92935)', 'MARVELD2 (153562)', 'MARVELD3 (91862)', 'MAS1 (4142)', 'MAS1L (116511)', 'MASP1 (5648)', 'MASP2 (10747)', 'MAST1 (22983)', 'MAST2 (23139)', 'MAST3 (23031)', 'MAST4 (375449)', 'MASTL (84930)', 'MAT1A (4143)', 'MAT2A (4144)', 'MAT2B (27430)', 'MATCAP1 (653319)', 'MATCAP2 (23366)', 'MATK (4145)', 'MATN1 (4146)', 'MATN2 (4147)', 'MATN3 (4148)', 'MATN4 (8785)', 'MATR3 (9782)', 'MAU2 (23383)', 'MAVS (57506)', 'MAX (4149)', 'MAZ (4150)', 'MB (4151)', 'MB21D2 (151963)', 'MBD1 (4152)', 'MBD2 (8932)', 'MBD3 (53615)', 'MBD3L1 (85509)', 'MBD4 (8930)', 'MBD5 (55777)', 'MBD6 (114785)', 'MBIP (51562)', 'MBL2 (4153)', 'MBLAC1 (255374)', 'MBLAC2 (153364)', 'MBNL1 (4154)', 'MBNL2 (10150)', 'MBNL3 (55796)', 'MBOAT1 (154141)', 'MBOAT2 (129642)', 'MBOAT4 (619373)', 'MBOAT7 (79143)', 'MBP (4155)', 'MBTD1 (54799)', 'MBTPS1 (8720)', 'MBTPS2 (51360)', 'MC1R (4157)', 'MC2R (4158)', 'MC3R (4159)', 'MC4R (4160)', 'MC5R (4161)', 'MCAM (4162)', 'MCAT (27349)', 'MCC (4163)', 'MCCC1 (56922)', 'MCCC2 (64087)', 'MCCD1 (401250)', 'MCEE (84693)', 'MCEMP1 (199675)', 'MCF2 (4168)', 'MCF2L (23263)', 'MCF2L2 (23101)', 'MCFD2 (90411)', 'MCHR1 (2847)', 'MCHR2 (84539)', 'MCIDAS (345643)', 'MCL1 (4170)', 'MCM10 (55388)', 'MCM2 (4171)', 'MCM3 (4172)', 'MCM3AP (8888)', 'MCM4 (4173)', 'MCM5 (4174)', 'MCM6 (4175)', 'MCM7 (4176)', 'MCM8 (84515)', 'MCM9 (254394)', 'MCMBP (79892)', 'MCMDC2 (157777)', 'MCOLN1 (57192)', 'MCOLN2 (255231)', 'MCOLN3 (55283)', 'MCPH1 (79648)', 'MCRIP1 (348262)', 'MCRIP2 (84331)', 'MCRS1 (10445)', 'MCTP1 (79772)', 'MCTP2 (55784)', 'MCTS1 (28985)', 'MCU (90550)', 'MCUB (55013)', 'MCUR1 (63933)', 'MDC1 (9656)', 'MDFI (4188)', 'MDFIC (29969)', 'MDGA1 (266727)', 'MDGA2 (161357)', 'MDH1 (4190)', 'MDH1B (130752)', 'MDH2 (4191)', 'MDK (4192)', 'MDM1 (56890)', 'MDM2 (4193)', 'MDM4 (4194)', 'MDN1 (23195)', 'MDP1 (145553)', 'ME1 (4199)', 'ME2 (4200)', 'ME3 (10873)', 'MEA1 (4201)', 'MEAF6 (64769)', 'MEAK7 (57707)', 'MECOM (2122)', 'MECP2 (4204)', 'MECR (51102)', 'MED1 (5469)', 'MED10 (84246)', 'MED11 (400569)', 'MED12 (9968)', 'MED12L (116931)', 'MED13 (9969)', 'MED13L (23389)', 'MED14 (9282)', 'MED15 (51586)', 'MED16 (10025)', 'MED17 (9440)', 'MED18 (54797)', 'MED19 (219541)', 'MED20 (9477)', 'MED21 (9412)', 'MED22 (6837)', 'MED23 (9439)', 'MED24 (9862)', 'MED25 (81857)', 'MED26 (9441)', 'MED27 (9442)', 'MED28 (80306)', 'MED29 (55588)', 'MED30 (90390)', 'MED31 (51003)', 'MED4 (29079)', 'MED6 (10001)', 'MED7 (9443)', 'MED8 (112950)', 'MED9 (55090)', 'MEDAG (84935)', 'MEF2A (4205)', 'MEF2B (100271849)', 'MEF2C (4208)', 'MEF2D (4209)', 'MEFV (4210)', 'MEGF10 (84466)', 'MEGF11 (84465)', 'MEGF6 (1953)', 'MEGF8 (1954)', 'MEGF9 (1955)', 'MEI1 (150365)', 'MEI4 (101928601)', 'MEIG1 (644890)', 'MEIOB (254528)', 'MEIOC (284071)', 'MEIS1 (4211)', 'MEIS2 (4212)', 'MEIS3 (56917)', 'MELK (9833)', 'MELTF (4241)', 'MEMO1 (51072)', 'MEN1 (4221)', 'MEOX1 (4222)', 'MEOX2 (4223)', 'MEP1A (4224)', 'MEP1B (4225)', 'MEPCE (56257)', 'MEPE (56955)', 'MERTK (10461)', 'MESD (23184)', 'MESP1 (55897)', 'MESP2 (145873)', 'MEST (4232)', 'MET (4233)', 'METAP1 (23173)', 'METAP1D (254042)', 'METAP2 (10988)', 'METRN (79006)', 'METRNL (284207)', 'METTL1 (4234)', 'METTL13 (51603)', 'METTL14 (57721)', 'METTL15 (196074)', 'METTL16 (79066)', 'METTL17 (64745)', 'METTL18 (92342)', 'METTL21A (151194)', 'METTL21C (196541)', 'METTL22 (79091)', 'METTL23 (124512)', 'METTL24 (728464)', 'METTL25 (84190)', 'METTL25B (51093)', 'METTL26 (84326)', 'METTL27 (155368)', 'METTL2A (339175)', 'METTL2B (55798)', 'METTL3 (56339)', 'METTL4 (64863)', 'METTL5 (29081)', 'METTL6 (131965)', 'METTL8 (79828)', 'METTL9 (51108)', 'MEX3A (92312)', 'MEX3B (84206)', 'MEX3C (51320)', 'MEX3D (399664)', 'MFAP1 (4236)', 'MFAP2 (4237)', 'MFAP3 (4238)', 'MFAP3L (9848)', 'MFAP4 (4239)', 'MFAP5 (8076)', 'MFF (56947)', 'MFGE8 (4240)', 'MFHAS1 (9258)', 'MFN1 (55669)', 'MFN2 (9927)', 'MFNG (4242)', 'MFRP (83552)', 'MFSD1 (64747)', 'MFSD10 (10227)', 'MFSD11 (79157)', 'MFSD12 (126321)', 'MFSD13A (79847)', 'MFSD14A (64645)', 'MFSD14B (84641)', 'MFSD2A (84879)', 'MFSD2B (388931)', 'MFSD3 (113655)', 'MFSD4A (148808)', 'MFSD4B (91749)', 'MFSD5 (84975)', 'MFSD6 (54842)', 'MFSD6L (162387)', 'MFSD8 (256471)', 'MFSD9 (84804)', 'MGA (23269)', 'MGAM (8972)', 'MGARP (84709)', 'MGAT1 (4245)', 'MGAT2 (4247)', 'MGAT3 (4248)', 'MGAT4A (11320)', 'MGAT4B (11282)', 'MGAT4C (25834)', 'MGAT4D (152586)', 'MGAT5 (4249)', 'MGAT5B (146664)', 'MGLL (11343)', 'MGME1 (92667)', 'MGMT (4255)', 'MGP (4256)', 'MGRN1 (23295)', 'MGST1 (4257)', 'MGST2 (4258)', 'MGST3 (4259)', 'MIA (8190)', 'MIA2 (4253)', 'MIA3 (375056)', 'MIB1 (57534)', 'MIB2 (142678)', 'MICA (100507436)', 'MICAL1 (64780)', 'MICAL2 (9645)', 'MICAL3 (57553)', 'MICALL1 (85377)', 'MICALL2 (79778)', 'MICB (4277)', 'MICOS10 (440574)', 'MICOS13 (125988)', 'MICU1 (10367)', 'MICU2 (221154)', 'MICU3 (286097)', 'MID1 (4281)', 'MID1IP1 (58526)', 'MID2 (11043)', 'MIDEAS (91748)', 'MIDN (90007)', 'MIEF1 (54471)', 'MIEF2 (125170)', 'MIEN1 (84299)', 'MIER1 (57708)', 'MIER2 (54531)', 'MIER3 (166968)', 'MIF (4282)', 'MIF4GD (57409)', 'MIGA1 (374986)', 'MIGA2 (84895)', 'MIIP (60672)', 'MINAR1 (23251)', 'MINAR2 (100127206)', 'MINDY1 (55793)', 'MINDY2 (54629)', 'MINDY3 (80013)', 'MINDY4 (84182)', 'MINK1 (50488)', 'MINPP1 (9562)', 'MIOS (54468)', 'MIOX (55586)', 'MIP (4284)', 'MIPEP (4285)', 'MIPOL1 (145282)', 'MIS12 (79003)', 'MIS18A (54069)', 'MIS18BP1 (55320)', 'MISP (126353)', 'MITD1 (129531)', 'MITF (4286)', 'MIX23 (131076)', 'MIXL1 (83881)', 'MKI67 (4288)', 'MKKS (8195)', 'MKLN1 (4289)', 'MKNK1 (8569)', 'MKNK2 (2872)', 'MKRN1 (23608)', 'MKRN2 (23609)', 'MKRN2OS (100129480)', 'MKRN3 (7681)', 'MKS1 (54903)', 'MKX (283078)', 'MLANA (2315)', 'MLC1 (23209)', 'MLEC (9761)', 'MLF1 (4291)', 'MLF2 (8079)', 'MLH1 (4292)', 'MLH3 (27030)', 'MLIP (90523)', 'MLKL (197259)', 'MLLT1 (4298)', 'MLLT10 (8028)', 'MLLT11 (10962)', 'MLLT3 (4300)', 'MLLT6 (4302)', 'MLN (4295)', 'MLNR (2862)', 'MLPH (79083)', 'MLST8 (64223)', 'MLX (6945)', 'MLXIPL (51085)', 'MLYCD (23417)', 'MMAA (166785)', 'MMAB (326625)', 'MMACHC (25974)', 'MMADHC (27249)', 'MMD (23531)', 'MMD2 (221938)', 'MME (4311)', 'MMEL1 (79258)', 'MMGT1 (93380)', 'MMP1 (4312)', 'MMP10 (4319)', 'MMP11 (4320)', 'MMP13 (4322)', 'MMP14 (4323)', 'MMP15 (4324)', 'MMP16 (4325)', 'MMP17 (4326)', 'MMP19 (4327)', 'MMP2 (4313)', 'MMP20 (9313)', 'MMP21 (118856)', 'MMP23B (8510)', 'MMP24 (10893)', 'MMP25 (64386)', 'MMP26 (56547)', 'MMP27 (64066)', 'MMP28 (79148)', 'MMP3 (4314)', 'MMP7 (4316)', 'MMP8 (4317)', 'MMP9 (4318)', 'MMRN1 (22915)', 'MMRN2 (79812)', 'MMS19 (64210)', 'MMS22L (253714)', 'MMUT (4594)', 'MN1 (4330)', 'MNAT1 (4331)', 'MND1 (84057)', 'MNDA (4332)', 'MNS1 (55329)', 'MNT (4335)', 'MNX1 (3110)', 'MOAP1 (64112)', 'MOB1A (55233)', 'MOB1B (92597)', 'MOB2 (81532)', 'MOB3A (126308)', 'MOB3B (79817)', 'MOB3C (148932)', 'MOB4 (25843)', 'MOBP (4336)', 'MOCOS (55034)', 'MOCS1 (4337)', 'MOCS2 (4338)', 'MOCS3 (27304)', 'MOG (4340)', 'MOGAT1 (116255)', 'MOGAT2 (80168)', 'MOGAT3 (346606)', 'MOGS (7841)', 'MOK (5891)', 'MON1A (84315)', 'MON1B (22879)', 'MON2 (23041)', 'MORC1 (27136)', 'MORC2 (22880)', 'MORC3 (23515)', 'MORC4 (79710)', 'MORF4L1 (10933)', 'MORF4L2 (9643)', 'MORN1 (79906)', 'MORN2 (729967)', 'MORN3 (283385)', 'MORN4 (118812)', 'MORN5 (254956)', 'MOS (4342)', 'MOSMO (730094)', 'MOSPD1 (56180)', 'MOSPD2 (158747)', 'MOSPD3 (64598)', 'MOV10 (4343)', 'MOV10L1 (54456)', 'MOXD1 (26002)', 'MPC1 (51660)', 'MPC2 (25874)', 'MPDU1 (9526)', 'MPDZ (8777)', 'MPEG1 (219972)', 'MPG (4350)', 'MPHOSPH10 (10199)', 'MPHOSPH6 (10200)', 'MPHOSPH8 (54737)', 'MPHOSPH9 (10198)', 'MPI (4351)', 'MPIG6B (80739)', 'MPL (4352)', 'MPLKIP (136647)', 'MPND (84954)', 'MPO (4353)', 'MPP1 (4354)', 'MPP2 (4355)', 'MPP3 (4356)', 'MPP4 (58538)', 'MPP7 (143098)', 'MPPE1 (65258)', 'MPPED1 (758)', 'MPPED2 (744)', 'MPRIP (23164)', 'MPST (4357)', 'MPV17 (4358)', 'MPV17L (255027)', 'MPV17L2 (84769)', 'MPZ (4359)', 'MPZL1 (9019)', 'MPZL2 (10205)', 'MPZL3 (196264)', 'MR1 (3140)', 'MRAP (56246)', 'MRAP2 (112609)', 'MRAS (22808)', 'MRC1 (4360)', 'MRC2 (9902)', 'MRE11 (4361)', 'MREG (55686)', 'MRFAP1 (93621)', 'MRFAP1L1 (114932)', 'MRGBP (55257)', 'MRGPRD (116512)', 'MRGPRE (116534)', 'MRGPRF (116535)', 'MRGPRG (386746)', 'MRGPRX1 (259249)', 'MRGPRX2 (117194)', 'MRGPRX3 (117195)', 'MRGPRX4 (117196)', 'MRI1 (84245)', 'MRM1 (79922)', 'MRM2 (29960)', 'MRM3 (55178)', 'MRNIP (51149)', 'MRO (83876)', 'MROH1 (727957)', 'MROH2B (133558)', 'MROH6 (642475)', 'MROH7 (374977)', 'MROH9 (80133)', 'MRPL1 (65008)', 'MRPL10 (124995)', 'MRPL11 (65003)', 'MRPL12 (6182)', 'MRPL13 (28998)', 'MRPL14 (64928)', 'MRPL15 (29088)', 'MRPL16 (54948)', 'MRPL17 (63875)', 'MRPL18 (29074)', 'MRPL19 (9801)', 'MRPL2 (51069)', 'MRPL20 (55052)', 'MRPL21 (219927)', 'MRPL22 (29093)', 'MRPL23 (6150)', 'MRPL24 (79590)', 'MRPL27 (51264)', 'MRPL28 (10573)', 'MRPL3 (11222)', 'MRPL30 (51263)', 'MRPL32 (64983)', 'MRPL33 (9553)', 'MRPL34 (64981)', 'MRPL35 (51318)', 'MRPL36 (64979)', 'MRPL37 (51253)', 'MRPL38 (64978)', 'MRPL39 (54148)', 'MRPL4 (51073)', 'MRPL40 (64976)', 'MRPL41 (64975)', 'MRPL42 (28977)', 'MRPL43 (84545)', 'MRPL44 (65080)', 'MRPL45 (84311)', 'MRPL46 (26589)', 'MRPL47 (57129)', 'MRPL48 (51642)', 'MRPL49 (740)', 'MRPL50 (54534)', 'MRPL51 (51258)', 'MRPL52 (122704)', 'MRPL53 (116540)', 'MRPL54 (116541)', 'MRPL55 (128308)', 'MRPL57 (78988)', 'MRPL58 (3396)', 'MRPL9 (65005)', 'MRPS10 (55173)', 'MRPS11 (64963)', 'MRPS12 (6183)', 'MRPS14 (63931)', 'MRPS15 (64960)', 'MRPS16 (51021)', 'MRPS17 (51373)', 'MRPS18A (55168)', 'MRPS18B (28973)', 'MRPS18C (51023)', 'MRPS2 (51116)', 'MRPS21 (54460)', 'MRPS22 (56945)', 'MRPS23 (51649)', 'MRPS24 (64951)', 'MRPS25 (64432)', 'MRPS26 (64949)', 'MRPS27 (23107)', 'MRPS28 (28957)', 'MRPS30 (10884)', 'MRPS31 (10240)', 'MRPS33 (51650)', 'MRPS34 (65993)', 'MRPS35 (60488)', 'MRPS5 (64969)', 'MRPS6 (64968)', 'MRPS7 (51081)', 'MRPS9 (64965)', 'MRRF (92399)', 'MRS2 (57380)', 'MRTFA (57591)', 'MRTFB (57496)', 'MRTO4 (51154)', 'MS4A1 (931)', 'MS4A10 (341116)', 'MS4A12 (54860)', 'MS4A13 (503497)', 'MS4A14 (84689)', 'MS4A15 (219995)', 'MS4A2 (2206)', 'MS4A3 (932)', 'MS4A4A (51338)', 'MS4A5 (64232)', 'MS4A6A (64231)', 'MS4A6E (245802)', 'MS4A7 (58475)', 'MS4A8 (83661)', 'MSANTD1 (345222)', 'MSANTD2 (79684)', 'MSANTD3 (91283)', 'MSANTD4 (84437)', 'MSC (9242)', 'MSGN1 (343930)', 'MSH2 (4436)', 'MSH3 (4437)', 'MSH4 (4438)', 'MSH5 (4439)', 'MSH6 (2956)', 'MSI1 (4440)', 'MSI2 (124540)', 'MSL1 (339287)', 'MSL2 (55167)', 'MSL3 (10943)', 'MSLN (10232)', 'MSMB (4477)', 'MSMO1 (6307)', 'MSMP (692094)', 'MSN (4478)', 'MSR1 (4481)', 'MSRA (4482)', 'MSRB1 (51734)', 'MSRB2 (22921)', 'MSRB3 (253827)', 'MSS51 (118490)', 'MST1 (4485)', 'MST1R (4486)', 'MSTN (2660)', 'MSTO1 (55154)', 'MSX1 (4487)', 'MSX2 (4488)', 'MT1A (4489)', 'MT1B (4490)', 'MT1E (4493)', 'MT1F (4494)', 'MT1G (4495)', 'MT1H (4496)', 'MT1HL1 (645745)', 'MT1M (4499)', 'MT1X (4501)', 'MT2A (4502)', 'MT3 (4504)', 'MT4 (84560)', 'MTA1 (9112)', 'MTA2 (9219)', 'MTA3 (57504)', 'MTAP (4507)', 'MTARC1 (64757)', 'MTARC2 (54996)', 'MTBP (27085)', 'MTCH1 (23787)', 'MTCH2 (23788)', 'MTCL1 (23255)', 'MTCL2 (140710)', 'MTCP1 (4515)', 'MTDH (92140)', 'MTERF1 (7978)', 'MTERF2 (80298)', 'MTERF3 (51001)', 'MTERF4 (130916)', 'MTF1 (4520)', 'MTF2 (22823)', 'MTFMT (123263)', 'MTFP1 (51537)', 'MTFR1 (9650)', 'MTFR1L (56181)', 'MTFR2 (113115)', 'MTG1 (92170)', 'MTG2 (26164)', 'MTHFD1 (4522)', 'MTHFD1L (25902)', 'MTHFD2 (10797)', 'MTHFD2L (441024)', 'MTHFR (4524)', 'MTHFS (10588)', 'MTHFSD (64779)', 'MTIF2 (4528)', 'MTIF3 (219402)', 'MTM1 (4534)', 'MTMR1 (8776)', 'MTMR10 (54893)', 'MTMR11 (10903)', 'MTMR12 (54545)', 'MTMR14 (64419)', 'MTMR2 (8898)', 'MTMR3 (8897)', 'MTMR4 (9110)', 'MTMR6 (9107)', 'MTMR7 (9108)', 'MTMR8 (55613)', 'MTMR9 (66036)', 'MTNAP1 (55028)', 'MTNR1A (4543)', 'MTNR1B (4544)', 'MTO1 (25821)', 'MTOR (2475)', 'MTPAP (55149)', 'MTPN (136319)', 'MTR (4548)', 'MTRES1 (51250)', 'MTREX (23517)', 'MTRF1 (9617)', 'MTRF1L (54516)', 'MTRFR (91574)', 'MTRR (4552)', 'MTSS1 (9788)', 'MTSS2 (92154)', 'MTTP (4547)', 'MTURN (222166)', 'MTUS1 (57509)', 'MTUS2 (23281)', 'MTX1 (4580)', 'MTX2 (10651)', 'MTX3 (345778)', 'MUC1 (4582)', 'MUC12 (10071)', 'MUC15 (143662)', 'MUC16 (94025)', 'MUC17 (140453)', 'MUC20 (200958)', 'MUC21 (394263)', 'MUC22 (100507679)', 'MUC4 (4585)', 'MUC5B (727897)', 'MUC6 (4588)', 'MUC7 (4589)', 'MUCL1 (118430)', 'MUCL3 (135656)', 'MUL1 (79594)', 'MUS81 (80198)', 'MUSK (4593)', 'MUSTN1 (389125)', 'MUTYH (4595)', 'MVB12A (93343)', 'MVB12B (89853)', 'MVD (4597)', 'MVK (4598)', 'MVP (9961)', 'MX1 (4599)', 'MX2 (4600)', 'MXD1 (4084)', 'MXD3 (83463)', 'MXD4 (10608)', 'MXI1 (4601)', 'MXRA5 (25878)', 'MXRA7 (439921)', 'MXRA8 (54587)', 'MYADM (91663)', 'MYADML2 (255275)', 'MYB (4602)', 'MYBBP1A (10514)', 'MYBL1 (4603)', 'MYBL2 (4605)', 'MYBPC1 (4604)', 'MYBPC2 (4606)', 'MYBPC3 (4607)', 'MYBPH (4608)', 'MYBPHL (343263)', 'MYC (4609)', 'MYCBP (26292)', 'MYCBPAP (84073)', 'MYCL (4610)', 'MYCN (4613)', 'MYCT1 (80177)', 'MYD88 (4615)', 'MYDGF (56005)', 'MYEF2 (50804)', 'MYEOV (26579)', 'MYF5 (4617)', 'MYF6 (4618)', 'MYG1 (60314)', 'MYH1 (4619)', 'MYH10 (4628)', 'MYH11 (4629)', 'MYH13 (8735)', 'MYH14 (79784)', 'MYH15 (22989)', 'MYH2 (4620)', 'MYH3 (4621)', 'MYH4 (4622)', 'MYH6 (4624)', 'MYH7 (4625)', 'MYH8 (4626)', 'MYH9 (4627)', 'MYL1 (4632)', 'MYL10 (93408)', 'MYL11 (29895)', 'MYL12A (10627)', 'MYL12B (103910)', 'MYL2 (4633)', 'MYL3 (4634)', 'MYL4 (4635)', 'MYL5 (4636)', 'MYL6 (4637)', 'MYL6B (140465)', 'MYL7 (58498)', 'MYL9 (10398)', 'MYLIP (29116)', 'MYLK (4638)', 'MYLK2 (85366)', 'MYLK3 (91807)', 'MYLK4 (340156)', 'MYMK (389827)', 'MYNN (55892)', 'MYO10 (4651)', 'MYO15A (51168)', 'MYO16 (23026)', 'MYO18A (399687)', 'MYO18B (84700)', 'MYO19 (80179)', 'MYO1A (4640)', 'MYO1B (4430)', 'MYO1C (4641)', 'MYO1D (4642)', 'MYO1E (4643)', 'MYO1F (4542)', 'MYO1G (64005)', 'MYO1H (283446)', 'MYO3A (53904)', 'MYO3B (140469)', 'MYO5A (4644)', 'MYO5B (4645)', 'MYO5C (55930)', 'MYO6 (4646)', 'MYO7A (4647)', 'MYO7B (4648)', 'MYO9A (4649)', 'MYO9B (4650)', 'MYOC (4653)', 'MYOCD (93649)', 'MYOD1 (4654)', 'MYOF (26509)', 'MYOG (4656)', 'MYOM1 (8736)', 'MYOM2 (9172)', 'MYOM3 (127294)', 'MYOT (9499)', 'MYOZ1 (58529)', 'MYOZ2 (51778)', 'MYOZ3 (91977)', 'MYPN (84665)', 'MYPOP (339344)', 'MYRF (745)', 'MYRIP (25924)', 'MYSM1 (114803)', 'MYT1 (4661)', 'MYT1L (23040)', 'MYZAP (100820829)', 'MZB1 (51237)', 'MZF1 (7593)', 'MZT1 (440145)', 'MZT2A (653784)', 'MZT2B (80097)', 'N4BP1 (9683)', 'N4BP2 (55728)', 'N4BP2L1 (90634)', 'N4BP2L2 (10443)', 'N4BP3 (23138)', 'N6AMT1 (29104)', 'NAA10 (8260)', 'NAA11 (84779)', 'NAA15 (80155)', 'NAA16 (79612)', 'NAA20 (51126)', 'NAA25 (80018)', 'NAA30 (122830)', 'NAA35 (60560)', 'NAA38 (84316)', 'NAA40 (79829)', 'NAA50 (80218)', 'NAA60 (79903)', 'NAA80 (24142)', 'NAAA (27163)', 'NAALAD2 (10003)', 'NAALADL1 (10004)', 'NAALADL2 (254827)', 'NAB1 (4664)', 'NAB2 (4665)', 'NABP1 (64859)', 'NABP2 (79035)', 'NACA (4666)', 'NACA2 (342538)', 'NACAD (23148)', 'NACC1 (112939)', 'NACC2 (138151)', 'NADK (65220)', 'NADK2 (133686)', 'NADSYN1 (55191)', 'NAE1 (8883)', 'NAF1 (92345)', 'NAGA (4668)', 'NAGK (55577)', 'NAGLU (4669)', 'NAGPA (51172)', 'NAGS (162417)', 'NAIF1 (203245)', 'NALCN (259232)', 'NALF1 (728215)', 'NALF2 (27112)', 'NAMPT (10135)', 'NANOG (79923)', 'NANOGNB (360030)', 'NANOS1 (340719)', 'NANOS2 (339345)', 'NANOS3 (342977)', 'NANP (140838)', 'NANS (54187)', 'NAP1L1 (4673)', 'NAP1L2 (4674)', 'NAP1L3 (4675)', 'NAP1L4 (4676)', 'NAP1L5 (266812)', 'NAPA (8775)', 'NAPB (63908)', 'NAPEPLD (222236)', 'NAPG (8774)', 'NAPRT (93100)', 'NAPSA (9476)', 'NARF (26502)', 'NARS1 (4677)', 'NARS2 (79731)', 'NASP (4678)', 'NAT1 (9)', 'NAT10 (55226)', 'NAT14 (57106)', 'NAT16 (375607)', 'NAT2 (10)', 'NAT8 (9027)', 'NAT8L (339983)', 'NAT9 (26151)', 'NAV1 (89796)', 'NAV2 (89797)', 'NAV3 (89795)', 'NAXD (55739)', 'NAXE (128240)', 'NBAS (51594)', 'NBEA (26960)', 'NBEAL1 (65065)', 'NBEAL2 (23218)', 'NBL1 (4681)', 'NBN (4683)', 'NBPF3 (84224)', 'NBR1 (4077)', 'NCALD (83988)', 'NCAM2 (4685)', 'NCAN (1463)', 'NCAPD2 (9918)', 'NCAPD3 (23310)', 'NCAPG (64151)', 'NCAPG2 (54892)', 'NCAPH (23397)', 'NCAPH2 (29781)', 'NCBP1 (4686)', 'NCBP2 (22916)', 'NCBP3 (55421)', 'NCCRP1 (342897)', 'NCDN (23154)', 'NCEH1 (57552)', 'NCF1 (653361)', 'NCF2 (4688)', 'NCF4 (4689)', 'NCK1 (4690)', 'NCK2 (8440)', 'NCKAP1 (10787)', 'NCKAP1L (3071)', 'NCKAP5 (344148)', 'NCKAP5L (57701)', 'NCKIPSD (51517)', 'NCL (4691)', 'NCLN (56926)', 'NCMAP (400746)', 'NCOA1 (8648)', 'NCOA2 (10499)', 'NCOA3 (8202)', 'NCOA4 (8031)', 'NCOA5 (57727)', 'NCOA6 (23054)', 'NCOA7 (135112)', 'NCOR1 (9611)', 'NCOR2 (9612)', 'NCR1 (9437)', 'NCR2 (9436)', 'NCR3 (259197)', 'NCR3LG1 (374383)', 'NCS1 (23413)', 'NCSTN (23385)', 'NDC1 (55706)', 'NDC80 (10403)', 'NDE1 (54820)', 'NDEL1 (81565)', 'NDFIP1 (80762)', 'NDFIP2 (54602)', 'NDN (4692)', 'NDNF (79625)', 'NDOR1 (27158)', 'NDP (4693)', 'NDRG1 (10397)', 'NDRG2 (57447)', 'NDRG3 (57446)', 'NDRG4 (65009)', 'NDST1 (3340)', 'NDST2 (8509)', 'NDST3 (9348)', 'NDST4 (64579)', 'NDUFA1 (4694)', 'NDUFA10 (4705)', 'NDUFA11 (126328)', 'NDUFA12 (55967)', 'NDUFA13 (51079)', 'NDUFA2 (4695)', 'NDUFA3 (4696)', 'NDUFA4 (4697)', 'NDUFA4L2 (56901)', 'NDUFA5 (4698)', 'NDUFA6 (4700)', 'NDUFA7 (4701)', 'NDUFA8 (4702)', 'NDUFA9 (4704)', 'NDUFAB1 (4706)', 'NDUFAF1 (51103)', 'NDUFAF2 (91942)', 'NDUFAF3 (25915)', 'NDUFAF4 (29078)', 'NDUFAF5 (79133)', 'NDUFAF6 (137682)', 'NDUFAF7 (55471)', 'NDUFAF8 (284184)', 'NDUFB1 (4707)', 'NDUFB10 (4716)', 'NDUFB11 (54539)', 'NDUFB2 (4708)', 'NDUFB3 (4709)', 'NDUFB4 (4710)', 'NDUFB5 (4711)', 'NDUFB6 (4712)', 'NDUFB7 (4713)', 'NDUFB8 (4714)', 'NDUFB9 (4715)', 'NDUFC1 (4717)', 'NDUFC2 (4718)', 'NDUFS1 (4719)', 'NDUFS2 (4720)', 'NDUFS3 (4722)', 'NDUFS4 (4724)', 'NDUFS5 (4725)', 'NDUFS6 (4726)', 'NDUFS7 (374291)', 'NDUFS8 (4728)', 'NDUFV1 (4723)', 'NDUFV2 (4729)', 'NDUFV3 (4731)', 'NEB (4703)', 'NEBL (10529)', 'NECAB1 (64168)', 'NECAB2 (54550)', 'NECAB3 (63941)', 'NECAP1 (25977)', 'NECAP2 (55707)', 'NECTIN1 (5818)', 'NECTIN2 (5819)', 'NECTIN3 (25945)', 'NECTIN4 (81607)', 'NEDD1 (121441)', 'NEDD4 (4734)', 'NEDD4L (23327)', 'NEDD8 (4738)', 'NEDD9 (4739)', 'NEFH (4744)', 'NEFM (4741)', 'NEGR1 (257194)', 'NEIL1 (79661)', 'NEIL2 (252969)', 'NEIL3 (55247)', 'NEK1 (4750)', 'NEK10 (152110)', 'NEK11 (79858)', 'NEK2 (4751)', 'NEK3 (4752)', 'NEK4 (6787)', 'NEK5 (341676)', 'NEK6 (10783)', 'NEK7 (140609)', 'NEK8 (284086)', 'NEK9 (91754)', 'NELFA (7469)', 'NELFB (25920)', 'NELFCD (51497)', 'NELFE (7936)', 'NELL1 (4745)', 'NELL2 (4753)', 'NEMF (9147)', 'NEMP1 (23306)', 'NEMP2 (100131211)', 'NENF (29937)', 'NEO1 (4756)', 'NEPRO (25871)', 'NES (10763)', 'NET1 (10276)', 'NETO1 (81832)', 'NETO2 (81831)', 'NEU1 (4758)', 'NEU2 (4759)', 'NEU3 (10825)', 'NEU4 (129807)', 'NEURL1 (9148)', 'NEURL1B (54492)', 'NEURL2 (140825)', 'NEURL4 (84461)', 'NEUROD1 (4760)', 'NEUROD2 (4761)', 'NEUROD4 (58158)', 'NEUROD6 (63974)', 'NEUROG1 (4762)', 'NEUROG2 (63973)', 'NEUROG3 (50674)', 'NEXMIF (340533)', 'NEXN (91624)', 'NF1 (4763)', 'NF2 (4771)', 'NFAM1 (150372)', 'NFASC (23114)', 'NFAT5 (10725)', 'NFATC1 (4772)', 'NFATC2 (4773)', 'NFATC2IP (84901)', 'NFATC3 (4775)', 'NFATC4 (4776)', 'NFE2 (4778)', 'NFE2L1 (4779)', 'NFE2L2 (4780)', 'NFE2L3 (9603)', 'NFIA (4774)', 'NFIB (4781)', 'NFIC (4782)', 'NFIL3 (4783)', 'NFIX (4784)', 'NFKB1 (4790)', 'NFKB2 (4791)', 'NFKBIA (4792)', 'NFKBIB (4793)', 'NFKBID (84807)', 'NFKBIE (4794)', 'NFKBIL1 (4795)', 'NFKBIZ (64332)', 'NFRKB (4798)', 'NFS1 (9054)', 'NFU1 (27247)', 'NFX1 (4799)', 'NFXL1 (152518)', 'NFYA (4800)', 'NFYB (4801)', 'NFYC (4802)', 'NGB (58157)', 'NGDN (25983)', 'NGEF (25791)', 'NGF (4803)', 'NGFR (4804)', 'NGLY1 (55768)', 'NGRN (51335)', 'NHEJ1 (79840)', 'NHERF1 (9368)', 'NHERF2 (9351)', 'NHERF4 (79849)', 'NHLH1 (4807)', 'NHLH2 (4808)', 'NHLRC1 (378884)', 'NHLRC2 (374354)', 'NHLRC3 (387921)', 'NHLRC4 (283948)', 'NHP2 (55651)', 'NHS (4810)', 'NHSL1 (57224)', 'NHSL3 (57648)', 'NIBAN1 (116496)', 'NIBAN2 (64855)', 'NIBAN3 (199786)', 'NICN1 (84276)', 'NICOL1 (401115)', 'NID1 (4811)', 'NID2 (22795)', 'NIF3L1 (60491)', 'NIFK (84365)', 'NIM1K (167359)', 'NIN (51199)', 'NINJ1 (4814)', 'NINJ2 (4815)', 'NINL (22981)', 'NIP7 (51388)', 'NIPA1 (123606)', 'NIPA2 (81614)', 'NIPAL1 (152519)', 'NIPAL2 (79815)', 'NIPAL3 (57185)', 'NIPAL4 (348938)', 'NIPBL (25836)', 'NIPSNAP1 (8508)', 'NIPSNAP2 (2631)', 'NIPSNAP3A (25934)', 'NIPSNAP3B (55335)', 'NISCH (11188)', 'NIT1 (4817)', 'NIT2 (56954)', 'NKAIN1 (79570)', 'NKAIN2 (154215)', 'NKAIN3 (286183)', 'NKAIN4 (128414)', 'NKAP (79576)', 'NKAPD1 (55216)', 'NKAPL (222698)', 'NKD1 (85407)', 'NKD2 (85409)', 'NKG7 (4818)', 'NKIRAS1 (28512)', 'NKIRAS2 (28511)', 'NKTR (4820)', 'NKX1-2 (390010)', 'NKX2-1 (7080)', 'NKX2-2 (4821)', 'NKX2-3 (159296)', 'NKX2-4 (644524)', 'NKX2-5 (1482)', 'NKX2-8 (26257)', 'NKX3-1 (4824)', 'NKX3-2 (579)', 'NKX6-1 (4825)', 'NKX6-2 (84504)', 'NKX6-3 (157848)', 'NLE1 (54475)', 'NLGN1 (22871)', 'NLGN2 (57555)', 'NLGN3 (54413)', 'NLGN4X (57502)', 'NLGN4Y (22829)', 'NLK (51701)', 'NLN (57486)', 'NLRC4 (58484)', 'NLRC5 (84166)', 'NLRP1 (22861)', 'NLRP10 (338322)', 'NLRP11 (204801)', 'NLRP12 (91662)', 'NLRP13 (126204)', 'NLRP14 (338323)', 'NLRP2 (55655)', 'NLRP3 (114548)', 'NLRP4 (147945)', 'NLRP5 (126206)', 'NLRP6 (171389)', 'NLRP7 (199713)', 'NLRP8 (126205)', 'NLRP9 (338321)', 'NLRX1 (79671)', 'NMB (4828)', 'NMBR (4829)', 'NMD3 (51068)', 'NME1 (4830)', 'NME2 (4831)', 'NME3 (4832)', 'NME4 (4833)', 'NME5 (8382)', 'NME6 (10201)', 'NME7 (29922)', 'NME8 (51314)', 'NME9 (347736)', 'NMI (9111)', 'NMNAT1 (64802)', 'NMNAT2 (23057)', 'NMNAT3 (349565)', 'NMRAL1 (57407)', 'NMRK1 (54981)', 'NMRK2 (27231)', 'NMS (129521)', 'NMT1 (4836)', 'NMT2 (9397)', 'NMU (10874)', 'NMUR1 (10316)', 'NMUR2 (56923)', 'NNAT (4826)', 'NNMT (4837)', 'NNT (23530)', 'NOA1 (84273)', 'NOB1 (28987)', 'NOC2L (26155)', 'NOC3L (64318)', 'NOC4L (79050)', 'NOCT (25819)', 'NOD1 (10392)', 'NOD2 (64127)', 'NODAL (4838)', 'NOG (9241)', 'NOL10 (79954)', 'NOL11 (25926)', 'NOL12 (79159)', 'NOL3 (8996)', 'NOL4 (8715)', 'NOL4L (140688)', 'NOL6 (65083)', 'NOL7 (51406)', 'NOL8 (55035)', 'NOL9 (79707)', 'NOLC1 (9221)', 'NOM1 (64434)', 'NOMO1 (23420)', 'NOMO2 (283820)', 'NONO (4841)', 'NOP10 (55505)', 'NOP14 (8602)', 'NOP16 (51491)', 'NOP2 (4839)', 'NOP53 (29997)', 'NOP56 (10528)', 'NOP58 (51602)', 'NOP9 (161424)', 'NOPCHAP1 (121053)', 'NOS1 (4842)', 'NOS1AP (9722)', 'NOS2 (4843)', 'NOS3 (4846)', 'NOSIP (51070)', 'NOSTRIN (115677)', 'NOTCH1 (4851)', 'NOTCH2 (4853)', 'NOTCH3 (4854)', 'NOTCH4 (4855)', 'NOTO (344022)', 'NOTUM (147111)', 'NOVA1 (4857)', 'NOVA2 (4858)', 'NOX1 (27035)', 'NOX3 (50508)', 'NOX4 (50507)', 'NOX5 (79400)', 'NOXA1 (10811)', 'NOXO1 (124056)', 'NOXRED1 (122945)', 'NPAP1 (23742)', 'NPAS1 (4861)', 'NPAS2 (4862)', 'NPAS3 (64067)', 'NPAS4 (266743)', 'NPAT (4863)', 'NPB (256933)', 'NPBWR1 (2831)', 'NPBWR2 (2832)', 'NPC1 (4864)', 'NPC1L1 (29881)', 'NPC2 (10577)', 'NPDC1 (56654)', 'NPEPL1 (79716)', 'NPEPPS (9520)', 'NPFF (8620)', 'NPFFR1 (64106)', 'NPFFR2 (10886)', 'NPHP1 (4867)', 'NPHP3 (27031)', 'NPHP4 (261734)', 'NPHS1 (4868)', 'NPHS2 (7827)', 'NPL (80896)', 'NPLOC4 (55666)', 'NPM1 (4869)', 'NPM2 (10361)', 'NPM3 (10360)', 'NPNT (255743)', 'NPPA (4878)', 'NPPB (4879)', 'NPPC (4880)', 'NPR1 (4881)', 'NPR2 (4882)', 'NPR3 (4883)', 'NPRL2 (10641)', 'NPS (594857)', 'NPSR1 (387129)', 'NPTN (27020)', 'NPTX1 (4884)', 'NPTX2 (4885)', 'NPTXR (23467)', 'NPVF (64111)', 'NPW (283869)', 'NPY (4852)', 'NPY1R (4886)', 'NPY2R (4887)', 'NPY5R (4889)', 'NQO1 (1728)', 'NQO2 (4835)', 'NR0B1 (190)', 'NR0B2 (8431)', 'NR1D1 (9572)', 'NR1D2 (9975)', 'NR1H2 (7376)', 'NR1H3 (10062)', 'NR1H4 (9971)', 'NR1I2 (8856)', 'NR1I3 (9970)', 'NR2C1 (7181)', 'NR2C2 (7182)', 'NR2C2AP (126382)', 'NR2E1 (7101)', 'NR2F1 (7025)', 'NR2F2 (7026)', 'NR2F6 (2063)', 'NR3C1 (2908)', 'NR3C2 (4306)', 'NR4A1 (3164)', 'NR4A2 (4929)', 'NR4A3 (8013)', 'NR5A1 (2516)', 'NR5A2 (2494)', 'NR6A1 (2649)', 'NRAP (4892)', 'NRARP (441478)', 'NRAS (4893)', 'NRBF2 (29982)', 'NRBP1 (29959)', 'NRBP2 (340371)', 'NRCAM (4897)', 'NRDC (4898)', 'NRDE2 (55051)', 'NREP (9315)', 'NRF1 (4899)', 'NRG1 (3084)', 'NRG2 (9542)', 'NRG3 (10718)', 'NRG4 (145957)', 'NRGN (4900)', 'NRIP1 (8204)', 'NRIP2 (83714)', 'NRIP3 (56675)', 'NRK (203447)', 'NRL (4901)', 'NRM (11270)', 'NRN1 (51299)', 'NRN1L (123904)', 'NRP1 (8829)', 'NRP2 (8828)', 'NRROS (375387)', 'NRSN1 (140767)', 'NRSN2 (80023)', 'NRTN (4902)', 'NRXN1 (9378)', 'NRXN2 (9379)', 'NRXN3 (9369)', 'NSA2 (10412)', 'NSD1 (64324)', 'NSD2 (7468)', 'NSD3 (54904)', 'NSDHL (50814)', 'NSF (4905)', 'NSFL1C (55968)', 'NSG1 (27065)', 'NSG2 (51617)', 'NSL1 (25936)', 'NSMAF (8439)', 'NSMCE1 (197370)', 'NSMCE2 (286053)', 'NSMCE3 (56160)', 'NSMCE4A (54780)', 'NSMF (26012)', 'NSRP1 (84081)', 'NSUN2 (54888)', 'NSUN3 (63899)', 'NSUN4 (387338)', 'NSUN5 (55695)', 'NSUN6 (221078)', 'NSUN7 (79730)', 'NT5C (30833)', 'NT5C1A (84618)', 'NT5C1B (93034)', 'NT5C2 (22978)', 'NT5C3A (51251)', 'NT5C3B (115024)', 'NT5DC1 (221294)', 'NT5DC2 (64943)', 'NT5DC3 (51559)', 'NT5E (4907)', 'NT5M (56953)', 'NTAN1 (123803)', 'NTAQ1 (55093)', 'NTF3 (4908)', 'NTF4 (4909)', 'NTHL1 (4913)', 'NTM (50863)', 'NTMT1 (28989)', 'NTMT2 (149281)', 'NTN1 (9423)', 'NTN3 (4917)', 'NTN4 (59277)', 'NTN5 (126147)', 'NTNG1 (22854)', 'NTNG2 (84628)', 'NTPCR (84284)', 'NTRK1 (4914)', 'NTRK2 (4915)', 'NTRK3 (4916)', 'NTS (4922)', 'NTSR1 (4923)', 'NTSR2 (23620)', 'NUAK1 (9891)', 'NUAK2 (81788)', 'NUB1 (51667)', 'NUBP1 (4682)', 'NUBP2 (10101)', 'NUBPL (80224)', 'NUCB1 (4924)', 'NUCB2 (4925)', 'NUCKS1 (64710)', 'NUDC (10726)', 'NUDCD1 (84955)', 'NUDCD2 (134492)', 'NUDCD3 (23386)', 'NUDT1 (4521)', 'NUDT10 (170685)', 'NUDT11 (55190)', 'NUDT12 (83594)', 'NUDT13 (25961)', 'NUDT14 (256281)', 'NUDT15 (55270)', 'NUDT16 (131870)', 'NUDT16L1 (84309)', 'NUDT17 (200035)', 'NUDT19 (390916)', 'NUDT2 (318)', 'NUDT21 (11051)', 'NUDT22 (84304)', 'NUDT3 (11165)', 'NUDT5 (11164)', 'NUDT6 (11162)', 'NUDT7 (283927)', 'NUDT8 (254552)', 'NUDT9 (53343)', 'NUF2 (83540)', 'NUFIP1 (26747)', 'NUFIP2 (57532)', 'NUGGC (389643)', 'NUMA1 (4926)', 'NUMB (8650)', 'NUMBL (9253)', 'NUP107 (57122)', 'NUP133 (55746)', 'NUP153 (9972)', 'NUP155 (9631)', 'NUP160 (23279)', 'NUP188 (23511)', 'NUP205 (23165)', 'NUP210 (23225)', 'NUP210L (91181)', 'NUP214 (8021)', 'NUP35 (129401)', 'NUP37 (79023)', 'NUP42 (11097)', 'NUP43 (348995)', 'NUP50 (10762)', 'NUP54 (53371)', 'NUP58 (9818)', 'NUP62 (23636)', 'NUP62CL (54830)', 'NUP85 (79902)', 'NUP88 (4927)', 'NUP93 (9688)', 'NUP98 (4928)', 'NUPR1 (26471)', 'NUPR2 (389493)', 'NUS1 (116150)', 'NUSAP1 (51203)', 'NUTF2 (10204)', 'NUTM1 (256646)', 'NUTM2A (728118)', 'NUTM2F (54754)', 'NUTM2G (441457)', 'NVL (4931)', 'NWD1 (284434)', 'NWD2 (57495)', 'NXF1 (10482)', 'NXF3 (56000)', 'NXN (64359)', 'NXNL1 (115861)', 'NXNL2 (158046)', 'NXPE1 (120400)', 'NXPE2 (120406)', 'NXPE3 (91775)', 'NXPE4 (54827)', 'NXPH1 (30010)', 'NXPH2 (11249)', 'NXPH3 (11248)', 'NXPH4 (11247)', 'NXT1 (29107)', 'NXT2 (55916)', 'NYAP1 (222950)', 'NYAP2 (57624)', 'NYNRIN (57523)', 'NYX (60506)', 'OAF (220323)', 'OARD1 (221443)', 'OAS1 (4938)', 'OAS2 (4939)', 'OAS3 (4940)', 'OASL (8638)', 'OAT (4942)', 'OAZ1 (4946)', 'OAZ2 (4947)', 'OAZ3 (51686)', 'OBI1 (79596)', 'OBP2A (29991)', 'OBSCN (84033)', 'OBSL1 (23363)', 'OC90 (729330)', 'OCA2 (4948)', 'OCEL1 (79629)', 'OCIAD1 (54940)', 'OCIAD2 (132299)', 'OCLN (100506658)', 'OCM (654231)', 'OCM2 (4951)', 'OCRL (4952)', 'OCSTAMP (128506)', 'ODAD1 (93233)', 'ODAD2 (55130)', 'ODAD3 (115948)', 'ODAM (54959)', 'ODAPH (152816)', 'ODC1 (4953)', 'ODF1 (4956)', 'ODF2 (4957)', 'ODF2L (57489)', 'ODF4 (146852)', 'ODR4 (54953)', 'OFD1 (8481)', 'OGA (10724)', 'OGDH (4967)', 'OGDHL (55753)', 'OGFOD1 (55239)', 'OGFOD2 (79676)', 'OGFOD3 (79701)', 'OGFR (11054)', 'OGFRL1 (79627)', 'OGG1 (4968)', 'OGN (4969)', 'OGT (8473)', 'OIP5 (11339)', 'OIT3 (170392)', 'OLA1 (29789)', 'OLAH (55301)', 'OLFM1 (10439)', 'OLFM2 (93145)', 'OLFM3 (118427)', 'OLFM4 (10562)', 'OLFML1 (283298)', 'OLFML2A (169611)', 'OLFML2B (25903)', 'OLFML3 (56944)', 'OLIG1 (116448)', 'OLIG2 (10215)', 'OLIG3 (167826)', 'OLR1 (4973)', 'OMA1 (115209)', 'OMD (4958)', 'OMG (4974)', 'OMP (4975)', 'ONECUT1 (3175)', 'ONECUT2 (9480)', 'ONECUT3 (390874)', 'OOEP (441161)', 'OOSP2 (219990)', 'OPA1 (4976)', 'OPA3 (80207)', 'OPALIN (93377)', 'OPCML (4978)', 'OPHN1 (4983)', 'OPN1LW (5956)', 'OPN1SW (611)', 'OPN3 (23596)', 'OPN4 (94233)', 'OPN5 (221391)', 'OPRD1 (4985)', 'OPRK1 (4986)', 'OPRL1 (4987)', 'OPRM1 (4988)', 'OPRPN (58503)', 'OPTC (26254)', 'OPTN (10133)', 'OR10A2 (341276)', 'OR10A3 (26496)', 'OR10A4 (283297)', 'OR10A5 (144124)', 'OR10A7 (121364)', 'OR10AD1 (121275)', 'OR10AG1 (282770)', 'OR10C1 (442194)', 'OR10G2 (26534)', 'OR10G3 (26533)', 'OR10G4 (390264)', 'OR10G7 (390265)', 'OR10G8 (219869)', 'OR10G9 (219870)', 'OR10H1 (26539)', 'OR10H2 (26538)', 'OR10H3 (26532)', 'OR10H4 (126541)', 'OR10H5 (284433)', 'OR10J1 (26476)', 'OR10J5 (127385)', 'OR10K1 (391109)', 'OR10K2 (391107)', 'OR10P1 (121130)', 'OR10Q1 (219960)', 'OR10R2 (343406)', 'OR10T2 (128360)', 'OR10V1 (390201)', 'OR10W1 (81341)', 'OR10Z1 (128368)', 'OR11A1 (26531)', 'OR11H4 (390442)', 'OR11H6 (122748)', 'OR11L1 (391189)', 'OR12D3 (81797)', 'OR13A1 (79290)', 'OR13C2 (392376)', 'OR13C3 (138803)', 'OR13C4 (138804)', 'OR13C5 (138799)', 'OR13C8 (138802)', 'OR13C9 (286362)', 'OR13D1 (286365)', 'OR13F1 (138805)', 'OR13G1 (441933)', 'OR13H1 (347468)', 'OR13J1 (392309)', 'OR14A16 (284532)', 'OR14C36 (127066)', 'OR14I1 (401994)', 'OR14J1 (442191)', 'OR1A1 (8383)', 'OR1A2 (26189)', 'OR1C1 (26188)', 'OR1D2 (4991)', 'OR1D5 (8386)', 'OR1E1 (8387)', 'OR1E2 (8388)', 'OR1F1 (4992)', 'OR1G1 (8390)', 'OR1I1 (126370)', 'OR1J1 (347168)', 'OR1J2 (26740)', 'OR1J4 (26219)', 'OR1K1 (392392)', 'OR1L1 (26737)', 'OR1L3 (26735)', 'OR1L4 (254973)', 'OR1L6 (392390)', 'OR1L8 (138881)', 'OR1M1 (125963)', 'OR1N1 (138883)', 'OR1N2 (138882)', 'OR1Q1 (158131)', 'OR2A12 (346525)', 'OR2A14 (135941)', 'OR2A2 (442361)', 'OR2A25 (392138)', 'OR2A5 (393046)', 'OR2AE1 (81392)', 'OR2AG1 (144125)', 'OR2AG2 (338755)', 'OR2AP1 (121129)', 'OR2AT4 (341152)', 'OR2B11 (127623)', 'OR2B2 (81697)', 'OR2B3 (442184)', 'OR2B6 (26212)', 'OR2C1 (4993)', 'OR2C3 (81472)', 'OR2D2 (120776)', 'OR2D3 (120775)', 'OR2F2 (135948)', 'OR2G2 (81470)', 'OR2G3 (81469)', 'OR2G6 (391211)', 'OR2H1 (26716)', 'OR2H2 (7932)', 'OR2J2 (26707)', 'OR2J3 (442186)', 'OR2K2 (26248)', 'OR2L13 (284521)', 'OR2L2 (26246)', 'OR2L3 (391192)', 'OR2L5 (81466)', 'OR2M2 (391194)', 'OR2M3 (127062)', 'OR2M4 (26245)', 'OR2M5 (127059)', 'OR2M7 (391196)', 'OR2S2 (56656)', 'OR2T10 (127069)', 'OR2T12 (127064)', 'OR2T2 (401992)', 'OR2T27 (403239)', 'OR2T29 (343563)', 'OR2T35 (403244)', 'OR2T4 (127074)', 'OR2T5 (401993)', 'OR2T6 (254879)', 'OR2T8 (343172)', 'OR2V1 (26693)', 'OR2V2 (285659)', 'OR2W1 (26692)', 'OR2W3 (343171)', 'OR2Y1 (134083)', 'OR2Z1 (284383)', 'OR3A1 (4994)', 'OR4A16 (81327)', 'OR4A47 (403253)', 'OR4A5 (81318)', 'OR4B1 (119765)', 'OR4C11 (219429)', 'OR4C12 (283093)', 'OR4C13 (283092)', 'OR4C15 (81309)', 'OR4C3 (256144)', 'OR4C46 (119749)', 'OR4C6 (219432)', 'OR4D1 (26689)', 'OR4D10 (390197)', 'OR4D11 (219986)', 'OR4D2 (124538)', 'OR4D5 (219875)', 'OR4D6 (219983)', 'OR4D9 (390199)', 'OR4E2 (26686)', 'OR4F15 (390649)', 'OR4F6 (390648)', 'OR4K1 (79544)', 'OR4K13 (390433)', 'OR4K14 (122740)', 'OR4K15 (81127)', 'OR4K17 (390436)', 'OR4K2 (390431)', 'OR4K5 (79317)', 'OR4L1 (122742)', 'OR4M1 (441670)', 'OR4N2 (390429)', 'OR4N5 (390437)', 'OR4P4 (81300)', 'OR4S1 (256148)', 'OR4S2 (219431)', 'OR4X1 (390113)', 'OR51A2 (401667)', 'OR51A4 (401666)', 'OR51A7 (119687)', 'OR51B4 (79339)', 'OR51B5 (282763)', 'OR51B6 (390058)', 'OR51D1 (390038)', 'OR51E1 (143503)', 'OR51E2 (81285)', 'OR51F2 (119694)', 'OR51G2 (81282)', 'OR51I1 (390063)', 'OR51I2 (390064)', 'OR51L1 (119682)', 'OR51M1 (390059)', 'OR51Q1 (390061)', 'OR51S1 (119692)', 'OR51T1 (401665)', 'OR51V1 (283111)', 'OR52A1 (23538)', 'OR52A5 (390054)', 'OR52B2 (255725)', 'OR52B6 (340980)', 'OR52D1 (390066)', 'OR52E2 (119678)', 'OR52E4 (390081)', 'OR52E6 (390078)', 'OR52E8 (390079)', 'OR52H1 (390067)', 'OR52I1 (390037)', 'OR52J3 (119679)', 'OR52K1 (390036)', 'OR52K2 (119774)', 'OR52L1 (338751)', 'OR52M1 (119772)', 'OR52N1 (79473)', 'OR52N2 (390077)', 'OR52N4 (390072)', 'OR52N5 (390075)', 'OR52W1 (120787)', 'OR56A1 (120796)', 'OR56A3 (390083)', 'OR56B1 (387748)', 'OR56B4 (196335)', 'OR5A1 (219982)', 'OR5A2 (219981)', 'OR5AC2 (81050)', 'OR5AK2 (390181)', 'OR5AN1 (390195)', 'OR5AP2 (338675)', 'OR5AS1 (219447)', 'OR5AU1 (390445)', 'OR5B12 (390191)', 'OR5B17 (219965)', 'OR5B2 (390190)', 'OR5B21 (219968)', 'OR5B3 (441608)', 'OR5C1 (392391)', 'OR5D14 (219436)', 'OR5D16 (390144)', 'OR5D18 (219438)', 'OR5F1 (338674)', 'OR5H14 (403273)', 'OR5H15 (403274)', 'OR5H2 (79310)', 'OR5H6 (79295)', 'OR5I1 (10798)', 'OR5J2 (282775)', 'OR5K1 (26339)', 'OR5K3 (403277)', 'OR5K4 (403278)', 'OR5L2 (26338)', 'OR5M1 (390168)', 'OR5M10 (390167)', 'OR5M11 (219487)', 'OR5M3 (219482)', 'OR5M8 (219484)', 'OR5M9 (390162)', 'OR5P2 (120065)', 'OR5P3 (120066)', 'OR5T1 (390155)', 'OR5T2 (219464)', 'OR5T3 (390154)', 'OR5V1 (81696)', 'OR5W2 (390148)', 'OR6A2 (8590)', 'OR6B1 (135946)', 'OR6B2 (389090)', 'OR6B3 (150681)', 'OR6C1 (390321)', 'OR6C2 (341416)', 'OR6C3 (254786)', 'OR6C4 (341418)', 'OR6C6 (283365)', 'OR6C65 (403282)', 'OR6C68 (403284)', 'OR6C70 (390327)', 'OR6C74 (254783)', 'OR6C75 (390323)', 'OR6C76 (390326)', 'OR6F1 (343169)', 'OR6K2 (81448)', 'OR6K3 (391114)', 'OR6M1 (390261)', 'OR6N1 (128372)', 'OR6N2 (81442)', 'OR6P1 (128366)', 'OR6Q1 (219952)', 'OR6S1 (341799)', 'OR6T1 (219874)', 'OR6V1 (346517)', 'OR6X1 (390260)', 'OR6Y1 (391112)', 'OR7A10 (390892)', 'OR7A17 (26333)', 'OR7A5 (26659)', 'OR7C1 (26664)', 'OR7C2 (26658)', 'OR7D2 (162998)', 'OR7D4 (125958)', 'OR7E24 (26648)', 'OR7G1 (125962)', 'OR7G2 (390882)', 'OR7G3 (390883)', 'OR8A1 (390275)', 'OR8B12 (219858)', 'OR8B2 (26595)', 'OR8B3 (390271)', 'OR8B4 (283162)', 'OR8B8 (26493)', 'OR8D1 (283159)', 'OR8D4 (338662)', 'OR8G5 (219865)', 'OR8H1 (219469)', 'OR8H2 (390151)', 'OR8H3 (390152)', 'OR8I2 (120586)', 'OR8J1 (219477)', 'OR8J3 (81168)', 'OR8K1 (390157)', 'OR8K5 (219453)', 'OR8S1 (341568)', 'OR8U1 (219417)', 'OR9A2 (135924)', 'OR9A4 (130075)', 'OR9G1 (390174)', 'OR9G4 (283189)', 'OR9I1 (219954)', 'OR9K2 (441639)', 'OR9Q1 (219956)', 'OR9Q2 (219957)', 'ORAI2 (80228)', 'ORAI3 (93129)', 'ORC1 (4998)', 'ORC2 (4999)', 'ORC3 (23595)', 'ORC4 (5000)', 'ORC5 (5001)', 'ORC6 (23594)', 'ORM1 (5004)', 'ORM2 (5005)', 'ORMDL1 (94101)', 'ORMDL2 (29095)', 'ORMDL3 (94103)', 'OS9 (10956)', 'OSBP (5007)', 'OSBP2 (23762)', 'OSBPL10 (114884)', 'OSBPL11 (114885)', 'OSBPL1A (114876)', 'OSBPL2 (9885)', 'OSBPL3 (26031)', 'OSBPL5 (114879)', 'OSBPL6 (114880)', 'OSBPL7 (114881)', 'OSBPL8 (114882)', 'OSBPL9 (114883)', 'OSCAR (126014)', 'OSCP1 (127700)', 'OSER1 (51526)', 'OSGEP (55644)', 'OSGEPL1 (64172)', 'OSGIN1 (29948)', 'OSGIN2 (734)', 'OSM (5008)', 'OSMR (9180)', 'OSR1 (130497)', 'OSR2 (116039)', 'OST4 (100128731)', 'OSTC (58505)', 'OSTF1 (26578)', 'OSTM1 (28962)', 'OSTN (344901)', 'OTC (5009)', 'OTOA (146183)', 'OTOF (9381)', 'OTOG (340990)', 'OTOL1 (131149)', 'OTOP1 (133060)', 'OTOP2 (92736)', 'OTOP3 (347741)', 'OTOR (56914)', 'OTOS (150677)', 'OTP (23440)', 'OTUB1 (55611)', 'OTUB2 (78990)', 'OTUD1 (220213)', 'OTUD3 (23252)', 'OTUD4 (54726)', 'OTUD5 (55593)', 'OTUD6A (139562)', 'OTUD6B (51633)', 'OTUD7A (161725)', 'OTUD7B (56957)', 'OTULIN (90268)', 'OTULINL (54491)', 'OTX1 (5013)', 'OTX2 (5015)', 'OVCA2 (124641)', 'OVGP1 (5016)', 'OVOL1 (5017)', 'OVOL2 (58495)', 'OXA1L (5018)', 'OXCT1 (5019)', 'OXCT2 (64064)', 'OXER1 (165140)', 'OXGR1 (27199)', 'OXLD1 (339229)', 'OXNAD1 (92106)', 'OXR1 (55074)', 'OXSM (54995)', 'OXSR1 (9943)', 'OXT (5020)', 'OXTR (5021)', 'P2RX1 (5023)', 'P2RX2 (22953)', 'P2RX3 (5024)', 'P2RX4 (5025)', 'P2RX5 (5026)', 'P2RX6 (9127)', 'P2RX7 (5027)', 'P2RY1 (5028)', 'P2RY10 (27334)', 'P2RY11 (5032)', 'P2RY12 (64805)', 'P2RY13 (53829)', 'P2RY14 (9934)', 'P2RY2 (5029)', 'P2RY4 (5030)', 'P2RY6 (5031)', 'P3H1 (64175)', 'P3H2 (55214)', 'P3H3 (10536)', 'P3H4 (10609)', 'P4HA1 (5033)', 'P4HA2 (8974)', 'P4HA3 (283208)', 'P4HB (5034)', 'P4HTM (54681)', 'PA2G4 (5036)', 'PAAF1 (80227)', 'PABIR1 (116224)', 'PABIR2 (159090)', 'PABIR3 (159091)', 'PABPC1 (26986)', 'PABPC1L (80336)', 'PABPC3 (5042)', 'PABPC4 (8761)', 'PABPC5 (140886)', 'PABPN1 (8106)', 'PABPN1L (390748)', 'PACC1 (55248)', 'PACRG (135138)', 'PACRGL (133015)', 'PACS1 (55690)', 'PACS2 (23241)', 'PACSIN1 (29993)', 'PACSIN2 (11252)', 'PACSIN3 (29763)', 'PADI1 (29943)', 'PADI2 (11240)', 'PADI3 (51702)', 'PADI4 (23569)', 'PAEP (5047)', 'PAF1 (54623)', 'PAFAH1B1 (5048)', 'PAFAH1B2 (5049)', 'PAFAH1B3 (5050)', 'PAFAH2 (5051)', 'PAG1 (55824)', 'PAGE1 (8712)', 'PAGE2 (203569)', 'PAGE4 (9506)', 'PAGE5 (90737)', 'PAGR1 (79447)', 'PAH (5053)', 'PAICS (10606)', 'PAIP1 (10605)', 'PAIP2 (51247)', 'PAIP2B (400961)', 'PAK1 (5058)', 'PAK1IP1 (55003)', 'PAK2 (5062)', 'PAK3 (5063)', 'PAK4 (10298)', 'PAK5 (57144)', 'PAK6 (56924)', 'PALB2 (79728)', 'PALD1 (27143)', 'PALLD (23022)', 'PALM (5064)', 'PALM2AKAP2 (445815)', 'PALM3 (342979)', 'PALMD (54873)', 'PALS1 (64398)', 'PALS2 (51678)', 'PAM (5066)', 'PAM16 (51025)', 'PAMR1 (25891)', 'PAN2 (9924)', 'PAN3 (255967)', 'PANK1 (53354)', 'PANK2 (80025)', 'PANK3 (79646)', 'PANK4 (55229)', 'PANX1 (24145)', 'PANX2 (56666)', 'PANX3 (116337)', 'PAOX (196743)', 'PAPLN (89932)', 'PAPOLA (10914)', 'PAPOLG (64895)', 'PAPPA (5069)', 'PAPPA2 (60676)', 'PAPSS1 (9061)', 'PAPSS2 (9060)', 'PAQR3 (152559)', 'PAQR4 (124222)', 'PAQR5 (54852)', 'PAQR6 (79957)', 'PAQR7 (164091)', 'PAQR8 (85315)', 'PAQR9 (344838)', 'PARD3 (56288)', 'PARD3B (117583)', 'PARD6A (50855)', 'PARD6B (84612)', 'PARD6G (84552)', 'PARK7 (11315)', 'PARL (55486)', 'PARM1 (25849)', 'PARN (5073)', 'PARP1 (142)', 'PARP10 (84875)', 'PARP11 (57097)', 'PARP12 (64761)', 'PARP14 (54625)', 'PARP15 (165631)', 'PARP16 (54956)', 'PARP2 (10038)', 'PARP3 (10039)', 'PARP4 (143)', 'PARP6 (56965)', 'PARP8 (79668)', 'PARP9 (83666)', 'PARPBP (55010)', 'PARS2 (25973)', 'PARVA (55742)', 'PARVB (29780)', 'PARVG (64098)', 'PASD1 (139135)', 'PASK (23178)', 'PATE1 (160065)', 'PATE2 (399967)', 'PATE3 (100169851)', 'PATE4 (399968)', 'PATJ (10207)', 'PATL1 (219988)', 'PATL2 (197135)', 'PATZ1 (23598)', 'PAWR (5074)', 'PAX1 (5075)', 'PAX2 (5076)', 'PAX3 (5077)', 'PAX4 (5078)', 'PAX5 (5079)', 'PAX6 (5080)', 'PAX7 (5081)', 'PAX8 (7849)', 'PAX9 (5083)', 'PAXBP1 (94104)', 'PAXIP1 (22976)', 'PAXX (286257)', 'PBDC1 (51260)', 'PBK (55872)', 'PBLD (64081)', 'PBOV1 (59351)', 'PBRM1 (55193)', 'PBX1 (5087)', 'PBX2 (5089)', 'PBX3 (5090)', 'PBX4 (80714)', 'PBXIP1 (57326)', 'PC (5091)', 'PCARE (388939)', 'PCBD1 (5092)', 'PCBD2 (84105)', 'PCBP1 (5093)', 'PCBP2 (5094)', 'PCBP3 (54039)', 'PCBP4 (57060)', 'PCCA (5095)', 'PCCB (5096)', 'PCDH1 (5097)', 'PCDH10 (57575)', 'PCDH11X (27328)', 'PCDH12 (51294)', 'PCDH15 (65217)', 'PCDH17 (27253)', 'PCDH18 (54510)', 'PCDH19 (57526)', 'PCDH20 (64881)', 'PCDH7 (5099)', 'PCDH8 (5100)', 'PCDH9 (5101)', 'PCDHA1 (56147)', 'PCDHA10 (56139)', 'PCDHA11 (56138)', 'PCDHA12 (56137)', 'PCDHA13 (56136)', 'PCDHA2 (56146)', 'PCDHA3 (56145)', 'PCDHA4 (56144)', 'PCDHA5 (56143)', 'PCDHA6 (56142)', 'PCDHA7 (56141)', 'PCDHA8 (56140)', 'PCDHA9 (9752)', 'PCDHAC1 (56135)', 'PCDHAC2 (56134)', 'PCDHB1 (29930)', 'PCDHB10 (56126)', 'PCDHB11 (56125)', 'PCDHB12 (56124)', 'PCDHB13 (56123)', 'PCDHB14 (56122)', 'PCDHB15 (56121)', 'PCDHB16 (57717)', 'PCDHB2 (56133)', 'PCDHB3 (56132)', 'PCDHB4 (56131)', 'PCDHB5 (26167)', 'PCDHB6 (56130)', 'PCDHB7 (56129)', 'PCDHB8 (56128)', 'PCDHGA1 (56114)', 'PCDHGA10 (56106)', 'PCDHGA11 (56105)', 'PCDHGA12 (26025)', 'PCDHGA2 (56113)', 'PCDHGA3 (56112)', 'PCDHGA4 (56111)', 'PCDHGA5 (56110)', 'PCDHGA6 (56109)', 'PCDHGA7 (56108)', 'PCDHGA8 (9708)', 'PCDHGA9 (56107)', 'PCDHGB1 (56104)', 'PCDHGB2 (56103)', 'PCDHGB3 (56102)', 'PCDHGB4 (8641)', 'PCDHGB6 (56100)', 'PCDHGB7 (56099)', 'PCDHGC3 (5098)', 'PCDHGC4 (56098)', 'PCDHGC5 (56097)', 'PCED1A (64773)', 'PCED1B (91523)', 'PCF11 (51585)', 'PCGF1 (84759)', 'PCGF2 (7703)', 'PCGF3 (10336)', 'PCGF5 (84333)', 'PCGF6 (84108)', 'PCID2 (55795)', 'PCIF1 (63935)', 'PCK1 (5105)', 'PCK2 (5106)', 'PCLAF (9768)', 'PCLO (27445)', 'PCM1 (5108)', 'PCMT1 (5110)', 'PCMTD1 (115294)', 'PCMTD2 (55251)', 'PCNA (5111)', 'PCNP (57092)', 'PCNT (5116)', 'PCNX1 (22990)', 'PCNX2 (80003)', 'PCNX3 (399909)', 'PCOLCE (5118)', 'PCOLCE2 (26577)', 'PCP2 (126006)', 'PCP4 (5121)', 'PCP4L1 (654790)', 'PCSK1 (5122)', 'PCSK1N (27344)', 'PCSK2 (5126)', 'PCSK4 (54760)', 'PCSK5 (5125)', 'PCSK7 (9159)', 'PCSK9 (255738)', 'PCTP (58488)', 'PCYOX1 (51449)', 'PCYOX1L (78991)', 'PCYT1A (5130)', 'PCYT1B (9468)', 'PCYT2 (5833)', 'PDAP1 (11333)', 'PDC (5132)', 'PDCD1 (5133)', 'PDCD10 (11235)', 'PDCD11 (22984)', 'PDCD1LG2 (80380)', 'PDCD2 (5134)', 'PDCD2L (84306)', 'PDCD4 (27250)', 'PDCD5 (9141)', 'PDCD6 (10016)', 'PDCD6IP (10015)', 'PDCD7 (10081)', 'PDCL (5082)', 'PDCL2 (132954)', 'PDCL3 (79031)', 'PDE10A (10846)', 'PDE11A (50940)', 'PDE12 (201626)', 'PDE1A (5136)', 'PDE1B (5153)', 'PDE1C (5137)', 'PDE2A (5138)', 'PDE3A (5139)', 'PDE3B (5140)', 'PDE4A (5141)', 'PDE4B (5142)', 'PDE4C (5143)', 'PDE4D (5144)', 'PDE4DIP (9659)', 'PDE5A (8654)', 'PDE6A (5145)', 'PDE6B (5158)', 'PDE6C (5146)', 'PDE6D (5147)', 'PDE6G (5148)', 'PDE6H (5149)', 'PDE7A (5150)', 'PDE7B (27115)', 'PDE8A (5151)', 'PDE8B (8622)', 'PDE9A (5152)', 'PDF (64146)', 'PDGFA (5154)', 'PDGFB (5155)', 'PDGFC (56034)', 'PDGFD (80310)', 'PDGFRA (5156)', 'PDGFRB (5159)', 'PDGFRL (5157)', 'PDHA1 (5160)', 'PDHA2 (5161)', 'PDHB (5162)', 'PDHX (8050)', 'PDIA2 (64714)', 'PDIA3 (2923)', 'PDIA4 (9601)', 'PDIA5 (10954)', 'PDIA6 (10130)', 'PDIK1L (149420)', 'PDILT (204474)', 'PDK1 (5163)', 'PDK2 (5164)', 'PDK3 (5165)', 'PDK4 (5166)', 'PDLIM1 (9124)', 'PDLIM2 (64236)', 'PDLIM3 (27295)', 'PDLIM4 (8572)', 'PDLIM5 (10611)', 'PDLIM7 (9260)', 'PDP1 (54704)', 'PDP2 (57546)', 'PDPK1 (5170)', 'PDPN (10630)', 'PDPR (55066)', 'PDRG1 (81572)', 'PDS5A (23244)', 'PDS5B (23047)', 'PDSS1 (23590)', 'PDSS2 (57107)', 'PDX1 (3651)', 'PDXDC1 (23042)', 'PDXK (8566)', 'PDXP (57026)', 'PDYN (5173)', 'PDZD11 (51248)', 'PDZD2 (23037)', 'PDZD4 (57595)', 'PDZD7 (79955)', 'PDZD8 (118987)', 'PDZD9 (255762)', 'PDZK1 (5174)', 'PDZK1IP1 (10158)', 'PDZRN3 (23024)', 'PDZRN4 (29951)', 'PEA15 (8682)', 'PEAK1 (79834)', 'PEAK3 (374872)', 'PEAR1 (375033)', 'PEBP1 (5037)', 'PEBP4 (157310)', 'PECR (55825)', 'PEDS1 (387521)', 'PEF1 (553115)', 'PEG10 (23089)', 'PEG3 (5178)', 'PELI1 (57162)', 'PELI2 (57161)', 'PELI3 (246330)', 'PELO (53918)', 'PELP1 (27043)', 'PEMT (10400)', 'PENK (5179)', 'PEPD (5184)', 'PER1 (5187)', 'PER2 (8864)', 'PER3 (8863)', 'PERP (64065)', 'PES1 (23481)', 'PET100 (100131801)', 'PET117 (100303755)', 'PEX1 (5189)', 'PEX10 (5192)', 'PEX11A (8800)', 'PEX11B (8799)', 'PEX11G (92960)', 'PEX12 (5193)', 'PEX13 (5194)', 'PEX14 (5195)', 'PEX16 (9409)', 'PEX19 (5824)', 'PEX2 (5828)', 'PEX26 (55670)', 'PEX3 (8504)', 'PEX5 (5830)', 'PEX5L (51555)', 'PEX6 (5190)', 'PEX7 (5191)', 'PF4 (5196)', 'PF4V1 (5197)', 'PFAS (5198)', 'PFDN1 (5201)', 'PFDN2 (5202)', 'PFDN4 (5203)', 'PFDN5 (5204)', 'PFDN6 (10471)', 'PFKFB1 (5207)', 'PFKFB2 (5208)', 'PFKFB3 (5209)', 'PFKFB4 (5210)', 'PFKL (5211)', 'PFKM (5213)', 'PFKP (5214)', 'PFN1 (5216)', 'PFN2 (5217)', 'PFN3 (345456)', 'PFN4 (375189)', 'PGAM2 (5224)', 'PGAM4 (441531)', 'PGAM5 (192111)', 'PGAP1 (80055)', 'PGAP2 (27315)', 'PGAP3 (93210)', 'PGAP4 (84302)', 'PGAP6 (58986)', 'PGBD1 (84547)', 'PGBD2 (267002)', 'PGBD4 (161779)', 'PGC (5225)', 'PGD (5226)', 'PGF (5228)', 'PGGHG (80162)', 'PGGT1B (5229)', 'PGK1 (5230)', 'PGK2 (5232)', 'PGLS (25796)', 'PGLYRP1 (8993)', 'PGLYRP2 (114770)', 'PGLYRP3 (114771)', 'PGLYRP4 (57115)', 'PGM1 (5236)', 'PGM2 (55276)', 'PGM2L1 (283209)', 'PGM3 (5238)', 'PGM5 (5239)', 'PGP (283871)', 'PGPEP1 (54858)', 'PGPEP1L (145814)', 'PGR (5241)', 'PGRMC1 (10857)', 'PGRMC2 (10424)', 'PGS1 (9489)', 'PHACTR2 (9749)', 'PHACTR3 (116154)', 'PHACTR4 (65979)', 'PHAF1 (80262)', 'PHAX (51808)', 'PHB1 (5245)', 'PHB2 (11331)', 'PHC1 (1911)', 'PHC2 (1912)', 'PHC3 (80012)', 'PHETA1 (144717)', 'PHETA2 (150368)', 'PHEX (5251)', 'PHF1 (5252)', 'PHF10 (55274)', 'PHF11 (51131)', 'PHF12 (57649)', 'PHF13 (148479)', 'PHF14 (9678)', 'PHF19 (26147)', 'PHF2 (5253)', 'PHF20 (51230)', 'PHF20L1 (51105)', 'PHF21A (51317)', 'PHF21B (112885)', 'PHF23 (79142)', 'PHF24 (23349)', 'PHF3 (23469)', 'PHF5A (84844)', 'PHF6 (84295)', 'PHF7 (51533)', 'PHF8 (23133)', 'PHGDH (26227)', 'PHGR1 (644844)', 'PHIP (55023)', 'PHKA1 (5255)', 'PHKA2 (5256)', 'PHKB (5257)', 'PHKG1 (5260)', 'PHKG2 (5261)', 'PHLDA1 (22822)', 'PHLDA2 (7262)', 'PHLDA3 (23612)', 'PHLDB1 (23187)', 'PHLDB2 (90102)', 'PHLDB3 (653583)', 'PHLPP1 (23239)', 'PHLPP2 (23035)', 'PHOSPHO1 (162466)', 'PHOSPHO2 (493911)', 'PHOX2A (401)', 'PHOX2B (8929)', 'PHPT1 (29085)', 'PHRF1 (57661)', 'PHTF1 (10745)', 'PHTF2 (57157)', 'PHYH (5264)', 'PHYHD1 (254295)', 'PHYHIP (9796)', 'PHYHIPL (84457)', 'PHYKPL (85007)', 'PI15 (51050)', 'PI16 (221476)', 'PI3 (5266)', 'PI4K2A (55361)', 'PI4K2B (55300)', 'PI4KA (5297)', 'PI4KB (5298)', 'PIANP (196500)', 'PIAS1 (8554)', 'PIAS2 (9063)', 'PIAS3 (10401)', 'PIAS4 (51588)', 'PIBF1 (10464)', 'PICALM (8301)', 'PICK1 (9463)', 'PID1 (55022)', 'PIDD1 (55367)', 'PIERCE1 (138162)', 'PIERCE2 (145788)', 'PIEZO1 (9780)', 'PIF1 (80119)', 'PIGA (5277)', 'PIGB (9488)', 'PIGC (5279)', 'PIGF (5281)', 'PIGG (54872)', 'PIGH (5283)', 'PIGK (10026)', 'PIGL (9487)', 'PIGM (93183)', 'PIGN (23556)', 'PIGO (84720)', 'PIGP (51227)', 'PIGQ (9091)', 'PIGR (5284)', 'PIGS (94005)', 'PIGT (51604)', 'PIGU (128869)', 'PIGV (55650)', 'PIGW (284098)', 'PIGX (54965)', 'PIGY (84992)', 'PIGZ (80235)', 'PIH1D1 (55011)', 'PIH1D2 (120379)', 'PIK3AP1 (118788)', 'PIK3C2A (5286)', 'PIK3C2B (5287)', 'PIK3C2G (5288)', 'PIK3C3 (5289)', 'PIK3CA (5290)', 'PIK3CB (5291)', 'PIK3CD (5293)', 'PIK3CG (5294)', 'PIK3IP1 (113791)', 'PIK3R1 (5295)', 'PIK3R2 (5296)', 'PIK3R3 (8503)', 'PIK3R4 (30849)', 'PIK3R5 (23533)', 'PIKFYVE (200576)', 'PILRA (29992)', 'PILRB (29990)', 'PIM1 (5292)', 'PIM2 (11040)', 'PIM3 (415116)', 'PIMREG (54478)', 'PIN1 (5300)', 'PIN4 (5303)', 'PINK1 (65018)', 'PINLYP (390940)', 'PINX1 (54984)', 'PIP (5304)', 'PIP4K2A (5305)', 'PIP4K2B (8396)', 'PIP4K2C (79837)', 'PIP4P1 (90809)', 'PIP4P2 (55529)', 'PIP5K1A (8394)', 'PIP5K1B (8395)', 'PIP5K1C (23396)', 'PIP5KL1 (138429)', 'PIPOX (51268)', 'PIR (8544)', 'PIRT (644139)', 'PISD (23761)', 'PITHD1 (57095)', 'PITPNA (5306)', 'PITPNB (23760)', 'PITPNC1 (26207)', 'PITPNM1 (9600)', 'PITPNM2 (57605)', 'PITPNM3 (83394)', 'PITRM1 (10531)', 'PITX1 (5307)', 'PITX2 (5308)', 'PITX3 (5309)', 'PIWIL1 (9271)', 'PIWIL2 (55124)', 'PIWIL3 (440822)', 'PIWIL4 (143689)', 'PJA1 (64219)', 'PJA2 (9867)', 'PJVK (494513)', 'PKD1 (5310)', 'PKD1L1 (168507)', 'PKD2 (5311)', 'PKD2L1 (9033)', 'PKD2L2 (27039)', 'PKDCC (91461)', 'PKDREJ (10343)', 'PKHD1 (5314)', 'PKHD1L1 (93035)', 'PKIA (5569)', 'PKIB (5570)', 'PKIG (11142)', 'PKLR (5313)', 'PKM (5315)', 'PKMYT1 (9088)', 'PKN1 (5585)', 'PKN2 (5586)', 'PKN3 (29941)', 'PKNOX1 (5316)', 'PKNOX2 (63876)', 'PKP1 (5317)', 'PKP2 (5318)', 'PKP3 (11187)', 'PKP4 (8502)', 'PLA1A (51365)', 'PLA2G12A (81579)', 'PLA2G12B (84647)', 'PLA2G15 (23659)', 'PLA2G1B (5319)', 'PLA2G2A (5320)', 'PLA2G2D (26279)', 'PLA2G2E (30814)', 'PLA2G2F (64600)', 'PLA2G3 (50487)', 'PLA2G4A (5321)', 'PLA2G4B (100137049)', 'PLA2G4C (8605)', 'PLA2G4D (283748)', 'PLA2G4E (123745)', 'PLA2G4F (255189)', 'PLA2G5 (5322)', 'PLA2G6 (8398)', 'PLA2G7 (7941)', 'PLA2R1 (22925)', 'PLAA (9373)', 'PLAAT1 (57110)', 'PLAAT2 (54979)', 'PLAAT3 (11145)', 'PLAAT4 (5920)', 'PLAAT5 (117245)', 'PLAC1 (10761)', 'PLAC8 (51316)', 'PLAC8L1 (153770)', 'PLAC9 (219348)', 'PLAG1 (5324)', 'PLAGL1 (5325)', 'PLAGL2 (5326)', 'PLAT (5327)', 'PLAU (5328)', 'PLAUR (5329)', 'PLB1 (151056)', 'PLBD1 (79887)', 'PLBD2 (196463)', 'PLCB1 (23236)', 'PLCB2 (5330)', 'PLCB3 (5331)', 'PLCB4 (5332)', 'PLCD1 (5333)', 'PLCD4 (84812)', 'PLCE1 (51196)', 'PLCG1 (5335)', 'PLCG2 (5336)', 'PLCH1 (23007)', 'PLCH2 (9651)', 'PLCL1 (5334)', 'PLCL2 (23228)', 'PLCXD2 (257068)', 'PLCXD3 (345557)', 'PLCZ1 (89869)', 'PLD1 (5337)', 'PLD2 (5338)', 'PLD3 (23646)', 'PLD4 (122618)', 'PLD5 (200150)', 'PLD6 (201164)', 'PLEC (5339)', 'PLEK (5341)', 'PLEK2 (26499)', 'PLEKHA1 (59338)', 'PLEKHA3 (65977)', 'PLEKHA4 (57664)', 'PLEKHA5 (54477)', 'PLEKHA6 (22874)', 'PLEKHA7 (144100)', 'PLEKHA8 (84725)', 'PLEKHB1 (58473)', 'PLEKHB2 (55041)', 'PLEKHD1 (400224)', 'PLEKHF1 (79156)', 'PLEKHF2 (79666)', 'PLEKHG1 (57480)', 'PLEKHG2 (64857)', 'PLEKHG3 (26030)', 'PLEKHG4 (25894)', 'PLEKHG4B (153478)', 'PLEKHG5 (57449)', 'PLEKHG6 (55200)', 'PLEKHG7 (440107)', 'PLEKHH1 (57475)', 'PLEKHH2 (130271)', 'PLEKHH3 (79990)', 'PLEKHJ1 (55111)', 'PLEKHM1 (9842)', 'PLEKHM2 (23207)', 'PLEKHM3 (389072)', 'PLEKHN1 (84069)', 'PLEKHO1 (51177)', 'PLEKHO2 (80301)', 'PLEKHS1 (79949)', 'PLG (5340)', 'PLGRKT (55848)', 'PLIN1 (5346)', 'PLIN2 (123)', 'PLIN3 (10226)', 'PLIN4 (729359)', 'PLIN5 (440503)', 'PLK1 (5347)', 'PLK2 (10769)', 'PLK3 (1263)', 'PLK4 (10733)', 'PLK5 (126520)', 'PLLP (51090)', 'PLN (5350)', 'PLOD1 (5351)', 'PLOD2 (5352)', 'PLOD3 (8985)', 'PLP1 (5354)', 'PLP2 (5355)', 'PLPBP (11212)', 'PLPP1 (8611)', 'PLPP2 (8612)', 'PLPP3 (8613)', 'PLPP4 (196051)', 'PLPP5 (84513)', 'PLPP6 (403313)', 'PLPP7 (84814)', 'PLPPR1 (54886)', 'PLPPR2 (64748)', 'PLPPR3 (79948)', 'PLPPR4 (9890)', 'PLPPR5 (163404)', 'PLRG1 (5356)', 'PLS1 (5357)', 'PLS3 (5358)', 'PLSCR1 (5359)', 'PLSCR2 (57047)', 'PLSCR3 (57048)', 'PLSCR4 (57088)', 'PLSCR5 (389158)', 'PLTP (5360)', 'PLVAP (83483)', 'PLXDC1 (57125)', 'PLXDC2 (84898)', 'PLXNA1 (5361)', 'PLXNA2 (5362)', 'PLXNA3 (55558)', 'PLXNA4 (91584)', 'PLXNB1 (5364)', 'PLXNB2 (23654)', 'PLXNB3 (5365)', 'PLXNC1 (10154)', 'PLXND1 (23129)', 'PM20D1 (148811)', 'PM20D2 (135293)', 'PMAIP1 (5366)', 'PMCH (5367)', 'PMEL (6490)', 'PMEPA1 (56937)', 'PMF1 (11243)', 'PMFBP1 (83449)', 'PML (5371)', 'PMM1 (5372)', 'PMM2 (5373)', 'PMP2 (5375)', 'PMP22 (5376)', 'PMPCA (23203)', 'PMPCB (9512)', 'PMS1 (5378)', 'PMS2 (5395)', 'PMVK (10654)', 'PNCK (139728)', 'PNISR (25957)', 'PNKD (25953)', 'PNKP (11284)', 'PNLDC1 (154197)', 'PNLIP (5406)', 'PNLIPRP1 (5407)', 'PNLIPRP3 (119548)', 'PNMA1 (9240)', 'PNMA2 (10687)', 'PNMA3 (29944)', 'PNMA5 (114824)', 'PNMA6A (84968)', 'PNMA8A (55228)', 'PNMA8B (57469)', 'PNMT (5409)', 'PNN (5411)', 'PNO1 (56902)', 'PNOC (5368)', 'PNP (4860)', 'PNPLA1 (285848)', 'PNPLA2 (57104)', 'PNPLA3 (80339)', 'PNPLA4 (8228)', 'PNPLA5 (150379)', 'PNPLA6 (10908)', 'PNPLA7 (375775)', 'PNPLA8 (50640)', 'PNPO (55163)', 'PNPT1 (87178)', 'PNRC1 (10957)', 'PNRC2 (55629)', 'POC1A (25886)', 'POC1B (282809)', 'POC5 (134359)', 'PODN (127435)', 'PODNL1 (79883)', 'PODXL (5420)', 'PODXL2 (50512)', 'POF1B (79983)', 'POFUT1 (23509)', 'POFUT2 (23275)', 'POGK (57645)', 'POGLUT1 (56983)', 'POGLUT2 (79070)', 'POGLUT3 (143888)', 'POGZ (23126)', 'POLA1 (5422)', 'POLA2 (23649)', 'POLB (5423)', 'POLD1 (5424)', 'POLD2 (5425)', 'POLD3 (10714)', 'POLD4 (57804)', 'POLDIP3 (84271)', 'POLE (5426)', 'POLE2 (5427)', 'POLE3 (54107)', 'POLE4 (56655)', 'POLG (5428)', 'POLG2 (11232)', 'POLH (5429)', 'POLI (11201)', 'POLK (51426)', 'POLL (27343)', 'POLM (27434)', 'POLN (353497)', 'POLQ (10721)', 'POLR1A (25885)', 'POLR1B (84172)', 'POLR1C (9533)', 'POLR1D (51082)', 'POLR1E (64425)', 'POLR1F (221830)', 'POLR1G (10849)', 'POLR1H (30834)', 'POLR2B (5431)', 'POLR2C (5432)', 'POLR2D (5433)', 'POLR2E (5434)', 'POLR2F (5435)', 'POLR2G (5436)', 'POLR2H (5437)', 'POLR2I (5438)', 'POLR2K (5440)', 'POLR2L (5441)', 'POLR2M (81488)', 'POLR3A (11128)', 'POLR3B (55703)', 'POLR3C (10623)', 'POLR3D (661)', 'POLR3E (55718)', 'POLR3F (10621)', 'POLR3G (10622)', 'POLR3GL (84265)', 'POLR3H (171568)', 'POLR3K (51728)', 'POLRMT (5442)', 'POM121 (9883)', 'POM121C (100101267)', 'POM121L12 (285877)', 'POM121L2 (94026)', 'POMC (5443)', 'POMGNT1 (55624)', 'POMGNT2 (84892)', 'POMK (84197)', 'POMP (51371)', 'POMT1 (10585)', 'POMT2 (29954)', 'PON1 (5444)', 'PON2 (5445)', 'PON3 (5446)', 'POP1 (10940)', 'POP4 (10775)', 'POP5 (51367)', 'POP7 (10248)', 'POPDC2 (64091)', 'POPDC3 (64208)', 'POR (5447)', 'PORCN (64840)', 'POSTN (10631)', 'POT1 (25913)', 'POTEC (388468)', 'POTED (317754)', 'POTEE (445582)', 'POTEF (728378)', 'POTEG (404785)', 'POTEH (23784)', 'POTEI (653269)', 'POTEM (641455)', 'POU1F1 (5449)', 'POU2AF1 (5450)', 'POU2AF2 (341032)', 'POU2AF3 (120376)', 'POU2F1 (5451)', 'POU2F2 (5452)', 'POU2F3 (25833)', 'POU3F1 (5453)', 'POU3F2 (5454)', 'POU3F3 (5455)', 'POU3F4 (5456)', 'POU4F1 (5457)', 'POU4F2 (5458)', 'POU4F3 (5459)', 'POU5F1 (5460)', 'POU5F1B (5462)', 'POU5F2 (134187)', 'POU6F1 (5463)', 'POU6F2 (11281)', 'PP2D1 (151649)', 'PPA1 (5464)', 'PPA2 (27068)', 'PPAN (56342)', 'PPARA (5465)', 'PPARD (5467)', 'PPARG (5468)', 'PPARGC1A (10891)', 'PPARGC1B (133522)', 'PPAT (5471)', 'PPBP (5473)', 'PPCDC (60490)', 'PPCS (79717)', 'PPDPF (79144)', 'PPDPFL (492307)', 'PPEF1 (5475)', 'PPEF2 (5470)', 'PPFIA1 (8500)', 'PPFIA2 (8499)', 'PPFIA3 (8541)', 'PPFIA4 (8497)', 'PPFIBP1 (8496)', 'PPFIBP2 (8495)', 'PPHLN1 (51535)', 'PPIA (5478)', 'PPIAL4G (644591)', 'PPIB (5479)', 'PPIC (5480)', 'PPID (5481)', 'PPIE (10450)', 'PPIF (10105)', 'PPIG (9360)', 'PPIH (10465)', 'PPIL1 (51645)', 'PPIL2 (23759)', 'PPIL3 (53938)', 'PPIL4 (85313)', 'PPIL6 (285755)', 'PPIP5K1 (9677)', 'PPIP5K2 (23262)', 'PPL (5493)', 'PPM1A (5494)', 'PPM1B (5495)', 'PPM1D (8493)', 'PPM1E (22843)', 'PPM1F (9647)', 'PPM1G (5496)', 'PPM1H (57460)', 'PPM1J (333926)', 'PPM1K (152926)', 'PPM1L (151742)', 'PPM1M (132160)', 'PPM1N (147699)', 'PPME1 (51400)', 'PPOX (5498)', 'PPP1CA (5499)', 'PPP1CB (5500)', 'PPP1CC (5501)', 'PPP1R10 (5514)', 'PPP1R11 (6992)', 'PPP1R12A (4659)', 'PPP1R12B (4660)', 'PPP1R12C (54776)', 'PPP1R13B (23368)', 'PPP1R13L (10848)', 'PPP1R14A (94274)', 'PPP1R14B (26472)', 'PPP1R14C (81706)', 'PPP1R14D (54866)', 'PPP1R15A (23645)', 'PPP1R15B (84919)', 'PPP1R16A (84988)', 'PPP1R16B (26051)', 'PPP1R17 (10842)', 'PPP1R18 (170954)', 'PPP1R1A (5502)', 'PPP1R1B (84152)', 'PPP1R1C (151242)', 'PPP1R2 (5504)', 'PPP1R21 (129285)', 'PPP1R26 (9858)', 'PPP1R27 (116729)', 'PPP1R35 (221908)', 'PPP1R36 (145376)', 'PPP1R37 (284352)', 'PPP1R3A (5506)', 'PPP1R3B (79660)', 'PPP1R3C (5507)', 'PPP1R3D (5509)', 'PPP1R3E (90673)', 'PPP1R3F (89801)', 'PPP1R3G (648791)', 'PPP1R42 (286187)', 'PPP1R7 (5510)', 'PPP1R8 (5511)', 'PPP1R9A (55607)', 'PPP2CA (5515)', 'PPP2CB (5516)', 'PPP2R1A (5518)', 'PPP2R1B (5519)', 'PPP2R2A (5520)', 'PPP2R2B (5521)', 'PPP2R2C (5522)', 'PPP2R3A (5523)', 'PPP2R3C (55012)', 'PPP2R5A (5525)', 'PPP2R5B (5526)', 'PPP2R5C (5527)', 'PPP2R5D (5528)', 'PPP2R5E (5529)', 'PPP3CA (5530)', 'PPP3CB (5532)', 'PPP3CC (5533)', 'PPP3R1 (5534)', 'PPP3R2 (5535)', 'PPP4C (5531)', 'PPP4R1 (9989)', 'PPP4R2 (151987)', 'PPP4R3A (55671)', 'PPP4R3B (57223)', 'PPP4R4 (57718)', 'PPP5C (5536)', 'PPP6C (5537)', 'PPP6R1 (22870)', 'PPP6R2 (9701)', 'PPP6R3 (55291)', 'PPRC1 (23082)', 'PPT1 (5538)', 'PPT2 (9374)', 'PPTC7 (160760)', 'PPWD1 (23398)', 'PPY (5539)', 'PQBP1 (10084)', 'PRAC1 (84366)', 'PRADC1 (84279)', 'PRAF2 (11230)', 'PRAG1 (157285)', 'PRAM1 (84106)', 'PRAME (23532)', 'PRAMEF1 (65121)', 'PRAMEF11 (440560)', 'PRAMEF12 (390999)', 'PRAMEF14 (729528)', 'PRAMEF18 (391003)', 'PRAMEF19 (645414)', 'PRAMEF2 (65122)', 'PRAMEF20 (645425)', 'PRAMEF4 (400735)', 'PRAP1 (118471)', 'PRB2 (653247)', 'PRB4 (5545)', 'PRC1 (9055)', 'PRCC (5546)', 'PRCD (768206)', 'PRCP (5547)', 'PRDM1 (639)', 'PRDM10 (56980)', 'PRDM11 (56981)', 'PRDM12 (59335)', 'PRDM13 (59336)', 'PRDM14 (63978)', 'PRDM15 (63977)', 'PRDM16 (63976)', 'PRDM2 (7799)', 'PRDM4 (11108)', 'PRDM5 (11107)', 'PRDM6 (93166)', 'PRDM7 (11105)', 'PRDM8 (56978)', 'PRDM9 (56979)', 'PRDX1 (5052)', 'PRDX2 (7001)', 'PRDX3 (10935)', 'PRDX4 (10549)', 'PRDX5 (25824)', 'PRDX6 (9588)', 'PREB (10113)', 'PRELID1 (27166)', 'PRELID2 (153768)', 'PRELID3A (10650)', 'PRELID3B (51012)', 'PRELP (5549)', 'PREP (5550)', 'PREPL (9581)', 'PREX1 (57580)', 'PREX2 (80243)', 'PRF1 (5551)', 'PRG2 (5553)', 'PRG3 (10394)', 'PRG4 (10216)', 'PRH2 (5555)', 'PRICKLE1 (144165)', 'PRICKLE2 (166336)', 'PRICKLE3 (4007)', 'PRICKLE4 (29964)', 'PRIM1 (5557)', 'PRIMA1 (145270)', 'PRIMPOL (201973)', 'PRKAA1 (5562)', 'PRKAA2 (5563)', 'PRKAB1 (5564)', 'PRKAB2 (5565)', 'PRKACA (5566)', 'PRKACB (5567)', 'PRKACG (5568)', 'PRKAG1 (5571)', 'PRKAG2 (51422)', 'PRKAG3 (53632)', 'PRKAR1A (5573)', 'PRKAR1B (5575)', 'PRKAR2A (5576)', 'PRKAR2B (5577)', 'PRKCA (5578)', 'PRKCB (5579)', 'PRKCD (5580)', 'PRKCE (5581)', 'PRKCG (5582)', 'PRKCH (5583)', 'PRKCI (5584)', 'PRKCQ (5588)', 'PRKCSH (5589)', 'PRKCZ (5590)', 'PRKD1 (5587)', 'PRKD2 (25865)', 'PRKD3 (23683)', 'PRKG1 (5592)', 'PRKG2 (5593)', 'PRKN (5071)', 'PRKRA (8575)', 'PRKRIP1 (79706)', 'PRKX (5613)', 'PRL (5617)', 'PRLH (51052)', 'PRLHR (2834)', 'PRLR (5618)', 'PRM1 (5619)', 'PRM2 (5620)', 'PRMT1 (3276)', 'PRMT2 (3275)', 'PRMT3 (10196)', 'PRMT5 (10419)', 'PRMT6 (55170)', 'PRMT7 (54496)', 'PRMT8 (56341)', 'PRMT9 (90826)', 'PRND (23627)', 'PRNP (5621)', 'PROB1 (389333)', 'PROC (5624)', 'PROCA1 (147011)', 'PROCR (10544)', 'PRODH (5625)', 'PRODH2 (58510)', 'PROK1 (84432)', 'PROK2 (60675)', 'PROKR1 (10887)', 'PROKR2 (128674)', 'PROM1 (8842)', 'PROM2 (150696)', 'PROP1 (5626)', 'PRORP (9692)', 'PROS1 (5627)', 'PROSER1 (80209)', 'PROSER2 (254427)', 'PROX1 (5629)', 'PROX2 (283571)', 'PROZ (8858)', 'PRP4K (8899)', 'PRPF18 (8559)', 'PRPF19 (27339)', 'PRPF3 (9129)', 'PRPF31 (26121)', 'PRPF38A (84950)', 'PRPF38B (55119)', 'PRPF39 (55015)', 'PRPF4 (9128)', 'PRPF40A (55660)', 'PRPF40B (25766)', 'PRPF6 (24148)', 'PRPF8 (10594)', 'PRPH (5630)', 'PRPH2 (5961)', 'PRPS1 (5631)', 'PRPS1L1 (221823)', 'PRPS2 (5634)', 'PRPSAP1 (5635)', 'PRPSAP2 (5636)', 'PRR11 (55771)', 'PRR12 (57479)', 'PRR13 (54458)', 'PRR14 (78994)', 'PRR14L (253143)', 'PRR15 (222171)', 'PRR15L (79170)', 'PRR16 (51334)', 'PRR18 (285800)', 'PRR19 (284338)', 'PRR22 (163154)', 'PRR23A (729627)', 'PRR23B (389151)', 'PRR23C (389152)', 'PRR27 (401137)', 'PRR29 (92340)', 'PRR3 (80742)', 'PRR30 (339779)', 'PRR32 (100130613)', 'PRR35 (146325)', 'PRR4 (11272)', 'PRR5 (55615)', 'PRR5L (79899)', 'PRR7 (80758)', 'PRR9 (574414)', 'PRRC1 (133619)', 'PRRC2A (7916)', 'PRRC2B (84726)', 'PRRC2C (23215)', 'PRRG1 (5638)', 'PRRG2 (5639)', 'PRRG3 (79057)', 'PRRG4 (79056)', 'PRRT1 (80863)', 'PRRT2 (112476)', 'PRRT3 (285368)', 'PRRT4 (401399)', 'PRRX1 (5396)', 'PRRX2 (51450)', 'PRSS1 (5644)', 'PRSS12 (8492)', 'PRSS16 (10279)', 'PRSS21 (10942)', 'PRSS22 (64063)', 'PRSS23 (11098)', 'PRSS27 (83886)', 'PRSS3 (5646)', 'PRSS33 (260429)', 'PRSS35 (167681)', 'PRSS36 (146547)', 'PRSS37 (136242)', 'PRSS38 (339501)', 'PRSS48 (345062)', 'PRSS50 (29122)', 'PRSS53 (339105)', 'PRSS54 (221191)', 'PRSS55 (203074)', 'PRSS57 (400668)', 'PRSS58 (136541)', 'PRSS8 (5652)', 'PRTFDC1 (56952)', 'PRTG (283659)', 'PRTN3 (5657)', 'PRUNE1 (58497)', 'PRUNE2 (158471)', 'PRX (57716)', 'PRXL2A (84293)', 'PRXL2B (127281)', 'PRXL2C (195827)', 'PSAP (5660)', 'PSAPL1 (768239)', 'PSAT1 (29968)', 'PSCA (8000)', 'PSD (5662)', 'PSD2 (84249)', 'PSD3 (23362)', 'PSD4 (23550)', 'PSEN1 (5663)', 'PSEN2 (5664)', 'PSENEN (55851)', 'PSG1 (5669)', 'PSG11 (5680)', 'PSG2 (5670)', 'PSG3 (5671)', 'PSG4 (5672)', 'PSG5 (5673)', 'PSG6 (5675)', 'PSG8 (440533)', 'PSG9 (5678)', 'PSIP1 (11168)', 'PSKH1 (5681)', 'PSKH2 (85481)', 'PSMA1 (5682)', 'PSMA2 (5683)', 'PSMA3 (5684)', 'PSMA4 (5685)', 'PSMA5 (5686)', 'PSMA6 (5687)', 'PSMA7 (5688)', 'PSMA8 (143471)', 'PSMB1 (5689)', 'PSMB10 (5699)', 'PSMB11 (122706)', 'PSMB2 (5690)', 'PSMB3 (5691)', 'PSMB4 (5692)', 'PSMB5 (5693)', 'PSMB6 (5694)', 'PSMB7 (5695)', 'PSMB8 (5696)', 'PSMB9 (5698)', 'PSMC1 (5700)', 'PSMC2 (5701)', 'PSMC3 (5702)', 'PSMC3IP (29893)', 'PSMC4 (5704)', 'PSMC5 (5705)', 'PSMC6 (5706)', 'PSMD1 (5707)', 'PSMD10 (5716)', 'PSMD11 (5717)', 'PSMD12 (5718)', 'PSMD13 (5719)', 'PSMD14 (10213)', 'PSMD2 (5708)', 'PSMD3 (5709)', 'PSMD4 (5710)', 'PSMD5 (5711)', 'PSMD6 (9861)', 'PSMD7 (5713)', 'PSMD8 (5714)', 'PSMD9 (5715)', 'PSME1 (5720)', 'PSME2 (5721)', 'PSME3 (10197)', 'PSME3IP1 (80011)', 'PSME4 (23198)', 'PSMF1 (9491)', 'PSMG1 (8624)', 'PSMG2 (56984)', 'PSMG3 (84262)', 'PSMG4 (389362)', 'PSORS1C1 (170679)', 'PSORS1C2 (170680)', 'PSPC1 (55269)', 'PSPH (5723)', 'PSPN (5623)', 'PSRC1 (84722)', 'PSTK (118672)', 'PSTPIP1 (9051)', 'PSTPIP2 (9050)', 'PTAFR (5724)', 'PTAR1 (375743)', 'PTBP1 (5725)', 'PTBP2 (58155)', 'PTBP3 (9991)', 'PTCD1 (26024)', 'PTCD2 (79810)', 'PTCD3 (55037)', 'PTCH1 (5727)', 'PTCH2 (8643)', 'PTCHD1 (139411)', 'PTCHD4 (442213)', 'PTCRA (171558)', 'PTDSS1 (9791)', 'PTDSS2 (81490)', 'PTEN (5728)', 'PTER (9317)', 'PTF1A (256297)', 'PTGDR (5729)', 'PTGDR2 (11251)', 'PTGDS (5730)', 'PTGER1 (5731)', 'PTGER2 (5732)', 'PTGER3 (5733)', 'PTGER4 (5734)', 'PTGES (9536)', 'PTGES2 (80142)', 'PTGES3 (10728)', 'PTGES3L (100885848)', 'PTGFR (5737)', 'PTGFRN (5738)', 'PTGIR (5739)', 'PTGIS (5740)', 'PTGR1 (22949)', 'PTGR2 (145482)', 'PTGR3 (284273)', 'PTGS1 (5742)', 'PTGS2 (5743)', 'PTH (5741)', 'PTH1R (5745)', 'PTH2 (113091)', 'PTH2R (5746)', 'PTHLH (5744)', 'PTK2 (5747)', 'PTK2B (2185)', 'PTK6 (5753)', 'PTK7 (5754)', 'PTMA (5757)', 'PTMS (5763)', 'PTN (5764)', 'PTOV1 (53635)', 'PTP4A1 (7803)', 'PTP4A2 (8073)', 'PTP4A3 (11156)', 'PTPA (5524)', 'PTPDC1 (138639)', 'PTPMT1 (114971)', 'PTPN1 (5770)', 'PTPN11 (5781)', 'PTPN12 (5782)', 'PTPN13 (5783)', 'PTPN14 (5784)', 'PTPN18 (26469)', 'PTPN2 (5771)', 'PTPN20 (26095)', 'PTPN21 (11099)', 'PTPN22 (26191)', 'PTPN23 (25930)', 'PTPN3 (5774)', 'PTPN4 (5775)', 'PTPN5 (84867)', 'PTPN6 (5777)', 'PTPN7 (5778)', 'PTPN9 (5780)', 'PTPRA (5786)', 'PTPRB (5787)', 'PTPRC (5788)', 'PTPRCAP (5790)', 'PTPRD (5789)', 'PTPRE (5791)', 'PTPRF (5792)', 'PTPRG (5793)', 'PTPRH (5794)', 'PTPRJ (5795)', 'PTPRK (5796)', 'PTPRM (5797)', 'PTPRN (5798)', 'PTPRN2 (5799)', 'PTPRO (5800)', 'PTPRR (5801)', 'PTPRS (5802)', 'PTPRT (11122)', 'PTPRU (10076)', 'PTPRZ1 (5803)', 'PTRH1 (138428)', 'PTRH2 (51651)', 'PTRHD1 (391356)', 'PTS (5805)', 'PTTG1 (9232)', 'PTTG1IP (754)', 'PTTG2 (10744)', 'PTX3 (5806)', 'PTX4 (390667)', 'PUDP (8226)', 'PUF60 (22827)', 'PUM1 (9698)', 'PUM2 (23369)', 'PUM3 (9933)', 'PURA (5813)', 'PURB (5814)', 'PURG (29942)', 'PUS1 (80324)', 'PUS10 (150962)', 'PUS3 (83480)', 'PUS7 (54517)', 'PUS7L (83448)', 'PUSL1 (126789)', 'PVALB (5816)', 'PVR (5817)', 'PVRIG (79037)', 'PWP1 (11137)', 'PWP2 (5822)', 'PWWP2A (114825)', 'PWWP2B (170394)', 'PWWP3A (84939)', 'PWWP3B (139221)', 'PXDC1 (221749)', 'PXDN (7837)', 'PXDNL (137902)', 'PXK (54899)', 'PXMP2 (5827)', 'PXMP4 (11264)', 'PXN (5829)', 'PXT1 (222659)', 'PXYLP1 (92370)', 'PYCARD (29108)', 'PYCR1 (5831)', 'PYCR2 (29920)', 'PYCR3 (65263)', 'PYDC1 (260434)', 'PYGB (5834)', 'PYGL (5836)', 'PYGM (5837)', 'PYGO1 (26108)', 'PYGO2 (90780)', 'PYHIN1 (149628)', 'PYM1 (84305)', 'PYROXD1 (79912)', 'PYROXD2 (84795)', 'PYURF (100996939)', 'PYY (5697)', 'PZP (5858)', 'QARS1 (5859)', 'QDPR (5860)', 'QKI (9444)', 'QNG1 (84267)', 'QPCT (25797)', 'QPCTL (54814)', 'QPRT (23475)', 'QRFP (347148)', 'QRFPR (84109)', 'QRICH1 (54870)', 'QRICH2 (84074)', 'QRSL1 (55278)', 'QSER1 (79832)', 'QSOX1 (5768)', 'QSOX2 (169714)', 'QTRT1 (81890)', 'QTRT2 (79691)', 'R3HCC1 (203069)', 'R3HCC1L (27291)', 'R3HDM1 (23518)', 'R3HDM2 (22864)', 'R3HDM4 (91300)', 'R3HDML (140902)', 'RAB10 (10890)', 'RAB11A (8766)', 'RAB11B (9230)', 'RAB11FIP1 (80223)', 'RAB11FIP2 (22841)', 'RAB11FIP3 (9727)', 'RAB11FIP4 (84440)', 'RAB11FIP5 (26056)', 'RAB12 (201475)', 'RAB13 (5872)', 'RAB14 (51552)', 'RAB15 (376267)', 'RAB17 (64284)', 'RAB18 (22931)', 'RAB19 (401409)', 'RAB1A (5861)', 'RAB1B (81876)', 'RAB20 (55647)', 'RAB21 (23011)', 'RAB22A (57403)', 'RAB23 (51715)', 'RAB24 (53917)', 'RAB25 (57111)', 'RAB26 (25837)', 'RAB27A (5873)', 'RAB27B (5874)', 'RAB28 (9364)', 'RAB29 (8934)', 'RAB2A (5862)', 'RAB2B (84932)', 'RAB30 (27314)', 'RAB31 (11031)', 'RAB32 (10981)', 'RAB33A (9363)', 'RAB33B (83452)', 'RAB34 (83871)', 'RAB35 (11021)', 'RAB36 (9609)', 'RAB37 (326624)', 'RAB38 (23682)', 'RAB39A (54734)', 'RAB39B (116442)', 'RAB3A (5864)', 'RAB3B (5865)', 'RAB3C (115827)', 'RAB3D (9545)', 'RAB3GAP1 (22930)', 'RAB3GAP2 (25782)', 'RAB3IL1 (5866)', 'RAB3IP (117177)', 'RAB40A (142684)', 'RAB40AL (282808)', 'RAB40B (10966)', 'RAB40C (57799)', 'RAB41 (347517)', 'RAB42 (115273)', 'RAB43 (339122)', 'RAB4A (5867)', 'RAB4B (53916)', 'RAB5A (5868)', 'RAB5B (5869)', 'RAB5C (5878)', 'RAB5IF (55969)', 'RAB6A (5870)', 'RAB6B (51560)', 'RAB7A (7879)', 'RAB8A (4218)', 'RAB8B (51762)', 'RAB9A (9367)', 'RAB9B (51209)', 'RABAC1 (10567)', 'RABEP1 (9135)', 'RABEP2 (79874)', 'RABEPK (10244)', 'RABGAP1 (23637)', 'RABGAP1L (9910)', 'RABGEF1 (27342)', 'RABGGTA (5875)', 'RABGGTB (5876)', 'RABIF (5877)', 'RABL2A (11159)', 'RABL2B (11158)', 'RABL3 (285282)', 'RABL6 (55684)', 'RAC1 (5879)', 'RAC2 (5880)', 'RAC3 (5881)', 'RACGAP1 (29127)', 'RACK1 (10399)', 'RAD1 (5810)', 'RAD17 (5884)', 'RAD18 (56852)', 'RAD21 (5885)', 'RAD21L1 (642636)', 'RAD23A (5886)', 'RAD23B (5887)', 'RAD50 (10111)', 'RAD51 (5888)', 'RAD51AP1 (10635)', 'RAD51AP2 (729475)', 'RAD51B (5890)', 'RAD51C (5889)', 'RAD51D (5892)', 'RAD52 (5893)', 'RAD54B (25788)', 'RAD54L (8438)', 'RAD54L2 (23132)', 'RAD9A (5883)', 'RAD9B (144715)', 'RADIL (55698)', 'RADX (55086)', 'RAE1 (8480)', 'RAET1E (135250)', 'RAET1G (353091)', 'RAET1L (154064)', 'RAF1 (5894)', 'RAG1 (5896)', 'RAG2 (5897)', 'RAI1 (10743)', 'RAI14 (26064)', 'RAI2 (10742)', 'RALA (5898)', 'RALB (5899)', 'RALBP1 (10928)', 'RALGAPA1 (253959)', 'RALGAPA2 (57186)', 'RALGAPB (57148)', 'RALGDS (5900)', 'RALGPS1 (9649)', 'RALGPS2 (55103)', 'RALY (22913)', 'RALYL (138046)', 'RAMP1 (10267)', 'RAMP2 (10266)', 'RAMP3 (10268)', 'RAN (5901)', 'RANBP1 (5902)', 'RANBP10 (57610)', 'RANBP17 (64901)', 'RANBP2 (5903)', 'RANBP3 (8498)', 'RANBP3L (202151)', 'RANBP6 (26953)', 'RANBP9 (10048)', 'RANGAP1 (5905)', 'RANGRF (29098)', 'RAP1A (5906)', 'RAP1B (5908)', 'RAP1GAP (5909)', 'RAP1GAP2 (23108)', 'RAP1GDS1 (5910)', 'RAP2A (5911)', 'RAP2B (5912)', 'RAP2C (57826)', 'RAPGEF1 (2889)', 'RAPGEF2 (9693)', 'RAPGEF3 (10411)', 'RAPGEF4 (11069)', 'RAPGEF5 (9771)', 'RAPGEF6 (51735)', 'RAPGEFL1 (51195)', 'RAPH1 (65059)', 'RAPSN (5913)', 'RARA (5914)', 'RARB (5915)', 'RARG (5916)', 'RARRES1 (5918)', 'RARRES2 (5919)', 'RARS1 (5917)', 'RARS2 (57038)', 'RASA1 (5921)', 'RASA2 (5922)', 'RASA3 (22821)', 'RASA4 (10156)', 'RASA4B (100271927)', 'RASAL1 (8437)', 'RASAL2 (9462)', 'RASAL3 (64926)', 'RASD1 (51655)', 'RASD2 (23551)', 'RASEF (158158)', 'RASGEF1A (221002)', 'RASGEF1B (153020)', 'RASGEF1C (255426)', 'RASGRF1 (5923)', 'RASGRF2 (5924)', 'RASGRP1 (10125)', 'RASGRP2 (10235)', 'RASGRP3 (25780)', 'RASGRP4 (115727)', 'RASIP1 (54922)', 'RASL10A (10633)', 'RASL10B (91608)', 'RASL11A (387496)', 'RASL11B (65997)', 'RASL12 (51285)', 'RASSF1 (11186)', 'RASSF2 (9770)', 'RASSF3 (283349)', 'RASSF4 (83937)', 'RASSF5 (83593)', 'RASSF6 (166824)', 'RASSF7 (8045)', 'RASSF8 (11228)', 'RASSF9 (9182)', 'RAVER1 (125950)', 'RAVER2 (55225)', 'RAX (30062)', 'RAX2 (84839)', 'RB1 (5925)', 'RB1CC1 (9821)', 'RBAK (57786)', 'RBBP4 (5928)', 'RBBP5 (5929)', 'RBBP6 (5930)', 'RBBP7 (5931)', 'RBBP8 (5932)', 'RBBP8NL (140893)', 'RBBP9 (10741)', 'RBCK1 (10616)', 'RBFA (79863)', 'RBFOX1 (54715)', 'RBFOX2 (23543)', 'RBFOX3 (146713)', 'RBIS (401466)', 'RBKS (64080)', 'RBL1 (5933)', 'RBL2 (5934)', 'RBM10 (8241)', 'RBM11 (54033)', 'RBM12 (10137)', 'RBM12B (389677)', 'RBM14 (10432)', 'RBM15 (64783)', 'RBM15B (29890)', 'RBM17 (84991)', 'RBM18 (92400)', 'RBM19 (9904)', 'RBM20 (282996)', 'RBM22 (55696)', 'RBM23 (55147)', 'RBM24 (221662)', 'RBM25 (58517)', 'RBM26 (64062)', 'RBM27 (54439)', 'RBM28 (55131)', 'RBM3 (5935)', 'RBM33 (155435)', 'RBM34 (23029)', 'RBM38 (55544)', 'RBM39 (9584)', 'RBM4 (5936)', 'RBM41 (55285)', 'RBM42 (79171)', 'RBM43 (375287)', 'RBM44 (375316)', 'RBM45 (129831)', 'RBM46 (166863)', 'RBM47 (54502)', 'RBM48 (84060)', 'RBM4B (83759)', 'RBM5 (10181)', 'RBM6 (10180)', 'RBM7 (10179)', 'RBM8A (9939)', 'RBMS1 (5937)', 'RBMS2 (5939)', 'RBMS3 (27303)', 'RBMX (27316)', 'RBMX2 (51634)', 'RBMXL1 (494115)', 'RBMXL2 (27288)', 'RBMXL3 (139804)', 'RBP1 (5947)', 'RBP2 (5948)', 'RBP3 (5949)', 'RBP4 (5950)', 'RBP5 (83758)', 'RBP7 (116362)', 'RBPJ (3516)', 'RBPJL (11317)', 'RBPMS (11030)', 'RBPMS2 (348093)', 'RBSN (64145)', 'RBX1 (9978)', 'RC3H1 (149041)', 'RC3H2 (54542)', 'RCAN1 (1827)', 'RCAN2 (10231)', 'RCAN3 (11123)', 'RCBTB1 (55213)', 'RCBTB2 (1102)', 'RCC1 (1104)', 'RCC1L (81554)', 'RCC2 (55920)', 'RCCD1 (91433)', 'RCE1 (9986)', 'RCHY1 (25898)', 'RCL1 (10171)', 'RCN1 (5954)', 'RCN2 (5955)', 'RCN3 (57333)', 'RCOR1 (23186)', 'RCOR2 (283248)', 'RCOR3 (55758)', 'RCSD1 (92241)', 'RCVRN (5957)', 'RD3 (343035)', 'RD3L (647286)', 'RDH10 (157506)', 'RDH11 (51109)', 'RDH12 (145226)', 'RDH13 (112724)', 'RDH14 (57665)', 'RDH16 (8608)', 'RDH5 (5959)', 'RDH8 (50700)', 'RDM1 (201299)', 'RDX (5962)', 'REC114 (283677)', 'REC8 (9985)', 'RECK (8434)', 'RECQL (5965)', 'RECQL5 (9400)', 'REDIC1 (283461)', 'REEP1 (65055)', 'REEP2 (51308)', 'REEP3 (221035)', 'REEP4 (80346)', 'REEP5 (7905)', 'REEP6 (92840)', 'REG1A (5967)', 'REG1B (5968)', 'REG3A (5068)', 'REG3G (130120)', 'REG4 (83998)', 'REL (5966)', 'RELA (5970)', 'RELB (5971)', 'RELCH (57614)', 'RELL1 (768211)', 'RELL2 (285613)', 'RELN (5649)', 'RELT (84957)', 'REM1 (28954)', 'REM2 (161253)', 'REN (5972)', 'RENBP (5973)', 'REP15 (387849)', 'REPIN1 (29803)', 'REPS1 (85021)', 'REPS2 (9185)', 'RER1 (11079)', 'RERE (473)', 'RERG (85004)', 'RERGL (79785)', 'RESF1 (55196)', 'RESP18 (389075)', 'REST (5978)', 'RET (5979)', 'RETN (56729)', 'RETNLB (84666)', 'RETREG1 (54463)', 'RETREG2 (79137)', 'RETREG3 (162427)', 'RETSAT (54884)', 'REV1 (51455)', 'REV3L (5980)', 'REX1BD (55049)', 'REXO1 (57455)', 'REXO2 (25996)', 'REXO4 (57109)', 'REXO5 (81691)', 'RFC1 (5981)', 'RFC2 (5982)', 'RFC3 (5983)', 'RFC4 (5984)', 'RFC5 (5985)', 'RFESD (317671)', 'RFFL (117584)', 'RFK (55312)', 'RFLNA (144347)', 'RFNG (5986)', 'RFPL1 (5988)', 'RFPL2 (10739)', 'RFPL3 (10738)', 'RFPL4B (442247)', 'RFT1 (91869)', 'RFTN1 (23180)', 'RFTN2 (130132)', 'RFWD3 (55159)', 'RFX1 (5989)', 'RFX2 (5990)', 'RFX3 (5991)', 'RFX4 (5992)', 'RFX5 (5993)', 'RFX6 (222546)', 'RFX8 (731220)', 'RFXANK (8625)', 'RFXAP (5994)', 'RGCC (28984)', 'RGL1 (23179)', 'RGL2 (5863)', 'RGL3 (57139)', 'RGL4 (266747)', 'RGMA (56963)', 'RGMB (285704)', 'RGN (9104)', 'RGP1 (9827)', 'RGPD4 (285190)', 'RGR (5995)', 'RGS1 (5996)', 'RGS10 (6001)', 'RGS11 (8786)', 'RGS12 (6002)', 'RGS13 (6003)', 'RGS14 (10636)', 'RGS16 (6004)', 'RGS17 (26575)', 'RGS18 (64407)', 'RGS19 (10287)', 'RGS2 (5997)', 'RGS20 (8601)', 'RGS21 (431704)', 'RGS22 (26166)', 'RGS3 (5998)', 'RGS4 (5999)', 'RGS5 (8490)', 'RGS6 (9628)', 'RGS7 (6000)', 'RGS7BP (401190)', 'RGS8 (85397)', 'RGS9 (8787)', 'RGS9BP (388531)', 'RGSL1 (353299)', 'RHAG (6005)', 'RHBDD1 (84236)', 'RHBDD2 (57414)', 'RHBDD3 (25807)', 'RHBDF1 (64285)', 'RHBDF2 (79651)', 'RHBDL1 (9028)', 'RHBDL2 (54933)', 'RHBDL3 (162494)', 'RHCE (6006)', 'RHCG (51458)', 'RHD (6007)', 'RHEB (6009)', 'RHEBL1 (121268)', 'RHEX (440712)', 'RHNO1 (83695)', 'RHO (6010)', 'RHOA (387)', 'RHOB (388)', 'RHOBTB1 (9886)', 'RHOBTB2 (23221)', 'RHOBTB3 (22836)', 'RHOC (389)', 'RHOD (29984)', 'RHOF (54509)', 'RHOG (391)', 'RHOH (399)', 'RHOJ (57381)', 'RHOQ (23433)', 'RHOT1 (55288)', 'RHOT2 (89941)', 'RHOU (58480)', 'RHOV (171177)', 'RHOXF1 (158800)', 'RHOXF2 (84528)', 'RHOXF2B (727940)', 'RHPN1 (114822)', 'RHPN2 (85415)', 'RIBC1 (158787)', 'RIC1 (57589)', 'RIC3 (79608)', 'RIC8A (60626)', 'RIC8B (55188)', 'RICTOR (253260)', 'RIDA (10247)', 'RIF1 (55183)', 'RIGI (23586)', 'RIIAD1 (284485)', 'RILP (83547)', 'RILPL1 (353116)', 'RILPL2 (196383)', 'RIMBP2 (23504)', 'RIMBP3 (85376)', 'RIMKLA (284716)', 'RIMKLB (57494)', 'RIMOC1 (285636)', 'RIMS1 (22999)', 'RIMS2 (9699)', 'RIMS3 (9783)', 'RIMS4 (140730)', 'RIN1 (9610)', 'RIN2 (54453)', 'RIN3 (79890)', 'RING1 (6015)', 'RINL (126432)', 'RINT1 (60561)', 'RIOK1 (83732)', 'RIOK2 (55781)', 'RIOK3 (8780)', 'RIOX2 (84864)', 'RIPK1 (8737)', 'RIPK2 (8767)', 'RIPK3 (11035)', 'RIPK4 (54101)', 'RIPOR1 (79567)', 'RIPOR2 (9750)', 'RIPOR3 (140876)', 'RIPPLY1 (92129)', 'RIPPLY2 (134701)', 'RIPPLY3 (53820)', 'RIT1 (6016)', 'RIT2 (6014)', 'RITA1 (84934)', 'RLBP1 (6017)', 'RLF (6018)', 'RLIG1 (91298)', 'RLIM (51132)', 'RLN1 (6013)', 'RLN2 (6019)', 'RLN3 (117579)', 'RMC1 (29919)', 'RMDN1 (51115)', 'RMDN2 (151393)', 'RMDN3 (55177)', 'RMI1 (80010)', 'RMI2 (116028)', 'RMND1 (55005)', 'RMND5A (64795)', 'RMND5B (64777)', 'RNASE1 (6035)', 'RNASE10 (338879)', 'RNASE11 (122651)', 'RNASE12 (493901)', 'RNASE13 (440163)', 'RNASE2 (6036)', 'RNASE3 (6037)', 'RNASE4 (6038)', 'RNASE6 (6039)', 'RNASE7 (84659)', 'RNASE8 (122665)', 'RNASE9 (390443)', 'RNASEH1 (246243)', 'RNASEH2A (10535)', 'RNASEH2B (79621)', 'RNASEH2C (84153)', 'RNASEK (440400)', 'RNASEL (6041)', 'RNASET2 (8635)', 'RND1 (27289)', 'RND2 (8153)', 'RND3 (390)', 'RNF10 (9921)', 'RNF103 (7844)', 'RNF11 (26994)', 'RNF111 (54778)', 'RNF112 (7732)', 'RNF113A (7737)', 'RNF113B (140432)', 'RNF114 (55905)', 'RNF115 (27246)', 'RNF121 (55298)', 'RNF122 (79845)', 'RNF123 (63891)', 'RNF125 (54941)', 'RNF126 (55658)', 'RNF128 (79589)', 'RNF13 (11342)', 'RNF130 (55819)', 'RNF133 (168433)', 'RNF135 (84282)', 'RNF138 (51444)', 'RNF139 (11236)', 'RNF14 (9604)', 'RNF141 (50862)', 'RNF144A (9781)', 'RNF144B (255488)', 'RNF145 (153830)', 'RNF146 (81847)', 'RNF148 (378925)', 'RNF149 (284996)', 'RNF150 (57484)', 'RNF151 (146310)', 'RNF152 (220441)', 'RNF157 (114804)', 'RNF166 (115992)', 'RNF167 (26001)', 'RNF168 (165918)', 'RNF169 (254225)', 'RNF17 (56163)', 'RNF170 (81790)', 'RNF175 (285533)', 'RNF180 (285671)', 'RNF181 (51255)', 'RNF182 (221687)', 'RNF183 (138065)', 'RNF185 (91445)', 'RNF186 (54546)', 'RNF19A (25897)', 'RNF19B (127544)', 'RNF2 (6045)', 'RNF20 (56254)', 'RNF207 (388591)', 'RNF208 (727800)', 'RNF212 (285498)', 'RNF212B (100507650)', 'RNF213 (57674)', 'RNF214 (257160)', 'RNF215 (200312)', 'RNF216 (54476)', 'RNF217 (154214)', 'RNF220 (55182)', 'RNF222 (643904)', 'RNF223 (401934)', 'RNF224 (643596)', 'RNF24 (11237)', 'RNF25 (64320)', 'RNF26 (79102)', 'RNF31 (55072)', 'RNF32 (140545)', 'RNF34 (80196)', 'RNF38 (152006)', 'RNF39 (80352)', 'RNF4 (6047)', 'RNF40 (9810)', 'RNF41 (10193)', 'RNF43 (54894)', 'RNF44 (22838)', 'RNF5 (6048)', 'RNF6 (6049)', 'RNF7 (9616)', 'RNF8 (9025)', 'RNFT1 (51136)', 'RNFT2 (84900)', 'RNGTT (8732)', 'RNH1 (6050)', 'RNLS (55328)', 'RNMT (8731)', 'RNPC3 (55599)', 'RNPEP (6051)', 'RNPS1 (10921)', 'RO60 (6738)', 'ROBO1 (6091)', 'ROBO2 (6092)', 'ROBO3 (64221)', 'ROBO4 (54538)', 'ROCK1 (6093)', 'ROCK2 (9475)', 'ROGDI (79641)', 'ROM1 (6094)', 'ROMO1 (140823)', 'ROPN1 (54763)', 'ROPN1B (152015)', 'ROPN1L (83853)', 'ROR1 (4919)', 'ROR2 (4920)', 'RORA (6095)', 'RORB (6096)', 'RORC (6097)', 'ROS1 (6098)', 'RP1 (6101)', 'RP1L1 (94137)', 'RP2 (6102)', 'RP9 (6100)', 'RPA1 (6117)', 'RPA2 (6118)', 'RPA3 (6119)', 'RPA4 (29935)', 'RPAIN (84268)', 'RPAP1 (26015)', 'RPAP2 (79871)', 'RPAP3 (79657)', 'RPE (6120)', 'RPE65 (6121)', 'RPEL1 (729020)', 'RPF1 (80135)', 'RPF2 (84154)', 'RPGR (6103)', 'RPGRIP1 (57096)', 'RPGRIP1L (23322)', 'RPH3A (22895)', 'RPH3AL (9501)', 'RPIA (22934)', 'RPL10 (6134)', 'RPL10A (4736)', 'RPL10L (140801)', 'RPL11 (6135)', 'RPL12 (6136)', 'RPL13 (6137)', 'RPL13A (23521)', 'RPL14 (9045)', 'RPL17 (6139)', 'RPL18 (6141)', 'RPL18A (6142)', 'RPL19 (6143)', 'RPL21 (6144)', 'RPL22 (6146)', 'RPL22L1 (200916)', 'RPL23 (9349)', 'RPL23A (6147)', 'RPL24 (6152)', 'RPL26 (6154)', 'RPL26L1 (51121)', 'RPL27 (6155)', 'RPL27A (6157)', 'RPL28 (6158)', 'RPL3 (6122)', 'RPL30 (6156)', 'RPL31 (6160)', 'RPL32 (6161)', 'RPL35 (11224)', 'RPL35A (6165)', 'RPL36 (25873)', 'RPL36A (6173)', 'RPL36AL (6166)', 'RPL37 (6167)', 'RPL37A (6168)', 'RPL38 (6169)', 'RPL39 (6170)', 'RPL39L (116832)', 'RPL3L (6123)', 'RPL4 (6124)', 'RPL41 (6171)', 'RPL5 (6125)', 'RPL6 (6128)', 'RPL7 (6129)', 'RPL7A (6130)', 'RPL7L1 (285855)', 'RPL8 (6132)', 'RPL9 (6133)', 'RPLP0 (6175)', 'RPLP1 (6176)', 'RPLP2 (6181)', 'RPN1 (6184)', 'RPN2 (6185)', 'RPP14 (11102)', 'RPP21 (79897)', 'RPP25 (54913)', 'RPP25L (138716)', 'RPP30 (10556)', 'RPP38 (10557)', 'RPP40 (10799)', 'RPRD1A (55197)', 'RPRD1B (58490)', 'RPRD2 (23248)', 'RPRM (56475)', 'RPRML (388394)', 'RPS10 (6204)', 'RPS11 (6205)', 'RPS12 (6206)', 'RPS13 (6207)', 'RPS14 (6208)', 'RPS15 (6209)', 'RPS15A (6210)', 'RPS16 (6217)', 'RPS17 (6218)', 'RPS18 (6222)', 'RPS19 (6223)', 'RPS19BP1 (91582)', 'RPS2 (6187)', 'RPS20 (6224)', 'RPS21 (6227)', 'RPS23 (6228)', 'RPS24 (6229)', 'RPS25 (6230)', 'RPS27A (6233)', 'RPS27L (51065)', 'RPS28 (6234)', 'RPS29 (6235)', 'RPS3 (6188)', 'RPS3A (6189)', 'RPS4X (6191)', 'RPS4Y1 (6192)', 'RPS4Y2 (140032)', 'RPS5 (6193)', 'RPS6 (6194)', 'RPS6KA1 (6195)', 'RPS6KA2 (6196)', 'RPS6KA3 (6197)', 'RPS6KA4 (8986)', 'RPS6KA5 (9252)', 'RPS6KA6 (27330)', 'RPS6KB1 (6198)', 'RPS6KB2 (6199)', 'RPS6KC1 (26750)', 'RPS6KL1 (83694)', 'RPS7 (6201)', 'RPS8 (6202)', 'RPS9 (6203)', 'RPTN (126638)', 'RPTOR (57521)', 'RPUSD1 (113000)', 'RPUSD2 (27079)', 'RPUSD3 (285367)', 'RPUSD4 (84881)', 'RRAD (6236)', 'RRAGA (10670)', 'RRAGB (10325)', 'RRAGC (64121)', 'RRAGD (58528)', 'RRAS (6237)', 'RRAS2 (22800)', 'RRBP1 (6238)', 'RREB1 (6239)', 'RRH (10692)', 'RRM1 (6240)', 'RRM2 (6241)', 'RRM2B (50484)', 'RRN3 (54700)', 'RRP1 (8568)', 'RRP12 (23223)', 'RRP15 (51018)', 'RRP1B (23076)', 'RRP36 (88745)', 'RRP7A (27341)', 'RRP8 (23378)', 'RRP9 (9136)', 'RRS1 (23212)', 'RS1 (6247)', 'RSAD1 (55316)', 'RSAD2 (91543)', 'RSBN1 (54665)', 'RSBN1L (222194)', 'RSC1A1 (6248)', 'RSF1 (51773)', 'RSL1D1 (26156)', 'RSL24D1 (51187)', 'RSPH1 (89765)', 'RSPH14 (27156)', 'RSPH3 (83861)', 'RSPH4A (345895)', 'RSPH6A (81492)', 'RSPH9 (221421)', 'RSPO1 (284654)', 'RSPO2 (340419)', 'RSPO3 (84870)', 'RSPO4 (343637)', 'RSPRY1 (89970)', 'RSRC1 (51319)', 'RSRC2 (65117)', 'RSRP1 (57035)', 'RSU1 (6251)', 'RTBDN (83546)', 'RTCA (8634)', 'RTCB (51493)', 'RTEL1 (51750)', 'RTF1 (23168)', 'RTF2 (51507)', 'RTKN (6242)', 'RTKN2 (219790)', 'RTL1 (388015)', 'RTL10 (79680)', 'RTL3 (203430)', 'RTL4 (340595)', 'RTL5 (340526)', 'RTL6 (84247)', 'RTL8A (26071)', 'RTL8B (441518)', 'RTL8C (8933)', 'RTL9 (57529)', 'RTN1 (6252)', 'RTN2 (6253)', 'RTN3 (10313)', 'RTN4 (57142)', 'RTN4IP1 (84816)', 'RTN4R (65078)', 'RTN4RL1 (146760)', 'RTN4RL2 (349667)', 'RTP1 (132112)', 'RTP2 (344892)', 'RTP3 (83597)', 'RTP4 (64108)', 'RTP5 (285093)', 'RTRAF (51637)', 'RTTN (25914)', 'RUBCN (9711)', 'RUBCNL (80183)', 'RUFY1 (80230)', 'RUFY2 (55680)', 'RUFY3 (22902)', 'RUNDC1 (146923)', 'RUNDC3A (10900)', 'RUNDC3B (154661)', 'RUNX1 (861)', 'RUNX1T1 (862)', 'RUNX2 (860)', 'RUNX3 (864)', 'RUSC1 (23623)', 'RUSC2 (9853)', 'RUSF1 (64755)', 'RUVBL1 (8607)', 'RUVBL2 (10856)', 'RWDD1 (51389)', 'RWDD2A (112611)', 'RWDD2B (10069)', 'RWDD3 (25950)', 'RWDD4 (201965)', 'RXFP1 (59350)', 'RXFP2 (122042)', 'RXFP3 (51289)', 'RXFP4 (339403)', 'RXRA (6256)', 'RXRB (6257)', 'RXRG (6258)', 'RXYLT1 (10329)', 'RYR1 (6261)', 'RYR2 (6262)', 'RYR3 (6263)', 'S100A1 (6271)', 'S100A10 (6281)', 'S100A11 (6282)', 'S100A12 (6283)', 'S100A13 (6284)', 'S100A14 (57402)', 'S100A16 (140576)', 'S100A2 (6273)', 'S100A3 (6274)', 'S100A4 (6275)', 'S100A5 (6276)', 'S100A6 (6277)', 'S100A7 (6278)', 'S100A8 (6279)', 'S100A9 (6280)', 'S100B (6285)', 'S100G (795)', 'S100P (6286)', 'S100PBP (64766)', 'S100Z (170591)', 'S1PR1 (1901)', 'S1PR2 (9294)', 'S1PR3 (1903)', 'S1PR4 (8698)', 'S1PR5 (53637)', 'SAA4 (6291)', 'SAAL1 (113174)', 'SAC3D1 (29901)', 'SACM1L (22908)', 'SACS (26278)', 'SAE1 (10055)', 'SAFB (6294)', 'SAFB2 (9667)', 'SAG (6295)', 'SAGE1 (55511)', 'SALL1 (6299)', 'SALL2 (6297)', 'SALL3 (27164)', 'SALL4 (57167)', 'SAMD10 (140700)', 'SAMD11 (148398)', 'SAMD12 (401474)', 'SAMD13 (148418)', 'SAMD14 (201191)', 'SAMD15 (161394)', 'SAMD3 (154075)', 'SAMD4A (23034)', 'SAMD4B (55095)', 'SAMD5 (389432)', 'SAMD7 (344658)', 'SAMD8 (142891)', 'SAMD9 (54809)', 'SAMD9L (219285)', 'SAMHD1 (25939)', 'SAMM50 (25813)', 'SAMSN1 (64092)', 'SANBR (84542)', 'SAP130 (79595)', 'SAP18 (10284)', 'SAP25 (100316904)', 'SAP30 (8819)', 'SAP30BP (29115)', 'SAP30L (79685)', 'SAPCD1 (401251)', 'SAPCD2 (89958)', 'SAR1A (56681)', 'SAR1B (51128)', 'SARAF (51669)', 'SARDH (1757)', 'SARNP (84324)', 'SARS1 (6301)', 'SARS2 (54938)', 'SART1 (9092)', 'SART3 (9733)', 'SASH1 (23328)', 'SASH3 (54440)', 'SASS6 (163786)', 'SAT1 (6303)', 'SAT2 (112483)', 'SATB1 (6304)', 'SATB2 (23314)', 'SATL1 (340562)', 'SAV1 (60485)', 'SAXO1 (158297)', 'SAXO2 (283726)', 'SAXO4 (220004)', 'SAXO5 (374877)', 'SAYSD1 (55776)', 'SBDS (51119)', 'SBF1 (6305)', 'SBF2 (81846)', 'SBK1 (388228)', 'SBK2 (646643)', 'SBNO1 (55206)', 'SBNO2 (22904)', 'SBSN (374897)', 'SBSPON (157869)', 'SC5D (6309)', 'SCAF1 (58506)', 'SCAF11 (9169)', 'SCAF4 (57466)', 'SCAF8 (22828)', 'SCAI (286205)', 'SCAMP2 (10066)', 'SCAMP3 (10067)', 'SCAMP4 (113178)', 'SCAMP5 (192683)', 'SCAND1 (51282)', 'SCAND3 (114821)', 'SCAP (22937)', 'SCAPER (49855)', 'SCARA3 (51435)', 'SCARA5 (286133)', 'SCARB1 (949)', 'SCARB2 (950)', 'SCARF1 (8578)', 'SCARF2 (91179)', 'SCCPDH (51097)', 'SCD (6319)', 'SCD5 (79966)', 'SCEL (8796)', 'SCFD1 (23256)', 'SCFD2 (152579)', 'SCG2 (7857)', 'SCG3 (29106)', 'SCG5 (6447)', 'SCGB1A1 (7356)', 'SCGB1D1 (10648)', 'SCGB1D2 (10647)', 'SCGB1D4 (404552)', 'SCGB2A1 (4246)', 'SCGB2A2 (4250)', 'SCGB2B2 (284402)', 'SCGB3A1 (92304)', 'SCGB3A2 (117156)', 'SCGN (10590)', 'SCHIP1 (29970)', 'SCIMP (388325)', 'SCIN (85477)', 'SCLT1 (132320)', 'SCLY (51540)', 'SCMH1 (22955)', 'SCML1 (6322)', 'SCML2 (10389)', 'SCML4 (256380)', 'SCN10A (6336)', 'SCN11A (11280)', 'SCN1A (6323)', 'SCN1B (6324)', 'SCN2A (6326)', 'SCN2B (6327)', 'SCN3A (6328)', 'SCN3B (55800)', 'SCN4A (6329)', 'SCN4B (6330)', 'SCN5A (6331)', 'SCN7A (6332)', 'SCN8A (6334)', 'SCN9A (6335)', 'SCNM1 (79005)', 'SCNN1A (6337)', 'SCNN1B (6338)', 'SCNN1D (6339)', 'SCNN1G (6340)', 'SCO1 (6341)', 'SCO2 (9997)', 'SCOC (60592)', 'SCP2 (6342)', 'SCP2D1 (140856)', 'SCPEP1 (59342)', 'SCRG1 (11341)', 'SCRIB (23513)', 'SCRN1 (9805)', 'SCRN2 (90507)', 'SCRN3 (79634)', 'SCRT1 (83482)', 'SCRT2 (85508)', 'SCT (6343)', 'SCTR (6344)', 'SCUBE1 (80274)', 'SCUBE2 (57758)', 'SCUBE3 (222663)', 'SCX (642658)', 'SCYL1 (57410)', 'SCYL2 (55681)', 'SCYL3 (57147)', 'SDAD1 (55153)', 'SDC1 (6382)', 'SDC2 (6383)', 'SDC3 (9672)', 'SDC4 (6385)', 'SDCBP (6386)', 'SDCBP2 (27111)', 'SDCCAG8 (10806)', 'SDE2 (163859)', 'SDF2 (6388)', 'SDF2L1 (23753)', 'SDF4 (51150)', 'SDHA (6389)', 'SDHAF1 (644096)', 'SDHAF2 (54949)', 'SDHAF3 (57001)', 'SDHAF4 (135154)', 'SDHB (6390)', 'SDHC (6391)', 'SDHD (6392)', 'SDK1 (221935)', 'SDK2 (54549)', 'SDR16C5 (195814)', 'SDR39U1 (56948)', 'SDR42E1 (93517)', 'SDR9C7 (121214)', 'SDS (10993)', 'SDSL (113675)', 'SEBOX (645832)', 'SEC11A (23478)', 'SEC11C (90701)', 'SEC13 (6396)', 'SEC14L1 (6397)', 'SEC14L2 (23541)', 'SEC14L3 (266629)', 'SEC14L4 (284904)', 'SEC14L5 (9717)', 'SEC14L6 (730005)', 'SEC16A (9919)', 'SEC16B (89866)', 'SEC22A (26984)', 'SEC22C (9117)', 'SEC23A (10484)', 'SEC23B (10483)', 'SEC23IP (11196)', 'SEC24A (10802)', 'SEC24B (10427)', 'SEC24C (9632)', 'SEC24D (9871)', 'SEC31A (22872)', 'SEC31B (25956)', 'SEC61A1 (29927)', 'SEC61A2 (55176)', 'SEC61B (10952)', 'SEC61G (23480)', 'SEC62 (7095)', 'SEC63 (11231)', 'SECISBP2 (79048)', 'SECISBP2L (9728)', 'SECTM1 (6398)', 'SEH1L (81929)', 'SEL1L (6400)', 'SEL1L2 (80343)', 'SEL1L3 (23231)', 'SELE (6401)', 'SELENBP1 (8991)', 'SELENOH (280636)', 'SELENOI (85465)', 'SELENOK (58515)', 'SELENOM (140606)', 'SELENON (57190)', 'SELENOO (83642)', 'SELENOP (6414)', 'SELENOS (55829)', 'SELENOT (51714)', 'SELENOV (348303)', 'SELENOW (6415)', 'SELL (6402)', 'SELP (6403)', 'SELPLG (6404)', 'SEM1 (7979)', 'SEMA3A (10371)', 'SEMA3C (10512)', 'SEMA3D (223117)', 'SEMA3E (9723)', 'SEMA3F (6405)', 'SEMA3G (56920)', 'SEMA4A (64218)', 'SEMA4B (10509)', 'SEMA4C (54910)', 'SEMA4D (10507)', 'SEMA4F (10505)', 'SEMA4G (57715)', 'SEMA5A (9037)', 'SEMA5B (54437)', 'SEMA6A (57556)', 'SEMA6B (10501)', 'SEMA6C (10500)', 'SEMA6D (80031)', 'SEMA7A (8482)', 'SEMG1 (6406)', 'SEMG2 (6407)', 'SENP1 (29843)', 'SENP2 (59343)', 'SENP5 (205564)', 'SENP6 (26054)', 'SENP7 (57337)', 'SENP8 (123228)', 'SEPHS1 (22929)', 'SEPHS2 (22928)', 'SEPSECS (51091)', 'SEPTIN10 (151011)', 'SEPTIN11 (55752)', 'SEPTIN12 (124404)', 'SEPTIN14 (346288)', 'SEPTIN2 (4735)', 'SEPTIN3 (55964)', 'SEPTIN4 (5414)', 'SEPTIN5 (5413)', 'SEPTIN6 (23157)', 'SEPTIN8 (23176)', 'SEPTIN9 (10801)', 'SERAC1 (84947)', 'SERBP1 (26135)', 'SERF2 (10169)', 'SERGEF (26297)', 'SERHL2 (253190)', 'SERINC1 (57515)', 'SERINC2 (347735)', 'SERINC3 (10955)', 'SERINC4 (619189)', 'SERINC5 (256987)', 'SERP1 (27230)', 'SERP2 (387923)', 'SERPINA1 (5265)', 'SERPINA10 (51156)', 'SERPINA11 (256394)', 'SERPINA12 (145264)', 'SERPINA3 (12)', 'SERPINA4 (5267)', 'SERPINA5 (5104)', 'SERPINA6 (866)', 'SERPINA7 (6906)', 'SERPINA9 (327657)', 'SERPINB1 (1992)', 'SERPINB10 (5273)', 'SERPINB12 (89777)', 'SERPINB13 (5275)', 'SERPINB2 (5055)', 'SERPINB3 (6317)', 'SERPINB4 (6318)', 'SERPINB5 (5268)', 'SERPINB6 (5269)', 'SERPINB7 (8710)', 'SERPINB8 (5271)', 'SERPINB9 (5272)', 'SERPINC1 (462)', 'SERPIND1 (3053)', 'SERPINE1 (5054)', 'SERPINE2 (5270)', 'SERPINE3 (647174)', 'SERPINF1 (5176)', 'SERPINF2 (5345)', 'SERPING1 (710)', 'SERPINH1 (871)', 'SERPINI1 (5274)', 'SERPINI2 (5276)', 'SERTAD1 (29950)', 'SERTAD2 (9792)', 'SERTAD3 (29946)', 'SERTAD4 (56256)', 'SERTM1 (400120)', 'SESN1 (27244)', 'SESN2 (83667)', 'SESN3 (143686)', 'SESTD1 (91404)', 'SET (6418)', 'SETBP1 (26040)', 'SETD1A (9739)', 'SETD1B (23067)', 'SETD2 (29072)', 'SETD3 (84193)', 'SETD4 (54093)', 'SETD5 (55209)', 'SETD6 (79918)', 'SETD7 (80854)', 'SETD9 (133383)', 'SETDB1 (9869)', 'SETDB2 (83852)', 'SETMAR (6419)', 'SETX (23064)', 'SEZ6 (124925)', 'SEZ6L (23544)', 'SEZ6L2 (26470)', 'SF1 (7536)', 'SF3A1 (10291)', 'SF3A2 (8175)', 'SF3A3 (10946)', 'SF3B1 (23451)', 'SF3B2 (10992)', 'SF3B3 (23450)', 'SF3B4 (10262)', 'SF3B5 (83443)', 'SF3B6 (51639)', 'SFI1 (9814)', 'SFMBT1 (51460)', 'SFMBT2 (57713)', 'SFN (2810)', 'SFPQ (6421)', 'SFR1 (119392)', 'SFRP1 (6422)', 'SFRP2 (6423)', 'SFRP4 (6424)', 'SFRP5 (6425)', 'SFSWAP (6433)', 'SFT2D1 (113402)', 'SFT2D2 (375035)', 'SFT2D3 (84826)', 'SFTA2 (389376)', 'SFTPA1 (653509)', 'SFTPA2 (729238)', 'SFTPB (6439)', 'SFTPC (6440)', 'SFTPD (6441)', 'SFXN1 (94081)', 'SFXN2 (118980)', 'SFXN3 (81855)', 'SFXN4 (119559)', 'SFXN5 (94097)', 'SGCA (6442)', 'SGCB (6443)', 'SGCD (6444)', 'SGCE (8910)', 'SGCG (6445)', 'SGCZ (137868)', 'SGF29 (112869)', 'SGIP1 (84251)', 'SGK1 (6446)', 'SGK2 (10110)', 'SGK3 (23678)', 'SGMS1 (259230)', 'SGMS2 (166929)', 'SGO1 (151648)', 'SGO2 (151246)', 'SGPL1 (8879)', 'SGPP1 (81537)', 'SGPP2 (130367)', 'SGSH (6448)', 'SGSM1 (129049)', 'SGSM2 (9905)', 'SGSM3 (27352)', 'SGTA (6449)', 'SGTB (54557)', 'SH2B1 (25970)', 'SH2B3 (10019)', 'SH2D1A (4068)', 'SH2D1B (117157)', 'SH2D2A (9047)', 'SH2D3A (10045)', 'SH2D3C (10044)', 'SH2D4A (63898)', 'SH2D4B (387694)', 'SH2D5 (400745)', 'SH2D6 (284948)', 'SH2D7 (646892)', 'SH3BGR (6450)', 'SH3BGRL (6451)', 'SH3BGRL2 (83699)', 'SH3BGRL3 (83442)', 'SH3BP1 (23616)', 'SH3BP2 (6452)', 'SH3BP4 (23677)', 'SH3BP5 (9467)', 'SH3BP5L (80851)', 'SH3D19 (152503)', 'SH3D21 (79729)', 'SH3GL1 (6455)', 'SH3GL2 (6456)', 'SH3GL3 (6457)', 'SH3GLB1 (51100)', 'SH3GLB2 (56904)', 'SH3KBP1 (30011)', 'SH3PXD2A (9644)', 'SH3PXD2B (285590)', 'SH3RF1 (57630)', 'SH3RF2 (153769)', 'SH3TC1 (54436)', 'SH3TC2 (79628)', 'SH3YL1 (26751)', 'SHANK1 (50944)', 'SHARPIN (81858)', 'SHB (6461)', 'SHBG (6462)', 'SHC1 (6464)', 'SHC2 (25759)', 'SHC3 (53358)', 'SHC4 (399694)', 'SHCBP1 (79801)', 'SHCBP1L (81626)', 'SHD (56961)', 'SHE (126669)', 'SHF (90525)', 'SHFL (55337)', 'SHH (6469)', 'SHISA2 (387914)', 'SHISA3 (152573)', 'SHISA4 (149345)', 'SHISA5 (51246)', 'SHISA6 (388336)', 'SHISA7 (729956)', 'SHISA9 (729993)', 'SHISAL1 (85352)', 'SHISAL2A (348378)', 'SHISAL2B (100132916)', 'SHKBP1 (92799)', 'SHLD1 (149840)', 'SHLD2 (54537)', 'SHMT1 (6470)', 'SHMT2 (6472)', 'SHOC1 (158401)', 'SHOC2 (8036)', 'SHOX2 (6474)', 'SHPK (23729)', 'SHPRH (257218)', 'SHQ1 (55164)', 'SHROOM1 (134549)', 'SHROOM2 (357)', 'SHROOM3 (57619)', 'SHROOM4 (57477)', 'SHTN1 (57698)', 'SI (6476)', 'SIAE (54414)', 'SIAH1 (6477)', 'SIAH2 (6478)', 'SIAH3 (283514)', 'SIDT1 (54847)', 'SIDT2 (51092)', 'SIGIRR (59307)', 'SIGLEC1 (6614)', 'SIGLEC10 (89790)', 'SIGLEC11 (114132)', 'SIGLEC12 (89858)', 'SIGLEC15 (284266)', 'SIGLEC5 (8778)', 'SIGLEC6 (946)', 'SIGLEC7 (27036)', 'SIGLEC8 (27181)', 'SIGLEC9 (27180)', 'SIGLECL1 (284369)', 'SIGMAR1 (10280)', 'SIK1 (150094)', 'SIK2 (23235)', 'SIK3 (23387)', 'SIKE1 (80143)', 'SIL1 (64374)', 'SIM1 (6492)', 'SIM2 (6493)', 'SIMC1 (375484)', 'SIN3A (25942)', 'SIN3B (23309)', 'SINHCAF (58516)', 'SIPA1 (6494)', 'SIPA1L1 (26037)', 'SIPA1L2 (57568)', 'SIPA1L3 (23094)', 'SIRPA (140885)', 'SIRPB1 (10326)', 'SIRPB2 (284759)', 'SIRPD (128646)', 'SIRPG (55423)', 'SIRT1 (23411)', 'SIRT2 (22933)', 'SIRT3 (23410)', 'SIRT4 (23409)', 'SIRT5 (23408)', 'SIRT6 (51548)', 'SIRT7 (51547)', 'SIT1 (27240)', 'SIVA1 (10572)', 'SIX1 (6495)', 'SIX2 (10736)', 'SIX3 (6496)', 'SIX4 (51804)', 'SIX5 (147912)', 'SIX6 (4990)', 'SKA1 (220134)', 'SKA2 (348235)', 'SKA3 (221150)', 'SKAP1 (8631)', 'SKAP2 (8935)', 'SKI (6497)', 'SKIC2 (6499)', 'SKIC3 (9652)', 'SKIC8 (80349)', 'SKIDA1 (387640)', 'SKIL (6498)', 'SKOR1 (390598)', 'SKOR2 (652991)', 'SKP1 (6500)', 'SKP2 (6502)', 'SLA (6503)', 'SLA2 (84174)', 'SLAIN1 (122060)', 'SLAIN2 (57606)', 'SLAMF1 (6504)', 'SLAMF6 (114836)', 'SLAMF7 (57823)', 'SLAMF8 (56833)', 'SLAMF9 (89886)', 'SLBP (7884)', 'SLC10A1 (6554)', 'SLC10A2 (6555)', 'SLC10A3 (8273)', 'SLC10A4 (201780)', 'SLC10A5 (347051)', 'SLC10A6 (345274)', 'SLC10A7 (84068)', 'SLC11A1 (6556)', 'SLC11A2 (4891)', 'SLC12A1 (6557)', 'SLC12A2 (6558)', 'SLC12A3 (6559)', 'SLC12A4 (6560)', 'SLC12A5 (57468)', 'SLC12A6 (9990)', 'SLC12A7 (10723)', 'SLC12A8 (84561)', 'SLC12A9 (56996)', 'SLC13A1 (6561)', 'SLC13A2 (9058)', 'SLC13A3 (64849)', 'SLC13A4 (26266)', 'SLC13A5 (284111)', 'SLC14A1 (6563)', 'SLC14A2 (8170)', 'SLC15A1 (6564)', 'SLC15A2 (6565)', 'SLC15A3 (51296)', 'SLC15A4 (121260)', 'SLC16A1 (6566)', 'SLC16A10 (117247)', 'SLC16A11 (162515)', 'SLC16A12 (387700)', 'SLC16A13 (201232)', 'SLC16A14 (151473)', 'SLC16A2 (6567)', 'SLC16A3 (9123)', 'SLC16A4 (9122)', 'SLC16A5 (9121)', 'SLC16A6 (9120)', 'SLC16A7 (9194)', 'SLC16A8 (23539)', 'SLC16A9 (220963)', 'SLC17A1 (6568)', 'SLC17A2 (10246)', 'SLC17A3 (10786)', 'SLC17A4 (10050)', 'SLC17A5 (26503)', 'SLC17A6 (57084)', 'SLC17A7 (57030)', 'SLC17A8 (246213)', 'SLC17A9 (63910)', 'SLC18A1 (6570)', 'SLC18A2 (6571)', 'SLC18A3 (6572)', 'SLC18B1 (116843)', 'SLC19A1 (6573)', 'SLC19A2 (10560)', 'SLC19A3 (80704)', 'SLC1A1 (6505)', 'SLC1A2 (6506)', 'SLC1A3 (6507)', 'SLC1A4 (6509)', 'SLC1A5 (6510)', 'SLC1A6 (6511)', 'SLC1A7 (6512)', 'SLC20A1 (6574)', 'SLC20A2 (6575)', 'SLC22A1 (6580)', 'SLC22A10 (387775)', 'SLC22A11 (55867)', 'SLC22A12 (116085)', 'SLC22A13 (9390)', 'SLC22A14 (9389)', 'SLC22A15 (55356)', 'SLC22A16 (85413)', 'SLC22A17 (51310)', 'SLC22A18 (5002)', 'SLC22A2 (6582)', 'SLC22A23 (63027)', 'SLC22A25 (387601)', 'SLC22A3 (6581)', 'SLC22A4 (6583)', 'SLC22A5 (6584)', 'SLC22A6 (9356)', 'SLC22A7 (10864)', 'SLC22A8 (9376)', 'SLC22A9 (114571)', 'SLC23A1 (9963)', 'SLC23A2 (9962)', 'SLC23A3 (151295)', 'SLC24A1 (9187)', 'SLC24A2 (25769)', 'SLC24A3 (57419)', 'SLC24A4 (123041)', 'SLC24A5 (283652)', 'SLC25A1 (6576)', 'SLC25A10 (1468)', 'SLC25A11 (8402)', 'SLC25A12 (8604)', 'SLC25A13 (10165)', 'SLC25A14 (9016)', 'SLC25A15 (10166)', 'SLC25A16 (8034)', 'SLC25A17 (10478)', 'SLC25A18 (83733)', 'SLC25A19 (60386)', 'SLC25A2 (83884)', 'SLC25A20 (788)', 'SLC25A21 (89874)', 'SLC25A22 (79751)', 'SLC25A23 (79085)', 'SLC25A24 (29957)', 'SLC25A25 (114789)', 'SLC25A26 (115286)', 'SLC25A27 (9481)', 'SLC25A28 (81894)', 'SLC25A29 (123096)', 'SLC25A3 (5250)', 'SLC25A30 (253512)', 'SLC25A31 (83447)', 'SLC25A32 (81034)', 'SLC25A33 (84275)', 'SLC25A34 (284723)', 'SLC25A35 (399512)', 'SLC25A36 (55186)', 'SLC25A37 (51312)', 'SLC25A38 (54977)', 'SLC25A39 (51629)', 'SLC25A4 (291)', 'SLC25A40 (55972)', 'SLC25A41 (284427)', 'SLC25A42 (284439)', 'SLC25A43 (203427)', 'SLC25A44 (9673)', 'SLC25A45 (283130)', 'SLC25A46 (91137)', 'SLC25A47 (283600)', 'SLC25A48 (153328)', 'SLC25A5 (292)', 'SLC25A51 (92014)', 'SLC25A52 (147407)', 'SLC25A53 (401612)', 'SLC26A1 (10861)', 'SLC26A11 (284129)', 'SLC26A2 (1836)', 'SLC26A3 (1811)', 'SLC26A4 (5172)', 'SLC26A5 (375611)', 'SLC26A6 (65010)', 'SLC26A7 (115111)', 'SLC26A8 (116369)', 'SLC26A9 (115019)', 'SLC27A1 (376497)', 'SLC27A2 (11001)', 'SLC27A3 (11000)', 'SLC27A4 (10999)', 'SLC27A5 (10998)', 'SLC27A6 (28965)', 'SLC28A1 (9154)', 'SLC28A2 (9153)', 'SLC28A3 (64078)', 'SLC29A1 (2030)', 'SLC29A2 (3177)', 'SLC29A3 (55315)', 'SLC29A4 (222962)', 'SLC2A1 (6513)', 'SLC2A10 (81031)', 'SLC2A11 (66035)', 'SLC2A12 (154091)', 'SLC2A13 (114134)', 'SLC2A14 (144195)', 'SLC2A2 (6514)', 'SLC2A3 (6515)', 'SLC2A4 (6517)', 'SLC2A4RG (56731)', 'SLC2A5 (6518)', 'SLC2A6 (11182)', 'SLC2A7 (155184)', 'SLC2A8 (29988)', 'SLC2A9 (56606)', 'SLC30A1 (7779)', 'SLC30A10 (55532)', 'SLC30A2 (7780)', 'SLC30A3 (7781)', 'SLC30A4 (7782)', 'SLC30A5 (64924)', 'SLC30A6 (55676)', 'SLC30A7 (148867)', 'SLC30A8 (169026)', 'SLC30A9 (10463)', 'SLC31A1 (1317)', 'SLC31A2 (1318)', 'SLC32A1 (140679)', 'SLC33A1 (9197)', 'SLC34A1 (6569)', 'SLC34A2 (10568)', 'SLC34A3 (142680)', 'SLC35A1 (10559)', 'SLC35A2 (7355)', 'SLC35A3 (23443)', 'SLC35A4 (113829)', 'SLC35A5 (55032)', 'SLC35B1 (10237)', 'SLC35B2 (347734)', 'SLC35B3 (51000)', 'SLC35B4 (84912)', 'SLC35C1 (55343)', 'SLC35C2 (51006)', 'SLC35D1 (23169)', 'SLC35D2 (11046)', 'SLC35D3 (340146)', 'SLC35E1 (79939)', 'SLC35E2B (728661)', 'SLC35E3 (55508)', 'SLC35E4 (339665)', 'SLC35F1 (222553)', 'SLC35F2 (54733)', 'SLC35F3 (148641)', 'SLC35F5 (80255)', 'SLC35F6 (54978)', 'SLC35G1 (159371)', 'SLC35G2 (80723)', 'SLC35G3 (146861)', 'SLC35G5 (83650)', 'SLC35G6 (643664)', 'SLC36A1 (206358)', 'SLC36A2 (153201)', 'SLC36A3 (285641)', 'SLC36A4 (120103)', 'SLC37A1 (54020)', 'SLC37A2 (219855)', 'SLC37A3 (84255)', 'SLC38A1 (81539)', 'SLC38A10 (124565)', 'SLC38A11 (151258)', 'SLC38A2 (54407)', 'SLC38A4 (55089)', 'SLC38A5 (92745)', 'SLC38A6 (145389)', 'SLC38A7 (55238)', 'SLC38A8 (146167)', 'SLC38A9 (153129)', 'SLC39A1 (27173)', 'SLC39A10 (57181)', 'SLC39A11 (201266)', 'SLC39A12 (221074)', 'SLC39A13 (91252)', 'SLC39A14 (23516)', 'SLC39A2 (29986)', 'SLC39A3 (29985)', 'SLC39A4 (55630)', 'SLC39A5 (283375)', 'SLC39A6 (25800)', 'SLC39A7 (7922)', 'SLC39A8 (64116)', 'SLC39A9 (55334)', 'SLC3A1 (6519)', 'SLC3A2 (6520)', 'SLC40A1 (30061)', 'SLC41A1 (254428)', 'SLC41A2 (84102)', 'SLC41A3 (54946)', 'SLC43A1 (8501)', 'SLC43A2 (124935)', 'SLC43A3 (29015)', 'SLC44A1 (23446)', 'SLC44A2 (57153)', 'SLC44A3 (126969)', 'SLC44A4 (80736)', 'SLC44A5 (204962)', 'SLC45A1 (50651)', 'SLC45A2 (51151)', 'SLC45A3 (85414)', 'SLC45A4 (57210)', 'SLC46A2 (57864)', 'SLC46A3 (283537)', 'SLC47A1 (55244)', 'SLC47A2 (146802)', 'SLC48A1 (55652)', 'SLC49A3 (84179)', 'SLC49A4 (84925)', 'SLC4A1 (6521)', 'SLC4A10 (57282)', 'SLC4A11 (83959)', 'SLC4A1AP (22950)', 'SLC4A2 (6522)', 'SLC4A3 (6508)', 'SLC4A4 (8671)', 'SLC4A5 (57835)', 'SLC4A7 (9497)', 'SLC4A8 (9498)', 'SLC4A9 (83697)', 'SLC50A1 (55974)', 'SLC51A (200931)', 'SLC51B (123264)', 'SLC52A1 (55065)', 'SLC52A2 (79581)', 'SLC52A3 (113278)', 'SLC5A1 (6523)', 'SLC5A10 (125206)', 'SLC5A11 (115584)', 'SLC5A12 (159963)', 'SLC5A2 (6524)', 'SLC5A3 (6526)', 'SLC5A4 (6527)', 'SLC5A5 (6528)', 'SLC5A6 (8884)', 'SLC5A7 (60482)', 'SLC5A8 (160728)', 'SLC5A9 (200010)', 'SLC66A1 (54896)', 'SLC66A2 (80148)', 'SLC66A3 (130814)', 'SLC6A1 (6529)', 'SLC6A11 (6538)', 'SLC6A12 (6539)', 'SLC6A13 (6540)', 'SLC6A14 (11254)', 'SLC6A15 (55117)', 'SLC6A16 (28968)', 'SLC6A17 (388662)', 'SLC6A18 (348932)', 'SLC6A19 (340024)', 'SLC6A2 (6530)', 'SLC6A20 (54716)', 'SLC6A3 (6531)', 'SLC6A4 (6532)', 'SLC6A5 (9152)', 'SLC6A6 (6533)', 'SLC6A7 (6534)', 'SLC6A8 (6535)', 'SLC6A9 (6536)', 'SLC7A1 (6541)', 'SLC7A10 (56301)', 'SLC7A11 (23657)', 'SLC7A13 (157724)', 'SLC7A14 (57709)', 'SLC7A2 (6542)', 'SLC7A3 (84889)', 'SLC7A4 (6545)', 'SLC7A5 (8140)', 'SLC7A6 (9057)', 'SLC7A6OS (84138)', 'SLC7A7 (9056)', 'SLC7A8 (23428)', 'SLC7A9 (11136)', 'SLC8A1 (6546)', 'SLC8A2 (6543)', 'SLC8A3 (6547)', 'SLC8B1 (80024)', 'SLC9A1 (6548)', 'SLC9A2 (6549)', 'SLC9A3 (6550)', 'SLC9A4 (389015)', 'SLC9A5 (6553)', 'SLC9A6 (10479)', 'SLC9A7 (84679)', 'SLC9A8 (23315)', 'SLC9A9 (285195)', 'SLC9B1 (150159)', 'SLC9B2 (133308)', 'SLC9C1 (285335)', 'SLC9C2 (284525)', 'SLCO1A2 (6579)', 'SLCO1B1 (10599)', 'SLCO1B3 (28234)', 'SLCO1C1 (53919)', 'SLCO2A1 (6578)', 'SLCO2B1 (11309)', 'SLCO3A1 (28232)', 'SLCO4A1 (28231)', 'SLCO4C1 (353189)', 'SLCO5A1 (81796)', 'SLCO6A1 (133482)', 'SLF1 (84250)', 'SLF2 (55719)', 'SLFN11 (91607)', 'SLFN12 (55106)', 'SLFN12L (100506736)', 'SLFN13 (146857)', 'SLFN14 (342618)', 'SLFN5 (162394)', 'SLFNL1 (200172)', 'SLIRP (81892)', 'SLIT1 (6585)', 'SLIT2 (9353)', 'SLIT3 (6586)', 'SLITRK1 (114798)', 'SLITRK2 (84631)', 'SLITRK3 (22865)', 'SLITRK4 (139065)', 'SLITRK5 (26050)', 'SLITRK6 (84189)', 'SLK (9748)', 'SLMAP (7871)', 'SLN (6588)', 'SLPI (6590)', 'SLTM (79811)', 'SLU7 (10569)', 'SLURP1 (57152)', 'SLURP2 (432355)', 'SLX4 (84464)', 'SLX4IP (128710)', 'SLX9 (85395)', 'SMAD1 (4086)', 'SMAD2 (4087)', 'SMAD3 (4088)', 'SMAD4 (4089)', 'SMAD6 (4091)', 'SMAD7 (4092)', 'SMAD9 (4093)', 'SMAGP (57228)', 'SMAP1 (60682)', 'SMAP2 (64744)', 'SMARCA1 (6594)', 'SMARCA2 (6595)', 'SMARCA4 (6597)', 'SMARCA5 (8467)', 'SMARCAD1 (56916)', 'SMARCAL1 (50485)', 'SMARCB1 (6598)', 'SMARCC1 (6599)', 'SMARCC2 (6601)', 'SMARCD1 (6602)', 'SMARCD2 (6603)', 'SMARCD3 (6604)', 'SMARCE1 (6605)', 'SMC1A (8243)', 'SMC1B (27127)', 'SMC2 (10592)', 'SMC3 (9126)', 'SMC4 (10051)', 'SMC5 (23137)', 'SMC6 (79677)', 'SMCHD1 (23347)', 'SMCO1 (255798)', 'SMCO2 (341346)', 'SMCO3 (440087)', 'SMCO4 (56935)', 'SMCP (4184)', 'SMCR8 (140775)', 'SMDT1 (91689)', 'SMG1 (23049)', 'SMG5 (23381)', 'SMG6 (23293)', 'SMG7 (9887)', 'SMG8 (55181)', 'SMG9 (56006)', 'SMIM1 (388588)', 'SMIM10 (644538)', 'SMIM11 (54065)', 'SMIM12 (113444)', 'SMIM13 (221710)', 'SMIM14 (201895)', 'SMIM15 (643155)', 'SMIM17 (147670)', 'SMIM18 (100507341)', 'SMIM19 (114926)', 'SMIM2 (79024)', 'SMIM20 (389203)', 'SMIM21 (284274)', 'SMIM22 (440335)', 'SMIM24 (284422)', 'SMIM29 (221491)', 'SMIM3 (85027)', 'SMIM5 (643008)', 'SMIM6 (100130933)', 'SMIM7 (79086)', 'SMIM8 (57150)', 'SMIM9 (100132963)', 'SMKR1 (100287482)', 'SMLR1 (100507203)', 'SMNDC1 (10285)', 'SMO (6608)', 'SMOC1 (64093)', 'SMOC2 (64094)', 'SMOX (54498)', 'SMPD1 (6609)', 'SMPD2 (6610)', 'SMPD3 (55512)', 'SMPD4 (55627)', 'SMPDL3A (10924)', 'SMPDL3B (27293)', 'SMPX (23676)', 'SMR3A (26952)', 'SMR3B (10879)', 'SMS (6611)', 'SMTN (6525)', 'SMTNL2 (342527)', 'SMU1 (55234)', 'SMUG1 (23583)', 'SMURF1 (57154)', 'SMURF2 (64750)', 'SMYD1 (150572)', 'SMYD2 (56950)', 'SMYD3 (64754)', 'SMYD4 (114826)', 'SMYD5 (10322)', 'SNAI1 (6615)', 'SNAI2 (6591)', 'SNAI3 (333929)', 'SNAP23 (8773)', 'SNAP25 (6616)', 'SNAP29 (9342)', 'SNAP47 (116841)', 'SNAP91 (9892)', 'SNAPC1 (6617)', 'SNAPC2 (6618)', 'SNAPC3 (6619)', 'SNAPC4 (6621)', 'SNAPC5 (10302)', 'SNAPIN (23557)', 'SNCA (6622)', 'SNCAIP (9627)', 'SNCB (6620)', 'SNCG (6623)', 'SND1 (27044)', 'SNED1 (25992)', 'SNF8 (11267)', 'SNIP1 (79753)', 'SNN (8303)', 'SNORC (389084)', 'SNPH (9751)', 'SNRK (54861)', 'SNRNP200 (23020)', 'SNRNP25 (79622)', 'SNRNP27 (11017)', 'SNRNP35 (11066)', 'SNRNP40 (9410)', 'SNRNP48 (154007)', 'SNRNP70 (6625)', 'SNRPA (6626)', 'SNRPA1 (6627)', 'SNRPB (6628)', 'SNRPB2 (6629)', 'SNRPC (6631)', 'SNRPD1 (6632)', 'SNRPD2 (6633)', 'SNRPD3 (6634)', 'SNRPE (6635)', 'SNRPF (6636)', 'SNRPG (6637)', 'SNRPN (6638)', 'SNTA1 (6640)', 'SNTB1 (6641)', 'SNTB2 (6645)', 'SNTG1 (54212)', 'SNTG2 (54221)', 'SNTN (132203)', 'SNU13 (4809)', 'SNUPN (10073)', 'SNURF (8926)', 'SNW1 (22938)', 'SNX1 (6642)', 'SNX10 (29887)', 'SNX11 (29916)', 'SNX12 (29934)', 'SNX13 (23161)', 'SNX14 (57231)', 'SNX15 (29907)', 'SNX16 (64089)', 'SNX17 (9784)', 'SNX18 (112574)', 'SNX19 (399979)', 'SNX2 (6643)', 'SNX20 (124460)', 'SNX21 (90203)', 'SNX22 (79856)', 'SNX24 (28966)', 'SNX25 (83891)', 'SNX27 (81609)', 'SNX29 (92017)', 'SNX3 (8724)', 'SNX30 (401548)', 'SNX31 (169166)', 'SNX32 (254122)', 'SNX33 (257364)', 'SNX4 (8723)', 'SNX5 (27131)', 'SNX6 (58533)', 'SNX7 (51375)', 'SNX8 (29886)', 'SNX9 (51429)', 'SOAT1 (6646)', 'SOAT2 (8435)', 'SOBP (55084)', 'SOCS1 (8651)', 'SOCS2 (8835)', 'SOCS3 (9021)', 'SOCS4 (122809)', 'SOCS5 (9655)', 'SOCS6 (9306)', 'SOCS7 (30837)', 'SOD1 (6647)', 'SOD2 (6648)', 'SOD3 (6649)', 'SOHLH1 (402381)', 'SOHLH2 (54937)', 'SON (6651)', 'SORBS1 (10580)', 'SORBS2 (8470)', 'SORBS3 (10174)', 'SORCS1 (114815)', 'SORCS2 (57537)', 'SORCS3 (22986)', 'SORD (6652)', 'SORL1 (6653)', 'SORT1 (6272)', 'SOS1 (6654)', 'SOS2 (6655)', 'SOST (50964)', 'SOSTDC1 (25928)', 'SOWAHA (134548)', 'SOWAHB (345079)', 'SOWAHC (65124)', 'SOWAHD (347454)', 'SOX1 (6656)', 'SOX10 (6663)', 'SOX11 (6664)', 'SOX12 (6666)', 'SOX13 (9580)', 'SOX14 (8403)', 'SOX15 (6665)', 'SOX17 (64321)', 'SOX18 (54345)', 'SOX2 (6657)', 'SOX21 (11166)', 'SOX3 (6658)', 'SOX30 (11063)', 'SOX4 (6659)', 'SOX5 (6660)', 'SOX6 (55553)', 'SOX7 (83595)', 'SOX8 (30812)', 'SOX9 (6662)', 'SP1 (6667)', 'SP100 (6672)', 'SP110 (3431)', 'SP140 (11262)', 'SP140L (93349)', 'SP2 (6668)', 'SP3 (6670)', 'SP4 (6671)', 'SP5 (389058)', 'SP6 (80320)', 'SP7 (121340)', 'SP8 (221833)', 'SP9 (100131390)', 'SPA17 (53340)', 'SPACA1 (81833)', 'SPACA3 (124912)', 'SPACA4 (171169)', 'SPACA7 (122258)', 'SPACA9 (11092)', 'SPACDR (402573)', 'SPAG1 (6674)', 'SPAG16 (79582)', 'SPAG17 (200162)', 'SPAG4 (6676)', 'SPAG5 (10615)', 'SPAG6 (9576)', 'SPAG7 (9552)', 'SPAG8 (26206)', 'SPAG9 (9043)', 'SPAM1 (6677)', 'SPANXB1 (728695)', 'SPANXN1 (494118)', 'SPANXN2 (494119)', 'SPANXN3 (139067)', 'SPANXN4 (441525)', 'SPANXN5 (494197)', 'SPARC (6678)', 'SPARCL1 (8404)', 'SPART (23111)', 'SPAST (6683)', 'SPATA12 (353324)', 'SPATA13 (221178)', 'SPATA16 (83893)', 'SPATA17 (128153)', 'SPATA18 (132671)', 'SPATA19 (219938)', 'SPATA2 (9825)', 'SPATA20 (64847)', 'SPATA21 (374955)', 'SPATA22 (84690)', 'SPATA24 (202051)', 'SPATA25 (128497)', 'SPATA2L (124044)', 'SPATA3 (130560)', 'SPATA31A1 (647060)', 'SPATA31A3 (727830)', 'SPATA31A6 (389730)', 'SPATA31A7 (26165)', 'SPATA31D1 (389763)', 'SPATA31E1 (286234)', 'SPATA31F1 (259308)', 'SPATA31G1 (138724)', 'SPATA32 (124783)', 'SPATA33 (124045)', 'SPATA4 (132851)', 'SPATA45 (149643)', 'SPATA46 (284680)', 'SPATA6 (54558)', 'SPATA6L (55064)', 'SPATA7 (55812)', 'SPATA9 (83890)', 'SPATC1 (375686)', 'SPATC1L (84221)', 'SPATS1 (221409)', 'SPATS2 (65244)', 'SPATS2L (26010)', 'SPC24 (147841)', 'SPC25 (57405)', 'SPCS1 (28972)', 'SPCS2 (9789)', 'SPCS3 (60559)', 'SPDEF (25803)', 'SPDL1 (54908)', 'SPDYA (245711)', 'SPDYE1 (285955)', 'SPDYE4 (388333)', 'SPECC1 (92521)', 'SPECC1L (23384)', 'SPEF1 (25876)', 'SPEF2 (79925)', 'SPEG (10290)', 'SPEM1 (374768)', 'SPEM2 (201243)', 'SPEN (23013)', 'SPESP1 (246777)', 'SPG11 (80208)', 'SPG21 (51324)', 'SPG7 (6687)', 'SPHK1 (8877)', 'SPHK2 (56848)', 'SPHKAP (80309)', 'SPI1 (6688)', 'SPIB (6689)', 'SPIC (121599)', 'SPICE1 (152185)', 'SPIDR (23514)', 'SPIN1 (10927)', 'SPIN3 (169981)', 'SPIN4 (139886)', 'SPINDOC (144097)', 'SPINK1 (6690)', 'SPINK13 (153218)', 'SPINK14 (408187)', 'SPINK2 (6691)', 'SPINK4 (27290)', 'SPINK5 (11005)', 'SPINK6 (404203)', 'SPINK7 (84651)', 'SPINK8 (646424)', 'SPINK9 (643394)', 'SPINT1 (6692)', 'SPINT2 (10653)', 'SPINT3 (10816)', 'SPINT4 (391253)', 'SPIRE1 (56907)', 'SPIRE2 (84501)', 'SPMAP1 (388381)', 'SPMAP2 (51298)', 'SPMAP2L (100506564)', 'SPMIP10 (389320)', 'SPMIP2 (152940)', 'SPMIP3 (200159)', 'SPMIP4 (136895)', 'SPMIP5 (143379)', 'SPMIP6 (84688)', 'SPMIP7 (100130988)', 'SPMIP8 (374739)', 'SPMIP9 (200523)', 'SPN (6693)', 'SPNS1 (83985)', 'SPNS2 (124976)', 'SPNS3 (201305)', 'SPO11 (23626)', 'SPOCD1 (90853)', 'SPOCK1 (6695)', 'SPOCK2 (9806)', 'SPOCK3 (50859)', 'SPOP (8405)', 'SPOPL (339745)', 'SPOUT1 (51490)', 'SPP1 (6696)', 'SPP2 (6694)', 'SPPL2A (84888)', 'SPPL2C (162540)', 'SPPL3 (121665)', 'SPR (6697)', 'SPRED1 (161742)', 'SPRED2 (200734)', 'SPRED3 (399473)', 'SPRING1 (79794)', 'SPRN (503542)', 'SPRR1A (6698)', 'SPRR1B (6699)', 'SPRR2F (6705)', 'SPRR2G (6706)', 'SPRR3 (6707)', 'SPRR4 (163778)', 'SPRTN (83932)', 'SPRY1 (10252)', 'SPRY2 (10253)', 'SPRY3 (10251)', 'SPRY4 (81848)', 'SPRYD3 (84926)', 'SPRYD4 (283377)', 'SPRYD7 (57213)', 'SPSB1 (80176)', 'SPSB2 (84727)', 'SPSB3 (90864)', 'SPSB4 (92369)', 'SPTA1 (6708)', 'SPTAN1 (6709)', 'SPTB (6710)', 'SPTBN1 (6711)', 'SPTBN2 (6712)', 'SPTBN4 (57731)', 'SPTBN5 (51332)', 'SPTLC1 (10558)', 'SPTLC2 (9517)', 'SPTLC3 (55304)', 'SPTSSA (171546)', 'SPTSSB (165679)', 'SPTY2D1 (144108)', 'SPX (80763)', 'SPZ1 (84654)', 'SQLE (6713)', 'SQOR (58472)', 'SQSTM1 (8878)', 'SRA1 (10011)', 'SRARP (149563)', 'SRBD1 (55133)', 'SRC (6714)', 'SRCAP (10847)', 'SRCIN1 (80725)', 'SRD5A1 (6715)', 'SRD5A3 (79644)', 'SREBF1 (6720)', 'SREBF2 (6721)', 'SREK1 (140890)', 'SREK1IP1 (285672)', 'SRF (6722)', 'SRFBP1 (153443)', 'SRGAP1 (57522)', 'SRGAP3 (9901)', 'SRGN (5552)', 'SRI (6717)', 'SRL (6345)', 'SRM (6723)', 'SRMS (6725)', 'SRP14 (6727)', 'SRP19 (6728)', 'SRP54 (6729)', 'SRP68 (6730)', 'SRP72 (6731)', 'SRP9 (6726)', 'SRPK1 (6732)', 'SRPK2 (6733)', 'SRPK3 (26576)', 'SRPRA (6734)', 'SRPRB (58477)', 'SRPX (8406)', 'SRPX2 (27286)', 'SRR (63826)', 'SRRD (402055)', 'SRRM1 (10250)', 'SRRM2 (23524)', 'SRRM4 (84530)', 'SRRM5 (100170229)', 'SRRT (51593)', 'SRSF1 (6426)', 'SRSF10 (10772)', 'SRSF11 (9295)', 'SRSF12 (135295)', 'SRSF2 (6427)', 'SRSF3 (6428)', 'SRSF4 (6429)', 'SRSF5 (6430)', 'SRSF6 (6431)', 'SRSF7 (6432)', 'SRSF9 (8683)', 'SRXN1 (140809)', 'SRY (6736)', 'SS18 (6760)', 'SS18L1 (26039)', 'SS18L2 (51188)', 'SSB (6741)', 'SSBP1 (6742)', 'SSBP2 (23635)', 'SSBP3 (23648)', 'SSBP4 (170463)', 'SSC4D (136853)', 'SSC5D (284297)', 'SSH1 (54434)', 'SSH2 (85464)', 'SSH3 (54961)', 'SSMEM1 (136263)', 'SSNA1 (8636)', 'SSPN (8082)', 'SSR1 (6745)', 'SSR2 (6746)', 'SSR3 (6747)', 'SSR4 (6748)', 'SSRP1 (6749)', 'SST (6750)', 'SSTR1 (6751)', 'SSTR2 (6752)', 'SSTR3 (6753)', 'SSTR4 (6754)', 'SSTR5 (6755)', 'SSU72 (29101)', 'SSUH2 (51066)', 'SSX1 (6756)', 'SSX2IP (117178)', 'SSX3 (10214)', 'SSX5 (6758)', 'SSX7 (280658)', 'ST13 (6767)', 'ST14 (6768)', 'ST18 (9705)', 'ST3GAL1 (6482)', 'ST3GAL2 (6483)', 'ST3GAL3 (6487)', 'ST3GAL4 (6484)', 'ST3GAL5 (8869)', 'ST3GAL6 (10402)', 'ST6GAL1 (6480)', 'ST6GAL2 (84620)', 'ST6GALNAC1 (55808)', 'ST6GALNAC2 (10610)', 'ST6GALNAC3 (256435)', 'ST6GALNAC4 (27090)', 'ST6GALNAC5 (81849)', 'ST6GALNAC6 (30815)', 'ST7 (7982)', 'ST7L (54879)', 'ST8SIA1 (6489)', 'ST8SIA2 (8128)', 'ST8SIA3 (51046)', 'ST8SIA4 (7903)', 'ST8SIA5 (29906)', 'ST8SIA6 (338596)', 'STAB1 (23166)', 'STAB2 (55576)', 'STAC (6769)', 'STAC2 (342667)', 'STAC3 (246329)', 'STAG1 (10274)', 'STAG2 (10735)', 'STAG3 (10734)', 'STAM (8027)', 'STAM2 (10254)', 'STAMBP (10617)', 'STAMBPL1 (57559)', 'STAP1 (26228)', 'STAP2 (55620)', 'STAR (6770)', 'STARD10 (10809)', 'STARD13 (90627)', 'STARD3 (10948)', 'STARD3NL (83930)', 'STARD4 (134429)', 'STARD5 (80765)', 'STARD6 (147323)', 'STARD7 (56910)', 'STARD8 (9754)', 'STARD9 (57519)', 'STAT1 (6772)', 'STAT2 (6773)', 'STAT3 (6774)', 'STAT4 (6775)', 'STAT5A (6776)', 'STAT5B (6777)', 'STAT6 (6778)', 'STATH (6779)', 'STAU1 (6780)', 'STAU2 (27067)', 'STBD1 (8987)', 'STC1 (6781)', 'STC2 (8614)', 'STEAP1 (26872)', 'STEAP1B (256227)', 'STEAP2 (261729)', 'STEAP3 (55240)', 'STEAP4 (79689)', 'STEEP1 (63932)', 'STH (246744)', 'STIL (6491)', 'STIM1 (6786)', 'STIM2 (57620)', 'STIMATE (375346)', 'STING1 (340061)', 'STIP1 (10963)', 'STK10 (6793)', 'STK11 (6794)', 'STK11IP (114790)', 'STK16 (8576)', 'STK17A (9263)', 'STK17B (9262)', 'STK19 (8859)', 'STK24 (8428)', 'STK25 (10494)', 'STK26 (51765)', 'STK3 (6788)', 'STK31 (56164)', 'STK32A (202374)', 'STK32B (55351)', 'STK32C (282974)', 'STK33 (65975)', 'STK35 (140901)', 'STK36 (27148)', 'STK38 (11329)', 'STK38L (23012)', 'STK39 (27347)', 'STK4 (6789)', 'STK40 (83931)', 'STKLD1 (169436)', 'STMN1 (3925)', 'STMN2 (11075)', 'STMN3 (50861)', 'STMN4 (81551)', 'STMND1 (401236)', 'STMP1 (647087)', 'STN1 (79991)', 'STOM (2040)', 'STOML1 (9399)', 'STOML2 (30968)', 'STOML3 (161003)', 'STON1 (11037)', 'STON2 (85439)', 'STOX1 (219736)', 'STOX2 (56977)', 'STPG1 (90529)', 'STPG2 (285555)', 'STPG3 (441476)', 'STPG4 (285051)', 'STRA6 (64220)', 'STRA8 (346673)', 'STRADA (92335)', 'STRADB (55437)', 'STRAP (11171)', 'STRBP (55342)', 'STRC (161497)', 'STRIP1 (85369)', 'STRIP2 (57464)', 'STRN (6801)', 'STRN3 (29966)', 'STRN4 (29888)', 'STS (412)', 'STT3A (3703)', 'STT3B (201595)', 'STUB1 (10273)', 'STUM (375057)', 'STX10 (8677)', 'STX11 (8676)', 'STX12 (23673)', 'STX16 (8675)', 'STX17 (55014)', 'STX18 (53407)', 'STX19 (415117)', 'STX1A (6804)', 'STX1B (112755)', 'STX2 (2054)', 'STX3 (6809)', 'STX4 (6810)', 'STX5 (6811)', 'STX6 (10228)', 'STX7 (8417)', 'STX8 (9482)', 'STXBP1 (6812)', 'STXBP2 (6813)', 'STXBP3 (6814)', 'STXBP4 (252983)', 'STXBP5 (134957)', 'STXBP5L (9515)', 'STXBP6 (29091)', 'STYK1 (55359)', 'STYX (6815)', 'STYXL1 (51657)', 'STYXL2 (92235)', 'SUB1 (10923)', 'SUCLA2 (8803)', 'SUCLG1 (8802)', 'SUCLG2 (8801)', 'SUCNR1 (56670)', 'SUCO (51430)', 'SUDS3 (64426)', 'SUFU (51684)', 'SUGCT (79783)', 'SUGP1 (57794)', 'SUGP2 (10147)', 'SUGT1 (10910)', 'SULF1 (23213)', 'SULF2 (55959)', 'SULT1A1 (6817)', 'SULT1A2 (6799)', 'SULT1B1 (27284)', 'SULT1C2 (6819)', 'SULT1C3 (442038)', 'SULT1C4 (27233)', 'SULT1E1 (6783)', 'SULT2A1 (6822)', 'SULT2B1 (6820)', 'SULT4A1 (25830)', 'SULT6B1 (391365)', 'SUMF1 (285362)', 'SUMF2 (25870)', 'SUMO1 (7341)', 'SUMO3 (6612)', 'SUMO4 (387082)', 'SUN1 (23353)', 'SUN2 (25777)', 'SUN3 (256979)', 'SUN5 (140732)', 'SUOX (6821)', 'SUPT16H (11198)', 'SUPT20H (55578)', 'SUPT3H (8464)', 'SUPT4H1 (6827)', 'SUPT5H (6829)', 'SUPT6H (6830)', 'SUPT7L (9913)', 'SUPV3L1 (6832)', 'SURF1 (6834)', 'SURF2 (6835)', 'SURF4 (6836)', 'SURF6 (6838)', 'SUSD1 (64420)', 'SUSD2 (56241)', 'SUSD3 (203328)', 'SUSD4 (55061)', 'SUSD5 (26032)', 'SUSD6 (9766)', 'SUV39H1 (6839)', 'SUV39H2 (79723)', 'SUZ12 (23512)', 'SV2A (9900)', 'SV2B (9899)', 'SV2C (22987)', 'SVBP (374969)', 'SVEP1 (79987)', 'SVIL (6840)', 'SVIP (258010)', 'SVOPL (136306)', 'SWAP70 (23075)', 'SWI5 (375757)', 'SWSAP1 (126074)', 'SWT1 (54823)', 'SYAP1 (94056)', 'SYBU (55638)', 'SYCE1 (93426)', 'SYCE1L (100130958)', 'SYCE2 (256126)', 'SYCE3 (644186)', 'SYCN (342898)', 'SYCP1 (6847)', 'SYCP2 (10388)', 'SYCP2L (221711)', 'SYCP3 (50511)', 'SYDE1 (85360)', 'SYDE2 (84144)', 'SYF2 (25949)', 'SYK (6850)', 'SYMPK (8189)', 'SYN1 (6853)', 'SYN3 (8224)', 'SYNC (81493)', 'SYNCRIP (10492)', 'SYNDIG1 (79953)', 'SYNDIG1L (646658)', 'SYNE1 (23345)', 'SYNE2 (23224)', 'SYNE3 (161176)', 'SYNE4 (163183)', 'SYNGAP1 (8831)', 'SYNGR1 (9145)', 'SYNGR2 (9144)', 'SYNGR3 (9143)', 'SYNGR4 (23546)', 'SYNJ1 (8867)', 'SYNJ2 (8871)', 'SYNJ2BP (55333)', 'SYNPO (11346)', 'SYNPO2 (171024)', 'SYNPO2L (79933)', 'SYNPR (132204)', 'SYNRG (11276)', 'SYP (6855)', 'SYPL1 (6856)', 'SYPL2 (284612)', 'SYS1 (90196)', 'SYT1 (6857)', 'SYT10 (341359)', 'SYT11 (23208)', 'SYT12 (91683)', 'SYT13 (57586)', 'SYT14 (255928)', 'SYT16 (83851)', 'SYT17 (51760)', 'SYT2 (127833)', 'SYT3 (84258)', 'SYT4 (6860)', 'SYT5 (6861)', 'SYT6 (148281)', 'SYT7 (9066)', 'SYT8 (90019)', 'SYT9 (143425)', 'SYTL1 (84958)', 'SYTL2 (54843)', 'SYTL3 (94120)', 'SYTL4 (94121)', 'SYTL5 (94122)', 'SYVN1 (84447)', 'SZRD1 (26099)', 'SZT2 (23334)', 'TAAR1 (134864)', 'TAAR2 (9287)', 'TAAR5 (9038)', 'TAAR6 (319100)', 'TAAR8 (83551)', 'TAB1 (10454)', 'TAB2 (23118)', 'TAB3 (257397)', 'TAC1 (6863)', 'TAC3 (6866)', 'TAC4 (255061)', 'TACC1 (6867)', 'TACC2 (10579)', 'TACC3 (10460)', 'TACO1 (51204)', 'TACR1 (6869)', 'TACR2 (6865)', 'TACR3 (6870)', 'TACSTD2 (4070)', 'TADA1 (117143)', 'TADA2A (6871)', 'TADA2B (93624)', 'TADA3 (10474)', 'TAF1 (6872)', 'TAF10 (6881)', 'TAF11 (6882)', 'TAF12 (6883)', 'TAF13 (6884)', 'TAF15 (8148)', 'TAF1A (9015)', 'TAF1B (9014)', 'TAF1C (9013)', 'TAF1D (79101)', 'TAF1L (138474)', 'TAF2 (6873)', 'TAF3 (83860)', 'TAF4 (6874)', 'TAF4B (6875)', 'TAF5 (6877)', 'TAF5L (27097)', 'TAF6 (6878)', 'TAF6L (10629)', 'TAF7 (6879)', 'TAF7L (54457)', 'TAF8 (129685)', 'TAF9 (6880)', 'TAF9B (51616)', 'TAFA1 (407738)', 'TAFA2 (338811)', 'TAFA3 (284467)', 'TAFA4 (151647)', 'TAFA5 (25817)', 'TAFAZZIN (6901)', 'TAGAP (117289)', 'TAGLN (6876)', 'TAGLN2 (8407)', 'TAGLN3 (29114)', 'TAL1 (6886)', 'TAL2 (6887)', 'TALDO1 (6888)', 'TAMALIN (160622)', 'TAMM41 (132001)', 'TANC1 (85461)', 'TANC2 (26115)', 'TANGO2 (128989)', 'TANGO6 (79613)', 'TANK (10010)', 'TAOK1 (57551)', 'TAOK2 (9344)', 'TAOK3 (51347)', 'TAP1 (6890)', 'TAP2 (6891)', 'TAPBP (6892)', 'TAPBPL (55080)', 'TAPT1 (202018)', 'TARBP1 (6894)', 'TARBP2 (6895)', 'TARDBP (23435)', 'TARM1 (441864)', 'TARS1 (6897)', 'TARS2 (80222)', 'TARS3 (123283)', 'TAS1R1 (80835)', 'TAS1R2 (80834)', 'TAS1R3 (83756)', 'TAS2R1 (50834)', 'TAS2R10 (50839)', 'TAS2R13 (50838)', 'TAS2R14 (50840)', 'TAS2R16 (50833)', 'TAS2R19 (259294)', 'TAS2R20 (259295)', 'TAS2R3 (50831)', 'TAS2R30 (259293)', 'TAS2R31 (259290)', 'TAS2R38 (5726)', 'TAS2R39 (259285)', 'TAS2R4 (50832)', 'TAS2R40 (259286)', 'TAS2R41 (259287)', 'TAS2R42 (353164)', 'TAS2R43 (259289)', 'TAS2R46 (259292)', 'TAS2R5 (54429)', 'TAS2R50 (259296)', 'TAS2R60 (338398)', 'TAS2R7 (50837)', 'TAS2R8 (50836)', 'TAS2R9 (50835)', 'TASL (80231)', 'TASOR (23272)', 'TASOR2 (54906)', 'TASP1 (55617)', 'TAT (6898)', 'TATDN1 (83940)', 'TATDN2 (9797)', 'TATDN3 (128387)', 'TAX1BP1 (8887)', 'TAX1BP3 (30851)', 'TBATA (219793)', 'TBC1D1 (23216)', 'TBC1D10A (83874)', 'TBC1D10B (26000)', 'TBC1D10C (374403)', 'TBC1D12 (23232)', 'TBC1D13 (54662)', 'TBC1D14 (57533)', 'TBC1D15 (64786)', 'TBC1D16 (125058)', 'TBC1D17 (79735)', 'TBC1D19 (55296)', 'TBC1D2 (55357)', 'TBC1D20 (128637)', 'TBC1D21 (161514)', 'TBC1D22A (25771)', 'TBC1D22B (55633)', 'TBC1D23 (55773)', 'TBC1D24 (57465)', 'TBC1D25 (4943)', 'TBC1D26 (353149)', 'TBC1D28 (254272)', 'TBC1D2B (23102)', 'TBC1D30 (23329)', 'TBC1D31 (93594)', 'TBC1D32 (221322)', 'TBC1D4 (9882)', 'TBC1D5 (9779)', 'TBC1D7 (51256)', 'TBC1D8 (11138)', 'TBC1D8B (54885)', 'TBC1D9 (23158)', 'TBC1D9B (23061)', 'TBCA (6902)', 'TBCB (1155)', 'TBCC (6903)', 'TBCCD1 (55171)', 'TBCD (6904)', 'TBCE (6905)', 'TBCEL (219899)', 'TBCK (93627)', 'TBK1 (29110)', 'TBKBP1 (9755)', 'TBL1X (6907)', 'TBL1XR1 (79718)', 'TBL1Y (90665)', 'TBL2 (26608)', 'TBL3 (10607)', 'TBP (6908)', 'TBPL1 (9519)', 'TBR1 (10716)', 'TBRG1 (84897)', 'TBRG4 (9238)', 'TBX1 (6899)', 'TBX10 (347853)', 'TBX15 (6913)', 'TBX18 (9096)', 'TBX19 (9095)', 'TBX2 (6909)', 'TBX20 (57057)', 'TBX21 (30009)', 'TBX22 (50945)', 'TBX3 (6926)', 'TBX4 (9496)', 'TBX5 (6910)', 'TBX6 (6911)', 'TBXA2R (6915)', 'TBXAS1 (6916)', 'TBXT (6862)', 'TC2N (123036)', 'TCAF1 (9747)', 'TCAIM (285343)', 'TCAP (8557)', 'TCEA1 (6917)', 'TCEA2 (6919)', 'TCEA3 (6920)', 'TCEAL1 (9338)', 'TCEAL2 (140597)', 'TCEAL3 (85012)', 'TCEAL4 (79921)', 'TCEAL5 (340543)', 'TCEAL7 (56849)', 'TCEAL8 (90843)', 'TCEAL9 (51186)', 'TCEANC (170082)', 'TCEANC2 (127428)', 'TCERG1 (10915)', 'TCERG1L (256536)', 'TCF12 (6938)', 'TCF15 (6939)', 'TCF19 (6941)', 'TCF20 (6942)', 'TCF21 (6943)', 'TCF23 (150921)', 'TCF24 (100129654)', 'TCF25 (22980)', 'TCF3 (6929)', 'TCF4 (6925)', 'TCF7 (6932)', 'TCF7L1 (83439)', 'TCF7L2 (6934)', 'TCFL5 (10732)', 'TCHH (7062)', 'TCHHL1 (126637)', 'TCHP (84260)', 'TCIM (56892)', 'TCIRG1 (10312)', 'TCL1A (8115)', 'TCL1B (9623)', 'TCN1 (6947)', 'TCN2 (6948)', 'TCOF1 (6949)', 'TCP1 (6950)', 'TCP10L (140290)', 'TCP11 (6954)', 'TCP11L1 (55346)', 'TCP11L2 (255394)', 'TCTA (6988)', 'TCTE1 (202500)', 'TCTN1 (79600)', 'TCTN2 (79867)', 'TCTN3 (26123)', 'TDG (6996)', 'TDO2 (6999)', 'TDP1 (55775)', 'TDP2 (51567)', 'TDRD1 (56165)', 'TDRD10 (126668)', 'TDRD12 (91646)', 'TDRD3 (81550)', 'TDRD5 (163589)', 'TDRD6 (221400)', 'TDRD7 (23424)', 'TDRD9 (122402)', 'TDRKH (11022)', 'TDRP (157695)', 'TEAD1 (7003)', 'TEAD2 (8463)', 'TEAD3 (7005)', 'TEAD4 (7004)', 'TEC (7006)', 'TECPR1 (25851)', 'TECPR2 (9895)', 'TECR (9524)', 'TECRL (253017)', 'TECTA (7007)', 'TECTB (6975)', 'TEDC1 (283643)', 'TEDC2 (80178)', 'TEDDM1 (127670)', 'TEF (7008)', 'TEFM (79736)', 'TEK (7010)', 'TEKT1 (83659)', 'TEKT2 (27285)', 'TEKT3 (64518)', 'TEKT4 (150483)', 'TEKT5 (146279)', 'TEKTIP1 (100128569)', 'TEKTL1 (126402)', 'TELO2 (9894)', 'TEN1 (100134934)', 'TENM1 (10178)', 'TENM3 (55714)', 'TENM4 (26011)', 'TENT2 (167153)', 'TENT4A (11044)', 'TENT4B (64282)', 'TENT5A (55603)', 'TENT5B (115572)', 'TENT5C (54855)', 'TENT5D (169966)', 'TEP1 (7011)', 'TEPSIN (146705)', 'TERB2 (145645)', 'TERF2 (7014)', 'TERF2IP (54386)', 'TERT (7015)', 'TES (26136)', 'TESC (54997)', 'TESK1 (7016)', 'TESK2 (10420)', 'TESMIN (9633)', 'TESPA1 (9840)', 'TET1 (80312)', 'TET2 (54790)', 'TET3 (200424)', 'TEX10 (54881)', 'TEX101 (83639)', 'TEX11 (56159)', 'TEX12 (56158)', 'TEX13B (56156)', 'TEX14 (56155)', 'TEX15 (56154)', 'TEX19 (400629)', 'TEX2 (55852)', 'TEX22 (647310)', 'TEX26 (122046)', 'TEX261 (113419)', 'TEX264 (51368)', 'TEX29 (121793)', 'TEX30 (93081)', 'TEX35 (84066)', 'TEX36 (387718)', 'TEX38 (374973)', 'TEX44 (165100)', 'TEX46 (729059)', 'TEX47 (219557)', 'TEX55 (152405)', 'TEX9 (374618)', 'TF (7018)', 'TFAM (7019)', 'TFAP2A (7020)', 'TFAP2B (7021)', 'TFAP2C (7022)', 'TFAP2D (83741)', 'TFAP2E (339488)', 'TFAP4 (7023)', 'TFB1M (51106)', 'TFB2M (64216)', 'TFCP2 (7024)', 'TFCP2L1 (29842)', 'TFDP1 (7027)', 'TFDP2 (7029)', 'TFDP3 (51270)', 'TFE3 (7030)', 'TFEB (7942)', 'TFEC (22797)', 'TFF1 (7031)', 'TFF2 (7032)', 'TFF3 (7033)', 'TFG (10342)', 'TFIP11 (24144)', 'TFPI (7035)', 'TFPI2 (7980)', 'TFPT (29844)', 'TFR2 (7036)', 'TFRC (7037)', 'TG (7038)', 'TGDS (23483)', 'TGFA (7039)', 'TGFB1 (7040)', 'TGFB1I1 (7041)', 'TGFB2 (7042)', 'TGFB3 (7043)', 'TGFBI (7045)', 'TGFBR1 (7046)', 'TGFBR2 (7048)', 'TGFBR3 (7049)', 'TGFBR3L (100507588)', 'TGFBRAP1 (9392)', 'TGIF1 (7050)', 'TGIF2 (60436)', 'TGIF2LX (90316)', 'TGM1 (7051)', 'TGM2 (7052)', 'TGM3 (7053)', 'TGM4 (7047)', 'TGM5 (9333)', 'TGM6 (343641)', 'TGM7 (116179)', 'TGOLN2 (10618)', 'TGS1 (96764)', 'TH (7054)', 'THADA (63892)', 'THAP1 (55145)', 'THAP10 (56906)', 'THAP11 (57215)', 'THAP12 (5612)', 'THAP2 (83591)', 'THAP3 (90326)', 'THAP4 (51078)', 'THAP5 (168451)', 'THAP6 (152815)', 'THAP7 (80764)', 'THAP8 (199745)', 'THAP9 (79725)', 'THBD (7056)', 'THBS1 (7057)', 'THBS2 (7058)', 'THBS3 (7059)', 'THBS4 (7060)', 'THEM4 (117145)', 'THEM5 (284486)', 'THEM6 (51337)', 'THEMIS (387357)', 'THEMIS2 (9473)', 'THG1L (54974)', 'THNSL1 (79896)', 'THNSL2 (55258)', 'THOC1 (9984)', 'THOC2 (57187)', 'THOC3 (84321)', 'THOC5 (8563)', 'THOC6 (79228)', 'THOC7 (80145)', 'THOP1 (7064)', 'THPO (7066)', 'THRA (7067)', 'THRAP3 (9967)', 'THRB (7068)', 'THRSP (7069)', 'THSD1 (55901)', 'THSD4 (79875)', 'THSD7A (221981)', 'THTPA (79178)', 'THUMPD1 (55623)', 'THUMPD2 (80745)', 'THUMPD3 (25917)', 'THY1 (7070)', 'THYN1 (29087)', 'TIA1 (7072)', 'TIAL1 (7073)', 'TIAM1 (7074)', 'TIAM2 (26230)', 'TICAM1 (148022)', 'TICAM2 (353376)', 'TICRR (90381)', 'TIE1 (7075)', 'TIFA (92610)', 'TIFAB (497189)', 'TIGAR (57103)', 'TIGD1 (200765)', 'TIGD2 (166815)', 'TIGD3 (220359)', 'TIGD4 (201798)', 'TIGD5 (84948)', 'TIGD6 (81789)', 'TIGD7 (91151)', 'TIGIT (201633)', 'TIMD4 (91937)', 'TIMELESS (8914)', 'TIMM10 (26519)', 'TIMM10B (26515)', 'TIMM13 (26517)', 'TIMM17A (10440)', 'TIMM17B (10245)', 'TIMM21 (29090)', 'TIMM22 (29928)', 'TIMM23 (100287932)', 'TIMM29 (90580)', 'TIMM44 (10469)', 'TIMM50 (92609)', 'TIMM8A (1678)', 'TIMM8B (26521)', 'TIMM9 (26520)', 'TIMMDC1 (51300)', 'TIMP1 (7076)', 'TIMP2 (7077)', 'TIMP3 (7078)', 'TIMP4 (7079)', 'TINAG (27283)', 'TINAGL1 (64129)', 'TINF2 (26277)', 'TIPARP (25976)', 'TIPIN (54962)', 'TIPRL (261726)', 'TIRAP (114609)', 'TJAP1 (93643)', 'TJP1 (7082)', 'TJP2 (9414)', 'TJP3 (27134)', 'TK1 (7083)', 'TK2 (7084)', 'TKFC (26007)', 'TKT (7086)', 'TKTL1 (8277)', 'TKTL2 (84076)', 'TLCD1 (116238)', 'TLCD2 (727910)', 'TLCD3A (79850)', 'TLCD3B (83723)', 'TLCD4 (148534)', 'TLCD5 (219902)', 'TLDC2 (140711)', 'TLE1 (7088)', 'TLE2 (7089)', 'TLE3 (7090)', 'TLE4 (7091)', 'TLE5 (166)', 'TLE6 (79816)', 'TLK1 (9874)', 'TLK2 (11011)', 'TLL1 (7092)', 'TLL2 (7093)', 'TLN1 (7094)', 'TLN2 (83660)', 'TLNRD1 (59274)', 'TLR1 (7096)', 'TLR10 (81793)', 'TLR2 (7097)', 'TLR3 (7098)', 'TLR4 (7099)', 'TLR5 (7100)', 'TLR6 (10333)', 'TLR7 (51284)', 'TLR8 (51311)', 'TLR9 (54106)', 'TLX1 (3195)', 'TLX2 (3196)', 'TLX3 (30012)', 'TM2D1 (83941)', 'TM2D2 (83877)', 'TM2D3 (80213)', 'TM4SF1 (4071)', 'TM4SF18 (116441)', 'TM4SF19 (116211)', 'TM4SF20 (79853)', 'TM4SF4 (7104)', 'TM4SF5 (9032)', 'TM6SF1 (53346)', 'TM6SF2 (53345)', 'TM7SF2 (7108)', 'TM7SF3 (51768)', 'TM9SF1 (10548)', 'TM9SF2 (9375)', 'TM9SF3 (56889)', 'TM9SF4 (9777)', 'TMA16 (55319)', 'TMA7 (51372)', 'TMBIM1 (64114)', 'TMBIM4 (51643)', 'TMBIM6 (7009)', 'TMC1 (117531)', 'TMC2 (117532)', 'TMC3 (342125)', 'TMC4 (147798)', 'TMC5 (79838)', 'TMC6 (11322)', 'TMC7 (79905)', 'TMC8 (147138)', 'TMCC1 (23023)', 'TMCC2 (9911)', 'TMCC3 (57458)', 'TMCO1 (54499)', 'TMCO2 (127391)', 'TMCO3 (55002)', 'TMCO4 (255104)', 'TMCO5A (145942)', 'TMCO6 (55374)', 'TMED1 (11018)', 'TMED10 (10972)', 'TMED2 (10959)', 'TMED3 (23423)', 'TMED4 (222068)', 'TMED5 (50999)', 'TMED6 (146456)', 'TMED7 (51014)', 'TMED8 (283578)', 'TMED9 (54732)', 'TMEFF1 (8577)', 'TMEFF2 (23671)', 'TMEM100 (55273)', 'TMEM101 (84336)', 'TMEM102 (284114)', 'TMEM104 (54868)', 'TMEM106A (113277)', 'TMEM106B (54664)', 'TMEM106C (79022)', 'TMEM107 (84314)', 'TMEM108 (66000)', 'TMEM109 (79073)', 'TMEM11 (8834)', 'TMEM115 (11070)', 'TMEM116 (89894)', 'TMEM117 (84216)', 'TMEM119 (338773)', 'TMEM120A (83862)', 'TMEM120B (144404)', 'TMEM121 (80757)', 'TMEM121B (27439)', 'TMEM123 (114908)', 'TMEM125 (128218)', 'TMEM126A (84233)', 'TMEM126B (55863)', 'TMEM127 (55654)', 'TMEM128 (85013)', 'TMEM129 (92305)', 'TMEM130 (222865)', 'TMEM131 (23505)', 'TMEM131L (23240)', 'TMEM132A (54972)', 'TMEM132B (114795)', 'TMEM132D (121256)', 'TMEM132E (124842)', 'TMEM134 (80194)', 'TMEM135 (65084)', 'TMEM138 (51524)', 'TMEM139 (135932)', 'TMEM140 (55281)', 'TMEM141 (85014)', 'TMEM143 (55260)', 'TMEM144 (55314)', 'TMEM145 (284339)', 'TMEM147 (10430)', 'TMEM14A (28978)', 'TMEM14B (81853)', 'TMEM14C (51522)', 'TMEM150A (129303)', 'TMEM150B (284417)', 'TMEM150C (441027)', 'TMEM151A (256472)', 'TMEM151B (441151)', 'TMEM154 (201799)', 'TMEM156 (80008)', 'TMEM158 (25907)', 'TMEM160 (54958)', 'TMEM161A (54929)', 'TMEM161B (153396)', 'TMEM163 (81615)', 'TMEM164 (84187)', 'TMEM165 (55858)', 'TMEM167A (153339)', 'TMEM167B (56900)', 'TMEM168 (64418)', 'TMEM169 (92691)', 'TMEM17 (200728)', 'TMEM170A (124491)', 'TMEM170B (100113407)', 'TMEM171 (134285)', 'TMEM174 (134288)', 'TMEM175 (84286)', 'TMEM176A (55365)', 'TMEM176B (28959)', 'TMEM177 (80775)', 'TMEM178A (130733)', 'TMEM178B (100507421)', 'TMEM179 (388021)', 'TMEM179B (374395)', 'TMEM18 (129787)', 'TMEM181 (57583)', 'TMEM182 (130827)', 'TMEM183A (92703)', 'TMEM184A (202915)', 'TMEM184B (25829)', 'TMEM184C (55751)', 'TMEM185A (84548)', 'TMEM185B (79134)', 'TMEM186 (25880)', 'TMEM187 (8269)', 'TMEM19 (55266)', 'TMEM190 (147744)', 'TMEM192 (201931)', 'TMEM196 (256130)', 'TMEM198 (130612)', 'TMEM199 (147007)', 'TMEM200A (114801)', 'TMEM200B (399474)', 'TMEM200C (645369)', 'TMEM201 (199953)', 'TMEM202 (338949)', 'TMEM203 (94107)', 'TMEM204 (79652)', 'TMEM205 (374882)', 'TMEM207 (131920)', 'TMEM208 (29100)', 'TMEM209 (84928)', 'TMEM210 (100505993)', 'TMEM212 (389177)', 'TMEM213 (155006)', 'TMEM214 (54867)', 'TMEM215 (401498)', 'TMEM216 (51259)', 'TMEM217 (221468)', 'TMEM218 (219854)', 'TMEM219 (124446)', 'TMEM220 (388335)', 'TMEM221 (100130519)', 'TMEM222 (84065)', 'TMEM223 (79064)', 'TMEM225 (338661)', 'TMEM229A (730130)', 'TMEM229B (161145)', 'TMEM230 (29058)', 'TMEM231 (79583)', 'TMEM232 (642987)', 'TMEM233 (387890)', 'TMEM234 (56063)', 'TMEM235 (283999)', 'TMEM236 (653567)', 'TMEM237 (65062)', 'TMEM238 (388564)', 'TMEM239 (100288797)', 'TMEM240 (339453)', 'TMEM241 (85019)', 'TMEM242 (729515)', 'TMEM243 (79161)', 'TMEM244 (253582)', 'TMEM245 (23731)', 'TMEM247 (388946)', 'TMEM248 (55069)', 'TMEM249 (340393)', 'TMEM25 (84866)', 'TMEM250 (90120)', 'TMEM252 (169693)', 'TMEM253 (643382)', 'TMEM254 (80195)', 'TMEM255A (55026)', 'TMEM255B (348013)', 'TMEM256 (254863)', 'TMEM258 (746)', 'TMEM259 (91304)', 'TMEM26 (219623)', 'TMEM260 (54916)', 'TMEM262 (100130348)', 'TMEM263 (90488)', 'TMEM266 (123591)', 'TMEM267 (64417)', 'TMEM268 (203197)', 'TMEM270 (135886)', 'TMEM273 (170371)', 'TMEM276 (84773)', 'TMEM30A (55754)', 'TMEM30B (161291)', 'TMEM31 (203562)', 'TMEM33 (55161)', 'TMEM35A (59353)', 'TMEM35B (100506144)', 'TMEM37 (140738)', 'TMEM38A (79041)', 'TMEM38B (55151)', 'TMEM39A (55254)', 'TMEM39B (55116)', 'TMEM40 (55287)', 'TMEM41A (90407)', 'TMEM41B (440026)', 'TMEM42 (131616)', 'TMEM43 (79188)', 'TMEM44 (93109)', 'TMEM45A (55076)', 'TMEM45B (120224)', 'TMEM47 (83604)', 'TMEM50A (23585)', 'TMEM50B (757)', 'TMEM51 (55092)', 'TMEM52 (339456)', 'TMEM52B (120939)', 'TMEM53 (79639)', 'TMEM54 (113452)', 'TMEM59 (9528)', 'TMEM59L (25789)', 'TMEM60 (85025)', 'TMEM61 (199964)', 'TMEM62 (80021)', 'TMEM63A (9725)', 'TMEM63B (55362)', 'TMEM63C (57156)', 'TMEM64 (169200)', 'TMEM65 (157378)', 'TMEM67 (91147)', 'TMEM68 (137695)', 'TMEM69 (51249)', 'TMEM70 (54968)', 'TMEM71 (137835)', 'TMEM72 (643236)', 'TMEM74 (157753)', 'TMEM74B (55321)', 'TMEM79 (84283)', 'TMEM80 (283232)', 'TMEM81 (388730)', 'TMEM82 (388595)', 'TMEM86A (144110)', 'TMEM86B (255043)', 'TMEM87A (25963)', 'TMEM87B (84910)', 'TMEM88 (92162)', 'TMEM88B (643965)', 'TMEM89 (440955)', 'TMEM8B (51754)', 'TMEM9 (252839)', 'TMEM91 (641649)', 'TMEM92 (162461)', 'TMEM94 (9772)', 'TMEM95 (339168)', 'TMEM97 (27346)', 'TMEM98 (26022)', 'TMEM9B (56674)', 'TMF1 (7110)', 'TMIE (259236)', 'TMIGD1 (388364)', 'TMIGD2 (126259)', 'TMIGD3 (57413)', 'TMLHE (55217)', 'TMOD1 (7111)', 'TMOD2 (29767)', 'TMOD3 (29766)', 'TMOD4 (29765)', 'TMPO (7112)', 'TMPPE (643853)', 'TMPRSS11A (339967)', 'TMPRSS11B (132724)', 'TMPRSS11D (9407)', 'TMPRSS11E (28983)', 'TMPRSS11F (389208)', 'TMPRSS12 (283471)', 'TMPRSS13 (84000)', 'TMPRSS15 (5651)', 'TMPRSS2 (7113)', 'TMPRSS3 (64699)', 'TMPRSS4 (56649)', 'TMPRSS5 (80975)', 'TMPRSS6 (164656)', 'TMPRSS7 (344805)', 'TMPRSS9 (360200)', 'TMSB10 (9168)', 'TMSB15A (11013)', 'TMSB4X (7114)', 'TMT1A (25840)', 'TMT1B (196410)', 'TMTC1 (83857)', 'TMTC2 (160335)', 'TMTC3 (160418)', 'TMTC4 (84899)', 'TMUB1 (83590)', 'TMUB2 (79089)', 'TMX1 (81542)', 'TMX2 (51075)', 'TMX3 (54495)', 'TMX4 (56255)', 'TNC (3371)', 'TNF (7124)', 'TNFAIP1 (7126)', 'TNFAIP2 (7127)', 'TNFAIP3 (7128)', 'TNFAIP6 (7130)', 'TNFAIP8 (25816)', 'TNFAIP8L1 (126282)', 'TNFAIP8L2 (79626)', 'TNFAIP8L3 (388121)', 'TNFRSF10A (8797)', 'TNFRSF10B (8795)', 'TNFRSF10C (8794)', 'TNFRSF10D (8793)', 'TNFRSF11A (8792)', 'TNFRSF11B (4982)', 'TNFRSF12A (51330)', 'TNFRSF13B (23495)', 'TNFRSF13C (115650)', 'TNFRSF14 (8764)', 'TNFRSF17 (608)', 'TNFRSF18 (8784)', 'TNFRSF19 (55504)', 'TNFRSF1A (7132)', 'TNFRSF1B (7133)', 'TNFRSF21 (27242)', 'TNFRSF25 (8718)', 'TNFRSF4 (7293)', 'TNFRSF6B (8771)', 'TNFRSF8 (943)', 'TNFRSF9 (3604)', 'TNFSF10 (8743)', 'TNFSF11 (8600)', 'TNFSF12 (8742)', 'TNFSF13 (8741)', 'TNFSF13B (10673)', 'TNFSF14 (8740)', 'TNFSF15 (9966)', 'TNFSF18 (8995)', 'TNFSF4 (7292)', 'TNFSF8 (944)', 'TNFSF9 (8744)', 'TNIK (23043)', 'TNIP1 (10318)', 'TNIP2 (79155)', 'TNIP3 (79931)', 'TNK1 (8711)', 'TNK2 (10188)', 'TNKS (8658)', 'TNKS1BP1 (85456)', 'TNKS2 (80351)', 'TNMD (64102)', 'TNN (63923)', 'TNNC1 (7134)', 'TNNC2 (7125)', 'TNNI1 (7135)', 'TNNI2 (7136)', 'TNNI3 (7137)', 'TNNI3K (51086)', 'TNNT1 (7138)', 'TNNT2 (7139)', 'TNNT3 (7140)', 'TNP1 (7141)', 'TNP2 (7142)', 'TNPO1 (3842)', 'TNPO2 (30000)', 'TNPO3 (23534)', 'TNR (7143)', 'TNRC18 (84629)', 'TNRC6A (27327)', 'TNRC6B (23112)', 'TNRC6C (57690)', 'TNS1 (7145)', 'TNS2 (23371)', 'TNS3 (64759)', 'TNS4 (84951)', 'TNXB (7148)', 'TOB1 (10140)', 'TOB2 (10766)', 'TOE1 (114034)', 'TOGARAM1 (23116)', 'TOGARAM2 (165186)', 'TOLLIP (54472)', 'TOM1 (10043)', 'TOM1L1 (10040)', 'TOM1L2 (146691)', 'TOMM20 (9804)', 'TOMM20L (387990)', 'TOMM22 (56993)', 'TOMM34 (10953)', 'TOMM40 (10452)', 'TOMM40L (84134)', 'TOMM5 (401505)', 'TOMM6 (100188893)', 'TOMM7 (54543)', 'TOMM70 (9868)', 'TONSL (4796)', 'TOP1 (7150)', 'TOP1MT (116447)', 'TOP2A (7153)', 'TOP2B (7155)', 'TOP3A (7156)', 'TOP3B (8940)', 'TOP6BL (79703)', 'TOPAZ1 (375337)', 'TOPBP1 (11073)', 'TOPORS (10210)', 'TOR1A (1861)', 'TOR1AIP1 (26092)', 'TOR1AIP2 (163590)', 'TOR1B (27348)', 'TOR2A (27433)', 'TOR3A (64222)', 'TOR4A (54863)', 'TOX (9760)', 'TOX2 (84969)', 'TOX3 (27324)', 'TOX4 (9878)', 'TP53 (7157)', 'TP53AIP1 (63970)', 'TP53BP1 (7158)', 'TP53BP2 (7159)', 'TP53I11 (9537)', 'TP53I13 (90313)', 'TP53I3 (9540)', 'TP53INP1 (94241)', 'TP53INP2 (58476)', 'TP53RK (112858)', 'TP53TG5 (27296)', 'TP63 (8626)', 'TP73 (7161)', 'TPBG (7162)', 'TPBGL (100507050)', 'TPCN1 (53373)', 'TPCN2 (219931)', 'TPD52 (7163)', 'TPD52L1 (7164)', 'TPD52L2 (7165)', 'TPD52L3 (89882)', 'TPGS1 (91978)', 'TPGS2 (25941)', 'TPH1 (7166)', 'TPH2 (121278)', 'TPI1 (7167)', 'TPK1 (27010)', 'TPM1 (7168)', 'TPM2 (7169)', 'TPM3 (7170)', 'TPM4 (7171)', 'TPMT (7172)', 'TPO (7173)', 'TPP1 (1200)', 'TPP2 (7174)', 'TPPP (11076)', 'TPPP2 (122664)', 'TPPP3 (51673)', 'TPR (7175)', 'TPRA1 (131601)', 'TPRG1 (285386)', 'TPRG1L (127262)', 'TPRKB (51002)', 'TPRN (286262)', 'TPRX1 (284355)', 'TPSAB1 (7177)', 'TPSD1 (23430)', 'TPSG1 (25823)', 'TPST1 (8460)', 'TPST2 (8459)', 'TPT1 (7178)', 'TPTE2 (93492)', 'TPX2 (22974)', 'TRA2A (29896)', 'TRA2B (6434)', 'TRABD (80305)', 'TRABD2A (129293)', 'TRABD2B (388630)', 'TRADD (8717)', 'TRAF1 (7185)', 'TRAF2 (7186)', 'TRAF3 (7187)', 'TRAF3IP1 (26146)', 'TRAF3IP2 (10758)', 'TRAF3IP3 (80342)', 'TRAF4 (9618)', 'TRAF5 (7188)', 'TRAF6 (7189)', 'TRAF7 (84231)', 'TRAFD1 (10906)', 'TRAIP (10293)', 'TRAK1 (22906)', 'TRAK2 (66008)', 'TRAM1 (23471)', 'TRAM1L1 (133022)', 'TRAM2 (9697)', 'TRANK1 (9881)', 'TRAP1 (10131)', 'TRAPPC1 (58485)', 'TRAPPC10 (7109)', 'TRAPPC11 (60684)', 'TRAPPC12 (51112)', 'TRAPPC13 (80006)', 'TRAPPC14 (55262)', 'TRAPPC2L (51693)', 'TRAPPC3 (27095)', 'TRAPPC3L (100128327)', 'TRAPPC4 (51399)', 'TRAPPC5 (126003)', 'TRAPPC6A (79090)', 'TRAPPC6B (122553)', 'TRAPPC8 (22878)', 'TRAPPC9 (83696)', 'TRARG1 (286753)', 'TRAT1 (50852)', 'TRDMT1 (1787)', 'TRDN (10345)', 'TREM1 (54210)', 'TREM2 (54209)', 'TREML1 (340205)', 'TREML2 (79865)', 'TREML4 (285852)', 'TRERF1 (55809)', 'TREX1 (11277)', 'TREX2 (11219)', 'TRH (7200)', 'TRHDE (29953)', 'TRHR (7201)', 'TRIAP1 (51499)', 'TRIB1 (10221)', 'TRIB2 (28951)', 'TRIB3 (57761)', 'TRIM10 (10107)', 'TRIM11 (81559)', 'TRIM13 (10206)', 'TRIM14 (9830)', 'TRIM15 (89870)', 'TRIM16 (10626)', 'TRIM17 (51127)', 'TRIM2 (23321)', 'TRIM21 (6737)', 'TRIM22 (10346)', 'TRIM23 (373)', 'TRIM24 (8805)', 'TRIM25 (7706)', 'TRIM26 (7726)', 'TRIM27 (5987)', 'TRIM28 (10155)', 'TRIM29 (23650)', 'TRIM3 (10612)', 'TRIM31 (11074)', 'TRIM32 (22954)', 'TRIM33 (51592)', 'TRIM34 (53840)', 'TRIM35 (23087)', 'TRIM36 (55521)', 'TRIM37 (4591)', 'TRIM38 (10475)', 'TRIM39 (56658)', 'TRIM4 (89122)', 'TRIM40 (135644)', 'TRIM41 (90933)', 'TRIM42 (287015)', 'TRIM43 (129868)', 'TRIM44 (54765)', 'TRIM45 (80263)', 'TRIM46 (80128)', 'TRIM47 (91107)', 'TRIM49B (283116)', 'TRIM49D1 (399939)', 'TRIM5 (85363)', 'TRIM50 (135892)', 'TRIM52 (84851)', 'TRIM54 (57159)', 'TRIM55 (84675)', 'TRIM56 (81844)', 'TRIM58 (25893)', 'TRIM59 (286827)', 'TRIM6 (117854)', 'TRIM60 (166655)', 'TRIM61 (391712)', 'TRIM62 (55223)', 'TRIM63 (84676)', 'TRIM65 (201292)', 'TRIM67 (440730)', 'TRIM68 (55128)', 'TRIM69 (140691)', 'TRIM7 (81786)', 'TRIM71 (131405)', 'TRIM72 (493829)', 'TRIM77 (390231)', 'TRIM8 (81603)', 'TRIM9 (114088)', 'TRIML1 (339976)', 'TRIML2 (205860)', 'TRIO (7204)', 'TRIOBP (11078)', 'TRIP10 (9322)', 'TRIP11 (9321)', 'TRIP12 (9320)', 'TRIP13 (9319)', 'TRIP4 (9325)', 'TRIP6 (7205)', 'TRIQK (286144)', 'TRIR (79002)', 'TRIT1 (54802)', 'TRMO (51531)', 'TRMT1 (55621)', 'TRMT10A (93587)', 'TRMT10B (158234)', 'TRMT10C (54931)', 'TRMT11 (60487)', 'TRMT112 (51504)', 'TRMT12 (55039)', 'TRMT13 (54482)', 'TRMT1L (81627)', 'TRMT2A (27037)', 'TRMT2B (79979)', 'TRMT44 (152992)', 'TRMT5 (57570)', 'TRMT6 (51605)', 'TRMT61A (115708)', 'TRMT61B (55006)', 'TRMT9B (57604)', 'TRMU (55687)', 'TRNAU1AP (54952)', 'TRNP1 (388610)', 'TRNT1 (51095)', 'TRO (7216)', 'TROAP (10024)', 'TRPA1 (8989)', 'TRPC1 (7220)', 'TRPC3 (7222)', 'TRPC4 (7223)', 'TRPC4AP (26133)', 'TRPC5 (7224)', 'TRPC6 (7225)', 'TRPC7 (57113)', 'TRPM1 (4308)', 'TRPM2 (7226)', 'TRPM3 (80036)', 'TRPM4 (54795)', 'TRPM5 (29850)', 'TRPM6 (140803)', 'TRPM7 (54822)', 'TRPM8 (79054)', 'TRPS1 (7227)', 'TRPT1 (83707)', 'TRPV1 (7442)', 'TRPV2 (51393)', 'TRPV3 (162514)', 'TRPV4 (59341)', 'TRPV5 (56302)', 'TRPV6 (55503)', 'TRRAP (8295)', 'TRUB1 (142940)', 'TRUB2 (26995)', 'TSACC (128229)', 'TSBP1 (10665)', 'TSC1 (7248)', 'TSC2 (7249)', 'TSC22D1 (8848)', 'TSC22D2 (9819)', 'TSC22D3 (1831)', 'TSC22D4 (81628)', 'TSEN15 (116461)', 'TSEN2 (80746)', 'TSEN34 (79042)', 'TSEN54 (283989)', 'TSFM (10102)', 'TSG101 (7251)', 'TSGA10 (80705)', 'TSGA10IP (254187)', 'TSGA13 (114960)', 'TSHB (7252)', 'TSHR (7253)', 'TSHZ1 (10194)', 'TSHZ2 (128553)', 'TSHZ3 (57616)', 'TSKS (60385)', 'TSKU (25987)', 'TSLP (85480)', 'TSN (7247)', 'TSNARE1 (203062)', 'TSNAX (7257)', 'TSNAXIP1 (55815)', 'TSPAN1 (10103)', 'TSPAN11 (441631)', 'TSPAN12 (23554)', 'TSPAN13 (27075)', 'TSPAN14 (81619)', 'TSPAN15 (23555)', 'TSPAN16 (26526)', 'TSPAN17 (26262)', 'TSPAN18 (90139)', 'TSPAN19 (144448)', 'TSPAN2 (10100)', 'TSPAN3 (10099)', 'TSPAN31 (6302)', 'TSPAN32 (10077)', 'TSPAN33 (340348)', 'TSPAN4 (7106)', 'TSPAN5 (10098)', 'TSPAN6 (7105)', 'TSPAN7 (7102)', 'TSPAN8 (7103)', 'TSPAN9 (10867)', 'TSPEAR (54084)', 'TSPO (706)', 'TSPO2 (222642)', 'TSPOAP1 (9256)', 'TSPYL1 (7259)', 'TSPYL2 (64061)', 'TSPYL4 (23270)', 'TSPYL5 (85453)', 'TSPYL6 (388951)', 'TSR1 (55720)', 'TSR2 (90121)', 'TSR3 (115939)', 'TSSC4 (10078)', 'TSSK1B (83942)', 'TSSK2 (23617)', 'TSSK3 (81629)', 'TSSK4 (283629)', 'TSSK6 (83983)', 'TST (7263)', 'TSTD1 (100131187)', 'TSTD2 (158427)', 'TTBK1 (84630)', 'TTBK2 (146057)', 'TTC1 (7265)', 'TTC12 (54970)', 'TTC13 (79573)', 'TTC14 (151613)', 'TTC16 (158248)', 'TTC17 (55761)', 'TTC19 (54902)', 'TTC21A (199223)', 'TTC21B (79809)', 'TTC22 (55001)', 'TTC23 (64927)', 'TTC23L (153657)', 'TTC24 (164118)', 'TTC27 (55622)', 'TTC28 (23331)', 'TTC29 (83894)', 'TTC3 (7267)', 'TTC31 (64427)', 'TTC32 (130502)', 'TTC33 (23548)', 'TTC34 (100287898)', 'TTC36 (143941)', 'TTC38 (55020)', 'TTC39A (22996)', 'TTC39B (158219)', 'TTC39C (125488)', 'TTC4 (7268)', 'TTC5 (91875)', 'TTC7A (57217)', 'TTC7B (145567)', 'TTC8 (123016)', 'TTC9 (23508)', 'TTC9B (148014)', 'TTC9C (283237)', 'TTF1 (7270)', 'TTF2 (8458)', 'TTI1 (9675)', 'TTI2 (80185)', 'TTK (7272)', 'TTL (150465)', 'TTLL1 (25809)', 'TTLL10 (254173)', 'TTLL11 (158135)', 'TTLL12 (23170)', 'TTLL2 (83887)', 'TTLL3 (26140)', 'TTLL4 (9654)', 'TTLL5 (23093)', 'TTLL6 (284076)', 'TTLL7 (79739)', 'TTLL9 (164395)', 'TTN (7273)', 'TTPA (7274)', 'TTPAL (79183)', 'TTR (7276)', 'TTYH1 (57348)', 'TTYH2 (94015)', 'TTYH3 (80727)', 'TUB (7275)', 'TUBA1A (7846)', 'TUBA1B (10376)', 'TUBA1C (84790)', 'TUBA3C (7278)', 'TUBA3D (113457)', 'TUBA3E (112714)', 'TUBA4A (7277)', 'TUBA8 (51807)', 'TUBAL3 (79861)', 'TUBB (203068)', 'TUBB1 (81027)', 'TUBB2A (7280)', 'TUBB2B (347733)', 'TUBB3 (10381)', 'TUBB4A (10382)', 'TUBB4B (10383)', 'TUBB6 (84617)', 'TUBD1 (51174)', 'TUBE1 (51175)', 'TUBG1 (7283)', 'TUBG2 (27175)', 'TUBGCP2 (10844)', 'TUBGCP3 (10426)', 'TUBGCP4 (27229)', 'TUBGCP5 (114791)', 'TUBGCP6 (85378)', 'TUFM (7284)', 'TUFT1 (7286)', 'TULP1 (7287)', 'TULP2 (7288)', 'TULP3 (7289)', 'TULP4 (56995)', 'TUSC1 (286319)', 'TUSC2 (11334)', 'TUSC3 (7991)', 'TUT1 (64852)', 'TUT4 (23318)', 'TUT7 (79670)', 'TVP23A (780776)', 'TWF1 (5756)', 'TWF2 (11344)', 'TWIST1 (7291)', 'TWIST2 (117581)', 'TWNK (56652)', 'TWSG1 (57045)', 'TXK (7294)', 'TXLNA (200081)', 'TXLNB (167838)', 'TXLNG (55787)', 'TXN (7295)', 'TXN2 (25828)', 'TXNDC11 (51061)', 'TXNDC12 (51060)', 'TXNDC15 (79770)', 'TXNDC16 (57544)', 'TXNDC17 (84817)', 'TXNDC2 (84203)', 'TXNDC5 (81567)', 'TXNDC8 (255220)', 'TXNDC9 (10190)', 'TXNIP (10628)', 'TXNL1 (9352)', 'TXNL4A (10907)', 'TXNL4B (54957)', 'TXNRD1 (7296)', 'TXNRD2 (10587)', 'TYK2 (7297)', 'TYMP (1890)', 'TYMS (7298)', 'TYR (7299)', 'TYRO3 (7301)', 'TYROBP (7305)', 'TYRP1 (7306)', 'TYSND1 (219743)', 'TYW1 (55253)', 'TYW1B (441250)', 'TYW3 (127253)', 'TYW5 (129450)', 'U2AF1 (7307)', 'U2AF1L4 (199746)', 'U2AF2 (11338)', 'U2SURP (23350)', 'UACA (55075)', 'UAP1 (6675)', 'UAP1L1 (91373)', 'UBA1 (7317)', 'UBA2 (10054)', 'UBA3 (9039)', 'UBA5 (79876)', 'UBA52 (7311)', 'UBA6 (55236)', 'UBA7 (7318)', 'UBAC1 (10422)', 'UBAC2 (337867)', 'UBALD1 (124402)', 'UBALD2 (283991)', 'UBAP1 (51271)', 'UBAP1L (390595)', 'UBAP2 (55833)', 'UBAP2L (9898)', 'UBASH3A (53347)', 'UBASH3B (84959)', 'UBB (7314)', 'UBC (7316)', 'UBD (10537)', 'UBE2A (7319)', 'UBE2B (7320)', 'UBE2C (11065)', 'UBE2D1 (7321)', 'UBE2D2 (7322)', 'UBE2D3 (7323)', 'UBE2D4 (51619)', 'UBE2E1 (7324)', 'UBE2E2 (7325)', 'UBE2E3 (10477)', 'UBE2F (140739)', 'UBE2G1 (7326)', 'UBE2G2 (7327)', 'UBE2H (7328)', 'UBE2I (7329)', 'UBE2J1 (51465)', 'UBE2J2 (118424)', 'UBE2K (3093)', 'UBE2L3 (7332)', 'UBE2L6 (9246)', 'UBE2M (9040)', 'UBE2N (7334)', 'UBE2O (63893)', 'UBE2Q1 (55585)', 'UBE2Q2 (92912)', 'UBE2QL1 (134111)', 'UBE2R2 (54926)', 'UBE2S (27338)', 'UBE2T (29089)', 'UBE2U (148581)', 'UBE2V1 (7335)', 'UBE2V2 (7336)', 'UBE2W (55284)', 'UBE2Z (65264)', 'UBE3A (7337)', 'UBE3B (89910)', 'UBE3C (9690)', 'UBE3D (90025)', 'UBE4A (9354)', 'UBE4B (10277)', 'UBFD1 (56061)', 'UBIAD1 (29914)', 'UBL3 (5412)', 'UBL4A (8266)', 'UBL4B (164153)', 'UBL5 (59286)', 'UBL7 (84993)', 'UBLCP1 (134510)', 'UBN1 (29855)', 'UBN2 (254048)', 'UBOX5 (22888)', 'UBP1 (7342)', 'UBQLN1 (29979)', 'UBQLN2 (29978)', 'UBQLN3 (50613)', 'UBQLN4 (56893)', 'UBQLNL (143630)', 'UBR1 (197131)', 'UBR2 (23304)', 'UBR3 (130507)', 'UBR4 (23352)', 'UBR5 (51366)', 'UBR7 (55148)', 'UBTD1 (80019)', 'UBTD2 (92181)', 'UBTF (7343)', 'UBXN1 (51035)', 'UBXN10 (127733)', 'UBXN11 (91544)', 'UBXN2A (165324)', 'UBXN2B (137886)', 'UBXN4 (23190)', 'UBXN6 (80700)', 'UBXN7 (26043)', 'UCHL1 (7345)', 'UCHL3 (7347)', 'UCHL5 (51377)', 'UCK1 (83549)', 'UCK2 (7371)', 'UCKL1 (54963)', 'UCMA (221044)', 'UCN (7349)', 'UCN2 (90226)', 'UCN3 (114131)', 'UCP1 (7350)', 'UCP2 (7351)', 'UCP3 (7352)', 'UEVLD (55293)', 'UFC1 (51506)', 'UFD1 (7353)', 'UFL1 (23376)', 'UFM1 (51569)', 'UFSP1 (402682)', 'UFSP2 (55325)', 'UGCG (7357)', 'UGDH (7358)', 'UGGT1 (56886)', 'UGGT2 (55757)', 'UGP2 (7360)', 'UGT1A1 (54658)', 'UGT1A10 (54575)', 'UGT1A3 (54659)', 'UGT1A4 (54657)', 'UGT1A5 (54579)', 'UGT1A7 (54577)', 'UGT1A8 (54576)', 'UGT1A9 (54600)', 'UGT2A1 (10941)', 'UGT2A3 (79799)', 'UGT2B11 (10720)', 'UGT2B15 (7366)', 'UGT2B17 (7367)', 'UGT2B28 (54490)', 'UGT2B4 (7363)', 'UGT2B7 (7364)', 'UGT3A1 (133688)', 'UGT3A2 (167127)', 'UGT8 (7368)', 'UHMK1 (127933)', 'UHRF2 (115426)', 'UIMC1 (51720)', 'ULBP1 (80329)', 'ULBP2 (80328)', 'ULBP3 (79465)', 'ULK1 (8408)', 'ULK2 (9706)', 'ULK3 (25989)', 'ULK4 (54986)', 'UMOD (7369)', 'UMODL1 (89766)', 'UMPS (7372)', 'UNC119 (9094)', 'UNC119B (84747)', 'UNC13A (23025)', 'UNC13B (10497)', 'UNC13C (440279)', 'UNC13D (201294)', 'UNC45A (55898)', 'UNC45B (146862)', 'UNC50 (25972)', 'UNC5A (90249)', 'UNC5B (219699)', 'UNC5C (8633)', 'UNC5CL (222643)', 'UNC5D (137970)', 'UNC79 (57578)', 'UNC80 (285175)', 'UNC93A (54346)', 'UNCX (340260)', 'UNG (7374)', 'UNK (85451)', 'UNKL (64718)', 'UPB1 (51733)', 'UPF1 (5976)', 'UPF2 (26019)', 'UPF3A (65110)', 'UPF3B (65109)', 'UPK1A (11045)', 'UPK1B (7348)', 'UPK2 (7379)', 'UPK3A (7380)', 'UPK3B (105375355)', 'UPP1 (7378)', 'UPP2 (151531)', 'UPRT (139596)', 'UQCC1 (55245)', 'UQCC2 (84300)', 'UQCC3 (790955)', 'UQCC4 (283951)', 'UQCC5 (440957)', 'UQCC6 (728568)', 'UQCR10 (29796)', 'UQCR11 (10975)', 'UQCRB (7381)', 'UQCRC1 (7384)', 'UQCRC2 (7385)', 'UQCRFS1 (7386)', 'UQCRQ (27089)', 'URAD (646625)', 'URB1 (9875)', 'URB2 (9816)', 'URGCP (55665)', 'URI1 (8725)', 'URM1 (81605)', 'UROC1 (131669)', 'UROD (7389)', 'UROS (7390)', 'USB1 (79650)', 'USE1 (55850)', 'USF1 (7391)', 'USF2 (7392)', 'USF3 (205717)', 'USH1C (10083)', 'USH1G (124590)', 'USH2A (7399)', 'USHBP1 (83878)', 'USP1 (7398)', 'USP10 (9100)', 'USP11 (8237)', 'USP12 (219333)', 'USP13 (8975)', 'USP14 (9097)', 'USP15 (9958)', 'USP16 (10600)', 'USP17L10 (100287144)', 'USP17L13 (100287238)', 'USP18 (11274)', 'USP19 (10869)', 'USP2 (9099)', 'USP20 (10868)', 'USP21 (27005)', 'USP22 (23326)', 'USP24 (23358)', 'USP25 (29761)', 'USP26 (83844)', 'USP27X (389856)', 'USP28 (57646)', 'USP29 (57663)', 'USP3 (9960)', 'USP30 (84749)', 'USP31 (57478)', 'USP32 (84669)', 'USP33 (23032)', 'USP34 (9736)', 'USP35 (57558)', 'USP36 (57602)', 'USP37 (57695)', 'USP38 (84640)', 'USP39 (10713)', 'USP4 (7375)', 'USP40 (55230)', 'USP42 (84132)', 'USP43 (124739)', 'USP44 (84101)', 'USP45 (85015)', 'USP46 (64854)', 'USP47 (55031)', 'USP48 (84196)', 'USP49 (25862)', 'USP5 (8078)', 'USP50 (373509)', 'USP51 (158880)', 'USP53 (54532)', 'USP54 (159195)', 'USP6 (9098)', 'USP6NL (9712)', 'USP7 (7874)', 'USP8 (9101)', 'USP9X (8239)', 'USP9Y (8287)', 'USPL1 (10208)', 'UST (10090)', 'UTF1 (8433)', 'UTP11 (51118)', 'UTP14A (10813)', 'UTP14C (9724)', 'UTP15 (84135)', 'UTP18 (51096)', 'UTP20 (27340)', 'UTP23 (84294)', 'UTP25 (27042)', 'UTP3 (57050)', 'UTP4 (84916)', 'UTP6 (55813)', 'UTRN (7402)', 'UTS2 (10911)', 'UTS2B (257313)', 'UTS2R (2837)', 'UTY (7404)', 'UVRAG (7405)', 'UVSSA (57654)', 'UXS1 (80146)', 'UXT (8409)', 'VAC14 (55697)', 'VAMP1 (6843)', 'VAMP2 (6844)', 'VAMP3 (9341)', 'VAMP4 (8674)', 'VAMP5 (10791)', 'VAMP7 (6845)', 'VAMP8 (8673)', 'VANGL1 (81839)', 'VANGL2 (57216)', 'VAPA (9218)', 'VAPB (9217)', 'VARS1 (7407)', 'VARS2 (57176)', 'VASH1 (22846)', 'VASH2 (79805)', 'VASN (114990)', 'VASP (7408)', 'VAT1 (10493)', 'VAT1L (57687)', 'VAV1 (7409)', 'VAV2 (7410)', 'VAV3 (10451)', 'VAX1 (11023)', 'VAX2 (25806)', 'VBP1 (7411)', 'VCAM1 (7412)', 'VCAN (1462)', 'VCF1 (84923)', 'VCF2 (90736)', 'VCL (7414)', 'VCP (7415)', 'VCPIP1 (80124)', 'VCPKMT (79609)', 'VDAC1 (7416)', 'VDAC2 (7417)', 'VDAC3 (7419)', 'VDR (7421)', 'VEGFA (7422)', 'VEGFB (7423)', 'VEGFC (7424)', 'VEGFD (2277)', 'VENTX (27287)', 'VEPH1 (79674)', 'VEZF1 (7716)', 'VEZT (55591)', 'VGF (7425)', 'VGLL1 (51442)', 'VGLL2 (245806)', 'VGLL3 (389136)', 'VGLL4 (9686)', 'VHL (7428)', 'VIL1 (7429)', 'VILL (50853)', 'VIM (7431)', 'VIP (7432)', 'VIPAS39 (63894)', 'VIPR1 (7433)', 'VIPR2 (7434)', 'VIRMA (25962)', 'VIT (5212)', 'VKORC1 (79001)', 'VKORC1L1 (154807)', 'VLDLR (7436)', 'VMA21 (203547)', 'VMAC (400673)', 'VMO1 (284013)', 'VMP1 (81671)', 'VN1R1 (57191)', 'VN1R2 (317701)', 'VN1R4 (317703)', 'VNN1 (8876)', 'VNN2 (8875)', 'VOPP1 (81552)', 'VPREB1 (7441)', 'VPREB3 (29802)', 'VPS13A (23230)', 'VPS13B (157680)', 'VPS13C (54832)', 'VPS13D (55187)', 'VPS16 (64601)', 'VPS18 (57617)', 'VPS25 (84313)', 'VPS26A (9559)', 'VPS26B (112936)', 'VPS26C (10311)', 'VPS28 (51160)', 'VPS29 (51699)', 'VPS33A (65082)', 'VPS33B (26276)', 'VPS35 (55737)', 'VPS35L (57020)', 'VPS36 (51028)', 'VPS37A (137492)', 'VPS37B (79720)', 'VPS37C (55048)', 'VPS37D (155382)', 'VPS39 (23339)', 'VPS41 (27072)', 'VPS45 (11311)', 'VPS4A (27183)', 'VPS4B (9525)', 'VPS50 (55610)', 'VPS51 (738)', 'VPS52 (6293)', 'VPS53 (55275)', 'VPS54 (51542)', 'VPS72 (6944)', 'VPS8 (23355)', 'VPS9D1 (9605)', 'VRK1 (7443)', 'VRK2 (7444)', 'VRK3 (51231)', 'VRTN (55237)', 'VSIG1 (340547)', 'VSIG10 (54621)', 'VSIG10L (147645)', 'VSIG2 (23584)', 'VSIG4 (11326)', 'VSIG8 (391123)', 'VSIR (64115)', 'VSNL1 (7447)', 'VSTM1 (284415)', 'VSTM2A (222008)', 'VSTM2B (342865)', 'VSTM2L (128434)', 'VSTM4 (196740)', 'VSTM5 (387804)', 'VSX1 (30813)', 'VSX2 (338917)', 'VTA1 (51534)', 'VTCN1 (79679)', 'VTI1A (143187)', 'VTI1B (10490)', 'VTN (7448)', 'VWA1 (64856)', 'VWA2 (340706)', 'VWA3A (146177)', 'VWA3B (200403)', 'VWA5A (4013)', 'VWA5B2 (90113)', 'VWA7 (80737)', 'VWA8 (23078)', 'VWC2 (375567)', 'VWC2L (402117)', 'VWCE (220001)', 'VWDE (221806)', 'VWF (7450)', 'VXN (254778)', 'WAC (51322)', 'WAPL (23063)', 'WARS1 (7453)', 'WARS2 (10352)', 'WAS (7454)', 'WASF1 (8936)', 'WASF2 (10163)', 'WASF3 (10810)', 'WASHC2A (387680)', 'WASHC2C (253725)', 'WASHC3 (51019)', 'WASHC4 (23325)', 'WASHC5 (9897)', 'WASL (8976)', 'WBP1 (23559)', 'WBP11 (51729)', 'WBP1L (54838)', 'WBP2 (23558)', 'WBP2NL (164684)', 'WBP4 (11193)', 'WDCP (80304)', 'WDFY1 (57590)', 'WDFY2 (115825)', 'WDFY3 (23001)', 'WDFY4 (57705)', 'WDHD1 (11169)', 'WDPCP (51057)', 'WDR1 (9948)', 'WDR11 (55717)', 'WDR12 (55759)', 'WDR13 (64743)', 'WDR17 (116966)', 'WDR18 (57418)', 'WDR19 (57728)', 'WDR20 (91833)', 'WDR24 (84219)', 'WDR25 (79446)', 'WDR26 (80232)', 'WDR27 (253769)', 'WDR3 (10885)', 'WDR31 (114987)', 'WDR33 (55339)', 'WDR35 (57539)', 'WDR36 (134430)', 'WDR37 (22884)', 'WDR38 (401551)', 'WDR4 (10785)', 'WDR41 (55255)', 'WDR43 (23160)', 'WDR44 (54521)', 'WDR45 (11152)', 'WDR45B (56270)', 'WDR46 (9277)', 'WDR47 (22911)', 'WDR48 (57599)', 'WDR49 (151790)', 'WDR5 (11091)', 'WDR53 (348793)', 'WDR54 (84058)', 'WDR55 (54853)', 'WDR59 (79726)', 'WDR5B (54554)', 'WDR6 (11180)', 'WDR62 (284403)', 'WDR7 (23335)', 'WDR70 (55100)', 'WDR72 (256764)', 'WDR73 (84942)', 'WDR74 (54663)', 'WDR75 (84128)', 'WDR76 (79968)', 'WDR77 (79084)', 'WDR81 (124997)', 'WDR82 (80335)', 'WDR83 (84292)', 'WDR83OS (51398)', 'WDR86 (349136)', 'WDR87 (83889)', 'WDR88 (126248)', 'WDR89 (112840)', 'WDR90 (197335)', 'WDR91 (29062)', 'WDR93 (56964)', 'WDSUB1 (151525)', 'WDTC1 (23038)', 'WEE1 (7465)', 'WEE2 (494551)', 'WFDC1 (58189)', 'WFDC10A (140832)', 'WFDC10B (280664)', 'WFDC11 (259239)', 'WFDC12 (128488)', 'WFDC13 (164237)', 'WFDC2 (10406)', 'WFDC3 (140686)', 'WFDC5 (149708)', 'WFDC6 (140870)', 'WFDC8 (90199)', 'WFDC9 (259240)', 'WFIKKN1 (117166)', 'WFIKKN2 (124857)', 'WFS1 (7466)', 'WHAMM (123720)', 'WHRN (25861)', 'WIF1 (11197)', 'WIPF1 (7456)', 'WIPF2 (147179)', 'WIPF3 (644150)', 'WIPI1 (55062)', 'WIPI2 (26100)', 'WIZ (58525)', 'WLS (79971)', 'WNK1 (65125)', 'WNK3 (65267)', 'WNK4 (65266)', 'WNT1 (7471)', 'WNT10A (80326)', 'WNT10B (7480)', 'WNT11 (7481)', 'WNT16 (51384)', 'WNT2 (7472)', 'WNT2B (7482)', 'WNT3 (7473)', 'WNT3A (89780)', 'WNT4 (54361)', 'WNT5A (7474)', 'WNT5B (81029)', 'WNT6 (7475)', 'WNT7A (7476)', 'WNT7B (7477)', 'WNT8A (7478)', 'WNT8B (7479)', 'WNT9A (7483)', 'WNT9B (7484)', 'WRAP53 (55135)', 'WRAP73 (49856)', 'WRN (7486)', 'WRNIP1 (56897)', 'WSB1 (26118)', 'WSB2 (55884)', 'WSCD1 (23302)', 'WSCD2 (9671)', 'WT1 (7490)', 'WTAP (9589)', 'WTIP (126374)', 'WWC1 (23286)', 'WWC2 (80014)', 'WWOX (51741)', 'WWP1 (11059)', 'WWP2 (11060)', 'WWTR1 (25937)', 'XAB2 (56949)', 'XAF1 (54739)', 'XAGE2 (9502)', 'XAGE3 (170626)', 'XAGE5 (170627)', 'XBP1 (7494)', 'XCL1 (6375)', 'XCL2 (6846)', 'XCR1 (2829)', 'XDH (7498)', 'XG (7499)', 'XIAP (331)', 'XIRP1 (165904)', 'XIRP2 (129446)', 'XK (7504)', 'XKR3 (150165)', 'XKR4 (114786)', 'XKR6 (286046)', 'XKR7 (343702)', 'XKR8 (55113)', 'XKR9 (389668)', 'XKRX (402415)', 'XPA (7507)', 'XPC (7508)', 'XPNPEP1 (7511)', 'XPNPEP2 (7512)', 'XPNPEP3 (63929)', 'XPO1 (7514)', 'XPO4 (64328)', 'XPO5 (57510)', 'XPO6 (23214)', 'XPO7 (23039)', 'XPOT (11260)', 'XPR1 (9213)', 'XRCC1 (7515)', 'XRCC2 (7516)', 'XRCC3 (7517)', 'XRCC4 (7518)', 'XRCC5 (7520)', 'XRCC6 (2547)', 'XRN1 (54464)', 'XRN2 (22803)', 'XRRA1 (143570)', 'XXYLT1 (152002)', 'XYLB (9942)', 'XYLT1 (64131)', 'XYLT2 (64132)', 'YAE1 (57002)', 'YAF2 (10138)', 'YAP1 (10413)', 'YARS1 (8565)', 'YARS2 (51067)', 'YBEY (54059)', 'YBX1 (4904)', 'YBX2 (51087)', 'YBX3 (8531)', 'YDJC (150223)', 'YEATS2 (55689)', 'YEATS4 (8089)', 'YES1 (7525)', 'YIF1A (10897)', 'YIF1B (90522)', 'YIPF1 (54432)', 'YIPF2 (78992)', 'YIPF3 (25844)', 'YIPF4 (84272)', 'YIPF5 (81555)', 'YIPF6 (286451)', 'YIPF7 (285525)', 'YJEFN3 (374887)', 'YJU2 (55702)', 'YJU2B (81576)', 'YKT6 (10652)', 'YLPM1 (56252)', 'YME1L1 (10730)', 'YOD1 (55432)', 'YPEL1 (29799)', 'YPEL2 (388403)', 'YPEL3 (83719)', 'YPEL4 (219539)', 'YPEL5 (51646)', 'YRDC (79693)', 'YTHDC1 (91746)', 'YTHDC2 (64848)', 'YTHDF1 (54915)', 'YTHDF2 (51441)', 'YWHAB (7529)', 'YWHAE (7531)', 'YWHAG (7532)', 'YWHAH (7533)', 'YWHAQ (10971)', 'YWHAZ (7534)', 'YY1 (7528)', 'YY1AP1 (55249)', 'YY2 (404281)', 'ZACN (353174)', 'ZAP70 (7535)', 'ZAR1 (326340)', 'ZAR1L (646799)', 'ZBBX (79740)', 'ZBED2 (79413)', 'ZBED3 (84327)', 'ZBED4 (9889)', 'ZBED6 (100381270)', 'ZBP1 (81030)', 'ZBTB1 (22890)', 'ZBTB10 (65986)', 'ZBTB11 (27107)', 'ZBTB12 (221527)', 'ZBTB14 (7541)', 'ZBTB16 (7704)', 'ZBTB17 (7709)', 'ZBTB18 (10472)', 'ZBTB2 (57621)', 'ZBTB20 (26137)', 'ZBTB21 (49854)', 'ZBTB22 (9278)', 'ZBTB24 (9841)', 'ZBTB25 (7597)', 'ZBTB26 (57684)', 'ZBTB3 (79842)', 'ZBTB32 (27033)', 'ZBTB33 (10009)', 'ZBTB34 (403341)', 'ZBTB37 (84614)', 'ZBTB38 (253461)', 'ZBTB39 (9880)', 'ZBTB4 (57659)', 'ZBTB40 (9923)', 'ZBTB41 (360023)', 'ZBTB42 (100128927)', 'ZBTB43 (23099)', 'ZBTB44 (29068)', 'ZBTB45 (84878)', 'ZBTB46 (140685)', 'ZBTB47 (92999)', 'ZBTB48 (3104)', 'ZBTB49 (166793)', 'ZBTB5 (9925)', 'ZBTB6 (10773)', 'ZBTB7A (51341)', 'ZBTB7B (51043)', 'ZBTB7C (201501)', 'ZBTB8A (653121)', 'ZBTB8B (728116)', 'ZBTB8OS (339487)', 'ZBTB9 (221504)', 'ZC2HC1A (51101)', 'ZC2HC1B (153918)', 'ZC2HC1C (79696)', 'ZC3H10 (84872)', 'ZC3H11A (9877)', 'ZC3H12A (80149)', 'ZC3H12B (340554)', 'ZC3H12C (85463)', 'ZC3H13 (23091)', 'ZC3H14 (79882)', 'ZC3H15 (55854)', 'ZC3H18 (124245)', 'ZC3H3 (23144)', 'ZC3H4 (23211)', 'ZC3H6 (376940)', 'ZC3H7A (29066)', 'ZC3H7B (23264)', 'ZC3H8 (84524)', 'ZC3HAV1 (56829)', 'ZC3HAV1L (92092)', 'ZC3HC1 (51530)', 'ZC4H2 (55906)', 'ZCCHC10 (54819)', 'ZCCHC12 (170261)', 'ZCCHC13 (389874)', 'ZCCHC14 (23174)', 'ZCCHC17 (51538)', 'ZCCHC18 (644353)', 'ZCCHC2 (54877)', 'ZCCHC24 (219654)', 'ZCCHC3 (85364)', 'ZCCHC4 (29063)', 'ZCCHC7 (84186)', 'ZCCHC9 (84240)', 'ZCRB1 (85437)', 'ZCWPW1 (55063)', 'ZCWPW2 (152098)', 'ZDBF2 (57683)', 'ZDHHC1 (29800)', 'ZDHHC11 (79844)', 'ZDHHC12 (84885)', 'ZDHHC13 (54503)', 'ZDHHC14 (79683)', 'ZDHHC15 (158866)', 'ZDHHC16 (84287)', 'ZDHHC17 (23390)', 'ZDHHC18 (84243)', 'ZDHHC19 (131540)', 'ZDHHC2 (51201)', 'ZDHHC20 (253832)', 'ZDHHC21 (340481)', 'ZDHHC22 (283576)', 'ZDHHC23 (254887)', 'ZDHHC24 (254359)', 'ZDHHC3 (51304)', 'ZDHHC4 (55146)', 'ZDHHC5 (25921)', 'ZDHHC6 (64429)', 'ZDHHC7 (55625)', 'ZDHHC8 (29801)', 'ZDHHC9 (51114)', 'ZEB1 (6935)', 'ZEB2 (9839)', 'ZER1 (10444)', 'ZFAND1 (79752)', 'ZFAND2A (90637)', 'ZFAND2B (130617)', 'ZFAND3 (60685)', 'ZFAND4 (93550)', 'ZFAND5 (7763)', 'ZFAND6 (54469)', 'ZFAT (57623)', 'ZFC3H1 (196441)', 'ZFHX2 (85446)', 'ZFHX3 (463)', 'ZFHX4 (79776)', 'ZFP1 (162239)', 'ZFP14 (57677)', 'ZFP2 (80108)', 'ZFP28 (140612)', 'ZFP3 (124961)', 'ZFP30 (22835)', 'ZFP36 (7538)', 'ZFP36L1 (677)', 'ZFP36L2 (678)', 'ZFP37 (7539)', 'ZFP41 (286128)', 'ZFP42 (132625)', 'ZFP57 (346171)', 'ZFP62 (643836)', 'ZFP64 (55734)', 'ZFP69 (339559)', 'ZFP69B (65243)', 'ZFP82 (284406)', 'ZFP90 (146198)', 'ZFP91 (80829)', 'ZFP92 (139735)', 'ZFPL1 (7542)', 'ZFPM1 (161882)', 'ZFPM2 (23414)', 'ZFR (51663)', 'ZFR2 (23217)', 'ZFX (7543)', 'ZFY (7544)', 'ZFYVE1 (53349)', 'ZFYVE16 (9765)', 'ZFYVE19 (84936)', 'ZFYVE21 (79038)', 'ZFYVE26 (23503)', 'ZFYVE27 (118813)', 'ZFYVE28 (57732)', 'ZFYVE9 (9372)', 'ZG16 (653808)', 'ZG16B (124220)', 'ZGLP1 (100125288)', 'ZGPAT (84619)', 'ZGRF1 (55345)', 'ZHX1 (11244)', 'ZHX2 (22882)', 'ZHX3 (23051)', 'ZIC1 (7545)', 'ZIC2 (7546)', 'ZIC3 (7547)', 'ZIC4 (84107)', 'ZIC5 (85416)', 'ZIK1 (284307)', 'ZIM2 (23619)', 'ZIM3 (114026)', 'ZKSCAN1 (7586)', 'ZKSCAN2 (342357)', 'ZKSCAN3 (80317)', 'ZKSCAN4 (387032)', 'ZKSCAN5 (23660)', 'ZKSCAN7 (55888)', 'ZKSCAN8 (7745)', 'ZMAT1 (84460)', 'ZMAT2 (153527)', 'ZMAT3 (64393)', 'ZMAT4 (79698)', 'ZMAT5 (55954)', 'ZMIZ1 (57178)', 'ZMIZ2 (83637)', 'ZMPSTE24 (10269)', 'ZMYM1 (79830)', 'ZMYM2 (7750)', 'ZMYM3 (9203)', 'ZMYM4 (9202)', 'ZMYM5 (9205)', 'ZMYM6 (9204)', 'ZMYND10 (51364)', 'ZMYND11 (10771)', 'ZMYND12 (84217)', 'ZMYND15 (84225)', 'ZMYND19 (116225)', 'ZMYND8 (23613)', 'ZNF10 (7556)', 'ZNF100 (163227)', 'ZNF101 (94039)', 'ZNF106 (64397)', 'ZNF107 (51427)', 'ZNF112 (7771)', 'ZNF114 (163071)', 'ZNF12 (7559)', 'ZNF121 (7675)', 'ZNF124 (7678)', 'ZNF131 (7690)', 'ZNF132 (7691)', 'ZNF133 (7692)', 'ZNF134 (7693)', 'ZNF135 (7694)', 'ZNF136 (7695)', 'ZNF138 (7697)', 'ZNF14 (7561)', 'ZNF140 (7699)', 'ZNF141 (7700)', 'ZNF142 (7701)', 'ZNF143 (7702)', 'ZNF146 (7705)', 'ZNF148 (7707)', 'ZNF154 (7710)', 'ZNF155 (7711)', 'ZNF157 (7712)', 'ZNF16 (7564)', 'ZNF160 (90338)', 'ZNF165 (7718)', 'ZNF169 (169841)', 'ZNF17 (7565)', 'ZNF174 (7727)', 'ZNF175 (7728)', 'ZNF177 (7730)', 'ZNF18 (7566)', 'ZNF180 (7733)', 'ZNF181 (339318)', 'ZNF182 (7569)', 'ZNF184 (7738)', 'ZNF185 (7739)', 'ZNF189 (7743)', 'ZNF19 (7567)', 'ZNF195 (7748)', 'ZNF2 (7549)', 'ZNF20 (7568)', 'ZNF200 (7752)', 'ZNF202 (7753)', 'ZNF205 (7755)', 'ZNF207 (7756)', 'ZNF208 (7757)', 'ZNF211 (10520)', 'ZNF212 (7988)', 'ZNF213 (7760)', 'ZNF214 (7761)', 'ZNF215 (7762)', 'ZNF217 (7764)', 'ZNF219 (51222)', 'ZNF22 (7570)', 'ZNF221 (7638)', 'ZNF222 (7673)', 'ZNF223 (7766)', 'ZNF224 (7767)', 'ZNF225 (7768)', 'ZNF226 (7769)', 'ZNF227 (7770)', 'ZNF229 (7772)', 'ZNF23 (7571)', 'ZNF230 (7773)', 'ZNF232 (7775)', 'ZNF233 (353355)', 'ZNF234 (10780)', 'ZNF235 (9310)', 'ZNF236 (7776)', 'ZNF239 (8187)', 'ZNF24 (7572)', 'ZNF248 (57209)', 'ZNF25 (219749)', 'ZNF250 (58500)', 'ZNF251 (90987)', 'ZNF253 (56242)', 'ZNF254 (9534)', 'ZNF256 (10172)', 'ZNF257 (113835)', 'ZNF26 (7574)', 'ZNF260 (339324)', 'ZNF263 (10127)', 'ZNF264 (9422)', 'ZNF266 (10781)', 'ZNF267 (10308)', 'ZNF268 (10795)', 'ZNF273 (10793)', 'ZNF276 (92822)', 'ZNF277 (11179)', 'ZNF28 (7576)', 'ZNF280A (129025)', 'ZNF280B (140883)', 'ZNF280C (55609)', 'ZNF280D (54816)', 'ZNF281 (23528)', 'ZNF282 (8427)', 'ZNF283 (284349)', 'ZNF284 (342909)', 'ZNF285 (26974)', 'ZNF286A (57335)', 'ZNF287 (57336)', 'ZNF292 (23036)', 'ZNF296 (162979)', 'ZNF3 (7551)', 'ZNF30 (90075)', 'ZNF300 (91975)', 'ZNF302 (55900)', 'ZNF304 (57343)', 'ZNF311 (282890)', 'ZNF317 (57693)', 'ZNF318 (24149)', 'ZNF319 (57567)', 'ZNF32 (7580)', 'ZNF320 (162967)', 'ZNF322 (79692)', 'ZNF324 (25799)', 'ZNF324B (388569)', 'ZNF326 (284695)', 'ZNF329 (79673)', 'ZNF330 (27309)', 'ZNF331 (55422)', 'ZNF333 (84449)', 'ZNF334 (55713)', 'ZNF335 (63925)', 'ZNF337 (26152)', 'ZNF33A (7581)', 'ZNF33B (7582)', 'ZNF34 (80778)', 'ZNF341 (84905)', 'ZNF343 (79175)', 'ZNF345 (25850)', 'ZNF346 (23567)', 'ZNF347 (84671)', 'ZNF35 (7584)', 'ZNF350 (59348)', 'ZNF354A (6940)', 'ZNF354B (117608)', 'ZNF354C (30832)', 'ZNF358 (140467)', 'ZNF362 (149076)', 'ZNF365 (22891)', 'ZNF366 (167465)', 'ZNF367 (195828)', 'ZNF37A (7587)', 'ZNF382 (84911)', 'ZNF383 (163087)', 'ZNF384 (171017)', 'ZNF385A (25946)', 'ZNF385B (151126)', 'ZNF385D (79750)', 'ZNF391 (346157)', 'ZNF394 (84124)', 'ZNF395 (55893)', 'ZNF396 (252884)', 'ZNF397 (84307)', 'ZNF398 (57541)', 'ZNF404 (342908)', 'ZNF407 (55628)', 'ZNF408 (79797)', 'ZNF41 (7592)', 'ZNF410 (57862)', 'ZNF414 (84330)', 'ZNF415 (55786)', 'ZNF416 (55659)', 'ZNF417 (147687)', 'ZNF418 (147686)', 'ZNF419 (79744)', 'ZNF420 (147923)', 'ZNF423 (23090)', 'ZNF425 (155054)', 'ZNF426 (79088)', 'ZNF428 (126299)', 'ZNF429 (353088)', 'ZNF43 (7594)', 'ZNF430 (80264)', 'ZNF432 (9668)', 'ZNF433 (163059)', 'ZNF436 (80818)', 'ZNF438 (220929)', 'ZNF439 (90594)', 'ZNF44 (51710)', 'ZNF440 (126070)', 'ZNF441 (126068)', 'ZNF442 (79973)', 'ZNF443 (10224)', 'ZNF444 (55311)', 'ZNF445 (353274)', 'ZNF446 (55663)', 'ZNF449 (203523)', 'ZNF45 (7596)', 'ZNF451 (26036)', 'ZNF454 (285676)', 'ZNF460 (10794)', 'ZNF461 (92283)', 'ZNF462 (58499)', 'ZNF467 (168544)', 'ZNF468 (90333)', 'ZNF469 (84627)', 'ZNF470 (388566)', 'ZNF471 (57573)', 'ZNF473 (25888)', 'ZNF474 (133923)', 'ZNF479 (90827)', 'ZNF48 (197407)', 'ZNF480 (147657)', 'ZNF483 (158399)', 'ZNF484 (83744)', 'ZNF485 (220992)', 'ZNF486 (90649)', 'ZNF488 (118738)', 'ZNF490 (57474)', 'ZNF491 (126069)', 'ZNF492 (57615)', 'ZNF493 (284443)', 'ZNF496 (84838)', 'ZNF497 (162968)', 'ZNF500 (26048)', 'ZNF501 (115560)', 'ZNF502 (91392)', 'ZNF503 (84858)', 'ZNF506 (440515)', 'ZNF507 (22847)', 'ZNF510 (22869)', 'ZNF511 (118472)', 'ZNF512 (84450)', 'ZNF512B (57473)', 'ZNF513 (130557)', 'ZNF514 (84874)', 'ZNF517 (340385)', 'ZNF518B (85460)', 'ZNF519 (162655)', 'ZNF521 (25925)', 'ZNF524 (147807)', 'ZNF526 (116115)', 'ZNF527 (84503)', 'ZNF528 (84436)', 'ZNF529 (57711)', 'ZNF530 (348327)', 'ZNF532 (55205)', 'ZNF534 (147658)', 'ZNF536 (9745)', 'ZNF540 (163255)', 'ZNF541 (84215)', 'ZNF543 (125919)', 'ZNF544 (27300)', 'ZNF546 (339327)', 'ZNF547 (284306)', 'ZNF548 (147694)', 'ZNF549 (256051)', 'ZNF550 (162972)', 'ZNF551 (90233)', 'ZNF552 (79818)', 'ZNF554 (115196)', 'ZNF555 (148254)', 'ZNF556 (80032)', 'ZNF557 (79230)', 'ZNF558 (148156)', 'ZNF559 (84527)', 'ZNF560 (147741)', 'ZNF561 (93134)', 'ZNF562 (54811)', 'ZNF563 (147837)', 'ZNF564 (163050)', 'ZNF565 (147929)', 'ZNF566 (84924)', 'ZNF567 (163081)', 'ZNF568 (374900)', 'ZNF569 (148266)', 'ZNF57 (126295)', 'ZNF570 (148268)', 'ZNF571 (51276)', 'ZNF572 (137209)', 'ZNF573 (126231)', 'ZNF574 (64763)', 'ZNF575 (284346)', 'ZNF576 (79177)', 'ZNF577 (84765)', 'ZNF578 (147660)', 'ZNF579 (163033)', 'ZNF580 (51157)', 'ZNF581 (51545)', 'ZNF582 (147948)', 'ZNF583 (147949)', 'ZNF584 (201514)', 'ZNF585A (199704)', 'ZNF585B (92285)', 'ZNF586 (54807)', 'ZNF587 (84914)', 'ZNF587B (100293516)', 'ZNF589 (51385)', 'ZNF592 (9640)', 'ZNF593 (51042)', 'ZNF594 (84622)', 'ZNF596 (169270)', 'ZNF597 (146434)', 'ZNF599 (148103)', 'ZNF600 (162966)', 'ZNF605 (100289635)', 'ZNF606 (80095)', 'ZNF607 (84775)', 'ZNF608 (57507)', 'ZNF609 (23060)', 'ZNF610 (162963)', 'ZNF611 (81856)', 'ZNF613 (79898)', 'ZNF614 (80110)', 'ZNF615 (284370)', 'ZNF616 (90317)', 'ZNF618 (114991)', 'ZNF619 (285267)', 'ZNF620 (253639)', 'ZNF621 (285268)', 'ZNF622 (90441)', 'ZNF623 (9831)', 'ZNF624 (57547)', 'ZNF625 (90589)', 'ZNF626 (199777)', 'ZNF627 (199692)', 'ZNF628 (89887)', 'ZNF629 (23361)', 'ZNF630 (57232)', 'ZNF638 (27332)', 'ZNF639 (51193)', 'ZNF641 (121274)', 'ZNF644 (84146)', 'ZNF646 (9726)', 'ZNF648 (127665)', 'ZNF649 (65251)', 'ZNF652 (22834)', 'ZNF653 (115950)', 'ZNF654 (55279)', 'ZNF655 (79027)', 'ZNF658 (26149)', 'ZNF660 (285349)', 'ZNF662 (389114)', 'ZNF664 (144348)', 'ZNF665 (79788)', 'ZNF667 (63934)', 'ZNF668 (79759)', 'ZNF669 (79862)', 'ZNF670 (93474)', 'ZNF671 (79891)', 'ZNF672 (79894)', 'ZNF674 (641339)', 'ZNF675 (171392)', 'ZNF676 (163223)', 'ZNF677 (342926)', 'ZNF679 (168417)', 'ZNF680 (340252)', 'ZNF681 (148213)', 'ZNF682 (91120)', 'ZNF683 (257101)', 'ZNF684 (127396)', 'ZNF687 (57592)', 'ZNF688 (146542)', 'ZNF689 (115509)', 'ZNF69 (7620)', 'ZNF691 (51058)', 'ZNF692 (55657)', 'ZNF695 (57116)', 'ZNF696 (79943)', 'ZNF697 (90874)', 'ZNF699 (374879)', 'ZNF7 (7553)', 'ZNF70 (7621)', 'ZNF700 (90592)', 'ZNF701 (55762)', 'ZNF703 (80139)', 'ZNF704 (619279)', 'ZNF705B (100132396)', 'ZNF705G (100131980)', 'ZNF706 (51123)', 'ZNF707 (286075)', 'ZNF708 (7562)', 'ZNF709 (163051)', 'ZNF71 (58491)', 'ZNF710 (374655)', 'ZNF711 (7552)', 'ZNF713 (349075)', 'ZNF714 (148206)', 'ZNF716 (441234)', 'ZNF721 (170960)', 'ZNF726 (730087)', 'ZNF727 (442319)', 'ZNF728 (388523)', 'ZNF729 (100287226)', 'ZNF730 (100129543)', 'ZNF732 (654254)', 'ZNF736 (728927)', 'ZNF737 (100129842)', 'ZNF74 (7625)', 'ZNF740 (283337)', 'ZNF746 (155061)', 'ZNF747 (65988)', 'ZNF749 (388567)', 'ZNF750 (79755)', 'ZNF75A (7627)', 'ZNF75D (7626)', 'ZNF76 (7629)', 'ZNF763 (284390)', 'ZNF764 (92595)', 'ZNF765 (91661)', 'ZNF766 (90321)', 'ZNF768 (79724)', 'ZNF77 (58492)', 'ZNF770 (54989)', 'ZNF771 (51333)', 'ZNF772 (400720)', 'ZNF773 (374928)', 'ZNF774 (342132)', 'ZNF775 (285971)', 'ZNF776 (284309)', 'ZNF777 (27153)', 'ZNF778 (197320)', 'ZNF780A (284323)', 'ZNF780B (163131)', 'ZNF782 (158431)', 'ZNF783 (100289678)', 'ZNF784 (147808)', 'ZNF785 (146540)', 'ZNF786 (136051)', 'ZNF787 (126208)', 'ZNF789 (285989)', 'ZNF79 (7633)', 'ZNF790 (388536)', 'ZNF791 (163049)', 'ZNF792 (126375)', 'ZNF793 (390927)', 'ZNF799 (90576)', 'ZNF8 (7554)', 'ZNF80 (7634)', 'ZNF800 (168850)', 'ZNF804A (91752)', 'ZNF804B (219578)', 'ZNF805 (390980)', 'ZNF808 (388558)', 'ZNF81 (347344)', 'ZNF813 (126017)', 'ZNF814 (730051)', 'ZNF816 (125893)', 'ZNF821 (55565)', 'ZNF823 (55552)', 'ZNF827 (152485)', 'ZNF829 (374899)', 'ZNF83 (55769)', 'ZNF830 (91603)', 'ZNF831 (128611)', 'ZNF835 (90485)', 'ZNF836 (162962)', 'ZNF837 (116412)', 'ZNF839 (55778)', 'ZNF84 (7637)', 'ZNF841 (284371)', 'ZNF843 (283933)', 'ZNF844 (284391)', 'ZNF845 (91664)', 'ZNF846 (162993)', 'ZNF85 (7639)', 'ZNF850 (342892)', 'ZNF853 (54753)', 'ZNF860 (344787)', 'ZNF862 (643641)', 'ZNF865 (100507290)', 'ZNF875 (284459)', 'ZNF878 (729747)', 'ZNF879 (345462)', 'ZNF880 (400713)', 'ZNF891 (101060200)', 'ZNF90 (7643)', 'ZNF91 (7644)', 'ZNF92 (168374)', 'ZNF93 (81931)', 'ZNF98 (148198)', 'ZNF99 (7652)', 'ZNFX1 (57169)', 'ZNG1A (55871)', 'ZNG1B (150472)', 'ZNHIT1 (10467)', 'ZNHIT2 (741)', 'ZNHIT3 (9326)', 'ZNHIT6 (54680)', 'ZNRD2 (10534)', 'ZNRF1 (84937)', 'ZNRF2 (223082)', 'ZNRF3 (84133)', 'ZNRF4 (148066)', 'ZP1 (22917)', 'ZP2 (7783)', 'ZP3 (7784)', 'ZP4 (57829)', 'ZPBP (11055)', 'ZPBP2 (124626)', 'ZPLD1 (131368)', 'ZPR1 (8882)', 'ZRANB1 (54764)', 'ZRANB2 (9406)', 'ZRANB3 (84083)', 'ZRSR2 (8233)', 'ZSCAN1 (284312)', 'ZSCAN10 (84891)', 'ZSCAN16 (80345)', 'ZSCAN18 (65982)', 'ZSCAN2 (54993)', 'ZSCAN20 (7579)', 'ZSCAN21 (7589)', 'ZSCAN22 (342945)', 'ZSCAN23 (222696)', 'ZSCAN25 (221785)', 'ZSCAN29 (146050)', 'ZSCAN30 (100101467)', 'ZSCAN31 (64288)', 'ZSCAN32 (54925)', 'ZSCAN4 (201516)', 'ZSCAN5A (79149)', 'ZSCAN5B (342933)', 'ZSCAN9 (7746)', 'ZSWIM1 (90204)', 'ZSWIM2 (151112)', 'ZSWIM3 (140831)', 'ZSWIM4 (65249)', 'ZSWIM5 (57643)', 'ZSWIM6 (57688)', 'ZSWIM7 (125150)', 'ZSWIM8 (23053)', 'ZUP1 (221302)', 'ZW10 (9183)', 'ZWILCH (55055)', 'ZWINT (11130)', 'ZXDA (7789)', 'ZXDB (158586)', 'ZXDC (79364)', 'ZYG11A (440590)', 'ZYG11B (79699)', 'ZYX (7791)', 'ZZEF1 (23140)', 'ZZZ3 (26009)']\n"
     ]
    }
   ],
   "source": [
    "# get the column names of the raw_crispr_df aside from the first column and convert this to a list\n",
    "raw_crispr_df_columns = raw_crispr_df.columns[1:].tolist()\n",
    "print(raw_crispr_df_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Coalign the samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 330,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['DMS53_LUNG', 'SW1116_LARGE_INTESTINE',\n",
      "       'P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE',\n",
      "       'HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE', 'UMUC3_URINARY_TRACT',\n",
      "       'HOS_BONE', 'AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE', 'RVH421_SKIN',\n",
      "       'NCIH1184_LUNG', 'HCC2157_BREAST',\n",
      "       ...\n",
      "       'JHUEM7_ENDOMETRIUM', 'OE21_OESOPHAGUS',\n",
      "       'MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE', 'HOP62_LUNG', 'EKVX_LUNG',\n",
      "       'UO31_KIDNEY', 'SF268_CENTRAL_NERVOUS_SYSTEM',\n",
      "       'SF539_CENTRAL_NERVOUS_SYSTEM', 'SNB75_CENTRAL_NERVOUS_SYSTEM',\n",
      "       'HOP92_LUNG'],\n",
      "      dtype='object', length=843)\n",
      "Index(['ACH-000628', 'ACH-000974', 'ACH-003071', 'ACH-001338', 'ACH-001608',\n",
      "       'ACH-000278', 'ACH-001350', 'ACH-000150', 'ACH-001132', 'ACH-000324',\n",
      "       ...\n",
      "       'ACH-000236', 'ACH-002177', 'ACH-002163', 'ACH-002316', 'ACH-001039',\n",
      "       'ACH-001389', 'ACH-001835', 'ACH-000296', 'ACH-001971', 'ACH-002276'],\n",
      "      dtype='object', length=1929)\n",
      "Index(['ACH-000873', 'ACH-000860', 'ACH-000439', 'ACH-000318', 'ACH-001142',\n",
      "       'ACH-000789', 'ACH-000544', 'ACH-000531', 'ACH-000695', 'ACH-000399',\n",
      "       ...\n",
      "       'ACH-002996', 'ACH-003061', 'ACH-001384', 'ACH-001700', 'ACH-002470',\n",
      "       'ACH-001843', 'ACH-002074', 'ACH-001164', 'ACH-003180', 'ACH-001979'],\n",
      "      dtype='object', length=1673)\n",
      "Index(['MDAMB468_BREAST_TenPx01', 'SH4_SKIN_TenPx01', 'AU565_BREAST_TenPx01',\n",
      "       'KMRC1_KIDNEY_TenPx01', 'CAL51_BREAST_TenPx01', 'RPMI7951_SKIN_TenPx01',\n",
      "       'RERFLCMS_LUNG_TenPx01', 'IGR37_SKIN_TenPx01', 'VMRCRCW_KIDNEY_TenPx01',\n",
      "       'HEP3B217_LIVER_TenPx02',\n",
      "       ...\n",
      "       'NCIH2030_LUNG_TenPx18', '22RV1_PROSTATE_TenPx24',\n",
      "       'A172_CENTRAL_NERVOUS_SYSTEM_TenPx24', 'BT20_BREAST_TenPx24',\n",
      "       'CALU6_LUNG_TenPx24', 'FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24',\n",
      "       'KP4_PANCREAS_TenPx24',\n",
      "       'MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24',\n",
      "       'OVCAR8_OVARY_TenPx24',\n",
      "       'THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24'],\n",
      "      dtype='object', length=378)\n",
      "['ACH-000878', 'ACH-000567', 'ACH-000617', 'ACH-002091', 'ACH-001365', 'ACH-001125', 'ACH-000298', 'ACH-000219', 'ACH-000335', 'ACH-002117', 'ACH-000767', 'ACH-002262', 'ACH-002152', 'ACH-002171', 'ACH-002161', 'ACH-002176', 'ACH-000365', 'ACH-000655', 'ACH-000876', 'ACH-001024', 'ACH-000072', 'ACH-000893', 'ACH-002124', 'ACH-000521', 'ACH-000641', 'ACH-000688', 'ACH-002263', 'ACH-000288', 'ACH-000015', 'ACH-002173', 'ACH-002129', 'ACH-001638', 'ACH-000145', 'ACH-000751', 'ACH-002267', 'ACH-000149', 'ACH-000577', 'ACH-000002', 'ACH-002092', 'ACH-002199', 'ACH-000795', 'ACH-001118', 'ACH-002252', 'ACH-002286', 'ACH-000099', 'ACH-002309', 'ACH-000819', 'ACH-000587', 'ACH-000091', 'ACH-000824', 'ACH-002137', 'ACH-000804', 'ACH-000605', 'ACH-002282', 'ACH-002165', 'ACH-001182', 'ACH-000148', 'ACH-002237', 'ACH-001338', 'ACH-000565', 'ACH-000440', 'ACH-000545', 'ACH-000238', 'ACH-000790', 'ACH-002258', 'ACH-000841', 'ACH-000574', 'ACH-002126', 'ACH-000581', 'ACH-000309', 'ACH-002095', 'ACH-000568', 'ACH-002290', 'ACH-000572', 'ACH-000525', 'ACH-000259', 'ACH-001127', 'ACH-002113', 'ACH-001136', 'ACH-000443', 'ACH-000573', 'ACH-000058', 'ACH-000947', 'ACH-000555', 'ACH-002114', 'ACH-000361', 'ACH-000704', 'ACH-002306', 'ACH-000196', 'ACH-000336', 'ACH-000935', 'ACH-000038', 'ACH-001023', 'ACH-001605', 'ACH-000186', 'ACH-000416', 'ACH-000470', 'ACH-001704', 'ACH-002202', 'ACH-000607', 'ACH-002277', 'ACH-000596', 'ACH-000716', 'ACH-000534', 'ACH-000422', 'ACH-001599', 'ACH-001617', 'ACH-000827', 'ACH-000902', 'ACH-002118', 'ACH-000394', 'ACH-000524', 'ACH-000560', 'ACH-000496', 'ACH-000101', 'ACH-000308', 'ACH-000428', 'ACH-000879', 'ACH-000424', 'ACH-000958', 'ACH-002264', 'ACH-000098', 'ACH-000362', 'ACH-000136', 'ACH-000338', 'ACH-002244', 'ACH-000697', 'ACH-000391', 'ACH-000969', 'ACH-000706', 'ACH-000983', 'ACH-001199', 'ACH-000341', 'ACH-000711', 'ACH-000838', 'ACH-000684', 'ACH-000950', 'ACH-000825', 'ACH-001544', 'ACH-002218', 'ACH-000987', 'ACH-000763', 'ACH-000563', 'ACH-002123', 'ACH-000067', 'ACH-000725', 'ACH-000865', 'ACH-002265', 'ACH-002146', 'ACH-000989', 'ACH-000124', 'ACH-000094', 'ACH-000636', 'ACH-002208', 'ACH-000622', 'ACH-002183', 'ACH-000880', 'ACH-000400', 'ACH-001350', 'ACH-000535', 'ACH-000161', 'ACH-000680', 'ACH-000910', 'ACH-000442', 'ACH-000754', 'ACH-000120', 'ACH-002317', 'ACH-002134', 'ACH-000427', 'ACH-002212', 'ACH-000282', 'ACH-002314', 'ACH-000662', 'ACH-000425', 'ACH-001519', 'ACH-000670', 'ACH-000757', 'ACH-000569', 'ACH-000116', 'ACH-001360', 'ACH-000359', 'ACH-002315', 'ACH-000190', 'ACH-001203', 'ACH-000008', 'ACH-002135', 'ACH-002200', 'ACH-002266', 'ACH-000830', 'ACH-000586', 'ACH-000613', 'ACH-002109', 'ACH-000483', 'ACH-001355', 'ACH-000327', 'ACH-000366', 'ACH-000696', 'ACH-002224', 'ACH-000846', 'ACH-000113', 'ACH-002261', 'ACH-000226', 'ACH-000271', 'ACH-000679', 'ACH-002174', 'ACH-000473', 'ACH-002211', 'ACH-000431', 'ACH-000823', 'ACH-000236', 'ACH-002207', 'ACH-001016', 'ACH-000489', 'ACH-002180', 'ACH-000775', 'ACH-000566', 'ACH-000747', 'ACH-000081', 'ACH-000776', 'ACH-000863', 'ACH-001703', 'ACH-000766', 'ACH-000675', 'ACH-000724', 'ACH-000423', 'ACH-000548', 'ACH-000421', 'ACH-000248', 'ACH-002089', 'ACH-000040', 'ACH-000408', 'ACH-000001', 'ACH-000376', 'ACH-000864', 'ACH-002247', 'ACH-000075', 'ACH-000380', 'ACH-002101', 'ACH-000892', 'ACH-000510', 'ACH-000225', 'ACH-000056', 'ACH-000866', 'ACH-000974', 'ACH-000432', 'ACH-000609', 'ACH-002291', 'ACH-000371', 'ACH-000903', 'ACH-002303', 'ACH-000287', 'ACH-000778', 'ACH-000501', 'ACH-001193', 'ACH-002110', 'ACH-000515', 'ACH-000968', 'ACH-002107', 'ACH-000348', 'ACH-000337', 'ACH-002298', 'ACH-001641', 'ACH-001328', 'ACH-000475', 'ACH-000349', 'ACH-000295', 'ACH-002108', 'ACH-000809', 'ACH-000449', 'ACH-001489', 'ACH-000508', 'ACH-000171', 'ACH-001021', 'ACH-000178', 'ACH-002242', 'ACH-002120', 'ACH-002145', 'ACH-000106', 'ACH-002191', 'ACH-000451', 'ACH-002132', 'ACH-000869', 'ACH-000759', 'ACH-002251', 'ACH-000621', 'ACH-000399', 'ACH-002133', 'ACH-002160', 'ACH-000474', 'ACH-000595', 'ACH-000664', 'ACH-000745', 'ACH-000059', 'ACH-002220', 'ACH-002153', 'ACH-000638', 'ACH-000648', 'ACH-000660', 'ACH-000121', 'ACH-002190', 'ACH-002305', 'ACH-000652', 'ACH-000934', 'ACH-000105', 'ACH-002245', 'ACH-000228', 'ACH-000319', 'ACH-000937', 'ACH-000441', 'ACH-000452', 'ACH-000417', 'ACH-000516', 'ACH-000379', 'ACH-001358', 'ACH-002281', 'ACH-002203', 'ACH-000476', 'ACH-002273', 'ACH-001333', 'ACH-000469', 'ACH-000433', 'ACH-000633', 'ACH-001373', 'ACH-000221', 'ACH-000187', 'ACH-000183', 'ACH-000949', 'ACH-000756', 'ACH-000018', 'ACH-000012', 'ACH-001039', 'ACH-002167', 'ACH-000843', 'ACH-000047', 'ACH-000146', 'ACH-002210', 'ACH-000847', 'ACH-001549', 'ACH-000254', 'ACH-000343', 'ACH-001702', 'ACH-002098', 'ACH-002154', 'ACH-000334', 'ACH-002310', 'ACH-000957', 'ACH-001064', 'ACH-000821', 'ACH-000914', 'ACH-000484', 'ACH-000111', 'ACH-000740', 'ACH-001442', 'ACH-002293', 'ACH-002169', 'ACH-000647', 'ACH-000538', 'ACH-002241', 'ACH-000584', 'ACH-001653', 'ACH-000304', 'ACH-000811', 'ACH-001443', 'ACH-002198', 'ACH-000087', 'ACH-000678', 'ACH-001131', 'ACH-000159', 'ACH-000019', 'ACH-001674', 'ACH-000703', 'ACH-000939', 'ACH-002105', 'ACH-000816', 'ACH-001525', 'ACH-000559', 'ACH-000554', 'ACH-000752', 'ACH-000730', 'ACH-000867', 'ACH-000674', 'ACH-000665', 'ACH-000315', 'ACH-000918', 'ACH-001359', 'ACH-000886', 'ACH-002232', 'ACH-000392', 'ACH-000045', 'ACH-000054', 'ACH-000439', 'ACH-002106', 'ACH-000644', 'ACH-000201', 'ACH-000360', 'ACH-000603', 'ACH-000979', 'ACH-001418', 'ACH-000583', 'ACH-002104', 'ACH-000233', 'ACH-000977', 'ACH-000881', 'ACH-000269', 'ACH-002159', 'ACH-000245', 'ACH-000073', 'ACH-000465', 'ACH-000021', 'ACH-000096', 'ACH-001709', 'ACH-000363', 'ACH-000789', 'ACH-000702', 'ACH-000938', 'ACH-000639', 'ACH-000875', 'ACH-002139', 'ACH-002243', 'ACH-000885', 'ACH-000556', 'ACH-002090', 'ACH-002233', 'ACH-000103', 'ACH-000155', 'ACH-000856', 'ACH-000326', 'ACH-002302', 'ACH-000657', 'ACH-002268', 'ACH-001530', 'ACH-000667', 'ACH-000132', 'ACH-000624', 'ACH-000713', 'ACH-000905', 'ACH-000530', 'ACH-001332', 'ACH-002162', 'ACH-002285', 'ACH-002312', 'ACH-000211', 'ACH-001494', 'ACH-000265', 'ACH-001307', 'ACH-000456', 'ACH-000227', 'ACH-000694', 'ACH-000486', 'ACH-002119', 'ACH-000978', 'ACH-001528', 'ACH-000200', 'ACH-002187', 'ACH-000060', 'ACH-000046', 'ACH-000848', 'ACH-000506', 'ACH-002275', 'ACH-000305', 'ACH-000625', 'ACH-000193', 'ACH-001715', 'ACH-002240', 'ACH-000970', 'ACH-000322', 'ACH-000480', 'ACH-000267', 'ACH-000527', 'ACH-000860', 'ACH-000207', 'ACH-000389', 'ACH-000142', 'ACH-000378', 'ACH-000532', 'ACH-000798', 'ACH-001496', 'ACH-000352', 'ACH-000112', 'ACH-000437', 'ACH-000784', 'ACH-000835', 'ACH-000074', 'ACH-000290', 'ACH-002142', 'ACH-000650', 'ACH-002196', 'ACH-002172', 'ACH-000151', 'ACH-000030', 'ACH-000102', 'ACH-002194', 'ACH-000357', 'ACH-001680', 'ACH-001049', 'ACH-000615', 'ACH-002294', 'ACH-000894', 'ACH-000364', 'ACH-002229', 'ACH-001497', 'ACH-001363', 'ACH-001233', 'ACH-000346', 'ACH-000927', 'ACH-000383', 'ACH-002100', 'ACH-001650', 'ACH-000176', 'ACH-000861', 'ACH-002216', 'ACH-001368', 'ACH-000722', 'ACH-001642', 'ACH-000920', 'ACH-002178', 'ACH-002283', 'ACH-000579', 'ACH-000137', 'ACH-000859', 'ACH-002127', 'ACH-002221', 'ACH-002192', 'ACH-000873', 'ACH-001500', 'ACH-000213', 'ACH-000988', 'ACH-000793', 'ACH-000998', 'ACH-001126', 'ACH-000276', 'ACH-000906', 'ACH-000973', 'ACH-000689', 'ACH-000518', 'ACH-000070', 'ACH-002163', 'ACH-000805', 'ACH-000329', 'ACH-000547', 'ACH-000810', 'ACH-001417', 'ACH-002215', 'ACH-000611', 'ACH-000589', 'ACH-000139', 'ACH-000482', 'ACH-000129', 'ACH-002158', 'ACH-000599', 'ACH-002280', 'ACH-000420', 'ACH-001089', 'ACH-001385', 'ACH-002130', 'ACH-000755', 'ACH-001334', 'ACH-000922', 'ACH-000744', 'ACH-000812', 'ACH-002121', 'ACH-000614', 'ACH-000546', 'ACH-000108', 'ACH-000849', 'ACH-000818', 'ACH-002209', 'ACH-000948', 'ACH-000372', 'ACH-002230', 'ACH-001137', 'ACH-000832', 'ACH-000890', 'ACH-001578', 'ACH-000963', 'ACH-000770', 'ACH-000980', 'ACH-002138', 'ACH-000197', 'ACH-002308', 'ACH-001081', 'ACH-000115', 'ACH-000356', 'ACH-000144', 'ACH-002278', 'ACH-000303', 'ACH-000320', 'ACH-002116', 'ACH-001526', 'ACH-000330', 'ACH-001090', 'ACH-000882', 'ACH-000677', 'ACH-000061', 'ACH-002269', 'ACH-001655', 'ACH-000375', 'ACH-000270', 'ACH-001654', 'ACH-000718', 'ACH-000871', 'ACH-001374', 'ACH-001336', 'ACH-002296', 'ACH-000828', 'ACH-000719', 'ACH-000802', 'ACH-000092', 'ACH-000999', 'ACH-000181', 'ACH-000971', 'ACH-000373', 'ACH-000772', 'ACH-001711', 'ACH-000395', 'ACH-000311', 'ACH-001341', 'ACH-002125', 'ACH-002228', 'ACH-000514', 'ACH-000273', 'ACH-001665', 'ACH-000561', 'ACH-002300', 'ACH-001364', 'ACH-000023', 'ACH-002270', 'ACH-000601', 'ACH-000258', 'ACH-000806', 'ACH-000457', 'ACH-001331', 'ACH-000157', 'ACH-000727', 'ACH-000078', 'ACH-000631', 'ACH-000616', 'ACH-000065', 'ACH-002307', 'ACH-000231', 'ACH-000826', 'ACH-001189', 'ACH-002102', 'ACH-000672', 'ACH-000472', 'ACH-000738', 'ACH-002164', 'ACH-000769', 'ACH-001517', 'ACH-000953', 'ACH-000962', 'ACH-000294', 'ACH-000858', 'ACH-000277', 'ACH-001591', 'ACH-001208', 'ACH-000222', 'ACH-002274', 'ACH-000007', 'ACH-000263', 'ACH-002276', 'ACH-000493', 'ACH-000926', 'ACH-000966', 'ACH-000386', 'ACH-002253', 'ACH-000976', 'ACH-001400', 'ACH-000936', 'ACH-002239', 'ACH-002249', 'ACH-000478', 'ACH-000114', 'ACH-000024', 'ACH-000930', 'ACH-000434', 'ACH-002259', 'ACH-000210', 'ACH-000985', 'ACH-000463', 'ACH-000118', 'ACH-000250', 'ACH-000232', 'ACH-000052', 'ACH-000082', 'ACH-000783', 'ACH-000429', 'ACH-002297', 'ACH-000090', 'ACH-000649', 'ACH-000733', 'ACH-002295', 'ACH-000628', 'ACH-000354', 'ACH-000162', 'ACH-000358', 'ACH-000260', 'ACH-000481', 'ACH-000350', 'ACH-000035', 'ACH-002311', 'ACH-000620', 'ACH-000189', 'ACH-000874', 'ACH-002301', 'ACH-002289', 'ACH-001402', 'ACH-000430', 'ACH-000669', 'ACH-002181', 'ACH-000857', 'ACH-000447', 'ACH-000911', 'ACH-002256', 'ACH-001345', 'ACH-000367', 'ACH-001529', 'ACH-000448', 'ACH-002150', 'ACH-002184', 'ACH-000153', 'ACH-000168', 'ACH-000244', 'ACH-000450', 'ACH-002166', 'ACH-001384', 'ACH-000042', 'ACH-000204', 'ACH-001498', 'ACH-002255', 'ACH-000243', 'ACH-002195', 'ACH-000020', 'ACH-002205', 'ACH-002111', 'ACH-002096', 'ACH-000786', 'ACH-002226', 'ACH-000610', 'ACH-002219', 'ACH-000138', 'ACH-001106', 'ACH-001075', 'ACH-000929', 'ACH-000995', 'ACH-002144', 'ACH-002206', 'ACH-000780', 'ACH-001344', 'ACH-000986', 'ACH-000956', 'ACH-000945', 'ACH-001626', 'ACH-002238', 'ACH-000750', 'ACH-000729', 'ACH-001306', 'ACH-002222', 'ACH-000585', 'ACH-000915', 'ACH-002147', 'ACH-000504', 'ACH-000743', 'ACH-001542', 'ACH-001490', 'ACH-002257', 'ACH-000135', 'ACH-000558', 'ACH-000768', 'ACH-000174', 'ACH-000794', 'ACH-000787', 'ACH-000188', 'ACH-001190', 'ACH-000169', 'ACH-002316', 'ACH-000817', 'ACH-000637', 'ACH-000396', 'ACH-000822', 'ACH-000803', 'ACH-002151', 'ACH-000055', 'ACH-002227', 'ACH-000932', 'ACH-000192', 'ACH-001738', 'ACH-001606', 'ACH-000749', 'ACH-000104', 'ACH-000698', 'ACH-000666', 'ACH-001086', 'ACH-001274', 'ACH-000351', 'ACH-002235', 'ACH-000640', 'ACH-001630', 'ACH-000536', 'ACH-000353', 'ACH-000053', 'ACH-000951', 'ACH-000924', 'ACH-000900', 'ACH-000681', 'ACH-002225', 'ACH-000714', 'ACH-001401', 'ACH-002149', 'ACH-000840', 'ACH-000792', 'ACH-000301', 'ACH-002177', 'ACH-002260', 'ACH-000522', 'ACH-000852', 'ACH-000965', 'ACH-002112', 'ACH-001639', 'ACH-002217', 'ACH-000883', 'ACH-000004', 'ACH-000048', 'ACH-002140', 'ACH-002143', 'ACH-000369', 'ACH-000318', 'ACH-001065', 'ACH-001625', 'ACH-000100', 'ACH-000943', 'ACH-002094', 'ACH-001398', 'ACH-001138', 'ACH-000955', 'ACH-002128', 'ACH-000781', 'ACH-000050', 'ACH-000663', 'ACH-000940', 'ACH-000651', 'ACH-000627', 'ACH-000691', 'ACH-002141', 'ACH-000384', 'ACH-002097', 'ACH-000062', 'ACH-002188', 'ACH-000403', 'ACH-000896', 'ACH-000235', 'ACH-002185', 'ACH-002234', 'ACH-002250', 'ACH-000888', 'ACH-000374', 'ACH-000401', 'ACH-000347', 'ACH-002272', 'ACH-000960', 'ACH-000619', 'ACH-000693', 'ACH-002148', 'ACH-002170', 'ACH-000252', 'ACH-000717', 'ACH-002197', 'ACH-000944', 'ACH-000643', 'ACH-000653', 'ACH-002246', 'ACH-000997', 'ACH-000855', 'ACH-000223', 'ACH-000580', 'ACH-002182', 'ACH-000551', 'ACH-000249', 'ACH-002254', 'ACH-002186', 'ACH-001002', 'ACH-000981', 'ACH-000009', 'ACH-000626', 'ACH-000668', 'ACH-001151', 'ACH-000570', 'ACH-000107', 'ACH-000323', 'ACH-000281', 'ACH-001716', 'ACH-000345', 'ACH-000410', 'ACH-002131', 'ACH-000257', 'ACH-002279', 'ACH-000571', 'ACH-002157', 'ACH-002168', 'ACH-001453', 'ACH-002136', 'ACH-000815', 'ACH-000217', 'ACH-000355', 'ACH-000006', 'ACH-002236', 'ACH-002299', 'ACH-001063', 'ACH-000913', 'ACH-000381', 'ACH-002155', 'ACH-000699', 'ACH-000264', 'ACH-000488', 'ACH-000147', 'ACH-001227', 'ACH-000991', 'ACH-001362', 'ACH-000438', 'ACH-002304', 'ACH-002099', 'ACH-000491', 'ACH-001107', 'ACH-002156', 'ACH-002248', 'ACH-000800', 'ACH-000198', 'ACH-000814', 'ACH-000292', 'ACH-002271', 'ACH-000788', 'ACH-000552', 'ACH-000917', 'ACH-002231', 'ACH-000895', 'ACH-000720', 'ACH-000562', 'ACH-002189', 'ACH-000919', 'ACH-002115', 'ACH-000853', 'ACH-001399', 'ACH-000656', 'ACH-000544', 'ACH-000916', 'ACH-002214', 'ACH-002288', 'ACH-002284', 'ACH-000654', 'ACH-001383', 'ACH-002213', 'ACH-000748', 'ACH-000291', 'ACH-000293', 'ACH-002093', 'ACH-002179', 'ACH-000464', 'ACH-000247', 'ACH-000212', 'ACH-000242', 'ACH-001603', 'ACH-000411', 'ACH-002313', 'ACH-000191', 'ACH-000332', 'ACH-000117', 'ACH-002201', 'ACH-000218', 'ACH-000528', 'ACH-000505', 'ACH-000414', 'ACH-002292', 'ACH-002103']\n"
     ]
    }
   ],
   "source": [
    "methy_samples = final_merged_methy_df.columns[1:]\n",
    "gene_samples = final_merged_gene_df.columns[1:]\n",
    "transcript_samples = final_merged_transcript_df.columns[1:]\n",
    "protein_samples = final_merged_protein_df.columns[1:]\n",
    "drug_samples = list(set(list(final_drug_df['ARXSPAN_ID'])))\n",
    "\n",
    "# print all samples\n",
    "print(methy_samples)\n",
    "print(gene_samples)\n",
    "print(transcript_samples)\n",
    "print(protein_samples)\n",
    "print(drug_samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['MDAMB468_BREAST', 'SH4_SKIN', 'AU565_BREAST', 'KMRC1_KIDNEY', 'CAL51_BREAST', 'RPMI7951_SKIN', 'RERFLCMS_LUNG', 'IGR37_SKIN', 'VMRCRCW_KIDNEY', 'HEP3B217_LIVER', 'HEPG2_LIVER', 'SKHEP1_LIVER', 'WM2664_SKIN', 'KMRC20_KIDNEY', 'HS294T_SKIN', 'CAOV3_OVARY', 'CAL120_BREAST', 'JMSU1_URINARY', 'PATU8988T_PANCREAS', 'JIMT1_BREAST', 'COV362_OVARY', 'KPL1_BREAST', 'A375_SKIN', 'A2058_SKIN', 'SW620_LARGE', 'MDST8_LARGE', 'HUH1_LIVER', 'HCT116_LARGE', 'HLF_LIVER', 'HT29_LARGE', 'ABC1_LUNG', 'CAMA1_BREAST', 'RKO_LARGE', 'HT115_LARGE', 'JHH5_LIVER', 'SKMEL3_SKIN', 'NCIH358_LUNG', 'HT1197_URINARY', 'HUH7_LIVER', 'HGC27_STOMACH', 'DMS273_LUNG', 'NCIH1299_LUNG', 'SKMEL28_SKIN', 'CAKI2_KIDNEY', 'HT55_LARGE', 'NCIH1048_LUNG', 'HSC3_UPPER', 'SW403_LARGE', 'BT549_BREAST', 'HCC56_LARGE', 'CAL851_BREAST', 'MDAMB157_BREAST', 'SW48_LARGE', 'IGR39_SKIN', 'MCF7_BREAST', 'BICR6_UPPER', 'WM1799_SKIN', 'IPC298_SKIN', 'NCIH1693_LUNG', 'HCC1954_BREAST', 'TYKNU_OVARY', 'DMS114_LUNG', 'TE1_OESOPHAGUS', 'TCCPAN2_PANCREAS', 'LUDLU1_LUNG', 'MELJUSO_SKIN', 'T47D_BREAST', 'LS411N_LARGE', 'PC3_PROSTATE', 'KYSE450_OESOPHAGUS', 'RT112_URINARY', 'LS513_LARGE', 'HCC827_LUNG', 'NCIH2110_LUNG', 'MIAPACA2_PANCREAS', 'NCIH520_LUNG', 'SW480_LARGE', 'OE33_OESOPHAGUS', 'ZR751_BREAST', 'COLO679_SKIN', '786O_KIDNEY', 'HCC38_BREAST', 'QGP1_PANCREAS', 'HCC95_LUNG', 'SW948_LARGE', 'LK2_LUNG', 'OVCAR4_OVARY', 'COLO741_SKIN', 'NCIH1650_LUNG', 'UACC257_SKIN', 'NCIH1573_LUNG', 'A549_LUNG', 'SU8686_PANCREAS', 'TE14_OESOPHAGUS', 'NCIH2170_LUNG', 'HCC1500_BREAST', 'MDAMB436_BREAST', 'DV90_LUNG', 'SNU61_LARGE', 'SBC5_LUNG', 'HT1376_URINARY', 'CFPAC1_PANCREAS', 'NCIH3255_LUNG', 'NCIH1975_LUNG', 'COLO678_LARGE', 'SNU119_OVARY', 'CORL105_LUNG', 'HCC70_BREAST', 'AGS_STOMACH', 'HCC1143_BREAST', 'NCIH1666_LUNG', 'NCIH2291_LUNG', 'NCIH747_LARGE', 'NCIH838_LUNG', 'FUOV1_OVARY', 'RERFLCSQ1_LUNG', 'DETROIT562_UPPER', 'L33_PANCREAS', 'MKN45_STOMACH', 'MKN7_STOMACH', 'NCIH1793_LUNG', 'NCIH2126_LUNG', 'NCIH2228_LUNG', 'NCIH661_LUNG', 'SNUC2A_LARGE', 'TE6_OESOPHAGUS', 'IGROV1_OVARY', 'LCLC103H_LUNG', 'LOXIMVI_SKIN', 'MDAMB453_BREAST', 'NCIH1703_LUNG', 'NCIH196_LUNG', 'NCIH2009_LUNG', 'NCIH2286_LUNG', 'SNUC1_LARGE', 'COLO205_LARGE', 'CORL47_LUNG', 'HCC1187_BREAST', 'HUH6_LIVER', 'NCIH146_LUNG', 'SNU1_STOMACH', 'NCIH23_LUNG', 'OSRC2_KIDNEY', 'SW948_LARGE', 'CORL88_LUNG', 'HCC2218_BREAST', 'HCC44_LUNG', 'HSC4_UPPER', 'NCIH2172_LUNG', 'TCCSUP_URINARY', 'TE11_OESOPHAGUS', 'UACC62_SKIN', 'VCAP_PROSTATE', 'EFM19_BREAST', 'HUG1N_STOMACH', 'KYSE510_OESOPHAGUS', 'LU65_LUNG', 'NCIH460_LUNG', 'NCIH522_LUNG', 'PECAPJ34CLONEC12_UPPER', 'SKMEL30_SKIN', 'SW1990_PANCREAS', 'A673_BONE', 'MDAMB231_BREAST', 'HEYA8_OVARY', 'RH41_SOFT', 'KMS12BM_HAEMATOPOIETIC', 'LN229_CENTRAL', 'LS180_LARGE', 'MOLM13_HAEMATOPOIETIC', 'KARPAS299_HAEMATOPOIETIC', 'CAL120_BREAST', 'DU145_PROSTATE', 'G401_SOFT', 'IM95_STOMACH', 'JHUEM2_ENDOMETRIUM', 'KARPAS422_HAEMATOPOIETIC', 'KASUMI1_HAEMATOPOIETIC', 'HCC15_LUNG', 'U87MG_CENTRAL', 'ASPC1_PANCREAS', 'SNU719_STOMACH', 'CORL23_LUNG', 'F36P_HAEMATOPOIETIC', 'HEC50B_ENDOMETRIUM', 'KNS81_CENTRAL', 'OCIAML5_HAEMATOPOIETIC', 'TE10_OESOPHAGUS', 'WM115_SKIN', 'EOL1_HAEMATOPOIETIC', 'SW1573_LUNG', 'HEL9217_HAEMATOPOIETIC', 'K029AX_SKIN', 'JHOS2_OVARY', 'SNU685_ENDOMETRIUM', 'NIHOVCAR3_OVARY', 'SAOS2_BONE', 'TE4_OESOPHAGUS', 'JHH1_LIVER', 'SW1271_LUNG', 'NCIH226_LUNG', 'REC1_HAEMATOPOIETIC', 'HT1080_SOFT', 'KYSE30_OESOPHAGUS', 'U2OS_BONE', 'GB1_CENTRAL', 'IALM_LUNG', 'CMK_HAEMATOPOIETIC', 'COLO829_SKIN', 'NCIH647_LUNG', 'OV56_OVARY', 'RH30_SOFT', 'RPMI8226_HAEMATOPOIETIC', 'SJSA1_BONE', 'HEC6_ENDOMETRIUM', 'SW837_LARGE', 'CAKI1_KIDNEY', 'KP2_PANCREAS', 'KU1919_URINARY', 'MFE296_ENDOMETRIUM', 'NCIH292_LUNG', 'NCIH716_LARGE', 'NUDHL1_HAEMATOPOIETIC', 'OCILY3_HAEMATOPOIETIC', 'SKNMC_BONE', 'CHAGOK1_LUNG', 'EFM192A_BREAST', 'HEC265_ENDOMETRIUM', 'MOLM16_HAEMATOPOIETIC', 'PANC1_PANCREAS', 'SCC25_UPPER', 'RT4_URINARY', 'T24_URINARY', 'TF1_HAEMATOPOIETIC', 'A101D_SKIN', 'DAOY_CENTRAL', 'JM1_HAEMATOPOIETIC', 'KASUMI2_HAEMATOPOIETIC', 'KYM1_SOFT', 'KYSE410_OESOPHAGUS', 'CL34_LARGE', 'NCIH446_LUNG', 'SKNAS_AUTONOMIC', 'DANG_PANCREAS', 'KNS42_CENTRAL', 'HEC108_ENDOMETRIUM', 'JEKO1_HAEMATOPOIETIC', 'JURKAT_HAEMATOPOIETIC', 'MSTO211H_PLEURA', 'OCUM1_STOMACH', 'CCK81_LARGE', 'SNU449_LIVER', '8505C_THYROID', 'HCC1833_LUNG', 'PANC0403_PANCREAS', 'J82_URINARY', 'JHH4_LIVER', 'OVSAHO_OVARY', 'RL_HAEMATOPOIETIC', 'SNUC5_LARGE', 'SUDHL6_HAEMATOPOIETIC', 'SKLU1_LUNG', 'ISHIKAWAHERAKLIO02ER_ENDOMETRIUM', 'SNGM_ENDOMETRIUM', 'GAMG_CENTRAL', 'NCIH929_HAEMATOPOIETIC', 'PANC0203_PANCREAS', 'RCHACV_HAEMATOPOIETIC', 'SNU423_LIVER', 'SNU1079_BILIARY', 'A204_SOFT', 'BXPC3_PANCREAS', 'CALU1_LUNG', 'HEC251_ENDOMETRIUM', 'HEL_HAEMATOPOIETIC', 'KYSE180_OESOPHAGUS', 'MONOMAC1_HAEMATOPOIETIC', 'SF295_CENTRAL', 'SKMEL5_SKIN', 'PC14_LUNG', 'EBC1_LUNG', 'SKMEL2_SKIN', 'MEWO_SKIN', 'HS944T_SKIN', 'JHH7_LIVER', 'RVH421_SKIN', 'A498_KIDNEY', 'HS695T_SKIN', '697_HAEMATOPOIETIC', '8305C_THYROID', 'HCT15_LARGE', 'IGR1_SKIN', 'SUIT2_PANCREAS', 'LN18_CENTRAL', 'LNCAPCLONEFGC_PROSTATE', 'NCIH650_LUNG', 'REH_HAEMATOPOIETIC', 'HEC1A_ENDOMETRIUM', 'HUPT3_PANCREAS', 'KMS27_HAEMATOPOIETIC', 'LXF289_LUNG', 'CAL27_UPPER', 'RMUGS_OVARY', 'SNU1105_CENTRAL', 'CAL33_UPPER', 'U937_HAEMATOPOIETIC', 'A704_KIDNEY', 'WM88_SKIN', 'KMS11_HAEMATOPOIETIC', 'NALM6_HAEMATOPOIETIC', 'NCIH2052_PLEURA', 'OV90_OVARY', 'SKES1_BONE', 'SNB19_CENTRAL', 'MFE280_ENDOMETRIUM', 'HCC1806_BREAST', 'COLO320_LARGE', 'NCIH1581_LUNG', 'NCIH1792_LUNG', 'NCIH2066_LUNG', 'SQ1_LUNG', 'A2780_OVARY', 'MKN1_STOMACH', 'BICR22_UPPER', 'HCC1395_BREAST', 'SW1417_LARGE', 'NCIH1435_LUNG', 'NCIH1568_LUNG', 'NCIH1944_LUNG', 'NCIH441_LUNG', 'KURAMOCHI_OVARY', 'KATOIII_STOMACH', 'UBLC1_URINARY', 'MFE319_ENDOMETRIUM', 'HDMYZ_HAEMATOPOIETIC', 'HDQP1_BREAST', 'JHH6_LIVER', 'L428_HAEMATOPOIETIC', 'LMSU_STOMACH', 'HUPT4_PANCREAS', 'SW1783_CENTRAL', 'UMUC3_URINARY', 'G402_SOFT', 'U118MG_CENTRAL', 'KYSE150_OESOPHAGUS', 'NCIH1155_LUNG', 'NCIN87_STOMACH', 'OPM2_HAEMATOPOIETIC', 'PL45_PANCREAS', 'SKCO1_LARGE', 'SUDHL4_HAEMATOPOIETIC', '769P_KIDNEY', 'C32_SKIN', 'HEC59_ENDOMETRIUM', 'K562_HAEMATOPOIETIC', 'NUGC3_STOMACH', 'RD_SOFT', 'SEM_HAEMATOPOIETIC', 'SHP77_LUNG', 'TC71_BONE', 'HCC1937_BREAST', 'HCT15_LARGE', 'KYSE70_OESOPHAGUS', 'NCIH1355_LUNG', 'NCIH1437_LUNG', 'NCIH2122_LUNG', 'TUHR4TKB_KIDNEY', 'WM793_SKIN', 'NCIH2030_LUNG', '22RV1_PROSTATE', 'A172_CENTRAL', 'BT20_BREAST', 'CALU6_LUNG', 'FADU_UPPER', 'KP4_PANCREAS', 'MONOMAC6_HAEMATOPOIETIC', 'OVCAR8_OVARY', 'THP1_HAEMATOPOIETIC']\n"
     ]
    }
   ],
   "source": [
    "# cell line names in the protein samples, remove the content after the second \"_\"\n",
    "# result = ['_'.join(s.split('_')[:2]) for s in strings]\n",
    "protein_split_samples = ['_'.join(s.split('_')[:2]) for s in protein_samples]\n",
    "print(protein_split_samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 332,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CCLE_ID</th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Site_Primary</th>\n",
       "      <th>Site_Subtype1</th>\n",
       "      <th>Site_Subtype2</th>\n",
       "      <th>Site_Subtype3</th>\n",
       "      <th>Histology</th>\n",
       "      <th>Hist_Subtype1</th>\n",
       "      <th>...</th>\n",
       "      <th>Growth.Medium</th>\n",
       "      <th>Supplements</th>\n",
       "      <th>Freezing.Medium</th>\n",
       "      <th>Doubling.Time.from.Vendor</th>\n",
       "      <th>Doubling.Time.Calculated.hrs</th>\n",
       "      <th>type</th>\n",
       "      <th>type_refined</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>mutRate</th>\n",
       "      <th>tcga_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DMS53_LUNG</td>\n",
       "      <td>ACH-000698</td>\n",
       "      <td>DMS 53</td>\n",
       "      <td>primary</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>small_cell_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>Waymouth's +10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>450.0</td>\n",
       "      <td>lung_small_cell</td>\n",
       "      <td>lung_small_cell</td>\n",
       "      <td>Lung:SCLC</td>\n",
       "      <td>157.241638</td>\n",
       "      <td>SCLC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SW1116_LARGE_INTESTINE</td>\n",
       "      <td>ACH-000489</td>\n",
       "      <td>SW1116</td>\n",
       "      <td>primary</td>\n",
       "      <td>large_intestine</td>\n",
       "      <td>colon</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>adenocarcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>L15+10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>123.2</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>94.373192</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NCIH1694_LUNG</td>\n",
       "      <td>ACH-000431</td>\n",
       "      <td>NCI-H1694</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>small_cell_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>DMEM:F12 (1:1) + 5% FBS</td>\n",
       "      <td>.005 mg/ml insulin, .01 mg/ml transferrin, 30n...</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>81.2</td>\n",
       "      <td>lung_small_cell</td>\n",
       "      <td>lung_small_cell</td>\n",
       "      <td>Lung:SCLC</td>\n",
       "      <td>109.534987</td>\n",
       "      <td>SCLC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>ACH-000707</td>\n",
       "      <td>P3HR-1</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>Burkitt_lymphoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lymphoma_Burkitt</td>\n",
       "      <td>lymphoma_Burkitt</td>\n",
       "      <td>Lymphoma:NH_B_cell</td>\n",
       "      <td>166.456406</td>\n",
       "      <td>DLBC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>ACH-000509</td>\n",
       "      <td>HuT 78</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>skin</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>mycosis_fungoides-Sezary_syndrome</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lymphoma_other</td>\n",
       "      <td>T-cell_lymphoma_other</td>\n",
       "      <td>Lymphoma:NH_T_cell</td>\n",
       "      <td>118.217569</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015</th>\n",
       "      <td>SF539_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>ACH-000273</td>\n",
       "      <td>SF539</td>\n",
       "      <td>primary</td>\n",
       "      <td>central_nervous_system</td>\n",
       "      <td>brain</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>glioma</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+ 5% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>105.3</td>\n",
       "      <td>glioma</td>\n",
       "      <td>glioma</td>\n",
       "      <td>CNS:Glioma_HighGrade</td>\n",
       "      <td>107.540583</td>\n",
       "      <td>LGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1016</th>\n",
       "      <td>SNB75_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>ACH-000504</td>\n",
       "      <td>SNB75</td>\n",
       "      <td>primary</td>\n",
       "      <td>central_nervous_system</td>\n",
       "      <td>brain</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>glioma</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+ 5% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.0</td>\n",
       "      <td>glioma</td>\n",
       "      <td>glioma</td>\n",
       "      <td>CNS:Glioma_HighGrade</td>\n",
       "      <td>106.364784</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1017</th>\n",
       "      <td>HOP92_LUNG</td>\n",
       "      <td>ACH-000825</td>\n",
       "      <td>HOP-92</td>\n",
       "      <td>primary</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>large_cell_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+ 5% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>122.2</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>Lung:NSCLC_Others</td>\n",
       "      <td>198.739234</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1018</th>\n",
       "      <td>ACCS_URINARY_TRACT</td>\n",
       "      <td>ACH-001011</td>\n",
       "      <td>ACCS</td>\n",
       "      <td>primary</td>\n",
       "      <td>urinary_tract</td>\n",
       "      <td>bladder</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>transitional_cell_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>urinary_tract</td>\n",
       "      <td>urinary_tract</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1019</th>\n",
       "      <td>MUTZ3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>ACH-000084</td>\n",
       "      <td>MUTZ-3</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>acute_myeloid_leukaemia</td>\n",
       "      <td>...</td>\n",
       "      <td>MEM + 20% FBS</td>\n",
       "      <td>10ng/ml SCF</td>\n",
       "      <td>10%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AML</td>\n",
       "      <td>AML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>83.534726</td>\n",
       "      <td>LAML</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1019 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       CCLE_ID    depMapID       Name  \\\n",
       "0                                   DMS53_LUNG  ACH-000698     DMS 53   \n",
       "1                       SW1116_LARGE_INTESTINE  ACH-000489     SW1116   \n",
       "2                                NCIH1694_LUNG  ACH-000431  NCI-H1694   \n",
       "3     P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ACH-000707     P3HR-1   \n",
       "4     HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ACH-000509     HuT 78   \n",
       "...                                        ...         ...        ...   \n",
       "1015              SF539_CENTRAL_NERVOUS_SYSTEM  ACH-000273      SF539   \n",
       "1016              SNB75_CENTRAL_NERVOUS_SYSTEM  ACH-000504      SNB75   \n",
       "1017                                HOP92_LUNG  ACH-000825     HOP-92   \n",
       "1018                        ACCS_URINARY_TRACT  ACH-001011       ACCS   \n",
       "1019  MUTZ3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ACH-000084     MUTZ-3   \n",
       "\n",
       "       Pathology                        Site_Primary Site_Subtype1  \\\n",
       "0        primary                                lung            NS   \n",
       "1        primary                     large_intestine         colon   \n",
       "2     metastasis                                lung            NS   \n",
       "3     metastasis  haematopoietic_and_lymphoid_tissue            NS   \n",
       "4        primary  haematopoietic_and_lymphoid_tissue          skin   \n",
       "...          ...                                 ...           ...   \n",
       "1015     primary              central_nervous_system         brain   \n",
       "1016     primary              central_nervous_system         brain   \n",
       "1017     primary                                lung            NS   \n",
       "1018     primary                       urinary_tract       bladder   \n",
       "1019     primary  haematopoietic_and_lymphoid_tissue            NS   \n",
       "\n",
       "     Site_Subtype2 Site_Subtype3                Histology  \\\n",
       "0               NS            NS                carcinoma   \n",
       "1               NS            NS                carcinoma   \n",
       "2               NS            NS                carcinoma   \n",
       "3               NS            NS        lymphoid_neoplasm   \n",
       "4               NS            NS        lymphoid_neoplasm   \n",
       "...            ...           ...                      ...   \n",
       "1015            NS            NS                   glioma   \n",
       "1016            NS            NS                   glioma   \n",
       "1017            NS            NS                carcinoma   \n",
       "1018            NS            NS                carcinoma   \n",
       "1019            NS            NS  haematopoietic_neoplasm   \n",
       "\n",
       "                          Hist_Subtype1  ...            Growth.Medium  \\\n",
       "0                  small_cell_carcinoma  ...       Waymouth's +10%FBS   \n",
       "1                        adenocarcinoma  ...               L15+10%FBS   \n",
       "2                  small_cell_carcinoma  ...  DMEM:F12 (1:1) + 5% FBS   \n",
       "3                      Burkitt_lymphoma  ...                      NaN   \n",
       "4     mycosis_fungoides-Sezary_syndrome  ...                      NaN   \n",
       "...                                 ...  ...                      ...   \n",
       "1015                                 NS  ...        RPMI-1640+ 5% FBS   \n",
       "1016                                 NS  ...        RPMI-1640+ 5% FBS   \n",
       "1017               large_cell_carcinoma  ...        RPMI-1640+ 5% FBS   \n",
       "1018        transitional_cell_carcinoma  ...                      NaN   \n",
       "1019            acute_myeloid_leukaemia  ...            MEM + 20% FBS   \n",
       "\n",
       "                                            Supplements Freezing.Medium  \\\n",
       "0                                                   NaN          5%DMSO   \n",
       "1                                                   NaN          5%DMSO   \n",
       "2     .005 mg/ml insulin, .01 mg/ml transferrin, 30n...         5% DMSO   \n",
       "3                                                   NaN             NaN   \n",
       "4                                                   NaN             NaN   \n",
       "...                                                 ...             ...   \n",
       "1015                                                NaN         5% DMSO   \n",
       "1016                                                NaN         5% DMSO   \n",
       "1017                                                NaN         5% DMSO   \n",
       "1018                                                NaN             NaN   \n",
       "1019                                        10ng/ml SCF         10%DMSO   \n",
       "\n",
       "     Doubling.Time.from.Vendor  Doubling.Time.Calculated.hrs  \\\n",
       "0                          NaN                         450.0   \n",
       "1                          NaN                         123.2   \n",
       "2                          NaN                          81.2   \n",
       "3                          NaN                           NaN   \n",
       "4                          NaN                           NaN   \n",
       "...                        ...                           ...   \n",
       "1015                       NaN                         105.3   \n",
       "1016                       NaN                          54.0   \n",
       "1017                       NaN                         122.2   \n",
       "1018                       NaN                           NaN   \n",
       "1019                       NaN                           NaN   \n",
       "\n",
       "                  type           type_refined PATHOLOGIST_ANNOTATION  \\\n",
       "0      lung_small_cell        lung_small_cell              Lung:SCLC   \n",
       "1           colorectal             colorectal   Colorectal:Carcinoma   \n",
       "2      lung_small_cell        lung_small_cell              Lung:SCLC   \n",
       "3     lymphoma_Burkitt       lymphoma_Burkitt     Lymphoma:NH_B_cell   \n",
       "4       lymphoma_other  T-cell_lymphoma_other     Lymphoma:NH_T_cell   \n",
       "...                ...                    ...                    ...   \n",
       "1015            glioma                 glioma   CNS:Glioma_HighGrade   \n",
       "1016            glioma                 glioma   CNS:Glioma_HighGrade   \n",
       "1017          lung_NSC               lung_NSC      Lung:NSCLC_Others   \n",
       "1018     urinary_tract          urinary_tract                    NaN   \n",
       "1019               AML                    AML                    NaN   \n",
       "\n",
       "         mutRate  tcga_code  \n",
       "0     157.241638       SCLC  \n",
       "1      94.373192  COAD/READ  \n",
       "2     109.534987       SCLC  \n",
       "3     166.456406       DLBC  \n",
       "4     118.217569        NaN  \n",
       "...          ...        ...  \n",
       "1015  107.540583        LGG  \n",
       "1016  106.364784        GBM  \n",
       "1017  198.739234        NaN  \n",
       "1018         NaN       BLCA  \n",
       "1019   83.534726       LAML  \n",
       "\n",
       "[1019 rows x 33 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell_line_anno_df = pd.read_csv('./data/raw_data/Cell_lines_annotations_20181226.txt', sep='\\t')\n",
    "# drop NaN in the depMapID\n",
    "cell_line_anno_df = cell_line_anno_df.dropna(subset=['depMapID']).reset_index(drop=True)\n",
    "# drop NaN in the Pathology\n",
    "cell_line_anno_df = cell_line_anno_df.dropna(subset=['Pathology']).reset_index(drop=True)\n",
    "# drop rows if the column 'PATHOLOGIST_ANNOTATION' contains 'benign'\n",
    "# First handle NaN values in PATHOLOGIST_ANNOTATION\n",
    "cell_line_anno_df = cell_line_anno_df[~(\n",
    "    cell_line_anno_df['PATHOLOGIST_ANNOTATION'].fillna('').str.contains('benign', case=False)\n",
    ")]\n",
    "display(cell_line_anno_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "375\n",
      "['CAL120_BREAST_TenPx02', 'SW948_LARGE_INTESTINE_TenPx11', 'SW948_LARGE_INTESTINE_TenPx20', 'CAL120_BREAST_TenPx28', 'HCT15_LARGE_INTESTINE_TenPx30', 'HCT15_LARGE_INTESTINE_TenPx18']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CAL120_BREAST_TenPx02</th>\n",
       "      <th>SW948_LARGE_INTESTINE_TenPx11</th>\n",
       "      <th>SW948_LARGE_INTESTINE_TenPx20</th>\n",
       "      <th>CAL120_BREAST_TenPx28</th>\n",
       "      <th>HCT15_LARGE_INTESTINE_TenPx30</th>\n",
       "      <th>HCT15_LARGE_INTESTINE_TenPx18</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CAL120_BREAST_TenPx02  SW948_LARGE_INTESTINE_TenPx11  \\\n",
       "0                         0.0                            0.0   \n",
       "1                         0.0                            0.0   \n",
       "2                         0.0                            0.0   \n",
       "3                         0.0                            0.0   \n",
       "4                         0.0                            0.0   \n",
       "...                       ...                            ...   \n",
       "121414                    0.0                            0.0   \n",
       "121415                    0.0                            0.0   \n",
       "121416                    0.0                            0.0   \n",
       "121417                    0.0                            0.0   \n",
       "121418                    0.0                            0.0   \n",
       "\n",
       "        SW948_LARGE_INTESTINE_TenPx20  CAL120_BREAST_TenPx28  \\\n",
       "0                                 0.0                    0.0   \n",
       "1                                 0.0                    0.0   \n",
       "2                                 0.0                    0.0   \n",
       "3                                 0.0                    0.0   \n",
       "4                                 0.0                    0.0   \n",
       "...                               ...                    ...   \n",
       "121414                            0.0                    0.0   \n",
       "121415                            0.0                    0.0   \n",
       "121416                            0.0                    0.0   \n",
       "121417                            0.0                    0.0   \n",
       "121418                            0.0                    0.0   \n",
       "\n",
       "        HCT15_LARGE_INTESTINE_TenPx30  HCT15_LARGE_INTESTINE_TenPx18  \n",
       "0                                 0.0                            0.0  \n",
       "1                                 0.0                            0.0  \n",
       "2                                 0.0                            0.0  \n",
       "3                                 0.0                            0.0  \n",
       "4                                 0.0                            0.0  \n",
       "...                               ...                            ...  \n",
       "121414                            0.0                            0.0  \n",
       "121415                            0.0                            0.0  \n",
       "121416                            0.0                            0.0  \n",
       "121417                            0.0                            0.0  \n",
       "121418                            0.0                            0.0  \n",
       "\n",
       "[121419 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>MDAMB468_BREAST_TenPx01</th>\n",
       "      <th>SH4_SKIN_TenPx01</th>\n",
       "      <th>AU565_BREAST_TenPx01</th>\n",
       "      <th>KMRC1_KIDNEY_TenPx01</th>\n",
       "      <th>CAL51_BREAST_TenPx01</th>\n",
       "      <th>RPMI7951_SKIN_TenPx01</th>\n",
       "      <th>RERFLCMS_LUNG_TenPx01</th>\n",
       "      <th>IGR37_SKIN_TenPx01</th>\n",
       "      <th>VMRCRCW_KIDNEY_TenPx01</th>\n",
       "      <th>...</th>\n",
       "      <th>NCIH2030_LUNG_TenPx18</th>\n",
       "      <th>22RV1_PROSTATE_TenPx24</th>\n",
       "      <th>A172_CENTRAL_NERVOUS_SYSTEM_TenPx24</th>\n",
       "      <th>BT20_BREAST_TenPx24</th>\n",
       "      <th>CALU6_LUNG_TenPx24</th>\n",
       "      <th>FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24</th>\n",
       "      <th>KP4_PANCREAS_TenPx24</th>\n",
       "      <th>MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "      <th>OVCAR8_OVARY_TenPx24</th>\n",
       "      <th>THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 373 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  MDAMB468_BREAST_TenPx01  SH4_SKIN_TenPx01  \\\n",
       "0               BMGC_PT000001                      0.0               0.0   \n",
       "1               BMGC_PT000002                      0.0               0.0   \n",
       "2               BMGC_PT000003                      0.0               0.0   \n",
       "3               BMGC_PT000004                      0.0               0.0   \n",
       "4               BMGC_PT000005                      0.0               0.0   \n",
       "...                       ...                      ...               ...   \n",
       "121414          BMGC_PT121415                      0.0               0.0   \n",
       "121415          BMGC_PT121416                      0.0               0.0   \n",
       "121416          BMGC_PT121417                      0.0               0.0   \n",
       "121417          BMGC_PT121418                      0.0               0.0   \n",
       "121418          BMGC_PT121419                      0.0               0.0   \n",
       "\n",
       "        AU565_BREAST_TenPx01  KMRC1_KIDNEY_TenPx01  CAL51_BREAST_TenPx01  \\\n",
       "0                        0.0                   0.0                   0.0   \n",
       "1                        0.0                   0.0                   0.0   \n",
       "2                        0.0                   0.0                   0.0   \n",
       "3                        0.0                   0.0                   0.0   \n",
       "4                        0.0                   0.0                   0.0   \n",
       "...                      ...                   ...                   ...   \n",
       "121414                   0.0                   0.0                   0.0   \n",
       "121415                   0.0                   0.0                   0.0   \n",
       "121416                   0.0                   0.0                   0.0   \n",
       "121417                   0.0                   0.0                   0.0   \n",
       "121418                   0.0                   0.0                   0.0   \n",
       "\n",
       "        RPMI7951_SKIN_TenPx01  RERFLCMS_LUNG_TenPx01  IGR37_SKIN_TenPx01  \\\n",
       "0                         0.0                    0.0                 0.0   \n",
       "1                         0.0                    0.0                 0.0   \n",
       "2                         0.0                    0.0                 0.0   \n",
       "3                         0.0                    0.0                 0.0   \n",
       "4                         0.0                    0.0                 0.0   \n",
       "...                       ...                    ...                 ...   \n",
       "121414                    0.0                    0.0                 0.0   \n",
       "121415                    0.0                    0.0                 0.0   \n",
       "121416                    0.0                    0.0                 0.0   \n",
       "121417                    0.0                    0.0                 0.0   \n",
       "121418                    0.0                    0.0                 0.0   \n",
       "\n",
       "        VMRCRCW_KIDNEY_TenPx01  ...  NCIH2030_LUNG_TenPx18  \\\n",
       "0                          0.0  ...                    0.0   \n",
       "1                          0.0  ...                    0.0   \n",
       "2                          0.0  ...                    0.0   \n",
       "3                          0.0  ...                    0.0   \n",
       "4                          0.0  ...                    0.0   \n",
       "...                        ...  ...                    ...   \n",
       "121414                     0.0  ...                    0.0   \n",
       "121415                     0.0  ...                    0.0   \n",
       "121416                     0.0  ...                    0.0   \n",
       "121417                     0.0  ...                    0.0   \n",
       "121418                     0.0  ...                    0.0   \n",
       "\n",
       "        22RV1_PROSTATE_TenPx24  A172_CENTRAL_NERVOUS_SYSTEM_TenPx24  \\\n",
       "0                          0.0                                  0.0   \n",
       "1                          0.0                                  0.0   \n",
       "2                          0.0                                  0.0   \n",
       "3                          0.0                                  0.0   \n",
       "4                          0.0                                  0.0   \n",
       "...                        ...                                  ...   \n",
       "121414                     0.0                                  0.0   \n",
       "121415                     0.0                                  0.0   \n",
       "121416                     0.0                                  0.0   \n",
       "121417                     0.0                                  0.0   \n",
       "121418                     0.0                                  0.0   \n",
       "\n",
       "        BT20_BREAST_TenPx24  CALU6_LUNG_TenPx24  \\\n",
       "0                       0.0                 0.0   \n",
       "1                       0.0                 0.0   \n",
       "2                       0.0                 0.0   \n",
       "3                       0.0                 0.0   \n",
       "4                       0.0                 0.0   \n",
       "...                     ...                 ...   \n",
       "121414                  0.0                 0.0   \n",
       "121415                  0.0                 0.0   \n",
       "121416                  0.0                 0.0   \n",
       "121417                  0.0                 0.0   \n",
       "121418                  0.0                 0.0   \n",
       "\n",
       "        FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24  KP4_PANCREAS_TenPx24  \\\n",
       "0                                          0.0                   0.0   \n",
       "1                                          0.0                   0.0   \n",
       "2                                          0.0                   0.0   \n",
       "3                                          0.0                   0.0   \n",
       "4                                          0.0                   0.0   \n",
       "...                                        ...                   ...   \n",
       "121414                                     0.0                   0.0   \n",
       "121415                                     0.0                   0.0   \n",
       "121416                                     0.0                   0.0   \n",
       "121417                                     0.0                   0.0   \n",
       "121418                                     0.0                   0.0   \n",
       "\n",
       "        MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \\\n",
       "0                                                     0.0     \n",
       "1                                                     0.0     \n",
       "2                                                     0.0     \n",
       "3                                                     0.0     \n",
       "4                                                     0.0     \n",
       "...                                                   ...     \n",
       "121414                                                0.0     \n",
       "121415                                                0.0     \n",
       "121416                                                0.0     \n",
       "121417                                                0.0     \n",
       "121418                                                0.0     \n",
       "\n",
       "        OVCAR8_OVARY_TenPx24  THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \n",
       "0                        0.0                                              0.0  \n",
       "1                        0.0                                              0.0  \n",
       "2                        0.0                                              0.0  \n",
       "3                        0.0                                              0.0  \n",
       "4                        0.0                                              0.0  \n",
       "...                      ...                                              ...  \n",
       "121414                   0.0                                              0.0  \n",
       "121415                   0.0                                              0.0  \n",
       "121416                   0.0                                              0.0  \n",
       "121417                   0.0                                              0.0  \n",
       "121418                   0.0                                              0.0  \n",
       "\n",
       "[121419 rows x 373 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(len(list(set(protein_split_samples))))\n",
    "# give me the duplicate elements in the protein_split_samples\n",
    "import collections\n",
    "rep_list = [item for item, count in collections.Counter(protein_split_samples).items() if count > 1]\n",
    "# for column names in final_merged_protein_df, if the column name contains the element in the rep_list, collect the column name in rep_col_list\n",
    "rep_col_list = [col for col in final_merged_protein_df.columns if any(rep in col for rep in rep_list)]\n",
    "print(rep_col_list)\n",
    "# display the rep_col_list in the final_merged_protein_df\n",
    "display(final_merged_protein_df[rep_col_list])\n",
    "\n",
    "# removed all of the rep_col_list in the final_merged_protein_df\n",
    "final_merged_protein_df = final_merged_protein_df.drop(columns=rep_col_list)\n",
    "display(final_merged_protein_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CCLE_Name</th>\n",
       "      <th>CCLE_ID</th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Site_Primary</th>\n",
       "      <th>Site_Subtype1</th>\n",
       "      <th>Site_Subtype2</th>\n",
       "      <th>Site_Subtype3</th>\n",
       "      <th>Histology</th>\n",
       "      <th>...</th>\n",
       "      <th>Growth.Medium</th>\n",
       "      <th>Supplements</th>\n",
       "      <th>Freezing.Medium</th>\n",
       "      <th>Doubling.Time.from.Vendor</th>\n",
       "      <th>Doubling.Time.Calculated.hrs</th>\n",
       "      <th>type</th>\n",
       "      <th>type_refined</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>mutRate</th>\n",
       "      <th>tcga_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DMS53_LUNG</td>\n",
       "      <td>DMS53_LUNG</td>\n",
       "      <td>ACH-000698</td>\n",
       "      <td>DMS 53</td>\n",
       "      <td>primary</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>Waymouth's +10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>450.0</td>\n",
       "      <td>lung_small_cell</td>\n",
       "      <td>lung_small_cell</td>\n",
       "      <td>Lung:SCLC</td>\n",
       "      <td>157.241638</td>\n",
       "      <td>SCLC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SW1116_LARGE_INTESTINE</td>\n",
       "      <td>SW1116_LARGE_INTESTINE</td>\n",
       "      <td>ACH-000489</td>\n",
       "      <td>SW1116</td>\n",
       "      <td>primary</td>\n",
       "      <td>large_intestine</td>\n",
       "      <td>colon</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>L15+10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>123.2</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>94.373192</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>ACH-000707</td>\n",
       "      <td>P3HR-1</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lymphoma_Burkitt</td>\n",
       "      <td>lymphoma_Burkitt</td>\n",
       "      <td>Lymphoma:NH_B_cell</td>\n",
       "      <td>166.456406</td>\n",
       "      <td>DLBC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>ACH-000509</td>\n",
       "      <td>HuT 78</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>skin</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lymphoma_other</td>\n",
       "      <td>T-cell_lymphoma_other</td>\n",
       "      <td>Lymphoma:NH_T_cell</td>\n",
       "      <td>118.217569</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>UMUC3_URINARY_TRACT</td>\n",
       "      <td>UMUC3_URINARY_TRACT</td>\n",
       "      <td>ACH-000522</td>\n",
       "      <td>UM-UC-3</td>\n",
       "      <td>primary</td>\n",
       "      <td>urinary_tract</td>\n",
       "      <td>bladder</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>EMEM +10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>386.7</td>\n",
       "      <td>urinary_tract</td>\n",
       "      <td>urinary_tract</td>\n",
       "      <td>Bladder:Carcinoma</td>\n",
       "      <td>124.707962</td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>UO31_KIDNEY</td>\n",
       "      <td>UO31_KIDNEY</td>\n",
       "      <td>ACH-000428</td>\n",
       "      <td>UO-31</td>\n",
       "      <td>primary</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+ 5% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.8</td>\n",
       "      <td>kidney</td>\n",
       "      <td>kidney</td>\n",
       "      <td>Kidney:Carcinoma</td>\n",
       "      <td>98.686744</td>\n",
       "      <td>KIRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>SF268_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>SF268_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>ACH-000655</td>\n",
       "      <td>SF268</td>\n",
       "      <td>primary</td>\n",
       "      <td>central_nervous_system</td>\n",
       "      <td>brain</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>glioma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+ 5% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>63.0</td>\n",
       "      <td>glioma</td>\n",
       "      <td>glioma</td>\n",
       "      <td>CNS:Glioma_HighGrade</td>\n",
       "      <td>157.809603</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>837</th>\n",
       "      <td>SF539_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>SF539_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>ACH-000273</td>\n",
       "      <td>SF539</td>\n",
       "      <td>primary</td>\n",
       "      <td>central_nervous_system</td>\n",
       "      <td>brain</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>glioma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+ 5% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>105.3</td>\n",
       "      <td>glioma</td>\n",
       "      <td>glioma</td>\n",
       "      <td>CNS:Glioma_HighGrade</td>\n",
       "      <td>107.540583</td>\n",
       "      <td>LGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>SNB75_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>SNB75_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>ACH-000504</td>\n",
       "      <td>SNB75</td>\n",
       "      <td>primary</td>\n",
       "      <td>central_nervous_system</td>\n",
       "      <td>brain</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>glioma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+ 5% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.0</td>\n",
       "      <td>glioma</td>\n",
       "      <td>glioma</td>\n",
       "      <td>CNS:Glioma_HighGrade</td>\n",
       "      <td>106.364784</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>839</th>\n",
       "      <td>HOP92_LUNG</td>\n",
       "      <td>HOP92_LUNG</td>\n",
       "      <td>ACH-000825</td>\n",
       "      <td>HOP-92</td>\n",
       "      <td>primary</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+ 5% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>122.2</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>Lung:NSCLC_Others</td>\n",
       "      <td>198.739234</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>840 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    CCLE_Name  \\\n",
       "0                                  DMS53_LUNG   \n",
       "1                      SW1116_LARGE_INTESTINE   \n",
       "2    P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE   \n",
       "3    HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE   \n",
       "4                         UMUC3_URINARY_TRACT   \n",
       "..                                        ...   \n",
       "835                               UO31_KIDNEY   \n",
       "836              SF268_CENTRAL_NERVOUS_SYSTEM   \n",
       "837              SF539_CENTRAL_NERVOUS_SYSTEM   \n",
       "838              SNB75_CENTRAL_NERVOUS_SYSTEM   \n",
       "839                                HOP92_LUNG   \n",
       "\n",
       "                                      CCLE_ID    depMapID     Name  \\\n",
       "0                                  DMS53_LUNG  ACH-000698   DMS 53   \n",
       "1                      SW1116_LARGE_INTESTINE  ACH-000489   SW1116   \n",
       "2    P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ACH-000707   P3HR-1   \n",
       "3    HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ACH-000509   HuT 78   \n",
       "4                         UMUC3_URINARY_TRACT  ACH-000522  UM-UC-3   \n",
       "..                                        ...         ...      ...   \n",
       "835                               UO31_KIDNEY  ACH-000428    UO-31   \n",
       "836              SF268_CENTRAL_NERVOUS_SYSTEM  ACH-000655    SF268   \n",
       "837              SF539_CENTRAL_NERVOUS_SYSTEM  ACH-000273    SF539   \n",
       "838              SNB75_CENTRAL_NERVOUS_SYSTEM  ACH-000504    SNB75   \n",
       "839                                HOP92_LUNG  ACH-000825   HOP-92   \n",
       "\n",
       "      Pathology                        Site_Primary Site_Subtype1  \\\n",
       "0       primary                                lung            NS   \n",
       "1       primary                     large_intestine         colon   \n",
       "2    metastasis  haematopoietic_and_lymphoid_tissue            NS   \n",
       "3       primary  haematopoietic_and_lymphoid_tissue          skin   \n",
       "4       primary                       urinary_tract       bladder   \n",
       "..          ...                                 ...           ...   \n",
       "835     primary                              kidney            NS   \n",
       "836     primary              central_nervous_system         brain   \n",
       "837     primary              central_nervous_system         brain   \n",
       "838     primary              central_nervous_system         brain   \n",
       "839     primary                                lung            NS   \n",
       "\n",
       "    Site_Subtype2 Site_Subtype3          Histology  ...       Growth.Medium  \\\n",
       "0              NS            NS          carcinoma  ...  Waymouth's +10%FBS   \n",
       "1              NS            NS          carcinoma  ...          L15+10%FBS   \n",
       "2              NS            NS  lymphoid_neoplasm  ...                 NaN   \n",
       "3              NS            NS  lymphoid_neoplasm  ...                 NaN   \n",
       "4              NS            NS          carcinoma  ...        EMEM +10%FBS   \n",
       "..            ...           ...                ...  ...                 ...   \n",
       "835            NS            NS          carcinoma  ...   RPMI-1640+ 5% FBS   \n",
       "836            NS            NS             glioma  ...   RPMI-1640+ 5% FBS   \n",
       "837            NS            NS             glioma  ...   RPMI-1640+ 5% FBS   \n",
       "838            NS            NS             glioma  ...   RPMI-1640+ 5% FBS   \n",
       "839            NS            NS          carcinoma  ...   RPMI-1640+ 5% FBS   \n",
       "\n",
       "    Supplements Freezing.Medium Doubling.Time.from.Vendor  \\\n",
       "0           NaN          5%DMSO                       NaN   \n",
       "1           NaN          5%DMSO                       NaN   \n",
       "2           NaN             NaN                       NaN   \n",
       "3           NaN             NaN                       NaN   \n",
       "4           NaN          5%DMSO                       NaN   \n",
       "..          ...             ...                       ...   \n",
       "835         NaN         5% DMSO                       NaN   \n",
       "836         NaN         5% DMSO                       NaN   \n",
       "837         NaN         5% DMSO                       NaN   \n",
       "838         NaN         5% DMSO                       NaN   \n",
       "839         NaN         5% DMSO                       NaN   \n",
       "\n",
       "    Doubling.Time.Calculated.hrs              type           type_refined  \\\n",
       "0                          450.0   lung_small_cell        lung_small_cell   \n",
       "1                          123.2        colorectal             colorectal   \n",
       "2                            NaN  lymphoma_Burkitt       lymphoma_Burkitt   \n",
       "3                            NaN    lymphoma_other  T-cell_lymphoma_other   \n",
       "4                          386.7     urinary_tract          urinary_tract   \n",
       "..                           ...               ...                    ...   \n",
       "835                         27.8            kidney                 kidney   \n",
       "836                         63.0            glioma                 glioma   \n",
       "837                        105.3            glioma                 glioma   \n",
       "838                         54.0            glioma                 glioma   \n",
       "839                        122.2          lung_NSC               lung_NSC   \n",
       "\n",
       "    PATHOLOGIST_ANNOTATION     mutRate  tcga_code  \n",
       "0                Lung:SCLC  157.241638       SCLC  \n",
       "1     Colorectal:Carcinoma   94.373192  COAD/READ  \n",
       "2       Lymphoma:NH_B_cell  166.456406       DLBC  \n",
       "3       Lymphoma:NH_T_cell  118.217569        NaN  \n",
       "4        Bladder:Carcinoma  124.707962       BLCA  \n",
       "..                     ...         ...        ...  \n",
       "835       Kidney:Carcinoma   98.686744       KIRC  \n",
       "836   CNS:Glioma_HighGrade  157.809603        GBM  \n",
       "837   CNS:Glioma_HighGrade  107.540583        LGG  \n",
       "838   CNS:Glioma_HighGrade  106.364784        GBM  \n",
       "839      Lung:NSCLC_Others  198.739234        NaN  \n",
       "\n",
       "[840 rows x 34 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'DMS53_LUNG': 'ACH-000698', 'SW1116_LARGE_INTESTINE': 'ACH-000489', 'P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000707', 'HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000509', 'UMUC3_URINARY_TRACT': 'ACH-000522', 'HOS_BONE': 'ACH-000613', 'AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000557', 'RVH421_SKIN': 'ACH-000614', 'NCIH1184_LUNG': 'ACH-000523', 'HCC2157_BREAST': 'ACH-000691', 'NCIH2227_LUNG': 'ACH-000610', 'SNU449_LIVER': 'ACH-000420', 'NCIH28_PLEURA': 'ACH-000648', 'OV56_OVARY': 'ACH-000091', 'RMUGS_OVARY': 'ACH-000701', 'LN229_CENTRAL_NERVOUS_SYSTEM': 'ACH-000595', 'P31FUJ_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000770', 'RKN_SOFT_TISSUE': 'ACH-000505', 'PATU8988S_PANCREAS': 'ACH-000022', 'NH6_AUTONOMIC_GANGLIA': 'ACH-000203', 'SF126_CENTRAL_NERVOUS_SYSTEM': 'ACH-000609', 'RERFLCAD2_LUNG': 'ACH-000774', 'OUMS23_LARGE_INTESTINE': 'ACH-000296', 'SNGM_ENDOMETRIUM': 'ACH-000974', 'OUMS27_BONE': 'ACH-001150', 'NCIH2347_LUNG': 'ACH-000875', 'SW1990_PANCREAS': 'ACH-000155', 'HS940T_FIBROBLAST': 'ACH-000135', 'HS611T_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000069', 'TOLEDO_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000285', 'RERFGC1B_STOMACH': 'ACH-000144', 'HT1080_SOFT_TISSUE': 'ACH-000054', 'NCIH2087_LUNG': 'ACH-000841', 'COV318_OVARY': 'ACH-000256', 'NCIH2085_LUNG': 'ACH-000451', 'NCIH510_LUNG': 'ACH-000871', 'CAKI1_KIDNEY': 'ACH-000433', 'NCIH716_LARGE_INTESTINE': 'ACH-000491', 'NCIH2066_LUNG': 'ACH-000727', 'NCIH1341_LUNG': 'ACH-000129', 'NCIH2029_LUNG': 'ACH-000298', 'SW480_LARGE_INTESTINE': 'ACH-000842', 'HS746T_STOMACH': 'ACH-000616', 'LU65_LUNG': 'ACH-000438', 'SNUC2A_LARGE_INTESTINE': 'ACH-000967', 'HEC151_ENDOMETRIUM': 'ACH-000972', 'MSTO211H_PLEURA': 'ACH-000335', 'RAJI_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000654', 'SKMEL1_SKIN': 'ACH-000465', 'SUPT1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000953', 'HUT102_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000612', '5637_URINARY_TRACT': 'ACH-000905', 'DU4475_BREAST': 'ACH-000258', 'NCIH508_LARGE_INTESTINE': 'ACH-000360', 'NCIH1651_LUNG': 'ACH-000893', 'SKUT1_SOFT_TISSUE': 'ACH-000939', 'MDAMB175VII_BREAST': 'ACH-000759', 'SUPB15_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000059', 'BECKER_CENTRAL_NERVOUS_SYSTEM': 'ACH-001016', 'NCIH2342_LUNG': 'ACH-000951', 'EBC1_LUNG': 'ACH-000563', 'HS936T_SKIN': 'ACH-000801', 'T173_FIBROBLAST': 'ACH-000529', 'NCIH322_LUNG': 'ACH-000837', 'HS618T_FIBROBLAST': 'ACH-000539', 'HS870T_FIBROBLAST': 'ACH-000340', 'HS706T_BONE': 'ACH-000071', 'PANC1_PANCREAS': 'ACH-000164', 'NCIH2291_LUNG': 'ACH-000718', 'T47D_BREAST': 'ACH-000147', 'BICR56_UPPER_AERODIGESTIVE_TRACT': 'ACH-000771', 'BICR16_UPPER_AERODIGESTIVE_TRACT': 'ACH-000503', 'TE10_OESOPHAGUS': 'ACH-000318', 'SKMEL28_SKIN': 'ACH-000615', 'BICR6_UPPER_AERODIGESTIVE_TRACT': 'ACH-000415', 'BHY_UPPER_AERODIGESTIVE_TRACT': 'ACH-000548', 'SH4_SKIN': 'ACH-000441', 'KPL1_BREAST': 'ACH-000028', 'TUHR4TKB_KIDNEY': 'ACH-000495', 'LS123_LARGE_INTESTINE': 'ACH-000501', 'HEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000004', 'HPAC_PANCREAS': 'ACH-000270', 'RCC10RGB_KIDNEY': 'ACH-000189', 'JHOM1_OVARY': 'ACH-000237', 'SU8686_PANCREAS': 'ACH-000114', 'HCC33_LUNG': 'ACH-000515', 'TE6_OESOPHAGUS': 'ACH-000605', 'JHOC5_OVARY': 'ACH-000324', 'ACCMESO1_PLEURA': 'ACH-000086', 'BL41_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000245', 'SNU1214_UPPER_AERODIGESTIVE_TRACT': 'ACH-000715', 'CORL105_LUNG': 'ACH-000161', 'LMSU_STOMACH': 'ACH-000255', 'SNU1197_LARGE_INTESTINE': 'ACH-000412', 'COLO668_LUNG': 'ACH-000803', 'OSRC2_KIDNEY': 'ACH-000159', 'HUH6_LIVER': 'ACH-000671', 'PANC0203_PANCREAS': 'ACH-000042', 'CJM_SKIN': 'ACH-000458', 'BICR22_UPPER_AERODIGESTIVE_TRACT': 'ACH-000794', 'TM31_CENTRAL_NERVOUS_SYSTEM': 'ACH-000592', 'SNU1272_KIDNEY': 'ACH-000513', 'KPNRTBM1_AUTONOMIC_GANGLIA': 'ACH-000345', 'BICR18_UPPER_AERODIGESTIVE_TRACT': 'ACH-000992', 'TE4_OESOPHAGUS': 'ACH-000917', 'PCM6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-001162', 'ECC10_STOMACH': 'ACH-000560', 'TUHR14TKB_KIDNEY': 'ACH-000317', 'YD15_SALIVARY_GLAND': 'ACH-000836', 'UBLC1_URINARY_TRACT': 'ACH-000741', 'ECGI10_OESOPHAGUS': 'ACH-000305', 'CW2_LARGE_INTESTINE': 'ACH-000998', 'SNU761_LIVER': 'ACH-000537', 'SNU61_LARGE_INTESTINE': 'ACH-000532', 'NCIH1915_LUNG': 'ACH-000434', 'SNU324_PANCREAS': 'ACH-000933', 'YD10B_UPPER_AERODIGESTIVE_TRACT': 'ACH-000723', 'HCC2279_LUNG': 'ACH-000731', 'TUHR10TKB_KIDNEY': 'ACH-000459', 'HEC1B_ENDOMETRIUM': 'ACH-000941', 'TGBC11TKB_STOMACH': 'ACH-000949', 'OVK18_OVARY': 'ACH-000947', 'SNU1077_ENDOMETRIUM': 'ACH-000302', 'SNU213_PANCREAS': 'ACH-000266', 'HT115_LARGE_INTESTINE': 'ACH-000986', 'TE14_OESOPHAGUS': 'ACH-000726', 'TE8_OESOPHAGUS': 'ACH-000452', 'SNU520_STOMACH': 'ACH-000908', 'JHUEM3_ENDOMETRIUM': 'ACH-000173', 'LI7_LIVER': 'ACH-000471', 'NCIH838_LUNG': 'ACH-000416', 'SNU119_OVARY': 'ACH-000635', 'SNUC5_LARGE_INTESTINE': 'ACH-000970', 'GSS_STOMACH': 'ACH-000746', 'TALL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000197', 'HCC1171_LUNG': 'ACH-000578', 'SNUC4_LARGE_INTESTINE': 'ACH-000959', 'SNU620_STOMACH': 'ACH-000325', 'ECC12_STOMACH': 'ACH-000225', 'GSU_STOMACH': 'ACH-000485', 'JHOM2B_OVARY': 'ACH-000333', 'MDAMB231_BREAST': 'ACH-000768', 'OV90_OVARY': 'ACH-000291', 'RS411_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000874', 'OVTOKO_OVARY': 'ACH-000663', 'OVCAR4_OVARY': 'ACH-000617', 'HLF_LIVER': 'ACH-000393', 'HCT15_LARGE_INTESTINE': 'ACH-000997', 'SW1463_LARGE_INTESTINE': 'ACH-000470', 'SF295_CENTRAL_NERVOUS_SYSTEM': 'ACH-000376', 'WM115_SKIN': 'ACH-000304', 'JHH4_LIVER': 'ACH-000476', 'CCK81_LARGE_INTESTINE': 'ACH-000963', 'SJSA1_BONE': 'ACH-000748', 'NCIH2228_LUNG': 'ACH-000447', 'RERFLCAI_LUNG': 'ACH-000261', 'SKMES1_LUNG': 'ACH-000665', 'FU97_STOMACH': 'ACH-000633', 'COLO800_SKIN': 'ACH-000401', 'NCIH1963_LUNG': 'ACH-000729', 'RL952_ENDOMETRIUM': 'ACH-000965', 'J82_URINARY_TRACT': 'ACH-000396', 'HCC15_LUNG': 'ACH-000878', 'M059K_CENTRAL_NERVOUS_SYSTEM': 'ACH-000152', 'CALU6_LUNG': 'ACH-000264', 'SKLU1_LUNG': 'ACH-000309', 'SW900_LUNG': 'ACH-000669', 'UACC62_SKIN': 'ACH-000425', 'MFE296_ENDOMETRIUM': 'ACH-000879', 'GOS3_CENTRAL_NERVOUS_SYSTEM': 'ACH-000027', 'GI1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000756', 'A204_SOFT_TISSUE': 'ACH-000201', 'PATU8902_PANCREAS': 'ACH-000599', 'NCIH810_LUNG': 'ACH-000789', 'RERFLCMS_LUNG': 'ACH-000062', 'L1236_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000702', 'DANG_PANCREAS': 'ACH-000243', 'NCIH460_LUNG': 'ACH-000463', 'KYSE140_OESOPHAGUS': 'ACH-000823', 'KYSE180_OESOPHAGUS': 'ACH-000693', 'HUPT3_PANCREAS': 'ACH-000118', 'EFE184_ENDOMETRIUM': 'ACH-000435', 'GCIY_STOMACH': 'ACH-000047', 'SH10TC_STOMACH': 'ACH-000764', 'CAOV4_OVARY': 'ACH-000103', 'HCC2218_BREAST': 'ACH-000755', 'KMS21BM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000598', 'KMS27_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000576', 'KO52_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000498', 'KMS28BM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000419', 'KNS60_CENTRAL_NERVOUS_SYSTEM': 'ACH-000445', 'KNS81_CENTRAL_NERVOUS_SYSTEM': 'ACH-000479', 'KPNSI9S_AUTONOMIC_GANGLIA': 'ACH-000446', 'LU99_LUNG': 'ACH-000444', 'KMS34_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000541', 'MKN1_STOMACH': 'ACH-000351', 'MKN74_STOMACH': 'ACH-000758', 'NCO2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000076', 'KHM1B_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000564', 'HSC2_UPPER_AERODIGESTIVE_TRACT': 'ACH-000472', 'KG1C_CENTRAL_NERVOUS_SYSTEM': 'ACH-000126', 'HUH7_LIVER': 'ACH-000480', 'KMBC2_URINARY_TRACT': 'ACH-000862', 'KMRC20_KIDNEY': 'ACH-000250', 'HUCCT1_BILIARY_TRACT': 'ACH-000976', 'KIJK_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000773', 'A4FUK_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000157', 'KALS1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000231', 'HEC108_ENDOMETRIUM': 'ACH-000990', 'HEL9217_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000005', 'HUH28_BILIARY_TRACT': 'ACH-000808', 'HEC265_ENDOMETRIUM': 'ACH-000946', 'CAL12T_LUNG': 'ACH-000826', 'THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000146', 'A3KAW_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000697', 'SKNSH_AUTONOMIC_GANGLIA': 'ACH-000149', 'HCC1954_BREAST': 'ACH-000859', 'CAMA1_BREAST': 'ACH-000783', 'HEC6_ENDOMETRIUM': 'ACH-000984', 'AM38_CENTRAL_NERVOUS_SYSTEM': 'ACH-000269', 'HEC50B_ENDOMETRIUM': 'ACH-000831', 'HSC3_UPPER_AERODIGESTIVE_TRACT': 'ACH-000778', 'HSC4_UPPER_AERODIGESTIVE_TRACT': 'ACH-000546', 'KMS11_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000714', 'COV504_OVARY': 'ACH-001048', 'VMRCRCW_KIDNEY': 'ACH-000484', 'HGC27_STOMACH': 'ACH-000847', 'LUDLU1_LUNG': 'ACH-000390', 'HUG1N_STOMACH': 'ACH-000239', 'U266B1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000626', 'NUGC4_STOMACH': 'ACH-000674', 'HMC18_BREAST': 'ACH-000721', 'CAS1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000464', 'KMM1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000889', 'CORL23_LUNG': 'ACH-000662', 'MORCPR_LUNG': 'ACH-000851', 'HARA_LUNG': 'ACH-000843', 'SKOV3_OVARY': 'ACH-000811', 'HPAFII_PANCREAS': 'ACH-000094', 'NCIH1568_LUNG': 'ACH-000869', 'CA46_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000440', 'A498_KIDNEY': 'ACH-000555', 'BCP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000923', 'A673_BONE': 'ACH-000052', 'AU565_BREAST': 'ACH-000248', 'COLO201_LARGE_INTESTINE': 'ACH-000253', 'NCIH358_LUNG': 'ACH-000860', 'CFPAC1_PANCREAS': 'ACH-000138', 'NIHOVCAR3_OVARY': 'ACH-000001', 'NCIH1650_LUNG': 'ACH-000035', 'NCIH1793_LUNG': 'ACH-000888', 'NCIH1437_LUNG': 'ACH-000589', 'KNS62_LUNG': 'ACH-000858', 'HS578T_BREAST': 'ACH-000148', 'BT20_BREAST': 'ACH-000536', 'SNU182_LIVER': 'ACH-000483', 'SKNMC_BONE': 'ACH-000039', 'HH_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000061', 'FADU_UPPER_AERODIGESTIVE_TRACT': 'ACH-000846', 'MDAMB361_BREAST': 'ACH-000934', 'MDAMB436_BREAST': 'ACH-000573', 'PANC0213_PANCREAS': 'ACH-000031', 'DB_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000334', 'RDES_BONE': 'ACH-000041', 'CAL27_UPPER_AERODIGESTIVE_TRACT': 'ACH-000832', 'MESSA_SOFT_TISSUE': 'ACH-000449', 'SW1271_LUNG': 'ACH-000890', 'NCIH1792_LUNG': 'ACH-000496', 'GCT_SOFT_TISSUE': 'ACH-000835', 'SKBR3_BREAST': 'ACH-000017', 'NCIH82_LUNG': 'ACH-000355', 'NCIN87_STOMACH': 'ACH-000427', 'U2OS_BONE': 'ACH-000364', 'H4_CENTRAL_NERVOUS_SYSTEM': 'ACH-000389', 'GDM1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000081', 'HCC1143_BREAST': 'ACH-000374', 'T24_URINARY_TRACT': 'ACH-000018', 'G401_SOFT_TISSUE': 'ACH-000096', 'HCC1937_BREAST': 'ACH-000223', 'HCC1569_BREAST': 'ACH-000930', 'OCUM1_STOMACH': 'ACH-000247', 'SAOS2_BONE': 'ACH-000410', 'D283MED_CENTRAL_NERVOUS_SYSTEM': 'ACH-000055', 'SCC9_UPPER_AERODIGESTIVE_TRACT': 'ACH-000181', 'HCC1395_BREAST': 'ACH-000699', 'NMCG1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000200', 'TEN_ENDOMETRIUM': 'ACH-000397', 'TE9_OESOPHAGUS': 'ACH-000694', 'KYSE520_OESOPHAGUS': 'ACH-000637', 'YH13_CENTRAL_NERVOUS_SYSTEM': 'ACH-000469', 'IALM_LUNG': 'ACH-000672', 'NCIH2052_PLEURA': 'ACH-000153', 'SKNFI_AUTONOMIC_GANGLIA': 'ACH-000341', 'SKNAS_AUTONOMIC_GANGLIA': 'ACH-000260', 'GA10_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000162', 'SNU398_LIVER': 'ACH-000221', 'SNU423_LIVER': 'ACH-000493', 'SCC4_UPPER_AERODIGESTIVE_TRACT': 'ACH-000238', 'EB1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000877', 'HCC1187_BREAST': 'ACH-000111', 'SNU387_LIVER': 'ACH-000478', 'DAUDI_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000786', 'HT_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000914', 'MJ_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000077', 'JM1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000151', 'C2BBE1_LARGE_INTESTINE': 'ACH-000009', 'SKES1_BONE': 'ACH-000087', 'EPLC272H_LUNG': 'ACH-000585', 'BXPC3_PANCREAS': 'ACH-000535', 'SNU5_STOMACH': 'ACH-000303', 'SNU475_LIVER': 'ACH-000422', 'HEC251_ENDOMETRIUM': 'ACH-000996', 'HCC95_LUNG': 'ACH-000454', '253J_URINARY_TRACT': 'ACH-000011', 'HCC1438_LUNG': 'ACH-000891', 'RPMI7951_SKIN': 'ACH-000348', 'HCC2108_LUNG': 'ACH-000314', 'COLO783_SKIN': 'ACH-000812', 'HCC1359_LUNG': 'ACH-000901', 'BICR31_UPPER_AERODIGESTIVE_TRACT': 'ACH-000228', 'COV362_OVARY': 'ACH-000278', 'COLO792_SKIN': 'ACH-000968', 'COV434_OVARY': 'ACH-000123', 'WM983B_SKIN': 'ACH-000765', 'HS751T_FIBROBLAST': 'ACH-000165', 'LS513_LARGE_INTESTINE': 'ACH-000007', 'NCIH2110_LUNG': 'ACH-000929', 'CHAGOK1_LUNG': 'ACH-000867', 'DMS454_LUNG': 'ACH-000844', 'CORL47_LUNG': 'ACH-000695', 'FTC238_THYROID': 'ACH-000897', 'DMS273_LUNG': 'ACH-000749', 'FTC133_THYROID': 'ACH-000903', 'CORL311_LUNG': 'ACH-000187', 'NCIH2286_LUNG': 'ACH-000912', 'NCIH1339_LUNG': 'ACH-000921', 'WM793_SKIN': 'ACH-000827', 'A704_KIDNEY': 'ACH-000429', 'WM1799_SKIN': 'ACH-000661', 'MEWO_SKIN': 'ACH-000987', 'WM88_SKIN': 'ACH-000899', 'NCIH2081_LUNG': 'ACH-000394', 'NCIH1092_LUNG': 'ACH-000514', 'SW837_LARGE_INTESTINE': 'ACH-000421', 'IGR1_SKIN': 'ACH-000882', 'NCIH1105_LUNG': 'ACH-000780', 'NCIH1436_LUNG': 'ACH-000830', 'NCIH1876_LUNG': 'ACH-000586', 'CORL24_LUNG': 'ACH-000382', 'COV644_OVARY': 'ACH-000608', 'GP2D_LARGE_INTESTINE': 'ACH-000982', 'PECAPJ15_UPPER_AERODIGESTIVE_TRACT': 'ACH-000619', 'CORL88_LUNG': 'ACH-000508', 'CORL95_LUNG': 'ACH-000743', 'PECAPJ49_UPPER_AERODIGESTIVE_TRACT': 'ACH-000735', 'PECAPJ34CLONEC12_UPPER_AERODIGESTIVE_TRACT': 'ACH-000606', 'PECAPJ41CLONED2_UPPER_AERODIGESTIVE_TRACT': 'ACH-000732', 'OV7_OVARY': 'ACH-000688', 'OAW42_OVARY': 'ACH-000704', 'MDST8_LARGE_INTESTINE': 'ACH-000935', 'OAW28_OVARY': 'ACH-000116', 'SUIT2_PANCREAS': 'ACH-000652', 'SUPM2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000226', 'SKHEP1_LIVER': 'ACH-000361', 'OVCAR8_OVARY': 'ACH-000696', 'NCIH2172_LUNG': 'ACH-000924', 'NCIH596_LUNG': 'ACH-000628', 'NCIH650_LUNG': 'ACH-000945', 'SW1573_LUNG': 'ACH-000677', 'OVMANA_OVARY': 'ACH-000646', 'SKCO1_LARGE_INTESTINE': 'ACH-000400', 'YAPC_PANCREAS': 'ACH-000332', 'HT1197_URINARY_TRACT': 'ACH-000547', 'A253_SALIVARY_GLAND': 'ACH-000740', 'HCC366_LUNG': 'ACH-000840', 'DKMG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000244', 'UMUC1_URINARY_TRACT': 'ACH-000834', 'NAMALWA_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000944', 'NCIH292_LUNG': 'ACH-001075', 'U118MG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000040', 'HCC1833_LUNG': 'ACH-000712', 'RH30_SOFT_TISSUE': 'ACH-000833', 'TCCSUP_URINARY_TRACT': 'ACH-000720', 'SNU601_STOMACH': 'ACH-000736', 'CAL33_UPPER_AERODIGESTIVE_TRACT': 'ACH-000518', 'SNU668_STOMACH': 'ACH-000344', 'CORL279_LUNG': 'ACH-000257', 'GB1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000738', 'COLO684_ENDOMETRIUM': 'ACH-000864', 'SNU685_ENDOMETRIUM': 'ACH-000407', 'SNU886_LIVER': 'ACH-000316', 'NCCSTCK140_STOMACH': 'ACH-000110', 'NCIH3255_LUNG': 'ACH-000109', 'SNU719_STOMACH': 'ACH-000898', 'SNU489_CENTRAL_NERVOUS_SYSTEM': 'ACH-000543', 'SNU81_LARGE_INTESTINE': 'ACH-000991', 'A2058_SKIN': 'ACH-000788', 'SNU1040_LARGE_INTESTINE': 'ACH-000999', 'SNU1076_UPPER_AERODIGESTIVE_TRACT': 'ACH-000549', 'SNU175_LARGE_INTESTINE': 'ACH-000989', 'SNU407_LARGE_INTESTINE': 'ACH-000955', 'SNU201_CENTRAL_NERVOUS_SYSTEM': 'ACH-000623', 'SNU216_STOMACH': 'ACH-000466', 'SNU1196_BILIARY_TRACT': 'ACH-000461', 'SNU1079_BILIARY_TRACT': 'ACH-000209', 'HCC1195_LUNG': 'ACH-000868', 'NCIH854_LUNG': 'ACH-000710', 'YD38_UPPER_AERODIGESTIVE_TRACT': 'ACH-000762', 'SNU1105_CENTRAL_NERVOUS_SYSTEM': 'ACH-000368', 'SNU283_LARGE_INTESTINE': 'ACH-000708', 'SNU46_UPPER_AERODIGESTIVE_TRACT': 'ACH-000500', 'SNU308_BILIARY_TRACT': 'ACH-000141', 'YD8_UPPER_AERODIGESTIVE_TRACT': 'ACH-000630', 'SNU478_BILIARY_TRACT': 'ACH-000377', 'SNU410_PANCREAS': 'ACH-000517', 'SNU899_UPPER_AERODIGESTIVE_TRACT': 'ACH-000692', 'SNU503_LARGE_INTESTINE': 'ACH-000683', 'SET2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000195', 'SNU738_CENTRAL_NERVOUS_SYSTEM': 'ACH-000807', 'SNU626_CENTRAL_NERVOUS_SYSTEM': 'ACH-000370', 'SNU869_BILIARY_TRACT': 'ACH-000182', 'SNU878_LIVER': 'ACH-000686', 'SNU840_OVARY': 'ACH-000280', 'SNU8_OVARY': 'ACH-000460', 'NCIH684_LIVER': 'ACH-000089', 'IGR39_SKIN': 'ACH-000550', 'DAOY_CENTRAL_NERVOUS_SYSTEM': 'ACH-000211', 'HEP3B217_LIVER': 'ACH-000625', 'KYM1_SOFT_TISSUE': 'ACH-000607', 'LC1F_LUNG': 'ACH-000705', 'CMK_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000641', 'JMSU1_URINARY_TRACT': 'ACH-000753', 'IMR32_AUTONOMIC_GANGLIA': 'ACH-000310', 'SUDHL8_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000656', 'T3M4_PANCREAS': 'ACH-000085', 'PK45H_PANCREAS': 'ACH-000468', 'HEC59_ENDOMETRIUM': 'ACH-000994', 'T3M10_LUNG': 'ACH-000813', 'NUGC2_STOMACH': 'ACH-000761', 'PLCPRF5_LIVER': 'ACH-001318', 'KPNYN_AUTONOMIC_GANGLIA': 'ACH-000227', 'CCFSTTG1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000329', 'NCIH929_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000050', 'KMRC3_KIDNEY': 'ACH-000313', 'NCIH2023_LUNG': 'ACH-000781', 'DMS153_LUNG': 'ACH-000594', 'NCIH1048_LUNG': 'ACH-000866', 'D341MED_CENTRAL_NERVOUS_SYSTEM': 'ACH-000095', 'NCIH660_PROSTATE': 'ACH-000177', 'NCIH1618_LUNG': 'ACH-000179', 'HS675T_FIBROBLAST': 'ACH-000214', 'PEER_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000519', 'SKM1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000373', 'HT144_SKIN': 'ACH-000322', 'SKMEL31_SKIN': 'ACH-000640', 'HS839T_FIBROBLAST': 'ACH-000180', 'HS939T_SKIN': 'ACH-000814', 'OCILY3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000158', 'NCIH69_LUNG': 'ACH-000358', 'DV90_LUNG': 'ACH-000925', 'ZR751_BREAST': 'ACH-000097', 'KG1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000386', 'SKNBE2_AUTONOMIC_GANGLIA': 'ACH-000312', 'HCC70_BREAST': 'ACH-000668', 'NCIH2452_PLEURA': 'ACH-000092', 'SNU1_STOMACH': 'ACH-000932', 'LOUCY_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000104', 'HCC56_LARGE_INTESTINE': 'ACH-000467', 'NCIH1648_LUNG': 'ACH-000766', 'TO175T_FIBROBLAST': 'ACH-000125', 'PANC0813_PANCREAS': 'ACH-000417', 'MG63_BONE': 'ACH-000359', 'PK1_PANCREAS': 'ACH-000307', 'NCIH23_LUNG': 'ACH-000900', 'TE617T_SOFT_TISSUE': 'ACH-000051', 'SW1088_CENTRAL_NERVOUS_SYSTEM': 'ACH-000437', 'ISTMES2_PLEURA': 'ACH-000331', 'NCIH1373_LUNG': 'ACH-000845', 'NCIH1930_LUNG': 'ACH-000870', 'NCIH1623_LUNG': 'ACH-000744', 'HS229T_FIBROBLAST': 'ACH-000131', 'NCIH1838_LUNG': 'ACH-000733', 'NCIH1836_LUNG': 'ACH-000559', 'HS688AT_FIBROBLAST': 'ACH-000306', 'PANC0504_PANCREAS': 'ACH-000093', 'BL70_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000402', 'CI1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000687', 'G292CLONEA141B1_BONE': 'ACH-000082', 'RCM1_LARGE_INTESTINE': 'ACH-000565', 'HS616T_FIBROBLAST': 'ACH-000063', 'A101D_SKIN': 'ACH-000008', 'RERFLCKJ_LUNG': 'ACH-000482', 'SW780_URINARY_TRACT': 'ACH-000384', 'HS683_CENTRAL_NERVOUS_SYSTEM': 'ACH-000067', 'RKO_LARGE_INTESTINE': 'ACH-000943', 'HS729_SOFT_TISSUE': 'ACH-000133', 'JHOS2_OVARY': 'ACH-000132', 'MCF7_BREAST': 'ACH-000019', 'U937_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000406', 'HCC202_BREAST': 'ACH-000725', 'MEG01_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000072', 'REH_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000960', 'ONS76_CENTRAL_NERVOUS_SYSTEM': 'ACH-000776', 'SNUC1_LARGE_INTESTINE': 'ACH-000722', 'SKNDZ_AUTONOMIC_GANGLIA': 'ACH-000366', 'DMS114_LUNG': 'ACH-000530', 'ST486_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000567', 'SW1353_BONE': 'ACH-000418', 'MDAMB157_BREAST': 'ACH-000621', '769P_KIDNEY': 'ACH-000411', '786O_KIDNEY': 'ACH-000649', 'CAOV3_OVARY': 'ACH-000713', 'HEPG2_LIVER': 'ACH-000739', 'NCIH524_LUNG': 'ACH-000816', 'NCIH209_LUNG': 'ACH-000290', 'MCAS_OVARY': 'ACH-000796', 'SBC5_LUNG': 'ACH-000670', 'COLO829_SKIN': 'ACH-000644', 'VCAP_PROSTATE': 'ACH-000115', 'IPC298_SKIN': 'ACH-000915', 'CAPAN2_PANCREAS': 'ACH-000107', 'EFM192A_BREAST': 'ACH-000117', 'HS766T_PANCREAS': 'ACH-000178', 'EFO27_OVARY': 'ACH-000936', 'A2780_OVARY': 'ACH-000657', 'NCIH196_LUNG': 'ACH-000752', 'IGROV1_OVARY': 'ACH-000966', 'NCIH889_LUNG': 'ACH-000297', 'NCIH211_LUNG': 'ACH-000639', 'MDAPCA2B_PROSTATE': 'ACH-000952', 'SKMEL24_SKIN': 'ACH-000822', 'K029AX_SKIN': 'ACH-000404', 'NCIH2171_LUNG': 'ACH-000525', 'PANC1005_PANCREAS': 'ACH-000060', 'HS695T_SKIN': 'ACH-000799', 'LOUNH91_LUNG': 'ACH-000176', 'TOV21G_OVARY': 'ACH-000885', 'SKMEL3_SKIN': 'ACH-000423', 'ASPC1_PANCREAS': 'ACH-000222', 'OVSAHO_OVARY': 'ACH-000409', 'NCIH441_LUNG': 'ACH-000638', 'NCIH2030_LUNG': 'ACH-000521', 'HMCB_SKIN': 'ACH-000931', 'FUOV1_OVARY': 'ACH-000574', 'MALME3M_SKIN': 'ACH-000477', 'KMS12BM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000380', 'HL60_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000002', 'DBTRG05MG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000863', 'HEYA8_OVARY': 'ACH-000542', 'NCIH2122_LUNG': 'ACH-000311', 'MELJUSO_SKIN': 'ACH-000881', 'SW620_LARGE_INTESTINE': 'ACH-000651', 'HCT116_LARGE_INTESTINE': 'ACH-000971', 'VMRCRCZ_KIDNEY': 'ACH-000171', 'DU145_PROSTATE': 'ACH-000979', 'MM1S_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000763', 'L33_PANCREAS': 'ACH-000685', 'SKMEL2_SKIN': 'ACH-001190', 'A549_LUNG': 'ACH-000681', 'A172_CENTRAL_NERVOUS_SYSTEM': 'ACH-000558', 'NCIH1975_LUNG': 'ACH-000587', 'COLO679_SKIN': 'ACH-000805', 'HUH1_LIVER': 'ACH-000475', 'MDAMB453_BREAST': 'ACH-000910', 'PANC0403_PANCREAS': 'ACH-000235', 'TYKNU_OVARY': 'ACH-000430', 'OVISE_OVARY': 'ACH-000527', 'EFM19_BREAST': 'ACH-000330', 'MELHO_SKIN': 'ACH-000450', 'RL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000371', 'G361_SKIN': 'ACH-000572', 'NCIH647_LUNG': 'ACH-000378', 'GMS10_CENTRAL_NERVOUS_SYSTEM': 'ACH-000102', 'SEM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000782', 'HLFA_FIBROBLAST': 'ACH-000797', '42MGBA_CENTRAL_NERVOUS_SYSTEM': 'ACH-000323', 'HCC44_LUNG': 'ACH-000667', 'KYSE70_OESOPHAGUS': 'ACH-000784', 'GAMG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000098', 'NCIH522_LUNG': 'ACH-000343', 'EN_ENDOMETRIUM': 'ACH-000978', 'PK59_PANCREAS': 'ACH-000205', 'COLO741_SKIN': 'ACH-000582', 'HDMYZ_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000190', 'NCIH661_LUNG': 'ACH-000853', 'LS411N_LARGE_INTESTINE': 'ACH-000985', 'HCC78_LUNG': 'ACH-000562', 'YKG1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000570', 'ESS1_ENDOMETRIUM': 'ACH-000913', 'U251MG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000232', 'TE11_OESOPHAGUS': 'ACH-000488', 'HEC1A_ENDOMETRIUM': 'ACH-000954', 'TE1_OESOPHAGUS': 'ACH-000647', 'JHUEM2_ENDOMETRIUM': 'ACH-000909', 'RERFLCAD1_LUNG': 'ACH-000791', 'TE5_OESOPHAGUS': 'ACH-000408', 'NCIH1155_LUNG': 'ACH-000980', 'TE15_OESOPHAGUS': 'ACH-000353', 'ABC1_LUNG': 'ACH-000528', '8MGBA_CENTRAL_NERVOUS_SYSTEM': 'ACH-000137', 'AGS_STOMACH': 'ACH-000880', 'QGP1_PANCREAS': 'ACH-000347', 'DETROIT562_UPPER_AERODIGESTIVE_TRACT': 'ACH-000207', 'BT474_BREAST': 'ACH-000927', 'ACHN_KIDNEY': 'ACH-000046', 'LOVO_LARGE_INTESTINE': 'ACH-000950', 'SCABER_URINARY_TRACT': 'ACH-000839', 'SKMEL30_SKIN': 'ACH-000810', 'OVKATE_OVARY': 'ACH-000443', '22RV1_PROSTATE': 'ACH-000956', 'C32_SKIN': 'ACH-000580', 'PANC0327_PANCREAS': 'ACH-000139', 'RPMI8226_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000817', 'ES2_OVARY': 'ACH-000906', 'NCIH1395_LUNG': 'ACH-000327', 'NCIH1781_LUNG': 'ACH-000379', 'G402_SOFT_TISSUE': 'ACH-000375', 'NCIH146_LUNG': 'ACH-000506', 'SW48_LARGE_INTESTINE': 'ACH-000958', 'CHP212_AUTONOMIC_GANGLIA': 'ACH-000120', 'LS180_LARGE_INTESTINE': 'ACH-000957', 'SW403_LARGE_INTESTINE': 'ACH-000820', 'NCIH747_LARGE_INTESTINE': 'ACH-000403', 'RT4_URINARY_TRACT': 'ACH-000242', 'KATOIII_STOMACH': 'ACH-000793', 'BDCM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000080', 'SW1417_LARGE_INTESTINE': 'ACH-000236', 'SW948_LARGE_INTESTINE': 'ACH-000680', 'HS606T_FIBROBLAST': 'ACH-000540', 'MDAMB415_BREAST': 'ACH-000876', 'DMS79_LUNG': 'ACH-000703', 'NCIH2106_LUNG': 'ACH-000904', 'SW579_THYROID': 'ACH-000163', 'MDAMB468_BREAST': 'ACH-000849', 'NCIH526_LUNG': 'ACH-000767', 'TOV112D_OVARY': 'ACH-000048', 'BT549_BREAST': 'ACH-000288', 'NCIH2196_LUNG': 'ACH-000399', 'CAL851_BREAST': 'ACH-000857', 'EFO21_OVARY': 'ACH-000308', 'A375_SKIN': 'ACH-000219', 'NCIH446_LUNG': 'ACH-000800', 'HS294T_SKIN': 'ACH-000014', 'SKMEL5_SKIN': 'ACH-000730', 'HS944T_SKIN': 'ACH-000632', 'TCCPAN2_PANCREAS': 'ACH-000502', 'DOV13_OVARY': 'ACH-001063', 'RMGI_OVARY': 'ACH-000719', 'PC3_PROSTATE': 'ACH-000090', 'K562_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000551', 'RT112_URINARY_TRACT': 'ACH-000473', 'NUGC3_STOMACH': 'ACH-000911', 'U87MG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000075', 'OE19_OESOPHAGUS': 'ACH-000679', 'KMRC2_KIDNEY': 'ACH-000709', 'WM2664_SKIN': 'ACH-001239', 'KMRC1_KIDNEY': 'ACH-000684', 'KM12_LARGE_INTESTINE': 'ACH-000969', 'T84_LARGE_INTESTINE': 'ACH-000381', 'CALU1_LUNG': 'ACH-000511', 'LOXIMVI_SKIN': 'ACH-000750', 'NCIH2170_LUNG': 'ACH-000481', 'SQ1_LUNG': 'ACH-000553', 'NCIH2444_LUNG': 'ACH-000186', 'UACC257_SKIN': 'ACH-000579', 'KYSE510_OESOPHAGUS': 'ACH-000824', 'MOLM13_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000362', 'KYSE30_OESOPHAGUS': 'ACH-000777', 'KYSE150_OESOPHAGUS': 'ACH-000855', 'PSN1_PANCREAS': 'ACH-000320', 'PATU8988T_PANCREAS': 'ACH-000023', 'IGR37_SKIN': 'ACH-000650', 'MOLP2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000453', 'SCC25_UPPER_AERODIGESTIVE_TRACT': 'ACH-000188', 'MHHNB11_AUTONOMIC_GANGLIA': 'ACH-000078', 'HCC1806_BREAST': 'ACH-000624', 'GRANTA519_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000073', 'MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000006', 'MKN7_STOMACH': 'ACH-000678', 'SW1710_URINARY_TRACT': 'ACH-000566', 'MOLT13_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000795', 'SR786_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000338', 'SCC15_UPPER_AERODIGESTIVE_TRACT': 'ACH-000254', 'NCIH2126_LUNG': 'ACH-000785', 'HCC827_LUNG': 'ACH-000012', 'RD_SOFT_TISSUE': 'ACH-000169', 'TT_THYROID': 'ACH-001321', 'TF1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000387', '8305C_THYROID': 'ACH-001306', 'UACC812_BREAST': 'ACH-000568', 'ALLSIL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000105', 'KASUMI2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000728', 'DND41_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000981', 'ME1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000439', 'JEKO1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000357', 'NUDHL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000388', 'CL14_LARGE_INTESTINE': 'ACH-000342', 'KYO1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000604', 'JL1_PLEURA': 'ACH-000645', 'NALM19_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000130', 'HPBALL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000942', 'SUPT11_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000122', 'CL40_LARGE_INTESTINE': 'ACH-000798', 'MDAMB134VI_BREAST': 'ACH-000044', 'HCC1500_BREAST': 'ACH-000349', 'HCC1599_BREAST': 'ACH-000196', 'MOTN1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-001130', 'OCIAML3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000336', 'SUDHL10_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000271', 'SUDHL4_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000365', 'OCILY19_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000124', 'MOLM6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000321', 'SKMM2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000363', 'KASUMI6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000166', 'HS698T_FIBROBLAST': 'ACH-000850', 'UACC893_BREAST': 'ACH-000554', 'MINO_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000220', 'CHP126_AUTONOMIC_GANGLIA': 'ACH-000136', 'COLO680N_OESOPHAGUS': 'ACH-000717', 'COLO678_LARGE_INTESTINE': 'ACH-000350', 'LAMA84_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000301', 'CMLT1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000920', 'EHEB_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000038', 'CAL54_KIDNEY': 'ACH-000457', 'BV173_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000432', 'DOHH2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000056', 'EOL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000198', 'NALM6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000938', 'VMCUB1_URINARY_TRACT': 'ACH-000545', 'LP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000204', 'LXF289_LUNG': 'ACH-000787', 'MONOMAC1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-001129', 'PL21_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000218', 'PF382_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000937', 'NALM1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000462', 'OCIAML2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000113', 'MHHCALL3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000032', 'JURLMK1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000326', 'OCIAML5_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000065', 'JK1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000241', 'RI1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000398', 'KARPAS422_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000315', 'L540_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000806', 'HDQP1_BREAST': 'ACH-000643', 'MHHCALL4_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000156', 'ONCODG1_OVARY': 'ACH-000013', 'HDLM2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000267', 'CAL29_URINARY_TRACT': 'ACH-000142', 'KOPN8_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-001106', 'SCLC21H_LUNG': 'ACH-000659', 'CALU3_LUNG': 'ACH-000392', 'NCIH1435_LUNG': 'ACH-000852', 'NCIH2009_LUNG': 'ACH-000886', 'HS852T_SKIN': 'ACH-000274', 'HCC1419_BREAST': 'ACH-000277', 'BT483_BREAST': 'ACH-000818', 'HS281T_FIBROBLAST': 'ACH-000083', 'HS934T_FIBROBLAST': 'ACH-000194', 'HCC4006_LUNG': 'ACH-000066', 'HCC1428_BREAST': 'ACH-000352', 'KU812_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000074', 'SHP77_LUNG': 'ACH-000790', 'TE441T_SOFT_TISSUE': 'ACH-000772', 'NB1_AUTONOMIC_GANGLIA': 'ACH-000804', 'PFEIFFER_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000140', 'ZR7530_BREAST': 'ACH-000828', 'NCIH841_LUNG': 'ACH-000292', 'HS821T_FIBROBLAST': 'ACH-000526', 'NCIH1944_LUNG': 'ACH-000414', 'NCIH2405_LUNG': 'ACH-000121', 'BEN_LUNG': 'ACH-000603', 'COLO320_LARGE_INTESTINE': 'ACH-000202', 'RPMI8402_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000636', 'SUDHL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000664', 'P12ICHIKAWA_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000372', 'BFTC905_URINARY_TRACT': 'ACH-000802', 'BFTC909_KIDNEY': 'ACH-000792', 'JVM3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000346', 'NOMO1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000168', 'NCIH727_LUNG': 'ACH-000775', 'HS343T_FIBROBLAST': 'ACH-000184', 'HCC2935_LUNG': 'ACH-000150', 'HS888T_FIBROBLAST': 'ACH-000154', 'MKN45_STOMACH': 'ACH-000356', '8505C_THYROID': 'ACH-001307', 'KU1919_URINARY_TRACT': 'ACH-000486', 'KCL22_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000983', 'CADOES1_BONE': 'ACH-000210', 'S117_THYROID': 'ACH-000037', 'EM2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000295', 'OPM2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000024', 'CAL62_THYROID': 'ACH-000174', 'LCLC103H_LUNG': 'ACH-000627', 'JURKAT_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000995', '697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000070', 'MHHCALL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000020', 'M07E_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000602', 'L428_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000754', 'MOLT16_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000918', 'NB4_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000294', 'LCLC97TM1_LUNG': 'ACH-000596', '2313287_STOMACH': 'ACH-000948', 'BHT101_THYROID': 'ACH-000191', 'JVM2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000106', 'CAL51_BREAST': 'ACH-000856', 'L363_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000183', 'KMH2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000815', 'SIMA_AUTONOMIC_GANGLIA': 'ACH-000099', 'NUDUL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000287', 'BC3C_URINARY_TRACT': 'ACH-000593', 'SUDHL6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000611', 'REC1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000068', 'OCIM1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000751', 'SIGM5_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000112', 'SUDHL5_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000660', 'BCPAP_THYROID': 'ACH-000456', '639V_URINARY_TRACT': 'ACH-000973', 'CL11_LARGE_INTESTINE': 'ACH-000249', 'TT2609C02_THYROID': 'ACH-000716', 'CL34_LARGE_INTESTINE': 'ACH-000895', 'RCHACV_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000922', 'CAL120_BREAST': 'ACH-000212', '647V_URINARY_TRACT': 'ACH-000896', 'RH18_SOFT_TISSUE': 'ACH-000689', 'MOLM16_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000369', 'F36P_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000487', 'EJM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000821', 'WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000534', 'ML1_THYROID': 'ACH-000058', 'AMO1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000838', 'MOLP8_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000745', 'JHUEM7_ENDOMETRIUM': 'ACH-000993', 'OE21_OESOPHAGUS': 'ACH-000544', 'HOP62_LUNG': 'ACH-000861', 'EKVX_LUNG': 'ACH-000706', 'UO31_KIDNEY': 'ACH-000428', 'SF268_CENTRAL_NERVOUS_SYSTEM': 'ACH-000655', 'SF539_CENTRAL_NERVOUS_SYSTEM': 'ACH-000273', 'SNB75_CENTRAL_NERVOUS_SYSTEM': 'ACH-000504', 'HOP92_LUNG': 'ACH-000825'}\n"
     ]
    }
   ],
   "source": [
    "# get the overlapped cell lines in methy_samples with cell_line_anno_df by merge on cell line id\n",
    "# first format the methy_samples as dataframe\n",
    "methy_samples_df = pd.DataFrame(methy_samples, columns=['CCLE_Name'])\n",
    "# merge the methy_samples_df with the cell_line_anno_df\n",
    "merged_methy_samples_df = pd.merge(methy_samples_df, cell_line_anno_df, left_on='CCLE_Name', right_on='CCLE_ID', how='inner')\n",
    "display(merged_methy_samples_df)\n",
    "\n",
    "# get the map dictionary from the ccle id to depmap id\n",
    "methy_map_dict = dict(zip(merged_methy_samples_df['CCLE_Name'], merged_methy_samples_df['depMapID']))\n",
    "print(methy_map_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CCLE_Name</th>\n",
       "      <th>cleaned_CCLE_Name</th>\n",
       "      <th>CCLE_ID</th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Site_Primary</th>\n",
       "      <th>Site_Subtype1</th>\n",
       "      <th>Site_Subtype2</th>\n",
       "      <th>Site_Subtype3</th>\n",
       "      <th>...</th>\n",
       "      <th>Growth.Medium</th>\n",
       "      <th>Supplements</th>\n",
       "      <th>Freezing.Medium</th>\n",
       "      <th>Doubling.Time.from.Vendor</th>\n",
       "      <th>Doubling.Time.Calculated.hrs</th>\n",
       "      <th>type</th>\n",
       "      <th>type_refined</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>mutRate</th>\n",
       "      <th>tcga_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MDAMB468_BREAST_TenPx01</td>\n",
       "      <td>MDAMB468_BREAST</td>\n",
       "      <td>MDAMB468_BREAST</td>\n",
       "      <td>ACH-000849</td>\n",
       "      <td>MDA-MB-468</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>breast</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>DMEM+10% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>63.5</td>\n",
       "      <td>breast</td>\n",
       "      <td>breast</td>\n",
       "      <td>Breast:Carcinoma</td>\n",
       "      <td>188.252652</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SH4_SKIN_TenPx01</td>\n",
       "      <td>SH4_SKIN</td>\n",
       "      <td>SH4_SKIN</td>\n",
       "      <td>ACH-000441</td>\n",
       "      <td>SH-4</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>skin</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>DMEM+10% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.9</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>Skin:Melanoma</td>\n",
       "      <td>109.541023</td>\n",
       "      <td>SKCM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AU565_BREAST_TenPx01</td>\n",
       "      <td>AU565_BREAST</td>\n",
       "      <td>AU565_BREAST</td>\n",
       "      <td>ACH-000248</td>\n",
       "      <td>AU565</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>breast</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI 1640 + 10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>74.1</td>\n",
       "      <td>breast</td>\n",
       "      <td>breast</td>\n",
       "      <td>Breast:Carcinoma</td>\n",
       "      <td>223.382430</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KMRC1_KIDNEY_TenPx01</td>\n",
       "      <td>KMRC1_KIDNEY</td>\n",
       "      <td>KMRC1_KIDNEY</td>\n",
       "      <td>ACH-000684</td>\n",
       "      <td>KMRC-1</td>\n",
       "      <td>primary</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>DMEM+10% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46.4</td>\n",
       "      <td>kidney</td>\n",
       "      <td>kidney</td>\n",
       "      <td>Kidney:Carcinoma</td>\n",
       "      <td>204.674772</td>\n",
       "      <td>KIRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CAL51_BREAST_TenPx01</td>\n",
       "      <td>CAL51_BREAST</td>\n",
       "      <td>CAL51_BREAST</td>\n",
       "      <td>ACH-000856</td>\n",
       "      <td>CAL-51</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>breast</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>80% Dulbecco's MEM(4.5g/L glucose) +20% h.i.FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10%DMSO</td>\n",
       "      <td>&gt;60 hrs</td>\n",
       "      <td>26.8</td>\n",
       "      <td>breast</td>\n",
       "      <td>breast</td>\n",
       "      <td>Breast:Carcinoma</td>\n",
       "      <td>206.841626</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>22RV1_PROSTATE_TenPx24</td>\n",
       "      <td>22RV1_PROSTATE</td>\n",
       "      <td>22RV1_PROSTATE</td>\n",
       "      <td>ACH-000956</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>primary</td>\n",
       "      <td>prostate</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+10% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58.4</td>\n",
       "      <td>prostate</td>\n",
       "      <td>prostate</td>\n",
       "      <td>Prostate:Carcinoma</td>\n",
       "      <td>539.830868</td>\n",
       "      <td>PRAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>BT20_BREAST_TenPx24</td>\n",
       "      <td>BT20_BREAST</td>\n",
       "      <td>BT20_BREAST</td>\n",
       "      <td>ACH-000536</td>\n",
       "      <td>BT-20</td>\n",
       "      <td>primary</td>\n",
       "      <td>breast</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>breast</td>\n",
       "      <td>breast</td>\n",
       "      <td>Breast:Carcinoma</td>\n",
       "      <td>114.140088</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>CALU6_LUNG_TenPx24</td>\n",
       "      <td>CALU6_LUNG</td>\n",
       "      <td>CALU6_LUNG</td>\n",
       "      <td>ACH-000264</td>\n",
       "      <td>Calu-6</td>\n",
       "      <td>primary</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>EMEM +10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>78.9</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>Lung:NSCLC_Others</td>\n",
       "      <td>87.777181</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>KP4_PANCREAS_TenPx24</td>\n",
       "      <td>KP4_PANCREAS</td>\n",
       "      <td>KP4_PANCREAS</td>\n",
       "      <td>ACH-000265</td>\n",
       "      <td>KP4</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>IMDM+10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>Pancreas:Carcinoma</td>\n",
       "      <td>143.110198</td>\n",
       "      <td>PAAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>OVCAR8_OVARY_TenPx24</td>\n",
       "      <td>OVCAR8_OVARY</td>\n",
       "      <td>OVCAR8_OVARY</td>\n",
       "      <td>ACH-000696</td>\n",
       "      <td>OVCAR-8</td>\n",
       "      <td>primary</td>\n",
       "      <td>ovary</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ovary</td>\n",
       "      <td>ovary</td>\n",
       "      <td>Ovary:Carcinoma</td>\n",
       "      <td>154.941601</td>\n",
       "      <td>OV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>258 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   CCLE_Name cleaned_CCLE_Name          CCLE_ID    depMapID  \\\n",
       "0    MDAMB468_BREAST_TenPx01   MDAMB468_BREAST  MDAMB468_BREAST  ACH-000849   \n",
       "1           SH4_SKIN_TenPx01          SH4_SKIN         SH4_SKIN  ACH-000441   \n",
       "2       AU565_BREAST_TenPx01      AU565_BREAST     AU565_BREAST  ACH-000248   \n",
       "3       KMRC1_KIDNEY_TenPx01      KMRC1_KIDNEY     KMRC1_KIDNEY  ACH-000684   \n",
       "4       CAL51_BREAST_TenPx01      CAL51_BREAST     CAL51_BREAST  ACH-000856   \n",
       "..                       ...               ...              ...         ...   \n",
       "253   22RV1_PROSTATE_TenPx24    22RV1_PROSTATE   22RV1_PROSTATE  ACH-000956   \n",
       "254      BT20_BREAST_TenPx24       BT20_BREAST      BT20_BREAST  ACH-000536   \n",
       "255       CALU6_LUNG_TenPx24        CALU6_LUNG       CALU6_LUNG  ACH-000264   \n",
       "256     KP4_PANCREAS_TenPx24      KP4_PANCREAS     KP4_PANCREAS  ACH-000265   \n",
       "257     OVCAR8_OVARY_TenPx24      OVCAR8_OVARY     OVCAR8_OVARY  ACH-000696   \n",
       "\n",
       "           Name   Pathology Site_Primary Site_Subtype1 Site_Subtype2  \\\n",
       "0    MDA-MB-468  metastasis       breast            NS            NS   \n",
       "1          SH-4  metastasis         skin            NS            NS   \n",
       "2         AU565  metastasis       breast            NS            NS   \n",
       "3        KMRC-1     primary       kidney            NS            NS   \n",
       "4        CAL-51  metastasis       breast            NS            NS   \n",
       "..          ...         ...          ...           ...           ...   \n",
       "253       22Rv1     primary     prostate            NS            NS   \n",
       "254       BT-20     primary       breast            NS            NS   \n",
       "255      Calu-6     primary         lung            NS            NS   \n",
       "256         KP4  metastasis     pancreas            NS            NS   \n",
       "257     OVCAR-8     primary        ovary            NS            NS   \n",
       "\n",
       "    Site_Subtype3  ...                                    Growth.Medium  \\\n",
       "0              NS  ...                                     DMEM+10% FBS   \n",
       "1              NS  ...                                     DMEM+10% FBS   \n",
       "2              NS  ...                               RPMI 1640 + 10%FBS   \n",
       "3              NS  ...                                     DMEM+10% FBS   \n",
       "4              NS  ...  80% Dulbecco's MEM(4.5g/L glucose) +20% h.i.FBS   \n",
       "..            ...  ...                                              ...   \n",
       "253            NS  ...                                RPMI-1640+10% FBS   \n",
       "254            NS  ...                                              NaN   \n",
       "255            NS  ...                                     EMEM +10%FBS   \n",
       "256            NS  ...                                      IMDM+10%FBS   \n",
       "257            NS  ...                                              NaN   \n",
       "\n",
       "    Supplements Freezing.Medium Doubling.Time.from.Vendor  \\\n",
       "0           NaN         5% DMSO                       NaN   \n",
       "1           NaN          5%DMSO                       NaN   \n",
       "2           NaN         5% DMSO                       NaN   \n",
       "3           NaN         5% DMSO                       NaN   \n",
       "4           NaN         10%DMSO                   >60 hrs   \n",
       "..          ...             ...                       ...   \n",
       "253         NaN          5%DMSO                       NaN   \n",
       "254         NaN             NaN                       NaN   \n",
       "255         NaN         5% DMSO                       NaN   \n",
       "256         NaN          5%DMSO                       NaN   \n",
       "257         NaN             NaN                       NaN   \n",
       "\n",
       "    Doubling.Time.Calculated.hrs      type  type_refined  \\\n",
       "0                           63.5    breast        breast   \n",
       "1                           68.9  melanoma      melanoma   \n",
       "2                           74.1    breast        breast   \n",
       "3                           46.4    kidney        kidney   \n",
       "4                           26.8    breast        breast   \n",
       "..                           ...       ...           ...   \n",
       "253                         58.4  prostate      prostate   \n",
       "254                          NaN    breast        breast   \n",
       "255                         78.9  lung_NSC      lung_NSC   \n",
       "256                          NaN  pancreas      pancreas   \n",
       "257                          NaN     ovary         ovary   \n",
       "\n",
       "    PATHOLOGIST_ANNOTATION     mutRate tcga_code  \n",
       "0         Breast:Carcinoma  188.252652      BRCA  \n",
       "1            Skin:Melanoma  109.541023      SKCM  \n",
       "2         Breast:Carcinoma  223.382430      BRCA  \n",
       "3         Kidney:Carcinoma  204.674772      KIRC  \n",
       "4         Breast:Carcinoma  206.841626      BRCA  \n",
       "..                     ...         ...       ...  \n",
       "253     Prostate:Carcinoma  539.830868      PRAD  \n",
       "254       Breast:Carcinoma  114.140088      BRCA  \n",
       "255      Lung:NSCLC_Others   87.777181      LUAD  \n",
       "256     Pancreas:Carcinoma  143.110198      PAAD  \n",
       "257        Ovary:Carcinoma  154.941601        OV  \n",
       "\n",
       "[258 rows x 35 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'MDAMB468_BREAST_TenPx01': 'ACH-000849', 'SH4_SKIN_TenPx01': 'ACH-000441', 'AU565_BREAST_TenPx01': 'ACH-000248', 'KMRC1_KIDNEY_TenPx01': 'ACH-000684', 'CAL51_BREAST_TenPx01': 'ACH-000856', 'RPMI7951_SKIN_TenPx01': 'ACH-000348', 'RERFLCMS_LUNG_TenPx01': 'ACH-000062', 'IGR37_SKIN_TenPx01': 'ACH-000650', 'VMRCRCW_KIDNEY_TenPx01': 'ACH-000484', 'HEP3B217_LIVER_TenPx02': 'ACH-000625', 'HEPG2_LIVER_TenPx02': 'ACH-000739', 'SKHEP1_LIVER_TenPx02': 'ACH-000361', 'WM2664_SKIN_TenPx02': 'ACH-001239', 'KMRC20_KIDNEY_TenPx02': 'ACH-000250', 'HS294T_SKIN_TenPx02': 'ACH-000014', 'CAOV3_OVARY_TenPx02': 'ACH-000713', 'JMSU1_URINARY_TRACT_TenPx02': 'ACH-000023', 'PATU8988T_PANCREAS_TenPx03': 'ACH-000711', 'JIMT1_BREAST_TenPx03': 'ACH-000278', 'COV362_OVARY_TenPx03': 'ACH-000028', 'KPL1_BREAST_TenPx03': 'ACH-000219', 'A375_SKIN_TenPx03': 'ACH-000788', 'A2058_SKIN_TenPx03': 'ACH-000475', 'SW620_LARGE_INTESTINE_TenPx03': 'ACH-000393', 'MDST8_LARGE_INTESTINE_TenPx03': 'ACH-000528', 'HUH1_LIVER_TenPx03': 'ACH-000783', 'HCT116_LARGE_INTESTINE_TenPx04': 'ACH-000734', 'HLF_LIVER_TenPx04': 'ACH-000423', 'HT29_LARGE_INTESTINE_TenPx04': 'ACH-000860', 'ABC1_LUNG_TenPx04': 'ACH-000480', 'CAMA1_BREAST_TenPx04': 'ACH-000847', 'RKO_LARGE_INTESTINE_TenPx04': 'ACH-000749', 'HT115_LARGE_INTESTINE_TenPx04': 'ACH-000510', 'JHH5_LIVER_TenPx04': 'ACH-000615', 'SKMEL3_SKIN_TenPx04': 'ACH-000234', 'NCIH358_LUNG_TenPx06': 'ACH-000866', 'HT1197_URINARY_TRACT_TenPx06': 'ACH-000288', 'HUH7_LIVER_TenPx06': 'ACH-000857', 'HGC27_STOMACH_TenPx06': 'ACH-000621', 'DMS273_LUNG_TenPx06': 'ACH-000550', 'NCIH1299_LUNG_TenPx06': 'ACH-000019', 'SKMEL28_SKIN_TenPx06': 'ACH-000661', 'CAKI2_KIDNEY_TenPx06': 'ACH-000915', 'HT55_LARGE_INTESTINE_TenPx06': 'ACH-000021', 'NCIH1048_LUNG_TenPx07': 'ACH-000859', 'HSC3_UPPER_AERODIGESTIVE_TRACT_TenPx07': 'ACH-000430', 'SW403_LARGE_INTESTINE_TenPx07': 'ACH-000530', 'BT549_BREAST_TenPx07': 'ACH-000647', 'HCC56_LARGE_INTESTINE_TenPx07': 'ACH-000502', 'CAL851_BREAST_TenPx07': 'ACH-000390', 'MDAMB157_BREAST_TenPx07': 'ACH-000881', 'SW48_LARGE_INTESTINE_TenPx07': 'ACH-000147', 'IGR39_SKIN_TenPx07': 'ACH-000090', 'MCF7_BREAST_TenPx08': 'ACH-000865', 'BICR6_UPPER_AERODIGESTIVE_TRACT_TenPx08': 'ACH-000012', 'WM1799_SKIN_TenPx08': 'ACH-000929', 'IPC298_SKIN_TenPx08': 'ACH-000601', 'NCIH1693_LUNG_TenPx08': 'ACH-000395', 'HCC1954_BREAST_TenPx08': 'ACH-000097', 'TYKNU_OVARY_TenPx08': 'ACH-000805', 'DMS114_LUNG_TenPx08': 'ACH-000649', 'TE1_OESOPHAGUS_TenPx08': 'ACH-000276', 'TCCPAN2_PANCREAS_TenPx09': 'ACH-000347', 'LUDLU1_LUNG_TenPx09': 'ACH-000454', 'MELJUSO_SKIN_TenPx09': 'ACH-000769', 'T47D_BREAST_TenPx09': 'ACH-000617', 'LS411N_LARGE_INTESTINE_TenPx09': 'ACH-000582', 'PC3_PROSTATE_TenPx09': 'ACH-000035', 'KYSE450_OESOPHAGUS_TenPx09': 'ACH-000579', 'RT112_URINARY_TRACT_TenPx09': 'ACH-000916', 'LS513_LARGE_INTESTINE_TenPx09': 'ACH-000681', 'HCC827_LUNG_TenPx10': 'ACH-000114', 'NCIH2110_LUNG_TenPx10': 'ACH-000726', 'MIAPACA2_PANCREAS_TenPx10': 'ACH-000481', 'NCIH520_LUNG_TenPx10': 'ACH-000349', 'SW480_LARGE_INTESTINE_TenPx10': 'ACH-000573', 'OE33_OESOPHAGUS_TenPx10': 'ACH-000925', 'ZR751_BREAST_TenPx10': 'ACH-000670', 'COLO679_SKIN_TenPx10': 'ACH-000138', '786O_KIDNEY_TenPx10': 'ACH-000109', 'HCC38_BREAST_TenPx11': 'ACH-000587', 'QGP1_PANCREAS_TenPx11': 'ACH-000635', 'HCC95_LUNG_TenPx11': 'ACH-000161', 'LK2_LUNG_TenPx11': 'ACH-000668', 'OVCAR4_OVARY_TenPx11': 'ACH-000880', 'COLO741_SKIN_TenPx11': 'ACH-000374', 'NCIH1650_LUNG_TenPx11': 'ACH-000448', 'UACC257_SKIN_TenPx11': 'ACH-000718', 'NCIH1573_LUNG_TenPx12': 'ACH-000416', 'A549_LUNG_TenPx12': 'ACH-000574', 'SU8686_PANCREAS_TenPx12': 'ACH-000442', 'TE14_OESOPHAGUS_TenPx12': 'ACH-000685', 'NCIH2170_LUNG_TenPx12': 'ACH-000356', 'HCC1500_BREAST_TenPx12': 'ACH-000678', 'MDAMB436_BREAST_TenPx12': 'ACH-000888', 'DV90_LUNG_TenPx12': 'ACH-000785', 'SNU61_LARGE_INTESTINE_TenPx12': 'ACH-000447', 'SBC5_LUNG_TenPx13': 'ACH-000853', 'HT1376_URINARY_TRACT_TenPx13': 'ACH-000605', 'CFPAC1_PANCREAS_TenPx13': 'ACH-000966', 'NCIH3255_LUNG_TenPx13': 'ACH-000627', 'NCIH1975_LUNG_TenPx13': 'ACH-000750', 'COLO678_LARGE_INTESTINE_TenPx13': 'ACH-000910', 'SNU119_OVARY_TenPx13': 'ACH-000747', 'CORL105_LUNG_TenPx13': 'ACH-000752', 'HCC70_BREAST_TenPx13': 'ACH-000886', 'AGS_STOMACH_TenPx16': 'ACH-000912', 'HCC1143_BREAST_TenPx16': 'ACH-000695', 'NCIH1666_LUNG_TenPx16': 'ACH-000111', 'NCIH2291_LUNG_TenPx16': 'ACH-000671', 'NCIH747_LARGE_INTESTINE_TenPx16': 'ACH-000506', 'NCIH838_LUNG_TenPx16': 'ACH-000932', 'FUOV1_OVARY_TenPx16': 'ACH-000900', 'RERFLCSQ1_LUNG_TenPx16': 'ACH-000159', 'DETROIT562_UPPER_AERODIGESTIVE_TRACT_TenPx16': 'ACH-000508', 'L33_PANCREAS_TenPx17': 'ACH-000755', 'MKN45_STOMACH_TenPx17': 'ACH-000667', 'MKN7_STOMACH_TenPx17': 'ACH-000924', 'NCIH1793_LUNG_TenPx17': 'ACH-000488', 'NCIH2126_LUNG_TenPx17': 'ACH-000425', 'NCIH2228_LUNG_TenPx17': 'ACH-000115', 'NCIH661_LUNG_TenPx17': 'ACH-000330', 'SNUC2A_LARGE_INTESTINE_TenPx17': 'ACH-000239', 'TE6_OESOPHAGUS_TenPx17': 'ACH-000824', 'IGROV1_OVARY_TenPx19': 'ACH-000438', 'LCLC103H_LUNG_TenPx19': 'ACH-000463', 'LOXIMVI_SKIN_TenPx19': 'ACH-000343', 'MDAMB453_BREAST_TenPx19': 'ACH-000810', 'NCIH1703_LUNG_TenPx19': 'ACH-000155', 'NCIH196_LUNG_TenPx19': 'ACH-000052', 'NCIH2009_LUNG_TenPx19': 'ACH-000768', 'NCIH2286_LUNG_TenPx19': 'ACH-000542', 'SNUC1_LARGE_INTESTINE_TenPx19': 'ACH-000979', 'COLO205_LARGE_INTESTINE_TenPx20': 'ACH-000919', 'CORL47_LUNG_TenPx20': 'ACH-000909', 'HCC1187_BREAST_TenPx20': 'ACH-000878', 'HUH6_LIVER_TenPx20': 'ACH-000222', 'NCIH146_LUNG_TenPx20': 'ACH-000898', 'SNU1_STOMACH_TenPx20': 'ACH-000662', 'NCIH23_LUNG_TenPx20': 'ACH-000831', 'OSRC2_KIDNEY_TenPx20': 'ACH-000318', 'CORL88_LUNG_TenPx21': 'ACH-000304', 'HCC2218_BREAST_TenPx21': 'ACH-000677', 'HCC44_LUNG_TenPx21': 'ACH-000404', 'HSC4_UPPER_AERODIGESTIVE_TRACT_TenPx21': 'ACH-000132', 'NCIH2172_LUNG_TenPx21': 'ACH-000407', 'TCCSUP_URINARY_TRACT_TenPx21': 'ACH-000001', 'TE11_OESOPHAGUS_TenPx21': 'ACH-000410', 'UACC62_SKIN_TenPx21': 'ACH-000917', 'VCAP_PROSTATE_TenPx21': 'ACH-000620', 'EFM19_BREAST_TenPx22': 'ACH-000890', 'HUG1N_STOMACH_TenPx22': 'ACH-000367', 'KYSE510_OESOPHAGUS_TenPx22': 'ACH-000777', 'LU65_LUNG_TenPx22': 'ACH-000364', 'NCIH460_LUNG_TenPx22': 'ACH-000672', 'NCIH522_LUNG_TenPx22': 'ACH-000644', 'PECAPJ34CLONEC12_UPPER_AERODIGESTIVE_TRACT_TenPx22': 'ACH-000378', 'SKMEL30_SKIN_TenPx22': 'ACH-000091', 'SW1990_PANCREAS_TenPx22': 'ACH-000748', 'A673_BONE_TenPx27': 'ACH-000984', 'MDAMB231_BREAST_TenPx27': 'ACH-000433', 'HEYA8_OVARY_TenPx27': 'ACH-000281', 'RH41_SOFT_TISSUE_TenPx27': 'ACH-000879', 'KMS12BM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx27': 'ACH-001075', 'LN229_CENTRAL_NERVOUS_SYSTEM_TenPx27': 'ACH-000039', 'LS180_LARGE_INTESTINE_TenPx27': 'ACH-000867', 'MOLM13_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx27': 'ACH-000117', 'KARPAS299_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx27': 'ACH-000946', 'DU145_PROSTATE_TenPx28': 'ACH-000164', 'G401_SOFT_TISSUE_TenPx28': 'ACH-000008', 'IM95_STOMACH_TenPx28': 'ACH-000809', 'JHUEM2_ENDOMETRIUM_TenPx28': 'ACH-000800', 'KARPAS422_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx28': 'ACH-000243', 'KASUMI1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx28': 'ACH-000990', 'HCC15_LUNG_TenPx28': 'ACH-000335', 'U87MG_CENTRAL_NERVOUS_SYSTEM_TenPx28': 'ACH-000247', 'ASPC1_PANCREAS_TenPx29': 'ACH-000420', 'SNU719_STOMACH_TenPx29': 'ACH-001307', 'CORL23_LUNG_TenPx29': 'ACH-000712', 'F36P_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx29': 'ACH-000235', 'HEC50B_ENDOMETRIUM_TenPx29': 'ACH-000476', 'KNS81_CENTRAL_NERVOUS_SYSTEM_TenPx29': 'ACH-000409', 'OCIAML5_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx29': 'ACH-000309', 'TE10_OESOPHAGUS_TenPx29': 'ACH-000961', 'WM115_SKIN_TenPx29': 'ACH-000974', 'EOL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx33': 'ACH-000042', 'SW1573_LUNG_TenPx33': 'ACH-000493', 'HEL9217_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx33': 'ACH-000535', 'K029AX_SKIN_TenPx33': 'ACH-000511', 'JHOS2_OVARY_TenPx33': 'ACH-000996', 'SNU685_ENDOMETRIUM_TenPx33': 'ACH-000693', 'NIHOVCAR3_OVARY_TenPx33': 'ACH-000730', 'SAOS2_BONE_TenPx33': 'ACH-000030', 'TE4_OESOPHAGUS_TenPx33': 'ACH-000563', 'JHH1_LIVER_TenPx34': 'ACH-001190', 'SW1271_LUNG_TenPx34': 'ACH-000987', 'NCIH226_LUNG_TenPx34': 'ACH-000632', 'REC1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx34': 'ACH-000848', 'HT1080_SOFT_TISSUE_TenPx34': 'ACH-000614', 'KYSE30_OESOPHAGUS_TenPx34': 'ACH-000555', 'U2OS_BONE_TenPx34': 'ACH-000799', 'GB1_CENTRAL_NERVOUS_SYSTEM_TenPx34': 'ACH-001306', 'IALM_LUNG_TenPx34': 'ACH-000882', 'CMK_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx35': 'ACH-000652', 'COLO829_SKIN_TenPx35': 'ACH-000977', 'NCIH647_LUNG_TenPx35': 'ACH-000945', 'OV56_OVARY_TenPx35': 'ACH-000954', 'RH30_SOFT_TISSUE_TenPx35': 'ACH-000118', 'RPMI8226_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx35': 'ACH-000787', 'SJSA1_BONE_TenPx35': 'ACH-000701', 'HEC6_ENDOMETRIUM_TenPx35': 'ACH-000429', 'SW837_LARGE_INTESTINE_TenPx35': 'ACH-000899', 'CAKI1_KIDNEY_TenPx36': 'ACH-000153', 'KP2_PANCREAS_TenPx36': 'ACH-000291', 'KU1919_URINARY_TRACT_TenPx36': 'ACH-000087', 'MFE296_ENDOMETRIUM_TenPx36': 'ACH-000192', 'NCIH292_LUNG_TenPx36': 'ACH-000624', 'NCIH716_LARGE_INTESTINE_TenPx36': 'ACH-000015', 'NUDHL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx36': 'ACH-000496', 'OCILY3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx36': 'ACH-000727', 'SKNMC_BONE_TenPx36': 'ACH-000553', 'CHAGOK1_LUNG_TenPx37': 'ACH-000657', 'EFM192A_BREAST_TenPx37': 'ACH-000351', 'HEC265_ENDOMETRIUM_TenPx37': 'ACH-000699', 'MOLM16_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx37': 'ACH-000852', 'PANC1_PANCREAS_TenPx37': 'ACH-000869', 'SCC25_UPPER_AERODIGESTIVE_TRACT_TenPx37': 'ACH-000414', 'RT4_URINARY_TRACT_TenPx37': 'ACH-000638', 'T24_URINARY_TRACT_TenPx37': 'ACH-000524', 'TF1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx37': 'ACH-000793', 'A101D_SKIN_TenPx38': 'ACH-000988', 'DAOY_CENTRAL_NERVOUS_SYSTEM_TenPx38': 'ACH-000643', 'JM1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx38': 'ACH-000217', 'KASUMI2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx38': 'ACH-000255', 'KYM1_SOFT_TISSUE_TenPx38': 'ACH-000213', 'KYSE410_OESOPHAGUS_TenPx38': 'ACH-000855', 'CL34_LARGE_INTESTINE_TenPx38': 'ACH-000980', 'NCIH446_LUNG_TenPx38': 'ACH-000427', 'SKNAS_AUTONOMIC_GANGLIA_TenPx38': 'ACH-001171', 'DANG_PANCREAS_TenPx39': 'ACH-000411', 'KNS42_CENTRAL_NERVOUS_SYSTEM_TenPx39': 'ACH-000580', 'HEC108_ENDOMETRIUM_TenPx39': 'ACH-000994', 'JEKO1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx39': 'ACH-000911', 'JURKAT_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx39': 'ACH-000790', 'MSTO211H_PLEURA_TenPx39': 'ACH-000424', 'OCUM1_STOMACH_TenPx39': 'ACH-000223', 'CCK81_LARGE_INTESTINE_TenPx39': 'ACH-000784', 'SNU449_LIVER_TenPx39': 'ACH-000666', '8505C_THYROID_TenPx40': 'ACH-000589', 'HCC1833_LUNG_TenPx40': 'ACH-000311', 'PANC0403_PANCREAS_TenPx40': 'ACH-000495', 'J82_URINARY_TRACT_TenPx40': 'ACH-000827', 'JHH4_LIVER_TenPx40': 'ACH-000521', 'OVSAHO_OVARY_TenPx40': 'ACH-000956', 'RL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx40': 'ACH-000536', 'SNUC5_LARGE_INTESTINE_TenPx40': 'ACH-000264', 'SUDHL6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx40': 'ACH-000265', 'SKLU1_LUNG_TenPx42': 'ACH-000696'}\n"
     ]
    }
   ],
   "source": [
    "# get the cleaned cell line names by removed the content after the second \"_\" in final_merged_protein_df.columns[1:]\n",
    "original_protein_samples = final_merged_protein_df.columns[1:]\n",
    "cleaned_protein_samples = ['_'.join(s.split('_')[:2]) for s in original_protein_samples]\n",
    "# get the cleaned_protein_samples_dict_df from dictionary\n",
    "cleaned_protein_samples_dict_df = pd.DataFrame({'CCLE_Name': original_protein_samples, 'cleaned_CCLE_Name': cleaned_protein_samples})\n",
    "# merge the cleaned_protein_samples_df with the cell_line_anno_df\n",
    "merged_cleaned_protein_samples_df = pd.merge(cleaned_protein_samples_dict_df, cell_line_anno_df, left_on='cleaned_CCLE_Name', right_on='CCLE_ID', how='inner')\n",
    "display(merged_cleaned_protein_samples_df)\n",
    "# get the depmap protein id\n",
    "depmap_protein_samples = merged_cleaned_protein_samples_df['depMapID'].to_list()\n",
    "\n",
    "# Create the map dictionary for the protein samples\n",
    "protein_map_dict = dict(zip(original_protein_samples, depmap_protein_samples))\n",
    "print(protein_map_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get the methy_mapped_samples and protein_mapped_samples from merged_methy_samples_df and merged_cleaned_protein_samples_df\n",
    "methy_mapped_samples = list(merged_methy_samples_df['depMapID'])\n",
    "protein_mapped_samples = list(merged_cleaned_protein_samples_df['depMapID'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 6.1 Aligning over omics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 6.1.1 Aligning with intersection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "length of omics overlapped samples:  223\n",
      "length of overlapped omics and annotation samples:  223\n"
     ]
    }
   ],
   "source": [
    "# get the overlapped samples only over omics samples\n",
    "omics_overlapped_samples = sorted(list(set(methy_mapped_samples) & set(gene_samples) & set(transcript_samples) & set(protein_mapped_samples)))\n",
    "print(\"length of omics overlapped samples: \", len(omics_overlapped_samples))\n",
    "# check the overlapped samples with annotation samples\n",
    "annotation_samples = list(cell_line_anno_df['depMapID'])\n",
    "overlapped_omics_annotation_samples = sorted(list(set(omics_overlapped_samples) & set(annotation_samples)))\n",
    "print(\"length of overlapped omics and annotation samples: \", len(overlapped_omics_annotation_samples))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 6.1.2 Aligning with union on omics and intersection with annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "length of omics union samples:  2007\n",
      "length of overlapped omics and annotation samples:  985\n",
      "['ACH-000001', 'ACH-000002', 'ACH-000004', 'ACH-000005', 'ACH-000006', 'ACH-000007', 'ACH-000008', 'ACH-000009', 'ACH-000011', 'ACH-000012', 'ACH-000013', 'ACH-000014', 'ACH-000015', 'ACH-000016', 'ACH-000017', 'ACH-000018', 'ACH-000019', 'ACH-000020', 'ACH-000021', 'ACH-000022', 'ACH-000023', 'ACH-000024', 'ACH-000026', 'ACH-000027', 'ACH-000028', 'ACH-000030', 'ACH-000031', 'ACH-000032', 'ACH-000035', 'ACH-000037', 'ACH-000038', 'ACH-000039', 'ACH-000040', 'ACH-000041', 'ACH-000042', 'ACH-000043', 'ACH-000044', 'ACH-000045', 'ACH-000046', 'ACH-000047', 'ACH-000048', 'ACH-000050', 'ACH-000051', 'ACH-000052', 'ACH-000053', 'ACH-000054', 'ACH-000055', 'ACH-000056', 'ACH-000058', 'ACH-000059', 'ACH-000060', 'ACH-000061', 'ACH-000062', 'ACH-000063', 'ACH-000065', 'ACH-000066', 'ACH-000067', 'ACH-000068', 'ACH-000069', 'ACH-000070', 'ACH-000071', 'ACH-000072', 'ACH-000073', 'ACH-000074', 'ACH-000075', 'ACH-000076', 'ACH-000077', 'ACH-000078', 'ACH-000079', 'ACH-000080', 'ACH-000081', 'ACH-000082', 'ACH-000083', 'ACH-000084', 'ACH-000085', 'ACH-000086', 'ACH-000087', 'ACH-000088', 'ACH-000089', 'ACH-000090', 'ACH-000091', 'ACH-000092', 'ACH-000093', 'ACH-000094', 'ACH-000095', 'ACH-000096', 'ACH-000097', 'ACH-000098', 'ACH-000099', 'ACH-000100', 'ACH-000101', 'ACH-000102', 'ACH-000103', 'ACH-000104', 'ACH-000105', 'ACH-000106', 'ACH-000107', 'ACH-000108', 'ACH-000109', 'ACH-000110', 'ACH-000111', 'ACH-000112', 'ACH-000113', 'ACH-000114', 'ACH-000115', 'ACH-000116', 'ACH-000117', 'ACH-000118', 'ACH-000119', 'ACH-000120', 'ACH-000121', 'ACH-000122', 'ACH-000123', 'ACH-000124', 'ACH-000125', 'ACH-000126', 'ACH-000127', 'ACH-000129', 'ACH-000130', 'ACH-000131', 'ACH-000132', 'ACH-000133', 'ACH-000134', 'ACH-000135', 'ACH-000136', 'ACH-000137', 'ACH-000138', 'ACH-000139', 'ACH-000140', 'ACH-000141', 'ACH-000142', 'ACH-000144', 'ACH-000145', 'ACH-000146', 'ACH-000147', 'ACH-000148', 'ACH-000149', 'ACH-000150', 'ACH-000151', 'ACH-000152', 'ACH-000153', 'ACH-000154', 'ACH-000155', 'ACH-000156', 'ACH-000157', 'ACH-000158', 'ACH-000159', 'ACH-000161', 'ACH-000162', 'ACH-000163', 'ACH-000164', 'ACH-000165', 'ACH-000166', 'ACH-000167', 'ACH-000168', 'ACH-000169', 'ACH-000171', 'ACH-000173', 'ACH-000174', 'ACH-000175', 'ACH-000176', 'ACH-000177', 'ACH-000178', 'ACH-000179', 'ACH-000180', 'ACH-000181', 'ACH-000182', 'ACH-000183', 'ACH-000184', 'ACH-000185', 'ACH-000186', 'ACH-000187', 'ACH-000188', 'ACH-000189', 'ACH-000190', 'ACH-000191', 'ACH-000192', 'ACH-000193', 'ACH-000194', 'ACH-000195', 'ACH-000196', 'ACH-000197', 'ACH-000198', 'ACH-000199', 'ACH-000200', 'ACH-000201', 'ACH-000202', 'ACH-000203', 'ACH-000204', 'ACH-000205', 'ACH-000207', 'ACH-000209', 'ACH-000210', 'ACH-000211', 'ACH-000212', 'ACH-000213', 'ACH-000214', 'ACH-000217', 'ACH-000218', 'ACH-000219', 'ACH-000220', 'ACH-000221', 'ACH-000222', 'ACH-000223', 'ACH-000224', 'ACH-000225', 'ACH-000226', 'ACH-000227', 'ACH-000228', 'ACH-000229', 'ACH-000230', 'ACH-000231', 'ACH-000232', 'ACH-000233', 'ACH-000234', 'ACH-000235', 'ACH-000236', 'ACH-000237', 'ACH-000238', 'ACH-000239', 'ACH-000240', 'ACH-000241', 'ACH-000242', 'ACH-000243', 'ACH-000244', 'ACH-000245', 'ACH-000246', 'ACH-000247', 'ACH-000248', 'ACH-000249', 'ACH-000250', 'ACH-000252', 'ACH-000253', 'ACH-000254', 'ACH-000255', 'ACH-000256', 'ACH-000257', 'ACH-000258', 'ACH-000259', 'ACH-000260', 'ACH-000261', 'ACH-000263', 'ACH-000264', 'ACH-000265', 'ACH-000266', 'ACH-000267', 'ACH-000268', 'ACH-000269', 'ACH-000270', 'ACH-000271', 'ACH-000272', 'ACH-000273', 'ACH-000274', 'ACH-000275', 'ACH-000276', 'ACH-000277', 'ACH-000278', 'ACH-000280', 'ACH-000281', 'ACH-000282', 'ACH-000284', 'ACH-000285', 'ACH-000286', 'ACH-000287', 'ACH-000288', 'ACH-000289', 'ACH-000290', 'ACH-000291', 'ACH-000292', 'ACH-000293', 'ACH-000294', 'ACH-000295', 'ACH-000296', 'ACH-000297', 'ACH-000298', 'ACH-000300', 'ACH-000301', 'ACH-000302', 'ACH-000303', 'ACH-000304', 'ACH-000305', 'ACH-000306', 'ACH-000307', 'ACH-000308', 'ACH-000309', 'ACH-000310', 'ACH-000311', 'ACH-000312', 'ACH-000313', 'ACH-000314', 'ACH-000315', 'ACH-000316', 'ACH-000317', 'ACH-000318', 'ACH-000319', 'ACH-000320', 'ACH-000321', 'ACH-000322', 'ACH-000323', 'ACH-000324', 'ACH-000325', 'ACH-000326', 'ACH-000327', 'ACH-000329', 'ACH-000330', 'ACH-000331', 'ACH-000332', 'ACH-000333', 'ACH-000334', 'ACH-000335', 'ACH-000336', 'ACH-000338', 'ACH-000340', 'ACH-000341', 'ACH-000342', 'ACH-000343', 'ACH-000344', 'ACH-000345', 'ACH-000346', 'ACH-000347', 'ACH-000348', 'ACH-000349', 'ACH-000350', 'ACH-000351', 'ACH-000352', 'ACH-000353', 'ACH-000354', 'ACH-000355', 'ACH-000356', 'ACH-000357', 'ACH-000358', 'ACH-000359', 'ACH-000360', 'ACH-000361', 'ACH-000362', 'ACH-000363', 'ACH-000364', 'ACH-000365', 'ACH-000366', 'ACH-000367', 'ACH-000368', 'ACH-000369', 'ACH-000370', 'ACH-000371', 'ACH-000372', 'ACH-000373', 'ACH-000374', 'ACH-000375', 'ACH-000376', 'ACH-000377', 'ACH-000378', 'ACH-000379', 'ACH-000380', 'ACH-000381', 'ACH-000382', 'ACH-000384', 'ACH-000386', 'ACH-000387', 'ACH-000388', 'ACH-000389', 'ACH-000390', 'ACH-000391', 'ACH-000392', 'ACH-000393', 'ACH-000394', 'ACH-000395', 'ACH-000396', 'ACH-000397', 'ACH-000398', 'ACH-000399', 'ACH-000400', 'ACH-000401', 'ACH-000402', 'ACH-000403', 'ACH-000404', 'ACH-000405', 'ACH-000406', 'ACH-000407', 'ACH-000408', 'ACH-000409', 'ACH-000410', 'ACH-000411', 'ACH-000412', 'ACH-000413', 'ACH-000414', 'ACH-000415', 'ACH-000416', 'ACH-000417', 'ACH-000418', 'ACH-000419', 'ACH-000420', 'ACH-000421', 'ACH-000422', 'ACH-000423', 'ACH-000424', 'ACH-000425', 'ACH-000426', 'ACH-000427', 'ACH-000428', 'ACH-000429', 'ACH-000430', 'ACH-000431', 'ACH-000432', 'ACH-000433', 'ACH-000434', 'ACH-000435', 'ACH-000437', 'ACH-000438', 'ACH-000439', 'ACH-000440', 'ACH-000441', 'ACH-000442', 'ACH-000443', 'ACH-000444', 'ACH-000445', 'ACH-000446', 'ACH-000447', 'ACH-000448', 'ACH-000449', 'ACH-000450', 'ACH-000451', 'ACH-000452', 'ACH-000453', 'ACH-000454', 'ACH-000456', 'ACH-000457', 'ACH-000458', 'ACH-000459', 'ACH-000460', 'ACH-000461', 'ACH-000462', 'ACH-000463', 'ACH-000464', 'ACH-000465', 'ACH-000466', 'ACH-000467', 'ACH-000468', 'ACH-000469', 'ACH-000470', 'ACH-000471', 'ACH-000472', 'ACH-000473', 'ACH-000475', 'ACH-000476', 'ACH-000477', 'ACH-000478', 'ACH-000479', 'ACH-000480', 'ACH-000481', 'ACH-000482', 'ACH-000483', 'ACH-000484', 'ACH-000485', 'ACH-000486', 'ACH-000487', 'ACH-000488', 'ACH-000489', 'ACH-000491', 'ACH-000492', 'ACH-000493', 'ACH-000495', 'ACH-000496', 'ACH-000498', 'ACH-000500', 'ACH-000501', 'ACH-000502', 'ACH-000503', 'ACH-000504', 'ACH-000505', 'ACH-000506', 'ACH-000507', 'ACH-000508', 'ACH-000509', 'ACH-000510', 'ACH-000511', 'ACH-000513', 'ACH-000514', 'ACH-000515', 'ACH-000516', 'ACH-000517', 'ACH-000518', 'ACH-000519', 'ACH-000520', 'ACH-000521', 'ACH-000522', 'ACH-000523', 'ACH-000524', 'ACH-000525', 'ACH-000526', 'ACH-000527', 'ACH-000528', 'ACH-000529', 'ACH-000530', 'ACH-000531', 'ACH-000532', 'ACH-000534', 'ACH-000535', 'ACH-000536', 'ACH-000537', 'ACH-000538', 'ACH-000539', 'ACH-000540', 'ACH-000541', 'ACH-000542', 'ACH-000543', 'ACH-000544', 'ACH-000545', 'ACH-000546', 'ACH-000547', 'ACH-000548', 'ACH-000549', 'ACH-000550', 'ACH-000551', 'ACH-000552', 'ACH-000553', 'ACH-000554', 'ACH-000555', 'ACH-000557', 'ACH-000558', 'ACH-000559', 'ACH-000560', 'ACH-000562', 'ACH-000563', 'ACH-000564', 'ACH-000565', 'ACH-000566', 'ACH-000567', 'ACH-000568', 'ACH-000569', 'ACH-000570', 'ACH-000571', 'ACH-000572', 'ACH-000573', 'ACH-000574', 'ACH-000576', 'ACH-000577', 'ACH-000578', 'ACH-000579', 'ACH-000580', 'ACH-000581', 'ACH-000582', 'ACH-000583', 'ACH-000584', 'ACH-000585', 'ACH-000586', 'ACH-000587', 'ACH-000588', 'ACH-000589', 'ACH-000590', 'ACH-000592', 'ACH-000593', 'ACH-000594', 'ACH-000595', 'ACH-000596', 'ACH-000598', 'ACH-000599', 'ACH-000600', 'ACH-000601', 'ACH-000602', 'ACH-000603', 'ACH-000604', 'ACH-000605', 'ACH-000606', 'ACH-000607', 'ACH-000608', 'ACH-000609', 'ACH-000610', 'ACH-000611', 'ACH-000612', 'ACH-000613', 'ACH-000614', 'ACH-000615', 'ACH-000616', 'ACH-000617', 'ACH-000618', 'ACH-000619', 'ACH-000620', 'ACH-000621', 'ACH-000622', 'ACH-000623', 'ACH-000624', 'ACH-000625', 'ACH-000626', 'ACH-000627', 'ACH-000628', 'ACH-000630', 'ACH-000631', 'ACH-000632', 'ACH-000633', 'ACH-000635', 'ACH-000636', 'ACH-000637', 'ACH-000638', 'ACH-000639', 'ACH-000640', 'ACH-000641', 'ACH-000643', 'ACH-000644', 'ACH-000645', 'ACH-000646', 'ACH-000647', 'ACH-000648', 'ACH-000649', 'ACH-000650', 'ACH-000651', 'ACH-000652', 'ACH-000653', 'ACH-000654', 'ACH-000655', 'ACH-000656', 'ACH-000657', 'ACH-000659', 'ACH-000660', 'ACH-000661', 'ACH-000662', 'ACH-000663', 'ACH-000664', 'ACH-000665', 'ACH-000666', 'ACH-000667', 'ACH-000668', 'ACH-000669', 'ACH-000670', 'ACH-000671', 'ACH-000672', 'ACH-000674', 'ACH-000675', 'ACH-000677', 'ACH-000678', 'ACH-000679', 'ACH-000680', 'ACH-000681', 'ACH-000682', 'ACH-000683', 'ACH-000684', 'ACH-000685', 'ACH-000686', 'ACH-000687', 'ACH-000688', 'ACH-000689', 'ACH-000691', 'ACH-000692', 'ACH-000693', 'ACH-000694', 'ACH-000695', 'ACH-000696', 'ACH-000697', 'ACH-000698', 'ACH-000699', 'ACH-000701', 'ACH-000702', 'ACH-000703', 'ACH-000704', 'ACH-000705', 'ACH-000706', 'ACH-000707', 'ACH-000708', 'ACH-000709', 'ACH-000710', 'ACH-000711', 'ACH-000712', 'ACH-000713', 'ACH-000714', 'ACH-000715', 'ACH-000716', 'ACH-000717', 'ACH-000718', 'ACH-000719', 'ACH-000720', 'ACH-000721', 'ACH-000722', 'ACH-000723', 'ACH-000724', 'ACH-000725', 'ACH-000726', 'ACH-000727', 'ACH-000728', 'ACH-000729', 'ACH-000730', 'ACH-000731', 'ACH-000732', 'ACH-000733', 'ACH-000734', 'ACH-000735', 'ACH-000736', 'ACH-000737', 'ACH-000738', 'ACH-000739', 'ACH-000740', 'ACH-000741', 'ACH-000742', 'ACH-000743', 'ACH-000744', 'ACH-000745', 'ACH-000746', 'ACH-000747', 'ACH-000748', 'ACH-000749', 'ACH-000750', 'ACH-000751', 'ACH-000752', 'ACH-000753', 'ACH-000754', 'ACH-000755', 'ACH-000756', 'ACH-000758', 'ACH-000759', 'ACH-000761', 'ACH-000762', 'ACH-000763', 'ACH-000764', 'ACH-000765', 'ACH-000766', 'ACH-000767', 'ACH-000768', 'ACH-000769', 'ACH-000770', 'ACH-000771', 'ACH-000772', 'ACH-000773', 'ACH-000774', 'ACH-000775', 'ACH-000776', 'ACH-000777', 'ACH-000778', 'ACH-000780', 'ACH-000781', 'ACH-000782', 'ACH-000783', 'ACH-000784', 'ACH-000785', 'ACH-000786', 'ACH-000787', 'ACH-000788', 'ACH-000789', 'ACH-000790', 'ACH-000791', 'ACH-000792', 'ACH-000793', 'ACH-000794', 'ACH-000795', 'ACH-000796', 'ACH-000797', 'ACH-000798', 'ACH-000799', 'ACH-000800', 'ACH-000801', 'ACH-000802', 'ACH-000803', 'ACH-000804', 'ACH-000805', 'ACH-000806', 'ACH-000807', 'ACH-000808', 'ACH-000809', 'ACH-000810', 'ACH-000811', 'ACH-000812', 'ACH-000813', 'ACH-000814', 'ACH-000815', 'ACH-000816', 'ACH-000817', 'ACH-000818', 'ACH-000819', 'ACH-000820', 'ACH-000821', 'ACH-000822', 'ACH-000823', 'ACH-000824', 'ACH-000825', 'ACH-000826', 'ACH-000827', 'ACH-000828', 'ACH-000829', 'ACH-000830', 'ACH-000831', 'ACH-000832', 'ACH-000833', 'ACH-000834', 'ACH-000835', 'ACH-000836', 'ACH-000837', 'ACH-000838', 'ACH-000839', 'ACH-000840', 'ACH-000841', 'ACH-000842', 'ACH-000843', 'ACH-000844', 'ACH-000845', 'ACH-000846', 'ACH-000847', 'ACH-000848', 'ACH-000849', 'ACH-000850', 'ACH-000851', 'ACH-000852', 'ACH-000853', 'ACH-000855', 'ACH-000856', 'ACH-000857', 'ACH-000858', 'ACH-000859', 'ACH-000860', 'ACH-000861', 'ACH-000862', 'ACH-000863', 'ACH-000864', 'ACH-000865', 'ACH-000866', 'ACH-000867', 'ACH-000868', 'ACH-000869', 'ACH-000870', 'ACH-000871', 'ACH-000873', 'ACH-000874', 'ACH-000875', 'ACH-000876', 'ACH-000877', 'ACH-000878', 'ACH-000879', 'ACH-000880', 'ACH-000881', 'ACH-000882', 'ACH-000883', 'ACH-000884', 'ACH-000885', 'ACH-000886', 'ACH-000888', 'ACH-000889', 'ACH-000890', 'ACH-000891', 'ACH-000892', 'ACH-000893', 'ACH-000894', 'ACH-000895', 'ACH-000896', 'ACH-000897', 'ACH-000898', 'ACH-000899', 'ACH-000900', 'ACH-000901', 'ACH-000902', 'ACH-000903', 'ACH-000904', 'ACH-000905', 'ACH-000906', 'ACH-000907', 'ACH-000908', 'ACH-000909', 'ACH-000910', 'ACH-000911', 'ACH-000912', 'ACH-000913', 'ACH-000914', 'ACH-000915', 'ACH-000916', 'ACH-000917', 'ACH-000918', 'ACH-000919', 'ACH-000920', 'ACH-000921', 'ACH-000922', 'ACH-000923', 'ACH-000924', 'ACH-000925', 'ACH-000926', 'ACH-000927', 'ACH-000928', 'ACH-000929', 'ACH-000930', 'ACH-000931', 'ACH-000932', 'ACH-000933', 'ACH-000934', 'ACH-000935', 'ACH-000936', 'ACH-000937', 'ACH-000938', 'ACH-000939', 'ACH-000940', 'ACH-000941', 'ACH-000942', 'ACH-000943', 'ACH-000944', 'ACH-000945', 'ACH-000946', 'ACH-000947', 'ACH-000948', 'ACH-000949', 'ACH-000950', 'ACH-000951', 'ACH-000952', 'ACH-000953', 'ACH-000954', 'ACH-000955', 'ACH-000956', 'ACH-000957', 'ACH-000958', 'ACH-000959', 'ACH-000960', 'ACH-000961', 'ACH-000962', 'ACH-000963', 'ACH-000965', 'ACH-000966', 'ACH-000967', 'ACH-000968', 'ACH-000969', 'ACH-000970', 'ACH-000971', 'ACH-000972', 'ACH-000973', 'ACH-000974', 'ACH-000976', 'ACH-000977', 'ACH-000978', 'ACH-000979', 'ACH-000980', 'ACH-000981', 'ACH-000982', 'ACH-000983', 'ACH-000984', 'ACH-000985', 'ACH-000986', 'ACH-000987', 'ACH-000988', 'ACH-000989', 'ACH-000990', 'ACH-000991', 'ACH-000992', 'ACH-000993', 'ACH-000994', 'ACH-000995', 'ACH-000996', 'ACH-000997', 'ACH-000998', 'ACH-000999', 'ACH-001001', 'ACH-001016', 'ACH-001021', 'ACH-001023', 'ACH-001024', 'ACH-001036', 'ACH-001039', 'ACH-001041', 'ACH-001048', 'ACH-001049', 'ACH-001061', 'ACH-001063', 'ACH-001064', 'ACH-001065', 'ACH-001075', 'ACH-001089', 'ACH-001093', 'ACH-001098', 'ACH-001106', 'ACH-001107', 'ACH-001113', 'ACH-001125', 'ACH-001126', 'ACH-001127', 'ACH-001129', 'ACH-001130', 'ACH-001138', 'ACH-001145', 'ACH-001150', 'ACH-001151', 'ACH-001162', 'ACH-001171', 'ACH-001182', 'ACH-001183', 'ACH-001188', 'ACH-001190', 'ACH-001194', 'ACH-001199', 'ACH-001203', 'ACH-001233', 'ACH-001239', 'ACH-001306', 'ACH-001307', 'ACH-001318', 'ACH-001321']\n"
     ]
    }
   ],
   "source": [
    "# get the union samples only over omics samples\n",
    "omics_union_samples = sorted(list(set(methy_mapped_samples) | set(gene_samples) | set(transcript_samples) | set(protein_mapped_samples)))\n",
    "print(\"length of omics union samples: \", len(omics_union_samples))\n",
    "# check the intersection of union samples with annotation samples\n",
    "overlapped_omics_union_annotation_samples = sorted(list(set(omics_union_samples) & set(annotation_samples)))\n",
    "print(\"length of overlapped omics and annotation samples: \", len(overlapped_omics_union_annotation_samples))\n",
    "print(overlapped_omics_union_annotation_samples)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 6.1.3 overlapped_omics_union_annotation_samples (cancerous / non-cancerous) Ratio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Depmap Id</th>\n",
       "      <th>Cell Line</th>\n",
       "      <th>Primary Disease</th>\n",
       "      <th>Tumor Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-002370</td>\n",
       "      <td>NCIBL2122</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-003133</td>\n",
       "      <td>ABMT0021</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-001207</td>\n",
       "      <td>TIG3TD</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000229</td>\n",
       "      <td>HS822T</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>Primary</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-002364</td>\n",
       "      <td>NCIBL1437</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>ACH-000214</td>\n",
       "      <td>HS675T</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>ACH-002368</td>\n",
       "      <td>NCIBL2087</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>ACH-000850</td>\n",
       "      <td>HS698T</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>Metastatic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>ACH-002463</td>\n",
       "      <td>RPE1SS77</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>ACH-003154</td>\n",
       "      <td>ABMT0795</td>\n",
       "      <td>Non-Cancerous</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>137 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Depmap Id  Cell Line Primary Disease  Tumor Type\n",
       "0    ACH-002370  NCIBL2122   Non-Cancerous         NaN\n",
       "1    ACH-003133   ABMT0021   Non-Cancerous         NaN\n",
       "2    ACH-001207     TIG3TD   Non-Cancerous         NaN\n",
       "3    ACH-000229     HS822T   Non-Cancerous     Primary\n",
       "4    ACH-002364  NCIBL1437   Non-Cancerous         NaN\n",
       "..          ...        ...             ...         ...\n",
       "132  ACH-000214     HS675T   Non-Cancerous         NaN\n",
       "133  ACH-002368  NCIBL2087   Non-Cancerous         NaN\n",
       "134  ACH-000850     HS698T   Non-Cancerous  Metastatic\n",
       "135  ACH-002463   RPE1SS77   Non-Cancerous         NaN\n",
       "136  ACH-003154   ABMT0795   Non-Cancerous         NaN\n",
       "\n",
       "[137 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "length of overlapped omics and annotation samples:  985\n",
      "length of overlapped non cancerous samples:  39\n"
     ]
    }
   ],
   "source": [
    "non_cancerous_samples_df = pd.read_csv('./data/raw_data/cell-lines-in-Non-Cancerous.csv')\n",
    "display(non_cancerous_samples_df)\n",
    "# get the depmap id from the non_cancerous_samples_df\n",
    "non_cancerous_samples = list(non_cancerous_samples_df['Depmap Id'])\n",
    "# get the overlapped non_cancerous samples with the overlapped_omics_union_annotation_samples\n",
    "overlapped_non_cancerous_samples = sorted(list(set(overlapped_omics_union_annotation_samples) & set(non_cancerous_samples)))\n",
    "print(\"length of overlapped omics and annotation samples: \", len(overlapped_omics_union_annotation_samples))\n",
    "print(\"length of overlapped non cancerous samples: \", len(overlapped_non_cancerous_samples))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 6.2 Overlapping over DTI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(overlapped_samples): 650\n",
      "len(overlapped_non_cancerous_samples): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>CCLE_ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Site_Primary</th>\n",
       "      <th>Site_Subtype1</th>\n",
       "      <th>Site_Subtype2</th>\n",
       "      <th>Site_Subtype3</th>\n",
       "      <th>Histology</th>\n",
       "      <th>Hist_Subtype1</th>\n",
       "      <th>...</th>\n",
       "      <th>Growth.Medium</th>\n",
       "      <th>Supplements</th>\n",
       "      <th>Freezing.Medium</th>\n",
       "      <th>Doubling.Time.from.Vendor</th>\n",
       "      <th>Doubling.Time.Calculated.hrs</th>\n",
       "      <th>type</th>\n",
       "      <th>type_refined</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>mutRate</th>\n",
       "      <th>tcga_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>NIHOVCAR3_OVARY</td>\n",
       "      <td>NIH:OVCAR-3</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>ovary</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI+20%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ovary</td>\n",
       "      <td>ovary</td>\n",
       "      <td>Ovary:Carcinoma</td>\n",
       "      <td>83.806951</td>\n",
       "      <td>OV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000002</td>\n",
       "      <td>HL60_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>HL-60</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>acute_myeloid_leukaemia</td>\n",
       "      <td>...</td>\n",
       "      <td>IMDM+20 % FBS</td>\n",
       "      <td>na</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>113.3</td>\n",
       "      <td>AML</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>81.133673</td>\n",
       "      <td>LAML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>HEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>HEL</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>acute_myeloid_leukaemia</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI1640+10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10%DMSO</td>\n",
       "      <td>36 hrs</td>\n",
       "      <td>38.4</td>\n",
       "      <td>AML</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>98.185377</td>\n",
       "      <td>LAML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000006</td>\n",
       "      <td>MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>MONO-MAC-6</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>acute_myeloid_leukaemia</td>\n",
       "      <td>...</td>\n",
       "      <td>90% RPMI 1640 +10% h.i. FBS</td>\n",
       "      <td>2mM L-glutamine + non-essential amino acids +1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>~  60 hrs</td>\n",
       "      <td>55.3</td>\n",
       "      <td>AML</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>77.065234</td>\n",
       "      <td>LAML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000007</td>\n",
       "      <td>LS513_LARGE_INTESTINE</td>\n",
       "      <td>LS513</td>\n",
       "      <td>primary</td>\n",
       "      <td>large_intestine</td>\n",
       "      <td>caecum</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>adenocarcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>112.349539</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>ACH-001199</td>\n",
       "      <td>SNUC2B_LARGE_INTESTINE</td>\n",
       "      <td>SNU-C2B</td>\n",
       "      <td>primary</td>\n",
       "      <td>large_intestine</td>\n",
       "      <td>caecum</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>adenocarcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>645</th>\n",
       "      <td>ACH-001203</td>\n",
       "      <td>SUPHD1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>SUP-HD1</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>lymph_node</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>Hodgkin_lymphoma</td>\n",
       "      <td>...</td>\n",
       "      <td>McCOY'S 5A+20% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lymphoma_Hodgkin</td>\n",
       "      <td>lymphoma_Hodgkin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>646</th>\n",
       "      <td>ACH-001233</td>\n",
       "      <td>VMRCLCD_LUNG</td>\n",
       "      <td>VMRC-LCD</td>\n",
       "      <td>primary</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>adenocarcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>EMEM+10% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>ACH-001306</td>\n",
       "      <td>8305C_THYROID</td>\n",
       "      <td>8305C</td>\n",
       "      <td>primary</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>anaplastic_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI1640+20%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10%DMSO</td>\n",
       "      <td>54hrs</td>\n",
       "      <td>121.6</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>Thyroid:Carcinoma</td>\n",
       "      <td>200.908236</td>\n",
       "      <td>THCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>ACH-001307</td>\n",
       "      <td>8505C_THYROID</td>\n",
       "      <td>8505C</td>\n",
       "      <td>primary</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>anaplastic_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>Eagle's minimal essential medium with 10% feta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>Thyroid:Carcinoma</td>\n",
       "      <td>141.208560</td>\n",
       "      <td>THCA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>649 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       depMapID                                      CCLE_ID         Name  \\\n",
       "0    ACH-000001                              NIHOVCAR3_OVARY  NIH:OVCAR-3   \n",
       "1    ACH-000002      HL60_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE        HL-60   \n",
       "2    ACH-000004       HEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE          HEL   \n",
       "3    ACH-000006  MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE   MONO-MAC-6   \n",
       "4    ACH-000007                        LS513_LARGE_INTESTINE        LS513   \n",
       "..          ...                                          ...          ...   \n",
       "644  ACH-001199                       SNUC2B_LARGE_INTESTINE      SNU-C2B   \n",
       "645  ACH-001203    SUPHD1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE      SUP-HD1   \n",
       "646  ACH-001233                                 VMRCLCD_LUNG     VMRC-LCD   \n",
       "647  ACH-001306                                8305C_THYROID        8305C   \n",
       "648  ACH-001307                                8505C_THYROID        8505C   \n",
       "\n",
       "      Pathology                        Site_Primary Site_Subtype1  \\\n",
       "0    metastasis                               ovary            NS   \n",
       "1       primary  haematopoietic_and_lymphoid_tissue            NS   \n",
       "2       primary  haematopoietic_and_lymphoid_tissue            NS   \n",
       "3       primary  haematopoietic_and_lymphoid_tissue            NS   \n",
       "4       primary                     large_intestine        caecum   \n",
       "..          ...                                 ...           ...   \n",
       "644     primary                     large_intestine        caecum   \n",
       "645     primary  haematopoietic_and_lymphoid_tissue    lymph_node   \n",
       "646     primary                                lung            NS   \n",
       "647     primary                             thyroid            NS   \n",
       "648     primary                             thyroid            NS   \n",
       "\n",
       "    Site_Subtype2 Site_Subtype3                Histology  \\\n",
       "0              NS            NS                carcinoma   \n",
       "1              NS            NS  haematopoietic_neoplasm   \n",
       "2              NS            NS  haematopoietic_neoplasm   \n",
       "3              NS            NS  haematopoietic_neoplasm   \n",
       "4              NS            NS                carcinoma   \n",
       "..            ...           ...                      ...   \n",
       "644            NS            NS                carcinoma   \n",
       "645            NS            NS        lymphoid_neoplasm   \n",
       "646            NS            NS                carcinoma   \n",
       "647            NS            NS                carcinoma   \n",
       "648            NS            NS                carcinoma   \n",
       "\n",
       "               Hist_Subtype1  ...  \\\n",
       "0                         NS  ...   \n",
       "1    acute_myeloid_leukaemia  ...   \n",
       "2    acute_myeloid_leukaemia  ...   \n",
       "3    acute_myeloid_leukaemia  ...   \n",
       "4             adenocarcinoma  ...   \n",
       "..                       ...  ...   \n",
       "644           adenocarcinoma  ...   \n",
       "645         Hodgkin_lymphoma  ...   \n",
       "646           adenocarcinoma  ...   \n",
       "647     anaplastic_carcinoma  ...   \n",
       "648     anaplastic_carcinoma  ...   \n",
       "\n",
       "                                         Growth.Medium  \\\n",
       "0                                          RPMI+20%FBS   \n",
       "1                                        IMDM+20 % FBS   \n",
       "2                                      RPMI1640+10%FBS   \n",
       "3                          90% RPMI 1640 +10% h.i. FBS   \n",
       "4                                                  NaN   \n",
       "..                                                 ...   \n",
       "644                                                NaN   \n",
       "645                                 McCOY'S 5A+20% FBS   \n",
       "646                                       EMEM+10% FBS   \n",
       "647                                    RPMI1640+20%FBS   \n",
       "648  Eagle's minimal essential medium with 10% feta...   \n",
       "\n",
       "                                           Supplements Freezing.Medium  \\\n",
       "0                                                  NaN          5%DMSO   \n",
       "1                                                   na          5%DMSO   \n",
       "2                                                  NaN         10%DMSO   \n",
       "3    2mM L-glutamine + non-essential amino acids +1...             NaN   \n",
       "4                                                  NaN             NaN   \n",
       "..                                                 ...             ...   \n",
       "644                                                NaN             NaN   \n",
       "645                                                NaN         5% DMSO   \n",
       "646                                                NaN         5% DMSO   \n",
       "647                                                NaN         10%DMSO   \n",
       "648                                                NaN             NaN   \n",
       "\n",
       "    Doubling.Time.from.Vendor  Doubling.Time.Calculated.hrs              type  \\\n",
       "0                         NaN                           NaN             ovary   \n",
       "1                         NaN                         113.3               AML   \n",
       "2                      36 hrs                          38.4               AML   \n",
       "3                   ~  60 hrs                          55.3               AML   \n",
       "4                         NaN                           NaN        colorectal   \n",
       "..                        ...                           ...               ...   \n",
       "644                       NaN                           NaN        colorectal   \n",
       "645                       NaN                           NaN  lymphoma_Hodgkin   \n",
       "646                       NaN                           NaN          lung_NSC   \n",
       "647                     54hrs                         121.6           thyroid   \n",
       "648                       NaN                           NaN           thyroid   \n",
       "\n",
       "         type_refined PATHOLOGIST_ANNOTATION     mutRate  tcga_code  \n",
       "0               ovary        Ovary:Carcinoma   83.806951         OV  \n",
       "1                 AML           Leukemia:AML   81.133673       LAML  \n",
       "2                 AML           Leukemia:AML   98.185377       LAML  \n",
       "3                 AML           Leukemia:AML   77.065234       LAML  \n",
       "4          colorectal   Colorectal:Carcinoma  112.349539  COAD/READ  \n",
       "..                ...                    ...         ...        ...  \n",
       "644        colorectal                    NaN         NaN  COAD/READ  \n",
       "645  lymphoma_Hodgkin                    NaN         NaN        NaN  \n",
       "646          lung_NSC                    NaN         NaN       LUAD  \n",
       "647           thyroid      Thyroid:Carcinoma  200.908236       THCA  \n",
       "648           thyroid      Thyroid:Carcinoma  141.208560       THCA  \n",
       "\n",
       "[649 rows x 33 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# get the overlapped samples among all samples (methy_mapped_samples, gene_samples, transcript_samples, protein_mapped_samples, drug_samples)\n",
    "dti_overlapped_samples = sorted(list(set(overlapped_omics_union_annotation_samples) & set(drug_samples)))\n",
    "print(\"len(overlapped_samples):\", len(dti_overlapped_samples))\n",
    "# need to confirm zero intersection between the overlapped samples and non_cancerous_samples\n",
    "dti_overlapped_non_cancerous_samples = sorted(list(set(dti_overlapped_samples) & set(non_cancerous_samples)))\n",
    "print(\"len(overlapped_non_cancerous_samples):\", len(dti_overlapped_non_cancerous_samples))\n",
    "# if there are overlapped samples between the dti_overlapped_samples and non_cancerous_samples, then remove these samples from the dti_overlapped_samples\n",
    "if len(dti_overlapped_non_cancerous_samples) > 0:\n",
    "    dti_overlapped_samples = sorted(list(set(dti_overlapped_samples) - set(dti_overlapped_non_cancerous_samples)))\n",
    "# convert the overlapped_samples to the dataframe with annotation\n",
    "dti_overlapped_samples_df = pd.merge(pd.DataFrame(dti_overlapped_samples, columns=['depMapID']), cell_line_anno_df, on='depMapID', how='inner')\n",
    "display(dti_overlapped_samples_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "metadata": {},
   "outputs": [],
   "source": [
    "# mkdir for processsing folder\n",
    "import os\n",
    "if not os.path.exists('./data/process_data'):\n",
    "    os.makedirs('./data/process_data')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "metadata": {},
   "outputs": [],
   "source": [
    "dti_overlapped_samples_df.to_csv('./data/process_data/dti_overlapped_samples.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['depMapID', 'CCLE_ID', 'Name', 'Pathology', 'Site_Primary',\n",
      "       'Site_Subtype1', 'Site_Subtype2', 'Site_Subtype3', 'Histology',\n",
      "       'Hist_Subtype1', 'Hist_Subtype2', 'Hist_Subtype3', 'Gender',\n",
      "       'Life_Stage', 'Age', 'Race', 'Geo_Loc', 'inferred_ethnicity',\n",
      "       'Site_Of_Finding', 'Disease', 'Annotation_Source',\n",
      "       'Original.Source.of.Cell.Line', 'Characteristics', 'Growth.Medium',\n",
      "       'Supplements', 'Freezing.Medium', 'Doubling.Time.from.Vendor',\n",
      "       'Doubling.Time.Calculated.hrs', 'type', 'type_refined',\n",
      "       'PATHOLOGIST_ANNOTATION', 'mutRate', 'tcga_code'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print(dti_overlapped_samples_df.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. Annotate the samples with disease"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 7.1 Add annotation from Cellosaurus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import pandas as pd\n",
    "# import requests\n",
    "\n",
    "# url = 'https://ftp.expasy.org/databases/cellosaurus/cellosaurus.obo'\n",
    "# response = requests.get(url)\n",
    "\n",
    "# with open('./data/raw_data/cellosaurus.obo', 'wb') as f:\n",
    "#     f.write(response.content)\n",
    "\n",
    "# # Path to the downloaded OBO file\n",
    "# obo_path = './data/raw_data/cellosaurus.obo'\n",
    "\n",
    "# entries = []\n",
    "# current_entry = {}\n",
    "\n",
    "# # Read and parse\n",
    "# with open(obo_path, 'r', encoding='utf-8') as file:\n",
    "#     for line in file:\n",
    "#         line = line.strip()\n",
    "\n",
    "#         if line == \"[Term]\":\n",
    "#             if current_entry.get(\"xref\") and any(\"NCBI_TaxID:9606\" in x for x in current_entry[\"xref\"]):\n",
    "#                 # Separate NCIt and ORDO xrefs\n",
    "#                 ncit_refs = [x for x in current_entry[\"xref\"] if x.startswith(\"NCIt:\")]\n",
    "#                 ordo_refs = [x for x in current_entry[\"xref\"] if x.startswith(\"ORDO:\")]\n",
    "#                 current_entry[\"xref_NCIt\"] = \"; \".join(ncit_refs)\n",
    "#                 current_entry[\"xref_ORDO\"] = \"; \".join(ordo_refs)\n",
    "#                 entries.append({\n",
    "#                     \"id\": current_entry.get(\"id\", \"\"),\n",
    "#                     \"name\": current_entry.get(\"name\", \"\"),\n",
    "#                     \"synonym\": \"; \".join(current_entry.get(\"synonym\", [])),\n",
    "#                     \"xref_NCIt\": current_entry.get(\"xref_NCIt\", \"\"),\n",
    "#                     \"xref_ORDO\": current_entry.get(\"xref_ORDO\", \"\")\n",
    "#                 })\n",
    "#             current_entry = {\"synonym\": [], \"xref\": []}\n",
    "\n",
    "#         elif line.startswith(\"id:\"):\n",
    "#             current_entry[\"id\"] = line.split(\"id:\")[1].strip()\n",
    "\n",
    "#         elif line.startswith(\"name:\"):\n",
    "#             current_entry[\"name\"] = line.split(\"name:\")[1].strip()\n",
    "\n",
    "#         elif line.startswith(\"synonym:\"):\n",
    "#             synonym = line.split(\"synonym:\")[1].split(\"RELATED\")[0].strip().strip('\"')\n",
    "#             current_entry[\"synonym\"].append(synonym)\n",
    "\n",
    "#         elif line.startswith(\"xref:\"):\n",
    "#             current_entry[\"xref\"].append(line.split(\"xref:\")[1].strip())\n",
    "\n",
    "# # Handle last entry\n",
    "# if current_entry.get(\"xref\") and any(\"NCBI_TaxID:9606\" in x for x in current_entry[\"xref\"]):\n",
    "#     ncit_refs = [x for x in current_entry[\"xref\"] if x.startswith(\"NCIt:\")]\n",
    "#     ordo_refs = [x for x in current_entry[\"xref\"] if x.startswith(\"ORDO:\")]\n",
    "#     entries.append({\n",
    "#         \"id\": current_entry.get(\"id\", \"\"),\n",
    "#         \"name\": current_entry.get(\"name\", \"\"),\n",
    "#         \"synonym\": \"; \".join(current_entry.get(\"synonym\", [])),\n",
    "#         \"xref_NCIt\": \"; \".join(ncit_refs),\n",
    "#         \"xref_ORDO\": \"; \".join(ordo_refs)\n",
    "#     })\n",
    "\n",
    "# # Create DataFrame\n",
    "# cellosaurus_parsed_df = pd.DataFrame(entries)\n",
    "# cellosaurus_parsed_df = cellosaurus_parsed_df[[\"id\", \"name\", \"synonym\", \"xref_NCIt\", \"xref_ORDO\"]]\n",
    "# cellosaurus_parsed_df.to_csv('./data/raw_data/cellosaurus_parsed.csv', index=False)\n",
    "# # Show preview\n",
    "# display(cellosaurus_parsed_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Step 1: Keep the desired columns\n",
    "# dti_overlapped_samples_desc_df = dti_overlapped_samples_df[['depMapID', 'Name', 'Pathology', 'Histology', 'type', 'PATHOLOGIST_ANNOTATION', 'tcga_code']]\n",
    "# display(dti_overlapped_samples_desc_df)\n",
    "\n",
    "# # Step 2: Try direct merge on Name == name using left join\n",
    "# matched_df = pd.merge(\n",
    "#     dti_overlapped_samples_desc_df,\n",
    "#     cellosaurus_parsed_df,\n",
    "#     left_on='Name',\n",
    "#     right_on='name',\n",
    "#     how='left'\n",
    "# )\n",
    "\n",
    "# # Step 3: Find rows where no match occurred (name is NaN)\n",
    "# unmatched_df = matched_df[matched_df['name'].isna()].copy()\n",
    "\n",
    "# # Step 4: Drop the Cellosaurus columns to prep for synonym match\n",
    "# cols_to_drop = [col for col in matched_df.columns if col in cellosaurus_parsed_df.columns]\n",
    "# unmatched_df = unmatched_df.drop(columns=cols_to_drop)\n",
    "\n",
    "# # Step 5: Expand cellosaurus synonyms\n",
    "# synonym_expanded_df = cellosaurus_parsed_df.copy()\n",
    "# synonym_expanded_df = synonym_expanded_df.dropna(subset=['synonym'])\n",
    "# synonym_expanded_df = synonym_expanded_df.assign(\n",
    "#     synonym=synonym_expanded_df['synonym'].str.split(';')\n",
    "# ).explode('synonym')\n",
    "# synonym_expanded_df['synonym'] = synonym_expanded_df['synonym'].str.strip()\n",
    "\n",
    "# # Step 6: Left join unmatched rows with synonym-expanded Cellosaurus\n",
    "# synonym_matched_df = pd.merge(\n",
    "#     unmatched_df,\n",
    "#     synonym_expanded_df,\n",
    "#     left_on='Name',\n",
    "#     right_on='synonym',\n",
    "#     how='left'\n",
    "# )\n",
    "\n",
    "# # Step 7: Combine direct-matched (non-NaN name) and synonym-matched rows\n",
    "# all_match_df = pd.concat(\n",
    "#     [matched_df[matched_df['name'].notna()], synonym_matched_df],\n",
    "#     ignore_index=True\n",
    "# )\n",
    "\n",
    "# # Optional: Drop duplicates based on depMapID if needed\n",
    "# all_match_df = all_match_df.drop_duplicates(subset=['depMapID'])\n",
    "# display(all_match_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Function to store primary and extra xref matches with better column handling\n",
    "# def process_xref_columns_with_extras(df):\n",
    "#     # Create a copy to avoid modifying the original\n",
    "#     result_df = df.copy()\n",
    "    \n",
    "#     # Always create all the columns, even if they might end up empty\n",
    "#     result_df['xref_NCIt_ID'] = ''\n",
    "#     result_df['xref_NCIt_name'] = ''\n",
    "#     result_df['xref_NCIt_extra_ID'] = ''\n",
    "#     result_df['xref_NCIt_extra_name'] = ''\n",
    "#     result_df['xref_ORDO_ID'] = ''\n",
    "#     result_df['xref_ORDO_name'] = ''\n",
    "#     result_df['xref_ORDO_extra_ID'] = ''\n",
    "#     result_df['xref_ORDO_extra_name'] = ''\n",
    "    \n",
    "#     # Process xref_NCIt if it exists\n",
    "#     if 'xref_NCIt' in result_df.columns:\n",
    "#         # Only process rows that have xref_NCIt values\n",
    "#         mask = result_df['xref_NCIt'].notna() & (result_df['xref_NCIt'] != '')\n",
    "        \n",
    "#         for idx in result_df[mask].index:\n",
    "#             xref_value = result_df.loc[idx, 'xref_NCIt']\n",
    "            \n",
    "#             if '; ' in xref_value:  # Multiple entries\n",
    "#                 entries = xref_value.split('; ')\n",
    "                \n",
    "#                 # Process first entry for primary columns\n",
    "#                 if ' ! ' in entries[0]:\n",
    "#                     id_val, name_val = entries[0].split(' ! ', 1)\n",
    "#                     result_df.loc[idx, 'xref_NCIt_ID'] = id_val\n",
    "#                     result_df.loc[idx, 'xref_NCIt_name'] = name_val\n",
    "                \n",
    "#                 # Process additional entries for extra columns\n",
    "#                 extra_ids = []\n",
    "#                 extra_names = []\n",
    "#                 for entry in entries[1:]:\n",
    "#                     if ' ! ' in entry:\n",
    "#                         id_val, name_val = entry.split(' ! ', 1)\n",
    "#                         extra_ids.append(id_val)\n",
    "#                         extra_names.append(name_val)\n",
    "                \n",
    "#                 if extra_ids:\n",
    "#                     result_df.loc[idx, 'xref_NCIt_extra_ID'] = '; '.join(extra_ids)\n",
    "#                     result_df.loc[idx, 'xref_NCIt_extra_name'] = '; '.join(extra_names)\n",
    "                    \n",
    "#             elif ' ! ' in xref_value:  # Single entry\n",
    "#                 id_val, name_val = xref_value.split(' ! ', 1)\n",
    "#                 result_df.loc[idx, 'xref_NCIt_ID'] = id_val\n",
    "#                 result_df.loc[idx, 'xref_NCIt_name'] = name_val\n",
    "        \n",
    "#         # Drop the original column\n",
    "#         result_df = result_df.drop(columns=['xref_NCIt'])\n",
    "    \n",
    "#     # Process xref_ORDO if it exists\n",
    "#     if 'xref_ORDO' in result_df.columns:\n",
    "#         # Only process rows that have xref_ORDO values\n",
    "#         mask = result_df['xref_ORDO'].notna() & (result_df['xref_ORDO'] != '')\n",
    "        \n",
    "#         for idx in result_df[mask].index:\n",
    "#             xref_value = result_df.loc[idx, 'xref_ORDO']\n",
    "            \n",
    "#             if '; ' in xref_value:  # Multiple entries\n",
    "#                 entries = xref_value.split('; ')\n",
    "                \n",
    "#                 # Process first entry for primary columns\n",
    "#                 if ' ! ' in entries[0]:\n",
    "#                     id_val, name_val = entries[0].split(' ! ', 1)\n",
    "#                     result_df.loc[idx, 'xref_ORDO_ID'] = id_val\n",
    "#                     result_df.loc[idx, 'xref_ORDO_name'] = name_val\n",
    "                \n",
    "#                 # Process additional entries for extra columns\n",
    "#                 extra_ids = []\n",
    "#                 extra_names = []\n",
    "#                 for entry in entries[1:]:\n",
    "#                     if ' ! ' in entry:\n",
    "#                         id_val, name_val = entry.split(' ! ', 1)\n",
    "#                         extra_ids.append(id_val)\n",
    "#                         extra_names.append(name_val)\n",
    "                \n",
    "#                 if extra_ids:\n",
    "#                     result_df.loc[idx, 'xref_ORDO_extra_ID'] = '; '.join(extra_ids)\n",
    "#                     result_df.loc[idx, 'xref_ORDO_extra_name'] = '; '.join(extra_names)\n",
    "                    \n",
    "#             elif ' ! ' in xref_value:  # Single entry\n",
    "#                 id_val, name_val = xref_value.split(' ! ', 1)\n",
    "#                 result_df.loc[idx, 'xref_ORDO_ID'] = id_val\n",
    "#                 result_df.loc[idx, 'xref_ORDO_name'] = name_val\n",
    "        \n",
    "#         # Drop the original column\n",
    "#         result_df = result_df.drop(columns=['xref_ORDO'])\n",
    "    \n",
    "#     return result_df\n",
    "\n",
    "# # Apply the function\n",
    "# all_match_df = process_xref_columns_with_extras(all_match_df)\n",
    "\n",
    "# # Update the column list to include the extra columns\n",
    "# dti_overlapped_samples_desc_co_df = all_match_df[['depMapID', 'Name', 'Pathology', 'Histology', \n",
    "#                                              'type', 'PATHOLOGIST_ANNOTATION', 'tcga_code', 'id', \n",
    "#                                              'xref_NCIt_ID', 'xref_NCIt_name', 'xref_NCIt_extra_ID', 'xref_NCIt_extra_name',\n",
    "#                                              'xref_ORDO_ID', 'xref_ORDO_name', 'xref_ORDO_extra_ID', 'xref_ORDO_extra_name']]\n",
    "# display(dti_overlapped_samples_desc_co_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 7.2 Cell line disease match"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 7.2.1 Cell line disease hard match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "metadata": {},
   "outputs": [],
   "source": [
    "# bmg_disease_name_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Disease/BioMedGraphica_Conn_Disease_GUI_Name.csv')\n",
    "# display(bmg_disease_name_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Step 1: Expand BMG_Disease_Name (still lowercased and exploded)\n",
    "# bmg_expanded = bmg_disease_name_df.copy()\n",
    "\n",
    "# # First, strip any surrounding quotes and fill NaN values with empty strings\n",
    "# bmg_expanded['Disease_Name_List'] = bmg_expanded['Disease_Name_List'].fillna('')\n",
    "# bmg_expanded['Disease_Name_List'] = bmg_expanded['Disease_Name_List'].str.replace('^\"|\"$', '', regex=True)\n",
    "\n",
    "# # Split on pipe with spaces\n",
    "# bmg_expanded['Disease_Name_List'] = bmg_expanded['Disease_Name_List'].str.split(' \\| ')\n",
    "\n",
    "# # Apply strip only to non-empty lists\n",
    "# bmg_expanded['Disease_Name_List'] = bmg_expanded['Disease_Name_List'].apply(\n",
    "#     lambda x: [name.strip() for name in x] if isinstance(x, list) else []\n",
    "# )\n",
    "\n",
    "# # Explode the list into separate rows\n",
    "# bmg_expanded = bmg_expanded.explode('Disease_Name_List')\n",
    "\n",
    "# # Remove rows with empty disease names after exploding\n",
    "# bmg_expanded = bmg_expanded[bmg_expanded['Disease_Name_List'].str.len() > 0]\n",
    "\n",
    "# # Create lowercase version for easier matching\n",
    "# bmg_expanded['Disease_Name_List_lower'] = bmg_expanded['Disease_Name_List'].str.lower()\n",
    "\n",
    "# display(bmg_expanded)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Create a lookup list of tuples: (Conn_ID, Original_Name, Lower_Name)\n",
    "# bmg_records = bmg_expanded[['BioMedGraphica_Conn_ID', 'Disease_Name_List', 'Disease_Name_List_lower']].to_records(index=False)\n",
    "\n",
    "# # Updated matching function: look for exact full-name match\n",
    "# def match_bmg_disease_exact(xref_name, bmg_records):\n",
    "#     if pd.isna(xref_name):\n",
    "#         return (None, None)\n",
    "#     xref_name_lower = str(xref_name).strip().lower()\n",
    "    \n",
    "#     for conn_id, original_name, bmg_name_lower in bmg_records:\n",
    "#         if xref_name_lower == bmg_name_lower:\n",
    "#             return (conn_id, original_name)\n",
    "    \n",
    "#     return (None, None)\n",
    "\n",
    "# # NCIt\n",
    "# ncit_matches = dti_overlapped_samples_desc_co_df['xref_NCIt_name'].apply(lambda x: match_bmg_disease_exact(x, bmg_records))\n",
    "# dti_overlapped_samples_desc_co_df['NCIt_BMGC_ID'] = ncit_matches.apply(lambda x: x[0])\n",
    "# dti_overlapped_samples_desc_co_df['NCIt_BMGC_name'] = ncit_matches.apply(lambda x: x[1])\n",
    "\n",
    "# # ORDO\n",
    "# ordo_matches = dti_overlapped_samples_desc_co_df['xref_ORDO_name'].apply(lambda x: match_bmg_disease_exact(x, bmg_records))\n",
    "# dti_overlapped_samples_desc_co_df['ORDO_BMGC_ID'] = ordo_matches.apply(lambda x: x[0])\n",
    "# dti_overlapped_samples_desc_co_df['ORDO_BMGC_name'] = ordo_matches.apply(lambda x: x[1])\n",
    "\n",
    "# display(dti_overlapped_samples_desc_co_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # keep the columns ['depMapID', 'Name', 'Pathology', 'Histology', 'type', 'PATHOLOGIST_ANNOTATION', 'tcga_code', 'id', 'xref_NCIt_ID', 'xref_NCIt_name', 'NCIt_BMGC_ID', 'NCIt_BMGC_name', 'xref_ORDO_ID', 'xref_ORDO_name', 'ORDO_BMGC_ID', 'ORDO_BMGC_name']\n",
    "# dti_overlapped_samples_desc_co_df = dti_overlapped_samples_desc_co_df[['depMapID', 'Name', 'Pathology', 'Histology', 'type', 'PATHOLOGIST_ANNOTATION', 'tcga_code', 'id', 'xref_NCIt_ID', 'xref_NCIt_name', 'NCIt_BMGC_ID', 'NCIt_BMGC_name', 'xref_ORDO_ID', 'xref_ORDO_name', 'ORDO_BMGC_ID', 'ORDO_BMGC_name']]\n",
    "# display(dti_overlapped_samples_desc_co_df)\n",
    "# dti_overlapped_samples_desc_co_df.to_csv('./data/process_data/dti_overlapped_samples_desc_co.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge NCIt_BMGC_ID/ORDO_BMGC_ID and NCIt_BMGC_name/ORDO_BMGC_name into BMGC_Matched columns\n",
    "# Strategy:\n",
    "# 1. Prioritize NCIt data when both are available\n",
    "# 2. Use NCIt data if only NCIt is available\n",
    "# 3. Use ORDO data if only ORDO is available\n",
    "# 4. Leave as null if both are null\n",
    "\n",
    "# dti_matched_df = dti_overlapped_samples_desc_co_df.copy()\n",
    "\n",
    "# # Create new merged ID column\n",
    "# dti_matched_df['BMGC_Matched_ID'] = (\n",
    "#     dti_matched_df['NCIt_BMGC_ID'].fillna(\n",
    "#         dti_matched_df['ORDO_BMGC_ID']\n",
    "#     )\n",
    "# )\n",
    "\n",
    "# # Create new merged name column\n",
    "# dti_matched_df['BMGC_Matched_name'] = (\n",
    "#     dti_matched_df['NCIt_BMGC_name'].fillna(\n",
    "#         dti_matched_df['ORDO_BMGC_name']\n",
    "#     )\n",
    "# )\n",
    "\n",
    "# # Display the dataframe with new columns\n",
    "# display(dti_matched_df)\n",
    "\n",
    "# # Check number of null values in new columns\n",
    "# print(f\"Null values in BMGC_Matched_ID: {dti_matched_df['BMGC_Matched_ID'].isna().sum()}\")\n",
    "# print(f\"Null values in BMGC_Matched_name: {dti_matched_df['BMGC_Matched_name'].isna().sum()}\")\n",
    "\n",
    "# # only keep the columns where the BMGC_Matched_ID is not null\n",
    "# dti_matched_df = dti_matched_df[dti_matched_df['BMGC_Matched_ID'].notna()].reset_index(drop=True)\n",
    "# display(dti_matched_df)\n",
    "\n",
    "# # Save to CSV\n",
    "# dti_matched_df.to_csv('./data/process_data/dti_matched.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 7.2.2 Cell line disease soft match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # check the unmatched samples in the dti_overlapped_samples_desc_co_df by checking the NaN in the NCIt_BMGC_ID and ORDO_BMGC_ID\n",
    "# unmatched_samples_df = dti_overlapped_samples_desc_co_df[dti_overlapped_samples_desc_co_df['NCIt_BMGC_ID'].isna() & dti_overlapped_samples_desc_co_df['ORDO_BMGC_ID'].isna()]\n",
    "# # keep the columns ['depMapID', 'Name', 'Pathology', 'Histology', 'type', 'PATHOLOGIST_ANNOTATION', 'tcga_code', 'id', 'xref_NCIt_name', , 'xref_ORDO_name']\n",
    "# unmatched_samples_df = unmatched_samples_df[['depMapID', 'Name', 'Pathology', 'Histology', 'type', 'PATHOLOGIST_ANNOTATION', 'tcga_code', 'id', 'xref_NCIt_name', 'xref_ORDO_name']]\n",
    "# display(unmatched_samples_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "metadata": {},
   "outputs": [],
   "source": [
    "# embed the disease name for bmg_disease_df by bert-based model\n",
    "import torch\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "from typing import List, Tuple, Dict\n",
    "from transformers import AutoTokenizer, AutoModel\n",
    "from torch.utils.data import DataLoader, Dataset\n",
    "\n",
    "class SentenceDataset(Dataset):\n",
    "    def __init__(self, sentences: List[str]):\n",
    "        self.sentences = sentences\n",
    "\n",
    "    def __len__(self):\n",
    "        return len(self.sentences)\n",
    "\n",
    "    def __getitem__(self, idx):\n",
    "        return self.sentences[idx]\n",
    "\n",
    "\n",
    "class TextEncoder():\n",
    "    def __init__(self, model_path: str = \"dmis-lab/biobert-v1.1\", device: str = \"cuda\"):\n",
    "        \"\"\"\n",
    "        Args:\n",
    "            model_path (str, optional): Path to the deberta model. Defaults to 'dmis-lab/biobert-v1.1'.\n",
    "            device (str, optional): Device to run the model on ('cpu' or 'cuda'). Defaults to 'cpu'.\n",
    "        \"\"\"\n",
    "        self.model_path = model_path\n",
    "        self.device = device\n",
    "        self.model = None\n",
    "        self.tokenizer = None\n",
    "\n",
    "    def load_model(self):\n",
    "        \"\"\"\n",
    "        Load the deberta model and tokenizer from the specified model path.\n",
    "        \"\"\"\n",
    "        self.tokenizer = AutoTokenizer.from_pretrained(self.model_path)\n",
    "        self.model = AutoModel.from_pretrained(self.model_path).to(self.device)\n",
    "\n",
    "    def generate_embeddings(self, sentences: List[str], batch_size: int = 32, seq_emb_dim: int = 64) -> torch.Tensor:\n",
    "        \"\"\"\n",
    "        Generate a single-dimensional embedding for each sentence.\n",
    "\n",
    "        Args:\n",
    "            sentences (List[str]): List of sentences to embed.\n",
    "            batch_size (int, optional): Batch size for DataLoader. Defaults to 32.\n",
    "\n",
    "        Returns:\n",
    "            List[float]: List of single-dimensional embeddings.\n",
    "        \"\"\"\n",
    "        dataset = SentenceDataset(sentences)\n",
    "        dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=False)\n",
    "\n",
    "        embedding_batches = []\n",
    "        for batch in tqdm(dataloader, desc=\"Embedding sentences\", unit=\"batch\"):\n",
    "            inputs = self.tokenizer(batch, return_tensors=\"pt\", padding=True, truncation=True, max_length=512).to(self.device)\n",
    "            with torch.no_grad():\n",
    "                outputs = self.model(**inputs)\n",
    "            # Handle single batch case properly\n",
    "            mean_embeddings = torch.mean(outputs.last_hidden_state, dim=1)  # [batch_size, hidden_dim]\n",
    "            \n",
    "            # For adaptive pooling, we need to reshape for 1D adaptive pooling\n",
    "            # [batch_size, 1, hidden_dim] -> [batch_size, 1, seq_emb_dim] -> [batch_size, seq_emb_dim]\n",
    "            batch_size = mean_embeddings.size(0)\n",
    "            reshaped = mean_embeddings.view(batch_size, 1, -1)\n",
    "            projected = torch.nn.functional.adaptive_avg_pool1d(reshaped, output_size=seq_emb_dim)\n",
    "            projected = projected.squeeze(1)  # Only squeeze dimension 1, keep batch dimension\n",
    "            embedding_batches.append(projected)\n",
    "        return torch.cat(embedding_batches, dim=0)\n",
    "\n",
    "    def save_embeddings(self, embeddings, output_npy_path):\n",
    "        \"\"\"\n",
    "        Save embeddings to a .npy file.\n",
    "        \n",
    "        Args:\n",
    "            embeddings (torch.Tensor): The embeddings to save.\n",
    "            output_npy_path (str): Path to save the embeddings file.\n",
    "        \"\"\"\n",
    "        # Move embeddings to CPU before converting to numpy\n",
    "        embeddings_cpu = embeddings.cpu().numpy()\n",
    "        np.save(output_npy_path, embeddings_cpu)\n",
    "        print(f\"Embeddings saved at {output_npy_path} with shape {embeddings_cpu.shape}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # convert the bmg_disease_name_df to the list of disease names of each rows\n",
    "# disease_names = list(bmg_disease_name_df['Disease_Name_List'])\n",
    "# # to make sure each disease name is a string\n",
    "# disease_names = [str(name) for name in disease_names]\n",
    "# print(len(disease_names))\n",
    "# print(disease_names[:5])\n",
    "# print(disease_names[-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  # Use language model to embed the name and description\n",
    "# name_sentence_list = disease_names\n",
    "# text_encoder = TextEncoder()\n",
    "# text_encoder.load_model()\n",
    "# name_embeddings = text_encoder.generate_embeddings(name_sentence_list, batch_size=32, seq_emb_dim=768)\n",
    "# print(name_embeddings.shape)\n",
    "# # mkdir folder BMG_emb\n",
    "# if not os.path.exists('./data/BMG_emb'):\n",
    "#     os.makedirs('./data/BMG_emb')\n",
    "# text_encoder.save_embeddings(name_embeddings, './data/BMG_emb/disease_name_embeddings.npy')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # convert each row in the unmatched_samples_df to a sentence like \"Name: NIH:OVCAR-3, Pathology: OVARY, Histology: CARCINOMA, type: CANCER, PATHOLOGIST_ANNOTATION: OVARY, tcga_code: OV\"\n",
    "# depmap_desc_sentence_list = []\n",
    "# for idx, row in unmatched_samples_df.iterrows():\n",
    "#     depmap_desc_sentence_list.append(f\" {row['type']}, {row['PATHOLOGIST_ANNOTATION']}\")\n",
    "\n",
    "# # and create the ncit_desc_sentence_list and ordo_desc_sentence_list\n",
    "# ncit_desc_sentence_list = list(unmatched_samples_df['xref_NCIt_name'])\n",
    "# ordo_desc_sentence_list = list(unmatched_samples_df['xref_ORDO_name'])\n",
    "\n",
    "# # make sure each element in the depmap_desc_sentence_list, ncit_desc_sentence_list and ordo_desc_sentence_list is a string\n",
    "# depmap_desc_sentence_list = [str(name) for name in depmap_desc_sentence_list]\n",
    "# ncit_desc_sentence_list = [str(name) for name in ncit_desc_sentence_list]\n",
    "# ordo_desc_sentence_list = [str(name) for name in ordo_desc_sentence_list]\n",
    "\n",
    "# print(depmap_desc_sentence_list)\n",
    "# print(ncit_desc_sentence_list)\n",
    "# print(ordo_desc_sentence_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Define the sentence lists to process and their corresponding prefixes\n",
    "# sentence_list_mapping = {\n",
    "#     'depmap': depmap_desc_sentence_list,\n",
    "#     'ncit': ncit_desc_sentence_list,\n",
    "#     'ordo': ordo_desc_sentence_list\n",
    "# }\n",
    "\n",
    "# # Define top_k for this matching task\n",
    "# top_k = 2\n",
    "\n",
    "# # Process each sentence list and add the matches to the DataFrame\n",
    "# for prefix, sentence_list in sentence_list_mapping.items():\n",
    "#     # Lists to store the matches for each description\n",
    "#     all_matched_lists = [[] for _ in range(top_k)]\n",
    "#     all_matched_name_lists = [[] for _ in range(top_k)]\n",
    "#     all_similarity_lists = [[] for _ in range(top_k)]\n",
    "    \n",
    "#     for desc in sentence_list:\n",
    "#         desc_embeddings = text_encoder.generate_embeddings([desc], batch_size=1, seq_emb_dim=768)\n",
    "#         # Calculate cosine similarity\n",
    "#         cos = torch.nn.CosineSimilarity(dim=1, eps=1e-6)\n",
    "#         similarity = cos(name_embeddings, desc_embeddings)\n",
    "#         # Find the index of top k most similar diseases\n",
    "#         top_k_idx = torch.argsort(similarity, descending=True)[:top_k]\n",
    "        \n",
    "#         # Process each of the top k matches\n",
    "#         for rank, idx in enumerate(top_k_idx):\n",
    "#             idx_int = idx.item()\n",
    "#             disease_id = bmg_disease_name_df.iloc[idx_int]['BioMedGraphica_Conn_ID']\n",
    "#             disease_name = disease_names[idx_int]\n",
    "#             sim_score = similarity[idx_int].item()\n",
    "            \n",
    "#             # Store in the corresponding lists\n",
    "#             all_matched_lists[rank].append(disease_id)\n",
    "#             all_matched_name_lists[rank].append(disease_name)\n",
    "#             all_similarity_lists[rank].append(sim_score)\n",
    "    \n",
    "#     # Add the matched data to the DataFrame with the appropriate prefix\n",
    "#     for i in range(top_k):\n",
    "#         rank = i + 1\n",
    "#         unmatched_samples_df[f'{prefix}_match_{rank}_disease'] = all_matched_lists[i]\n",
    "#         unmatched_samples_df[f'{prefix}_match_{rank}_disease_name'] = all_matched_name_lists[i]\n",
    "#         unmatched_samples_df[f'{prefix}_match_{rank}_similarity'] = all_similarity_lists[i]\n",
    "    \n",
    "#     print(f\"Finished processing {prefix} descriptions\")\n",
    "\n",
    "# # Display the updated DataFrame\n",
    "# display(unmatched_samples_df)\n",
    "\n",
    "# # Save the updated DataFrame to CSV\n",
    "# unmatched_samples_df.to_csv('./data/process_data/unmatched_samples_softmatch.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 7.3 Manual filter and match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "metadata": {},
   "outputs": [],
   "source": [
    "# unmatched_samples_df = pd.read_csv('./data/process_data/unmatched_samples_softmatch.csv')\n",
    "# # add the 2 empty columnns 'BMGC_manual_ID' and 'BMGC_manual_name' to the unmatched_samples_manual_df\n",
    "# unmatched_samples_manual_df = unmatched_samples_df.copy()\n",
    "# unmatched_samples_manual_df['BMGC_manual_ID'] = None\n",
    "# unmatched_samples_manual_df['BMGC_manual_name'] = None\n",
    "# # change the order of the columns to make the 'BMGC_manual_ID' and 'BMGC_manual_name' after id\n",
    "# unmatched_samples_manual_df = unmatched_samples_manual_df[['depMapID', 'Name', 'Pathology', 'Histology', 'type', 'PATHOLOGIST_ANNOTATION', 'tcga_code', 'id', 'BMGC_manual_ID', 'BMGC_manual_name', 'xref_NCIt_name', 'xref_ORDO_name', 'depmap_match_1_disease', 'depmap_match_1_disease_name', 'depmap_match_2_disease', 'depmap_match_2_disease_name', 'ncit_match_1_disease', 'ncit_match_1_disease_name', 'ncit_match_2_disease', 'ncit_match_2_disease_name', 'ordo_match_1_disease', 'ordo_match_1_disease_name', 'ordo_match_2_disease', 'ordo_match_2_disease_name']]\n",
    "# display(unmatched_samples_manual_df)\n",
    "# unmatched_samples_manual_df.to_csv('./data/process_data/unmatched_samples_manual.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 7.4 Combine matched and manual_unmatached"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>BMGC_Disease_name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Histology</th>\n",
       "      <th>type</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>tcga_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000002</td>\n",
       "      <td>HL-60</td>\n",
       "      <td>BMGC_DS01407</td>\n",
       "      <td>acute myeloid leukemia</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>LAML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>HEL</td>\n",
       "      <td>BMGC_DS16646</td>\n",
       "      <td>erythroleukemia</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>LAML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000007</td>\n",
       "      <td>LS513</td>\n",
       "      <td>BMGC_DS08517</td>\n",
       "      <td>cecum adenocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000008</td>\n",
       "      <td>A101D</td>\n",
       "      <td>BMGC_DS01559</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>malignant_melanoma</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>Skin:Melanoma</td>\n",
       "      <td>SKCM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000009</td>\n",
       "      <td>C2BBe1</td>\n",
       "      <td>BMGC_DS05624</td>\n",
       "      <td>colon adenocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>549</th>\n",
       "      <td>ACH-000976</td>\n",
       "      <td>HuCCT1</td>\n",
       "      <td>BMGC_DS05988</td>\n",
       "      <td>intrahepatic cholangiocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>bile_duct</td>\n",
       "      <td>Biliary_Tract:Carcinoma</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>550</th>\n",
       "      <td>ACH-000979</td>\n",
       "      <td>DU 145</td>\n",
       "      <td>BMGC_DS06988</td>\n",
       "      <td>prostate carcinoma</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>prostate</td>\n",
       "      <td>Prostate:Carcinoma</td>\n",
       "      <td>PRAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>551</th>\n",
       "      <td>ACH-000997</td>\n",
       "      <td>HCT-15</td>\n",
       "      <td>BMGC_DS05624</td>\n",
       "      <td>colon adenocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>ACH-001021</td>\n",
       "      <td>C3A</td>\n",
       "      <td>BMGC_DS04044</td>\n",
       "      <td>hepatoblastoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIHC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553</th>\n",
       "      <td>ACH-001065</td>\n",
       "      <td>EVSA-T</td>\n",
       "      <td>BMGC_DS07015</td>\n",
       "      <td>breast carcinoma</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>breast</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>554 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       depMapID    Name BMGC_Disease_ID                BMGC_Disease_name  \\\n",
       "0    ACH-000002   HL-60    BMGC_DS01407           acute myeloid leukemia   \n",
       "1    ACH-000004     HEL    BMGC_DS16646                  erythroleukemia   \n",
       "2    ACH-000007   LS513    BMGC_DS08517             cecum adenocarcinoma   \n",
       "3    ACH-000008   A101D    BMGC_DS01559                         melanoma   \n",
       "4    ACH-000009  C2BBe1    BMGC_DS05624             colon adenocarcinoma   \n",
       "..          ...     ...             ...                              ...   \n",
       "549  ACH-000976  HuCCT1    BMGC_DS05988  intrahepatic cholangiocarcinoma   \n",
       "550  ACH-000979  DU 145    BMGC_DS06988               prostate carcinoma   \n",
       "551  ACH-000997  HCT-15    BMGC_DS05624             colon adenocarcinoma   \n",
       "552  ACH-001021     C3A    BMGC_DS04044                   hepatoblastoma   \n",
       "553  ACH-001065  EVSA-T    BMGC_DS07015                 breast carcinoma   \n",
       "\n",
       "      Pathology                Histology        type   PATHOLOGIST_ANNOTATION  \\\n",
       "0       primary  haematopoietic_neoplasm         AML             Leukemia:AML   \n",
       "1       primary  haematopoietic_neoplasm         AML             Leukemia:AML   \n",
       "2       primary                carcinoma  colorectal     Colorectal:Carcinoma   \n",
       "3       primary       malignant_melanoma    melanoma            Skin:Melanoma   \n",
       "4       primary                carcinoma  colorectal     Colorectal:Carcinoma   \n",
       "..          ...                      ...         ...                      ...   \n",
       "549     primary                carcinoma   bile_duct  Biliary_Tract:Carcinoma   \n",
       "550  metastasis                carcinoma    prostate       Prostate:Carcinoma   \n",
       "551     primary                carcinoma  colorectal     Colorectal:Carcinoma   \n",
       "552     primary                carcinoma       liver                      NaN   \n",
       "553  metastasis                carcinoma      breast                      NaN   \n",
       "\n",
       "     tcga_code  \n",
       "0         LAML  \n",
       "1         LAML  \n",
       "2    COAD/READ  \n",
       "3         SKCM  \n",
       "4    COAD/READ  \n",
       "..         ...  \n",
       "549        NaN  \n",
       "550       PRAD  \n",
       "551  COAD/READ  \n",
       "552       LIHC  \n",
       "553       BRCA  \n",
       "\n",
       "[554 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Histology</th>\n",
       "      <th>type</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>tcga_code</th>\n",
       "      <th>id</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>BMGC_Disease_name</th>\n",
       "      <th>...</th>\n",
       "      <th>depmap_match_2_disease</th>\n",
       "      <th>depmap_match_2_disease_name</th>\n",
       "      <th>ncit_match_1_disease</th>\n",
       "      <th>ncit_match_1_disease_name</th>\n",
       "      <th>ncit_match_2_disease</th>\n",
       "      <th>ncit_match_2_disease_name</th>\n",
       "      <th>ordo_match_1_disease</th>\n",
       "      <th>ordo_match_1_disease_name</th>\n",
       "      <th>ordo_match_2_disease</th>\n",
       "      <th>ordo_match_2_disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000038</td>\n",
       "      <td>EHEB</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>lymphoma_other</td>\n",
       "      <td>Leukemia:CLL</td>\n",
       "      <td>CLL</td>\n",
       "      <td>CVCL_1194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20800</td>\n",
       "      <td>Leukemia, Myeloid, Acute</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000070</td>\n",
       "      <td>697</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>leukemia_other</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>ALL</td>\n",
       "      <td>CVCL_0079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20478</td>\n",
       "      <td>Leukemia, Megakaryoblastic, Acute</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS07386</td>\n",
       "      <td>adult acute lymphoblastic leukemia | adult acu...</td>\n",
       "      <td>BMGC_DS13122</td>\n",
       "      <td>B-cell acute lymphoblastic leukemia with t(1;1...</td>\n",
       "      <td>BMGC_DS06807</td>\n",
       "      <td>acute myeloid leukemia with inv(16)(p13.1q22) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000101</td>\n",
       "      <td>KE-37</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>leukemia_other</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>ALL</td>\n",
       "      <td>CVCL_1327</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20478</td>\n",
       "      <td>Leukemia, Megakaryoblastic, Acute</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS05322</td>\n",
       "      <td>T-cell adult acute lymphocytic leukemia</td>\n",
       "      <td>BMGC_DS20794</td>\n",
       "      <td>Precursor B-Cell Lymphoblastic Leukemia-Lymphoma</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000103</td>\n",
       "      <td>Caov-4</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>ovary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OV</td>\n",
       "      <td>CVCL_0202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS03249</td>\n",
       "      <td>uterine adnexa cancer</td>\n",
       "      <td>BMGC_DS08979</td>\n",
       "      <td>ovarian serous surface papillary adenocarcinom...</td>\n",
       "      <td>BMGC_DS09417</td>\n",
       "      <td>peritoneal serous adenocarcinoma | primary per...</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000104</td>\n",
       "      <td>Loucy</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>T-cell_ALL</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>ALL</td>\n",
       "      <td>CVCL_1380</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20800</td>\n",
       "      <td>Leukemia, Myeloid, Acute</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS05322</td>\n",
       "      <td>T-cell adult acute lymphocytic leukemia</td>\n",
       "      <td>BMGC_DS20794</td>\n",
       "      <td>Precursor B-Cell Lymphoblastic Leukemia-Lymphoma</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>ACH-000926</td>\n",
       "      <td>HT55</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>COAD/READ</td>\n",
       "      <td>CVCL_1294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS08620</td>\n",
       "      <td>colon squamous cell carcinoma | squamous cell ...</td>\n",
       "      <td>BMGC_DS05296</td>\n",
       "      <td>duodenal adenocarcinoma | duodenum adenocarcinoma</td>\n",
       "      <td>BMGC_DS05340</td>\n",
       "      <td>rectum mucinous adenocarcinoma</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>ACH-000973</td>\n",
       "      <td>639-V</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>urinary_tract</td>\n",
       "      <td>Bladder:Carcinoma</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>CVCL_1048</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20762</td>\n",
       "      <td>Urinary Tract Infections | urinary tract infec...</td>\n",
       "      <td>BMGC_DS09400</td>\n",
       "      <td>infiltrating renal pelvis/ureter urothelial ca...</td>\n",
       "      <td>BMGC_DS08797</td>\n",
       "      <td>inverted urothelial papilloma | urinary tract ...</td>\n",
       "      <td>BMGC_DS08797</td>\n",
       "      <td>inverted urothelial papilloma | urinary tract ...</td>\n",
       "      <td>BMGC_DS09400</td>\n",
       "      <td>infiltrating renal pelvis/ureter urothelial ca...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>ACH-000995</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>T-cell_ALL</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>ALL</td>\n",
       "      <td>CVCL_0065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20800</td>\n",
       "      <td>Leukemia, Myeloid, Acute</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS07386</td>\n",
       "      <td>adult acute lymphoblastic leukemia | adult acu...</td>\n",
       "      <td>BMGC_DS20794</td>\n",
       "      <td>Precursor B-Cell Lymphoblastic Leukemia-Lymphoma</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>ACH-001064</td>\n",
       "      <td>EB2</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>lymphoma_Burkitt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>ACH-001182</td>\n",
       "      <td>RPMI 6666</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoblastoid</td>\n",
       "      <td>lymphoma_Hodgkin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CVCL_1665</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      depMapID       Name   Pathology          Histology              type  \\\n",
       "0   ACH-000038       EHEB     primary  lymphoid_neoplasm    lymphoma_other   \n",
       "1   ACH-000070        697     primary  lymphoid_neoplasm    leukemia_other   \n",
       "2   ACH-000101      KE-37     primary  lymphoid_neoplasm    leukemia_other   \n",
       "3   ACH-000103     Caov-4  metastasis          carcinoma             ovary   \n",
       "4   ACH-000104      Loucy     primary  lymphoid_neoplasm        T-cell_ALL   \n",
       "..         ...        ...         ...                ...               ...   \n",
       "90  ACH-000926       HT55     primary          carcinoma        colorectal   \n",
       "91  ACH-000973      639-V     primary          carcinoma     urinary_tract   \n",
       "92  ACH-000995     JURKAT     primary  lymphoid_neoplasm        T-cell_ALL   \n",
       "93  ACH-001064        EB2  metastasis  lymphoid_neoplasm  lymphoma_Burkitt   \n",
       "94  ACH-001182  RPMI 6666     primary     lymphoblastoid  lymphoma_Hodgkin   \n",
       "\n",
       "   PATHOLOGIST_ANNOTATION  tcga_code         id  BMGC_Disease_ID  \\\n",
       "0            Leukemia:CLL        CLL  CVCL_1194              NaN   \n",
       "1            Leukemia:ALL        ALL  CVCL_0079              NaN   \n",
       "2            Leukemia:ALL        ALL  CVCL_1327              NaN   \n",
       "3                     NaN         OV  CVCL_0202              NaN   \n",
       "4            Leukemia:ALL        ALL  CVCL_1380              NaN   \n",
       "..                    ...        ...        ...              ...   \n",
       "90   Colorectal:Carcinoma  COAD/READ  CVCL_1294              NaN   \n",
       "91      Bladder:Carcinoma       BLCA  CVCL_1048              NaN   \n",
       "92           Leukemia:ALL        ALL  CVCL_0065              NaN   \n",
       "93                    NaN        NaN        NaN              NaN   \n",
       "94                    NaN        NaN  CVCL_1665              NaN   \n",
       "\n",
       "    BMGC_Disease_name  ... depmap_match_2_disease  \\\n",
       "0                 NaN  ...           BMGC_DS20800   \n",
       "1                 NaN  ...           BMGC_DS20478   \n",
       "2                 NaN  ...           BMGC_DS20478   \n",
       "3                 NaN  ...           BMGC_DS03249   \n",
       "4                 NaN  ...           BMGC_DS20800   \n",
       "..                ...  ...                    ...   \n",
       "90                NaN  ...           BMGC_DS08620   \n",
       "91                NaN  ...           BMGC_DS20762   \n",
       "92                NaN  ...           BMGC_DS20800   \n",
       "93                NaN  ...           BMGC_DS20499   \n",
       "94                NaN  ...           BMGC_DS20499   \n",
       "\n",
       "                          depmap_match_2_disease_name ncit_match_1_disease  \\\n",
       "0                            Leukemia, Myeloid, Acute         BMGC_DS20323   \n",
       "1                   Leukemia, Megakaryoblastic, Acute         BMGC_DS09029   \n",
       "2                   Leukemia, Megakaryoblastic, Acute         BMGC_DS09029   \n",
       "3                               uterine adnexa cancer         BMGC_DS08979   \n",
       "4                            Leukemia, Myeloid, Acute         BMGC_DS09029   \n",
       "..                                                ...                  ...   \n",
       "90  colon squamous cell carcinoma | squamous cell ...         BMGC_DS05296   \n",
       "91  Urinary Tract Infections | urinary tract infec...         BMGC_DS09400   \n",
       "92                           Leukemia, Myeloid, Acute         BMGC_DS09029   \n",
       "93                              Lymphoma, Non-Hodgkin         BMGC_DS20323   \n",
       "94                              Lymphoma, Non-Hodgkin         BMGC_DS20323   \n",
       "\n",
       "                            ncit_match_1_disease_name ncit_match_2_disease  \\\n",
       "0                                               Fever         BMGC_DS22241   \n",
       "1   acute lymphoblastic leukemia | precursor lymph...         BMGC_DS07386   \n",
       "2   acute lymphoblastic leukemia | precursor lymph...         BMGC_DS05322   \n",
       "3   ovarian serous surface papillary adenocarcinom...         BMGC_DS09417   \n",
       "4   acute lymphoblastic leukemia | precursor lymph...         BMGC_DS05322   \n",
       "..                                                ...                  ...   \n",
       "90  duodenal adenocarcinoma | duodenum adenocarcinoma         BMGC_DS05340   \n",
       "91  infiltrating renal pelvis/ureter urothelial ca...         BMGC_DS08797   \n",
       "92  acute lymphoblastic leukemia | precursor lymph...         BMGC_DS07386   \n",
       "93                                              Fever         BMGC_DS22241   \n",
       "94                                              Fever         BMGC_DS22241   \n",
       "\n",
       "                            ncit_match_2_disease_name ordo_match_1_disease  \\\n",
       "0                                                 NaN         BMGC_DS20323   \n",
       "1   adult acute lymphoblastic leukemia | adult acu...         BMGC_DS13122   \n",
       "2             T-cell adult acute lymphocytic leukemia         BMGC_DS20794   \n",
       "3   peritoneal serous adenocarcinoma | primary per...         BMGC_DS20323   \n",
       "4             T-cell adult acute lymphocytic leukemia         BMGC_DS20794   \n",
       "..                                                ...                  ...   \n",
       "90                     rectum mucinous adenocarcinoma         BMGC_DS20323   \n",
       "91  inverted urothelial papilloma | urinary tract ...         BMGC_DS08797   \n",
       "92  adult acute lymphoblastic leukemia | adult acu...         BMGC_DS20794   \n",
       "93                                                NaN         BMGC_DS20323   \n",
       "94                                                NaN         BMGC_DS20323   \n",
       "\n",
       "                            ordo_match_1_disease_name ordo_match_2_disease  \\\n",
       "0                                               Fever         BMGC_DS22241   \n",
       "1   B-cell acute lymphoblastic leukemia with t(1;1...         BMGC_DS06807   \n",
       "2    Precursor B-Cell Lymphoblastic Leukemia-Lymphoma         BMGC_DS09029   \n",
       "3                                               Fever         BMGC_DS22241   \n",
       "4    Precursor B-Cell Lymphoblastic Leukemia-Lymphoma         BMGC_DS09029   \n",
       "..                                                ...                  ...   \n",
       "90                                              Fever         BMGC_DS22241   \n",
       "91  inverted urothelial papilloma | urinary tract ...         BMGC_DS09400   \n",
       "92   Precursor B-Cell Lymphoblastic Leukemia-Lymphoma         BMGC_DS09029   \n",
       "93                                              Fever         BMGC_DS22241   \n",
       "94                                              Fever         BMGC_DS22241   \n",
       "\n",
       "                            ordo_match_2_disease_name  \n",
       "0                                                 NaN  \n",
       "1   acute myeloid leukemia with inv(16)(p13.1q22) ...  \n",
       "2   acute lymphoblastic leukemia | precursor lymph...  \n",
       "3                                                 NaN  \n",
       "4   acute lymphoblastic leukemia | precursor lymph...  \n",
       "..                                                ...  \n",
       "90                                                NaN  \n",
       "91  infiltrating renal pelvis/ureter urothelial ca...  \n",
       "92  acute lymphoblastic leukemia | precursor lymph...  \n",
       "93                                                NaN  \n",
       "94                                                NaN  \n",
       "\n",
       "[95 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Matched samples, rename BMGC_Matched_ID and BMGC_Matched_name to BMGC_ID and BMGC_name\n",
    "dti_matched_df = pd.read_csv('./data/process_data/dti_matched.csv').rename(columns={'BMGC_Matched_ID': 'BMGC_Disease_ID', 'BMGC_Matched_name': 'BMGC_Disease_name'})\n",
    "# keep the columns ['depMapID', 'Name', 'BMGC_Disease_ID', 'BMGC_Disease_name', 'Pathology', 'Histology', 'type', 'PATHOLOGIST_ANNOTATION', 'tcga_code', 'xref_NCIt_ID', 'xref_NCIt_name', 'xref_ORDO_ID', 'xref_ORDO_name']\n",
    "dti_matched_df = dti_matched_df[['depMapID', 'Name', 'BMGC_Disease_ID', 'BMGC_Disease_name', 'Pathology', 'Histology', 'type', 'PATHOLOGIST_ANNOTATION', 'tcga_code']]\n",
    "display(dti_matched_df)\n",
    "# Unmatched samples, rename BMGC_manual_ID and BMGC_manual_name to BMGC_Disease_ID and BMGC_Disease_name\n",
    "dti_unmatched_manual_df = pd.read_csv('./data/process_data/unmatched_samples_manual.csv').rename(columns={'BMGC_manual_ID': 'BMGC_Disease_ID', 'BMGC_manual_name': 'BMGC_Disease_name'})\n",
    "display(dti_unmatched_manual_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 360,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Histology</th>\n",
       "      <th>type</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>tcga_code</th>\n",
       "      <th>id</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>BMGC_Disease_name</th>\n",
       "      <th>...</th>\n",
       "      <th>depmap_match_2_disease</th>\n",
       "      <th>depmap_match_2_disease_name</th>\n",
       "      <th>ncit_match_1_disease</th>\n",
       "      <th>ncit_match_1_disease_name</th>\n",
       "      <th>ncit_match_2_disease</th>\n",
       "      <th>ncit_match_2_disease_name</th>\n",
       "      <th>ordo_match_1_disease</th>\n",
       "      <th>ordo_match_1_disease_name</th>\n",
       "      <th>ordo_match_2_disease</th>\n",
       "      <th>ordo_match_2_disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000038</td>\n",
       "      <td>EHEB</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>lymphoma_other</td>\n",
       "      <td>Leukemia:CLL</td>\n",
       "      <td>CLL</td>\n",
       "      <td>CVCL_1194</td>\n",
       "      <td>BMGC_DS01395</td>\n",
       "      <td>B-cell chronic lymphocytic leukemia | chronic ...</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20800</td>\n",
       "      <td>Leukemia, Myeloid, Acute</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000070</td>\n",
       "      <td>697</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>leukemia_other</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>ALL</td>\n",
       "      <td>CVCL_0079</td>\n",
       "      <td>BMGC_DS13122</td>\n",
       "      <td>B-cell acute lymphoblastic leukemia with t(1;1...</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20478</td>\n",
       "      <td>Leukemia, Megakaryoblastic, Acute</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS07386</td>\n",
       "      <td>adult acute lymphoblastic leukemia | adult acu...</td>\n",
       "      <td>BMGC_DS13122</td>\n",
       "      <td>B-cell acute lymphoblastic leukemia with t(1;1...</td>\n",
       "      <td>BMGC_DS06807</td>\n",
       "      <td>acute myeloid leukemia with inv(16)(p13.1q22) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000101</td>\n",
       "      <td>KE-37</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>leukemia_other</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>ALL</td>\n",
       "      <td>CVCL_1327</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20478</td>\n",
       "      <td>Leukemia, Megakaryoblastic, Acute</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS05322</td>\n",
       "      <td>T-cell adult acute lymphocytic leukemia</td>\n",
       "      <td>BMGC_DS20794</td>\n",
       "      <td>Precursor B-Cell Lymphoblastic Leukemia-Lymphoma</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000103</td>\n",
       "      <td>Caov-4</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>ovary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OV</td>\n",
       "      <td>CVCL_0202</td>\n",
       "      <td>BMGC_DS07934</td>\n",
       "      <td>ovarian adenocarcinoma | ovary adenocarcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS03249</td>\n",
       "      <td>uterine adnexa cancer</td>\n",
       "      <td>BMGC_DS08979</td>\n",
       "      <td>ovarian serous surface papillary adenocarcinom...</td>\n",
       "      <td>BMGC_DS09417</td>\n",
       "      <td>peritoneal serous adenocarcinoma | primary per...</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000104</td>\n",
       "      <td>Loucy</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>T-cell_ALL</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>ALL</td>\n",
       "      <td>CVCL_1380</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20800</td>\n",
       "      <td>Leukemia, Myeloid, Acute</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS05322</td>\n",
       "      <td>T-cell adult acute lymphocytic leukemia</td>\n",
       "      <td>BMGC_DS20794</td>\n",
       "      <td>Precursor B-Cell Lymphoblastic Leukemia-Lymphoma</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>ACH-000926</td>\n",
       "      <td>HT55</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>COAD/READ</td>\n",
       "      <td>CVCL_1294</td>\n",
       "      <td>BMGC_DS08359</td>\n",
       "      <td>colorectal adenocarcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS08620</td>\n",
       "      <td>colon squamous cell carcinoma | squamous cell ...</td>\n",
       "      <td>BMGC_DS05296</td>\n",
       "      <td>duodenal adenocarcinoma | duodenum adenocarcinoma</td>\n",
       "      <td>BMGC_DS05340</td>\n",
       "      <td>rectum mucinous adenocarcinoma</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>ACH-000973</td>\n",
       "      <td>639-V</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>urinary_tract</td>\n",
       "      <td>Bladder:Carcinoma</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>CVCL_1048</td>\n",
       "      <td>BMGC_DS07053</td>\n",
       "      <td>bladder carcinoma | urinary bladder carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20762</td>\n",
       "      <td>Urinary Tract Infections | urinary tract infec...</td>\n",
       "      <td>BMGC_DS09400</td>\n",
       "      <td>infiltrating renal pelvis/ureter urothelial ca...</td>\n",
       "      <td>BMGC_DS08797</td>\n",
       "      <td>inverted urothelial papilloma | urinary tract ...</td>\n",
       "      <td>BMGC_DS08797</td>\n",
       "      <td>inverted urothelial papilloma | urinary tract ...</td>\n",
       "      <td>BMGC_DS09400</td>\n",
       "      <td>infiltrating renal pelvis/ureter urothelial ca...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>ACH-000995</td>\n",
       "      <td>JURKAT</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>T-cell_ALL</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>ALL</td>\n",
       "      <td>CVCL_0065</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20800</td>\n",
       "      <td>Leukemia, Myeloid, Acute</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS07386</td>\n",
       "      <td>adult acute lymphoblastic leukemia | adult acu...</td>\n",
       "      <td>BMGC_DS20794</td>\n",
       "      <td>Precursor B-Cell Lymphoblastic Leukemia-Lymphoma</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>ACH-001182</td>\n",
       "      <td>RPMI 6666</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoblastoid</td>\n",
       "      <td>lymphoma_Hodgkin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CVCL_1665</td>\n",
       "      <td>BMGC_DS20405</td>\n",
       "      <td>Hodgkin Disease | Hodgkins lymphoma</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>ACH-001064</td>\n",
       "      <td>EB2</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>lymphoma_Burkitt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_DS20177</td>\n",
       "      <td>Burkitt Lymphoma</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      depMapID       Name   Pathology          Histology              type  \\\n",
       "0   ACH-000038       EHEB     primary  lymphoid_neoplasm    lymphoma_other   \n",
       "1   ACH-000070        697     primary  lymphoid_neoplasm    leukemia_other   \n",
       "2   ACH-000101      KE-37     primary  lymphoid_neoplasm    leukemia_other   \n",
       "3   ACH-000103     Caov-4  metastasis          carcinoma             ovary   \n",
       "4   ACH-000104      Loucy     primary  lymphoid_neoplasm        T-cell_ALL   \n",
       "..         ...        ...         ...                ...               ...   \n",
       "90  ACH-000926       HT55     primary          carcinoma        colorectal   \n",
       "91  ACH-000973      639-V     primary          carcinoma     urinary_tract   \n",
       "92  ACH-000995     JURKAT     primary  lymphoid_neoplasm        T-cell_ALL   \n",
       "93  ACH-001182  RPMI 6666     primary     lymphoblastoid  lymphoma_Hodgkin   \n",
       "94  ACH-001064        EB2  metastasis  lymphoid_neoplasm  lymphoma_Burkitt   \n",
       "\n",
       "   PATHOLOGIST_ANNOTATION  tcga_code         id BMGC_Disease_ID  \\\n",
       "0            Leukemia:CLL        CLL  CVCL_1194    BMGC_DS01395   \n",
       "1            Leukemia:ALL        ALL  CVCL_0079    BMGC_DS13122   \n",
       "2            Leukemia:ALL        ALL  CVCL_1327    BMGC_DS09029   \n",
       "3                     NaN         OV  CVCL_0202    BMGC_DS07934   \n",
       "4            Leukemia:ALL        ALL  CVCL_1380    BMGC_DS09029   \n",
       "..                    ...        ...        ...             ...   \n",
       "90   Colorectal:Carcinoma  COAD/READ  CVCL_1294    BMGC_DS08359   \n",
       "91      Bladder:Carcinoma       BLCA  CVCL_1048    BMGC_DS07053   \n",
       "92           Leukemia:ALL        ALL  CVCL_0065    BMGC_DS09029   \n",
       "93                    NaN        NaN  CVCL_1665    BMGC_DS20405   \n",
       "94                    NaN        NaN        NaN    BMGC_DS20177   \n",
       "\n",
       "                                    BMGC_Disease_name  ...  \\\n",
       "0   B-cell chronic lymphocytic leukemia | chronic ...  ...   \n",
       "1   B-cell acute lymphoblastic leukemia with t(1;1...  ...   \n",
       "2   acute lymphoblastic leukemia | precursor lymph...  ...   \n",
       "3       ovarian adenocarcinoma | ovary adenocarcinoma  ...   \n",
       "4   acute lymphoblastic leukemia | precursor lymph...  ...   \n",
       "..                                                ...  ...   \n",
       "90                          colorectal adenocarcinoma  ...   \n",
       "91      bladder carcinoma | urinary bladder carcinoma  ...   \n",
       "92  acute lymphoblastic leukemia | precursor lymph...  ...   \n",
       "93                Hodgkin Disease | Hodgkins lymphoma  ...   \n",
       "94                                   Burkitt Lymphoma  ...   \n",
       "\n",
       "   depmap_match_2_disease                        depmap_match_2_disease_name  \\\n",
       "0            BMGC_DS20800                           Leukemia, Myeloid, Acute   \n",
       "1            BMGC_DS20478                  Leukemia, Megakaryoblastic, Acute   \n",
       "2            BMGC_DS20478                  Leukemia, Megakaryoblastic, Acute   \n",
       "3            BMGC_DS03249                              uterine adnexa cancer   \n",
       "4            BMGC_DS20800                           Leukemia, Myeloid, Acute   \n",
       "..                    ...                                                ...   \n",
       "90           BMGC_DS08620  colon squamous cell carcinoma | squamous cell ...   \n",
       "91           BMGC_DS20762  Urinary Tract Infections | urinary tract infec...   \n",
       "92           BMGC_DS20800                           Leukemia, Myeloid, Acute   \n",
       "93           BMGC_DS20499                              Lymphoma, Non-Hodgkin   \n",
       "94           BMGC_DS20499                              Lymphoma, Non-Hodgkin   \n",
       "\n",
       "   ncit_match_1_disease                          ncit_match_1_disease_name  \\\n",
       "0          BMGC_DS20323                                              Fever   \n",
       "1          BMGC_DS09029  acute lymphoblastic leukemia | precursor lymph...   \n",
       "2          BMGC_DS09029  acute lymphoblastic leukemia | precursor lymph...   \n",
       "3          BMGC_DS08979  ovarian serous surface papillary adenocarcinom...   \n",
       "4          BMGC_DS09029  acute lymphoblastic leukemia | precursor lymph...   \n",
       "..                  ...                                                ...   \n",
       "90         BMGC_DS05296  duodenal adenocarcinoma | duodenum adenocarcinoma   \n",
       "91         BMGC_DS09400  infiltrating renal pelvis/ureter urothelial ca...   \n",
       "92         BMGC_DS09029  acute lymphoblastic leukemia | precursor lymph...   \n",
       "93         BMGC_DS20323                                              Fever   \n",
       "94         BMGC_DS20323                                              Fever   \n",
       "\n",
       "   ncit_match_2_disease                          ncit_match_2_disease_name  \\\n",
       "0          BMGC_DS22241                                                NaN   \n",
       "1          BMGC_DS07386  adult acute lymphoblastic leukemia | adult acu...   \n",
       "2          BMGC_DS05322            T-cell adult acute lymphocytic leukemia   \n",
       "3          BMGC_DS09417  peritoneal serous adenocarcinoma | primary per...   \n",
       "4          BMGC_DS05322            T-cell adult acute lymphocytic leukemia   \n",
       "..                  ...                                                ...   \n",
       "90         BMGC_DS05340                     rectum mucinous adenocarcinoma   \n",
       "91         BMGC_DS08797  inverted urothelial papilloma | urinary tract ...   \n",
       "92         BMGC_DS07386  adult acute lymphoblastic leukemia | adult acu...   \n",
       "93         BMGC_DS22241                                                NaN   \n",
       "94         BMGC_DS22241                                                NaN   \n",
       "\n",
       "   ordo_match_1_disease                          ordo_match_1_disease_name  \\\n",
       "0          BMGC_DS20323                                              Fever   \n",
       "1          BMGC_DS13122  B-cell acute lymphoblastic leukemia with t(1;1...   \n",
       "2          BMGC_DS20794   Precursor B-Cell Lymphoblastic Leukemia-Lymphoma   \n",
       "3          BMGC_DS20323                                              Fever   \n",
       "4          BMGC_DS20794   Precursor B-Cell Lymphoblastic Leukemia-Lymphoma   \n",
       "..                  ...                                                ...   \n",
       "90         BMGC_DS20323                                              Fever   \n",
       "91         BMGC_DS08797  inverted urothelial papilloma | urinary tract ...   \n",
       "92         BMGC_DS20794   Precursor B-Cell Lymphoblastic Leukemia-Lymphoma   \n",
       "93         BMGC_DS20323                                              Fever   \n",
       "94         BMGC_DS20323                                              Fever   \n",
       "\n",
       "   ordo_match_2_disease                          ordo_match_2_disease_name  \n",
       "0          BMGC_DS22241                                                NaN  \n",
       "1          BMGC_DS06807  acute myeloid leukemia with inv(16)(p13.1q22) ...  \n",
       "2          BMGC_DS09029  acute lymphoblastic leukemia | precursor lymph...  \n",
       "3          BMGC_DS22241                                                NaN  \n",
       "4          BMGC_DS09029  acute lymphoblastic leukemia | precursor lymph...  \n",
       "..                  ...                                                ...  \n",
       "90         BMGC_DS22241                                                NaN  \n",
       "91         BMGC_DS09400  infiltrating renal pelvis/ureter urothelial ca...  \n",
       "92         BMGC_DS09029  acute lymphoblastic leukemia | precursor lymph...  \n",
       "93         BMGC_DS22241                                                NaN  \n",
       "94         BMGC_DS22241                                                NaN  \n",
       "\n",
       "[95 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BMGC_Disease_ID      0\n",
      "BMGC_Disease_name    0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>BMGC_Disease_name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Histology</th>\n",
       "      <th>type</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>tcga_code</th>\n",
       "      <th>id</th>\n",
       "      <th>...</th>\n",
       "      <th>depmap_match_2_disease</th>\n",
       "      <th>depmap_match_2_disease_name</th>\n",
       "      <th>ncit_match_1_disease</th>\n",
       "      <th>ncit_match_1_disease_name</th>\n",
       "      <th>ncit_match_2_disease</th>\n",
       "      <th>ncit_match_2_disease_name</th>\n",
       "      <th>ordo_match_1_disease</th>\n",
       "      <th>ordo_match_1_disease_name</th>\n",
       "      <th>ordo_match_2_disease</th>\n",
       "      <th>ordo_match_2_disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>NIH:OVCAR-3</td>\n",
       "      <td>BMGC_DS07934</td>\n",
       "      <td>ovarian adenocarcinoma | ovary adenocarcinoma</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>ovary</td>\n",
       "      <td>Ovary:Carcinoma</td>\n",
       "      <td>OV</td>\n",
       "      <td>CVCL_0465</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS08982</td>\n",
       "      <td>ovarian transitional cell carcinoma | ovary tr...</td>\n",
       "      <td>BMGC_DS08979</td>\n",
       "      <td>ovarian serous surface papillary adenocarcinom...</td>\n",
       "      <td>BMGC_DS09417</td>\n",
       "      <td>peritoneal serous adenocarcinoma | primary per...</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000002</td>\n",
       "      <td>HL-60</td>\n",
       "      <td>BMGC_DS01407</td>\n",
       "      <td>acute myeloid leukemia</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>LAML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>HEL</td>\n",
       "      <td>BMGC_DS16646</td>\n",
       "      <td>erythroleukemia</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>LAML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000006</td>\n",
       "      <td>MONO-MAC-6</td>\n",
       "      <td>BMGC_DS20800</td>\n",
       "      <td>leukemia, myeloid, acute</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>LAML</td>\n",
       "      <td>CVCL_1426</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20480</td>\n",
       "      <td>Leukemia, Myeloid</td>\n",
       "      <td>BMGC_DS05322</td>\n",
       "      <td>T-cell adult acute lymphocytic leukemia</td>\n",
       "      <td>BMGC_DS07386</td>\n",
       "      <td>adult acute lymphoblastic leukemia | adult acu...</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS07386</td>\n",
       "      <td>adult acute lymphoblastic leukemia | adult acu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000007</td>\n",
       "      <td>LS513</td>\n",
       "      <td>BMGC_DS08517</td>\n",
       "      <td>cecum adenocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>COAD/READ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>ACH-001199</td>\n",
       "      <td>SNU-C2B</td>\n",
       "      <td>BMGC_DS08517</td>\n",
       "      <td>cecum adenocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COAD/READ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>645</th>\n",
       "      <td>ACH-001203</td>\n",
       "      <td>SUP-HD1</td>\n",
       "      <td>BMGC_DS20405</td>\n",
       "      <td>hodgkin disease | hodgkins lymphoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>lymphoma_Hodgkin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CVCL_2208</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "      <td>BMGC_DS20405</td>\n",
       "      <td>Hodgkin Disease | Hodgkins lymphoma</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "      <td>BMGC_DS20405</td>\n",
       "      <td>Hodgkin Disease | Hodgkins lymphoma</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>646</th>\n",
       "      <td>ACH-001233</td>\n",
       "      <td>VMRC-LCD</td>\n",
       "      <td>BMGC_DS02944</td>\n",
       "      <td>lung adenocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>ACH-001306</td>\n",
       "      <td>8305C</td>\n",
       "      <td>BMGC_DS21147</td>\n",
       "      <td>thyroid carcinoma, anaplastic</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>Thyroid:Carcinoma</td>\n",
       "      <td>THCA</td>\n",
       "      <td>CVCL_1053</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS21147</td>\n",
       "      <td>Thyroid Carcinoma, Anaplastic</td>\n",
       "      <td>BMGC_DS04082</td>\n",
       "      <td>follicular thyroid carcinoma | thyroid gland f...</td>\n",
       "      <td>BMGC_DS03290</td>\n",
       "      <td>malignant tumor of parathyroid gland | parathy...</td>\n",
       "      <td>BMGC_DS03290</td>\n",
       "      <td>malignant tumor of parathyroid gland | parathy...</td>\n",
       "      <td>BMGC_DS04082</td>\n",
       "      <td>follicular thyroid carcinoma | thyroid gland f...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>ACH-001307</td>\n",
       "      <td>8505C</td>\n",
       "      <td>BMGC_DS21147</td>\n",
       "      <td>thyroid carcinoma, anaplastic</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>Thyroid:Carcinoma</td>\n",
       "      <td>THCA</td>\n",
       "      <td>CVCL_1054</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS21147</td>\n",
       "      <td>Thyroid Carcinoma, Anaplastic</td>\n",
       "      <td>BMGC_DS04082</td>\n",
       "      <td>follicular thyroid carcinoma | thyroid gland f...</td>\n",
       "      <td>BMGC_DS03290</td>\n",
       "      <td>malignant tumor of parathyroid gland | parathy...</td>\n",
       "      <td>BMGC_DS03290</td>\n",
       "      <td>malignant tumor of parathyroid gland | parathy...</td>\n",
       "      <td>BMGC_DS04082</td>\n",
       "      <td>follicular thyroid carcinoma | thyroid gland f...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>649 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       depMapID         Name BMGC_Disease_ID  \\\n",
       "0    ACH-000001  NIH:OVCAR-3    BMGC_DS07934   \n",
       "1    ACH-000002        HL-60    BMGC_DS01407   \n",
       "2    ACH-000004          HEL    BMGC_DS16646   \n",
       "3    ACH-000006   MONO-MAC-6    BMGC_DS20800   \n",
       "4    ACH-000007        LS513    BMGC_DS08517   \n",
       "..          ...          ...             ...   \n",
       "644  ACH-001199      SNU-C2B    BMGC_DS08517   \n",
       "645  ACH-001203      SUP-HD1    BMGC_DS20405   \n",
       "646  ACH-001233     VMRC-LCD    BMGC_DS02944   \n",
       "647  ACH-001306        8305C    BMGC_DS21147   \n",
       "648  ACH-001307        8505C    BMGC_DS21147   \n",
       "\n",
       "                                 BMGC_Disease_name   Pathology  \\\n",
       "0    ovarian adenocarcinoma | ovary adenocarcinoma  metastasis   \n",
       "1                           acute myeloid leukemia     primary   \n",
       "2                                  erythroleukemia     primary   \n",
       "3                         leukemia, myeloid, acute     primary   \n",
       "4                             cecum adenocarcinoma     primary   \n",
       "..                                             ...         ...   \n",
       "644                           cecum adenocarcinoma     primary   \n",
       "645            hodgkin disease | hodgkins lymphoma     primary   \n",
       "646                            lung adenocarcinoma     primary   \n",
       "647                  thyroid carcinoma, anaplastic     primary   \n",
       "648                  thyroid carcinoma, anaplastic     primary   \n",
       "\n",
       "                   Histology              type PATHOLOGIST_ANNOTATION  \\\n",
       "0                  carcinoma             ovary        Ovary:Carcinoma   \n",
       "1    haematopoietic_neoplasm               AML           Leukemia:AML   \n",
       "2    haematopoietic_neoplasm               AML           Leukemia:AML   \n",
       "3    haematopoietic_neoplasm               AML           Leukemia:AML   \n",
       "4                  carcinoma        colorectal   Colorectal:Carcinoma   \n",
       "..                       ...               ...                    ...   \n",
       "644                carcinoma        colorectal                    NaN   \n",
       "645        lymphoid_neoplasm  lymphoma_Hodgkin                    NaN   \n",
       "646                carcinoma          lung_NSC                    NaN   \n",
       "647                carcinoma           thyroid      Thyroid:Carcinoma   \n",
       "648                carcinoma           thyroid      Thyroid:Carcinoma   \n",
       "\n",
       "     tcga_code         id  ... depmap_match_2_disease  \\\n",
       "0           OV  CVCL_0465  ...           BMGC_DS08982   \n",
       "1         LAML        NaN  ...                    NaN   \n",
       "2         LAML        NaN  ...                    NaN   \n",
       "3         LAML  CVCL_1426  ...           BMGC_DS20480   \n",
       "4    COAD/READ        NaN  ...                    NaN   \n",
       "..         ...        ...  ...                    ...   \n",
       "644  COAD/READ        NaN  ...                    NaN   \n",
       "645        NaN  CVCL_2208  ...           BMGC_DS20499   \n",
       "646       LUAD        NaN  ...                    NaN   \n",
       "647       THCA  CVCL_1053  ...           BMGC_DS21147   \n",
       "648       THCA  CVCL_1054  ...           BMGC_DS21147   \n",
       "\n",
       "                           depmap_match_2_disease_name ncit_match_1_disease  \\\n",
       "0    ovarian transitional cell carcinoma | ovary tr...         BMGC_DS08979   \n",
       "1                                                  NaN                  NaN   \n",
       "2                                                  NaN                  NaN   \n",
       "3                                    Leukemia, Myeloid         BMGC_DS05322   \n",
       "4                                                  NaN                  NaN   \n",
       "..                                                 ...                  ...   \n",
       "644                                                NaN                  NaN   \n",
       "645                              Lymphoma, Non-Hodgkin         BMGC_DS20405   \n",
       "646                                                NaN                  NaN   \n",
       "647                      Thyroid Carcinoma, Anaplastic         BMGC_DS04082   \n",
       "648                      Thyroid Carcinoma, Anaplastic         BMGC_DS04082   \n",
       "\n",
       "                             ncit_match_1_disease_name ncit_match_2_disease  \\\n",
       "0    ovarian serous surface papillary adenocarcinom...         BMGC_DS09417   \n",
       "1                                                  NaN                  NaN   \n",
       "2                                                  NaN                  NaN   \n",
       "3              T-cell adult acute lymphocytic leukemia         BMGC_DS07386   \n",
       "4                                                  NaN                  NaN   \n",
       "..                                                 ...                  ...   \n",
       "644                                                NaN                  NaN   \n",
       "645                Hodgkin Disease | Hodgkins lymphoma         BMGC_DS20499   \n",
       "646                                                NaN                  NaN   \n",
       "647  follicular thyroid carcinoma | thyroid gland f...         BMGC_DS03290   \n",
       "648  follicular thyroid carcinoma | thyroid gland f...         BMGC_DS03290   \n",
       "\n",
       "                             ncit_match_2_disease_name ordo_match_1_disease  \\\n",
       "0    peritoneal serous adenocarcinoma | primary per...         BMGC_DS20323   \n",
       "1                                                  NaN                  NaN   \n",
       "2                                                  NaN                  NaN   \n",
       "3    adult acute lymphoblastic leukemia | adult acu...         BMGC_DS09029   \n",
       "4                                                  NaN                  NaN   \n",
       "..                                                 ...                  ...   \n",
       "644                                                NaN                  NaN   \n",
       "645                              Lymphoma, Non-Hodgkin         BMGC_DS20405   \n",
       "646                                                NaN                  NaN   \n",
       "647  malignant tumor of parathyroid gland | parathy...         BMGC_DS03290   \n",
       "648  malignant tumor of parathyroid gland | parathy...         BMGC_DS03290   \n",
       "\n",
       "                             ordo_match_1_disease_name ordo_match_2_disease  \\\n",
       "0                                                Fever         BMGC_DS22241   \n",
       "1                                                  NaN                  NaN   \n",
       "2                                                  NaN                  NaN   \n",
       "3    acute lymphoblastic leukemia | precursor lymph...         BMGC_DS07386   \n",
       "4                                                  NaN                  NaN   \n",
       "..                                                 ...                  ...   \n",
       "644                                                NaN                  NaN   \n",
       "645                Hodgkin Disease | Hodgkins lymphoma         BMGC_DS20499   \n",
       "646                                                NaN                  NaN   \n",
       "647  malignant tumor of parathyroid gland | parathy...         BMGC_DS04082   \n",
       "648  malignant tumor of parathyroid gland | parathy...         BMGC_DS04082   \n",
       "\n",
       "                             ordo_match_2_disease_name  \n",
       "0                                                  NaN  \n",
       "1                                                  NaN  \n",
       "2                                                  NaN  \n",
       "3    adult acute lymphoblastic leukemia | adult acu...  \n",
       "4                                                  NaN  \n",
       "..                                                 ...  \n",
       "644                                                NaN  \n",
       "645                              Lymphoma, Non-Hodgkin  \n",
       "646                                                NaN  \n",
       "647  follicular thyroid carcinoma | thyroid gland f...  \n",
       "648  follicular thyroid carcinoma | thyroid gland f...  \n",
       "\n",
       "[649 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dti_unmatched_manual_valid_df = pd.read_csv('./data/manual_match_data/unmatched_samples_manual_valid.csv').rename(columns={'BMGC_manual_ID': 'BMGC_Disease_ID', 'BMGC_manual_name': 'BMGC_Disease_name'})\n",
    "display(dti_unmatched_manual_valid_df)\n",
    "# concatenate the dti_matched_df and dti_unmatched_manual_df and sort by depMapID\n",
    "dti_combined_df = pd.concat([dti_matched_df, dti_unmatched_manual_valid_df], ignore_index=True)\n",
    "# sort the dti_combined_df by depMapID\n",
    "dti_combined_df = dti_combined_df.sort_values(by='depMapID').reset_index(drop=True)\n",
    "# check if there is any empty value in the BMGC_ID and BMGC_name columns\n",
    "print(dti_combined_df[['BMGC_Disease_ID', 'BMGC_Disease_name']].isnull().sum())\n",
    "# convert all content in 'BMGC_Disease_name' to lower case\n",
    "dti_combined_df['BMGC_Disease_name'] = dti_combined_df['BMGC_Disease_name'].str.lower()\n",
    "display(dti_combined_df)\n",
    "dti_combined_df.to_csv('./data/process_data/dti_combined_samples.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 8. CRISPR/RNAi Biomarkers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.makedirs('./data/pretrain_plain_data', exist_ok=True)\n",
    "os.makedirs('./data/pretrain_status_data', exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Null values in bmgc_omics_name_id_df:\n",
      "BioMedGraphica_Conn_ID    0\n",
      "Names_and_IDs             0\n",
      "dtype: int64\n",
      "Null values in bmgc_omics_name_id_df:\n",
      "BioMedGraphica_Conn_ID    0\n",
      "Names_and_IDs             0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Names_and_IDs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>NCBI Gene ID:1 | HGNC:5 | OMIM ID:138670 | Ref...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>N-acetyltransferase 2 Promoter | RefSeq ID:NG_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>RefSeq ID:NC_000020.11 | OMIM ID:608958 | NCBI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>NCBI Gene ID:1000 | HGNC:1759 | RefSeq ID:NC_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>RefSeq ID:NG_029764.2 | RefSeq ID:NT_187519.1 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705929</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>ENSP00000520928.1 | ENSP00000520928 | NCBI Gen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705930</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>RefSeq ID:NP_001366093 | NCBI Gene ID:1438.0 |...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705931</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>NCBI Gene ID:1438.0 | CSF2RA | RefSeq ID:NP_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705932</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>GTPBP6 | ENSP00000520931.1 | NCBI Gene ID:8225...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705933</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>NCBI Gene ID:3581.0 | RefSeq ID:NP_789743 | IL...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>705934 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  \\\n",
       "0                BMGC_PM00001   \n",
       "1                BMGC_PM00002   \n",
       "2                BMGC_PM00003   \n",
       "3                BMGC_PM00004   \n",
       "4                BMGC_PM00005   \n",
       "...                       ...   \n",
       "705929          BMGC_PT121415   \n",
       "705930          BMGC_PT121416   \n",
       "705931          BMGC_PT121417   \n",
       "705932          BMGC_PT121418   \n",
       "705933          BMGC_PT121419   \n",
       "\n",
       "                                            Names_and_IDs  \n",
       "0       NCBI Gene ID:1 | HGNC:5 | OMIM ID:138670 | Ref...  \n",
       "1       N-acetyltransferase 2 Promoter | RefSeq ID:NG_...  \n",
       "2       RefSeq ID:NC_000020.11 | OMIM ID:608958 | NCBI...  \n",
       "3       NCBI Gene ID:1000 | HGNC:1759 | RefSeq ID:NC_0...  \n",
       "4       RefSeq ID:NG_029764.2 | RefSeq ID:NT_187519.1 ...  \n",
       "...                                                   ...  \n",
       "705929  ENSP00000520928.1 | ENSP00000520928 | NCBI Gen...  \n",
       "705930  RefSeq ID:NP_001366093 | NCBI Gene ID:1438.0 |...  \n",
       "705931  NCBI Gene ID:1438.0 | CSF2RA | RefSeq ID:NP_00...  \n",
       "705932  GTPBP6 | ENSP00000520931.1 | NCBI Gene ID:8225...  \n",
       "705933  NCBI Gene ID:3581.0 | RefSeq ID:NP_789743 | IL...  \n",
       "\n",
       "[705934 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Create the name and desc files\n",
    "bmgc_promoter_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Promoter/BioMedGraphica_Conn_Promoter_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_gene_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_transcript_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Transcript/BioMedGraphica_Conn_Transcript_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_protein_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "\n",
    "bmgc_omics_name_id_tmp_df = pd.concat([bmgc_promoter_name_id_df, bmgc_gene_name_id_df, bmgc_transcript_name_id_df, bmgc_protein_name_id_df], axis=0).reset_index(drop=True)\n",
    "bmgc_omics_name_id_df = pd.merge(bmgc_omics_df[['BioMedGraphica_Conn_ID']], bmgc_omics_name_id_tmp_df, on='BioMedGraphica_Conn_ID', how='left')\n",
    "# check the null values in the bmgc_omics_name_id_df\n",
    "print(\"Null values in bmgc_omics_name_id_df:\")\n",
    "print(bmgc_omics_name_id_df.isnull().sum())\n",
    "# fill the NaN values in the Names_and_IDs column with empty string\n",
    "bmgc_omics_name_id_df['Names_and_IDs'] = bmgc_omics_name_id_df['Names_and_IDs'].fillna(' ')\n",
    "# recheck the null values in the bmgc_omics_name_id_dff\n",
    "print(\"Null values in bmgc_omics_name_id_df:\")\n",
    "print(bmgc_omics_name_id_df.isnull().sum())\n",
    "display(bmgc_omics_name_id_df)\n",
    "bmgc_omics_name_id_df.to_csv('./data/pretrain_plain_data/bmgc_omics_name.csv', index=False)\n",
    "bmgc_omics_name_id_df.to_csv('./data/pretrain_status_data/bmgc_omics_name.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BioMedGraphica_Conn_ID         0\n",
      "Description               193184\n",
      "dtype: int64\n",
      "BioMedGraphica_Conn_ID         0\n",
      "Description               193184\n",
      "dtype: int64\n",
      "BioMedGraphica_Conn_ID    0\n",
      "Description               0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705929</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705930</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705931</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705932</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705933</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>705934 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID Description\n",
       "0                BMGC_PM00001            \n",
       "1                BMGC_PM00002            \n",
       "2                BMGC_PM00003            \n",
       "3                BMGC_PM00004            \n",
       "4                BMGC_PM00005            \n",
       "...                       ...         ...\n",
       "705929          BMGC_PT121415            \n",
       "705930          BMGC_PT121416            \n",
       "705931          BMGC_PT121417            \n",
       "705932          BMGC_PT121418            \n",
       "705933          BMGC_PT121419            \n",
       "\n",
       "[705934 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bmgc_promoter_desc_df = bmgc_promoter_name_id_df.drop(columns=['Names_and_IDs'], axis=1).copy()\n",
    "bmgc_promoter_desc_df['Description'] = np.nan # add the Description column to bmgc_promoter_name_df with NaN values\n",
    "bmgc_gene_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'], axis=1)\n",
    "bmgc_transcript_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Transcript/BioMedGraphica_Conn_Transcript_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'], axis=1)\n",
    "bmgc_protein_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'], axis=1)\n",
    "\n",
    "# concat and drop BioMedGraphica_ID column\n",
    "bmgc_omics_desc_tmp_df = pd.concat([bmgc_promoter_desc_df, bmgc_gene_desc_df, bmgc_transcript_desc_df, bmgc_protein_desc_df], axis=0).reset_index(drop=True)\n",
    "# check the null values in the bmgc_omics_desc_tmp_df\n",
    "print(bmgc_omics_desc_tmp_df.isnull().sum())\n",
    "bmgc_omics_desc_df = pd.merge(bmgc_omics_df[['BioMedGraphica_Conn_ID']], bmgc_omics_desc_tmp_df, on='BioMedGraphica_Conn_ID', how='left')\n",
    "# check the null values in the bmgc_omics_desc_df\n",
    "print(bmgc_omics_desc_df.isnull().sum())\n",
    "# fill NaN values with empty string in Description column\n",
    "bmgc_omics_desc_df['Description'] = bmgc_omics_desc_df['Description'].fillna(' ')\n",
    "# recheck the null values in the bmgc_omics_desc_df\n",
    "print(bmgc_omics_desc_df.isnull().sum())\n",
    "display(bmgc_omics_desc_df)\n",
    "\n",
    "bmgc_omics_desc_df.to_csv('./data/pretrain_plain_data/bmgc_omics_desc.csv', index=False)\n",
    "bmgc_omics_desc_df.to_csv('./data/pretrain_status_data/bmgc_omics_desc.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 8.0 Entity markers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>BioMedGraphica_ID</th>\n",
       "      <th>BMGC_From_ID</th>\n",
       "      <th>BMGC_To_ID</th>\n",
       "      <th>BMG_From_ID</th>\n",
       "      <th>BMG_To_ID</th>\n",
       "      <th>Source</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_ED_PMGN00001</td>\n",
       "      <td>BMG_ED_PMGN00001</td>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMG_PM000001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_ED_PMGN00002</td>\n",
       "      <td>BMG_ED_PMGN00002</td>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMG_PM000002</td>\n",
       "      <td>BMG_GN000002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_ED_PMGN00003</td>\n",
       "      <td>BMG_ED_PMGN00003</td>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>BMG_PM000003</td>\n",
       "      <td>BMG_GN000003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_ED_PMGN00004</td>\n",
       "      <td>BMG_ED_PMGN00004</td>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>BMG_PM000004</td>\n",
       "      <td>BMG_GN000004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_ED_PMGN00005</td>\n",
       "      <td>BMG_ED_PMGN00005</td>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>BMG_PM000005</td>\n",
       "      <td>BMG_GN000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Promoter-Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151448</th>\n",
       "      <td>BMGC_ED_PTPT16484816</td>\n",
       "      <td>BMG_ED_PTPT16484816</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT117260</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT169819</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151449</th>\n",
       "      <td>BMGC_ED_PTPT16484817</td>\n",
       "      <td>BMG_ED_PTPT16484817</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120142</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT172701</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151450</th>\n",
       "      <td>BMGC_ED_PTPT16484818</td>\n",
       "      <td>BMG_ED_PTPT16484818</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120825</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT173384</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151451</th>\n",
       "      <td>BMGC_ED_PTPT16484819</td>\n",
       "      <td>BMG_ED_PTPT16484819</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120826</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT173385</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17151452</th>\n",
       "      <td>BMGC_ED_PTPT16484820</td>\n",
       "      <td>BMG_ED_PTPT16484820</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMGC_PT120827</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>BMG_PT173386</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Protein-Protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17151453 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         BioMedGraphica_Conn_ID    BioMedGraphica_ID   BMGC_From_ID  \\\n",
       "0             BMGC_ED_PMGN00001     BMG_ED_PMGN00001   BMGC_PM00001   \n",
       "1             BMGC_ED_PMGN00002     BMG_ED_PMGN00002   BMGC_PM00002   \n",
       "2             BMGC_ED_PMGN00003     BMG_ED_PMGN00003   BMGC_PM00003   \n",
       "3             BMGC_ED_PMGN00004     BMG_ED_PMGN00004   BMGC_PM00004   \n",
       "4             BMGC_ED_PMGN00005     BMG_ED_PMGN00005   BMGC_PM00005   \n",
       "...                         ...                  ...            ...   \n",
       "17151448   BMGC_ED_PTPT16484816  BMG_ED_PTPT16484816  BMGC_PT121419   \n",
       "17151449   BMGC_ED_PTPT16484817  BMG_ED_PTPT16484817  BMGC_PT121419   \n",
       "17151450   BMGC_ED_PTPT16484818  BMG_ED_PTPT16484818  BMGC_PT121419   \n",
       "17151451   BMGC_ED_PTPT16484819  BMG_ED_PTPT16484819  BMGC_PT121419   \n",
       "17151452   BMGC_ED_PTPT16484820  BMG_ED_PTPT16484820  BMGC_PT121419   \n",
       "\n",
       "             BMGC_To_ID   BMG_From_ID     BMG_To_ID Source             Type  \n",
       "0          BMGC_GN00001  BMG_PM000001  BMG_GN000001    NaN    Promoter-Gene  \n",
       "1          BMGC_GN00002  BMG_PM000002  BMG_GN000002    NaN    Promoter-Gene  \n",
       "2          BMGC_GN00003  BMG_PM000003  BMG_GN000003    NaN    Promoter-Gene  \n",
       "3          BMGC_GN00004  BMG_PM000004  BMG_GN000004    NaN    Promoter-Gene  \n",
       "4          BMGC_GN00005  BMG_PM000005  BMG_GN000005    NaN    Promoter-Gene  \n",
       "...                 ...           ...           ...    ...              ...  \n",
       "17151448  BMGC_PT117260  BMG_PT173978  BMG_PT169819   KEGG  Protein-Protein  \n",
       "17151449  BMGC_PT120142  BMG_PT173978  BMG_PT172701   KEGG  Protein-Protein  \n",
       "17151450  BMGC_PT120825  BMG_PT173978  BMG_PT173384   KEGG  Protein-Protein  \n",
       "17151451  BMGC_PT120826  BMG_PT173978  BMG_PT173385   KEGG  Protein-Protein  \n",
       "17151452  BMGC_PT120827  BMG_PT173978  BMG_PT173386   KEGG  Protein-Protein  \n",
       "\n",
       "[17151453 rows x 8 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Null values in bmgc_omics_relation_df:\n",
      "BioMedGraphica_Conn_ID        0\n",
      "BioMedGraphica_ID             0\n",
      "BMGC_From_ID                  0\n",
      "BMGC_To_ID                    0\n",
      "BMG_From_ID                   0\n",
      "BMG_To_ID                     0\n",
      "Source                    86238\n",
      "Type                          0\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# translation chain converging to the same node\n",
    "# fetch the promoter, gene, transcript and protein entity alone\n",
    "promoter_entity_df = bmgc_omics_df[bmgc_omics_df['Type'] == 'Promoter'].copy()\n",
    "gene_entity_df = bmgc_omics_df[bmgc_omics_df['Type'] == 'Gene'].copy()\n",
    "transcript_entity_df = bmgc_omics_df[bmgc_omics_df['Type'] == 'Transcript'].copy()\n",
    "protein_entity_df = bmgc_omics_df[bmgc_omics_df['Type'] == 'Protein'].copy()\n",
    "\n",
    "display(bmgc_omics_relation_df)\n",
    "# recheck the null values in bmgc_omics_relation_df\n",
    "print(\"Null values in bmgc_omics_relation_df:\")\n",
    "print(bmgc_omics_relation_df.isnull().sum())\n",
    "\n",
    "# fetch the Promoter-Gene, Gene-Transcript, Transcript-Protein relation alone\n",
    "promoter_gene_relation_df = bmgc_omics_relation_df[bmgc_omics_relation_df['Type'] == 'Promoter-Gene'].copy()\n",
    "gene_transcript_relation_df = bmgc_omics_relation_df[bmgc_omics_relation_df['Type'] == 'Gene-Transcript'].copy()\n",
    "transcript_protein_relation_df = bmgc_omics_relation_df[bmgc_omics_relation_df['Type'] == 'Transcript-Protein'].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_GN_ID</th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_PT013542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_PT091702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT070222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT041299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT038889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126009</th>\n",
       "      <td>BMGC_GN70323</td>\n",
       "      <td>BMGC_PT121177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126010</th>\n",
       "      <td>BMGC_GN70327</td>\n",
       "      <td>BMGC_PT121280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126011</th>\n",
       "      <td>BMGC_GN70329</td>\n",
       "      <td>BMGC_PT121282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126012</th>\n",
       "      <td>BMGC_GN76774</td>\n",
       "      <td>BMGC_PT007612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126013</th>\n",
       "      <td>BMGC_GN86213</td>\n",
       "      <td>BMGC_PT082937</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>126014 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BMGC_GN_ID     BMGC_PT_ID\n",
       "0       BMGC_GN00001  BMGC_PT013542\n",
       "1       BMGC_GN00001  BMGC_PT091702\n",
       "2       BMGC_GN00002  BMGC_PT070222\n",
       "3       BMGC_GN00002  BMGC_PT041299\n",
       "4       BMGC_GN00002  BMGC_PT038889\n",
       "...              ...            ...\n",
       "126009  BMGC_GN70323  BMGC_PT121177\n",
       "126010  BMGC_GN70327  BMGC_PT121280\n",
       "126011  BMGC_GN70329  BMGC_PT121282\n",
       "126012  BMGC_GN76774  BMGC_PT007612\n",
       "126013  BMGC_GN86213  BMGC_PT082937\n",
       "\n",
       "[126014 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4082382977.py:11: DtypeWarning: Columns (4,5,6,11,12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmgc_promoter_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Promoter/BioMedGraphica_Conn_Promoter.csv')\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4082382977.py:12: DtypeWarning: Columns (4,5,6,11,12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmgc_gene_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_PM_ID</th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_PT013542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_PT091702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMGC_PT070222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMGC_PT041299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>BMGC_PT038889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126009</th>\n",
       "      <td>BMGC_PM70323</td>\n",
       "      <td>BMGC_PT121177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126010</th>\n",
       "      <td>BMGC_PM70327</td>\n",
       "      <td>BMGC_PT121280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126011</th>\n",
       "      <td>BMGC_PM70329</td>\n",
       "      <td>BMGC_PT121282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126012</th>\n",
       "      <td>BMGC_PM76774</td>\n",
       "      <td>BMGC_PT007612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126013</th>\n",
       "      <td>BMGC_PM86213</td>\n",
       "      <td>BMGC_PT082937</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>126014 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BMGC_PM_ID     BMGC_PT_ID\n",
       "0       BMGC_PM00001  BMGC_PT013542\n",
       "1       BMGC_PM00001  BMGC_PT091702\n",
       "2       BMGC_PM00002  BMGC_PT070222\n",
       "3       BMGC_PM00002  BMGC_PT041299\n",
       "4       BMGC_PM00002  BMGC_PT038889\n",
       "...              ...            ...\n",
       "126009  BMGC_PM70323  BMGC_PT121177\n",
       "126010  BMGC_PM70327  BMGC_PT121280\n",
       "126011  BMGC_PM70329  BMGC_PT121282\n",
       "126012  BMGC_PM76774  BMGC_PT007612\n",
       "126013  BMGC_PM86213  BMGC_PT082937\n",
       "\n",
       "[126014 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_TS_ID</th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>BMGC_PT017571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>BMGC_PT014978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>BMGC_PT014403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>BMGC_PT017807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>BMGC_PT043262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152580</th>\n",
       "      <td>BMGC_TS411885</td>\n",
       "      <td>BMGC_PT121415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152581</th>\n",
       "      <td>BMGC_TS411886</td>\n",
       "      <td>BMGC_PT121416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152582</th>\n",
       "      <td>BMGC_TS411887</td>\n",
       "      <td>BMGC_PT121417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152583</th>\n",
       "      <td>BMGC_TS412018</td>\n",
       "      <td>BMGC_PT121418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152584</th>\n",
       "      <td>BMGC_TS412019</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>152585 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           BMGC_TS_ID     BMGC_PT_ID\n",
       "0       BMGC_TS000001  BMGC_PT017571\n",
       "1       BMGC_TS000002  BMGC_PT014978\n",
       "2       BMGC_TS000003  BMGC_PT014403\n",
       "3       BMGC_TS000004  BMGC_PT017807\n",
       "4       BMGC_TS000005  BMGC_PT043262\n",
       "...               ...            ...\n",
       "152580  BMGC_TS411885  BMGC_PT121415\n",
       "152581  BMGC_TS411886  BMGC_PT121416\n",
       "152582  BMGC_TS411887  BMGC_PT121417\n",
       "152583  BMGC_TS412018  BMGC_PT121418\n",
       "152584  BMGC_TS412019  BMGC_PT121419\n",
       "\n",
       "[152585 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_transcript_entity_df = pd.merge(gene_entity_df, gene_transcript_relation_df[['BMGC_From_ID', 'BMGC_To_ID']], left_on='BioMedGraphica_Conn_ID', right_on='BMGC_From_ID', how='outer')\n",
    "gene_transcript_protein_entity_df = pd.merge(gene_transcript_entity_df, transcript_protein_relation_df[['BMGC_From_ID', 'BMGC_To_ID']], left_on='BMGC_To_ID', right_on='BMGC_From_ID', how='outer')\n",
    "# drop NaN values in BMGC_From_ID_x\tBMGC_To_ID_x BMGC_From_ID_y\tBMGC_To_ID_y\n",
    "gene_transcript_protein_entity_df = gene_transcript_protein_entity_df.dropna(subset=['BMGC_From_ID_x', 'BMGC_To_ID_x', 'BMGC_From_ID_y', 'BMGC_To_ID_y']).reset_index(drop=True)\n",
    "# keep the columns ['BioMedGraphica_Conn_ID', 'BMGC_To_ID_x', 'BMGC_To_ID_y'] and rename the columns to ['BMGC_GN_ID', 'BMGC_TS_ID', 'BMGC_PT_ID']\n",
    "gene_transcript_protein_entity_df = gene_transcript_protein_entity_df[['BioMedGraphica_Conn_ID', 'BMGC_To_ID_y']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_GN_ID', 'BMGC_To_ID_y': 'BMGC_PT_ID'}).sort_values(by='BMGC_GN_ID').reset_index(drop=True)\n",
    "# drop duplicates rows in gene_transcript_protein_entity_df\n",
    "gene_transcript_protein_entity_df = gene_transcript_protein_entity_df.drop_duplicates().reset_index(drop=True)\n",
    "display(gene_transcript_protein_entity_df)\n",
    "# just copy gene_transcript_protein_entity_df as promoter_gene_transcript_protein_entity_df\n",
    "bmgc_promoter_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Promoter/BioMedGraphica_Conn_Promoter.csv')\n",
    "bmgc_gene_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene.csv')\n",
    "promoter_gene_transcript_protein_entity_df = gene_transcript_protein_entity_df.copy()\n",
    "promoter_gene_df = pd.concat([bmgc_promoter_df[['BioMedGraphica_Conn_ID']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_PM_ID'}), bmgc_gene_df[['BioMedGraphica_Conn_ID']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_GN_ID'})], axis=1)\n",
    "promoter_protein_entity_df = pd.merge(promoter_gene_transcript_protein_entity_df, promoter_gene_df, left_on='BMGC_GN_ID', right_on='BMGC_GN_ID', how='left').drop(columns=['BMGC_GN_ID'])\n",
    "promoter_protein_entity_df = promoter_protein_entity_df[['BMGC_PM_ID', 'BMGC_PT_ID']]\n",
    "display(promoter_protein_entity_df)\n",
    "transcript_protein_entity_df = pd.merge(transcript_entity_df, transcript_protein_relation_df[['BMGC_From_ID', 'BMGC_To_ID']], left_on='BioMedGraphica_Conn_ID', right_on='BMGC_From_ID', how='outer')\n",
    "# drop NaN rows in the BMGC_From_ID and\tBMGC_To_ID\n",
    "transcript_protein_entity_df = transcript_protein_entity_df.dropna(subset=['BMGC_From_ID', 'BMGC_To_ID']).reset_index(drop=True)\n",
    "# keep the columns ['BioMedGraphica_Conn_ID', 'BMGC_To_ID'] and rename the columns to ['BMGC_TS_ID', 'BMGC_PT_ID']\n",
    "transcript_protein_entity_df = transcript_protein_entity_df[['BioMedGraphica_Conn_ID', 'BMGC_To_ID']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_TS_ID', 'BMGC_To_ID': 'BMGC_PT_ID'}).sort_values(by='BMGC_TS_ID').reset_index(drop=True)\n",
    "# drop duplicates rows in transcript_protein_entity_df\n",
    "transcript_protein_entity_df = transcript_protein_entity_df.drop_duplicates().reset_index(drop=True)\n",
    "display(transcript_protein_entity_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           BMGC_PT_ID\n",
       "0       BMGC_PT000001\n",
       "1       BMGC_PT000002\n",
       "2       BMGC_PT000003\n",
       "3       BMGC_PT000004\n",
       "4       BMGC_PT000005\n",
       "...               ...\n",
       "121414  BMGC_PT121415\n",
       "121415  BMGC_PT121416\n",
       "121416  BMGC_PT121417\n",
       "121417  BMGC_PT121418\n",
       "121418  BMGC_PT121419\n",
       "\n",
       "[121419 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# keep the columns ['BioMedGraphica_Conn_ID'] and rename the columns to ['BMGC_PT_ID']\n",
    "only_protein_entity_df = protein_entity_df[['BioMedGraphica_Conn_ID']].rename(columns={'BioMedGraphica_Conn_ID': 'BMGC_PT_ID'}).sort_values(by='BMGC_PT_ID').reset_index(drop=True)\n",
    "display(only_protein_entity_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 8.1 CRISPR top 100 gene entities"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 367,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>A1BG (1)</th>\n",
       "      <th>A1CF (29974)</th>\n",
       "      <th>A2M (2)</th>\n",
       "      <th>A2ML1 (144568)</th>\n",
       "      <th>A3GALT2 (127550)</th>\n",
       "      <th>A4GALT (53947)</th>\n",
       "      <th>A4GNT (51146)</th>\n",
       "      <th>AAAS (8086)</th>\n",
       "      <th>AACS (65985)</th>\n",
       "      <th>...</th>\n",
       "      <th>ZWILCH (55055)</th>\n",
       "      <th>ZWINT (11130)</th>\n",
       "      <th>ZXDA (7789)</th>\n",
       "      <th>ZXDB (158586)</th>\n",
       "      <th>ZXDC (79364)</th>\n",
       "      <th>ZYG11A (440590)</th>\n",
       "      <th>ZYG11B (79699)</th>\n",
       "      <th>ZYX (7791)</th>\n",
       "      <th>ZZEF1 (23140)</th>\n",
       "      <th>ZZZ3 (26009)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>0.042647</td>\n",
       "      <td>0.026102</td>\n",
       "      <td>-0.147614</td>\n",
       "      <td>-0.020425</td>\n",
       "      <td>0.184014</td>\n",
       "      <td>-0.076020</td>\n",
       "      <td>-0.097068</td>\n",
       "      <td>0.182759</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.120536</td>\n",
       "      <td>0.107286</td>\n",
       "      <td>0.166390</td>\n",
       "      <td>0.262275</td>\n",
       "      <td>0.163876</td>\n",
       "      <td>-0.249508</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>0.067986</td>\n",
       "      <td>0.136285</td>\n",
       "      <td>-0.438536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.088500</td>\n",
       "      <td>-0.088198</td>\n",
       "      <td>-0.041225</td>\n",
       "      <td>-0.028074</td>\n",
       "      <td>-0.157206</td>\n",
       "      <td>0.280101</td>\n",
       "      <td>-0.106904</td>\n",
       "      <td>0.178125</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.192527</td>\n",
       "      <td>-0.324059</td>\n",
       "      <td>0.230377</td>\n",
       "      <td>0.087609</td>\n",
       "      <td>0.074897</td>\n",
       "      <td>0.054335</td>\n",
       "      <td>-0.330343</td>\n",
       "      <td>0.099067</td>\n",
       "      <td>0.274566</td>\n",
       "      <td>0.001871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000005</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>0.003300</td>\n",
       "      <td>0.160881</td>\n",
       "      <td>0.086224</td>\n",
       "      <td>-0.149315</td>\n",
       "      <td>-0.253837</td>\n",
       "      <td>0.167011</td>\n",
       "      <td>-0.101209</td>\n",
       "      <td>-0.129827</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.312827</td>\n",
       "      <td>-0.338023</td>\n",
       "      <td>-0.039700</td>\n",
       "      <td>-0.055349</td>\n",
       "      <td>-0.000367</td>\n",
       "      <td>-0.205605</td>\n",
       "      <td>-0.066032</td>\n",
       "      <td>-0.054518</td>\n",
       "      <td>0.035579</td>\n",
       "      <td>-0.150486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000007</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.113717</td>\n",
       "      <td>0.082872</td>\n",
       "      <td>0.099633</td>\n",
       "      <td>-0.008378</td>\n",
       "      <td>-0.022310</td>\n",
       "      <td>0.014416</td>\n",
       "      <td>-0.184977</td>\n",
       "      <td>-0.173739</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.334843</td>\n",
       "      <td>-0.355499</td>\n",
       "      <td>-0.014183</td>\n",
       "      <td>0.230944</td>\n",
       "      <td>0.044628</td>\n",
       "      <td>-0.081863</td>\n",
       "      <td>-0.390748</td>\n",
       "      <td>-0.036547</td>\n",
       "      <td>-0.273129</td>\n",
       "      <td>-0.382723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000009</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>-0.134669</td>\n",
       "      <td>0.060323</td>\n",
       "      <td>0.076647</td>\n",
       "      <td>0.078922</td>\n",
       "      <td>-0.100243</td>\n",
       "      <td>0.047559</td>\n",
       "      <td>-0.136988</td>\n",
       "      <td>0.037759</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.299593</td>\n",
       "      <td>-0.194427</td>\n",
       "      <td>-0.027365</td>\n",
       "      <td>0.236591</td>\n",
       "      <td>-0.084224</td>\n",
       "      <td>-0.098271</td>\n",
       "      <td>-0.510495</td>\n",
       "      <td>0.052938</td>\n",
       "      <td>0.018623</td>\n",
       "      <td>-0.258353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1173</th>\n",
       "      <td>ACH-002926</td>\n",
       "      <td>-0.107883</td>\n",
       "      <td>-0.101501</td>\n",
       "      <td>0.046677</td>\n",
       "      <td>0.096868</td>\n",
       "      <td>-0.118650</td>\n",
       "      <td>-0.065258</td>\n",
       "      <td>-0.064944</td>\n",
       "      <td>-0.175912</td>\n",
       "      <td>-0.044707</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.409541</td>\n",
       "      <td>-0.579376</td>\n",
       "      <td>0.038706</td>\n",
       "      <td>0.338610</td>\n",
       "      <td>-0.174379</td>\n",
       "      <td>-0.040264</td>\n",
       "      <td>-0.239763</td>\n",
       "      <td>0.065168</td>\n",
       "      <td>-0.144088</td>\n",
       "      <td>-0.334424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1174</th>\n",
       "      <td>ACH-002928</td>\n",
       "      <td>-0.287606</td>\n",
       "      <td>-0.179101</td>\n",
       "      <td>-0.135824</td>\n",
       "      <td>-0.046579</td>\n",
       "      <td>-0.249888</td>\n",
       "      <td>-0.097376</td>\n",
       "      <td>0.070385</td>\n",
       "      <td>0.049444</td>\n",
       "      <td>-0.244937</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.095328</td>\n",
       "      <td>-0.440452</td>\n",
       "      <td>0.303500</td>\n",
       "      <td>0.097690</td>\n",
       "      <td>0.151790</td>\n",
       "      <td>0.039994</td>\n",
       "      <td>0.098240</td>\n",
       "      <td>-0.266654</td>\n",
       "      <td>-0.115108</td>\n",
       "      <td>-0.080421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1175</th>\n",
       "      <td>ACH-003012</td>\n",
       "      <td>-0.145878</td>\n",
       "      <td>-0.004000</td>\n",
       "      <td>0.091933</td>\n",
       "      <td>0.191332</td>\n",
       "      <td>-0.065385</td>\n",
       "      <td>-0.134152</td>\n",
       "      <td>0.012353</td>\n",
       "      <td>-0.321012</td>\n",
       "      <td>0.001571</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.280119</td>\n",
       "      <td>-0.662673</td>\n",
       "      <td>-0.180936</td>\n",
       "      <td>0.169997</td>\n",
       "      <td>-0.036326</td>\n",
       "      <td>0.158702</td>\n",
       "      <td>-0.213281</td>\n",
       "      <td>-0.017884</td>\n",
       "      <td>-0.195395</td>\n",
       "      <td>-0.309021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>ACH-003177</td>\n",
       "      <td>-0.038435</td>\n",
       "      <td>-0.055660</td>\n",
       "      <td>0.018989</td>\n",
       "      <td>0.117182</td>\n",
       "      <td>0.063397</td>\n",
       "      <td>0.049993</td>\n",
       "      <td>0.175520</td>\n",
       "      <td>-0.085752</td>\n",
       "      <td>0.078435</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.080700</td>\n",
       "      <td>-0.476797</td>\n",
       "      <td>-0.158375</td>\n",
       "      <td>-0.008999</td>\n",
       "      <td>-0.030058</td>\n",
       "      <td>-0.103513</td>\n",
       "      <td>-0.119234</td>\n",
       "      <td>-0.153744</td>\n",
       "      <td>0.032584</td>\n",
       "      <td>-0.536071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1177</th>\n",
       "      <td>ACH-003180</td>\n",
       "      <td>-0.007314</td>\n",
       "      <td>-0.212478</td>\n",
       "      <td>0.129900</td>\n",
       "      <td>-0.023112</td>\n",
       "      <td>-0.068613</td>\n",
       "      <td>0.067406</td>\n",
       "      <td>0.003678</td>\n",
       "      <td>-0.312161</td>\n",
       "      <td>0.009587</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.038622</td>\n",
       "      <td>-0.667881</td>\n",
       "      <td>-0.090500</td>\n",
       "      <td>0.077756</td>\n",
       "      <td>0.055249</td>\n",
       "      <td>0.040616</td>\n",
       "      <td>-0.378505</td>\n",
       "      <td>-0.143920</td>\n",
       "      <td>-0.039985</td>\n",
       "      <td>-0.439905</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1178 rows × 17917 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0  A1BG (1)  A1CF (29974)   A2M (2)  A2ML1 (144568)  \\\n",
       "0     ACH-000001 -0.121964      0.042647  0.026102       -0.147614   \n",
       "1     ACH-000004 -0.016484     -0.088500 -0.088198       -0.041225   \n",
       "2     ACH-000005 -0.184847      0.003300  0.160881        0.086224   \n",
       "3     ACH-000007 -0.071921     -0.113717  0.082872        0.099633   \n",
       "4     ACH-000009 -0.019163     -0.134669  0.060323        0.076647   \n",
       "...          ...       ...           ...       ...             ...   \n",
       "1173  ACH-002926 -0.107883     -0.101501  0.046677        0.096868   \n",
       "1174  ACH-002928 -0.287606     -0.179101 -0.135824       -0.046579   \n",
       "1175  ACH-003012 -0.145878     -0.004000  0.091933        0.191332   \n",
       "1176  ACH-003177 -0.038435     -0.055660  0.018989        0.117182   \n",
       "1177  ACH-003180 -0.007314     -0.212478  0.129900       -0.023112   \n",
       "\n",
       "      A3GALT2 (127550)  A4GALT (53947)  A4GNT (51146)  AAAS (8086)  \\\n",
       "0            -0.020425        0.184014      -0.076020    -0.097068   \n",
       "1            -0.028074       -0.157206       0.280101    -0.106904   \n",
       "2            -0.149315       -0.253837       0.167011    -0.101209   \n",
       "3            -0.008378       -0.022310       0.014416    -0.184977   \n",
       "4             0.078922       -0.100243       0.047559    -0.136988   \n",
       "...                ...             ...            ...          ...   \n",
       "1173         -0.118650       -0.065258      -0.064944    -0.175912   \n",
       "1174         -0.249888       -0.097376       0.070385     0.049444   \n",
       "1175         -0.065385       -0.134152       0.012353    -0.321012   \n",
       "1176          0.063397        0.049993       0.175520    -0.085752   \n",
       "1177         -0.068613        0.067406       0.003678    -0.312161   \n",
       "\n",
       "      AACS (65985)  ...  ZWILCH (55055)  ZWINT (11130)  ZXDA (7789)  \\\n",
       "0         0.182759  ...       -0.120536       0.107286     0.166390   \n",
       "1         0.178125  ...       -0.192527      -0.324059     0.230377   \n",
       "2        -0.129827  ...       -0.312827      -0.338023    -0.039700   \n",
       "3        -0.173739  ...       -0.334843      -0.355499    -0.014183   \n",
       "4         0.037759  ...       -0.299593      -0.194427    -0.027365   \n",
       "...            ...  ...             ...            ...          ...   \n",
       "1173     -0.044707  ...       -0.409541      -0.579376     0.038706   \n",
       "1174     -0.244937  ...       -0.095328      -0.440452     0.303500   \n",
       "1175      0.001571  ...       -0.280119      -0.662673    -0.180936   \n",
       "1176      0.078435  ...       -0.080700      -0.476797    -0.158375   \n",
       "1177      0.009587  ...       -0.038622      -0.667881    -0.090500   \n",
       "\n",
       "      ZXDB (158586)  ZXDC (79364)  ZYG11A (440590)  ZYG11B (79699)  \\\n",
       "0          0.262275      0.163876        -0.249508        0.053725   \n",
       "1          0.087609      0.074897         0.054335       -0.330343   \n",
       "2         -0.055349     -0.000367        -0.205605       -0.066032   \n",
       "3          0.230944      0.044628        -0.081863       -0.390748   \n",
       "4          0.236591     -0.084224        -0.098271       -0.510495   \n",
       "...             ...           ...              ...             ...   \n",
       "1173       0.338610     -0.174379        -0.040264       -0.239763   \n",
       "1174       0.097690      0.151790         0.039994        0.098240   \n",
       "1175       0.169997     -0.036326         0.158702       -0.213281   \n",
       "1176      -0.008999     -0.030058        -0.103513       -0.119234   \n",
       "1177       0.077756      0.055249         0.040616       -0.378505   \n",
       "\n",
       "      ZYX (7791)  ZZEF1 (23140)  ZZZ3 (26009)  \n",
       "0       0.067986       0.136285     -0.438536  \n",
       "1       0.099067       0.274566      0.001871  \n",
       "2      -0.054518       0.035579     -0.150486  \n",
       "3      -0.036547      -0.273129     -0.382723  \n",
       "4       0.052938       0.018623     -0.258353  \n",
       "...          ...            ...           ...  \n",
       "1173    0.065168      -0.144088     -0.334424  \n",
       "1174   -0.266654      -0.115108     -0.080421  \n",
       "1175   -0.017884      -0.195395     -0.309021  \n",
       "1176   -0.153744       0.032584     -0.536071  \n",
       "1177   -0.143920      -0.039985     -0.439905  \n",
       "\n",
       "[1178 rows x 17917 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "raw_crispr_df = pd.read_csv('./data/raw_data/CRISPRGeneEffect.csv')\n",
    "display(raw_crispr_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 368,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['A1BG (1)', 'A1CF (29974)', 'A2M (2)', 'A2ML1 (144568)', 'A3GALT2 (127550)', 'A4GALT (53947)', 'A4GNT (51146)', 'AAAS (8086)', 'AACS (65985)', 'AADAC (13)', 'AADACL2 (344752)', 'AADACL3 (126767)', 'AADACL4 (343066)', 'AADAT (51166)', 'AAGAB (79719)', 'AAK1 (22848)', 'AAMDC (28971)', 'AAMP (14)', 'AANAT (15)', 'AAR2 (25980)', 'AARD (441376)', 'AARS1 (16)', 'AARS2 (57505)', 'AARSD1 (80755)', 'AASDH (132949)', 'AASDHPPT (60496)', 'AASS (10157)', 'AATF (26574)', 'AATK (9625)', 'ABAT (18)', 'ABCA1 (19)', 'ABCA10 (10349)', 'ABCA12 (26154)', 'ABCA13 (154664)', 'ABCA2 (20)', 'ABCA3 (21)', 'ABCA4 (24)', 'ABCA5 (23461)', 'ABCA6 (23460)', 'ABCA7 (10347)', 'ABCA8 (10351)', 'ABCA9 (10350)', 'ABCB1 (5243)', 'ABCB10 (23456)', 'ABCB11 (8647)', 'ABCB4 (5244)', 'ABCB5 (340273)', 'ABCB6 (10058)', 'ABCB7 (22)', 'ABCB8 (11194)', 'ABCB9 (23457)', 'ABCC1 (4363)', 'ABCC10 (89845)', 'ABCC11 (85320)', 'ABCC12 (94160)', 'ABCC2 (1244)', 'ABCC3 (8714)', 'ABCC4 (10257)', 'ABCC5 (10057)', 'ABCC6 (368)', 'ABCC8 (6833)', 'ABCC9 (10060)', 'ABCD1 (215)', 'ABCD2 (225)', 'ABCD3 (5825)', 'ABCD4 (5826)', 'ABCE1 (6059)', 'ABCF1 (23)', 'ABCF3 (55324)', 'ABCG1 (9619)', 'ABCG2 (9429)', 'ABCG4 (64137)', 'ABCG5 (64240)', 'ABCG8 (64241)', 'ABHD1 (84696)', 'ABHD10 (55347)', 'ABHD11 (83451)', 'ABHD12 (26090)', 'ABHD12B (145447)', 'ABHD13 (84945)', 'ABHD14A (25864)', 'ABHD14B (84836)', 'ABHD15 (116236)', 'ABHD16A (7920)', 'ABHD16B (140701)', 'ABHD17A (81926)', 'ABHD17B (51104)', 'ABHD17C (58489)', 'ABHD18 (80167)', 'ABHD2 (11057)', 'ABHD3 (171586)', 'ABHD4 (63874)', 'ABHD5 (51099)', 'ABHD6 (57406)', 'ABHD8 (79575)', 'ABI1 (10006)', 'ABI2 (10152)', 'ABI3 (51225)', 'ABI3BP (25890)', 'ABITRAM (54942)', 'ABL1 (25)', 'ABL2 (27)', 'ABLIM1 (3983)', 'ABLIM2 (84448)', 'ABLIM3 (22885)', 'ABR (29)', 'ABRA (137735)', 'ABRACL (58527)', 'ABRAXAS1 (84142)', 'ABRAXAS2 (23172)', 'ABT1 (29777)', 'ABTB1 (80325)', 'ABTB2 (25841)', 'ABTB3 (121551)', 'ACAA1 (30)', 'ACAA2 (10449)', 'ACACA (31)', 'ACACB (32)', 'ACAD10 (80724)', 'ACAD11 (84129)', 'ACAD8 (27034)', 'ACAD9 (28976)', 'ACADL (33)', 'ACADM (34)', 'ACADS (35)', 'ACADSB (36)', 'ACADVL (37)', 'ACAN (176)', 'ACAP1 (9744)', 'ACAP2 (23527)', 'ACAP3 (116983)', 'ACAT1 (38)', 'ACAT2 (39)', 'ACBD3 (64746)', 'ACBD4 (79777)', 'ACBD5 (91452)', 'ACBD6 (84320)', 'ACBD7 (414149)', 'ACCS (84680)', 'ACCSL (390110)', 'ACD (65057)', 'ACE (1636)', 'ACE2 (59272)', 'ACER1 (125981)', 'ACER2 (340485)', 'ACER3 (55331)', 'ACHE (43)', 'ACIN1 (22985)', 'ACKR1 (2532)', 'ACKR2 (1238)', 'ACKR3 (57007)', 'ACKR4 (51554)', 'ACLY (47)', 'ACMSD (130013)', 'ACO1 (48)', 'ACO2 (50)', 'ACOD1 (730249)', 'ACOT11 (26027)', 'ACOT12 (134526)', 'ACOT13 (55856)', 'ACOT2 (10965)', 'ACOT4 (122970)', 'ACOT6 (641372)', 'ACOT7 (11332)', 'ACOT8 (10005)', 'ACOT9 (23597)', 'ACOX1 (51)', 'ACOX2 (8309)', 'ACOX3 (8310)', 'ACOXL (55289)', 'ACP1 (52)', 'ACP2 (53)', 'ACP3 (55)', 'ACP4 (93650)', 'ACP5 (54)', 'ACP6 (51205)', 'ACP7 (390928)', 'ACR (49)', 'ACRBP (84519)', 'ACRV1 (56)', 'ACSBG1 (23205)', 'ACSBG2 (81616)', 'ACSF2 (80221)', 'ACSF3 (197322)', 'ACSL1 (2180)', 'ACSL3 (2181)', 'ACSL4 (2182)', 'ACSL5 (51703)', 'ACSL6 (23305)', 'ACSM1 (116285)', 'ACSM2A (123876)', 'ACSM2B (348158)', 'ACSM3 (6296)', 'ACSM4 (341392)', 'ACSM5 (54988)', 'ACSM6 (142827)', 'ACSS1 (84532)', 'ACSS2 (55902)', 'ACSS3 (79611)', 'ACTA1 (58)', 'ACTA2 (59)', 'ACTB (60)', 'ACTBL2 (345651)', 'ACTC1 (70)', 'ACTG1 (71)', 'ACTG2 (72)', 'ACTL10 (170487)', 'ACTL6A (86)', 'ACTL6B (51412)', 'ACTL7A (10881)', 'ACTL7B (10880)', 'ACTL8 (81569)', 'ACTL9 (284382)', 'ACTMAP (284325)', 'ACTN1 (87)', 'ACTN2 (88)', 'ACTN4 (81)', 'ACTR10 (55860)', 'ACTR1A (10121)', 'ACTR1B (10120)', 'ACTR2 (10097)', 'ACTR3 (10096)', 'ACTR3B (57180)', 'ACTR3C (653857)', 'ACTR5 (79913)', 'ACTR6 (64431)', 'ACTR8 (93973)', 'ACTRT1 (139741)', 'ACTRT2 (140625)', 'ACTRT3 (84517)', 'ACVR1 (90)', 'ACVR1B (91)', 'ACVR1C (130399)', 'ACVR2A (92)', 'ACVR2B (93)', 'ACVRL1 (94)', 'ACY1 (95)', 'ACY3 (91703)', 'ACYP1 (97)', 'ACYP2 (98)', 'ADA (100)', 'ADA2 (51816)', 'ADAD1 (132612)', 'ADAD2 (161931)', 'ADAL (161823)', 'ADAM10 (102)', 'ADAM11 (4185)', 'ADAM12 (8038)', 'ADAM15 (8751)', 'ADAM17 (6868)', 'ADAM18 (8749)', 'ADAM19 (8728)', 'ADAM2 (2515)', 'ADAM20 (8748)', 'ADAM21 (8747)', 'ADAM22 (53616)', 'ADAM23 (8745)', 'ADAM28 (10863)', 'ADAM29 (11086)', 'ADAM30 (11085)', 'ADAM32 (203102)', 'ADAM33 (80332)', 'ADAM7 (8756)', 'ADAM8 (101)', 'ADAM9 (8754)', 'ADAMDEC1 (27299)', 'ADAMTS1 (9510)', 'ADAMTS10 (81794)', 'ADAMTS12 (81792)', 'ADAMTS13 (11093)', 'ADAMTS14 (140766)', 'ADAMTS15 (170689)', 'ADAMTS16 (170690)', 'ADAMTS17 (170691)', 'ADAMTS18 (170692)', 'ADAMTS19 (171019)', 'ADAMTS2 (9509)', 'ADAMTS20 (80070)', 'ADAMTS3 (9508)', 'ADAMTS4 (9507)', 'ADAMTS5 (11096)', 'ADAMTS6 (11174)', 'ADAMTS7 (11173)', 'ADAMTS8 (11095)', 'ADAMTS9 (56999)', 'ADAMTSL1 (92949)', 'ADAMTSL2 (9719)', 'ADAMTSL3 (57188)', 'ADAMTSL4 (54507)', 'ADAMTSL5 (339366)', 'ADAP1 (11033)', 'ADAP2 (55803)', 'ADAR (103)', 'ADARB1 (104)', 'ADARB2 (105)', 'ADAT1 (23536)', 'ADAT2 (134637)', 'ADAT3 (113179)', 'ADCK1 (57143)', 'ADCK2 (90956)', 'ADCK5 (203054)', 'ADCY1 (107)', 'ADCY10 (55811)', 'ADCY2 (108)', 'ADCY3 (109)', 'ADCY4 (196883)', 'ADCY5 (111)', 'ADCY6 (112)', 'ADCY7 (113)', 'ADCY8 (114)', 'ADCY9 (115)', 'ADCYAP1 (116)', 'ADCYAP1R1 (117)', 'ADD1 (118)', 'ADD2 (119)', 'ADD3 (120)', 'ADGRA1 (84435)', 'ADGRA2 (25960)', 'ADGRA3 (166647)', 'ADGRB1 (575)', 'ADGRB2 (576)', 'ADGRB3 (577)', 'ADGRD1 (283383)', 'ADGRE1 (2015)', 'ADGRE2 (30817)', 'ADGRE3 (84658)', 'ADGRE5 (976)', 'ADGRF1 (266977)', 'ADGRF3 (165082)', 'ADGRF4 (221393)', 'ADGRF5 (221395)', 'ADGRG1 (9289)', 'ADGRG2 (10149)', 'ADGRG3 (222487)', 'ADGRG4 (139378)', 'ADGRG5 (221188)', 'ADGRG6 (57211)', 'ADGRG7 (84873)', 'ADGRL1 (22859)', 'ADGRL2 (23266)', 'ADGRL3 (23284)', 'ADGRL4 (64123)', 'ADGRV1 (84059)', 'ADH1A (124)', 'ADH1B (125)', 'ADH4 (127)', 'ADH5 (128)', 'ADH6 (130)', 'ADH7 (131)', 'ADHFE1 (137872)', 'ADI1 (55256)', 'ADIG (149685)', 'ADIPOQ (9370)', 'ADIPOR1 (51094)', 'ADIPOR2 (79602)', 'ADIRF (10974)', 'ADISSP (54976)', 'ADK (132)', 'ADM (133)', 'ADM2 (79924)', 'ADM5 (199800)', 'ADNP (23394)', 'ADNP2 (22850)', 'ADO (84890)', 'ADORA1 (134)', 'ADORA2A (135)', 'ADORA2B (136)', 'ADORA3 (140)', 'ADPGK (83440)', 'ADPRH (141)', 'ADPRHL1 (113622)', 'ADPRM (56985)', 'ADPRS (54936)', 'ADRA1A (148)', 'ADRA1B (147)', 'ADRA1D (146)', 'ADRA2A (150)', 'ADRA2B (151)', 'ADRA2C (152)', 'ADRB1 (153)', 'ADRB2 (154)', 'ADRB3 (155)', 'ADRM1 (11047)', 'ADSL (158)', 'ADSS1 (122622)', 'ADSS2 (159)', 'ADTRP (84830)', 'AEBP1 (165)', 'AEBP2 (121536)', 'AEN (64782)', 'AFAP1 (60312)', 'AFAP1L1 (134265)', 'AFAP1L2 (84632)', 'AFDN (4301)', 'AFF1 (4299)', 'AFF2 (2334)', 'AFF3 (3899)', 'AFF4 (27125)', 'AFG1L (246269)', 'AFG2A (166378)', 'AFG2B (79029)', 'AFG3L2 (10939)', 'AFM (173)', 'AFMID (125061)', 'AFP (174)', 'AFTPH (54812)', 'AGA (175)', 'AGAP1 (116987)', 'AGAP2 (116986)', 'AGAP3 (116988)', 'AGAP5 (729092)', 'AGAP6 (414189)', 'AGAP9 (642517)', 'AGBL1 (123624)', 'AGBL2 (79841)', 'AGBL3 (340351)', 'AGBL4 (84871)', 'AGBL5 (60509)', 'AGER (177)', 'AGFG1 (3267)', 'AGFG2 (3268)', 'AGGF1 (55109)', 'AGK (55750)', 'AGL (178)', 'AGMAT (79814)', 'AGMO (392636)', 'AGO1 (26523)', 'AGO2 (27161)', 'AGO3 (192669)', 'AGO4 (192670)', 'AGPAT1 (10554)', 'AGPAT2 (10555)', 'AGPAT3 (56894)', 'AGPAT4 (56895)', 'AGPAT5 (55326)', 'AGPS (8540)', 'AGR2 (10551)', 'AGR3 (155465)', 'AGRN (375790)', 'AGRP (181)', 'AGT (183)', 'AGTPBP1 (23287)', 'AGTR1 (185)', 'AGTR2 (186)', 'AGTRAP (57085)', 'AGXT (189)', 'AGXT2 (64902)', 'AHCTF1 (25909)', 'AHCY (191)', 'AHCYL1 (10768)', 'AHCYL2 (23382)', 'AHDC1 (27245)', 'AHI1 (54806)', 'AHNAK (79026)', 'AHNAK2 (113146)', 'AHR (196)', 'AHRR (57491)', 'AHSA1 (10598)', 'AHSG (197)', 'AHSP (51327)', 'AICDA (57379)', 'AIDA (64853)', 'AIF1 (199)', 'AIF1L (83543)', 'AIFM1 (9131)', 'AIFM2 (84883)', 'AIFM3 (150209)', 'AIG1 (51390)', 'AIM2 (9447)', 'AIMP1 (9255)', 'AIMP2 (7965)', 'AIP (9049)', 'AIPL1 (23746)', 'AIRE (326)', 'AIRIM (54955)', 'AJAP1 (55966)', 'AJM1 (389813)', 'AJUBA (84962)', 'AK1 (203)', 'AK2 (204)', 'AK3 (50808)', 'AK4 (205)', 'AK5 (26289)', 'AK6 (102157402)', 'AK7 (122481)', 'AK8 (158067)', 'AK9 (221264)', 'AKAIN1 (642597)', 'AKAP1 (8165)', 'AKAP10 (11216)', 'AKAP11 (11215)', 'AKAP12 (9590)', 'AKAP13 (11214)', 'AKAP14 (158798)', 'AKAP3 (10566)', 'AKAP4 (8852)', 'AKAP5 (9495)', 'AKAP6 (9472)', 'AKAP7 (9465)', 'AKAP8 (10270)', 'AKAP8L (26993)', 'AKAP9 (10142)', 'AKIP1 (56672)', 'AKIRIN1 (79647)', 'AKIRIN2 (55122)', 'AKNA (80709)', 'AKNAD1 (254268)', 'AKR1A1 (10327)', 'AKR1B1 (231)', 'AKR1B10 (57016)', 'AKR1B15 (441282)', 'AKR1C2 (1646)', 'AKR1C3 (8644)', 'AKR1C4 (1109)', 'AKR1D1 (6718)', 'AKR1E2 (83592)', 'AKR7A2 (8574)', 'AKR7A3 (22977)', 'AKT1 (207)', 'AKT1S1 (84335)', 'AKT2 (208)', 'AKT3 (10000)', 'AKTIP (64400)', 'ALAD (210)', 'ALAS1 (211)', 'ALAS2 (212)', 'ALB (213)', 'ALCAM (214)', 'ALDH16A1 (126133)', 'ALDH18A1 (5832)', 'ALDH1A1 (216)', 'ALDH1A2 (8854)', 'ALDH1A3 (220)', 'ALDH1B1 (219)', 'ALDH1L1 (10840)', 'ALDH1L2 (160428)', 'ALDH2 (217)', 'ALDH3A1 (218)', 'ALDH3A2 (224)', 'ALDH3B2 (222)', 'ALDH4A1 (8659)', 'ALDH5A1 (7915)', 'ALDH6A1 (4329)', 'ALDH7A1 (501)', 'ALDH8A1 (64577)', 'ALDH9A1 (223)', 'ALDOA (226)', 'ALDOB (229)', 'ALDOC (230)', 'ALG1 (56052)', 'ALG10 (84920)', 'ALG10B (144245)', 'ALG11 (440138)', 'ALG12 (79087)', 'ALG13 (79868)', 'ALG14 (199857)', 'ALG2 (85365)', 'ALG3 (10195)', 'ALG5 (29880)', 'ALG6 (29929)', 'ALG8 (79053)', 'ALG9 (79796)', 'ALK (238)', 'ALKAL1 (389658)', 'ALKAL2 (285016)', 'ALKBH1 (8846)', 'ALKBH2 (121642)', 'ALKBH3 (221120)', 'ALKBH4 (54784)', 'ALKBH5 (54890)', 'ALKBH6 (84964)', 'ALKBH7 (84266)', 'ALKBH8 (91801)', 'ALLC (55821)', 'ALMS1 (7840)', 'ALOX12 (239)', 'ALOX12B (242)', 'ALOX15 (246)', 'ALOX15B (247)', 'ALOX5 (240)', 'ALOX5AP (241)', 'ALOXE3 (59344)', 'ALPG (251)', 'ALPI (248)', 'ALPK1 (80216)', 'ALPK2 (115701)', 'ALPK3 (57538)', 'ALPL (249)', 'ALPP (250)', 'ALS2 (57679)', 'ALS2CL (259173)', 'ALX1 (8092)', 'ALX3 (257)', 'ALX4 (60529)', 'ALYREF (10189)', 'AMACR (23600)', 'AMBN (258)', 'AMBP (259)', 'AMBRA1 (55626)', 'AMD1 (262)', 'AMDHD1 (144193)', 'AMDHD2 (51005)', 'AMELX (265)', 'AMELY (266)', 'AMER1 (139285)', 'AMER2 (219287)', 'AMER3 (205147)', 'AMFR (267)', 'AMH (268)', 'AMHR2 (269)', 'AMIGO1 (57463)', 'AMIGO2 (347902)', 'AMIGO3 (386724)', 'AMMECR1 (9949)', 'AMMECR1L (83607)', 'AMN (81693)', 'AMN1 (196394)', 'AMOT (154796)', 'AMOTL1 (154810)', 'AMOTL2 (51421)', 'AMPD1 (270)', 'AMPD2 (271)', 'AMPD3 (272)', 'AMPH (273)', 'AMT (275)', 'AMTN (401138)', 'AMY2A (279)', 'AMZ1 (155185)', 'AMZ2 (51321)', 'ANAPC1 (64682)', 'ANAPC10 (10393)', 'ANAPC11 (51529)', 'ANAPC13 (25847)', 'ANAPC15 (25906)', 'ANAPC16 (119504)', 'ANAPC2 (29882)', 'ANAPC4 (29945)', 'ANAPC5 (51433)', 'ANAPC7 (51434)', 'ANG (283)', 'ANGEL1 (23357)', 'ANGEL2 (90806)', 'ANGPT1 (284)', 'ANGPT2 (285)', 'ANGPT4 (51378)', 'ANGPTL1 (9068)', 'ANGPTL2 (23452)', 'ANGPTL3 (27329)', 'ANGPTL4 (51129)', 'ANGPTL5 (253935)', 'ANGPTL6 (83854)', 'ANGPTL7 (10218)', 'ANGPTL8 (55908)', 'ANHX (647589)', 'ANK1 (286)', 'ANK2 (287)', 'ANK3 (288)', 'ANKAR (150709)', 'ANKDD1A (348094)', 'ANKDD1B (728780)', 'ANKEF1 (63926)', 'ANKFN1 (162282)', 'ANKFY1 (51479)', 'ANKH (56172)', 'ANKHD1 (54882)', 'ANKIB1 (54467)', 'ANKK1 (255239)', 'ANKLE1 (126549)', 'ANKLE2 (23141)', 'ANKMY1 (51281)', 'ANKMY2 (57037)', 'ANKRA2 (57763)', 'ANKRD1 (27063)', 'ANKRD10 (55608)', 'ANKRD11 (29123)', 'ANKRD12 (23253)', 'ANKRD13A (88455)', 'ANKRD13B (124930)', 'ANKRD13C (81573)', 'ANKRD13D (338692)', 'ANKRD16 (54522)', 'ANKRD17 (26057)', 'ANKRD18A (253650)', 'ANKRD2 (26287)', 'ANKRD22 (118932)', 'ANKRD23 (200539)', 'ANKRD24 (170961)', 'ANKRD26 (22852)', 'ANKRD27 (84079)', 'ANKRD28 (23243)', 'ANKRD29 (147463)', 'ANKRD30A (91074)', 'ANKRD30B (374860)', 'ANKRD31 (256006)', 'ANKRD33 (341405)', 'ANKRD33B (651746)', 'ANKRD34A (284615)', 'ANKRD34B (340120)', 'ANKRD34C (390616)', 'ANKRD35 (148741)', 'ANKRD36 (375248)', 'ANKRD37 (353322)', 'ANKRD39 (51239)', 'ANKRD40 (91369)', 'ANKRD42 (338699)', 'ANKRD44 (91526)', 'ANKRD45 (339416)', 'ANKRD46 (157567)', 'ANKRD49 (54851)', 'ANKRD50 (57182)', 'ANKRD52 (283373)', 'ANKRD53 (79998)', 'ANKRD54 (129138)', 'ANKRD55 (79722)', 'ANKRD6 (22881)', 'ANKRD61 (100310846)', 'ANKRD62 (342850)', 'ANKRD65 (441869)', 'ANKRD66 (100287718)', 'ANKRD7 (56311)', 'ANKRD9 (122416)', 'ANKS1A (23294)', 'ANKS1B (56899)', 'ANKS3 (124401)', 'ANKS4B (257629)', 'ANKS6 (203286)', 'ANKZF1 (55139)', 'ANLN (54443)', 'ANO1 (55107)', 'ANO10 (55129)', 'ANO2 (57101)', 'ANO3 (63982)', 'ANO4 (121601)', 'ANO5 (203859)', 'ANO6 (196527)', 'ANO7 (50636)', 'ANO8 (57719)', 'ANO9 (338440)', 'ANOS1 (3730)', 'ANP32A (8125)', 'ANP32B (10541)', 'ANP32D (23519)', 'ANP32E (81611)', 'ANPEP (290)', 'ANTKMT (65990)', 'ANTXR1 (84168)', 'ANTXR2 (118429)', 'ANTXRL (195977)', 'ANXA1 (301)', 'ANXA10 (11199)', 'ANXA11 (311)', 'ANXA13 (312)', 'ANXA2 (302)', 'ANXA2R (389289)', 'ANXA3 (306)', 'ANXA4 (307)', 'ANXA5 (308)', 'ANXA6 (309)', 'ANXA7 (310)', 'ANXA8 (653145)', 'ANXA8L1 (728113)', 'ANXA9 (8416)', 'AOAH (313)', 'AOC1 (26)', 'AOC2 (314)', 'AOC3 (8639)', 'AOPEP (84909)', 'AOX1 (316)', 'AP1AR (55435)', 'AP1B1 (162)', 'AP1G1 (164)', 'AP1G2 (8906)', 'AP1M1 (8907)', 'AP1M2 (10053)', 'AP1S1 (1174)', 'AP1S2 (8905)', 'AP1S3 (130340)', 'AP2A1 (160)', 'AP2A2 (161)', 'AP2B1 (163)', 'AP2M1 (1173)', 'AP2S1 (1175)', 'AP3B1 (8546)', 'AP3B2 (8120)', 'AP3D1 (8943)', 'AP3M1 (26985)', 'AP3M2 (10947)', 'AP3S1 (1176)', 'AP3S2 (10239)', 'AP4B1 (10717)', 'AP4E1 (23431)', 'AP4M1 (9179)', 'AP4S1 (11154)', 'AP5B1 (91056)', 'AP5M1 (55745)', 'AP5S1 (55317)', 'AP5Z1 (9907)', 'APAF1 (317)', 'APBA1 (320)', 'APBA2 (321)', 'APBA3 (9546)', 'APBB1 (322)', 'APBB1IP (54518)', 'APBB2 (323)', 'APBB3 (10307)', 'APC (324)', 'APC2 (10297)', 'APCDD1 (147495)', 'APCDD1L (164284)', 'APCS (325)', 'APEH (327)', 'APEX1 (328)', 'APEX2 (27301)', 'APH1A (51107)', 'APH1B (83464)', 'API5 (8539)', 'APIP (51074)', 'APLF (200558)', 'APLN (8862)', 'APLNR (187)', 'APLP1 (333)', 'APLP2 (334)', 'APMAP (57136)', 'APOA1 (335)', 'APOA2 (336)', 'APOA4 (337)', 'APOA5 (116519)', 'APOB (338)', 'APOBEC1 (339)', 'APOBEC2 (10930)', 'APOBEC3A (200315)', 'APOBEC3B (9582)', 'APOBEC3C (27350)', 'APOBEC3D (140564)', 'APOBEC3F (200316)', 'APOBEC3G (60489)', 'APOBEC3H (164668)', 'APOBEC4 (403314)', 'APOBR (55911)', 'APOC1 (341)', 'APOC2 (344)', 'APOC3 (345)', 'APOC4 (346)', 'APOD (347)', 'APOE (348)', 'APOF (319)', 'APOH (350)', 'APOL1 (8542)', 'APOL2 (23780)', 'APOL3 (80833)', 'APOL5 (80831)', 'APOL6 (80830)', 'APOLD1 (81575)', 'APOM (55937)', 'APOO (79135)', 'APOOL (139322)', 'APP (351)', 'APPBP2 (10513)', 'APPL1 (26060)', 'APPL2 (55198)', 'APRT (353)', 'APTX (54840)', 'AQP1 (358)', 'AQP10 (89872)', 'AQP11 (282679)', 'AQP12B (653437)', 'AQP2 (359)', 'AQP3 (360)', 'AQP4 (361)', 'AQP5 (362)', 'AQP6 (363)', 'AQP7 (364)', 'AQP8 (343)', 'AQP9 (366)', 'AQR (9716)', 'AR (367)', 'ARAF (369)', 'ARAP1 (116985)', 'ARAP2 (116984)', 'ARAP3 (64411)', 'ARB2A (83989)', 'ARC (23237)', 'ARCN1 (372)', 'AREG (374)', 'AREL1 (9870)', 'ARF1 (375)', 'ARF3 (377)', 'ARF4 (378)', 'ARF5 (381)', 'ARF6 (382)', 'ARFGAP1 (55738)', 'ARFGAP2 (84364)', 'ARFGAP3 (26286)', 'ARFGEF1 (10565)', 'ARFGEF2 (10564)', 'ARFGEF3 (57221)', 'ARFIP1 (27236)', 'ARFIP2 (23647)', 'ARFRP1 (10139)', 'ARG1 (383)', 'ARG2 (384)', 'ARGFX (503582)', 'ARGLU1 (55082)', 'ARHGAP1 (392)', 'ARHGAP10 (79658)', 'ARHGAP11A (9824)', 'ARHGAP12 (94134)', 'ARHGAP15 (55843)', 'ARHGAP17 (55114)', 'ARHGAP18 (93663)', 'ARHGAP19 (84986)', 'ARHGAP20 (57569)', 'ARHGAP21 (57584)', 'ARHGAP22 (58504)', 'ARHGAP23 (57636)', 'ARHGAP24 (83478)', 'ARHGAP25 (9938)', 'ARHGAP26 (23092)', 'ARHGAP27 (201176)', 'ARHGAP28 (79822)', 'ARHGAP29 (9411)', 'ARHGAP30 (257106)', 'ARHGAP31 (57514)', 'ARHGAP32 (9743)', 'ARHGAP33 (115703)', 'ARHGAP35 (2909)', 'ARHGAP36 (158763)', 'ARHGAP39 (80728)', 'ARHGAP4 (393)', 'ARHGAP44 (9912)', 'ARHGAP45 (23526)', 'ARHGAP5 (394)', 'ARHGAP6 (395)', 'ARHGAP8 (23779)', 'ARHGAP9 (64333)', 'ARHGDIA (396)', 'ARHGDIB (397)', 'ARHGDIG (398)', 'ARHGEF1 (9138)', 'ARHGEF10 (9639)', 'ARHGEF10L (55160)', 'ARHGEF11 (9826)', 'ARHGEF12 (23365)', 'ARHGEF15 (22899)', 'ARHGEF16 (27237)', 'ARHGEF17 (9828)', 'ARHGEF18 (23370)', 'ARHGEF19 (128272)', 'ARHGEF2 (9181)', 'ARHGEF25 (115557)', 'ARHGEF26 (26084)', 'ARHGEF28 (64283)', 'ARHGEF3 (50650)', 'ARHGEF33 (100271715)', 'ARHGEF35 (445328)', 'ARHGEF37 (389337)', 'ARHGEF38 (54848)', 'ARHGEF39 (84904)', 'ARHGEF4 (50649)', 'ARHGEF40 (55701)', 'ARHGEF5 (7984)', 'ARHGEF6 (9459)', 'ARHGEF7 (8874)', 'ARHGEF9 (23229)', 'ARID1A (8289)', 'ARID1B (57492)', 'ARID2 (196528)', 'ARID3A (1820)', 'ARID3B (10620)', 'ARID3C (138715)', 'ARID4A (5926)', 'ARID4B (51742)', 'ARID5A (10865)', 'ARID5B (84159)', 'ARIH1 (25820)', 'ARIH2 (10425)', 'ARK2C (494470)', 'ARK2N (147339)', 'ARL1 (400)', 'ARL10 (285598)', 'ARL11 (115761)', 'ARL13A (392509)', 'ARL13B (200894)', 'ARL14 (80117)', 'ARL14EP (120534)', 'ARL14EPL (644100)', 'ARL15 (54622)', 'ARL16 (339231)', 'ARL2 (402)', 'ARL2BP (23568)', 'ARL3 (403)', 'ARL4A (10124)', 'ARL4C (10123)', 'ARL4D (379)', 'ARL5A (26225)', 'ARL5B (221079)', 'ARL5C (390790)', 'ARL6 (84100)', 'ARL6IP1 (23204)', 'ARL6IP5 (10550)', 'ARL6IP6 (151188)', 'ARL8A (127829)', 'ARL8B (55207)', 'ARL9 (132946)', 'ARMC1 (55156)', 'ARMC10 (83787)', 'ARMC12 (221481)', 'ARMC2 (84071)', 'ARMC3 (219681)', 'ARMC5 (79798)', 'ARMC6 (93436)', 'ARMC7 (79637)', 'ARMC8 (25852)', 'ARMC9 (80210)', 'ARMCX1 (51309)', 'ARMCX2 (9823)', 'ARMCX3 (51566)', 'ARMCX4 (100131755)', 'ARMCX5 (64860)', 'ARMCX6 (54470)', 'ARMH1 (339541)', 'ARMH2 (101928603)', 'ARMH3 (79591)', 'ARMH4 (145407)', 'ARMS2 (387715)', 'ARMT1 (79624)', 'ARNT (405)', 'ARNT2 (9915)', 'ARPC1A (10552)', 'ARPC1B (10095)', 'ARPC2 (10109)', 'ARPC3 (10094)', 'ARPC4 (10093)', 'ARPC5 (10092)', 'ARPC5L (81873)', 'ARPIN (348110)', 'ARPP19 (10776)', 'ARPP21 (10777)', 'ARR3 (407)', 'ARRB1 (408)', 'ARRB2 (409)', 'ARRDC1 (92714)', 'ARRDC2 (27106)', 'ARRDC3 (57561)', 'ARRDC4 (91947)', 'ARRDC5 (645432)', 'ARSA (410)', 'ARSB (411)', 'ARSD (414)', 'ARSF (416)', 'ARSG (22901)', 'ARSH (347527)', 'ARSI (340075)', 'ARSJ (79642)', 'ARSK (153642)', 'ARSL (415)', 'ART1 (417)', 'ART3 (419)', 'ART4 (420)', 'ART5 (116969)', 'ARTN (9048)', 'ARV1 (64801)', 'ARVCF (421)', 'ARX (170302)', 'AS3MT (57412)', 'ASAH1 (427)', 'ASAH2 (56624)', 'ASAP1 (50807)', 'ASAP2 (8853)', 'ASAP3 (55616)', 'ASB1 (51665)', 'ASB10 (136371)', 'ASB11 (140456)', 'ASB12 (142689)', 'ASB13 (79754)', 'ASB14 (142686)', 'ASB15 (142685)', 'ASB16 (92591)', 'ASB17 (127247)', 'ASB18 (401036)', 'ASB2 (51676)', 'ASB3 (51130)', 'ASB4 (51666)', 'ASB5 (140458)', 'ASB6 (140459)', 'ASB7 (140460)', 'ASB8 (140461)', 'ASB9 (140462)', 'ASCC1 (51008)', 'ASCC2 (84164)', 'ASCC3 (10973)', 'ASCL1 (429)', 'ASCL2 (430)', 'ASCL3 (56676)', 'ASCL4 (121549)', 'ASCL5 (647219)', 'ASF1A (25842)', 'ASF1B (55723)', 'ASGR1 (432)', 'ASGR2 (433)', 'ASH1L (55870)', 'ASH2L (9070)', 'ASIC1 (41)', 'ASIC2 (40)', 'ASIC3 (9311)', 'ASIC4 (55515)', 'ASIC5 (51802)', 'ASIP (434)', 'ASL (435)', 'ASNS (440)', 'ASNSD1 (54529)', 'ASPA (443)', 'ASPDH (554235)', 'ASPG (374569)', 'ASPH (444)', 'ASPHD1 (253982)', 'ASPHD2 (57168)', 'ASPM (259266)', 'ASPRV1 (151516)', 'ASPSCR1 (79058)', 'ASRGL1 (80150)', 'ASS1 (445)', 'ASTE1 (28990)', 'ASTL (431705)', 'ASTN1 (460)', 'ASTN2 (23245)', 'ASXL1 (171023)', 'ASXL3 (80816)', 'ASZ1 (136991)', 'ATAD1 (84896)', 'ATAD2 (29028)', 'ATAD2B (54454)', 'ATAD3A (55210)', 'ATAD3B (83858)', 'ATAD3C (219293)', 'ATAD5 (79915)', 'ATAT1 (79969)', 'ATCAY (85300)', 'ATE1 (11101)', 'ATF1 (466)', 'ATF2 (1386)', 'ATF3 (467)', 'ATF4 (468)', 'ATF5 (22809)', 'ATF6 (22926)', 'ATF6B (1388)', 'ATF7 (11016)', 'ATF7IP (55729)', 'ATF7IP2 (80063)', 'ATG10 (83734)', 'ATG101 (60673)', 'ATG12 (9140)', 'ATG13 (9776)', 'ATG14 (22863)', 'ATG16L1 (55054)', 'ATG16L2 (89849)', 'ATG2A (23130)', 'ATG2B (55102)', 'ATG3 (64422)', 'ATG4A (115201)', 'ATG4B (23192)', 'ATG4C (84938)', 'ATG4D (84971)', 'ATG5 (9474)', 'ATG7 (10533)', 'ATG9A (79065)', 'ATIC (471)', 'ATL1 (51062)', 'ATL2 (64225)', 'ATL3 (25923)', 'ATM (472)', 'ATMIN (23300)', 'ATN1 (1822)', 'ATOH1 (474)', 'ATOH7 (220202)', 'ATOH8 (84913)', 'ATOSA (56204)', 'ATOSB (80256)', 'ATOX1 (475)', 'ATP10A (57194)', 'ATP10B (23120)', 'ATP10D (57205)', 'ATP11A (23250)', 'ATP11B (23200)', 'ATP11C (286410)', 'ATP12A (479)', 'ATP13A1 (57130)', 'ATP13A2 (23400)', 'ATP13A3 (79572)', 'ATP13A4 (84239)', 'ATP13A5 (344905)', 'ATP1A1 (476)', 'ATP1A2 (477)', 'ATP1A3 (478)', 'ATP1A4 (480)', 'ATP1B1 (481)', 'ATP1B2 (482)', 'ATP1B3 (483)', 'ATP1B4 (23439)', 'ATP23 (91419)', 'ATP2A1 (487)', 'ATP2A2 (488)', 'ATP2A3 (489)', 'ATP2B1 (490)', 'ATP2B2 (491)', 'ATP2B3 (492)', 'ATP2B4 (493)', 'ATP2C1 (27032)', 'ATP2C2 (9914)', 'ATP4A (495)', 'ATP4B (496)', 'ATP5F1A (498)', 'ATP5F1B (506)', 'ATP5F1C (509)', 'ATP5F1D (513)', 'ATP5F1E (514)', 'ATP5IF1 (93974)', 'ATP5MC1 (516)', 'ATP5MC2 (517)', 'ATP5MC3 (518)', 'ATP5ME (521)', 'ATP5MF (9551)', 'ATP5MG (10632)', 'ATP5MGL (267020)', 'ATP5MJ (9556)', 'ATP5MK (84833)', 'ATP5PB (515)', 'ATP5PD (10476)', 'ATP5PF (522)', 'ATP5PO (539)', 'ATP6AP1 (537)', 'ATP6AP2 (10159)', 'ATP6V0A1 (535)', 'ATP6V0A2 (23545)', 'ATP6V0A4 (50617)', 'ATP6V0B (533)', 'ATP6V0C (527)', 'ATP6V0D1 (9114)', 'ATP6V0D2 (245972)', 'ATP6V0E1 (8992)', 'ATP6V0E2 (155066)', 'ATP6V1A (523)', 'ATP6V1B1 (525)', 'ATP6V1B2 (526)', 'ATP6V1C1 (528)', 'ATP6V1C2 (245973)', 'ATP6V1D (51382)', 'ATP6V1E1 (529)', 'ATP6V1E2 (90423)', 'ATP6V1F (9296)', 'ATP6V1G1 (9550)', 'ATP6V1G2 (534)', 'ATP6V1G3 (127124)', 'ATP6V1H (51606)', 'ATP7A (538)', 'ATP7B (540)', 'ATP8A1 (10396)', 'ATP8A2 (51761)', 'ATP8B1 (5205)', 'ATP8B2 (57198)', 'ATP8B3 (148229)', 'ATP8B4 (79895)', 'ATP9A (10079)', 'ATP9B (374868)', 'ATPAF1 (64756)', 'ATPAF2 (91647)', 'ATPSCKMT (134145)', 'ATR (545)', 'ATRAID (51374)', 'ATRIP (84126)', 'ATRN (8455)', 'ATRNL1 (26033)', 'ATRX (546)', 'ATXN1 (6310)', 'ATXN10 (25814)', 'ATXN1L (342371)', 'ATXN2 (6311)', 'ATXN2L (11273)', 'ATXN3 (4287)', 'ATXN3L (92552)', 'ATXN7 (6314)', 'ATXN7L1 (222255)', 'ATXN7L2 (127002)', 'ATXN7L3 (56970)', 'ATXN7L3B (552889)', 'AUH (549)', 'AUNIP (79000)', 'AUP1 (550)', 'AURKA (6790)', 'AURKAIP1 (54998)', 'AURKB (9212)', 'AURKC (6795)', 'AUTS2 (26053)', 'AVEN (57099)', 'AVIL (10677)', 'AVL9 (23080)', 'AVP (551)', 'AVPI1 (60370)', 'AVPR1A (552)', 'AVPR1B (553)', 'AVPR2 (554)', 'AWAT1 (158833)', 'AWAT2 (158835)', 'AXDND1 (126859)', 'AXIN1 (8312)', 'AXIN2 (8313)', 'AXL (558)', 'AZGP1 (563)', 'AZI2 (64343)', 'AZIN1 (51582)', 'AZIN2 (113451)', 'AZU1 (566)', 'B2M (567)', 'B3GALNT1 (8706)', 'B3GALNT2 (148789)', 'B3GALT1 (8708)', 'B3GALT2 (8707)', 'B3GALT4 (8705)', 'B3GALT5 (10317)', 'B3GALT6 (126792)', 'B3GAT1 (27087)', 'B3GAT2 (135152)', 'B3GAT3 (26229)', 'B3GLCT (145173)', 'B3GNT2 (10678)', 'B3GNT3 (10331)', 'B3GNT4 (79369)', 'B3GNT5 (84002)', 'B3GNT6 (192134)', 'B3GNT7 (93010)', 'B3GNT8 (374907)', 'B3GNT9 (84752)', 'B3GNTL1 (146712)', 'B4GALNT1 (2583)', 'B4GALNT2 (124872)', 'B4GALNT3 (283358)', 'B4GALNT4 (338707)', 'B4GALT1 (2683)', 'B4GALT2 (8704)', 'B4GALT3 (8703)', 'B4GALT4 (8702)', 'B4GALT5 (9334)', 'B4GALT6 (9331)', 'B4GALT7 (11285)', 'B4GAT1 (11041)', 'B9D1 (27077)', 'B9D2 (80776)', 'BAALC (79870)', 'BAAT (570)', 'BABAM1 (29086)', 'BABAM2 (9577)', 'BACE1 (23621)', 'BACE2 (25825)', 'BACH1 (571)', 'BACH2 (60468)', 'BAD (572)', 'BAG1 (573)', 'BAG2 (9532)', 'BAG3 (9531)', 'BAG4 (9530)', 'BAG5 (9529)', 'BAG6 (7917)', 'BAHD1 (22893)', 'BAIAP2 (10458)', 'BAIAP2L1 (55971)', 'BAIAP2L2 (80115)', 'BAIAP3 (8938)', 'BAK1 (578)', 'BAMBI (25805)', 'BANF1 (8815)', 'BANF2 (140836)', 'BANK1 (55024)', 'BANP (54971)', 'BAP1 (8314)', 'BARD1 (580)', 'BARHL1 (56751)', 'BARHL2 (343472)', 'BARX1 (56033)', 'BARX2 (8538)', 'BASP1 (10409)', 'BATF (10538)', 'BATF2 (116071)', 'BATF3 (55509)', 'BAX (581)', 'BAZ1A (11177)', 'BAZ1B (9031)', 'BAZ2A (11176)', 'BAZ2B (29994)', 'BBC3 (27113)', 'BBIP1 (92482)', 'BBLN (79095)', 'BBOF1 (80127)', 'BBOX1 (8424)', 'BBS1 (582)', 'BBS10 (79738)', 'BBS12 (166379)', 'BBS2 (583)', 'BBS4 (585)', 'BBS5 (129880)', 'BBS7 (55212)', 'BBS9 (27241)', 'BBX (56987)', 'BCAM (4059)', 'BCAN (63827)', 'BCAP29 (55973)', 'BCAP31 (10134)', 'BCAR1 (9564)', 'BCAR3 (8412)', 'BCAS1 (8537)', 'BCAS2 (10286)', 'BCAS3 (54828)', 'BCAS4 (55653)', 'BCAT1 (586)', 'BCAT2 (587)', 'BCCIP (56647)', 'BCDIN3D (144233)', 'BCHE (590)', 'BCKDHA (593)', 'BCKDHB (594)', 'BCKDK (10295)', 'BCL10 (8915)', 'BCL11A (53335)', 'BCL11B (64919)', 'BCL2 (596)', 'BCL2A1 (597)', 'BCL2L1 (598)', 'BCL2L10 (10017)', 'BCL2L11 (10018)', 'BCL2L12 (83596)', 'BCL2L13 (23786)', 'BCL2L14 (79370)', 'BCL2L15 (440603)', 'BCL2L2 (599)', 'BCL3 (602)', 'BCL6 (604)', 'BCL6B (255877)', 'BCL7A (605)', 'BCL7B (9275)', 'BCL7C (9274)', 'BCL9 (607)', 'BCL9L (283149)', 'BCLAF1 (9774)', 'BCLAF3 (256643)', 'BCO1 (53630)', 'BCO2 (83875)', 'BCOR (54880)', 'BCORL1 (63035)', 'BCR (613)', 'BCS1L (617)', 'BDH1 (622)', 'BDH2 (56898)', 'BDKRB1 (623)', 'BDKRB2 (624)', 'BDNF (627)', 'BDP1 (55814)', 'BEAN1 (146227)', 'BECN1 (8678)', 'BEGAIN (57596)', 'BEND2 (139105)', 'BEND3 (57673)', 'BEND4 (389206)', 'BEND5 (79656)', 'BEND6 (221336)', 'BEND7 (222389)', 'BEST1 (7439)', 'BEST2 (54831)', 'BEST3 (144453)', 'BEST4 (266675)', 'BET1 (10282)', 'BET1L (51272)', 'BEX1 (55859)', 'BEX2 (84707)', 'BEX3 (27018)', 'BEX4 (56271)', 'BEX5 (340542)', 'BFAR (51283)', 'BFSP1 (631)', 'BFSP2 (8419)', 'BGLAP (632)', 'BGN (633)', 'BHLHA15 (168620)', 'BHLHA9 (727857)', 'BHLHE22 (27319)', 'BHLHE23 (128408)', 'BHLHE40 (8553)', 'BHLHE41 (79365)', 'BHMT (635)', 'BHMT2 (23743)', 'BICC1 (80114)', 'BICD1 (636)', 'BICD2 (23299)', 'BICDL1 (92558)', 'BICDL2 (146439)', 'BICRA (29998)', 'BICRAL (23506)', 'BID (637)', 'BIK (638)', 'BIN1 (274)', 'BIN2 (51411)', 'BIN3 (55909)', 'BIRC2 (329)', 'BIRC3 (330)', 'BIRC5 (332)', 'BIRC6 (57448)', 'BIRC7 (79444)', 'BIVM (54841)', 'BLCAP (10904)', 'BLID (414899)', 'BLK (640)', 'BLM (641)', 'BLMH (642)', 'BLNK (29760)', 'BLOC1S1 (2647)', 'BLOC1S2 (282991)', 'BLOC1S3 (388552)', 'BLOC1S4 (55330)', 'BLOC1S5 (63915)', 'BLOC1S6 (26258)', 'BLTP1 (84162)', 'BLTP2 (9703)', 'BLTP3A (54887)', 'BLTP3B (23074)', 'BLVRA (644)', 'BLVRB (645)', 'BLZF1 (8548)', 'BMAL1 (406)', 'BMAL2 (56938)', 'BMERB1 (89927)', 'BMF (90427)', 'BMI1 (648)', 'BMP1 (649)', 'BMP10 (27302)', 'BMP15 (9210)', 'BMP2 (650)', 'BMP2K (55589)', 'BMP3 (651)', 'BMP4 (652)', 'BMP5 (653)', 'BMP6 (654)', 'BMP7 (655)', 'BMP8A (353500)', 'BMP8B (656)', 'BMPER (168667)', 'BMPR1A (657)', 'BMPR1B (658)', 'BMPR2 (659)', 'BMS1 (9790)', 'BMT2 (154743)', 'BMX (660)', 'BNC1 (646)', 'BNC2 (54796)', 'BNIP1 (662)', 'BNIP2 (663)', 'BNIP3 (664)', 'BNIP3L (665)', 'BNIP5 (389384)', 'BNIPL (149428)', 'BOC (91653)', 'BOD1 (91272)', 'BOD1L1 (259282)', 'BOD1L2 (284257)', 'BOK (666)', 'BOLA1 (51027)', 'BOLA3 (388962)', 'BOLL (66037)', 'BOP1 (23246)', 'BORA (79866)', 'BORCS5 (118426)', 'BORCS6 (54785)', 'BORCS7 (119032)', 'BORCS8 (729991)', 'BPGM (669)', 'BPHL (670)', 'BPI (671)', 'BPIFA1 (51297)', 'BPIFA2 (140683)', 'BPIFA3 (128861)', 'BPIFB1 (92747)', 'BPIFB2 (80341)', 'BPIFB3 (359710)', 'BPIFB4 (149954)', 'BPIFB6 (128859)', 'BPIFC (254240)', 'BPNT1 (10380)', 'BPNT2 (54928)', 'BPTF (2186)', 'BRAF (673)', 'BRAP (8315)', 'BRAT1 (221927)', 'BRCA1 (672)', 'BRCA2 (675)', 'BRCC3 (79184)', 'BRD1 (23774)', 'BRD2 (6046)', 'BRD3 (8019)', 'BRD4 (23476)', 'BRD7 (29117)', 'BRD8 (10902)', 'BRD9 (65980)', 'BRDT (676)', 'BRF1 (2972)', 'BRF2 (55290)', 'BRI3 (25798)', 'BRI3BP (140707)', 'BRICD5 (283870)', 'BRINP1 (1620)', 'BRINP2 (57795)', 'BRINP3 (339479)', 'BRIP1 (83990)', 'BRIX1 (55299)', 'BRK1 (55845)', 'BRME1 (79173)', 'BRMS1 (25855)', 'BRMS1L (84312)', 'BROX (148362)', 'BRPF1 (7862)', 'BRPF3 (27154)', 'BRS3 (680)', 'BRSK1 (84446)', 'BRSK2 (9024)', 'BRWD1 (54014)', 'BRWD3 (254065)', 'BSCL2 (26580)', 'BSDC1 (55108)', 'BSG (682)', 'BSN (8927)', 'BSND (7809)', 'BSPH1 (100131137)', 'BSPRY (54836)', 'BST1 (683)', 'BST2 (684)', 'BSX (390259)', 'BTAF1 (9044)', 'BTBD1 (53339)', 'BTBD10 (84280)', 'BTBD16 (118663)', 'BTBD17 (388419)', 'BTBD18 (643376)', 'BTBD2 (55643)', 'BTBD3 (22903)', 'BTBD6 (90135)', 'BTBD7 (55727)', 'BTBD8 (284697)', 'BTBD9 (114781)', 'BTC (685)', 'BTD (686)', 'BTF3 (689)', 'BTF3L4 (91408)', 'BTG1 (694)', 'BTG2 (7832)', 'BTG3 (10950)', 'BTG4 (54766)', 'BTK (695)', 'BTLA (151888)', 'BTN1A1 (696)', 'BTN2A1 (11120)', 'BTN2A2 (10385)', 'BTN3A1 (11119)', 'BTN3A2 (11118)', 'BTN3A3 (10384)', 'BTNL3 (10917)', 'BTNL8 (79908)', 'BTNL9 (153579)', 'BTRC (8945)', 'BUB1 (699)', 'BUB1B (701)', 'BUB3 (9184)', 'BUD13 (84811)', 'BUD23 (114049)', 'BUD31 (8896)', 'BVES (11149)', 'BYSL (705)', 'BZW1 (9689)', 'BZW2 (28969)', 'C10orf105 (414152)', 'C10orf120 (399814)', 'C10orf53 (282966)', 'C10orf62 (414157)', 'C10orf67 (256815)', 'C10orf71 (118461)', 'C10orf88 (80007)', 'C10orf90 (118611)', 'C10orf95 (79946)', 'C11orf16 (56673)', 'C11orf21 (29125)', 'C11orf24 (53838)', 'C11orf42 (160298)', 'C11orf52 (91894)', 'C11orf54 (28970)', 'C11orf58 (10944)', 'C11orf65 (160140)', 'C11orf68 (83638)', 'C11orf71 (54494)', 'C11orf86 (254439)', 'C11orf87 (399947)', 'C11orf91 (100131378)', 'C12orf42 (374470)', 'C12orf43 (64897)', 'C12orf50 (160419)', 'C12orf54 (121273)', 'C12orf56 (115749)', 'C12orf57 (113246)', 'C12orf60 (144608)', 'C12orf71 (728858)', 'C12orf75 (387882)', 'C14orf119 (55017)', 'C14orf180 (400258)', 'C14orf28 (122525)', 'C14orf39 (317761)', 'C14orf93 (60686)', 'C15orf39 (56905)', 'C15orf40 (123207)', 'C15orf48 (84419)', 'C15orf61 (145853)', 'C15orf62 (643338)', 'C16orf46 (123775)', 'C16orf54 (283897)', 'C16orf74 (404550)', 'C16orf78 (123970)', 'C16orf86 (388284)', 'C16orf87 (388272)', 'C16orf89 (146556)', 'C16orf90 (646174)', 'C16orf92 (146378)', 'C16orf95 (100506581)', 'C16orf96 (342346)', 'C17orf107 (100130311)', 'C17orf49 (124944)', 'C17orf50 (146853)', 'C17orf58 (284018)', 'C17orf67 (339210)', 'C17orf75 (64149)', 'C17orf78 (284099)', 'C17orf99 (100141515)', 'C18orf21 (83608)', 'C18orf32 (497661)', 'C18orf54 (162681)', 'C18orf63 (644041)', 'C19orf12 (83636)', 'C19orf18 (147685)', 'C19orf25 (148223)', 'C19orf33 (64073)', 'C19orf38 (255809)', 'C19orf44 (84167)', 'C19orf47 (126526)', 'C19orf53 (28974)', 'C19orf67 (646457)', 'C19orf73 (55150)', 'C19orf81 (342918)', 'C19orf84 (147646)', 'C1D (10438)', 'C1GALT1 (56913)', 'C1GALT1C1 (29071)', 'C1QA (712)', 'C1QB (713)', 'C1QBP (708)', 'C1QC (714)', 'C1QL1 (10882)', 'C1QL2 (165257)', 'C1QL3 (389941)', 'C1QL4 (338761)', 'C1QTNF1 (114897)', 'C1QTNF12 (388581)', 'C1QTNF2 (114898)', 'C1QTNF3 (114899)', 'C1QTNF4 (114900)', 'C1QTNF5 (114902)', 'C1QTNF6 (114904)', 'C1QTNF7 (114905)', 'C1QTNF8 (390664)', 'C1QTNF9 (338872)', 'C1QTNF9B (387911)', 'C1RL (51279)', 'C1S (716)', 'C1orf105 (92346)', 'C1orf115 (79762)', 'C1orf116 (79098)', 'C1orf122 (127687)', 'C1orf127 (148345)', 'C1orf131 (128061)', 'C1orf141 (400757)', 'C1orf146 (388649)', 'C1orf159 (54991)', 'C1orf162 (128346)', 'C1orf174 (339448)', 'C1orf185 (284546)', 'C1orf198 (84886)', 'C1orf21 (81563)', 'C1orf210 (149466)', 'C1orf216 (127703)', 'C1orf226 (400793)', 'C1orf35 (79169)', 'C1orf43 (25912)', 'C1orf50 (79078)', 'C1orf52 (148423)', 'C1orf53 (388722)', 'C1orf54 (79630)', 'C1orf56 (54964)', 'C1orf74 (148304)', 'C1orf87 (127795)', 'C1orf94 (84970)', 'C2 (717)', 'C20orf141 (128653)', 'C20orf144 (128864)', 'C20orf173 (140873)', 'C20orf202 (400831)', 'C20orf96 (140680)', 'C21orf140 (101928147)', 'C21orf58 (54058)', 'C21orf91 (54149)', 'C22orf15 (150248)', 'C22orf23 (84645)', 'C22orf31 (25770)', 'C22orf39 (128977)', 'C22orf42 (150297)', 'C2CD2 (25966)', 'C2CD2L (9854)', 'C2CD3 (26005)', 'C2CD4A (145741)', 'C2CD4B (388125)', 'C2CD4C (126567)', 'C2CD4D (100191040)', 'C2CD5 (9847)', 'C2CD6 (151254)', 'C2orf15 (150590)', 'C2orf42 (54980)', 'C2orf49 (79074)', 'C2orf50 (130813)', 'C2orf66 (401027)', 'C2orf68 (388969)', 'C2orf69 (205327)', 'C2orf72 (257407)', 'C2orf73 (129852)', 'C2orf76 (130355)', 'C2orf78 (388960)', 'C2orf80 (389073)', 'C2orf88 (84281)', 'C3 (718)', 'C3AR1 (719)', 'C3orf18 (51161)', 'C3orf20 (84077)', 'C3orf22 (152065)', 'C3orf33 (285315)', 'C3orf38 (285237)', 'C3orf52 (79669)', 'C3orf62 (375341)', 'C3orf70 (285382)', 'C3orf80 (401097)', 'C3orf84 (646498)', 'C4BPA (722)', 'C4BPB (725)', 'C4orf17 (84103)', 'C4orf19 (55286)', 'C4orf3 (401152)', 'C4orf33 (132321)', 'C4orf36 (132989)', 'C4orf46 (201725)', 'C4orf51 (646603)', 'C5 (727)', 'C5AR1 (728)', 'C5AR2 (27202)', 'C5orf15 (56951)', 'C5orf22 (55322)', 'C5orf24 (134553)', 'C5orf34 (375444)', 'C5orf46 (389336)', 'C5orf47 (133491)', 'C5orf52 (100190949)', 'C5orf58 (133874)', 'C5orf63 (401207)', 'C6 (729)', 'C6orf118 (168090)', 'C6orf120 (387263)', 'C6orf132 (647024)', 'C6orf136 (221545)', 'C6orf141 (135398)', 'C6orf15 (29113)', 'C6orf163 (206412)', 'C6orf226 (441150)', 'C6orf47 (57827)', 'C6orf52 (347744)', 'C6orf58 (352999)', 'C6orf62 (81688)', 'C6orf89 (221477)', 'C7 (730)', 'C7orf25 (79020)', 'C7orf33 (202865)', 'C7orf50 (84310)', 'C7orf57 (136288)', 'C8A (731)', 'C8B (732)', 'C8G (733)', 'C8orf33 (65265)', 'C8orf34 (116328)', 'C8orf48 (157773)', 'C8orf58 (541565)', 'C8orf74 (203076)', 'C8orf76 (84933)', 'C8orf82 (414919)', 'C8orf88 (100127983)', 'C9 (735)', 'C9orf152 (401546)', 'C9orf153 (389766)', 'C9orf40 (55071)', 'C9orf43 (257169)', 'C9orf50 (375759)', 'C9orf57 (138240)', 'C9orf72 (203228)', 'C9orf78 (51759)', 'C9orf85 (138241)', 'CA1 (759)', 'CA10 (56934)', 'CA11 (770)', 'CA12 (771)', 'CA13 (377677)', 'CA14 (23632)', 'CA2 (760)', 'CA3 (761)', 'CA4 (762)', 'CA5A (763)', 'CA5B (11238)', 'CA6 (765)', 'CA7 (766)', 'CA8 (767)', 'CA9 (768)', 'CAAP1 (79886)', 'CAB39 (51719)', 'CAB39L (81617)', 'CABCOCO1 (219621)', 'CABIN1 (23523)', 'CABLES1 (91768)', 'CABLES2 (81928)', 'CABP1 (9478)', 'CABP2 (51475)', 'CABP4 (57010)', 'CABP5 (56344)', 'CABP7 (164633)', 'CABS1 (85438)', 'CABYR (26256)', 'CACFD1 (11094)', 'CACHD1 (57685)', 'CACNA1A (773)', 'CACNA1B (774)', 'CACNA1C (775)', 'CACNA1D (776)', 'CACNA1E (777)', 'CACNA1F (778)', 'CACNA1G (8913)', 'CACNA1H (8912)', 'CACNA1I (8911)', 'CACNA1S (779)', 'CACNA2D1 (781)', 'CACNA2D2 (9254)', 'CACNA2D3 (55799)', 'CACNA2D4 (93589)', 'CACNB1 (782)', 'CACNB2 (783)', 'CACNB3 (784)', 'CACNB4 (785)', 'CACNG1 (786)', 'CACNG2 (10369)', 'CACNG3 (10368)', 'CACNG4 (27092)', 'CACNG5 (27091)', 'CACNG6 (59285)', 'CACNG7 (59284)', 'CACNG8 (59283)', 'CACTIN (58509)', 'CACUL1 (143384)', 'CACYBP (27101)', 'CAD (790)', 'CADM1 (23705)', 'CADM2 (253559)', 'CADM3 (57863)', 'CADM4 (199731)', 'CADPS (8618)', 'CADPS2 (93664)', 'CAGE1 (285782)', 'CALB1 (793)', 'CALB2 (794)', 'CALCA (796)', 'CALCB (797)', 'CALCOCO1 (57658)', 'CALCOCO2 (10241)', 'CALCR (799)', 'CALCRL (10203)', 'CALD1 (800)', 'CALHM1 (255022)', 'CALHM2 (51063)', 'CALHM3 (119395)', 'CALHM4 (221301)', 'CALHM5 (254228)', 'CALHM6 (441168)', 'CALM1 (801)', 'CALM2 (805)', 'CALM3 (808)', 'CALML3 (810)', 'CALML4 (91860)', 'CALML5 (51806)', 'CALML6 (163688)', 'CALN1 (83698)', 'CALR (811)', 'CALR3 (125972)', 'CALU (813)', 'CALY (50632)', 'CAMK1 (8536)', 'CAMK1D (57118)', 'CAMK1G (57172)', 'CAMK2A (815)', 'CAMK2B (816)', 'CAMK2D (817)', 'CAMK2G (818)', 'CAMK2N1 (55450)', 'CAMK2N2 (94032)', 'CAMK4 (814)', 'CAMKK1 (84254)', 'CAMKK2 (10645)', 'CAMKMT (79823)', 'CAMKV (79012)', 'CAMLG (819)', 'CAMP (820)', 'CAMSAP1 (157922)', 'CAMSAP2 (23271)', 'CAMSAP3 (57662)', 'CAMTA1 (23261)', 'CAMTA2 (23125)', 'CAND1 (55832)', 'CAND2 (23066)', 'CANT1 (124583)', 'CANX (821)', 'CAP1 (10487)', 'CAP2 (10486)', 'CAPG (822)', 'CAPN1 (823)', 'CAPN10 (11132)', 'CAPN11 (11131)', 'CAPN12 (147968)', 'CAPN13 (92291)', 'CAPN14 (440854)', 'CAPN15 (6650)', 'CAPN2 (824)', 'CAPN3 (825)', 'CAPN5 (726)', 'CAPN6 (827)', 'CAPN7 (23473)', 'CAPN9 (10753)', 'CAPNS1 (826)', 'CAPNS2 (84290)', 'CAPRIN1 (4076)', 'CAPRIN2 (65981)', 'CAPS (828)', 'CAPS2 (84698)', 'CAPSL (133690)', 'CAPZA1 (829)', 'CAPZA2 (830)', 'CAPZA3 (93661)', 'CAPZB (832)', 'CARD10 (29775)', 'CARD11 (84433)', 'CARD14 (79092)', 'CARD16 (114769)', 'CARD18 (59082)', 'CARD19 (84270)', 'CARD6 (84674)', 'CARD8 (22900)', 'CARD9 (64170)', 'CARF (79800)', 'CARHSP1 (23589)', 'CARM1 (10498)', 'CARMIL1 (55604)', 'CARMIL2 (146206)', 'CARMIL3 (90668)', 'CARNMT1 (138199)', 'CARNS1 (57571)', 'CARS1 (833)', 'CARS2 (79587)', 'CARTPT (9607)', 'CASC3 (22794)', 'CASD1 (64921)', 'CASK (8573)', 'CASKIN1 (57524)', 'CASKIN2 (57513)', 'CASP1 (834)', 'CASP10 (843)', 'CASP14 (23581)', 'CASP2 (835)', 'CASP3 (836)', 'CASP4 (837)', 'CASP5 (838)', 'CASP6 (839)', 'CASP7 (840)', 'CASP8 (841)', 'CASP9 (842)', 'CASQ1 (844)', 'CASQ2 (845)', 'CASR (846)', 'CASS4 (57091)', 'CAST (831)', 'CASTOR1 (652968)', 'CASTOR2 (729438)', 'CASZ1 (54897)', 'CAT (847)', 'CATIP (375307)', 'CATSPER1 (117144)', 'CATSPER2 (117155)', 'CATSPER3 (347732)', 'CATSPER4 (378807)', 'CATSPERB (79820)', 'CATSPERD (257062)', 'CATSPERE (257044)', 'CATSPERG (57828)', 'CAV1 (857)', 'CAV2 (858)', 'CAV3 (859)', 'CAVIN1 (284119)', 'CAVIN2 (8436)', 'CAVIN3 (112464)', 'CAVIN4 (347273)', 'CBARP (255057)', 'CBFA2T2 (9139)', 'CBFA2T3 (863)', 'CBFB (865)', 'CBL (867)', 'CBLB (868)', 'CBLC (23624)', 'CBLIF (2694)', 'CBLL1 (79872)', 'CBLL2 (158506)', 'CBLN1 (869)', 'CBLN2 (147381)', 'CBLN3 (643866)', 'CBLN4 (140689)', 'CBR1 (873)', 'CBR3 (874)', 'CBR4 (84869)', 'CBS (875)', 'CBX1 (10951)', 'CBX2 (84733)', 'CBX3 (11335)', 'CBX4 (8535)', 'CBX5 (23468)', 'CBX6 (23466)', 'CBX7 (23492)', 'CBX8 (57332)', 'CBY1 (25776)', 'CBY2 (220082)', 'CBY3 (646019)', 'CC2D1A (54862)', 'CC2D1B (200014)', 'CC2D2A (57545)', 'CC2D2B (387707)', 'CCAR1 (55749)', 'CCAR2 (57805)', 'CCBE1 (147372)', 'CCDC102A (92922)', 'CCDC102B (79839)', 'CCDC103 (388389)', 'CCDC106 (29903)', 'CCDC107 (203260)', 'CCDC110 (256309)', 'CCDC112 (153733)', 'CCDC115 (84317)', 'CCDC116 (164592)', 'CCDC117 (150275)', 'CCDC12 (151903)', 'CCDC120 (90060)', 'CCDC121 (79635)', 'CCDC122 (160857)', 'CCDC124 (115098)', 'CCDC125 (202243)', 'CCDC126 (90693)', 'CCDC127 (133957)', 'CCDC13 (152206)', 'CCDC134 (79879)', 'CCDC136 (64753)', 'CCDC137 (339230)', 'CCDC138 (165055)', 'CCDC14 (64770)', 'CCDC142 (84865)', 'CCDC144A (9720)', 'CCDC146 (57639)', 'CCDC148 (130940)', 'CCDC149 (91050)', 'CCDC15 (80071)', 'CCDC150 (284992)', 'CCDC152 (100129792)', 'CCDC157 (550631)', 'CCDC158 (339965)', 'CCDC159 (126075)', 'CCDC160 (347475)', 'CCDC166 (100130274)', 'CCDC167 (154467)', 'CCDC169 (728591)', 'CCDC17 (149483)', 'CCDC170 (80129)', 'CCDC171 (203238)', 'CCDC172 (374355)', 'CCDC174 (51244)', 'CCDC175 (729665)', 'CCDC177 (56936)', 'CCDC178 (374864)', 'CCDC179 (100500938)', 'CCDC180 (100499483)', 'CCDC181 (57821)', 'CCDC182 (101927581)', 'CCDC183 (84960)', 'CCDC184 (387856)', 'CCDC185 (164127)', 'CCDC186 (55088)', 'CCDC190 (339512)', 'CCDC191 (57577)', 'CCDC198 (55195)', 'CCDC22 (28952)', 'CCDC24 (149473)', 'CCDC25 (55246)', 'CCDC27 (148870)', 'CCDC28A (25901)', 'CCDC28B (79140)', 'CCDC3 (83643)', 'CCDC30 (728621)', 'CCDC32 (90416)', 'CCDC33 (80125)', 'CCDC34 (91057)', 'CCDC38 (120935)', 'CCDC39 (339829)', 'CCDC40 (55036)', 'CCDC42 (146849)', 'CCDC43 (124808)', 'CCDC47 (57003)', 'CCDC50 (152137)', 'CCDC51 (79714)', 'CCDC54 (84692)', 'CCDC59 (29080)', 'CCDC6 (8030)', 'CCDC60 (160777)', 'CCDC61 (729440)', 'CCDC62 (84660)', 'CCDC63 (160762)', 'CCDC65 (85478)', 'CCDC66 (285331)', 'CCDC68 (80323)', 'CCDC69 (26112)', 'CCDC7 (79741)', 'CCDC70 (83446)', 'CCDC71 (64925)', 'CCDC71L (168455)', 'CCDC73 (493860)', 'CCDC74A (90557)', 'CCDC74B (91409)', 'CCDC77 (84318)', 'CCDC78 (124093)', 'CCDC8 (83987)', 'CCDC80 (151887)', 'CCDC81 (60494)', 'CCDC82 (79780)', 'CCDC83 (220047)', 'CCDC85A (114800)', 'CCDC85B (11007)', 'CCDC85C (317762)', 'CCDC86 (79080)', 'CCDC87 (55231)', 'CCDC88A (55704)', 'CCDC88B (283234)', 'CCDC88C (440193)', 'CCDC89 (220388)', 'CCDC9 (26093)', 'CCDC90B (60492)', 'CCDC91 (55297)', 'CCDC92 (80212)', 'CCDC93 (54520)', 'CCDC97 (90324)', 'CCDC9B (388115)', 'CCER1 (196477)', 'CCER2 (643669)', 'CCHCR1 (54535)', 'CCIN (881)', 'CCK (885)', 'CCKAR (886)', 'CCKBR (887)', 'CCL1 (6346)', 'CCL11 (6356)', 'CCL13 (6357)', 'CCL14 (6358)', 'CCL15 (6359)', 'CCL16 (6360)', 'CCL17 (6361)', 'CCL18 (6362)', 'CCL19 (6363)', 'CCL2 (6347)', 'CCL20 (6364)', 'CCL21 (6366)', 'CCL22 (6367)', 'CCL23 (6368)', 'CCL24 (6369)', 'CCL25 (6370)', 'CCL26 (10344)', 'CCL27 (10850)', 'CCL28 (56477)', 'CCL3 (6348)', 'CCL3L3 (414062)', 'CCL4 (6351)', 'CCL5 (6352)', 'CCL7 (6354)', 'CCL8 (6355)', 'CCM2 (83605)', 'CCM2L (140706)', 'CCN1 (3491)', 'CCN2 (1490)', 'CCN3 (4856)', 'CCN4 (8840)', 'CCN5 (8839)', 'CCN6 (8838)', 'CCNA1 (8900)', 'CCNA2 (890)', 'CCNB1 (891)', 'CCNB1IP1 (57820)', 'CCNB2 (9133)', 'CCNB3 (85417)', 'CCNC (892)', 'CCND1 (595)', 'CCND2 (894)', 'CCND3 (896)', 'CCNDBP1 (23582)', 'CCNE1 (898)', 'CCNE2 (9134)', 'CCNF (899)', 'CCNG1 (900)', 'CCNG2 (901)', 'CCNH (902)', 'CCNI (10983)', 'CCNI2 (645121)', 'CCNJ (54619)', 'CCNJL (79616)', 'CCNK (8812)', 'CCNL1 (57018)', 'CCNL2 (81669)', 'CCNO (10309)', 'CCNP (79935)', 'CCNT1 (904)', 'CCNT2 (905)', 'CCNY (219771)', 'CCNYL1 (151195)', 'CCP110 (9738)', 'CCPG1 (9236)', 'CCR1 (1230)', 'CCR10 (2826)', 'CCR2 (729230)', 'CCR3 (1232)', 'CCR4 (1233)', 'CCR5 (1234)', 'CCR6 (1235)', 'CCR7 (1236)', 'CCR8 (1237)', 'CCR9 (10803)', 'CCRL2 (9034)', 'CCS (9973)', 'CCSAP (126731)', 'CCSER1 (401145)', 'CCSER2 (54462)', 'CCT2 (10576)', 'CCT3 (7203)', 'CCT4 (10575)', 'CCT5 (22948)', 'CCT6A (908)', 'CCT6B (10693)', 'CCT7 (10574)', 'CCT8 (10694)', 'CCT8L2 (150160)', 'CD101 (9398)', 'CD109 (135228)', 'CD14 (929)', 'CD151 (977)', 'CD160 (11126)', 'CD163 (9332)', 'CD163L1 (283316)', 'CD164 (8763)', 'CD164L2 (388611)', 'CD177 (57126)', 'CD180 (4064)', 'CD19 (930)', 'CD1A (909)', 'CD1B (910)', 'CD1C (911)', 'CD1D (912)', 'CD1E (913)', 'CD2 (914)', 'CD200 (4345)', 'CD200R1 (131450)', 'CD200R1L (344807)', 'CD209 (30835)', 'CD22 (933)', 'CD226 (10666)', 'CD244 (51744)', 'CD247 (919)', 'CD248 (57124)', 'CD27 (939)', 'CD274 (29126)', 'CD276 (80381)', 'CD28 (940)', 'CD2AP (23607)', 'CD2BP2 (10421)', 'CD300A (11314)', 'CD300C (10871)', 'CD300E (342510)', 'CD300LB (124599)', 'CD300LD (100131439)', 'CD300LF (146722)', 'CD300LG (146894)', 'CD302 (9936)', 'CD320 (51293)', 'CD33 (945)', 'CD34 (947)', 'CD36 (948)', 'CD37 (951)', 'CD38 (952)', 'CD3D (915)', 'CD3E (916)', 'CD3G (917)', 'CD4 (920)', 'CD40 (958)', 'CD40LG (959)', 'CD44 (960)', 'CD46 (4179)', 'CD47 (961)', 'CD48 (962)', 'CD5 (921)', 'CD52 (1043)', 'CD53 (963)', 'CD55 (1604)', 'CD58 (965)', 'CD59 (966)', 'CD5L (922)', 'CD6 (923)', 'CD63 (967)', 'CD68 (968)', 'CD69 (969)', 'CD7 (924)', 'CD70 (970)', 'CD72 (971)', 'CD74 (972)', 'CD79A (973)', 'CD79B (974)', 'CD80 (941)', 'CD81 (975)', 'CD82 (3732)', 'CD83 (9308)', 'CD84 (8832)', 'CD86 (942)', 'CD8A (925)', 'CD9 (928)', 'CD93 (22918)', 'CD96 (10225)', 'CD99L2 (83692)', 'CDA (978)', 'CDADC1 (81602)', 'CDAN1 (146059)', 'CDC123 (8872)', 'CDC14A (8556)', 'CDC14B (8555)', 'CDC16 (8881)', 'CDC20 (991)', 'CDC20B (166979)', 'CDC23 (8697)', 'CDC25A (993)', 'CDC25B (994)', 'CDC25C (995)', 'CDC26 (246184)', 'CDC27 (996)', 'CDC34 (997)', 'CDC37 (11140)', 'CDC37L1 (55664)', 'CDC40 (51362)', 'CDC42 (998)', 'CDC42BPA (8476)', 'CDC42BPB (9578)', 'CDC42BPG (55561)', 'CDC42EP1 (11135)', 'CDC42EP2 (10435)', 'CDC42EP3 (10602)', 'CDC42EP4 (23580)', 'CDC42EP5 (148170)', 'CDC42SE1 (56882)', 'CDC42SE2 (56990)', 'CDC45 (8318)', 'CDC5L (988)', 'CDC6 (990)', 'CDC7 (8317)', 'CDC73 (79577)', 'CDCA2 (157313)', 'CDCA3 (83461)', 'CDCA4 (55038)', 'CDCA5 (113130)', 'CDCA7 (83879)', 'CDCA7L (55536)', 'CDCA8 (55143)', 'CDCP1 (64866)', 'CDCP2 (200008)', 'CDH1 (999)', 'CDH10 (1008)', 'CDH11 (1009)', 'CDH12 (1010)', 'CDH13 (1012)', 'CDH15 (1013)', 'CDH16 (1014)', 'CDH17 (1015)', 'CDH18 (1016)', 'CDH19 (28513)', 'CDH2 (1000)', 'CDH20 (28316)', 'CDH22 (64405)', 'CDH23 (64072)', 'CDH24 (64403)', 'CDH26 (60437)', 'CDH3 (1001)', 'CDH4 (1002)', 'CDH5 (1003)', 'CDH6 (1004)', 'CDH7 (1005)', 'CDH8 (1006)', 'CDH9 (1007)', 'CDHR1 (92211)', 'CDHR2 (54825)', 'CDHR3 (222256)', 'CDHR4 (389118)', 'CDHR5 (53841)', 'CDIN1 (84529)', 'CDIP1 (29965)', 'CDIPT (10423)', 'CDK1 (983)', 'CDK10 (8558)', 'CDK12 (51755)', 'CDK13 (8621)', 'CDK14 (5218)', 'CDK15 (65061)', 'CDK16 (5127)', 'CDK17 (5128)', 'CDK18 (5129)', 'CDK19 (23097)', 'CDK2 (1017)', 'CDK20 (23552)', 'CDK2AP1 (8099)', 'CDK2AP2 (10263)', 'CDK3 (1018)', 'CDK4 (1019)', 'CDK5 (1020)', 'CDK5R1 (8851)', 'CDK5R2 (8941)', 'CDK5RAP1 (51654)', 'CDK5RAP2 (55755)', 'CDK5RAP3 (80279)', 'CDK6 (1021)', 'CDK7 (1022)', 'CDK8 (1024)', 'CDK9 (1025)', 'CDKAL1 (54901)', 'CDKL1 (8814)', 'CDKL2 (8999)', 'CDKL3 (51265)', 'CDKL4 (344387)', 'CDKL5 (6792)', 'CDKN1A (1026)', 'CDKN1B (1027)', 'CDKN1C (1028)', 'CDKN2A (1029)', 'CDKN2AIP (55602)', 'CDKN2AIPNL (91368)', 'CDKN2B (1030)', 'CDKN2C (1031)', 'CDKN2D (1032)', 'CDKN3 (1033)', 'CDNF (441549)', 'CDO1 (1036)', 'CDON (50937)', 'CDPF1 (150383)', 'CDR2 (1039)', 'CDR2L (30850)', 'CDRT15 (146822)', 'CDRT15L2 (256223)', 'CDS1 (1040)', 'CDS2 (8760)', 'CDSN (1041)', 'CDT1 (81620)', 'CDV3 (55573)', 'CDX1 (1044)', 'CDX2 (1045)', 'CDX4 (1046)', 'CDYL (9425)', 'CDYL2 (124359)', 'CEACAM1 (634)', 'CEACAM16 (388551)', 'CEACAM19 (56971)', 'CEACAM21 (90273)', 'CEACAM3 (1084)', 'CEACAM4 (1089)', 'CEACAM5 (1048)', 'CEACAM6 (4680)', 'CEACAM7 (1087)', 'CEACAM8 (1088)', 'CEBPA (1050)', 'CEBPB (1051)', 'CEBPD (1052)', 'CEBPE (1053)', 'CEBPG (1054)', 'CEBPZ (10153)', 'CEL (1056)', 'CELA1 (1990)', 'CELA2A (63036)', 'CELA2B (51032)', 'CELA3A (10136)', 'CELA3B (23436)', 'CELF1 (10658)', 'CELF2 (10659)', 'CELF3 (11189)', 'CELF4 (56853)', 'CELF5 (60680)', 'CELF6 (60677)', 'CELSR1 (9620)', 'CELSR2 (1952)', 'CELSR3 (1951)', 'CEMIP (57214)', 'CEMIP2 (23670)', 'CEMP1 (752014)', 'CENATAC (338657)', 'CEND1 (51286)', 'CENPA (1058)', 'CENPB (1059)', 'CENPC (1060)', 'CENPE (1062)', 'CENPF (1063)', 'CENPH (64946)', 'CENPI (2491)', 'CENPJ (55835)', 'CENPK (64105)', 'CENPL (91687)', 'CENPM (79019)', 'CENPN (55839)', 'CENPO (79172)', 'CENPP (401541)', 'CENPQ (55166)', 'CENPT (80152)', 'CENPU (79682)', 'CENPV (201161)', 'CENPW (387103)', 'CENPX (201254)', 'CEP104 (9731)', 'CEP112 (201134)', 'CEP120 (153241)', 'CEP126 (57562)', 'CEP128 (145508)', 'CEP131 (22994)', 'CEP135 (9662)', 'CEP15 (57415)', 'CEP152 (22995)', 'CEP162 (22832)', 'CEP164 (22897)', 'CEP170 (9859)', 'CEP170B (283638)', 'CEP19 (84984)', 'CEP192 (55125)', 'CEP20 (123811)', 'CEP250 (11190)', 'CEP290 (80184)', 'CEP295 (85459)', 'CEP295NL (100653515)', 'CEP350 (9857)', 'CEP41 (95681)', 'CEP43 (11116)', 'CEP44 (80817)', 'CEP55 (55165)', 'CEP57 (9702)', 'CEP57L1 (285753)', 'CEP63 (80254)', 'CEP68 (23177)', 'CEP70 (80321)', 'CEP72 (55722)', 'CEP76 (79959)', 'CEP78 (84131)', 'CEP83 (51134)', 'CEP85 (64793)', 'CEP85L (387119)', 'CEP89 (84902)', 'CEP95 (90799)', 'CEP97 (79598)', 'CEPT1 (10390)', 'CER1 (9350)', 'CERCAM (51148)', 'CERK (64781)', 'CERKL (375298)', 'CERS1 (10715)', 'CERS2 (29956)', 'CERS3 (204219)', 'CERS4 (79603)', 'CERS5 (91012)', 'CERS6 (253782)', 'CERT1 (10087)', 'CES1 (1066)', 'CES2 (8824)', 'CES3 (23491)', 'CES4A (283848)', 'CES5A (221223)', 'CETN1 (1068)', 'CETN2 (1069)', 'CETN3 (1070)', 'CETP (1071)', 'CFAP100 (348807)', 'CFAP107 (93190)', 'CFAP119 (90835)', 'CFAP126 (257177)', 'CFAP141 (388701)', 'CFAP144 (440585)', 'CFAP157 (286207)', 'CFAP161 (161502)', 'CFAP184 (257236)', 'CFAP20 (29105)', 'CFAP206 (154313)', 'CFAP20DC (200844)', 'CFAP210 (129881)', 'CFAP221 (200373)', 'CFAP251 (144406)', 'CFAP263 (29070)', 'CFAP276 (127003)', 'CFAP298 (56683)', 'CFAP299 (255119)', 'CFAP300 (85016)', 'CFAP36 (112942)', 'CFAP410 (755)', 'CFAP418 (157657)', 'CFAP43 (80217)', 'CFAP44 (55779)', 'CFAP45 (25790)', 'CFAP46 (54777)', 'CFAP47 (286464)', 'CFAP52 (146845)', 'CFAP53 (220136)', 'CFAP57 (149465)', 'CFAP58 (159686)', 'CFAP61 (26074)', 'CFAP65 (255101)', 'CFAP68 (64776)', 'CFAP69 (79846)', 'CFAP70 (118491)', 'CFAP73 (387885)', 'CFAP77 (389799)', 'CFAP90 (134121)', 'CFAP91 (89876)', 'CFAP92 (57501)', 'CFAP95 (138255)', 'CFAP96 (441054)', 'CFAP97 (57587)', 'CFAP97D1 (284067)', 'CFB (629)', 'CFC1 (55997)', 'CFC1B (653275)', 'CFD (1675)', 'CFDP1 (10428)', 'CFH (3075)', 'CFHR1 (3078)', 'CFHR2 (3080)', 'CFHR3 (10878)', 'CFHR4 (10877)', 'CFHR5 (81494)', 'CFI (3426)', 'CFL1 (1072)', 'CFL2 (1073)', 'CFLAR (8837)', 'CFP (5199)', 'CFTR (1080)', 'CGA (1081)', 'CGAS (115004)', 'CGB7 (94027)', 'CGGBP1 (8545)', 'CGN (57530)', 'CGNL1 (84952)', 'CGREF1 (10669)', 'CGRRF1 (10668)', 'CH25H (9023)', 'CHAC1 (79094)', 'CHAC2 (494143)', 'CHAD (1101)', 'CHADL (150356)', 'CHAF1A (10036)', 'CHAF1B (8208)', 'CHAMP1 (283489)', 'CHAT (1103)', 'CHCHD1 (118487)', 'CHCHD10 (400916)', 'CHCHD2 (51142)', 'CHCHD3 (54927)', 'CHCHD4 (131474)', 'CHCHD5 (84269)', 'CHCHD6 (84303)', 'CHCHD7 (79145)', 'CHCT1 (124773)', 'CHD1 (1105)', 'CHD1L (9557)', 'CHD2 (1106)', 'CHD3 (1107)', 'CHD4 (1108)', 'CHD5 (26038)', 'CHD6 (84181)', 'CHD7 (55636)', 'CHD8 (57680)', 'CHD9 (80205)', 'CHDH (55349)', 'CHEK1 (1111)', 'CHEK2 (11200)', 'CHERP (10523)', 'CHFR (55743)', 'CHGA (1113)', 'CHGB (1114)', 'CHI3L1 (1116)', 'CHI3L2 (1117)', 'CHIA (27159)', 'CHIC1 (53344)', 'CHIC2 (26511)', 'CHID1 (66005)', 'CHIT1 (1118)', 'CHKA (1119)', 'CHKB (1120)', 'CHL1 (10752)', 'CHM (1121)', 'CHML (1122)', 'CHMP1A (5119)', 'CHMP1B (57132)', 'CHMP2A (27243)', 'CHMP2B (25978)', 'CHMP3 (51652)', 'CHMP4A (29082)', 'CHMP4B (128866)', 'CHMP4C (92421)', 'CHMP5 (51510)', 'CHMP6 (79643)', 'CHMP7 (91782)', 'CHN1 (1123)', 'CHN2 (1124)', 'CHODL (140578)', 'CHORDC1 (26973)', 'CHP1 (11261)', 'CHP2 (63928)', 'CHPF (79586)', 'CHPF2 (54480)', 'CHPT1 (56994)', 'CHRAC1 (54108)', 'CHRD (8646)', 'CHRDL1 (91851)', 'CHRDL2 (25884)', 'CHRFAM7A (89832)', 'CHRM1 (1128)', 'CHRM2 (1129)', 'CHRM3 (1131)', 'CHRM4 (1132)', 'CHRM5 (1133)', 'CHRNA1 (1134)', 'CHRNA10 (57053)', 'CHRNA2 (1135)', 'CHRNA3 (1136)', 'CHRNA4 (1137)', 'CHRNA5 (1138)', 'CHRNA6 (8973)', 'CHRNA7 (1139)', 'CHRNA9 (55584)', 'CHRNB1 (1140)', 'CHRNB2 (1141)', 'CHRNB3 (1142)', 'CHRNB4 (1143)', 'CHRND (1144)', 'CHRNE (1145)', 'CHRNG (1146)', 'CHST1 (8534)', 'CHST10 (9486)', 'CHST11 (50515)', 'CHST12 (55501)', 'CHST13 (166012)', 'CHST14 (113189)', 'CHST15 (51363)', 'CHST2 (9435)', 'CHST3 (9469)', 'CHST4 (10164)', 'CHST5 (23563)', 'CHST6 (4166)', 'CHST7 (56548)', 'CHST8 (64377)', 'CHST9 (83539)', 'CHSY1 (22856)', 'CHSY3 (337876)', 'CHTF18 (63922)', 'CHTF8 (54921)', 'CHTOP (26097)', 'CHUK (1147)', 'CHURC1 (91612)', 'CIAO1 (9391)', 'CIAO2A (84191)', 'CIAO2B (51647)', 'CIAO3 (64428)', 'CIAPIN1 (57019)', 'CIART (148523)', 'CIB1 (10519)', 'CIB2 (10518)', 'CIB3 (117286)', 'CIB4 (130106)', 'CIBAR1 (137392)', 'CIBAR2 (339145)', 'CIC (23152)', 'CIDEA (1149)', 'CIDEB (27141)', 'CIDEC (63924)', 'CIITA (4261)', 'CILK1 (22858)', 'CILP (8483)', 'CILP2 (148113)', 'CIMAP1A (113746)', 'CIMAP1B (440836)', 'CIMAP1C (161753)', 'CIMAP1D (284451)', 'CIMAP2 (163747)', 'CIMAP3 (128344)', 'CIMIP1 (128602)', 'CIMIP2A (401565)', 'CIMIP2B (730112)', 'CIMIP2C (339778)', 'CIMIP4 (339669)', 'CINP (51550)', 'CIP2A (57650)', 'CIPC (85457)', 'CIR1 (9541)', 'CIRBP (1153)', 'CISD1 (55847)', 'CISD2 (493856)', 'CISD3 (284106)', 'CISH (1154)', 'CIT (11113)', 'CITED1 (4435)', 'CITED2 (10370)', 'CITED4 (163732)', 'CIZ1 (25792)', 'CKAP2 (26586)', 'CKAP2L (150468)', 'CKAP4 (10970)', 'CKAP5 (9793)', 'CKB (1152)', 'CKLF (51192)', 'CKM (1158)', 'CKMT2 (1160)', 'CKS1B (1163)', 'CKS2 (1164)', 'CLASRP (11129)', 'CLBA1 (122616)', 'CLC (1178)', 'CLCA1 (1179)', 'CLCA2 (9635)', 'CLCA4 (22802)', 'CLCC1 (23155)', 'CLCF1 (23529)', 'CLCN1 (1180)', 'CLCN2 (1181)', 'CLCN3 (1182)', 'CLCN4 (1183)', 'CLCN5 (1184)', 'CLCN6 (1185)', 'CLCN7 (1186)', 'CLCNKA (1187)', 'CLCNKB (1188)', 'CLDN1 (9076)', 'CLDN10 (9071)', 'CLDN11 (5010)', 'CLDN12 (9069)', 'CLDN14 (23562)', 'CLDN15 (24146)', 'CLDN16 (10686)', 'CLDN17 (26285)', 'CLDN18 (51208)', 'CLDN19 (149461)', 'CLDN2 (9075)', 'CLDN20 (49861)', 'CLDN22 (53842)', 'CLDN23 (137075)', 'CLDN25 (644672)', 'CLDN3 (1365)', 'CLDN4 (1364)', 'CLDN5 (7122)', 'CLDN6 (9074)', 'CLDN7 (1366)', 'CLDN8 (9073)', 'CLDN9 (9080)', 'CLDND1 (56650)', 'CLDND2 (125875)', 'CLEC10A (10462)', 'CLEC11A (6320)', 'CLEC12A (160364)', 'CLEC12B (387837)', 'CLEC14A (161198)', 'CLEC16A (23274)', 'CLEC17A (388512)', 'CLEC18A (348174)', 'CLEC18B (497190)', 'CLEC1A (51267)', 'CLEC1B (51266)', 'CLEC2A (387836)', 'CLEC2B (9976)', 'CLEC2D (29121)', 'CLEC2L (154790)', 'CLEC3A (10143)', 'CLEC3B (7123)', 'CLEC4A (50856)', 'CLEC4C (170482)', 'CLEC4D (338339)', 'CLEC4E (26253)', 'CLEC4F (165530)', 'CLEC4G (339390)', 'CLEC4M (10332)', 'CLEC5A (23601)', 'CLEC6A (93978)', 'CLEC7A (64581)', 'CLEC9A (283420)', 'CLGN (1047)', 'CLHC1 (130162)', 'CLIC1 (1192)', 'CLIC2 (1193)', 'CLIC3 (9022)', 'CLIC4 (25932)', 'CLIC5 (53405)', 'CLIC6 (54102)', 'CLINT1 (9685)', 'CLIP1 (6249)', 'CLIP2 (7461)', 'CLIP3 (25999)', 'CLIP4 (79745)', 'CLK1 (1195)', 'CLK2 (1196)', 'CLK3 (1198)', 'CLK4 (57396)', 'CLMN (79789)', 'CLMP (79827)', 'CLN3 (1201)', 'CLN5 (1203)', 'CLN6 (54982)', 'CLN8 (2055)', 'CLNK (116449)', 'CLNS1A (1207)', 'CLOCK (9575)', 'CLP1 (10978)', 'CLPB (81570)', 'CLPP (8192)', 'CLPS (1208)', 'CLPSL1 (340204)', 'CLPSL2 (389383)', 'CLPTM1 (1209)', 'CLPTM1L (81037)', 'CLPX (10845)', 'CLRN1 (7401)', 'CLRN2 (645104)', 'CLRN3 (119467)', 'CLSPN (63967)', 'CLSTN1 (22883)', 'CLSTN2 (64084)', 'CLSTN3 (9746)', 'CLTA (1211)', 'CLTB (1212)', 'CLTC (1213)', 'CLTCL1 (8218)', 'CLTRN (57393)', 'CLU (1191)', 'CLUAP1 (23059)', 'CLUH (23277)', 'CLUL1 (27098)', 'CLVS1 (157807)', 'CLVS2 (134829)', 'CLXN (79645)', 'CLYBL (171425)', 'CMA1 (1215)', 'CMAS (55907)', 'CMBL (134147)', 'CMC1 (152100)', 'CMC2 (56942)', 'CMC4 (100272147)', 'CMIP (80790)', 'CMKLR1 (1240)', 'CMKLR2 (2825)', 'CMPK1 (51727)', 'CMPK2 (129607)', 'CMSS1 (84319)', 'CMTM1 (113540)', 'CMTM2 (146225)', 'CMTM3 (123920)', 'CMTM4 (146223)', 'CMTM5 (116173)', 'CMTM6 (54918)', 'CMTM7 (112616)', 'CMTM8 (152189)', 'CMTR1 (23070)', 'CMTR2 (55783)', 'CMYA5 (202333)', 'CNBD1 (168975)', 'CNBD2 (140894)', 'CNBP (7555)', 'CNDP1 (84735)', 'CNDP2 (55748)', 'CNEP1R1 (255919)', 'CNFN (84518)', 'CNGA1 (1259)', 'CNGA2 (1260)', 'CNGA3 (1261)', 'CNGA4 (1262)', 'CNGB1 (1258)', 'CNGB3 (54714)', 'CNIH1 (10175)', 'CNIH2 (254263)', 'CNIH3 (149111)', 'CNIH4 (29097)', 'CNKSR1 (10256)', 'CNKSR2 (22866)', 'CNKSR3 (154043)', 'CNMD (11061)', 'CNN1 (1264)', 'CNN2 (1265)', 'CNN3 (1266)', 'CNNM1 (26507)', 'CNNM2 (54805)', 'CNNM3 (26505)', 'CNNM4 (26504)', 'CNOT1 (23019)', 'CNOT10 (25904)', 'CNOT11 (55571)', 'CNOT2 (4848)', 'CNOT3 (4849)', 'CNOT4 (4850)', 'CNOT6 (57472)', 'CNOT6L (246175)', 'CNOT7 (29883)', 'CNOT8 (9337)', 'CNOT9 (9125)', 'CNP (1267)', 'CNPPD1 (27013)', 'CNPY1 (285888)', 'CNPY2 (10330)', 'CNPY3 (10695)', 'CNPY4 (245812)', 'CNR1 (1268)', 'CNR2 (1269)', 'CNRIP1 (25927)', 'CNST (163882)', 'CNTD1 (124817)', 'CNTF (1270)', 'CNTFR (1271)', 'CNTLN (54875)', 'CNTN1 (1272)', 'CNTN2 (6900)', 'CNTN3 (5067)', 'CNTN4 (152330)', 'CNTN5 (53942)', 'CNTN6 (27255)', 'CNTNAP1 (8506)', 'CNTNAP2 (26047)', 'CNTNAP3 (79937)', 'CNTNAP3B (728577)', 'CNTNAP4 (85445)', 'CNTNAP5 (129684)', 'CNTRL (11064)', 'CNTROB (116840)', 'COA1 (55744)', 'COA3 (28958)', 'COA4 (51287)', 'COA5 (493753)', 'COA6 (388753)', 'COA7 (65260)', 'COA8 (84334)', 'COASY (80347)', 'COBL (23242)', 'COBLL1 (22837)', 'COCH (1690)', 'COG1 (9382)', 'COG2 (22796)', 'COG3 (83548)', 'COG4 (25839)', 'COG5 (10466)', 'COG6 (57511)', 'COG7 (91949)', 'COG8 (84342)', 'COIL (8161)', 'COL10A1 (1300)', 'COL11A1 (1301)', 'COL11A2 (1302)', 'COL12A1 (1303)', 'COL13A1 (1305)', 'COL14A1 (7373)', 'COL15A1 (1306)', 'COL16A1 (1307)', 'COL17A1 (1308)', 'COL18A1 (80781)', 'COL19A1 (1310)', 'COL1A1 (1277)', 'COL1A2 (1278)', 'COL20A1 (57642)', 'COL21A1 (81578)', 'COL22A1 (169044)', 'COL23A1 (91522)', 'COL24A1 (255631)', 'COL25A1 (84570)', 'COL26A1 (136227)', 'COL27A1 (85301)', 'COL28A1 (340267)', 'COL2A1 (1280)', 'COL3A1 (1281)', 'COL4A1 (1282)', 'COL4A2 (1284)', 'COL4A3 (1285)', 'COL4A4 (1286)', 'COL4A5 (1287)', 'COL4A6 (1288)', 'COL5A1 (1289)', 'COL5A2 (1290)', 'COL5A3 (50509)', 'COL6A1 (1291)', 'COL6A2 (1292)', 'COL6A3 (1293)', 'COL6A6 (131873)', 'COL7A1 (1294)', 'COL8A1 (1295)', 'COL8A2 (1296)', 'COL9A1 (1297)', 'COL9A2 (1298)', 'COL9A3 (1299)', 'COLEC10 (10584)', 'COLEC11 (78989)', 'COLEC12 (81035)', 'COLGALT1 (79709)', 'COLGALT2 (23127)', 'COLQ (8292)', 'COMMD1 (150684)', 'COMMD10 (51397)', 'COMMD2 (51122)', 'COMMD3 (23412)', 'COMMD4 (54939)', 'COMMD5 (28991)', 'COMMD6 (170622)', 'COMMD7 (149951)', 'COMMD8 (54951)', 'COMMD9 (29099)', 'COMP (1311)', 'COMT (1312)', 'COMTD1 (118881)', 'COP1 (64326)', 'COPA (1314)', 'COPB1 (1315)', 'COPB2 (9276)', 'COPE (11316)', 'COPG1 (22820)', 'COPRS (55352)', 'COPS2 (9318)', 'COPS3 (8533)', 'COPS4 (51138)', 'COPS5 (10987)', 'COPS6 (10980)', 'COPS7A (50813)', 'COPS7B (64708)', 'COPS8 (10920)', 'COPS9 (150678)', 'COPZ1 (22818)', 'COQ10A (93058)', 'COQ10B (80219)', 'COQ2 (27235)', 'COQ3 (51805)', 'COQ4 (51117)', 'COQ5 (84274)', 'COQ6 (51004)', 'COQ7 (10229)', 'COQ8A (56997)', 'COQ8B (79934)', 'COQ9 (57017)', 'CORIN (10699)', 'CORO1A (11151)', 'CORO1B (57175)', 'CORO1C (23603)', 'CORO2A (7464)', 'CORO2B (10391)', 'CORO6 (84940)', 'CORO7 (79585)', 'CORT (1325)', 'COTL1 (23406)', 'COX10 (1352)', 'COX11 (1353)', 'COX14 (84987)', 'COX15 (1355)', 'COX16 (51241)', 'COX17 (10063)', 'COX18 (285521)', 'COX19 (90639)', 'COX20 (116228)', 'COX4I1 (1327)', 'COX4I2 (84701)', 'COX5A (9377)', 'COX5B (1329)', 'COX6A1 (1337)', 'COX6A2 (1339)', 'COX6B1 (1340)', 'COX6B2 (125965)', 'COX6C (1345)', 'COX7A1 (1346)', 'COX7A2 (1347)', 'COX7A2L (9167)', 'COX7B (1349)', 'COX7B2 (170712)', 'COX7C (1350)', 'COX8A (1351)', 'COX8C (341947)', 'CP (1356)', 'CPA1 (1357)', 'CPA2 (1358)', 'CPA3 (1359)', 'CPA4 (51200)', 'CPA5 (93979)', 'CPA6 (57094)', 'CPAMD8 (27151)', 'CPB1 (1360)', 'CPB2 (1361)', 'CPD (1362)', 'CPE (1363)', 'CPEB1 (64506)', 'CPEB2 (132864)', 'CPEB3 (22849)', 'CPEB4 (80315)', 'CPED1 (79974)', 'CPLANE1 (65250)', 'CPLANE2 (79363)', 'CPLX1 (10815)', 'CPLX2 (10814)', 'CPLX3 (594855)', 'CPLX4 (339302)', 'CPM (1368)', 'CPN1 (1369)', 'CPN2 (1370)', 'CPNE1 (8904)', 'CPNE2 (221184)', 'CPNE3 (8895)', 'CPNE4 (131034)', 'CPNE5 (57699)', 'CPNE6 (9362)', 'CPNE7 (27132)', 'CPNE8 (144402)', 'CPNE9 (151835)', 'CPO (130749)', 'CPOX (1371)', 'CPPED1 (55313)', 'CPQ (10404)', 'CPS1 (1373)', 'CPSF1 (29894)', 'CPSF2 (53981)', 'CPSF3 (51692)', 'CPSF4 (10898)', 'CPSF4L (642843)', 'CPSF6 (11052)', 'CPSF7 (79869)', 'CPT1A (1374)', 'CPT1B (1375)', 'CPT1C (126129)', 'CPT2 (1376)', 'CPTP (80772)', 'CPVL (54504)', 'CPXCR1 (53336)', 'CPXM1 (56265)', 'CPXM2 (119587)', 'CPZ (8532)', 'CR1 (1378)', 'CR1L (1379)', 'CR2 (1380)', 'CRABP1 (1381)', 'CRABP2 (1382)', 'CRACD (57482)', 'CRACDL (343990)', 'CRACR2A (84766)', 'CRACR2B (283229)', 'CRADD (8738)', 'CRAMP1 (57585)', 'CRAT (1384)', 'CRB1 (23418)', 'CRB2 (286204)', 'CRB3 (92359)', 'CRBN (51185)', 'CRCP (27297)', 'CRCT1 (54544)', 'CREB1 (1385)', 'CREB3 (10488)', 'CREB3L1 (90993)', 'CREB3L2 (64764)', 'CREB3L3 (84699)', 'CREB3L4 (148327)', 'CREB5 (9586)', 'CREBBP (1387)', 'CREBL2 (1389)', 'CREBRF (153222)', 'CREBZF (58487)', 'CREG1 (8804)', 'CREG2 (200407)', 'CRELD1 (78987)', 'CRELD2 (79174)', 'CREM (1390)', 'CRH (1392)', 'CRHBP (1393)', 'CRHR1 (1394)', 'CRHR2 (1395)', 'CRIM1 (51232)', 'CRIP1 (1396)', 'CRIP2 (1397)', 'CRIP3 (401262)', 'CRIPT (9419)', 'CRIPTO (6997)', 'CRISP1 (167)', 'CRISP2 (7180)', 'CRISP3 (10321)', 'CRISPLD1 (83690)', 'CRISPLD2 (83716)', 'CRK (1398)', 'CRKL (1399)', 'CRLF1 (9244)', 'CRLF3 (51379)', 'CRLS1 (54675)', 'CRMP1 (1400)', 'CRNKL1 (51340)', 'CRNN (49860)', 'CROCC (9696)', 'CROT (54677)', 'CRP (1401)', 'CRTAC1 (55118)', 'CRTAM (56253)', 'CRTAP (10491)', 'CRTC1 (23373)', 'CRTC2 (200186)', 'CRTC3 (64784)', 'CRX (1406)', 'CRY1 (1407)', 'CRY2 (1408)', 'CRYAA (1409)', 'CRYAB (1410)', 'CRYBA1 (1411)', 'CRYBA2 (1412)', 'CRYBA4 (1413)', 'CRYBB1 (1414)', 'CRYBB2 (1415)', 'CRYBB3 (1417)', 'CRYBG1 (202)', 'CRYGA (1418)', 'CRYGB (1419)', 'CRYGC (1420)', 'CRYGD (1421)', 'CRYGN (155051)', 'CRYGS (1427)', 'CRYL1 (51084)', 'CRYM (1428)', 'CRYZ (1429)', 'CRYZL1 (9946)', 'CS (1431)', 'CSAD (51380)', 'CSAG1 (158511)', 'CSDC2 (27254)', 'CSDE1 (7812)', 'CSE1L (1434)', 'CSF1 (1435)', 'CSF1R (1436)', 'CSF2 (1437)', 'CSF2RB (1439)', 'CSF3 (1440)', 'CSF3R (1441)', 'CSGALNACT1 (55790)', 'CSGALNACT2 (55454)', 'CSH2 (1443)', 'CSHL1 (1444)', 'CSK (1445)', 'CSKMT (751071)', 'CSMD1 (64478)', 'CSMD2 (114784)', 'CSMD3 (114788)', 'CSN1S1 (1446)', 'CSN2 (1447)', 'CSN3 (1448)', 'CSNK1A1 (1452)', 'CSNK1A1L (122011)', 'CSNK1D (1453)', 'CSNK1G1 (53944)', 'CSNK1G2 (1455)', 'CSNK1G3 (1456)', 'CSNK2A1 (1457)', 'CSNK2A2 (1459)', 'CSNK2A3 (283106)', 'CSNK2B (1460)', 'CSPG4 (1464)', 'CSPG5 (10675)', 'CSPP1 (79848)', 'CSRNP1 (64651)', 'CSRNP2 (81566)', 'CSRNP3 (80034)', 'CSRP1 (1465)', 'CSRP2 (1466)', 'CSRP3 (8048)', 'CST1 (1469)', 'CST11 (140880)', 'CST2 (1470)', 'CST3 (1471)', 'CST4 (1472)', 'CST5 (1473)', 'CST6 (1474)', 'CST7 (8530)', 'CST8 (10047)', 'CST9 (128822)', 'CST9L (128821)', 'CSTA (1475)', 'CSTB (1476)', 'CSTF1 (1477)', 'CSTF2 (1478)', 'CSTF2T (23283)', 'CSTF3 (1479)', 'CSTL1 (128817)', 'CSTPP1 (79096)', 'CT47B1 (643311)', 'CT55 (54967)', 'CT83 (203413)', 'CTAG2 (30848)', 'CTAGE1 (64693)', 'CTAGE9 (643854)', 'CTBP1 (1487)', 'CTBP2 (1488)', 'CTBS (1486)', 'CTC1 (80169)', 'CTCF (10664)', 'CTCFL (140690)', 'CTDNEP1 (23399)', 'CTDP1 (9150)', 'CTDSP1 (58190)', 'CTDSP2 (10106)', 'CTDSPL (10217)', 'CTDSPL2 (51496)', 'CTF1 (1489)', 'CTH (1491)', 'CTHRC1 (115908)', 'CTIF (9811)', 'CTLA4 (1493)', 'CTNNA1 (1495)', 'CTNNA2 (1496)', 'CTNNA3 (29119)', 'CTNNAL1 (8727)', 'CTNNB1 (1499)', 'CTNNBIP1 (56998)', 'CTNNBL1 (56259)', 'CTNND1 (1500)', 'CTNND2 (1501)', 'CTNS (1497)', 'CTPS1 (1503)', 'CTPS2 (56474)', 'CTR9 (9646)', 'CTRB1 (1504)', 'CTRB2 (440387)', 'CTRC (11330)', 'CTRL (1506)', 'CTSA (5476)', 'CTSB (1508)', 'CTSC (1075)', 'CTSD (1509)', 'CTSE (1510)', 'CTSF (8722)', 'CTSG (1511)', 'CTSH (1512)', 'CTSK (1513)', 'CTSL (1514)', 'CTSO (1519)', 'CTSS (1520)', 'CTSV (1515)', 'CTSW (1521)', 'CTSZ (1522)', 'CTTN (2017)', 'CTTNBP2 (83992)', 'CTTNBP2NL (55917)', 'CTU1 (90353)', 'CTU2 (348180)', 'CTXN1 (404217)', 'CTXN2 (399697)', 'CTXN3 (613212)', 'CUBN (8029)', 'CUEDC1 (404093)', 'CUEDC2 (79004)', 'CUL1 (8454)', 'CUL2 (8453)', 'CUL3 (8452)', 'CUL4A (8451)', 'CUL4B (8450)', 'CUL5 (8065)', 'CUL7 (9820)', 'CUL9 (23113)', 'CUTA (51596)', 'CUTC (51076)', 'CUX1 (1523)', 'CUX2 (23316)', 'CUZD1 (50624)', 'CWC22 (57703)', 'CWC25 (54883)', 'CWC27 (10283)', 'CWF19L1 (55280)', 'CWF19L2 (143884)', 'CWH43 (80157)', 'CX3CL1 (6376)', 'CX3CR1 (1524)', 'CXADR (1525)', 'CXCL1 (2919)', 'CXCL10 (3627)', 'CXCL11 (6373)', 'CXCL12 (6387)', 'CXCL13 (10563)', 'CXCL14 (9547)', 'CXCL16 (58191)', 'CXCL17 (284340)', 'CXCL2 (2920)', 'CXCL3 (2921)', 'CXCL5 (6374)', 'CXCL6 (6372)', 'CXCL8 (3576)', 'CXCL9 (4283)', 'CXCR1 (3577)', 'CXCR2 (3579)', 'CXCR3 (2833)', 'CXCR4 (7852)', 'CXCR5 (643)', 'CXCR6 (10663)', 'CXXC1 (30827)', 'CXXC4 (80319)', 'CXXC5 (51523)', 'CXorf38 (159013)', 'CXorf58 (254158)', 'CXorf65 (158830)', 'CXorf66 (347487)', 'CYB561 (1534)', 'CYB561A3 (220002)', 'CYB561D1 (284613)', 'CYB561D2 (11068)', 'CYB5A (1528)', 'CYB5B (80777)', 'CYB5D1 (124637)', 'CYB5D2 (124936)', 'CYB5R1 (51706)', 'CYB5R2 (51700)', 'CYB5R3 (1727)', 'CYB5R4 (51167)', 'CYB5RL (606495)', 'CYBA (1535)', 'CYBB (1536)', 'CYBC1 (79415)', 'CYBRD1 (79901)', 'CYC1 (1537)', 'CYCS (54205)', 'CYFIP1 (23191)', 'CYGB (114757)', 'CYLC1 (1538)', 'CYLC2 (1539)', 'CYLD (1540)', 'CYP11A1 (1583)', 'CYP11B1 (1584)', 'CYP11B2 (1585)', 'CYP17A1 (1586)', 'CYP19A1 (1588)', 'CYP1A1 (1543)', 'CYP1A2 (1544)', 'CYP1B1 (1545)', 'CYP20A1 (57404)', 'CYP21A2 (1589)', 'CYP24A1 (1591)', 'CYP26A1 (1592)', 'CYP26B1 (56603)', 'CYP26C1 (340665)', 'CYP27A1 (1593)', 'CYP27B1 (1594)', 'CYP27C1 (339761)', 'CYP2A13 (1553)', 'CYP2A6 (1548)', 'CYP2A7 (1549)', 'CYP2B6 (1555)', 'CYP2C18 (1562)', 'CYP2C19 (1557)', 'CYP2C8 (1558)', 'CYP2C9 (1559)', 'CYP2D6 (1565)', 'CYP2E1 (1571)', 'CYP2F1 (1572)', 'CYP2J2 (1573)', 'CYP2R1 (120227)', 'CYP2S1 (29785)', 'CYP2U1 (113612)', 'CYP2W1 (54905)', 'CYP39A1 (51302)', 'CYP3A4 (1576)', 'CYP3A43 (64816)', 'CYP3A5 (1577)', 'CYP3A7 (1551)', 'CYP46A1 (10858)', 'CYP4A11 (1579)', 'CYP4A22 (284541)', 'CYP4B1 (1580)', 'CYP4F11 (57834)', 'CYP4F12 (66002)', 'CYP4F2 (8529)', 'CYP4F22 (126410)', 'CYP4F3 (4051)', 'CYP4V2 (285440)', 'CYP4X1 (260293)', 'CYP4Z1 (199974)', 'CYP51A1 (1595)', 'CYP7A1 (1581)', 'CYP7B1 (9420)', 'CYP8B1 (1582)', 'CYREN (78996)', 'CYRIA (81553)', 'CYRIB (51571)', 'CYS1 (192668)', 'CYSLTR1 (10800)', 'CYSLTR2 (57105)', 'CYSRT1 (375791)', 'CYSTM1 (84418)', 'CYTH1 (9267)', 'CYTH2 (9266)', 'CYTH3 (9265)', 'CYTH4 (27128)', 'CYTIP (9595)', 'CYTL1 (54360)', 'CYYR1 (116159)', 'CZIB (54987)', 'D2HGDH (728294)', 'DAAM1 (23002)', 'DAAM2 (23500)', 'DAB1 (1600)', 'DAB2 (1601)', 'DAB2IP (153090)', 'DACH1 (1602)', 'DACH2 (117154)', 'DACT1 (51339)', 'DACT2 (168002)', 'DACT3 (147906)', 'DAD1 (1603)', 'DAG1 (1605)', 'DAGLA (747)', 'DAGLB (221955)', 'DALRD3 (55152)', 'DAND5 (199699)', 'DAO (1610)', 'DAOA (267012)', 'DAP (1611)', 'DAP3 (7818)', 'DAPK1 (1612)', 'DAPK2 (23604)', 'DAPK3 (1613)', 'DAPL1 (92196)', 'DAPP1 (27071)', 'DARS1 (1615)', 'DARS2 (55157)', 'DAW1 (164781)', 'DAXX (1616)', 'DAZAP1 (26528)', 'DAZAP2 (9802)', 'DAZL (1618)', 'DBF4 (10926)', 'DBF4B (80174)', 'DBH (1621)', 'DBI (1622)', 'DBN1 (1627)', 'DBNDD1 (79007)', 'DBNDD2 (55861)', 'DBNL (28988)', 'DBP (1628)', 'DBR1 (51163)', 'DBT (1629)', 'DBX2 (440097)', 'DCAF10 (79269)', 'DCAF11 (80344)', 'DCAF12 (25853)', 'DCAF12L1 (139170)', 'DCAF12L2 (340578)', 'DCAF15 (90379)', 'DCAF16 (54876)', 'DCAF17 (80067)', 'DCAF4 (26094)', 'DCAF4L1 (285429)', 'DCAF4L2 (138009)', 'DCAF5 (8816)', 'DCAF6 (55827)', 'DCAF8 (50717)', 'DCAF8L1 (139425)', 'DCAF8L2 (347442)', 'DCAKD (79877)', 'DCANP1 (140947)', 'DCBLD1 (285761)', 'DCBLD2 (131566)', 'DCC (1630)', 'DCD (117159)', 'DCDC1 (341019)', 'DCDC2 (51473)', 'DCDC2B (149069)', 'DCHS1 (8642)', 'DCHS2 (54798)', 'DCK (1633)', 'DCLK1 (9201)', 'DCLK2 (166614)', 'DCLK3 (85443)', 'DCLRE1A (9937)', 'DCLRE1B (64858)', 'DCLRE1C (64421)', 'DCN (1634)', 'DCP1B (196513)', 'DCP2 (167227)', 'DCPS (28960)', 'DCST1 (149095)', 'DCST2 (127579)', 'DCSTAMP (81501)', 'DCT (1638)', 'DCTD (1635)', 'DCTN1 (1639)', 'DCTN2 (10540)', 'DCTN3 (11258)', 'DCTN4 (51164)', 'DCTN5 (84516)', 'DCTN6 (10671)', 'DCTPP1 (79077)', 'DCUN1D1 (54165)', 'DCUN1D2 (55208)', 'DCUN1D3 (123879)', 'DCUN1D4 (23142)', 'DCUN1D5 (84259)', 'DCX (1641)', 'DCXR (51181)', 'DDA1 (79016)', 'DDAH1 (23576)', 'DDAH2 (23564)', 'DDB1 (1642)', 'DDB2 (1643)', 'DDC (1644)', 'DDHD1 (80821)', 'DDHD2 (23259)', 'DDI1 (414301)', 'DDI2 (84301)', 'DDIAS (220042)', 'DDIT3 (1649)', 'DDIT4 (54541)', 'DDIT4L (115265)', 'DDN (23109)', 'DDO (8528)', 'DDOST (1650)', 'DDR1 (780)', 'DDR2 (4921)', 'DDRGK1 (65992)', 'DDX1 (1653)', 'DDX10 (1662)', 'DDX11 (1663)', 'DDX17 (10521)', 'DDX18 (8886)', 'DDX19A (55308)', 'DDX19B (11269)', 'DDX20 (11218)', 'DDX21 (9188)', 'DDX23 (9416)', 'DDX24 (57062)', 'DDX25 (29118)', 'DDX27 (55661)', 'DDX28 (55794)', 'DDX31 (64794)', 'DDX39A (10212)', 'DDX39B (7919)', 'DDX3X (1654)', 'DDX3Y (8653)', 'DDX4 (54514)', 'DDX41 (51428)', 'DDX42 (11325)', 'DDX43 (55510)', 'DDX46 (9879)', 'DDX47 (51202)', 'DDX49 (54555)', 'DDX5 (1655)', 'DDX50 (79009)', 'DDX51 (317781)', 'DDX52 (11056)', 'DDX53 (168400)', 'DDX54 (79039)', 'DDX55 (57696)', 'DDX56 (54606)', 'DDX59 (83479)', 'DDX6 (1656)', 'DDX60 (55601)', 'DDX60L (91351)', 'DEAF1 (10522)', 'DECR1 (1666)', 'DECR2 (26063)', 'DEDD (9191)', 'DEDD2 (162989)', 'DEF6 (50619)', 'DEF8 (54849)', 'DEFA4 (1669)', 'DEFA5 (1670)', 'DEFA6 (1671)', 'DEFB1 (1672)', 'DEFB110 (245913)', 'DEFB112 (245915)', 'DEFB113 (245927)', 'DEFB114 (245928)', 'DEFB115 (245929)', 'DEFB116 (245930)', 'DEFB118 (117285)', 'DEFB119 (245932)', 'DEFB121 (245934)', 'DEFB123 (245936)', 'DEFB124 (245937)', 'DEFB125 (245938)', 'DEFB126 (81623)', 'DEFB127 (140850)', 'DEFB128 (245939)', 'DEFB129 (140881)', 'DEFB131A (644414)', 'DEFB132 (400830)', 'DEFB134 (613211)', 'DEFB135 (613209)', 'DEFB136 (613210)', 'DEGS1 (8560)', 'DEGS2 (123099)', 'DEK (7913)', 'DELE1 (9812)', 'DENND10 (404636)', 'DENND11 (57189)', 'DENND1A (57706)', 'DENND1B (163486)', 'DENND1C (79958)', 'DENND2A (27147)', 'DENND2B (6764)', 'DENND2C (163259)', 'DENND2D (79961)', 'DENND3 (22898)', 'DENND4A (10260)', 'DENND4B (9909)', 'DENND4C (55667)', 'DENND5A (23258)', 'DENND5B (160518)', 'DENND6A (201627)', 'DENND6B (414918)', 'DENR (8562)', 'DEPDC1 (55635)', 'DEPDC1B (55789)', 'DEPDC4 (120863)', 'DEPDC5 (9681)', 'DEPDC7 (91614)', 'DEPP1 (11067)', 'DEPTOR (64798)', 'DERA (51071)', 'DERL1 (79139)', 'DERL2 (51009)', 'DERL3 (91319)', 'DES (1674)', 'DESI1 (27351)', 'DESI2 (51029)', 'DET1 (55070)', 'DEUP1 (159989)', 'DFFA (1676)', 'DFFB (1677)', 'DGAT1 (8694)', 'DGAT2 (84649)', 'DGAT2L6 (347516)', 'DGCR2 (9993)', 'DGCR6 (8214)', 'DGCR6L (85359)', 'DGCR8 (54487)', 'DGKA (1606)', 'DGKB (1607)', 'DGKD (8527)', 'DGKE (8526)', 'DGKG (1608)', 'DGKH (160851)', 'DGKI (9162)', 'DGKQ (1609)', 'DGKZ (8525)', 'DGLUCY (80017)', 'DGUOK (1716)', 'DHCR24 (1718)', 'DHCR7 (1717)', 'DHDDS (79947)', 'DHDH (27294)', 'DHFR (1719)', 'DHFR2 (200895)', 'DHH (50846)', 'DHODH (1723)', 'DHPS (1725)', 'DHRS1 (115817)', 'DHRS11 (79154)', 'DHRS12 (79758)', 'DHRS13 (147015)', 'DHRS2 (10202)', 'DHRS3 (9249)', 'DHRS4 (10901)', 'DHRS4L2 (317749)', 'DHRS7 (51635)', 'DHRS7B (25979)', 'DHRS7C (201140)', 'DHRS9 (10170)', 'DHTKD1 (55526)', 'DHX15 (1665)', 'DHX16 (8449)', 'DHX29 (54505)', 'DHX30 (22907)', 'DHX32 (55760)', 'DHX33 (56919)', 'DHX34 (9704)', 'DHX35 (60625)', 'DHX36 (170506)', 'DHX37 (57647)', 'DHX38 (9785)', 'DHX40 (79665)', 'DHX57 (90957)', 'DHX58 (79132)', 'DHX8 (1659)', 'DHX9 (1660)', 'DIABLO (56616)', 'DIAPH1 (1729)', 'DIAPH2 (1730)', 'DIAPH3 (81624)', 'DICER1 (23405)', 'DIDO1 (11083)', 'DIMT1 (27292)', 'DIO1 (1733)', 'DIO2 (1734)', 'DIO3 (1735)', 'DIP2A (23181)', 'DIP2B (57609)', 'DIP2C (22982)', 'DIPK1A (388650)', 'DIPK1B (138311)', 'DIPK1C (125704)', 'DIPK2A (205428)', 'DIPK2B (79742)', 'DIRAS1 (148252)', 'DIRAS2 (54769)', 'DIRAS3 (9077)', 'DIS3 (22894)', 'DIS3L (115752)', 'DIS3L2 (129563)', 'DISC1 (27185)', 'DISP1 (84976)', 'DISP2 (85455)', 'DISP3 (57540)', 'DIXDC1 (85458)', 'DKC1 (1736)', 'DKK1 (22943)', 'DKK2 (27123)', 'DKK3 (27122)', 'DKK4 (27121)', 'DKKL1 (27120)', 'DLAT (1737)', 'DLC1 (10395)', 'DLD (1738)', 'DLEC1 (9940)', 'DLEU7 (220107)', 'DLG1 (1739)', 'DLG2 (1740)', 'DLG3 (1741)', 'DLG4 (1742)', 'DLG5 (9231)', 'DLGAP1 (9229)', 'DLGAP2 (9228)', 'DLGAP3 (58512)', 'DLGAP4 (22839)', 'DLGAP5 (9787)', 'DLK1 (8788)', 'DLK2 (65989)', 'DLL1 (28514)', 'DLL3 (10683)', 'DLL4 (54567)', 'DLST (1743)', 'DLX1 (1745)', 'DLX2 (1746)', 'DLX3 (1747)', 'DLX4 (1748)', 'DLX5 (1749)', 'DLX6 (1750)', 'DMAC1 (90871)', 'DMAC2 (55101)', 'DMAC2L (27109)', 'DMAP1 (55929)', 'DMBT1 (1755)', 'DMBX1 (127343)', 'DMC1 (11144)', 'DMD (1756)', 'DMGDH (29958)', 'DMKN (93099)', 'DMP1 (1758)', 'DMPK (1760)', 'DMRT1 (1761)', 'DMRT2 (10655)', 'DMRT3 (58524)', 'DMRTA1 (63951)', 'DMRTA2 (63950)', 'DMRTB1 (63948)', 'DMRTC2 (63946)', 'DMTF1 (9988)', 'DMTN (2039)', 'DMWD (1762)', 'DMXL1 (1657)', 'DMXL2 (23312)', 'DNAAF1 (123872)', 'DNAAF10 (116143)', 'DNAAF11 (23639)', 'DNAAF2 (55172)', 'DNAAF3 (352909)', 'DNAAF4 (161582)', 'DNAAF5 (54919)', 'DNAAF6 (139212)', 'DNAAF8 (146562)', 'DNAAF9 (25943)', 'DNAH1 (25981)', 'DNAH10 (196385)', 'DNAH11 (8701)', 'DNAH12 (201625)', 'DNAH14 (127602)', 'DNAH2 (146754)', 'DNAH3 (55567)', 'DNAH5 (1767)', 'DNAH6 (1768)', 'DNAH7 (56171)', 'DNAH9 (1770)', 'DNAI1 (27019)', 'DNAI2 (64446)', 'DNAI3 (126820)', 'DNAI4 (79819)', 'DNAI7 (55259)', 'DNAJA1 (3301)', 'DNAJA2 (10294)', 'DNAJA3 (9093)', 'DNAJA4 (55466)', 'DNAJB1 (3337)', 'DNAJB11 (51726)', 'DNAJB12 (54788)', 'DNAJB13 (374407)', 'DNAJB14 (79982)', 'DNAJB2 (3300)', 'DNAJB4 (11080)', 'DNAJB5 (25822)', 'DNAJB6 (10049)', 'DNAJB7 (150353)', 'DNAJB8 (165721)', 'DNAJB9 (4189)', 'DNAJC1 (64215)', 'DNAJC10 (54431)', 'DNAJC11 (55735)', 'DNAJC12 (56521)', 'DNAJC13 (23317)', 'DNAJC14 (85406)', 'DNAJC15 (29103)', 'DNAJC16 (23341)', 'DNAJC17 (55192)', 'DNAJC18 (202052)', 'DNAJC19 (131118)', 'DNAJC2 (27000)', 'DNAJC21 (134218)', 'DNAJC22 (79962)', 'DNAJC24 (120526)', 'DNAJC25 (548645)', 'DNAJC27 (51277)', 'DNAJC28 (54943)', 'DNAJC3 (5611)', 'DNAJC30 (84277)', 'DNAJC4 (3338)', 'DNAJC5 (80331)', 'DNAJC5B (85479)', 'DNAJC5G (285126)', 'DNAJC6 (9829)', 'DNAJC7 (7266)', 'DNAJC8 (22826)', 'DNAJC9 (23234)', 'DNAL1 (83544)', 'DNAL4 (10126)', 'DNALI1 (7802)', 'DNASE1 (1773)', 'DNASE1L1 (1774)', 'DNASE1L2 (1775)', 'DNASE1L3 (1776)', 'DNASE2 (1777)', 'DNASE2B (58511)', 'DND1 (373863)', 'DNER (92737)', 'DNHD1 (144132)', 'DNLZ (728489)', 'DNM1 (1759)', 'DNM1L (10059)', 'DNM2 (1785)', 'DNM3 (26052)', 'DNMBP (23268)', 'DNMT1 (1786)', 'DNMT3A (1788)', 'DNMT3B (1789)', 'DNMT3L (29947)', 'DNPEP (23549)', 'DNPH1 (10591)', 'DNTT (1791)', 'DNTTIP1 (116092)', 'DNTTIP2 (30836)', 'DOC2A (8448)', 'DOCK10 (55619)', 'DOCK11 (139818)', 'DOCK2 (1794)', 'DOCK3 (1795)', 'DOCK4 (9732)', 'DOCK5 (80005)', 'DOCK6 (57572)', 'DOCK7 (85440)', 'DOCK8 (81704)', 'DOCK9 (23348)', 'DOHH (83475)', 'DOK1 (1796)', 'DOK2 (9046)', 'DOK3 (79930)', 'DOK4 (55715)', 'DOK5 (55816)', 'DOK6 (220164)', 'DOK7 (285489)', 'DOLK (22845)', 'DOLPP1 (57171)', 'DONSON (29980)', 'DOP1A (23033)', 'DOP1B (9980)', 'DOT1L (84444)', 'DPAGT1 (1798)', 'DPCD (25911)', 'DPEP1 (1800)', 'DPEP2 (64174)', 'DPEP3 (64180)', 'DPF1 (8193)', 'DPF2 (5977)', 'DPF3 (8110)', 'DPH1 (1801)', 'DPH2 (1802)', 'DPH3 (285381)', 'DPH5 (51611)', 'DPH6 (89978)', 'DPH7 (92715)', 'DPM1 (8813)', 'DPM2 (8818)', 'DPM3 (54344)', 'DPP10 (57628)', 'DPP3 (10072)', 'DPP4 (1803)', 'DPP7 (29952)', 'DPP8 (54878)', 'DPP9 (91039)', 'DPPA2 (151871)', 'DPPA3 (359787)', 'DPPA4 (55211)', 'DPPA5 (340168)', 'DPRX (503834)', 'DPT (1805)', 'DPY19L1 (23333)', 'DPY19L2 (283417)', 'DPY19L3 (147991)', 'DPY19L4 (286148)', 'DPY30 (84661)', 'DPYD (1806)', 'DPYS (1807)', 'DPYSL2 (1808)', 'DPYSL3 (1809)', 'DPYSL4 (10570)', 'DPYSL5 (56896)', 'DQX1 (165545)', 'DR1 (1810)', 'DRAM1 (55332)', 'DRAM2 (128338)', 'DRAP1 (10589)', 'DRAXIN (374946)', 'DRC1 (92749)', 'DRC12 (283152)', 'DRC3 (83450)', 'DRC7 (84229)', 'DRD1 (1812)', 'DRD2 (1813)', 'DRD3 (1814)', 'DRD4 (1815)', 'DRD5 (1816)', 'DRG1 (4733)', 'DRG2 (1819)', 'DRGX (644168)', 'DRICH1 (51233)', 'DROSHA (29102)', 'DRP2 (1821)', 'DSC1 (1823)', 'DSC2 (1824)', 'DSC3 (1825)', 'DSCAM (1826)', 'DSCAML1 (57453)', 'DSCC1 (79075)', 'DSE (29940)', 'DSEL (92126)', 'DSG1 (1828)', 'DSG2 (1829)', 'DSG3 (1830)', 'DSG4 (147409)', 'DSN1 (79980)', 'DSP (1832)', 'DSPP (1834)', 'DST (667)', 'DSTN (11034)', 'DSTYK (25778)', 'DTD1 (92675)', 'DTD2 (112487)', 'DTHD1 (401124)', 'DTL (51514)', 'DTNA (1837)', 'DTNB (1838)', 'DTNBP1 (84062)', 'DTWD1 (56986)', 'DTWD2 (285605)', 'DTX1 (1840)', 'DTX2 (113878)', 'DTX3 (196403)', 'DTX3L (151636)', 'DTX4 (23220)', 'DTYMK (1841)', 'DUOX1 (53905)', 'DUOX2 (50506)', 'DUOXA1 (90527)', 'DUOXA2 (405753)', 'DUS1L (64118)', 'DUS2 (54920)', 'DUS3L (56931)', 'DUS4L (11062)', 'DUSP1 (1843)', 'DUSP10 (11221)', 'DUSP12 (11266)', 'DUSP13B (51207)', 'DUSP14 (11072)', 'DUSP15 (128853)', 'DUSP16 (80824)', 'DUSP18 (150290)', 'DUSP19 (142679)', 'DUSP2 (1844)', 'DUSP21 (63904)', 'DUSP22 (56940)', 'DUSP23 (54935)', 'DUSP26 (78986)', 'DUSP28 (285193)', 'DUSP29 (338599)', 'DUSP3 (1845)', 'DUSP4 (1846)', 'DUSP5 (1847)', 'DUSP6 (1848)', 'DUSP7 (1849)', 'DUSP8 (1850)', 'DUSP9 (1852)', 'DUT (1854)', 'DUXA (503835)', 'DVL1 (1855)', 'DVL2 (1856)', 'DVL3 (1857)', 'DXO (1797)', 'DYDC1 (143241)', 'DYDC2 (84332)', 'DYM (54808)', 'DYNAP (284254)', 'DYNC1H1 (1778)', 'DYNC1I1 (1780)', 'DYNC1I2 (1781)', 'DYNC1LI1 (51143)', 'DYNC1LI2 (1783)', 'DYNC2H1 (79659)', 'DYNC2I1 (55112)', 'DYNC2I2 (89891)', 'DYNC2LI1 (51626)', 'DYNLL1 (8655)', 'DYNLL2 (140735)', 'DYNLRB1 (83658)', 'DYNLRB2 (83657)', 'DYNLT1 (6993)', 'DYNLT2 (6991)', 'DYNLT2B (255758)', 'DYNLT3 (6990)', 'DYNLT4 (343521)', 'DYNLT5 (200132)', 'DYRK1A (1859)', 'DYRK1B (9149)', 'DYRK2 (8445)', 'DYRK3 (8444)', 'DYRK4 (8798)', 'DYSF (8291)', 'DYTN (391475)', 'DZANK1 (55184)', 'DZIP1 (22873)', 'DZIP1L (199221)', 'DZIP3 (9666)', 'E2F1 (1869)', 'E2F2 (1870)', 'E2F3 (1871)', 'E2F4 (1874)', 'E2F5 (1875)', 'E2F6 (1876)', 'E2F7 (144455)', 'E2F8 (79733)', 'E4F1 (1877)', 'EAF1 (85403)', 'EAF2 (55840)', 'EAPP (55837)', 'EARS2 (124454)', 'EBAG9 (9166)', 'EBF1 (1879)', 'EBF2 (64641)', 'EBF3 (253738)', 'EBF4 (57593)', 'EBI3 (10148)', 'EBLN1 (340900)', 'EBLN2 (55096)', 'EBNA1BP2 (10969)', 'EBP (10682)', 'EBPL (84650)', 'ECD (11319)', 'ECE1 (1889)', 'ECE2 (9718)', 'ECEL1 (9427)', 'ECH1 (1891)', 'ECHDC1 (55862)', 'ECHDC2 (55268)', 'ECHDC3 (79746)', 'ECHS1 (1892)', 'ECI1 (1632)', 'ECI2 (10455)', 'ECM1 (1893)', 'ECM2 (1842)', 'ECPAS (23392)', 'ECRG4 (84417)', 'ECSIT (51295)', 'ECT2 (1894)', 'ECT2L (345930)', 'EDA (1896)', 'EDA2R (60401)', 'EDAR (10913)', 'EDARADD (128178)', 'EDC3 (80153)', 'EDC4 (23644)', 'EDDM3A (10876)', 'EDDM3B (64184)', 'EDEM1 (9695)', 'EDEM2 (55741)', 'EDEM3 (80267)', 'EDF1 (8721)', 'EDIL3 (10085)', 'EDN1 (1906)', 'EDN2 (1907)', 'EDN3 (1908)', 'EDNRA (1909)', 'EDNRB (1910)', 'EDRF1 (26098)', 'EEA1 (8411)', 'EED (8726)', 'EEF1A1 (1915)', 'EEF1A2 (1917)', 'EEF1AKMT1 (221143)', 'EEF1AKMT2 (399818)', 'EEF1AKMT3 (25895)', 'EEF1AKMT4 (110599564)', 'EEF1B2 (1933)', 'EEF1D (1936)', 'EEF1E1 (9521)', 'EEF1G (1937)', 'EEF2 (1938)', 'EEF2K (29904)', 'EEF2KMT (196483)', 'EEFSEC (60678)', 'EEIG1 (399665)', 'EEIG2 (284611)', 'EEPD1 (80820)', 'EFCAB11 (90141)', 'EFCAB12 (90288)', 'EFCAB13 (124989)', 'EFCAB14 (9813)', 'EFCAB2 (84288)', 'EFCAB3 (146779)', 'EFCAB5 (374786)', 'EFCAB6 (64800)', 'EFCAB7 (84455)', 'EFCAB9 (285588)', 'EFCC1 (79825)', 'EFEMP1 (2202)', 'EFEMP2 (30008)', 'EFHB (151651)', 'EFHC1 (114327)', 'EFHC2 (80258)', 'EFHD1 (80303)', 'EFHD2 (79180)', 'EFL1 (79631)', 'EFNA1 (1942)', 'EFNA2 (1943)', 'EFNA3 (1944)', 'EFNA4 (1945)', 'EFNA5 (1946)', 'EFNB1 (1947)', 'EFNB2 (1948)', 'EFNB3 (1949)', 'EFR3A (23167)', 'EFR3B (22979)', 'EFS (10278)', 'EFTUD2 (9343)', 'EGF (1950)', 'EGFL6 (25975)', 'EGFL7 (51162)', 'EGFL8 (80864)', 'EGFLAM (133584)', 'EGFR (1956)', 'EGLN1 (54583)', 'EGLN2 (112398)', 'EGLN3 (112399)', 'EGR1 (1958)', 'EGR2 (1959)', 'EGR3 (1960)', 'EGR4 (1961)', 'EHBP1 (23301)', 'EHBP1L1 (254102)', 'EHD1 (10938)', 'EHD2 (30846)', 'EHD3 (30845)', 'EHD4 (30844)', 'EHF (26298)', 'EHHADH (1962)', 'EHMT1 (79813)', 'EHMT2 (10919)', 'EID1 (23741)', 'EID2 (163126)', 'EID2B (126272)', 'EID3 (493861)', 'EIF1 (10209)', 'EIF1AD (84285)', 'EIF1AX (1964)', 'EIF1AY (9086)', 'EIF1B (10289)', 'EIF2A (83939)', 'EIF2AK1 (27102)', 'EIF2AK2 (5610)', 'EIF2AK3 (9451)', 'EIF2AK4 (440275)', 'EIF2B1 (1967)', 'EIF2B2 (8892)', 'EIF2B3 (8891)', 'EIF2B4 (8890)', 'EIF2B5 (8893)', 'EIF2D (1939)', 'EIF2S1 (1965)', 'EIF2S2 (8894)', 'EIF2S3 (1968)', 'EIF3A (8661)', 'EIF3B (8662)', 'EIF3D (8664)', 'EIF3E (3646)', 'EIF3F (8665)', 'EIF3G (8666)', 'EIF3H (8667)', 'EIF3I (8668)', 'EIF3J (8669)', 'EIF3K (27335)', 'EIF3L (51386)', 'EIF3M (10480)', 'EIF4A1 (1973)', 'EIF4A2 (1974)', 'EIF4A3 (9775)', 'EIF4B (1975)', 'EIF4E (1977)', 'EIF4E1B (253314)', 'EIF4E2 (9470)', 'EIF4E3 (317649)', 'EIF4EBP1 (1978)', 'EIF4EBP2 (1979)', 'EIF4EBP3 (8637)', 'EIF4ENIF1 (56478)', 'EIF4G1 (1981)', 'EIF4G2 (1982)', 'EIF4G3 (8672)', 'EIF4H (7458)', 'EIF5 (1983)', 'EIF5A (1984)', 'EIF5A2 (56648)', 'EIF5AL1 (143244)', 'EIF5B (9669)', 'EIF6 (3692)', 'EIPR1 (7260)', 'ELAC1 (55520)', 'ELAC2 (60528)', 'ELANE (1991)', 'ELAPOR1 (57535)', 'ELAPOR2 (222223)', 'ELAVL1 (1994)', 'ELAVL2 (1993)', 'ELAVL3 (1995)', 'ELAVL4 (1996)', 'ELF1 (1997)', 'ELF2 (1998)', 'ELF3 (1999)', 'ELF4 (2000)', 'ELF5 (2001)', 'ELFN1 (392617)', 'ELFN2 (114794)', 'ELK1 (2002)', 'ELK3 (2004)', 'ELK4 (2005)', 'ELL (8178)', 'ELL2 (22936)', 'ELL3 (80237)', 'ELMO1 (9844)', 'ELMO2 (63916)', 'ELMO3 (79767)', 'ELMOD1 (55531)', 'ELMOD2 (255520)', 'ELMOD3 (84173)', 'ELN (2006)', 'ELOA (6924)', 'ELOA2 (51224)', 'ELOB (6923)', 'ELOC (6921)', 'ELOF1 (84337)', 'ELOVL1 (64834)', 'ELOVL2 (54898)', 'ELOVL3 (83401)', 'ELOVL4 (6785)', 'ELOVL5 (60481)', 'ELOVL6 (79071)', 'ELOVL7 (79993)', 'ELP1 (8518)', 'ELP2 (55250)', 'ELP3 (55140)', 'ELP4 (26610)', 'ELP5 (23587)', 'ELP6 (54859)', 'ELSPBP1 (64100)', 'EMB (133418)', 'EMC1 (23065)', 'EMC10 (284361)', 'EMC2 (9694)', 'EMC3 (55831)', 'EMC4 (51234)', 'EMC6 (83460)', 'EMC7 (56851)', 'EMC8 (10328)', 'EMC9 (51016)', 'EMCN (51705)', 'EMD (2010)', 'EME1 (146956)', 'EME2 (197342)', 'EMID1 (129080)', 'EMILIN1 (11117)', 'EMILIN2 (84034)', 'EMILIN3 (90187)', 'EML1 (2009)', 'EML2 (24139)', 'EML3 (256364)', 'EML4 (27436)', 'EML5 (161436)', 'EML6 (400954)', 'EMP1 (2012)', 'EMP2 (2013)', 'EMP3 (2014)', 'EMSY (56946)', 'EMX1 (2016)', 'EMX2 (2018)', 'EN1 (2019)', 'EN2 (2020)', 'ENAH (55740)', 'ENAM (10117)', 'ENC1 (8507)', 'ENDOD1 (23052)', 'ENDOG (2021)', 'ENDOU (8909)', 'ENDOV (284131)', 'ENG (2022)', 'ENGASE (64772)', 'ENHO (375704)', 'ENKD1 (84080)', 'ENKUR (219670)', 'ENO1 (2023)', 'ENO2 (2026)', 'ENO3 (2027)', 'ENOPH1 (58478)', 'ENOSF1 (55556)', 'ENOX1 (55068)', 'ENOX2 (10495)', 'ENPEP (2028)', 'ENPP1 (5167)', 'ENPP2 (5168)', 'ENPP3 (5169)', 'ENPP4 (22875)', 'ENPP5 (59084)', 'ENPP6 (133121)', 'ENPP7 (339221)', 'ENSA (2029)', 'ENTHD1 (150350)', 'ENTPD1 (953)', 'ENTPD2 (954)', 'ENTPD3 (956)', 'ENTPD4 (9583)', 'ENTPD5 (957)', 'ENTPD6 (955)', 'ENTPD7 (57089)', 'ENTPD8 (377841)', 'ENTR1 (10807)', 'ENTREP1 (9413)', 'ENTREP2 (23359)', 'ENTREP3 (10712)', 'ENY2 (56943)', 'EOGT (285203)', 'EOLA1 (91966)', 'EOLA2 (541578)', 'EOMES (8320)', 'EP300 (2033)', 'EP400 (57634)', 'EPAS1 (2034)', 'EPB41 (2035)', 'EPB41L1 (2036)', 'EPB41L2 (2037)', 'EPB41L3 (23136)', 'EPB41L4A (64097)', 'EPB41L4B (54566)', 'EPB41L5 (57669)', 'EPB42 (2038)', 'EPC1 (80314)', 'EPC2 (26122)', 'EPCAM (4072)', 'EPCIP (56245)', 'EPDR1 (54749)', 'EPG5 (57724)', 'EPGN (255324)', 'EPHA1 (2041)', 'EPHA10 (284656)', 'EPHA2 (1969)', 'EPHA3 (2042)', 'EPHA4 (2043)', 'EPHA5 (2044)', 'EPHA6 (285220)', 'EPHA7 (2045)', 'EPHA8 (2046)', 'EPHB1 (2047)', 'EPHB2 (2048)', 'EPHB3 (2049)', 'EPHB4 (2050)', 'EPHB6 (2051)', 'EPHX1 (2052)', 'EPHX2 (2053)', 'EPHX3 (79852)', 'EPHX4 (253152)', 'EPM2A (7957)', 'EPM2AIP1 (9852)', 'EPN1 (29924)', 'EPN2 (22905)', 'EPN3 (55040)', 'EPO (2056)', 'EPOP (100170841)', 'EPOR (2057)', 'EPPIN (57119)', 'EPRS1 (2058)', 'EPS15 (2060)', 'EPS15L1 (58513)', 'EPS8 (2059)', 'EPS8L1 (54869)', 'EPS8L2 (64787)', 'EPS8L3 (79574)', 'EPSTI1 (94240)', 'EPX (8288)', 'EPYC (1833)', 'EQTN (54586)', 'ERAL1 (26284)', 'ERAP1 (51752)', 'ERAP2 (64167)', 'ERAS (3266)', 'ERBB2 (2064)', 'ERBB3 (2065)', 'ERBB4 (2066)', 'ERBIN (55914)', 'ERC1 (23085)', 'ERC2 (26059)', 'ERCC1 (2067)', 'ERCC2 (2068)', 'ERCC3 (2071)', 'ERCC4 (2072)', 'ERCC5 (2073)', 'ERCC6 (2074)', 'ERCC6L (54821)', 'ERCC6L2 (375748)', 'ERCC8 (1161)', 'EREG (2069)', 'ERF (2077)', 'ERG (2078)', 'ERG28 (11161)', 'ERGIC1 (57222)', 'ERGIC2 (51290)', 'ERGIC3 (51614)', 'ERH (2079)', 'ERI1 (90459)', 'ERI2 (112479)', 'ERI3 (79033)', 'ERICH1 (157697)', 'ERICH3 (127254)', 'ERICH4 (100170765)', 'ERICH5 (203111)', 'ERICH6 (131831)', 'ERICH6B (220081)', 'ERLEC1 (27248)', 'ERLIN1 (10613)', 'ERLIN2 (11160)', 'ERMAP (114625)', 'ERMARD (55780)', 'ERMN (57471)', 'ERMP1 (79956)', 'ERN1 (2081)', 'ERN2 (10595)', 'ERO1A (30001)', 'ERO1B (56605)', 'ERP27 (121506)', 'ERP29 (10961)', 'ERP44 (23071)', 'ERRFI1 (54206)', 'ERV3-1 (2086)', 'ERVFRD-1 (405754)', 'ERVW-1 (30816)', 'ESAM (90952)', 'ESCO1 (114799)', 'ESCO2 (157570)', 'ESD (2098)', 'ESF1 (51575)', 'ESM1 (11082)', 'ESPL1 (9700)', 'ESPN (83715)', 'ESPNL (339768)', 'ESR1 (2099)', 'ESR2 (2100)', 'ESRP1 (54845)', 'ESRP2 (80004)', 'ESRRA (2101)', 'ESRRB (2103)', 'ESRRG (2104)', 'ESS2 (8220)', 'ESX1 (80712)', 'ESYT1 (23344)', 'ESYT2 (57488)', 'ESYT3 (83850)', 'ETAA1 (54465)', 'ETF1 (2107)', 'ETFA (2108)', 'ETFB (2109)', 'ETFBKMT (254013)', 'ETFDH (2110)', 'ETFRF1 (144363)', 'ETHE1 (23474)', 'ETNK1 (55500)', 'ETNK2 (55224)', 'ETNPPL (64850)', 'ETS1 (2113)', 'ETS2 (2114)', 'ETV1 (2115)', 'ETV2 (2116)', 'ETV3 (2117)', 'ETV3L (440695)', 'ETV4 (2118)', 'ETV5 (2119)', 'ETV6 (2120)', 'ETV7 (51513)', 'EVA1A (84141)', 'EVA1B (55194)', 'EVA1C (59271)', 'EVC (2121)', 'EVC2 (132884)', 'EVI2A (2123)', 'EVI2B (2124)', 'EVI5 (7813)', 'EVI5L (115704)', 'EVL (51466)', 'EVPL (2125)', 'EVPLL (645027)', 'EVX1 (2128)', 'EVX2 (344191)', 'EWSR1 (2130)', 'EXD1 (161829)', 'EXD2 (55218)', 'EXD3 (54932)', 'EXO1 (9156)', 'EXO5 (64789)', 'EXOC1 (55763)', 'EXOC2 (55770)', 'EXOC3 (11336)', 'EXOC3L1 (283849)', 'EXOC3L2 (90332)', 'EXOC3L4 (91828)', 'EXOC4 (60412)', 'EXOC5 (10640)', 'EXOC6 (54536)', 'EXOC6B (23233)', 'EXOC7 (23265)', 'EXOC8 (149371)', 'EXOG (9941)', 'EXOSC1 (51013)', 'EXOSC10 (5394)', 'EXOSC2 (23404)', 'EXOSC3 (51010)', 'EXOSC4 (54512)', 'EXOSC5 (56915)', 'EXOSC6 (118460)', 'EXOSC7 (23016)', 'EXOSC8 (11340)', 'EXOSC9 (5393)', 'EXPH5 (23086)', 'EXT1 (2131)', 'EXT2 (2132)', 'EXTL1 (2134)', 'EXTL2 (2135)', 'EXTL3 (2137)', 'EYA1 (2138)', 'EYA2 (2139)', 'EYA3 (2140)', 'EYA4 (2070)', 'EYS (346007)', 'EZH1 (2145)', 'EZH2 (2146)', 'EZR (7430)', 'F10 (2159)', 'F11 (2160)', 'F11R (50848)', 'F12 (2161)', 'F13A1 (2162)', 'F13B (2165)', 'F2 (2147)', 'F2R (2149)', 'F2RL1 (2150)', 'F2RL2 (2151)', 'F2RL3 (9002)', 'F3 (2152)', 'F5 (2153)', 'F7 (2155)', 'F8 (2157)', 'F9 (2158)', 'FA2H (79152)', 'FAAH (2166)', 'FAAH2 (158584)', 'FAAP100 (80233)', 'FAAP20 (199990)', 'FAAP24 (91442)', 'FABP1 (2168)', 'FABP12 (646486)', 'FABP2 (2169)', 'FABP3 (2170)', 'FABP4 (2167)', 'FABP6 (2172)', 'FABP7 (2173)', 'FADD (8772)', 'FADS1 (3992)', 'FADS2 (9415)', 'FADS3 (3995)', 'FADS6 (283985)', 'FAF1 (11124)', 'FAF2 (23197)', 'FAH (2184)', 'FAHD1 (81889)', 'FAHD2A (51011)', 'FAHD2B (151313)', 'FAIM (55179)', 'FAIM2 (23017)', 'FAM107A (11170)', 'FAM107B (83641)', 'FAM110A (83541)', 'FAM110B (90362)', 'FAM110C (642273)', 'FAM110D (79927)', 'FAM111A (63901)', 'FAM111B (374393)', 'FAM114A1 (92689)', 'FAM114A2 (10827)', 'FAM117A (81558)', 'FAM117B (150864)', 'FAM118A (55007)', 'FAM118B (79607)', 'FAM120A (23196)', 'FAM120AOS (158293)', 'FAM120B (84498)', 'FAM120C (54954)', 'FAM124A (220108)', 'FAM124B (79843)', 'FAM131A (131408)', 'FAM131B (9715)', 'FAM131C (348487)', 'FAM133A (286499)', 'FAM133B (257415)', 'FAM135A (57579)', 'FAM135B (51059)', 'FAM136A (84908)', 'FAM13A (10144)', 'FAM13B (51306)', 'FAM13C (220965)', 'FAM149A (25854)', 'FAM149B1 (317662)', 'FAM151A (338094)', 'FAM151B (167555)', 'FAM153A (285596)', 'FAM161A (84140)', 'FAM161B (145483)', 'FAM162A (26355)', 'FAM162B (221303)', 'FAM163A (148753)', 'FAM163B (642968)', 'FAM167A (83648)', 'FAM167B (84734)', 'FAM168A (23201)', 'FAM168B (130074)', 'FAM169A (26049)', 'FAM170A (340069)', 'FAM170B (170370)', 'FAM171A1 (221061)', 'FAM171A2 (284069)', 'FAM171B (165215)', 'FAM174A (345757)', 'FAM174B (400451)', 'FAM174C (55009)', 'FAM177A1 (283635)', 'FAM177B (400823)', 'FAM178B (51252)', 'FAM180A (389558)', 'FAM181A (90050)', 'FAM181B (220382)', 'FAM184A (79632)', 'FAM184B (27146)', 'FAM185A (222234)', 'FAM186A (121006)', 'FAM186B (84070)', 'FAM187B (148109)', 'FAM193A (8603)', 'FAM193B (54540)', 'FAM199X (139231)', 'FAM200A (221786)', 'FAM200B (285550)', 'FAM200C (63920)', 'FAM204A (63877)', 'FAM209A (200232)', 'FAM209B (388799)', 'FAM20A (54757)', 'FAM20B (9917)', 'FAM20C (56975)', 'FAM210A (125228)', 'FAM210B (116151)', 'FAM216A (29902)', 'FAM216B (144809)', 'FAM217A (222826)', 'FAM217B (63939)', 'FAM219A (203259)', 'FAM219B (57184)', 'FAM220A (84792)', 'FAM221A (340277)', 'FAM221B (392307)', 'FAM222A (84915)', 'FAM222B (55731)', 'FAM227B (196951)', 'FAM228A (653140)', 'FAM229A (100128071)', 'FAM229B (619208)', 'FAM234A (83986)', 'FAM234B (57613)', 'FAM241A (132720)', 'FAM241B (219738)', 'FAM24A (118670)', 'FAM24B (196792)', 'FAM25A (643161)', 'FAM32A (26017)', 'FAM3A (60343)', 'FAM3B (54097)', 'FAM3C (10447)', 'FAM3D (131177)', 'FAM43A (131583)', 'FAM43B (163933)', 'FAM47A (158724)', 'FAM47B (170062)', 'FAM47C (442444)', 'FAM47E (100129583)', 'FAM50A (9130)', 'FAM50B (26240)', 'FAM53A (152877)', 'FAM53B (9679)', 'FAM53C (51307)', 'FAM76A (199870)', 'FAM76B (143684)', 'FAM78A (286336)', 'FAM78B (149297)', 'FAM81A (145773)', 'FAM81B (153643)', 'FAM83A (84985)', 'FAM83B (222584)', 'FAM83C (128876)', 'FAM83D (81610)', 'FAM83E (54854)', 'FAM83F (113828)', 'FAM83G (644815)', 'FAM83H (286077)', 'FAM86B2 (653333)', 'FAM89A (375061)', 'FAM89B (23625)', 'FAM8A1 (51439)', 'FAM90A1 (55138)', 'FAM91A1 (157769)', 'FAM98A (25940)', 'FAM98B (283742)', 'FAM98C (147965)', 'FAM9A (171482)', 'FAM9B (171483)', 'FAM9C (171484)', 'FAN1 (22909)', 'FANCA (2175)', 'FANCB (2187)', 'FANCC (2176)', 'FANCD2 (2177)', 'FANCD2OS (115795)', 'FANCE (2178)', 'FANCF (2188)', 'FANCG (2189)', 'FANCI (55215)', 'FANCL (55120)', 'FANCM (57697)', 'FANK1 (92565)', 'FAP (2191)', 'FAR1 (84188)', 'FAR2 (55711)', 'FARP1 (10160)', 'FARP2 (9855)', 'FARS2 (10667)', 'FARSA (2193)', 'FARSB (10056)', 'FAS (355)', 'FASLG (356)', 'FASN (2194)', 'FASTK (10922)', 'FASTKD1 (79675)', 'FASTKD2 (22868)', 'FASTKD3 (79072)', 'FASTKD5 (60493)', 'FAT1 (2195)', 'FAT2 (2196)', 'FAT4 (79633)', 'FATE1 (89885)', 'FAU (2197)', 'FAXC (84553)', 'FAXDC2 (10826)', 'FBF1 (85302)', 'FBH1 (84893)', 'FBL (2091)', 'FBLIM1 (54751)', 'FBLN1 (2192)', 'FBLN2 (2199)', 'FBLN5 (10516)', 'FBLN7 (129804)', 'FBN1 (2200)', 'FBN2 (2201)', 'FBN3 (84467)', 'FBP1 (2203)', 'FBP2 (8789)', 'FBRSL1 (57666)', 'FBXL12 (54850)', 'FBXL13 (222235)', 'FBXL14 (144699)', 'FBXL15 (79176)', 'FBXL16 (146330)', 'FBXL17 (64839)', 'FBXL18 (80028)', 'FBXL19 (54620)', 'FBXL2 (25827)', 'FBXL20 (84961)', 'FBXL22 (283807)', 'FBXL3 (26224)', 'FBXL4 (26235)', 'FBXL5 (26234)', 'FBXL6 (26233)', 'FBXL7 (23194)', 'FBXL8 (55336)', 'FBXO10 (26267)', 'FBXO11 (80204)', 'FBXO15 (201456)', 'FBXO16 (157574)', 'FBXO17 (115290)', 'FBXO2 (26232)', 'FBXO21 (23014)', 'FBXO22 (26263)', 'FBXO24 (26261)', 'FBXO25 (26260)', 'FBXO27 (126433)', 'FBXO28 (23219)', 'FBXO3 (26273)', 'FBXO30 (84085)', 'FBXO31 (79791)', 'FBXO32 (114907)', 'FBXO33 (254170)', 'FBXO34 (55030)', 'FBXO36 (130888)', 'FBXO38 (81545)', 'FBXO39 (162517)', 'FBXO4 (26272)', 'FBXO40 (51725)', 'FBXO41 (150726)', 'FBXO42 (54455)', 'FBXO43 (286151)', 'FBXO44 (93611)', 'FBXO45 (200933)', 'FBXO46 (23403)', 'FBXO47 (494188)', 'FBXO48 (554251)', 'FBXO5 (26271)', 'FBXO6 (26270)', 'FBXO7 (25793)', 'FBXO8 (26269)', 'FBXO9 (26268)', 'FBXW10 (10517)', 'FBXW10B (374286)', 'FBXW11 (23291)', 'FBXW12 (285231)', 'FBXW2 (26190)', 'FBXW4 (6468)', 'FBXW5 (54461)', 'FBXW7 (55294)', 'FBXW8 (26259)', 'FBXW9 (84261)', 'FCAMR (83953)', 'FCAR (2204)', 'FCER1A (2205)', 'FCER1G (2207)', 'FCER2 (2208)', 'FCF1 (51077)', 'FCGR1A (2209)', 'FCGR2A (2212)', 'FCGR2B (2213)', 'FCGR3A (2214)', 'FCGR3B (2215)', 'FCGRT (2217)', 'FCHO1 (23149)', 'FCHO2 (115548)', 'FCHSD1 (89848)', 'FCHSD2 (9873)', 'FCMR (9214)', 'FCN1 (2219)', 'FCN2 (2220)', 'FCN3 (8547)', 'FCRL1 (115350)', 'FCRL2 (79368)', 'FCRL3 (115352)', 'FCRL4 (83417)', 'FCRL5 (83416)', 'FCRL6 (343413)', 'FCRLA (84824)', 'FCRLB (127943)', 'FCSK (197258)', 'FDCSP (260436)', 'FDFT1 (2222)', 'FDPS (2224)', 'FDX1 (2230)', 'FDX2 (112812)', 'FDXACB1 (91893)', 'FDXR (2232)', 'FECH (2235)', 'FEM1A (55527)', 'FEM1B (10116)', 'FEM1C (56929)', 'FEN1 (2237)', 'FER (2241)', 'FER1L6 (654463)', 'FERD3L (222894)', 'FERMT1 (55612)', 'FERMT2 (10979)', 'FERMT3 (83706)', 'FERRY3 (57102)', 'FES (2242)', 'FETUB (26998)', 'FEV (54738)', 'FEZ1 (9638)', 'FEZ2 (9637)', 'FEZF1 (389549)', 'FEZF2 (55079)', 'FFAR1 (2864)', 'FFAR2 (2867)', 'FFAR4 (338557)', 'FGA (2243)', 'FGB (2244)', 'FGD1 (2245)', 'FGD2 (221472)', 'FGD3 (89846)', 'FGD4 (121512)', 'FGD5 (152273)', 'FGD6 (55785)', 'FGF1 (2246)', 'FGF10 (2255)', 'FGF11 (2256)', 'FGF12 (2257)', 'FGF13 (2258)', 'FGF14 (2259)', 'FGF17 (8822)', 'FGF18 (8817)', 'FGF19 (9965)', 'FGF2 (2247)', 'FGF20 (26281)', 'FGF21 (26291)', 'FGF22 (27006)', 'FGF23 (8074)', 'FGF3 (2248)', 'FGF4 (2249)', 'FGF5 (2250)', 'FGF6 (2251)', 'FGF7 (2252)', 'FGF8 (2253)', 'FGF9 (2254)', 'FGFBP1 (9982)', 'FGFBP2 (83888)', 'FGFBP3 (143282)', 'FGFR1 (2260)', 'FGFR1OP2 (26127)', 'FGFR2 (2263)', 'FGFR3 (2261)', 'FGFR4 (2264)', 'FGFRL1 (53834)', 'FGG (2266)', 'FGGY (55277)', 'FGL1 (2267)', 'FGL2 (10875)', 'FGR (2268)', 'FH (2271)', 'FHDC1 (85462)', 'FHIP1A (729830)', 'FHIP1B (84067)', 'FHIP2A (57700)', 'FHIP2B (64760)', 'FHIT (2272)', 'FHL1 (2273)', 'FHL2 (2274)', 'FHL3 (2275)', 'FHL5 (9457)', 'FHOD1 (29109)', 'FHOD3 (80206)', 'FIBCD1 (84929)', 'FIBIN (387758)', 'FIBP (9158)', 'FICD (11153)', 'FIG4 (9896)', 'FIGLA (344018)', 'FIGN (55137)', 'FIGNL1 (63979)', 'FILIP1 (27145)', 'FILIP1L (11259)', 'FIP1L1 (81608)', 'FIRRM (55732)', 'FIS1 (51024)', 'FITM1 (161247)', 'FITM2 (128486)', 'FIZ1 (84922)', 'FJX1 (24147)', 'FKBP10 (60681)', 'FKBP11 (51303)', 'FKBP14 (55033)', 'FKBP15 (23307)', 'FKBP1A (2280)', 'FKBP1B (2281)', 'FKBP2 (2286)', 'FKBP3 (2287)', 'FKBP4 (2288)', 'FKBP5 (2289)', 'FKBP6 (8468)', 'FKBP7 (51661)', 'FKBP8 (23770)', 'FKBP9 (11328)', 'FKBPL (63943)', 'FKRP (79147)', 'FKTN (2218)', 'FLACC1 (130540)', 'FLAD1 (80308)', 'FLCN (201163)', 'FLG (2312)', 'FLG2 (388698)', 'FLI1 (2313)', 'FLII (2314)', 'FLNA (2316)', 'FLNB (2317)', 'FLNC (2318)', 'FLOT1 (10211)', 'FLOT2 (2319)', 'FLRT1 (23769)', 'FLRT2 (23768)', 'FLRT3 (23767)', 'FLT1 (2321)', 'FLT3 (2322)', 'FLT3LG (2323)', 'FLT4 (2324)', 'FLVCR1 (28982)', 'FLVCR2 (55640)', 'FLYWCH1 (84256)', 'FLYWCH2 (114984)', 'FMC1 (154791)', 'FMN1 (342184)', 'FMN2 (56776)', 'FMNL1 (752)', 'FMNL2 (114793)', 'FMNL3 (91010)', 'FMO1 (2326)', 'FMO2 (2327)', 'FMO3 (2328)', 'FMO4 (2329)', 'FMO5 (2330)', 'FMOD (2331)', 'FMR1 (2332)', 'FMR1NB (158521)', 'FN1 (2335)', 'FN3K (64122)', 'FN3KRP (79672)', 'FNBP1 (23048)', 'FNBP1L (54874)', 'FNBP4 (23360)', 'FNDC1 (84624)', 'FNDC10 (643988)', 'FNDC11 (79025)', 'FNDC3A (22862)', 'FNDC3B (64778)', 'FNDC4 (64838)', 'FNDC5 (252995)', 'FNDC7 (163479)', 'FNDC8 (54752)', 'FNDC9 (408263)', 'FNIP1 (96459)', 'FNIP2 (57600)', 'FNTA (2339)', 'FNTB (2342)', 'FOCAD (54914)', 'FOLH1 (2346)', 'FOLR1 (2348)', 'FOLR2 (2350)', 'FOS (2353)', 'FOSB (2354)', 'FOSL1 (8061)', 'FOSL2 (2355)', 'FOXA1 (3169)', 'FOXA2 (3170)', 'FOXA3 (3171)', 'FOXB1 (27023)', 'FOXB2 (442425)', 'FOXC1 (2296)', 'FOXC2 (2303)', 'FOXD2 (2306)', 'FOXD3 (27022)', 'FOXD4 (2298)', 'FOXD4L5 (653427)', 'FOXE1 (2304)', 'FOXE3 (2301)', 'FOXF1 (2294)', 'FOXF2 (2295)', 'FOXG1 (2290)', 'FOXH1 (8928)', 'FOXI1 (2299)', 'FOXI2 (399823)', 'FOXJ1 (2302)', 'FOXJ2 (55810)', 'FOXJ3 (22887)', 'FOXK1 (221937)', 'FOXK2 (3607)', 'FOXL1 (2300)', 'FOXL2 (668)', 'FOXL2NB (401089)', 'FOXM1 (2305)', 'FOXN1 (8456)', 'FOXN2 (3344)', 'FOXN3 (1112)', 'FOXN4 (121643)', 'FOXO1 (2308)', 'FOXO3 (2309)', 'FOXO3B (2310)', 'FOXO4 (4303)', 'FOXP1 (27086)', 'FOXP2 (93986)', 'FOXP3 (50943)', 'FOXP4 (116113)', 'FOXQ1 (94234)', 'FOXR1 (283150)', 'FOXR2 (139628)', 'FOXRED1 (55572)', 'FOXRED2 (80020)', 'FOXS1 (2307)', 'FPGS (2356)', 'FPGT (8790)', 'FPR1 (2357)', 'FPR2 (2358)', 'FPR3 (2359)', 'FRA10AC1 (118924)', 'FRAS1 (80144)', 'FRAT1 (10023)', 'FRAT2 (23401)', 'FREM1 (158326)', 'FREM2 (341640)', 'FREM3 (166752)', 'FREY1 (143678)', 'FRG1 (2483)', 'FRG2B (441581)', 'FRG2C (100288801)', 'FRK (2444)', 'FRMD1 (79981)', 'FRMD3 (257019)', 'FRMD4A (55691)', 'FRMD4B (23150)', 'FRMD5 (84978)', 'FRMD6 (122786)', 'FRMD7 (90167)', 'FRMD8 (83786)', 'FRMPD1 (22844)', 'FRMPD2 (143162)', 'FRMPD4 (9758)', 'FRRS1 (391059)', 'FRRS1L (23732)', 'FRS2 (10818)', 'FRS3 (10817)', 'FRY (10129)', 'FRYL (285527)', 'FRZB (2487)', 'FSBP (100861412)', 'FSCB (84075)', 'FSCN1 (6624)', 'FSCN2 (25794)', 'FSCN3 (29999)', 'FSD1 (79187)', 'FSD1L (83856)', 'FSD2 (123722)', 'FSHB (2488)', 'FSHR (2492)', 'FSIP1 (161835)', 'FSIP2 (401024)', 'FST (10468)', 'FSTL1 (11167)', 'FSTL3 (10272)', 'FSTL4 (23105)', 'FSTL5 (56884)', 'FTCD (10841)', 'FTH1 (2495)', 'FTHL17 (53940)', 'FTL (2512)', 'FTMT (94033)', 'FTO (79068)', 'FTSJ1 (24140)', 'FTSJ3 (117246)', 'FUBP1 (8880)', 'FUBP3 (8939)', 'FUCA1 (2517)', 'FUCA2 (2519)', 'FUNDC1 (139341)', 'FUNDC2 (65991)', 'FUOM (282969)', 'FURIN (5045)', 'FUS (2521)', 'FUT1 (2523)', 'FUT10 (84750)', 'FUT11 (170384)', 'FUT2 (2524)', 'FUT3 (2525)', 'FUT4 (2526)', 'FUT5 (2527)', 'FUT6 (2528)', 'FUT7 (2529)', 'FUT8 (2530)', 'FUT9 (10690)', 'FUZ (80199)', 'FXN (2395)', 'FXR1 (8087)', 'FXR2 (9513)', 'FXYD1 (5348)', 'FXYD2 (486)', 'FXYD3 (5349)', 'FXYD4 (53828)', 'FXYD5 (53827)', 'FXYD6 (53826)', 'FXYD7 (53822)', 'FYB1 (2533)', 'FYB2 (199920)', 'FYCO1 (79443)', 'FYN (2534)', 'FYTTD1 (84248)', 'FZD1 (8321)', 'FZD10 (11211)', 'FZD2 (2535)', 'FZD3 (7976)', 'FZD4 (8322)', 'FZD5 (7855)', 'FZD6 (8323)', 'FZD7 (8324)', 'FZD8 (8325)', 'FZD9 (8326)', 'FZR1 (51343)', 'G0S2 (50486)', 'G2E3 (55632)', 'G3BP1 (10146)', 'G3BP2 (9908)', 'G6PC1 (2538)', 'G6PC2 (57818)', 'G6PC3 (92579)', 'G6PD (2539)', 'GAA (2548)', 'GAB1 (2549)', 'GAB2 (9846)', 'GAB3 (139716)', 'GAB4 (128954)', 'GABARAP (11337)', 'GABARAPL1 (23710)', 'GABARAPL2 (11345)', 'GABBR1 (2550)', 'GABBR2 (9568)', 'GABPA (2551)', 'GABPB1 (2553)', 'GABPB2 (126626)', 'GABRA1 (2554)', 'GABRA2 (2555)', 'GABRA3 (2556)', 'GABRA4 (2557)', 'GABRA5 (2558)', 'GABRA6 (2559)', 'GABRB1 (2560)', 'GABRB2 (2561)', 'GABRB3 (2562)', 'GABRD (2563)', 'GABRE (2564)', 'GABRG1 (2565)', 'GABRG2 (2566)', 'GABRG3 (2567)', 'GABRP (2568)', 'GABRQ (55879)', 'GABRR1 (2569)', 'GABRR2 (2570)', 'GAD1 (2571)', 'GAD2 (2572)', 'GADD45A (1647)', 'GADD45B (4616)', 'GADD45G (10912)', 'GADD45GIP1 (90480)', 'GADL1 (339896)', 'GAGE12H (729442)', 'GAK (2580)', 'GAL (51083)', 'GAL3ST1 (9514)', 'GAL3ST2 (64090)', 'GAL3ST3 (89792)', 'GAL3ST4 (79690)', 'GALC (2581)', 'GALE (2582)', 'GALK1 (2584)', 'GALK2 (2585)', 'GALM (130589)', 'GALNS (2588)', 'GALNT1 (2589)', 'GALNT10 (55568)', 'GALNT11 (63917)', 'GALNT12 (79695)', 'GALNT13 (114805)', 'GALNT14 (79623)', 'GALNT15 (117248)', 'GALNT16 (57452)', 'GALNT17 (64409)', 'GALNT18 (374378)', 'GALNT2 (2590)', 'GALNT3 (2591)', 'GALNT4 (8693)', 'GALNT5 (11227)', 'GALNT6 (11226)', 'GALNT7 (51809)', 'GALNT8 (26290)', 'GALNT9 (50614)', 'GALNTL5 (168391)', 'GALNTL6 (442117)', 'GALP (85569)', 'GALR1 (2587)', 'GALR2 (8811)', 'GALR3 (8484)', 'GALT (2592)', 'GAMT (2593)', 'GAN (8139)', 'GANAB (23193)', 'GANC (2595)', 'GAP43 (2596)', 'GAPDH (2597)', 'GAPDHS (26330)', 'GAPT (202309)', 'GAPVD1 (26130)', 'GAR1 (54433)', 'GAREM1 (64762)', 'GAREM2 (150946)', 'GARIN1A (346653)', 'GARIN1B (84691)', 'GARIN2 (161142)', 'GARIN3 (153745)', 'GARIN4 (149647)', 'GARIN5A (112703)', 'GARIN6 (196472)', 'GARNL3 (84253)', 'GARRE1 (9710)', 'GARS1 (2617)', 'GART (2618)', 'GAS1 (2619)', 'GAS2 (2620)', 'GAS2L1 (10634)', 'GAS2L2 (246176)', 'GAS2L3 (283431)', 'GAS6 (2621)', 'GAS7 (8522)', 'GAS8 (2622)', 'GASK1A (729085)', 'GASK1B (51313)', 'GAST (2520)', 'GATA1 (2623)', 'GATA2 (2624)', 'GATA3 (2625)', 'GATA4 (2626)', 'GATA5 (140628)', 'GATA6 (2627)', 'GATAD1 (57798)', 'GATAD2A (54815)', 'GATAD2B (57459)', 'GATB (5188)', 'GATC (283459)', 'GATD1 (347862)', 'GATD3 (8209)', 'GATM (2628)', 'GBA1 (2629)', 'GBA2 (57704)', 'GBE1 (2632)', 'GBF1 (8729)', 'GBGT1 (26301)', 'GBP1 (2633)', 'GBP2 (2634)', 'GBP3 (2635)', 'GBP4 (115361)', 'GBP5 (115362)', 'GBP6 (163351)', 'GBP7 (388646)', 'GBX1 (2636)', 'GBX2 (2637)', 'GC (2638)', 'GCA (25801)', 'GCAT (23464)', 'GCC1 (79571)', 'GCC2 (9648)', 'GCDH (2639)', 'GCFC2 (6936)', 'GCG (2641)', 'GCGR (2642)', 'GCH1 (2643)', 'GCHFR (2644)', 'GCK (2645)', 'GCKR (2646)', 'GCLC (2729)', 'GCLM (2730)', 'GCM1 (8521)', 'GCM2 (9247)', 'GCN1 (10985)', 'GCNA (93953)', 'GCNT1 (2650)', 'GCNT2 (2651)', 'GCNT3 (9245)', 'GCNT4 (51301)', 'GCSAM (257144)', 'GCSAML (148823)', 'GCSH (2653)', 'GDA (9615)', 'GDAP1 (54332)', 'GDAP1L1 (78997)', 'GDAP2 (54834)', 'GDE1 (51573)', 'GDF1 (2657)', 'GDF10 (2662)', 'GDF11 (10220)', 'GDF15 (9518)', 'GDF2 (2658)', 'GDF3 (9573)', 'GDF5 (8200)', 'GDF6 (392255)', 'GDF7 (151449)', 'GDF9 (2661)', 'GDI1 (2664)', 'GDI2 (2665)', 'GDNF (2668)', 'GDPD1 (284161)', 'GDPD2 (54857)', 'GDPD3 (79153)', 'GDPD4 (220032)', 'GDPD5 (81544)', 'GDPGP1 (390637)', 'GEM (2669)', 'GEMIN2 (8487)', 'GEMIN4 (50628)', 'GEMIN5 (25929)', 'GEMIN6 (79833)', 'GEMIN7 (79760)', 'GEMIN8 (54960)', 'GEN1 (348654)', 'GET1 (7485)', 'GET3 (439)', 'GET4 (51608)', 'GFAP (2670)', 'GFER (2671)', 'GFI1 (2672)', 'GFI1B (8328)', 'GFM1 (85476)', 'GFM2 (84340)', 'GFOD1 (54438)', 'GFOD2 (81577)', 'GFPT1 (2673)', 'GFPT2 (9945)', 'GFRA1 (2674)', 'GFRA2 (2675)', 'GFRA3 (2676)', 'GFRA4 (64096)', 'GFRAL (389400)', 'GFUS (7264)', 'GFY (100507003)', 'GGA1 (26088)', 'GGA2 (23062)', 'GGA3 (23163)', 'GGACT (87769)', 'GGCT (79017)', 'GGCX (2677)', 'GGH (8836)', 'GGN (199720)', 'GGNBP2 (79893)', 'GGPS1 (9453)', 'GGT1 (2678)', 'GGT5 (2687)', 'GGT6 (124975)', 'GGT7 (2686)', 'GGTLC1 (92086)', 'GGTLC2 (91227)', 'GH1 (2688)', 'GH2 (2689)', 'GHDC (84514)', 'GHITM (27069)', 'GHR (2690)', 'GHRH (2691)', 'GHRHR (2692)', 'GHRL (51738)', 'GHSR (2693)', 'GID4 (79018)', 'GID8 (54994)', 'GIGYF1 (64599)', 'GIGYF2 (26058)', 'GIMAP1 (170575)', 'GIMAP2 (26157)', 'GIMAP4 (55303)', 'GIMAP5 (55340)', 'GIMAP6 (474344)', 'GIMAP7 (168537)', 'GIMAP8 (155038)', 'GIN1 (54826)', 'GINM1 (116254)', 'GINS1 (9837)', 'GINS2 (51659)', 'GINS3 (64785)', 'GINS4 (84296)', 'GIP (2695)', 'GIPC1 (10755)', 'GIPC2 (54810)', 'GIPC3 (126326)', 'GIPR (2696)', 'GIT1 (28964)', 'GIT2 (9815)', 'GJA1 (2697)', 'GJA10 (84694)', 'GJA3 (2700)', 'GJA4 (2701)', 'GJA5 (2702)', 'GJA8 (2703)', 'GJA9 (81025)', 'GJB1 (2705)', 'GJB2 (2706)', 'GJB3 (2707)', 'GJB4 (127534)', 'GJB5 (2709)', 'GJB6 (10804)', 'GJB7 (375519)', 'GJC1 (10052)', 'GJC2 (57165)', 'GJC3 (349149)', 'GJD2 (57369)', 'GJD3 (125111)', 'GJD4 (219770)', 'GK (2710)', 'GK2 (2712)', 'GK5 (256356)', 'GKAP1 (80318)', 'GKN1 (56287)', 'GKN2 (200504)', 'GLA (2717)', 'GLB1 (2720)', 'GLB1L (79411)', 'GLB1L2 (89944)', 'GLB1L3 (112937)', 'GLCCI1 (113263)', 'GLCE (26035)', 'GLDC (2731)', 'GLDN (342035)', 'GLE1 (2733)', 'GLG1 (2734)', 'GLI1 (2735)', 'GLI2 (2736)', 'GLI3 (2737)', 'GLI4 (2738)', 'GLIPR1 (11010)', 'GLIPR1L1 (256710)', 'GLIPR1L2 (144321)', 'GLIPR2 (152007)', 'GLIS1 (148979)', 'GLIS2 (84662)', 'GLIS3 (169792)', 'GLMN (11146)', 'GLMP (112770)', 'GLO1 (2739)', 'GLOD4 (51031)', 'GLOD5 (392465)', 'GLP1R (2740)', 'GLP2R (9340)', 'GLRA1 (2741)', 'GLRA2 (2742)', 'GLRA3 (8001)', 'GLRB (2743)', 'GLRX (2745)', 'GLRX2 (51022)', 'GLRX3 (10539)', 'GLRX5 (51218)', 'GLS (2744)', 'GLS2 (27165)', 'GLT1D1 (144423)', 'GLT6D1 (360203)', 'GLT8D1 (55830)', 'GLT8D2 (83468)', 'GLTP (51228)', 'GLTPD2 (388323)', 'GLUD1 (2746)', 'GLUD2 (2747)', 'GLUL (2752)', 'GLYAT (10249)', 'GLYATL1 (92292)', 'GLYATL2 (219970)', 'GLYATL3 (389396)', 'GLYCTK (132158)', 'GLYR1 (84656)', 'GM2A (2760)', 'GMCL1 (64395)', 'GMDS (2762)', 'GMEB1 (10691)', 'GMEB2 (26205)', 'GMFB (2764)', 'GMFG (9535)', 'GMIP (51291)', 'GML (2765)', 'GMNC (647309)', 'GMNN (51053)', 'GMPPA (29926)', 'GMPPB (29925)', 'GMPR (2766)', 'GMPR2 (51292)', 'GMPS (8833)', 'GNA11 (2767)', 'GNA12 (2768)', 'GNA13 (10672)', 'GNA14 (9630)', 'GNA15 (2769)', 'GNAI1 (2770)', 'GNAI2 (2771)', 'GNAI3 (2773)', 'GNAL (2774)', 'GNAO1 (2775)', 'GNAQ (2776)', 'GNAS (2778)', 'GNAT1 (2779)', 'GNAT2 (2780)', 'GNAT3 (346562)', 'GNAZ (2781)', 'GNB1 (2782)', 'GNB1L (54584)', 'GNB2 (2783)', 'GNB3 (2784)', 'GNB4 (59345)', 'GNB5 (10681)', 'GNE (10020)', 'GNG10 (2790)', 'GNG11 (2791)', 'GNG12 (55970)', 'GNG13 (51764)', 'GNG2 (54331)', 'GNG3 (2785)', 'GNG4 (2786)', 'GNG5 (2787)', 'GNG7 (2788)', 'GNG8 (94235)', 'GNGT1 (2792)', 'GNGT2 (2793)', 'GNL1 (2794)', 'GNL2 (29889)', 'GNL3 (26354)', 'GNL3L (54552)', 'GNLY (10578)', 'GNMT (27232)', 'GNPAT (8443)', 'GNPDA1 (10007)', 'GNPDA2 (132789)', 'GNPNAT1 (64841)', 'GNPTAB (79158)', 'GNPTG (84572)', 'GNRH1 (2796)', 'GNRH2 (2797)', 'GNRHR (2798)', 'GNS (2799)', 'GOLGA1 (2800)', 'GOLGA2 (2801)', 'GOLGA3 (2802)', 'GOLGA4 (2803)', 'GOLGA5 (9950)', 'GOLGA6B (55889)', 'GOLGA6D (653643)', 'GOLGA6L6 (727832)', 'GOLGA7 (51125)', 'GOLGA7B (401647)', 'GOLGA8M (653720)', 'GOLGA8N (643699)', 'GOLGA8R (101059918)', 'GOLGB1 (2804)', 'GOLIM4 (27333)', 'GOLM1 (51280)', 'GOLM2 (113201)', 'GOLPH3 (64083)', 'GOLPH3L (55204)', 'GOLT1A (127845)', 'GOLT1B (51026)', 'GON4L (54856)', 'GON7 (84520)', 'GOPC (57120)', 'GORAB (92344)', 'GORASP1 (64689)', 'GORASP2 (26003)', 'GOSR1 (9527)', 'GOSR2 (9570)', 'GOT1 (2805)', 'GOT1L1 (137362)', 'GOT2 (2806)', 'GP1BA (2811)', 'GP1BB (2812)', 'GP2 (2813)', 'GP5 (2814)', 'GP6 (51206)', 'GP9 (2815)', 'GPA33 (10223)', 'GPAA1 (8733)', 'GPALPP1 (55425)', 'GPAM (57678)', 'GPANK1 (7918)', 'GPAT2 (150763)', 'GPAT3 (84803)', 'GPAT4 (137964)', 'GPATCH1 (55094)', 'GPATCH11 (253635)', 'GPATCH2 (55105)', 'GPATCH2L (55668)', 'GPATCH3 (63906)', 'GPATCH8 (23131)', 'GPBAR1 (151306)', 'GPBP1 (65056)', 'GPBP1L1 (60313)', 'GPC1 (2817)', 'GPC2 (221914)', 'GPC3 (2719)', 'GPC4 (2239)', 'GPC5 (2262)', 'GPC6 (10082)', 'GPCPD1 (56261)', 'GPD1 (2819)', 'GPD1L (23171)', 'GPD2 (2820)', 'GPER1 (2852)', 'GPHA2 (170589)', 'GPHN (10243)', 'GPI (2821)', 'GPIHBP1 (338328)', 'GPKOW (27238)', 'GPLD1 (2822)', 'GPM6A (2823)', 'GPM6B (2824)', 'GPN1 (11321)', 'GPN2 (54707)', 'GPN3 (51184)', 'GPNMB (10457)', 'GPR101 (83550)', 'GPR107 (57720)', 'GPR108 (56927)', 'GPR119 (139760)', 'GPR12 (2835)', 'GPR132 (29933)', 'GPR135 (64582)', 'GPR137 (56834)', 'GPR137B (7107)', 'GPR137C (283554)', 'GPR139 (124274)', 'GPR141 (353345)', 'GPR142 (350383)', 'GPR143 (4935)', 'GPR146 (115330)', 'GPR148 (344561)', 'GPR149 (344758)', 'GPR15 (2838)', 'GPR150 (285601)', 'GPR151 (134391)', 'GPR152 (390212)', 'GPR153 (387509)', 'GPR155 (151556)', 'GPR156 (165829)', 'GPR157 (80045)', 'GPR158 (57512)', 'GPR15LG (387695)', 'GPR160 (26996)', 'GPR161 (23432)', 'GPR162 (27239)', 'GPR17 (2840)', 'GPR171 (29909)', 'GPR173 (54328)', 'GPR174 (84636)', 'GPR176 (11245)', 'GPR179 (440435)', 'GPR18 (2841)', 'GPR180 (160897)', 'GPR182 (11318)', 'GPR183 (1880)', 'GPR19 (2842)', 'GPR20 (2843)', 'GPR21 (2844)', 'GPR22 (2845)', 'GPR25 (2848)', 'GPR26 (2849)', 'GPR27 (2850)', 'GPR3 (2827)', 'GPR31 (2853)', 'GPR32 (2854)', 'GPR34 (2857)', 'GPR35 (2859)', 'GPR37 (2861)', 'GPR37L1 (9283)', 'GPR39 (2863)', 'GPR4 (2828)', 'GPR45 (11250)', 'GPR50 (9248)', 'GPR52 (9293)', 'GPR55 (9290)', 'GPR6 (2830)', 'GPR61 (83873)', 'GPR62 (118442)', 'GPR63 (81491)', 'GPR65 (8477)', 'GPR68 (8111)', 'GPR75 (10936)', 'GPR78 (27201)', 'GPR82 (27197)', 'GPR83 (10888)', 'GPR84 (53831)', 'GPR85 (54329)', 'GPR87 (53836)', 'GPR88 (54112)', 'GPRASP1 (9737)', 'GPRC5A (9052)', 'GPRC5B (51704)', 'GPRC5C (55890)', 'GPRC5D (55507)', 'GPRC6A (222545)', 'GPRIN1 (114787)', 'GPRIN2 (9721)', 'GPRIN3 (285513)', 'GPS1 (2873)', 'GPS2 (2874)', 'GPSM1 (26086)', 'GPSM2 (29899)', 'GPSM3 (63940)', 'GPT (2875)', 'GPT2 (84706)', 'GPX1 (2876)', 'GPX2 (2877)', 'GPX3 (2878)', 'GPX4 (2879)', 'GPX5 (2880)', 'GPX6 (257202)', 'GPX7 (2882)', 'GPX8 (493869)', 'GRAMD1A (57655)', 'GRAMD1B (57476)', 'GRAMD1C (54762)', 'GRAMD2A (196996)', 'GRAMD2B (65983)', 'GRAMD4 (23151)', 'GRAP2 (9402)', 'GRB10 (2887)', 'GRB14 (2888)', 'GRB2 (2885)', 'GRB7 (2886)', 'GREB1 (9687)', 'GREB1L (80000)', 'GREM1 (26585)', 'GREM2 (64388)', 'GRHL1 (29841)', 'GRHL2 (79977)', 'GRHL3 (57822)', 'GRHPR (9380)', 'GRIA1 (2890)', 'GRIA2 (2891)', 'GRIA3 (2892)', 'GRIA4 (2893)', 'GRID1 (2894)', 'GRID2 (2895)', 'GRID2IP (392862)', 'GRIK1 (2897)', 'GRIK2 (2898)', 'GRIK3 (2899)', 'GRIK4 (2900)', 'GRIK5 (2901)', 'GRIN1 (2902)', 'GRIN2A (2903)', 'GRIN2B (2904)', 'GRIN2C (2905)', 'GRIN2D (2906)', 'GRIN3A (116443)', 'GRIN3B (116444)', 'GRINA (2907)', 'GRIP1 (23426)', 'GRIPAP1 (56850)', 'GRK2 (156)', 'GRK3 (157)', 'GRK4 (2868)', 'GRK5 (2869)', 'GRK6 (2870)', 'GRK7 (131890)', 'GRM1 (2911)', 'GRM2 (2912)', 'GRM3 (2913)', 'GRM4 (2914)', 'GRM5 (2915)', 'GRM6 (2916)', 'GRM7 (2917)', 'GRM8 (2918)', 'GRN (2896)', 'GRP (2922)', 'GRPEL1 (80273)', 'GRPEL2 (134266)', 'GRPR (2925)', 'GRSF1 (2926)', 'GRTP1 (79774)', 'GRWD1 (83743)', 'GRXCR1 (389207)', 'GRXCR2 (643226)', 'GSAP (54103)', 'GSC (145258)', 'GSC2 (2928)', 'GSDMA (284110)', 'GSDMB (55876)', 'GSDMC (56169)', 'GSDMD (79792)', 'GSDME (1687)', 'GSE1 (23199)', 'GSG1 (83445)', 'GSG1L (146395)', 'GSK3A (2931)', 'GSK3B (2932)', 'GSKIP (51527)', 'GSN (2934)', 'GSPT1 (2935)', 'GSPT2 (23708)', 'GSR (2936)', 'GSS (2937)', 'GSTA1 (2938)', 'GSTA2 (2939)', 'GSTA3 (2940)', 'GSTA4 (2941)', 'GSTA5 (221357)', 'GSTCD (79807)', 'GSTK1 (373156)', 'GSTM1 (2944)', 'GSTM2 (2946)', 'GSTM3 (2947)', 'GSTM4 (2948)', 'GSTM5 (2949)', 'GSTO1 (9446)', 'GSTO2 (119391)', 'GSTP1 (2950)', 'GSTT2B (653689)', 'GSTZ1 (2954)', 'GSX1 (219409)', 'GSX2 (170825)', 'GTDC1 (79712)', 'GTF2A1 (2957)', 'GTF2A1L (11036)', 'GTF2A2 (2958)', 'GTF2B (2959)', 'GTF2E1 (2960)', 'GTF2E2 (2961)', 'GTF2F1 (2962)', 'GTF2F2 (2963)', 'GTF2H1 (2965)', 'GTF2H3 (2967)', 'GTF2H4 (2968)', 'GTF2H5 (404672)', 'GTF2I (2969)', 'GTF2IRD1 (9569)', 'GTF3A (2971)', 'GTF3C1 (2975)', 'GTF3C2 (2976)', 'GTF3C3 (9330)', 'GTF3C4 (9329)', 'GTF3C5 (9328)', 'GTF3C6 (112495)', 'GTPBP1 (9567)', 'GTPBP10 (85865)', 'GTPBP2 (54676)', 'GTPBP3 (84705)', 'GTPBP4 (23560)', 'GTPBP8 (29083)', 'GTSE1 (51512)', 'GTSF1 (121355)', 'GTSF1L (149699)', 'GUCA1ANB-GUCA1A (118142757)', 'GUCA1B (2979)', 'GUCA1C (9626)', 'GUCA2A (2980)', 'GUCA2B (2981)', 'GUCD1 (83606)', 'GUCY1A1 (2982)', 'GUCY1A2 (2977)', 'GUCY1B1 (2983)', 'GUCY2C (2984)', 'GUCY2D (3000)', 'GUCY2F (2986)', 'GUF1 (60558)', 'GUK1 (2987)', 'GULP1 (51454)', 'GUSB (2990)', 'GXYLT1 (283464)', 'GXYLT2 (727936)', 'GYG1 (2992)', 'GYG2 (8908)', 'GYPA (2993)', 'GYPC (2995)', 'GYPE (2996)', 'GYS1 (2997)', 'GYS2 (2998)', 'GZF1 (64412)', 'GZMA (3001)', 'GZMB (3002)', 'GZMH (2999)', 'GZMK (3003)', 'GZMM (3004)', 'H1-0 (3005)', 'H1-1 (3024)', 'H1-10 (8971)', 'H1-2 (3006)', 'H1-3 (3007)', 'H1-4 (3008)', 'H1-5 (3009)', 'H1-6 (3010)', 'H1-7 (341567)', 'H1-8 (132243)', 'H2AC1 (221613)', 'H2AC11 (8969)', 'H2AC12 (85235)', 'H2AC15 (8330)', 'H2AC16 (8332)', 'H2AC17 (8336)', 'H2AC20 (8338)', 'H2AC21 (317772)', 'H2AC25 (92815)', 'H2AC4 (8335)', 'H2AC6 (8334)', 'H2AC7 (3013)', 'H2AC8 (3012)', 'H2AJ (55766)', 'H2AP (25763)', 'H2AX (3014)', 'H2AZ1 (3015)', 'H2AZ2 (94239)', 'H2BC1 (255626)', 'H2BC10 (8346)', 'H2BC11 (8970)', 'H2BC13 (8340)', 'H2BC14 (8342)', 'H2BC15 (8341)', 'H2BC17 (8348)', 'H2BC18 (440689)', 'H2BC21 (8349)', 'H2BC26 (128312)', 'H2BC3 (3018)', 'H2BC4 (8347)', 'H2BC5 (3017)', 'H2BC6 (8344)', 'H2BC7 (8343)', 'H2BC8 (8339)', 'H2BC9 (8345)', 'H2BW1 (158983)', 'H2BW2 (286436)', 'H3-3A (3020)', 'H3-3B (3021)', 'H3-4 (8290)', 'H3-5 (440093)', 'H3C1 (8350)', 'H3C10 (8357)', 'H3C11 (8354)', 'H3C12 (8356)', 'H3C2 (8358)', 'H3C3 (8352)', 'H3C4 (8351)', 'H3C6 (8353)', 'H3C7 (8968)', 'H3C8 (8355)', 'H4C1 (8359)', 'H4C13 (8368)', 'H4C16 (121504)', 'H4C2 (8366)', 'H4C3 (8364)', 'H4C4 (8360)', 'H4C5 (8367)', 'H4C6 (8361)', 'H4C7 (8369)', 'H4C8 (8365)', 'H4C9 (8294)', 'H6PD (9563)', 'HAAO (23498)', 'HABP2 (3026)', 'HABP4 (22927)', 'HACD1 (9200)', 'HACD2 (201562)', 'HACD3 (51495)', 'HACD4 (401494)', 'HACE1 (57531)', 'HACL1 (26061)', 'HADH (3033)', 'HADHA (3030)', 'HADHB (3032)', 'HAGH (3029)', 'HAGHL (84264)', 'HAL (3034)', 'HAMP (57817)', 'HAND1 (9421)', 'HAND2 (9464)', 'HAO1 (54363)', 'HAO2 (51179)', 'HAP1 (9001)', 'HAPLN1 (1404)', 'HAPLN2 (60484)', 'HAPLN3 (145864)', 'HAPLN4 (404037)', 'HAPSTR1 (29035)', 'HARBI1 (283254)', 'HARS1 (3035)', 'HARS2 (23438)', 'HAS1 (3036)', 'HAS2 (3037)', 'HAS3 (3038)', 'HASPIN (83903)', 'HAT1 (8520)', 'HAUS1 (115106)', 'HAUS2 (55142)', 'HAUS3 (79441)', 'HAUS4 (54930)', 'HAUS5 (23354)', 'HAUS6 (54801)', 'HAUS7 (55559)', 'HAUS8 (93323)', 'HAVCR1 (26762)', 'HAVCR2 (84868)', 'HAX1 (10456)', 'HBB (3043)', 'HBD (3045)', 'HBE1 (3046)', 'HBEGF (1839)', 'HBM (3042)', 'HBP1 (26959)', 'HBQ1 (3049)', 'HBS1L (10767)', 'HBZ (3050)', 'HCAR1 (27198)', 'HCAR2 (338442)', 'HCAR3 (8843)', 'HCCS (3052)', 'HCFC1 (3054)', 'HCFC1R1 (54985)', 'HCFC2 (29915)', 'HCK (3055)', 'HCLS1 (3059)', 'HCN1 (348980)', 'HCN2 (610)', 'HCN3 (57657)', 'HCN4 (10021)', 'HCRT (3060)', 'HCRTR1 (3061)', 'HCRTR2 (3062)', 'HCST (10870)', 'HDAC1 (3065)', 'HDAC10 (83933)', 'HDAC11 (79885)', 'HDAC2 (3066)', 'HDAC3 (8841)', 'HDAC4 (9759)', 'HDAC5 (10014)', 'HDAC6 (10013)', 'HDAC7 (51564)', 'HDAC8 (55869)', 'HDAC9 (9734)', 'HDC (3067)', 'HDDC2 (51020)', 'HDDC3 (374659)', 'HDGF (3068)', 'HDGFL1 (154150)', 'HDGFL2 (84717)', 'HDGFL3 (50810)', 'HDHD2 (84064)', 'HDHD3 (81932)', 'HDHD5 (27440)', 'HDLBP (3069)', 'HDX (139324)', 'HEATR1 (55127)', 'HEATR3 (55027)', 'HEATR4 (399671)', 'HEATR5B (54497)', 'HEATR6 (63897)', 'HEATR9 (256957)', 'HEBP1 (50865)', 'HEBP2 (23593)', 'HECA (51696)', 'HECTD1 (25831)', 'HECTD2 (143279)', 'HECTD3 (79654)', 'HECW1 (23072)', 'HECW2 (57520)', 'HEG1 (57493)', 'HELB (92797)', 'HELLS (3070)', 'HELQ (113510)', 'HELT (391723)', 'HELZ (9931)', 'HELZ2 (85441)', 'HEMGN (55363)', 'HEMK1 (51409)', 'HENMT1 (113802)', 'HEPACAM (220296)', 'HEPACAM2 (253012)', 'HEPH (9843)', 'HEPHL1 (341208)', 'HERC1 (8925)', 'HERC2 (8924)', 'HERC3 (8916)', 'HERC4 (26091)', 'HERC5 (51191)', 'HERC6 (55008)', 'HERPUD1 (9709)', 'HERPUD2 (64224)', 'HES1 (3280)', 'HES2 (54626)', 'HES3 (390992)', 'HES4 (57801)', 'HES5 (388585)', 'HES6 (55502)', 'HES7 (84667)', 'HESX1 (8820)', 'HEXA (3073)', 'HEXB (3074)', 'HEXD (284004)', 'HEXIM1 (10614)', 'HEXIM2 (124790)', 'HEY1 (23462)', 'HEY2 (23493)', 'HEYL (26508)', 'HFE (3077)', 'HFM1 (164045)', 'HGD (3081)', 'HGF (3082)', 'HGFAC (3083)', 'HGH1 (51236)', 'HGS (9146)', 'HGSNAT (138050)', 'HHAT (55733)', 'HHATL (57467)', 'HHEX (3087)', 'HHIP (64399)', 'HHIPL1 (84439)', 'HHIPL2 (79802)', 'HHLA2 (11148)', 'HIBADH (11112)', 'HIBCH (26275)', 'HIC1 (3090)', 'HIC2 (23119)', 'HID1 (283987)', 'HIF1A (3091)', 'HIF1AN (55662)', 'HIF3A (64344)', 'HIGD1A (25994)', 'HIGD1B (51751)', 'HIGD1C (613227)', 'HIGD2A (192286)', 'HIKESHI (51501)', 'HILPDA (29923)', 'HINFP (25988)', 'HINT1 (3094)', 'HINT2 (84681)', 'HINT3 (135114)', 'HIP1 (3092)', 'HIP1R (9026)', 'HIPK1 (204851)', 'HIPK3 (10114)', 'HIPK4 (147746)', 'HIRA (7290)', 'HIRIP3 (8479)', 'HIVEP1 (3096)', 'HIVEP2 (3097)', 'HIVEP3 (59269)', 'HJURP (55355)', 'HJV (148738)', 'HK1 (3098)', 'HK2 (3099)', 'HK3 (3101)', 'HKDC1 (80201)', 'HLA-A (3105)', 'HLA-B (3106)', 'HLA-C (3107)', 'HLA-DMA (3108)', 'HLA-DMB (3109)', 'HLA-DOA (3111)', 'HLA-DOB (3112)', 'HLA-DPA1 (3113)', 'HLA-DPB1 (3115)', 'HLA-DQA1 (3117)', 'HLA-DQA2 (3118)', 'HLA-DQB1 (3119)', 'HLA-DQB2 (3120)', 'HLA-DRA (3122)', 'HLA-DRB1 (3123)', 'HLA-DRB5 (3127)', 'HLA-E (3133)', 'HLA-F (3134)', 'HLA-G (3135)', 'HLCS (3141)', 'HLF (3131)', 'HLTF (6596)', 'HLX (3142)', 'HM13 (81502)', 'HMBOX1 (79618)', 'HMBS (3145)', 'HMCES (56941)', 'HMCN1 (83872)', 'HMG20A (10363)', 'HMG20B (10362)', 'HMGA1 (3159)', 'HMGA2 (8091)', 'HMGB1 (3146)', 'HMGB2 (3148)', 'HMGB3 (3149)', 'HMGB4 (127540)', 'HMGCL (3155)', 'HMGCLL1 (54511)', 'HMGCR (3156)', 'HMGCS1 (3157)', 'HMGCS2 (3158)', 'HMGN1 (3150)', 'HMGN2 (3151)', 'HMGN3 (9324)', 'HMGN4 (10473)', 'HMGN5 (79366)', 'HMGXB3 (22993)', 'HMGXB4 (10042)', 'HMHB1 (57824)', 'HMMR (3161)', 'HMOX1 (3162)', 'HMOX2 (3163)', 'HMSD (284293)', 'HMX1 (3166)', 'HMX2 (3167)', 'HMX3 (340784)', 'HNF1A (6927)', 'HNF1B (6928)', 'HNF4A (3172)', 'HNF4G (3174)', 'HNMT (3176)', 'HNRNPA0 (10949)', 'HNRNPA1 (3178)', 'HNRNPA1L2 (144983)', 'HNRNPA2B1 (3181)', 'HNRNPA3 (220988)', 'HNRNPAB (3182)', 'HNRNPC (3183)', 'HNRNPD (3184)', 'HNRNPDL (9987)', 'HNRNPF (3185)', 'HNRNPH1 (3187)', 'HNRNPH2 (3188)', 'HNRNPH3 (3189)', 'HNRNPK (3190)', 'HNRNPL (3191)', 'HNRNPLL (92906)', 'HNRNPM (4670)', 'HNRNPR (10236)', 'HNRNPU (3192)', 'HNRNPUL1 (11100)', 'HNRNPUL2 (221092)', 'HOATZ (399949)', 'HOGA1 (112817)', 'HOMER1 (9456)', 'HOMER2 (9455)', 'HOMER3 (9454)', 'HOMEZ (57594)', 'HOOK1 (51361)', 'HOOK2 (29911)', 'HOOK3 (84376)', 'HOPX (84525)', 'HORMAD1 (84072)', 'HORMAD2 (150280)', 'HOXA1 (3198)', 'HOXA10 (3206)', 'HOXA11 (3207)', 'HOXA13 (3209)', 'HOXA2 (3199)', 'HOXA3 (3200)', 'HOXA4 (3201)', 'HOXA5 (3202)', 'HOXA6 (3203)', 'HOXA7 (3204)', 'HOXA9 (3205)', 'HOXB1 (3211)', 'HOXB13 (10481)', 'HOXB2 (3212)', 'HOXB3 (3213)', 'HOXB4 (3214)', 'HOXB5 (3215)', 'HOXB6 (3216)', 'HOXB7 (3217)', 'HOXB8 (3218)', 'HOXB9 (3219)', 'HOXC10 (3226)', 'HOXC11 (3227)', 'HOXC12 (3228)', 'HOXC13 (3229)', 'HOXC4 (3221)', 'HOXC5 (3222)', 'HOXC6 (3223)', 'HOXC8 (3224)', 'HOXC9 (3225)', 'HOXD1 (3231)', 'HOXD10 (3236)', 'HOXD11 (3237)', 'HOXD12 (3238)', 'HOXD13 (3239)', 'HOXD3 (3232)', 'HOXD4 (3233)', 'HOXD8 (3234)', 'HOXD9 (3235)', 'HP (3240)', 'HP1BP3 (50809)', 'HPCA (3208)', 'HPCAL1 (3241)', 'HPCAL4 (51440)', 'HPD (3242)', 'HPDL (84842)', 'HPF1 (54969)', 'HPGD (3248)', 'HPGDS (27306)', 'HPN (3249)', 'HPR (3250)', 'HPRT1 (3251)', 'HPS1 (3257)', 'HPS3 (84343)', 'HPS4 (89781)', 'HPS5 (11234)', 'HPS6 (79803)', 'HPSE (10855)', 'HPSE2 (60495)', 'HPX (3263)', 'HR (55806)', 'HRAS (3265)', 'HRC (3270)', 'HRCT1 (646962)', 'HRG (3273)', 'HRH1 (3269)', 'HRH2 (3274)', 'HRH3 (11255)', 'HRH4 (59340)', 'HRK (8739)', 'HRNR (388697)', 'HROB (78995)', 'HS1BP3 (64342)', 'HS2ST1 (9653)', 'HS3ST1 (9957)', 'HS3ST2 (9956)', 'HS3ST3A1 (9955)', 'HS3ST3B1 (9953)', 'HS3ST4 (9951)', 'HS3ST5 (222537)', 'HS3ST6 (64711)', 'HS6ST1 (9394)', 'HS6ST2 (90161)', 'HS6ST3 (266722)', 'HSBP1 (3281)', 'HSBP1L1 (440498)', 'HSCB (150274)', 'HSD11B1 (3290)', 'HSD11B1L (374875)', 'HSD11B2 (3291)', 'HSD17B1 (3292)', 'HSD17B10 (3028)', 'HSD17B11 (51170)', 'HSD17B12 (51144)', 'HSD17B13 (345275)', 'HSD17B14 (51171)', 'HSD17B2 (3294)', 'HSD17B3 (3293)', 'HSD17B4 (3295)', 'HSD17B6 (8630)', 'HSD17B7 (51478)', 'HSD17B8 (7923)', 'HSD3B1 (3283)', 'HSD3B2 (3284)', 'HSD3B7 (80270)', 'HSDL1 (83693)', 'HSDL2 (84263)', 'HSF1 (3297)', 'HSF2 (3298)', 'HSF2BP (11077)', 'HSF4 (3299)', 'HSF5 (124535)', 'HSP90AA1 (3320)', 'HSP90AB1 (3326)', 'HSP90B1 (7184)', 'HSPA12A (259217)', 'HSPA12B (116835)', 'HSPA13 (6782)', 'HSPA14 (51182)', 'HSPA1L (3305)', 'HSPA2 (3306)', 'HSPA4 (3308)', 'HSPA4L (22824)', 'HSPA5 (3309)', 'HSPA6 (3310)', 'HSPA8 (3312)', 'HSPA9 (3313)', 'HSPB1 (3315)', 'HSPB2 (3316)', 'HSPB3 (8988)', 'HSPB6 (126393)', 'HSPB7 (27129)', 'HSPB8 (26353)', 'HSPB9 (94086)', 'HSPBAP1 (79663)', 'HSPBP1 (23640)', 'HSPD1 (3329)', 'HSPE1 (3336)', 'HSPG2 (3339)', 'HSPH1 (10808)', 'HTATIP2 (10553)', 'HTATSF1 (27336)', 'HTN1 (3346)', 'HTN3 (3347)', 'HTR1A (3350)', 'HTR1B (3351)', 'HTR1D (3352)', 'HTR1E (3354)', 'HTR1F (3355)', 'HTR2A (3356)', 'HTR2B (3357)', 'HTR2C (3358)', 'HTR3A (3359)', 'HTR3B (9177)', 'HTR3C (170572)', 'HTR3D (200909)', 'HTR3E (285242)', 'HTR4 (3360)', 'HTR5A (3361)', 'HTR6 (3362)', 'HTR7 (3363)', 'HTRA1 (5654)', 'HTRA2 (27429)', 'HTRA3 (94031)', 'HTRA4 (203100)', 'HTT (3064)', 'HUNK (30811)', 'HUS1 (3364)', 'HUS1B (135458)', 'HUWE1 (10075)', 'HVCN1 (84329)', 'HYAL1 (3373)', 'HYAL2 (8692)', 'HYAL3 (8372)', 'HYAL4 (23553)', 'HYCC1 (84668)', 'HYCC2 (285172)', 'HYDIN (54768)', 'HYI (81888)', 'HYKK (123688)', 'HYLS1 (219844)', 'HYOU1 (10525)', 'HYPK (25764)', 'IAH1 (285148)', 'IAPP (3375)', 'IARS1 (3376)', 'IARS2 (55699)', 'IBA57 (200205)', 'IBSP (3381)', 'IBTK (25998)', 'ICA1 (3382)', 'ICA1L (130026)', 'ICAM1 (3383)', 'ICAM2 (3384)', 'ICAM3 (3385)', 'ICAM4 (3386)', 'ICAM5 (7087)', 'ICE1 (23379)', 'ICE2 (79664)', 'ICMT (23463)', 'ICOS (29851)', 'ICOSLG (23308)', 'ID1 (3397)', 'ID2 (3398)', 'ID3 (3399)', 'ID4 (3400)', 'IDE (3416)', 'IDH1 (3417)', 'IDH2 (3418)', 'IDH3A (3419)', 'IDH3B (3420)', 'IDH3G (3421)', 'IDI1 (3422)', 'IDI2 (91734)', 'IDNK (414328)', 'IDO1 (3620)', 'IDS (3423)', 'IDUA (3425)', 'IER2 (9592)', 'IER3 (8870)', 'IER3IP1 (51124)', 'IER5 (51278)', 'IER5L (389792)', 'IFFO1 (25900)', 'IFFO2 (126917)', 'IFI16 (3428)', 'IFI27 (3429)', 'IFI27L1 (122509)', 'IFI27L2 (83982)', 'IFI30 (10437)', 'IFI35 (3430)', 'IFI44 (10561)', 'IFI44L (10964)', 'IFI6 (2537)', 'IFIH1 (64135)', 'IFIT1 (3434)', 'IFIT1B (439996)', 'IFIT2 (3433)', 'IFIT3 (3437)', 'IFIT5 (24138)', 'IFITM1 (8519)', 'IFITM10 (402778)', 'IFITM2 (10581)', 'IFITM3 (10410)', 'IFITM5 (387733)', 'IFNA1 (3439)', 'IFNA10 (3446)', 'IFNA14 (3448)', 'IFNA16 (3449)', 'IFNA17 (3451)', 'IFNA2 (3440)', 'IFNA21 (3452)', 'IFNA4 (3441)', 'IFNA5 (3442)', 'IFNA6 (3443)', 'IFNA7 (3444)', 'IFNA8 (3445)', 'IFNAR1 (3454)', 'IFNAR2 (3455)', 'IFNB1 (3456)', 'IFNE (338376)', 'IFNG (3458)', 'IFNGR1 (3459)', 'IFNGR2 (3460)', 'IFNK (56832)', 'IFNL1 (282618)', 'IFNL2 (282616)', 'IFNL3 (282617)', 'IFNLR1 (163702)', 'IFNW1 (3467)', 'IFRD1 (3475)', 'IFRD2 (7866)', 'IFT122 (55764)', 'IFT140 (9742)', 'IFT172 (26160)', 'IFT20 (90410)', 'IFT22 (64792)', 'IFT25 (51668)', 'IFT27 (11020)', 'IFT43 (112752)', 'IFT46 (56912)', 'IFT52 (51098)', 'IFT56 (79989)', 'IFT57 (55081)', 'IFT70A (92104)', 'IFT70B (150737)', 'IFT74 (80173)', 'IFT80 (57560)', 'IFT81 (28981)', 'IFT88 (8100)', 'IFTAP (119710)', 'IGBP1 (3476)', 'IGDCC3 (9543)', 'IGDCC4 (57722)', 'IGF1 (3479)', 'IGF1R (3480)', 'IGF2 (3481)', 'IGF2BP1 (10642)', 'IGF2BP2 (10644)', 'IGF2BP3 (10643)', 'IGF2R (3482)', 'IGFALS (3483)', 'IGFBP1 (3484)', 'IGFBP2 (3485)', 'IGFBP3 (3486)', 'IGFBP4 (3487)', 'IGFBP5 (3488)', 'IGFBP6 (3489)', 'IGFBP7 (3490)', 'IGFBPL1 (347252)', 'IGFL1 (374918)', 'IGFL2 (147920)', 'IGFL3 (388555)', 'IGFL4 (444882)', 'IGFLR1 (79713)', 'IGFN1 (91156)', 'IGHMBP2 (3508)', 'IGIP (492311)', 'IGLL1 (3543)', 'IGLL5 (100423062)', 'IGLON5 (402665)', 'IGSF1 (3547)', 'IGSF10 (285313)', 'IGSF11 (152404)', 'IGSF21 (84966)', 'IGSF22 (283284)', 'IGSF23 (147710)', 'IGSF3 (3321)', 'IGSF5 (150084)', 'IGSF6 (10261)', 'IGSF8 (93185)', 'IGSF9 (57549)', 'IGSF9B (22997)', 'IHH (3549)', 'IHO1 (339834)', 'IK (3550)', 'IKBIP (121457)', 'IKBKB (3551)', 'IKBKE (9641)', 'IKBKG (8517)', 'IKZF1 (10320)', 'IKZF2 (22807)', 'IKZF3 (22806)', 'IKZF4 (64375)', 'IKZF5 (64376)', 'IL10 (3586)', 'IL10RA (3587)', 'IL10RB (3588)', 'IL11 (3589)', 'IL11RA (3590)', 'IL12A (3592)', 'IL12B (3593)', 'IL12RB1 (3594)', 'IL12RB2 (3595)', 'IL13 (3596)', 'IL13RA1 (3597)', 'IL13RA2 (3598)', 'IL15 (3600)', 'IL15RA (3601)', 'IL16 (3603)', 'IL17A (3605)', 'IL17B (27190)', 'IL17C (27189)', 'IL17D (53342)', 'IL17F (112744)', 'IL17RA (23765)', 'IL17RB (55540)', 'IL17RC (84818)', 'IL17RD (54756)', 'IL17RE (132014)', 'IL17REL (400935)', 'IL18 (3606)', 'IL18BP (10068)', 'IL18R1 (8809)', 'IL18RAP (8807)', 'IL19 (29949)', 'IL1A (3552)', 'IL1B (3553)', 'IL1F10 (84639)', 'IL1R1 (3554)', 'IL1R2 (7850)', 'IL1RAP (3556)', 'IL1RAPL1 (11141)', 'IL1RAPL2 (26280)', 'IL1RL1 (9173)', 'IL1RL2 (8808)', 'IL1RN (3557)', 'IL2 (3558)', 'IL20 (50604)', 'IL20RA (53832)', 'IL20RB (53833)', 'IL21 (59067)', 'IL21R (50615)', 'IL22 (50616)', 'IL22RA1 (58985)', 'IL22RA2 (116379)', 'IL23A (51561)', 'IL23R (149233)', 'IL24 (11009)', 'IL25 (64806)', 'IL26 (55801)', 'IL27 (246778)', 'IL27RA (9466)', 'IL2RA (3559)', 'IL2RB (3560)', 'IL2RG (3561)', 'IL3 (3562)', 'IL31 (386653)', 'IL31RA (133396)', 'IL32 (9235)', 'IL33 (90865)', 'IL34 (146433)', 'IL36A (27179)', 'IL36B (27177)', 'IL36G (56300)', 'IL36RN (26525)', 'IL37 (27178)', 'IL4 (3565)', 'IL4I1 (259307)', 'IL4R (3566)', 'IL5 (3567)', 'IL5RA (3568)', 'IL6 (3569)', 'IL6R (3570)', 'IL6ST (3572)', 'IL7 (3574)', 'IL7R (3575)', 'IL9 (3578)', 'IL9R (3581)', 'ILDR1 (286676)', 'ILDR2 (387597)', 'ILF2 (3608)', 'ILF3 (3609)', 'ILK (3611)', 'ILKAP (80895)', 'ILRUN (64771)', 'ILVBL (10994)', 'IMMP1L (196294)', 'IMMP2L (83943)', 'IMMT (10989)', 'IMP3 (55272)', 'IMP4 (92856)', 'IMPA1 (3612)', 'IMPA2 (3613)', 'IMPACT (55364)', 'IMPDH1 (3614)', 'IMPDH2 (3615)', 'IMPG1 (3617)', 'IMPG2 (50939)', 'INA (9118)', 'INAFM1 (255783)', 'INAVA (55765)', 'INCA1 (388324)', 'INCENP (3619)', 'INF2 (64423)', 'ING1 (3621)', 'ING2 (3622)', 'ING3 (54556)', 'ING4 (51147)', 'ING5 (84289)', 'INHA (3623)', 'INHBA (3624)', 'INHBB (3625)', 'INHBC (3626)', 'INHBE (83729)', 'INIP (58493)', 'INKA1 (389119)', 'INKA2 (55924)', 'INMT (11185)', 'INO80 (54617)', 'INO80B (83444)', 'INO80C (125476)', 'INO80D (54891)', 'INO80E (283899)', 'INPP1 (3628)', 'INPP4A (3631)', 'INPP4B (8821)', 'INPP5A (3632)', 'INPP5B (3633)', 'INPP5E (56623)', 'INPP5F (22876)', 'INPP5J (27124)', 'INPP5K (51763)', 'INPPL1 (3636)', 'INS (3630)', 'INSC (387755)', 'INSIG1 (3638)', 'INSIG2 (51141)', 'INSL3 (3640)', 'INSL4 (3641)', 'INSL5 (10022)', 'INSL6 (11172)', 'INSM1 (3642)', 'INSM2 (84684)', 'INSR (3643)', 'INSRR (3645)', 'INSYN1 (388135)', 'INSYN2A (642938)', 'INSYN2B (100131897)', 'INTS1 (26173)', 'INTS10 (55174)', 'INTS11 (54973)', 'INTS12 (57117)', 'INTS13 (55726)', 'INTS14 (81556)', 'INTS15 (79034)', 'INTS2 (57508)', 'INTS3 (65123)', 'INTS4 (92105)', 'INTS5 (80789)', 'INTS6 (26512)', 'INTS6L (203522)', 'INTS7 (25896)', 'INTS8 (55656)', 'INTS9 (55756)', 'INTU (27152)', 'INVS (27130)', 'IP6K1 (9807)', 'IP6K2 (51447)', 'IP6K3 (117283)', 'IPCEF1 (26034)', 'IPMK (253430)', 'IPO11 (51194)', 'IPO13 (9670)', 'IPO4 (79711)', 'IPO5 (3843)', 'IPO7 (10527)', 'IPO8 (10526)', 'IPO9 (55705)', 'IPP (3652)', 'IPPK (64768)', 'IQCA1 (79781)', 'IQCB1 (9657)', 'IQCC (55721)', 'IQCD (115811)', 'IQCE (23288)', 'IQCF1 (132141)', 'IQCF2 (389123)', 'IQCF3 (401067)', 'IQCF5 (389124)', 'IQCF6 (440956)', 'IQCG (84223)', 'IQCH (64799)', 'IQCJ (654502)', 'IQCK (124152)', 'IQCN (80726)', 'IQGAP1 (8826)', 'IQGAP2 (10788)', 'IQGAP3 (128239)', 'IQSEC1 (9922)', 'IQSEC2 (23096)', 'IQSEC3 (440073)', 'IQUB (154865)', 'IRAG1 (10335)', 'IRAG2 (4033)', 'IRAK1 (3654)', 'IRAK1BP1 (134728)', 'IRAK2 (3656)', 'IRAK3 (11213)', 'IRAK4 (51135)', 'IREB2 (3658)', 'IRF1 (3659)', 'IRF2 (3660)', 'IRF2BP1 (26145)', 'IRF2BP2 (359948)', 'IRF2BPL (64207)', 'IRF3 (3661)', 'IRF4 (3662)', 'IRF5 (3663)', 'IRF6 (3664)', 'IRF7 (3665)', 'IRF8 (3394)', 'IRF9 (10379)', 'IRGC (56269)', 'IRGM (345611)', 'IRGQ (126298)', 'IRS1 (3667)', 'IRS2 (8660)', 'IRS4 (8471)', 'IRX1 (79192)', 'IRX2 (153572)', 'IRX3 (79191)', 'IRX4 (50805)', 'IRX5 (10265)', 'IRX6 (79190)', 'ISCA1 (81689)', 'ISCA2 (122961)', 'ISCU (23479)', 'ISG15 (9636)', 'ISG20 (3669)', 'ISG20L2 (81875)', 'ISL1 (3670)', 'ISL2 (64843)', 'ISLR (3671)', 'ISLR2 (57611)', 'ISM1 (140862)', 'ISM2 (145501)', 'ISOC1 (51015)', 'ISOC2 (79763)', 'IST1 (9798)', 'ISX (91464)', 'ISY1 (57461)', 'ISYNA1 (51477)', 'ITCH (83737)', 'ITFG1 (81533)', 'ITFG2 (55846)', 'ITGA1 (3672)', 'ITGA10 (8515)', 'ITGA11 (22801)', 'ITGA2 (3673)', 'ITGA2B (3674)', 'ITGA3 (3675)', 'ITGA4 (3676)', 'ITGA5 (3678)', 'ITGA6 (3655)', 'ITGA7 (3679)', 'ITGA8 (8516)', 'ITGA9 (3680)', 'ITGAD (3681)', 'ITGAE (3682)', 'ITGAL (3683)', 'ITGAM (3684)', 'ITGAV (3685)', 'ITGAX (3687)', 'ITGB1 (3688)', 'ITGB1BP1 (9270)', 'ITGB1BP2 (26548)', 'ITGB2 (3689)', 'ITGB3 (3690)', 'ITGB3BP (23421)', 'ITGB4 (3691)', 'ITGB5 (3693)', 'ITGB6 (3694)', 'ITGB7 (3695)', 'ITGB8 (3696)', 'ITGBL1 (9358)', 'ITIH1 (3697)', 'ITIH2 (3698)', 'ITIH3 (3699)', 'ITIH4 (3700)', 'ITIH6 (347365)', 'ITK (3702)', 'ITLN1 (55600)', 'ITLN2 (142683)', 'ITM2A (9452)', 'ITM2B (9445)', 'ITM2C (81618)', 'ITPA (3704)', 'ITPK1 (3705)', 'ITPKA (3706)', 'ITPKB (3707)', 'ITPKC (80271)', 'ITPR1 (3708)', 'ITPR2 (3709)', 'ITPR3 (3710)', 'ITPRID1 (223075)', 'ITPRID2 (6744)', 'ITPRIP (85450)', 'ITPRIPL1 (150771)', 'ITPRIPL2 (162073)', 'ITSN1 (6453)', 'ITSN2 (50618)', 'IVD (3712)', 'IVL (3713)', 'IVNS1ABP (10625)', 'IWS1 (55677)', 'IYD (389434)', 'IZUMO1 (284359)', 'IZUMO2 (126123)', 'IZUMO3 (100129669)', 'IZUMO4 (113177)', 'JADE1 (79960)', 'JADE2 (23338)', 'JADE3 (9767)', 'JAG1 (182)', 'JAG2 (3714)', 'JAGN1 (84522)', 'JAK1 (3716)', 'JAK2 (3717)', 'JAK3 (3718)', 'JAKMIP1 (152789)', 'JAKMIP2 (9832)', 'JAKMIP3 (282973)', 'JAM2 (58494)', 'JAM3 (83700)', 'JAML (120425)', 'JARID2 (3720)', 'JAZF1 (221895)', 'JCAD (57608)', 'JCHAIN (3512)', 'JDP2 (122953)', 'JHY (79864)', 'JKAMP (51528)', 'JMJD1C (221037)', 'JMJD4 (65094)', 'JMJD6 (23210)', 'JMJD7 (100137047)', 'JMJD8 (339123)', 'JMY (133746)', 'JOSD1 (9929)', 'JOSD2 (126119)', 'JPH1 (56704)', 'JPH2 (57158)', 'JPH3 (57338)', 'JPH4 (84502)', 'JPT1 (51155)', 'JPT2 (90861)', 'JRKL (8690)', 'JSRP1 (126306)', 'JTB (10899)', 'JUN (3725)', 'JUNB (3726)', 'JUND (3727)', 'JUP (3728)', 'KALRN (8997)', 'KANK1 (23189)', 'KANK2 (25959)', 'KANK3 (256949)', 'KANK4 (163782)', 'KANSL1 (284058)', 'KANSL1L (151050)', 'KANSL2 (54934)', 'KANSL3 (55683)', 'KARS1 (3735)', 'KASH5 (147872)', 'KAT14 (57325)', 'KAT2A (2648)', 'KAT2B (8850)', 'KAT5 (10524)', 'KAT6A (7994)', 'KAT6B (23522)', 'KAT7 (11143)', 'KAT8 (84148)', 'KATNA1 (11104)', 'KATNAL1 (84056)', 'KATNAL2 (83473)', 'KATNB1 (10300)', 'KATNBL1 (79768)', 'KATNIP (23247)', 'KAZALD1 (81621)', 'KAZN (23254)', 'KBTBD11 (9920)', 'KBTBD12 (166348)', 'KBTBD13 (390594)', 'KBTBD2 (25948)', 'KBTBD3 (143879)', 'KBTBD4 (55709)', 'KBTBD6 (89890)', 'KBTBD7 (84078)', 'KBTBD8 (84541)', 'KCMF1 (56888)', 'KCNA1 (3736)', 'KCNA10 (3744)', 'KCNA2 (3737)', 'KCNA3 (3738)', 'KCNA4 (3739)', 'KCNA5 (3741)', 'KCNA6 (3742)', 'KCNA7 (3743)', 'KCNAB1 (7881)', 'KCNAB2 (8514)', 'KCNAB3 (9196)', 'KCNB1 (3745)', 'KCNB2 (9312)', 'KCNC1 (3746)', 'KCNC2 (3747)', 'KCNC3 (3748)', 'KCNC4 (3749)', 'KCND1 (3750)', 'KCND2 (3751)', 'KCND3 (3752)', 'KCNE1 (3753)', 'KCNE2 (9992)', 'KCNE3 (10008)', 'KCNE4 (23704)', 'KCNE5 (23630)', 'KCNF1 (3754)', 'KCNG1 (3755)', 'KCNG2 (26251)', 'KCNG3 (170850)', 'KCNG4 (93107)', 'KCNH1 (3756)', 'KCNH2 (3757)', 'KCNH3 (23416)', 'KCNH4 (23415)', 'KCNH5 (27133)', 'KCNH6 (81033)', 'KCNH7 (90134)', 'KCNH8 (131096)', 'KCNIP1 (30820)', 'KCNIP2 (30819)', 'KCNIP3 (30818)', 'KCNIP4 (80333)', 'KCNJ1 (3758)', 'KCNJ10 (3766)', 'KCNJ11 (3767)', 'KCNJ13 (3769)', 'KCNJ14 (3770)', 'KCNJ15 (3772)', 'KCNJ16 (3773)', 'KCNJ2 (3759)', 'KCNJ3 (3760)', 'KCNJ4 (3761)', 'KCNJ5 (3762)', 'KCNJ6 (3763)', 'KCNJ8 (3764)', 'KCNJ9 (3765)', 'KCNK1 (3775)', 'KCNK10 (54207)', 'KCNK12 (56660)', 'KCNK13 (56659)', 'KCNK15 (60598)', 'KCNK16 (83795)', 'KCNK17 (89822)', 'KCNK18 (338567)', 'KCNK2 (3776)', 'KCNK3 (3777)', 'KCNK4 (50801)', 'KCNK5 (8645)', 'KCNK6 (9424)', 'KCNK7 (10089)', 'KCNK9 (51305)', 'KCNMA1 (3778)', 'KCNMB1 (3779)', 'KCNMB2 (10242)', 'KCNMB3 (27094)', 'KCNMB4 (27345)', 'KCNN1 (3780)', 'KCNN2 (3781)', 'KCNN3 (3782)', 'KCNN4 (3783)', 'KCNQ1 (3784)', 'KCNQ2 (3785)', 'KCNQ3 (3786)', 'KCNQ4 (9132)', 'KCNQ5 (56479)', 'KCNRG (283518)', 'KCNS1 (3787)', 'KCNS2 (3788)', 'KCNS3 (3790)', 'KCNT1 (57582)', 'KCNT2 (343450)', 'KCNU1 (157855)', 'KCNV1 (27012)', 'KCNV2 (169522)', 'KCTD1 (284252)', 'KCTD10 (83892)', 'KCTD11 (147040)', 'KCTD12 (115207)', 'KCTD13 (253980)', 'KCTD14 (65987)', 'KCTD15 (79047)', 'KCTD16 (57528)', 'KCTD17 (79734)', 'KCTD18 (130535)', 'KCTD19 (146212)', 'KCTD2 (23510)', 'KCTD20 (222658)', 'KCTD21 (283219)', 'KCTD3 (51133)', 'KCTD4 (386618)', 'KCTD5 (54442)', 'KCTD6 (200845)', 'KCTD7 (154881)', 'KCTD8 (386617)', 'KCTD9 (54793)', 'KDELR1 (10945)', 'KDELR2 (11014)', 'KDELR3 (11015)', 'KDF1 (126695)', 'KDM1A (23028)', 'KDM1B (221656)', 'KDM2A (22992)', 'KDM2B (84678)', 'KDM3A (55818)', 'KDM3B (51780)', 'KDM4A (9682)', 'KDM4B (23030)', 'KDM4C (23081)', 'KDM4D (55693)', 'KDM4E (390245)', 'KDM5A (5927)', 'KDM5B (10765)', 'KDM5C (8242)', 'KDM5D (8284)', 'KDM6A (7403)', 'KDM6B (23135)', 'KDM7A (80853)', 'KDM8 (79831)', 'KDR (3791)', 'KDSR (2531)', 'KEAP1 (9817)', 'KEL (3792)', 'KERA (11081)', 'KGD4 (92259)', 'KHDC1 (80759)', 'KHDC1L (100129128)', 'KHDC3L (154288)', 'KHDC4 (22889)', 'KHDRBS1 (10657)', 'KHDRBS2 (202559)', 'KHDRBS3 (10656)', 'KHK (3795)', 'KHNYN (23351)', 'KHSRP (8570)', 'KIAA0232 (9778)', 'KIAA0319 (9856)', 'KIAA0319L (79932)', 'KIAA0408 (9729)', 'KIAA0513 (9764)', 'KIAA0586 (9786)', 'KIAA0753 (9851)', 'KIAA0825 (285600)', 'KIAA0930 (23313)', 'KIAA1143 (57456)', 'KIAA1191 (57179)', 'KIAA1210 (57481)', 'KIAA1217 (56243)', 'KIAA1328 (57536)', 'KIAA1549 (57670)', 'KIAA1549L (25758)', 'KIAA1586 (57691)', 'KIAA1614 (57710)', 'KIAA1671 (85379)', 'KIAA1755 (85449)', 'KIAA1958 (158405)', 'KIAA2013 (90231)', 'KICS2 (144577)', 'KIDINS220 (57498)', 'KIF11 (3832)', 'KIF12 (113220)', 'KIF13A (63971)', 'KIF13B (23303)', 'KIF14 (9928)', 'KIF15 (56992)', 'KIF16B (55614)', 'KIF17 (57576)', 'KIF18A (81930)', 'KIF18B (146909)', 'KIF19 (124602)', 'KIF1A (547)', 'KIF1B (23095)', 'KIF1C (10749)', 'KIF20A (10112)', 'KIF20B (9585)', 'KIF21A (55605)', 'KIF21B (23046)', 'KIF22 (3835)', 'KIF23 (9493)', 'KIF24 (347240)', 'KIF25 (3834)', 'KIF26A (26153)', 'KIF26B (55083)', 'KIF27 (55582)', 'KIF2A (3796)', 'KIF2B (84643)', 'KIF2C (11004)', 'KIF3A (11127)', 'KIF3B (9371)', 'KIF3C (3797)', 'KIF4A (24137)', 'KIF4B (285643)', 'KIF5A (3798)', 'KIF5B (3799)', 'KIF6 (221458)', 'KIF7 (374654)', 'KIF9 (64147)', 'KIFAP3 (22920)', 'KIFBP (26128)', 'KIFC1 (3833)', 'KIFC2 (90990)', 'KIFC3 (3801)', 'KIN (22944)', 'KIR2DL1 (3802)', 'KIR2DL3 (3804)', 'KIR2DL4 (3805)', 'KIR3DL1 (3811)', 'KIR3DL2 (3812)', 'KIR3DL3 (115653)', 'KIRREL1 (55243)', 'KIRREL2 (84063)', 'KIRREL3 (84623)', 'KISS1 (3814)', 'KISS1R (84634)', 'KIT (3815)', 'KITLG (4254)', 'KL (9365)', 'KLB (152831)', 'KLC1 (3831)', 'KLC2 (64837)', 'KLC3 (147700)', 'KLC4 (89953)', 'KLF1 (10661)', 'KLF10 (7071)', 'KLF11 (8462)', 'KLF12 (11278)', 'KLF13 (51621)', 'KLF14 (136259)', 'KLF15 (28999)', 'KLF16 (83855)', 'KLF17 (128209)', 'KLF2 (10365)', 'KLF3 (51274)', 'KLF4 (9314)', 'KLF5 (688)', 'KLF6 (1316)', 'KLF7 (8609)', 'KLF8 (11279)', 'KLF9 (687)', 'KLHDC1 (122773)', 'KLHDC10 (23008)', 'KLHDC2 (23588)', 'KLHDC3 (116138)', 'KLHDC4 (54758)', 'KLHDC7A (127707)', 'KLHDC7B (113730)', 'KLHDC8A (55220)', 'KLHDC8B (200942)', 'KLHDC9 (126823)', 'KLHL1 (57626)', 'KLHL10 (317719)', 'KLHL11 (55175)', 'KLHL12 (59349)', 'KLHL13 (90293)', 'KLHL14 (57565)', 'KLHL15 (80311)', 'KLHL17 (339451)', 'KLHL18 (23276)', 'KLHL2 (11275)', 'KLHL20 (27252)', 'KLHL21 (9903)', 'KLHL22 (84861)', 'KLHL24 (54800)', 'KLHL25 (64410)', 'KLHL26 (55295)', 'KLHL28 (54813)', 'KLHL29 (114818)', 'KLHL3 (26249)', 'KLHL30 (377007)', 'KLHL31 (401265)', 'KLHL32 (114792)', 'KLHL33 (123103)', 'KLHL34 (257240)', 'KLHL35 (283212)', 'KLHL36 (79786)', 'KLHL38 (340359)', 'KLHL4 (56062)', 'KLHL40 (131377)', 'KLHL41 (10324)', 'KLHL42 (57542)', 'KLHL5 (51088)', 'KLHL6 (89857)', 'KLHL7 (55975)', 'KLHL8 (57563)', 'KLHL9 (55958)', 'KLK1 (3816)', 'KLK10 (5655)', 'KLK11 (11012)', 'KLK12 (43849)', 'KLK13 (26085)', 'KLK14 (43847)', 'KLK15 (55554)', 'KLK2 (3817)', 'KLK3 (354)', 'KLK4 (9622)', 'KLK5 (25818)', 'KLK6 (5653)', 'KLK7 (5650)', 'KLK8 (11202)', 'KLK9 (284366)', 'KLKB1 (3818)', 'KLLN (100144748)', 'KLRB1 (3820)', 'KLRC1 (3821)', 'KLRC2 (3822)', 'KLRC3 (3823)', 'KLRC4 (8302)', 'KLRD1 (3824)', 'KLRF1 (51348)', 'KLRF2 (100431172)', 'KLRG1 (10219)', 'KLRG2 (346689)', 'KMO (8564)', 'KMT2A (4297)', 'KMT2B (9757)', 'KMT2C (58508)', 'KMT2D (8085)', 'KMT2E (55904)', 'KMT5A (387893)', 'KMT5B (51111)', 'KMT5C (84787)', 'KNCN (148930)', 'KNDC1 (85442)', 'KNG1 (3827)', 'KNL1 (57082)', 'KNOP1 (400506)', 'KNSTRN (90417)', 'KNTC1 (9735)', 'KPLCE (100129271)', 'KPNA1 (3836)', 'KPNA2 (3838)', 'KPNA3 (3839)', 'KPNA4 (3840)', 'KPNA5 (3841)', 'KPNA6 (23633)', 'KPNA7 (402569)', 'KPNB1 (3837)', 'KPRP (448834)', 'KPTN (11133)', 'KRAS (3845)', 'KRBA2 (124751)', 'KRBOX1 (100506243)', 'KRBOX5 (124411)', 'KRCC1 (51315)', 'KREMEN1 (83999)', 'KREMEN2 (79412)', 'KRI1 (65095)', 'KRIT1 (889)', 'KRR1 (11103)', 'KRT1 (3848)', 'KRT10 (3858)', 'KRT12 (3859)', 'KRT13 (3860)', 'KRT14 (3861)', 'KRT15 (3866)', 'KRT16 (3868)', 'KRT18 (3875)', 'KRT19 (3880)', 'KRT2 (3849)', 'KRT20 (54474)', 'KRT222 (125113)', 'KRT23 (25984)', 'KRT24 (192666)', 'KRT25 (147183)', 'KRT26 (353288)', 'KRT27 (342574)', 'KRT28 (162605)', 'KRT3 (3850)', 'KRT31 (3881)', 'KRT32 (3882)', 'KRT33A (3883)', 'KRT33B (3884)', 'KRT34 (3885)', 'KRT35 (3886)', 'KRT36 (8689)', 'KRT37 (8688)', 'KRT38 (8687)', 'KRT39 (390792)', 'KRT4 (3851)', 'KRT40 (125115)', 'KRT5 (3852)', 'KRT6A (3853)', 'KRT6B (3854)', 'KRT6C (286887)', 'KRT7 (3855)', 'KRT71 (112802)', 'KRT72 (140807)', 'KRT73 (319101)', 'KRT74 (121391)', 'KRT75 (9119)', 'KRT76 (51350)', 'KRT77 (374454)', 'KRT78 (196374)', 'KRT79 (338785)', 'KRT8 (3856)', 'KRT80 (144501)', 'KRT81 (3887)', 'KRT82 (3888)', 'KRT83 (3889)', 'KRT84 (3890)', 'KRT85 (3891)', 'KRT86 (3892)', 'KRT9 (3857)', 'KRTAP1-1 (81851)', 'KRTAP1-4 (728255)', 'KRTAP1-5 (83895)', 'KRTAP10-1 (386677)', 'KRTAP10-10 (353333)', 'KRTAP10-11 (386678)', 'KRTAP10-2 (386679)', 'KRTAP10-3 (386682)', 'KRTAP10-5 (386680)', 'KRTAP10-8 (386681)', 'KRTAP10-9 (386676)', 'KRTAP11-1 (337880)', 'KRTAP12-1 (353332)', 'KRTAP12-2 (353323)', 'KRTAP12-3 (386683)', 'KRTAP12-4 (386684)', 'KRTAP13-1 (140258)', 'KRTAP13-2 (337959)', 'KRTAP13-3 (337960)', 'KRTAP13-4 (284827)', 'KRTAP15-1 (254950)', 'KRTAP16-1 (100505753)', 'KRTAP17-1 (83902)', 'KRTAP19-1 (337882)', 'KRTAP19-2 (337969)', 'KRTAP19-3 (337970)', 'KRTAP19-4 (337971)', 'KRTAP19-5 (337972)', 'KRTAP19-6 (337973)', 'KRTAP19-7 (337974)', 'KRTAP19-8 (728299)', 'KRTAP20-1 (337975)', 'KRTAP20-2 (337976)', 'KRTAP20-3 (337985)', 'KRTAP21-1 (337977)', 'KRTAP21-2 (337978)', 'KRTAP21-3 (100288323)', 'KRTAP22-1 (337979)', 'KRTAP22-2 (100288287)', 'KRTAP23-1 (337963)', 'KRTAP24-1 (643803)', 'KRTAP25-1 (100131902)', 'KRTAP26-1 (388818)', 'KRTAP27-1 (643812)', 'KRTAP29-1 (100533177)', 'KRTAP3-1 (83896)', 'KRTAP3-2 (83897)', 'KRTAP3-3 (85293)', 'KRTAP4-11 (653240)', 'KRTAP4-2 (85291)', 'KRTAP4-3 (85290)', 'KRTAP4-4 (84616)', 'KRTAP4-5 (85289)', 'KRTAP4-6 (81871)', 'KRTAP4-8 (728224)', 'KRTAP4-9 (100132386)', 'KRTAP5-1 (387264)', 'KRTAP5-10 (387273)', 'KRTAP5-11 (440051)', 'KRTAP5-2 (440021)', 'KRTAP5-3 (387266)', 'KRTAP5-5 (439915)', 'KRTAP5-6 (440023)', 'KRTAP5-7 (440050)', 'KRTAP5-9 (3846)', 'KRTAP6-1 (337966)', 'KRTAP6-2 (337967)', 'KRTAP8-1 (337879)', 'KRTAP9-1 (728318)', 'KRTAP9-6 (100507608)', 'KRTAP9-8 (83901)', 'KRTAP9-9 (81870)', 'KRTCAP2 (200185)', 'KRTCAP3 (200634)', 'KRTDAP (388533)', 'KSR1 (8844)', 'KSR2 (283455)', 'KTI12 (112970)', 'KTN1 (3895)', 'KXD1 (79036)', 'KY (339855)', 'KYAT1 (883)', 'KYAT3 (56267)', 'KYNU (8942)', 'L1CAM (3897)', 'L1TD1 (54596)', 'L2HGDH (79944)', 'L3HYPDH (112849)', 'L3MBTL1 (26013)', 'L3MBTL2 (83746)', 'L3MBTL3 (84456)', 'L3MBTL4 (91133)', 'LACC1 (144811)', 'LACRT (90070)', 'LACTB (114294)', 'LACTB2 (51110)', 'LAD1 (3898)', 'LAG3 (3902)', 'LAGE3 (8270)', 'LAIR1 (3903)', 'LAIR2 (3904)', 'LALBA (3906)', 'LAMA1 (284217)', 'LAMA2 (3908)', 'LAMA3 (3909)', 'LAMA4 (3910)', 'LAMA5 (3911)', 'LAMB1 (3912)', 'LAMB2 (3913)', 'LAMB3 (3914)', 'LAMB4 (22798)', 'LAMC1 (3915)', 'LAMC2 (3918)', 'LAMC3 (10319)', 'LAMP1 (3916)', 'LAMP2 (3920)', 'LAMP3 (27074)', 'LAMP5 (24141)', 'LAMTOR1 (55004)', 'LAMTOR2 (28956)', 'LAMTOR3 (8649)', 'LAMTOR4 (389541)', 'LAMTOR5 (10542)', 'LANCL1 (10314)', 'LANCL2 (55915)', 'LANCL3 (347404)', 'LAP3 (51056)', 'LAPTM4A (9741)', 'LAPTM4B (55353)', 'LAPTM5 (7805)', 'LARGE1 (9215)', 'LARGE2 (120071)', 'LARP1 (23367)', 'LARP1B (55132)', 'LARP4 (113251)', 'LARP4B (23185)', 'LARP6 (55323)', 'LARP7 (51574)', 'LARS1 (51520)', 'LARS2 (23395)', 'LAS1L (81887)', 'LASP1 (3927)', 'LAT (27040)', 'LAT2 (7462)', 'LATS1 (9113)', 'LATS2 (26524)', 'LAX1 (54900)', 'LAYN (143903)', 'LBH (81606)', 'LBHD1 (79081)', 'LBP (3929)', 'LBR (3930)', 'LBX1 (10660)', 'LBX2 (85474)', 'LCA5 (167691)', 'LCA5L (150082)', 'LCAT (3931)', 'LCE1B (353132)', 'LCE1D (353134)', 'LCE1F (353137)', 'LCE2A (353139)', 'LCE2B (26239)', 'LCE2D (353141)', 'LCE3A (353142)', 'LCE3B (353143)', 'LCE3C (353144)', 'LCE3D (84648)', 'LCE3E (353145)', 'LCE4A (199834)', 'LCE5A (254910)', 'LCE6A (448835)', 'LCK (3932)', 'LCLAT1 (253558)', 'LCMT1 (51451)', 'LCMT2 (9836)', 'LCN1 (3933)', 'LCN10 (414332)', 'LCN12 (286256)', 'LCN15 (389812)', 'LCN2 (3934)', 'LCN6 (158062)', 'LCN8 (138307)', 'LCN9 (392399)', 'LCNL1 (401562)', 'LCOR (84458)', 'LCORL (254251)', 'LCP1 (3936)', 'LCP2 (3937)', 'LCT (3938)', 'LCTL (197021)', 'LDAF1 (57146)', 'LDAH (60526)', 'LDB1 (8861)', 'LDB2 (9079)', 'LDB3 (11155)', 'LDHA (3939)', 'LDHAL6A (160287)', 'LDHAL6B (92483)', 'LDHB (3945)', 'LDHC (3948)', 'LDHD (197257)', 'LDLR (3949)', 'LDLRAD1 (388633)', 'LDLRAD2 (401944)', 'LDLRAD3 (143458)', 'LDLRAD4 (753)', 'LDLRAP1 (26119)', 'LDOC1 (23641)', 'LEAP2 (116842)', 'LECT2 (3950)', 'LEF1 (51176)', 'LEFTY1 (10637)', 'LEFTY2 (7044)', 'LEKR1 (389170)', 'LELP1 (149018)', 'LEMD1 (93273)', 'LEMD2 (221496)', 'LEMD3 (23592)', 'LENEP (55891)', 'LENG1 (79165)', 'LENG8 (114823)', 'LENG9 (94059)', 'LEO1 (123169)', 'LEP (3952)', 'LEPR (3953)', 'LEPROT (54741)', 'LEPROTL1 (23484)', 'LETM1 (3954)', 'LETM2 (137994)', 'LETMD1 (25875)', 'LFNG (3955)', 'LGALS1 (3956)', 'LGALS12 (85329)', 'LGALS13 (29124)', 'LGALS14 (56891)', 'LGALS16 (148003)', 'LGALS2 (3957)', 'LGALS3 (3958)', 'LGALS3BP (3959)', 'LGALS4 (3960)', 'LGALS7 (3963)', 'LGALS7B (653499)', 'LGALS8 (3964)', 'LGALS9 (3965)', 'LGALS9B (284194)', 'LGALS9C (654346)', 'LGALSL (29094)', 'LGI1 (9211)', 'LGI2 (55203)', 'LGI3 (203190)', 'LGI4 (163175)', 'LGMN (5641)', 'LGR4 (55366)', 'LGR5 (8549)', 'LGR6 (59352)', 'LGSN (51557)', 'LHB (3972)', 'LHCGR (3973)', 'LHFPL1 (340596)', 'LHFPL2 (10184)', 'LHFPL4 (375323)', 'LHFPL5 (222662)', 'LHFPL6 (10186)', 'LHFPL7 (255349)', 'LHPP (64077)', 'LHX1 (3975)', 'LHX2 (9355)', 'LHX3 (8022)', 'LHX4 (89884)', 'LHX5 (64211)', 'LHX6 (26468)', 'LHX8 (431707)', 'LHX9 (56956)', 'LIAS (11019)', 'LIAT1 (400566)', 'LIF (3976)', 'LIFR (3977)', 'LIG1 (3978)', 'LIG3 (3980)', 'LIG4 (3981)', 'LILRA1 (11024)', 'LILRA2 (11027)', 'LILRA4 (23547)', 'LILRA5 (353514)', 'LILRB1 (10859)', 'LILRB2 (10288)', 'LILRB3 (11025)', 'LILRB4 (11006)', 'LILRB5 (10990)', 'LIM2 (3982)', 'LIMA1 (51474)', 'LIMCH1 (22998)', 'LIMD1 (8994)', 'LIMD2 (80774)', 'LIME1 (54923)', 'LIMK1 (3984)', 'LIMK2 (3985)', 'LIMS2 (55679)', 'LIN28A (79727)', 'LIN28B (389421)', 'LIN37 (55957)', 'LIN52 (91750)', 'LIN54 (132660)', 'LIN7A (8825)', 'LIN7B (64130)', 'LIN7C (55327)', 'LIN9 (286826)', 'LINGO1 (84894)', 'LINGO2 (158038)', 'LINGO3 (645191)', 'LINGO4 (339398)', 'LINS1 (55180)', 'LIPA (3988)', 'LIPC (3990)', 'LIPE (3991)', 'LIPF (8513)', 'LIPG (9388)', 'LIPH (200879)', 'LIPI (149998)', 'LIPJ (142910)', 'LIPK (643414)', 'LIPM (340654)', 'LIPN (643418)', 'LIPT1 (51601)', 'LIPT2 (387787)', 'LITAF (9516)', 'LIX1 (167410)', 'LIX1L (128077)', 'LKAAEAR1 (198437)', 'LLCFC1 (135927)', 'LLGL1 (3996)', 'LLGL2 (3993)', 'LLPH (84298)', 'LMAN1 (3998)', 'LMAN1L (79748)', 'LMAN2 (10960)', 'LMAN2L (81562)', 'LMBR1 (64327)', 'LMBR1L (55716)', 'LMBRD1 (55788)', 'LMBRD2 (92255)', 'LMCD1 (29995)', 'LMF1 (64788)', 'LMF2 (91289)', 'LMNA (4000)', 'LMNB1 (4001)', 'LMNB2 (84823)', 'LMNTD1 (160492)', 'LMNTD2 (256329)', 'LMO1 (4004)', 'LMO2 (4005)', 'LMO3 (55885)', 'LMO4 (8543)', 'LMO7 (4008)', 'LMOD1 (25802)', 'LMOD2 (442721)', 'LMOD3 (56203)', 'LMTK2 (22853)', 'LMTK3 (114783)', 'LMX1A (4009)', 'LMX1B (4010)', 'LNP1 (348801)', 'LNPEP (4012)', 'LNPK (80856)', 'LNX1 (84708)', 'LNX2 (222484)', 'LONP1 (9361)', 'LONP2 (83752)', 'LONRF1 (91694)', 'LONRF2 (164832)', 'LONRF3 (79836)', 'LORICRIN (4014)', 'LOX (4015)', 'LOXHD1 (125336)', 'LOXL1 (4016)', 'LOXL2 (4017)', 'LOXL3 (84695)', 'LOXL4 (84171)', 'LPA (4018)', 'LPAR1 (1902)', 'LPAR3 (23566)', 'LPAR4 (2846)', 'LPAR5 (57121)', 'LPAR6 (10161)', 'LPCAT1 (79888)', 'LPCAT2 (54947)', 'LPCAT3 (10162)', 'LPCAT4 (254531)', 'LPGAT1 (9926)', 'LPIN1 (23175)', 'LPIN2 (9663)', 'LPIN3 (64900)', 'LPL (4023)', 'LPO (4025)', 'LPP (4026)', 'LPXN (9404)', 'LRAT (9227)', 'LRATD1 (151354)', 'LRATD2 (157638)', 'LRBA (987)', 'LRCH1 (23143)', 'LRCH2 (57631)', 'LRCH3 (84859)', 'LRCH4 (4034)', 'LRFN1 (57622)', 'LRFN2 (57497)', 'LRFN3 (79414)', 'LRFN4 (78999)', 'LRFN5 (145581)', 'LRG1 (116844)', 'LRGUK (136332)', 'LRIF1 (55791)', 'LRIG1 (26018)', 'LRIG2 (9860)', 'LRIG3 (121227)', 'LRIT1 (26103)', 'LRIT2 (340745)', 'LRIT3 (345193)', 'LRMDA (83938)', 'LRP1 (4035)', 'LRP10 (26020)', 'LRP11 (84918)', 'LRP12 (29967)', 'LRP1B (53353)', 'LRP2 (4036)', 'LRP2BP (55805)', 'LRP3 (4037)', 'LRP4 (4038)', 'LRP5 (4041)', 'LRP6 (4040)', 'LRP8 (7804)', 'LRPAP1 (4043)', 'LRPPRC (10128)', 'LRR1 (122769)', 'LRRC1 (55227)', 'LRRC10 (376132)', 'LRRC10B (390205)', 'LRRC14 (9684)', 'LRRC14B (389257)', 'LRRC15 (131578)', 'LRRC17 (10234)', 'LRRC18 (474354)', 'LRRC19 (64922)', 'LRRC2 (79442)', 'LRRC20 (55222)', 'LRRC23 (10233)', 'LRRC24 (441381)', 'LRRC25 (126364)', 'LRRC26 (389816)', 'LRRC27 (80313)', 'LRRC28 (123355)', 'LRRC3 (81543)', 'LRRC30 (339291)', 'LRRC31 (79782)', 'LRRC32 (2615)', 'LRRC34 (151827)', 'LRRC36 (55282)', 'LRRC37A2 (474170)', 'LRRC37A3 (374819)', 'LRRC37B (114659)', 'LRRC38 (126755)', 'LRRC39 (127495)', 'LRRC3B (116135)', 'LRRC3C (100505591)', 'LRRC4 (64101)', 'LRRC40 (55631)', 'LRRC41 (10489)', 'LRRC42 (115353)', 'LRRC43 (254050)', 'LRRC45 (201255)', 'LRRC46 (90506)', 'LRRC47 (57470)', 'LRRC49 (54839)', 'LRRC4B (94030)', 'LRRC4C (57689)', 'LRRC51 (120356739)', 'LRRC52 (440699)', 'LRRC55 (219527)', 'LRRC56 (115399)', 'LRRC57 (255252)', 'LRRC58 (116064)', 'LRRC59 (55379)', 'LRRC61 (65999)', 'LRRC63 (220416)', 'LRRC66 (339977)', 'LRRC7 (57554)', 'LRRC70 (100130733)', 'LRRC71 (149499)', 'LRRC72 (100506049)', 'LRRC73 (221424)', 'LRRC74A (145497)', 'LRRC75A (388341)', 'LRRC75B (388886)', 'LRRC8A (56262)', 'LRRC8B (23507)', 'LRRC8C (84230)', 'LRRC8D (55144)', 'LRRC8E (80131)', 'LRRCC1 (85444)', 'LRRD1 (401387)', 'LRRFIP1 (9208)', 'LRRFIP2 (9209)', 'LRRIQ1 (84125)', 'LRRIQ3 (127255)', 'LRRIQ4 (344657)', 'LRRK1 (79705)', 'LRRK2 (120892)', 'LRRN1 (57633)', 'LRRN2 (10446)', 'LRRN3 (54674)', 'LRRN4 (164312)', 'LRRN4CL (221091)', 'LRRTM1 (347730)', 'LRRTM2 (26045)', 'LRRTM3 (347731)', 'LRRTM4 (80059)', 'LRSAM1 (90678)', 'LRTM1 (57408)', 'LRTM2 (654429)', 'LRWD1 (222229)', 'LSAMP (4045)', 'LSG1 (55341)', 'LSM1 (27257)', 'LSM10 (84967)', 'LSM11 (134353)', 'LSM12 (124801)', 'LSM14A (26065)', 'LSM14B (149986)', 'LSM2 (57819)', 'LSM3 (27258)', 'LSM4 (25804)', 'LSM5 (23658)', 'LSM6 (11157)', 'LSM7 (51690)', 'LSM8 (51691)', 'LSMEM1 (286006)', 'LSMEM2 (132228)', 'LSP1 (4046)', 'LSR (51599)', 'LSS (4047)', 'LST1 (7940)', 'LTA (4049)', 'LTA4H (4048)', 'LTB (4050)', 'LTB4R (1241)', 'LTB4R2 (56413)', 'LTBP1 (4052)', 'LTBP2 (4053)', 'LTBP3 (4054)', 'LTBR (4055)', 'LTC4S (4056)', 'LTF (4057)', 'LTK (4058)', 'LTN1 (26046)', 'LTO1 (220064)', 'LTV1 (84946)', 'LUC7L (55692)', 'LUC7L2 (51631)', 'LUC7L3 (51747)', 'LUM (4060)', 'LURAP1 (541468)', 'LURAP1L (286343)', 'LUZP1 (7798)', 'LUZP2 (338645)', 'LUZP4 (51213)', 'LVRN (206338)', 'LXN (56925)', 'LY6D (8581)', 'LY6E (4061)', 'LY6G5B (58496)', 'LY6G5C (80741)', 'LY6G6C (80740)', 'LY6H (4062)', 'LY6K (54742)', 'LY75 (4065)', 'LY86 (9450)', 'LY9 (4063)', 'LY96 (23643)', 'LYAR (55646)', 'LYG1 (129530)', 'LYG2 (254773)', 'LYL1 (4066)', 'LYN (4067)', 'LYNX1 (66004)', 'LYPD1 (116372)', 'LYPD2 (137797)', 'LYPD3 (27076)', 'LYPD4 (147719)', 'LYPD5 (284348)', 'LYPD6 (130574)', 'LYPD6B (130576)', 'LYPLA1 (10434)', 'LYPLA2 (11313)', 'LYPLAL1 (127018)', 'LYRM1 (57149)', 'LYRM2 (57226)', 'LYRM4 (57128)', 'LYRM7 (90624)', 'LYRM9 (201229)', 'LYSET (26175)', 'LYSMD1 (388695)', 'LYSMD2 (256586)', 'LYSMD3 (116068)', 'LYSMD4 (145748)', 'LYST (1130)', 'LYVE1 (10894)', 'LYZ (4069)', 'LYZL4 (131375)', 'LYZL6 (57151)', 'LZIC (84328)', 'LZTFL1 (54585)', 'LZTR1 (8216)', 'LZTS1 (11178)', 'LZTS2 (84445)', 'LZTS3 (9762)', 'M1AP (130951)', 'M6PR (4074)', 'MAB21L1 (4081)', 'MAB21L2 (10586)', 'MAB21L3 (126868)', 'MAB21L4 (79919)', 'MACC1 (346389)', 'MACF1 (23499)', 'MACIR (90355)', 'MACO1 (55219)', 'MACROD1 (28992)', 'MACROD2 (140733)', 'MACROH2A1 (9555)', 'MACROH2A2 (55506)', 'MAD1L1 (8379)', 'MAD2L1 (4085)', 'MAD2L1BP (9587)', 'MAD2L2 (10459)', 'MADCAM1 (8174)', 'MADD (8567)', 'MAEA (10296)', 'MAEL (84944)', 'MAF (4094)', 'MAF1 (84232)', 'MAFA (389692)', 'MAFB (9935)', 'MAFF (23764)', 'MAFG (4097)', 'MAFK (7975)', 'MAG (4099)', 'MAGEA1 (4100)', 'MAGEA10 (4109)', 'MAGEA11 (4110)', 'MAGEA12 (4111)', 'MAGEA4 (4103)', 'MAGEA6 (4105)', 'MAGEA8 (4107)', 'MAGEB1 (4112)', 'MAGEB10 (139422)', 'MAGEB16 (139604)', 'MAGEB17 (645864)', 'MAGEB18 (286514)', 'MAGEB2 (4113)', 'MAGEB3 (4114)', 'MAGEB4 (4115)', 'MAGEB5 (347541)', 'MAGEB6 (158809)', 'MAGEC1 (9947)', 'MAGEC2 (51438)', 'MAGEC3 (139081)', 'MAGED1 (9500)', 'MAGED2 (10916)', 'MAGEE1 (57692)', 'MAGEE2 (139599)', 'MAGEF1 (64110)', 'MAGEH1 (28986)', 'MAGI1 (9223)', 'MAGI2 (9863)', 'MAGI3 (260425)', 'MAGIX (79917)', 'MAGOH (4116)', 'MAGOHB (55110)', 'MAGT1 (84061)', 'MAIP1 (79568)', 'MAJIN (283129)', 'MAK (4117)', 'MAK16 (84549)', 'MAL (4118)', 'MALL (7851)', 'MALSU1 (115416)', 'MALT1 (10892)', 'MAMDC2 (256691)', 'MAMDC4 (158056)', 'MAML1 (9794)', 'MAML2 (84441)', 'MAML3 (55534)', 'MAMLD1 (10046)', 'MAMSTR (284358)', 'MAN1A1 (4121)', 'MAN1A2 (10905)', 'MAN1B1 (11253)', 'MAN1C1 (57134)', 'MAN2A1 (4124)', 'MAN2A2 (4122)', 'MAN2B1 (4125)', 'MAN2B2 (23324)', 'MAN2C1 (4123)', 'MANBA (4126)', 'MANBAL (63905)', 'MANEA (79694)', 'MANEAL (149175)', 'MANF (7873)', 'MANSC1 (54682)', 'MANSC4 (100287284)', 'MAOA (4128)', 'MAOB (4129)', 'MAP10 (54627)', 'MAP1A (4130)', 'MAP1B (4131)', 'MAP1LC3A (84557)', 'MAP1LC3B (81631)', 'MAP1LC3B2 (643246)', 'MAP1LC3C (440738)', 'MAP1S (55201)', 'MAP2 (4133)', 'MAP2K1 (5604)', 'MAP2K2 (5605)', 'MAP2K3 (5606)', 'MAP2K4 (6416)', 'MAP2K5 (5607)', 'MAP2K6 (5608)', 'MAP2K7 (5609)', 'MAP3K1 (4214)', 'MAP3K10 (4294)', 'MAP3K11 (4296)', 'MAP3K12 (7786)', 'MAP3K13 (9175)', 'MAP3K19 (80122)', 'MAP3K2 (10746)', 'MAP3K20 (51776)', 'MAP3K21 (84451)', 'MAP3K3 (4215)', 'MAP3K4 (4216)', 'MAP3K5 (4217)', 'MAP3K6 (9064)', 'MAP3K7 (6885)', 'MAP3K7CL (56911)', 'MAP3K8 (1326)', 'MAP3K9 (4293)', 'MAP4 (4134)', 'MAP4K1 (11184)', 'MAP4K2 (5871)', 'MAP4K3 (8491)', 'MAP4K4 (9448)', 'MAP6 (4135)', 'MAP6D1 (79929)', 'MAP7 (9053)', 'MAP7D1 (55700)', 'MAP7D2 (256714)', 'MAP7D3 (79649)', 'MAP9 (79884)', 'MAPK1 (5594)', 'MAPK10 (5602)', 'MAPK11 (5600)', 'MAPK12 (6300)', 'MAPK13 (5603)', 'MAPK14 (1432)', 'MAPK15 (225689)', 'MAPK1IP1L (93487)', 'MAPK3 (5595)', 'MAPK4 (5596)', 'MAPK6 (5597)', 'MAPK7 (5598)', 'MAPK8 (5599)', 'MAPK8IP1 (9479)', 'MAPK8IP3 (23162)', 'MAPK9 (5601)', 'MAPKAP1 (79109)', 'MAPKAPK2 (9261)', 'MAPKAPK3 (7867)', 'MAPKAPK5 (8550)', 'MAPKBP1 (23005)', 'MAPRE1 (22919)', 'MAPRE2 (10982)', 'MAPRE3 (22924)', 'MAPT (4137)', 'MARCHF1 (55016)', 'MARCHF10 (162333)', 'MARCHF11 (441061)', 'MARCHF2 (51257)', 'MARCHF3 (115123)', 'MARCHF4 (57574)', 'MARCHF5 (54708)', 'MARCHF6 (10299)', 'MARCHF7 (64844)', 'MARCHF8 (220972)', 'MARCHF9 (92979)', 'MARCKS (4082)', 'MARCKSL1 (65108)', 'MARCO (8685)', 'MARF1 (9665)', 'MARK1 (4139)', 'MARK2 (2011)', 'MARK3 (4140)', 'MARK4 (57787)', 'MARS1 (4141)', 'MARS2 (92935)', 'MARVELD2 (153562)', 'MARVELD3 (91862)', 'MAS1 (4142)', 'MAS1L (116511)', 'MASP1 (5648)', 'MASP2 (10747)', 'MAST1 (22983)', 'MAST2 (23139)', 'MAST3 (23031)', 'MAST4 (375449)', 'MASTL (84930)', 'MAT1A (4143)', 'MAT2A (4144)', 'MAT2B (27430)', 'MATCAP1 (653319)', 'MATCAP2 (23366)', 'MATK (4145)', 'MATN1 (4146)', 'MATN2 (4147)', 'MATN3 (4148)', 'MATN4 (8785)', 'MATR3 (9782)', 'MAU2 (23383)', 'MAVS (57506)', 'MAX (4149)', 'MAZ (4150)', 'MB (4151)', 'MB21D2 (151963)', 'MBD1 (4152)', 'MBD2 (8932)', 'MBD3 (53615)', 'MBD3L1 (85509)', 'MBD4 (8930)', 'MBD5 (55777)', 'MBD6 (114785)', 'MBIP (51562)', 'MBL2 (4153)', 'MBLAC1 (255374)', 'MBLAC2 (153364)', 'MBNL1 (4154)', 'MBNL2 (10150)', 'MBNL3 (55796)', 'MBOAT1 (154141)', 'MBOAT2 (129642)', 'MBOAT4 (619373)', 'MBOAT7 (79143)', 'MBP (4155)', 'MBTD1 (54799)', 'MBTPS1 (8720)', 'MBTPS2 (51360)', 'MC1R (4157)', 'MC2R (4158)', 'MC3R (4159)', 'MC4R (4160)', 'MC5R (4161)', 'MCAM (4162)', 'MCAT (27349)', 'MCC (4163)', 'MCCC1 (56922)', 'MCCC2 (64087)', 'MCCD1 (401250)', 'MCEE (84693)', 'MCEMP1 (199675)', 'MCF2 (4168)', 'MCF2L (23263)', 'MCF2L2 (23101)', 'MCFD2 (90411)', 'MCHR1 (2847)', 'MCHR2 (84539)', 'MCIDAS (345643)', 'MCL1 (4170)', 'MCM10 (55388)', 'MCM2 (4171)', 'MCM3 (4172)', 'MCM3AP (8888)', 'MCM4 (4173)', 'MCM5 (4174)', 'MCM6 (4175)', 'MCM7 (4176)', 'MCM8 (84515)', 'MCM9 (254394)', 'MCMBP (79892)', 'MCMDC2 (157777)', 'MCOLN1 (57192)', 'MCOLN2 (255231)', 'MCOLN3 (55283)', 'MCPH1 (79648)', 'MCRIP1 (348262)', 'MCRIP2 (84331)', 'MCRS1 (10445)', 'MCTP1 (79772)', 'MCTP2 (55784)', 'MCTS1 (28985)', 'MCU (90550)', 'MCUB (55013)', 'MCUR1 (63933)', 'MDC1 (9656)', 'MDFI (4188)', 'MDFIC (29969)', 'MDGA1 (266727)', 'MDGA2 (161357)', 'MDH1 (4190)', 'MDH1B (130752)', 'MDH2 (4191)', 'MDK (4192)', 'MDM1 (56890)', 'MDM2 (4193)', 'MDM4 (4194)', 'MDN1 (23195)', 'MDP1 (145553)', 'ME1 (4199)', 'ME2 (4200)', 'ME3 (10873)', 'MEA1 (4201)', 'MEAF6 (64769)', 'MEAK7 (57707)', 'MECOM (2122)', 'MECP2 (4204)', 'MECR (51102)', 'MED1 (5469)', 'MED10 (84246)', 'MED11 (400569)', 'MED12 (9968)', 'MED12L (116931)', 'MED13 (9969)', 'MED13L (23389)', 'MED14 (9282)', 'MED15 (51586)', 'MED16 (10025)', 'MED17 (9440)', 'MED18 (54797)', 'MED19 (219541)', 'MED20 (9477)', 'MED21 (9412)', 'MED22 (6837)', 'MED23 (9439)', 'MED24 (9862)', 'MED25 (81857)', 'MED26 (9441)', 'MED27 (9442)', 'MED28 (80306)', 'MED29 (55588)', 'MED30 (90390)', 'MED31 (51003)', 'MED4 (29079)', 'MED6 (10001)', 'MED7 (9443)', 'MED8 (112950)', 'MED9 (55090)', 'MEDAG (84935)', 'MEF2A (4205)', 'MEF2B (100271849)', 'MEF2C (4208)', 'MEF2D (4209)', 'MEFV (4210)', 'MEGF10 (84466)', 'MEGF11 (84465)', 'MEGF6 (1953)', 'MEGF8 (1954)', 'MEGF9 (1955)', 'MEI1 (150365)', 'MEI4 (101928601)', 'MEIG1 (644890)', 'MEIOB (254528)', 'MEIOC (284071)', 'MEIS1 (4211)', 'MEIS2 (4212)', 'MEIS3 (56917)', 'MELK (9833)', 'MELTF (4241)', 'MEMO1 (51072)', 'MEN1 (4221)', 'MEOX1 (4222)', 'MEOX2 (4223)', 'MEP1A (4224)', 'MEP1B (4225)', 'MEPCE (56257)', 'MEPE (56955)', 'MERTK (10461)', 'MESD (23184)', 'MESP1 (55897)', 'MESP2 (145873)', 'MEST (4232)', 'MET (4233)', 'METAP1 (23173)', 'METAP1D (254042)', 'METAP2 (10988)', 'METRN (79006)', 'METRNL (284207)', 'METTL1 (4234)', 'METTL13 (51603)', 'METTL14 (57721)', 'METTL15 (196074)', 'METTL16 (79066)', 'METTL17 (64745)', 'METTL18 (92342)', 'METTL21A (151194)', 'METTL21C (196541)', 'METTL22 (79091)', 'METTL23 (124512)', 'METTL24 (728464)', 'METTL25 (84190)', 'METTL25B (51093)', 'METTL26 (84326)', 'METTL27 (155368)', 'METTL2A (339175)', 'METTL2B (55798)', 'METTL3 (56339)', 'METTL4 (64863)', 'METTL5 (29081)', 'METTL6 (131965)', 'METTL8 (79828)', 'METTL9 (51108)', 'MEX3A (92312)', 'MEX3B (84206)', 'MEX3C (51320)', 'MEX3D (399664)', 'MFAP1 (4236)', 'MFAP2 (4237)', 'MFAP3 (4238)', 'MFAP3L (9848)', 'MFAP4 (4239)', 'MFAP5 (8076)', 'MFF (56947)', 'MFGE8 (4240)', 'MFHAS1 (9258)', 'MFN1 (55669)', 'MFN2 (9927)', 'MFNG (4242)', 'MFRP (83552)', 'MFSD1 (64747)', 'MFSD10 (10227)', 'MFSD11 (79157)', 'MFSD12 (126321)', 'MFSD13A (79847)', 'MFSD14A (64645)', 'MFSD14B (84641)', 'MFSD2A (84879)', 'MFSD2B (388931)', 'MFSD3 (113655)', 'MFSD4A (148808)', 'MFSD4B (91749)', 'MFSD5 (84975)', 'MFSD6 (54842)', 'MFSD6L (162387)', 'MFSD8 (256471)', 'MFSD9 (84804)', 'MGA (23269)', 'MGAM (8972)', 'MGARP (84709)', 'MGAT1 (4245)', 'MGAT2 (4247)', 'MGAT3 (4248)', 'MGAT4A (11320)', 'MGAT4B (11282)', 'MGAT4C (25834)', 'MGAT4D (152586)', 'MGAT5 (4249)', 'MGAT5B (146664)', 'MGLL (11343)', 'MGME1 (92667)', 'MGMT (4255)', 'MGP (4256)', 'MGRN1 (23295)', 'MGST1 (4257)', 'MGST2 (4258)', 'MGST3 (4259)', 'MIA (8190)', 'MIA2 (4253)', 'MIA3 (375056)', 'MIB1 (57534)', 'MIB2 (142678)', 'MICA (100507436)', 'MICAL1 (64780)', 'MICAL2 (9645)', 'MICAL3 (57553)', 'MICALL1 (85377)', 'MICALL2 (79778)', 'MICB (4277)', 'MICOS10 (440574)', 'MICOS13 (125988)', 'MICU1 (10367)', 'MICU2 (221154)', 'MICU3 (286097)', 'MID1 (4281)', 'MID1IP1 (58526)', 'MID2 (11043)', 'MIDEAS (91748)', 'MIDN (90007)', 'MIEF1 (54471)', 'MIEF2 (125170)', 'MIEN1 (84299)', 'MIER1 (57708)', 'MIER2 (54531)', 'MIER3 (166968)', 'MIF (4282)', 'MIF4GD (57409)', 'MIGA1 (374986)', 'MIGA2 (84895)', 'MIIP (60672)', 'MINAR1 (23251)', 'MINAR2 (100127206)', 'MINDY1 (55793)', 'MINDY2 (54629)', 'MINDY3 (80013)', 'MINDY4 (84182)', 'MINK1 (50488)', 'MINPP1 (9562)', 'MIOS (54468)', 'MIOX (55586)', 'MIP (4284)', 'MIPEP (4285)', 'MIPOL1 (145282)', 'MIS12 (79003)', 'MIS18A (54069)', 'MIS18BP1 (55320)', 'MISP (126353)', 'MITD1 (129531)', 'MITF (4286)', 'MIX23 (131076)', 'MIXL1 (83881)', 'MKI67 (4288)', 'MKKS (8195)', 'MKLN1 (4289)', 'MKNK1 (8569)', 'MKNK2 (2872)', 'MKRN1 (23608)', 'MKRN2 (23609)', 'MKRN2OS (100129480)', 'MKRN3 (7681)', 'MKS1 (54903)', 'MKX (283078)', 'MLANA (2315)', 'MLC1 (23209)', 'MLEC (9761)', 'MLF1 (4291)', 'MLF2 (8079)', 'MLH1 (4292)', 'MLH3 (27030)', 'MLIP (90523)', 'MLKL (197259)', 'MLLT1 (4298)', 'MLLT10 (8028)', 'MLLT11 (10962)', 'MLLT3 (4300)', 'MLLT6 (4302)', 'MLN (4295)', 'MLNR (2862)', 'MLPH (79083)', 'MLST8 (64223)', 'MLX (6945)', 'MLXIPL (51085)', 'MLYCD (23417)', 'MMAA (166785)', 'MMAB (326625)', 'MMACHC (25974)', 'MMADHC (27249)', 'MMD (23531)', 'MMD2 (221938)', 'MME (4311)', 'MMEL1 (79258)', 'MMGT1 (93380)', 'MMP1 (4312)', 'MMP10 (4319)', 'MMP11 (4320)', 'MMP13 (4322)', 'MMP14 (4323)', 'MMP15 (4324)', 'MMP16 (4325)', 'MMP17 (4326)', 'MMP19 (4327)', 'MMP2 (4313)', 'MMP20 (9313)', 'MMP21 (118856)', 'MMP23B (8510)', 'MMP24 (10893)', 'MMP25 (64386)', 'MMP26 (56547)', 'MMP27 (64066)', 'MMP28 (79148)', 'MMP3 (4314)', 'MMP7 (4316)', 'MMP8 (4317)', 'MMP9 (4318)', 'MMRN1 (22915)', 'MMRN2 (79812)', 'MMS19 (64210)', 'MMS22L (253714)', 'MMUT (4594)', 'MN1 (4330)', 'MNAT1 (4331)', 'MND1 (84057)', 'MNDA (4332)', 'MNS1 (55329)', 'MNT (4335)', 'MNX1 (3110)', 'MOAP1 (64112)', 'MOB1A (55233)', 'MOB1B (92597)', 'MOB2 (81532)', 'MOB3A (126308)', 'MOB3B (79817)', 'MOB3C (148932)', 'MOB4 (25843)', 'MOBP (4336)', 'MOCOS (55034)', 'MOCS1 (4337)', 'MOCS2 (4338)', 'MOCS3 (27304)', 'MOG (4340)', 'MOGAT1 (116255)', 'MOGAT2 (80168)', 'MOGAT3 (346606)', 'MOGS (7841)', 'MOK (5891)', 'MON1A (84315)', 'MON1B (22879)', 'MON2 (23041)', 'MORC1 (27136)', 'MORC2 (22880)', 'MORC3 (23515)', 'MORC4 (79710)', 'MORF4L1 (10933)', 'MORF4L2 (9643)', 'MORN1 (79906)', 'MORN2 (729967)', 'MORN3 (283385)', 'MORN4 (118812)', 'MORN5 (254956)', 'MOS (4342)', 'MOSMO (730094)', 'MOSPD1 (56180)', 'MOSPD2 (158747)', 'MOSPD3 (64598)', 'MOV10 (4343)', 'MOV10L1 (54456)', 'MOXD1 (26002)', 'MPC1 (51660)', 'MPC2 (25874)', 'MPDU1 (9526)', 'MPDZ (8777)', 'MPEG1 (219972)', 'MPG (4350)', 'MPHOSPH10 (10199)', 'MPHOSPH6 (10200)', 'MPHOSPH8 (54737)', 'MPHOSPH9 (10198)', 'MPI (4351)', 'MPIG6B (80739)', 'MPL (4352)', 'MPLKIP (136647)', 'MPND (84954)', 'MPO (4353)', 'MPP1 (4354)', 'MPP2 (4355)', 'MPP3 (4356)', 'MPP4 (58538)', 'MPP7 (143098)', 'MPPE1 (65258)', 'MPPED1 (758)', 'MPPED2 (744)', 'MPRIP (23164)', 'MPST (4357)', 'MPV17 (4358)', 'MPV17L (255027)', 'MPV17L2 (84769)', 'MPZ (4359)', 'MPZL1 (9019)', 'MPZL2 (10205)', 'MPZL3 (196264)', 'MR1 (3140)', 'MRAP (56246)', 'MRAP2 (112609)', 'MRAS (22808)', 'MRC1 (4360)', 'MRC2 (9902)', 'MRE11 (4361)', 'MREG (55686)', 'MRFAP1 (93621)', 'MRFAP1L1 (114932)', 'MRGBP (55257)', 'MRGPRD (116512)', 'MRGPRE (116534)', 'MRGPRF (116535)', 'MRGPRG (386746)', 'MRGPRX1 (259249)', 'MRGPRX2 (117194)', 'MRGPRX3 (117195)', 'MRGPRX4 (117196)', 'MRI1 (84245)', 'MRM1 (79922)', 'MRM2 (29960)', 'MRM3 (55178)', 'MRNIP (51149)', 'MRO (83876)', 'MROH1 (727957)', 'MROH2B (133558)', 'MROH6 (642475)', 'MROH7 (374977)', 'MROH9 (80133)', 'MRPL1 (65008)', 'MRPL10 (124995)', 'MRPL11 (65003)', 'MRPL12 (6182)', 'MRPL13 (28998)', 'MRPL14 (64928)', 'MRPL15 (29088)', 'MRPL16 (54948)', 'MRPL17 (63875)', 'MRPL18 (29074)', 'MRPL19 (9801)', 'MRPL2 (51069)', 'MRPL20 (55052)', 'MRPL21 (219927)', 'MRPL22 (29093)', 'MRPL23 (6150)', 'MRPL24 (79590)', 'MRPL27 (51264)', 'MRPL28 (10573)', 'MRPL3 (11222)', 'MRPL30 (51263)', 'MRPL32 (64983)', 'MRPL33 (9553)', 'MRPL34 (64981)', 'MRPL35 (51318)', 'MRPL36 (64979)', 'MRPL37 (51253)', 'MRPL38 (64978)', 'MRPL39 (54148)', 'MRPL4 (51073)', 'MRPL40 (64976)', 'MRPL41 (64975)', 'MRPL42 (28977)', 'MRPL43 (84545)', 'MRPL44 (65080)', 'MRPL45 (84311)', 'MRPL46 (26589)', 'MRPL47 (57129)', 'MRPL48 (51642)', 'MRPL49 (740)', 'MRPL50 (54534)', 'MRPL51 (51258)', 'MRPL52 (122704)', 'MRPL53 (116540)', 'MRPL54 (116541)', 'MRPL55 (128308)', 'MRPL57 (78988)', 'MRPL58 (3396)', 'MRPL9 (65005)', 'MRPS10 (55173)', 'MRPS11 (64963)', 'MRPS12 (6183)', 'MRPS14 (63931)', 'MRPS15 (64960)', 'MRPS16 (51021)', 'MRPS17 (51373)', 'MRPS18A (55168)', 'MRPS18B (28973)', 'MRPS18C (51023)', 'MRPS2 (51116)', 'MRPS21 (54460)', 'MRPS22 (56945)', 'MRPS23 (51649)', 'MRPS24 (64951)', 'MRPS25 (64432)', 'MRPS26 (64949)', 'MRPS27 (23107)', 'MRPS28 (28957)', 'MRPS30 (10884)', 'MRPS31 (10240)', 'MRPS33 (51650)', 'MRPS34 (65993)', 'MRPS35 (60488)', 'MRPS5 (64969)', 'MRPS6 (64968)', 'MRPS7 (51081)', 'MRPS9 (64965)', 'MRRF (92399)', 'MRS2 (57380)', 'MRTFA (57591)', 'MRTFB (57496)', 'MRTO4 (51154)', 'MS4A1 (931)', 'MS4A10 (341116)', 'MS4A12 (54860)', 'MS4A13 (503497)', 'MS4A14 (84689)', 'MS4A15 (219995)', 'MS4A2 (2206)', 'MS4A3 (932)', 'MS4A4A (51338)', 'MS4A5 (64232)', 'MS4A6A (64231)', 'MS4A6E (245802)', 'MS4A7 (58475)', 'MS4A8 (83661)', 'MSANTD1 (345222)', 'MSANTD2 (79684)', 'MSANTD3 (91283)', 'MSANTD4 (84437)', 'MSC (9242)', 'MSGN1 (343930)', 'MSH2 (4436)', 'MSH3 (4437)', 'MSH4 (4438)', 'MSH5 (4439)', 'MSH6 (2956)', 'MSI1 (4440)', 'MSI2 (124540)', 'MSL1 (339287)', 'MSL2 (55167)', 'MSL3 (10943)', 'MSLN (10232)', 'MSMB (4477)', 'MSMO1 (6307)', 'MSMP (692094)', 'MSN (4478)', 'MSR1 (4481)', 'MSRA (4482)', 'MSRB1 (51734)', 'MSRB2 (22921)', 'MSRB3 (253827)', 'MSS51 (118490)', 'MST1 (4485)', 'MST1R (4486)', 'MSTN (2660)', 'MSTO1 (55154)', 'MSX1 (4487)', 'MSX2 (4488)', 'MT1A (4489)', 'MT1B (4490)', 'MT1E (4493)', 'MT1F (4494)', 'MT1G (4495)', 'MT1H (4496)', 'MT1HL1 (645745)', 'MT1M (4499)', 'MT1X (4501)', 'MT2A (4502)', 'MT3 (4504)', 'MT4 (84560)', 'MTA1 (9112)', 'MTA2 (9219)', 'MTA3 (57504)', 'MTAP (4507)', 'MTARC1 (64757)', 'MTARC2 (54996)', 'MTBP (27085)', 'MTCH1 (23787)', 'MTCH2 (23788)', 'MTCL1 (23255)', 'MTCL2 (140710)', 'MTCP1 (4515)', 'MTDH (92140)', 'MTERF1 (7978)', 'MTERF2 (80298)', 'MTERF3 (51001)', 'MTERF4 (130916)', 'MTF1 (4520)', 'MTF2 (22823)', 'MTFMT (123263)', 'MTFP1 (51537)', 'MTFR1 (9650)', 'MTFR1L (56181)', 'MTFR2 (113115)', 'MTG1 (92170)', 'MTG2 (26164)', 'MTHFD1 (4522)', 'MTHFD1L (25902)', 'MTHFD2 (10797)', 'MTHFD2L (441024)', 'MTHFR (4524)', 'MTHFS (10588)', 'MTHFSD (64779)', 'MTIF2 (4528)', 'MTIF3 (219402)', 'MTM1 (4534)', 'MTMR1 (8776)', 'MTMR10 (54893)', 'MTMR11 (10903)', 'MTMR12 (54545)', 'MTMR14 (64419)', 'MTMR2 (8898)', 'MTMR3 (8897)', 'MTMR4 (9110)', 'MTMR6 (9107)', 'MTMR7 (9108)', 'MTMR8 (55613)', 'MTMR9 (66036)', 'MTNAP1 (55028)', 'MTNR1A (4543)', 'MTNR1B (4544)', 'MTO1 (25821)', 'MTOR (2475)', 'MTPAP (55149)', 'MTPN (136319)', 'MTR (4548)', 'MTRES1 (51250)', 'MTREX (23517)', 'MTRF1 (9617)', 'MTRF1L (54516)', 'MTRFR (91574)', 'MTRR (4552)', 'MTSS1 (9788)', 'MTSS2 (92154)', 'MTTP (4547)', 'MTURN (222166)', 'MTUS1 (57509)', 'MTUS2 (23281)', 'MTX1 (4580)', 'MTX2 (10651)', 'MTX3 (345778)', 'MUC1 (4582)', 'MUC12 (10071)', 'MUC15 (143662)', 'MUC16 (94025)', 'MUC17 (140453)', 'MUC20 (200958)', 'MUC21 (394263)', 'MUC22 (100507679)', 'MUC4 (4585)', 'MUC5B (727897)', 'MUC6 (4588)', 'MUC7 (4589)', 'MUCL1 (118430)', 'MUCL3 (135656)', 'MUL1 (79594)', 'MUS81 (80198)', 'MUSK (4593)', 'MUSTN1 (389125)', 'MUTYH (4595)', 'MVB12A (93343)', 'MVB12B (89853)', 'MVD (4597)', 'MVK (4598)', 'MVP (9961)', 'MX1 (4599)', 'MX2 (4600)', 'MXD1 (4084)', 'MXD3 (83463)', 'MXD4 (10608)', 'MXI1 (4601)', 'MXRA5 (25878)', 'MXRA7 (439921)', 'MXRA8 (54587)', 'MYADM (91663)', 'MYADML2 (255275)', 'MYB (4602)', 'MYBBP1A (10514)', 'MYBL1 (4603)', 'MYBL2 (4605)', 'MYBPC1 (4604)', 'MYBPC2 (4606)', 'MYBPC3 (4607)', 'MYBPH (4608)', 'MYBPHL (343263)', 'MYC (4609)', 'MYCBP (26292)', 'MYCBPAP (84073)', 'MYCL (4610)', 'MYCN (4613)', 'MYCT1 (80177)', 'MYD88 (4615)', 'MYDGF (56005)', 'MYEF2 (50804)', 'MYEOV (26579)', 'MYF5 (4617)', 'MYF6 (4618)', 'MYG1 (60314)', 'MYH1 (4619)', 'MYH10 (4628)', 'MYH11 (4629)', 'MYH13 (8735)', 'MYH14 (79784)', 'MYH15 (22989)', 'MYH2 (4620)', 'MYH3 (4621)', 'MYH4 (4622)', 'MYH6 (4624)', 'MYH7 (4625)', 'MYH8 (4626)', 'MYH9 (4627)', 'MYL1 (4632)', 'MYL10 (93408)', 'MYL11 (29895)', 'MYL12A (10627)', 'MYL12B (103910)', 'MYL2 (4633)', 'MYL3 (4634)', 'MYL4 (4635)', 'MYL5 (4636)', 'MYL6 (4637)', 'MYL6B (140465)', 'MYL7 (58498)', 'MYL9 (10398)', 'MYLIP (29116)', 'MYLK (4638)', 'MYLK2 (85366)', 'MYLK3 (91807)', 'MYLK4 (340156)', 'MYMK (389827)', 'MYNN (55892)', 'MYO10 (4651)', 'MYO15A (51168)', 'MYO16 (23026)', 'MYO18A (399687)', 'MYO18B (84700)', 'MYO19 (80179)', 'MYO1A (4640)', 'MYO1B (4430)', 'MYO1C (4641)', 'MYO1D (4642)', 'MYO1E (4643)', 'MYO1F (4542)', 'MYO1G (64005)', 'MYO1H (283446)', 'MYO3A (53904)', 'MYO3B (140469)', 'MYO5A (4644)', 'MYO5B (4645)', 'MYO5C (55930)', 'MYO6 (4646)', 'MYO7A (4647)', 'MYO7B (4648)', 'MYO9A (4649)', 'MYO9B (4650)', 'MYOC (4653)', 'MYOCD (93649)', 'MYOD1 (4654)', 'MYOF (26509)', 'MYOG (4656)', 'MYOM1 (8736)', 'MYOM2 (9172)', 'MYOM3 (127294)', 'MYOT (9499)', 'MYOZ1 (58529)', 'MYOZ2 (51778)', 'MYOZ3 (91977)', 'MYPN (84665)', 'MYPOP (339344)', 'MYRF (745)', 'MYRIP (25924)', 'MYSM1 (114803)', 'MYT1 (4661)', 'MYT1L (23040)', 'MYZAP (100820829)', 'MZB1 (51237)', 'MZF1 (7593)', 'MZT1 (440145)', 'MZT2A (653784)', 'MZT2B (80097)', 'N4BP1 (9683)', 'N4BP2 (55728)', 'N4BP2L1 (90634)', 'N4BP2L2 (10443)', 'N4BP3 (23138)', 'N6AMT1 (29104)', 'NAA10 (8260)', 'NAA11 (84779)', 'NAA15 (80155)', 'NAA16 (79612)', 'NAA20 (51126)', 'NAA25 (80018)', 'NAA30 (122830)', 'NAA35 (60560)', 'NAA38 (84316)', 'NAA40 (79829)', 'NAA50 (80218)', 'NAA60 (79903)', 'NAA80 (24142)', 'NAAA (27163)', 'NAALAD2 (10003)', 'NAALADL1 (10004)', 'NAALADL2 (254827)', 'NAB1 (4664)', 'NAB2 (4665)', 'NABP1 (64859)', 'NABP2 (79035)', 'NACA (4666)', 'NACA2 (342538)', 'NACAD (23148)', 'NACC1 (112939)', 'NACC2 (138151)', 'NADK (65220)', 'NADK2 (133686)', 'NADSYN1 (55191)', 'NAE1 (8883)', 'NAF1 (92345)', 'NAGA (4668)', 'NAGK (55577)', 'NAGLU (4669)', 'NAGPA (51172)', 'NAGS (162417)', 'NAIF1 (203245)', 'NALCN (259232)', 'NALF1 (728215)', 'NALF2 (27112)', 'NAMPT (10135)', 'NANOG (79923)', 'NANOGNB (360030)', 'NANOS1 (340719)', 'NANOS2 (339345)', 'NANOS3 (342977)', 'NANP (140838)', 'NANS (54187)', 'NAP1L1 (4673)', 'NAP1L2 (4674)', 'NAP1L3 (4675)', 'NAP1L4 (4676)', 'NAP1L5 (266812)', 'NAPA (8775)', 'NAPB (63908)', 'NAPEPLD (222236)', 'NAPG (8774)', 'NAPRT (93100)', 'NAPSA (9476)', 'NARF (26502)', 'NARS1 (4677)', 'NARS2 (79731)', 'NASP (4678)', 'NAT1 (9)', 'NAT10 (55226)', 'NAT14 (57106)', 'NAT16 (375607)', 'NAT2 (10)', 'NAT8 (9027)', 'NAT8L (339983)', 'NAT9 (26151)', 'NAV1 (89796)', 'NAV2 (89797)', 'NAV3 (89795)', 'NAXD (55739)', 'NAXE (128240)', 'NBAS (51594)', 'NBEA (26960)', 'NBEAL1 (65065)', 'NBEAL2 (23218)', 'NBL1 (4681)', 'NBN (4683)', 'NBPF3 (84224)', 'NBR1 (4077)', 'NCALD (83988)', 'NCAM2 (4685)', 'NCAN (1463)', 'NCAPD2 (9918)', 'NCAPD3 (23310)', 'NCAPG (64151)', 'NCAPG2 (54892)', 'NCAPH (23397)', 'NCAPH2 (29781)', 'NCBP1 (4686)', 'NCBP2 (22916)', 'NCBP3 (55421)', 'NCCRP1 (342897)', 'NCDN (23154)', 'NCEH1 (57552)', 'NCF1 (653361)', 'NCF2 (4688)', 'NCF4 (4689)', 'NCK1 (4690)', 'NCK2 (8440)', 'NCKAP1 (10787)', 'NCKAP1L (3071)', 'NCKAP5 (344148)', 'NCKAP5L (57701)', 'NCKIPSD (51517)', 'NCL (4691)', 'NCLN (56926)', 'NCMAP (400746)', 'NCOA1 (8648)', 'NCOA2 (10499)', 'NCOA3 (8202)', 'NCOA4 (8031)', 'NCOA5 (57727)', 'NCOA6 (23054)', 'NCOA7 (135112)', 'NCOR1 (9611)', 'NCOR2 (9612)', 'NCR1 (9437)', 'NCR2 (9436)', 'NCR3 (259197)', 'NCR3LG1 (374383)', 'NCS1 (23413)', 'NCSTN (23385)', 'NDC1 (55706)', 'NDC80 (10403)', 'NDE1 (54820)', 'NDEL1 (81565)', 'NDFIP1 (80762)', 'NDFIP2 (54602)', 'NDN (4692)', 'NDNF (79625)', 'NDOR1 (27158)', 'NDP (4693)', 'NDRG1 (10397)', 'NDRG2 (57447)', 'NDRG3 (57446)', 'NDRG4 (65009)', 'NDST1 (3340)', 'NDST2 (8509)', 'NDST3 (9348)', 'NDST4 (64579)', 'NDUFA1 (4694)', 'NDUFA10 (4705)', 'NDUFA11 (126328)', 'NDUFA12 (55967)', 'NDUFA13 (51079)', 'NDUFA2 (4695)', 'NDUFA3 (4696)', 'NDUFA4 (4697)', 'NDUFA4L2 (56901)', 'NDUFA5 (4698)', 'NDUFA6 (4700)', 'NDUFA7 (4701)', 'NDUFA8 (4702)', 'NDUFA9 (4704)', 'NDUFAB1 (4706)', 'NDUFAF1 (51103)', 'NDUFAF2 (91942)', 'NDUFAF3 (25915)', 'NDUFAF4 (29078)', 'NDUFAF5 (79133)', 'NDUFAF6 (137682)', 'NDUFAF7 (55471)', 'NDUFAF8 (284184)', 'NDUFB1 (4707)', 'NDUFB10 (4716)', 'NDUFB11 (54539)', 'NDUFB2 (4708)', 'NDUFB3 (4709)', 'NDUFB4 (4710)', 'NDUFB5 (4711)', 'NDUFB6 (4712)', 'NDUFB7 (4713)', 'NDUFB8 (4714)', 'NDUFB9 (4715)', 'NDUFC1 (4717)', 'NDUFC2 (4718)', 'NDUFS1 (4719)', 'NDUFS2 (4720)', 'NDUFS3 (4722)', 'NDUFS4 (4724)', 'NDUFS5 (4725)', 'NDUFS6 (4726)', 'NDUFS7 (374291)', 'NDUFS8 (4728)', 'NDUFV1 (4723)', 'NDUFV2 (4729)', 'NDUFV3 (4731)', 'NEB (4703)', 'NEBL (10529)', 'NECAB1 (64168)', 'NECAB2 (54550)', 'NECAB3 (63941)', 'NECAP1 (25977)', 'NECAP2 (55707)', 'NECTIN1 (5818)', 'NECTIN2 (5819)', 'NECTIN3 (25945)', 'NECTIN4 (81607)', 'NEDD1 (121441)', 'NEDD4 (4734)', 'NEDD4L (23327)', 'NEDD8 (4738)', 'NEDD9 (4739)', 'NEFH (4744)', 'NEFM (4741)', 'NEGR1 (257194)', 'NEIL1 (79661)', 'NEIL2 (252969)', 'NEIL3 (55247)', 'NEK1 (4750)', 'NEK10 (152110)', 'NEK11 (79858)', 'NEK2 (4751)', 'NEK3 (4752)', 'NEK4 (6787)', 'NEK5 (341676)', 'NEK6 (10783)', 'NEK7 (140609)', 'NEK8 (284086)', 'NEK9 (91754)', 'NELFA (7469)', 'NELFB (25920)', 'NELFCD (51497)', 'NELFE (7936)', 'NELL1 (4745)', 'NELL2 (4753)', 'NEMF (9147)', 'NEMP1 (23306)', 'NEMP2 (100131211)', 'NENF (29937)', 'NEO1 (4756)', 'NEPRO (25871)', 'NES (10763)', 'NET1 (10276)', 'NETO1 (81832)', 'NETO2 (81831)', 'NEU1 (4758)', 'NEU2 (4759)', 'NEU3 (10825)', 'NEU4 (129807)', 'NEURL1 (9148)', 'NEURL1B (54492)', 'NEURL2 (140825)', 'NEURL4 (84461)', 'NEUROD1 (4760)', 'NEUROD2 (4761)', 'NEUROD4 (58158)', 'NEUROD6 (63974)', 'NEUROG1 (4762)', 'NEUROG2 (63973)', 'NEUROG3 (50674)', 'NEXMIF (340533)', 'NEXN (91624)', 'NF1 (4763)', 'NF2 (4771)', 'NFAM1 (150372)', 'NFASC (23114)', 'NFAT5 (10725)', 'NFATC1 (4772)', 'NFATC2 (4773)', 'NFATC2IP (84901)', 'NFATC3 (4775)', 'NFATC4 (4776)', 'NFE2 (4778)', 'NFE2L1 (4779)', 'NFE2L2 (4780)', 'NFE2L3 (9603)', 'NFIA (4774)', 'NFIB (4781)', 'NFIC (4782)', 'NFIL3 (4783)', 'NFIX (4784)', 'NFKB1 (4790)', 'NFKB2 (4791)', 'NFKBIA (4792)', 'NFKBIB (4793)', 'NFKBID (84807)', 'NFKBIE (4794)', 'NFKBIL1 (4795)', 'NFKBIZ (64332)', 'NFRKB (4798)', 'NFS1 (9054)', 'NFU1 (27247)', 'NFX1 (4799)', 'NFXL1 (152518)', 'NFYA (4800)', 'NFYB (4801)', 'NFYC (4802)', 'NGB (58157)', 'NGDN (25983)', 'NGEF (25791)', 'NGF (4803)', 'NGFR (4804)', 'NGLY1 (55768)', 'NGRN (51335)', 'NHEJ1 (79840)', 'NHERF1 (9368)', 'NHERF2 (9351)', 'NHERF4 (79849)', 'NHLH1 (4807)', 'NHLH2 (4808)', 'NHLRC1 (378884)', 'NHLRC2 (374354)', 'NHLRC3 (387921)', 'NHLRC4 (283948)', 'NHP2 (55651)', 'NHS (4810)', 'NHSL1 (57224)', 'NHSL3 (57648)', 'NIBAN1 (116496)', 'NIBAN2 (64855)', 'NIBAN3 (199786)', 'NICN1 (84276)', 'NICOL1 (401115)', 'NID1 (4811)', 'NID2 (22795)', 'NIF3L1 (60491)', 'NIFK (84365)', 'NIM1K (167359)', 'NIN (51199)', 'NINJ1 (4814)', 'NINJ2 (4815)', 'NINL (22981)', 'NIP7 (51388)', 'NIPA1 (123606)', 'NIPA2 (81614)', 'NIPAL1 (152519)', 'NIPAL2 (79815)', 'NIPAL3 (57185)', 'NIPAL4 (348938)', 'NIPBL (25836)', 'NIPSNAP1 (8508)', 'NIPSNAP2 (2631)', 'NIPSNAP3A (25934)', 'NIPSNAP3B (55335)', 'NISCH (11188)', 'NIT1 (4817)', 'NIT2 (56954)', 'NKAIN1 (79570)', 'NKAIN2 (154215)', 'NKAIN3 (286183)', 'NKAIN4 (128414)', 'NKAP (79576)', 'NKAPD1 (55216)', 'NKAPL (222698)', 'NKD1 (85407)', 'NKD2 (85409)', 'NKG7 (4818)', 'NKIRAS1 (28512)', 'NKIRAS2 (28511)', 'NKTR (4820)', 'NKX1-2 (390010)', 'NKX2-1 (7080)', 'NKX2-2 (4821)', 'NKX2-3 (159296)', 'NKX2-4 (644524)', 'NKX2-5 (1482)', 'NKX2-8 (26257)', 'NKX3-1 (4824)', 'NKX3-2 (579)', 'NKX6-1 (4825)', 'NKX6-2 (84504)', 'NKX6-3 (157848)', 'NLE1 (54475)', 'NLGN1 (22871)', 'NLGN2 (57555)', 'NLGN3 (54413)', 'NLGN4X (57502)', 'NLGN4Y (22829)', 'NLK (51701)', 'NLN (57486)', 'NLRC4 (58484)', 'NLRC5 (84166)', 'NLRP1 (22861)', 'NLRP10 (338322)', 'NLRP11 (204801)', 'NLRP12 (91662)', 'NLRP13 (126204)', 'NLRP14 (338323)', 'NLRP2 (55655)', 'NLRP3 (114548)', 'NLRP4 (147945)', 'NLRP5 (126206)', 'NLRP6 (171389)', 'NLRP7 (199713)', 'NLRP8 (126205)', 'NLRP9 (338321)', 'NLRX1 (79671)', 'NMB (4828)', 'NMBR (4829)', 'NMD3 (51068)', 'NME1 (4830)', 'NME2 (4831)', 'NME3 (4832)', 'NME4 (4833)', 'NME5 (8382)', 'NME6 (10201)', 'NME7 (29922)', 'NME8 (51314)', 'NME9 (347736)', 'NMI (9111)', 'NMNAT1 (64802)', 'NMNAT2 (23057)', 'NMNAT3 (349565)', 'NMRAL1 (57407)', 'NMRK1 (54981)', 'NMRK2 (27231)', 'NMS (129521)', 'NMT1 (4836)', 'NMT2 (9397)', 'NMU (10874)', 'NMUR1 (10316)', 'NMUR2 (56923)', 'NNAT (4826)', 'NNMT (4837)', 'NNT (23530)', 'NOA1 (84273)', 'NOB1 (28987)', 'NOC2L (26155)', 'NOC3L (64318)', 'NOC4L (79050)', 'NOCT (25819)', 'NOD1 (10392)', 'NOD2 (64127)', 'NODAL (4838)', 'NOG (9241)', 'NOL10 (79954)', 'NOL11 (25926)', 'NOL12 (79159)', 'NOL3 (8996)', 'NOL4 (8715)', 'NOL4L (140688)', 'NOL6 (65083)', 'NOL7 (51406)', 'NOL8 (55035)', 'NOL9 (79707)', 'NOLC1 (9221)', 'NOM1 (64434)', 'NOMO1 (23420)', 'NOMO2 (283820)', 'NONO (4841)', 'NOP10 (55505)', 'NOP14 (8602)', 'NOP16 (51491)', 'NOP2 (4839)', 'NOP53 (29997)', 'NOP56 (10528)', 'NOP58 (51602)', 'NOP9 (161424)', 'NOPCHAP1 (121053)', 'NOS1 (4842)', 'NOS1AP (9722)', 'NOS2 (4843)', 'NOS3 (4846)', 'NOSIP (51070)', 'NOSTRIN (115677)', 'NOTCH1 (4851)', 'NOTCH2 (4853)', 'NOTCH3 (4854)', 'NOTCH4 (4855)', 'NOTO (344022)', 'NOTUM (147111)', 'NOVA1 (4857)', 'NOVA2 (4858)', 'NOX1 (27035)', 'NOX3 (50508)', 'NOX4 (50507)', 'NOX5 (79400)', 'NOXA1 (10811)', 'NOXO1 (124056)', 'NOXRED1 (122945)', 'NPAP1 (23742)', 'NPAS1 (4861)', 'NPAS2 (4862)', 'NPAS3 (64067)', 'NPAS4 (266743)', 'NPAT (4863)', 'NPB (256933)', 'NPBWR1 (2831)', 'NPBWR2 (2832)', 'NPC1 (4864)', 'NPC1L1 (29881)', 'NPC2 (10577)', 'NPDC1 (56654)', 'NPEPL1 (79716)', 'NPEPPS (9520)', 'NPFF (8620)', 'NPFFR1 (64106)', 'NPFFR2 (10886)', 'NPHP1 (4867)', 'NPHP3 (27031)', 'NPHP4 (261734)', 'NPHS1 (4868)', 'NPHS2 (7827)', 'NPL (80896)', 'NPLOC4 (55666)', 'NPM1 (4869)', 'NPM2 (10361)', 'NPM3 (10360)', 'NPNT (255743)', 'NPPA (4878)', 'NPPB (4879)', 'NPPC (4880)', 'NPR1 (4881)', 'NPR2 (4882)', 'NPR3 (4883)', 'NPRL2 (10641)', 'NPS (594857)', 'NPSR1 (387129)', 'NPTN (27020)', 'NPTX1 (4884)', 'NPTX2 (4885)', 'NPTXR (23467)', 'NPVF (64111)', 'NPW (283869)', 'NPY (4852)', 'NPY1R (4886)', 'NPY2R (4887)', 'NPY5R (4889)', 'NQO1 (1728)', 'NQO2 (4835)', 'NR0B1 (190)', 'NR0B2 (8431)', 'NR1D1 (9572)', 'NR1D2 (9975)', 'NR1H2 (7376)', 'NR1H3 (10062)', 'NR1H4 (9971)', 'NR1I2 (8856)', 'NR1I3 (9970)', 'NR2C1 (7181)', 'NR2C2 (7182)', 'NR2C2AP (126382)', 'NR2E1 (7101)', 'NR2F1 (7025)', 'NR2F2 (7026)', 'NR2F6 (2063)', 'NR3C1 (2908)', 'NR3C2 (4306)', 'NR4A1 (3164)', 'NR4A2 (4929)', 'NR4A3 (8013)', 'NR5A1 (2516)', 'NR5A2 (2494)', 'NR6A1 (2649)', 'NRAP (4892)', 'NRARP (441478)', 'NRAS (4893)', 'NRBF2 (29982)', 'NRBP1 (29959)', 'NRBP2 (340371)', 'NRCAM (4897)', 'NRDC (4898)', 'NRDE2 (55051)', 'NREP (9315)', 'NRF1 (4899)', 'NRG1 (3084)', 'NRG2 (9542)', 'NRG3 (10718)', 'NRG4 (145957)', 'NRGN (4900)', 'NRIP1 (8204)', 'NRIP2 (83714)', 'NRIP3 (56675)', 'NRK (203447)', 'NRL (4901)', 'NRM (11270)', 'NRN1 (51299)', 'NRN1L (123904)', 'NRP1 (8829)', 'NRP2 (8828)', 'NRROS (375387)', 'NRSN1 (140767)', 'NRSN2 (80023)', 'NRTN (4902)', 'NRXN1 (9378)', 'NRXN2 (9379)', 'NRXN3 (9369)', 'NSA2 (10412)', 'NSD1 (64324)', 'NSD2 (7468)', 'NSD3 (54904)', 'NSDHL (50814)', 'NSF (4905)', 'NSFL1C (55968)', 'NSG1 (27065)', 'NSG2 (51617)', 'NSL1 (25936)', 'NSMAF (8439)', 'NSMCE1 (197370)', 'NSMCE2 (286053)', 'NSMCE3 (56160)', 'NSMCE4A (54780)', 'NSMF (26012)', 'NSRP1 (84081)', 'NSUN2 (54888)', 'NSUN3 (63899)', 'NSUN4 (387338)', 'NSUN5 (55695)', 'NSUN6 (221078)', 'NSUN7 (79730)', 'NT5C (30833)', 'NT5C1A (84618)', 'NT5C1B (93034)', 'NT5C2 (22978)', 'NT5C3A (51251)', 'NT5C3B (115024)', 'NT5DC1 (221294)', 'NT5DC2 (64943)', 'NT5DC3 (51559)', 'NT5E (4907)', 'NT5M (56953)', 'NTAN1 (123803)', 'NTAQ1 (55093)', 'NTF3 (4908)', 'NTF4 (4909)', 'NTHL1 (4913)', 'NTM (50863)', 'NTMT1 (28989)', 'NTMT2 (149281)', 'NTN1 (9423)', 'NTN3 (4917)', 'NTN4 (59277)', 'NTN5 (126147)', 'NTNG1 (22854)', 'NTNG2 (84628)', 'NTPCR (84284)', 'NTRK1 (4914)', 'NTRK2 (4915)', 'NTRK3 (4916)', 'NTS (4922)', 'NTSR1 (4923)', 'NTSR2 (23620)', 'NUAK1 (9891)', 'NUAK2 (81788)', 'NUB1 (51667)', 'NUBP1 (4682)', 'NUBP2 (10101)', 'NUBPL (80224)', 'NUCB1 (4924)', 'NUCB2 (4925)', 'NUCKS1 (64710)', 'NUDC (10726)', 'NUDCD1 (84955)', 'NUDCD2 (134492)', 'NUDCD3 (23386)', 'NUDT1 (4521)', 'NUDT10 (170685)', 'NUDT11 (55190)', 'NUDT12 (83594)', 'NUDT13 (25961)', 'NUDT14 (256281)', 'NUDT15 (55270)', 'NUDT16 (131870)', 'NUDT16L1 (84309)', 'NUDT17 (200035)', 'NUDT19 (390916)', 'NUDT2 (318)', 'NUDT21 (11051)', 'NUDT22 (84304)', 'NUDT3 (11165)', 'NUDT5 (11164)', 'NUDT6 (11162)', 'NUDT7 (283927)', 'NUDT8 (254552)', 'NUDT9 (53343)', 'NUF2 (83540)', 'NUFIP1 (26747)', 'NUFIP2 (57532)', 'NUGGC (389643)', 'NUMA1 (4926)', 'NUMB (8650)', 'NUMBL (9253)', 'NUP107 (57122)', 'NUP133 (55746)', 'NUP153 (9972)', 'NUP155 (9631)', 'NUP160 (23279)', 'NUP188 (23511)', 'NUP205 (23165)', 'NUP210 (23225)', 'NUP210L (91181)', 'NUP214 (8021)', 'NUP35 (129401)', 'NUP37 (79023)', 'NUP42 (11097)', 'NUP43 (348995)', 'NUP50 (10762)', 'NUP54 (53371)', 'NUP58 (9818)', 'NUP62 (23636)', 'NUP62CL (54830)', 'NUP85 (79902)', 'NUP88 (4927)', 'NUP93 (9688)', 'NUP98 (4928)', 'NUPR1 (26471)', 'NUPR2 (389493)', 'NUS1 (116150)', 'NUSAP1 (51203)', 'NUTF2 (10204)', 'NUTM1 (256646)', 'NUTM2A (728118)', 'NUTM2F (54754)', 'NUTM2G (441457)', 'NVL (4931)', 'NWD1 (284434)', 'NWD2 (57495)', 'NXF1 (10482)', 'NXF3 (56000)', 'NXN (64359)', 'NXNL1 (115861)', 'NXNL2 (158046)', 'NXPE1 (120400)', 'NXPE2 (120406)', 'NXPE3 (91775)', 'NXPE4 (54827)', 'NXPH1 (30010)', 'NXPH2 (11249)', 'NXPH3 (11248)', 'NXPH4 (11247)', 'NXT1 (29107)', 'NXT2 (55916)', 'NYAP1 (222950)', 'NYAP2 (57624)', 'NYNRIN (57523)', 'NYX (60506)', 'OAF (220323)', 'OARD1 (221443)', 'OAS1 (4938)', 'OAS2 (4939)', 'OAS3 (4940)', 'OASL (8638)', 'OAT (4942)', 'OAZ1 (4946)', 'OAZ2 (4947)', 'OAZ3 (51686)', 'OBI1 (79596)', 'OBP2A (29991)', 'OBSCN (84033)', 'OBSL1 (23363)', 'OC90 (729330)', 'OCA2 (4948)', 'OCEL1 (79629)', 'OCIAD1 (54940)', 'OCIAD2 (132299)', 'OCLN (100506658)', 'OCM (654231)', 'OCM2 (4951)', 'OCRL (4952)', 'OCSTAMP (128506)', 'ODAD1 (93233)', 'ODAD2 (55130)', 'ODAD3 (115948)', 'ODAM (54959)', 'ODAPH (152816)', 'ODC1 (4953)', 'ODF1 (4956)', 'ODF2 (4957)', 'ODF2L (57489)', 'ODF4 (146852)', 'ODR4 (54953)', 'OFD1 (8481)', 'OGA (10724)', 'OGDH (4967)', 'OGDHL (55753)', 'OGFOD1 (55239)', 'OGFOD2 (79676)', 'OGFOD3 (79701)', 'OGFR (11054)', 'OGFRL1 (79627)', 'OGG1 (4968)', 'OGN (4969)', 'OGT (8473)', 'OIP5 (11339)', 'OIT3 (170392)', 'OLA1 (29789)', 'OLAH (55301)', 'OLFM1 (10439)', 'OLFM2 (93145)', 'OLFM3 (118427)', 'OLFM4 (10562)', 'OLFML1 (283298)', 'OLFML2A (169611)', 'OLFML2B (25903)', 'OLFML3 (56944)', 'OLIG1 (116448)', 'OLIG2 (10215)', 'OLIG3 (167826)', 'OLR1 (4973)', 'OMA1 (115209)', 'OMD (4958)', 'OMG (4974)', 'OMP (4975)', 'ONECUT1 (3175)', 'ONECUT2 (9480)', 'ONECUT3 (390874)', 'OOEP (441161)', 'OOSP2 (219990)', 'OPA1 (4976)', 'OPA3 (80207)', 'OPALIN (93377)', 'OPCML (4978)', 'OPHN1 (4983)', 'OPN1LW (5956)', 'OPN1SW (611)', 'OPN3 (23596)', 'OPN4 (94233)', 'OPN5 (221391)', 'OPRD1 (4985)', 'OPRK1 (4986)', 'OPRL1 (4987)', 'OPRM1 (4988)', 'OPRPN (58503)', 'OPTC (26254)', 'OPTN (10133)', 'OR10A2 (341276)', 'OR10A3 (26496)', 'OR10A4 (283297)', 'OR10A5 (144124)', 'OR10A7 (121364)', 'OR10AD1 (121275)', 'OR10AG1 (282770)', 'OR10C1 (442194)', 'OR10G2 (26534)', 'OR10G3 (26533)', 'OR10G4 (390264)', 'OR10G7 (390265)', 'OR10G8 (219869)', 'OR10G9 (219870)', 'OR10H1 (26539)', 'OR10H2 (26538)', 'OR10H3 (26532)', 'OR10H4 (126541)', 'OR10H5 (284433)', 'OR10J1 (26476)', 'OR10J5 (127385)', 'OR10K1 (391109)', 'OR10K2 (391107)', 'OR10P1 (121130)', 'OR10Q1 (219960)', 'OR10R2 (343406)', 'OR10T2 (128360)', 'OR10V1 (390201)', 'OR10W1 (81341)', 'OR10Z1 (128368)', 'OR11A1 (26531)', 'OR11H4 (390442)', 'OR11H6 (122748)', 'OR11L1 (391189)', 'OR12D3 (81797)', 'OR13A1 (79290)', 'OR13C2 (392376)', 'OR13C3 (138803)', 'OR13C4 (138804)', 'OR13C5 (138799)', 'OR13C8 (138802)', 'OR13C9 (286362)', 'OR13D1 (286365)', 'OR13F1 (138805)', 'OR13G1 (441933)', 'OR13H1 (347468)', 'OR13J1 (392309)', 'OR14A16 (284532)', 'OR14C36 (127066)', 'OR14I1 (401994)', 'OR14J1 (442191)', 'OR1A1 (8383)', 'OR1A2 (26189)', 'OR1C1 (26188)', 'OR1D2 (4991)', 'OR1D5 (8386)', 'OR1E1 (8387)', 'OR1E2 (8388)', 'OR1F1 (4992)', 'OR1G1 (8390)', 'OR1I1 (126370)', 'OR1J1 (347168)', 'OR1J2 (26740)', 'OR1J4 (26219)', 'OR1K1 (392392)', 'OR1L1 (26737)', 'OR1L3 (26735)', 'OR1L4 (254973)', 'OR1L6 (392390)', 'OR1L8 (138881)', 'OR1M1 (125963)', 'OR1N1 (138883)', 'OR1N2 (138882)', 'OR1Q1 (158131)', 'OR2A12 (346525)', 'OR2A14 (135941)', 'OR2A2 (442361)', 'OR2A25 (392138)', 'OR2A5 (393046)', 'OR2AE1 (81392)', 'OR2AG1 (144125)', 'OR2AG2 (338755)', 'OR2AP1 (121129)', 'OR2AT4 (341152)', 'OR2B11 (127623)', 'OR2B2 (81697)', 'OR2B3 (442184)', 'OR2B6 (26212)', 'OR2C1 (4993)', 'OR2C3 (81472)', 'OR2D2 (120776)', 'OR2D3 (120775)', 'OR2F2 (135948)', 'OR2G2 (81470)', 'OR2G3 (81469)', 'OR2G6 (391211)', 'OR2H1 (26716)', 'OR2H2 (7932)', 'OR2J2 (26707)', 'OR2J3 (442186)', 'OR2K2 (26248)', 'OR2L13 (284521)', 'OR2L2 (26246)', 'OR2L3 (391192)', 'OR2L5 (81466)', 'OR2M2 (391194)', 'OR2M3 (127062)', 'OR2M4 (26245)', 'OR2M5 (127059)', 'OR2M7 (391196)', 'OR2S2 (56656)', 'OR2T10 (127069)', 'OR2T12 (127064)', 'OR2T2 (401992)', 'OR2T27 (403239)', 'OR2T29 (343563)', 'OR2T35 (403244)', 'OR2T4 (127074)', 'OR2T5 (401993)', 'OR2T6 (254879)', 'OR2T8 (343172)', 'OR2V1 (26693)', 'OR2V2 (285659)', 'OR2W1 (26692)', 'OR2W3 (343171)', 'OR2Y1 (134083)', 'OR2Z1 (284383)', 'OR3A1 (4994)', 'OR4A16 (81327)', 'OR4A47 (403253)', 'OR4A5 (81318)', 'OR4B1 (119765)', 'OR4C11 (219429)', 'OR4C12 (283093)', 'OR4C13 (283092)', 'OR4C15 (81309)', 'OR4C3 (256144)', 'OR4C46 (119749)', 'OR4C6 (219432)', 'OR4D1 (26689)', 'OR4D10 (390197)', 'OR4D11 (219986)', 'OR4D2 (124538)', 'OR4D5 (219875)', 'OR4D6 (219983)', 'OR4D9 (390199)', 'OR4E2 (26686)', 'OR4F15 (390649)', 'OR4F6 (390648)', 'OR4K1 (79544)', 'OR4K13 (390433)', 'OR4K14 (122740)', 'OR4K15 (81127)', 'OR4K17 (390436)', 'OR4K2 (390431)', 'OR4K5 (79317)', 'OR4L1 (122742)', 'OR4M1 (441670)', 'OR4N2 (390429)', 'OR4N5 (390437)', 'OR4P4 (81300)', 'OR4S1 (256148)', 'OR4S2 (219431)', 'OR4X1 (390113)', 'OR51A2 (401667)', 'OR51A4 (401666)', 'OR51A7 (119687)', 'OR51B4 (79339)', 'OR51B5 (282763)', 'OR51B6 (390058)', 'OR51D1 (390038)', 'OR51E1 (143503)', 'OR51E2 (81285)', 'OR51F2 (119694)', 'OR51G2 (81282)', 'OR51I1 (390063)', 'OR51I2 (390064)', 'OR51L1 (119682)', 'OR51M1 (390059)', 'OR51Q1 (390061)', 'OR51S1 (119692)', 'OR51T1 (401665)', 'OR51V1 (283111)', 'OR52A1 (23538)', 'OR52A5 (390054)', 'OR52B2 (255725)', 'OR52B6 (340980)', 'OR52D1 (390066)', 'OR52E2 (119678)', 'OR52E4 (390081)', 'OR52E6 (390078)', 'OR52E8 (390079)', 'OR52H1 (390067)', 'OR52I1 (390037)', 'OR52J3 (119679)', 'OR52K1 (390036)', 'OR52K2 (119774)', 'OR52L1 (338751)', 'OR52M1 (119772)', 'OR52N1 (79473)', 'OR52N2 (390077)', 'OR52N4 (390072)', 'OR52N5 (390075)', 'OR52W1 (120787)', 'OR56A1 (120796)', 'OR56A3 (390083)', 'OR56B1 (387748)', 'OR56B4 (196335)', 'OR5A1 (219982)', 'OR5A2 (219981)', 'OR5AC2 (81050)', 'OR5AK2 (390181)', 'OR5AN1 (390195)', 'OR5AP2 (338675)', 'OR5AS1 (219447)', 'OR5AU1 (390445)', 'OR5B12 (390191)', 'OR5B17 (219965)', 'OR5B2 (390190)', 'OR5B21 (219968)', 'OR5B3 (441608)', 'OR5C1 (392391)', 'OR5D14 (219436)', 'OR5D16 (390144)', 'OR5D18 (219438)', 'OR5F1 (338674)', 'OR5H14 (403273)', 'OR5H15 (403274)', 'OR5H2 (79310)', 'OR5H6 (79295)', 'OR5I1 (10798)', 'OR5J2 (282775)', 'OR5K1 (26339)', 'OR5K3 (403277)', 'OR5K4 (403278)', 'OR5L2 (26338)', 'OR5M1 (390168)', 'OR5M10 (390167)', 'OR5M11 (219487)', 'OR5M3 (219482)', 'OR5M8 (219484)', 'OR5M9 (390162)', 'OR5P2 (120065)', 'OR5P3 (120066)', 'OR5T1 (390155)', 'OR5T2 (219464)', 'OR5T3 (390154)', 'OR5V1 (81696)', 'OR5W2 (390148)', 'OR6A2 (8590)', 'OR6B1 (135946)', 'OR6B2 (389090)', 'OR6B3 (150681)', 'OR6C1 (390321)', 'OR6C2 (341416)', 'OR6C3 (254786)', 'OR6C4 (341418)', 'OR6C6 (283365)', 'OR6C65 (403282)', 'OR6C68 (403284)', 'OR6C70 (390327)', 'OR6C74 (254783)', 'OR6C75 (390323)', 'OR6C76 (390326)', 'OR6F1 (343169)', 'OR6K2 (81448)', 'OR6K3 (391114)', 'OR6M1 (390261)', 'OR6N1 (128372)', 'OR6N2 (81442)', 'OR6P1 (128366)', 'OR6Q1 (219952)', 'OR6S1 (341799)', 'OR6T1 (219874)', 'OR6V1 (346517)', 'OR6X1 (390260)', 'OR6Y1 (391112)', 'OR7A10 (390892)', 'OR7A17 (26333)', 'OR7A5 (26659)', 'OR7C1 (26664)', 'OR7C2 (26658)', 'OR7D2 (162998)', 'OR7D4 (125958)', 'OR7E24 (26648)', 'OR7G1 (125962)', 'OR7G2 (390882)', 'OR7G3 (390883)', 'OR8A1 (390275)', 'OR8B12 (219858)', 'OR8B2 (26595)', 'OR8B3 (390271)', 'OR8B4 (283162)', 'OR8B8 (26493)', 'OR8D1 (283159)', 'OR8D4 (338662)', 'OR8G5 (219865)', 'OR8H1 (219469)', 'OR8H2 (390151)', 'OR8H3 (390152)', 'OR8I2 (120586)', 'OR8J1 (219477)', 'OR8J3 (81168)', 'OR8K1 (390157)', 'OR8K5 (219453)', 'OR8S1 (341568)', 'OR8U1 (219417)', 'OR9A2 (135924)', 'OR9A4 (130075)', 'OR9G1 (390174)', 'OR9G4 (283189)', 'OR9I1 (219954)', 'OR9K2 (441639)', 'OR9Q1 (219956)', 'OR9Q2 (219957)', 'ORAI2 (80228)', 'ORAI3 (93129)', 'ORC1 (4998)', 'ORC2 (4999)', 'ORC3 (23595)', 'ORC4 (5000)', 'ORC5 (5001)', 'ORC6 (23594)', 'ORM1 (5004)', 'ORM2 (5005)', 'ORMDL1 (94101)', 'ORMDL2 (29095)', 'ORMDL3 (94103)', 'OS9 (10956)', 'OSBP (5007)', 'OSBP2 (23762)', 'OSBPL10 (114884)', 'OSBPL11 (114885)', 'OSBPL1A (114876)', 'OSBPL2 (9885)', 'OSBPL3 (26031)', 'OSBPL5 (114879)', 'OSBPL6 (114880)', 'OSBPL7 (114881)', 'OSBPL8 (114882)', 'OSBPL9 (114883)', 'OSCAR (126014)', 'OSCP1 (127700)', 'OSER1 (51526)', 'OSGEP (55644)', 'OSGEPL1 (64172)', 'OSGIN1 (29948)', 'OSGIN2 (734)', 'OSM (5008)', 'OSMR (9180)', 'OSR1 (130497)', 'OSR2 (116039)', 'OST4 (100128731)', 'OSTC (58505)', 'OSTF1 (26578)', 'OSTM1 (28962)', 'OSTN (344901)', 'OTC (5009)', 'OTOA (146183)', 'OTOF (9381)', 'OTOG (340990)', 'OTOL1 (131149)', 'OTOP1 (133060)', 'OTOP2 (92736)', 'OTOP3 (347741)', 'OTOR (56914)', 'OTOS (150677)', 'OTP (23440)', 'OTUB1 (55611)', 'OTUB2 (78990)', 'OTUD1 (220213)', 'OTUD3 (23252)', 'OTUD4 (54726)', 'OTUD5 (55593)', 'OTUD6A (139562)', 'OTUD6B (51633)', 'OTUD7A (161725)', 'OTUD7B (56957)', 'OTULIN (90268)', 'OTULINL (54491)', 'OTX1 (5013)', 'OTX2 (5015)', 'OVCA2 (124641)', 'OVGP1 (5016)', 'OVOL1 (5017)', 'OVOL2 (58495)', 'OXA1L (5018)', 'OXCT1 (5019)', 'OXCT2 (64064)', 'OXER1 (165140)', 'OXGR1 (27199)', 'OXLD1 (339229)', 'OXNAD1 (92106)', 'OXR1 (55074)', 'OXSM (54995)', 'OXSR1 (9943)', 'OXT (5020)', 'OXTR (5021)', 'P2RX1 (5023)', 'P2RX2 (22953)', 'P2RX3 (5024)', 'P2RX4 (5025)', 'P2RX5 (5026)', 'P2RX6 (9127)', 'P2RX7 (5027)', 'P2RY1 (5028)', 'P2RY10 (27334)', 'P2RY11 (5032)', 'P2RY12 (64805)', 'P2RY13 (53829)', 'P2RY14 (9934)', 'P2RY2 (5029)', 'P2RY4 (5030)', 'P2RY6 (5031)', 'P3H1 (64175)', 'P3H2 (55214)', 'P3H3 (10536)', 'P3H4 (10609)', 'P4HA1 (5033)', 'P4HA2 (8974)', 'P4HA3 (283208)', 'P4HB (5034)', 'P4HTM (54681)', 'PA2G4 (5036)', 'PAAF1 (80227)', 'PABIR1 (116224)', 'PABIR2 (159090)', 'PABIR3 (159091)', 'PABPC1 (26986)', 'PABPC1L (80336)', 'PABPC3 (5042)', 'PABPC4 (8761)', 'PABPC5 (140886)', 'PABPN1 (8106)', 'PABPN1L (390748)', 'PACC1 (55248)', 'PACRG (135138)', 'PACRGL (133015)', 'PACS1 (55690)', 'PACS2 (23241)', 'PACSIN1 (29993)', 'PACSIN2 (11252)', 'PACSIN3 (29763)', 'PADI1 (29943)', 'PADI2 (11240)', 'PADI3 (51702)', 'PADI4 (23569)', 'PAEP (5047)', 'PAF1 (54623)', 'PAFAH1B1 (5048)', 'PAFAH1B2 (5049)', 'PAFAH1B3 (5050)', 'PAFAH2 (5051)', 'PAG1 (55824)', 'PAGE1 (8712)', 'PAGE2 (203569)', 'PAGE4 (9506)', 'PAGE5 (90737)', 'PAGR1 (79447)', 'PAH (5053)', 'PAICS (10606)', 'PAIP1 (10605)', 'PAIP2 (51247)', 'PAIP2B (400961)', 'PAK1 (5058)', 'PAK1IP1 (55003)', 'PAK2 (5062)', 'PAK3 (5063)', 'PAK4 (10298)', 'PAK5 (57144)', 'PAK6 (56924)', 'PALB2 (79728)', 'PALD1 (27143)', 'PALLD (23022)', 'PALM (5064)', 'PALM2AKAP2 (445815)', 'PALM3 (342979)', 'PALMD (54873)', 'PALS1 (64398)', 'PALS2 (51678)', 'PAM (5066)', 'PAM16 (51025)', 'PAMR1 (25891)', 'PAN2 (9924)', 'PAN3 (255967)', 'PANK1 (53354)', 'PANK2 (80025)', 'PANK3 (79646)', 'PANK4 (55229)', 'PANX1 (24145)', 'PANX2 (56666)', 'PANX3 (116337)', 'PAOX (196743)', 'PAPLN (89932)', 'PAPOLA (10914)', 'PAPOLG (64895)', 'PAPPA (5069)', 'PAPPA2 (60676)', 'PAPSS1 (9061)', 'PAPSS2 (9060)', 'PAQR3 (152559)', 'PAQR4 (124222)', 'PAQR5 (54852)', 'PAQR6 (79957)', 'PAQR7 (164091)', 'PAQR8 (85315)', 'PAQR9 (344838)', 'PARD3 (56288)', 'PARD3B (117583)', 'PARD6A (50855)', 'PARD6B (84612)', 'PARD6G (84552)', 'PARK7 (11315)', 'PARL (55486)', 'PARM1 (25849)', 'PARN (5073)', 'PARP1 (142)', 'PARP10 (84875)', 'PARP11 (57097)', 'PARP12 (64761)', 'PARP14 (54625)', 'PARP15 (165631)', 'PARP16 (54956)', 'PARP2 (10038)', 'PARP3 (10039)', 'PARP4 (143)', 'PARP6 (56965)', 'PARP8 (79668)', 'PARP9 (83666)', 'PARPBP (55010)', 'PARS2 (25973)', 'PARVA (55742)', 'PARVB (29780)', 'PARVG (64098)', 'PASD1 (139135)', 'PASK (23178)', 'PATE1 (160065)', 'PATE2 (399967)', 'PATE3 (100169851)', 'PATE4 (399968)', 'PATJ (10207)', 'PATL1 (219988)', 'PATL2 (197135)', 'PATZ1 (23598)', 'PAWR (5074)', 'PAX1 (5075)', 'PAX2 (5076)', 'PAX3 (5077)', 'PAX4 (5078)', 'PAX5 (5079)', 'PAX6 (5080)', 'PAX7 (5081)', 'PAX8 (7849)', 'PAX9 (5083)', 'PAXBP1 (94104)', 'PAXIP1 (22976)', 'PAXX (286257)', 'PBDC1 (51260)', 'PBK (55872)', 'PBLD (64081)', 'PBOV1 (59351)', 'PBRM1 (55193)', 'PBX1 (5087)', 'PBX2 (5089)', 'PBX3 (5090)', 'PBX4 (80714)', 'PBXIP1 (57326)', 'PC (5091)', 'PCARE (388939)', 'PCBD1 (5092)', 'PCBD2 (84105)', 'PCBP1 (5093)', 'PCBP2 (5094)', 'PCBP3 (54039)', 'PCBP4 (57060)', 'PCCA (5095)', 'PCCB (5096)', 'PCDH1 (5097)', 'PCDH10 (57575)', 'PCDH11X (27328)', 'PCDH12 (51294)', 'PCDH15 (65217)', 'PCDH17 (27253)', 'PCDH18 (54510)', 'PCDH19 (57526)', 'PCDH20 (64881)', 'PCDH7 (5099)', 'PCDH8 (5100)', 'PCDH9 (5101)', 'PCDHA1 (56147)', 'PCDHA10 (56139)', 'PCDHA11 (56138)', 'PCDHA12 (56137)', 'PCDHA13 (56136)', 'PCDHA2 (56146)', 'PCDHA3 (56145)', 'PCDHA4 (56144)', 'PCDHA5 (56143)', 'PCDHA6 (56142)', 'PCDHA7 (56141)', 'PCDHA8 (56140)', 'PCDHA9 (9752)', 'PCDHAC1 (56135)', 'PCDHAC2 (56134)', 'PCDHB1 (29930)', 'PCDHB10 (56126)', 'PCDHB11 (56125)', 'PCDHB12 (56124)', 'PCDHB13 (56123)', 'PCDHB14 (56122)', 'PCDHB15 (56121)', 'PCDHB16 (57717)', 'PCDHB2 (56133)', 'PCDHB3 (56132)', 'PCDHB4 (56131)', 'PCDHB5 (26167)', 'PCDHB6 (56130)', 'PCDHB7 (56129)', 'PCDHB8 (56128)', 'PCDHGA1 (56114)', 'PCDHGA10 (56106)', 'PCDHGA11 (56105)', 'PCDHGA12 (26025)', 'PCDHGA2 (56113)', 'PCDHGA3 (56112)', 'PCDHGA4 (56111)', 'PCDHGA5 (56110)', 'PCDHGA6 (56109)', 'PCDHGA7 (56108)', 'PCDHGA8 (9708)', 'PCDHGA9 (56107)', 'PCDHGB1 (56104)', 'PCDHGB2 (56103)', 'PCDHGB3 (56102)', 'PCDHGB4 (8641)', 'PCDHGB6 (56100)', 'PCDHGB7 (56099)', 'PCDHGC3 (5098)', 'PCDHGC4 (56098)', 'PCDHGC5 (56097)', 'PCED1A (64773)', 'PCED1B (91523)', 'PCF11 (51585)', 'PCGF1 (84759)', 'PCGF2 (7703)', 'PCGF3 (10336)', 'PCGF5 (84333)', 'PCGF6 (84108)', 'PCID2 (55795)', 'PCIF1 (63935)', 'PCK1 (5105)', 'PCK2 (5106)', 'PCLAF (9768)', 'PCLO (27445)', 'PCM1 (5108)', 'PCMT1 (5110)', 'PCMTD1 (115294)', 'PCMTD2 (55251)', 'PCNA (5111)', 'PCNP (57092)', 'PCNT (5116)', 'PCNX1 (22990)', 'PCNX2 (80003)', 'PCNX3 (399909)', 'PCOLCE (5118)', 'PCOLCE2 (26577)', 'PCP2 (126006)', 'PCP4 (5121)', 'PCP4L1 (654790)', 'PCSK1 (5122)', 'PCSK1N (27344)', 'PCSK2 (5126)', 'PCSK4 (54760)', 'PCSK5 (5125)', 'PCSK7 (9159)', 'PCSK9 (255738)', 'PCTP (58488)', 'PCYOX1 (51449)', 'PCYOX1L (78991)', 'PCYT1A (5130)', 'PCYT1B (9468)', 'PCYT2 (5833)', 'PDAP1 (11333)', 'PDC (5132)', 'PDCD1 (5133)', 'PDCD10 (11235)', 'PDCD11 (22984)', 'PDCD1LG2 (80380)', 'PDCD2 (5134)', 'PDCD2L (84306)', 'PDCD4 (27250)', 'PDCD5 (9141)', 'PDCD6 (10016)', 'PDCD6IP (10015)', 'PDCD7 (10081)', 'PDCL (5082)', 'PDCL2 (132954)', 'PDCL3 (79031)', 'PDE10A (10846)', 'PDE11A (50940)', 'PDE12 (201626)', 'PDE1A (5136)', 'PDE1B (5153)', 'PDE1C (5137)', 'PDE2A (5138)', 'PDE3A (5139)', 'PDE3B (5140)', 'PDE4A (5141)', 'PDE4B (5142)', 'PDE4C (5143)', 'PDE4D (5144)', 'PDE4DIP (9659)', 'PDE5A (8654)', 'PDE6A (5145)', 'PDE6B (5158)', 'PDE6C (5146)', 'PDE6D (5147)', 'PDE6G (5148)', 'PDE6H (5149)', 'PDE7A (5150)', 'PDE7B (27115)', 'PDE8A (5151)', 'PDE8B (8622)', 'PDE9A (5152)', 'PDF (64146)', 'PDGFA (5154)', 'PDGFB (5155)', 'PDGFC (56034)', 'PDGFD (80310)', 'PDGFRA (5156)', 'PDGFRB (5159)', 'PDGFRL (5157)', 'PDHA1 (5160)', 'PDHA2 (5161)', 'PDHB (5162)', 'PDHX (8050)', 'PDIA2 (64714)', 'PDIA3 (2923)', 'PDIA4 (9601)', 'PDIA5 (10954)', 'PDIA6 (10130)', 'PDIK1L (149420)', 'PDILT (204474)', 'PDK1 (5163)', 'PDK2 (5164)', 'PDK3 (5165)', 'PDK4 (5166)', 'PDLIM1 (9124)', 'PDLIM2 (64236)', 'PDLIM3 (27295)', 'PDLIM4 (8572)', 'PDLIM5 (10611)', 'PDLIM7 (9260)', 'PDP1 (54704)', 'PDP2 (57546)', 'PDPK1 (5170)', 'PDPN (10630)', 'PDPR (55066)', 'PDRG1 (81572)', 'PDS5A (23244)', 'PDS5B (23047)', 'PDSS1 (23590)', 'PDSS2 (57107)', 'PDX1 (3651)', 'PDXDC1 (23042)', 'PDXK (8566)', 'PDXP (57026)', 'PDYN (5173)', 'PDZD11 (51248)', 'PDZD2 (23037)', 'PDZD4 (57595)', 'PDZD7 (79955)', 'PDZD8 (118987)', 'PDZD9 (255762)', 'PDZK1 (5174)', 'PDZK1IP1 (10158)', 'PDZRN3 (23024)', 'PDZRN4 (29951)', 'PEA15 (8682)', 'PEAK1 (79834)', 'PEAK3 (374872)', 'PEAR1 (375033)', 'PEBP1 (5037)', 'PEBP4 (157310)', 'PECR (55825)', 'PEDS1 (387521)', 'PEF1 (553115)', 'PEG10 (23089)', 'PEG3 (5178)', 'PELI1 (57162)', 'PELI2 (57161)', 'PELI3 (246330)', 'PELO (53918)', 'PELP1 (27043)', 'PEMT (10400)', 'PENK (5179)', 'PEPD (5184)', 'PER1 (5187)', 'PER2 (8864)', 'PER3 (8863)', 'PERP (64065)', 'PES1 (23481)', 'PET100 (100131801)', 'PET117 (100303755)', 'PEX1 (5189)', 'PEX10 (5192)', 'PEX11A (8800)', 'PEX11B (8799)', 'PEX11G (92960)', 'PEX12 (5193)', 'PEX13 (5194)', 'PEX14 (5195)', 'PEX16 (9409)', 'PEX19 (5824)', 'PEX2 (5828)', 'PEX26 (55670)', 'PEX3 (8504)', 'PEX5 (5830)', 'PEX5L (51555)', 'PEX6 (5190)', 'PEX7 (5191)', 'PF4 (5196)', 'PF4V1 (5197)', 'PFAS (5198)', 'PFDN1 (5201)', 'PFDN2 (5202)', 'PFDN4 (5203)', 'PFDN5 (5204)', 'PFDN6 (10471)', 'PFKFB1 (5207)', 'PFKFB2 (5208)', 'PFKFB3 (5209)', 'PFKFB4 (5210)', 'PFKL (5211)', 'PFKM (5213)', 'PFKP (5214)', 'PFN1 (5216)', 'PFN2 (5217)', 'PFN3 (345456)', 'PFN4 (375189)', 'PGAM2 (5224)', 'PGAM4 (441531)', 'PGAM5 (192111)', 'PGAP1 (80055)', 'PGAP2 (27315)', 'PGAP3 (93210)', 'PGAP4 (84302)', 'PGAP6 (58986)', 'PGBD1 (84547)', 'PGBD2 (267002)', 'PGBD4 (161779)', 'PGC (5225)', 'PGD (5226)', 'PGF (5228)', 'PGGHG (80162)', 'PGGT1B (5229)', 'PGK1 (5230)', 'PGK2 (5232)', 'PGLS (25796)', 'PGLYRP1 (8993)', 'PGLYRP2 (114770)', 'PGLYRP3 (114771)', 'PGLYRP4 (57115)', 'PGM1 (5236)', 'PGM2 (55276)', 'PGM2L1 (283209)', 'PGM3 (5238)', 'PGM5 (5239)', 'PGP (283871)', 'PGPEP1 (54858)', 'PGPEP1L (145814)', 'PGR (5241)', 'PGRMC1 (10857)', 'PGRMC2 (10424)', 'PGS1 (9489)', 'PHACTR2 (9749)', 'PHACTR3 (116154)', 'PHACTR4 (65979)', 'PHAF1 (80262)', 'PHAX (51808)', 'PHB1 (5245)', 'PHB2 (11331)', 'PHC1 (1911)', 'PHC2 (1912)', 'PHC3 (80012)', 'PHETA1 (144717)', 'PHETA2 (150368)', 'PHEX (5251)', 'PHF1 (5252)', 'PHF10 (55274)', 'PHF11 (51131)', 'PHF12 (57649)', 'PHF13 (148479)', 'PHF14 (9678)', 'PHF19 (26147)', 'PHF2 (5253)', 'PHF20 (51230)', 'PHF20L1 (51105)', 'PHF21A (51317)', 'PHF21B (112885)', 'PHF23 (79142)', 'PHF24 (23349)', 'PHF3 (23469)', 'PHF5A (84844)', 'PHF6 (84295)', 'PHF7 (51533)', 'PHF8 (23133)', 'PHGDH (26227)', 'PHGR1 (644844)', 'PHIP (55023)', 'PHKA1 (5255)', 'PHKA2 (5256)', 'PHKB (5257)', 'PHKG1 (5260)', 'PHKG2 (5261)', 'PHLDA1 (22822)', 'PHLDA2 (7262)', 'PHLDA3 (23612)', 'PHLDB1 (23187)', 'PHLDB2 (90102)', 'PHLDB3 (653583)', 'PHLPP1 (23239)', 'PHLPP2 (23035)', 'PHOSPHO1 (162466)', 'PHOSPHO2 (493911)', 'PHOX2A (401)', 'PHOX2B (8929)', 'PHPT1 (29085)', 'PHRF1 (57661)', 'PHTF1 (10745)', 'PHTF2 (57157)', 'PHYH (5264)', 'PHYHD1 (254295)', 'PHYHIP (9796)', 'PHYHIPL (84457)', 'PHYKPL (85007)', 'PI15 (51050)', 'PI16 (221476)', 'PI3 (5266)', 'PI4K2A (55361)', 'PI4K2B (55300)', 'PI4KA (5297)', 'PI4KB (5298)', 'PIANP (196500)', 'PIAS1 (8554)', 'PIAS2 (9063)', 'PIAS3 (10401)', 'PIAS4 (51588)', 'PIBF1 (10464)', 'PICALM (8301)', 'PICK1 (9463)', 'PID1 (55022)', 'PIDD1 (55367)', 'PIERCE1 (138162)', 'PIERCE2 (145788)', 'PIEZO1 (9780)', 'PIF1 (80119)', 'PIGA (5277)', 'PIGB (9488)', 'PIGC (5279)', 'PIGF (5281)', 'PIGG (54872)', 'PIGH (5283)', 'PIGK (10026)', 'PIGL (9487)', 'PIGM (93183)', 'PIGN (23556)', 'PIGO (84720)', 'PIGP (51227)', 'PIGQ (9091)', 'PIGR (5284)', 'PIGS (94005)', 'PIGT (51604)', 'PIGU (128869)', 'PIGV (55650)', 'PIGW (284098)', 'PIGX (54965)', 'PIGY (84992)', 'PIGZ (80235)', 'PIH1D1 (55011)', 'PIH1D2 (120379)', 'PIK3AP1 (118788)', 'PIK3C2A (5286)', 'PIK3C2B (5287)', 'PIK3C2G (5288)', 'PIK3C3 (5289)', 'PIK3CA (5290)', 'PIK3CB (5291)', 'PIK3CD (5293)', 'PIK3CG (5294)', 'PIK3IP1 (113791)', 'PIK3R1 (5295)', 'PIK3R2 (5296)', 'PIK3R3 (8503)', 'PIK3R4 (30849)', 'PIK3R5 (23533)', 'PIKFYVE (200576)', 'PILRA (29992)', 'PILRB (29990)', 'PIM1 (5292)', 'PIM2 (11040)', 'PIM3 (415116)', 'PIMREG (54478)', 'PIN1 (5300)', 'PIN4 (5303)', 'PINK1 (65018)', 'PINLYP (390940)', 'PINX1 (54984)', 'PIP (5304)', 'PIP4K2A (5305)', 'PIP4K2B (8396)', 'PIP4K2C (79837)', 'PIP4P1 (90809)', 'PIP4P2 (55529)', 'PIP5K1A (8394)', 'PIP5K1B (8395)', 'PIP5K1C (23396)', 'PIP5KL1 (138429)', 'PIPOX (51268)', 'PIR (8544)', 'PIRT (644139)', 'PISD (23761)', 'PITHD1 (57095)', 'PITPNA (5306)', 'PITPNB (23760)', 'PITPNC1 (26207)', 'PITPNM1 (9600)', 'PITPNM2 (57605)', 'PITPNM3 (83394)', 'PITRM1 (10531)', 'PITX1 (5307)', 'PITX2 (5308)', 'PITX3 (5309)', 'PIWIL1 (9271)', 'PIWIL2 (55124)', 'PIWIL3 (440822)', 'PIWIL4 (143689)', 'PJA1 (64219)', 'PJA2 (9867)', 'PJVK (494513)', 'PKD1 (5310)', 'PKD1L1 (168507)', 'PKD2 (5311)', 'PKD2L1 (9033)', 'PKD2L2 (27039)', 'PKDCC (91461)', 'PKDREJ (10343)', 'PKHD1 (5314)', 'PKHD1L1 (93035)', 'PKIA (5569)', 'PKIB (5570)', 'PKIG (11142)', 'PKLR (5313)', 'PKM (5315)', 'PKMYT1 (9088)', 'PKN1 (5585)', 'PKN2 (5586)', 'PKN3 (29941)', 'PKNOX1 (5316)', 'PKNOX2 (63876)', 'PKP1 (5317)', 'PKP2 (5318)', 'PKP3 (11187)', 'PKP4 (8502)', 'PLA1A (51365)', 'PLA2G12A (81579)', 'PLA2G12B (84647)', 'PLA2G15 (23659)', 'PLA2G1B (5319)', 'PLA2G2A (5320)', 'PLA2G2D (26279)', 'PLA2G2E (30814)', 'PLA2G2F (64600)', 'PLA2G3 (50487)', 'PLA2G4A (5321)', 'PLA2G4B (100137049)', 'PLA2G4C (8605)', 'PLA2G4D (283748)', 'PLA2G4E (123745)', 'PLA2G4F (255189)', 'PLA2G5 (5322)', 'PLA2G6 (8398)', 'PLA2G7 (7941)', 'PLA2R1 (22925)', 'PLAA (9373)', 'PLAAT1 (57110)', 'PLAAT2 (54979)', 'PLAAT3 (11145)', 'PLAAT4 (5920)', 'PLAAT5 (117245)', 'PLAC1 (10761)', 'PLAC8 (51316)', 'PLAC8L1 (153770)', 'PLAC9 (219348)', 'PLAG1 (5324)', 'PLAGL1 (5325)', 'PLAGL2 (5326)', 'PLAT (5327)', 'PLAU (5328)', 'PLAUR (5329)', 'PLB1 (151056)', 'PLBD1 (79887)', 'PLBD2 (196463)', 'PLCB1 (23236)', 'PLCB2 (5330)', 'PLCB3 (5331)', 'PLCB4 (5332)', 'PLCD1 (5333)', 'PLCD4 (84812)', 'PLCE1 (51196)', 'PLCG1 (5335)', 'PLCG2 (5336)', 'PLCH1 (23007)', 'PLCH2 (9651)', 'PLCL1 (5334)', 'PLCL2 (23228)', 'PLCXD2 (257068)', 'PLCXD3 (345557)', 'PLCZ1 (89869)', 'PLD1 (5337)', 'PLD2 (5338)', 'PLD3 (23646)', 'PLD4 (122618)', 'PLD5 (200150)', 'PLD6 (201164)', 'PLEC (5339)', 'PLEK (5341)', 'PLEK2 (26499)', 'PLEKHA1 (59338)', 'PLEKHA3 (65977)', 'PLEKHA4 (57664)', 'PLEKHA5 (54477)', 'PLEKHA6 (22874)', 'PLEKHA7 (144100)', 'PLEKHA8 (84725)', 'PLEKHB1 (58473)', 'PLEKHB2 (55041)', 'PLEKHD1 (400224)', 'PLEKHF1 (79156)', 'PLEKHF2 (79666)', 'PLEKHG1 (57480)', 'PLEKHG2 (64857)', 'PLEKHG3 (26030)', 'PLEKHG4 (25894)', 'PLEKHG4B (153478)', 'PLEKHG5 (57449)', 'PLEKHG6 (55200)', 'PLEKHG7 (440107)', 'PLEKHH1 (57475)', 'PLEKHH2 (130271)', 'PLEKHH3 (79990)', 'PLEKHJ1 (55111)', 'PLEKHM1 (9842)', 'PLEKHM2 (23207)', 'PLEKHM3 (389072)', 'PLEKHN1 (84069)', 'PLEKHO1 (51177)', 'PLEKHO2 (80301)', 'PLEKHS1 (79949)', 'PLG (5340)', 'PLGRKT (55848)', 'PLIN1 (5346)', 'PLIN2 (123)', 'PLIN3 (10226)', 'PLIN4 (729359)', 'PLIN5 (440503)', 'PLK1 (5347)', 'PLK2 (10769)', 'PLK3 (1263)', 'PLK4 (10733)', 'PLK5 (126520)', 'PLLP (51090)', 'PLN (5350)', 'PLOD1 (5351)', 'PLOD2 (5352)', 'PLOD3 (8985)', 'PLP1 (5354)', 'PLP2 (5355)', 'PLPBP (11212)', 'PLPP1 (8611)', 'PLPP2 (8612)', 'PLPP3 (8613)', 'PLPP4 (196051)', 'PLPP5 (84513)', 'PLPP6 (403313)', 'PLPP7 (84814)', 'PLPPR1 (54886)', 'PLPPR2 (64748)', 'PLPPR3 (79948)', 'PLPPR4 (9890)', 'PLPPR5 (163404)', 'PLRG1 (5356)', 'PLS1 (5357)', 'PLS3 (5358)', 'PLSCR1 (5359)', 'PLSCR2 (57047)', 'PLSCR3 (57048)', 'PLSCR4 (57088)', 'PLSCR5 (389158)', 'PLTP (5360)', 'PLVAP (83483)', 'PLXDC1 (57125)', 'PLXDC2 (84898)', 'PLXNA1 (5361)', 'PLXNA2 (5362)', 'PLXNA3 (55558)', 'PLXNA4 (91584)', 'PLXNB1 (5364)', 'PLXNB2 (23654)', 'PLXNB3 (5365)', 'PLXNC1 (10154)', 'PLXND1 (23129)', 'PM20D1 (148811)', 'PM20D2 (135293)', 'PMAIP1 (5366)', 'PMCH (5367)', 'PMEL (6490)', 'PMEPA1 (56937)', 'PMF1 (11243)', 'PMFBP1 (83449)', 'PML (5371)', 'PMM1 (5372)', 'PMM2 (5373)', 'PMP2 (5375)', 'PMP22 (5376)', 'PMPCA (23203)', 'PMPCB (9512)', 'PMS1 (5378)', 'PMS2 (5395)', 'PMVK (10654)', 'PNCK (139728)', 'PNISR (25957)', 'PNKD (25953)', 'PNKP (11284)', 'PNLDC1 (154197)', 'PNLIP (5406)', 'PNLIPRP1 (5407)', 'PNLIPRP3 (119548)', 'PNMA1 (9240)', 'PNMA2 (10687)', 'PNMA3 (29944)', 'PNMA5 (114824)', 'PNMA6A (84968)', 'PNMA8A (55228)', 'PNMA8B (57469)', 'PNMT (5409)', 'PNN (5411)', 'PNO1 (56902)', 'PNOC (5368)', 'PNP (4860)', 'PNPLA1 (285848)', 'PNPLA2 (57104)', 'PNPLA3 (80339)', 'PNPLA4 (8228)', 'PNPLA5 (150379)', 'PNPLA6 (10908)', 'PNPLA7 (375775)', 'PNPLA8 (50640)', 'PNPO (55163)', 'PNPT1 (87178)', 'PNRC1 (10957)', 'PNRC2 (55629)', 'POC1A (25886)', 'POC1B (282809)', 'POC5 (134359)', 'PODN (127435)', 'PODNL1 (79883)', 'PODXL (5420)', 'PODXL2 (50512)', 'POF1B (79983)', 'POFUT1 (23509)', 'POFUT2 (23275)', 'POGK (57645)', 'POGLUT1 (56983)', 'POGLUT2 (79070)', 'POGLUT3 (143888)', 'POGZ (23126)', 'POLA1 (5422)', 'POLA2 (23649)', 'POLB (5423)', 'POLD1 (5424)', 'POLD2 (5425)', 'POLD3 (10714)', 'POLD4 (57804)', 'POLDIP3 (84271)', 'POLE (5426)', 'POLE2 (5427)', 'POLE3 (54107)', 'POLE4 (56655)', 'POLG (5428)', 'POLG2 (11232)', 'POLH (5429)', 'POLI (11201)', 'POLK (51426)', 'POLL (27343)', 'POLM (27434)', 'POLN (353497)', 'POLQ (10721)', 'POLR1A (25885)', 'POLR1B (84172)', 'POLR1C (9533)', 'POLR1D (51082)', 'POLR1E (64425)', 'POLR1F (221830)', 'POLR1G (10849)', 'POLR1H (30834)', 'POLR2B (5431)', 'POLR2C (5432)', 'POLR2D (5433)', 'POLR2E (5434)', 'POLR2F (5435)', 'POLR2G (5436)', 'POLR2H (5437)', 'POLR2I (5438)', 'POLR2K (5440)', 'POLR2L (5441)', 'POLR2M (81488)', 'POLR3A (11128)', 'POLR3B (55703)', 'POLR3C (10623)', 'POLR3D (661)', 'POLR3E (55718)', 'POLR3F (10621)', 'POLR3G (10622)', 'POLR3GL (84265)', 'POLR3H (171568)', 'POLR3K (51728)', 'POLRMT (5442)', 'POM121 (9883)', 'POM121C (100101267)', 'POM121L12 (285877)', 'POM121L2 (94026)', 'POMC (5443)', 'POMGNT1 (55624)', 'POMGNT2 (84892)', 'POMK (84197)', 'POMP (51371)', 'POMT1 (10585)', 'POMT2 (29954)', 'PON1 (5444)', 'PON2 (5445)', 'PON3 (5446)', 'POP1 (10940)', 'POP4 (10775)', 'POP5 (51367)', 'POP7 (10248)', 'POPDC2 (64091)', 'POPDC3 (64208)', 'POR (5447)', 'PORCN (64840)', 'POSTN (10631)', 'POT1 (25913)', 'POTEC (388468)', 'POTED (317754)', 'POTEE (445582)', 'POTEF (728378)', 'POTEG (404785)', 'POTEH (23784)', 'POTEI (653269)', 'POTEM (641455)', 'POU1F1 (5449)', 'POU2AF1 (5450)', 'POU2AF2 (341032)', 'POU2AF3 (120376)', 'POU2F1 (5451)', 'POU2F2 (5452)', 'POU2F3 (25833)', 'POU3F1 (5453)', 'POU3F2 (5454)', 'POU3F3 (5455)', 'POU3F4 (5456)', 'POU4F1 (5457)', 'POU4F2 (5458)', 'POU4F3 (5459)', 'POU5F1 (5460)', 'POU5F1B (5462)', 'POU5F2 (134187)', 'POU6F1 (5463)', 'POU6F2 (11281)', 'PP2D1 (151649)', 'PPA1 (5464)', 'PPA2 (27068)', 'PPAN (56342)', 'PPARA (5465)', 'PPARD (5467)', 'PPARG (5468)', 'PPARGC1A (10891)', 'PPARGC1B (133522)', 'PPAT (5471)', 'PPBP (5473)', 'PPCDC (60490)', 'PPCS (79717)', 'PPDPF (79144)', 'PPDPFL (492307)', 'PPEF1 (5475)', 'PPEF2 (5470)', 'PPFIA1 (8500)', 'PPFIA2 (8499)', 'PPFIA3 (8541)', 'PPFIA4 (8497)', 'PPFIBP1 (8496)', 'PPFIBP2 (8495)', 'PPHLN1 (51535)', 'PPIA (5478)', 'PPIAL4G (644591)', 'PPIB (5479)', 'PPIC (5480)', 'PPID (5481)', 'PPIE (10450)', 'PPIF (10105)', 'PPIG (9360)', 'PPIH (10465)', 'PPIL1 (51645)', 'PPIL2 (23759)', 'PPIL3 (53938)', 'PPIL4 (85313)', 'PPIL6 (285755)', 'PPIP5K1 (9677)', 'PPIP5K2 (23262)', 'PPL (5493)', 'PPM1A (5494)', 'PPM1B (5495)', 'PPM1D (8493)', 'PPM1E (22843)', 'PPM1F (9647)', 'PPM1G (5496)', 'PPM1H (57460)', 'PPM1J (333926)', 'PPM1K (152926)', 'PPM1L (151742)', 'PPM1M (132160)', 'PPM1N (147699)', 'PPME1 (51400)', 'PPOX (5498)', 'PPP1CA (5499)', 'PPP1CB (5500)', 'PPP1CC (5501)', 'PPP1R10 (5514)', 'PPP1R11 (6992)', 'PPP1R12A (4659)', 'PPP1R12B (4660)', 'PPP1R12C (54776)', 'PPP1R13B (23368)', 'PPP1R13L (10848)', 'PPP1R14A (94274)', 'PPP1R14B (26472)', 'PPP1R14C (81706)', 'PPP1R14D (54866)', 'PPP1R15A (23645)', 'PPP1R15B (84919)', 'PPP1R16A (84988)', 'PPP1R16B (26051)', 'PPP1R17 (10842)', 'PPP1R18 (170954)', 'PPP1R1A (5502)', 'PPP1R1B (84152)', 'PPP1R1C (151242)', 'PPP1R2 (5504)', 'PPP1R21 (129285)', 'PPP1R26 (9858)', 'PPP1R27 (116729)', 'PPP1R35 (221908)', 'PPP1R36 (145376)', 'PPP1R37 (284352)', 'PPP1R3A (5506)', 'PPP1R3B (79660)', 'PPP1R3C (5507)', 'PPP1R3D (5509)', 'PPP1R3E (90673)', 'PPP1R3F (89801)', 'PPP1R3G (648791)', 'PPP1R42 (286187)', 'PPP1R7 (5510)', 'PPP1R8 (5511)', 'PPP1R9A (55607)', 'PPP2CA (5515)', 'PPP2CB (5516)', 'PPP2R1A (5518)', 'PPP2R1B (5519)', 'PPP2R2A (5520)', 'PPP2R2B (5521)', 'PPP2R2C (5522)', 'PPP2R3A (5523)', 'PPP2R3C (55012)', 'PPP2R5A (5525)', 'PPP2R5B (5526)', 'PPP2R5C (5527)', 'PPP2R5D (5528)', 'PPP2R5E (5529)', 'PPP3CA (5530)', 'PPP3CB (5532)', 'PPP3CC (5533)', 'PPP3R1 (5534)', 'PPP3R2 (5535)', 'PPP4C (5531)', 'PPP4R1 (9989)', 'PPP4R2 (151987)', 'PPP4R3A (55671)', 'PPP4R3B (57223)', 'PPP4R4 (57718)', 'PPP5C (5536)', 'PPP6C (5537)', 'PPP6R1 (22870)', 'PPP6R2 (9701)', 'PPP6R3 (55291)', 'PPRC1 (23082)', 'PPT1 (5538)', 'PPT2 (9374)', 'PPTC7 (160760)', 'PPWD1 (23398)', 'PPY (5539)', 'PQBP1 (10084)', 'PRAC1 (84366)', 'PRADC1 (84279)', 'PRAF2 (11230)', 'PRAG1 (157285)', 'PRAM1 (84106)', 'PRAME (23532)', 'PRAMEF1 (65121)', 'PRAMEF11 (440560)', 'PRAMEF12 (390999)', 'PRAMEF14 (729528)', 'PRAMEF18 (391003)', 'PRAMEF19 (645414)', 'PRAMEF2 (65122)', 'PRAMEF20 (645425)', 'PRAMEF4 (400735)', 'PRAP1 (118471)', 'PRB2 (653247)', 'PRB4 (5545)', 'PRC1 (9055)', 'PRCC (5546)', 'PRCD (768206)', 'PRCP (5547)', 'PRDM1 (639)', 'PRDM10 (56980)', 'PRDM11 (56981)', 'PRDM12 (59335)', 'PRDM13 (59336)', 'PRDM14 (63978)', 'PRDM15 (63977)', 'PRDM16 (63976)', 'PRDM2 (7799)', 'PRDM4 (11108)', 'PRDM5 (11107)', 'PRDM6 (93166)', 'PRDM7 (11105)', 'PRDM8 (56978)', 'PRDM9 (56979)', 'PRDX1 (5052)', 'PRDX2 (7001)', 'PRDX3 (10935)', 'PRDX4 (10549)', 'PRDX5 (25824)', 'PRDX6 (9588)', 'PREB (10113)', 'PRELID1 (27166)', 'PRELID2 (153768)', 'PRELID3A (10650)', 'PRELID3B (51012)', 'PRELP (5549)', 'PREP (5550)', 'PREPL (9581)', 'PREX1 (57580)', 'PREX2 (80243)', 'PRF1 (5551)', 'PRG2 (5553)', 'PRG3 (10394)', 'PRG4 (10216)', 'PRH2 (5555)', 'PRICKLE1 (144165)', 'PRICKLE2 (166336)', 'PRICKLE3 (4007)', 'PRICKLE4 (29964)', 'PRIM1 (5557)', 'PRIMA1 (145270)', 'PRIMPOL (201973)', 'PRKAA1 (5562)', 'PRKAA2 (5563)', 'PRKAB1 (5564)', 'PRKAB2 (5565)', 'PRKACA (5566)', 'PRKACB (5567)', 'PRKACG (5568)', 'PRKAG1 (5571)', 'PRKAG2 (51422)', 'PRKAG3 (53632)', 'PRKAR1A (5573)', 'PRKAR1B (5575)', 'PRKAR2A (5576)', 'PRKAR2B (5577)', 'PRKCA (5578)', 'PRKCB (5579)', 'PRKCD (5580)', 'PRKCE (5581)', 'PRKCG (5582)', 'PRKCH (5583)', 'PRKCI (5584)', 'PRKCQ (5588)', 'PRKCSH (5589)', 'PRKCZ (5590)', 'PRKD1 (5587)', 'PRKD2 (25865)', 'PRKD3 (23683)', 'PRKG1 (5592)', 'PRKG2 (5593)', 'PRKN (5071)', 'PRKRA (8575)', 'PRKRIP1 (79706)', 'PRKX (5613)', 'PRL (5617)', 'PRLH (51052)', 'PRLHR (2834)', 'PRLR (5618)', 'PRM1 (5619)', 'PRM2 (5620)', 'PRMT1 (3276)', 'PRMT2 (3275)', 'PRMT3 (10196)', 'PRMT5 (10419)', 'PRMT6 (55170)', 'PRMT7 (54496)', 'PRMT8 (56341)', 'PRMT9 (90826)', 'PRND (23627)', 'PRNP (5621)', 'PROB1 (389333)', 'PROC (5624)', 'PROCA1 (147011)', 'PROCR (10544)', 'PRODH (5625)', 'PRODH2 (58510)', 'PROK1 (84432)', 'PROK2 (60675)', 'PROKR1 (10887)', 'PROKR2 (128674)', 'PROM1 (8842)', 'PROM2 (150696)', 'PROP1 (5626)', 'PRORP (9692)', 'PROS1 (5627)', 'PROSER1 (80209)', 'PROSER2 (254427)', 'PROX1 (5629)', 'PROX2 (283571)', 'PROZ (8858)', 'PRP4K (8899)', 'PRPF18 (8559)', 'PRPF19 (27339)', 'PRPF3 (9129)', 'PRPF31 (26121)', 'PRPF38A (84950)', 'PRPF38B (55119)', 'PRPF39 (55015)', 'PRPF4 (9128)', 'PRPF40A (55660)', 'PRPF40B (25766)', 'PRPF6 (24148)', 'PRPF8 (10594)', 'PRPH (5630)', 'PRPH2 (5961)', 'PRPS1 (5631)', 'PRPS1L1 (221823)', 'PRPS2 (5634)', 'PRPSAP1 (5635)', 'PRPSAP2 (5636)', 'PRR11 (55771)', 'PRR12 (57479)', 'PRR13 (54458)', 'PRR14 (78994)', 'PRR14L (253143)', 'PRR15 (222171)', 'PRR15L (79170)', 'PRR16 (51334)', 'PRR18 (285800)', 'PRR19 (284338)', 'PRR22 (163154)', 'PRR23A (729627)', 'PRR23B (389151)', 'PRR23C (389152)', 'PRR27 (401137)', 'PRR29 (92340)', 'PRR3 (80742)', 'PRR30 (339779)', 'PRR32 (100130613)', 'PRR35 (146325)', 'PRR4 (11272)', 'PRR5 (55615)', 'PRR5L (79899)', 'PRR7 (80758)', 'PRR9 (574414)', 'PRRC1 (133619)', 'PRRC2A (7916)', 'PRRC2B (84726)', 'PRRC2C (23215)', 'PRRG1 (5638)', 'PRRG2 (5639)', 'PRRG3 (79057)', 'PRRG4 (79056)', 'PRRT1 (80863)', 'PRRT2 (112476)', 'PRRT3 (285368)', 'PRRT4 (401399)', 'PRRX1 (5396)', 'PRRX2 (51450)', 'PRSS1 (5644)', 'PRSS12 (8492)', 'PRSS16 (10279)', 'PRSS21 (10942)', 'PRSS22 (64063)', 'PRSS23 (11098)', 'PRSS27 (83886)', 'PRSS3 (5646)', 'PRSS33 (260429)', 'PRSS35 (167681)', 'PRSS36 (146547)', 'PRSS37 (136242)', 'PRSS38 (339501)', 'PRSS48 (345062)', 'PRSS50 (29122)', 'PRSS53 (339105)', 'PRSS54 (221191)', 'PRSS55 (203074)', 'PRSS57 (400668)', 'PRSS58 (136541)', 'PRSS8 (5652)', 'PRTFDC1 (56952)', 'PRTG (283659)', 'PRTN3 (5657)', 'PRUNE1 (58497)', 'PRUNE2 (158471)', 'PRX (57716)', 'PRXL2A (84293)', 'PRXL2B (127281)', 'PRXL2C (195827)', 'PSAP (5660)', 'PSAPL1 (768239)', 'PSAT1 (29968)', 'PSCA (8000)', 'PSD (5662)', 'PSD2 (84249)', 'PSD3 (23362)', 'PSD4 (23550)', 'PSEN1 (5663)', 'PSEN2 (5664)', 'PSENEN (55851)', 'PSG1 (5669)', 'PSG11 (5680)', 'PSG2 (5670)', 'PSG3 (5671)', 'PSG4 (5672)', 'PSG5 (5673)', 'PSG6 (5675)', 'PSG8 (440533)', 'PSG9 (5678)', 'PSIP1 (11168)', 'PSKH1 (5681)', 'PSKH2 (85481)', 'PSMA1 (5682)', 'PSMA2 (5683)', 'PSMA3 (5684)', 'PSMA4 (5685)', 'PSMA5 (5686)', 'PSMA6 (5687)', 'PSMA7 (5688)', 'PSMA8 (143471)', 'PSMB1 (5689)', 'PSMB10 (5699)', 'PSMB11 (122706)', 'PSMB2 (5690)', 'PSMB3 (5691)', 'PSMB4 (5692)', 'PSMB5 (5693)', 'PSMB6 (5694)', 'PSMB7 (5695)', 'PSMB8 (5696)', 'PSMB9 (5698)', 'PSMC1 (5700)', 'PSMC2 (5701)', 'PSMC3 (5702)', 'PSMC3IP (29893)', 'PSMC4 (5704)', 'PSMC5 (5705)', 'PSMC6 (5706)', 'PSMD1 (5707)', 'PSMD10 (5716)', 'PSMD11 (5717)', 'PSMD12 (5718)', 'PSMD13 (5719)', 'PSMD14 (10213)', 'PSMD2 (5708)', 'PSMD3 (5709)', 'PSMD4 (5710)', 'PSMD5 (5711)', 'PSMD6 (9861)', 'PSMD7 (5713)', 'PSMD8 (5714)', 'PSMD9 (5715)', 'PSME1 (5720)', 'PSME2 (5721)', 'PSME3 (10197)', 'PSME3IP1 (80011)', 'PSME4 (23198)', 'PSMF1 (9491)', 'PSMG1 (8624)', 'PSMG2 (56984)', 'PSMG3 (84262)', 'PSMG4 (389362)', 'PSORS1C1 (170679)', 'PSORS1C2 (170680)', 'PSPC1 (55269)', 'PSPH (5723)', 'PSPN (5623)', 'PSRC1 (84722)', 'PSTK (118672)', 'PSTPIP1 (9051)', 'PSTPIP2 (9050)', 'PTAFR (5724)', 'PTAR1 (375743)', 'PTBP1 (5725)', 'PTBP2 (58155)', 'PTBP3 (9991)', 'PTCD1 (26024)', 'PTCD2 (79810)', 'PTCD3 (55037)', 'PTCH1 (5727)', 'PTCH2 (8643)', 'PTCHD1 (139411)', 'PTCHD4 (442213)', 'PTCRA (171558)', 'PTDSS1 (9791)', 'PTDSS2 (81490)', 'PTEN (5728)', 'PTER (9317)', 'PTF1A (256297)', 'PTGDR (5729)', 'PTGDR2 (11251)', 'PTGDS (5730)', 'PTGER1 (5731)', 'PTGER2 (5732)', 'PTGER3 (5733)', 'PTGER4 (5734)', 'PTGES (9536)', 'PTGES2 (80142)', 'PTGES3 (10728)', 'PTGES3L (100885848)', 'PTGFR (5737)', 'PTGFRN (5738)', 'PTGIR (5739)', 'PTGIS (5740)', 'PTGR1 (22949)', 'PTGR2 (145482)', 'PTGR3 (284273)', 'PTGS1 (5742)', 'PTGS2 (5743)', 'PTH (5741)', 'PTH1R (5745)', 'PTH2 (113091)', 'PTH2R (5746)', 'PTHLH (5744)', 'PTK2 (5747)', 'PTK2B (2185)', 'PTK6 (5753)', 'PTK7 (5754)', 'PTMA (5757)', 'PTMS (5763)', 'PTN (5764)', 'PTOV1 (53635)', 'PTP4A1 (7803)', 'PTP4A2 (8073)', 'PTP4A3 (11156)', 'PTPA (5524)', 'PTPDC1 (138639)', 'PTPMT1 (114971)', 'PTPN1 (5770)', 'PTPN11 (5781)', 'PTPN12 (5782)', 'PTPN13 (5783)', 'PTPN14 (5784)', 'PTPN18 (26469)', 'PTPN2 (5771)', 'PTPN20 (26095)', 'PTPN21 (11099)', 'PTPN22 (26191)', 'PTPN23 (25930)', 'PTPN3 (5774)', 'PTPN4 (5775)', 'PTPN5 (84867)', 'PTPN6 (5777)', 'PTPN7 (5778)', 'PTPN9 (5780)', 'PTPRA (5786)', 'PTPRB (5787)', 'PTPRC (5788)', 'PTPRCAP (5790)', 'PTPRD (5789)', 'PTPRE (5791)', 'PTPRF (5792)', 'PTPRG (5793)', 'PTPRH (5794)', 'PTPRJ (5795)', 'PTPRK (5796)', 'PTPRM (5797)', 'PTPRN (5798)', 'PTPRN2 (5799)', 'PTPRO (5800)', 'PTPRR (5801)', 'PTPRS (5802)', 'PTPRT (11122)', 'PTPRU (10076)', 'PTPRZ1 (5803)', 'PTRH1 (138428)', 'PTRH2 (51651)', 'PTRHD1 (391356)', 'PTS (5805)', 'PTTG1 (9232)', 'PTTG1IP (754)', 'PTTG2 (10744)', 'PTX3 (5806)', 'PTX4 (390667)', 'PUDP (8226)', 'PUF60 (22827)', 'PUM1 (9698)', 'PUM2 (23369)', 'PUM3 (9933)', 'PURA (5813)', 'PURB (5814)', 'PURG (29942)', 'PUS1 (80324)', 'PUS10 (150962)', 'PUS3 (83480)', 'PUS7 (54517)', 'PUS7L (83448)', 'PUSL1 (126789)', 'PVALB (5816)', 'PVR (5817)', 'PVRIG (79037)', 'PWP1 (11137)', 'PWP2 (5822)', 'PWWP2A (114825)', 'PWWP2B (170394)', 'PWWP3A (84939)', 'PWWP3B (139221)', 'PXDC1 (221749)', 'PXDN (7837)', 'PXDNL (137902)', 'PXK (54899)', 'PXMP2 (5827)', 'PXMP4 (11264)', 'PXN (5829)', 'PXT1 (222659)', 'PXYLP1 (92370)', 'PYCARD (29108)', 'PYCR1 (5831)', 'PYCR2 (29920)', 'PYCR3 (65263)', 'PYDC1 (260434)', 'PYGB (5834)', 'PYGL (5836)', 'PYGM (5837)', 'PYGO1 (26108)', 'PYGO2 (90780)', 'PYHIN1 (149628)', 'PYM1 (84305)', 'PYROXD1 (79912)', 'PYROXD2 (84795)', 'PYURF (100996939)', 'PYY (5697)', 'PZP (5858)', 'QARS1 (5859)', 'QDPR (5860)', 'QKI (9444)', 'QNG1 (84267)', 'QPCT (25797)', 'QPCTL (54814)', 'QPRT (23475)', 'QRFP (347148)', 'QRFPR (84109)', 'QRICH1 (54870)', 'QRICH2 (84074)', 'QRSL1 (55278)', 'QSER1 (79832)', 'QSOX1 (5768)', 'QSOX2 (169714)', 'QTRT1 (81890)', 'QTRT2 (79691)', 'R3HCC1 (203069)', 'R3HCC1L (27291)', 'R3HDM1 (23518)', 'R3HDM2 (22864)', 'R3HDM4 (91300)', 'R3HDML (140902)', 'RAB10 (10890)', 'RAB11A (8766)', 'RAB11B (9230)', 'RAB11FIP1 (80223)', 'RAB11FIP2 (22841)', 'RAB11FIP3 (9727)', 'RAB11FIP4 (84440)', 'RAB11FIP5 (26056)', 'RAB12 (201475)', 'RAB13 (5872)', 'RAB14 (51552)', 'RAB15 (376267)', 'RAB17 (64284)', 'RAB18 (22931)', 'RAB19 (401409)', 'RAB1A (5861)', 'RAB1B (81876)', 'RAB20 (55647)', 'RAB21 (23011)', 'RAB22A (57403)', 'RAB23 (51715)', 'RAB24 (53917)', 'RAB25 (57111)', 'RAB26 (25837)', 'RAB27A (5873)', 'RAB27B (5874)', 'RAB28 (9364)', 'RAB29 (8934)', 'RAB2A (5862)', 'RAB2B (84932)', 'RAB30 (27314)', 'RAB31 (11031)', 'RAB32 (10981)', 'RAB33A (9363)', 'RAB33B (83452)', 'RAB34 (83871)', 'RAB35 (11021)', 'RAB36 (9609)', 'RAB37 (326624)', 'RAB38 (23682)', 'RAB39A (54734)', 'RAB39B (116442)', 'RAB3A (5864)', 'RAB3B (5865)', 'RAB3C (115827)', 'RAB3D (9545)', 'RAB3GAP1 (22930)', 'RAB3GAP2 (25782)', 'RAB3IL1 (5866)', 'RAB3IP (117177)', 'RAB40A (142684)', 'RAB40AL (282808)', 'RAB40B (10966)', 'RAB40C (57799)', 'RAB41 (347517)', 'RAB42 (115273)', 'RAB43 (339122)', 'RAB4A (5867)', 'RAB4B (53916)', 'RAB5A (5868)', 'RAB5B (5869)', 'RAB5C (5878)', 'RAB5IF (55969)', 'RAB6A (5870)', 'RAB6B (51560)', 'RAB7A (7879)', 'RAB8A (4218)', 'RAB8B (51762)', 'RAB9A (9367)', 'RAB9B (51209)', 'RABAC1 (10567)', 'RABEP1 (9135)', 'RABEP2 (79874)', 'RABEPK (10244)', 'RABGAP1 (23637)', 'RABGAP1L (9910)', 'RABGEF1 (27342)', 'RABGGTA (5875)', 'RABGGTB (5876)', 'RABIF (5877)', 'RABL2A (11159)', 'RABL2B (11158)', 'RABL3 (285282)', 'RABL6 (55684)', 'RAC1 (5879)', 'RAC2 (5880)', 'RAC3 (5881)', 'RACGAP1 (29127)', 'RACK1 (10399)', 'RAD1 (5810)', 'RAD17 (5884)', 'RAD18 (56852)', 'RAD21 (5885)', 'RAD21L1 (642636)', 'RAD23A (5886)', 'RAD23B (5887)', 'RAD50 (10111)', 'RAD51 (5888)', 'RAD51AP1 (10635)', 'RAD51AP2 (729475)', 'RAD51B (5890)', 'RAD51C (5889)', 'RAD51D (5892)', 'RAD52 (5893)', 'RAD54B (25788)', 'RAD54L (8438)', 'RAD54L2 (23132)', 'RAD9A (5883)', 'RAD9B (144715)', 'RADIL (55698)', 'RADX (55086)', 'RAE1 (8480)', 'RAET1E (135250)', 'RAET1G (353091)', 'RAET1L (154064)', 'RAF1 (5894)', 'RAG1 (5896)', 'RAG2 (5897)', 'RAI1 (10743)', 'RAI14 (26064)', 'RAI2 (10742)', 'RALA (5898)', 'RALB (5899)', 'RALBP1 (10928)', 'RALGAPA1 (253959)', 'RALGAPA2 (57186)', 'RALGAPB (57148)', 'RALGDS (5900)', 'RALGPS1 (9649)', 'RALGPS2 (55103)', 'RALY (22913)', 'RALYL (138046)', 'RAMP1 (10267)', 'RAMP2 (10266)', 'RAMP3 (10268)', 'RAN (5901)', 'RANBP1 (5902)', 'RANBP10 (57610)', 'RANBP17 (64901)', 'RANBP2 (5903)', 'RANBP3 (8498)', 'RANBP3L (202151)', 'RANBP6 (26953)', 'RANBP9 (10048)', 'RANGAP1 (5905)', 'RANGRF (29098)', 'RAP1A (5906)', 'RAP1B (5908)', 'RAP1GAP (5909)', 'RAP1GAP2 (23108)', 'RAP1GDS1 (5910)', 'RAP2A (5911)', 'RAP2B (5912)', 'RAP2C (57826)', 'RAPGEF1 (2889)', 'RAPGEF2 (9693)', 'RAPGEF3 (10411)', 'RAPGEF4 (11069)', 'RAPGEF5 (9771)', 'RAPGEF6 (51735)', 'RAPGEFL1 (51195)', 'RAPH1 (65059)', 'RAPSN (5913)', 'RARA (5914)', 'RARB (5915)', 'RARG (5916)', 'RARRES1 (5918)', 'RARRES2 (5919)', 'RARS1 (5917)', 'RARS2 (57038)', 'RASA1 (5921)', 'RASA2 (5922)', 'RASA3 (22821)', 'RASA4 (10156)', 'RASA4B (100271927)', 'RASAL1 (8437)', 'RASAL2 (9462)', 'RASAL3 (64926)', 'RASD1 (51655)', 'RASD2 (23551)', 'RASEF (158158)', 'RASGEF1A (221002)', 'RASGEF1B (153020)', 'RASGEF1C (255426)', 'RASGRF1 (5923)', 'RASGRF2 (5924)', 'RASGRP1 (10125)', 'RASGRP2 (10235)', 'RASGRP3 (25780)', 'RASGRP4 (115727)', 'RASIP1 (54922)', 'RASL10A (10633)', 'RASL10B (91608)', 'RASL11A (387496)', 'RASL11B (65997)', 'RASL12 (51285)', 'RASSF1 (11186)', 'RASSF2 (9770)', 'RASSF3 (283349)', 'RASSF4 (83937)', 'RASSF5 (83593)', 'RASSF6 (166824)', 'RASSF7 (8045)', 'RASSF8 (11228)', 'RASSF9 (9182)', 'RAVER1 (125950)', 'RAVER2 (55225)', 'RAX (30062)', 'RAX2 (84839)', 'RB1 (5925)', 'RB1CC1 (9821)', 'RBAK (57786)', 'RBBP4 (5928)', 'RBBP5 (5929)', 'RBBP6 (5930)', 'RBBP7 (5931)', 'RBBP8 (5932)', 'RBBP8NL (140893)', 'RBBP9 (10741)', 'RBCK1 (10616)', 'RBFA (79863)', 'RBFOX1 (54715)', 'RBFOX2 (23543)', 'RBFOX3 (146713)', 'RBIS (401466)', 'RBKS (64080)', 'RBL1 (5933)', 'RBL2 (5934)', 'RBM10 (8241)', 'RBM11 (54033)', 'RBM12 (10137)', 'RBM12B (389677)', 'RBM14 (10432)', 'RBM15 (64783)', 'RBM15B (29890)', 'RBM17 (84991)', 'RBM18 (92400)', 'RBM19 (9904)', 'RBM20 (282996)', 'RBM22 (55696)', 'RBM23 (55147)', 'RBM24 (221662)', 'RBM25 (58517)', 'RBM26 (64062)', 'RBM27 (54439)', 'RBM28 (55131)', 'RBM3 (5935)', 'RBM33 (155435)', 'RBM34 (23029)', 'RBM38 (55544)', 'RBM39 (9584)', 'RBM4 (5936)', 'RBM41 (55285)', 'RBM42 (79171)', 'RBM43 (375287)', 'RBM44 (375316)', 'RBM45 (129831)', 'RBM46 (166863)', 'RBM47 (54502)', 'RBM48 (84060)', 'RBM4B (83759)', 'RBM5 (10181)', 'RBM6 (10180)', 'RBM7 (10179)', 'RBM8A (9939)', 'RBMS1 (5937)', 'RBMS2 (5939)', 'RBMS3 (27303)', 'RBMX (27316)', 'RBMX2 (51634)', 'RBMXL1 (494115)', 'RBMXL2 (27288)', 'RBMXL3 (139804)', 'RBP1 (5947)', 'RBP2 (5948)', 'RBP3 (5949)', 'RBP4 (5950)', 'RBP5 (83758)', 'RBP7 (116362)', 'RBPJ (3516)', 'RBPJL (11317)', 'RBPMS (11030)', 'RBPMS2 (348093)', 'RBSN (64145)', 'RBX1 (9978)', 'RC3H1 (149041)', 'RC3H2 (54542)', 'RCAN1 (1827)', 'RCAN2 (10231)', 'RCAN3 (11123)', 'RCBTB1 (55213)', 'RCBTB2 (1102)', 'RCC1 (1104)', 'RCC1L (81554)', 'RCC2 (55920)', 'RCCD1 (91433)', 'RCE1 (9986)', 'RCHY1 (25898)', 'RCL1 (10171)', 'RCN1 (5954)', 'RCN2 (5955)', 'RCN3 (57333)', 'RCOR1 (23186)', 'RCOR2 (283248)', 'RCOR3 (55758)', 'RCSD1 (92241)', 'RCVRN (5957)', 'RD3 (343035)', 'RD3L (647286)', 'RDH10 (157506)', 'RDH11 (51109)', 'RDH12 (145226)', 'RDH13 (112724)', 'RDH14 (57665)', 'RDH16 (8608)', 'RDH5 (5959)', 'RDH8 (50700)', 'RDM1 (201299)', 'RDX (5962)', 'REC114 (283677)', 'REC8 (9985)', 'RECK (8434)', 'RECQL (5965)', 'RECQL5 (9400)', 'REDIC1 (283461)', 'REEP1 (65055)', 'REEP2 (51308)', 'REEP3 (221035)', 'REEP4 (80346)', 'REEP5 (7905)', 'REEP6 (92840)', 'REG1A (5967)', 'REG1B (5968)', 'REG3A (5068)', 'REG3G (130120)', 'REG4 (83998)', 'REL (5966)', 'RELA (5970)', 'RELB (5971)', 'RELCH (57614)', 'RELL1 (768211)', 'RELL2 (285613)', 'RELN (5649)', 'RELT (84957)', 'REM1 (28954)', 'REM2 (161253)', 'REN (5972)', 'RENBP (5973)', 'REP15 (387849)', 'REPIN1 (29803)', 'REPS1 (85021)', 'REPS2 (9185)', 'RER1 (11079)', 'RERE (473)', 'RERG (85004)', 'RERGL (79785)', 'RESF1 (55196)', 'RESP18 (389075)', 'REST (5978)', 'RET (5979)', 'RETN (56729)', 'RETNLB (84666)', 'RETREG1 (54463)', 'RETREG2 (79137)', 'RETREG3 (162427)', 'RETSAT (54884)', 'REV1 (51455)', 'REV3L (5980)', 'REX1BD (55049)', 'REXO1 (57455)', 'REXO2 (25996)', 'REXO4 (57109)', 'REXO5 (81691)', 'RFC1 (5981)', 'RFC2 (5982)', 'RFC3 (5983)', 'RFC4 (5984)', 'RFC5 (5985)', 'RFESD (317671)', 'RFFL (117584)', 'RFK (55312)', 'RFLNA (144347)', 'RFNG (5986)', 'RFPL1 (5988)', 'RFPL2 (10739)', 'RFPL3 (10738)', 'RFPL4B (442247)', 'RFT1 (91869)', 'RFTN1 (23180)', 'RFTN2 (130132)', 'RFWD3 (55159)', 'RFX1 (5989)', 'RFX2 (5990)', 'RFX3 (5991)', 'RFX4 (5992)', 'RFX5 (5993)', 'RFX6 (222546)', 'RFX8 (731220)', 'RFXANK (8625)', 'RFXAP (5994)', 'RGCC (28984)', 'RGL1 (23179)', 'RGL2 (5863)', 'RGL3 (57139)', 'RGL4 (266747)', 'RGMA (56963)', 'RGMB (285704)', 'RGN (9104)', 'RGP1 (9827)', 'RGPD4 (285190)', 'RGR (5995)', 'RGS1 (5996)', 'RGS10 (6001)', 'RGS11 (8786)', 'RGS12 (6002)', 'RGS13 (6003)', 'RGS14 (10636)', 'RGS16 (6004)', 'RGS17 (26575)', 'RGS18 (64407)', 'RGS19 (10287)', 'RGS2 (5997)', 'RGS20 (8601)', 'RGS21 (431704)', 'RGS22 (26166)', 'RGS3 (5998)', 'RGS4 (5999)', 'RGS5 (8490)', 'RGS6 (9628)', 'RGS7 (6000)', 'RGS7BP (401190)', 'RGS8 (85397)', 'RGS9 (8787)', 'RGS9BP (388531)', 'RGSL1 (353299)', 'RHAG (6005)', 'RHBDD1 (84236)', 'RHBDD2 (57414)', 'RHBDD3 (25807)', 'RHBDF1 (64285)', 'RHBDF2 (79651)', 'RHBDL1 (9028)', 'RHBDL2 (54933)', 'RHBDL3 (162494)', 'RHCE (6006)', 'RHCG (51458)', 'RHD (6007)', 'RHEB (6009)', 'RHEBL1 (121268)', 'RHEX (440712)', 'RHNO1 (83695)', 'RHO (6010)', 'RHOA (387)', 'RHOB (388)', 'RHOBTB1 (9886)', 'RHOBTB2 (23221)', 'RHOBTB3 (22836)', 'RHOC (389)', 'RHOD (29984)', 'RHOF (54509)', 'RHOG (391)', 'RHOH (399)', 'RHOJ (57381)', 'RHOQ (23433)', 'RHOT1 (55288)', 'RHOT2 (89941)', 'RHOU (58480)', 'RHOV (171177)', 'RHOXF1 (158800)', 'RHOXF2 (84528)', 'RHOXF2B (727940)', 'RHPN1 (114822)', 'RHPN2 (85415)', 'RIBC1 (158787)', 'RIC1 (57589)', 'RIC3 (79608)', 'RIC8A (60626)', 'RIC8B (55188)', 'RICTOR (253260)', 'RIDA (10247)', 'RIF1 (55183)', 'RIGI (23586)', 'RIIAD1 (284485)', 'RILP (83547)', 'RILPL1 (353116)', 'RILPL2 (196383)', 'RIMBP2 (23504)', 'RIMBP3 (85376)', 'RIMKLA (284716)', 'RIMKLB (57494)', 'RIMOC1 (285636)', 'RIMS1 (22999)', 'RIMS2 (9699)', 'RIMS3 (9783)', 'RIMS4 (140730)', 'RIN1 (9610)', 'RIN2 (54453)', 'RIN3 (79890)', 'RING1 (6015)', 'RINL (126432)', 'RINT1 (60561)', 'RIOK1 (83732)', 'RIOK2 (55781)', 'RIOK3 (8780)', 'RIOX2 (84864)', 'RIPK1 (8737)', 'RIPK2 (8767)', 'RIPK3 (11035)', 'RIPK4 (54101)', 'RIPOR1 (79567)', 'RIPOR2 (9750)', 'RIPOR3 (140876)', 'RIPPLY1 (92129)', 'RIPPLY2 (134701)', 'RIPPLY3 (53820)', 'RIT1 (6016)', 'RIT2 (6014)', 'RITA1 (84934)', 'RLBP1 (6017)', 'RLF (6018)', 'RLIG1 (91298)', 'RLIM (51132)', 'RLN1 (6013)', 'RLN2 (6019)', 'RLN3 (117579)', 'RMC1 (29919)', 'RMDN1 (51115)', 'RMDN2 (151393)', 'RMDN3 (55177)', 'RMI1 (80010)', 'RMI2 (116028)', 'RMND1 (55005)', 'RMND5A (64795)', 'RMND5B (64777)', 'RNASE1 (6035)', 'RNASE10 (338879)', 'RNASE11 (122651)', 'RNASE12 (493901)', 'RNASE13 (440163)', 'RNASE2 (6036)', 'RNASE3 (6037)', 'RNASE4 (6038)', 'RNASE6 (6039)', 'RNASE7 (84659)', 'RNASE8 (122665)', 'RNASE9 (390443)', 'RNASEH1 (246243)', 'RNASEH2A (10535)', 'RNASEH2B (79621)', 'RNASEH2C (84153)', 'RNASEK (440400)', 'RNASEL (6041)', 'RNASET2 (8635)', 'RND1 (27289)', 'RND2 (8153)', 'RND3 (390)', 'RNF10 (9921)', 'RNF103 (7844)', 'RNF11 (26994)', 'RNF111 (54778)', 'RNF112 (7732)', 'RNF113A (7737)', 'RNF113B (140432)', 'RNF114 (55905)', 'RNF115 (27246)', 'RNF121 (55298)', 'RNF122 (79845)', 'RNF123 (63891)', 'RNF125 (54941)', 'RNF126 (55658)', 'RNF128 (79589)', 'RNF13 (11342)', 'RNF130 (55819)', 'RNF133 (168433)', 'RNF135 (84282)', 'RNF138 (51444)', 'RNF139 (11236)', 'RNF14 (9604)', 'RNF141 (50862)', 'RNF144A (9781)', 'RNF144B (255488)', 'RNF145 (153830)', 'RNF146 (81847)', 'RNF148 (378925)', 'RNF149 (284996)', 'RNF150 (57484)', 'RNF151 (146310)', 'RNF152 (220441)', 'RNF157 (114804)', 'RNF166 (115992)', 'RNF167 (26001)', 'RNF168 (165918)', 'RNF169 (254225)', 'RNF17 (56163)', 'RNF170 (81790)', 'RNF175 (285533)', 'RNF180 (285671)', 'RNF181 (51255)', 'RNF182 (221687)', 'RNF183 (138065)', 'RNF185 (91445)', 'RNF186 (54546)', 'RNF19A (25897)', 'RNF19B (127544)', 'RNF2 (6045)', 'RNF20 (56254)', 'RNF207 (388591)', 'RNF208 (727800)', 'RNF212 (285498)', 'RNF212B (100507650)', 'RNF213 (57674)', 'RNF214 (257160)', 'RNF215 (200312)', 'RNF216 (54476)', 'RNF217 (154214)', 'RNF220 (55182)', 'RNF222 (643904)', 'RNF223 (401934)', 'RNF224 (643596)', 'RNF24 (11237)', 'RNF25 (64320)', 'RNF26 (79102)', 'RNF31 (55072)', 'RNF32 (140545)', 'RNF34 (80196)', 'RNF38 (152006)', 'RNF39 (80352)', 'RNF4 (6047)', 'RNF40 (9810)', 'RNF41 (10193)', 'RNF43 (54894)', 'RNF44 (22838)', 'RNF5 (6048)', 'RNF6 (6049)', 'RNF7 (9616)', 'RNF8 (9025)', 'RNFT1 (51136)', 'RNFT2 (84900)', 'RNGTT (8732)', 'RNH1 (6050)', 'RNLS (55328)', 'RNMT (8731)', 'RNPC3 (55599)', 'RNPEP (6051)', 'RNPS1 (10921)', 'RO60 (6738)', 'ROBO1 (6091)', 'ROBO2 (6092)', 'ROBO3 (64221)', 'ROBO4 (54538)', 'ROCK1 (6093)', 'ROCK2 (9475)', 'ROGDI (79641)', 'ROM1 (6094)', 'ROMO1 (140823)', 'ROPN1 (54763)', 'ROPN1B (152015)', 'ROPN1L (83853)', 'ROR1 (4919)', 'ROR2 (4920)', 'RORA (6095)', 'RORB (6096)', 'RORC (6097)', 'ROS1 (6098)', 'RP1 (6101)', 'RP1L1 (94137)', 'RP2 (6102)', 'RP9 (6100)', 'RPA1 (6117)', 'RPA2 (6118)', 'RPA3 (6119)', 'RPA4 (29935)', 'RPAIN (84268)', 'RPAP1 (26015)', 'RPAP2 (79871)', 'RPAP3 (79657)', 'RPE (6120)', 'RPE65 (6121)', 'RPEL1 (729020)', 'RPF1 (80135)', 'RPF2 (84154)', 'RPGR (6103)', 'RPGRIP1 (57096)', 'RPGRIP1L (23322)', 'RPH3A (22895)', 'RPH3AL (9501)', 'RPIA (22934)', 'RPL10 (6134)', 'RPL10A (4736)', 'RPL10L (140801)', 'RPL11 (6135)', 'RPL12 (6136)', 'RPL13 (6137)', 'RPL13A (23521)', 'RPL14 (9045)', 'RPL17 (6139)', 'RPL18 (6141)', 'RPL18A (6142)', 'RPL19 (6143)', 'RPL21 (6144)', 'RPL22 (6146)', 'RPL22L1 (200916)', 'RPL23 (9349)', 'RPL23A (6147)', 'RPL24 (6152)', 'RPL26 (6154)', 'RPL26L1 (51121)', 'RPL27 (6155)', 'RPL27A (6157)', 'RPL28 (6158)', 'RPL3 (6122)', 'RPL30 (6156)', 'RPL31 (6160)', 'RPL32 (6161)', 'RPL35 (11224)', 'RPL35A (6165)', 'RPL36 (25873)', 'RPL36A (6173)', 'RPL36AL (6166)', 'RPL37 (6167)', 'RPL37A (6168)', 'RPL38 (6169)', 'RPL39 (6170)', 'RPL39L (116832)', 'RPL3L (6123)', 'RPL4 (6124)', 'RPL41 (6171)', 'RPL5 (6125)', 'RPL6 (6128)', 'RPL7 (6129)', 'RPL7A (6130)', 'RPL7L1 (285855)', 'RPL8 (6132)', 'RPL9 (6133)', 'RPLP0 (6175)', 'RPLP1 (6176)', 'RPLP2 (6181)', 'RPN1 (6184)', 'RPN2 (6185)', 'RPP14 (11102)', 'RPP21 (79897)', 'RPP25 (54913)', 'RPP25L (138716)', 'RPP30 (10556)', 'RPP38 (10557)', 'RPP40 (10799)', 'RPRD1A (55197)', 'RPRD1B (58490)', 'RPRD2 (23248)', 'RPRM (56475)', 'RPRML (388394)', 'RPS10 (6204)', 'RPS11 (6205)', 'RPS12 (6206)', 'RPS13 (6207)', 'RPS14 (6208)', 'RPS15 (6209)', 'RPS15A (6210)', 'RPS16 (6217)', 'RPS17 (6218)', 'RPS18 (6222)', 'RPS19 (6223)', 'RPS19BP1 (91582)', 'RPS2 (6187)', 'RPS20 (6224)', 'RPS21 (6227)', 'RPS23 (6228)', 'RPS24 (6229)', 'RPS25 (6230)', 'RPS27A (6233)', 'RPS27L (51065)', 'RPS28 (6234)', 'RPS29 (6235)', 'RPS3 (6188)', 'RPS3A (6189)', 'RPS4X (6191)', 'RPS4Y1 (6192)', 'RPS4Y2 (140032)', 'RPS5 (6193)', 'RPS6 (6194)', 'RPS6KA1 (6195)', 'RPS6KA2 (6196)', 'RPS6KA3 (6197)', 'RPS6KA4 (8986)', 'RPS6KA5 (9252)', 'RPS6KA6 (27330)', 'RPS6KB1 (6198)', 'RPS6KB2 (6199)', 'RPS6KC1 (26750)', 'RPS6KL1 (83694)', 'RPS7 (6201)', 'RPS8 (6202)', 'RPS9 (6203)', 'RPTN (126638)', 'RPTOR (57521)', 'RPUSD1 (113000)', 'RPUSD2 (27079)', 'RPUSD3 (285367)', 'RPUSD4 (84881)', 'RRAD (6236)', 'RRAGA (10670)', 'RRAGB (10325)', 'RRAGC (64121)', 'RRAGD (58528)', 'RRAS (6237)', 'RRAS2 (22800)', 'RRBP1 (6238)', 'RREB1 (6239)', 'RRH (10692)', 'RRM1 (6240)', 'RRM2 (6241)', 'RRM2B (50484)', 'RRN3 (54700)', 'RRP1 (8568)', 'RRP12 (23223)', 'RRP15 (51018)', 'RRP1B (23076)', 'RRP36 (88745)', 'RRP7A (27341)', 'RRP8 (23378)', 'RRP9 (9136)', 'RRS1 (23212)', 'RS1 (6247)', 'RSAD1 (55316)', 'RSAD2 (91543)', 'RSBN1 (54665)', 'RSBN1L (222194)', 'RSC1A1 (6248)', 'RSF1 (51773)', 'RSL1D1 (26156)', 'RSL24D1 (51187)', 'RSPH1 (89765)', 'RSPH14 (27156)', 'RSPH3 (83861)', 'RSPH4A (345895)', 'RSPH6A (81492)', 'RSPH9 (221421)', 'RSPO1 (284654)', 'RSPO2 (340419)', 'RSPO3 (84870)', 'RSPO4 (343637)', 'RSPRY1 (89970)', 'RSRC1 (51319)', 'RSRC2 (65117)', 'RSRP1 (57035)', 'RSU1 (6251)', 'RTBDN (83546)', 'RTCA (8634)', 'RTCB (51493)', 'RTEL1 (51750)', 'RTF1 (23168)', 'RTF2 (51507)', 'RTKN (6242)', 'RTKN2 (219790)', 'RTL1 (388015)', 'RTL10 (79680)', 'RTL3 (203430)', 'RTL4 (340595)', 'RTL5 (340526)', 'RTL6 (84247)', 'RTL8A (26071)', 'RTL8B (441518)', 'RTL8C (8933)', 'RTL9 (57529)', 'RTN1 (6252)', 'RTN2 (6253)', 'RTN3 (10313)', 'RTN4 (57142)', 'RTN4IP1 (84816)', 'RTN4R (65078)', 'RTN4RL1 (146760)', 'RTN4RL2 (349667)', 'RTP1 (132112)', 'RTP2 (344892)', 'RTP3 (83597)', 'RTP4 (64108)', 'RTP5 (285093)', 'RTRAF (51637)', 'RTTN (25914)', 'RUBCN (9711)', 'RUBCNL (80183)', 'RUFY1 (80230)', 'RUFY2 (55680)', 'RUFY3 (22902)', 'RUNDC1 (146923)', 'RUNDC3A (10900)', 'RUNDC3B (154661)', 'RUNX1 (861)', 'RUNX1T1 (862)', 'RUNX2 (860)', 'RUNX3 (864)', 'RUSC1 (23623)', 'RUSC2 (9853)', 'RUSF1 (64755)', 'RUVBL1 (8607)', 'RUVBL2 (10856)', 'RWDD1 (51389)', 'RWDD2A (112611)', 'RWDD2B (10069)', 'RWDD3 (25950)', 'RWDD4 (201965)', 'RXFP1 (59350)', 'RXFP2 (122042)', 'RXFP3 (51289)', 'RXFP4 (339403)', 'RXRA (6256)', 'RXRB (6257)', 'RXRG (6258)', 'RXYLT1 (10329)', 'RYR1 (6261)', 'RYR2 (6262)', 'RYR3 (6263)', 'S100A1 (6271)', 'S100A10 (6281)', 'S100A11 (6282)', 'S100A12 (6283)', 'S100A13 (6284)', 'S100A14 (57402)', 'S100A16 (140576)', 'S100A2 (6273)', 'S100A3 (6274)', 'S100A4 (6275)', 'S100A5 (6276)', 'S100A6 (6277)', 'S100A7 (6278)', 'S100A8 (6279)', 'S100A9 (6280)', 'S100B (6285)', 'S100G (795)', 'S100P (6286)', 'S100PBP (64766)', 'S100Z (170591)', 'S1PR1 (1901)', 'S1PR2 (9294)', 'S1PR3 (1903)', 'S1PR4 (8698)', 'S1PR5 (53637)', 'SAA4 (6291)', 'SAAL1 (113174)', 'SAC3D1 (29901)', 'SACM1L (22908)', 'SACS (26278)', 'SAE1 (10055)', 'SAFB (6294)', 'SAFB2 (9667)', 'SAG (6295)', 'SAGE1 (55511)', 'SALL1 (6299)', 'SALL2 (6297)', 'SALL3 (27164)', 'SALL4 (57167)', 'SAMD10 (140700)', 'SAMD11 (148398)', 'SAMD12 (401474)', 'SAMD13 (148418)', 'SAMD14 (201191)', 'SAMD15 (161394)', 'SAMD3 (154075)', 'SAMD4A (23034)', 'SAMD4B (55095)', 'SAMD5 (389432)', 'SAMD7 (344658)', 'SAMD8 (142891)', 'SAMD9 (54809)', 'SAMD9L (219285)', 'SAMHD1 (25939)', 'SAMM50 (25813)', 'SAMSN1 (64092)', 'SANBR (84542)', 'SAP130 (79595)', 'SAP18 (10284)', 'SAP25 (100316904)', 'SAP30 (8819)', 'SAP30BP (29115)', 'SAP30L (79685)', 'SAPCD1 (401251)', 'SAPCD2 (89958)', 'SAR1A (56681)', 'SAR1B (51128)', 'SARAF (51669)', 'SARDH (1757)', 'SARNP (84324)', 'SARS1 (6301)', 'SARS2 (54938)', 'SART1 (9092)', 'SART3 (9733)', 'SASH1 (23328)', 'SASH3 (54440)', 'SASS6 (163786)', 'SAT1 (6303)', 'SAT2 (112483)', 'SATB1 (6304)', 'SATB2 (23314)', 'SATL1 (340562)', 'SAV1 (60485)', 'SAXO1 (158297)', 'SAXO2 (283726)', 'SAXO4 (220004)', 'SAXO5 (374877)', 'SAYSD1 (55776)', 'SBDS (51119)', 'SBF1 (6305)', 'SBF2 (81846)', 'SBK1 (388228)', 'SBK2 (646643)', 'SBNO1 (55206)', 'SBNO2 (22904)', 'SBSN (374897)', 'SBSPON (157869)', 'SC5D (6309)', 'SCAF1 (58506)', 'SCAF11 (9169)', 'SCAF4 (57466)', 'SCAF8 (22828)', 'SCAI (286205)', 'SCAMP2 (10066)', 'SCAMP3 (10067)', 'SCAMP4 (113178)', 'SCAMP5 (192683)', 'SCAND1 (51282)', 'SCAND3 (114821)', 'SCAP (22937)', 'SCAPER (49855)', 'SCARA3 (51435)', 'SCARA5 (286133)', 'SCARB1 (949)', 'SCARB2 (950)', 'SCARF1 (8578)', 'SCARF2 (91179)', 'SCCPDH (51097)', 'SCD (6319)', 'SCD5 (79966)', 'SCEL (8796)', 'SCFD1 (23256)', 'SCFD2 (152579)', 'SCG2 (7857)', 'SCG3 (29106)', 'SCG5 (6447)', 'SCGB1A1 (7356)', 'SCGB1D1 (10648)', 'SCGB1D2 (10647)', 'SCGB1D4 (404552)', 'SCGB2A1 (4246)', 'SCGB2A2 (4250)', 'SCGB2B2 (284402)', 'SCGB3A1 (92304)', 'SCGB3A2 (117156)', 'SCGN (10590)', 'SCHIP1 (29970)', 'SCIMP (388325)', 'SCIN (85477)', 'SCLT1 (132320)', 'SCLY (51540)', 'SCMH1 (22955)', 'SCML1 (6322)', 'SCML2 (10389)', 'SCML4 (256380)', 'SCN10A (6336)', 'SCN11A (11280)', 'SCN1A (6323)', 'SCN1B (6324)', 'SCN2A (6326)', 'SCN2B (6327)', 'SCN3A (6328)', 'SCN3B (55800)', 'SCN4A (6329)', 'SCN4B (6330)', 'SCN5A (6331)', 'SCN7A (6332)', 'SCN8A (6334)', 'SCN9A (6335)', 'SCNM1 (79005)', 'SCNN1A (6337)', 'SCNN1B (6338)', 'SCNN1D (6339)', 'SCNN1G (6340)', 'SCO1 (6341)', 'SCO2 (9997)', 'SCOC (60592)', 'SCP2 (6342)', 'SCP2D1 (140856)', 'SCPEP1 (59342)', 'SCRG1 (11341)', 'SCRIB (23513)', 'SCRN1 (9805)', 'SCRN2 (90507)', 'SCRN3 (79634)', 'SCRT1 (83482)', 'SCRT2 (85508)', 'SCT (6343)', 'SCTR (6344)', 'SCUBE1 (80274)', 'SCUBE2 (57758)', 'SCUBE3 (222663)', 'SCX (642658)', 'SCYL1 (57410)', 'SCYL2 (55681)', 'SCYL3 (57147)', 'SDAD1 (55153)', 'SDC1 (6382)', 'SDC2 (6383)', 'SDC3 (9672)', 'SDC4 (6385)', 'SDCBP (6386)', 'SDCBP2 (27111)', 'SDCCAG8 (10806)', 'SDE2 (163859)', 'SDF2 (6388)', 'SDF2L1 (23753)', 'SDF4 (51150)', 'SDHA (6389)', 'SDHAF1 (644096)', 'SDHAF2 (54949)', 'SDHAF3 (57001)', 'SDHAF4 (135154)', 'SDHB (6390)', 'SDHC (6391)', 'SDHD (6392)', 'SDK1 (221935)', 'SDK2 (54549)', 'SDR16C5 (195814)', 'SDR39U1 (56948)', 'SDR42E1 (93517)', 'SDR9C7 (121214)', 'SDS (10993)', 'SDSL (113675)', 'SEBOX (645832)', 'SEC11A (23478)', 'SEC11C (90701)', 'SEC13 (6396)', 'SEC14L1 (6397)', 'SEC14L2 (23541)', 'SEC14L3 (266629)', 'SEC14L4 (284904)', 'SEC14L5 (9717)', 'SEC14L6 (730005)', 'SEC16A (9919)', 'SEC16B (89866)', 'SEC22A (26984)', 'SEC22C (9117)', 'SEC23A (10484)', 'SEC23B (10483)', 'SEC23IP (11196)', 'SEC24A (10802)', 'SEC24B (10427)', 'SEC24C (9632)', 'SEC24D (9871)', 'SEC31A (22872)', 'SEC31B (25956)', 'SEC61A1 (29927)', 'SEC61A2 (55176)', 'SEC61B (10952)', 'SEC61G (23480)', 'SEC62 (7095)', 'SEC63 (11231)', 'SECISBP2 (79048)', 'SECISBP2L (9728)', 'SECTM1 (6398)', 'SEH1L (81929)', 'SEL1L (6400)', 'SEL1L2 (80343)', 'SEL1L3 (23231)', 'SELE (6401)', 'SELENBP1 (8991)', 'SELENOH (280636)', 'SELENOI (85465)', 'SELENOK (58515)', 'SELENOM (140606)', 'SELENON (57190)', 'SELENOO (83642)', 'SELENOP (6414)', 'SELENOS (55829)', 'SELENOT (51714)', 'SELENOV (348303)', 'SELENOW (6415)', 'SELL (6402)', 'SELP (6403)', 'SELPLG (6404)', 'SEM1 (7979)', 'SEMA3A (10371)', 'SEMA3C (10512)', 'SEMA3D (223117)', 'SEMA3E (9723)', 'SEMA3F (6405)', 'SEMA3G (56920)', 'SEMA4A (64218)', 'SEMA4B (10509)', 'SEMA4C (54910)', 'SEMA4D (10507)', 'SEMA4F (10505)', 'SEMA4G (57715)', 'SEMA5A (9037)', 'SEMA5B (54437)', 'SEMA6A (57556)', 'SEMA6B (10501)', 'SEMA6C (10500)', 'SEMA6D (80031)', 'SEMA7A (8482)', 'SEMG1 (6406)', 'SEMG2 (6407)', 'SENP1 (29843)', 'SENP2 (59343)', 'SENP5 (205564)', 'SENP6 (26054)', 'SENP7 (57337)', 'SENP8 (123228)', 'SEPHS1 (22929)', 'SEPHS2 (22928)', 'SEPSECS (51091)', 'SEPTIN10 (151011)', 'SEPTIN11 (55752)', 'SEPTIN12 (124404)', 'SEPTIN14 (346288)', 'SEPTIN2 (4735)', 'SEPTIN3 (55964)', 'SEPTIN4 (5414)', 'SEPTIN5 (5413)', 'SEPTIN6 (23157)', 'SEPTIN8 (23176)', 'SEPTIN9 (10801)', 'SERAC1 (84947)', 'SERBP1 (26135)', 'SERF2 (10169)', 'SERGEF (26297)', 'SERHL2 (253190)', 'SERINC1 (57515)', 'SERINC2 (347735)', 'SERINC3 (10955)', 'SERINC4 (619189)', 'SERINC5 (256987)', 'SERP1 (27230)', 'SERP2 (387923)', 'SERPINA1 (5265)', 'SERPINA10 (51156)', 'SERPINA11 (256394)', 'SERPINA12 (145264)', 'SERPINA3 (12)', 'SERPINA4 (5267)', 'SERPINA5 (5104)', 'SERPINA6 (866)', 'SERPINA7 (6906)', 'SERPINA9 (327657)', 'SERPINB1 (1992)', 'SERPINB10 (5273)', 'SERPINB12 (89777)', 'SERPINB13 (5275)', 'SERPINB2 (5055)', 'SERPINB3 (6317)', 'SERPINB4 (6318)', 'SERPINB5 (5268)', 'SERPINB6 (5269)', 'SERPINB7 (8710)', 'SERPINB8 (5271)', 'SERPINB9 (5272)', 'SERPINC1 (462)', 'SERPIND1 (3053)', 'SERPINE1 (5054)', 'SERPINE2 (5270)', 'SERPINE3 (647174)', 'SERPINF1 (5176)', 'SERPINF2 (5345)', 'SERPING1 (710)', 'SERPINH1 (871)', 'SERPINI1 (5274)', 'SERPINI2 (5276)', 'SERTAD1 (29950)', 'SERTAD2 (9792)', 'SERTAD3 (29946)', 'SERTAD4 (56256)', 'SERTM1 (400120)', 'SESN1 (27244)', 'SESN2 (83667)', 'SESN3 (143686)', 'SESTD1 (91404)', 'SET (6418)', 'SETBP1 (26040)', 'SETD1A (9739)', 'SETD1B (23067)', 'SETD2 (29072)', 'SETD3 (84193)', 'SETD4 (54093)', 'SETD5 (55209)', 'SETD6 (79918)', 'SETD7 (80854)', 'SETD9 (133383)', 'SETDB1 (9869)', 'SETDB2 (83852)', 'SETMAR (6419)', 'SETX (23064)', 'SEZ6 (124925)', 'SEZ6L (23544)', 'SEZ6L2 (26470)', 'SF1 (7536)', 'SF3A1 (10291)', 'SF3A2 (8175)', 'SF3A3 (10946)', 'SF3B1 (23451)', 'SF3B2 (10992)', 'SF3B3 (23450)', 'SF3B4 (10262)', 'SF3B5 (83443)', 'SF3B6 (51639)', 'SFI1 (9814)', 'SFMBT1 (51460)', 'SFMBT2 (57713)', 'SFN (2810)', 'SFPQ (6421)', 'SFR1 (119392)', 'SFRP1 (6422)', 'SFRP2 (6423)', 'SFRP4 (6424)', 'SFRP5 (6425)', 'SFSWAP (6433)', 'SFT2D1 (113402)', 'SFT2D2 (375035)', 'SFT2D3 (84826)', 'SFTA2 (389376)', 'SFTPA1 (653509)', 'SFTPA2 (729238)', 'SFTPB (6439)', 'SFTPC (6440)', 'SFTPD (6441)', 'SFXN1 (94081)', 'SFXN2 (118980)', 'SFXN3 (81855)', 'SFXN4 (119559)', 'SFXN5 (94097)', 'SGCA (6442)', 'SGCB (6443)', 'SGCD (6444)', 'SGCE (8910)', 'SGCG (6445)', 'SGCZ (137868)', 'SGF29 (112869)', 'SGIP1 (84251)', 'SGK1 (6446)', 'SGK2 (10110)', 'SGK3 (23678)', 'SGMS1 (259230)', 'SGMS2 (166929)', 'SGO1 (151648)', 'SGO2 (151246)', 'SGPL1 (8879)', 'SGPP1 (81537)', 'SGPP2 (130367)', 'SGSH (6448)', 'SGSM1 (129049)', 'SGSM2 (9905)', 'SGSM3 (27352)', 'SGTA (6449)', 'SGTB (54557)', 'SH2B1 (25970)', 'SH2B3 (10019)', 'SH2D1A (4068)', 'SH2D1B (117157)', 'SH2D2A (9047)', 'SH2D3A (10045)', 'SH2D3C (10044)', 'SH2D4A (63898)', 'SH2D4B (387694)', 'SH2D5 (400745)', 'SH2D6 (284948)', 'SH2D7 (646892)', 'SH3BGR (6450)', 'SH3BGRL (6451)', 'SH3BGRL2 (83699)', 'SH3BGRL3 (83442)', 'SH3BP1 (23616)', 'SH3BP2 (6452)', 'SH3BP4 (23677)', 'SH3BP5 (9467)', 'SH3BP5L (80851)', 'SH3D19 (152503)', 'SH3D21 (79729)', 'SH3GL1 (6455)', 'SH3GL2 (6456)', 'SH3GL3 (6457)', 'SH3GLB1 (51100)', 'SH3GLB2 (56904)', 'SH3KBP1 (30011)', 'SH3PXD2A (9644)', 'SH3PXD2B (285590)', 'SH3RF1 (57630)', 'SH3RF2 (153769)', 'SH3TC1 (54436)', 'SH3TC2 (79628)', 'SH3YL1 (26751)', 'SHANK1 (50944)', 'SHARPIN (81858)', 'SHB (6461)', 'SHBG (6462)', 'SHC1 (6464)', 'SHC2 (25759)', 'SHC3 (53358)', 'SHC4 (399694)', 'SHCBP1 (79801)', 'SHCBP1L (81626)', 'SHD (56961)', 'SHE (126669)', 'SHF (90525)', 'SHFL (55337)', 'SHH (6469)', 'SHISA2 (387914)', 'SHISA3 (152573)', 'SHISA4 (149345)', 'SHISA5 (51246)', 'SHISA6 (388336)', 'SHISA7 (729956)', 'SHISA9 (729993)', 'SHISAL1 (85352)', 'SHISAL2A (348378)', 'SHISAL2B (100132916)', 'SHKBP1 (92799)', 'SHLD1 (149840)', 'SHLD2 (54537)', 'SHMT1 (6470)', 'SHMT2 (6472)', 'SHOC1 (158401)', 'SHOC2 (8036)', 'SHOX2 (6474)', 'SHPK (23729)', 'SHPRH (257218)', 'SHQ1 (55164)', 'SHROOM1 (134549)', 'SHROOM2 (357)', 'SHROOM3 (57619)', 'SHROOM4 (57477)', 'SHTN1 (57698)', 'SI (6476)', 'SIAE (54414)', 'SIAH1 (6477)', 'SIAH2 (6478)', 'SIAH3 (283514)', 'SIDT1 (54847)', 'SIDT2 (51092)', 'SIGIRR (59307)', 'SIGLEC1 (6614)', 'SIGLEC10 (89790)', 'SIGLEC11 (114132)', 'SIGLEC12 (89858)', 'SIGLEC15 (284266)', 'SIGLEC5 (8778)', 'SIGLEC6 (946)', 'SIGLEC7 (27036)', 'SIGLEC8 (27181)', 'SIGLEC9 (27180)', 'SIGLECL1 (284369)', 'SIGMAR1 (10280)', 'SIK1 (150094)', 'SIK2 (23235)', 'SIK3 (23387)', 'SIKE1 (80143)', 'SIL1 (64374)', 'SIM1 (6492)', 'SIM2 (6493)', 'SIMC1 (375484)', 'SIN3A (25942)', 'SIN3B (23309)', 'SINHCAF (58516)', 'SIPA1 (6494)', 'SIPA1L1 (26037)', 'SIPA1L2 (57568)', 'SIPA1L3 (23094)', 'SIRPA (140885)', 'SIRPB1 (10326)', 'SIRPB2 (284759)', 'SIRPD (128646)', 'SIRPG (55423)', 'SIRT1 (23411)', 'SIRT2 (22933)', 'SIRT3 (23410)', 'SIRT4 (23409)', 'SIRT5 (23408)', 'SIRT6 (51548)', 'SIRT7 (51547)', 'SIT1 (27240)', 'SIVA1 (10572)', 'SIX1 (6495)', 'SIX2 (10736)', 'SIX3 (6496)', 'SIX4 (51804)', 'SIX5 (147912)', 'SIX6 (4990)', 'SKA1 (220134)', 'SKA2 (348235)', 'SKA3 (221150)', 'SKAP1 (8631)', 'SKAP2 (8935)', 'SKI (6497)', 'SKIC2 (6499)', 'SKIC3 (9652)', 'SKIC8 (80349)', 'SKIDA1 (387640)', 'SKIL (6498)', 'SKOR1 (390598)', 'SKOR2 (652991)', 'SKP1 (6500)', 'SKP2 (6502)', 'SLA (6503)', 'SLA2 (84174)', 'SLAIN1 (122060)', 'SLAIN2 (57606)', 'SLAMF1 (6504)', 'SLAMF6 (114836)', 'SLAMF7 (57823)', 'SLAMF8 (56833)', 'SLAMF9 (89886)', 'SLBP (7884)', 'SLC10A1 (6554)', 'SLC10A2 (6555)', 'SLC10A3 (8273)', 'SLC10A4 (201780)', 'SLC10A5 (347051)', 'SLC10A6 (345274)', 'SLC10A7 (84068)', 'SLC11A1 (6556)', 'SLC11A2 (4891)', 'SLC12A1 (6557)', 'SLC12A2 (6558)', 'SLC12A3 (6559)', 'SLC12A4 (6560)', 'SLC12A5 (57468)', 'SLC12A6 (9990)', 'SLC12A7 (10723)', 'SLC12A8 (84561)', 'SLC12A9 (56996)', 'SLC13A1 (6561)', 'SLC13A2 (9058)', 'SLC13A3 (64849)', 'SLC13A4 (26266)', 'SLC13A5 (284111)', 'SLC14A1 (6563)', 'SLC14A2 (8170)', 'SLC15A1 (6564)', 'SLC15A2 (6565)', 'SLC15A3 (51296)', 'SLC15A4 (121260)', 'SLC16A1 (6566)', 'SLC16A10 (117247)', 'SLC16A11 (162515)', 'SLC16A12 (387700)', 'SLC16A13 (201232)', 'SLC16A14 (151473)', 'SLC16A2 (6567)', 'SLC16A3 (9123)', 'SLC16A4 (9122)', 'SLC16A5 (9121)', 'SLC16A6 (9120)', 'SLC16A7 (9194)', 'SLC16A8 (23539)', 'SLC16A9 (220963)', 'SLC17A1 (6568)', 'SLC17A2 (10246)', 'SLC17A3 (10786)', 'SLC17A4 (10050)', 'SLC17A5 (26503)', 'SLC17A6 (57084)', 'SLC17A7 (57030)', 'SLC17A8 (246213)', 'SLC17A9 (63910)', 'SLC18A1 (6570)', 'SLC18A2 (6571)', 'SLC18A3 (6572)', 'SLC18B1 (116843)', 'SLC19A1 (6573)', 'SLC19A2 (10560)', 'SLC19A3 (80704)', 'SLC1A1 (6505)', 'SLC1A2 (6506)', 'SLC1A3 (6507)', 'SLC1A4 (6509)', 'SLC1A5 (6510)', 'SLC1A6 (6511)', 'SLC1A7 (6512)', 'SLC20A1 (6574)', 'SLC20A2 (6575)', 'SLC22A1 (6580)', 'SLC22A10 (387775)', 'SLC22A11 (55867)', 'SLC22A12 (116085)', 'SLC22A13 (9390)', 'SLC22A14 (9389)', 'SLC22A15 (55356)', 'SLC22A16 (85413)', 'SLC22A17 (51310)', 'SLC22A18 (5002)', 'SLC22A2 (6582)', 'SLC22A23 (63027)', 'SLC22A25 (387601)', 'SLC22A3 (6581)', 'SLC22A4 (6583)', 'SLC22A5 (6584)', 'SLC22A6 (9356)', 'SLC22A7 (10864)', 'SLC22A8 (9376)', 'SLC22A9 (114571)', 'SLC23A1 (9963)', 'SLC23A2 (9962)', 'SLC23A3 (151295)', 'SLC24A1 (9187)', 'SLC24A2 (25769)', 'SLC24A3 (57419)', 'SLC24A4 (123041)', 'SLC24A5 (283652)', 'SLC25A1 (6576)', 'SLC25A10 (1468)', 'SLC25A11 (8402)', 'SLC25A12 (8604)', 'SLC25A13 (10165)', 'SLC25A14 (9016)', 'SLC25A15 (10166)', 'SLC25A16 (8034)', 'SLC25A17 (10478)', 'SLC25A18 (83733)', 'SLC25A19 (60386)', 'SLC25A2 (83884)', 'SLC25A20 (788)', 'SLC25A21 (89874)', 'SLC25A22 (79751)', 'SLC25A23 (79085)', 'SLC25A24 (29957)', 'SLC25A25 (114789)', 'SLC25A26 (115286)', 'SLC25A27 (9481)', 'SLC25A28 (81894)', 'SLC25A29 (123096)', 'SLC25A3 (5250)', 'SLC25A30 (253512)', 'SLC25A31 (83447)', 'SLC25A32 (81034)', 'SLC25A33 (84275)', 'SLC25A34 (284723)', 'SLC25A35 (399512)', 'SLC25A36 (55186)', 'SLC25A37 (51312)', 'SLC25A38 (54977)', 'SLC25A39 (51629)', 'SLC25A4 (291)', 'SLC25A40 (55972)', 'SLC25A41 (284427)', 'SLC25A42 (284439)', 'SLC25A43 (203427)', 'SLC25A44 (9673)', 'SLC25A45 (283130)', 'SLC25A46 (91137)', 'SLC25A47 (283600)', 'SLC25A48 (153328)', 'SLC25A5 (292)', 'SLC25A51 (92014)', 'SLC25A52 (147407)', 'SLC25A53 (401612)', 'SLC26A1 (10861)', 'SLC26A11 (284129)', 'SLC26A2 (1836)', 'SLC26A3 (1811)', 'SLC26A4 (5172)', 'SLC26A5 (375611)', 'SLC26A6 (65010)', 'SLC26A7 (115111)', 'SLC26A8 (116369)', 'SLC26A9 (115019)', 'SLC27A1 (376497)', 'SLC27A2 (11001)', 'SLC27A3 (11000)', 'SLC27A4 (10999)', 'SLC27A5 (10998)', 'SLC27A6 (28965)', 'SLC28A1 (9154)', 'SLC28A2 (9153)', 'SLC28A3 (64078)', 'SLC29A1 (2030)', 'SLC29A2 (3177)', 'SLC29A3 (55315)', 'SLC29A4 (222962)', 'SLC2A1 (6513)', 'SLC2A10 (81031)', 'SLC2A11 (66035)', 'SLC2A12 (154091)', 'SLC2A13 (114134)', 'SLC2A14 (144195)', 'SLC2A2 (6514)', 'SLC2A3 (6515)', 'SLC2A4 (6517)', 'SLC2A4RG (56731)', 'SLC2A5 (6518)', 'SLC2A6 (11182)', 'SLC2A7 (155184)', 'SLC2A8 (29988)', 'SLC2A9 (56606)', 'SLC30A1 (7779)', 'SLC30A10 (55532)', 'SLC30A2 (7780)', 'SLC30A3 (7781)', 'SLC30A4 (7782)', 'SLC30A5 (64924)', 'SLC30A6 (55676)', 'SLC30A7 (148867)', 'SLC30A8 (169026)', 'SLC30A9 (10463)', 'SLC31A1 (1317)', 'SLC31A2 (1318)', 'SLC32A1 (140679)', 'SLC33A1 (9197)', 'SLC34A1 (6569)', 'SLC34A2 (10568)', 'SLC34A3 (142680)', 'SLC35A1 (10559)', 'SLC35A2 (7355)', 'SLC35A3 (23443)', 'SLC35A4 (113829)', 'SLC35A5 (55032)', 'SLC35B1 (10237)', 'SLC35B2 (347734)', 'SLC35B3 (51000)', 'SLC35B4 (84912)', 'SLC35C1 (55343)', 'SLC35C2 (51006)', 'SLC35D1 (23169)', 'SLC35D2 (11046)', 'SLC35D3 (340146)', 'SLC35E1 (79939)', 'SLC35E2B (728661)', 'SLC35E3 (55508)', 'SLC35E4 (339665)', 'SLC35F1 (222553)', 'SLC35F2 (54733)', 'SLC35F3 (148641)', 'SLC35F5 (80255)', 'SLC35F6 (54978)', 'SLC35G1 (159371)', 'SLC35G2 (80723)', 'SLC35G3 (146861)', 'SLC35G5 (83650)', 'SLC35G6 (643664)', 'SLC36A1 (206358)', 'SLC36A2 (153201)', 'SLC36A3 (285641)', 'SLC36A4 (120103)', 'SLC37A1 (54020)', 'SLC37A2 (219855)', 'SLC37A3 (84255)', 'SLC38A1 (81539)', 'SLC38A10 (124565)', 'SLC38A11 (151258)', 'SLC38A2 (54407)', 'SLC38A4 (55089)', 'SLC38A5 (92745)', 'SLC38A6 (145389)', 'SLC38A7 (55238)', 'SLC38A8 (146167)', 'SLC38A9 (153129)', 'SLC39A1 (27173)', 'SLC39A10 (57181)', 'SLC39A11 (201266)', 'SLC39A12 (221074)', 'SLC39A13 (91252)', 'SLC39A14 (23516)', 'SLC39A2 (29986)', 'SLC39A3 (29985)', 'SLC39A4 (55630)', 'SLC39A5 (283375)', 'SLC39A6 (25800)', 'SLC39A7 (7922)', 'SLC39A8 (64116)', 'SLC39A9 (55334)', 'SLC3A1 (6519)', 'SLC3A2 (6520)', 'SLC40A1 (30061)', 'SLC41A1 (254428)', 'SLC41A2 (84102)', 'SLC41A3 (54946)', 'SLC43A1 (8501)', 'SLC43A2 (124935)', 'SLC43A3 (29015)', 'SLC44A1 (23446)', 'SLC44A2 (57153)', 'SLC44A3 (126969)', 'SLC44A4 (80736)', 'SLC44A5 (204962)', 'SLC45A1 (50651)', 'SLC45A2 (51151)', 'SLC45A3 (85414)', 'SLC45A4 (57210)', 'SLC46A2 (57864)', 'SLC46A3 (283537)', 'SLC47A1 (55244)', 'SLC47A2 (146802)', 'SLC48A1 (55652)', 'SLC49A3 (84179)', 'SLC49A4 (84925)', 'SLC4A1 (6521)', 'SLC4A10 (57282)', 'SLC4A11 (83959)', 'SLC4A1AP (22950)', 'SLC4A2 (6522)', 'SLC4A3 (6508)', 'SLC4A4 (8671)', 'SLC4A5 (57835)', 'SLC4A7 (9497)', 'SLC4A8 (9498)', 'SLC4A9 (83697)', 'SLC50A1 (55974)', 'SLC51A (200931)', 'SLC51B (123264)', 'SLC52A1 (55065)', 'SLC52A2 (79581)', 'SLC52A3 (113278)', 'SLC5A1 (6523)', 'SLC5A10 (125206)', 'SLC5A11 (115584)', 'SLC5A12 (159963)', 'SLC5A2 (6524)', 'SLC5A3 (6526)', 'SLC5A4 (6527)', 'SLC5A5 (6528)', 'SLC5A6 (8884)', 'SLC5A7 (60482)', 'SLC5A8 (160728)', 'SLC5A9 (200010)', 'SLC66A1 (54896)', 'SLC66A2 (80148)', 'SLC66A3 (130814)', 'SLC6A1 (6529)', 'SLC6A11 (6538)', 'SLC6A12 (6539)', 'SLC6A13 (6540)', 'SLC6A14 (11254)', 'SLC6A15 (55117)', 'SLC6A16 (28968)', 'SLC6A17 (388662)', 'SLC6A18 (348932)', 'SLC6A19 (340024)', 'SLC6A2 (6530)', 'SLC6A20 (54716)', 'SLC6A3 (6531)', 'SLC6A4 (6532)', 'SLC6A5 (9152)', 'SLC6A6 (6533)', 'SLC6A7 (6534)', 'SLC6A8 (6535)', 'SLC6A9 (6536)', 'SLC7A1 (6541)', 'SLC7A10 (56301)', 'SLC7A11 (23657)', 'SLC7A13 (157724)', 'SLC7A14 (57709)', 'SLC7A2 (6542)', 'SLC7A3 (84889)', 'SLC7A4 (6545)', 'SLC7A5 (8140)', 'SLC7A6 (9057)', 'SLC7A6OS (84138)', 'SLC7A7 (9056)', 'SLC7A8 (23428)', 'SLC7A9 (11136)', 'SLC8A1 (6546)', 'SLC8A2 (6543)', 'SLC8A3 (6547)', 'SLC8B1 (80024)', 'SLC9A1 (6548)', 'SLC9A2 (6549)', 'SLC9A3 (6550)', 'SLC9A4 (389015)', 'SLC9A5 (6553)', 'SLC9A6 (10479)', 'SLC9A7 (84679)', 'SLC9A8 (23315)', 'SLC9A9 (285195)', 'SLC9B1 (150159)', 'SLC9B2 (133308)', 'SLC9C1 (285335)', 'SLC9C2 (284525)', 'SLCO1A2 (6579)', 'SLCO1B1 (10599)', 'SLCO1B3 (28234)', 'SLCO1C1 (53919)', 'SLCO2A1 (6578)', 'SLCO2B1 (11309)', 'SLCO3A1 (28232)', 'SLCO4A1 (28231)', 'SLCO4C1 (353189)', 'SLCO5A1 (81796)', 'SLCO6A1 (133482)', 'SLF1 (84250)', 'SLF2 (55719)', 'SLFN11 (91607)', 'SLFN12 (55106)', 'SLFN12L (100506736)', 'SLFN13 (146857)', 'SLFN14 (342618)', 'SLFN5 (162394)', 'SLFNL1 (200172)', 'SLIRP (81892)', 'SLIT1 (6585)', 'SLIT2 (9353)', 'SLIT3 (6586)', 'SLITRK1 (114798)', 'SLITRK2 (84631)', 'SLITRK3 (22865)', 'SLITRK4 (139065)', 'SLITRK5 (26050)', 'SLITRK6 (84189)', 'SLK (9748)', 'SLMAP (7871)', 'SLN (6588)', 'SLPI (6590)', 'SLTM (79811)', 'SLU7 (10569)', 'SLURP1 (57152)', 'SLURP2 (432355)', 'SLX4 (84464)', 'SLX4IP (128710)', 'SLX9 (85395)', 'SMAD1 (4086)', 'SMAD2 (4087)', 'SMAD3 (4088)', 'SMAD4 (4089)', 'SMAD6 (4091)', 'SMAD7 (4092)', 'SMAD9 (4093)', 'SMAGP (57228)', 'SMAP1 (60682)', 'SMAP2 (64744)', 'SMARCA1 (6594)', 'SMARCA2 (6595)', 'SMARCA4 (6597)', 'SMARCA5 (8467)', 'SMARCAD1 (56916)', 'SMARCAL1 (50485)', 'SMARCB1 (6598)', 'SMARCC1 (6599)', 'SMARCC2 (6601)', 'SMARCD1 (6602)', 'SMARCD2 (6603)', 'SMARCD3 (6604)', 'SMARCE1 (6605)', 'SMC1A (8243)', 'SMC1B (27127)', 'SMC2 (10592)', 'SMC3 (9126)', 'SMC4 (10051)', 'SMC5 (23137)', 'SMC6 (79677)', 'SMCHD1 (23347)', 'SMCO1 (255798)', 'SMCO2 (341346)', 'SMCO3 (440087)', 'SMCO4 (56935)', 'SMCP (4184)', 'SMCR8 (140775)', 'SMDT1 (91689)', 'SMG1 (23049)', 'SMG5 (23381)', 'SMG6 (23293)', 'SMG7 (9887)', 'SMG8 (55181)', 'SMG9 (56006)', 'SMIM1 (388588)', 'SMIM10 (644538)', 'SMIM11 (54065)', 'SMIM12 (113444)', 'SMIM13 (221710)', 'SMIM14 (201895)', 'SMIM15 (643155)', 'SMIM17 (147670)', 'SMIM18 (100507341)', 'SMIM19 (114926)', 'SMIM2 (79024)', 'SMIM20 (389203)', 'SMIM21 (284274)', 'SMIM22 (440335)', 'SMIM24 (284422)', 'SMIM29 (221491)', 'SMIM3 (85027)', 'SMIM5 (643008)', 'SMIM6 (100130933)', 'SMIM7 (79086)', 'SMIM8 (57150)', 'SMIM9 (100132963)', 'SMKR1 (100287482)', 'SMLR1 (100507203)', 'SMNDC1 (10285)', 'SMO (6608)', 'SMOC1 (64093)', 'SMOC2 (64094)', 'SMOX (54498)', 'SMPD1 (6609)', 'SMPD2 (6610)', 'SMPD3 (55512)', 'SMPD4 (55627)', 'SMPDL3A (10924)', 'SMPDL3B (27293)', 'SMPX (23676)', 'SMR3A (26952)', 'SMR3B (10879)', 'SMS (6611)', 'SMTN (6525)', 'SMTNL2 (342527)', 'SMU1 (55234)', 'SMUG1 (23583)', 'SMURF1 (57154)', 'SMURF2 (64750)', 'SMYD1 (150572)', 'SMYD2 (56950)', 'SMYD3 (64754)', 'SMYD4 (114826)', 'SMYD5 (10322)', 'SNAI1 (6615)', 'SNAI2 (6591)', 'SNAI3 (333929)', 'SNAP23 (8773)', 'SNAP25 (6616)', 'SNAP29 (9342)', 'SNAP47 (116841)', 'SNAP91 (9892)', 'SNAPC1 (6617)', 'SNAPC2 (6618)', 'SNAPC3 (6619)', 'SNAPC4 (6621)', 'SNAPC5 (10302)', 'SNAPIN (23557)', 'SNCA (6622)', 'SNCAIP (9627)', 'SNCB (6620)', 'SNCG (6623)', 'SND1 (27044)', 'SNED1 (25992)', 'SNF8 (11267)', 'SNIP1 (79753)', 'SNN (8303)', 'SNORC (389084)', 'SNPH (9751)', 'SNRK (54861)', 'SNRNP200 (23020)', 'SNRNP25 (79622)', 'SNRNP27 (11017)', 'SNRNP35 (11066)', 'SNRNP40 (9410)', 'SNRNP48 (154007)', 'SNRNP70 (6625)', 'SNRPA (6626)', 'SNRPA1 (6627)', 'SNRPB (6628)', 'SNRPB2 (6629)', 'SNRPC (6631)', 'SNRPD1 (6632)', 'SNRPD2 (6633)', 'SNRPD3 (6634)', 'SNRPE (6635)', 'SNRPF (6636)', 'SNRPG (6637)', 'SNRPN (6638)', 'SNTA1 (6640)', 'SNTB1 (6641)', 'SNTB2 (6645)', 'SNTG1 (54212)', 'SNTG2 (54221)', 'SNTN (132203)', 'SNU13 (4809)', 'SNUPN (10073)', 'SNURF (8926)', 'SNW1 (22938)', 'SNX1 (6642)', 'SNX10 (29887)', 'SNX11 (29916)', 'SNX12 (29934)', 'SNX13 (23161)', 'SNX14 (57231)', 'SNX15 (29907)', 'SNX16 (64089)', 'SNX17 (9784)', 'SNX18 (112574)', 'SNX19 (399979)', 'SNX2 (6643)', 'SNX20 (124460)', 'SNX21 (90203)', 'SNX22 (79856)', 'SNX24 (28966)', 'SNX25 (83891)', 'SNX27 (81609)', 'SNX29 (92017)', 'SNX3 (8724)', 'SNX30 (401548)', 'SNX31 (169166)', 'SNX32 (254122)', 'SNX33 (257364)', 'SNX4 (8723)', 'SNX5 (27131)', 'SNX6 (58533)', 'SNX7 (51375)', 'SNX8 (29886)', 'SNX9 (51429)', 'SOAT1 (6646)', 'SOAT2 (8435)', 'SOBP (55084)', 'SOCS1 (8651)', 'SOCS2 (8835)', 'SOCS3 (9021)', 'SOCS4 (122809)', 'SOCS5 (9655)', 'SOCS6 (9306)', 'SOCS7 (30837)', 'SOD1 (6647)', 'SOD2 (6648)', 'SOD3 (6649)', 'SOHLH1 (402381)', 'SOHLH2 (54937)', 'SON (6651)', 'SORBS1 (10580)', 'SORBS2 (8470)', 'SORBS3 (10174)', 'SORCS1 (114815)', 'SORCS2 (57537)', 'SORCS3 (22986)', 'SORD (6652)', 'SORL1 (6653)', 'SORT1 (6272)', 'SOS1 (6654)', 'SOS2 (6655)', 'SOST (50964)', 'SOSTDC1 (25928)', 'SOWAHA (134548)', 'SOWAHB (345079)', 'SOWAHC (65124)', 'SOWAHD (347454)', 'SOX1 (6656)', 'SOX10 (6663)', 'SOX11 (6664)', 'SOX12 (6666)', 'SOX13 (9580)', 'SOX14 (8403)', 'SOX15 (6665)', 'SOX17 (64321)', 'SOX18 (54345)', 'SOX2 (6657)', 'SOX21 (11166)', 'SOX3 (6658)', 'SOX30 (11063)', 'SOX4 (6659)', 'SOX5 (6660)', 'SOX6 (55553)', 'SOX7 (83595)', 'SOX8 (30812)', 'SOX9 (6662)', 'SP1 (6667)', 'SP100 (6672)', 'SP110 (3431)', 'SP140 (11262)', 'SP140L (93349)', 'SP2 (6668)', 'SP3 (6670)', 'SP4 (6671)', 'SP5 (389058)', 'SP6 (80320)', 'SP7 (121340)', 'SP8 (221833)', 'SP9 (100131390)', 'SPA17 (53340)', 'SPACA1 (81833)', 'SPACA3 (124912)', 'SPACA4 (171169)', 'SPACA7 (122258)', 'SPACA9 (11092)', 'SPACDR (402573)', 'SPAG1 (6674)', 'SPAG16 (79582)', 'SPAG17 (200162)', 'SPAG4 (6676)', 'SPAG5 (10615)', 'SPAG6 (9576)', 'SPAG7 (9552)', 'SPAG8 (26206)', 'SPAG9 (9043)', 'SPAM1 (6677)', 'SPANXB1 (728695)', 'SPANXN1 (494118)', 'SPANXN2 (494119)', 'SPANXN3 (139067)', 'SPANXN4 (441525)', 'SPANXN5 (494197)', 'SPARC (6678)', 'SPARCL1 (8404)', 'SPART (23111)', 'SPAST (6683)', 'SPATA12 (353324)', 'SPATA13 (221178)', 'SPATA16 (83893)', 'SPATA17 (128153)', 'SPATA18 (132671)', 'SPATA19 (219938)', 'SPATA2 (9825)', 'SPATA20 (64847)', 'SPATA21 (374955)', 'SPATA22 (84690)', 'SPATA24 (202051)', 'SPATA25 (128497)', 'SPATA2L (124044)', 'SPATA3 (130560)', 'SPATA31A1 (647060)', 'SPATA31A3 (727830)', 'SPATA31A6 (389730)', 'SPATA31A7 (26165)', 'SPATA31D1 (389763)', 'SPATA31E1 (286234)', 'SPATA31F1 (259308)', 'SPATA31G1 (138724)', 'SPATA32 (124783)', 'SPATA33 (124045)', 'SPATA4 (132851)', 'SPATA45 (149643)', 'SPATA46 (284680)', 'SPATA6 (54558)', 'SPATA6L (55064)', 'SPATA7 (55812)', 'SPATA9 (83890)', 'SPATC1 (375686)', 'SPATC1L (84221)', 'SPATS1 (221409)', 'SPATS2 (65244)', 'SPATS2L (26010)', 'SPC24 (147841)', 'SPC25 (57405)', 'SPCS1 (28972)', 'SPCS2 (9789)', 'SPCS3 (60559)', 'SPDEF (25803)', 'SPDL1 (54908)', 'SPDYA (245711)', 'SPDYE1 (285955)', 'SPDYE4 (388333)', 'SPECC1 (92521)', 'SPECC1L (23384)', 'SPEF1 (25876)', 'SPEF2 (79925)', 'SPEG (10290)', 'SPEM1 (374768)', 'SPEM2 (201243)', 'SPEN (23013)', 'SPESP1 (246777)', 'SPG11 (80208)', 'SPG21 (51324)', 'SPG7 (6687)', 'SPHK1 (8877)', 'SPHK2 (56848)', 'SPHKAP (80309)', 'SPI1 (6688)', 'SPIB (6689)', 'SPIC (121599)', 'SPICE1 (152185)', 'SPIDR (23514)', 'SPIN1 (10927)', 'SPIN3 (169981)', 'SPIN4 (139886)', 'SPINDOC (144097)', 'SPINK1 (6690)', 'SPINK13 (153218)', 'SPINK14 (408187)', 'SPINK2 (6691)', 'SPINK4 (27290)', 'SPINK5 (11005)', 'SPINK6 (404203)', 'SPINK7 (84651)', 'SPINK8 (646424)', 'SPINK9 (643394)', 'SPINT1 (6692)', 'SPINT2 (10653)', 'SPINT3 (10816)', 'SPINT4 (391253)', 'SPIRE1 (56907)', 'SPIRE2 (84501)', 'SPMAP1 (388381)', 'SPMAP2 (51298)', 'SPMAP2L (100506564)', 'SPMIP10 (389320)', 'SPMIP2 (152940)', 'SPMIP3 (200159)', 'SPMIP4 (136895)', 'SPMIP5 (143379)', 'SPMIP6 (84688)', 'SPMIP7 (100130988)', 'SPMIP8 (374739)', 'SPMIP9 (200523)', 'SPN (6693)', 'SPNS1 (83985)', 'SPNS2 (124976)', 'SPNS3 (201305)', 'SPO11 (23626)', 'SPOCD1 (90853)', 'SPOCK1 (6695)', 'SPOCK2 (9806)', 'SPOCK3 (50859)', 'SPOP (8405)', 'SPOPL (339745)', 'SPOUT1 (51490)', 'SPP1 (6696)', 'SPP2 (6694)', 'SPPL2A (84888)', 'SPPL2C (162540)', 'SPPL3 (121665)', 'SPR (6697)', 'SPRED1 (161742)', 'SPRED2 (200734)', 'SPRED3 (399473)', 'SPRING1 (79794)', 'SPRN (503542)', 'SPRR1A (6698)', 'SPRR1B (6699)', 'SPRR2F (6705)', 'SPRR2G (6706)', 'SPRR3 (6707)', 'SPRR4 (163778)', 'SPRTN (83932)', 'SPRY1 (10252)', 'SPRY2 (10253)', 'SPRY3 (10251)', 'SPRY4 (81848)', 'SPRYD3 (84926)', 'SPRYD4 (283377)', 'SPRYD7 (57213)', 'SPSB1 (80176)', 'SPSB2 (84727)', 'SPSB3 (90864)', 'SPSB4 (92369)', 'SPTA1 (6708)', 'SPTAN1 (6709)', 'SPTB (6710)', 'SPTBN1 (6711)', 'SPTBN2 (6712)', 'SPTBN4 (57731)', 'SPTBN5 (51332)', 'SPTLC1 (10558)', 'SPTLC2 (9517)', 'SPTLC3 (55304)', 'SPTSSA (171546)', 'SPTSSB (165679)', 'SPTY2D1 (144108)', 'SPX (80763)', 'SPZ1 (84654)', 'SQLE (6713)', 'SQOR (58472)', 'SQSTM1 (8878)', 'SRA1 (10011)', 'SRARP (149563)', 'SRBD1 (55133)', 'SRC (6714)', 'SRCAP (10847)', 'SRCIN1 (80725)', 'SRD5A1 (6715)', 'SRD5A3 (79644)', 'SREBF1 (6720)', 'SREBF2 (6721)', 'SREK1 (140890)', 'SREK1IP1 (285672)', 'SRF (6722)', 'SRFBP1 (153443)', 'SRGAP1 (57522)', 'SRGAP3 (9901)', 'SRGN (5552)', 'SRI (6717)', 'SRL (6345)', 'SRM (6723)', 'SRMS (6725)', 'SRP14 (6727)', 'SRP19 (6728)', 'SRP54 (6729)', 'SRP68 (6730)', 'SRP72 (6731)', 'SRP9 (6726)', 'SRPK1 (6732)', 'SRPK2 (6733)', 'SRPK3 (26576)', 'SRPRA (6734)', 'SRPRB (58477)', 'SRPX (8406)', 'SRPX2 (27286)', 'SRR (63826)', 'SRRD (402055)', 'SRRM1 (10250)', 'SRRM2 (23524)', 'SRRM4 (84530)', 'SRRM5 (100170229)', 'SRRT (51593)', 'SRSF1 (6426)', 'SRSF10 (10772)', 'SRSF11 (9295)', 'SRSF12 (135295)', 'SRSF2 (6427)', 'SRSF3 (6428)', 'SRSF4 (6429)', 'SRSF5 (6430)', 'SRSF6 (6431)', 'SRSF7 (6432)', 'SRSF9 (8683)', 'SRXN1 (140809)', 'SRY (6736)', 'SS18 (6760)', 'SS18L1 (26039)', 'SS18L2 (51188)', 'SSB (6741)', 'SSBP1 (6742)', 'SSBP2 (23635)', 'SSBP3 (23648)', 'SSBP4 (170463)', 'SSC4D (136853)', 'SSC5D (284297)', 'SSH1 (54434)', 'SSH2 (85464)', 'SSH3 (54961)', 'SSMEM1 (136263)', 'SSNA1 (8636)', 'SSPN (8082)', 'SSR1 (6745)', 'SSR2 (6746)', 'SSR3 (6747)', 'SSR4 (6748)', 'SSRP1 (6749)', 'SST (6750)', 'SSTR1 (6751)', 'SSTR2 (6752)', 'SSTR3 (6753)', 'SSTR4 (6754)', 'SSTR5 (6755)', 'SSU72 (29101)', 'SSUH2 (51066)', 'SSX1 (6756)', 'SSX2IP (117178)', 'SSX3 (10214)', 'SSX5 (6758)', 'SSX7 (280658)', 'ST13 (6767)', 'ST14 (6768)', 'ST18 (9705)', 'ST3GAL1 (6482)', 'ST3GAL2 (6483)', 'ST3GAL3 (6487)', 'ST3GAL4 (6484)', 'ST3GAL5 (8869)', 'ST3GAL6 (10402)', 'ST6GAL1 (6480)', 'ST6GAL2 (84620)', 'ST6GALNAC1 (55808)', 'ST6GALNAC2 (10610)', 'ST6GALNAC3 (256435)', 'ST6GALNAC4 (27090)', 'ST6GALNAC5 (81849)', 'ST6GALNAC6 (30815)', 'ST7 (7982)', 'ST7L (54879)', 'ST8SIA1 (6489)', 'ST8SIA2 (8128)', 'ST8SIA3 (51046)', 'ST8SIA4 (7903)', 'ST8SIA5 (29906)', 'ST8SIA6 (338596)', 'STAB1 (23166)', 'STAB2 (55576)', 'STAC (6769)', 'STAC2 (342667)', 'STAC3 (246329)', 'STAG1 (10274)', 'STAG2 (10735)', 'STAG3 (10734)', 'STAM (8027)', 'STAM2 (10254)', 'STAMBP (10617)', 'STAMBPL1 (57559)', 'STAP1 (26228)', 'STAP2 (55620)', 'STAR (6770)', 'STARD10 (10809)', 'STARD13 (90627)', 'STARD3 (10948)', 'STARD3NL (83930)', 'STARD4 (134429)', 'STARD5 (80765)', 'STARD6 (147323)', 'STARD7 (56910)', 'STARD8 (9754)', 'STARD9 (57519)', 'STAT1 (6772)', 'STAT2 (6773)', 'STAT3 (6774)', 'STAT4 (6775)', 'STAT5A (6776)', 'STAT5B (6777)', 'STAT6 (6778)', 'STATH (6779)', 'STAU1 (6780)', 'STAU2 (27067)', 'STBD1 (8987)', 'STC1 (6781)', 'STC2 (8614)', 'STEAP1 (26872)', 'STEAP1B (256227)', 'STEAP2 (261729)', 'STEAP3 (55240)', 'STEAP4 (79689)', 'STEEP1 (63932)', 'STH (246744)', 'STIL (6491)', 'STIM1 (6786)', 'STIM2 (57620)', 'STIMATE (375346)', 'STING1 (340061)', 'STIP1 (10963)', 'STK10 (6793)', 'STK11 (6794)', 'STK11IP (114790)', 'STK16 (8576)', 'STK17A (9263)', 'STK17B (9262)', 'STK19 (8859)', 'STK24 (8428)', 'STK25 (10494)', 'STK26 (51765)', 'STK3 (6788)', 'STK31 (56164)', 'STK32A (202374)', 'STK32B (55351)', 'STK32C (282974)', 'STK33 (65975)', 'STK35 (140901)', 'STK36 (27148)', 'STK38 (11329)', 'STK38L (23012)', 'STK39 (27347)', 'STK4 (6789)', 'STK40 (83931)', 'STKLD1 (169436)', 'STMN1 (3925)', 'STMN2 (11075)', 'STMN3 (50861)', 'STMN4 (81551)', 'STMND1 (401236)', 'STMP1 (647087)', 'STN1 (79991)', 'STOM (2040)', 'STOML1 (9399)', 'STOML2 (30968)', 'STOML3 (161003)', 'STON1 (11037)', 'STON2 (85439)', 'STOX1 (219736)', 'STOX2 (56977)', 'STPG1 (90529)', 'STPG2 (285555)', 'STPG3 (441476)', 'STPG4 (285051)', 'STRA6 (64220)', 'STRA8 (346673)', 'STRADA (92335)', 'STRADB (55437)', 'STRAP (11171)', 'STRBP (55342)', 'STRC (161497)', 'STRIP1 (85369)', 'STRIP2 (57464)', 'STRN (6801)', 'STRN3 (29966)', 'STRN4 (29888)', 'STS (412)', 'STT3A (3703)', 'STT3B (201595)', 'STUB1 (10273)', 'STUM (375057)', 'STX10 (8677)', 'STX11 (8676)', 'STX12 (23673)', 'STX16 (8675)', 'STX17 (55014)', 'STX18 (53407)', 'STX19 (415117)', 'STX1A (6804)', 'STX1B (112755)', 'STX2 (2054)', 'STX3 (6809)', 'STX4 (6810)', 'STX5 (6811)', 'STX6 (10228)', 'STX7 (8417)', 'STX8 (9482)', 'STXBP1 (6812)', 'STXBP2 (6813)', 'STXBP3 (6814)', 'STXBP4 (252983)', 'STXBP5 (134957)', 'STXBP5L (9515)', 'STXBP6 (29091)', 'STYK1 (55359)', 'STYX (6815)', 'STYXL1 (51657)', 'STYXL2 (92235)', 'SUB1 (10923)', 'SUCLA2 (8803)', 'SUCLG1 (8802)', 'SUCLG2 (8801)', 'SUCNR1 (56670)', 'SUCO (51430)', 'SUDS3 (64426)', 'SUFU (51684)', 'SUGCT (79783)', 'SUGP1 (57794)', 'SUGP2 (10147)', 'SUGT1 (10910)', 'SULF1 (23213)', 'SULF2 (55959)', 'SULT1A1 (6817)', 'SULT1A2 (6799)', 'SULT1B1 (27284)', 'SULT1C2 (6819)', 'SULT1C3 (442038)', 'SULT1C4 (27233)', 'SULT1E1 (6783)', 'SULT2A1 (6822)', 'SULT2B1 (6820)', 'SULT4A1 (25830)', 'SULT6B1 (391365)', 'SUMF1 (285362)', 'SUMF2 (25870)', 'SUMO1 (7341)', 'SUMO3 (6612)', 'SUMO4 (387082)', 'SUN1 (23353)', 'SUN2 (25777)', 'SUN3 (256979)', 'SUN5 (140732)', 'SUOX (6821)', 'SUPT16H (11198)', 'SUPT20H (55578)', 'SUPT3H (8464)', 'SUPT4H1 (6827)', 'SUPT5H (6829)', 'SUPT6H (6830)', 'SUPT7L (9913)', 'SUPV3L1 (6832)', 'SURF1 (6834)', 'SURF2 (6835)', 'SURF4 (6836)', 'SURF6 (6838)', 'SUSD1 (64420)', 'SUSD2 (56241)', 'SUSD3 (203328)', 'SUSD4 (55061)', 'SUSD5 (26032)', 'SUSD6 (9766)', 'SUV39H1 (6839)', 'SUV39H2 (79723)', 'SUZ12 (23512)', 'SV2A (9900)', 'SV2B (9899)', 'SV2C (22987)', 'SVBP (374969)', 'SVEP1 (79987)', 'SVIL (6840)', 'SVIP (258010)', 'SVOPL (136306)', 'SWAP70 (23075)', 'SWI5 (375757)', 'SWSAP1 (126074)', 'SWT1 (54823)', 'SYAP1 (94056)', 'SYBU (55638)', 'SYCE1 (93426)', 'SYCE1L (100130958)', 'SYCE2 (256126)', 'SYCE3 (644186)', 'SYCN (342898)', 'SYCP1 (6847)', 'SYCP2 (10388)', 'SYCP2L (221711)', 'SYCP3 (50511)', 'SYDE1 (85360)', 'SYDE2 (84144)', 'SYF2 (25949)', 'SYK (6850)', 'SYMPK (8189)', 'SYN1 (6853)', 'SYN3 (8224)', 'SYNC (81493)', 'SYNCRIP (10492)', 'SYNDIG1 (79953)', 'SYNDIG1L (646658)', 'SYNE1 (23345)', 'SYNE2 (23224)', 'SYNE3 (161176)', 'SYNE4 (163183)', 'SYNGAP1 (8831)', 'SYNGR1 (9145)', 'SYNGR2 (9144)', 'SYNGR3 (9143)', 'SYNGR4 (23546)', 'SYNJ1 (8867)', 'SYNJ2 (8871)', 'SYNJ2BP (55333)', 'SYNPO (11346)', 'SYNPO2 (171024)', 'SYNPO2L (79933)', 'SYNPR (132204)', 'SYNRG (11276)', 'SYP (6855)', 'SYPL1 (6856)', 'SYPL2 (284612)', 'SYS1 (90196)', 'SYT1 (6857)', 'SYT10 (341359)', 'SYT11 (23208)', 'SYT12 (91683)', 'SYT13 (57586)', 'SYT14 (255928)', 'SYT16 (83851)', 'SYT17 (51760)', 'SYT2 (127833)', 'SYT3 (84258)', 'SYT4 (6860)', 'SYT5 (6861)', 'SYT6 (148281)', 'SYT7 (9066)', 'SYT8 (90019)', 'SYT9 (143425)', 'SYTL1 (84958)', 'SYTL2 (54843)', 'SYTL3 (94120)', 'SYTL4 (94121)', 'SYTL5 (94122)', 'SYVN1 (84447)', 'SZRD1 (26099)', 'SZT2 (23334)', 'TAAR1 (134864)', 'TAAR2 (9287)', 'TAAR5 (9038)', 'TAAR6 (319100)', 'TAAR8 (83551)', 'TAB1 (10454)', 'TAB2 (23118)', 'TAB3 (257397)', 'TAC1 (6863)', 'TAC3 (6866)', 'TAC4 (255061)', 'TACC1 (6867)', 'TACC2 (10579)', 'TACC3 (10460)', 'TACO1 (51204)', 'TACR1 (6869)', 'TACR2 (6865)', 'TACR3 (6870)', 'TACSTD2 (4070)', 'TADA1 (117143)', 'TADA2A (6871)', 'TADA2B (93624)', 'TADA3 (10474)', 'TAF1 (6872)', 'TAF10 (6881)', 'TAF11 (6882)', 'TAF12 (6883)', 'TAF13 (6884)', 'TAF15 (8148)', 'TAF1A (9015)', 'TAF1B (9014)', 'TAF1C (9013)', 'TAF1D (79101)', 'TAF1L (138474)', 'TAF2 (6873)', 'TAF3 (83860)', 'TAF4 (6874)', 'TAF4B (6875)', 'TAF5 (6877)', 'TAF5L (27097)', 'TAF6 (6878)', 'TAF6L (10629)', 'TAF7 (6879)', 'TAF7L (54457)', 'TAF8 (129685)', 'TAF9 (6880)', 'TAF9B (51616)', 'TAFA1 (407738)', 'TAFA2 (338811)', 'TAFA3 (284467)', 'TAFA4 (151647)', 'TAFA5 (25817)', 'TAFAZZIN (6901)', 'TAGAP (117289)', 'TAGLN (6876)', 'TAGLN2 (8407)', 'TAGLN3 (29114)', 'TAL1 (6886)', 'TAL2 (6887)', 'TALDO1 (6888)', 'TAMALIN (160622)', 'TAMM41 (132001)', 'TANC1 (85461)', 'TANC2 (26115)', 'TANGO2 (128989)', 'TANGO6 (79613)', 'TANK (10010)', 'TAOK1 (57551)', 'TAOK2 (9344)', 'TAOK3 (51347)', 'TAP1 (6890)', 'TAP2 (6891)', 'TAPBP (6892)', 'TAPBPL (55080)', 'TAPT1 (202018)', 'TARBP1 (6894)', 'TARBP2 (6895)', 'TARDBP (23435)', 'TARM1 (441864)', 'TARS1 (6897)', 'TARS2 (80222)', 'TARS3 (123283)', 'TAS1R1 (80835)', 'TAS1R2 (80834)', 'TAS1R3 (83756)', 'TAS2R1 (50834)', 'TAS2R10 (50839)', 'TAS2R13 (50838)', 'TAS2R14 (50840)', 'TAS2R16 (50833)', 'TAS2R19 (259294)', 'TAS2R20 (259295)', 'TAS2R3 (50831)', 'TAS2R30 (259293)', 'TAS2R31 (259290)', 'TAS2R38 (5726)', 'TAS2R39 (259285)', 'TAS2R4 (50832)', 'TAS2R40 (259286)', 'TAS2R41 (259287)', 'TAS2R42 (353164)', 'TAS2R43 (259289)', 'TAS2R46 (259292)', 'TAS2R5 (54429)', 'TAS2R50 (259296)', 'TAS2R60 (338398)', 'TAS2R7 (50837)', 'TAS2R8 (50836)', 'TAS2R9 (50835)', 'TASL (80231)', 'TASOR (23272)', 'TASOR2 (54906)', 'TASP1 (55617)', 'TAT (6898)', 'TATDN1 (83940)', 'TATDN2 (9797)', 'TATDN3 (128387)', 'TAX1BP1 (8887)', 'TAX1BP3 (30851)', 'TBATA (219793)', 'TBC1D1 (23216)', 'TBC1D10A (83874)', 'TBC1D10B (26000)', 'TBC1D10C (374403)', 'TBC1D12 (23232)', 'TBC1D13 (54662)', 'TBC1D14 (57533)', 'TBC1D15 (64786)', 'TBC1D16 (125058)', 'TBC1D17 (79735)', 'TBC1D19 (55296)', 'TBC1D2 (55357)', 'TBC1D20 (128637)', 'TBC1D21 (161514)', 'TBC1D22A (25771)', 'TBC1D22B (55633)', 'TBC1D23 (55773)', 'TBC1D24 (57465)', 'TBC1D25 (4943)', 'TBC1D26 (353149)', 'TBC1D28 (254272)', 'TBC1D2B (23102)', 'TBC1D30 (23329)', 'TBC1D31 (93594)', 'TBC1D32 (221322)', 'TBC1D4 (9882)', 'TBC1D5 (9779)', 'TBC1D7 (51256)', 'TBC1D8 (11138)', 'TBC1D8B (54885)', 'TBC1D9 (23158)', 'TBC1D9B (23061)', 'TBCA (6902)', 'TBCB (1155)', 'TBCC (6903)', 'TBCCD1 (55171)', 'TBCD (6904)', 'TBCE (6905)', 'TBCEL (219899)', 'TBCK (93627)', 'TBK1 (29110)', 'TBKBP1 (9755)', 'TBL1X (6907)', 'TBL1XR1 (79718)', 'TBL1Y (90665)', 'TBL2 (26608)', 'TBL3 (10607)', 'TBP (6908)', 'TBPL1 (9519)', 'TBR1 (10716)', 'TBRG1 (84897)', 'TBRG4 (9238)', 'TBX1 (6899)', 'TBX10 (347853)', 'TBX15 (6913)', 'TBX18 (9096)', 'TBX19 (9095)', 'TBX2 (6909)', 'TBX20 (57057)', 'TBX21 (30009)', 'TBX22 (50945)', 'TBX3 (6926)', 'TBX4 (9496)', 'TBX5 (6910)', 'TBX6 (6911)', 'TBXA2R (6915)', 'TBXAS1 (6916)', 'TBXT (6862)', 'TC2N (123036)', 'TCAF1 (9747)', 'TCAIM (285343)', 'TCAP (8557)', 'TCEA1 (6917)', 'TCEA2 (6919)', 'TCEA3 (6920)', 'TCEAL1 (9338)', 'TCEAL2 (140597)', 'TCEAL3 (85012)', 'TCEAL4 (79921)', 'TCEAL5 (340543)', 'TCEAL7 (56849)', 'TCEAL8 (90843)', 'TCEAL9 (51186)', 'TCEANC (170082)', 'TCEANC2 (127428)', 'TCERG1 (10915)', 'TCERG1L (256536)', 'TCF12 (6938)', 'TCF15 (6939)', 'TCF19 (6941)', 'TCF20 (6942)', 'TCF21 (6943)', 'TCF23 (150921)', 'TCF24 (100129654)', 'TCF25 (22980)', 'TCF3 (6929)', 'TCF4 (6925)', 'TCF7 (6932)', 'TCF7L1 (83439)', 'TCF7L2 (6934)', 'TCFL5 (10732)', 'TCHH (7062)', 'TCHHL1 (126637)', 'TCHP (84260)', 'TCIM (56892)', 'TCIRG1 (10312)', 'TCL1A (8115)', 'TCL1B (9623)', 'TCN1 (6947)', 'TCN2 (6948)', 'TCOF1 (6949)', 'TCP1 (6950)', 'TCP10L (140290)', 'TCP11 (6954)', 'TCP11L1 (55346)', 'TCP11L2 (255394)', 'TCTA (6988)', 'TCTE1 (202500)', 'TCTN1 (79600)', 'TCTN2 (79867)', 'TCTN3 (26123)', 'TDG (6996)', 'TDO2 (6999)', 'TDP1 (55775)', 'TDP2 (51567)', 'TDRD1 (56165)', 'TDRD10 (126668)', 'TDRD12 (91646)', 'TDRD3 (81550)', 'TDRD5 (163589)', 'TDRD6 (221400)', 'TDRD7 (23424)', 'TDRD9 (122402)', 'TDRKH (11022)', 'TDRP (157695)', 'TEAD1 (7003)', 'TEAD2 (8463)', 'TEAD3 (7005)', 'TEAD4 (7004)', 'TEC (7006)', 'TECPR1 (25851)', 'TECPR2 (9895)', 'TECR (9524)', 'TECRL (253017)', 'TECTA (7007)', 'TECTB (6975)', 'TEDC1 (283643)', 'TEDC2 (80178)', 'TEDDM1 (127670)', 'TEF (7008)', 'TEFM (79736)', 'TEK (7010)', 'TEKT1 (83659)', 'TEKT2 (27285)', 'TEKT3 (64518)', 'TEKT4 (150483)', 'TEKT5 (146279)', 'TEKTIP1 (100128569)', 'TEKTL1 (126402)', 'TELO2 (9894)', 'TEN1 (100134934)', 'TENM1 (10178)', 'TENM3 (55714)', 'TENM4 (26011)', 'TENT2 (167153)', 'TENT4A (11044)', 'TENT4B (64282)', 'TENT5A (55603)', 'TENT5B (115572)', 'TENT5C (54855)', 'TENT5D (169966)', 'TEP1 (7011)', 'TEPSIN (146705)', 'TERB2 (145645)', 'TERF2 (7014)', 'TERF2IP (54386)', 'TERT (7015)', 'TES (26136)', 'TESC (54997)', 'TESK1 (7016)', 'TESK2 (10420)', 'TESMIN (9633)', 'TESPA1 (9840)', 'TET1 (80312)', 'TET2 (54790)', 'TET3 (200424)', 'TEX10 (54881)', 'TEX101 (83639)', 'TEX11 (56159)', 'TEX12 (56158)', 'TEX13B (56156)', 'TEX14 (56155)', 'TEX15 (56154)', 'TEX19 (400629)', 'TEX2 (55852)', 'TEX22 (647310)', 'TEX26 (122046)', 'TEX261 (113419)', 'TEX264 (51368)', 'TEX29 (121793)', 'TEX30 (93081)', 'TEX35 (84066)', 'TEX36 (387718)', 'TEX38 (374973)', 'TEX44 (165100)', 'TEX46 (729059)', 'TEX47 (219557)', 'TEX55 (152405)', 'TEX9 (374618)', 'TF (7018)', 'TFAM (7019)', 'TFAP2A (7020)', 'TFAP2B (7021)', 'TFAP2C (7022)', 'TFAP2D (83741)', 'TFAP2E (339488)', 'TFAP4 (7023)', 'TFB1M (51106)', 'TFB2M (64216)', 'TFCP2 (7024)', 'TFCP2L1 (29842)', 'TFDP1 (7027)', 'TFDP2 (7029)', 'TFDP3 (51270)', 'TFE3 (7030)', 'TFEB (7942)', 'TFEC (22797)', 'TFF1 (7031)', 'TFF2 (7032)', 'TFF3 (7033)', 'TFG (10342)', 'TFIP11 (24144)', 'TFPI (7035)', 'TFPI2 (7980)', 'TFPT (29844)', 'TFR2 (7036)', 'TFRC (7037)', 'TG (7038)', 'TGDS (23483)', 'TGFA (7039)', 'TGFB1 (7040)', 'TGFB1I1 (7041)', 'TGFB2 (7042)', 'TGFB3 (7043)', 'TGFBI (7045)', 'TGFBR1 (7046)', 'TGFBR2 (7048)', 'TGFBR3 (7049)', 'TGFBR3L (100507588)', 'TGFBRAP1 (9392)', 'TGIF1 (7050)', 'TGIF2 (60436)', 'TGIF2LX (90316)', 'TGM1 (7051)', 'TGM2 (7052)', 'TGM3 (7053)', 'TGM4 (7047)', 'TGM5 (9333)', 'TGM6 (343641)', 'TGM7 (116179)', 'TGOLN2 (10618)', 'TGS1 (96764)', 'TH (7054)', 'THADA (63892)', 'THAP1 (55145)', 'THAP10 (56906)', 'THAP11 (57215)', 'THAP12 (5612)', 'THAP2 (83591)', 'THAP3 (90326)', 'THAP4 (51078)', 'THAP5 (168451)', 'THAP6 (152815)', 'THAP7 (80764)', 'THAP8 (199745)', 'THAP9 (79725)', 'THBD (7056)', 'THBS1 (7057)', 'THBS2 (7058)', 'THBS3 (7059)', 'THBS4 (7060)', 'THEM4 (117145)', 'THEM5 (284486)', 'THEM6 (51337)', 'THEMIS (387357)', 'THEMIS2 (9473)', 'THG1L (54974)', 'THNSL1 (79896)', 'THNSL2 (55258)', 'THOC1 (9984)', 'THOC2 (57187)', 'THOC3 (84321)', 'THOC5 (8563)', 'THOC6 (79228)', 'THOC7 (80145)', 'THOP1 (7064)', 'THPO (7066)', 'THRA (7067)', 'THRAP3 (9967)', 'THRB (7068)', 'THRSP (7069)', 'THSD1 (55901)', 'THSD4 (79875)', 'THSD7A (221981)', 'THTPA (79178)', 'THUMPD1 (55623)', 'THUMPD2 (80745)', 'THUMPD3 (25917)', 'THY1 (7070)', 'THYN1 (29087)', 'TIA1 (7072)', 'TIAL1 (7073)', 'TIAM1 (7074)', 'TIAM2 (26230)', 'TICAM1 (148022)', 'TICAM2 (353376)', 'TICRR (90381)', 'TIE1 (7075)', 'TIFA (92610)', 'TIFAB (497189)', 'TIGAR (57103)', 'TIGD1 (200765)', 'TIGD2 (166815)', 'TIGD3 (220359)', 'TIGD4 (201798)', 'TIGD5 (84948)', 'TIGD6 (81789)', 'TIGD7 (91151)', 'TIGIT (201633)', 'TIMD4 (91937)', 'TIMELESS (8914)', 'TIMM10 (26519)', 'TIMM10B (26515)', 'TIMM13 (26517)', 'TIMM17A (10440)', 'TIMM17B (10245)', 'TIMM21 (29090)', 'TIMM22 (29928)', 'TIMM23 (100287932)', 'TIMM29 (90580)', 'TIMM44 (10469)', 'TIMM50 (92609)', 'TIMM8A (1678)', 'TIMM8B (26521)', 'TIMM9 (26520)', 'TIMMDC1 (51300)', 'TIMP1 (7076)', 'TIMP2 (7077)', 'TIMP3 (7078)', 'TIMP4 (7079)', 'TINAG (27283)', 'TINAGL1 (64129)', 'TINF2 (26277)', 'TIPARP (25976)', 'TIPIN (54962)', 'TIPRL (261726)', 'TIRAP (114609)', 'TJAP1 (93643)', 'TJP1 (7082)', 'TJP2 (9414)', 'TJP3 (27134)', 'TK1 (7083)', 'TK2 (7084)', 'TKFC (26007)', 'TKT (7086)', 'TKTL1 (8277)', 'TKTL2 (84076)', 'TLCD1 (116238)', 'TLCD2 (727910)', 'TLCD3A (79850)', 'TLCD3B (83723)', 'TLCD4 (148534)', 'TLCD5 (219902)', 'TLDC2 (140711)', 'TLE1 (7088)', 'TLE2 (7089)', 'TLE3 (7090)', 'TLE4 (7091)', 'TLE5 (166)', 'TLE6 (79816)', 'TLK1 (9874)', 'TLK2 (11011)', 'TLL1 (7092)', 'TLL2 (7093)', 'TLN1 (7094)', 'TLN2 (83660)', 'TLNRD1 (59274)', 'TLR1 (7096)', 'TLR10 (81793)', 'TLR2 (7097)', 'TLR3 (7098)', 'TLR4 (7099)', 'TLR5 (7100)', 'TLR6 (10333)', 'TLR7 (51284)', 'TLR8 (51311)', 'TLR9 (54106)', 'TLX1 (3195)', 'TLX2 (3196)', 'TLX3 (30012)', 'TM2D1 (83941)', 'TM2D2 (83877)', 'TM2D3 (80213)', 'TM4SF1 (4071)', 'TM4SF18 (116441)', 'TM4SF19 (116211)', 'TM4SF20 (79853)', 'TM4SF4 (7104)', 'TM4SF5 (9032)', 'TM6SF1 (53346)', 'TM6SF2 (53345)', 'TM7SF2 (7108)', 'TM7SF3 (51768)', 'TM9SF1 (10548)', 'TM9SF2 (9375)', 'TM9SF3 (56889)', 'TM9SF4 (9777)', 'TMA16 (55319)', 'TMA7 (51372)', 'TMBIM1 (64114)', 'TMBIM4 (51643)', 'TMBIM6 (7009)', 'TMC1 (117531)', 'TMC2 (117532)', 'TMC3 (342125)', 'TMC4 (147798)', 'TMC5 (79838)', 'TMC6 (11322)', 'TMC7 (79905)', 'TMC8 (147138)', 'TMCC1 (23023)', 'TMCC2 (9911)', 'TMCC3 (57458)', 'TMCO1 (54499)', 'TMCO2 (127391)', 'TMCO3 (55002)', 'TMCO4 (255104)', 'TMCO5A (145942)', 'TMCO6 (55374)', 'TMED1 (11018)', 'TMED10 (10972)', 'TMED2 (10959)', 'TMED3 (23423)', 'TMED4 (222068)', 'TMED5 (50999)', 'TMED6 (146456)', 'TMED7 (51014)', 'TMED8 (283578)', 'TMED9 (54732)', 'TMEFF1 (8577)', 'TMEFF2 (23671)', 'TMEM100 (55273)', 'TMEM101 (84336)', 'TMEM102 (284114)', 'TMEM104 (54868)', 'TMEM106A (113277)', 'TMEM106B (54664)', 'TMEM106C (79022)', 'TMEM107 (84314)', 'TMEM108 (66000)', 'TMEM109 (79073)', 'TMEM11 (8834)', 'TMEM115 (11070)', 'TMEM116 (89894)', 'TMEM117 (84216)', 'TMEM119 (338773)', 'TMEM120A (83862)', 'TMEM120B (144404)', 'TMEM121 (80757)', 'TMEM121B (27439)', 'TMEM123 (114908)', 'TMEM125 (128218)', 'TMEM126A (84233)', 'TMEM126B (55863)', 'TMEM127 (55654)', 'TMEM128 (85013)', 'TMEM129 (92305)', 'TMEM130 (222865)', 'TMEM131 (23505)', 'TMEM131L (23240)', 'TMEM132A (54972)', 'TMEM132B (114795)', 'TMEM132D (121256)', 'TMEM132E (124842)', 'TMEM134 (80194)', 'TMEM135 (65084)', 'TMEM138 (51524)', 'TMEM139 (135932)', 'TMEM140 (55281)', 'TMEM141 (85014)', 'TMEM143 (55260)', 'TMEM144 (55314)', 'TMEM145 (284339)', 'TMEM147 (10430)', 'TMEM14A (28978)', 'TMEM14B (81853)', 'TMEM14C (51522)', 'TMEM150A (129303)', 'TMEM150B (284417)', 'TMEM150C (441027)', 'TMEM151A (256472)', 'TMEM151B (441151)', 'TMEM154 (201799)', 'TMEM156 (80008)', 'TMEM158 (25907)', 'TMEM160 (54958)', 'TMEM161A (54929)', 'TMEM161B (153396)', 'TMEM163 (81615)', 'TMEM164 (84187)', 'TMEM165 (55858)', 'TMEM167A (153339)', 'TMEM167B (56900)', 'TMEM168 (64418)', 'TMEM169 (92691)', 'TMEM17 (200728)', 'TMEM170A (124491)', 'TMEM170B (100113407)', 'TMEM171 (134285)', 'TMEM174 (134288)', 'TMEM175 (84286)', 'TMEM176A (55365)', 'TMEM176B (28959)', 'TMEM177 (80775)', 'TMEM178A (130733)', 'TMEM178B (100507421)', 'TMEM179 (388021)', 'TMEM179B (374395)', 'TMEM18 (129787)', 'TMEM181 (57583)', 'TMEM182 (130827)', 'TMEM183A (92703)', 'TMEM184A (202915)', 'TMEM184B (25829)', 'TMEM184C (55751)', 'TMEM185A (84548)', 'TMEM185B (79134)', 'TMEM186 (25880)', 'TMEM187 (8269)', 'TMEM19 (55266)', 'TMEM190 (147744)', 'TMEM192 (201931)', 'TMEM196 (256130)', 'TMEM198 (130612)', 'TMEM199 (147007)', 'TMEM200A (114801)', 'TMEM200B (399474)', 'TMEM200C (645369)', 'TMEM201 (199953)', 'TMEM202 (338949)', 'TMEM203 (94107)', 'TMEM204 (79652)', 'TMEM205 (374882)', 'TMEM207 (131920)', 'TMEM208 (29100)', 'TMEM209 (84928)', 'TMEM210 (100505993)', 'TMEM212 (389177)', 'TMEM213 (155006)', 'TMEM214 (54867)', 'TMEM215 (401498)', 'TMEM216 (51259)', 'TMEM217 (221468)', 'TMEM218 (219854)', 'TMEM219 (124446)', 'TMEM220 (388335)', 'TMEM221 (100130519)', 'TMEM222 (84065)', 'TMEM223 (79064)', 'TMEM225 (338661)', 'TMEM229A (730130)', 'TMEM229B (161145)', 'TMEM230 (29058)', 'TMEM231 (79583)', 'TMEM232 (642987)', 'TMEM233 (387890)', 'TMEM234 (56063)', 'TMEM235 (283999)', 'TMEM236 (653567)', 'TMEM237 (65062)', 'TMEM238 (388564)', 'TMEM239 (100288797)', 'TMEM240 (339453)', 'TMEM241 (85019)', 'TMEM242 (729515)', 'TMEM243 (79161)', 'TMEM244 (253582)', 'TMEM245 (23731)', 'TMEM247 (388946)', 'TMEM248 (55069)', 'TMEM249 (340393)', 'TMEM25 (84866)', 'TMEM250 (90120)', 'TMEM252 (169693)', 'TMEM253 (643382)', 'TMEM254 (80195)', 'TMEM255A (55026)', 'TMEM255B (348013)', 'TMEM256 (254863)', 'TMEM258 (746)', 'TMEM259 (91304)', 'TMEM26 (219623)', 'TMEM260 (54916)', 'TMEM262 (100130348)', 'TMEM263 (90488)', 'TMEM266 (123591)', 'TMEM267 (64417)', 'TMEM268 (203197)', 'TMEM270 (135886)', 'TMEM273 (170371)', 'TMEM276 (84773)', 'TMEM30A (55754)', 'TMEM30B (161291)', 'TMEM31 (203562)', 'TMEM33 (55161)', 'TMEM35A (59353)', 'TMEM35B (100506144)', 'TMEM37 (140738)', 'TMEM38A (79041)', 'TMEM38B (55151)', 'TMEM39A (55254)', 'TMEM39B (55116)', 'TMEM40 (55287)', 'TMEM41A (90407)', 'TMEM41B (440026)', 'TMEM42 (131616)', 'TMEM43 (79188)', 'TMEM44 (93109)', 'TMEM45A (55076)', 'TMEM45B (120224)', 'TMEM47 (83604)', 'TMEM50A (23585)', 'TMEM50B (757)', 'TMEM51 (55092)', 'TMEM52 (339456)', 'TMEM52B (120939)', 'TMEM53 (79639)', 'TMEM54 (113452)', 'TMEM59 (9528)', 'TMEM59L (25789)', 'TMEM60 (85025)', 'TMEM61 (199964)', 'TMEM62 (80021)', 'TMEM63A (9725)', 'TMEM63B (55362)', 'TMEM63C (57156)', 'TMEM64 (169200)', 'TMEM65 (157378)', 'TMEM67 (91147)', 'TMEM68 (137695)', 'TMEM69 (51249)', 'TMEM70 (54968)', 'TMEM71 (137835)', 'TMEM72 (643236)', 'TMEM74 (157753)', 'TMEM74B (55321)', 'TMEM79 (84283)', 'TMEM80 (283232)', 'TMEM81 (388730)', 'TMEM82 (388595)', 'TMEM86A (144110)', 'TMEM86B (255043)', 'TMEM87A (25963)', 'TMEM87B (84910)', 'TMEM88 (92162)', 'TMEM88B (643965)', 'TMEM89 (440955)', 'TMEM8B (51754)', 'TMEM9 (252839)', 'TMEM91 (641649)', 'TMEM92 (162461)', 'TMEM94 (9772)', 'TMEM95 (339168)', 'TMEM97 (27346)', 'TMEM98 (26022)', 'TMEM9B (56674)', 'TMF1 (7110)', 'TMIE (259236)', 'TMIGD1 (388364)', 'TMIGD2 (126259)', 'TMIGD3 (57413)', 'TMLHE (55217)', 'TMOD1 (7111)', 'TMOD2 (29767)', 'TMOD3 (29766)', 'TMOD4 (29765)', 'TMPO (7112)', 'TMPPE (643853)', 'TMPRSS11A (339967)', 'TMPRSS11B (132724)', 'TMPRSS11D (9407)', 'TMPRSS11E (28983)', 'TMPRSS11F (389208)', 'TMPRSS12 (283471)', 'TMPRSS13 (84000)', 'TMPRSS15 (5651)', 'TMPRSS2 (7113)', 'TMPRSS3 (64699)', 'TMPRSS4 (56649)', 'TMPRSS5 (80975)', 'TMPRSS6 (164656)', 'TMPRSS7 (344805)', 'TMPRSS9 (360200)', 'TMSB10 (9168)', 'TMSB15A (11013)', 'TMSB4X (7114)', 'TMT1A (25840)', 'TMT1B (196410)', 'TMTC1 (83857)', 'TMTC2 (160335)', 'TMTC3 (160418)', 'TMTC4 (84899)', 'TMUB1 (83590)', 'TMUB2 (79089)', 'TMX1 (81542)', 'TMX2 (51075)', 'TMX3 (54495)', 'TMX4 (56255)', 'TNC (3371)', 'TNF (7124)', 'TNFAIP1 (7126)', 'TNFAIP2 (7127)', 'TNFAIP3 (7128)', 'TNFAIP6 (7130)', 'TNFAIP8 (25816)', 'TNFAIP8L1 (126282)', 'TNFAIP8L2 (79626)', 'TNFAIP8L3 (388121)', 'TNFRSF10A (8797)', 'TNFRSF10B (8795)', 'TNFRSF10C (8794)', 'TNFRSF10D (8793)', 'TNFRSF11A (8792)', 'TNFRSF11B (4982)', 'TNFRSF12A (51330)', 'TNFRSF13B (23495)', 'TNFRSF13C (115650)', 'TNFRSF14 (8764)', 'TNFRSF17 (608)', 'TNFRSF18 (8784)', 'TNFRSF19 (55504)', 'TNFRSF1A (7132)', 'TNFRSF1B (7133)', 'TNFRSF21 (27242)', 'TNFRSF25 (8718)', 'TNFRSF4 (7293)', 'TNFRSF6B (8771)', 'TNFRSF8 (943)', 'TNFRSF9 (3604)', 'TNFSF10 (8743)', 'TNFSF11 (8600)', 'TNFSF12 (8742)', 'TNFSF13 (8741)', 'TNFSF13B (10673)', 'TNFSF14 (8740)', 'TNFSF15 (9966)', 'TNFSF18 (8995)', 'TNFSF4 (7292)', 'TNFSF8 (944)', 'TNFSF9 (8744)', 'TNIK (23043)', 'TNIP1 (10318)', 'TNIP2 (79155)', 'TNIP3 (79931)', 'TNK1 (8711)', 'TNK2 (10188)', 'TNKS (8658)', 'TNKS1BP1 (85456)', 'TNKS2 (80351)', 'TNMD (64102)', 'TNN (63923)', 'TNNC1 (7134)', 'TNNC2 (7125)', 'TNNI1 (7135)', 'TNNI2 (7136)', 'TNNI3 (7137)', 'TNNI3K (51086)', 'TNNT1 (7138)', 'TNNT2 (7139)', 'TNNT3 (7140)', 'TNP1 (7141)', 'TNP2 (7142)', 'TNPO1 (3842)', 'TNPO2 (30000)', 'TNPO3 (23534)', 'TNR (7143)', 'TNRC18 (84629)', 'TNRC6A (27327)', 'TNRC6B (23112)', 'TNRC6C (57690)', 'TNS1 (7145)', 'TNS2 (23371)', 'TNS3 (64759)', 'TNS4 (84951)', 'TNXB (7148)', 'TOB1 (10140)', 'TOB2 (10766)', 'TOE1 (114034)', 'TOGARAM1 (23116)', 'TOGARAM2 (165186)', 'TOLLIP (54472)', 'TOM1 (10043)', 'TOM1L1 (10040)', 'TOM1L2 (146691)', 'TOMM20 (9804)', 'TOMM20L (387990)', 'TOMM22 (56993)', 'TOMM34 (10953)', 'TOMM40 (10452)', 'TOMM40L (84134)', 'TOMM5 (401505)', 'TOMM6 (100188893)', 'TOMM7 (54543)', 'TOMM70 (9868)', 'TONSL (4796)', 'TOP1 (7150)', 'TOP1MT (116447)', 'TOP2A (7153)', 'TOP2B (7155)', 'TOP3A (7156)', 'TOP3B (8940)', 'TOP6BL (79703)', 'TOPAZ1 (375337)', 'TOPBP1 (11073)', 'TOPORS (10210)', 'TOR1A (1861)', 'TOR1AIP1 (26092)', 'TOR1AIP2 (163590)', 'TOR1B (27348)', 'TOR2A (27433)', 'TOR3A (64222)', 'TOR4A (54863)', 'TOX (9760)', 'TOX2 (84969)', 'TOX3 (27324)', 'TOX4 (9878)', 'TP53 (7157)', 'TP53AIP1 (63970)', 'TP53BP1 (7158)', 'TP53BP2 (7159)', 'TP53I11 (9537)', 'TP53I13 (90313)', 'TP53I3 (9540)', 'TP53INP1 (94241)', 'TP53INP2 (58476)', 'TP53RK (112858)', 'TP53TG5 (27296)', 'TP63 (8626)', 'TP73 (7161)', 'TPBG (7162)', 'TPBGL (100507050)', 'TPCN1 (53373)', 'TPCN2 (219931)', 'TPD52 (7163)', 'TPD52L1 (7164)', 'TPD52L2 (7165)', 'TPD52L3 (89882)', 'TPGS1 (91978)', 'TPGS2 (25941)', 'TPH1 (7166)', 'TPH2 (121278)', 'TPI1 (7167)', 'TPK1 (27010)', 'TPM1 (7168)', 'TPM2 (7169)', 'TPM3 (7170)', 'TPM4 (7171)', 'TPMT (7172)', 'TPO (7173)', 'TPP1 (1200)', 'TPP2 (7174)', 'TPPP (11076)', 'TPPP2 (122664)', 'TPPP3 (51673)', 'TPR (7175)', 'TPRA1 (131601)', 'TPRG1 (285386)', 'TPRG1L (127262)', 'TPRKB (51002)', 'TPRN (286262)', 'TPRX1 (284355)', 'TPSAB1 (7177)', 'TPSD1 (23430)', 'TPSG1 (25823)', 'TPST1 (8460)', 'TPST2 (8459)', 'TPT1 (7178)', 'TPTE2 (93492)', 'TPX2 (22974)', 'TRA2A (29896)', 'TRA2B (6434)', 'TRABD (80305)', 'TRABD2A (129293)', 'TRABD2B (388630)', 'TRADD (8717)', 'TRAF1 (7185)', 'TRAF2 (7186)', 'TRAF3 (7187)', 'TRAF3IP1 (26146)', 'TRAF3IP2 (10758)', 'TRAF3IP3 (80342)', 'TRAF4 (9618)', 'TRAF5 (7188)', 'TRAF6 (7189)', 'TRAF7 (84231)', 'TRAFD1 (10906)', 'TRAIP (10293)', 'TRAK1 (22906)', 'TRAK2 (66008)', 'TRAM1 (23471)', 'TRAM1L1 (133022)', 'TRAM2 (9697)', 'TRANK1 (9881)', 'TRAP1 (10131)', 'TRAPPC1 (58485)', 'TRAPPC10 (7109)', 'TRAPPC11 (60684)', 'TRAPPC12 (51112)', 'TRAPPC13 (80006)', 'TRAPPC14 (55262)', 'TRAPPC2L (51693)', 'TRAPPC3 (27095)', 'TRAPPC3L (100128327)', 'TRAPPC4 (51399)', 'TRAPPC5 (126003)', 'TRAPPC6A (79090)', 'TRAPPC6B (122553)', 'TRAPPC8 (22878)', 'TRAPPC9 (83696)', 'TRARG1 (286753)', 'TRAT1 (50852)', 'TRDMT1 (1787)', 'TRDN (10345)', 'TREM1 (54210)', 'TREM2 (54209)', 'TREML1 (340205)', 'TREML2 (79865)', 'TREML4 (285852)', 'TRERF1 (55809)', 'TREX1 (11277)', 'TREX2 (11219)', 'TRH (7200)', 'TRHDE (29953)', 'TRHR (7201)', 'TRIAP1 (51499)', 'TRIB1 (10221)', 'TRIB2 (28951)', 'TRIB3 (57761)', 'TRIM10 (10107)', 'TRIM11 (81559)', 'TRIM13 (10206)', 'TRIM14 (9830)', 'TRIM15 (89870)', 'TRIM16 (10626)', 'TRIM17 (51127)', 'TRIM2 (23321)', 'TRIM21 (6737)', 'TRIM22 (10346)', 'TRIM23 (373)', 'TRIM24 (8805)', 'TRIM25 (7706)', 'TRIM26 (7726)', 'TRIM27 (5987)', 'TRIM28 (10155)', 'TRIM29 (23650)', 'TRIM3 (10612)', 'TRIM31 (11074)', 'TRIM32 (22954)', 'TRIM33 (51592)', 'TRIM34 (53840)', 'TRIM35 (23087)', 'TRIM36 (55521)', 'TRIM37 (4591)', 'TRIM38 (10475)', 'TRIM39 (56658)', 'TRIM4 (89122)', 'TRIM40 (135644)', 'TRIM41 (90933)', 'TRIM42 (287015)', 'TRIM43 (129868)', 'TRIM44 (54765)', 'TRIM45 (80263)', 'TRIM46 (80128)', 'TRIM47 (91107)', 'TRIM49B (283116)', 'TRIM49D1 (399939)', 'TRIM5 (85363)', 'TRIM50 (135892)', 'TRIM52 (84851)', 'TRIM54 (57159)', 'TRIM55 (84675)', 'TRIM56 (81844)', 'TRIM58 (25893)', 'TRIM59 (286827)', 'TRIM6 (117854)', 'TRIM60 (166655)', 'TRIM61 (391712)', 'TRIM62 (55223)', 'TRIM63 (84676)', 'TRIM65 (201292)', 'TRIM67 (440730)', 'TRIM68 (55128)', 'TRIM69 (140691)', 'TRIM7 (81786)', 'TRIM71 (131405)', 'TRIM72 (493829)', 'TRIM77 (390231)', 'TRIM8 (81603)', 'TRIM9 (114088)', 'TRIML1 (339976)', 'TRIML2 (205860)', 'TRIO (7204)', 'TRIOBP (11078)', 'TRIP10 (9322)', 'TRIP11 (9321)', 'TRIP12 (9320)', 'TRIP13 (9319)', 'TRIP4 (9325)', 'TRIP6 (7205)', 'TRIQK (286144)', 'TRIR (79002)', 'TRIT1 (54802)', 'TRMO (51531)', 'TRMT1 (55621)', 'TRMT10A (93587)', 'TRMT10B (158234)', 'TRMT10C (54931)', 'TRMT11 (60487)', 'TRMT112 (51504)', 'TRMT12 (55039)', 'TRMT13 (54482)', 'TRMT1L (81627)', 'TRMT2A (27037)', 'TRMT2B (79979)', 'TRMT44 (152992)', 'TRMT5 (57570)', 'TRMT6 (51605)', 'TRMT61A (115708)', 'TRMT61B (55006)', 'TRMT9B (57604)', 'TRMU (55687)', 'TRNAU1AP (54952)', 'TRNP1 (388610)', 'TRNT1 (51095)', 'TRO (7216)', 'TROAP (10024)', 'TRPA1 (8989)', 'TRPC1 (7220)', 'TRPC3 (7222)', 'TRPC4 (7223)', 'TRPC4AP (26133)', 'TRPC5 (7224)', 'TRPC6 (7225)', 'TRPC7 (57113)', 'TRPM1 (4308)', 'TRPM2 (7226)', 'TRPM3 (80036)', 'TRPM4 (54795)', 'TRPM5 (29850)', 'TRPM6 (140803)', 'TRPM7 (54822)', 'TRPM8 (79054)', 'TRPS1 (7227)', 'TRPT1 (83707)', 'TRPV1 (7442)', 'TRPV2 (51393)', 'TRPV3 (162514)', 'TRPV4 (59341)', 'TRPV5 (56302)', 'TRPV6 (55503)', 'TRRAP (8295)', 'TRUB1 (142940)', 'TRUB2 (26995)', 'TSACC (128229)', 'TSBP1 (10665)', 'TSC1 (7248)', 'TSC2 (7249)', 'TSC22D1 (8848)', 'TSC22D2 (9819)', 'TSC22D3 (1831)', 'TSC22D4 (81628)', 'TSEN15 (116461)', 'TSEN2 (80746)', 'TSEN34 (79042)', 'TSEN54 (283989)', 'TSFM (10102)', 'TSG101 (7251)', 'TSGA10 (80705)', 'TSGA10IP (254187)', 'TSGA13 (114960)', 'TSHB (7252)', 'TSHR (7253)', 'TSHZ1 (10194)', 'TSHZ2 (128553)', 'TSHZ3 (57616)', 'TSKS (60385)', 'TSKU (25987)', 'TSLP (85480)', 'TSN (7247)', 'TSNARE1 (203062)', 'TSNAX (7257)', 'TSNAXIP1 (55815)', 'TSPAN1 (10103)', 'TSPAN11 (441631)', 'TSPAN12 (23554)', 'TSPAN13 (27075)', 'TSPAN14 (81619)', 'TSPAN15 (23555)', 'TSPAN16 (26526)', 'TSPAN17 (26262)', 'TSPAN18 (90139)', 'TSPAN19 (144448)', 'TSPAN2 (10100)', 'TSPAN3 (10099)', 'TSPAN31 (6302)', 'TSPAN32 (10077)', 'TSPAN33 (340348)', 'TSPAN4 (7106)', 'TSPAN5 (10098)', 'TSPAN6 (7105)', 'TSPAN7 (7102)', 'TSPAN8 (7103)', 'TSPAN9 (10867)', 'TSPEAR (54084)', 'TSPO (706)', 'TSPO2 (222642)', 'TSPOAP1 (9256)', 'TSPYL1 (7259)', 'TSPYL2 (64061)', 'TSPYL4 (23270)', 'TSPYL5 (85453)', 'TSPYL6 (388951)', 'TSR1 (55720)', 'TSR2 (90121)', 'TSR3 (115939)', 'TSSC4 (10078)', 'TSSK1B (83942)', 'TSSK2 (23617)', 'TSSK3 (81629)', 'TSSK4 (283629)', 'TSSK6 (83983)', 'TST (7263)', 'TSTD1 (100131187)', 'TSTD2 (158427)', 'TTBK1 (84630)', 'TTBK2 (146057)', 'TTC1 (7265)', 'TTC12 (54970)', 'TTC13 (79573)', 'TTC14 (151613)', 'TTC16 (158248)', 'TTC17 (55761)', 'TTC19 (54902)', 'TTC21A (199223)', 'TTC21B (79809)', 'TTC22 (55001)', 'TTC23 (64927)', 'TTC23L (153657)', 'TTC24 (164118)', 'TTC27 (55622)', 'TTC28 (23331)', 'TTC29 (83894)', 'TTC3 (7267)', 'TTC31 (64427)', 'TTC32 (130502)', 'TTC33 (23548)', 'TTC34 (100287898)', 'TTC36 (143941)', 'TTC38 (55020)', 'TTC39A (22996)', 'TTC39B (158219)', 'TTC39C (125488)', 'TTC4 (7268)', 'TTC5 (91875)', 'TTC7A (57217)', 'TTC7B (145567)', 'TTC8 (123016)', 'TTC9 (23508)', 'TTC9B (148014)', 'TTC9C (283237)', 'TTF1 (7270)', 'TTF2 (8458)', 'TTI1 (9675)', 'TTI2 (80185)', 'TTK (7272)', 'TTL (150465)', 'TTLL1 (25809)', 'TTLL10 (254173)', 'TTLL11 (158135)', 'TTLL12 (23170)', 'TTLL2 (83887)', 'TTLL3 (26140)', 'TTLL4 (9654)', 'TTLL5 (23093)', 'TTLL6 (284076)', 'TTLL7 (79739)', 'TTLL9 (164395)', 'TTN (7273)', 'TTPA (7274)', 'TTPAL (79183)', 'TTR (7276)', 'TTYH1 (57348)', 'TTYH2 (94015)', 'TTYH3 (80727)', 'TUB (7275)', 'TUBA1A (7846)', 'TUBA1B (10376)', 'TUBA1C (84790)', 'TUBA3C (7278)', 'TUBA3D (113457)', 'TUBA3E (112714)', 'TUBA4A (7277)', 'TUBA8 (51807)', 'TUBAL3 (79861)', 'TUBB (203068)', 'TUBB1 (81027)', 'TUBB2A (7280)', 'TUBB2B (347733)', 'TUBB3 (10381)', 'TUBB4A (10382)', 'TUBB4B (10383)', 'TUBB6 (84617)', 'TUBD1 (51174)', 'TUBE1 (51175)', 'TUBG1 (7283)', 'TUBG2 (27175)', 'TUBGCP2 (10844)', 'TUBGCP3 (10426)', 'TUBGCP4 (27229)', 'TUBGCP5 (114791)', 'TUBGCP6 (85378)', 'TUFM (7284)', 'TUFT1 (7286)', 'TULP1 (7287)', 'TULP2 (7288)', 'TULP3 (7289)', 'TULP4 (56995)', 'TUSC1 (286319)', 'TUSC2 (11334)', 'TUSC3 (7991)', 'TUT1 (64852)', 'TUT4 (23318)', 'TUT7 (79670)', 'TVP23A (780776)', 'TWF1 (5756)', 'TWF2 (11344)', 'TWIST1 (7291)', 'TWIST2 (117581)', 'TWNK (56652)', 'TWSG1 (57045)', 'TXK (7294)', 'TXLNA (200081)', 'TXLNB (167838)', 'TXLNG (55787)', 'TXN (7295)', 'TXN2 (25828)', 'TXNDC11 (51061)', 'TXNDC12 (51060)', 'TXNDC15 (79770)', 'TXNDC16 (57544)', 'TXNDC17 (84817)', 'TXNDC2 (84203)', 'TXNDC5 (81567)', 'TXNDC8 (255220)', 'TXNDC9 (10190)', 'TXNIP (10628)', 'TXNL1 (9352)', 'TXNL4A (10907)', 'TXNL4B (54957)', 'TXNRD1 (7296)', 'TXNRD2 (10587)', 'TYK2 (7297)', 'TYMP (1890)', 'TYMS (7298)', 'TYR (7299)', 'TYRO3 (7301)', 'TYROBP (7305)', 'TYRP1 (7306)', 'TYSND1 (219743)', 'TYW1 (55253)', 'TYW1B (441250)', 'TYW3 (127253)', 'TYW5 (129450)', 'U2AF1 (7307)', 'U2AF1L4 (199746)', 'U2AF2 (11338)', 'U2SURP (23350)', 'UACA (55075)', 'UAP1 (6675)', 'UAP1L1 (91373)', 'UBA1 (7317)', 'UBA2 (10054)', 'UBA3 (9039)', 'UBA5 (79876)', 'UBA52 (7311)', 'UBA6 (55236)', 'UBA7 (7318)', 'UBAC1 (10422)', 'UBAC2 (337867)', 'UBALD1 (124402)', 'UBALD2 (283991)', 'UBAP1 (51271)', 'UBAP1L (390595)', 'UBAP2 (55833)', 'UBAP2L (9898)', 'UBASH3A (53347)', 'UBASH3B (84959)', 'UBB (7314)', 'UBC (7316)', 'UBD (10537)', 'UBE2A (7319)', 'UBE2B (7320)', 'UBE2C (11065)', 'UBE2D1 (7321)', 'UBE2D2 (7322)', 'UBE2D3 (7323)', 'UBE2D4 (51619)', 'UBE2E1 (7324)', 'UBE2E2 (7325)', 'UBE2E3 (10477)', 'UBE2F (140739)', 'UBE2G1 (7326)', 'UBE2G2 (7327)', 'UBE2H (7328)', 'UBE2I (7329)', 'UBE2J1 (51465)', 'UBE2J2 (118424)', 'UBE2K (3093)', 'UBE2L3 (7332)', 'UBE2L6 (9246)', 'UBE2M (9040)', 'UBE2N (7334)', 'UBE2O (63893)', 'UBE2Q1 (55585)', 'UBE2Q2 (92912)', 'UBE2QL1 (134111)', 'UBE2R2 (54926)', 'UBE2S (27338)', 'UBE2T (29089)', 'UBE2U (148581)', 'UBE2V1 (7335)', 'UBE2V2 (7336)', 'UBE2W (55284)', 'UBE2Z (65264)', 'UBE3A (7337)', 'UBE3B (89910)', 'UBE3C (9690)', 'UBE3D (90025)', 'UBE4A (9354)', 'UBE4B (10277)', 'UBFD1 (56061)', 'UBIAD1 (29914)', 'UBL3 (5412)', 'UBL4A (8266)', 'UBL4B (164153)', 'UBL5 (59286)', 'UBL7 (84993)', 'UBLCP1 (134510)', 'UBN1 (29855)', 'UBN2 (254048)', 'UBOX5 (22888)', 'UBP1 (7342)', 'UBQLN1 (29979)', 'UBQLN2 (29978)', 'UBQLN3 (50613)', 'UBQLN4 (56893)', 'UBQLNL (143630)', 'UBR1 (197131)', 'UBR2 (23304)', 'UBR3 (130507)', 'UBR4 (23352)', 'UBR5 (51366)', 'UBR7 (55148)', 'UBTD1 (80019)', 'UBTD2 (92181)', 'UBTF (7343)', 'UBXN1 (51035)', 'UBXN10 (127733)', 'UBXN11 (91544)', 'UBXN2A (165324)', 'UBXN2B (137886)', 'UBXN4 (23190)', 'UBXN6 (80700)', 'UBXN7 (26043)', 'UCHL1 (7345)', 'UCHL3 (7347)', 'UCHL5 (51377)', 'UCK1 (83549)', 'UCK2 (7371)', 'UCKL1 (54963)', 'UCMA (221044)', 'UCN (7349)', 'UCN2 (90226)', 'UCN3 (114131)', 'UCP1 (7350)', 'UCP2 (7351)', 'UCP3 (7352)', 'UEVLD (55293)', 'UFC1 (51506)', 'UFD1 (7353)', 'UFL1 (23376)', 'UFM1 (51569)', 'UFSP1 (402682)', 'UFSP2 (55325)', 'UGCG (7357)', 'UGDH (7358)', 'UGGT1 (56886)', 'UGGT2 (55757)', 'UGP2 (7360)', 'UGT1A1 (54658)', 'UGT1A10 (54575)', 'UGT1A3 (54659)', 'UGT1A4 (54657)', 'UGT1A5 (54579)', 'UGT1A7 (54577)', 'UGT1A8 (54576)', 'UGT1A9 (54600)', 'UGT2A1 (10941)', 'UGT2A3 (79799)', 'UGT2B11 (10720)', 'UGT2B15 (7366)', 'UGT2B17 (7367)', 'UGT2B28 (54490)', 'UGT2B4 (7363)', 'UGT2B7 (7364)', 'UGT3A1 (133688)', 'UGT3A2 (167127)', 'UGT8 (7368)', 'UHMK1 (127933)', 'UHRF2 (115426)', 'UIMC1 (51720)', 'ULBP1 (80329)', 'ULBP2 (80328)', 'ULBP3 (79465)', 'ULK1 (8408)', 'ULK2 (9706)', 'ULK3 (25989)', 'ULK4 (54986)', 'UMOD (7369)', 'UMODL1 (89766)', 'UMPS (7372)', 'UNC119 (9094)', 'UNC119B (84747)', 'UNC13A (23025)', 'UNC13B (10497)', 'UNC13C (440279)', 'UNC13D (201294)', 'UNC45A (55898)', 'UNC45B (146862)', 'UNC50 (25972)', 'UNC5A (90249)', 'UNC5B (219699)', 'UNC5C (8633)', 'UNC5CL (222643)', 'UNC5D (137970)', 'UNC79 (57578)', 'UNC80 (285175)', 'UNC93A (54346)', 'UNCX (340260)', 'UNG (7374)', 'UNK (85451)', 'UNKL (64718)', 'UPB1 (51733)', 'UPF1 (5976)', 'UPF2 (26019)', 'UPF3A (65110)', 'UPF3B (65109)', 'UPK1A (11045)', 'UPK1B (7348)', 'UPK2 (7379)', 'UPK3A (7380)', 'UPK3B (105375355)', 'UPP1 (7378)', 'UPP2 (151531)', 'UPRT (139596)', 'UQCC1 (55245)', 'UQCC2 (84300)', 'UQCC3 (790955)', 'UQCC4 (283951)', 'UQCC5 (440957)', 'UQCC6 (728568)', 'UQCR10 (29796)', 'UQCR11 (10975)', 'UQCRB (7381)', 'UQCRC1 (7384)', 'UQCRC2 (7385)', 'UQCRFS1 (7386)', 'UQCRQ (27089)', 'URAD (646625)', 'URB1 (9875)', 'URB2 (9816)', 'URGCP (55665)', 'URI1 (8725)', 'URM1 (81605)', 'UROC1 (131669)', 'UROD (7389)', 'UROS (7390)', 'USB1 (79650)', 'USE1 (55850)', 'USF1 (7391)', 'USF2 (7392)', 'USF3 (205717)', 'USH1C (10083)', 'USH1G (124590)', 'USH2A (7399)', 'USHBP1 (83878)', 'USP1 (7398)', 'USP10 (9100)', 'USP11 (8237)', 'USP12 (219333)', 'USP13 (8975)', 'USP14 (9097)', 'USP15 (9958)', 'USP16 (10600)', 'USP17L10 (100287144)', 'USP17L13 (100287238)', 'USP18 (11274)', 'USP19 (10869)', 'USP2 (9099)', 'USP20 (10868)', 'USP21 (27005)', 'USP22 (23326)', 'USP24 (23358)', 'USP25 (29761)', 'USP26 (83844)', 'USP27X (389856)', 'USP28 (57646)', 'USP29 (57663)', 'USP3 (9960)', 'USP30 (84749)', 'USP31 (57478)', 'USP32 (84669)', 'USP33 (23032)', 'USP34 (9736)', 'USP35 (57558)', 'USP36 (57602)', 'USP37 (57695)', 'USP38 (84640)', 'USP39 (10713)', 'USP4 (7375)', 'USP40 (55230)', 'USP42 (84132)', 'USP43 (124739)', 'USP44 (84101)', 'USP45 (85015)', 'USP46 (64854)', 'USP47 (55031)', 'USP48 (84196)', 'USP49 (25862)', 'USP5 (8078)', 'USP50 (373509)', 'USP51 (158880)', 'USP53 (54532)', 'USP54 (159195)', 'USP6 (9098)', 'USP6NL (9712)', 'USP7 (7874)', 'USP8 (9101)', 'USP9X (8239)', 'USP9Y (8287)', 'USPL1 (10208)', 'UST (10090)', 'UTF1 (8433)', 'UTP11 (51118)', 'UTP14A (10813)', 'UTP14C (9724)', 'UTP15 (84135)', 'UTP18 (51096)', 'UTP20 (27340)', 'UTP23 (84294)', 'UTP25 (27042)', 'UTP3 (57050)', 'UTP4 (84916)', 'UTP6 (55813)', 'UTRN (7402)', 'UTS2 (10911)', 'UTS2B (257313)', 'UTS2R (2837)', 'UTY (7404)', 'UVRAG (7405)', 'UVSSA (57654)', 'UXS1 (80146)', 'UXT (8409)', 'VAC14 (55697)', 'VAMP1 (6843)', 'VAMP2 (6844)', 'VAMP3 (9341)', 'VAMP4 (8674)', 'VAMP5 (10791)', 'VAMP7 (6845)', 'VAMP8 (8673)', 'VANGL1 (81839)', 'VANGL2 (57216)', 'VAPA (9218)', 'VAPB (9217)', 'VARS1 (7407)', 'VARS2 (57176)', 'VASH1 (22846)', 'VASH2 (79805)', 'VASN (114990)', 'VASP (7408)', 'VAT1 (10493)', 'VAT1L (57687)', 'VAV1 (7409)', 'VAV2 (7410)', 'VAV3 (10451)', 'VAX1 (11023)', 'VAX2 (25806)', 'VBP1 (7411)', 'VCAM1 (7412)', 'VCAN (1462)', 'VCF1 (84923)', 'VCF2 (90736)', 'VCL (7414)', 'VCP (7415)', 'VCPIP1 (80124)', 'VCPKMT (79609)', 'VDAC1 (7416)', 'VDAC2 (7417)', 'VDAC3 (7419)', 'VDR (7421)', 'VEGFA (7422)', 'VEGFB (7423)', 'VEGFC (7424)', 'VEGFD (2277)', 'VENTX (27287)', 'VEPH1 (79674)', 'VEZF1 (7716)', 'VEZT (55591)', 'VGF (7425)', 'VGLL1 (51442)', 'VGLL2 (245806)', 'VGLL3 (389136)', 'VGLL4 (9686)', 'VHL (7428)', 'VIL1 (7429)', 'VILL (50853)', 'VIM (7431)', 'VIP (7432)', 'VIPAS39 (63894)', 'VIPR1 (7433)', 'VIPR2 (7434)', 'VIRMA (25962)', 'VIT (5212)', 'VKORC1 (79001)', 'VKORC1L1 (154807)', 'VLDLR (7436)', 'VMA21 (203547)', 'VMAC (400673)', 'VMO1 (284013)', 'VMP1 (81671)', 'VN1R1 (57191)', 'VN1R2 (317701)', 'VN1R4 (317703)', 'VNN1 (8876)', 'VNN2 (8875)', 'VOPP1 (81552)', 'VPREB1 (7441)', 'VPREB3 (29802)', 'VPS13A (23230)', 'VPS13B (157680)', 'VPS13C (54832)', 'VPS13D (55187)', 'VPS16 (64601)', 'VPS18 (57617)', 'VPS25 (84313)', 'VPS26A (9559)', 'VPS26B (112936)', 'VPS26C (10311)', 'VPS28 (51160)', 'VPS29 (51699)', 'VPS33A (65082)', 'VPS33B (26276)', 'VPS35 (55737)', 'VPS35L (57020)', 'VPS36 (51028)', 'VPS37A (137492)', 'VPS37B (79720)', 'VPS37C (55048)', 'VPS37D (155382)', 'VPS39 (23339)', 'VPS41 (27072)', 'VPS45 (11311)', 'VPS4A (27183)', 'VPS4B (9525)', 'VPS50 (55610)', 'VPS51 (738)', 'VPS52 (6293)', 'VPS53 (55275)', 'VPS54 (51542)', 'VPS72 (6944)', 'VPS8 (23355)', 'VPS9D1 (9605)', 'VRK1 (7443)', 'VRK2 (7444)', 'VRK3 (51231)', 'VRTN (55237)', 'VSIG1 (340547)', 'VSIG10 (54621)', 'VSIG10L (147645)', 'VSIG2 (23584)', 'VSIG4 (11326)', 'VSIG8 (391123)', 'VSIR (64115)', 'VSNL1 (7447)', 'VSTM1 (284415)', 'VSTM2A (222008)', 'VSTM2B (342865)', 'VSTM2L (128434)', 'VSTM4 (196740)', 'VSTM5 (387804)', 'VSX1 (30813)', 'VSX2 (338917)', 'VTA1 (51534)', 'VTCN1 (79679)', 'VTI1A (143187)', 'VTI1B (10490)', 'VTN (7448)', 'VWA1 (64856)', 'VWA2 (340706)', 'VWA3A (146177)', 'VWA3B (200403)', 'VWA5A (4013)', 'VWA5B2 (90113)', 'VWA7 (80737)', 'VWA8 (23078)', 'VWC2 (375567)', 'VWC2L (402117)', 'VWCE (220001)', 'VWDE (221806)', 'VWF (7450)', 'VXN (254778)', 'WAC (51322)', 'WAPL (23063)', 'WARS1 (7453)', 'WARS2 (10352)', 'WAS (7454)', 'WASF1 (8936)', 'WASF2 (10163)', 'WASF3 (10810)', 'WASHC2A (387680)', 'WASHC2C (253725)', 'WASHC3 (51019)', 'WASHC4 (23325)', 'WASHC5 (9897)', 'WASL (8976)', 'WBP1 (23559)', 'WBP11 (51729)', 'WBP1L (54838)', 'WBP2 (23558)', 'WBP2NL (164684)', 'WBP4 (11193)', 'WDCP (80304)', 'WDFY1 (57590)', 'WDFY2 (115825)', 'WDFY3 (23001)', 'WDFY4 (57705)', 'WDHD1 (11169)', 'WDPCP (51057)', 'WDR1 (9948)', 'WDR11 (55717)', 'WDR12 (55759)', 'WDR13 (64743)', 'WDR17 (116966)', 'WDR18 (57418)', 'WDR19 (57728)', 'WDR20 (91833)', 'WDR24 (84219)', 'WDR25 (79446)', 'WDR26 (80232)', 'WDR27 (253769)', 'WDR3 (10885)', 'WDR31 (114987)', 'WDR33 (55339)', 'WDR35 (57539)', 'WDR36 (134430)', 'WDR37 (22884)', 'WDR38 (401551)', 'WDR4 (10785)', 'WDR41 (55255)', 'WDR43 (23160)', 'WDR44 (54521)', 'WDR45 (11152)', 'WDR45B (56270)', 'WDR46 (9277)', 'WDR47 (22911)', 'WDR48 (57599)', 'WDR49 (151790)', 'WDR5 (11091)', 'WDR53 (348793)', 'WDR54 (84058)', 'WDR55 (54853)', 'WDR59 (79726)', 'WDR5B (54554)', 'WDR6 (11180)', 'WDR62 (284403)', 'WDR7 (23335)', 'WDR70 (55100)', 'WDR72 (256764)', 'WDR73 (84942)', 'WDR74 (54663)', 'WDR75 (84128)', 'WDR76 (79968)', 'WDR77 (79084)', 'WDR81 (124997)', 'WDR82 (80335)', 'WDR83 (84292)', 'WDR83OS (51398)', 'WDR86 (349136)', 'WDR87 (83889)', 'WDR88 (126248)', 'WDR89 (112840)', 'WDR90 (197335)', 'WDR91 (29062)', 'WDR93 (56964)', 'WDSUB1 (151525)', 'WDTC1 (23038)', 'WEE1 (7465)', 'WEE2 (494551)', 'WFDC1 (58189)', 'WFDC10A (140832)', 'WFDC10B (280664)', 'WFDC11 (259239)', 'WFDC12 (128488)', 'WFDC13 (164237)', 'WFDC2 (10406)', 'WFDC3 (140686)', 'WFDC5 (149708)', 'WFDC6 (140870)', 'WFDC8 (90199)', 'WFDC9 (259240)', 'WFIKKN1 (117166)', 'WFIKKN2 (124857)', 'WFS1 (7466)', 'WHAMM (123720)', 'WHRN (25861)', 'WIF1 (11197)', 'WIPF1 (7456)', 'WIPF2 (147179)', 'WIPF3 (644150)', 'WIPI1 (55062)', 'WIPI2 (26100)', 'WIZ (58525)', 'WLS (79971)', 'WNK1 (65125)', 'WNK3 (65267)', 'WNK4 (65266)', 'WNT1 (7471)', 'WNT10A (80326)', 'WNT10B (7480)', 'WNT11 (7481)', 'WNT16 (51384)', 'WNT2 (7472)', 'WNT2B (7482)', 'WNT3 (7473)', 'WNT3A (89780)', 'WNT4 (54361)', 'WNT5A (7474)', 'WNT5B (81029)', 'WNT6 (7475)', 'WNT7A (7476)', 'WNT7B (7477)', 'WNT8A (7478)', 'WNT8B (7479)', 'WNT9A (7483)', 'WNT9B (7484)', 'WRAP53 (55135)', 'WRAP73 (49856)', 'WRN (7486)', 'WRNIP1 (56897)', 'WSB1 (26118)', 'WSB2 (55884)', 'WSCD1 (23302)', 'WSCD2 (9671)', 'WT1 (7490)', 'WTAP (9589)', 'WTIP (126374)', 'WWC1 (23286)', 'WWC2 (80014)', 'WWOX (51741)', 'WWP1 (11059)', 'WWP2 (11060)', 'WWTR1 (25937)', 'XAB2 (56949)', 'XAF1 (54739)', 'XAGE2 (9502)', 'XAGE3 (170626)', 'XAGE5 (170627)', 'XBP1 (7494)', 'XCL1 (6375)', 'XCL2 (6846)', 'XCR1 (2829)', 'XDH (7498)', 'XG (7499)', 'XIAP (331)', 'XIRP1 (165904)', 'XIRP2 (129446)', 'XK (7504)', 'XKR3 (150165)', 'XKR4 (114786)', 'XKR6 (286046)', 'XKR7 (343702)', 'XKR8 (55113)', 'XKR9 (389668)', 'XKRX (402415)', 'XPA (7507)', 'XPC (7508)', 'XPNPEP1 (7511)', 'XPNPEP2 (7512)', 'XPNPEP3 (63929)', 'XPO1 (7514)', 'XPO4 (64328)', 'XPO5 (57510)', 'XPO6 (23214)', 'XPO7 (23039)', 'XPOT (11260)', 'XPR1 (9213)', 'XRCC1 (7515)', 'XRCC2 (7516)', 'XRCC3 (7517)', 'XRCC4 (7518)', 'XRCC5 (7520)', 'XRCC6 (2547)', 'XRN1 (54464)', 'XRN2 (22803)', 'XRRA1 (143570)', 'XXYLT1 (152002)', 'XYLB (9942)', 'XYLT1 (64131)', 'XYLT2 (64132)', 'YAE1 (57002)', 'YAF2 (10138)', 'YAP1 (10413)', 'YARS1 (8565)', 'YARS2 (51067)', 'YBEY (54059)', 'YBX1 (4904)', 'YBX2 (51087)', 'YBX3 (8531)', 'YDJC (150223)', 'YEATS2 (55689)', 'YEATS4 (8089)', 'YES1 (7525)', 'YIF1A (10897)', 'YIF1B (90522)', 'YIPF1 (54432)', 'YIPF2 (78992)', 'YIPF3 (25844)', 'YIPF4 (84272)', 'YIPF5 (81555)', 'YIPF6 (286451)', 'YIPF7 (285525)', 'YJEFN3 (374887)', 'YJU2 (55702)', 'YJU2B (81576)', 'YKT6 (10652)', 'YLPM1 (56252)', 'YME1L1 (10730)', 'YOD1 (55432)', 'YPEL1 (29799)', 'YPEL2 (388403)', 'YPEL3 (83719)', 'YPEL4 (219539)', 'YPEL5 (51646)', 'YRDC (79693)', 'YTHDC1 (91746)', 'YTHDC2 (64848)', 'YTHDF1 (54915)', 'YTHDF2 (51441)', 'YWHAB (7529)', 'YWHAE (7531)', 'YWHAG (7532)', 'YWHAH (7533)', 'YWHAQ (10971)', 'YWHAZ (7534)', 'YY1 (7528)', 'YY1AP1 (55249)', 'YY2 (404281)', 'ZACN (353174)', 'ZAP70 (7535)', 'ZAR1 (326340)', 'ZAR1L (646799)', 'ZBBX (79740)', 'ZBED2 (79413)', 'ZBED3 (84327)', 'ZBED4 (9889)', 'ZBED6 (100381270)', 'ZBP1 (81030)', 'ZBTB1 (22890)', 'ZBTB10 (65986)', 'ZBTB11 (27107)', 'ZBTB12 (221527)', 'ZBTB14 (7541)', 'ZBTB16 (7704)', 'ZBTB17 (7709)', 'ZBTB18 (10472)', 'ZBTB2 (57621)', 'ZBTB20 (26137)', 'ZBTB21 (49854)', 'ZBTB22 (9278)', 'ZBTB24 (9841)', 'ZBTB25 (7597)', 'ZBTB26 (57684)', 'ZBTB3 (79842)', 'ZBTB32 (27033)', 'ZBTB33 (10009)', 'ZBTB34 (403341)', 'ZBTB37 (84614)', 'ZBTB38 (253461)', 'ZBTB39 (9880)', 'ZBTB4 (57659)', 'ZBTB40 (9923)', 'ZBTB41 (360023)', 'ZBTB42 (100128927)', 'ZBTB43 (23099)', 'ZBTB44 (29068)', 'ZBTB45 (84878)', 'ZBTB46 (140685)', 'ZBTB47 (92999)', 'ZBTB48 (3104)', 'ZBTB49 (166793)', 'ZBTB5 (9925)', 'ZBTB6 (10773)', 'ZBTB7A (51341)', 'ZBTB7B (51043)', 'ZBTB7C (201501)', 'ZBTB8A (653121)', 'ZBTB8B (728116)', 'ZBTB8OS (339487)', 'ZBTB9 (221504)', 'ZC2HC1A (51101)', 'ZC2HC1B (153918)', 'ZC2HC1C (79696)', 'ZC3H10 (84872)', 'ZC3H11A (9877)', 'ZC3H12A (80149)', 'ZC3H12B (340554)', 'ZC3H12C (85463)', 'ZC3H13 (23091)', 'ZC3H14 (79882)', 'ZC3H15 (55854)', 'ZC3H18 (124245)', 'ZC3H3 (23144)', 'ZC3H4 (23211)', 'ZC3H6 (376940)', 'ZC3H7A (29066)', 'ZC3H7B (23264)', 'ZC3H8 (84524)', 'ZC3HAV1 (56829)', 'ZC3HAV1L (92092)', 'ZC3HC1 (51530)', 'ZC4H2 (55906)', 'ZCCHC10 (54819)', 'ZCCHC12 (170261)', 'ZCCHC13 (389874)', 'ZCCHC14 (23174)', 'ZCCHC17 (51538)', 'ZCCHC18 (644353)', 'ZCCHC2 (54877)', 'ZCCHC24 (219654)', 'ZCCHC3 (85364)', 'ZCCHC4 (29063)', 'ZCCHC7 (84186)', 'ZCCHC9 (84240)', 'ZCRB1 (85437)', 'ZCWPW1 (55063)', 'ZCWPW2 (152098)', 'ZDBF2 (57683)', 'ZDHHC1 (29800)', 'ZDHHC11 (79844)', 'ZDHHC12 (84885)', 'ZDHHC13 (54503)', 'ZDHHC14 (79683)', 'ZDHHC15 (158866)', 'ZDHHC16 (84287)', 'ZDHHC17 (23390)', 'ZDHHC18 (84243)', 'ZDHHC19 (131540)', 'ZDHHC2 (51201)', 'ZDHHC20 (253832)', 'ZDHHC21 (340481)', 'ZDHHC22 (283576)', 'ZDHHC23 (254887)', 'ZDHHC24 (254359)', 'ZDHHC3 (51304)', 'ZDHHC4 (55146)', 'ZDHHC5 (25921)', 'ZDHHC6 (64429)', 'ZDHHC7 (55625)', 'ZDHHC8 (29801)', 'ZDHHC9 (51114)', 'ZEB1 (6935)', 'ZEB2 (9839)', 'ZER1 (10444)', 'ZFAND1 (79752)', 'ZFAND2A (90637)', 'ZFAND2B (130617)', 'ZFAND3 (60685)', 'ZFAND4 (93550)', 'ZFAND5 (7763)', 'ZFAND6 (54469)', 'ZFAT (57623)', 'ZFC3H1 (196441)', 'ZFHX2 (85446)', 'ZFHX3 (463)', 'ZFHX4 (79776)', 'ZFP1 (162239)', 'ZFP14 (57677)', 'ZFP2 (80108)', 'ZFP28 (140612)', 'ZFP3 (124961)', 'ZFP30 (22835)', 'ZFP36 (7538)', 'ZFP36L1 (677)', 'ZFP36L2 (678)', 'ZFP37 (7539)', 'ZFP41 (286128)', 'ZFP42 (132625)', 'ZFP57 (346171)', 'ZFP62 (643836)', 'ZFP64 (55734)', 'ZFP69 (339559)', 'ZFP69B (65243)', 'ZFP82 (284406)', 'ZFP90 (146198)', 'ZFP91 (80829)', 'ZFP92 (139735)', 'ZFPL1 (7542)', 'ZFPM1 (161882)', 'ZFPM2 (23414)', 'ZFR (51663)', 'ZFR2 (23217)', 'ZFX (7543)', 'ZFY (7544)', 'ZFYVE1 (53349)', 'ZFYVE16 (9765)', 'ZFYVE19 (84936)', 'ZFYVE21 (79038)', 'ZFYVE26 (23503)', 'ZFYVE27 (118813)', 'ZFYVE28 (57732)', 'ZFYVE9 (9372)', 'ZG16 (653808)', 'ZG16B (124220)', 'ZGLP1 (100125288)', 'ZGPAT (84619)', 'ZGRF1 (55345)', 'ZHX1 (11244)', 'ZHX2 (22882)', 'ZHX3 (23051)', 'ZIC1 (7545)', 'ZIC2 (7546)', 'ZIC3 (7547)', 'ZIC4 (84107)', 'ZIC5 (85416)', 'ZIK1 (284307)', 'ZIM2 (23619)', 'ZIM3 (114026)', 'ZKSCAN1 (7586)', 'ZKSCAN2 (342357)', 'ZKSCAN3 (80317)', 'ZKSCAN4 (387032)', 'ZKSCAN5 (23660)', 'ZKSCAN7 (55888)', 'ZKSCAN8 (7745)', 'ZMAT1 (84460)', 'ZMAT2 (153527)', 'ZMAT3 (64393)', 'ZMAT4 (79698)', 'ZMAT5 (55954)', 'ZMIZ1 (57178)', 'ZMIZ2 (83637)', 'ZMPSTE24 (10269)', 'ZMYM1 (79830)', 'ZMYM2 (7750)', 'ZMYM3 (9203)', 'ZMYM4 (9202)', 'ZMYM5 (9205)', 'ZMYM6 (9204)', 'ZMYND10 (51364)', 'ZMYND11 (10771)', 'ZMYND12 (84217)', 'ZMYND15 (84225)', 'ZMYND19 (116225)', 'ZMYND8 (23613)', 'ZNF10 (7556)', 'ZNF100 (163227)', 'ZNF101 (94039)', 'ZNF106 (64397)', 'ZNF107 (51427)', 'ZNF112 (7771)', 'ZNF114 (163071)', 'ZNF12 (7559)', 'ZNF121 (7675)', 'ZNF124 (7678)', 'ZNF131 (7690)', 'ZNF132 (7691)', 'ZNF133 (7692)', 'ZNF134 (7693)', 'ZNF135 (7694)', 'ZNF136 (7695)', 'ZNF138 (7697)', 'ZNF14 (7561)', 'ZNF140 (7699)', 'ZNF141 (7700)', 'ZNF142 (7701)', 'ZNF143 (7702)', 'ZNF146 (7705)', 'ZNF148 (7707)', 'ZNF154 (7710)', 'ZNF155 (7711)', 'ZNF157 (7712)', 'ZNF16 (7564)', 'ZNF160 (90338)', 'ZNF165 (7718)', 'ZNF169 (169841)', 'ZNF17 (7565)', 'ZNF174 (7727)', 'ZNF175 (7728)', 'ZNF177 (7730)', 'ZNF18 (7566)', 'ZNF180 (7733)', 'ZNF181 (339318)', 'ZNF182 (7569)', 'ZNF184 (7738)', 'ZNF185 (7739)', 'ZNF189 (7743)', 'ZNF19 (7567)', 'ZNF195 (7748)', 'ZNF2 (7549)', 'ZNF20 (7568)', 'ZNF200 (7752)', 'ZNF202 (7753)', 'ZNF205 (7755)', 'ZNF207 (7756)', 'ZNF208 (7757)', 'ZNF211 (10520)', 'ZNF212 (7988)', 'ZNF213 (7760)', 'ZNF214 (7761)', 'ZNF215 (7762)', 'ZNF217 (7764)', 'ZNF219 (51222)', 'ZNF22 (7570)', 'ZNF221 (7638)', 'ZNF222 (7673)', 'ZNF223 (7766)', 'ZNF224 (7767)', 'ZNF225 (7768)', 'ZNF226 (7769)', 'ZNF227 (7770)', 'ZNF229 (7772)', 'ZNF23 (7571)', 'ZNF230 (7773)', 'ZNF232 (7775)', 'ZNF233 (353355)', 'ZNF234 (10780)', 'ZNF235 (9310)', 'ZNF236 (7776)', 'ZNF239 (8187)', 'ZNF24 (7572)', 'ZNF248 (57209)', 'ZNF25 (219749)', 'ZNF250 (58500)', 'ZNF251 (90987)', 'ZNF253 (56242)', 'ZNF254 (9534)', 'ZNF256 (10172)', 'ZNF257 (113835)', 'ZNF26 (7574)', 'ZNF260 (339324)', 'ZNF263 (10127)', 'ZNF264 (9422)', 'ZNF266 (10781)', 'ZNF267 (10308)', 'ZNF268 (10795)', 'ZNF273 (10793)', 'ZNF276 (92822)', 'ZNF277 (11179)', 'ZNF28 (7576)', 'ZNF280A (129025)', 'ZNF280B (140883)', 'ZNF280C (55609)', 'ZNF280D (54816)', 'ZNF281 (23528)', 'ZNF282 (8427)', 'ZNF283 (284349)', 'ZNF284 (342909)', 'ZNF285 (26974)', 'ZNF286A (57335)', 'ZNF287 (57336)', 'ZNF292 (23036)', 'ZNF296 (162979)', 'ZNF3 (7551)', 'ZNF30 (90075)', 'ZNF300 (91975)', 'ZNF302 (55900)', 'ZNF304 (57343)', 'ZNF311 (282890)', 'ZNF317 (57693)', 'ZNF318 (24149)', 'ZNF319 (57567)', 'ZNF32 (7580)', 'ZNF320 (162967)', 'ZNF322 (79692)', 'ZNF324 (25799)', 'ZNF324B (388569)', 'ZNF326 (284695)', 'ZNF329 (79673)', 'ZNF330 (27309)', 'ZNF331 (55422)', 'ZNF333 (84449)', 'ZNF334 (55713)', 'ZNF335 (63925)', 'ZNF337 (26152)', 'ZNF33A (7581)', 'ZNF33B (7582)', 'ZNF34 (80778)', 'ZNF341 (84905)', 'ZNF343 (79175)', 'ZNF345 (25850)', 'ZNF346 (23567)', 'ZNF347 (84671)', 'ZNF35 (7584)', 'ZNF350 (59348)', 'ZNF354A (6940)', 'ZNF354B (117608)', 'ZNF354C (30832)', 'ZNF358 (140467)', 'ZNF362 (149076)', 'ZNF365 (22891)', 'ZNF366 (167465)', 'ZNF367 (195828)', 'ZNF37A (7587)', 'ZNF382 (84911)', 'ZNF383 (163087)', 'ZNF384 (171017)', 'ZNF385A (25946)', 'ZNF385B (151126)', 'ZNF385D (79750)', 'ZNF391 (346157)', 'ZNF394 (84124)', 'ZNF395 (55893)', 'ZNF396 (252884)', 'ZNF397 (84307)', 'ZNF398 (57541)', 'ZNF404 (342908)', 'ZNF407 (55628)', 'ZNF408 (79797)', 'ZNF41 (7592)', 'ZNF410 (57862)', 'ZNF414 (84330)', 'ZNF415 (55786)', 'ZNF416 (55659)', 'ZNF417 (147687)', 'ZNF418 (147686)', 'ZNF419 (79744)', 'ZNF420 (147923)', 'ZNF423 (23090)', 'ZNF425 (155054)', 'ZNF426 (79088)', 'ZNF428 (126299)', 'ZNF429 (353088)', 'ZNF43 (7594)', 'ZNF430 (80264)', 'ZNF432 (9668)', 'ZNF433 (163059)', 'ZNF436 (80818)', 'ZNF438 (220929)', 'ZNF439 (90594)', 'ZNF44 (51710)', 'ZNF440 (126070)', 'ZNF441 (126068)', 'ZNF442 (79973)', 'ZNF443 (10224)', 'ZNF444 (55311)', 'ZNF445 (353274)', 'ZNF446 (55663)', 'ZNF449 (203523)', 'ZNF45 (7596)', 'ZNF451 (26036)', 'ZNF454 (285676)', 'ZNF460 (10794)', 'ZNF461 (92283)', 'ZNF462 (58499)', 'ZNF467 (168544)', 'ZNF468 (90333)', 'ZNF469 (84627)', 'ZNF470 (388566)', 'ZNF471 (57573)', 'ZNF473 (25888)', 'ZNF474 (133923)', 'ZNF479 (90827)', 'ZNF48 (197407)', 'ZNF480 (147657)', 'ZNF483 (158399)', 'ZNF484 (83744)', 'ZNF485 (220992)', 'ZNF486 (90649)', 'ZNF488 (118738)', 'ZNF490 (57474)', 'ZNF491 (126069)', 'ZNF492 (57615)', 'ZNF493 (284443)', 'ZNF496 (84838)', 'ZNF497 (162968)', 'ZNF500 (26048)', 'ZNF501 (115560)', 'ZNF502 (91392)', 'ZNF503 (84858)', 'ZNF506 (440515)', 'ZNF507 (22847)', 'ZNF510 (22869)', 'ZNF511 (118472)', 'ZNF512 (84450)', 'ZNF512B (57473)', 'ZNF513 (130557)', 'ZNF514 (84874)', 'ZNF517 (340385)', 'ZNF518B (85460)', 'ZNF519 (162655)', 'ZNF521 (25925)', 'ZNF524 (147807)', 'ZNF526 (116115)', 'ZNF527 (84503)', 'ZNF528 (84436)', 'ZNF529 (57711)', 'ZNF530 (348327)', 'ZNF532 (55205)', 'ZNF534 (147658)', 'ZNF536 (9745)', 'ZNF540 (163255)', 'ZNF541 (84215)', 'ZNF543 (125919)', 'ZNF544 (27300)', 'ZNF546 (339327)', 'ZNF547 (284306)', 'ZNF548 (147694)', 'ZNF549 (256051)', 'ZNF550 (162972)', 'ZNF551 (90233)', 'ZNF552 (79818)', 'ZNF554 (115196)', 'ZNF555 (148254)', 'ZNF556 (80032)', 'ZNF557 (79230)', 'ZNF558 (148156)', 'ZNF559 (84527)', 'ZNF560 (147741)', 'ZNF561 (93134)', 'ZNF562 (54811)', 'ZNF563 (147837)', 'ZNF564 (163050)', 'ZNF565 (147929)', 'ZNF566 (84924)', 'ZNF567 (163081)', 'ZNF568 (374900)', 'ZNF569 (148266)', 'ZNF57 (126295)', 'ZNF570 (148268)', 'ZNF571 (51276)', 'ZNF572 (137209)', 'ZNF573 (126231)', 'ZNF574 (64763)', 'ZNF575 (284346)', 'ZNF576 (79177)', 'ZNF577 (84765)', 'ZNF578 (147660)', 'ZNF579 (163033)', 'ZNF580 (51157)', 'ZNF581 (51545)', 'ZNF582 (147948)', 'ZNF583 (147949)', 'ZNF584 (201514)', 'ZNF585A (199704)', 'ZNF585B (92285)', 'ZNF586 (54807)', 'ZNF587 (84914)', 'ZNF587B (100293516)', 'ZNF589 (51385)', 'ZNF592 (9640)', 'ZNF593 (51042)', 'ZNF594 (84622)', 'ZNF596 (169270)', 'ZNF597 (146434)', 'ZNF599 (148103)', 'ZNF600 (162966)', 'ZNF605 (100289635)', 'ZNF606 (80095)', 'ZNF607 (84775)', 'ZNF608 (57507)', 'ZNF609 (23060)', 'ZNF610 (162963)', 'ZNF611 (81856)', 'ZNF613 (79898)', 'ZNF614 (80110)', 'ZNF615 (284370)', 'ZNF616 (90317)', 'ZNF618 (114991)', 'ZNF619 (285267)', 'ZNF620 (253639)', 'ZNF621 (285268)', 'ZNF622 (90441)', 'ZNF623 (9831)', 'ZNF624 (57547)', 'ZNF625 (90589)', 'ZNF626 (199777)', 'ZNF627 (199692)', 'ZNF628 (89887)', 'ZNF629 (23361)', 'ZNF630 (57232)', 'ZNF638 (27332)', 'ZNF639 (51193)', 'ZNF641 (121274)', 'ZNF644 (84146)', 'ZNF646 (9726)', 'ZNF648 (127665)', 'ZNF649 (65251)', 'ZNF652 (22834)', 'ZNF653 (115950)', 'ZNF654 (55279)', 'ZNF655 (79027)', 'ZNF658 (26149)', 'ZNF660 (285349)', 'ZNF662 (389114)', 'ZNF664 (144348)', 'ZNF665 (79788)', 'ZNF667 (63934)', 'ZNF668 (79759)', 'ZNF669 (79862)', 'ZNF670 (93474)', 'ZNF671 (79891)', 'ZNF672 (79894)', 'ZNF674 (641339)', 'ZNF675 (171392)', 'ZNF676 (163223)', 'ZNF677 (342926)', 'ZNF679 (168417)', 'ZNF680 (340252)', 'ZNF681 (148213)', 'ZNF682 (91120)', 'ZNF683 (257101)', 'ZNF684 (127396)', 'ZNF687 (57592)', 'ZNF688 (146542)', 'ZNF689 (115509)', 'ZNF69 (7620)', 'ZNF691 (51058)', 'ZNF692 (55657)', 'ZNF695 (57116)', 'ZNF696 (79943)', 'ZNF697 (90874)', 'ZNF699 (374879)', 'ZNF7 (7553)', 'ZNF70 (7621)', 'ZNF700 (90592)', 'ZNF701 (55762)', 'ZNF703 (80139)', 'ZNF704 (619279)', 'ZNF705B (100132396)', 'ZNF705G (100131980)', 'ZNF706 (51123)', 'ZNF707 (286075)', 'ZNF708 (7562)', 'ZNF709 (163051)', 'ZNF71 (58491)', 'ZNF710 (374655)', 'ZNF711 (7552)', 'ZNF713 (349075)', 'ZNF714 (148206)', 'ZNF716 (441234)', 'ZNF721 (170960)', 'ZNF726 (730087)', 'ZNF727 (442319)', 'ZNF728 (388523)', 'ZNF729 (100287226)', 'ZNF730 (100129543)', 'ZNF732 (654254)', 'ZNF736 (728927)', 'ZNF737 (100129842)', 'ZNF74 (7625)', 'ZNF740 (283337)', 'ZNF746 (155061)', 'ZNF747 (65988)', 'ZNF749 (388567)', 'ZNF750 (79755)', 'ZNF75A (7627)', 'ZNF75D (7626)', 'ZNF76 (7629)', 'ZNF763 (284390)', 'ZNF764 (92595)', 'ZNF765 (91661)', 'ZNF766 (90321)', 'ZNF768 (79724)', 'ZNF77 (58492)', 'ZNF770 (54989)', 'ZNF771 (51333)', 'ZNF772 (400720)', 'ZNF773 (374928)', 'ZNF774 (342132)', 'ZNF775 (285971)', 'ZNF776 (284309)', 'ZNF777 (27153)', 'ZNF778 (197320)', 'ZNF780A (284323)', 'ZNF780B (163131)', 'ZNF782 (158431)', 'ZNF783 (100289678)', 'ZNF784 (147808)', 'ZNF785 (146540)', 'ZNF786 (136051)', 'ZNF787 (126208)', 'ZNF789 (285989)', 'ZNF79 (7633)', 'ZNF790 (388536)', 'ZNF791 (163049)', 'ZNF792 (126375)', 'ZNF793 (390927)', 'ZNF799 (90576)', 'ZNF8 (7554)', 'ZNF80 (7634)', 'ZNF800 (168850)', 'ZNF804A (91752)', 'ZNF804B (219578)', 'ZNF805 (390980)', 'ZNF808 (388558)', 'ZNF81 (347344)', 'ZNF813 (126017)', 'ZNF814 (730051)', 'ZNF816 (125893)', 'ZNF821 (55565)', 'ZNF823 (55552)', 'ZNF827 (152485)', 'ZNF829 (374899)', 'ZNF83 (55769)', 'ZNF830 (91603)', 'ZNF831 (128611)', 'ZNF835 (90485)', 'ZNF836 (162962)', 'ZNF837 (116412)', 'ZNF839 (55778)', 'ZNF84 (7637)', 'ZNF841 (284371)', 'ZNF843 (283933)', 'ZNF844 (284391)', 'ZNF845 (91664)', 'ZNF846 (162993)', 'ZNF85 (7639)', 'ZNF850 (342892)', 'ZNF853 (54753)', 'ZNF860 (344787)', 'ZNF862 (643641)', 'ZNF865 (100507290)', 'ZNF875 (284459)', 'ZNF878 (729747)', 'ZNF879 (345462)', 'ZNF880 (400713)', 'ZNF891 (101060200)', 'ZNF90 (7643)', 'ZNF91 (7644)', 'ZNF92 (168374)', 'ZNF93 (81931)', 'ZNF98 (148198)', 'ZNF99 (7652)', 'ZNFX1 (57169)', 'ZNG1A (55871)', 'ZNG1B (150472)', 'ZNHIT1 (10467)', 'ZNHIT2 (741)', 'ZNHIT3 (9326)', 'ZNHIT6 (54680)', 'ZNRD2 (10534)', 'ZNRF1 (84937)', 'ZNRF2 (223082)', 'ZNRF3 (84133)', 'ZNRF4 (148066)', 'ZP1 (22917)', 'ZP2 (7783)', 'ZP3 (7784)', 'ZP4 (57829)', 'ZPBP (11055)', 'ZPBP2 (124626)', 'ZPLD1 (131368)', 'ZPR1 (8882)', 'ZRANB1 (54764)', 'ZRANB2 (9406)', 'ZRANB3 (84083)', 'ZRSR2 (8233)', 'ZSCAN1 (284312)', 'ZSCAN10 (84891)', 'ZSCAN16 (80345)', 'ZSCAN18 (65982)', 'ZSCAN2 (54993)', 'ZSCAN20 (7579)', 'ZSCAN21 (7589)', 'ZSCAN22 (342945)', 'ZSCAN23 (222696)', 'ZSCAN25 (221785)', 'ZSCAN29 (146050)', 'ZSCAN30 (100101467)', 'ZSCAN31 (64288)', 'ZSCAN32 (54925)', 'ZSCAN4 (201516)', 'ZSCAN5A (79149)', 'ZSCAN5B (342933)', 'ZSCAN9 (7746)', 'ZSWIM1 (90204)', 'ZSWIM2 (151112)', 'ZSWIM3 (140831)', 'ZSWIM4 (65249)', 'ZSWIM5 (57643)', 'ZSWIM6 (57688)', 'ZSWIM7 (125150)', 'ZSWIM8 (23053)', 'ZUP1 (221302)', 'ZW10 (9183)', 'ZWILCH (55055)', 'ZWINT (11130)', 'ZXDA (7789)', 'ZXDB (158586)', 'ZXDC (79364)', 'ZYG11A (440590)', 'ZYG11B (79699)', 'ZYX (7791)', 'ZZEF1 (23140)', 'ZZZ3 (26009)']\n"
     ]
    }
   ],
   "source": [
    "# get the column names of the raw_crispr_df aside from the first column and convert this to a list\n",
    "raw_crispr_df_columns = raw_crispr_df.columns[1:].tolist()\n",
    "print(raw_crispr_df_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-002806</th>\n",
       "      <th>ACH-002834</th>\n",
       "      <th>ACH-002847</th>\n",
       "      <th>ACH-002922</th>\n",
       "      <th>ACH-002925</th>\n",
       "      <th>ACH-002926</th>\n",
       "      <th>ACH-002928</th>\n",
       "      <th>ACH-003012</th>\n",
       "      <th>ACH-003177</th>\n",
       "      <th>ACH-003180</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>0.023031</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>-0.071322</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.051500</td>\n",
       "      <td>-0.072687</td>\n",
       "      <td>-0.057531</td>\n",
       "      <td>-0.050743</td>\n",
       "      <td>-0.163957</td>\n",
       "      <td>-0.107883</td>\n",
       "      <td>-0.287606</td>\n",
       "      <td>-0.145878</td>\n",
       "      <td>-0.038435</td>\n",
       "      <td>-0.007314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.042647</td>\n",
       "      <td>-0.088500</td>\n",
       "      <td>0.003300</td>\n",
       "      <td>-0.113717</td>\n",
       "      <td>-0.134669</td>\n",
       "      <td>-0.053814</td>\n",
       "      <td>-0.024648</td>\n",
       "      <td>-0.139422</td>\n",
       "      <td>-0.162024</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.117224</td>\n",
       "      <td>-0.107839</td>\n",
       "      <td>-0.120021</td>\n",
       "      <td>-0.003851</td>\n",
       "      <td>-0.042068</td>\n",
       "      <td>-0.101501</td>\n",
       "      <td>-0.179101</td>\n",
       "      <td>-0.004000</td>\n",
       "      <td>-0.055660</td>\n",
       "      <td>-0.212478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.026102</td>\n",
       "      <td>-0.088198</td>\n",
       "      <td>0.160881</td>\n",
       "      <td>0.082872</td>\n",
       "      <td>0.060323</td>\n",
       "      <td>0.065476</td>\n",
       "      <td>0.034448</td>\n",
       "      <td>-0.039279</td>\n",
       "      <td>0.109822</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002730</td>\n",
       "      <td>0.034674</td>\n",
       "      <td>0.005217</td>\n",
       "      <td>-0.029880</td>\n",
       "      <td>0.118830</td>\n",
       "      <td>0.046677</td>\n",
       "      <td>-0.135824</td>\n",
       "      <td>0.091933</td>\n",
       "      <td>0.018989</td>\n",
       "      <td>0.129900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>-0.147614</td>\n",
       "      <td>-0.041225</td>\n",
       "      <td>0.086224</td>\n",
       "      <td>0.099633</td>\n",
       "      <td>0.076647</td>\n",
       "      <td>0.222308</td>\n",
       "      <td>0.060509</td>\n",
       "      <td>0.021942</td>\n",
       "      <td>0.076680</td>\n",
       "      <td>...</td>\n",
       "      <td>0.070590</td>\n",
       "      <td>0.000908</td>\n",
       "      <td>0.027493</td>\n",
       "      <td>0.243014</td>\n",
       "      <td>-0.042774</td>\n",
       "      <td>0.096868</td>\n",
       "      <td>-0.046579</td>\n",
       "      <td>0.191332</td>\n",
       "      <td>0.117182</td>\n",
       "      <td>-0.023112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>-0.020425</td>\n",
       "      <td>-0.028074</td>\n",
       "      <td>-0.149315</td>\n",
       "      <td>-0.008378</td>\n",
       "      <td>0.078922</td>\n",
       "      <td>-0.119423</td>\n",
       "      <td>0.046351</td>\n",
       "      <td>-0.089072</td>\n",
       "      <td>-0.013627</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.152683</td>\n",
       "      <td>-0.187834</td>\n",
       "      <td>-0.108456</td>\n",
       "      <td>-0.197960</td>\n",
       "      <td>-0.090302</td>\n",
       "      <td>-0.118650</td>\n",
       "      <td>-0.249888</td>\n",
       "      <td>-0.065385</td>\n",
       "      <td>0.063397</td>\n",
       "      <td>-0.068613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17911</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>-0.249508</td>\n",
       "      <td>0.054335</td>\n",
       "      <td>-0.205605</td>\n",
       "      <td>-0.081863</td>\n",
       "      <td>-0.098271</td>\n",
       "      <td>0.031686</td>\n",
       "      <td>0.028423</td>\n",
       "      <td>-0.099067</td>\n",
       "      <td>-0.015455</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.033863</td>\n",
       "      <td>0.011158</td>\n",
       "      <td>-0.091727</td>\n",
       "      <td>-0.095130</td>\n",
       "      <td>-0.046472</td>\n",
       "      <td>-0.040264</td>\n",
       "      <td>0.039994</td>\n",
       "      <td>0.158702</td>\n",
       "      <td>-0.103513</td>\n",
       "      <td>0.040616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17912</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>-0.330343</td>\n",
       "      <td>-0.066032</td>\n",
       "      <td>-0.390748</td>\n",
       "      <td>-0.510495</td>\n",
       "      <td>-0.200174</td>\n",
       "      <td>-0.059623</td>\n",
       "      <td>-0.062914</td>\n",
       "      <td>-0.117128</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.135263</td>\n",
       "      <td>0.096450</td>\n",
       "      <td>-0.186492</td>\n",
       "      <td>-0.250546</td>\n",
       "      <td>-0.267548</td>\n",
       "      <td>-0.239763</td>\n",
       "      <td>0.098240</td>\n",
       "      <td>-0.213281</td>\n",
       "      <td>-0.119234</td>\n",
       "      <td>-0.378505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17913</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.067986</td>\n",
       "      <td>0.099067</td>\n",
       "      <td>-0.054518</td>\n",
       "      <td>-0.036547</td>\n",
       "      <td>0.052938</td>\n",
       "      <td>0.058936</td>\n",
       "      <td>-0.114437</td>\n",
       "      <td>-0.067185</td>\n",
       "      <td>-0.136238</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.202681</td>\n",
       "      <td>-0.311926</td>\n",
       "      <td>-0.188271</td>\n",
       "      <td>-0.115048</td>\n",
       "      <td>-0.240009</td>\n",
       "      <td>0.065168</td>\n",
       "      <td>-0.266654</td>\n",
       "      <td>-0.017884</td>\n",
       "      <td>-0.153744</td>\n",
       "      <td>-0.143920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17914</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.136285</td>\n",
       "      <td>0.274566</td>\n",
       "      <td>0.035579</td>\n",
       "      <td>-0.273129</td>\n",
       "      <td>0.018623</td>\n",
       "      <td>-0.186168</td>\n",
       "      <td>-0.117129</td>\n",
       "      <td>-0.111276</td>\n",
       "      <td>0.222765</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.087873</td>\n",
       "      <td>-0.094481</td>\n",
       "      <td>-0.148179</td>\n",
       "      <td>-0.032583</td>\n",
       "      <td>-0.041099</td>\n",
       "      <td>-0.144088</td>\n",
       "      <td>-0.115108</td>\n",
       "      <td>-0.195395</td>\n",
       "      <td>0.032584</td>\n",
       "      <td>-0.039985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17915</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-0.438536</td>\n",
       "      <td>0.001871</td>\n",
       "      <td>-0.150486</td>\n",
       "      <td>-0.382723</td>\n",
       "      <td>-0.258353</td>\n",
       "      <td>-0.300830</td>\n",
       "      <td>-0.150634</td>\n",
       "      <td>-0.463948</td>\n",
       "      <td>-0.070226</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.221564</td>\n",
       "      <td>-0.510165</td>\n",
       "      <td>-0.417894</td>\n",
       "      <td>-0.280342</td>\n",
       "      <td>-0.132683</td>\n",
       "      <td>-0.334424</td>\n",
       "      <td>-0.080421</td>\n",
       "      <td>-0.309021</td>\n",
       "      <td>-0.536071</td>\n",
       "      <td>-0.439905</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17916 rows × 1179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 gene_name  ACH-000001  ACH-000004  ACH-000005  ACH-000007  \\\n",
       "0               A1BG   -0.121964   -0.016484   -0.184847   -0.071921   \n",
       "1               A1CF    0.042647   -0.088500    0.003300   -0.113717   \n",
       "2                A2M    0.026102   -0.088198    0.160881    0.082872   \n",
       "3              A2ML1   -0.147614   -0.041225    0.086224    0.099633   \n",
       "4            A3GALT2   -0.020425   -0.028074   -0.149315   -0.008378   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "17911         ZYG11A   -0.249508    0.054335   -0.205605   -0.081863   \n",
       "17912         ZYG11B    0.053725   -0.330343   -0.066032   -0.390748   \n",
       "17913            ZYX    0.067986    0.099067   -0.054518   -0.036547   \n",
       "17914          ZZEF1    0.136285    0.274566    0.035579   -0.273129   \n",
       "17915           ZZZ3   -0.438536    0.001871   -0.150486   -0.382723   \n",
       "\n",
       "Unnamed: 0  ACH-000009  ACH-000011  ACH-000012  ACH-000013  ACH-000014  ...  \\\n",
       "0            -0.019163    0.023031   -0.139496    0.014120   -0.071322  ...   \n",
       "1            -0.134669   -0.053814   -0.024648   -0.139422   -0.162024  ...   \n",
       "2             0.060323    0.065476    0.034448   -0.039279    0.109822  ...   \n",
       "3             0.076647    0.222308    0.060509    0.021942    0.076680  ...   \n",
       "4             0.078922   -0.119423    0.046351   -0.089072   -0.013627  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "17911        -0.098271    0.031686    0.028423   -0.099067   -0.015455  ...   \n",
       "17912        -0.510495   -0.200174   -0.059623   -0.062914   -0.117128  ...   \n",
       "17913         0.052938    0.058936   -0.114437   -0.067185   -0.136238  ...   \n",
       "17914         0.018623   -0.186168   -0.117129   -0.111276    0.222765  ...   \n",
       "17915        -0.258353   -0.300830   -0.150634   -0.463948   -0.070226  ...   \n",
       "\n",
       "Unnamed: 0  ACH-002806  ACH-002834  ACH-002847  ACH-002922  ACH-002925  \\\n",
       "0            -0.051500   -0.072687   -0.057531   -0.050743   -0.163957   \n",
       "1            -0.117224   -0.107839   -0.120021   -0.003851   -0.042068   \n",
       "2             0.002730    0.034674    0.005217   -0.029880    0.118830   \n",
       "3             0.070590    0.000908    0.027493    0.243014   -0.042774   \n",
       "4            -0.152683   -0.187834   -0.108456   -0.197960   -0.090302   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "17911        -0.033863    0.011158   -0.091727   -0.095130   -0.046472   \n",
       "17912        -0.135263    0.096450   -0.186492   -0.250546   -0.267548   \n",
       "17913        -0.202681   -0.311926   -0.188271   -0.115048   -0.240009   \n",
       "17914        -0.087873   -0.094481   -0.148179   -0.032583   -0.041099   \n",
       "17915        -0.221564   -0.510165   -0.417894   -0.280342   -0.132683   \n",
       "\n",
       "Unnamed: 0  ACH-002926  ACH-002928  ACH-003012  ACH-003177  ACH-003180  \n",
       "0            -0.107883   -0.287606   -0.145878   -0.038435   -0.007314  \n",
       "1            -0.101501   -0.179101   -0.004000   -0.055660   -0.212478  \n",
       "2             0.046677   -0.135824    0.091933    0.018989    0.129900  \n",
       "3             0.096868   -0.046579    0.191332    0.117182   -0.023112  \n",
       "4            -0.118650   -0.249888   -0.065385    0.063397   -0.068613  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "17911        -0.040264    0.039994    0.158702   -0.103513    0.040616  \n",
       "17912        -0.239763    0.098240   -0.213281   -0.119234   -0.378505  \n",
       "17913         0.065168   -0.266654   -0.017884   -0.153744   -0.143920  \n",
       "17914        -0.144088   -0.115108   -0.195395    0.032584   -0.039985  \n",
       "17915        -0.334424   -0.080421   -0.309021   -0.536071   -0.439905  \n",
       "\n",
       "[17916 rows x 1179 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# First, set the \"Unnamed: 0\" column as the index:\n",
    "raw_crispr_t_df = raw_crispr_df.set_index(\"Unnamed: 0\")\n",
    "# Then transpose the DataFrame:\n",
    "raw_crispr_t_df = raw_crispr_t_df.transpose()\n",
    "# move the index to a column and rename it as the gene_name\n",
    "raw_crispr_t_df.reset_index(inplace=True)\n",
    "raw_crispr_t_df.rename(columns={'index': 'gene_name'}, inplace=True)\n",
    "# column gene_name should be kept with the content by removing \"()\" and remove and \" \" content\n",
    "raw_crispr_t_df['gene_name'] = raw_crispr_t_df['gene_name'].apply(lambda x: x.split('(')[0].strip())\n",
    "# Optionally, if you want to view the result:\n",
    "display(raw_crispr_t_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-002806</th>\n",
       "      <th>ACH-002834</th>\n",
       "      <th>ACH-002847</th>\n",
       "      <th>ACH-002922</th>\n",
       "      <th>ACH-002925</th>\n",
       "      <th>ACH-002926</th>\n",
       "      <th>ACH-002928</th>\n",
       "      <th>ACH-003012</th>\n",
       "      <th>ACH-003177</th>\n",
       "      <th>ACH-003180</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>0.023031</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>-0.071322</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.051500</td>\n",
       "      <td>-0.072687</td>\n",
       "      <td>-0.057531</td>\n",
       "      <td>-0.050743</td>\n",
       "      <td>-0.163957</td>\n",
       "      <td>-0.107883</td>\n",
       "      <td>-0.287606</td>\n",
       "      <td>-0.145878</td>\n",
       "      <td>-0.038435</td>\n",
       "      <td>-0.007314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.042647</td>\n",
       "      <td>-0.088500</td>\n",
       "      <td>0.003300</td>\n",
       "      <td>-0.113717</td>\n",
       "      <td>-0.134669</td>\n",
       "      <td>-0.053814</td>\n",
       "      <td>-0.024648</td>\n",
       "      <td>-0.139422</td>\n",
       "      <td>-0.162024</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.117224</td>\n",
       "      <td>-0.107839</td>\n",
       "      <td>-0.120021</td>\n",
       "      <td>-0.003851</td>\n",
       "      <td>-0.042068</td>\n",
       "      <td>-0.101501</td>\n",
       "      <td>-0.179101</td>\n",
       "      <td>-0.004000</td>\n",
       "      <td>-0.055660</td>\n",
       "      <td>-0.212478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.026102</td>\n",
       "      <td>-0.088198</td>\n",
       "      <td>0.160881</td>\n",
       "      <td>0.082872</td>\n",
       "      <td>0.060323</td>\n",
       "      <td>0.065476</td>\n",
       "      <td>0.034448</td>\n",
       "      <td>-0.039279</td>\n",
       "      <td>0.109822</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002730</td>\n",
       "      <td>0.034674</td>\n",
       "      <td>0.005217</td>\n",
       "      <td>-0.029880</td>\n",
       "      <td>0.118830</td>\n",
       "      <td>0.046677</td>\n",
       "      <td>-0.135824</td>\n",
       "      <td>0.091933</td>\n",
       "      <td>0.018989</td>\n",
       "      <td>0.129900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>-0.147614</td>\n",
       "      <td>-0.041225</td>\n",
       "      <td>0.086224</td>\n",
       "      <td>0.099633</td>\n",
       "      <td>0.076647</td>\n",
       "      <td>0.222308</td>\n",
       "      <td>0.060509</td>\n",
       "      <td>0.021942</td>\n",
       "      <td>0.076680</td>\n",
       "      <td>...</td>\n",
       "      <td>0.070590</td>\n",
       "      <td>0.000908</td>\n",
       "      <td>0.027493</td>\n",
       "      <td>0.243014</td>\n",
       "      <td>-0.042774</td>\n",
       "      <td>0.096868</td>\n",
       "      <td>-0.046579</td>\n",
       "      <td>0.191332</td>\n",
       "      <td>0.117182</td>\n",
       "      <td>-0.023112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>-0.020425</td>\n",
       "      <td>-0.028074</td>\n",
       "      <td>-0.149315</td>\n",
       "      <td>-0.008378</td>\n",
       "      <td>0.078922</td>\n",
       "      <td>-0.119423</td>\n",
       "      <td>0.046351</td>\n",
       "      <td>-0.089072</td>\n",
       "      <td>-0.013627</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.152683</td>\n",
       "      <td>-0.187834</td>\n",
       "      <td>-0.108456</td>\n",
       "      <td>-0.197960</td>\n",
       "      <td>-0.090302</td>\n",
       "      <td>-0.118650</td>\n",
       "      <td>-0.249888</td>\n",
       "      <td>-0.065385</td>\n",
       "      <td>0.063397</td>\n",
       "      <td>-0.068613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17911</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>-0.249508</td>\n",
       "      <td>0.054335</td>\n",
       "      <td>-0.205605</td>\n",
       "      <td>-0.081863</td>\n",
       "      <td>-0.098271</td>\n",
       "      <td>0.031686</td>\n",
       "      <td>0.028423</td>\n",
       "      <td>-0.099067</td>\n",
       "      <td>-0.015455</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.033863</td>\n",
       "      <td>0.011158</td>\n",
       "      <td>-0.091727</td>\n",
       "      <td>-0.095130</td>\n",
       "      <td>-0.046472</td>\n",
       "      <td>-0.040264</td>\n",
       "      <td>0.039994</td>\n",
       "      <td>0.158702</td>\n",
       "      <td>-0.103513</td>\n",
       "      <td>0.040616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17912</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>-0.330343</td>\n",
       "      <td>-0.066032</td>\n",
       "      <td>-0.390748</td>\n",
       "      <td>-0.510495</td>\n",
       "      <td>-0.200174</td>\n",
       "      <td>-0.059623</td>\n",
       "      <td>-0.062914</td>\n",
       "      <td>-0.117128</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.135263</td>\n",
       "      <td>0.096450</td>\n",
       "      <td>-0.186492</td>\n",
       "      <td>-0.250546</td>\n",
       "      <td>-0.267548</td>\n",
       "      <td>-0.239763</td>\n",
       "      <td>0.098240</td>\n",
       "      <td>-0.213281</td>\n",
       "      <td>-0.119234</td>\n",
       "      <td>-0.378505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17913</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.067986</td>\n",
       "      <td>0.099067</td>\n",
       "      <td>-0.054518</td>\n",
       "      <td>-0.036547</td>\n",
       "      <td>0.052938</td>\n",
       "      <td>0.058936</td>\n",
       "      <td>-0.114437</td>\n",
       "      <td>-0.067185</td>\n",
       "      <td>-0.136238</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.202681</td>\n",
       "      <td>-0.311926</td>\n",
       "      <td>-0.188271</td>\n",
       "      <td>-0.115048</td>\n",
       "      <td>-0.240009</td>\n",
       "      <td>0.065168</td>\n",
       "      <td>-0.266654</td>\n",
       "      <td>-0.017884</td>\n",
       "      <td>-0.153744</td>\n",
       "      <td>-0.143920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17914</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.136285</td>\n",
       "      <td>0.274566</td>\n",
       "      <td>0.035579</td>\n",
       "      <td>-0.273129</td>\n",
       "      <td>0.018623</td>\n",
       "      <td>-0.186168</td>\n",
       "      <td>-0.117129</td>\n",
       "      <td>-0.111276</td>\n",
       "      <td>0.222765</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.087873</td>\n",
       "      <td>-0.094481</td>\n",
       "      <td>-0.148179</td>\n",
       "      <td>-0.032583</td>\n",
       "      <td>-0.041099</td>\n",
       "      <td>-0.144088</td>\n",
       "      <td>-0.115108</td>\n",
       "      <td>-0.195395</td>\n",
       "      <td>0.032584</td>\n",
       "      <td>-0.039985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17915</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-0.438536</td>\n",
       "      <td>0.001871</td>\n",
       "      <td>-0.150486</td>\n",
       "      <td>-0.382723</td>\n",
       "      <td>-0.258353</td>\n",
       "      <td>-0.300830</td>\n",
       "      <td>-0.150634</td>\n",
       "      <td>-0.463948</td>\n",
       "      <td>-0.070226</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.221564</td>\n",
       "      <td>-0.510165</td>\n",
       "      <td>-0.417894</td>\n",
       "      <td>-0.280342</td>\n",
       "      <td>-0.132683</td>\n",
       "      <td>-0.334424</td>\n",
       "      <td>-0.080421</td>\n",
       "      <td>-0.309021</td>\n",
       "      <td>-0.536071</td>\n",
       "      <td>-0.439905</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17916 rows × 1179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 gene_name  ACH-000001  ACH-000004  ACH-000005  ACH-000007  \\\n",
       "0               A1BG   -0.121964   -0.016484   -0.184847   -0.071921   \n",
       "1               A1CF    0.042647   -0.088500    0.003300   -0.113717   \n",
       "2                A2M    0.026102   -0.088198    0.160881    0.082872   \n",
       "3              A2ML1   -0.147614   -0.041225    0.086224    0.099633   \n",
       "4            A3GALT2   -0.020425   -0.028074   -0.149315   -0.008378   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "17911         ZYG11A   -0.249508    0.054335   -0.205605   -0.081863   \n",
       "17912         ZYG11B    0.053725   -0.330343   -0.066032   -0.390748   \n",
       "17913            ZYX    0.067986    0.099067   -0.054518   -0.036547   \n",
       "17914          ZZEF1    0.136285    0.274566    0.035579   -0.273129   \n",
       "17915           ZZZ3   -0.438536    0.001871   -0.150486   -0.382723   \n",
       "\n",
       "Unnamed: 0  ACH-000009  ACH-000011  ACH-000012  ACH-000013  ACH-000014  ...  \\\n",
       "0            -0.019163    0.023031   -0.139496    0.014120   -0.071322  ...   \n",
       "1            -0.134669   -0.053814   -0.024648   -0.139422   -0.162024  ...   \n",
       "2             0.060323    0.065476    0.034448   -0.039279    0.109822  ...   \n",
       "3             0.076647    0.222308    0.060509    0.021942    0.076680  ...   \n",
       "4             0.078922   -0.119423    0.046351   -0.089072   -0.013627  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "17911        -0.098271    0.031686    0.028423   -0.099067   -0.015455  ...   \n",
       "17912        -0.510495   -0.200174   -0.059623   -0.062914   -0.117128  ...   \n",
       "17913         0.052938    0.058936   -0.114437   -0.067185   -0.136238  ...   \n",
       "17914         0.018623   -0.186168   -0.117129   -0.111276    0.222765  ...   \n",
       "17915        -0.258353   -0.300830   -0.150634   -0.463948   -0.070226  ...   \n",
       "\n",
       "Unnamed: 0  ACH-002806  ACH-002834  ACH-002847  ACH-002922  ACH-002925  \\\n",
       "0            -0.051500   -0.072687   -0.057531   -0.050743   -0.163957   \n",
       "1            -0.117224   -0.107839   -0.120021   -0.003851   -0.042068   \n",
       "2             0.002730    0.034674    0.005217   -0.029880    0.118830   \n",
       "3             0.070590    0.000908    0.027493    0.243014   -0.042774   \n",
       "4            -0.152683   -0.187834   -0.108456   -0.197960   -0.090302   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "17911        -0.033863    0.011158   -0.091727   -0.095130   -0.046472   \n",
       "17912        -0.135263    0.096450   -0.186492   -0.250546   -0.267548   \n",
       "17913        -0.202681   -0.311926   -0.188271   -0.115048   -0.240009   \n",
       "17914        -0.087873   -0.094481   -0.148179   -0.032583   -0.041099   \n",
       "17915        -0.221564   -0.510165   -0.417894   -0.280342   -0.132683   \n",
       "\n",
       "Unnamed: 0  ACH-002926  ACH-002928  ACH-003012  ACH-003177  ACH-003180  \n",
       "0            -0.107883   -0.287606   -0.145878   -0.038435   -0.007314  \n",
       "1            -0.101501   -0.179101   -0.004000   -0.055660   -0.212478  \n",
       "2             0.046677   -0.135824    0.091933    0.018989    0.129900  \n",
       "3             0.096868   -0.046579    0.191332    0.117182   -0.023112  \n",
       "4            -0.118650   -0.249888   -0.065385    0.063397   -0.068613  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "17911        -0.040264    0.039994    0.158702   -0.103513    0.040616  \n",
       "17912        -0.239763    0.098240   -0.213281   -0.119234   -0.378505  \n",
       "17913         0.065168   -0.266654   -0.017884   -0.153744   -0.143920  \n",
       "17914        -0.144088   -0.115108   -0.195395    0.032584   -0.039985  \n",
       "17915        -0.334424   -0.080421   -0.309021   -0.536071   -0.439905  \n",
       "\n",
       "[17916 rows x 1179 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\1869562081.py:8: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  crispr_df = raw_crispr_t_df.groupby('gene_name', as_index=False).mean()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-002806</th>\n",
       "      <th>ACH-002834</th>\n",
       "      <th>ACH-002847</th>\n",
       "      <th>ACH-002922</th>\n",
       "      <th>ACH-002925</th>\n",
       "      <th>ACH-002926</th>\n",
       "      <th>ACH-002928</th>\n",
       "      <th>ACH-003012</th>\n",
       "      <th>ACH-003177</th>\n",
       "      <th>ACH-003180</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>0.023031</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>-0.071322</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.051500</td>\n",
       "      <td>-0.072687</td>\n",
       "      <td>-0.057531</td>\n",
       "      <td>-0.050743</td>\n",
       "      <td>-0.163957</td>\n",
       "      <td>-0.107883</td>\n",
       "      <td>-0.287606</td>\n",
       "      <td>-0.145878</td>\n",
       "      <td>-0.038435</td>\n",
       "      <td>-0.007314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.042647</td>\n",
       "      <td>-0.088500</td>\n",
       "      <td>0.003300</td>\n",
       "      <td>-0.113717</td>\n",
       "      <td>-0.134669</td>\n",
       "      <td>-0.053814</td>\n",
       "      <td>-0.024648</td>\n",
       "      <td>-0.139422</td>\n",
       "      <td>-0.162024</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.117224</td>\n",
       "      <td>-0.107839</td>\n",
       "      <td>-0.120021</td>\n",
       "      <td>-0.003851</td>\n",
       "      <td>-0.042068</td>\n",
       "      <td>-0.101501</td>\n",
       "      <td>-0.179101</td>\n",
       "      <td>-0.004000</td>\n",
       "      <td>-0.055660</td>\n",
       "      <td>-0.212478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.026102</td>\n",
       "      <td>-0.088198</td>\n",
       "      <td>0.160881</td>\n",
       "      <td>0.082872</td>\n",
       "      <td>0.060323</td>\n",
       "      <td>0.065476</td>\n",
       "      <td>0.034448</td>\n",
       "      <td>-0.039279</td>\n",
       "      <td>0.109822</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002730</td>\n",
       "      <td>0.034674</td>\n",
       "      <td>0.005217</td>\n",
       "      <td>-0.029880</td>\n",
       "      <td>0.118830</td>\n",
       "      <td>0.046677</td>\n",
       "      <td>-0.135824</td>\n",
       "      <td>0.091933</td>\n",
       "      <td>0.018989</td>\n",
       "      <td>0.129900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>-0.147614</td>\n",
       "      <td>-0.041225</td>\n",
       "      <td>0.086224</td>\n",
       "      <td>0.099633</td>\n",
       "      <td>0.076647</td>\n",
       "      <td>0.222308</td>\n",
       "      <td>0.060509</td>\n",
       "      <td>0.021942</td>\n",
       "      <td>0.076680</td>\n",
       "      <td>...</td>\n",
       "      <td>0.070590</td>\n",
       "      <td>0.000908</td>\n",
       "      <td>0.027493</td>\n",
       "      <td>0.243014</td>\n",
       "      <td>-0.042774</td>\n",
       "      <td>0.096868</td>\n",
       "      <td>-0.046579</td>\n",
       "      <td>0.191332</td>\n",
       "      <td>0.117182</td>\n",
       "      <td>-0.023112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>-0.020425</td>\n",
       "      <td>-0.028074</td>\n",
       "      <td>-0.149315</td>\n",
       "      <td>-0.008378</td>\n",
       "      <td>0.078922</td>\n",
       "      <td>-0.119423</td>\n",
       "      <td>0.046351</td>\n",
       "      <td>-0.089072</td>\n",
       "      <td>-0.013627</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.152683</td>\n",
       "      <td>-0.187834</td>\n",
       "      <td>-0.108456</td>\n",
       "      <td>-0.197960</td>\n",
       "      <td>-0.090302</td>\n",
       "      <td>-0.118650</td>\n",
       "      <td>-0.249888</td>\n",
       "      <td>-0.065385</td>\n",
       "      <td>0.063397</td>\n",
       "      <td>-0.068613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17911</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>-0.249508</td>\n",
       "      <td>0.054335</td>\n",
       "      <td>-0.205605</td>\n",
       "      <td>-0.081863</td>\n",
       "      <td>-0.098271</td>\n",
       "      <td>0.031686</td>\n",
       "      <td>0.028423</td>\n",
       "      <td>-0.099067</td>\n",
       "      <td>-0.015455</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.033863</td>\n",
       "      <td>0.011158</td>\n",
       "      <td>-0.091727</td>\n",
       "      <td>-0.095130</td>\n",
       "      <td>-0.046472</td>\n",
       "      <td>-0.040264</td>\n",
       "      <td>0.039994</td>\n",
       "      <td>0.158702</td>\n",
       "      <td>-0.103513</td>\n",
       "      <td>0.040616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17912</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>-0.330343</td>\n",
       "      <td>-0.066032</td>\n",
       "      <td>-0.390748</td>\n",
       "      <td>-0.510495</td>\n",
       "      <td>-0.200174</td>\n",
       "      <td>-0.059623</td>\n",
       "      <td>-0.062914</td>\n",
       "      <td>-0.117128</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.135263</td>\n",
       "      <td>0.096450</td>\n",
       "      <td>-0.186492</td>\n",
       "      <td>-0.250546</td>\n",
       "      <td>-0.267548</td>\n",
       "      <td>-0.239763</td>\n",
       "      <td>0.098240</td>\n",
       "      <td>-0.213281</td>\n",
       "      <td>-0.119234</td>\n",
       "      <td>-0.378505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17913</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.067986</td>\n",
       "      <td>0.099067</td>\n",
       "      <td>-0.054518</td>\n",
       "      <td>-0.036547</td>\n",
       "      <td>0.052938</td>\n",
       "      <td>0.058936</td>\n",
       "      <td>-0.114437</td>\n",
       "      <td>-0.067185</td>\n",
       "      <td>-0.136238</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.202681</td>\n",
       "      <td>-0.311926</td>\n",
       "      <td>-0.188271</td>\n",
       "      <td>-0.115048</td>\n",
       "      <td>-0.240009</td>\n",
       "      <td>0.065168</td>\n",
       "      <td>-0.266654</td>\n",
       "      <td>-0.017884</td>\n",
       "      <td>-0.153744</td>\n",
       "      <td>-0.143920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17914</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.136285</td>\n",
       "      <td>0.274566</td>\n",
       "      <td>0.035579</td>\n",
       "      <td>-0.273129</td>\n",
       "      <td>0.018623</td>\n",
       "      <td>-0.186168</td>\n",
       "      <td>-0.117129</td>\n",
       "      <td>-0.111276</td>\n",
       "      <td>0.222765</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.087873</td>\n",
       "      <td>-0.094481</td>\n",
       "      <td>-0.148179</td>\n",
       "      <td>-0.032583</td>\n",
       "      <td>-0.041099</td>\n",
       "      <td>-0.144088</td>\n",
       "      <td>-0.115108</td>\n",
       "      <td>-0.195395</td>\n",
       "      <td>0.032584</td>\n",
       "      <td>-0.039985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17915</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-0.438536</td>\n",
       "      <td>0.001871</td>\n",
       "      <td>-0.150486</td>\n",
       "      <td>-0.382723</td>\n",
       "      <td>-0.258353</td>\n",
       "      <td>-0.300830</td>\n",
       "      <td>-0.150634</td>\n",
       "      <td>-0.463948</td>\n",
       "      <td>-0.070226</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.221564</td>\n",
       "      <td>-0.510165</td>\n",
       "      <td>-0.417894</td>\n",
       "      <td>-0.280342</td>\n",
       "      <td>-0.132683</td>\n",
       "      <td>-0.334424</td>\n",
       "      <td>-0.080421</td>\n",
       "      <td>-0.309021</td>\n",
       "      <td>-0.536071</td>\n",
       "      <td>-0.439905</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17916 rows × 1179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 gene_name  ACH-000001  ACH-000004  ACH-000005  ACH-000007  \\\n",
       "0               A1BG   -0.121964   -0.016484   -0.184847   -0.071921   \n",
       "1               A1CF    0.042647   -0.088500    0.003300   -0.113717   \n",
       "2                A2M    0.026102   -0.088198    0.160881    0.082872   \n",
       "3              A2ML1   -0.147614   -0.041225    0.086224    0.099633   \n",
       "4            A3GALT2   -0.020425   -0.028074   -0.149315   -0.008378   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "17911         ZYG11A   -0.249508    0.054335   -0.205605   -0.081863   \n",
       "17912         ZYG11B    0.053725   -0.330343   -0.066032   -0.390748   \n",
       "17913            ZYX    0.067986    0.099067   -0.054518   -0.036547   \n",
       "17914          ZZEF1    0.136285    0.274566    0.035579   -0.273129   \n",
       "17915           ZZZ3   -0.438536    0.001871   -0.150486   -0.382723   \n",
       "\n",
       "Unnamed: 0  ACH-000009  ACH-000011  ACH-000012  ACH-000013  ACH-000014  ...  \\\n",
       "0            -0.019163    0.023031   -0.139496    0.014120   -0.071322  ...   \n",
       "1            -0.134669   -0.053814   -0.024648   -0.139422   -0.162024  ...   \n",
       "2             0.060323    0.065476    0.034448   -0.039279    0.109822  ...   \n",
       "3             0.076647    0.222308    0.060509    0.021942    0.076680  ...   \n",
       "4             0.078922   -0.119423    0.046351   -0.089072   -0.013627  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "17911        -0.098271    0.031686    0.028423   -0.099067   -0.015455  ...   \n",
       "17912        -0.510495   -0.200174   -0.059623   -0.062914   -0.117128  ...   \n",
       "17913         0.052938    0.058936   -0.114437   -0.067185   -0.136238  ...   \n",
       "17914         0.018623   -0.186168   -0.117129   -0.111276    0.222765  ...   \n",
       "17915        -0.258353   -0.300830   -0.150634   -0.463948   -0.070226  ...   \n",
       "\n",
       "Unnamed: 0  ACH-002806  ACH-002834  ACH-002847  ACH-002922  ACH-002925  \\\n",
       "0            -0.051500   -0.072687   -0.057531   -0.050743   -0.163957   \n",
       "1            -0.117224   -0.107839   -0.120021   -0.003851   -0.042068   \n",
       "2             0.002730    0.034674    0.005217   -0.029880    0.118830   \n",
       "3             0.070590    0.000908    0.027493    0.243014   -0.042774   \n",
       "4            -0.152683   -0.187834   -0.108456   -0.197960   -0.090302   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "17911        -0.033863    0.011158   -0.091727   -0.095130   -0.046472   \n",
       "17912        -0.135263    0.096450   -0.186492   -0.250546   -0.267548   \n",
       "17913        -0.202681   -0.311926   -0.188271   -0.115048   -0.240009   \n",
       "17914        -0.087873   -0.094481   -0.148179   -0.032583   -0.041099   \n",
       "17915        -0.221564   -0.510165   -0.417894   -0.280342   -0.132683   \n",
       "\n",
       "Unnamed: 0  ACH-002926  ACH-002928  ACH-003012  ACH-003177  ACH-003180  \n",
       "0            -0.107883   -0.287606   -0.145878   -0.038435   -0.007314  \n",
       "1            -0.101501   -0.179101   -0.004000   -0.055660   -0.212478  \n",
       "2             0.046677   -0.135824    0.091933    0.018989    0.129900  \n",
       "3             0.096868   -0.046579    0.191332    0.117182   -0.023112  \n",
       "4            -0.118650   -0.249888   -0.065385    0.063397   -0.068613  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "17911        -0.040264    0.039994    0.158702   -0.103513    0.040616  \n",
       "17912        -0.239763    0.098240   -0.213281   -0.119234   -0.378505  \n",
       "17913         0.065168   -0.266654   -0.017884   -0.153744   -0.143920  \n",
       "17914        -0.144088   -0.115108   -0.195395    0.032584   -0.039985  \n",
       "17915        -0.334424   -0.080421   -0.309021   -0.536071   -0.439905  \n",
       "\n",
       "[17916 rows x 1179 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Identify columns to convert (all except 'gene_name')\n",
    "cols_to_convert = raw_crispr_t_df.columns.difference(['gene_name'])\n",
    "# Convert these columns to numeric (using pd.to_numeric for safety, which can handle errors)\n",
    "raw_crispr_t_df[cols_to_convert] = raw_crispr_t_df[cols_to_convert].apply(pd.to_numeric, errors='coerce')\n",
    "raw_crispr_t_df = raw_crispr_t_df.fillna(0.0)\n",
    "display(raw_crispr_t_df)\n",
    "# groupby the gene_name and calculate the mean\n",
    "crispr_df = raw_crispr_t_df.groupby('gene_name', as_index=False).mean()\n",
    "display(crispr_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\2142238278.py:1: DtypeWarning: Columns (4,5,6,11,12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmg_gene_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>HGNC_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>NAT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>ADA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>CDH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>AKT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_GN86234</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_GN86235</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_GN86236</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID HGNC_Symbol\n",
       "0               BMGC_GN00001        A1BG\n",
       "1               BMGC_GN00002        NAT2\n",
       "2               BMGC_GN00003         ADA\n",
       "3               BMGC_GN00004        CDH2\n",
       "4               BMGC_GN00005        AKT3\n",
       "...                      ...         ...\n",
       "86233           BMGC_GN86234         NaN\n",
       "86234           BMGC_GN86235         NaN\n",
       "86235           BMGC_GN86236         NaN\n",
       "86236           BMGC_GN86237         NaN\n",
       "86237           BMGC_GN86238         NaN\n",
       "\n",
       "[86238 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-002806</th>\n",
       "      <th>ACH-002834</th>\n",
       "      <th>ACH-002847</th>\n",
       "      <th>ACH-002922</th>\n",
       "      <th>ACH-002925</th>\n",
       "      <th>ACH-002926</th>\n",
       "      <th>ACH-002928</th>\n",
       "      <th>ACH-003012</th>\n",
       "      <th>ACH-003177</th>\n",
       "      <th>ACH-003180</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>0.023031</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>-0.071322</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.051500</td>\n",
       "      <td>-0.072687</td>\n",
       "      <td>-0.057531</td>\n",
       "      <td>-0.050743</td>\n",
       "      <td>-0.163957</td>\n",
       "      <td>-0.107883</td>\n",
       "      <td>-0.287606</td>\n",
       "      <td>-0.145878</td>\n",
       "      <td>-0.038435</td>\n",
       "      <td>-0.007314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>0.196378</td>\n",
       "      <td>0.359769</td>\n",
       "      <td>0.310169</td>\n",
       "      <td>0.198303</td>\n",
       "      <td>0.113667</td>\n",
       "      <td>0.157864</td>\n",
       "      <td>0.154384</td>\n",
       "      <td>0.300938</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131496</td>\n",
       "      <td>0.165664</td>\n",
       "      <td>0.156277</td>\n",
       "      <td>0.198406</td>\n",
       "      <td>0.209907</td>\n",
       "      <td>0.192162</td>\n",
       "      <td>0.068782</td>\n",
       "      <td>0.143209</td>\n",
       "      <td>0.074829</td>\n",
       "      <td>-0.262441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>-0.116327</td>\n",
       "      <td>-0.200383</td>\n",
       "      <td>-0.211542</td>\n",
       "      <td>-0.120459</td>\n",
       "      <td>-0.100498</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>0.084591</td>\n",
       "      <td>0.025797</td>\n",
       "      <td>-0.141183</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.155286</td>\n",
       "      <td>-0.027161</td>\n",
       "      <td>-0.107837</td>\n",
       "      <td>-0.023697</td>\n",
       "      <td>-0.347253</td>\n",
       "      <td>-0.092996</td>\n",
       "      <td>-0.246541</td>\n",
       "      <td>-0.065361</td>\n",
       "      <td>0.092299</td>\n",
       "      <td>-0.089303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>-0.019295</td>\n",
       "      <td>0.099193</td>\n",
       "      <td>-0.057229</td>\n",
       "      <td>-0.213060</td>\n",
       "      <td>-0.100742</td>\n",
       "      <td>-0.558269</td>\n",
       "      <td>-0.027933</td>\n",
       "      <td>0.033106</td>\n",
       "      <td>-0.120098</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.303145</td>\n",
       "      <td>-0.125616</td>\n",
       "      <td>-0.520756</td>\n",
       "      <td>-0.194251</td>\n",
       "      <td>0.065170</td>\n",
       "      <td>-0.098423</td>\n",
       "      <td>0.167748</td>\n",
       "      <td>0.053459</td>\n",
       "      <td>-0.204036</td>\n",
       "      <td>-0.120517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>-0.100900</td>\n",
       "      <td>0.151491</td>\n",
       "      <td>0.293747</td>\n",
       "      <td>0.191130</td>\n",
       "      <td>0.025839</td>\n",
       "      <td>0.127902</td>\n",
       "      <td>0.055322</td>\n",
       "      <td>0.074227</td>\n",
       "      <td>-0.331538</td>\n",
       "      <td>...</td>\n",
       "      <td>0.079258</td>\n",
       "      <td>0.102221</td>\n",
       "      <td>0.036739</td>\n",
       "      <td>0.099489</td>\n",
       "      <td>0.121007</td>\n",
       "      <td>0.165328</td>\n",
       "      <td>-0.097967</td>\n",
       "      <td>0.239757</td>\n",
       "      <td>0.193184</td>\n",
       "      <td>0.077518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20283</th>\n",
       "      <td>BMGC_GN69808</td>\n",
       "      <td>0.094246</td>\n",
       "      <td>0.021657</td>\n",
       "      <td>0.196685</td>\n",
       "      <td>0.392400</td>\n",
       "      <td>0.293312</td>\n",
       "      <td>0.404930</td>\n",
       "      <td>0.124325</td>\n",
       "      <td>0.173397</td>\n",
       "      <td>-0.021022</td>\n",
       "      <td>...</td>\n",
       "      <td>0.102379</td>\n",
       "      <td>0.197394</td>\n",
       "      <td>0.081715</td>\n",
       "      <td>0.114349</td>\n",
       "      <td>0.180772</td>\n",
       "      <td>0.017811</td>\n",
       "      <td>0.291685</td>\n",
       "      <td>0.200229</td>\n",
       "      <td>0.209375</td>\n",
       "      <td>0.096925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20284</th>\n",
       "      <td>BMGC_GN69841</td>\n",
       "      <td>-0.137149</td>\n",
       "      <td>-0.601468</td>\n",
       "      <td>-0.147378</td>\n",
       "      <td>-0.320952</td>\n",
       "      <td>0.022595</td>\n",
       "      <td>-0.610531</td>\n",
       "      <td>-0.075869</td>\n",
       "      <td>-0.091422</td>\n",
       "      <td>-0.258779</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.100619</td>\n",
       "      <td>-0.022613</td>\n",
       "      <td>0.084076</td>\n",
       "      <td>-0.180020</td>\n",
       "      <td>-0.017108</td>\n",
       "      <td>-0.325764</td>\n",
       "      <td>0.144469</td>\n",
       "      <td>0.009504</td>\n",
       "      <td>-0.118627</td>\n",
       "      <td>-0.348614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20285</th>\n",
       "      <td>BMGC_GN69848</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.333310</td>\n",
       "      <td>-0.151258</td>\n",
       "      <td>0.090807</td>\n",
       "      <td>0.130973</td>\n",
       "      <td>0.035430</td>\n",
       "      <td>0.057994</td>\n",
       "      <td>0.127841</td>\n",
       "      <td>-0.073217</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.097166</td>\n",
       "      <td>-0.125821</td>\n",
       "      <td>0.056233</td>\n",
       "      <td>-0.057404</td>\n",
       "      <td>0.140328</td>\n",
       "      <td>0.182520</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.198007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20286</th>\n",
       "      <td>BMGC_GN69876</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.316461</td>\n",
       "      <td>-0.021494</td>\n",
       "      <td>-0.556799</td>\n",
       "      <td>-0.350030</td>\n",
       "      <td>-0.136289</td>\n",
       "      <td>-0.579060</td>\n",
       "      <td>-0.200260</td>\n",
       "      <td>-0.171833</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.288444</td>\n",
       "      <td>-0.376535</td>\n",
       "      <td>-0.083581</td>\n",
       "      <td>-0.059077</td>\n",
       "      <td>-0.246367</td>\n",
       "      <td>-0.165150</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.199642</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20287</th>\n",
       "      <td>BMGC_GN70238</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.019036</td>\n",
       "      <td>-0.202153</td>\n",
       "      <td>-0.158997</td>\n",
       "      <td>0.331191</td>\n",
       "      <td>-0.110372</td>\n",
       "      <td>0.074430</td>\n",
       "      <td>0.147567</td>\n",
       "      <td>0.139878</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.243601</td>\n",
       "      <td>0.276185</td>\n",
       "      <td>0.135168</td>\n",
       "      <td>0.063313</td>\n",
       "      <td>-0.226531</td>\n",
       "      <td>0.050751</td>\n",
       "      <td>0.212870</td>\n",
       "      <td>-0.237297</td>\n",
       "      <td>0.123344</td>\n",
       "      <td>0.172626</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20288 rows × 1179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000001  ACH-000004  ACH-000005  ACH-000007  \\\n",
       "0               BMGC_GN00001   -0.121964   -0.016484   -0.184847   -0.071921   \n",
       "1               BMGC_GN00002   -0.022127    0.196378    0.359769    0.310169   \n",
       "2               BMGC_GN00003   -0.116327   -0.200383   -0.211542   -0.120459   \n",
       "3               BMGC_GN00004   -0.019295    0.099193   -0.057229   -0.213060   \n",
       "4               BMGC_GN00005   -0.100900    0.151491    0.293747    0.191130   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "20283           BMGC_GN69808    0.094246    0.021657    0.196685    0.392400   \n",
       "20284           BMGC_GN69841   -0.137149   -0.601468   -0.147378   -0.320952   \n",
       "20285           BMGC_GN69848    0.000000    0.333310   -0.151258    0.090807   \n",
       "20286           BMGC_GN69876    0.000000   -0.316461   -0.021494   -0.556799   \n",
       "20287           BMGC_GN70238    0.000000   -0.019036   -0.202153   -0.158997   \n",
       "\n",
       "       ACH-000009  ACH-000011  ACH-000012  ACH-000013  ACH-000014  ...  \\\n",
       "0       -0.019163    0.023031   -0.139496    0.014120   -0.071322  ...   \n",
       "1        0.198303    0.113667    0.157864    0.154384    0.300938  ...   \n",
       "2       -0.100498    0.000700    0.084591    0.025797   -0.141183  ...   \n",
       "3       -0.100742   -0.558269   -0.027933    0.033106   -0.120098  ...   \n",
       "4        0.025839    0.127902    0.055322    0.074227   -0.331538  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "20283    0.293312    0.404930    0.124325    0.173397   -0.021022  ...   \n",
       "20284    0.022595   -0.610531   -0.075869   -0.091422   -0.258779  ...   \n",
       "20285    0.130973    0.035430    0.057994    0.127841   -0.073217  ...   \n",
       "20286   -0.350030   -0.136289   -0.579060   -0.200260   -0.171833  ...   \n",
       "20287    0.331191   -0.110372    0.074430    0.147567    0.139878  ...   \n",
       "\n",
       "       ACH-002806  ACH-002834  ACH-002847  ACH-002922  ACH-002925  ACH-002926  \\\n",
       "0       -0.051500   -0.072687   -0.057531   -0.050743   -0.163957   -0.107883   \n",
       "1        0.131496    0.165664    0.156277    0.198406    0.209907    0.192162   \n",
       "2       -0.155286   -0.027161   -0.107837   -0.023697   -0.347253   -0.092996   \n",
       "3       -0.303145   -0.125616   -0.520756   -0.194251    0.065170   -0.098423   \n",
       "4        0.079258    0.102221    0.036739    0.099489    0.121007    0.165328   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "20283    0.102379    0.197394    0.081715    0.114349    0.180772    0.017811   \n",
       "20284   -0.100619   -0.022613    0.084076   -0.180020   -0.017108   -0.325764   \n",
       "20285   -0.097166   -0.125821    0.056233   -0.057404    0.140328    0.182520   \n",
       "20286   -0.288444   -0.376535   -0.083581   -0.059077   -0.246367   -0.165150   \n",
       "20287   -0.243601    0.276185    0.135168    0.063313   -0.226531    0.050751   \n",
       "\n",
       "       ACH-002928  ACH-003012  ACH-003177  ACH-003180  \n",
       "0       -0.287606   -0.145878   -0.038435   -0.007314  \n",
       "1        0.068782    0.143209    0.074829   -0.262441  \n",
       "2       -0.246541   -0.065361    0.092299   -0.089303  \n",
       "3        0.167748    0.053459   -0.204036   -0.120517  \n",
       "4       -0.097967    0.239757    0.193184    0.077518  \n",
       "...           ...         ...         ...         ...  \n",
       "20283    0.291685    0.200229    0.209375    0.096925  \n",
       "20284    0.144469    0.009504   -0.118627   -0.348614  \n",
       "20285    0.000000    0.198007    0.000000    0.000000  \n",
       "20286    0.000000   -0.199642    0.000000    0.000000  \n",
       "20287    0.212870   -0.237297    0.123344    0.172626  \n",
       "\n",
       "[20288 rows x 1179 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bmg_gene_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene.csv')\n",
    "# keep BioMedGraphica_ID and HGNC_Symbol\n",
    "bmg_gene_df = bmg_gene_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']]\n",
    "display(bmg_gene_df)\n",
    "# merge the biomedgraphica_id with the crispr_df\n",
    "merged_crispr_df = pd.merge(bmg_gene_df, crispr_df, left_on='HGNC_Symbol', right_on='gene_name', how='inner')\n",
    "merged_crispr_df.drop(columns=['HGNC_Symbol', 'gene_name'], inplace=True)\n",
    "display(merged_crispr_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_GN_ID</th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-002806</th>\n",
       "      <th>ACH-002834</th>\n",
       "      <th>ACH-002847</th>\n",
       "      <th>ACH-002922</th>\n",
       "      <th>ACH-002925</th>\n",
       "      <th>ACH-002926</th>\n",
       "      <th>ACH-002928</th>\n",
       "      <th>ACH-003012</th>\n",
       "      <th>ACH-003177</th>\n",
       "      <th>ACH-003180</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_PT013542</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>0.023031</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.051500</td>\n",
       "      <td>-0.072687</td>\n",
       "      <td>-0.057531</td>\n",
       "      <td>-0.050743</td>\n",
       "      <td>-0.163957</td>\n",
       "      <td>-0.107883</td>\n",
       "      <td>-0.287606</td>\n",
       "      <td>-0.145878</td>\n",
       "      <td>-0.038435</td>\n",
       "      <td>-0.007314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_PT091702</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>0.023031</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.051500</td>\n",
       "      <td>-0.072687</td>\n",
       "      <td>-0.057531</td>\n",
       "      <td>-0.050743</td>\n",
       "      <td>-0.163957</td>\n",
       "      <td>-0.107883</td>\n",
       "      <td>-0.287606</td>\n",
       "      <td>-0.145878</td>\n",
       "      <td>-0.038435</td>\n",
       "      <td>-0.007314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT070222</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>0.196378</td>\n",
       "      <td>0.359769</td>\n",
       "      <td>0.310169</td>\n",
       "      <td>0.198303</td>\n",
       "      <td>0.113667</td>\n",
       "      <td>0.157864</td>\n",
       "      <td>0.154384</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131496</td>\n",
       "      <td>0.165664</td>\n",
       "      <td>0.156277</td>\n",
       "      <td>0.198406</td>\n",
       "      <td>0.209907</td>\n",
       "      <td>0.192162</td>\n",
       "      <td>0.068782</td>\n",
       "      <td>0.143209</td>\n",
       "      <td>0.074829</td>\n",
       "      <td>-0.262441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT041299</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>0.196378</td>\n",
       "      <td>0.359769</td>\n",
       "      <td>0.310169</td>\n",
       "      <td>0.198303</td>\n",
       "      <td>0.113667</td>\n",
       "      <td>0.157864</td>\n",
       "      <td>0.154384</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131496</td>\n",
       "      <td>0.165664</td>\n",
       "      <td>0.156277</td>\n",
       "      <td>0.198406</td>\n",
       "      <td>0.209907</td>\n",
       "      <td>0.192162</td>\n",
       "      <td>0.068782</td>\n",
       "      <td>0.143209</td>\n",
       "      <td>0.074829</td>\n",
       "      <td>-0.262441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT038889</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>0.196378</td>\n",
       "      <td>0.359769</td>\n",
       "      <td>0.310169</td>\n",
       "      <td>0.198303</td>\n",
       "      <td>0.113667</td>\n",
       "      <td>0.157864</td>\n",
       "      <td>0.154384</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131496</td>\n",
       "      <td>0.165664</td>\n",
       "      <td>0.156277</td>\n",
       "      <td>0.198406</td>\n",
       "      <td>0.209907</td>\n",
       "      <td>0.192162</td>\n",
       "      <td>0.068782</td>\n",
       "      <td>0.143209</td>\n",
       "      <td>0.074829</td>\n",
       "      <td>-0.262441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117025</th>\n",
       "      <td>BMGC_GN69841</td>\n",
       "      <td>BMGC_PT120188</td>\n",
       "      <td>-0.137149</td>\n",
       "      <td>-0.601468</td>\n",
       "      <td>-0.147378</td>\n",
       "      <td>-0.320952</td>\n",
       "      <td>0.022595</td>\n",
       "      <td>-0.610531</td>\n",
       "      <td>-0.075869</td>\n",
       "      <td>-0.091422</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.100619</td>\n",
       "      <td>-0.022613</td>\n",
       "      <td>0.084076</td>\n",
       "      <td>-0.180020</td>\n",
       "      <td>-0.017108</td>\n",
       "      <td>-0.325764</td>\n",
       "      <td>0.144469</td>\n",
       "      <td>0.009504</td>\n",
       "      <td>-0.118627</td>\n",
       "      <td>-0.348614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117026</th>\n",
       "      <td>BMGC_GN69841</td>\n",
       "      <td>BMGC_PT120187</td>\n",
       "      <td>-0.137149</td>\n",
       "      <td>-0.601468</td>\n",
       "      <td>-0.147378</td>\n",
       "      <td>-0.320952</td>\n",
       "      <td>0.022595</td>\n",
       "      <td>-0.610531</td>\n",
       "      <td>-0.075869</td>\n",
       "      <td>-0.091422</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.100619</td>\n",
       "      <td>-0.022613</td>\n",
       "      <td>0.084076</td>\n",
       "      <td>-0.180020</td>\n",
       "      <td>-0.017108</td>\n",
       "      <td>-0.325764</td>\n",
       "      <td>0.144469</td>\n",
       "      <td>0.009504</td>\n",
       "      <td>-0.118627</td>\n",
       "      <td>-0.348614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117027</th>\n",
       "      <td>BMGC_GN69848</td>\n",
       "      <td>BMGC_PT104310</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.333310</td>\n",
       "      <td>-0.151258</td>\n",
       "      <td>0.090807</td>\n",
       "      <td>0.130973</td>\n",
       "      <td>0.035430</td>\n",
       "      <td>0.057994</td>\n",
       "      <td>0.127841</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.097166</td>\n",
       "      <td>-0.125821</td>\n",
       "      <td>0.056233</td>\n",
       "      <td>-0.057404</td>\n",
       "      <td>0.140328</td>\n",
       "      <td>0.182520</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.198007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117028</th>\n",
       "      <td>BMGC_GN69876</td>\n",
       "      <td>BMGC_PT120255</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.316461</td>\n",
       "      <td>-0.021494</td>\n",
       "      <td>-0.556799</td>\n",
       "      <td>-0.350030</td>\n",
       "      <td>-0.136289</td>\n",
       "      <td>-0.579060</td>\n",
       "      <td>-0.200260</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.288444</td>\n",
       "      <td>-0.376535</td>\n",
       "      <td>-0.083581</td>\n",
       "      <td>-0.059077</td>\n",
       "      <td>-0.246367</td>\n",
       "      <td>-0.165150</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.199642</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117029</th>\n",
       "      <td>BMGC_GN70238</td>\n",
       "      <td>BMGC_PT014744</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.019036</td>\n",
       "      <td>-0.202153</td>\n",
       "      <td>-0.158997</td>\n",
       "      <td>0.331191</td>\n",
       "      <td>-0.110372</td>\n",
       "      <td>0.074430</td>\n",
       "      <td>0.147567</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.243601</td>\n",
       "      <td>0.276185</td>\n",
       "      <td>0.135168</td>\n",
       "      <td>0.063313</td>\n",
       "      <td>-0.226531</td>\n",
       "      <td>0.050751</td>\n",
       "      <td>0.212870</td>\n",
       "      <td>-0.237297</td>\n",
       "      <td>0.123344</td>\n",
       "      <td>0.172626</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117030 rows × 1180 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BMGC_GN_ID     BMGC_PT_ID  ACH-000001  ACH-000004  ACH-000005  \\\n",
       "0       BMGC_GN00001  BMGC_PT013542   -0.121964   -0.016484   -0.184847   \n",
       "1       BMGC_GN00001  BMGC_PT091702   -0.121964   -0.016484   -0.184847   \n",
       "2       BMGC_GN00002  BMGC_PT070222   -0.022127    0.196378    0.359769   \n",
       "3       BMGC_GN00002  BMGC_PT041299   -0.022127    0.196378    0.359769   \n",
       "4       BMGC_GN00002  BMGC_PT038889   -0.022127    0.196378    0.359769   \n",
       "...              ...            ...         ...         ...         ...   \n",
       "117025  BMGC_GN69841  BMGC_PT120188   -0.137149   -0.601468   -0.147378   \n",
       "117026  BMGC_GN69841  BMGC_PT120187   -0.137149   -0.601468   -0.147378   \n",
       "117027  BMGC_GN69848  BMGC_PT104310    0.000000    0.333310   -0.151258   \n",
       "117028  BMGC_GN69876  BMGC_PT120255    0.000000   -0.316461   -0.021494   \n",
       "117029  BMGC_GN70238  BMGC_PT014744    0.000000   -0.019036   -0.202153   \n",
       "\n",
       "        ACH-000007  ACH-000009  ACH-000011  ACH-000012  ACH-000013  ...  \\\n",
       "0        -0.071921   -0.019163    0.023031   -0.139496    0.014120  ...   \n",
       "1        -0.071921   -0.019163    0.023031   -0.139496    0.014120  ...   \n",
       "2         0.310169    0.198303    0.113667    0.157864    0.154384  ...   \n",
       "3         0.310169    0.198303    0.113667    0.157864    0.154384  ...   \n",
       "4         0.310169    0.198303    0.113667    0.157864    0.154384  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "117025   -0.320952    0.022595   -0.610531   -0.075869   -0.091422  ...   \n",
       "117026   -0.320952    0.022595   -0.610531   -0.075869   -0.091422  ...   \n",
       "117027    0.090807    0.130973    0.035430    0.057994    0.127841  ...   \n",
       "117028   -0.556799   -0.350030   -0.136289   -0.579060   -0.200260  ...   \n",
       "117029   -0.158997    0.331191   -0.110372    0.074430    0.147567  ...   \n",
       "\n",
       "        ACH-002806  ACH-002834  ACH-002847  ACH-002922  ACH-002925  \\\n",
       "0        -0.051500   -0.072687   -0.057531   -0.050743   -0.163957   \n",
       "1        -0.051500   -0.072687   -0.057531   -0.050743   -0.163957   \n",
       "2         0.131496    0.165664    0.156277    0.198406    0.209907   \n",
       "3         0.131496    0.165664    0.156277    0.198406    0.209907   \n",
       "4         0.131496    0.165664    0.156277    0.198406    0.209907   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "117025   -0.100619   -0.022613    0.084076   -0.180020   -0.017108   \n",
       "117026   -0.100619   -0.022613    0.084076   -0.180020   -0.017108   \n",
       "117027   -0.097166   -0.125821    0.056233   -0.057404    0.140328   \n",
       "117028   -0.288444   -0.376535   -0.083581   -0.059077   -0.246367   \n",
       "117029   -0.243601    0.276185    0.135168    0.063313   -0.226531   \n",
       "\n",
       "        ACH-002926  ACH-002928  ACH-003012  ACH-003177  ACH-003180  \n",
       "0        -0.107883   -0.287606   -0.145878   -0.038435   -0.007314  \n",
       "1        -0.107883   -0.287606   -0.145878   -0.038435   -0.007314  \n",
       "2         0.192162    0.068782    0.143209    0.074829   -0.262441  \n",
       "3         0.192162    0.068782    0.143209    0.074829   -0.262441  \n",
       "4         0.192162    0.068782    0.143209    0.074829   -0.262441  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "117025   -0.325764    0.144469    0.009504   -0.118627   -0.348614  \n",
       "117026   -0.325764    0.144469    0.009504   -0.118627   -0.348614  \n",
       "117027    0.182520    0.000000    0.198007    0.000000    0.000000  \n",
       "117028   -0.165150    0.000000   -0.199642    0.000000    0.000000  \n",
       "117029    0.050751    0.212870   -0.237297    0.123344    0.172626  \n",
       "\n",
       "[117030 rows x 1180 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# map the gene entity with gene_transcript_protein_entity_df\n",
    "protein_crispr_df = pd.merge(gene_transcript_protein_entity_df, merged_crispr_df, left_on='BMGC_GN_ID', right_on='BioMedGraphica_Conn_ID', how='left').drop(columns=['BioMedGraphica_Conn_ID'])\n",
    "# and drop nan in any rows from protein_crispr_df\n",
    "protein_crispr_df = protein_crispr_df.dropna().reset_index(drop=True)\n",
    "display(protein_crispr_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ACH-000001',\n",
       " 'ACH-000002',\n",
       " 'ACH-000004',\n",
       " 'ACH-000005',\n",
       " 'ACH-000006',\n",
       " 'ACH-000007',\n",
       " 'ACH-000008',\n",
       " 'ACH-000009',\n",
       " 'ACH-000011',\n",
       " 'ACH-000012',\n",
       " 'ACH-000013',\n",
       " 'ACH-000014',\n",
       " 'ACH-000015',\n",
       " 'ACH-000016',\n",
       " 'ACH-000017',\n",
       " 'ACH-000018',\n",
       " 'ACH-000019',\n",
       " 'ACH-000020',\n",
       " 'ACH-000021',\n",
       " 'ACH-000022',\n",
       " 'ACH-000023',\n",
       " 'ACH-000024',\n",
       " 'ACH-000026',\n",
       " 'ACH-000027',\n",
       " 'ACH-000028',\n",
       " 'ACH-000030',\n",
       " 'ACH-000031',\n",
       " 'ACH-000032',\n",
       " 'ACH-000035',\n",
       " 'ACH-000037',\n",
       " 'ACH-000038',\n",
       " 'ACH-000039',\n",
       " 'ACH-000040',\n",
       " 'ACH-000041',\n",
       " 'ACH-000042',\n",
       " 'ACH-000043',\n",
       " 'ACH-000044',\n",
       " 'ACH-000045',\n",
       " 'ACH-000046',\n",
       " 'ACH-000047',\n",
       " 'ACH-000048',\n",
       " 'ACH-000050',\n",
       " 'ACH-000051',\n",
       " 'ACH-000052',\n",
       " 'ACH-000053',\n",
       " 'ACH-000054',\n",
       " 'ACH-000055',\n",
       " 'ACH-000056',\n",
       " 'ACH-000058',\n",
       " 'ACH-000059',\n",
       " 'ACH-000060',\n",
       " 'ACH-000061',\n",
       " 'ACH-000062',\n",
       " 'ACH-000063',\n",
       " 'ACH-000065',\n",
       " 'ACH-000066',\n",
       " 'ACH-000067',\n",
       " 'ACH-000068',\n",
       " 'ACH-000069',\n",
       " 'ACH-000070',\n",
       " 'ACH-000071',\n",
       " 'ACH-000072',\n",
       " 'ACH-000073',\n",
       " 'ACH-000074',\n",
       " 'ACH-000075',\n",
       " 'ACH-000076',\n",
       " 'ACH-000077',\n",
       " 'ACH-000078',\n",
       " 'ACH-000079',\n",
       " 'ACH-000080',\n",
       " 'ACH-000081',\n",
       " 'ACH-000082',\n",
       " 'ACH-000083',\n",
       " 'ACH-000084',\n",
       " 'ACH-000085',\n",
       " 'ACH-000086',\n",
       " 'ACH-000087',\n",
       " 'ACH-000088',\n",
       " 'ACH-000089',\n",
       " 'ACH-000090',\n",
       " 'ACH-000091',\n",
       " 'ACH-000092',\n",
       " 'ACH-000093',\n",
       " 'ACH-000094',\n",
       " 'ACH-000095',\n",
       " 'ACH-000096',\n",
       " 'ACH-000097',\n",
       " 'ACH-000098',\n",
       " 'ACH-000099',\n",
       " 'ACH-000100',\n",
       " 'ACH-000101',\n",
       " 'ACH-000102',\n",
       " 'ACH-000103',\n",
       " 'ACH-000104',\n",
       " 'ACH-000105',\n",
       " 'ACH-000106',\n",
       " 'ACH-000107',\n",
       " 'ACH-000108',\n",
       " 'ACH-000109',\n",
       " 'ACH-000110',\n",
       " 'ACH-000111',\n",
       " 'ACH-000112',\n",
       " 'ACH-000113',\n",
       " 'ACH-000114',\n",
       " 'ACH-000115',\n",
       " 'ACH-000116',\n",
       " 'ACH-000117',\n",
       " 'ACH-000118',\n",
       " 'ACH-000119',\n",
       " 'ACH-000120',\n",
       " 'ACH-000121',\n",
       " 'ACH-000122',\n",
       " 'ACH-000123',\n",
       " 'ACH-000124',\n",
       " 'ACH-000125',\n",
       " 'ACH-000126',\n",
       " 'ACH-000127',\n",
       " 'ACH-000129',\n",
       " 'ACH-000130',\n",
       " 'ACH-000131',\n",
       " 'ACH-000132',\n",
       " 'ACH-000133',\n",
       " 'ACH-000134',\n",
       " 'ACH-000135',\n",
       " 'ACH-000136',\n",
       " 'ACH-000137',\n",
       " 'ACH-000138',\n",
       " 'ACH-000139',\n",
       " 'ACH-000140',\n",
       " 'ACH-000141',\n",
       " 'ACH-000142',\n",
       " 'ACH-000144',\n",
       " 'ACH-000145',\n",
       " 'ACH-000146',\n",
       " 'ACH-000147',\n",
       " 'ACH-000148',\n",
       " 'ACH-000149',\n",
       " 'ACH-000150',\n",
       " 'ACH-000151',\n",
       " 'ACH-000152',\n",
       " 'ACH-000153',\n",
       " 'ACH-000154',\n",
       " 'ACH-000155',\n",
       " 'ACH-000156',\n",
       " 'ACH-000157',\n",
       " 'ACH-000158',\n",
       " 'ACH-000159',\n",
       " 'ACH-000161',\n",
       " 'ACH-000162',\n",
       " 'ACH-000163',\n",
       " 'ACH-000164',\n",
       " 'ACH-000165',\n",
       " 'ACH-000166',\n",
       " 'ACH-000167',\n",
       " 'ACH-000168',\n",
       " 'ACH-000169',\n",
       " 'ACH-000171',\n",
       " 'ACH-000173',\n",
       " 'ACH-000174',\n",
       " 'ACH-000175',\n",
       " 'ACH-000176',\n",
       " 'ACH-000177',\n",
       " 'ACH-000178',\n",
       " 'ACH-000179',\n",
       " 'ACH-000180',\n",
       " 'ACH-000181',\n",
       " 'ACH-000182',\n",
       " 'ACH-000183',\n",
       " 'ACH-000184',\n",
       " 'ACH-000185',\n",
       " 'ACH-000186',\n",
       " 'ACH-000187',\n",
       " 'ACH-000188',\n",
       " 'ACH-000189',\n",
       " 'ACH-000190',\n",
       " 'ACH-000191',\n",
       " 'ACH-000192',\n",
       " 'ACH-000193',\n",
       " 'ACH-000194',\n",
       " 'ACH-000195',\n",
       " 'ACH-000196',\n",
       " 'ACH-000197',\n",
       " 'ACH-000198',\n",
       " 'ACH-000199',\n",
       " 'ACH-000200',\n",
       " 'ACH-000201',\n",
       " 'ACH-000202',\n",
       " 'ACH-000203',\n",
       " 'ACH-000204',\n",
       " 'ACH-000205',\n",
       " 'ACH-000207',\n",
       " 'ACH-000209',\n",
       " 'ACH-000210',\n",
       " 'ACH-000211',\n",
       " 'ACH-000212',\n",
       " 'ACH-000213',\n",
       " 'ACH-000214',\n",
       " 'ACH-000217',\n",
       " 'ACH-000218',\n",
       " 'ACH-000219',\n",
       " 'ACH-000220',\n",
       " 'ACH-000221',\n",
       " 'ACH-000222',\n",
       " 'ACH-000223',\n",
       " 'ACH-000224',\n",
       " 'ACH-000225',\n",
       " 'ACH-000226',\n",
       " 'ACH-000227',\n",
       " 'ACH-000228',\n",
       " 'ACH-000229',\n",
       " 'ACH-000230',\n",
       " 'ACH-000231',\n",
       " 'ACH-000232',\n",
       " 'ACH-000233',\n",
       " 'ACH-000234',\n",
       " 'ACH-000235',\n",
       " 'ACH-000236',\n",
       " 'ACH-000237',\n",
       " 'ACH-000238',\n",
       " 'ACH-000239',\n",
       " 'ACH-000240',\n",
       " 'ACH-000241',\n",
       " 'ACH-000242',\n",
       " 'ACH-000243',\n",
       " 'ACH-000244',\n",
       " 'ACH-000245',\n",
       " 'ACH-000246',\n",
       " 'ACH-000247',\n",
       " 'ACH-000248',\n",
       " 'ACH-000249',\n",
       " 'ACH-000250',\n",
       " 'ACH-000252',\n",
       " 'ACH-000253',\n",
       " 'ACH-000254',\n",
       " 'ACH-000255',\n",
       " 'ACH-000256',\n",
       " 'ACH-000257',\n",
       " 'ACH-000258',\n",
       " 'ACH-000259',\n",
       " 'ACH-000260',\n",
       " 'ACH-000261',\n",
       " 'ACH-000263',\n",
       " 'ACH-000264',\n",
       " 'ACH-000265',\n",
       " 'ACH-000266',\n",
       " 'ACH-000267',\n",
       " 'ACH-000268',\n",
       " 'ACH-000269',\n",
       " 'ACH-000270',\n",
       " 'ACH-000271',\n",
       " 'ACH-000272',\n",
       " 'ACH-000273',\n",
       " 'ACH-000274',\n",
       " 'ACH-000275',\n",
       " 'ACH-000276',\n",
       " 'ACH-000277',\n",
       " 'ACH-000278',\n",
       " 'ACH-000280',\n",
       " 'ACH-000281',\n",
       " 'ACH-000282',\n",
       " 'ACH-000284',\n",
       " 'ACH-000285',\n",
       " 'ACH-000286',\n",
       " 'ACH-000287',\n",
       " 'ACH-000288',\n",
       " 'ACH-000289',\n",
       " 'ACH-000290',\n",
       " 'ACH-000291',\n",
       " 'ACH-000292',\n",
       " 'ACH-000293',\n",
       " 'ACH-000294',\n",
       " 'ACH-000295',\n",
       " 'ACH-000296',\n",
       " 'ACH-000297',\n",
       " 'ACH-000298',\n",
       " 'ACH-000300',\n",
       " 'ACH-000301',\n",
       " 'ACH-000302',\n",
       " 'ACH-000303',\n",
       " 'ACH-000304',\n",
       " 'ACH-000305',\n",
       " 'ACH-000306',\n",
       " 'ACH-000307',\n",
       " 'ACH-000308',\n",
       " 'ACH-000309',\n",
       " 'ACH-000310',\n",
       " 'ACH-000311',\n",
       " 'ACH-000312',\n",
       " 'ACH-000313',\n",
       " 'ACH-000314',\n",
       " 'ACH-000315',\n",
       " 'ACH-000316',\n",
       " 'ACH-000317',\n",
       " 'ACH-000318',\n",
       " 'ACH-000319',\n",
       " 'ACH-000320',\n",
       " 'ACH-000321',\n",
       " 'ACH-000322',\n",
       " 'ACH-000323',\n",
       " 'ACH-000324',\n",
       " 'ACH-000325',\n",
       " 'ACH-000326',\n",
       " 'ACH-000327',\n",
       " 'ACH-000329',\n",
       " 'ACH-000330',\n",
       " 'ACH-000331',\n",
       " 'ACH-000332',\n",
       " 'ACH-000333',\n",
       " 'ACH-000334',\n",
       " 'ACH-000335',\n",
       " 'ACH-000336',\n",
       " 'ACH-000338',\n",
       " 'ACH-000340',\n",
       " 'ACH-000341',\n",
       " 'ACH-000342',\n",
       " 'ACH-000343',\n",
       " 'ACH-000344',\n",
       " 'ACH-000345',\n",
       " 'ACH-000346',\n",
       " 'ACH-000347',\n",
       " 'ACH-000348',\n",
       " 'ACH-000349',\n",
       " 'ACH-000350',\n",
       " 'ACH-000351',\n",
       " 'ACH-000352',\n",
       " 'ACH-000353',\n",
       " 'ACH-000354',\n",
       " 'ACH-000355',\n",
       " 'ACH-000356',\n",
       " 'ACH-000357',\n",
       " 'ACH-000358',\n",
       " 'ACH-000359',\n",
       " 'ACH-000360',\n",
       " 'ACH-000361',\n",
       " 'ACH-000362',\n",
       " 'ACH-000363',\n",
       " 'ACH-000364',\n",
       " 'ACH-000365',\n",
       " 'ACH-000366',\n",
       " 'ACH-000367',\n",
       " 'ACH-000368',\n",
       " 'ACH-000369',\n",
       " 'ACH-000370',\n",
       " 'ACH-000371',\n",
       " 'ACH-000372',\n",
       " 'ACH-000373',\n",
       " 'ACH-000374',\n",
       " 'ACH-000375',\n",
       " 'ACH-000376',\n",
       " 'ACH-000377',\n",
       " 'ACH-000378',\n",
       " 'ACH-000379',\n",
       " 'ACH-000380',\n",
       " 'ACH-000381',\n",
       " 'ACH-000382',\n",
       " 'ACH-000384',\n",
       " 'ACH-000386',\n",
       " 'ACH-000387',\n",
       " 'ACH-000388',\n",
       " 'ACH-000389',\n",
       " 'ACH-000390',\n",
       " 'ACH-000391',\n",
       " 'ACH-000392',\n",
       " 'ACH-000393',\n",
       " 'ACH-000394',\n",
       " 'ACH-000395',\n",
       " 'ACH-000396',\n",
       " 'ACH-000397',\n",
       " 'ACH-000398',\n",
       " 'ACH-000399',\n",
       " 'ACH-000400',\n",
       " 'ACH-000401',\n",
       " 'ACH-000402',\n",
       " 'ACH-000403',\n",
       " 'ACH-000404',\n",
       " 'ACH-000405',\n",
       " 'ACH-000406',\n",
       " 'ACH-000407',\n",
       " 'ACH-000408',\n",
       " 'ACH-000409',\n",
       " 'ACH-000410',\n",
       " 'ACH-000411',\n",
       " 'ACH-000412',\n",
       " 'ACH-000413',\n",
       " 'ACH-000414',\n",
       " 'ACH-000415',\n",
       " 'ACH-000416',\n",
       " 'ACH-000417',\n",
       " 'ACH-000418',\n",
       " 'ACH-000419',\n",
       " 'ACH-000420',\n",
       " 'ACH-000421',\n",
       " 'ACH-000422',\n",
       " 'ACH-000423',\n",
       " 'ACH-000424',\n",
       " 'ACH-000425',\n",
       " 'ACH-000426',\n",
       " 'ACH-000427',\n",
       " 'ACH-000428',\n",
       " 'ACH-000429',\n",
       " 'ACH-000430',\n",
       " 'ACH-000431',\n",
       " 'ACH-000432',\n",
       " 'ACH-000433',\n",
       " 'ACH-000434',\n",
       " 'ACH-000435',\n",
       " 'ACH-000437',\n",
       " 'ACH-000438',\n",
       " 'ACH-000439',\n",
       " 'ACH-000440',\n",
       " 'ACH-000441',\n",
       " 'ACH-000442',\n",
       " 'ACH-000443',\n",
       " 'ACH-000444',\n",
       " 'ACH-000445',\n",
       " 'ACH-000446',\n",
       " 'ACH-000447',\n",
       " 'ACH-000448',\n",
       " 'ACH-000449',\n",
       " 'ACH-000450',\n",
       " 'ACH-000451',\n",
       " 'ACH-000452',\n",
       " 'ACH-000453',\n",
       " 'ACH-000454',\n",
       " 'ACH-000456',\n",
       " 'ACH-000457',\n",
       " 'ACH-000458',\n",
       " 'ACH-000459',\n",
       " 'ACH-000460',\n",
       " 'ACH-000461',\n",
       " 'ACH-000462',\n",
       " 'ACH-000463',\n",
       " 'ACH-000464',\n",
       " 'ACH-000465',\n",
       " 'ACH-000466',\n",
       " 'ACH-000467',\n",
       " 'ACH-000468',\n",
       " 'ACH-000469',\n",
       " 'ACH-000470',\n",
       " 'ACH-000471',\n",
       " 'ACH-000472',\n",
       " 'ACH-000473',\n",
       " 'ACH-000475',\n",
       " 'ACH-000476',\n",
       " 'ACH-000477',\n",
       " 'ACH-000478',\n",
       " 'ACH-000479',\n",
       " 'ACH-000480',\n",
       " 'ACH-000481',\n",
       " 'ACH-000482',\n",
       " 'ACH-000483',\n",
       " 'ACH-000484',\n",
       " 'ACH-000485',\n",
       " 'ACH-000486',\n",
       " 'ACH-000487',\n",
       " 'ACH-000488',\n",
       " 'ACH-000489',\n",
       " 'ACH-000491',\n",
       " 'ACH-000492',\n",
       " 'ACH-000493',\n",
       " 'ACH-000495',\n",
       " 'ACH-000496',\n",
       " 'ACH-000498',\n",
       " 'ACH-000500',\n",
       " 'ACH-000501',\n",
       " 'ACH-000502',\n",
       " 'ACH-000503',\n",
       " 'ACH-000504',\n",
       " 'ACH-000505',\n",
       " 'ACH-000506',\n",
       " 'ACH-000507',\n",
       " 'ACH-000508',\n",
       " 'ACH-000509',\n",
       " 'ACH-000510',\n",
       " 'ACH-000511',\n",
       " 'ACH-000513',\n",
       " 'ACH-000514',\n",
       " 'ACH-000515',\n",
       " 'ACH-000516',\n",
       " 'ACH-000517',\n",
       " 'ACH-000518',\n",
       " 'ACH-000519',\n",
       " 'ACH-000520',\n",
       " 'ACH-000521',\n",
       " 'ACH-000522',\n",
       " 'ACH-000523',\n",
       " 'ACH-000524',\n",
       " 'ACH-000525',\n",
       " 'ACH-000526',\n",
       " 'ACH-000527',\n",
       " 'ACH-000528',\n",
       " 'ACH-000529',\n",
       " 'ACH-000530',\n",
       " 'ACH-000531',\n",
       " 'ACH-000532',\n",
       " 'ACH-000534',\n",
       " 'ACH-000535',\n",
       " 'ACH-000536',\n",
       " 'ACH-000537',\n",
       " 'ACH-000538',\n",
       " 'ACH-000539',\n",
       " 'ACH-000540',\n",
       " 'ACH-000541',\n",
       " 'ACH-000542',\n",
       " 'ACH-000543',\n",
       " 'ACH-000544',\n",
       " 'ACH-000545',\n",
       " 'ACH-000546',\n",
       " 'ACH-000547',\n",
       " 'ACH-000548',\n",
       " 'ACH-000549',\n",
       " 'ACH-000550',\n",
       " 'ACH-000551',\n",
       " 'ACH-000552',\n",
       " 'ACH-000553',\n",
       " 'ACH-000554',\n",
       " 'ACH-000555',\n",
       " 'ACH-000557',\n",
       " 'ACH-000558',\n",
       " 'ACH-000559',\n",
       " 'ACH-000560',\n",
       " 'ACH-000562',\n",
       " 'ACH-000563',\n",
       " 'ACH-000564',\n",
       " 'ACH-000565',\n",
       " 'ACH-000566',\n",
       " 'ACH-000567',\n",
       " 'ACH-000568',\n",
       " 'ACH-000569',\n",
       " 'ACH-000570',\n",
       " 'ACH-000571',\n",
       " 'ACH-000572',\n",
       " 'ACH-000573',\n",
       " 'ACH-000574',\n",
       " 'ACH-000576',\n",
       " 'ACH-000577',\n",
       " 'ACH-000578',\n",
       " 'ACH-000579',\n",
       " 'ACH-000580',\n",
       " 'ACH-000581',\n",
       " 'ACH-000582',\n",
       " 'ACH-000583',\n",
       " 'ACH-000584',\n",
       " 'ACH-000585',\n",
       " 'ACH-000586',\n",
       " 'ACH-000587',\n",
       " 'ACH-000588',\n",
       " 'ACH-000589',\n",
       " 'ACH-000590',\n",
       " 'ACH-000592',\n",
       " 'ACH-000593',\n",
       " 'ACH-000594',\n",
       " 'ACH-000595',\n",
       " 'ACH-000596',\n",
       " 'ACH-000598',\n",
       " 'ACH-000599',\n",
       " 'ACH-000600',\n",
       " 'ACH-000601',\n",
       " 'ACH-000602',\n",
       " 'ACH-000603',\n",
       " 'ACH-000604',\n",
       " 'ACH-000605',\n",
       " 'ACH-000606',\n",
       " 'ACH-000607',\n",
       " 'ACH-000608',\n",
       " 'ACH-000609',\n",
       " 'ACH-000610',\n",
       " 'ACH-000611',\n",
       " 'ACH-000612',\n",
       " 'ACH-000613',\n",
       " 'ACH-000614',\n",
       " 'ACH-000615',\n",
       " 'ACH-000616',\n",
       " 'ACH-000617',\n",
       " 'ACH-000618',\n",
       " 'ACH-000619',\n",
       " 'ACH-000620',\n",
       " 'ACH-000621',\n",
       " 'ACH-000622',\n",
       " 'ACH-000623',\n",
       " 'ACH-000624',\n",
       " 'ACH-000625',\n",
       " 'ACH-000626',\n",
       " 'ACH-000627',\n",
       " 'ACH-000628',\n",
       " 'ACH-000630',\n",
       " 'ACH-000631',\n",
       " 'ACH-000632',\n",
       " 'ACH-000633',\n",
       " 'ACH-000635',\n",
       " 'ACH-000636',\n",
       " 'ACH-000637',\n",
       " 'ACH-000638',\n",
       " 'ACH-000639',\n",
       " 'ACH-000640',\n",
       " 'ACH-000641',\n",
       " 'ACH-000643',\n",
       " 'ACH-000644',\n",
       " 'ACH-000645',\n",
       " 'ACH-000646',\n",
       " 'ACH-000647',\n",
       " 'ACH-000648',\n",
       " 'ACH-000649',\n",
       " 'ACH-000650',\n",
       " 'ACH-000651',\n",
       " 'ACH-000652',\n",
       " 'ACH-000653',\n",
       " 'ACH-000654',\n",
       " 'ACH-000655',\n",
       " 'ACH-000656',\n",
       " 'ACH-000657',\n",
       " 'ACH-000659',\n",
       " 'ACH-000660',\n",
       " 'ACH-000661',\n",
       " 'ACH-000662',\n",
       " 'ACH-000663',\n",
       " 'ACH-000664',\n",
       " 'ACH-000665',\n",
       " 'ACH-000666',\n",
       " 'ACH-000667',\n",
       " 'ACH-000668',\n",
       " 'ACH-000669',\n",
       " 'ACH-000670',\n",
       " 'ACH-000671',\n",
       " 'ACH-000672',\n",
       " 'ACH-000674',\n",
       " 'ACH-000675',\n",
       " 'ACH-000677',\n",
       " 'ACH-000678',\n",
       " 'ACH-000679',\n",
       " 'ACH-000680',\n",
       " 'ACH-000681',\n",
       " 'ACH-000682',\n",
       " 'ACH-000683',\n",
       " 'ACH-000684',\n",
       " 'ACH-000685',\n",
       " 'ACH-000686',\n",
       " 'ACH-000687',\n",
       " 'ACH-000688',\n",
       " 'ACH-000689',\n",
       " 'ACH-000691',\n",
       " 'ACH-000692',\n",
       " 'ACH-000693',\n",
       " 'ACH-000694',\n",
       " 'ACH-000695',\n",
       " 'ACH-000696',\n",
       " 'ACH-000697',\n",
       " 'ACH-000698',\n",
       " 'ACH-000699',\n",
       " 'ACH-000701',\n",
       " 'ACH-000702',\n",
       " 'ACH-000703',\n",
       " 'ACH-000704',\n",
       " 'ACH-000705',\n",
       " 'ACH-000706',\n",
       " 'ACH-000707',\n",
       " 'ACH-000708',\n",
       " 'ACH-000709',\n",
       " 'ACH-000710',\n",
       " 'ACH-000711',\n",
       " 'ACH-000712',\n",
       " 'ACH-000713',\n",
       " 'ACH-000714',\n",
       " 'ACH-000715',\n",
       " 'ACH-000716',\n",
       " 'ACH-000717',\n",
       " 'ACH-000718',\n",
       " 'ACH-000719',\n",
       " 'ACH-000720',\n",
       " 'ACH-000721',\n",
       " 'ACH-000722',\n",
       " 'ACH-000723',\n",
       " 'ACH-000724',\n",
       " 'ACH-000725',\n",
       " 'ACH-000726',\n",
       " 'ACH-000727',\n",
       " 'ACH-000728',\n",
       " 'ACH-000729',\n",
       " 'ACH-000730',\n",
       " 'ACH-000731',\n",
       " 'ACH-000732',\n",
       " 'ACH-000733',\n",
       " 'ACH-000734',\n",
       " 'ACH-000735',\n",
       " 'ACH-000736',\n",
       " 'ACH-000737',\n",
       " 'ACH-000738',\n",
       " 'ACH-000739',\n",
       " 'ACH-000740',\n",
       " 'ACH-000741',\n",
       " 'ACH-000742',\n",
       " 'ACH-000743',\n",
       " 'ACH-000744',\n",
       " 'ACH-000745',\n",
       " 'ACH-000746',\n",
       " 'ACH-000747',\n",
       " 'ACH-000748',\n",
       " 'ACH-000749',\n",
       " 'ACH-000750',\n",
       " 'ACH-000751',\n",
       " 'ACH-000752',\n",
       " 'ACH-000753',\n",
       " 'ACH-000754',\n",
       " 'ACH-000755',\n",
       " 'ACH-000756',\n",
       " 'ACH-000758',\n",
       " 'ACH-000759',\n",
       " 'ACH-000761',\n",
       " 'ACH-000762',\n",
       " 'ACH-000763',\n",
       " 'ACH-000764',\n",
       " 'ACH-000765',\n",
       " 'ACH-000766',\n",
       " 'ACH-000767',\n",
       " 'ACH-000768',\n",
       " 'ACH-000769',\n",
       " 'ACH-000770',\n",
       " 'ACH-000771',\n",
       " 'ACH-000772',\n",
       " 'ACH-000773',\n",
       " 'ACH-000774',\n",
       " 'ACH-000775',\n",
       " 'ACH-000776',\n",
       " 'ACH-000777',\n",
       " 'ACH-000778',\n",
       " 'ACH-000780',\n",
       " 'ACH-000781',\n",
       " 'ACH-000782',\n",
       " 'ACH-000783',\n",
       " 'ACH-000784',\n",
       " 'ACH-000785',\n",
       " 'ACH-000786',\n",
       " 'ACH-000787',\n",
       " 'ACH-000788',\n",
       " 'ACH-000789',\n",
       " 'ACH-000790',\n",
       " 'ACH-000791',\n",
       " 'ACH-000792',\n",
       " 'ACH-000793',\n",
       " 'ACH-000794',\n",
       " 'ACH-000795',\n",
       " 'ACH-000796',\n",
       " 'ACH-000797',\n",
       " 'ACH-000798',\n",
       " 'ACH-000799',\n",
       " 'ACH-000800',\n",
       " 'ACH-000801',\n",
       " 'ACH-000802',\n",
       " 'ACH-000803',\n",
       " 'ACH-000804',\n",
       " 'ACH-000805',\n",
       " 'ACH-000806',\n",
       " 'ACH-000807',\n",
       " 'ACH-000808',\n",
       " 'ACH-000809',\n",
       " 'ACH-000810',\n",
       " 'ACH-000811',\n",
       " 'ACH-000812',\n",
       " 'ACH-000813',\n",
       " 'ACH-000814',\n",
       " 'ACH-000815',\n",
       " 'ACH-000816',\n",
       " 'ACH-000817',\n",
       " 'ACH-000818',\n",
       " 'ACH-000819',\n",
       " 'ACH-000820',\n",
       " 'ACH-000821',\n",
       " 'ACH-000822',\n",
       " 'ACH-000823',\n",
       " 'ACH-000824',\n",
       " 'ACH-000825',\n",
       " 'ACH-000826',\n",
       " 'ACH-000827',\n",
       " 'ACH-000828',\n",
       " 'ACH-000829',\n",
       " 'ACH-000830',\n",
       " 'ACH-000831',\n",
       " 'ACH-000832',\n",
       " 'ACH-000833',\n",
       " 'ACH-000834',\n",
       " 'ACH-000835',\n",
       " 'ACH-000836',\n",
       " 'ACH-000837',\n",
       " 'ACH-000838',\n",
       " 'ACH-000839',\n",
       " 'ACH-000840',\n",
       " 'ACH-000841',\n",
       " 'ACH-000842',\n",
       " 'ACH-000843',\n",
       " 'ACH-000844',\n",
       " 'ACH-000845',\n",
       " 'ACH-000846',\n",
       " 'ACH-000847',\n",
       " 'ACH-000848',\n",
       " 'ACH-000849',\n",
       " 'ACH-000850',\n",
       " 'ACH-000851',\n",
       " 'ACH-000852',\n",
       " 'ACH-000853',\n",
       " 'ACH-000855',\n",
       " 'ACH-000856',\n",
       " 'ACH-000857',\n",
       " 'ACH-000858',\n",
       " 'ACH-000859',\n",
       " 'ACH-000860',\n",
       " 'ACH-000861',\n",
       " 'ACH-000862',\n",
       " 'ACH-000863',\n",
       " 'ACH-000864',\n",
       " 'ACH-000865',\n",
       " 'ACH-000866',\n",
       " 'ACH-000867',\n",
       " 'ACH-000868',\n",
       " 'ACH-000869',\n",
       " 'ACH-000870',\n",
       " 'ACH-000871',\n",
       " 'ACH-000873',\n",
       " 'ACH-000874',\n",
       " 'ACH-000875',\n",
       " 'ACH-000876',\n",
       " 'ACH-000877',\n",
       " 'ACH-000878',\n",
       " 'ACH-000879',\n",
       " 'ACH-000880',\n",
       " 'ACH-000881',\n",
       " 'ACH-000882',\n",
       " 'ACH-000883',\n",
       " 'ACH-000884',\n",
       " 'ACH-000885',\n",
       " 'ACH-000886',\n",
       " 'ACH-000888',\n",
       " 'ACH-000889',\n",
       " 'ACH-000890',\n",
       " 'ACH-000891',\n",
       " 'ACH-000892',\n",
       " 'ACH-000893',\n",
       " 'ACH-000894',\n",
       " 'ACH-000895',\n",
       " 'ACH-000896',\n",
       " 'ACH-000897',\n",
       " 'ACH-000898',\n",
       " 'ACH-000899',\n",
       " 'ACH-000900',\n",
       " 'ACH-000901',\n",
       " 'ACH-000902',\n",
       " 'ACH-000903',\n",
       " 'ACH-000904',\n",
       " 'ACH-000905',\n",
       " 'ACH-000906',\n",
       " 'ACH-000907',\n",
       " 'ACH-000908',\n",
       " 'ACH-000909',\n",
       " 'ACH-000910',\n",
       " 'ACH-000911',\n",
       " 'ACH-000912',\n",
       " 'ACH-000913',\n",
       " 'ACH-000914',\n",
       " 'ACH-000915',\n",
       " 'ACH-000916',\n",
       " 'ACH-000917',\n",
       " 'ACH-000918',\n",
       " 'ACH-000919',\n",
       " 'ACH-000920',\n",
       " 'ACH-000921',\n",
       " 'ACH-000922',\n",
       " 'ACH-000923',\n",
       " 'ACH-000924',\n",
       " 'ACH-000925',\n",
       " 'ACH-000926',\n",
       " 'ACH-000927',\n",
       " 'ACH-000928',\n",
       " 'ACH-000929',\n",
       " 'ACH-000930',\n",
       " 'ACH-000931',\n",
       " 'ACH-000932',\n",
       " 'ACH-000933',\n",
       " 'ACH-000934',\n",
       " 'ACH-000935',\n",
       " 'ACH-000936',\n",
       " 'ACH-000937',\n",
       " 'ACH-000938',\n",
       " 'ACH-000939',\n",
       " 'ACH-000940',\n",
       " 'ACH-000941',\n",
       " 'ACH-000942',\n",
       " 'ACH-000943',\n",
       " 'ACH-000944',\n",
       " 'ACH-000945',\n",
       " 'ACH-000946',\n",
       " 'ACH-000947',\n",
       " 'ACH-000948',\n",
       " 'ACH-000949',\n",
       " 'ACH-000950',\n",
       " 'ACH-000951',\n",
       " 'ACH-000952',\n",
       " 'ACH-000953',\n",
       " 'ACH-000954',\n",
       " 'ACH-000955',\n",
       " 'ACH-000956',\n",
       " 'ACH-000957',\n",
       " 'ACH-000958',\n",
       " 'ACH-000959',\n",
       " 'ACH-000960',\n",
       " 'ACH-000961',\n",
       " 'ACH-000962',\n",
       " 'ACH-000963',\n",
       " 'ACH-000965',\n",
       " 'ACH-000966',\n",
       " 'ACH-000967',\n",
       " 'ACH-000968',\n",
       " 'ACH-000969',\n",
       " 'ACH-000970',\n",
       " 'ACH-000971',\n",
       " 'ACH-000972',\n",
       " 'ACH-000973',\n",
       " 'ACH-000974',\n",
       " 'ACH-000976',\n",
       " 'ACH-000977',\n",
       " 'ACH-000978',\n",
       " 'ACH-000979',\n",
       " 'ACH-000980',\n",
       " 'ACH-000981',\n",
       " 'ACH-000982',\n",
       " 'ACH-000983',\n",
       " 'ACH-000984',\n",
       " 'ACH-000985',\n",
       " 'ACH-000986',\n",
       " 'ACH-000987',\n",
       " 'ACH-000988',\n",
       " 'ACH-000989',\n",
       " 'ACH-000990',\n",
       " 'ACH-000991',\n",
       " 'ACH-000992',\n",
       " 'ACH-000993',\n",
       " 'ACH-000994',\n",
       " 'ACH-000995',\n",
       " 'ACH-000996',\n",
       " 'ACH-000997',\n",
       " 'ACH-000998',\n",
       " 'ACH-000999',\n",
       " 'ACH-001001',\n",
       " 'ACH-001016',\n",
       " 'ACH-001021',\n",
       " 'ACH-001023',\n",
       " 'ACH-001024',\n",
       " 'ACH-001036',\n",
       " 'ACH-001039',\n",
       " 'ACH-001041',\n",
       " 'ACH-001048',\n",
       " 'ACH-001049',\n",
       " 'ACH-001061',\n",
       " 'ACH-001063',\n",
       " 'ACH-001064',\n",
       " 'ACH-001065',\n",
       " 'ACH-001075',\n",
       " 'ACH-001089',\n",
       " 'ACH-001093',\n",
       " 'ACH-001098',\n",
       " 'ACH-001106',\n",
       " 'ACH-001107',\n",
       " 'ACH-001113',\n",
       " 'ACH-001125',\n",
       " 'ACH-001126',\n",
       " 'ACH-001127',\n",
       " 'ACH-001129',\n",
       " 'ACH-001130',\n",
       " 'ACH-001138',\n",
       " 'ACH-001145',\n",
       " 'ACH-001150',\n",
       " 'ACH-001151',\n",
       " 'ACH-001162',\n",
       " 'ACH-001171',\n",
       " 'ACH-001182',\n",
       " 'ACH-001183',\n",
       " 'ACH-001188',\n",
       " 'ACH-001190',\n",
       " 'ACH-001194',\n",
       " 'ACH-001199',\n",
       " 'ACH-001203',\n",
       " 'ACH-001233',\n",
       " 'ACH-001239',\n",
       " 'ACH-001306',\n",
       " 'ACH-001307',\n",
       " 'ACH-001318',\n",
       " 'ACH-001321']"
      ]
     },
     "execution_count": 373,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlapped_omics_union_annotation_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_GN_ID</th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001183</th>\n",
       "      <th>ACH-001188</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001194</th>\n",
       "      <th>ACH-001233</th>\n",
       "      <th>ACH-001239</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "      <th>ACH-001318</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_PT013542</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>0.023031</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.102453</td>\n",
       "      <td>-0.054710</td>\n",
       "      <td>-0.110358</td>\n",
       "      <td>-0.042641</td>\n",
       "      <td>-0.009775</td>\n",
       "      <td>-0.121936</td>\n",
       "      <td>-0.080115</td>\n",
       "      <td>-0.031081</td>\n",
       "      <td>0.012357</td>\n",
       "      <td>-0.084400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_PT091702</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>0.023031</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.102453</td>\n",
       "      <td>-0.054710</td>\n",
       "      <td>-0.110358</td>\n",
       "      <td>-0.042641</td>\n",
       "      <td>-0.009775</td>\n",
       "      <td>-0.121936</td>\n",
       "      <td>-0.080115</td>\n",
       "      <td>-0.031081</td>\n",
       "      <td>0.012357</td>\n",
       "      <td>-0.084400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT070222</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>0.196378</td>\n",
       "      <td>0.359769</td>\n",
       "      <td>0.310169</td>\n",
       "      <td>0.198303</td>\n",
       "      <td>0.113667</td>\n",
       "      <td>0.157864</td>\n",
       "      <td>0.154384</td>\n",
       "      <td>...</td>\n",
       "      <td>0.198034</td>\n",
       "      <td>0.123040</td>\n",
       "      <td>0.168877</td>\n",
       "      <td>0.185244</td>\n",
       "      <td>0.180496</td>\n",
       "      <td>0.086880</td>\n",
       "      <td>0.303579</td>\n",
       "      <td>0.250509</td>\n",
       "      <td>0.223775</td>\n",
       "      <td>0.184682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT041299</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>0.196378</td>\n",
       "      <td>0.359769</td>\n",
       "      <td>0.310169</td>\n",
       "      <td>0.198303</td>\n",
       "      <td>0.113667</td>\n",
       "      <td>0.157864</td>\n",
       "      <td>0.154384</td>\n",
       "      <td>...</td>\n",
       "      <td>0.198034</td>\n",
       "      <td>0.123040</td>\n",
       "      <td>0.168877</td>\n",
       "      <td>0.185244</td>\n",
       "      <td>0.180496</td>\n",
       "      <td>0.086880</td>\n",
       "      <td>0.303579</td>\n",
       "      <td>0.250509</td>\n",
       "      <td>0.223775</td>\n",
       "      <td>0.184682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>BMGC_PT038889</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>0.196378</td>\n",
       "      <td>0.359769</td>\n",
       "      <td>0.310169</td>\n",
       "      <td>0.198303</td>\n",
       "      <td>0.113667</td>\n",
       "      <td>0.157864</td>\n",
       "      <td>0.154384</td>\n",
       "      <td>...</td>\n",
       "      <td>0.198034</td>\n",
       "      <td>0.123040</td>\n",
       "      <td>0.168877</td>\n",
       "      <td>0.185244</td>\n",
       "      <td>0.180496</td>\n",
       "      <td>0.086880</td>\n",
       "      <td>0.303579</td>\n",
       "      <td>0.250509</td>\n",
       "      <td>0.223775</td>\n",
       "      <td>0.184682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117025</th>\n",
       "      <td>BMGC_GN69841</td>\n",
       "      <td>BMGC_PT120188</td>\n",
       "      <td>-0.137149</td>\n",
       "      <td>-0.601468</td>\n",
       "      <td>-0.147378</td>\n",
       "      <td>-0.320952</td>\n",
       "      <td>0.022595</td>\n",
       "      <td>-0.610531</td>\n",
       "      <td>-0.075869</td>\n",
       "      <td>-0.091422</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.289655</td>\n",
       "      <td>0.190473</td>\n",
       "      <td>-0.057121</td>\n",
       "      <td>0.105202</td>\n",
       "      <td>0.051115</td>\n",
       "      <td>-0.141930</td>\n",
       "      <td>-0.141818</td>\n",
       "      <td>-0.048947</td>\n",
       "      <td>-0.077387</td>\n",
       "      <td>-0.068879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117026</th>\n",
       "      <td>BMGC_GN69841</td>\n",
       "      <td>BMGC_PT120187</td>\n",
       "      <td>-0.137149</td>\n",
       "      <td>-0.601468</td>\n",
       "      <td>-0.147378</td>\n",
       "      <td>-0.320952</td>\n",
       "      <td>0.022595</td>\n",
       "      <td>-0.610531</td>\n",
       "      <td>-0.075869</td>\n",
       "      <td>-0.091422</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.289655</td>\n",
       "      <td>0.190473</td>\n",
       "      <td>-0.057121</td>\n",
       "      <td>0.105202</td>\n",
       "      <td>0.051115</td>\n",
       "      <td>-0.141930</td>\n",
       "      <td>-0.141818</td>\n",
       "      <td>-0.048947</td>\n",
       "      <td>-0.077387</td>\n",
       "      <td>-0.068879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117027</th>\n",
       "      <td>BMGC_GN69848</td>\n",
       "      <td>BMGC_PT104310</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.333310</td>\n",
       "      <td>-0.151258</td>\n",
       "      <td>0.090807</td>\n",
       "      <td>0.130973</td>\n",
       "      <td>0.035430</td>\n",
       "      <td>0.057994</td>\n",
       "      <td>0.127841</td>\n",
       "      <td>...</td>\n",
       "      <td>0.293546</td>\n",
       "      <td>0.275511</td>\n",
       "      <td>-0.006515</td>\n",
       "      <td>0.276861</td>\n",
       "      <td>0.300316</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.151090</td>\n",
       "      <td>-0.006914</td>\n",
       "      <td>0.126152</td>\n",
       "      <td>0.291545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117028</th>\n",
       "      <td>BMGC_GN69876</td>\n",
       "      <td>BMGC_PT120255</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.316461</td>\n",
       "      <td>-0.021494</td>\n",
       "      <td>-0.556799</td>\n",
       "      <td>-0.350030</td>\n",
       "      <td>-0.136289</td>\n",
       "      <td>-0.579060</td>\n",
       "      <td>-0.200260</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.202856</td>\n",
       "      <td>-0.170992</td>\n",
       "      <td>-0.407035</td>\n",
       "      <td>-0.067530</td>\n",
       "      <td>-0.153063</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.361657</td>\n",
       "      <td>-0.110224</td>\n",
       "      <td>-0.190182</td>\n",
       "      <td>-0.046852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117029</th>\n",
       "      <td>BMGC_GN70238</td>\n",
       "      <td>BMGC_PT014744</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.019036</td>\n",
       "      <td>-0.202153</td>\n",
       "      <td>-0.158997</td>\n",
       "      <td>0.331191</td>\n",
       "      <td>-0.110372</td>\n",
       "      <td>0.074430</td>\n",
       "      <td>0.147567</td>\n",
       "      <td>...</td>\n",
       "      <td>0.086688</td>\n",
       "      <td>0.075748</td>\n",
       "      <td>0.107002</td>\n",
       "      <td>-0.095637</td>\n",
       "      <td>-0.058258</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.078426</td>\n",
       "      <td>0.082939</td>\n",
       "      <td>0.047632</td>\n",
       "      <td>0.099352</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117030 rows × 669 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BMGC_GN_ID     BMGC_PT_ID  ACH-000001  ACH-000004  ACH-000005  \\\n",
       "0       BMGC_GN00001  BMGC_PT013542   -0.121964   -0.016484   -0.184847   \n",
       "1       BMGC_GN00001  BMGC_PT091702   -0.121964   -0.016484   -0.184847   \n",
       "2       BMGC_GN00002  BMGC_PT070222   -0.022127    0.196378    0.359769   \n",
       "3       BMGC_GN00002  BMGC_PT041299   -0.022127    0.196378    0.359769   \n",
       "4       BMGC_GN00002  BMGC_PT038889   -0.022127    0.196378    0.359769   \n",
       "...              ...            ...         ...         ...         ...   \n",
       "117025  BMGC_GN69841  BMGC_PT120188   -0.137149   -0.601468   -0.147378   \n",
       "117026  BMGC_GN69841  BMGC_PT120187   -0.137149   -0.601468   -0.147378   \n",
       "117027  BMGC_GN69848  BMGC_PT104310    0.000000    0.333310   -0.151258   \n",
       "117028  BMGC_GN69876  BMGC_PT120255    0.000000   -0.316461   -0.021494   \n",
       "117029  BMGC_GN70238  BMGC_PT014744    0.000000   -0.019036   -0.202153   \n",
       "\n",
       "        ACH-000007  ACH-000009  ACH-000011  ACH-000012  ACH-000013  ...  \\\n",
       "0        -0.071921   -0.019163    0.023031   -0.139496    0.014120  ...   \n",
       "1        -0.071921   -0.019163    0.023031   -0.139496    0.014120  ...   \n",
       "2         0.310169    0.198303    0.113667    0.157864    0.154384  ...   \n",
       "3         0.310169    0.198303    0.113667    0.157864    0.154384  ...   \n",
       "4         0.310169    0.198303    0.113667    0.157864    0.154384  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "117025   -0.320952    0.022595   -0.610531   -0.075869   -0.091422  ...   \n",
       "117026   -0.320952    0.022595   -0.610531   -0.075869   -0.091422  ...   \n",
       "117027    0.090807    0.130973    0.035430    0.057994    0.127841  ...   \n",
       "117028   -0.556799   -0.350030   -0.136289   -0.579060   -0.200260  ...   \n",
       "117029   -0.158997    0.331191   -0.110372    0.074430    0.147567  ...   \n",
       "\n",
       "        ACH-001151  ACH-001183  ACH-001188  ACH-001190  ACH-001194  \\\n",
       "0        -0.102453   -0.054710   -0.110358   -0.042641   -0.009775   \n",
       "1        -0.102453   -0.054710   -0.110358   -0.042641   -0.009775   \n",
       "2         0.198034    0.123040    0.168877    0.185244    0.180496   \n",
       "3         0.198034    0.123040    0.168877    0.185244    0.180496   \n",
       "4         0.198034    0.123040    0.168877    0.185244    0.180496   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "117025   -0.289655    0.190473   -0.057121    0.105202    0.051115   \n",
       "117026   -0.289655    0.190473   -0.057121    0.105202    0.051115   \n",
       "117027    0.293546    0.275511   -0.006515    0.276861    0.300316   \n",
       "117028   -0.202856   -0.170992   -0.407035   -0.067530   -0.153063   \n",
       "117029    0.086688    0.075748    0.107002   -0.095637   -0.058258   \n",
       "\n",
       "        ACH-001233  ACH-001239  ACH-001306  ACH-001307  ACH-001318  \n",
       "0        -0.121936   -0.080115   -0.031081    0.012357   -0.084400  \n",
       "1        -0.121936   -0.080115   -0.031081    0.012357   -0.084400  \n",
       "2         0.086880    0.303579    0.250509    0.223775    0.184682  \n",
       "3         0.086880    0.303579    0.250509    0.223775    0.184682  \n",
       "4         0.086880    0.303579    0.250509    0.223775    0.184682  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "117025   -0.141930   -0.141818   -0.048947   -0.077387   -0.068879  \n",
       "117026   -0.141930   -0.141818   -0.048947   -0.077387   -0.068879  \n",
       "117027    0.000000    0.151090   -0.006914    0.126152    0.291545  \n",
       "117028    0.000000   -0.361657   -0.110224   -0.190182   -0.046852  \n",
       "117029    0.000000    0.078426    0.082939    0.047632    0.099352  \n",
       "\n",
       "[117030 rows x 669 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Keep BioMedGraphica_Conn_ID and columns that are in overlapped_omics_union_annotation_samples\n",
    "# For CRISPR data\n",
    "columns_to_keep = ['BMGC_GN_ID', 'BMGC_PT_ID'] + [col for col in protein_crispr_df.columns if col in overlapped_omics_union_annotation_samples]\n",
    "processed_crispr_df = protein_crispr_df[columns_to_keep]\n",
    "display(processed_crispr_df)\n",
    "processed_crispr_df.to_csv('./data/pretrain_plain_data/processed_crispr.csv', index=False)\n",
    "processed_crispr_df.to_csv('./data/pretrain_status_data/processed_crispr.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 8.2 RNAi top 100 transcript entities"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 375,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>127399_SOFT_TISSUE</th>\n",
       "      <th>1321N1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>143B_BONE</th>\n",
       "      <th>184A1_BREAST</th>\n",
       "      <th>184B5_BREAST</th>\n",
       "      <th>22RV1_PROSTATE</th>\n",
       "      <th>2313287_STOMACH</th>\n",
       "      <th>600MPE_BREAST</th>\n",
       "      <th>697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>...</th>\n",
       "      <th>WM88_SKIN</th>\n",
       "      <th>WM983B_SKIN</th>\n",
       "      <th>WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>YD38_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>YD8_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>YKG1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>YMB1_BREAST</th>\n",
       "      <th>ZR751_BREAST</th>\n",
       "      <th>ZR7530_BREAST</th>\n",
       "      <th>ZR75B_BREAST</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG (1)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.146042</td>\n",
       "      <td>-0.190388</td>\n",
       "      <td>0.907063</td>\n",
       "      <td>-0.019331</td>\n",
       "      <td>-0.016734</td>\n",
       "      <td>0.091580</td>\n",
       "      <td>0.035023</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.088267</td>\n",
       "      <td>0.002171</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.120294</td>\n",
       "      <td>0.012540</td>\n",
       "      <td>0.111530</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.079313</td>\n",
       "      <td>-0.141559</td>\n",
       "      <td>0.214268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NAT2 (10)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.102854</td>\n",
       "      <td>0.384106</td>\n",
       "      <td>0.403192</td>\n",
       "      <td>0.001925</td>\n",
       "      <td>-0.153933</td>\n",
       "      <td>-0.317969</td>\n",
       "      <td>0.012341</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.003747</td>\n",
       "      <td>-0.321445</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.003256</td>\n",
       "      <td>-0.220472</td>\n",
       "      <td>0.073460</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.130921</td>\n",
       "      <td>0.127358</td>\n",
       "      <td>-0.405974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ADA (100)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.168839</td>\n",
       "      <td>-0.120700</td>\n",
       "      <td>0.004394</td>\n",
       "      <td>-0.188700</td>\n",
       "      <td>-0.060818</td>\n",
       "      <td>-0.755058</td>\n",
       "      <td>0.129770</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.014085</td>\n",
       "      <td>0.039679</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.076521</td>\n",
       "      <td>0.106995</td>\n",
       "      <td>0.227977</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.134479</td>\n",
       "      <td>0.083506</td>\n",
       "      <td>-0.404291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CDH2 (1000)</td>\n",
       "      <td>-0.194962</td>\n",
       "      <td>-0.028171</td>\n",
       "      <td>0.063047</td>\n",
       "      <td>-0.237251</td>\n",
       "      <td>-0.017059</td>\n",
       "      <td>-0.103047</td>\n",
       "      <td>-0.049460</td>\n",
       "      <td>0.130107</td>\n",
       "      <td>0.146864</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.073435</td>\n",
       "      <td>-0.140041</td>\n",
       "      <td>-0.154436</td>\n",
       "      <td>-0.040308</td>\n",
       "      <td>-0.078707</td>\n",
       "      <td>0.000769</td>\n",
       "      <td>-0.139126</td>\n",
       "      <td>0.047022</td>\n",
       "      <td>-0.097644</td>\n",
       "      <td>-0.062622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT3 (10000)</td>\n",
       "      <td>-0.256108</td>\n",
       "      <td>0.100751</td>\n",
       "      <td>-0.008077</td>\n",
       "      <td>0.060267</td>\n",
       "      <td>-0.094749</td>\n",
       "      <td>-0.066591</td>\n",
       "      <td>0.166029</td>\n",
       "      <td>0.149969</td>\n",
       "      <td>-0.053022</td>\n",
       "      <td>...</td>\n",
       "      <td>0.028714</td>\n",
       "      <td>-0.054628</td>\n",
       "      <td>0.450581</td>\n",
       "      <td>0.002932</td>\n",
       "      <td>0.129679</td>\n",
       "      <td>-0.072564</td>\n",
       "      <td>0.017161</td>\n",
       "      <td>0.123615</td>\n",
       "      <td>0.046846</td>\n",
       "      <td>0.125711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17304</th>\n",
       "      <td>SLC12A6 (9990)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.106165</td>\n",
       "      <td>0.227968</td>\n",
       "      <td>0.088744</td>\n",
       "      <td>-0.060731</td>\n",
       "      <td>0.127754</td>\n",
       "      <td>-0.049485</td>\n",
       "      <td>0.381034</td>\n",
       "      <td>...</td>\n",
       "      <td>0.170658</td>\n",
       "      <td>-0.004165</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.118544</td>\n",
       "      <td>0.273135</td>\n",
       "      <td>0.141616</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.100142</td>\n",
       "      <td>0.160643</td>\n",
       "      <td>0.089555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17305</th>\n",
       "      <td>KCNE2 (9992)</td>\n",
       "      <td>0.088853</td>\n",
       "      <td>-0.110628</td>\n",
       "      <td>0.057286</td>\n",
       "      <td>0.028095</td>\n",
       "      <td>0.159467</td>\n",
       "      <td>0.197974</td>\n",
       "      <td>0.048420</td>\n",
       "      <td>-0.008985</td>\n",
       "      <td>-0.092494</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.149664</td>\n",
       "      <td>-0.262675</td>\n",
       "      <td>-0.209937</td>\n",
       "      <td>-0.110529</td>\n",
       "      <td>-0.196629</td>\n",
       "      <td>0.109206</td>\n",
       "      <td>-0.048763</td>\n",
       "      <td>0.128882</td>\n",
       "      <td>-0.179656</td>\n",
       "      <td>-0.455001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17306</th>\n",
       "      <td>DGCR2 (9993)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.025596</td>\n",
       "      <td>-0.080611</td>\n",
       "      <td>0.014071</td>\n",
       "      <td>0.141779</td>\n",
       "      <td>-0.044246</td>\n",
       "      <td>0.055804</td>\n",
       "      <td>-0.489630</td>\n",
       "      <td>...</td>\n",
       "      <td>0.170620</td>\n",
       "      <td>-0.012766</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.011414</td>\n",
       "      <td>-0.243745</td>\n",
       "      <td>0.153414</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.159781</td>\n",
       "      <td>0.286456</td>\n",
       "      <td>0.047184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17307</th>\n",
       "      <td>CASP8AP2 (9994)</td>\n",
       "      <td>-0.843295</td>\n",
       "      <td>-0.307031</td>\n",
       "      <td>-0.413669</td>\n",
       "      <td>-1.849696</td>\n",
       "      <td>-0.414154</td>\n",
       "      <td>-0.457898</td>\n",
       "      <td>-0.372166</td>\n",
       "      <td>-0.037574</td>\n",
       "      <td>-0.468437</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.066150</td>\n",
       "      <td>-0.676503</td>\n",
       "      <td>-0.476991</td>\n",
       "      <td>-0.437871</td>\n",
       "      <td>-0.200516</td>\n",
       "      <td>-0.046700</td>\n",
       "      <td>-0.865486</td>\n",
       "      <td>-1.039110</td>\n",
       "      <td>-0.301415</td>\n",
       "      <td>-0.865502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17308</th>\n",
       "      <td>SCO2 (9997)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.122669</td>\n",
       "      <td>-0.078856</td>\n",
       "      <td>0.032661</td>\n",
       "      <td>0.034858</td>\n",
       "      <td>0.016732</td>\n",
       "      <td>0.234413</td>\n",
       "      <td>-0.091241</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.091525</td>\n",
       "      <td>0.187555</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.123356</td>\n",
       "      <td>-0.224527</td>\n",
       "      <td>0.075238</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.100361</td>\n",
       "      <td>-0.117268</td>\n",
       "      <td>-0.036066</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17309 rows × 713 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Unnamed: 0  127399_SOFT_TISSUE  1321N1_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "0             A1BG (1)            0.000000                       0.000000   \n",
       "1            NAT2 (10)            0.000000                       0.000000   \n",
       "2            ADA (100)            0.000000                       0.000000   \n",
       "3          CDH2 (1000)           -0.194962                      -0.028171   \n",
       "4         AKT3 (10000)           -0.256108                       0.100751   \n",
       "...                ...                 ...                            ...   \n",
       "17304   SLC12A6 (9990)            0.000000                       0.000000   \n",
       "17305     KCNE2 (9992)            0.088853                      -0.110628   \n",
       "17306     DGCR2 (9993)            0.000000                       0.000000   \n",
       "17307  CASP8AP2 (9994)           -0.843295                      -0.307031   \n",
       "17308      SCO2 (9997)            0.000000                       0.000000   \n",
       "\n",
       "       143B_BONE  184A1_BREAST  184B5_BREAST  22RV1_PROSTATE  2313287_STOMACH  \\\n",
       "0       0.146042     -0.190388      0.907063       -0.019331        -0.016734   \n",
       "1       0.102854      0.384106      0.403192        0.001925        -0.153933   \n",
       "2       0.168839     -0.120700      0.004394       -0.188700        -0.060818   \n",
       "3       0.063047     -0.237251     -0.017059       -0.103047        -0.049460   \n",
       "4      -0.008077      0.060267     -0.094749       -0.066591         0.166029   \n",
       "...          ...           ...           ...             ...              ...   \n",
       "17304   0.106165      0.227968      0.088744       -0.060731         0.127754   \n",
       "17305   0.057286      0.028095      0.159467        0.197974         0.048420   \n",
       "17306   0.025596     -0.080611      0.014071        0.141779        -0.044246   \n",
       "17307  -0.413669     -1.849696     -0.414154       -0.457898        -0.372166   \n",
       "17308   0.122669     -0.078856      0.032661        0.034858         0.016732   \n",
       "\n",
       "       600MPE_BREAST  697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ...  WM88_SKIN  \\\n",
       "0           0.091580                                0.035023  ...  -0.088267   \n",
       "1          -0.317969                                0.012341  ...  -0.003747   \n",
       "2          -0.755058                                0.129770  ...  -0.014085   \n",
       "3           0.130107                                0.146864  ...  -0.073435   \n",
       "4           0.149969                               -0.053022  ...   0.028714   \n",
       "...              ...                                     ...  ...        ...   \n",
       "17304      -0.049485                                0.381034  ...   0.170658   \n",
       "17305      -0.008985                               -0.092494  ...  -0.149664   \n",
       "17306       0.055804                               -0.489630  ...   0.170620   \n",
       "17307      -0.037574                               -0.468437  ...  -1.066150   \n",
       "17308       0.234413                               -0.091241  ...  -0.091525   \n",
       "\n",
       "       WM983B_SKIN  WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0         0.002171                                     0.000000   \n",
       "1        -0.321445                                     0.000000   \n",
       "2         0.039679                                     0.000000   \n",
       "3        -0.140041                                    -0.154436   \n",
       "4        -0.054628                                     0.450581   \n",
       "...            ...                                          ...   \n",
       "17304    -0.004165                                     0.000000   \n",
       "17305    -0.262675                                    -0.209937   \n",
       "17306    -0.012766                                     0.000000   \n",
       "17307    -0.676503                                    -0.476991   \n",
       "17308     0.187555                                     0.000000   \n",
       "\n",
       "       YD38_UPPER_AERODIGESTIVE_TRACT  YD8_UPPER_AERODIGESTIVE_TRACT  \\\n",
       "0                            0.120294                       0.012540   \n",
       "1                           -0.003256                      -0.220472   \n",
       "2                            0.076521                       0.106995   \n",
       "3                           -0.040308                      -0.078707   \n",
       "4                            0.002932                       0.129679   \n",
       "...                               ...                            ...   \n",
       "17304                        0.118544                       0.273135   \n",
       "17305                       -0.110529                      -0.196629   \n",
       "17306                       -0.011414                      -0.243745   \n",
       "17307                       -0.437871                      -0.200516   \n",
       "17308                        0.123356                      -0.224527   \n",
       "\n",
       "       YKG1_CENTRAL_NERVOUS_SYSTEM  YMB1_BREAST  ZR751_BREAST  ZR7530_BREAST  \\\n",
       "0                         0.111530     0.000000     -0.079313      -0.141559   \n",
       "1                         0.073460     0.000000     -0.130921       0.127358   \n",
       "2                         0.227977     0.000000     -0.134479       0.083506   \n",
       "3                         0.000769    -0.139126      0.047022      -0.097644   \n",
       "4                        -0.072564     0.017161      0.123615       0.046846   \n",
       "...                            ...          ...           ...            ...   \n",
       "17304                     0.141616     0.000000      0.100142       0.160643   \n",
       "17305                     0.109206    -0.048763      0.128882      -0.179656   \n",
       "17306                     0.153414     0.000000      0.159781       0.286456   \n",
       "17307                    -0.046700    -0.865486     -1.039110      -0.301415   \n",
       "17308                     0.075238     0.000000      0.100361      -0.117268   \n",
       "\n",
       "       ZR75B_BREAST  \n",
       "0          0.214268  \n",
       "1         -0.405974  \n",
       "2         -0.404291  \n",
       "3         -0.062622  \n",
       "4          0.125711  \n",
       "...             ...  \n",
       "17304      0.089555  \n",
       "17305     -0.455001  \n",
       "17306      0.047184  \n",
       "17307     -0.865502  \n",
       "17308     -0.036066  \n",
       "\n",
       "[17309 rows x 713 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>127399_SOFT_TISSUE</th>\n",
       "      <th>1321N1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>143B_BONE</th>\n",
       "      <th>184A1_BREAST</th>\n",
       "      <th>184B5_BREAST</th>\n",
       "      <th>22RV1_PROSTATE</th>\n",
       "      <th>2313287_STOMACH</th>\n",
       "      <th>600MPE_BREAST</th>\n",
       "      <th>697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>...</th>\n",
       "      <th>WM88_SKIN</th>\n",
       "      <th>WM983B_SKIN</th>\n",
       "      <th>WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>YD38_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>YD8_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>YKG1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>YMB1_BREAST</th>\n",
       "      <th>ZR751_BREAST</th>\n",
       "      <th>ZR7530_BREAST</th>\n",
       "      <th>ZR75B_BREAST</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.146042</td>\n",
       "      <td>-0.190388</td>\n",
       "      <td>0.907063</td>\n",
       "      <td>-0.019331</td>\n",
       "      <td>-0.016734</td>\n",
       "      <td>0.091580</td>\n",
       "      <td>0.035023</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.088267</td>\n",
       "      <td>0.002171</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.120294</td>\n",
       "      <td>0.012540</td>\n",
       "      <td>0.111530</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.079313</td>\n",
       "      <td>-0.141559</td>\n",
       "      <td>0.214268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NAT2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.102854</td>\n",
       "      <td>0.384106</td>\n",
       "      <td>0.403192</td>\n",
       "      <td>0.001925</td>\n",
       "      <td>-0.153933</td>\n",
       "      <td>-0.317969</td>\n",
       "      <td>0.012341</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.003747</td>\n",
       "      <td>-0.321445</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.003256</td>\n",
       "      <td>-0.220472</td>\n",
       "      <td>0.073460</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.130921</td>\n",
       "      <td>0.127358</td>\n",
       "      <td>-0.405974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ADA</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.168839</td>\n",
       "      <td>-0.120700</td>\n",
       "      <td>0.004394</td>\n",
       "      <td>-0.188700</td>\n",
       "      <td>-0.060818</td>\n",
       "      <td>-0.755058</td>\n",
       "      <td>0.129770</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.014085</td>\n",
       "      <td>0.039679</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.076521</td>\n",
       "      <td>0.106995</td>\n",
       "      <td>0.227977</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.134479</td>\n",
       "      <td>0.083506</td>\n",
       "      <td>-0.404291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CDH2</td>\n",
       "      <td>-0.194962</td>\n",
       "      <td>-0.028171</td>\n",
       "      <td>0.063047</td>\n",
       "      <td>-0.237251</td>\n",
       "      <td>-0.017059</td>\n",
       "      <td>-0.103047</td>\n",
       "      <td>-0.049460</td>\n",
       "      <td>0.130107</td>\n",
       "      <td>0.146864</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.073435</td>\n",
       "      <td>-0.140041</td>\n",
       "      <td>-0.154436</td>\n",
       "      <td>-0.040308</td>\n",
       "      <td>-0.078707</td>\n",
       "      <td>0.000769</td>\n",
       "      <td>-0.139126</td>\n",
       "      <td>0.047022</td>\n",
       "      <td>-0.097644</td>\n",
       "      <td>-0.062622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>-0.256108</td>\n",
       "      <td>0.100751</td>\n",
       "      <td>-0.008077</td>\n",
       "      <td>0.060267</td>\n",
       "      <td>-0.094749</td>\n",
       "      <td>-0.066591</td>\n",
       "      <td>0.166029</td>\n",
       "      <td>0.149969</td>\n",
       "      <td>-0.053022</td>\n",
       "      <td>...</td>\n",
       "      <td>0.028714</td>\n",
       "      <td>-0.054628</td>\n",
       "      <td>0.450581</td>\n",
       "      <td>0.002932</td>\n",
       "      <td>0.129679</td>\n",
       "      <td>-0.072564</td>\n",
       "      <td>0.017161</td>\n",
       "      <td>0.123615</td>\n",
       "      <td>0.046846</td>\n",
       "      <td>0.125711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17304</th>\n",
       "      <td>SLC12A6</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.106165</td>\n",
       "      <td>0.227968</td>\n",
       "      <td>0.088744</td>\n",
       "      <td>-0.060731</td>\n",
       "      <td>0.127754</td>\n",
       "      <td>-0.049485</td>\n",
       "      <td>0.381034</td>\n",
       "      <td>...</td>\n",
       "      <td>0.170658</td>\n",
       "      <td>-0.004165</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.118544</td>\n",
       "      <td>0.273135</td>\n",
       "      <td>0.141616</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.100142</td>\n",
       "      <td>0.160643</td>\n",
       "      <td>0.089555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17305</th>\n",
       "      <td>KCNE2</td>\n",
       "      <td>0.088853</td>\n",
       "      <td>-0.110628</td>\n",
       "      <td>0.057286</td>\n",
       "      <td>0.028095</td>\n",
       "      <td>0.159467</td>\n",
       "      <td>0.197974</td>\n",
       "      <td>0.048420</td>\n",
       "      <td>-0.008985</td>\n",
       "      <td>-0.092494</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.149664</td>\n",
       "      <td>-0.262675</td>\n",
       "      <td>-0.209937</td>\n",
       "      <td>-0.110529</td>\n",
       "      <td>-0.196629</td>\n",
       "      <td>0.109206</td>\n",
       "      <td>-0.048763</td>\n",
       "      <td>0.128882</td>\n",
       "      <td>-0.179656</td>\n",
       "      <td>-0.455001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17306</th>\n",
       "      <td>DGCR2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.025596</td>\n",
       "      <td>-0.080611</td>\n",
       "      <td>0.014071</td>\n",
       "      <td>0.141779</td>\n",
       "      <td>-0.044246</td>\n",
       "      <td>0.055804</td>\n",
       "      <td>-0.489630</td>\n",
       "      <td>...</td>\n",
       "      <td>0.170620</td>\n",
       "      <td>-0.012766</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.011414</td>\n",
       "      <td>-0.243745</td>\n",
       "      <td>0.153414</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.159781</td>\n",
       "      <td>0.286456</td>\n",
       "      <td>0.047184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17307</th>\n",
       "      <td>CASP8AP2</td>\n",
       "      <td>-0.843295</td>\n",
       "      <td>-0.307031</td>\n",
       "      <td>-0.413669</td>\n",
       "      <td>-1.849696</td>\n",
       "      <td>-0.414154</td>\n",
       "      <td>-0.457898</td>\n",
       "      <td>-0.372166</td>\n",
       "      <td>-0.037574</td>\n",
       "      <td>-0.468437</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.066150</td>\n",
       "      <td>-0.676503</td>\n",
       "      <td>-0.476991</td>\n",
       "      <td>-0.437871</td>\n",
       "      <td>-0.200516</td>\n",
       "      <td>-0.046700</td>\n",
       "      <td>-0.865486</td>\n",
       "      <td>-1.039110</td>\n",
       "      <td>-0.301415</td>\n",
       "      <td>-0.865502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17308</th>\n",
       "      <td>SCO2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.122669</td>\n",
       "      <td>-0.078856</td>\n",
       "      <td>0.032661</td>\n",
       "      <td>0.034858</td>\n",
       "      <td>0.016732</td>\n",
       "      <td>0.234413</td>\n",
       "      <td>-0.091241</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.091525</td>\n",
       "      <td>0.187555</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.123356</td>\n",
       "      <td>-0.224527</td>\n",
       "      <td>0.075238</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.100361</td>\n",
       "      <td>-0.117268</td>\n",
       "      <td>-0.036066</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17309 rows × 713 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_name  127399_SOFT_TISSUE  1321N1_CENTRAL_NERVOUS_SYSTEM  143B_BONE  \\\n",
       "0          A1BG            0.000000                       0.000000   0.146042   \n",
       "1          NAT2            0.000000                       0.000000   0.102854   \n",
       "2           ADA            0.000000                       0.000000   0.168839   \n",
       "3          CDH2           -0.194962                      -0.028171   0.063047   \n",
       "4          AKT3           -0.256108                       0.100751  -0.008077   \n",
       "...         ...                 ...                            ...        ...   \n",
       "17304   SLC12A6            0.000000                       0.000000   0.106165   \n",
       "17305     KCNE2            0.088853                      -0.110628   0.057286   \n",
       "17306     DGCR2            0.000000                       0.000000   0.025596   \n",
       "17307  CASP8AP2           -0.843295                      -0.307031  -0.413669   \n",
       "17308      SCO2            0.000000                       0.000000   0.122669   \n",
       "\n",
       "       184A1_BREAST  184B5_BREAST  22RV1_PROSTATE  2313287_STOMACH  \\\n",
       "0         -0.190388      0.907063       -0.019331        -0.016734   \n",
       "1          0.384106      0.403192        0.001925        -0.153933   \n",
       "2         -0.120700      0.004394       -0.188700        -0.060818   \n",
       "3         -0.237251     -0.017059       -0.103047        -0.049460   \n",
       "4          0.060267     -0.094749       -0.066591         0.166029   \n",
       "...             ...           ...             ...              ...   \n",
       "17304      0.227968      0.088744       -0.060731         0.127754   \n",
       "17305      0.028095      0.159467        0.197974         0.048420   \n",
       "17306     -0.080611      0.014071        0.141779        -0.044246   \n",
       "17307     -1.849696     -0.414154       -0.457898        -0.372166   \n",
       "17308     -0.078856      0.032661        0.034858         0.016732   \n",
       "\n",
       "       600MPE_BREAST  697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ...  WM88_SKIN  \\\n",
       "0           0.091580                                0.035023  ...  -0.088267   \n",
       "1          -0.317969                                0.012341  ...  -0.003747   \n",
       "2          -0.755058                                0.129770  ...  -0.014085   \n",
       "3           0.130107                                0.146864  ...  -0.073435   \n",
       "4           0.149969                               -0.053022  ...   0.028714   \n",
       "...              ...                                     ...  ...        ...   \n",
       "17304      -0.049485                                0.381034  ...   0.170658   \n",
       "17305      -0.008985                               -0.092494  ...  -0.149664   \n",
       "17306       0.055804                               -0.489630  ...   0.170620   \n",
       "17307      -0.037574                               -0.468437  ...  -1.066150   \n",
       "17308       0.234413                               -0.091241  ...  -0.091525   \n",
       "\n",
       "       WM983B_SKIN  WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0         0.002171                                     0.000000   \n",
       "1        -0.321445                                     0.000000   \n",
       "2         0.039679                                     0.000000   \n",
       "3        -0.140041                                    -0.154436   \n",
       "4        -0.054628                                     0.450581   \n",
       "...            ...                                          ...   \n",
       "17304    -0.004165                                     0.000000   \n",
       "17305    -0.262675                                    -0.209937   \n",
       "17306    -0.012766                                     0.000000   \n",
       "17307    -0.676503                                    -0.476991   \n",
       "17308     0.187555                                     0.000000   \n",
       "\n",
       "       YD38_UPPER_AERODIGESTIVE_TRACT  YD8_UPPER_AERODIGESTIVE_TRACT  \\\n",
       "0                            0.120294                       0.012540   \n",
       "1                           -0.003256                      -0.220472   \n",
       "2                            0.076521                       0.106995   \n",
       "3                           -0.040308                      -0.078707   \n",
       "4                            0.002932                       0.129679   \n",
       "...                               ...                            ...   \n",
       "17304                        0.118544                       0.273135   \n",
       "17305                       -0.110529                      -0.196629   \n",
       "17306                       -0.011414                      -0.243745   \n",
       "17307                       -0.437871                      -0.200516   \n",
       "17308                        0.123356                      -0.224527   \n",
       "\n",
       "       YKG1_CENTRAL_NERVOUS_SYSTEM  YMB1_BREAST  ZR751_BREAST  ZR7530_BREAST  \\\n",
       "0                         0.111530     0.000000     -0.079313      -0.141559   \n",
       "1                         0.073460     0.000000     -0.130921       0.127358   \n",
       "2                         0.227977     0.000000     -0.134479       0.083506   \n",
       "3                         0.000769    -0.139126      0.047022      -0.097644   \n",
       "4                        -0.072564     0.017161      0.123615       0.046846   \n",
       "...                            ...          ...           ...            ...   \n",
       "17304                     0.141616     0.000000      0.100142       0.160643   \n",
       "17305                     0.109206    -0.048763      0.128882      -0.179656   \n",
       "17306                     0.153414     0.000000      0.159781       0.286456   \n",
       "17307                    -0.046700    -0.865486     -1.039110      -0.301415   \n",
       "17308                     0.075238     0.000000      0.100361      -0.117268   \n",
       "\n",
       "       ZR75B_BREAST  \n",
       "0          0.214268  \n",
       "1         -0.405974  \n",
       "2         -0.404291  \n",
       "3         -0.062622  \n",
       "4          0.125711  \n",
       "...             ...  \n",
       "17304      0.089555  \n",
       "17305     -0.455001  \n",
       "17306      0.047184  \n",
       "17307     -0.865502  \n",
       "17308     -0.036066  \n",
       "\n",
       "[17309 rows x 713 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "raw_rna_combined_df = pd.read_csv('./data/raw_data/D2_combined_gene_dep_scores.csv')\n",
    "# fill NaN values with 0.0\n",
    "raw_rna_combined_df = raw_rna_combined_df.fillna(0.0)\n",
    "display(raw_rna_combined_df)\n",
    "# Move the Unnamed: 0 to a column and rename it as the gene_name\n",
    "raw_rna_combined_df.rename(columns={'Unnamed: 0': 'gene_name'}, inplace=True)\n",
    "# Clean the gene_name column by removing parentheses and extra spaces\n",
    "raw_rna_combined_df['gene_name'] = raw_rna_combined_df['gene_name'].apply(lambda x: x.split('(')[0].strip())\n",
    "# Display the result\n",
    "display(raw_rna_combined_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 376,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CCLE_Name</th>\n",
       "      <th>CCLE_ID</th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Site_Primary</th>\n",
       "      <th>Site_Subtype1</th>\n",
       "      <th>Site_Subtype2</th>\n",
       "      <th>Site_Subtype3</th>\n",
       "      <th>Histology</th>\n",
       "      <th>...</th>\n",
       "      <th>Growth.Medium</th>\n",
       "      <th>Supplements</th>\n",
       "      <th>Freezing.Medium</th>\n",
       "      <th>Doubling.Time.from.Vendor</th>\n",
       "      <th>Doubling.Time.Calculated.hrs</th>\n",
       "      <th>type</th>\n",
       "      <th>type_refined</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>mutRate</th>\n",
       "      <th>tcga_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1321N1_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>1321N1_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>ACH-001000</td>\n",
       "      <td>1321N1</td>\n",
       "      <td>primary</td>\n",
       "      <td>central_nervous_system</td>\n",
       "      <td>brain</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>glioma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glioma</td>\n",
       "      <td>glioma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>143B_BONE</td>\n",
       "      <td>143B_BONE</td>\n",
       "      <td>ACH-001001</td>\n",
       "      <td>143B</td>\n",
       "      <td>primary</td>\n",
       "      <td>bone</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>osteosarcoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>osteosarcoma</td>\n",
       "      <td>osteosarcoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>22RV1_PROSTATE</td>\n",
       "      <td>22RV1_PROSTATE</td>\n",
       "      <td>ACH-000956</td>\n",
       "      <td>22Rv1</td>\n",
       "      <td>primary</td>\n",
       "      <td>prostate</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640+10% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58.4</td>\n",
       "      <td>prostate</td>\n",
       "      <td>prostate</td>\n",
       "      <td>Prostate:Carcinoma</td>\n",
       "      <td>539.830868</td>\n",
       "      <td>PRAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2313287_STOMACH</td>\n",
       "      <td>2313287_STOMACH</td>\n",
       "      <td>ACH-000948</td>\n",
       "      <td>23132/87</td>\n",
       "      <td>primary</td>\n",
       "      <td>stomach</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>90%RPMI 1640+10% h.i FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10%DMSO</td>\n",
       "      <td>50 hrs</td>\n",
       "      <td>97.6</td>\n",
       "      <td>stomach</td>\n",
       "      <td>stomach</td>\n",
       "      <td>Gastric:Carcinoma</td>\n",
       "      <td>521.355253</td>\n",
       "      <td>STAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>ACH-000070</td>\n",
       "      <td>697</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI + 10 % FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46.6</td>\n",
       "      <td>leukemia_other</td>\n",
       "      <td>leukemia_other</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>75.699456</td>\n",
       "      <td>ALL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>597</th>\n",
       "      <td>YD8_UPPER_AERODIGESTIVE_TRACT</td>\n",
       "      <td>YD8_UPPER_AERODIGESTIVE_TRACT</td>\n",
       "      <td>ACH-000630</td>\n",
       "      <td>YD-8</td>\n",
       "      <td>primary</td>\n",
       "      <td>upper_aerodigestive_tract</td>\n",
       "      <td>mouth</td>\n",
       "      <td>tongue</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI 1640  with L-glutamine (300mg/L), 90%; he...</td>\n",
       "      <td>25mM Hepes and 25mM NaHCo3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.3 hrs</td>\n",
       "      <td>NaN</td>\n",
       "      <td>upper_aerodigestive</td>\n",
       "      <td>upper_aerodigestive</td>\n",
       "      <td>Upper_Aerodigestive_Tract:Carcinoma</td>\n",
       "      <td>176.583324</td>\n",
       "      <td>HNSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>598</th>\n",
       "      <td>YKG1_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>YKG1_CENTRAL_NERVOUS_SYSTEM</td>\n",
       "      <td>ACH-000570</td>\n",
       "      <td>YKG1</td>\n",
       "      <td>primary</td>\n",
       "      <td>central_nervous_system</td>\n",
       "      <td>temporal_lobe</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>glioma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glioma</td>\n",
       "      <td>glioma</td>\n",
       "      <td>CNS:Glioma_HighGrade</td>\n",
       "      <td>162.475553</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>YMB1_BREAST</td>\n",
       "      <td>YMB1_BREAST</td>\n",
       "      <td>ACH-001249</td>\n",
       "      <td>YMB-1</td>\n",
       "      <td>primary</td>\n",
       "      <td>breast</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>breast</td>\n",
       "      <td>breast</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>ZR751_BREAST</td>\n",
       "      <td>ZR751_BREAST</td>\n",
       "      <td>ACH-000097</td>\n",
       "      <td>ZR-75-1</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>breast</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640 + 10% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.7</td>\n",
       "      <td>breast</td>\n",
       "      <td>breast</td>\n",
       "      <td>Breast:Carcinoma</td>\n",
       "      <td>65.390119</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>601</th>\n",
       "      <td>ZR7530_BREAST</td>\n",
       "      <td>ZR7530_BREAST</td>\n",
       "      <td>ACH-000828</td>\n",
       "      <td>ZR-75-30</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>breast</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>110 hrs</td>\n",
       "      <td>NaN</td>\n",
       "      <td>breast</td>\n",
       "      <td>breast</td>\n",
       "      <td>Breast:Carcinoma</td>\n",
       "      <td>175.876541</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>602 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  CCLE_Name  \\\n",
       "0             1321N1_CENTRAL_NERVOUS_SYSTEM   \n",
       "1                                 143B_BONE   \n",
       "2                            22RV1_PROSTATE   \n",
       "3                           2313287_STOMACH   \n",
       "4    697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE   \n",
       "..                                      ...   \n",
       "597           YD8_UPPER_AERODIGESTIVE_TRACT   \n",
       "598             YKG1_CENTRAL_NERVOUS_SYSTEM   \n",
       "599                             YMB1_BREAST   \n",
       "600                            ZR751_BREAST   \n",
       "601                           ZR7530_BREAST   \n",
       "\n",
       "                                    CCLE_ID    depMapID      Name   Pathology  \\\n",
       "0             1321N1_CENTRAL_NERVOUS_SYSTEM  ACH-001000    1321N1     primary   \n",
       "1                                 143B_BONE  ACH-001001      143B     primary   \n",
       "2                            22RV1_PROSTATE  ACH-000956     22Rv1     primary   \n",
       "3                           2313287_STOMACH  ACH-000948  23132/87     primary   \n",
       "4    697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  ACH-000070       697     primary   \n",
       "..                                      ...         ...       ...         ...   \n",
       "597           YD8_UPPER_AERODIGESTIVE_TRACT  ACH-000630      YD-8     primary   \n",
       "598             YKG1_CENTRAL_NERVOUS_SYSTEM  ACH-000570      YKG1     primary   \n",
       "599                             YMB1_BREAST  ACH-001249     YMB-1     primary   \n",
       "600                            ZR751_BREAST  ACH-000097   ZR-75-1  metastasis   \n",
       "601                           ZR7530_BREAST  ACH-000828  ZR-75-30  metastasis   \n",
       "\n",
       "                           Site_Primary  Site_Subtype1 Site_Subtype2  \\\n",
       "0                central_nervous_system          brain            NS   \n",
       "1                                  bone             NS            NS   \n",
       "2                              prostate             NS            NS   \n",
       "3                               stomach             NS            NS   \n",
       "4    haematopoietic_and_lymphoid_tissue             NS            NS   \n",
       "..                                  ...            ...           ...   \n",
       "597           upper_aerodigestive_tract          mouth        tongue   \n",
       "598              central_nervous_system  temporal_lobe            NS   \n",
       "599                              breast             NS            NS   \n",
       "600                              breast             NS            NS   \n",
       "601                              breast             NS            NS   \n",
       "\n",
       "    Site_Subtype3          Histology  ...  \\\n",
       "0              NS             glioma  ...   \n",
       "1              NS       osteosarcoma  ...   \n",
       "2              NS          carcinoma  ...   \n",
       "3              NS          carcinoma  ...   \n",
       "4              NS  lymphoid_neoplasm  ...   \n",
       "..            ...                ...  ...   \n",
       "597            NS          carcinoma  ...   \n",
       "598            NS             glioma  ...   \n",
       "599            NS          carcinoma  ...   \n",
       "600            NS          carcinoma  ...   \n",
       "601            NS          carcinoma  ...   \n",
       "\n",
       "                                         Growth.Medium  \\\n",
       "0                                                  NaN   \n",
       "1                                                  NaN   \n",
       "2                                    RPMI-1640+10% FBS   \n",
       "3                             90%RPMI 1640+10% h.i FBS   \n",
       "4                                      RPMI + 10 % FBS   \n",
       "..                                                 ...   \n",
       "597  RPMI 1640  with L-glutamine (300mg/L), 90%; he...   \n",
       "598                                                NaN   \n",
       "599                                                NaN   \n",
       "600                                RPMI-1640 + 10% FBS   \n",
       "601                                                NaN   \n",
       "\n",
       "                    Supplements Freezing.Medium Doubling.Time.from.Vendor  \\\n",
       "0                           NaN             NaN                       NaN   \n",
       "1                           NaN             NaN                       NaN   \n",
       "2                           NaN          5%DMSO                       NaN   \n",
       "3                           NaN         10%DMSO                    50 hrs   \n",
       "4                           NaN         5% DMSO                       NaN   \n",
       "..                          ...             ...                       ...   \n",
       "597  25mM Hepes and 25mM NaHCo3             NaN                  29.3 hrs   \n",
       "598                         NaN             NaN                       NaN   \n",
       "599                         NaN             NaN                       NaN   \n",
       "600                         NaN          5%DMSO                       NaN   \n",
       "601                         NaN             NaN                   110 hrs   \n",
       "\n",
       "    Doubling.Time.Calculated.hrs                 type         type_refined  \\\n",
       "0                            NaN               glioma               glioma   \n",
       "1                            NaN         osteosarcoma         osteosarcoma   \n",
       "2                           58.4             prostate             prostate   \n",
       "3                           97.6              stomach              stomach   \n",
       "4                           46.6       leukemia_other       leukemia_other   \n",
       "..                           ...                  ...                  ...   \n",
       "597                          NaN  upper_aerodigestive  upper_aerodigestive   \n",
       "598                          NaN               glioma               glioma   \n",
       "599                          NaN               breast               breast   \n",
       "600                         88.7               breast               breast   \n",
       "601                          NaN               breast               breast   \n",
       "\n",
       "                  PATHOLOGIST_ANNOTATION     mutRate tcga_code  \n",
       "0                                    NaN         NaN       NaN  \n",
       "1                                    NaN         NaN       NaN  \n",
       "2                     Prostate:Carcinoma  539.830868      PRAD  \n",
       "3                      Gastric:Carcinoma  521.355253      STAD  \n",
       "4                           Leukemia:ALL   75.699456       ALL  \n",
       "..                                   ...         ...       ...  \n",
       "597  Upper_Aerodigestive_Tract:Carcinoma  176.583324      HNSC  \n",
       "598                 CNS:Glioma_HighGrade  162.475553       GBM  \n",
       "599                                  NaN         NaN      BRCA  \n",
       "600                     Breast:Carcinoma   65.390119      BRCA  \n",
       "601                     Breast:Carcinoma  175.876541      BRCA  \n",
       "\n",
       "[602 rows x 34 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'1321N1_CENTRAL_NERVOUS_SYSTEM': 'ACH-001000', '143B_BONE': 'ACH-001001', '22RV1_PROSTATE': 'ACH-000956', '2313287_STOMACH': 'ACH-000948', '697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000070', '769P_KIDNEY': 'ACH-000411', '786O_KIDNEY': 'ACH-000649', '8305C_THYROID': 'ACH-001306', '8505C_THYROID': 'ACH-001307', 'A101D_SKIN': 'ACH-000008', 'A172_CENTRAL_NERVOUS_SYSTEM': 'ACH-000558', 'A204_SOFT_TISSUE': 'ACH-000201', 'A2058_SKIN': 'ACH-000788', 'A253_SALIVARY_GLAND': 'ACH-000740', 'A2780_OVARY': 'ACH-000657', 'A375_SKIN': 'ACH-000219', 'A3KAW_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000697', 'A498_KIDNEY': 'ACH-000555', 'A4FUK_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000157', 'A549_LUNG': 'ACH-000681', 'A673_BONE': 'ACH-000052', 'A704_KIDNEY': 'ACH-000429', 'ABC1_LUNG': 'ACH-000528', 'ACCMESO1_PLEURA': 'ACH-000086', 'ACHN_KIDNEY': 'ACH-000046', 'AGS_STOMACH': 'ACH-000880', 'AM38_CENTRAL_NERVOUS_SYSTEM': 'ACH-000269', 'AML193_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000557', 'AMO1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000838', 'AN3CA_ENDOMETRIUM': 'ACH-000940', 'ASPC1_PANCREAS': 'ACH-000222', 'AU565_BREAST': 'ACH-000248', 'BC3C_URINARY_TRACT': 'ACH-000593', 'BCPAP_THYROID': 'ACH-000456', 'BFTC909_KIDNEY': 'ACH-000792', 'BHT101_THYROID': 'ACH-000191', 'BHY_UPPER_AERODIGESTIVE_TRACT': 'ACH-000548', 'BICR22_UPPER_AERODIGESTIVE_TRACT': 'ACH-000794', 'BICR56_UPPER_AERODIGESTIVE_TRACT': 'ACH-000771', 'BICR6_UPPER_AERODIGESTIVE_TRACT': 'ACH-000415', 'BT20_BREAST': 'ACH-000536', 'BT474_BREAST': 'ACH-000927', 'BT483_BREAST': 'ACH-000818', 'BT549_BREAST': 'ACH-000288', 'BXPC3_PANCREAS': 'ACH-000535', 'C2BBE1_LARGE_INTESTINE': 'ACH-000009', 'C32_SKIN': 'ACH-000580', 'CADOES1_BONE': 'ACH-000210', 'CAKI1_KIDNEY': 'ACH-000433', 'CAKI2_KIDNEY': 'ACH-000234', 'CAL120_BREAST': 'ACH-000212', 'CAL12T_LUNG': 'ACH-000826', 'CAL148_BREAST': 'ACH-000902', 'CAL27_UPPER_AERODIGESTIVE_TRACT': 'ACH-000832', 'CAL29_URINARY_TRACT': 'ACH-000142', 'CAL33_UPPER_AERODIGESTIVE_TRACT': 'ACH-000518', 'CAL51_BREAST': 'ACH-000856', 'CAL54_KIDNEY': 'ACH-000457', 'CAL851_BREAST': 'ACH-000857', 'CALU1_LUNG': 'ACH-000511', 'CALU3_LUNG': 'ACH-000392', 'CALU6_LUNG': 'ACH-000264', 'CAMA1_BREAST': 'ACH-000783', 'CAOV3_OVARY': 'ACH-000713', 'CAOV4_OVARY': 'ACH-000103', 'CAS1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000464', 'CCK81_LARGE_INTESTINE': 'ACH-000963', 'CFPAC1_PANCREAS': 'ACH-000138', 'CHAGOK1_LUNG': 'ACH-000867', 'CJM_SKIN': 'ACH-000458', 'CL11_LARGE_INTESTINE': 'ACH-000249', 'CL34_LARGE_INTESTINE': 'ACH-000895', 'CMK115_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-001036', 'CMK_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000641', 'COLO201_LARGE_INTESTINE': 'ACH-000253', 'COLO205_LARGE_INTESTINE': 'ACH-001039', 'COLO320_LARGE_INTESTINE': 'ACH-000202', 'COLO678_LARGE_INTESTINE': 'ACH-000350', 'COLO679_SKIN': 'ACH-000805', 'COLO680N_OESOPHAGUS': 'ACH-000717', 'COLO684_ENDOMETRIUM': 'ACH-000864', 'COLO699_LUNG': 'ACH-001041', 'COLO704_OVARY': 'ACH-001042', 'COLO741_SKIN': 'ACH-000582', 'COLO783_SKIN': 'ACH-000812', 'COLO800_SKIN': 'ACH-000401', 'COLO829_SKIN': 'ACH-000644', 'CORL105_LUNG': 'ACH-000161', 'CORL23_LUNG': 'ACH-000662', 'CORL279_LUNG': 'ACH-000257', 'CORL88_LUNG': 'ACH-000508', 'COV318_OVARY': 'ACH-000256', 'COV362_OVARY': 'ACH-000278', 'COV434_OVARY': 'ACH-000123', 'COV504_OVARY': 'ACH-001048', 'COV644_OVARY': 'ACH-000608', 'CW2_LARGE_INTESTINE': 'ACH-000998', 'D283MED_CENTRAL_NERVOUS_SYSTEM': 'ACH-000055', 'D341MED_CENTRAL_NERVOUS_SYSTEM': 'ACH-000095', 'DANG_PANCREAS': 'ACH-000243', 'DAOY_CENTRAL_NERVOUS_SYSTEM': 'ACH-000211', 'DBTRG05MG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000863', 'DB_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000334', 'DETROIT562_UPPER_AERODIGESTIVE_TRACT': 'ACH-000207', 'DKMG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000244', 'DLD1_LARGE_INTESTINE': 'ACH-001061', 'DMS114_LUNG': 'ACH-000530', 'DMS273_LUNG': 'ACH-000749', 'DMS53_LUNG': 'ACH-000698', 'DND41_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000981', 'DU145_PROSTATE': 'ACH-000979', 'DU4475_BREAST': 'ACH-000258', 'EBC1_LUNG': 'ACH-000563', 'ECC10_STOMACH': 'ACH-000560', 'ECGI10_OESOPHAGUS': 'ACH-000305', 'EFE184_ENDOMETRIUM': 'ACH-000435', 'EFM192A_BREAST': 'ACH-000117', 'EFM19_BREAST': 'ACH-000330', 'EFO21_OVARY': 'ACH-000308', 'EFO27_OVARY': 'ACH-000936', 'EKVX_LUNG': 'ACH-000706', 'EN_ENDOMETRIUM': 'ACH-000978', 'EPLC272H_LUNG': 'ACH-000585', 'ES2_OVARY': 'ACH-000906', 'ESS1_ENDOMETRIUM': 'ACH-000913', 'EVSAT_BREAST': 'ACH-001065', 'F36P_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000487', 'FADU_UPPER_AERODIGESTIVE_TRACT': 'ACH-000846', 'FU97_STOMACH': 'ACH-000633', 'FUOV1_OVARY': 'ACH-000574', 'G361_SKIN': 'ACH-000572', 'G401_SOFT_TISSUE': 'ACH-000096', 'G402_SOFT_TISSUE': 'ACH-000375', 'GAMG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000098', 'GB1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000738', 'GI1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000756', 'GP2D_LARGE_INTESTINE': 'ACH-000982', 'GSS_STOMACH': 'ACH-000746', 'GSU_STOMACH': 'ACH-000485', 'HCC1143_BREAST': 'ACH-000374', 'HCC1187_BREAST': 'ACH-000111', 'HCC1359_LUNG': 'ACH-000901', 'HCC1395_BREAST': 'ACH-000699', 'HCC1419_BREAST': 'ACH-000277', 'HCC1428_BREAST': 'ACH-000352', 'HCC1500_BREAST': 'ACH-000349', 'HCC1569_BREAST': 'ACH-000930', 'HCC1599_BREAST': 'ACH-000196', 'HCC15_LUNG': 'ACH-000878', 'HCC1806_BREAST': 'ACH-000624', 'HCC1833_LUNG': 'ACH-000712', 'HCC1937_BREAST': 'ACH-000223', 'HCC1954_BREAST': 'ACH-000859', 'HCC202_BREAST': 'ACH-000725', 'HCC2218_BREAST': 'ACH-000755', 'HCC33_LUNG': 'ACH-000515', 'HCC366_LUNG': 'ACH-000840', 'HCC38_BREAST': 'ACH-000276', 'HCC4006_LUNG': 'ACH-000066', 'HCC44_LUNG': 'ACH-000667', 'HCC70_BREAST': 'ACH-000668', 'HCC827_LUNG': 'ACH-000012', 'HCC95_LUNG': 'ACH-000454', 'HCT116_LARGE_INTESTINE': 'ACH-000971', 'HCT15_LARGE_INTESTINE': 'ACH-000997', 'HDQP1_BREAST': 'ACH-000643', 'HEC1A_ENDOMETRIUM': 'ACH-000954', 'HEC251_ENDOMETRIUM': 'ACH-000996', 'HEC265_ENDOMETRIUM': 'ACH-000946', 'HEC50B_ENDOMETRIUM': 'ACH-000831', 'HEC59_ENDOMETRIUM': 'ACH-000994', 'HEC6_ENDOMETRIUM': 'ACH-000984', 'HEL9217_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000005', 'HEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000004', 'HEP3B217_LIVER': 'ACH-000625', 'HEPG2_LIVER': 'ACH-000739', 'HEYA8_OVARY': 'ACH-000542', 'HGC27_STOMACH': 'ACH-000847', 'HL60_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000002', 'HLC1_LUNG': 'ACH-001088', 'HLE_LIVER': 'ACH-001089', 'HLF_LIVER': 'ACH-000393', 'HMC18_BREAST': 'ACH-000721', 'HOP62_LUNG': 'ACH-000861', 'HPAC_PANCREAS': 'ACH-000270', 'HPAFII_PANCREAS': 'ACH-000094', 'HRT18_LARGE_INTESTINE': 'ACH-001091', 'HS294T_SKIN': 'ACH-000014', 'HS578T_BREAST': 'ACH-000148', 'HS683_CENTRAL_NERVOUS_SYSTEM': 'ACH-000067', 'HS695T_SKIN': 'ACH-000799', 'HS729_SOFT_TISSUE': 'ACH-000133', 'HS766T_PANCREAS': 'ACH-000178', 'HS852T_SKIN': 'ACH-000274', 'HS939T_SKIN': 'ACH-000814', 'HS944T_SKIN': 'ACH-000632', 'HSC2_UPPER_AERODIGESTIVE_TRACT': 'ACH-000472', 'HSC3_UPPER_AERODIGESTIVE_TRACT': 'ACH-000778', 'HT1080_SOFT_TISSUE': 'ACH-000054', 'HT115_LARGE_INTESTINE': 'ACH-000986', 'HT1197_URINARY_TRACT': 'ACH-000547', 'HT1376_URINARY_TRACT': 'ACH-000724', 'HT144_SKIN': 'ACH-000322', 'HT29_LARGE_INTESTINE': 'ACH-000552', 'HT55_LARGE_INTESTINE': 'ACH-000926', 'HUG1N_STOMACH': 'ACH-000239', 'HUH1_LIVER': 'ACH-000475', 'HUH6_LIVER': 'ACH-000671', 'HUH7_LIVER': 'ACH-000480', 'HUPT3_PANCREAS': 'ACH-000118', 'HUPT4_PANCREAS': 'ACH-000213', 'HUTU80_SMALL_INTESTINE': 'ACH-000538', 'IALM_LUNG': 'ACH-000672', 'IGR1_SKIN': 'ACH-000882', 'IGR37_SKIN': 'ACH-000650', 'IGR39_SKIN': 'ACH-000550', 'IGROV1_OVARY': 'ACH-000966', 'IM95_STOMACH': 'ACH-000919', 'IPC298_SKIN': 'ACH-000915', 'ISHIKAWAHERAKLIO02ER_ENDOMETRIUM': 'ACH-000961', 'JHH1_LIVER': 'ACH-000620', 'JHH5_LIVER': 'ACH-000734', 'JHH6_LIVER': 'ACH-000217', 'JHH7_LIVER': 'ACH-000848', 'JHOC5_OVARY': 'ACH-000324', 'JHOM1_OVARY': 'ACH-000237', 'JHOS2_OVARY': 'ACH-000132', 'JHUEM2_ENDOMETRIUM': 'ACH-000909', 'JIMT1_BREAST': 'ACH-000711', 'JJN3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000653', 'JL1_PLEURA': 'ACH-000645', 'JMSU1_URINARY_TRACT': 'ACH-000753', 'K029AX_SKIN': 'ACH-000404', 'K562_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000551', 'KALS1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000231', 'KARPAS422_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000315', 'KASUMI1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000263', 'KASUMI2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000728', 'KATOIII_STOMACH': 'ACH-000793', 'KE39_STOMACH': 'ACH-000507', 'KELLY_AUTONOMIC_GANGLIA': 'ACH-000259', 'KM12_LARGE_INTESTINE': 'ACH-000969', 'KMM1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000889', 'KMRC1_KIDNEY': 'ACH-000684', 'KMRC20_KIDNEY': 'ACH-000250', 'KMRC2_KIDNEY': 'ACH-000709', 'KMS12BM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000380', 'KMS18_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000658', 'KMS27_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000576', 'KMS28BM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000419', 'KMS34_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000541', 'KNS42_CENTRAL_NERVOUS_SYSTEM': 'ACH-000622', 'KNS60_CENTRAL_NERVOUS_SYSTEM': 'ACH-000445', 'KNS62_LUNG': 'ACH-000858', 'KNS81_CENTRAL_NERVOUS_SYSTEM': 'ACH-000479', 'KP1NL_PANCREAS': 'ACH-001108', 'KP1N_PANCREAS': 'ACH-001107', 'KP2_PANCREAS': 'ACH-000281', 'KP3_PANCREAS': 'ACH-000108', 'KP4_PANCREAS': 'ACH-000265', 'KPL1_BREAST': 'ACH-000028', 'KPNSI9S_AUTONOMIC_GANGLIA': 'ACH-000446', 'KPNYN_AUTONOMIC_GANGLIA': 'ACH-000227', 'KS1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000631', 'KU1919_URINARY_TRACT': 'ACH-000486', 'KURAMOCHI_OVARY': 'ACH-000524', 'KYM1_SOFT_TISSUE': 'ACH-000607', 'KYSE140_OESOPHAGUS': 'ACH-000823', 'KYSE150_OESOPHAGUS': 'ACH-000855', 'KYSE180_OESOPHAGUS': 'ACH-000693', 'KYSE30_OESOPHAGUS': 'ACH-000777', 'KYSE410_OESOPHAGUS': 'ACH-000809', 'KYSE450_OESOPHAGUS': 'ACH-000865', 'KYSE510_OESOPHAGUS': 'ACH-000824', 'KYSE520_OESOPHAGUS': 'ACH-000637', 'KYSE70_OESOPHAGUS': 'ACH-000784', 'L33_PANCREAS': 'ACH-000685', 'L363_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000183', 'LAMA84_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000301', 'LC1SQSF_LUNG': 'ACH-001113', 'LCLC103H_LUNG': 'ACH-000627', 'LCLC97TM1_LUNG': 'ACH-000596', 'LI7_LIVER': 'ACH-000471', 'LK2_LUNG': 'ACH-000769', 'LMSU_STOMACH': 'ACH-000255', 'LN18_CENTRAL_NERVOUS_SYSTEM': 'ACH-000819', 'LN229_CENTRAL_NERVOUS_SYSTEM': 'ACH-000595', 'LNCAPCLONEFGC_PROSTATE': 'ACH-000977', 'LOUNH91_LUNG': 'ACH-000176', 'LOVO_LARGE_INTESTINE': 'ACH-000950', 'LOXIMVI_SKIN': 'ACH-000750', 'LS180_LARGE_INTESTINE': 'ACH-000957', 'LS411N_LARGE_INTESTINE': 'ACH-000985', 'LS513_LARGE_INTESTINE': 'ACH-000007', 'LU65_LUNG': 'ACH-000438', 'LU99_LUNG': 'ACH-000444', 'LUDLU1_LUNG': 'ACH-000390', 'LXF289_LUNG': 'ACH-000787', 'M059K_CENTRAL_NERVOUS_SYSTEM': 'ACH-000152', 'MB157_BREAST': 'ACH-001120', 'MCF7_BREAST': 'ACH-000019', 'MDAMB134VI_BREAST': 'ACH-000044', 'MDAMB157_BREAST': 'ACH-000621', 'MDAMB175VII_BREAST': 'ACH-000759', 'MDAMB231_BREAST': 'ACH-000768', 'MDAMB361_BREAST': 'ACH-000934', 'MDAMB415_BREAST': 'ACH-000876', 'MDAMB435S_SKIN': 'ACH-000884', 'MDAMB436_BREAST': 'ACH-000573', 'MDAMB453_BREAST': 'ACH-000910', 'MDAMB468_BREAST': 'ACH-000849', 'MDAPCA2B_PROSTATE': 'ACH-000952', 'MDST8_LARGE_INTESTINE': 'ACH-000935', 'MELHO_SKIN': 'ACH-000450', 'MELJUSO_SKIN': 'ACH-000881', 'MEWO_SKIN': 'ACH-000987', 'MFE280_ENDOMETRIUM': 'ACH-000192', 'MFE296_ENDOMETRIUM': 'ACH-000879', 'MFE319_ENDOMETRIUM': 'ACH-000988', 'MHHES1_BONE': 'ACH-000391', 'MIAPACA2_PANCREAS': 'ACH-000601', 'MKN1_STOMACH': 'ACH-000351', 'MKN45_STOMACH': 'ACH-000356', 'MKN7_STOMACH': 'ACH-000678', 'MM1S_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000763', 'MOLM13_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000362', 'MOLM16_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000369', 'MONOMAC1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-001129', 'MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000006', 'MORCPR_LUNG': 'ACH-000851', 'MPP89_PLEURA': 'ACH-000319', 'MV411_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000045', 'NALM6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000938', 'NB1_AUTONOMIC_GANGLIA': 'ACH-000804', 'NB4_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000294', 'NCCSTCK140_STOMACH': 'ACH-000110', 'NCIH1048_LUNG': 'ACH-000866', 'NCIH1105_LUNG': 'ACH-000780', 'NCIH1155_LUNG': 'ACH-000980', 'NCIH1299_LUNG': 'ACH-000510', 'NCIH1339_LUNG': 'ACH-000921', 'NCIH1355_LUNG': 'ACH-000666', 'NCIH1373_LUNG': 'ACH-000845', 'NCIH1395_LUNG': 'ACH-000327', 'NCIH1435_LUNG': 'ACH-000852', 'NCIH1437_LUNG': 'ACH-000589', 'NCIH146_LUNG': 'ACH-000506', 'NCIH1568_LUNG': 'ACH-000869', 'NCIH1573_LUNG': 'ACH-000916', 'NCIH1581_LUNG': 'ACH-000015', 'NCIH1623_LUNG': 'ACH-000744', 'NCIH1648_LUNG': 'ACH-000766', 'NCIH1650_LUNG': 'ACH-000035', 'NCIH1666_LUNG': 'ACH-000448', 'NCIH1693_LUNG': 'ACH-000021', 'NCIH1694_LUNG': 'ACH-000431', 'NCIH1703_LUNG': 'ACH-000747', 'NCIH1792_LUNG': 'ACH-000496', 'NCIH1793_LUNG': 'ACH-000888', 'NCIH1836_LUNG': 'ACH-000559', 'NCIH1838_LUNG': 'ACH-000733', 'NCIH1876_LUNG': 'ACH-000586', 'NCIH1915_LUNG': 'ACH-000434', 'NCIH1944_LUNG': 'ACH-000414', 'NCIH196_LUNG': 'ACH-000752', 'NCIH1975_LUNG': 'ACH-000587', 'NCIH2009_LUNG': 'ACH-000886', 'NCIH2023_LUNG': 'ACH-000781', 'NCIH2030_LUNG': 'ACH-000521', 'NCIH2052_PLEURA': 'ACH-000153', 'NCIH2066_LUNG': 'ACH-000727', 'NCIH2087_LUNG': 'ACH-000841', 'NCIH209_LUNG': 'ACH-000290', 'NCIH2110_LUNG': 'ACH-000929', 'NCIH211_LUNG': 'ACH-000639', 'NCIH2122_LUNG': 'ACH-000311', 'NCIH2126_LUNG': 'ACH-000785', 'NCIH2170_LUNG': 'ACH-000481', 'NCIH2171_LUNG': 'ACH-000525', 'NCIH2172_LUNG': 'ACH-000924', 'NCIH2228_LUNG': 'ACH-000447', 'NCIH2286_LUNG': 'ACH-000912', 'NCIH2291_LUNG': 'ACH-000718', 'NCIH23_LUNG': 'ACH-000900', 'NCIH2405_LUNG': 'ACH-000121', 'NCIH2452_PLEURA': 'ACH-000092', 'NCIH28_PLEURA': 'ACH-000648', 'NCIH292_LUNG': 'ACH-001075', 'NCIH322_LUNG': 'ACH-000837', 'NCIH3255_LUNG': 'ACH-000109', 'NCIH358_LUNG': 'ACH-000860', 'NCIH441_LUNG': 'ACH-000638', 'NCIH446_LUNG': 'ACH-000800', 'NCIH460_LUNG': 'ACH-000463', 'NCIH508_LARGE_INTESTINE': 'ACH-000360', 'NCIH520_LUNG': 'ACH-000395', 'NCIH522_LUNG': 'ACH-000343', 'NCIH526_LUNG': 'ACH-000767', 'NCIH647_LUNG': 'ACH-000378', 'NCIH650_LUNG': 'ACH-000945', 'NCIH660_PROSTATE': 'ACH-000177', 'NCIH661_LUNG': 'ACH-000853', 'NCIH716_LARGE_INTESTINE': 'ACH-000491', 'NCIH727_LUNG': 'ACH-000775', 'NCIH747_LARGE_INTESTINE': 'ACH-000403', 'NCIH838_LUNG': 'ACH-000416', 'NCIH841_LUNG': 'ACH-000292', 'NCIH854_LUNG': 'ACH-000710', 'NCIN87_STOMACH': 'ACH-000427', 'NIHOVCAR3_OVARY': 'ACH-000001', 'NOMO1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000168', 'NUGC3_STOMACH': 'ACH-000911', 'OAW42_OVARY': 'ACH-000704', 'OC314_OVARY': 'ACH-000962', 'OCIAML2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000113', 'OCIAML3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000336', 'OCIAML5_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000065', 'OCUM1_STOMACH': 'ACH-000247', 'OE21_OESOPHAGUS': 'ACH-000544', 'OPM2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000024', 'OSRC2_KIDNEY': 'ACH-000159', 'OUMS23_LARGE_INTESTINE': 'ACH-000296', 'OV7_OVARY': 'ACH-000688', 'OV90_OVARY': 'ACH-000291', 'OVCAR4_OVARY': 'ACH-000617', 'OVCAR8_OVARY': 'ACH-000696', 'OVISE_OVARY': 'ACH-000527', 'OVSAHO_OVARY': 'ACH-000409', 'OVTOKO_OVARY': 'ACH-000663', 'PANC0203_PANCREAS': 'ACH-000042', 'PANC0327_PANCREAS': 'ACH-000139', 'PANC0403_PANCREAS': 'ACH-000235', 'PANC0504_PANCREAS': 'ACH-000093', 'PANC0813_PANCREAS': 'ACH-000417', 'PANC1005_PANCREAS': 'ACH-000060', 'PANC1_PANCREAS': 'ACH-000164', 'PATU8902_PANCREAS': 'ACH-000599', 'PATU8988T_PANCREAS': 'ACH-000023', 'PC14_LUNG': 'ACH-000030', 'PC3_PROSTATE': 'ACH-000090', 'PECAPJ34CLONEC12_UPPER_AERODIGESTIVE_TRACT': 'ACH-000606', 'PECAPJ41CLONED2_UPPER_AERODIGESTIVE_TRACT': 'ACH-000732', 'PECAPJ49_UPPER_AERODIGESTIVE_TRACT': 'ACH-000735', 'PFEIFFER_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000140', 'PK1_PANCREAS': 'ACH-000307', 'PLCPRF5_LIVER': 'ACH-001318', 'PSN1_PANCREAS': 'ACH-000320', 'QGP1_PANCREAS': 'ACH-000347', 'RCHACV_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000922', 'RDES_BONE': 'ACH-000041', 'RD_SOFT_TISSUE': 'ACH-000169', 'REH_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000960', 'RERFLCAD2_LUNG': 'ACH-000774', 'RERFLCAI_LUNG': 'ACH-000261', 'RERFLCMS_LUNG': 'ACH-000062', 'RH30_SOFT_TISSUE': 'ACH-000833', 'RH41_SOFT_TISSUE': 'ACH-000100', 'RKN_SOFT_TISSUE': 'ACH-000505', 'RKO_LARGE_INTESTINE': 'ACH-000943', 'RL952_ENDOMETRIUM': 'ACH-000965', 'RL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000371', 'RMGI_OVARY': 'ACH-000719', 'RMUGS_OVARY': 'ACH-000701', 'RPMI7951_SKIN': 'ACH-000348', 'RPMI8226_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000817', 'RS411_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000874', 'RT112_URINARY_TRACT': 'ACH-000473', 'RVH421_SKIN': 'ACH-000614', 'SAOS2_BONE': 'ACH-000410', 'SBC5_LUNG': 'ACH-000670', 'SCC25_UPPER_AERODIGESTIVE_TRACT': 'ACH-000188', 'SCC4_UPPER_AERODIGESTIVE_TRACT': 'ACH-000238', 'SEM_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000782', 'SF126_CENTRAL_NERVOUS_SYSTEM': 'ACH-000609', 'SF268_CENTRAL_NERVOUS_SYSTEM': 'ACH-000655', 'SF295_CENTRAL_NERVOUS_SYSTEM': 'ACH-000376', 'SH4_SKIN': 'ACH-000441', 'SHP77_LUNG': 'ACH-000790', 'SJRH30_SOFT_TISSUE': 'ACH-001189', 'SJSA1_BONE': 'ACH-000748', 'SKBR3_BREAST': 'ACH-000017', 'SKCO1_LARGE_INTESTINE': 'ACH-000400', 'SKES1_BONE': 'ACH-000087', 'SKHEP1_LIVER': 'ACH-000361', 'SKMEL28_SKIN': 'ACH-000615', 'SKMEL2_SKIN': 'ACH-001190', 'SKMEL30_SKIN': 'ACH-000810', 'SKMEL3_SKIN': 'ACH-000423', 'SKMEL5_SKIN': 'ACH-000730', 'SKMES1_LUNG': 'ACH-000665', 'SKMM2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000363', 'SKNAS_AUTONOMIC_GANGLIA': 'ACH-000260', 'SKNBE2_AUTONOMIC_GANGLIA': 'ACH-000312', 'SKNDZ_AUTONOMIC_GANGLIA': 'ACH-000366', 'SKNFI_AUTONOMIC_GANGLIA': 'ACH-000341', 'SKNMC_BONE': 'ACH-000039', 'SKNSH_AUTONOMIC_GANGLIA': 'ACH-000149', 'SKOV3_OVARY': 'ACH-000811', 'SLR20_KIDNEY': 'ACH-000127', 'SLR21_KIDNEY': 'ACH-000016', 'SLR23_KIDNEY': 'ACH-000246', 'SLR24_KIDNEY': 'ACH-000272', 'SLR25_KIDNEY': 'ACH-000300', 'SLR26_KIDNEY': 'ACH-000600', 'SNB19_CENTRAL_NERVOUS_SYSTEM': 'ACH-001198', 'SNGM_ENDOMETRIUM': 'ACH-000974', 'SNU1076_UPPER_AERODIGESTIVE_TRACT': 'ACH-000549', 'SNU1079_BILIARY_TRACT': 'ACH-000209', 'SNU1105_CENTRAL_NERVOUS_SYSTEM': 'ACH-000368', 'SNU1_STOMACH': 'ACH-000932', 'SNU216_STOMACH': 'ACH-000466', 'SNU349_KIDNEY': 'ACH-000907', 'SNU407_LARGE_INTESTINE': 'ACH-000955', 'SNU410_PANCREAS': 'ACH-000517', 'SNU423_LIVER': 'ACH-000493', 'SNU449_LIVER': 'ACH-000420', 'SNU5_STOMACH': 'ACH-000303', 'SNU601_STOMACH': 'ACH-000736', 'SNU61_LARGE_INTESTINE': 'ACH-000532', 'SNU685_ENDOMETRIUM': 'ACH-000407', 'SNU719_STOMACH': 'ACH-000898', 'SNU738_CENTRAL_NERVOUS_SYSTEM': 'ACH-000807', 'SNU761_LIVER': 'ACH-000537', 'SNU81_LARGE_INTESTINE': 'ACH-000991', 'SNU840_OVARY': 'ACH-000280', 'SNU886_LIVER': 'ACH-000316', 'SNU8_OVARY': 'ACH-000460', 'SNUC1_LARGE_INTESTINE': 'ACH-000722', 'SNUC2A_LARGE_INTESTINE': 'ACH-000967', 'SNUC4_LARGE_INTESTINE': 'ACH-000959', 'SQ1_LUNG': 'ACH-000553', 'SU8686_PANCREAS': 'ACH-000114', 'SUDHL10_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000271', 'SUDHL4_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000365', 'SUDHL6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000611', 'SUIT2_PANCREAS': 'ACH-000652', 'SUPT1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000953', 'SW1088_CENTRAL_NERVOUS_SYSTEM': 'ACH-000437', 'SW1271_LUNG': 'ACH-000890', 'SW1353_BONE': 'ACH-000418', 'SW1417_LARGE_INTESTINE': 'ACH-000236', 'SW1463_LARGE_INTESTINE': 'ACH-000470', 'SW1573_LUNG': 'ACH-000677', 'SW1783_CENTRAL_NERVOUS_SYSTEM': 'ACH-000883', 'SW1990_PANCREAS': 'ACH-000155', 'SW403_LARGE_INTESTINE': 'ACH-000820', 'SW480_LARGE_INTESTINE': 'ACH-000842', 'SW48_LARGE_INTESTINE': 'ACH-000958', 'SW579_THYROID': 'ACH-000163', 'SW620_LARGE_INTESTINE': 'ACH-000651', 'SW780_URINARY_TRACT': 'ACH-000384', 'SW837_LARGE_INTESTINE': 'ACH-000421', 'SW948_LARGE_INTESTINE': 'ACH-000680', 'T24_URINARY_TRACT': 'ACH-000018', 'T47D_BREAST': 'ACH-000147', 'T98G_CENTRAL_NERVOUS_SYSTEM': 'ACH-000571', 'TC71_BONE': 'ACH-000424', 'TCCPAN2_PANCREAS': 'ACH-000502', 'TCCSUP_URINARY_TRACT': 'ACH-000720', 'TE10_OESOPHAGUS': 'ACH-000318', 'TE11_OESOPHAGUS': 'ACH-000488', 'TE14_OESOPHAGUS': 'ACH-000726', 'TE15_OESOPHAGUS': 'ACH-000353', 'TE1_OESOPHAGUS': 'ACH-000647', 'TE4_OESOPHAGUS': 'ACH-000917', 'TE6_OESOPHAGUS': 'ACH-000605', 'TE8_OESOPHAGUS': 'ACH-000452', 'TE9_OESOPHAGUS': 'ACH-000694', 'TEN_ENDOMETRIUM': 'ACH-000397', 'THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000146', 'TOV112D_OVARY': 'ACH-000048', 'TOV21G_OVARY': 'ACH-000885', 'TUHR10TKB_KIDNEY': 'ACH-000459', 'TUHR14TKB_KIDNEY': 'ACH-000317', 'TUHR4TKB_KIDNEY': 'ACH-000495', 'TYKNU_OVARY': 'ACH-000430', 'U118MG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000040', 'U251MG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000232', 'U2OS_BONE': 'ACH-000364', 'U87MG_CENTRAL_NERVOUS_SYSTEM': 'ACH-000075', 'U937_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000406', 'UACC257_SKIN': 'ACH-000579', 'UACC62_SKIN': 'ACH-000425', 'UACC812_BREAST': 'ACH-000568', 'UACC893_BREAST': 'ACH-000554', 'UBLC1_URINARY_TRACT': 'ACH-000741', 'UMUC3_URINARY_TRACT': 'ACH-000522', 'VCAP_PROSTATE': 'ACH-000115', 'VMRCRCW_KIDNEY': 'ACH-000484', 'WM115_SKIN': 'ACH-000304', 'WM1799_SKIN': 'ACH-000661', 'WM2664_SKIN': 'ACH-001239', 'WM793_SKIN': 'ACH-000827', 'WM88_SKIN': 'ACH-000899', 'WM983B_SKIN': 'ACH-000765', 'WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE': 'ACH-000534', 'YD38_UPPER_AERODIGESTIVE_TRACT': 'ACH-000762', 'YD8_UPPER_AERODIGESTIVE_TRACT': 'ACH-000630', 'YKG1_CENTRAL_NERVOUS_SYSTEM': 'ACH-000570', 'YMB1_BREAST': 'ACH-001249', 'ZR751_BREAST': 'ACH-000097', 'ZR7530_BREAST': 'ACH-000828'}\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001089</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001113</th>\n",
       "      <th>ACH-001129</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001239</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "      <th>ACH-001318</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.036726</td>\n",
       "      <td>0.231996</td>\n",
       "      <td>-0.136186</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.570520</td>\n",
       "      <td>-0.012940</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.664572</td>\n",
       "      <td>-0.084851</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.044965</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.188873</td>\n",
       "      <td>-0.008834</td>\n",
       "      <td>-0.042931</td>\n",
       "      <td>-0.326934</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.365715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NAT2</td>\n",
       "      <td>0.191694</td>\n",
       "      <td>-0.311668</td>\n",
       "      <td>-0.193766</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.169935</td>\n",
       "      <td>-0.290193</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.086928</td>\n",
       "      <td>-0.018109</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.120774</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.193814</td>\n",
       "      <td>-0.079915</td>\n",
       "      <td>0.146653</td>\n",
       "      <td>-0.107795</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.083265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ADA</td>\n",
       "      <td>0.223975</td>\n",
       "      <td>-0.576057</td>\n",
       "      <td>0.131067</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.055516</td>\n",
       "      <td>-0.282557</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.164582</td>\n",
       "      <td>0.320907</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.214594</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.136263</td>\n",
       "      <td>0.135128</td>\n",
       "      <td>-0.075160</td>\n",
       "      <td>0.029538</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.183369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CDH2</td>\n",
       "      <td>0.025377</td>\n",
       "      <td>0.072624</td>\n",
       "      <td>0.172407</td>\n",
       "      <td>-0.209676</td>\n",
       "      <td>-0.060467</td>\n",
       "      <td>0.136671</td>\n",
       "      <td>0.095038</td>\n",
       "      <td>0.193100</td>\n",
       "      <td>-0.105455</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.070584</td>\n",
       "      <td>-0.286068</td>\n",
       "      <td>-0.017196</td>\n",
       "      <td>-0.024994</td>\n",
       "      <td>-0.024949</td>\n",
       "      <td>0.221152</td>\n",
       "      <td>-0.166268</td>\n",
       "      <td>-0.295108</td>\n",
       "      <td>-0.101788</td>\n",
       "      <td>0.158220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>0.167998</td>\n",
       "      <td>-0.253519</td>\n",
       "      <td>0.114353</td>\n",
       "      <td>0.105309</td>\n",
       "      <td>0.028471</td>\n",
       "      <td>0.140191</td>\n",
       "      <td>0.084754</td>\n",
       "      <td>-0.010944</td>\n",
       "      <td>0.133982</td>\n",
       "      <td>...</td>\n",
       "      <td>0.019461</td>\n",
       "      <td>-0.583573</td>\n",
       "      <td>0.069328</td>\n",
       "      <td>0.210320</td>\n",
       "      <td>0.304147</td>\n",
       "      <td>0.040289</td>\n",
       "      <td>-0.121801</td>\n",
       "      <td>-0.005521</td>\n",
       "      <td>-0.021668</td>\n",
       "      <td>0.183108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17304</th>\n",
       "      <td>SLC12A6</td>\n",
       "      <td>0.111539</td>\n",
       "      <td>0.029433</td>\n",
       "      <td>0.040104</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.044931</td>\n",
       "      <td>0.076166</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.123799</td>\n",
       "      <td>-0.093000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.147727</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.038196</td>\n",
       "      <td>-0.007167</td>\n",
       "      <td>0.046325</td>\n",
       "      <td>0.144315</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.098176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17305</th>\n",
       "      <td>KCNE2</td>\n",
       "      <td>-0.112684</td>\n",
       "      <td>0.059434</td>\n",
       "      <td>-0.071567</td>\n",
       "      <td>-0.166129</td>\n",
       "      <td>-0.079283</td>\n",
       "      <td>0.042259</td>\n",
       "      <td>-0.250895</td>\n",
       "      <td>0.207326</td>\n",
       "      <td>-0.050655</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.142186</td>\n",
       "      <td>0.139844</td>\n",
       "      <td>0.078532</td>\n",
       "      <td>0.392288</td>\n",
       "      <td>-0.074156</td>\n",
       "      <td>-0.054130</td>\n",
       "      <td>-0.137692</td>\n",
       "      <td>0.114692</td>\n",
       "      <td>0.060095</td>\n",
       "      <td>-0.038330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17306</th>\n",
       "      <td>DGCR2</td>\n",
       "      <td>0.050646</td>\n",
       "      <td>-0.323283</td>\n",
       "      <td>0.120482</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.117080</td>\n",
       "      <td>0.177101</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.353728</td>\n",
       "      <td>-0.366567</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.018230</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.234106</td>\n",
       "      <td>0.115288</td>\n",
       "      <td>-0.212524</td>\n",
       "      <td>0.223209</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.109311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17307</th>\n",
       "      <td>CASP8AP2</td>\n",
       "      <td>-1.228253</td>\n",
       "      <td>-0.414654</td>\n",
       "      <td>-0.050006</td>\n",
       "      <td>0.076652</td>\n",
       "      <td>-0.360068</td>\n",
       "      <td>-0.868595</td>\n",
       "      <td>-0.461431</td>\n",
       "      <td>-0.301565</td>\n",
       "      <td>-0.543784</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.433340</td>\n",
       "      <td>-0.293064</td>\n",
       "      <td>-0.454495</td>\n",
       "      <td>-0.867408</td>\n",
       "      <td>-0.930363</td>\n",
       "      <td>-0.828188</td>\n",
       "      <td>-0.138393</td>\n",
       "      <td>-0.983701</td>\n",
       "      <td>-0.626359</td>\n",
       "      <td>-0.440665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17308</th>\n",
       "      <td>SCO2</td>\n",
       "      <td>-0.077570</td>\n",
       "      <td>0.475836</td>\n",
       "      <td>-0.148953</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.088060</td>\n",
       "      <td>0.246722</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.307953</td>\n",
       "      <td>-0.172032</td>\n",
       "      <td>...</td>\n",
       "      <td>0.170232</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.082845</td>\n",
       "      <td>-0.073901</td>\n",
       "      <td>0.001383</td>\n",
       "      <td>0.377378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.080429</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17309 rows × 593 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_name  ACH-000001  ACH-000002  ACH-000004  ACH-000005  ACH-000006  \\\n",
       "0          A1BG    0.036726    0.231996   -0.136186    0.000000   -0.570520   \n",
       "1          NAT2    0.191694   -0.311668   -0.193766    0.000000   -0.169935   \n",
       "2           ADA    0.223975   -0.576057    0.131067    0.000000   -0.055516   \n",
       "3          CDH2    0.025377    0.072624    0.172407   -0.209676   -0.060467   \n",
       "4          AKT3    0.167998   -0.253519    0.114353    0.105309    0.028471   \n",
       "...         ...         ...         ...         ...         ...         ...   \n",
       "17304   SLC12A6    0.111539    0.029433    0.040104    0.000000    0.044931   \n",
       "17305     KCNE2   -0.112684    0.059434   -0.071567   -0.166129   -0.079283   \n",
       "17306     DGCR2    0.050646   -0.323283    0.120482    0.000000   -0.117080   \n",
       "17307  CASP8AP2   -1.228253   -0.414654   -0.050006    0.076652   -0.360068   \n",
       "17308      SCO2   -0.077570    0.475836   -0.148953    0.000000    0.088060   \n",
       "\n",
       "       ACH-000007  ACH-000008  ACH-000009  ACH-000012  ...  ACH-001075  \\\n",
       "0       -0.012940    0.000000   -0.664572   -0.084851  ...   -0.044965   \n",
       "1       -0.290193    0.000000   -0.086928   -0.018109  ...   -0.120774   \n",
       "2       -0.282557    0.000000    0.164582    0.320907  ...   -0.214594   \n",
       "3        0.136671    0.095038    0.193100   -0.105455  ...   -0.070584   \n",
       "4        0.140191    0.084754   -0.010944    0.133982  ...    0.019461   \n",
       "...           ...         ...         ...         ...  ...         ...   \n",
       "17304    0.076166    0.000000    0.123799   -0.093000  ...    0.147727   \n",
       "17305    0.042259   -0.250895    0.207326   -0.050655  ...   -0.142186   \n",
       "17306    0.177101    0.000000    0.353728   -0.366567  ...   -0.018230   \n",
       "17307   -0.868595   -0.461431   -0.301565   -0.543784  ...   -0.433340   \n",
       "17308    0.246722    0.000000   -0.307953   -0.172032  ...    0.170232   \n",
       "\n",
       "       ACH-001089  ACH-001107  ACH-001113  ACH-001129  ACH-001190  ACH-001239  \\\n",
       "0        0.000000    0.000000    0.188873   -0.008834   -0.042931   -0.326934   \n",
       "1        0.000000    0.000000   -0.193814   -0.079915    0.146653   -0.107795   \n",
       "2        0.000000    0.000000    0.136263    0.135128   -0.075160    0.029538   \n",
       "3       -0.286068   -0.017196   -0.024994   -0.024949    0.221152   -0.166268   \n",
       "4       -0.583573    0.069328    0.210320    0.304147    0.040289   -0.121801   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "17304    0.000000    0.000000    0.038196   -0.007167    0.046325    0.144315   \n",
       "17305    0.139844    0.078532    0.392288   -0.074156   -0.054130   -0.137692   \n",
       "17306    0.000000    0.000000    0.234106    0.115288   -0.212524    0.223209   \n",
       "17307   -0.293064   -0.454495   -0.867408   -0.930363   -0.828188   -0.138393   \n",
       "17308    0.000000    0.000000   -0.082845   -0.073901    0.001383    0.377378   \n",
       "\n",
       "       ACH-001306  ACH-001307  ACH-001318  \n",
       "0        0.000000    0.000000    0.365715  \n",
       "1        0.000000    0.000000    0.083265  \n",
       "2        0.000000    0.000000   -0.183369  \n",
       "3       -0.295108   -0.101788    0.158220  \n",
       "4       -0.005521   -0.021668    0.183108  \n",
       "...           ...         ...         ...  \n",
       "17304    0.000000    0.000000   -0.098176  \n",
       "17305    0.114692    0.060095   -0.038330  \n",
       "17306    0.000000    0.000000    0.109311  \n",
       "17307   -0.983701   -0.626359   -0.440665  \n",
       "17308    0.000000    0.000000   -0.080429  \n",
       "\n",
       "[17309 rows x 593 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# get the overlapped cell lines in methy_samples with cell_line_anno_df by merge on cell line id\n",
    "rna_combined_samples = raw_rna_combined_df.columns[1:].tolist()\n",
    "# first format the methy_samples as dataframe\n",
    "rna_combined_samples_df = pd.DataFrame(rna_combined_samples, columns=['CCLE_Name'])\n",
    "# merge the methy_samples_df with the cell_line_anno_df\n",
    "rna_combined_samples_anno_df = pd.merge(rna_combined_samples_df, cell_line_anno_df, left_on='CCLE_Name', right_on='CCLE_ID', how='inner')\n",
    "display(rna_combined_samples_anno_df)\n",
    "\n",
    "# get the map dictionary from the ccle id to depmap id\n",
    "rna_combined_samples_map_dict = dict(zip(rna_combined_samples_anno_df['CCLE_Name'], rna_combined_samples_anno_df['depMapID']))\n",
    "print(rna_combined_samples_map_dict)\n",
    "maped_rna_combined_df = raw_rna_combined_df.rename(columns=rna_combined_samples_map_dict)\n",
    "columns_to_keep = ['gene_name'] + sorted([col for col in maped_rna_combined_df.columns if col in overlapped_omics_union_annotation_samples])\n",
    "filtered_rna_df = maped_rna_combined_df[columns_to_keep]\n",
    "display(filtered_rna_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 377,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\2511689865.py:1: DtypeWarning: Columns (14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmg_transcript_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Transcript/BioMedGraphica_Conn_Transcript.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>HGNC_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>ARF5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>M6PR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>ESRRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>FKBP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>CYP26B1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412034</th>\n",
       "      <td>BMGC_TS412035</td>\n",
       "      <td>LINC02968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412035</th>\n",
       "      <td>BMGC_TS412036</td>\n",
       "      <td>LINC03112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412036</th>\n",
       "      <td>BMGC_TS412037</td>\n",
       "      <td>LINC03112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412037</th>\n",
       "      <td>BMGC_TS412038</td>\n",
       "      <td>LINC03112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412038</th>\n",
       "      <td>BMGC_TS412039</td>\n",
       "      <td>CD99P1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412039 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID HGNC_Symbol\n",
       "0               BMGC_TS000001        ARF5\n",
       "1               BMGC_TS000002        M6PR\n",
       "2               BMGC_TS000003       ESRRA\n",
       "3               BMGC_TS000004       FKBP4\n",
       "4               BMGC_TS000005     CYP26B1\n",
       "...                       ...         ...\n",
       "412034          BMGC_TS412035   LINC02968\n",
       "412035          BMGC_TS412036   LINC03112\n",
       "412036          BMGC_TS412037   LINC03112\n",
       "412037          BMGC_TS412038   LINC03112\n",
       "412038          BMGC_TS412039      CD99P1\n",
       "\n",
       "[412039 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001089</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001113</th>\n",
       "      <th>ACH-001129</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001239</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "      <th>ACH-001318</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>0.123267</td>\n",
       "      <td>-0.184446</td>\n",
       "      <td>-0.072956</td>\n",
       "      <td>0.000381</td>\n",
       "      <td>-0.037266</td>\n",
       "      <td>-0.029936</td>\n",
       "      <td>0.102107</td>\n",
       "      <td>-0.108996</td>\n",
       "      <td>-0.099402</td>\n",
       "      <td>...</td>\n",
       "      <td>0.127145</td>\n",
       "      <td>0.037874</td>\n",
       "      <td>-0.107370</td>\n",
       "      <td>-0.105618</td>\n",
       "      <td>-0.115614</td>\n",
       "      <td>0.031604</td>\n",
       "      <td>-0.080667</td>\n",
       "      <td>-0.181498</td>\n",
       "      <td>0.048788</td>\n",
       "      <td>0.201115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>-0.025559</td>\n",
       "      <td>0.113514</td>\n",
       "      <td>0.141620</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.137914</td>\n",
       "      <td>-0.179942</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.028500</td>\n",
       "      <td>-0.324958</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.202727</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.198776</td>\n",
       "      <td>0.371099</td>\n",
       "      <td>-0.111130</td>\n",
       "      <td>0.308401</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.245889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>-0.047525</td>\n",
       "      <td>-0.535418</td>\n",
       "      <td>0.024752</td>\n",
       "      <td>-0.012339</td>\n",
       "      <td>0.115913</td>\n",
       "      <td>0.135700</td>\n",
       "      <td>0.104687</td>\n",
       "      <td>-0.274726</td>\n",
       "      <td>0.046911</td>\n",
       "      <td>...</td>\n",
       "      <td>0.043630</td>\n",
       "      <td>-0.082785</td>\n",
       "      <td>-0.084723</td>\n",
       "      <td>0.009095</td>\n",
       "      <td>-0.198575</td>\n",
       "      <td>0.065699</td>\n",
       "      <td>-0.078132</td>\n",
       "      <td>-0.152112</td>\n",
       "      <td>0.103321</td>\n",
       "      <td>0.119976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>-0.526937</td>\n",
       "      <td>-0.723573</td>\n",
       "      <td>-0.492790</td>\n",
       "      <td>-0.513913</td>\n",
       "      <td>-0.591701</td>\n",
       "      <td>-0.246384</td>\n",
       "      <td>-0.531943</td>\n",
       "      <td>-0.149865</td>\n",
       "      <td>-0.217154</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.463727</td>\n",
       "      <td>-0.367015</td>\n",
       "      <td>-0.232642</td>\n",
       "      <td>-0.409513</td>\n",
       "      <td>-0.587943</td>\n",
       "      <td>-0.119636</td>\n",
       "      <td>-0.400796</td>\n",
       "      <td>-0.118495</td>\n",
       "      <td>-0.052553</td>\n",
       "      <td>-0.383631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>0.116400</td>\n",
       "      <td>0.240852</td>\n",
       "      <td>-0.270499</td>\n",
       "      <td>-0.183875</td>\n",
       "      <td>0.023891</td>\n",
       "      <td>-0.204737</td>\n",
       "      <td>-0.102901</td>\n",
       "      <td>-0.053493</td>\n",
       "      <td>-0.095810</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.208952</td>\n",
       "      <td>0.039113</td>\n",
       "      <td>-0.080288</td>\n",
       "      <td>-0.022177</td>\n",
       "      <td>-0.232425</td>\n",
       "      <td>-0.033772</td>\n",
       "      <td>-0.245861</td>\n",
       "      <td>-0.045608</td>\n",
       "      <td>-0.141908</td>\n",
       "      <td>-0.201729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170386</th>\n",
       "      <td>BMGC_TS411885</td>\n",
       "      <td>-0.236575</td>\n",
       "      <td>-0.155037</td>\n",
       "      <td>0.115897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.168277</td>\n",
       "      <td>0.041007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.335707</td>\n",
       "      <td>-0.561136</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054384</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002146</td>\n",
       "      <td>-0.072585</td>\n",
       "      <td>-0.254140</td>\n",
       "      <td>-0.124046</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.364009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170387</th>\n",
       "      <td>BMGC_TS411886</td>\n",
       "      <td>-0.236575</td>\n",
       "      <td>-0.155037</td>\n",
       "      <td>0.115897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.168277</td>\n",
       "      <td>0.041007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.335707</td>\n",
       "      <td>-0.561136</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054384</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002146</td>\n",
       "      <td>-0.072585</td>\n",
       "      <td>-0.254140</td>\n",
       "      <td>-0.124046</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.364009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170388</th>\n",
       "      <td>BMGC_TS411887</td>\n",
       "      <td>-0.236575</td>\n",
       "      <td>-0.155037</td>\n",
       "      <td>0.115897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.168277</td>\n",
       "      <td>0.041007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.335707</td>\n",
       "      <td>-0.561136</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054384</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002146</td>\n",
       "      <td>-0.072585</td>\n",
       "      <td>-0.254140</td>\n",
       "      <td>-0.124046</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.364009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170389</th>\n",
       "      <td>BMGC_TS412018</td>\n",
       "      <td>0.028071</td>\n",
       "      <td>-0.129326</td>\n",
       "      <td>0.091669</td>\n",
       "      <td>-0.004241</td>\n",
       "      <td>0.055351</td>\n",
       "      <td>-0.016189</td>\n",
       "      <td>0.269712</td>\n",
       "      <td>0.139049</td>\n",
       "      <td>-0.242594</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.127993</td>\n",
       "      <td>-0.081723</td>\n",
       "      <td>0.001105</td>\n",
       "      <td>0.048276</td>\n",
       "      <td>-0.008880</td>\n",
       "      <td>-0.076799</td>\n",
       "      <td>-0.011490</td>\n",
       "      <td>0.257041</td>\n",
       "      <td>0.024479</td>\n",
       "      <td>0.076093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170390</th>\n",
       "      <td>BMGC_TS412019</td>\n",
       "      <td>0.090240</td>\n",
       "      <td>-0.107814</td>\n",
       "      <td>0.132475</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.249785</td>\n",
       "      <td>-0.030092</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.262196</td>\n",
       "      <td>-0.074556</td>\n",
       "      <td>...</td>\n",
       "      <td>0.014026</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.059573</td>\n",
       "      <td>-0.234128</td>\n",
       "      <td>-0.274055</td>\n",
       "      <td>0.183369</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.055520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170391 rows × 593 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000001  ACH-000002  ACH-000004  ACH-000005  \\\n",
       "0               BMGC_TS000001    0.123267   -0.184446   -0.072956    0.000381   \n",
       "1               BMGC_TS000002   -0.025559    0.113514    0.141620    0.000000   \n",
       "2               BMGC_TS000003   -0.047525   -0.535418    0.024752   -0.012339   \n",
       "3               BMGC_TS000004   -0.526937   -0.723573   -0.492790   -0.513913   \n",
       "4               BMGC_TS000005    0.116400    0.240852   -0.270499   -0.183875   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "170386          BMGC_TS411885   -0.236575   -0.155037    0.115897    0.000000   \n",
       "170387          BMGC_TS411886   -0.236575   -0.155037    0.115897    0.000000   \n",
       "170388          BMGC_TS411887   -0.236575   -0.155037    0.115897    0.000000   \n",
       "170389          BMGC_TS412018    0.028071   -0.129326    0.091669   -0.004241   \n",
       "170390          BMGC_TS412019    0.090240   -0.107814    0.132475    0.000000   \n",
       "\n",
       "        ACH-000006  ACH-000007  ACH-000008  ACH-000009  ACH-000012  ...  \\\n",
       "0        -0.037266   -0.029936    0.102107   -0.108996   -0.099402  ...   \n",
       "1         0.137914   -0.179942    0.000000   -0.028500   -0.324958  ...   \n",
       "2         0.115913    0.135700    0.104687   -0.274726    0.046911  ...   \n",
       "3        -0.591701   -0.246384   -0.531943   -0.149865   -0.217154  ...   \n",
       "4         0.023891   -0.204737   -0.102901   -0.053493   -0.095810  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "170386   -0.168277    0.041007    0.000000   -0.335707   -0.561136  ...   \n",
       "170387   -0.168277    0.041007    0.000000   -0.335707   -0.561136  ...   \n",
       "170388   -0.168277    0.041007    0.000000   -0.335707   -0.561136  ...   \n",
       "170389    0.055351   -0.016189    0.269712    0.139049   -0.242594  ...   \n",
       "170390   -0.249785   -0.030092    0.000000   -0.262196   -0.074556  ...   \n",
       "\n",
       "        ACH-001075  ACH-001089  ACH-001107  ACH-001113  ACH-001129  \\\n",
       "0         0.127145    0.037874   -0.107370   -0.105618   -0.115614   \n",
       "1        -0.202727    0.000000    0.000000   -0.198776    0.371099   \n",
       "2         0.043630   -0.082785   -0.084723    0.009095   -0.198575   \n",
       "3        -0.463727   -0.367015   -0.232642   -0.409513   -0.587943   \n",
       "4        -0.208952    0.039113   -0.080288   -0.022177   -0.232425   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "170386    0.054384    0.000000    0.000000    0.002146   -0.072585   \n",
       "170387    0.054384    0.000000    0.000000    0.002146   -0.072585   \n",
       "170388    0.054384    0.000000    0.000000    0.002146   -0.072585   \n",
       "170389   -0.127993   -0.081723    0.001105    0.048276   -0.008880   \n",
       "170390    0.014026    0.000000    0.000000    0.059573   -0.234128   \n",
       "\n",
       "        ACH-001190  ACH-001239  ACH-001306  ACH-001307  ACH-001318  \n",
       "0         0.031604   -0.080667   -0.181498    0.048788    0.201115  \n",
       "1        -0.111130    0.308401    0.000000    0.000000    0.245889  \n",
       "2         0.065699   -0.078132   -0.152112    0.103321    0.119976  \n",
       "3        -0.119636   -0.400796   -0.118495   -0.052553   -0.383631  \n",
       "4        -0.033772   -0.245861   -0.045608   -0.141908   -0.201729  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "170386   -0.254140   -0.124046    0.000000    0.000000   -0.364009  \n",
       "170387   -0.254140   -0.124046    0.000000    0.000000   -0.364009  \n",
       "170388   -0.254140   -0.124046    0.000000    0.000000   -0.364009  \n",
       "170389   -0.076799   -0.011490    0.257041    0.024479    0.076093  \n",
       "170390   -0.274055    0.183369    0.000000    0.000000    0.055520  \n",
       "\n",
       "[170391 rows x 593 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bmg_transcript_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Transcript/BioMedGraphica_Conn_Transcript.csv')\n",
    "# keep BioMedGraphica_ID and HGNC_Symbol\n",
    "bmg_transcript_df = bmg_transcript_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']]\n",
    "display(bmg_transcript_df)\n",
    "# merge the biomedgraphica_id with the raw_transcript_df\n",
    "merge_rna_df = pd.merge(bmg_transcript_df, filtered_rna_df, left_on='HGNC_Symbol', right_on='gene_name', how='inner')\n",
    "merge_rna_df.drop(columns=['HGNC_Symbol', 'gene_name'], inplace=True)\n",
    "display(merge_rna_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 378,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_TS_ID</th>\n",
       "      <th>BMGC_PT_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001089</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001113</th>\n",
       "      <th>ACH-001129</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001239</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "      <th>ACH-001318</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>BMGC_PT017571</td>\n",
       "      <td>0.123267</td>\n",
       "      <td>-0.184446</td>\n",
       "      <td>-0.072956</td>\n",
       "      <td>0.000381</td>\n",
       "      <td>-0.037266</td>\n",
       "      <td>-0.029936</td>\n",
       "      <td>0.102107</td>\n",
       "      <td>-0.108996</td>\n",
       "      <td>...</td>\n",
       "      <td>0.127145</td>\n",
       "      <td>0.037874</td>\n",
       "      <td>-0.107370</td>\n",
       "      <td>-0.105618</td>\n",
       "      <td>-0.115614</td>\n",
       "      <td>0.031604</td>\n",
       "      <td>-0.080667</td>\n",
       "      <td>-0.181498</td>\n",
       "      <td>0.048788</td>\n",
       "      <td>0.201115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>BMGC_PT014978</td>\n",
       "      <td>-0.025559</td>\n",
       "      <td>0.113514</td>\n",
       "      <td>0.141620</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.137914</td>\n",
       "      <td>-0.179942</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.028500</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.202727</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.198776</td>\n",
       "      <td>0.371099</td>\n",
       "      <td>-0.111130</td>\n",
       "      <td>0.308401</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.245889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>BMGC_PT014403</td>\n",
       "      <td>-0.047525</td>\n",
       "      <td>-0.535418</td>\n",
       "      <td>0.024752</td>\n",
       "      <td>-0.012339</td>\n",
       "      <td>0.115913</td>\n",
       "      <td>0.135700</td>\n",
       "      <td>0.104687</td>\n",
       "      <td>-0.274726</td>\n",
       "      <td>...</td>\n",
       "      <td>0.043630</td>\n",
       "      <td>-0.082785</td>\n",
       "      <td>-0.084723</td>\n",
       "      <td>0.009095</td>\n",
       "      <td>-0.198575</td>\n",
       "      <td>0.065699</td>\n",
       "      <td>-0.078132</td>\n",
       "      <td>-0.152112</td>\n",
       "      <td>0.103321</td>\n",
       "      <td>0.119976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>BMGC_PT017807</td>\n",
       "      <td>-0.526937</td>\n",
       "      <td>-0.723573</td>\n",
       "      <td>-0.492790</td>\n",
       "      <td>-0.513913</td>\n",
       "      <td>-0.591701</td>\n",
       "      <td>-0.246384</td>\n",
       "      <td>-0.531943</td>\n",
       "      <td>-0.149865</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.463727</td>\n",
       "      <td>-0.367015</td>\n",
       "      <td>-0.232642</td>\n",
       "      <td>-0.409513</td>\n",
       "      <td>-0.587943</td>\n",
       "      <td>-0.119636</td>\n",
       "      <td>-0.400796</td>\n",
       "      <td>-0.118495</td>\n",
       "      <td>-0.052553</td>\n",
       "      <td>-0.383631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>BMGC_PT043262</td>\n",
       "      <td>0.116400</td>\n",
       "      <td>0.240852</td>\n",
       "      <td>-0.270499</td>\n",
       "      <td>-0.183875</td>\n",
       "      <td>0.023891</td>\n",
       "      <td>-0.204737</td>\n",
       "      <td>-0.102901</td>\n",
       "      <td>-0.053493</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.208952</td>\n",
       "      <td>0.039113</td>\n",
       "      <td>-0.080288</td>\n",
       "      <td>-0.022177</td>\n",
       "      <td>-0.232425</td>\n",
       "      <td>-0.033772</td>\n",
       "      <td>-0.245861</td>\n",
       "      <td>-0.045608</td>\n",
       "      <td>-0.141908</td>\n",
       "      <td>-0.201729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132566</th>\n",
       "      <td>BMGC_TS411885</td>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>-0.236575</td>\n",
       "      <td>-0.155037</td>\n",
       "      <td>0.115897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.168277</td>\n",
       "      <td>0.041007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.335707</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054384</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002146</td>\n",
       "      <td>-0.072585</td>\n",
       "      <td>-0.254140</td>\n",
       "      <td>-0.124046</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.364009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132567</th>\n",
       "      <td>BMGC_TS411886</td>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>-0.236575</td>\n",
       "      <td>-0.155037</td>\n",
       "      <td>0.115897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.168277</td>\n",
       "      <td>0.041007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.335707</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054384</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002146</td>\n",
       "      <td>-0.072585</td>\n",
       "      <td>-0.254140</td>\n",
       "      <td>-0.124046</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.364009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132568</th>\n",
       "      <td>BMGC_TS411887</td>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>-0.236575</td>\n",
       "      <td>-0.155037</td>\n",
       "      <td>0.115897</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.168277</td>\n",
       "      <td>0.041007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.335707</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054384</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002146</td>\n",
       "      <td>-0.072585</td>\n",
       "      <td>-0.254140</td>\n",
       "      <td>-0.124046</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.364009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132569</th>\n",
       "      <td>BMGC_TS412018</td>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.028071</td>\n",
       "      <td>-0.129326</td>\n",
       "      <td>0.091669</td>\n",
       "      <td>-0.004241</td>\n",
       "      <td>0.055351</td>\n",
       "      <td>-0.016189</td>\n",
       "      <td>0.269712</td>\n",
       "      <td>0.139049</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.127993</td>\n",
       "      <td>-0.081723</td>\n",
       "      <td>0.001105</td>\n",
       "      <td>0.048276</td>\n",
       "      <td>-0.008880</td>\n",
       "      <td>-0.076799</td>\n",
       "      <td>-0.011490</td>\n",
       "      <td>0.257041</td>\n",
       "      <td>0.024479</td>\n",
       "      <td>0.076093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132570</th>\n",
       "      <td>BMGC_TS412019</td>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.090240</td>\n",
       "      <td>-0.107814</td>\n",
       "      <td>0.132475</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.249785</td>\n",
       "      <td>-0.030092</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.262196</td>\n",
       "      <td>...</td>\n",
       "      <td>0.014026</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.059573</td>\n",
       "      <td>-0.234128</td>\n",
       "      <td>-0.274055</td>\n",
       "      <td>0.183369</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.055520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>132571 rows × 594 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           BMGC_TS_ID     BMGC_PT_ID  ACH-000001  ACH-000002  ACH-000004  \\\n",
       "0       BMGC_TS000001  BMGC_PT017571    0.123267   -0.184446   -0.072956   \n",
       "1       BMGC_TS000002  BMGC_PT014978   -0.025559    0.113514    0.141620   \n",
       "2       BMGC_TS000003  BMGC_PT014403   -0.047525   -0.535418    0.024752   \n",
       "3       BMGC_TS000004  BMGC_PT017807   -0.526937   -0.723573   -0.492790   \n",
       "4       BMGC_TS000005  BMGC_PT043262    0.116400    0.240852   -0.270499   \n",
       "...               ...            ...         ...         ...         ...   \n",
       "132566  BMGC_TS411885  BMGC_PT121415   -0.236575   -0.155037    0.115897   \n",
       "132567  BMGC_TS411886  BMGC_PT121416   -0.236575   -0.155037    0.115897   \n",
       "132568  BMGC_TS411887  BMGC_PT121417   -0.236575   -0.155037    0.115897   \n",
       "132569  BMGC_TS412018  BMGC_PT121418    0.028071   -0.129326    0.091669   \n",
       "132570  BMGC_TS412019  BMGC_PT121419    0.090240   -0.107814    0.132475   \n",
       "\n",
       "        ACH-000005  ACH-000006  ACH-000007  ACH-000008  ACH-000009  ...  \\\n",
       "0         0.000381   -0.037266   -0.029936    0.102107   -0.108996  ...   \n",
       "1         0.000000    0.137914   -0.179942    0.000000   -0.028500  ...   \n",
       "2        -0.012339    0.115913    0.135700    0.104687   -0.274726  ...   \n",
       "3        -0.513913   -0.591701   -0.246384   -0.531943   -0.149865  ...   \n",
       "4        -0.183875    0.023891   -0.204737   -0.102901   -0.053493  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "132566    0.000000   -0.168277    0.041007    0.000000   -0.335707  ...   \n",
       "132567    0.000000   -0.168277    0.041007    0.000000   -0.335707  ...   \n",
       "132568    0.000000   -0.168277    0.041007    0.000000   -0.335707  ...   \n",
       "132569   -0.004241    0.055351   -0.016189    0.269712    0.139049  ...   \n",
       "132570    0.000000   -0.249785   -0.030092    0.000000   -0.262196  ...   \n",
       "\n",
       "        ACH-001075  ACH-001089  ACH-001107  ACH-001113  ACH-001129  \\\n",
       "0         0.127145    0.037874   -0.107370   -0.105618   -0.115614   \n",
       "1        -0.202727    0.000000    0.000000   -0.198776    0.371099   \n",
       "2         0.043630   -0.082785   -0.084723    0.009095   -0.198575   \n",
       "3        -0.463727   -0.367015   -0.232642   -0.409513   -0.587943   \n",
       "4        -0.208952    0.039113   -0.080288   -0.022177   -0.232425   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "132566    0.054384    0.000000    0.000000    0.002146   -0.072585   \n",
       "132567    0.054384    0.000000    0.000000    0.002146   -0.072585   \n",
       "132568    0.054384    0.000000    0.000000    0.002146   -0.072585   \n",
       "132569   -0.127993   -0.081723    0.001105    0.048276   -0.008880   \n",
       "132570    0.014026    0.000000    0.000000    0.059573   -0.234128   \n",
       "\n",
       "        ACH-001190  ACH-001239  ACH-001306  ACH-001307  ACH-001318  \n",
       "0         0.031604   -0.080667   -0.181498    0.048788    0.201115  \n",
       "1        -0.111130    0.308401    0.000000    0.000000    0.245889  \n",
       "2         0.065699   -0.078132   -0.152112    0.103321    0.119976  \n",
       "3        -0.119636   -0.400796   -0.118495   -0.052553   -0.383631  \n",
       "4        -0.033772   -0.245861   -0.045608   -0.141908   -0.201729  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "132566   -0.254140   -0.124046    0.000000    0.000000   -0.364009  \n",
       "132567   -0.254140   -0.124046    0.000000    0.000000   -0.364009  \n",
       "132568   -0.254140   -0.124046    0.000000    0.000000   -0.364009  \n",
       "132569   -0.076799   -0.011490    0.257041    0.024479    0.076093  \n",
       "132570   -0.274055    0.183369    0.000000    0.000000    0.055520  \n",
       "\n",
       "[132571 rows x 594 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# map the transcript entity with transcript_protein_entity_df\n",
    "protein_rna_df = pd.merge(transcript_protein_entity_df, merge_rna_df, left_on='BMGC_TS_ID', right_on='BioMedGraphica_Conn_ID', how='left').drop(columns=['BioMedGraphica_Conn_ID'])\n",
    "# and drop nan in any rows from protein_rna_df\n",
    "processed_rna_df = protein_rna_df.dropna().reset_index(drop=True)\n",
    "display(processed_rna_df)\n",
    "processed_rna_df.to_csv('./data/pretrain_plain_data/processed_rna.csv', index=False)\n",
    "processed_rna_df.to_csv('./data/pretrain_status_data/processed_rna.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 379,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The overlapped samples between protein_crispr_samples and protein_rna_samples are:\n",
      "489\n",
      "['ACH-000001', 'ACH-000004', 'ACH-000005', 'ACH-000007', 'ACH-000009', 'ACH-000012', 'ACH-000014', 'ACH-000015', 'ACH-000017', 'ACH-000018', 'ACH-000019', 'ACH-000021', 'ACH-000023', 'ACH-000024', 'ACH-000028', 'ACH-000030', 'ACH-000035', 'ACH-000039', 'ACH-000040', 'ACH-000041', 'ACH-000042', 'ACH-000045', 'ACH-000048', 'ACH-000052', 'ACH-000054', 'ACH-000055', 'ACH-000060', 'ACH-000067', 'ACH-000070', 'ACH-000075', 'ACH-000086', 'ACH-000087', 'ACH-000090', 'ACH-000092', 'ACH-000093', 'ACH-000094', 'ACH-000095', 'ACH-000096', 'ACH-000097', 'ACH-000098', 'ACH-000100', 'ACH-000103', 'ACH-000108', 'ACH-000111', 'ACH-000113', 'ACH-000114', 'ACH-000115', 'ACH-000118', 'ACH-000123', 'ACH-000127', 'ACH-000132', 'ACH-000133', 'ACH-000138', 'ACH-000139', 'ACH-000142', 'ACH-000146', 'ACH-000147', 'ACH-000148', 'ACH-000149', 'ACH-000152', 'ACH-000153', 'ACH-000155', 'ACH-000159', 'ACH-000161', 'ACH-000163', 'ACH-000164', 'ACH-000168', 'ACH-000169', 'ACH-000176', 'ACH-000178', 'ACH-000183', 'ACH-000188', 'ACH-000191', 'ACH-000192', 'ACH-000202', 'ACH-000207', 'ACH-000209', 'ACH-000210', 'ACH-000211', 'ACH-000212', 'ACH-000213', 'ACH-000217', 'ACH-000219', 'ACH-000222', 'ACH-000223', 'ACH-000227', 'ACH-000231', 'ACH-000232', 'ACH-000234', 'ACH-000235', 'ACH-000237', 'ACH-000238', 'ACH-000239', 'ACH-000243', 'ACH-000244', 'ACH-000246', 'ACH-000247', 'ACH-000248', 'ACH-000249', 'ACH-000250', 'ACH-000253', 'ACH-000255', 'ACH-000256', 'ACH-000257', 'ACH-000258', 'ACH-000259', 'ACH-000260', 'ACH-000261', 'ACH-000263', 'ACH-000264', 'ACH-000265', 'ACH-000269', 'ACH-000271', 'ACH-000272', 'ACH-000274', 'ACH-000276', 'ACH-000277', 'ACH-000278', 'ACH-000280', 'ACH-000281', 'ACH-000288', 'ACH-000290', 'ACH-000291', 'ACH-000292', 'ACH-000294', 'ACH-000296', 'ACH-000303', 'ACH-000304', 'ACH-000305', 'ACH-000307', 'ACH-000308', 'ACH-000311', 'ACH-000312', 'ACH-000315', 'ACH-000316', 'ACH-000317', 'ACH-000318', 'ACH-000319', 'ACH-000320', 'ACH-000322', 'ACH-000324', 'ACH-000330', 'ACH-000334', 'ACH-000336', 'ACH-000341', 'ACH-000343', 'ACH-000348', 'ACH-000350', 'ACH-000352', 'ACH-000356', 'ACH-000361', 'ACH-000362', 'ACH-000363', 'ACH-000364', 'ACH-000365', 'ACH-000366', 'ACH-000368', 'ACH-000371', 'ACH-000374', 'ACH-000375', 'ACH-000376', 'ACH-000378', 'ACH-000390', 'ACH-000391', 'ACH-000393', 'ACH-000395', 'ACH-000397', 'ACH-000401', 'ACH-000403', 'ACH-000404', 'ACH-000406', 'ACH-000407', 'ACH-000410', 'ACH-000411', 'ACH-000414', 'ACH-000415', 'ACH-000416', 'ACH-000417', 'ACH-000420', 'ACH-000421', 'ACH-000424', 'ACH-000425', 'ACH-000427', 'ACH-000430', 'ACH-000431', 'ACH-000433', 'ACH-000434', 'ACH-000435', 'ACH-000437', 'ACH-000438', 'ACH-000441', 'ACH-000444', 'ACH-000445', 'ACH-000446', 'ACH-000448', 'ACH-000450', 'ACH-000452', 'ACH-000454', 'ACH-000456', 'ACH-000458', 'ACH-000459', 'ACH-000460', 'ACH-000463', 'ACH-000464', 'ACH-000466', 'ACH-000470', 'ACH-000471', 'ACH-000472', 'ACH-000473', 'ACH-000475', 'ACH-000479', 'ACH-000480', 'ACH-000481', 'ACH-000484', 'ACH-000485', 'ACH-000486', 'ACH-000487', 'ACH-000488', 'ACH-000491', 'ACH-000493', 'ACH-000495', 'ACH-000496', 'ACH-000502', 'ACH-000505', 'ACH-000507', 'ACH-000510', 'ACH-000511', 'ACH-000515', 'ACH-000517', 'ACH-000518', 'ACH-000521', 'ACH-000522', 'ACH-000524', 'ACH-000525', 'ACH-000527', 'ACH-000528', 'ACH-000532', 'ACH-000535', 'ACH-000537', 'ACH-000541', 'ACH-000542', 'ACH-000544', 'ACH-000547', 'ACH-000548', 'ACH-000549', 'ACH-000550', 'ACH-000551', 'ACH-000552', 'ACH-000554', 'ACH-000557', 'ACH-000558', 'ACH-000559', 'ACH-000563', 'ACH-000570', 'ACH-000571', 'ACH-000573', 'ACH-000576', 'ACH-000579', 'ACH-000580', 'ACH-000585', 'ACH-000587', 'ACH-000589', 'ACH-000593', 'ACH-000595', 'ACH-000596', 'ACH-000599', 'ACH-000601', 'ACH-000605', 'ACH-000606', 'ACH-000607', 'ACH-000608', 'ACH-000609', 'ACH-000614', 'ACH-000620', 'ACH-000621', 'ACH-000622', 'ACH-000624', 'ACH-000627', 'ACH-000630', 'ACH-000631', 'ACH-000632', 'ACH-000633', 'ACH-000637', 'ACH-000638', 'ACH-000639', 'ACH-000645', 'ACH-000647', 'ACH-000648', 'ACH-000649', 'ACH-000651', 'ACH-000652', 'ACH-000653', 'ACH-000655', 'ACH-000657', 'ACH-000661', 'ACH-000662', 'ACH-000663', 'ACH-000665', 'ACH-000666', 'ACH-000667', 'ACH-000668', 'ACH-000671', 'ACH-000672', 'ACH-000677', 'ACH-000681', 'ACH-000684', 'ACH-000685', 'ACH-000688', 'ACH-000693', 'ACH-000694', 'ACH-000696', 'ACH-000697', 'ACH-000698', 'ACH-000699', 'ACH-000701', 'ACH-000704', 'ACH-000706', 'ACH-000709', 'ACH-000711', 'ACH-000713', 'ACH-000717', 'ACH-000718', 'ACH-000719', 'ACH-000720', 'ACH-000721', 'ACH-000722', 'ACH-000724', 'ACH-000725', 'ACH-000732', 'ACH-000734', 'ACH-000735', 'ACH-000736', 'ACH-000738', 'ACH-000739', 'ACH-000740', 'ACH-000741', 'ACH-000746', 'ACH-000747', 'ACH-000748', 'ACH-000749', 'ACH-000753', 'ACH-000756', 'ACH-000762', 'ACH-000763', 'ACH-000765', 'ACH-000766', 'ACH-000767', 'ACH-000768', 'ACH-000769', 'ACH-000771', 'ACH-000774', 'ACH-000775', 'ACH-000777', 'ACH-000778', 'ACH-000781', 'ACH-000782', 'ACH-000783', 'ACH-000784', 'ACH-000785', 'ACH-000787', 'ACH-000788', 'ACH-000790', 'ACH-000792', 'ACH-000793', 'ACH-000794', 'ACH-000799', 'ACH-000800', 'ACH-000804', 'ACH-000805', 'ACH-000807', 'ACH-000809', 'ACH-000810', 'ACH-000811', 'ACH-000817', 'ACH-000819', 'ACH-000820', 'ACH-000823', 'ACH-000824', 'ACH-000826', 'ACH-000827', 'ACH-000831', 'ACH-000832', 'ACH-000833', 'ACH-000837', 'ACH-000838', 'ACH-000840', 'ACH-000841', 'ACH-000845', 'ACH-000846', 'ACH-000847', 'ACH-000848', 'ACH-000849', 'ACH-000851', 'ACH-000853', 'ACH-000855', 'ACH-000856', 'ACH-000858', 'ACH-000859', 'ACH-000860', 'ACH-000861', 'ACH-000863', 'ACH-000864', 'ACH-000865', 'ACH-000866', 'ACH-000867', 'ACH-000869', 'ACH-000874', 'ACH-000876', 'ACH-000878', 'ACH-000879', 'ACH-000880', 'ACH-000881', 'ACH-000882', 'ACH-000884', 'ACH-000885', 'ACH-000886', 'ACH-000888', 'ACH-000889', 'ACH-000890', 'ACH-000898', 'ACH-000899', 'ACH-000900', 'ACH-000901', 'ACH-000906', 'ACH-000907', 'ACH-000910', 'ACH-000911', 'ACH-000912', 'ACH-000913', 'ACH-000915', 'ACH-000916', 'ACH-000917', 'ACH-000919', 'ACH-000921', 'ACH-000922', 'ACH-000924', 'ACH-000926', 'ACH-000929', 'ACH-000932', 'ACH-000934', 'ACH-000935', 'ACH-000936', 'ACH-000938', 'ACH-000943', 'ACH-000945', 'ACH-000946', 'ACH-000948', 'ACH-000950', 'ACH-000953', 'ACH-000956', 'ACH-000957', 'ACH-000958', 'ACH-000959', 'ACH-000960', 'ACH-000961', 'ACH-000962', 'ACH-000963', 'ACH-000965', 'ACH-000966', 'ACH-000969', 'ACH-000971', 'ACH-000974', 'ACH-000977', 'ACH-000978', 'ACH-000979', 'ACH-000980', 'ACH-000984', 'ACH-000985', 'ACH-000986', 'ACH-000988', 'ACH-000991', 'ACH-000994', 'ACH-000996', 'ACH-000997', 'ACH-001001', 'ACH-001036', 'ACH-001039', 'ACH-001041', 'ACH-001048', 'ACH-001061', 'ACH-001065', 'ACH-001075', 'ACH-001107', 'ACH-001129', 'ACH-001190', 'ACH-001239', 'ACH-001306', 'ACH-001307', 'ACH-001318']\n",
      "The union samples between protein_crispr_samples and protein_rna_samples are:\n",
      "770\n",
      "['ACH-000001', 'ACH-000002', 'ACH-000004', 'ACH-000005', 'ACH-000006', 'ACH-000007', 'ACH-000008', 'ACH-000009', 'ACH-000011', 'ACH-000012', 'ACH-000013', 'ACH-000014', 'ACH-000015', 'ACH-000016', 'ACH-000017', 'ACH-000018', 'ACH-000019', 'ACH-000021', 'ACH-000022', 'ACH-000023', 'ACH-000024', 'ACH-000026', 'ACH-000028', 'ACH-000030', 'ACH-000035', 'ACH-000037', 'ACH-000039', 'ACH-000040', 'ACH-000041', 'ACH-000042', 'ACH-000044', 'ACH-000045', 'ACH-000046', 'ACH-000047', 'ACH-000048', 'ACH-000052', 'ACH-000053', 'ACH-000054', 'ACH-000055', 'ACH-000060', 'ACH-000062', 'ACH-000065', 'ACH-000066', 'ACH-000067', 'ACH-000070', 'ACH-000074', 'ACH-000075', 'ACH-000076', 'ACH-000077', 'ACH-000078', 'ACH-000082', 'ACH-000085', 'ACH-000086', 'ACH-000087', 'ACH-000090', 'ACH-000092', 'ACH-000093', 'ACH-000094', 'ACH-000095', 'ACH-000096', 'ACH-000097', 'ACH-000098', 'ACH-000099', 'ACH-000100', 'ACH-000101', 'ACH-000103', 'ACH-000106', 'ACH-000108', 'ACH-000109', 'ACH-000110', 'ACH-000111', 'ACH-000113', 'ACH-000114', 'ACH-000115', 'ACH-000116', 'ACH-000117', 'ACH-000118', 'ACH-000120', 'ACH-000121', 'ACH-000123', 'ACH-000124', 'ACH-000127', 'ACH-000132', 'ACH-000133', 'ACH-000137', 'ACH-000138', 'ACH-000139', 'ACH-000140', 'ACH-000141', 'ACH-000142', 'ACH-000144', 'ACH-000145', 'ACH-000146', 'ACH-000147', 'ACH-000148', 'ACH-000149', 'ACH-000151', 'ACH-000152', 'ACH-000153', 'ACH-000155', 'ACH-000157', 'ACH-000159', 'ACH-000161', 'ACH-000163', 'ACH-000164', 'ACH-000167', 'ACH-000168', 'ACH-000169', 'ACH-000174', 'ACH-000176', 'ACH-000177', 'ACH-000178', 'ACH-000181', 'ACH-000182', 'ACH-000183', 'ACH-000187', 'ACH-000188', 'ACH-000189', 'ACH-000190', 'ACH-000191', 'ACH-000192', 'ACH-000195', 'ACH-000196', 'ACH-000198', 'ACH-000200', 'ACH-000201', 'ACH-000202', 'ACH-000203', 'ACH-000204', 'ACH-000207', 'ACH-000209', 'ACH-000210', 'ACH-000211', 'ACH-000212', 'ACH-000213', 'ACH-000217', 'ACH-000219', 'ACH-000221', 'ACH-000222', 'ACH-000223', 'ACH-000226', 'ACH-000227', 'ACH-000228', 'ACH-000231', 'ACH-000232', 'ACH-000233', 'ACH-000234', 'ACH-000235', 'ACH-000236', 'ACH-000237', 'ACH-000238', 'ACH-000239', 'ACH-000242', 'ACH-000243', 'ACH-000244', 'ACH-000246', 'ACH-000247', 'ACH-000248', 'ACH-000249', 'ACH-000250', 'ACH-000252', 'ACH-000253', 'ACH-000255', 'ACH-000256', 'ACH-000257', 'ACH-000258', 'ACH-000259', 'ACH-000260', 'ACH-000261', 'ACH-000263', 'ACH-000264', 'ACH-000265', 'ACH-000266', 'ACH-000268', 'ACH-000269', 'ACH-000270', 'ACH-000271', 'ACH-000272', 'ACH-000273', 'ACH-000274', 'ACH-000276', 'ACH-000277', 'ACH-000278', 'ACH-000280', 'ACH-000281', 'ACH-000286', 'ACH-000288', 'ACH-000290', 'ACH-000291', 'ACH-000292', 'ACH-000293', 'ACH-000294', 'ACH-000295', 'ACH-000296', 'ACH-000300', 'ACH-000301', 'ACH-000302', 'ACH-000303', 'ACH-000304', 'ACH-000305', 'ACH-000307', 'ACH-000308', 'ACH-000310', 'ACH-000311', 'ACH-000312', 'ACH-000313', 'ACH-000315', 'ACH-000316', 'ACH-000317', 'ACH-000318', 'ACH-000319', 'ACH-000320', 'ACH-000322', 'ACH-000323', 'ACH-000324', 'ACH-000325', 'ACH-000326', 'ACH-000327', 'ACH-000329', 'ACH-000330', 'ACH-000331', 'ACH-000332', 'ACH-000334', 'ACH-000335', 'ACH-000336', 'ACH-000338', 'ACH-000341', 'ACH-000343', 'ACH-000344', 'ACH-000345', 'ACH-000346', 'ACH-000347', 'ACH-000348', 'ACH-000349', 'ACH-000350', 'ACH-000351', 'ACH-000352', 'ACH-000353', 'ACH-000354', 'ACH-000355', 'ACH-000356', 'ACH-000359', 'ACH-000360', 'ACH-000361', 'ACH-000362', 'ACH-000363', 'ACH-000364', 'ACH-000365', 'ACH-000366', 'ACH-000367', 'ACH-000368', 'ACH-000369', 'ACH-000370', 'ACH-000371', 'ACH-000373', 'ACH-000374', 'ACH-000375', 'ACH-000376', 'ACH-000377', 'ACH-000378', 'ACH-000380', 'ACH-000381', 'ACH-000384', 'ACH-000387', 'ACH-000389', 'ACH-000390', 'ACH-000391', 'ACH-000392', 'ACH-000393', 'ACH-000395', 'ACH-000396', 'ACH-000397', 'ACH-000398', 'ACH-000400', 'ACH-000401', 'ACH-000402', 'ACH-000403', 'ACH-000404', 'ACH-000405', 'ACH-000406', 'ACH-000407', 'ACH-000408', 'ACH-000409', 'ACH-000410', 'ACH-000411', 'ACH-000414', 'ACH-000415', 'ACH-000416', 'ACH-000417', 'ACH-000418', 'ACH-000419', 'ACH-000420', 'ACH-000421', 'ACH-000422', 'ACH-000423', 'ACH-000424', 'ACH-000425', 'ACH-000426', 'ACH-000427', 'ACH-000429', 'ACH-000430', 'ACH-000431', 'ACH-000433', 'ACH-000434', 'ACH-000435', 'ACH-000437', 'ACH-000438', 'ACH-000441', 'ACH-000442', 'ACH-000444', 'ACH-000445', 'ACH-000446', 'ACH-000447', 'ACH-000448', 'ACH-000450', 'ACH-000452', 'ACH-000454', 'ACH-000456', 'ACH-000457', 'ACH-000458', 'ACH-000459', 'ACH-000460', 'ACH-000461', 'ACH-000463', 'ACH-000464', 'ACH-000466', 'ACH-000467', 'ACH-000468', 'ACH-000469', 'ACH-000470', 'ACH-000471', 'ACH-000472', 'ACH-000473', 'ACH-000475', 'ACH-000476', 'ACH-000477', 'ACH-000478', 'ACH-000479', 'ACH-000480', 'ACH-000481', 'ACH-000483', 'ACH-000484', 'ACH-000485', 'ACH-000486', 'ACH-000487', 'ACH-000488', 'ACH-000489', 'ACH-000491', 'ACH-000493', 'ACH-000495', 'ACH-000496', 'ACH-000498', 'ACH-000500', 'ACH-000501', 'ACH-000502', 'ACH-000503', 'ACH-000504', 'ACH-000505', 'ACH-000506', 'ACH-000507', 'ACH-000508', 'ACH-000510', 'ACH-000511', 'ACH-000514', 'ACH-000515', 'ACH-000516', 'ACH-000517', 'ACH-000518', 'ACH-000519', 'ACH-000520', 'ACH-000521', 'ACH-000522', 'ACH-000524', 'ACH-000525', 'ACH-000527', 'ACH-000528', 'ACH-000530', 'ACH-000532', 'ACH-000534', 'ACH-000535', 'ACH-000536', 'ACH-000537', 'ACH-000538', 'ACH-000541', 'ACH-000542', 'ACH-000544', 'ACH-000545', 'ACH-000546', 'ACH-000547', 'ACH-000548', 'ACH-000549', 'ACH-000550', 'ACH-000551', 'ACH-000552', 'ACH-000553', 'ACH-000554', 'ACH-000555', 'ACH-000557', 'ACH-000558', 'ACH-000559', 'ACH-000560', 'ACH-000562', 'ACH-000563', 'ACH-000565', 'ACH-000567', 'ACH-000568', 'ACH-000569', 'ACH-000570', 'ACH-000571', 'ACH-000572', 'ACH-000573', 'ACH-000574', 'ACH-000576', 'ACH-000577', 'ACH-000579', 'ACH-000580', 'ACH-000581', 'ACH-000582', 'ACH-000583', 'ACH-000584', 'ACH-000585', 'ACH-000586', 'ACH-000587', 'ACH-000588', 'ACH-000589', 'ACH-000592', 'ACH-000593', 'ACH-000595', 'ACH-000596', 'ACH-000599', 'ACH-000600', 'ACH-000601', 'ACH-000602', 'ACH-000604', 'ACH-000605', 'ACH-000606', 'ACH-000607', 'ACH-000608', 'ACH-000609', 'ACH-000611', 'ACH-000614', 'ACH-000615', 'ACH-000616', 'ACH-000617', 'ACH-000618', 'ACH-000619', 'ACH-000620', 'ACH-000621', 'ACH-000622', 'ACH-000623', 'ACH-000624', 'ACH-000625', 'ACH-000627', 'ACH-000630', 'ACH-000631', 'ACH-000632', 'ACH-000633', 'ACH-000637', 'ACH-000638', 'ACH-000639', 'ACH-000641', 'ACH-000643', 'ACH-000644', 'ACH-000645', 'ACH-000646', 'ACH-000647', 'ACH-000648', 'ACH-000649', 'ACH-000650', 'ACH-000651', 'ACH-000652', 'ACH-000653', 'ACH-000654', 'ACH-000655', 'ACH-000656', 'ACH-000657', 'ACH-000660', 'ACH-000661', 'ACH-000662', 'ACH-000663', 'ACH-000664', 'ACH-000665', 'ACH-000666', 'ACH-000667', 'ACH-000668', 'ACH-000670', 'ACH-000671', 'ACH-000672', 'ACH-000677', 'ACH-000678', 'ACH-000680', 'ACH-000681', 'ACH-000682', 'ACH-000683', 'ACH-000684', 'ACH-000685', 'ACH-000688', 'ACH-000693', 'ACH-000694', 'ACH-000695', 'ACH-000696', 'ACH-000697', 'ACH-000698', 'ACH-000699', 'ACH-000701', 'ACH-000702', 'ACH-000704', 'ACH-000706', 'ACH-000709', 'ACH-000710', 'ACH-000711', 'ACH-000712', 'ACH-000713', 'ACH-000714', 'ACH-000716', 'ACH-000717', 'ACH-000718', 'ACH-000719', 'ACH-000720', 'ACH-000721', 'ACH-000722', 'ACH-000724', 'ACH-000725', 'ACH-000726', 'ACH-000727', 'ACH-000728', 'ACH-000730', 'ACH-000732', 'ACH-000733', 'ACH-000734', 'ACH-000735', 'ACH-000736', 'ACH-000738', 'ACH-000739', 'ACH-000740', 'ACH-000741', 'ACH-000744', 'ACH-000746', 'ACH-000747', 'ACH-000748', 'ACH-000749', 'ACH-000750', 'ACH-000752', 'ACH-000753', 'ACH-000754', 'ACH-000755', 'ACH-000756', 'ACH-000758', 'ACH-000759', 'ACH-000762', 'ACH-000763', 'ACH-000764', 'ACH-000765', 'ACH-000766', 'ACH-000767', 'ACH-000768', 'ACH-000769', 'ACH-000770', 'ACH-000771', 'ACH-000773', 'ACH-000774', 'ACH-000775', 'ACH-000776', 'ACH-000777', 'ACH-000778', 'ACH-000780', 'ACH-000781', 'ACH-000782', 'ACH-000783', 'ACH-000784', 'ACH-000785', 'ACH-000787', 'ACH-000788', 'ACH-000790', 'ACH-000791', 'ACH-000792', 'ACH-000793', 'ACH-000794', 'ACH-000796', 'ACH-000798', 'ACH-000799', 'ACH-000800', 'ACH-000801', 'ACH-000802', 'ACH-000804', 'ACH-000805', 'ACH-000807', 'ACH-000809', 'ACH-000810', 'ACH-000811', 'ACH-000812', 'ACH-000814', 'ACH-000817', 'ACH-000818', 'ACH-000819', 'ACH-000820', 'ACH-000821', 'ACH-000822', 'ACH-000823', 'ACH-000824', 'ACH-000826', 'ACH-000827', 'ACH-000828', 'ACH-000829', 'ACH-000831', 'ACH-000832', 'ACH-000833', 'ACH-000834', 'ACH-000835', 'ACH-000836', 'ACH-000837', 'ACH-000838', 'ACH-000839', 'ACH-000840', 'ACH-000841', 'ACH-000842', 'ACH-000843', 'ACH-000845', 'ACH-000846', 'ACH-000847', 'ACH-000848', 'ACH-000849', 'ACH-000851', 'ACH-000852', 'ACH-000853', 'ACH-000855', 'ACH-000856', 'ACH-000857', 'ACH-000858', 'ACH-000859', 'ACH-000860', 'ACH-000861', 'ACH-000862', 'ACH-000863', 'ACH-000864', 'ACH-000865', 'ACH-000866', 'ACH-000867', 'ACH-000869', 'ACH-000871', 'ACH-000873', 'ACH-000874', 'ACH-000876', 'ACH-000878', 'ACH-000879', 'ACH-000880', 'ACH-000881', 'ACH-000882', 'ACH-000883', 'ACH-000884', 'ACH-000885', 'ACH-000886', 'ACH-000888', 'ACH-000889', 'ACH-000890', 'ACH-000894', 'ACH-000895', 'ACH-000896', 'ACH-000898', 'ACH-000899', 'ACH-000900', 'ACH-000901', 'ACH-000902', 'ACH-000903', 'ACH-000905', 'ACH-000906', 'ACH-000907', 'ACH-000909', 'ACH-000910', 'ACH-000911', 'ACH-000912', 'ACH-000913', 'ACH-000914', 'ACH-000915', 'ACH-000916', 'ACH-000917', 'ACH-000919', 'ACH-000920', 'ACH-000921', 'ACH-000922', 'ACH-000924', 'ACH-000926', 'ACH-000927', 'ACH-000928', 'ACH-000929', 'ACH-000930', 'ACH-000932', 'ACH-000934', 'ACH-000935', 'ACH-000936', 'ACH-000937', 'ACH-000938', 'ACH-000939', 'ACH-000940', 'ACH-000941', 'ACH-000943', 'ACH-000944', 'ACH-000945', 'ACH-000946', 'ACH-000947', 'ACH-000948', 'ACH-000950', 'ACH-000952', 'ACH-000953', 'ACH-000954', 'ACH-000955', 'ACH-000956', 'ACH-000957', 'ACH-000958', 'ACH-000959', 'ACH-000960', 'ACH-000961', 'ACH-000962', 'ACH-000963', 'ACH-000965', 'ACH-000966', 'ACH-000967', 'ACH-000968', 'ACH-000969', 'ACH-000970', 'ACH-000971', 'ACH-000972', 'ACH-000973', 'ACH-000974', 'ACH-000976', 'ACH-000977', 'ACH-000978', 'ACH-000979', 'ACH-000980', 'ACH-000981', 'ACH-000982', 'ACH-000983', 'ACH-000984', 'ACH-000985', 'ACH-000986', 'ACH-000987', 'ACH-000988', 'ACH-000991', 'ACH-000993', 'ACH-000994', 'ACH-000995', 'ACH-000996', 'ACH-000997', 'ACH-000998', 'ACH-001001', 'ACH-001016', 'ACH-001036', 'ACH-001039', 'ACH-001041', 'ACH-001048', 'ACH-001061', 'ACH-001065', 'ACH-001075', 'ACH-001089', 'ACH-001098', 'ACH-001107', 'ACH-001113', 'ACH-001126', 'ACH-001129', 'ACH-001145', 'ACH-001151', 'ACH-001183', 'ACH-001188', 'ACH-001190', 'ACH-001194', 'ACH-001233', 'ACH-001239', 'ACH-001306', 'ACH-001307', 'ACH-001318']\n"
     ]
    }
   ],
   "source": [
    "processed_crispr_samples = processed_crispr_df.columns[2:].tolist()\n",
    "processed_rna_samples = processed_rna_df.columns[2:].tolist()\n",
    "\n",
    "# get the union and intersection samples of the two lists\n",
    "overlapped_crispr_rna_samples = sorted(list(set(processed_crispr_samples) & set(processed_rna_samples)))\n",
    "print(\"The overlapped samples between protein_crispr_samples and protein_rna_samples are:\")\n",
    "print(len(overlapped_crispr_rna_samples))\n",
    "print(overlapped_crispr_rna_samples)\n",
    "\n",
    "# get the union set of samples of the two lists\n",
    "union_crispr_rna_samples = sorted(list(set(processed_crispr_samples) | set(processed_rna_samples)))\n",
    "print(\"The union samples between protein_crispr_samples and protein_rna_samples are:\")\n",
    "print(len(union_crispr_rna_samples))\n",
    "print(union_crispr_rna_samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 380,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The overlapped samples between protein_crispr_samples and protein_rna_samples are:\n",
      "178\n",
      "['ACH-000001', 'ACH-000012', 'ACH-000014', 'ACH-000019', 'ACH-000023', 'ACH-000028', 'ACH-000035', 'ACH-000039', 'ACH-000042', 'ACH-000052', 'ACH-000087', 'ACH-000090', 'ACH-000097', 'ACH-000111', 'ACH-000114', 'ACH-000115', 'ACH-000118', 'ACH-000132', 'ACH-000138', 'ACH-000147', 'ACH-000153', 'ACH-000155', 'ACH-000159', 'ACH-000161', 'ACH-000164', 'ACH-000219', 'ACH-000222', 'ACH-000223', 'ACH-000235', 'ACH-000239', 'ACH-000243', 'ACH-000247', 'ACH-000248', 'ACH-000250', 'ACH-000255', 'ACH-000264', 'ACH-000278', 'ACH-000288', 'ACH-000291', 'ACH-000304', 'ACH-000311', 'ACH-000318', 'ACH-000330', 'ACH-000343', 'ACH-000348', 'ACH-000356', 'ACH-000361', 'ACH-000364', 'ACH-000374', 'ACH-000378', 'ACH-000390', 'ACH-000393', 'ACH-000404', 'ACH-000407', 'ACH-000410', 'ACH-000411', 'ACH-000414', 'ACH-000416', 'ACH-000420', 'ACH-000425', 'ACH-000427', 'ACH-000430', 'ACH-000433', 'ACH-000438', 'ACH-000441', 'ACH-000454', 'ACH-000463', 'ACH-000475', 'ACH-000480', 'ACH-000481', 'ACH-000484', 'ACH-000488', 'ACH-000493', 'ACH-000495', 'ACH-000496', 'ACH-000502', 'ACH-000511', 'ACH-000521', 'ACH-000528', 'ACH-000535', 'ACH-000542', 'ACH-000550', 'ACH-000563', 'ACH-000573', 'ACH-000579', 'ACH-000580', 'ACH-000587', 'ACH-000589', 'ACH-000605', 'ACH-000614', 'ACH-000621', 'ACH-000624', 'ACH-000627', 'ACH-000632', 'ACH-000638', 'ACH-000647', 'ACH-000649', 'ACH-000652', 'ACH-000657', 'ACH-000661', 'ACH-000662', 'ACH-000667', 'ACH-000668', 'ACH-000671', 'ACH-000672', 'ACH-000677', 'ACH-000681', 'ACH-000684', 'ACH-000685', 'ACH-000693', 'ACH-000696', 'ACH-000699', 'ACH-000701', 'ACH-000713', 'ACH-000718', 'ACH-000739', 'ACH-000748', 'ACH-000749', 'ACH-000768', 'ACH-000777', 'ACH-000783', 'ACH-000784', 'ACH-000785', 'ACH-000787', 'ACH-000788', 'ACH-000790', 'ACH-000793', 'ACH-000799', 'ACH-000800', 'ACH-000805', 'ACH-000810', 'ACH-000824', 'ACH-000827', 'ACH-000831', 'ACH-000847', 'ACH-000849', 'ACH-000853', 'ACH-000855', 'ACH-000856', 'ACH-000859', 'ACH-000860', 'ACH-000866', 'ACH-000867', 'ACH-000869', 'ACH-000878', 'ACH-000879', 'ACH-000880', 'ACH-000881', 'ACH-000882', 'ACH-000886', 'ACH-000888', 'ACH-000890', 'ACH-000898', 'ACH-000899', 'ACH-000900', 'ACH-000910', 'ACH-000911', 'ACH-000912', 'ACH-000915', 'ACH-000917', 'ACH-000924', 'ACH-000929', 'ACH-000932', 'ACH-000945', 'ACH-000946', 'ACH-000956', 'ACH-000966', 'ACH-000974', 'ACH-000979', 'ACH-000980', 'ACH-000984', 'ACH-000994', 'ACH-000996', 'ACH-001075', 'ACH-001190', 'ACH-001239', 'ACH-001306', 'ACH-001307']\n",
      "The overlapped samples between protein_crispr_samples and protein_rna_samples are:\n",
      "219\n",
      "['ACH-000001', 'ACH-000008', 'ACH-000012', 'ACH-000014', 'ACH-000019', 'ACH-000023', 'ACH-000028', 'ACH-000035', 'ACH-000039', 'ACH-000042', 'ACH-000052', 'ACH-000062', 'ACH-000087', 'ACH-000090', 'ACH-000097', 'ACH-000109', 'ACH-000111', 'ACH-000114', 'ACH-000115', 'ACH-000117', 'ACH-000118', 'ACH-000132', 'ACH-000138', 'ACH-000147', 'ACH-000153', 'ACH-000155', 'ACH-000159', 'ACH-000161', 'ACH-000164', 'ACH-000219', 'ACH-000222', 'ACH-000223', 'ACH-000235', 'ACH-000239', 'ACH-000243', 'ACH-000247', 'ACH-000248', 'ACH-000250', 'ACH-000255', 'ACH-000264', 'ACH-000278', 'ACH-000288', 'ACH-000291', 'ACH-000304', 'ACH-000311', 'ACH-000318', 'ACH-000330', 'ACH-000335', 'ACH-000343', 'ACH-000347', 'ACH-000348', 'ACH-000349', 'ACH-000351', 'ACH-000356', 'ACH-000361', 'ACH-000364', 'ACH-000374', 'ACH-000378', 'ACH-000390', 'ACH-000393', 'ACH-000404', 'ACH-000407', 'ACH-000409', 'ACH-000410', 'ACH-000411', 'ACH-000414', 'ACH-000416', 'ACH-000420', 'ACH-000423', 'ACH-000425', 'ACH-000427', 'ACH-000429', 'ACH-000430', 'ACH-000433', 'ACH-000438', 'ACH-000441', 'ACH-000447', 'ACH-000454', 'ACH-000463', 'ACH-000475', 'ACH-000476', 'ACH-000480', 'ACH-000481', 'ACH-000484', 'ACH-000488', 'ACH-000493', 'ACH-000495', 'ACH-000496', 'ACH-000502', 'ACH-000506', 'ACH-000508', 'ACH-000511', 'ACH-000521', 'ACH-000528', 'ACH-000530', 'ACH-000535', 'ACH-000536', 'ACH-000542', 'ACH-000550', 'ACH-000553', 'ACH-000555', 'ACH-000563', 'ACH-000573', 'ACH-000574', 'ACH-000579', 'ACH-000580', 'ACH-000582', 'ACH-000587', 'ACH-000589', 'ACH-000605', 'ACH-000614', 'ACH-000615', 'ACH-000617', 'ACH-000621', 'ACH-000624', 'ACH-000625', 'ACH-000627', 'ACH-000632', 'ACH-000638', 'ACH-000643', 'ACH-000644', 'ACH-000647', 'ACH-000649', 'ACH-000650', 'ACH-000652', 'ACH-000657', 'ACH-000661', 'ACH-000662', 'ACH-000667', 'ACH-000668', 'ACH-000670', 'ACH-000671', 'ACH-000672', 'ACH-000677', 'ACH-000678', 'ACH-000681', 'ACH-000684', 'ACH-000685', 'ACH-000693', 'ACH-000695', 'ACH-000696', 'ACH-000699', 'ACH-000701', 'ACH-000712', 'ACH-000713', 'ACH-000718', 'ACH-000726', 'ACH-000730', 'ACH-000739', 'ACH-000748', 'ACH-000749', 'ACH-000750', 'ACH-000752', 'ACH-000755', 'ACH-000768', 'ACH-000777', 'ACH-000783', 'ACH-000784', 'ACH-000785', 'ACH-000787', 'ACH-000788', 'ACH-000790', 'ACH-000793', 'ACH-000799', 'ACH-000800', 'ACH-000805', 'ACH-000810', 'ACH-000824', 'ACH-000827', 'ACH-000831', 'ACH-000847', 'ACH-000849', 'ACH-000852', 'ACH-000853', 'ACH-000855', 'ACH-000856', 'ACH-000857', 'ACH-000859', 'ACH-000860', 'ACH-000866', 'ACH-000867', 'ACH-000869', 'ACH-000878', 'ACH-000879', 'ACH-000880', 'ACH-000881', 'ACH-000882', 'ACH-000886', 'ACH-000888', 'ACH-000890', 'ACH-000898', 'ACH-000899', 'ACH-000900', 'ACH-000909', 'ACH-000910', 'ACH-000911', 'ACH-000912', 'ACH-000915', 'ACH-000917', 'ACH-000924', 'ACH-000929', 'ACH-000932', 'ACH-000945', 'ACH-000946', 'ACH-000954', 'ACH-000956', 'ACH-000966', 'ACH-000974', 'ACH-000979', 'ACH-000980', 'ACH-000984', 'ACH-000987', 'ACH-000994', 'ACH-000996', 'ACH-001075', 'ACH-001190', 'ACH-001239', 'ACH-001306', 'ACH-001307']\n"
     ]
    }
   ],
   "source": [
    "# get the intersection based on the overlapped over omics not union over omics by overlapped_omics_annotation_samples\n",
    "overlapped_omics_crispr_rna_samples = sorted(list(set(overlapped_crispr_rna_samples) & set(overlapped_omics_annotation_samples)))\n",
    "print(\"The overlapped samples between protein_crispr_samples and protein_rna_samples are:\")\n",
    "print(len(overlapped_omics_crispr_rna_samples))\n",
    "print(overlapped_omics_crispr_rna_samples)\n",
    "\n",
    "# get the intersection based on the overlapped over omics not union over omics by overlapped_omics_annotation_samples for union_crispr_rna_samples\n",
    "overlapped_omics_union_crispr_rna_samples = sorted(list(set(union_crispr_rna_samples) & set(overlapped_omics_annotation_samples)))\n",
    "print(\"The overlapped samples between protein_crispr_samples and protein_rna_samples are:\")\n",
    "print(len(overlapped_omics_union_crispr_rna_samples))\n",
    "print(overlapped_omics_union_crispr_rna_samples)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 9. Sample Splits"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 9.1 Pretrain, Drug and Target Sample Splits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ACH-000001', 'ACH-000002', 'ACH-000004', 'ACH-000006', 'ACH-000007', 'ACH-000008', 'ACH-000009', 'ACH-000012', 'ACH-000015', 'ACH-000018', 'ACH-000019', 'ACH-000020', 'ACH-000021', 'ACH-000023', 'ACH-000024', 'ACH-000030', 'ACH-000035', 'ACH-000038', 'ACH-000040', 'ACH-000042', 'ACH-000045', 'ACH-000046', 'ACH-000047', 'ACH-000048', 'ACH-000050', 'ACH-000052', 'ACH-000053', 'ACH-000054', 'ACH-000055', 'ACH-000056', 'ACH-000058', 'ACH-000059', 'ACH-000060', 'ACH-000061', 'ACH-000062', 'ACH-000065', 'ACH-000067', 'ACH-000070', 'ACH-000072', 'ACH-000073', 'ACH-000074', 'ACH-000075', 'ACH-000078', 'ACH-000081', 'ACH-000082', 'ACH-000087', 'ACH-000090', 'ACH-000091', 'ACH-000092', 'ACH-000094', 'ACH-000096', 'ACH-000098', 'ACH-000099', 'ACH-000100', 'ACH-000101', 'ACH-000102', 'ACH-000103', 'ACH-000104', 'ACH-000105', 'ACH-000106', 'ACH-000107', 'ACH-000108', 'ACH-000111', 'ACH-000112', 'ACH-000113', 'ACH-000114', 'ACH-000115', 'ACH-000116', 'ACH-000117', 'ACH-000118', 'ACH-000120', 'ACH-000121', 'ACH-000124', 'ACH-000129', 'ACH-000132', 'ACH-000136', 'ACH-000137', 'ACH-000138', 'ACH-000139', 'ACH-000142', 'ACH-000144', 'ACH-000145', 'ACH-000146', 'ACH-000147', 'ACH-000148', 'ACH-000149', 'ACH-000151', 'ACH-000153', 'ACH-000155', 'ACH-000157', 'ACH-000159', 'ACH-000161', 'ACH-000162', 'ACH-000168', 'ACH-000169', 'ACH-000171', 'ACH-000174', 'ACH-000176', 'ACH-000178', 'ACH-000181', 'ACH-000183', 'ACH-000186', 'ACH-000187', 'ACH-000188', 'ACH-000189', 'ACH-000190', 'ACH-000191', 'ACH-000192', 'ACH-000193', 'ACH-000196', 'ACH-000197', 'ACH-000198', 'ACH-000200', 'ACH-000201', 'ACH-000204', 'ACH-000207', 'ACH-000210', 'ACH-000211', 'ACH-000212', 'ACH-000213', 'ACH-000217', 'ACH-000218', 'ACH-000219', 'ACH-000221', 'ACH-000222', 'ACH-000223', 'ACH-000225', 'ACH-000226', 'ACH-000227', 'ACH-000228', 'ACH-000231', 'ACH-000232', 'ACH-000233', 'ACH-000235', 'ACH-000236', 'ACH-000238', 'ACH-000242', 'ACH-000243', 'ACH-000244', 'ACH-000245', 'ACH-000247', 'ACH-000248', 'ACH-000249', 'ACH-000250', 'ACH-000252', 'ACH-000254', 'ACH-000257', 'ACH-000258', 'ACH-000259', 'ACH-000260', 'ACH-000263', 'ACH-000264', 'ACH-000265', 'ACH-000267', 'ACH-000269', 'ACH-000270', 'ACH-000271', 'ACH-000273', 'ACH-000276', 'ACH-000277', 'ACH-000281', 'ACH-000282', 'ACH-000287', 'ACH-000288', 'ACH-000290', 'ACH-000291', 'ACH-000292', 'ACH-000293', 'ACH-000294', 'ACH-000295', 'ACH-000298', 'ACH-000301', 'ACH-000303', 'ACH-000304', 'ACH-000305', 'ACH-000308', 'ACH-000309', 'ACH-000311', 'ACH-000315', 'ACH-000318', 'ACH-000319', 'ACH-000320', 'ACH-000322', 'ACH-000323', 'ACH-000326', 'ACH-000327', 'ACH-000329', 'ACH-000330', 'ACH-000332', 'ACH-000334', 'ACH-000335', 'ACH-000336', 'ACH-000338', 'ACH-000341', 'ACH-000343', 'ACH-000345', 'ACH-000346', 'ACH-000347', 'ACH-000348', 'ACH-000349', 'ACH-000350', 'ACH-000351', 'ACH-000352', 'ACH-000353', 'ACH-000354', 'ACH-000355', 'ACH-000356', 'ACH-000357', 'ACH-000358', 'ACH-000359', 'ACH-000360', 'ACH-000361', 'ACH-000362', 'ACH-000363', 'ACH-000364', 'ACH-000365', 'ACH-000366', 'ACH-000367', 'ACH-000369', 'ACH-000371', 'ACH-000372', 'ACH-000373', 'ACH-000374', 'ACH-000375', 'ACH-000376', 'ACH-000378', 'ACH-000379', 'ACH-000380', 'ACH-000381', 'ACH-000384', 'ACH-000386', 'ACH-000389', 'ACH-000391', 'ACH-000392', 'ACH-000394', 'ACH-000395', 'ACH-000396', 'ACH-000399', 'ACH-000400', 'ACH-000401', 'ACH-000403', 'ACH-000408', 'ACH-000410', 'ACH-000411', 'ACH-000414', 'ACH-000416', 'ACH-000417', 'ACH-000420', 'ACH-000421', 'ACH-000422', 'ACH-000423', 'ACH-000424', 'ACH-000425', 'ACH-000427', 'ACH-000428', 'ACH-000429', 'ACH-000430', 'ACH-000431', 'ACH-000432', 'ACH-000433', 'ACH-000434', 'ACH-000437', 'ACH-000438', 'ACH-000439', 'ACH-000440', 'ACH-000441', 'ACH-000442', 'ACH-000443', 'ACH-000447', 'ACH-000448', 'ACH-000449', 'ACH-000450', 'ACH-000451', 'ACH-000452', 'ACH-000456', 'ACH-000457', 'ACH-000463', 'ACH-000464', 'ACH-000465', 'ACH-000469', 'ACH-000470', 'ACH-000472', 'ACH-000473', 'ACH-000475', 'ACH-000476', 'ACH-000478', 'ACH-000480', 'ACH-000481', 'ACH-000482', 'ACH-000483', 'ACH-000484', 'ACH-000486', 'ACH-000488', 'ACH-000489', 'ACH-000491', 'ACH-000493', 'ACH-000496', 'ACH-000501', 'ACH-000504', 'ACH-000505', 'ACH-000506', 'ACH-000508', 'ACH-000510', 'ACH-000514', 'ACH-000515', 'ACH-000516', 'ACH-000518', 'ACH-000521', 'ACH-000522', 'ACH-000524', 'ACH-000525', 'ACH-000527', 'ACH-000528', 'ACH-000530', 'ACH-000532', 'ACH-000534', 'ACH-000535', 'ACH-000536', 'ACH-000538', 'ACH-000544', 'ACH-000545', 'ACH-000546', 'ACH-000547', 'ACH-000548', 'ACH-000551', 'ACH-000552', 'ACH-000554', 'ACH-000555', 'ACH-000558', 'ACH-000559', 'ACH-000560', 'ACH-000562', 'ACH-000563', 'ACH-000565', 'ACH-000566', 'ACH-000567', 'ACH-000568', 'ACH-000569', 'ACH-000570', 'ACH-000571', 'ACH-000572', 'ACH-000573', 'ACH-000574', 'ACH-000577', 'ACH-000579', 'ACH-000580', 'ACH-000581', 'ACH-000583', 'ACH-000584', 'ACH-000585', 'ACH-000586', 'ACH-000587', 'ACH-000589', 'ACH-000595', 'ACH-000596', 'ACH-000599', 'ACH-000601', 'ACH-000603', 'ACH-000605', 'ACH-000607', 'ACH-000609', 'ACH-000610', 'ACH-000611', 'ACH-000613', 'ACH-000614', 'ACH-000615', 'ACH-000616', 'ACH-000617', 'ACH-000619', 'ACH-000620', 'ACH-000621', 'ACH-000622', 'ACH-000624', 'ACH-000625', 'ACH-000626', 'ACH-000627', 'ACH-000628', 'ACH-000631', 'ACH-000633', 'ACH-000636', 'ACH-000637', 'ACH-000638', 'ACH-000639', 'ACH-000640', 'ACH-000641', 'ACH-000643', 'ACH-000644', 'ACH-000647', 'ACH-000648', 'ACH-000649', 'ACH-000650', 'ACH-000651', 'ACH-000652', 'ACH-000653', 'ACH-000654', 'ACH-000655', 'ACH-000656', 'ACH-000657', 'ACH-000660', 'ACH-000662', 'ACH-000663', 'ACH-000664', 'ACH-000665', 'ACH-000666', 'ACH-000667', 'ACH-000668', 'ACH-000669', 'ACH-000670', 'ACH-000672', 'ACH-000674', 'ACH-000675', 'ACH-000677', 'ACH-000678', 'ACH-000679', 'ACH-000680', 'ACH-000681', 'ACH-000684', 'ACH-000688', 'ACH-000689', 'ACH-000691', 'ACH-000693', 'ACH-000694', 'ACH-000696', 'ACH-000697', 'ACH-000698', 'ACH-000699', 'ACH-000702', 'ACH-000703', 'ACH-000704', 'ACH-000706', 'ACH-000711', 'ACH-000713', 'ACH-000714', 'ACH-000716', 'ACH-000717', 'ACH-000718', 'ACH-000719', 'ACH-000720', 'ACH-000722', 'ACH-000724', 'ACH-000725', 'ACH-000727', 'ACH-000729', 'ACH-000730', 'ACH-000733', 'ACH-000738', 'ACH-000740', 'ACH-000743', 'ACH-000744', 'ACH-000745', 'ACH-000747', 'ACH-000748', 'ACH-000749', 'ACH-000750', 'ACH-000751', 'ACH-000752', 'ACH-000754', 'ACH-000755', 'ACH-000756', 'ACH-000759', 'ACH-000763', 'ACH-000766', 'ACH-000767', 'ACH-000768', 'ACH-000769', 'ACH-000770', 'ACH-000772', 'ACH-000775', 'ACH-000776', 'ACH-000778', 'ACH-000780', 'ACH-000781', 'ACH-000783', 'ACH-000784', 'ACH-000786', 'ACH-000787', 'ACH-000788', 'ACH-000789', 'ACH-000790', 'ACH-000792', 'ACH-000793', 'ACH-000794', 'ACH-000795', 'ACH-000798', 'ACH-000800', 'ACH-000802', 'ACH-000803', 'ACH-000804', 'ACH-000805', 'ACH-000806', 'ACH-000809', 'ACH-000810', 'ACH-000811', 'ACH-000812', 'ACH-000814', 'ACH-000815', 'ACH-000816', 'ACH-000817', 'ACH-000818', 'ACH-000819', 'ACH-000821', 'ACH-000822', 'ACH-000823', 'ACH-000824', 'ACH-000825', 'ACH-000826', 'ACH-000827', 'ACH-000828', 'ACH-000830', 'ACH-000832', 'ACH-000835', 'ACH-000838', 'ACH-000840', 'ACH-000841', 'ACH-000843', 'ACH-000846', 'ACH-000847', 'ACH-000848', 'ACH-000849', 'ACH-000852', 'ACH-000853', 'ACH-000855', 'ACH-000856', 'ACH-000857', 'ACH-000858', 'ACH-000859', 'ACH-000860', 'ACH-000861', 'ACH-000863', 'ACH-000864', 'ACH-000865', 'ACH-000866', 'ACH-000867', 'ACH-000869', 'ACH-000871', 'ACH-000873', 'ACH-000874', 'ACH-000875', 'ACH-000876', 'ACH-000878', 'ACH-000879', 'ACH-000880', 'ACH-000881', 'ACH-000882', 'ACH-000883', 'ACH-000885', 'ACH-000886', 'ACH-000888', 'ACH-000890', 'ACH-000892', 'ACH-000893', 'ACH-000894', 'ACH-000895', 'ACH-000896', 'ACH-000900', 'ACH-000902', 'ACH-000903', 'ACH-000905', 'ACH-000906', 'ACH-000910', 'ACH-000911', 'ACH-000913', 'ACH-000914', 'ACH-000915', 'ACH-000916', 'ACH-000917', 'ACH-000918', 'ACH-000919', 'ACH-000920', 'ACH-000922', 'ACH-000924', 'ACH-000926', 'ACH-000927', 'ACH-000929', 'ACH-000930', 'ACH-000932', 'ACH-000934', 'ACH-000935', 'ACH-000936', 'ACH-000937', 'ACH-000938', 'ACH-000939', 'ACH-000940', 'ACH-000943', 'ACH-000944', 'ACH-000945', 'ACH-000947', 'ACH-000948', 'ACH-000949', 'ACH-000950', 'ACH-000951', 'ACH-000953', 'ACH-000955', 'ACH-000956', 'ACH-000957', 'ACH-000958', 'ACH-000960', 'ACH-000962', 'ACH-000963', 'ACH-000965', 'ACH-000966', 'ACH-000968', 'ACH-000969', 'ACH-000970', 'ACH-000971', 'ACH-000973', 'ACH-000974', 'ACH-000976', 'ACH-000977', 'ACH-000978', 'ACH-000979', 'ACH-000980', 'ACH-000981', 'ACH-000983', 'ACH-000985', 'ACH-000986', 'ACH-000987', 'ACH-000988', 'ACH-000989', 'ACH-000991', 'ACH-000995', 'ACH-000997', 'ACH-000998', 'ACH-000999', 'ACH-001016', 'ACH-001021', 'ACH-001023', 'ACH-001024', 'ACH-001039', 'ACH-001049', 'ACH-001063', 'ACH-001064', 'ACH-001065', 'ACH-001075', 'ACH-001089', 'ACH-001106', 'ACH-001107', 'ACH-001125', 'ACH-001126', 'ACH-001127', 'ACH-001138', 'ACH-001151', 'ACH-001182', 'ACH-001190', 'ACH-001199', 'ACH-001203', 'ACH-001233', 'ACH-001306', 'ACH-001307']\n",
      "length of dti_overlapped_samples:  649\n"
     ]
    }
   ],
   "source": [
    "dti_overlapped_samples = list(dti_overlapped_samples_df['depMapID'])\n",
    "print(dti_overlapped_samples)\n",
    "print(\"length of dti_overlapped_samples: \", len(set(dti_overlapped_samples)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 9.1.1 Pretrain Samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 382,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(rest_samples): 336\n",
      "['ACH-000005', 'ACH-000011', 'ACH-000013', 'ACH-000014', 'ACH-000016']\n",
      "len(non_cancerous_rest_samples): 39\n",
      "['ACH-000043', 'ACH-000063', 'ACH-000071', 'ACH-000079', 'ACH-000083']\n",
      "len(cancerous_rest_samples): 297\n",
      "['ACH-000005', 'ACH-000011', 'ACH-000013', 'ACH-000014', 'ACH-000016']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>cancerous_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000005</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000011</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000013</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000014</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000016</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>ACH-001188</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>ACH-001194</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>ACH-001239</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>ACH-001318</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>ACH-001321</td>\n",
       "      <td>cancerous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>336 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       depMapID cancerous_status\n",
       "0    ACH-000005        cancerous\n",
       "1    ACH-000011        cancerous\n",
       "2    ACH-000013        cancerous\n",
       "3    ACH-000014        cancerous\n",
       "4    ACH-000016        cancerous\n",
       "..          ...              ...\n",
       "331  ACH-001188        cancerous\n",
       "332  ACH-001194        cancerous\n",
       "333  ACH-001239        cancerous\n",
       "334  ACH-001318        cancerous\n",
       "335  ACH-001321        cancerous\n",
       "\n",
       "[336 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "📊 Pretrain Sample Split Summary:\n",
      "Total rest samples: 336\n",
      "Pretrain plain samples: 67 (19.9%)\n",
      "Pretrain status samples: 269 (80.1%)\n",
      "\n",
      "📋 Pretrain Plain Sample Distribution:\n",
      "cancerous_status\n",
      "cancerous        59\n",
      "non-cancerous     8\n",
      "Name: count, dtype: int64\n",
      "cancerous_status\n",
      "cancerous        0.880597\n",
      "non-cancerous    0.119403\n",
      "Name: proportion, dtype: float64\n",
      "\n",
      "📋 Pretrain Status Sample Distribution:\n",
      "cancerous_status\n",
      "cancerous        238\n",
      "non-cancerous     31\n",
      "Name: count, dtype: int64\n",
      "cancerous_status\n",
      "cancerous        0.884758\n",
      "non-cancerous    0.115242\n",
      "Name: proportion, dtype: float64\n",
      "\n",
      "✅ Saved pretrain sample splits to:\n",
      "- ./data/pretrain_plain_data/pretrain_plain_samples.csv (67 samples)\n",
      "- ./data/pretrain_status_data/pretrain_status_samples.csv (269 samples)\n",
      "\n",
      "🔍 Overlap check: 0 overlapping samples (should be 0)\n",
      "\n",
      "📈 Final Cancerous/Non-cancerous Distribution:\n",
      "Pretrain Plain - Cancerous: 59, Non-cancerous: 8\n",
      "Pretrain Status - Cancerous: 238, Non-cancerous: 31\n",
      "\n",
      "📊 Pretrain Status Sample Split Summary:\n",
      "Total pretrain status samples: 269\n",
      "Training samples: 215 (79.9%)\n",
      "Test samples: 54 (20.1%)\n",
      "\n",
      "📋 Training Sample Distribution:\n",
      "cancerous_status\n",
      "cancerous        190\n",
      "non-cancerous     25\n",
      "Name: count, dtype: int64\n",
      "cancerous_status\n",
      "cancerous        0.883721\n",
      "non-cancerous    0.116279\n",
      "Name: proportion, dtype: float64\n",
      "\n",
      "📋 Test Sample Distribution:\n",
      "cancerous_status\n",
      "cancerous        48\n",
      "non-cancerous     6\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "# Your existing code for rest_samples is good\n",
    "rest_samples = sorted(list(set(overlapped_omics_union_annotation_samples) - set(dti_overlapped_samples)))\n",
    "print(\"len(rest_samples):\", len(rest_samples))\n",
    "print(rest_samples[:5])\n",
    "\n",
    "# check the non-cancerous samples in the rest_samples\n",
    "non_cancerous_rest_samples = sorted(list(set(non_cancerous_samples_df['Depmap Id']) & set(rest_samples)))\n",
    "print(\"len(non_cancerous_rest_samples):\", len(non_cancerous_rest_samples))\n",
    "print(non_cancerous_rest_samples[:5])\n",
    "\n",
    "cancerous_rest_samples = sorted(list(set(rest_samples) - set(non_cancerous_rest_samples)))\n",
    "print(\"len(cancerous_rest_samples):\", len(cancerous_rest_samples))\n",
    "print(cancerous_rest_samples[:5])\n",
    "\n",
    "# convert the rest_samples to a dataframe with marking the sample cancerous or non-cancerous status\n",
    "rest_samples_df = pd.DataFrame({\n",
    "    'depMapID': rest_samples,\n",
    "    'cancerous_status': ['cancerous' if sample in cancerous_rest_samples else 'non-cancerous' for sample in rest_samples]\n",
    "})\n",
    "display(rest_samples_df)\n",
    "\n",
    "# Split the rest_samples into pretrain_plain_samples and pretrain_status_samples\n",
    "# Sample 20% of the rest_samples for pretrain_plain_samples while maintaining the cancerous/non-cancerous ratio\n",
    "\n",
    "def stratified_split_pretrain_samples(rest_samples_df, pretrain_plain_ratio=0.2, random_state=42):\n",
    "    \"\"\"\n",
    "    Split rest samples into pretrain_plain and pretrain_status while maintaining \n",
    "    the cancerous/non-cancerous ratio in both splits.\n",
    "    \n",
    "    Args:\n",
    "        rest_samples_df: DataFrame with depMapID and cancerous_status columns\n",
    "        pretrain_plain_ratio: Ratio of samples to allocate to pretrain_plain (default 0.2)\n",
    "        random_state: Random seed for reproducibility\n",
    "    \n",
    "    Returns:\n",
    "        tuple: (pretrain_plain_samples, pretrain_status_samples)\n",
    "    \"\"\"\n",
    "    # Use stratified split to maintain the ratio\n",
    "    pretrain_plain_df, pretrain_status_df = train_test_split(\n",
    "        rest_samples_df,\n",
    "        test_size=1-pretrain_plain_ratio,  # 0.8 for pretrain_status\n",
    "        stratify=rest_samples_df['cancerous_status'],\n",
    "        random_state=random_state\n",
    "    )\n",
    "    \n",
    "    # Extract sample lists\n",
    "    pretrain_plain_samples = sorted(pretrain_plain_df['depMapID'].tolist())\n",
    "    pretrain_status_samples = sorted(pretrain_status_df['depMapID'].tolist())\n",
    "    \n",
    "    return pretrain_plain_samples, pretrain_status_samples, pretrain_plain_df, pretrain_status_df\n",
    "\n",
    "# Perform the split\n",
    "pretrain_plain_samples, pretrain_status_samples, pretrain_plain_df, pretrain_status_df = stratified_split_pretrain_samples(\n",
    "    rest_samples_df, \n",
    "    pretrain_plain_ratio=0.2, \n",
    "    random_state=42\n",
    ")\n",
    "\n",
    "# Display results\n",
    "print(f\"\\n📊 Pretrain Sample Split Summary:\")\n",
    "print(f\"Total rest samples: {len(rest_samples_df)}\")\n",
    "print(f\"Pretrain plain samples: {len(pretrain_plain_samples)} ({len(pretrain_plain_samples)/len(rest_samples_df)*100:.1f}%)\")\n",
    "print(f\"Pretrain status samples: {len(pretrain_status_samples)} ({len(pretrain_status_samples)/len(rest_samples_df)*100:.1f}%)\")\n",
    "\n",
    "print(f\"\\n📋 Pretrain Plain Sample Distribution:\")\n",
    "print(pretrain_plain_df['cancerous_status'].value_counts())\n",
    "print(pretrain_plain_df['cancerous_status'].value_counts(normalize=True))\n",
    "\n",
    "print(f\"\\n📋 Pretrain Status Sample Distribution:\")\n",
    "print(pretrain_status_df['cancerous_status'].value_counts())\n",
    "print(pretrain_status_df['cancerous_status'].value_counts(normalize=True))\n",
    "\n",
    "# Save the sample lists and detailed DataFrames\n",
    "pretrain_plain_df.to_csv('./data/pretrain_plain_data/pretrain_plain_samples.csv', index=False)\n",
    "pretrain_status_df.to_csv('./data/pretrain_status_data/pretrain_status_samples.csv', index=False)\n",
    "\n",
    "# Also save just the sample IDs for convenience\n",
    "pd.DataFrame(pretrain_plain_samples, columns=['depMapID']).to_csv('./data/pretrain_plain_data/pretrain_plain_sample_ids.csv', index=False)\n",
    "pd.DataFrame(pretrain_status_samples, columns=['depMapID']).to_csv('./data/pretrain_status_data/pretrain_status_sample_ids.csv', index=False)\n",
    "\n",
    "print(f\"\\n✅ Saved pretrain sample splits to:\")\n",
    "print(f\"- ./data/pretrain_plain_data/pretrain_plain_samples.csv ({len(pretrain_plain_samples)} samples)\")\n",
    "print(f\"- ./data/pretrain_status_data/pretrain_status_samples.csv ({len(pretrain_status_samples)} samples)\")\n",
    "\n",
    "# Verify the splits don't overlap\n",
    "overlap_check = set(pretrain_plain_samples) & set(pretrain_status_samples)\n",
    "print(f\"\\n🔍 Overlap check: {len(overlap_check)} overlapping samples (should be 0)\")\n",
    "\n",
    "# Show sample cancerous/non-cancerous ratios\n",
    "pretrain_plain_cancerous = pretrain_plain_df['cancerous_status'].value_counts()\n",
    "pretrain_status_cancerous = pretrain_status_df['cancerous_status'].value_counts()\n",
    "\n",
    "print(f\"\\n📈 Final Cancerous/Non-cancerous Distribution:\")\n",
    "print(f\"Pretrain Plain - Cancerous: {pretrain_plain_cancerous.get('cancerous', 0)}, Non-cancerous: {pretrain_plain_cancerous.get('non-cancerous', 0)}\")\n",
    "print(f\"Pretrain Status - Cancerous: {pretrain_status_cancerous.get('cancerous', 0)}, Non-cancerous: {pretrain_status_cancerous.get('non-cancerous', 0)}\")\n",
    "\n",
    "# Split the pretrain_status_samples into training and test sets according to the cancerous status\n",
    "def stratified_split_status_samples(pretrain_status_df, test_size=0.2, random_state=42):\n",
    "    \"\"\"\n",
    "    Split pretrain status samples into training and test sets while maintaining \n",
    "    the cancerous/non-cancerous ratio.\n",
    "    \n",
    "    Args:\n",
    "        pretrain_status_df: DataFrame with depMapID and cancerous_status columns\n",
    "        test_size: Proportion of the dataset to include in the test split (default 0.2)\n",
    "        random_state: Random seed for reproducibility\n",
    "    \n",
    "    Returns:\n",
    "        tuple: (train_samples, test_samples, train_df, test_df)\n",
    "    \"\"\"\n",
    "    # Use stratified split to maintain the ratio\n",
    "    train_df, test_df = train_test_split(\n",
    "        pretrain_status_df,\n",
    "        test_size=test_size,\n",
    "        stratify=pretrain_status_df['cancerous_status'],\n",
    "        random_state=random_state\n",
    "    )\n",
    "    \n",
    "    # Extract sample lists\n",
    "    train_samples = sorted(train_df['depMapID'].tolist())\n",
    "    test_samples = sorted(test_df['depMapID'].tolist())\n",
    "    \n",
    "    return train_samples, test_samples, train_df, test_df\n",
    "\n",
    "# Perform the split\n",
    "pretrain_status_train_samples, pretrain_status_test_samples, pretrain_status_train_df, pretrain_status_test_df = stratified_split_status_samples(\n",
    "    pretrain_status_df, \n",
    "    test_size=0.2, \n",
    "    random_state=42\n",
    ")\n",
    "\n",
    "# Display results for training and test sets for cancerous and non-cancerous counts\n",
    "print(f\"\\n📊 Pretrain Status Sample Split Summary:\"\n",
    "      f\"\\nTotal pretrain status samples: {len(pretrain_status_df)}\")\n",
    "print(f\"Training samples: {len(pretrain_status_train_samples)} ({len(pretrain_status_train_samples)/len(pretrain_status_df)*100:.1f}%)\")\n",
    "print(f\"Test samples: {len(pretrain_status_test_samples)} ({len(pretrain_status_test_samples)/len(pretrain_status_df)*100:.1f}%)\")\n",
    "print(f\"\\n📋 Training Sample Distribution:\"\n",
    "      f\"\\n{pretrain_status_train_df['cancerous_status'].value_counts()}\")\n",
    "print(pretrain_status_train_df['cancerous_status'].value_counts(normalize=True))\n",
    "print(f\"\\n📋 Test Sample Distribution:\"\n",
    "      f\"\\n{pretrain_status_test_df['cancerous_status'].value_counts()}\")\n",
    "\n",
    "# Save the training and test sample lists and detailed DataFrames\n",
    "pretrain_status_train_df.to_csv('./data/pretrain_status_data/pretrain_status_train_samples.csv', index=False)\n",
    "pretrain_status_test_df.to_csv('./data/pretrain_status_data/pretrain_status_test_samples.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 9.1.2 Target-CRISPR Samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The overlapped samples between dti_overlapped_samples and overlapped_crispr_rna_samples are:\n",
      "363\n",
      "['ACH-000001', 'ACH-000004', 'ACH-000007', 'ACH-000009', 'ACH-000012', 'ACH-000015', 'ACH-000018', 'ACH-000019', 'ACH-000021', 'ACH-000023', 'ACH-000024', 'ACH-000030', 'ACH-000035', 'ACH-000040', 'ACH-000042', 'ACH-000045', 'ACH-000048', 'ACH-000052', 'ACH-000054', 'ACH-000055', 'ACH-000060', 'ACH-000067', 'ACH-000070', 'ACH-000075', 'ACH-000087', 'ACH-000090', 'ACH-000092', 'ACH-000094', 'ACH-000096', 'ACH-000098', 'ACH-000100', 'ACH-000103', 'ACH-000108', 'ACH-000111', 'ACH-000113', 'ACH-000114', 'ACH-000115', 'ACH-000118', 'ACH-000132', 'ACH-000138', 'ACH-000139', 'ACH-000142', 'ACH-000146', 'ACH-000147', 'ACH-000148', 'ACH-000149', 'ACH-000153', 'ACH-000155', 'ACH-000159', 'ACH-000161', 'ACH-000168', 'ACH-000169', 'ACH-000176', 'ACH-000178', 'ACH-000183', 'ACH-000188', 'ACH-000191', 'ACH-000192', 'ACH-000207', 'ACH-000210', 'ACH-000211', 'ACH-000212', 'ACH-000213', 'ACH-000217', 'ACH-000219', 'ACH-000222', 'ACH-000223', 'ACH-000227', 'ACH-000231', 'ACH-000232', 'ACH-000235', 'ACH-000238', 'ACH-000243', 'ACH-000244', 'ACH-000247', 'ACH-000248', 'ACH-000249', 'ACH-000250', 'ACH-000257', 'ACH-000258', 'ACH-000259', 'ACH-000260', 'ACH-000263', 'ACH-000264', 'ACH-000265', 'ACH-000269', 'ACH-000271', 'ACH-000276', 'ACH-000277', 'ACH-000281', 'ACH-000288', 'ACH-000290', 'ACH-000291', 'ACH-000292', 'ACH-000294', 'ACH-000303', 'ACH-000304', 'ACH-000305', 'ACH-000308', 'ACH-000311', 'ACH-000315', 'ACH-000318', 'ACH-000319', 'ACH-000320', 'ACH-000322', 'ACH-000330', 'ACH-000334', 'ACH-000336', 'ACH-000341', 'ACH-000343', 'ACH-000348', 'ACH-000350', 'ACH-000352', 'ACH-000356', 'ACH-000361', 'ACH-000362', 'ACH-000363', 'ACH-000364', 'ACH-000365', 'ACH-000366', 'ACH-000371', 'ACH-000374', 'ACH-000375', 'ACH-000376', 'ACH-000378', 'ACH-000391', 'ACH-000395', 'ACH-000401', 'ACH-000403', 'ACH-000410', 'ACH-000411', 'ACH-000414', 'ACH-000416', 'ACH-000417', 'ACH-000420', 'ACH-000421', 'ACH-000424', 'ACH-000425', 'ACH-000427', 'ACH-000430', 'ACH-000431', 'ACH-000433', 'ACH-000434', 'ACH-000437', 'ACH-000438', 'ACH-000441', 'ACH-000448', 'ACH-000450', 'ACH-000452', 'ACH-000456', 'ACH-000463', 'ACH-000464', 'ACH-000470', 'ACH-000472', 'ACH-000473', 'ACH-000475', 'ACH-000480', 'ACH-000481', 'ACH-000484', 'ACH-000486', 'ACH-000488', 'ACH-000491', 'ACH-000493', 'ACH-000496', 'ACH-000505', 'ACH-000510', 'ACH-000515', 'ACH-000518', 'ACH-000521', 'ACH-000522', 'ACH-000524', 'ACH-000525', 'ACH-000527', 'ACH-000528', 'ACH-000532', 'ACH-000535', 'ACH-000544', 'ACH-000547', 'ACH-000548', 'ACH-000551', 'ACH-000552', 'ACH-000554', 'ACH-000558', 'ACH-000559', 'ACH-000563', 'ACH-000570', 'ACH-000571', 'ACH-000573', 'ACH-000579', 'ACH-000580', 'ACH-000585', 'ACH-000587', 'ACH-000589', 'ACH-000595', 'ACH-000596', 'ACH-000599', 'ACH-000601', 'ACH-000605', 'ACH-000607', 'ACH-000609', 'ACH-000614', 'ACH-000620', 'ACH-000621', 'ACH-000622', 'ACH-000624', 'ACH-000627', 'ACH-000631', 'ACH-000633', 'ACH-000637', 'ACH-000638', 'ACH-000639', 'ACH-000647', 'ACH-000648', 'ACH-000649', 'ACH-000651', 'ACH-000652', 'ACH-000653', 'ACH-000655', 'ACH-000657', 'ACH-000662', 'ACH-000663', 'ACH-000665', 'ACH-000666', 'ACH-000667', 'ACH-000668', 'ACH-000672', 'ACH-000677', 'ACH-000681', 'ACH-000684', 'ACH-000688', 'ACH-000693', 'ACH-000694', 'ACH-000696', 'ACH-000697', 'ACH-000698', 'ACH-000699', 'ACH-000704', 'ACH-000706', 'ACH-000711', 'ACH-000713', 'ACH-000717', 'ACH-000718', 'ACH-000719', 'ACH-000720', 'ACH-000722', 'ACH-000724', 'ACH-000725', 'ACH-000738', 'ACH-000740', 'ACH-000747', 'ACH-000748', 'ACH-000749', 'ACH-000756', 'ACH-000763', 'ACH-000766', 'ACH-000767', 'ACH-000768', 'ACH-000769', 'ACH-000775', 'ACH-000778', 'ACH-000781', 'ACH-000783', 'ACH-000784', 'ACH-000787', 'ACH-000788', 'ACH-000790', 'ACH-000792', 'ACH-000793', 'ACH-000794', 'ACH-000800', 'ACH-000804', 'ACH-000805', 'ACH-000809', 'ACH-000810', 'ACH-000811', 'ACH-000817', 'ACH-000819', 'ACH-000823', 'ACH-000824', 'ACH-000826', 'ACH-000827', 'ACH-000832', 'ACH-000838', 'ACH-000840', 'ACH-000841', 'ACH-000846', 'ACH-000847', 'ACH-000848', 'ACH-000849', 'ACH-000853', 'ACH-000855', 'ACH-000856', 'ACH-000858', 'ACH-000859', 'ACH-000860', 'ACH-000861', 'ACH-000863', 'ACH-000864', 'ACH-000865', 'ACH-000866', 'ACH-000867', 'ACH-000869', 'ACH-000874', 'ACH-000876', 'ACH-000878', 'ACH-000879', 'ACH-000880', 'ACH-000881', 'ACH-000882', 'ACH-000885', 'ACH-000886', 'ACH-000888', 'ACH-000890', 'ACH-000900', 'ACH-000906', 'ACH-000910', 'ACH-000911', 'ACH-000913', 'ACH-000915', 'ACH-000916', 'ACH-000917', 'ACH-000919', 'ACH-000922', 'ACH-000924', 'ACH-000926', 'ACH-000929', 'ACH-000932', 'ACH-000934', 'ACH-000935', 'ACH-000936', 'ACH-000938', 'ACH-000943', 'ACH-000945', 'ACH-000948', 'ACH-000950', 'ACH-000953', 'ACH-000956', 'ACH-000957', 'ACH-000958', 'ACH-000960', 'ACH-000962', 'ACH-000963', 'ACH-000965', 'ACH-000966', 'ACH-000969', 'ACH-000971', 'ACH-000974', 'ACH-000977', 'ACH-000978', 'ACH-000979', 'ACH-000980', 'ACH-000985', 'ACH-000986', 'ACH-000988', 'ACH-000991', 'ACH-000997', 'ACH-001039', 'ACH-001065', 'ACH-001075', 'ACH-001107', 'ACH-001190', 'ACH-001306', 'ACH-001307']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>CCLE_ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Site_Primary</th>\n",
       "      <th>Site_Subtype1</th>\n",
       "      <th>Site_Subtype2</th>\n",
       "      <th>Site_Subtype3</th>\n",
       "      <th>Histology</th>\n",
       "      <th>Hist_Subtype1</th>\n",
       "      <th>...</th>\n",
       "      <th>Growth.Medium</th>\n",
       "      <th>Supplements</th>\n",
       "      <th>Freezing.Medium</th>\n",
       "      <th>Doubling.Time.from.Vendor</th>\n",
       "      <th>Doubling.Time.Calculated.hrs</th>\n",
       "      <th>type</th>\n",
       "      <th>type_refined</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>mutRate</th>\n",
       "      <th>tcga_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>NIHOVCAR3_OVARY</td>\n",
       "      <td>NIH:OVCAR-3</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>ovary</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI+20%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ovary</td>\n",
       "      <td>ovary</td>\n",
       "      <td>Ovary:Carcinoma</td>\n",
       "      <td>83.806951</td>\n",
       "      <td>OV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>HEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>HEL</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_and_lymphoid_tissue</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>acute_myeloid_leukaemia</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI1640+10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10%DMSO</td>\n",
       "      <td>36 hrs</td>\n",
       "      <td>38.4</td>\n",
       "      <td>AML</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>98.185377</td>\n",
       "      <td>LAML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000007</td>\n",
       "      <td>LS513_LARGE_INTESTINE</td>\n",
       "      <td>LS513</td>\n",
       "      <td>primary</td>\n",
       "      <td>large_intestine</td>\n",
       "      <td>caecum</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>adenocarcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>112.349539</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000009</td>\n",
       "      <td>C2BBE1_LARGE_INTESTINE</td>\n",
       "      <td>C2BBe1</td>\n",
       "      <td>primary</td>\n",
       "      <td>large_intestine</td>\n",
       "      <td>colon</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>adenocarcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>DMEM +10%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5% DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>110.9</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>Colorectal:Carcinoma</td>\n",
       "      <td>115.886197</td>\n",
       "      <td>COAD/READ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000012</td>\n",
       "      <td>HCC827_LUNG</td>\n",
       "      <td>HCC827</td>\n",
       "      <td>primary</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>adenocarcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI-1640 10% FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>117.6</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>Lung:NSCLC_Adeno</td>\n",
       "      <td>82.926540</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>ACH-001075</td>\n",
       "      <td>NCIH292_LUNG</td>\n",
       "      <td>NCI-H292</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>lung</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>mucoepidermoid_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>Lung:Others</td>\n",
       "      <td>117.719142</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>ACH-001107</td>\n",
       "      <td>KP1N_PANCREAS</td>\n",
       "      <td>KP-1N</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>ductal_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PAAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360</th>\n",
       "      <td>ACH-001190</td>\n",
       "      <td>SKMEL2_SKIN</td>\n",
       "      <td>SK-MEL-2</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>skin</td>\n",
       "      <td>thigh</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>malignant_melanoma</td>\n",
       "      <td>NS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5%DMSO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98.8</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>Skin:Melanoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SKCM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>ACH-001306</td>\n",
       "      <td>8305C_THYROID</td>\n",
       "      <td>8305C</td>\n",
       "      <td>primary</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>anaplastic_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>RPMI1640+20%FBS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10%DMSO</td>\n",
       "      <td>54hrs</td>\n",
       "      <td>121.6</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>Thyroid:Carcinoma</td>\n",
       "      <td>200.908236</td>\n",
       "      <td>THCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362</th>\n",
       "      <td>ACH-001307</td>\n",
       "      <td>8505C_THYROID</td>\n",
       "      <td>8505C</td>\n",
       "      <td>primary</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>NS</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>anaplastic_carcinoma</td>\n",
       "      <td>...</td>\n",
       "      <td>Eagle's minimal essential medium with 10% feta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>Thyroid:Carcinoma</td>\n",
       "      <td>141.208560</td>\n",
       "      <td>THCA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>363 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       depMapID                                 CCLE_ID         Name  \\\n",
       "0    ACH-000001                         NIHOVCAR3_OVARY  NIH:OVCAR-3   \n",
       "1    ACH-000004  HEL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE          HEL   \n",
       "2    ACH-000007                   LS513_LARGE_INTESTINE        LS513   \n",
       "3    ACH-000009                  C2BBE1_LARGE_INTESTINE       C2BBe1   \n",
       "4    ACH-000012                             HCC827_LUNG       HCC827   \n",
       "..          ...                                     ...          ...   \n",
       "358  ACH-001075                            NCIH292_LUNG     NCI-H292   \n",
       "359  ACH-001107                           KP1N_PANCREAS        KP-1N   \n",
       "360  ACH-001190                             SKMEL2_SKIN     SK-MEL-2   \n",
       "361  ACH-001306                           8305C_THYROID        8305C   \n",
       "362  ACH-001307                           8505C_THYROID        8505C   \n",
       "\n",
       "      Pathology                        Site_Primary Site_Subtype1  \\\n",
       "0    metastasis                               ovary            NS   \n",
       "1       primary  haematopoietic_and_lymphoid_tissue            NS   \n",
       "2       primary                     large_intestine        caecum   \n",
       "3       primary                     large_intestine         colon   \n",
       "4       primary                                lung            NS   \n",
       "..          ...                                 ...           ...   \n",
       "358  metastasis                                lung            NS   \n",
       "359  metastasis                            pancreas            NS   \n",
       "360  metastasis                                skin         thigh   \n",
       "361     primary                             thyroid            NS   \n",
       "362     primary                             thyroid            NS   \n",
       "\n",
       "    Site_Subtype2 Site_Subtype3                Histology  \\\n",
       "0              NS            NS                carcinoma   \n",
       "1              NS            NS  haematopoietic_neoplasm   \n",
       "2              NS            NS                carcinoma   \n",
       "3              NS            NS                carcinoma   \n",
       "4              NS            NS                carcinoma   \n",
       "..            ...           ...                      ...   \n",
       "358            NS            NS                carcinoma   \n",
       "359            NS            NS                carcinoma   \n",
       "360            NS            NS       malignant_melanoma   \n",
       "361            NS            NS                carcinoma   \n",
       "362            NS            NS                carcinoma   \n",
       "\n",
       "                Hist_Subtype1  ...  \\\n",
       "0                          NS  ...   \n",
       "1     acute_myeloid_leukaemia  ...   \n",
       "2              adenocarcinoma  ...   \n",
       "3              adenocarcinoma  ...   \n",
       "4              adenocarcinoma  ...   \n",
       "..                        ...  ...   \n",
       "358  mucoepidermoid_carcinoma  ...   \n",
       "359          ductal_carcinoma  ...   \n",
       "360                        NS  ...   \n",
       "361      anaplastic_carcinoma  ...   \n",
       "362      anaplastic_carcinoma  ...   \n",
       "\n",
       "                                         Growth.Medium Supplements  \\\n",
       "0                                          RPMI+20%FBS         NaN   \n",
       "1                                      RPMI1640+10%FBS         NaN   \n",
       "2                                                  NaN         NaN   \n",
       "3                                         DMEM +10%FBS         NaN   \n",
       "4                                    RPMI-1640 10% FBS         NaN   \n",
       "..                                                 ...         ...   \n",
       "358                                                NaN         NaN   \n",
       "359                                                NaN         NaN   \n",
       "360                                                NaN         NaN   \n",
       "361                                    RPMI1640+20%FBS         NaN   \n",
       "362  Eagle's minimal essential medium with 10% feta...         NaN   \n",
       "\n",
       "    Freezing.Medium Doubling.Time.from.Vendor  Doubling.Time.Calculated.hrs  \\\n",
       "0            5%DMSO                       NaN                           NaN   \n",
       "1           10%DMSO                    36 hrs                          38.4   \n",
       "2               NaN                       NaN                           NaN   \n",
       "3           5% DMSO                       NaN                         110.9   \n",
       "4            5%DMSO                       NaN                         117.6   \n",
       "..              ...                       ...                           ...   \n",
       "358             NaN                       NaN                           NaN   \n",
       "359             NaN                       NaN                           NaN   \n",
       "360          5%DMSO                       NaN                          98.8   \n",
       "361         10%DMSO                     54hrs                         121.6   \n",
       "362             NaN                       NaN                           NaN   \n",
       "\n",
       "           type type_refined PATHOLOGIST_ANNOTATION     mutRate  tcga_code  \n",
       "0         ovary        ovary        Ovary:Carcinoma   83.806951         OV  \n",
       "1           AML          AML           Leukemia:AML   98.185377       LAML  \n",
       "2    colorectal   colorectal   Colorectal:Carcinoma  112.349539  COAD/READ  \n",
       "3    colorectal   colorectal   Colorectal:Carcinoma  115.886197  COAD/READ  \n",
       "4      lung_NSC     lung_NSC       Lung:NSCLC_Adeno   82.926540       LUAD  \n",
       "..          ...          ...                    ...         ...        ...  \n",
       "358    lung_NSC     lung_NSC            Lung:Others  117.719142    Unknown  \n",
       "359    pancreas     pancreas                    NaN         NaN       PAAD  \n",
       "360    melanoma     melanoma          Skin:Melanoma         NaN       SKCM  \n",
       "361     thyroid      thyroid      Thyroid:Carcinoma  200.908236       THCA  \n",
       "362     thyroid      thyroid      Thyroid:Carcinoma  141.208560       THCA  \n",
       "\n",
       "[363 rows x 33 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Check the overlapped samples between dti_overlapped_samples and overlapped_crispr_rna_samples\n",
    "overlapped_dti_crispr_rna_samples = sorted(list(set(dti_overlapped_samples) & set(overlapped_crispr_rna_samples)))\n",
    "print(\"The overlapped samples between dti_overlapped_samples and overlapped_crispr_rna_samples are:\")\n",
    "print(len(overlapped_dti_crispr_rna_samples))\n",
    "print(overlapped_dti_crispr_rna_samples)\n",
    "\n",
    "# Add the sample information to overlapped_dti_crispr_rna_samples with tcga_code, type, and PATHOLOGIST_ANNOTATION\n",
    "overlapped_dti_crispr_rna_samples_df = dti_overlapped_samples_df[dti_overlapped_samples_df['depMapID'].isin(overlapped_dti_crispr_rna_samples)].reset_index(drop=True)\n",
    "# fill NaN values in the columns with 'Unknown'\n",
    "overlapped_dti_crispr_rna_samples_df['tcga_code'] = overlapped_dti_crispr_rna_samples_df['tcga_code'].fillna('Unknown')\n",
    "display(overlapped_dti_crispr_rna_samples_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 384,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA Code Distribution and Split Strategy for Target Samples:\n",
      "----------------------------------------------------------------------\n",
      "TCGA ALL: 6 samples -> 1 test, 5 train\n",
      "TCGA BLCA: 8 samples -> 1 test, 7 train\n",
      "TCGA BRCA: 31 samples -> 6 test, 25 train\n",
      "TCGA COAD/READ: 26 samples -> 5 test, 21 train\n",
      "TCGA DLBC: 6 samples -> 1 test, 5 train\n",
      "TCGA ESCA: 18 samples -> 3 test, 15 train\n",
      "TCGA GBM: 18 samples -> 3 test, 15 train\n",
      "TCGA HNSC: 11 samples -> 2 test, 9 train\n",
      "TCGA KIRC: 8 samples -> 1 test, 7 train\n",
      "TCGA LAML: 9 samples -> 1 test, 8 train\n",
      "TCGA LCML: 1 sample -> train only\n",
      "TCGA LGG: 5 samples -> 1 test, 4 train\n",
      "TCGA LIHC: 8 samples -> 1 test, 7 train\n",
      "TCGA LUAD: 37 samples -> 7 test, 30 train\n",
      "TCGA LUSC: 10 samples -> 2 test, 8 train\n",
      "TCGA MB: 2 samples -> 1 test, 1 train\n",
      "TCGA MESO: 4 samples -> 1 test, 3 train\n",
      "TCGA MM: 7 samples -> 1 test, 6 train\n",
      "TCGA NB: 7 samples -> 1 test, 6 train\n",
      "TCGA OV: 19 samples -> 3 test, 16 train\n",
      "TCGA PAAD: 23 samples -> 4 test, 19 train\n",
      "TCGA PRAD: 5 samples -> 1 test, 4 train\n",
      "TCGA SARC: 15 samples -> 3 test, 12 train\n",
      "TCGA SCLC: 15 samples -> 3 test, 12 train\n",
      "TCGA SKCM: 19 samples -> 3 test, 16 train\n",
      "TCGA STAD: 12 samples -> 2 test, 10 train\n",
      "TCGA THCA: 4 samples -> 1 test, 3 train\n",
      "TCGA UCEC: 8 samples -> 1 test, 7 train\n",
      "TCGA UNABLE TO CLASSIFY: 8 samples -> 1 test, 7 train\n",
      "TCGA Unknown: 13 samples -> 2 test, 11 train\n",
      "----------------------------------------------------------------------\n",
      "Total train samples: 300\n",
      "Total test samples: 63\n",
      "Train ratio: 0.826\n",
      "Test ratio: 0.174\n",
      "\n",
      "📊 Target Sample Count Summary:\n",
      "Total samples in dataset: 363\n",
      "Training samples: 300\n",
      "Test samples: 63\n",
      "Training percentage: 82.6%\n",
      "Test percentage: 17.4%\n",
      "\n",
      "📋 Train set TCGA distribution:\n",
      "tcga_code\n",
      "ALL                    5\n",
      "BLCA                   7\n",
      "BRCA                  25\n",
      "COAD/READ             21\n",
      "DLBC                   5\n",
      "ESCA                  15\n",
      "GBM                   15\n",
      "HNSC                   9\n",
      "KIRC                   7\n",
      "LAML                   8\n",
      "LCML                   1\n",
      "LGG                    4\n",
      "LIHC                   7\n",
      "LUAD                  30\n",
      "LUSC                   8\n",
      "MB                     1\n",
      "MESO                   3\n",
      "MM                     6\n",
      "NB                     6\n",
      "OV                    16\n",
      "PAAD                  19\n",
      "PRAD                   4\n",
      "SARC                  12\n",
      "SCLC                  12\n",
      "SKCM                  16\n",
      "STAD                  10\n",
      "THCA                   3\n",
      "UCEC                   7\n",
      "UNABLE TO CLASSIFY     7\n",
      "Unknown               11\n",
      "Name: count, dtype: int64\n",
      "\n",
      "📋 Test set TCGA distribution:\n",
      "tcga_code\n",
      "ALL                   1\n",
      "BLCA                  1\n",
      "BRCA                  6\n",
      "COAD/READ             5\n",
      "DLBC                  1\n",
      "ESCA                  3\n",
      "GBM                   3\n",
      "HNSC                  2\n",
      "KIRC                  1\n",
      "LAML                  1\n",
      "LGG                   1\n",
      "LIHC                  1\n",
      "LUAD                  7\n",
      "LUSC                  2\n",
      "MB                    1\n",
      "MESO                  1\n",
      "MM                    1\n",
      "NB                    1\n",
      "OV                    3\n",
      "PAAD                  4\n",
      "PRAD                  1\n",
      "SARC                  3\n",
      "SCLC                  3\n",
      "SKCM                  3\n",
      "STAD                  2\n",
      "THCA                  1\n",
      "UCEC                  1\n",
      "UNABLE TO CLASSIFY    1\n",
      "Unknown               2\n",
      "Name: count, dtype: int64\n",
      "\n",
      "✅ Saved stratified train and test splits to:\n",
      "- ./data/TargetQA/train_samples.csv (300 samples)\n",
      "- ./data/TargetQA/test_samples.csv (63 samples)\n",
      "- ./data/TargetScreen/train_samples.csv (300 samples)\n",
      "- ./data/TargetScreen/test_samples.csv (63 samples)\n",
      "- Detailed sample information with TCGA annotations also saved\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "\n",
    "def stratified_tcga_split_target(df, test_ratio=0.2, random_state=42):\n",
    "    \"\"\"\n",
    "    Split samples by TCGA code with special handling for small groups:\n",
    "    - For TCGA codes with >=5 samples: split according to test_ratio (0.8/0.2)\n",
    "    - For TCGA codes with 2-4 samples: keep 1 sample for test\n",
    "    - For TCGA codes with 1 sample: keep that sample for train\n",
    "    \"\"\"\n",
    "    np.random.seed(random_state)\n",
    "    \n",
    "    train_samples = []\n",
    "    test_samples = []\n",
    "    \n",
    "    # Group by TCGA code\n",
    "    tcga_groups = df.groupby('tcga_code')\n",
    "    \n",
    "    print(\"TCGA Code Distribution and Split Strategy for Target Samples:\")\n",
    "    print(\"-\" * 70)\n",
    "    \n",
    "    for tcga_code, group in tcga_groups:\n",
    "        n_samples = len(group)\n",
    "        samples = group['depMapID'].tolist()\n",
    "        \n",
    "        # Shuffle samples within each TCGA group\n",
    "        np.random.shuffle(samples)\n",
    "        \n",
    "        if n_samples == 1:\n",
    "            # Single sample: keep for train\n",
    "            train_samples.extend(samples)\n",
    "            test_samples_for_group = []\n",
    "            strategy = \"1 sample -> train only\"\n",
    "            \n",
    "        elif n_samples < 5:\n",
    "            # 2-4 samples: keep 1 for test, rest for train\n",
    "            test_samples_for_group = samples[:1]\n",
    "            train_samples_for_group = samples[1:]\n",
    "            train_samples.extend(train_samples_for_group)\n",
    "            test_samples.extend(test_samples_for_group)\n",
    "            strategy = f\"{n_samples} samples -> 1 test, {n_samples-1} train\"\n",
    "            \n",
    "        else:\n",
    "            # 5+ samples: use the specified ratio\n",
    "            n_test = max(1, int(n_samples * test_ratio))\n",
    "            test_samples_for_group = samples[:n_test]\n",
    "            train_samples_for_group = samples[n_test:]\n",
    "            train_samples.extend(train_samples_for_group)\n",
    "            test_samples.extend(test_samples_for_group)\n",
    "            strategy = f\"{n_samples} samples -> {n_test} test, {len(train_samples_for_group)} train\"\n",
    "        \n",
    "        print(f\"TCGA {tcga_code}: {strategy}\")\n",
    "    \n",
    "    print(\"-\" * 70)\n",
    "    print(f\"Total train samples: {len(train_samples)}\")\n",
    "    print(f\"Total test samples: {len(test_samples)}\")\n",
    "    print(f\"Train ratio: {len(train_samples)/(len(train_samples)+len(test_samples)):.3f}\")\n",
    "    print(f\"Test ratio: {len(test_samples)/(len(train_samples)+len(test_samples)):.3f}\")\n",
    "    \n",
    "    return train_samples, test_samples\n",
    "\n",
    "# Prepare the overlapped samples with TCGA annotation\n",
    "overlapped_dti_crispr_rna_samples_df_annotated = overlapped_dti_crispr_rna_samples_df.copy()\n",
    "# Fill NaN TCGA codes with 'Unknown' if any exist\n",
    "overlapped_dti_crispr_rna_samples_df_annotated['tcga_code'] = overlapped_dti_crispr_rna_samples_df_annotated['tcga_code'].fillna('Unknown')\n",
    "\n",
    "# Perform stratified split based on TCGA codes\n",
    "target_crispr_train_samples, target_crispr_test_samples = stratified_tcga_split_target(\n",
    "    overlapped_dti_crispr_rna_samples_df_annotated, \n",
    "    test_ratio=0.2, \n",
    "    random_state=42\n",
    ")\n",
    "\n",
    "# Count the samples\n",
    "print(f\"\\n📊 Target Sample Count Summary:\")\n",
    "print(f\"Total samples in dataset: {len(overlapped_dti_crispr_rna_samples_df_annotated)}\")\n",
    "print(f\"Training samples: {len(target_crispr_train_samples)}\")\n",
    "print(f\"Test samples: {len(target_crispr_test_samples)}\")\n",
    "print(f\"Training percentage: {len(target_crispr_train_samples)/len(overlapped_dti_crispr_rna_samples_df_annotated)*100:.1f}%\")\n",
    "print(f\"Test percentage: {len(target_crispr_test_samples)/len(overlapped_dti_crispr_rna_samples_df_annotated)*100:.1f}%\")\n",
    "\n",
    "# Create directories\n",
    "if not os.path.exists('./data/TargetQA'):\n",
    "    os.makedirs('./data/TargetQA')\n",
    "if not os.path.exists('./data/TargetScreen'):\n",
    "    os.makedirs('./data/TargetScreen')\n",
    "\n",
    "# Sort the samples before saving\n",
    "train_samples_sorted = sorted(target_crispr_train_samples)\n",
    "test_samples_sorted = sorted(target_crispr_test_samples)\n",
    "\n",
    "# Save the sorted train and test samples to CSV files\n",
    "pd.DataFrame(train_samples_sorted, columns=['depMapID']).to_csv('./data/TargetQA/train_samples.csv', index=False)\n",
    "pd.DataFrame(test_samples_sorted, columns=['depMapID']).to_csv('./data/TargetQA/test_samples.csv', index=False)\n",
    "pd.DataFrame(train_samples_sorted, columns=['depMapID']).to_csv('./data/TargetScreen/train_samples.csv', index=False)\n",
    "pd.DataFrame(test_samples_sorted, columns=['depMapID']).to_csv('./data/TargetScreen/test_samples.csv', index=False)\n",
    "\n",
    "# Create detailed split information DataFrames\n",
    "train_target_samples_df = overlapped_dti_crispr_rna_samples_df_annotated[\n",
    "    overlapped_dti_crispr_rna_samples_df_annotated['depMapID'].isin(target_crispr_train_samples)\n",
    "].reset_index(drop=True)\n",
    "\n",
    "test_target_samples_df = overlapped_dti_crispr_rna_samples_df_annotated[\n",
    "    overlapped_dti_crispr_rna_samples_df_annotated['depMapID'].isin(target_crispr_test_samples)\n",
    "].reset_index(drop=True)\n",
    "\n",
    "# Display the TCGA distributions in train and test sets\n",
    "print(\"\\n📋 Train set TCGA distribution:\")\n",
    "print(train_target_samples_df['tcga_code'].value_counts().sort_index())\n",
    "print(\"\\n📋 Test set TCGA distribution:\")\n",
    "print(test_target_samples_df['tcga_code'].value_counts().sort_index())\n",
    "\n",
    "# Save detailed sample information\n",
    "train_target_samples_df.to_csv('./data/TargetQA/train_samples_detailed.csv', index=False)\n",
    "test_target_samples_df.to_csv('./data/TargetQA/test_samples_detailed.csv', index=False)\n",
    "train_target_samples_df.to_csv('./data/TargetScreen/train_samples_detailed.csv', index=False)\n",
    "test_target_samples_df.to_csv('./data/TargetScreen/test_samples_detailed.csv', index=False)\n",
    "\n",
    "print(f\"\\n✅ Saved stratified train and test splits to:\")\n",
    "print(f\"- ./data/TargetQA/train_samples.csv ({len(train_samples_sorted)} samples)\")\n",
    "print(f\"- ./data/TargetQA/test_samples.csv ({len(test_samples_sorted)} samples)\")\n",
    "print(f\"- ./data/TargetScreen/train_samples.csv ({len(train_samples_sorted)} samples)\")\n",
    "print(f\"- ./data/TargetScreen/test_samples.csv ({len(test_samples_sorted)} samples)\")\n",
    "print(f\"- Detailed sample information with TCGA annotations also saved\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 9.1.3 Drug Samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 385,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The remaining samples in dti_overlapped_samples after removing overlapped_dti_crispr_rna_samples are:\n",
      "286\n"
     ]
    }
   ],
   "source": [
    "# Keep the left samples dti_overlapped_samples - overlapped_dti_crispr_rna_samples\n",
    "remaining_dti_samples = sorted(list(set(dti_overlapped_samples) - set(overlapped_dti_crispr_rna_samples)))\n",
    "print(\"The remaining samples in dti_overlapped_samples after removing overlapped_dti_crispr_rna_samples are:\")\n",
    "print(len(remaining_dti_samples))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 394,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The remaining samples in dti_overlapped_samples after removing overlapped_dti_crispr_rna_samples are:\n",
      "286\n",
      "TCGA Code Distribution and Split Strategy for DTI Drug Samples:\n",
      "----------------------------------------------------------------------\n",
      "TCGA ALL: 16 samples -> 3 test, 13 train\n",
      "TCGA BLCA: 9 samples -> 1 test, 8 train\n",
      "TCGA BRCA: 15 samples -> 3 test, 12 train\n",
      "TCGA CLL: 3 samples -> 1 test, 2 train\n",
      "TCGA COAD/READ: 17 samples -> 3 test, 14 train\n",
      "TCGA DLBC: 20 samples -> 4 test, 16 train\n",
      "TCGA ESCA: 4 samples -> 1 test, 3 train\n",
      "TCGA GBM: 7 samples -> 1 test, 6 train\n",
      "TCGA HNSC: 5 samples -> 1 test, 4 train\n",
      "TCGA KIRC: 7 samples -> 1 test, 6 train\n",
      "TCGA LAML: 14 samples -> 2 test, 12 train\n",
      "TCGA LCML: 8 samples -> 1 test, 7 train\n",
      "TCGA LGG: 6 samples -> 1 test, 5 train\n",
      "TCGA LIHC: 9 samples -> 1 test, 8 train\n",
      "TCGA LUAD: 21 samples -> 4 test, 17 train\n",
      "TCGA LUSC: 5 samples -> 1 test, 4 train\n",
      "TCGA MB: 1 sample -> train only\n",
      "TCGA MESO: 2 samples -> 1 test, 1 train\n",
      "TCGA MM: 8 samples -> 1 test, 7 train\n",
      "TCGA NB: 5 samples -> 1 test, 4 train\n",
      "TCGA OV: 9 samples -> 1 test, 8 train\n",
      "TCGA PAAD: 5 samples -> 1 test, 4 train\n",
      "TCGA SARC: 10 samples -> 2 test, 8 train\n",
      "TCGA SCLC: 26 samples -> 5 test, 21 train\n",
      "TCGA SKCM: 16 samples -> 3 test, 13 train\n",
      "TCGA STAD: 10 samples -> 2 test, 8 train\n",
      "TCGA THCA: 5 samples -> 1 test, 4 train\n",
      "TCGA UCEC: 2 samples -> 1 test, 1 train\n",
      "TCGA UNABLE TO CLASSIFY: 2 samples -> 1 test, 1 train\n",
      "TCGA Unknown: 19 samples -> 3 test, 16 train\n",
      "----------------------------------------------------------------------\n",
      "Total train samples: 234\n",
      "Total test samples: 52\n",
      "Train ratio: 0.818\n",
      "Test ratio: 0.182\n",
      "\n",
      "📊 DTI Drug Sample Count Summary:\n",
      "Total remaining DTI samples: 286\n",
      "Training samples: 234\n",
      "Test samples: 52\n",
      "Training percentage: 81.8%\n",
      "Test percentage: 18.2%\n",
      "\n",
      "📋 DTI Drug Train set TCGA distribution:\n",
      "tcga_code\n",
      "ALL                   13\n",
      "BLCA                   8\n",
      "BRCA                  12\n",
      "CLL                    2\n",
      "COAD/READ             14\n",
      "DLBC                  16\n",
      "ESCA                   3\n",
      "GBM                    6\n",
      "HNSC                   4\n",
      "KIRC                   6\n",
      "LAML                  12\n",
      "LCML                   7\n",
      "LGG                    5\n",
      "LIHC                   8\n",
      "LUAD                  17\n",
      "LUSC                   4\n",
      "MB                     1\n",
      "MESO                   1\n",
      "MM                     7\n",
      "NB                     4\n",
      "OV                     8\n",
      "PAAD                   4\n",
      "SARC                   8\n",
      "SCLC                  21\n",
      "SKCM                  13\n",
      "STAD                   8\n",
      "THCA                   4\n",
      "UCEC                   1\n",
      "UNABLE TO CLASSIFY     1\n",
      "Unknown               16\n",
      "Name: count, dtype: int64\n",
      "\n",
      "📋 DTI Drug Test set TCGA distribution:\n",
      "tcga_code\n",
      "ALL                   3\n",
      "BLCA                  1\n",
      "BRCA                  3\n",
      "CLL                   1\n",
      "COAD/READ             3\n",
      "DLBC                  4\n",
      "ESCA                  1\n",
      "GBM                   1\n",
      "HNSC                  1\n",
      "KIRC                  1\n",
      "LAML                  2\n",
      "LCML                  1\n",
      "LGG                   1\n",
      "LIHC                  1\n",
      "LUAD                  4\n",
      "LUSC                  1\n",
      "MESO                  1\n",
      "MM                    1\n",
      "NB                    1\n",
      "OV                    1\n",
      "PAAD                  1\n",
      "SARC                  2\n",
      "SCLC                  5\n",
      "SKCM                  3\n",
      "STAD                  2\n",
      "THCA                  1\n",
      "UCEC                  1\n",
      "UNABLE TO CLASSIFY    1\n",
      "Unknown               3\n",
      "Name: count, dtype: int64\n",
      "\n",
      "✅ Saved stratified DTI drug train and test splits to:\n",
      "- ./data/DrugQA/train_samples.csv (234 samples)\n",
      "- ./data/DrugQA/test_samples.csv (52 samples)\n",
      "- ./data/DrugScreen/train_samples.csv (234 samples)\n",
      "- ./data/DrugScreen/test_samples.csv (52 samples)\n",
      "- Detailed sample information with TCGA annotations also saved\n",
      "\n",
      "📈 Complete Sample Split Summary:\n",
      "Original DTI overlapped samples: 649\n",
      "Target identification samples (CRISPR/RNA): 363\n",
      "Drug screening samples (remaining): 286\n",
      "  - DTI Drug Train: 234\n",
      "  - DTI Drug Test: 52\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "\n",
    "# Get the remaining DTI samples\n",
    "remaining_dti_samples = sorted(list(set(dti_overlapped_samples) - set(overlapped_dti_crispr_rna_samples)))\n",
    "print(\"The remaining samples in dti_overlapped_samples after removing overlapped_dti_crispr_rna_samples are:\")\n",
    "print(len(remaining_dti_samples))\n",
    "\n",
    "# Create DataFrame with remaining samples and their TCGA annotations\n",
    "remaining_dti_samples_df = dti_overlapped_samples_df[\n",
    "    dti_overlapped_samples_df['depMapID'].isin(remaining_dti_samples)\n",
    "].reset_index(drop=True)\n",
    "\n",
    "# Fill NaN TCGA codes with 'Unknown' if any exist\n",
    "remaining_dti_samples_df['tcga_code'] = remaining_dti_samples_df['tcga_code'].fillna('Unknown')\n",
    "\n",
    "def stratified_tcga_split_dti(df, test_ratio=0.2, random_state=42):\n",
    "    \"\"\"\n",
    "    Split samples by TCGA code with special handling for small groups:\n",
    "    - For TCGA codes with >=5 samples: split according to test_ratio (0.8/0.2)\n",
    "    - For TCGA codes with 2-4 samples: keep 1 sample for test\n",
    "    - For TCGA codes with 1 sample: keep that sample for train\n",
    "    \"\"\"\n",
    "    np.random.seed(random_state)\n",
    "    \n",
    "    train_samples = []\n",
    "    test_samples = []\n",
    "    \n",
    "    # Group by TCGA code\n",
    "    tcga_groups = df.groupby('tcga_code')\n",
    "    \n",
    "    print(\"TCGA Code Distribution and Split Strategy for DTI Drug Samples:\")\n",
    "    print(\"-\" * 70)\n",
    "    \n",
    "    for tcga_code, group in tcga_groups:\n",
    "        n_samples = len(group)\n",
    "        samples = group['depMapID'].tolist()\n",
    "        \n",
    "        # Shuffle samples within each TCGA group\n",
    "        np.random.shuffle(samples)\n",
    "        \n",
    "        if n_samples == 1:\n",
    "            # Single sample: keep for train\n",
    "            train_samples.extend(samples)\n",
    "            test_samples_for_group = []\n",
    "            strategy = \"1 sample -> train only\"\n",
    "            \n",
    "        elif n_samples < 5:\n",
    "            # 2-4 samples: keep 1 for test, rest for train\n",
    "            test_samples_for_group = samples[:1]\n",
    "            train_samples_for_group = samples[1:]\n",
    "            train_samples.extend(train_samples_for_group)\n",
    "            test_samples.extend(test_samples_for_group)\n",
    "            strategy = f\"{n_samples} samples -> 1 test, {n_samples-1} train\"\n",
    "            \n",
    "        else:\n",
    "            # 5+ samples: use the specified ratio\n",
    "            n_test = max(1, int(n_samples * test_ratio))\n",
    "            test_samples_for_group = samples[:n_test]\n",
    "            train_samples_for_group = samples[n_test:]\n",
    "            train_samples.extend(train_samples_for_group)\n",
    "            test_samples.extend(test_samples_for_group)\n",
    "            strategy = f\"{n_samples} samples -> {n_test} test, {len(train_samples_for_group)} train\"\n",
    "        \n",
    "        print(f\"TCGA {tcga_code}: {strategy}\")\n",
    "    \n",
    "    print(\"-\" * 70)\n",
    "    print(f\"Total train samples: {len(train_samples)}\")\n",
    "    print(f\"Total test samples: {len(test_samples)}\")\n",
    "    print(f\"Train ratio: {len(train_samples)/(len(train_samples)+len(test_samples)):.3f}\")\n",
    "    print(f\"Test ratio: {len(test_samples)/(len(train_samples)+len(test_samples)):.3f}\")\n",
    "    \n",
    "    return train_samples, test_samples\n",
    "\n",
    "# Perform stratified split based on TCGA codes for remaining DTI samples\n",
    "dti_train_samples, dti_test_samples = stratified_tcga_split_dti(\n",
    "    remaining_dti_samples_df, \n",
    "    test_ratio=0.2, \n",
    "    random_state=42\n",
    ")\n",
    "\n",
    "# Count the samples\n",
    "print(f\"\\n📊 DTI Drug Sample Count Summary:\")\n",
    "print(f\"Total remaining DTI samples: {len(remaining_dti_samples_df)}\")\n",
    "print(f\"Training samples: {len(dti_train_samples)}\")\n",
    "print(f\"Test samples: {len(dti_test_samples)}\")\n",
    "print(f\"Training percentage: {len(dti_train_samples)/len(remaining_dti_samples_df)*100:.1f}%\")\n",
    "print(f\"Test percentage: {len(dti_test_samples)/len(remaining_dti_samples_df)*100:.1f}%\")\n",
    "\n",
    "# Create directories\n",
    "if not os.path.exists('./data/DrugQA'):\n",
    "    os.makedirs('./data/DrugQA')\n",
    "if not os.path.exists('./data/DrugScreen'):\n",
    "    os.makedirs('./data/DrugScreen')\n",
    "\n",
    "# Sort the samples before saving\n",
    "dti_train_samples_sorted = sorted(dti_train_samples)\n",
    "dti_test_samples_sorted = sorted(dti_test_samples)\n",
    "\n",
    "remaining_dti_samples_df.to_csv('./data/DrugQA/remaining_dti_samples.csv', index=False)\n",
    "remaining_dti_samples_df.to_csv('./data/DrugScreen/remaining_dti_samples.csv', index=False)\n",
    "# Save the sorted train and test samples to CSV files\n",
    "pd.DataFrame(dti_train_samples_sorted, columns=['depMapID']).to_csv('./data/DrugQA/train_samples.csv', index=False)\n",
    "pd.DataFrame(dti_test_samples_sorted, columns=['depMapID']).to_csv('./data/DrugQA/test_samples.csv', index=False)\n",
    "pd.DataFrame(dti_train_samples_sorted, columns=['depMapID']).to_csv('./data/DrugScreen/train_samples.csv', index=False)\n",
    "pd.DataFrame(dti_test_samples_sorted, columns=['depMapID']).to_csv('./data/DrugScreen/test_samples.csv', index=False)\n",
    "\n",
    "# Create detailed split information DataFrames\n",
    "train_dti_samples_df = remaining_dti_samples_df[\n",
    "    remaining_dti_samples_df['depMapID'].isin(dti_train_samples)\n",
    "].reset_index(drop=True)\n",
    "\n",
    "test_dti_samples_df = remaining_dti_samples_df[\n",
    "    remaining_dti_samples_df['depMapID'].isin(dti_test_samples)\n",
    "].reset_index(drop=True)\n",
    "\n",
    "# Display the TCGA distributions in train and test sets\n",
    "print(\"\\n📋 DTI Drug Train set TCGA distribution:\")\n",
    "print(train_dti_samples_df['tcga_code'].value_counts().sort_index())\n",
    "print(\"\\n📋 DTI Drug Test set TCGA distribution:\")\n",
    "print(test_dti_samples_df['tcga_code'].value_counts().sort_index())\n",
    "\n",
    "# Save detailed sample information\n",
    "train_dti_samples_df.to_csv('./data/DrugQA/train_samples_detailed.csv', index=False)\n",
    "test_dti_samples_df.to_csv('./data/DrugQA/test_samples_detailed.csv', index=False)\n",
    "train_dti_samples_df.to_csv('./data/DrugScreen/train_samples_detailed.csv', index=False)\n",
    "test_dti_samples_df.to_csv('./data/DrugScreen/test_samples_detailed.csv', index=False)\n",
    "\n",
    "print(f\"\\n✅ Saved stratified DTI drug train and test splits to:\")\n",
    "print(f\"- ./data/DrugQA/train_samples.csv ({len(dti_train_samples_sorted)} samples)\")\n",
    "print(f\"- ./data/DrugQA/test_samples.csv ({len(dti_test_samples_sorted)} samples)\")\n",
    "print(f\"- ./data/DrugScreen/train_samples.csv ({len(dti_train_samples_sorted)} samples)\")\n",
    "print(f\"- ./data/DrugScreen/test_samples.csv ({len(dti_test_samples_sorted)} samples)\")\n",
    "print(f\"- Detailed sample information with TCGA annotations also saved\")\n",
    "\n",
    "# Summary of all splits\n",
    "print(f\"\\n📈 Complete Sample Split Summary:\")\n",
    "print(f\"Original DTI overlapped samples: {len(dti_overlapped_samples)}\")\n",
    "print(f\"Target identification samples (CRISPR/RNA): {len(overlapped_dti_crispr_rna_samples)}\")\n",
    "print(f\"Drug screening samples (remaining): {len(remaining_dti_samples)}\")\n",
    "print(f\"  - DTI Drug Train: {len(dti_train_samples)}\")\n",
    "print(f\"  - DTI Drug Test: {len(dti_test_samples)}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 9.2 Data Integration"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 9.2.0 Data Preparation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 387,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000698</th>\n",
       "      <th>ACH-000489</th>\n",
       "      <th>ACH-000707</th>\n",
       "      <th>ACH-000509</th>\n",
       "      <th>ACH-000522</th>\n",
       "      <th>ACH-000613</th>\n",
       "      <th>ACH-000557</th>\n",
       "      <th>ACH-000614</th>\n",
       "      <th>ACH-000523</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000993</th>\n",
       "      <th>ACH-000544</th>\n",
       "      <th>MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>ACH-000861</th>\n",
       "      <th>ACH-000706</th>\n",
       "      <th>ACH-000428</th>\n",
       "      <th>ACH-000655</th>\n",
       "      <th>ACH-000273</th>\n",
       "      <th>ACH-000504</th>\n",
       "      <th>ACH-000825</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.51504</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.98288</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.55295</td>\n",
       "      <td>0.8701</td>\n",
       "      <td>0.90770</td>\n",
       "      <td>0.86540</td>\n",
       "      <td>0.71970</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.89135</td>\n",
       "      <td>0.28806</td>\n",
       "      <td>0.50752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.01768</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.09933</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04551</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00385</td>\n",
       "      <td>0.00499</td>\n",
       "      <td>0.01187</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00306</td>\n",
       "      <td>0.00160</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.00481</td>\n",
       "      <td>0.85522</td>\n",
       "      <td>0.96297</td>\n",
       "      <td>0.68518</td>\n",
       "      <td>0.11157</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.03094</td>\n",
       "      <td>0.00935</td>\n",
       "      <td>0.00457</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00704</td>\n",
       "      <td>0.00862</td>\n",
       "      <td>0.91237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_PM86234</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_PM86235</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_PM86236</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 844 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000698  ACH-000489  ACH-000707  ACH-000509  \\\n",
       "0               BMGC_PM00001     0.51504     0.50000     0.98288     1.00000   \n",
       "1               BMGC_PM00002     0.00000     0.00000     0.00000     0.00000   \n",
       "2               BMGC_PM00003     0.01768     0.00000     0.09933     0.00000   \n",
       "3               BMGC_PM00004     0.00481     0.85522     0.96297     0.68518   \n",
       "4               BMGC_PM00005     0.00000     0.00000     0.00000     0.00000   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_PM86234     0.00000     0.00000     0.00000     0.00000   \n",
       "86234           BMGC_PM86235     0.00000     0.00000     0.00000     0.00000   \n",
       "86235           BMGC_PM86236     0.00000     0.00000     0.00000     0.00000   \n",
       "86236           BMGC_PM86237     0.00000     0.00000     0.00000     0.00000   \n",
       "86237           BMGC_PM86238     0.00000     0.00000     0.00000     0.00000   \n",
       "\n",
       "       ACH-000522  ACH-000613  ACH-000557  ACH-000614  ACH-000523  ...  \\\n",
       "0         0.55295      0.8701     0.90770     0.86540     0.71970  ...   \n",
       "1         0.00000      0.0000     0.00000     0.00000     0.00000  ...   \n",
       "2         0.04551      0.0000     0.00385     0.00499     0.01187  ...   \n",
       "3         0.11157      0.0000     0.03094     0.00935     0.00457  ...   \n",
       "4         0.00000      0.0000     0.00000     0.00000     0.00000  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233     0.00000      0.0000     0.00000     0.00000     0.00000  ...   \n",
       "86234     0.00000      0.0000     0.00000     0.00000     0.00000  ...   \n",
       "86235     0.00000      0.0000     0.00000     0.00000     0.00000  ...   \n",
       "86236     0.00000      0.0000     0.00000     0.00000     0.00000  ...   \n",
       "86237     0.00000      0.0000     0.00000     0.00000     0.00000  ...   \n",
       "\n",
       "       ACH-000993  ACH-000544  MOLT3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0             0.0         0.0                                       0.0   \n",
       "1             0.0         0.0                                       0.0   \n",
       "2             0.0         0.0                                       0.0   \n",
       "3             0.0         0.0                                       0.0   \n",
       "4             0.0         0.0                                       0.0   \n",
       "...           ...         ...                                       ...   \n",
       "86233         0.0         0.0                                       0.0   \n",
       "86234         0.0         0.0                                       0.0   \n",
       "86235         0.0         0.0                                       0.0   \n",
       "86236         0.0         0.0                                       0.0   \n",
       "86237         0.0         0.0                                       0.0   \n",
       "\n",
       "       ACH-000861  ACH-000706  ACH-000428  ACH-000655  ACH-000273  ACH-000504  \\\n",
       "0             0.0         0.0         0.0     0.50000     0.89135     0.28806   \n",
       "1             0.0         0.0         0.0     0.00000     0.00000     0.00000   \n",
       "2             0.0         0.0         0.0     0.00306     0.00160     0.00000   \n",
       "3             0.0         0.0         0.0     0.00000     0.00704     0.00862   \n",
       "4             0.0         0.0         0.0     0.00000     0.00000     0.00000   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233         0.0         0.0         0.0     0.00000     0.00000     0.00000   \n",
       "86234         0.0         0.0         0.0     0.00000     0.00000     0.00000   \n",
       "86235         0.0         0.0         0.0     0.00000     0.00000     0.00000   \n",
       "86236         0.0         0.0         0.0     0.00000     0.00000     0.00000   \n",
       "86237         0.0         0.0         0.0     0.00000     0.00000     0.00000   \n",
       "\n",
       "       ACH-000825  \n",
       "0         0.50752  \n",
       "1         0.00000  \n",
       "2         0.00375  \n",
       "3         0.91237  \n",
       "4         0.00000  \n",
       "...           ...  \n",
       "86233     0.00000  \n",
       "86234     0.00000  \n",
       "86235     0.00000  \n",
       "86236     0.00000  \n",
       "86237     0.00000  \n",
       "\n",
       "[86238 rows x 844 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000849</th>\n",
       "      <th>ACH-000441</th>\n",
       "      <th>ACH-000248</th>\n",
       "      <th>ACH-000684</th>\n",
       "      <th>ACH-000856</th>\n",
       "      <th>ACH-000348</th>\n",
       "      <th>ACH-000062</th>\n",
       "      <th>ACH-000650</th>\n",
       "      <th>ACH-000484</th>\n",
       "      <th>...</th>\n",
       "      <th>NCIH2030_LUNG_TenPx18</th>\n",
       "      <th>22RV1_PROSTATE_TenPx24</th>\n",
       "      <th>A172_CENTRAL_NERVOUS_SYSTEM_TenPx24</th>\n",
       "      <th>BT20_BREAST_TenPx24</th>\n",
       "      <th>CALU6_LUNG_TenPx24</th>\n",
       "      <th>FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24</th>\n",
       "      <th>KP4_PANCREAS_TenPx24</th>\n",
       "      <th>MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "      <th>OVCAR8_OVARY_TenPx24</th>\n",
       "      <th>THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 373 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000849  ACH-000441  ACH-000248  ACH-000684  \\\n",
       "0               BMGC_PT000001         0.0         0.0         0.0         0.0   \n",
       "1               BMGC_PT000002         0.0         0.0         0.0         0.0   \n",
       "2               BMGC_PT000003         0.0         0.0         0.0         0.0   \n",
       "3               BMGC_PT000004         0.0         0.0         0.0         0.0   \n",
       "4               BMGC_PT000005         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "121414          BMGC_PT121415         0.0         0.0         0.0         0.0   \n",
       "121415          BMGC_PT121416         0.0         0.0         0.0         0.0   \n",
       "121416          BMGC_PT121417         0.0         0.0         0.0         0.0   \n",
       "121417          BMGC_PT121418         0.0         0.0         0.0         0.0   \n",
       "121418          BMGC_PT121419         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-000856  ACH-000348  ACH-000062  ACH-000650  ACH-000484  ...  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "1              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "2              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "3              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "4              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121415         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121416         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121417         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121418         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "\n",
       "        NCIH2030_LUNG_TenPx18  22RV1_PROSTATE_TenPx24  \\\n",
       "0                         0.0                     0.0   \n",
       "1                         0.0                     0.0   \n",
       "2                         0.0                     0.0   \n",
       "3                         0.0                     0.0   \n",
       "4                         0.0                     0.0   \n",
       "...                       ...                     ...   \n",
       "121414                    0.0                     0.0   \n",
       "121415                    0.0                     0.0   \n",
       "121416                    0.0                     0.0   \n",
       "121417                    0.0                     0.0   \n",
       "121418                    0.0                     0.0   \n",
       "\n",
       "        A172_CENTRAL_NERVOUS_SYSTEM_TenPx24  BT20_BREAST_TenPx24  \\\n",
       "0                                       0.0                  0.0   \n",
       "1                                       0.0                  0.0   \n",
       "2                                       0.0                  0.0   \n",
       "3                                       0.0                  0.0   \n",
       "4                                       0.0                  0.0   \n",
       "...                                     ...                  ...   \n",
       "121414                                  0.0                  0.0   \n",
       "121415                                  0.0                  0.0   \n",
       "121416                                  0.0                  0.0   \n",
       "121417                                  0.0                  0.0   \n",
       "121418                                  0.0                  0.0   \n",
       "\n",
       "        CALU6_LUNG_TenPx24  FADU_UPPER_AERODIGESTIVE_TRACT_TenPx24  \\\n",
       "0                      0.0                                     0.0   \n",
       "1                      0.0                                     0.0   \n",
       "2                      0.0                                     0.0   \n",
       "3                      0.0                                     0.0   \n",
       "4                      0.0                                     0.0   \n",
       "...                    ...                                     ...   \n",
       "121414                 0.0                                     0.0   \n",
       "121415                 0.0                                     0.0   \n",
       "121416                 0.0                                     0.0   \n",
       "121417                 0.0                                     0.0   \n",
       "121418                 0.0                                     0.0   \n",
       "\n",
       "        KP4_PANCREAS_TenPx24  \\\n",
       "0                        0.0   \n",
       "1                        0.0   \n",
       "2                        0.0   \n",
       "3                        0.0   \n",
       "4                        0.0   \n",
       "...                      ...   \n",
       "121414                   0.0   \n",
       "121415                   0.0   \n",
       "121416                   0.0   \n",
       "121417                   0.0   \n",
       "121418                   0.0   \n",
       "\n",
       "        MONOMAC6_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \\\n",
       "0                                                     0.0     \n",
       "1                                                     0.0     \n",
       "2                                                     0.0     \n",
       "3                                                     0.0     \n",
       "4                                                     0.0     \n",
       "...                                                   ...     \n",
       "121414                                                0.0     \n",
       "121415                                                0.0     \n",
       "121416                                                0.0     \n",
       "121417                                                0.0     \n",
       "121418                                                0.0     \n",
       "\n",
       "        OVCAR8_OVARY_TenPx24  THP1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE_TenPx24  \n",
       "0                        0.0                                              0.0  \n",
       "1                        0.0                                              0.0  \n",
       "2                        0.0                                              0.0  \n",
       "3                        0.0                                              0.0  \n",
       "4                        0.0                                              0.0  \n",
       "...                      ...                                              ...  \n",
       "121414                   0.0                                              0.0  \n",
       "121415                   0.0                                              0.0  \n",
       "121416                   0.0                                              0.0  \n",
       "121417                   0.0                                              0.0  \n",
       "121418                   0.0                                              0.0  \n",
       "\n",
       "[121419 rows x 373 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sum of all values in mapped_protein_df: -88617.17531696183\n",
      "Are all values in mapped_protein_df zero? False\n"
     ]
    }
   ],
   "source": [
    "maped_methy_df = final_merged_methy_df.rename(columns=methy_map_dict)\n",
    "display(maped_methy_df)\n",
    "mapped_protein_df = final_merged_protein_df.rename(columns=protein_map_dict)\n",
    "display(mapped_protein_df)\n",
    "print(\"Sum of all values in mapped_protein_df:\", mapped_protein_df.iloc[:, 1:].sum().sum())\n",
    "# check if all values in mapped_protein_df are zero\n",
    "print(\"Are all values in mapped_protein_df zero?\", (mapped_protein_df.iloc[:, 1:] == 0).all().all())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 9.2.0 All tasks shared the same edge_index and nodes_index system"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Node</th>\n",
       "      <th>Index</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>1</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>2</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>3</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>4</td>\n",
       "      <td>Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>834804</td>\n",
       "      <td>Drug</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>834805</td>\n",
       "      <td>Drug</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>834806</td>\n",
       "      <td>Drug</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>834807</td>\n",
       "      <td>Drug</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>834808</td>\n",
       "      <td>Drug</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Node   Index      Type\n",
       "0       BMGC_PM00001       0  Promoter\n",
       "1       BMGC_PM00002       1  Promoter\n",
       "2       BMGC_PM00003       2  Promoter\n",
       "3       BMGC_PM00004       3  Promoter\n",
       "4       BMGC_PM00005       4  Promoter\n",
       "...              ...     ...       ...\n",
       "834804  BMGC_DG20914  834804      Drug\n",
       "834805  BMGC_DG20915  834805      Drug\n",
       "834806  BMGC_DG20916  834806      Drug\n",
       "834807  BMGC_DG20917  834807      Drug\n",
       "834808  BMGC_DG20918  834808      Drug\n",
       "\n",
       "[834809 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "First 10 items in node_index_dict:\n",
      "[('BMGC_PM00001', 0), ('BMGC_PM00002', 1), ('BMGC_PM00003', 2), ('BMGC_PM00004', 3), ('BMGC_PM00005', 4), ('BMGC_PM00006', 5), ('BMGC_PM00007', 6), ('BMGC_PM00008', 7), ('BMGC_PM00009', 8), ('BMGC_PM00010', 9)]\n",
      "Null values in edge_index_df:\n",
      "BMGC_From_ID    0\n",
      "BMGC_To_ID      0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMGC_From_ID</th>\n",
       "      <th>BMGC_To_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>86238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>86239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>86240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>86241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>86242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27087966</th>\n",
       "      <td>834808</td>\n",
       "      <td>819233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27087967</th>\n",
       "      <td>834808</td>\n",
       "      <td>833344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27087968</th>\n",
       "      <td>834808</td>\n",
       "      <td>826065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27087969</th>\n",
       "      <td>834808</td>\n",
       "      <td>833596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27087970</th>\n",
       "      <td>834808</td>\n",
       "      <td>833612</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27087971 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BMGC_From_ID  BMGC_To_ID\n",
       "0                    0       86238\n",
       "1                    1       86239\n",
       "2                    2       86240\n",
       "3                    3       86241\n",
       "4                    4       86242\n",
       "...                ...         ...\n",
       "27087966        834808      819233\n",
       "27087967        834808      833344\n",
       "27087968        834808      826065\n",
       "27087969        834808      833596\n",
       "27087970        834808      833612\n",
       "\n",
       "[27087971 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The shape of edge_index_array is: (2, 27087971)\n",
      "[[     0      1      2 ... 834808 834808 834808]\n",
      " [ 86238  86239  86240 ... 826065 833596 833612]]\n",
      "Null values in internal_edge_index_df:\n",
      "BMGC_From_ID    0\n",
      "BMGC_To_ID      0\n",
      "dtype: int64\n",
      "The shape of internal_edge_index_array is: (2, 666633)\n",
      "[[     0      1      2 ... 584362 584493 584494]\n",
      " [ 86238  86239  86240 ... 705931 705932 705933]]\n",
      "Null values in ppi_edge_index_df:\n",
      "BMGC_From_ID    0\n",
      "BMGC_To_ID      0\n",
      "dtype: int64\n",
      "The shape of ppi_edge_index_array is: (2, 16484820)\n",
      "[[584539 584539 584539 ... 705933 705933 705933]\n",
      " [594942 596590 596896 ... 705339 705340 705341]]\n"
     ]
    }
   ],
   "source": [
    "# create an index data file for nodes and save it to the DTI_data folder\n",
    "nodes = bmgc_entity_df['BioMedGraphica_Conn_ID'].tolist()\n",
    "# create nodes index ranging from 0 to len(nodes)-1\n",
    "nodes_index = np.arange(len(nodes))\n",
    "nodes_index_data = pd.DataFrame({'Node': nodes, 'Index': nodes_index})\n",
    "nodes_index_data = pd.merge(nodes_index_data, bmgc_entity_df[['BioMedGraphica_Conn_ID', 'Type']], left_on='Node', right_on='BioMedGraphica_Conn_ID', how='left').drop(columns=['BioMedGraphica_Conn_ID'])\n",
    "display(nodes_index_data)\n",
    "nodes_index_data.to_csv('./data/pretrain_plain_data/nodes_index.csv', index=False)\n",
    "nodes_index_data.to_csv('./data/pretrain_status_data/nodes_index.csv', index=False)\n",
    "nodes_index_data.to_csv('./data/TargetQA/nodes_index.csv', index=False)\n",
    "nodes_index_data.to_csv('./data/TargetScreen/nodes_index.csv', index=False)\n",
    "nodes_index_data.to_csv('./data/DrugQA/nodes_index.csv', index=False)\n",
    "nodes_index_data.to_csv('./data/DrugScreen/nodes_index.csv', index=False)\n",
    "node_index_dict = dict(zip(nodes_index_data['Node'], nodes_index_data['Index']))\n",
    "# Print the first 10 items of node_index_dict\n",
    "print(\"\\nFirst 10 items in node_index_dict:\")\n",
    "print(list(node_index_dict.items())[:10])\n",
    "\n",
    "# Convert the bmgc_relation_df to numpy array and save it to the DTI_data folder as the edge_index.npy\n",
    "# keep the columns ['BMGC_From_ID', 'BMGC_To_ID']\n",
    "edge_index_df = bmgc_relation_df[['BMGC_From_ID', 'BMGC_To_ID']].copy()\n",
    "# Map the BMGC_From_ID and BMGC_To_ID by the node_index_dict\n",
    "edge_index_df['BMGC_From_ID'] = edge_index_df['BMGC_From_ID'].map(node_index_dict)\n",
    "edge_index_df['BMGC_To_ID'] = edge_index_df['BMGC_To_ID'].map(node_index_dict)\n",
    "# check the null values in the edge_index_df\n",
    "print(\"Null values in edge_index_df:\")\n",
    "print(edge_index_df.isnull().sum())\n",
    "display(edge_index_df)\n",
    "# convert the edge_index_df to numpy array and save it to the DTI_data folder\n",
    "edge_index_array = edge_index_df.to_numpy().T\n",
    "print('The shape of edge_index_array is:', edge_index_array.shape)\n",
    "print(edge_index_array)\n",
    "# save the numpy array to the DTI_data folder\n",
    "np.save('./data/pretrain_plain_data/edge_index.npy', edge_index_array)\n",
    "np.save('./data/pretrain_status_data/edge_index.npy', edge_index_array)\n",
    "np.save('./data/TargetQA/edge_index.npy', edge_index_array)\n",
    "np.save('./data/TargetScreen/edge_index.npy', edge_index_array)\n",
    "np.save('./data/DrugQA/edge_index.npy', edge_index_array)\n",
    "np.save('./data/DrugScreen/edge_index.npy', edge_index_array)\n",
    "# generate the internal_edge_index by selecting bmgc_relation_df in Type ['Promoter-Gene', 'Gene-Transcript', 'Transcript-Protein']\n",
    "internal_edge_index_df = bmgc_relation_df[bmgc_relation_df['Type'].isin(['Promoter-Gene', 'Gene-Transcript', 'Transcript-Protein'])].copy()\n",
    "# keep the columns ['BMGC_From_ID', 'BMGC_To_ID']\n",
    "internal_edge_index_df = internal_edge_index_df[['BMGC_From_ID', 'BMGC_To_ID']].copy()\n",
    "# Map the BMGC_From_ID and BMGC_To_ID by the node_index_dict\n",
    "internal_edge_index_df['BMGC_From_ID'] = internal_edge_index_df['BMGC_From_ID'].map(node_index_dict)\n",
    "internal_edge_index_df['BMGC_To_ID'] = internal_edge_index_df['BMGC_To_ID'].map(node_index_dict)\n",
    "# check the null values in the internal_edge_index_df\n",
    "print(\"Null values in internal_edge_index_df:\")\n",
    "print(internal_edge_index_df.isnull().sum())\n",
    "# convert the internal_edge_index_df to numpy array and save it to the DTI_data folder\n",
    "internal_edge_index_array = internal_edge_index_df.to_numpy().T\n",
    "print('The shape of internal_edge_index_array is:', internal_edge_index_array.shape)\n",
    "print(internal_edge_index_array)\n",
    "# save the numpy array to the DTI_data folder\n",
    "np.save('./data/pretrain_plain_data/internal_edge_index.npy', internal_edge_index_array)\n",
    "np.save('./data/pretrain_status_data/internal_edge_index.npy', internal_edge_index_array)\n",
    "np.save('./data/TargetQA/internal_edge_index.npy', internal_edge_index_array)\n",
    "np.save('./data/TargetScreen/internal_edge_index.npy', internal_edge_index_array)\n",
    "np.save('./data/DrugQA/internal_edge_index.npy', internal_edge_index_array)\n",
    "np.save('./data/DrugScreen/internal_edge_index.npy', internal_edge_index_array)\n",
    "# generate the ppi_edge_index by selecting bmgc_relation_df in Type ['Protein-Protein']\n",
    "ppi_edge_index_df = bmgc_relation_df[bmgc_relation_df['Type'].isin(['Protein-Protein'])].copy()\n",
    "# keep the columns ['BMGC_From_ID', 'BMGC_To_ID']\n",
    "ppi_edge_index_df = ppi_edge_index_df[['BMGC_From_ID', 'BMGC_To_ID']].copy()\n",
    "# Map the BMGC_From_ID and BMGC_To_ID by the node_index_dict\n",
    "ppi_edge_index_df['BMGC_From_ID'] = ppi_edge_index_df['BMGC_From_ID'].map(node_index_dict)\n",
    "ppi_edge_index_df['BMGC_To_ID'] = ppi_edge_index_df['BMGC_To_ID'].map(node_index_dict)\n",
    "# check the null values in the ppi_edge_index_df\n",
    "print(\"Null values in ppi_edge_index_df:\")\n",
    "print(ppi_edge_index_df.isnull().sum())\n",
    "# convert the ppi_edge_index_df to numpy array and save it to the DTI_data folder\n",
    "ppi_edge_index_array = ppi_edge_index_df.to_numpy().T\n",
    "print('The shape of ppi_edge_index_array is:', ppi_edge_index_array.shape)\n",
    "print(ppi_edge_index_array)\n",
    "# save the numpy array to the DTI_data folder\n",
    "np.save('./data/pretrain_plain_data/ppi_edge_index.npy', ppi_edge_index_array)\n",
    "np.save('./data/pretrain_status_data/ppi_edge_index.npy', ppi_edge_index_array)\n",
    "np.save('./data/TargetQA/ppi_edge_index.npy', ppi_edge_index_array)\n",
    "np.save('./data/TargetScreen/ppi_edge_index.npy', ppi_edge_index_array)\n",
    "np.save('./data/DrugQA/ppi_edge_index.npy', ppi_edge_index_array)\n",
    "np.save('./data/DrugScreen/ppi_edge_index.npy', ppi_edge_index_array)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 9.2.0 All tasks shared the same textual description"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 474,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bmgc_promoter_name_id_df: 86238\n",
      "bmgc_gene_name_id_df: 86238\n",
      "bmgc_transcript_name_id_df: 412039\n",
      "bmgc_protein_name_id_df: 121419\n",
      "bmgc_pathway_name_id_df: 1930\n",
      "bmgc_metabolite_name_id_df: 62364\n",
      "bmgc_microbiota_name_id_df: 1119\n",
      "bmgc_exposure_name_id_df: 1037\n",
      "bmgc_phenotype_name_id_df: 19078\n",
      "bmgc_disease_name_id_df: 22429\n",
      "bmgc_drug_name_id_df: 20918\n",
      "Total number of rows in all dataframes: 834809\n"
     ]
    }
   ],
   "source": [
    "# Create the name and desc files\n",
    "bmgc_promoter_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Promoter/BioMedGraphica_Conn_Promoter_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_gene_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_transcript_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Transcript/BioMedGraphica_Conn_Transcript_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_protein_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_pathway_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Pathway/BioMedGraphica_Conn_Pathway_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_metabolite_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Metabolite/BioMedGraphica_Conn_Metabolite_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_microbiota_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Microbiota/BioMedGraphica_Conn_Microbiota_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_exposure_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Exposure/BioMedGraphica_Conn_Exposure_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_phenotype_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Phenotype/BioMedGraphica_Conn_Phenotype_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_disease_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Disease/BioMedGraphica_Conn_Disease_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_drug_name_id_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Drug/BioMedGraphica_Conn_Drug_LLM_Name_ID_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "# get the number of rows in each dataframe and sum the number of rows for those dataframes\n",
    "print(\"bmgc_promoter_name_id_df:\", len(bmgc_promoter_name_id_df))\n",
    "print(\"bmgc_gene_name_id_df:\", len(bmgc_gene_name_id_df))\n",
    "print(\"bmgc_transcript_name_id_df:\", len(bmgc_transcript_name_id_df))\n",
    "print(\"bmgc_protein_name_id_df:\", len(bmgc_protein_name_id_df))\n",
    "print(\"bmgc_pathway_name_id_df:\", len(bmgc_pathway_name_id_df))\n",
    "print(\"bmgc_metabolite_name_id_df:\", len(bmgc_metabolite_name_id_df))\n",
    "print(\"bmgc_microbiota_name_id_df:\", len(bmgc_microbiota_name_id_df))\n",
    "print(\"bmgc_exposure_name_id_df:\", len(bmgc_exposure_name_id_df))\n",
    "print(\"bmgc_phenotype_name_id_df:\", len(bmgc_phenotype_name_id_df))\n",
    "print(\"bmgc_disease_name_id_df:\", len(bmgc_disease_name_id_df))\n",
    "print(\"bmgc_drug_name_id_df:\", len(bmgc_drug_name_id_df))\n",
    "# sum the number of rows for those dataframes\n",
    "print(\"Total number of rows in all dataframes:\", len(bmgc_promoter_name_id_df) + len(bmgc_gene_name_id_df) + len(bmgc_transcript_name_id_df) + len(bmgc_protein_name_id_df) + len(bmgc_pathway_name_id_df) + len(bmgc_metabolite_name_id_df) + len(bmgc_microbiota_name_id_df) + len(bmgc_exposure_name_id_df) + len(bmgc_phenotype_name_id_df) + len(bmgc_disease_name_id_df) + len(bmgc_drug_name_id_df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 475,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID\n",
       "0                BMGC_PM00001\n",
       "1                BMGC_PM00002\n",
       "2                BMGC_PM00003\n",
       "3                BMGC_PM00004\n",
       "4                BMGC_PM00005\n",
       "...                       ...\n",
       "834804           BMGC_DG20914\n",
       "834805           BMGC_DG20915\n",
       "834806           BMGC_DG20916\n",
       "834807           BMGC_DG20917\n",
       "834808           BMGC_DG20918\n",
       "\n",
       "[834809 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Names_and_IDs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>NCBI Gene ID:1 | HGNC:5 | OMIM ID:138670 | Ref...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>N-acetyltransferase 2 Promoter | RefSeq ID:NG_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>RefSeq ID:NC_000020.11 | OMIM ID:608958 | NCBI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>NCBI Gene ID:1000 | HGNC:1759 | RefSeq ID:NC_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>RefSeq ID:NG_029764.2 | RefSeq ID:NT_187519.1 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>Zoxazolamine | DrugBank ID:DB19372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>Megestrol | DrugBank ID:DB19378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>DrugBank ID:DB19381 | AZACYCLONOL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>DrugBank ID:DB19397 | AZD8630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>DrugBank ID:DB19413 | Influenza A Virus A/Thai...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  \\\n",
       "0                BMGC_PM00001   \n",
       "1                BMGC_PM00002   \n",
       "2                BMGC_PM00003   \n",
       "3                BMGC_PM00004   \n",
       "4                BMGC_PM00005   \n",
       "...                       ...   \n",
       "834804           BMGC_DG20914   \n",
       "834805           BMGC_DG20915   \n",
       "834806           BMGC_DG20916   \n",
       "834807           BMGC_DG20917   \n",
       "834808           BMGC_DG20918   \n",
       "\n",
       "                                            Names_and_IDs  \n",
       "0       NCBI Gene ID:1 | HGNC:5 | OMIM ID:138670 | Ref...  \n",
       "1       N-acetyltransferase 2 Promoter | RefSeq ID:NG_...  \n",
       "2       RefSeq ID:NC_000020.11 | OMIM ID:608958 | NCBI...  \n",
       "3       NCBI Gene ID:1000 | HGNC:1759 | RefSeq ID:NC_0...  \n",
       "4       RefSeq ID:NG_029764.2 | RefSeq ID:NT_187519.1 ...  \n",
       "...                                                   ...  \n",
       "834804                 Zoxazolamine | DrugBank ID:DB19372  \n",
       "834805                    Megestrol | DrugBank ID:DB19378  \n",
       "834806                  DrugBank ID:DB19381 | AZACYCLONOL  \n",
       "834807                      DrugBank ID:DB19397 | AZD8630  \n",
       "834808  DrugBank ID:DB19413 | Influenza A Virus A/Thai...  \n",
       "\n",
       "[834809 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Names_and_IDs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>NCBI Gene ID:1 | HGNC:5 | OMIM ID:138670 | Ref...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>N-acetyltransferase 2 Promoter | RefSeq ID:NG_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>RefSeq ID:NC_000020.11 | OMIM ID:608958 | NCBI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>NCBI Gene ID:1000 | HGNC:1759 | RefSeq ID:NC_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>RefSeq ID:NG_029764.2 | RefSeq ID:NT_187519.1 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>Zoxazolamine | DrugBank ID:DB19372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>Megestrol | DrugBank ID:DB19378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>DrugBank ID:DB19381 | AZACYCLONOL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>DrugBank ID:DB19397 | AZD8630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>DrugBank ID:DB19413 | Influenza A Virus A/Thai...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  \\\n",
       "0                BMGC_PM00001   \n",
       "1                BMGC_PM00002   \n",
       "2                BMGC_PM00003   \n",
       "3                BMGC_PM00004   \n",
       "4                BMGC_PM00005   \n",
       "...                       ...   \n",
       "834804           BMGC_DG20914   \n",
       "834805           BMGC_DG20915   \n",
       "834806           BMGC_DG20916   \n",
       "834807           BMGC_DG20917   \n",
       "834808           BMGC_DG20918   \n",
       "\n",
       "                                            Names_and_IDs  \n",
       "0       NCBI Gene ID:1 | HGNC:5 | OMIM ID:138670 | Ref...  \n",
       "1       N-acetyltransferase 2 Promoter | RefSeq ID:NG_...  \n",
       "2       RefSeq ID:NC_000020.11 | OMIM ID:608958 | NCBI...  \n",
       "3       NCBI Gene ID:1000 | HGNC:1759 | RefSeq ID:NC_0...  \n",
       "4       RefSeq ID:NG_029764.2 | RefSeq ID:NT_187519.1 ...  \n",
       "...                                                   ...  \n",
       "834804                 Zoxazolamine | DrugBank ID:DB19372  \n",
       "834805                    Megestrol | DrugBank ID:DB19378  \n",
       "834806                  DrugBank ID:DB19381 | AZACYCLONOL  \n",
       "834807                      DrugBank ID:DB19397 | AZD8630  \n",
       "834808  DrugBank ID:DB19413 | Influenza A Virus A/Thai...  \n",
       "\n",
       "[834809 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# left join those name_df into bmgc_entity_df\n",
    "bmgc_entity_cp_df = bmgc_entity_df[['BioMedGraphica_Conn_ID']].copy()\n",
    "display(bmgc_entity_cp_df)\n",
    "# concatenate all the name_id_df into one dataframe\n",
    "bmgc_name_id_tmp_df = pd.concat([bmgc_promoter_name_id_df, bmgc_gene_name_id_df, bmgc_transcript_name_id_df, bmgc_protein_name_id_df, bmgc_pathway_name_id_df, bmgc_metabolite_name_id_df, bmgc_microbiota_name_id_df, bmgc_exposure_name_id_df, bmgc_phenotype_name_id_df, bmgc_disease_name_id_df, bmgc_drug_name_id_df], ignore_index=True)\n",
    "display(bmgc_name_id_tmp_df)\n",
    "# left join the bmgc_name_id_tmp_df into bmgc_entity_cp_df on BioMedGraphica_Conn_ID\n",
    "bmgc_entity_cp_df = pd.merge(bmgc_entity_cp_df, bmgc_name_id_tmp_df, on='BioMedGraphica_Conn_ID', how='left')\n",
    "display(bmgc_entity_cp_df)\n",
    "bmgc_entity_cp_df.to_csv('./data/pretrain_plain_data/bmgc_name.csv', index=False)\n",
    "bmgc_entity_cp_df.to_csv('./data/pretrain_status_data/bmgc_name.csv', index=False)\n",
    "bmgc_entity_cp_df.to_csv('./data/TargetScreen/bmgc_name.csv', index=False)\n",
    "bmgc_entity_cp_df.to_csv('./data/TargetQA/bmgc_name.csv', index=False)\n",
    "bmgc_entity_cp_df.to_csv('./data/DrugScreen/bmgc_name.csv', index=False)\n",
    "bmgc_entity_cp_df.to_csv('./data/DrugQA/bmgc_name.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 476,
   "metadata": {},
   "outputs": [],
   "source": [
    "bmgc_promoter_desc_df = bmgc_promoter_name_id_df.drop(columns=['Names_and_IDs'], axis=1).copy()\n",
    "bmgc_promoter_desc_df['Description'] = np.nan # add the Description column to bmgc_promoter_name_df with NaN values\n",
    "bmgc_gene_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Gene/BioMedGraphica_Conn_Gene_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_transcript_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Transcript/BioMedGraphica_Conn_Transcript_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_protein_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_pathway_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Pathway/BioMedGraphica_Conn_Pathway_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_metabolite_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Metabolite/BioMedGraphica_Conn_Metabolite_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_microbiota_desc_df = bmgc_microbiota_name_id_df.drop(columns=['Names_and_IDs'], axis=1).copy()\n",
    "bmgc_microbiota_desc_df['Description'] = np.nan # add the Description column to bmgc_microbiota_name_df with NaN values\n",
    "bmgc_exposure_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Exposure/BioMedGraphica_Conn_Exposure_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_phenotype_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Phenotype/BioMedGraphica_Conn_Phenotype_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_disease_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Disease/BioMedGraphica_Conn_Disease_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'])\n",
    "bmgc_drug_desc_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Drug/BioMedGraphica_Conn_Drug_Description_Combined.csv').drop(columns=['BioMedGraphica_ID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 477,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>DrugBank: AZD8630 is an experimental drug unde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>DrugBank: A seasonally-specific component of t...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  \\\n",
       "0                BMGC_PM00001   \n",
       "1                BMGC_PM00002   \n",
       "2                BMGC_PM00003   \n",
       "3                BMGC_PM00004   \n",
       "4                BMGC_PM00005   \n",
       "...                       ...   \n",
       "834804           BMGC_DG20914   \n",
       "834805           BMGC_DG20915   \n",
       "834806           BMGC_DG20916   \n",
       "834807           BMGC_DG20917   \n",
       "834808           BMGC_DG20918   \n",
       "\n",
       "                                              Description  \n",
       "0                                                     NaN  \n",
       "1                                                     NaN  \n",
       "2                                                     NaN  \n",
       "3                                                     NaN  \n",
       "4                                                     NaN  \n",
       "...                                                   ...  \n",
       "834804                                                NaN  \n",
       "834805                                                NaN  \n",
       "834806                                                NaN  \n",
       "834807  DrugBank: AZD8630 is an experimental drug unde...  \n",
       "834808  DrugBank: A seasonally-specific component of t...  \n",
       "\n",
       "[834809 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Null values in bmgc_desc_df:\n",
      "BioMedGraphica_Conn_ID         0\n",
      "Description               221047\n",
      "dtype: int64\n",
      "Null values in bmgc_desc_df:\n",
      "BioMedGraphica_Conn_ID    0\n",
      "Description               0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>DrugBank: AZD8630 is an experimental drug unde...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>DrugBank: A seasonally-specific component of t...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  \\\n",
       "0                BMGC_PM00001   \n",
       "1                BMGC_PM00002   \n",
       "2                BMGC_PM00003   \n",
       "3                BMGC_PM00004   \n",
       "4                BMGC_PM00005   \n",
       "...                       ...   \n",
       "834804           BMGC_DG20914   \n",
       "834805           BMGC_DG20915   \n",
       "834806           BMGC_DG20916   \n",
       "834807           BMGC_DG20917   \n",
       "834808           BMGC_DG20918   \n",
       "\n",
       "                                              Description  \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "3                                                          \n",
       "4                                                          \n",
       "...                                                   ...  \n",
       "834804                                                     \n",
       "834805                                                     \n",
       "834806                                                     \n",
       "834807  DrugBank: AZD8630 is an experimental drug unde...  \n",
       "834808  DrugBank: A seasonally-specific component of t...  \n",
       "\n",
       "[834809 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bmgc_desc_tmp_df = pd.concat([bmgc_promoter_desc_df, bmgc_gene_desc_df, bmgc_transcript_desc_df, bmgc_protein_desc_df, bmgc_pathway_desc_df, bmgc_metabolite_desc_df, bmgc_microbiota_desc_df, bmgc_exposure_desc_df, bmgc_phenotype_desc_df, bmgc_disease_desc_df, bmgc_drug_desc_df], ignore_index=True)\n",
    "display(bmgc_desc_tmp_df)\n",
    "# left join the bmgc_desc_tmp_df into bmgc_entity_cp_df on BioMedGraphica_Conn_ID\n",
    "bmgc_entity_cp_df = bmgc_entity_df[['BioMedGraphica_Conn_ID']].copy()\n",
    "bmgc_desc_df = pd.merge(bmgc_entity_cp_df, bmgc_desc_tmp_df, on='BioMedGraphica_Conn_ID', how='left')\n",
    "# check the null values in the bmgc_desc_df\n",
    "print(\"Null values in bmgc_desc_df:\")\n",
    "print(bmgc_desc_df.isnull().sum())\n",
    "# fill the NaN values in the Description column with empty string\n",
    "bmgc_desc_df['Description'] = bmgc_desc_df['Description'].fillna(' ')\n",
    "# recheck the null values in the bmgc_desc_df\n",
    "print(\"Null values in bmgc_desc_df:\")\n",
    "print(bmgc_desc_df.isnull().sum())\n",
    "display(bmgc_desc_df)\n",
    "bmgc_desc_df.to_csv('./data/pretrain_plain_data/bmgc_desc.csv', index=False)\n",
    "bmgc_desc_df.to_csv('./data/pretrain_status_data/bmgc_desc.csv', index=False)\n",
    "bmgc_desc_df.to_csv('./data/TargetScreen/bmgc_desc.csv', index=False)\n",
    "bmgc_desc_df.to_csv('./data/TargetQA/bmgc_desc.csv', index=False)\n",
    "bmgc_desc_df.to_csv('./data/DrugScreen/bmgc_desc.csv', index=False)\n",
    "bmgc_desc_df.to_csv('./data/DrugQA/bmgc_desc.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 9.2.1 Pretrain data integration (plain + status)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The overlapped samples dataframe for methylation between pretrain_plain_samples and pretrain_status_samples are:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000017</th>\n",
       "      <th>ACH-000022</th>\n",
       "      <th>ACH-000026</th>\n",
       "      <th>ACH-000028</th>\n",
       "      <th>ACH-000063</th>\n",
       "      <th>ACH-000066</th>\n",
       "      <th>ACH-000085</th>\n",
       "      <th>ACH-000086</th>\n",
       "      <th>ACH-000119</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000844</th>\n",
       "      <th>ACH-000889</th>\n",
       "      <th>ACH-000942</th>\n",
       "      <th>ACH-000993</th>\n",
       "      <th>ACH-001001</th>\n",
       "      <th>ACH-001036</th>\n",
       "      <th>ACH-001098</th>\n",
       "      <th>ACH-001129</th>\n",
       "      <th>ACH-001188</th>\n",
       "      <th>ACH-001239</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.50165</td>\n",
       "      <td>0.49325</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.84212</td>\n",
       "      <td>0.48636</td>\n",
       "      <td>0.65552</td>\n",
       "      <td>0.09454</td>\n",
       "      <td>0.69810</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.05468</td>\n",
       "      <td>0.86674</td>\n",
       "      <td>0.45000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.72600</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.80595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.00300</td>\n",
       "      <td>0.00649</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00141</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00943</td>\n",
       "      <td>0.00064</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.02074</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.01952</td>\n",
       "      <td>0.00237</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.03815</td>\n",
       "      <td>0.02166</td>\n",
       "      <td>0.03068</td>\n",
       "      <td>0.01391</td>\n",
       "      <td>0.01534</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.01450</td>\n",
       "      <td>0.87374</td>\n",
       "      <td>0.01292</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.26234</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_PM86234</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_PM86235</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_PM86236</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000017  ACH-000022  ACH-000026  ACH-000028  \\\n",
       "0               BMGC_PM00001     0.50165     0.49325         0.0     0.84212   \n",
       "1               BMGC_PM00002     0.00000     0.00000         0.0     0.00000   \n",
       "2               BMGC_PM00003     0.00300     0.00649         0.0     0.00141   \n",
       "3               BMGC_PM00004     0.01952     0.00237         0.0     0.03815   \n",
       "4               BMGC_PM00005     0.00000     0.00000         0.0     0.00000   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_PM86234     0.00000     0.00000         0.0     0.00000   \n",
       "86234           BMGC_PM86235     0.00000     0.00000         0.0     0.00000   \n",
       "86235           BMGC_PM86236     0.00000     0.00000         0.0     0.00000   \n",
       "86236           BMGC_PM86237     0.00000     0.00000         0.0     0.00000   \n",
       "86237           BMGC_PM86238     0.00000     0.00000         0.0     0.00000   \n",
       "\n",
       "       ACH-000063  ACH-000066  ACH-000085  ACH-000086  ACH-000119  ...  \\\n",
       "0         0.48636     0.65552     0.09454     0.69810         0.0  ...   \n",
       "1         0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "2         0.00000     0.00000     0.00943     0.00064         0.0  ...   \n",
       "3         0.02166     0.03068     0.01391     0.01534         0.0  ...   \n",
       "4         0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "86234     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "86235     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "86236     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "86237     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "       ACH-000844  ACH-000889  ACH-000942  ACH-000993  ACH-001001  ACH-001036  \\\n",
       "0         0.05468     0.86674     0.45000         0.0         0.0         0.0   \n",
       "1         0.00000     0.00000     0.00000         0.0         0.0         0.0   \n",
       "2         0.02074     0.00000     0.00000         0.0         0.0         0.0   \n",
       "3         0.01450     0.87374     0.01292         0.0         0.0         0.0   \n",
       "4         0.00000     0.00000     0.00000         0.0         0.0         0.0   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233     0.00000     0.00000     0.00000         0.0         0.0         0.0   \n",
       "86234     0.00000     0.00000     0.00000         0.0         0.0         0.0   \n",
       "86235     0.00000     0.00000     0.00000         0.0         0.0         0.0   \n",
       "86236     0.00000     0.00000     0.00000         0.0         0.0         0.0   \n",
       "86237     0.00000     0.00000     0.00000         0.0         0.0         0.0   \n",
       "\n",
       "       ACH-001098  ACH-001129  ACH-001188  ACH-001239  \n",
       "0             0.0     0.72600         0.0     0.80595  \n",
       "1             0.0     0.00000         0.0     0.00000  \n",
       "2             0.0     0.00000         0.0     0.01409  \n",
       "3             0.0     0.26234         0.0     0.00195  \n",
       "4             0.0     0.00000         0.0     0.00000  \n",
       "...           ...         ...         ...         ...  \n",
       "86233         0.0     0.00000         0.0     0.00000  \n",
       "86234         0.0     0.00000         0.0     0.00000  \n",
       "86235         0.0     0.00000         0.0     0.00000  \n",
       "86236         0.0     0.00000         0.0     0.00000  \n",
       "86237         0.0     0.00000         0.0     0.00000  \n",
       "\n",
       "[86238 rows x 68 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>ACH-000016</th>\n",
       "      <th>ACH-000027</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000032</th>\n",
       "      <th>ACH-000037</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001113</th>\n",
       "      <th>ACH-001130</th>\n",
       "      <th>ACH-001145</th>\n",
       "      <th>ACH-001150</th>\n",
       "      <th>ACH-001162</th>\n",
       "      <th>ACH-001171</th>\n",
       "      <th>ACH-001183</th>\n",
       "      <th>ACH-001194</th>\n",
       "      <th>ACH-001318</th>\n",
       "      <th>ACH-001321</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.70874</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.66057</td>\n",
       "      <td>0.88330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.51282</td>\n",
       "      <td>0.12250</td>\n",
       "      <td>0.70000</td>\n",
       "      <td>0.15385</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.47703</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.42555</td>\n",
       "      <td>0.47785</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50254</td>\n",
       "      <td>0.63336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.06977</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00507</td>\n",
       "      <td>0.01536</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00260</td>\n",
       "      <td>0.00351</td>\n",
       "      <td>0.01941</td>\n",
       "      <td>0.01643</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00206</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01197</td>\n",
       "      <td>0.00757</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00453</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.59602</td>\n",
       "      <td>0.01294</td>\n",
       "      <td>0.04124</td>\n",
       "      <td>0.01022</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01732</td>\n",
       "      <td>0.00056</td>\n",
       "      <td>0.55973</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05561</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01421</td>\n",
       "      <td>0.73156</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01198</td>\n",
       "      <td>0.01604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_PM86234</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_PM86235</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_PM86236</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000005  ACH-000011  ACH-000013  ACH-000014  \\\n",
       "0               BMGC_PM00001     0.70874     1.00000     0.66057     0.88330   \n",
       "1               BMGC_PM00002     0.00000     0.00000     0.00000     0.00000   \n",
       "2               BMGC_PM00003     0.06977     0.00000     0.00507     0.01536   \n",
       "3               BMGC_PM00004     0.59602     0.01294     0.04124     0.01022   \n",
       "4               BMGC_PM00005     0.00000     0.00000     0.00000     0.00000   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_PM86234     0.00000     0.00000     0.00000     0.00000   \n",
       "86234           BMGC_PM86235     0.00000     0.00000     0.00000     0.00000   \n",
       "86235           BMGC_PM86236     0.00000     0.00000     0.00000     0.00000   \n",
       "86236           BMGC_PM86237     0.00000     0.00000     0.00000     0.00000   \n",
       "86237           BMGC_PM86238     0.00000     0.00000     0.00000     0.00000   \n",
       "\n",
       "       ACH-000016  ACH-000027  ACH-000031  ACH-000032  ACH-000037  ...  \\\n",
       "0             0.0     0.51282     0.12250     0.70000     0.15385  ...   \n",
       "1             0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "2             0.0     0.00260     0.00351     0.01941     0.01643  ...   \n",
       "3             0.0     0.01732     0.00056     0.55973     0.00000  ...   \n",
       "4             0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "86234         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "86235         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "86236         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "86237         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "\n",
       "       ACH-001113  ACH-001130  ACH-001145  ACH-001150  ACH-001162  ACH-001171  \\\n",
       "0             0.0     0.47703         0.0     0.42555     0.47785         0.0   \n",
       "1             0.0     0.00000         0.0     0.00000     0.00000         0.0   \n",
       "2             0.0     0.00206         0.0     0.01197     0.00757         0.0   \n",
       "3             0.0     0.05561         0.0     0.01421     0.73156         0.0   \n",
       "4             0.0     0.00000         0.0     0.00000     0.00000         0.0   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233         0.0     0.00000         0.0     0.00000     0.00000         0.0   \n",
       "86234         0.0     0.00000         0.0     0.00000     0.00000         0.0   \n",
       "86235         0.0     0.00000         0.0     0.00000     0.00000         0.0   \n",
       "86236         0.0     0.00000         0.0     0.00000     0.00000         0.0   \n",
       "86237         0.0     0.00000         0.0     0.00000     0.00000         0.0   \n",
       "\n",
       "       ACH-001183  ACH-001194  ACH-001318  ACH-001321  \n",
       "0             0.0         0.0     0.50254     0.63336  \n",
       "1             0.0         0.0     0.00000     0.00000  \n",
       "2             0.0         0.0     0.00453     0.00000  \n",
       "3             0.0         0.0     0.01198     0.01604  \n",
       "4             0.0         0.0     0.00000     0.00000  \n",
       "...           ...         ...         ...         ...  \n",
       "86233         0.0         0.0     0.00000     0.00000  \n",
       "86234         0.0         0.0     0.00000     0.00000  \n",
       "86235         0.0         0.0     0.00000     0.00000  \n",
       "86236         0.0         0.0     0.00000     0.00000  \n",
       "86237         0.0         0.0     0.00000     0.00000  \n",
       "\n",
       "[86238 rows x 270 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The overlapped samples dataframe for protein between pretrain_plain_samples and pretrain_status_samples are:\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:21: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000017</th>\n",
       "      <th>ACH-000022</th>\n",
       "      <th>ACH-000026</th>\n",
       "      <th>ACH-000028</th>\n",
       "      <th>ACH-000063</th>\n",
       "      <th>ACH-000066</th>\n",
       "      <th>ACH-000085</th>\n",
       "      <th>ACH-000086</th>\n",
       "      <th>ACH-000119</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000844</th>\n",
       "      <th>ACH-000889</th>\n",
       "      <th>ACH-000942</th>\n",
       "      <th>ACH-000993</th>\n",
       "      <th>ACH-001001</th>\n",
       "      <th>ACH-001036</th>\n",
       "      <th>ACH-001098</th>\n",
       "      <th>ACH-001129</th>\n",
       "      <th>ACH-001188</th>\n",
       "      <th>ACH-001239</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000017  ACH-000022  ACH-000026  ACH-000028  \\\n",
       "0               BMGC_PT000001         0.0         0.0         0.0         0.0   \n",
       "1               BMGC_PT000002         0.0         0.0         0.0         0.0   \n",
       "2               BMGC_PT000003         0.0         0.0         0.0         0.0   \n",
       "3               BMGC_PT000004         0.0         0.0         0.0         0.0   \n",
       "4               BMGC_PT000005         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "121414          BMGC_PT121415         0.0         0.0         0.0         0.0   \n",
       "121415          BMGC_PT121416         0.0         0.0         0.0         0.0   \n",
       "121416          BMGC_PT121417         0.0         0.0         0.0         0.0   \n",
       "121417          BMGC_PT121418         0.0         0.0         0.0         0.0   \n",
       "121418          BMGC_PT121419         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-000063  ACH-000066  ACH-000085  ACH-000086  ACH-000119  ...  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "1              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "2              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "3              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "4              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121415         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121416         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121417         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121418         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "\n",
       "        ACH-000844  ACH-000889  ACH-000942  ACH-000993  ACH-001001  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0   \n",
       "1              0.0         0.0         0.0         0.0         0.0   \n",
       "2              0.0         0.0         0.0         0.0         0.0   \n",
       "3              0.0         0.0         0.0         0.0         0.0   \n",
       "4              0.0         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0   \n",
       "121415         0.0         0.0         0.0         0.0         0.0   \n",
       "121416         0.0         0.0         0.0         0.0         0.0   \n",
       "121417         0.0         0.0         0.0         0.0         0.0   \n",
       "121418         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001036  ACH-001098  ACH-001129  ACH-001188  ACH-001239  \n",
       "0              0.0         0.0         0.0         0.0         0.0  \n",
       "1              0.0         0.0         0.0         0.0         0.0  \n",
       "2              0.0         0.0         0.0         0.0         0.0  \n",
       "3              0.0         0.0         0.0         0.0         0.0  \n",
       "4              0.0         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "121414         0.0         0.0         0.0         0.0         0.0  \n",
       "121415         0.0         0.0         0.0         0.0         0.0  \n",
       "121416         0.0         0.0         0.0         0.0         0.0  \n",
       "121417         0.0         0.0         0.0         0.0         0.0  \n",
       "121418         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[121419 rows x 68 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\4084772069.py:28: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>ACH-000016</th>\n",
       "      <th>ACH-000027</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000032</th>\n",
       "      <th>ACH-000037</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001113</th>\n",
       "      <th>ACH-001130</th>\n",
       "      <th>ACH-001145</th>\n",
       "      <th>ACH-001150</th>\n",
       "      <th>ACH-001162</th>\n",
       "      <th>ACH-001171</th>\n",
       "      <th>ACH-001183</th>\n",
       "      <th>ACH-001194</th>\n",
       "      <th>ACH-001318</th>\n",
       "      <th>ACH-001321</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000005  ACH-000011  ACH-000013  ACH-000014  \\\n",
       "0               BMGC_PT000001         0.0         0.0         0.0         0.0   \n",
       "1               BMGC_PT000002         0.0         0.0         0.0         0.0   \n",
       "2               BMGC_PT000003         0.0         0.0         0.0         0.0   \n",
       "3               BMGC_PT000004         0.0         0.0         0.0         0.0   \n",
       "4               BMGC_PT000005         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "121414          BMGC_PT121415         0.0         0.0         0.0         0.0   \n",
       "121415          BMGC_PT121416         0.0         0.0         0.0         0.0   \n",
       "121416          BMGC_PT121417         0.0         0.0         0.0         0.0   \n",
       "121417          BMGC_PT121418         0.0         0.0         0.0         0.0   \n",
       "121418          BMGC_PT121419         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-000016  ACH-000027  ACH-000031  ACH-000032  ACH-000037  ...  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "1              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "2              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "3              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "4              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121415         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121416         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121417         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121418         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "\n",
       "        ACH-001113  ACH-001130  ACH-001145  ACH-001150  ACH-001162  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0   \n",
       "1              0.0         0.0         0.0         0.0         0.0   \n",
       "2              0.0         0.0         0.0         0.0         0.0   \n",
       "3              0.0         0.0         0.0         0.0         0.0   \n",
       "4              0.0         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0   \n",
       "121415         0.0         0.0         0.0         0.0         0.0   \n",
       "121416         0.0         0.0         0.0         0.0         0.0   \n",
       "121417         0.0         0.0         0.0         0.0         0.0   \n",
       "121418         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001171  ACH-001183  ACH-001194  ACH-001318  ACH-001321  \n",
       "0              0.0         0.0         0.0         0.0         0.0  \n",
       "1              0.0         0.0         0.0         0.0         0.0  \n",
       "2              0.0         0.0         0.0         0.0         0.0  \n",
       "3              0.0         0.0         0.0         0.0         0.0  \n",
       "4              0.0         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "121414         0.0         0.0         0.0         0.0         0.0  \n",
       "121415         0.0         0.0         0.0         0.0         0.0  \n",
       "121416         0.0         0.0         0.0         0.0         0.0  \n",
       "121417         0.0         0.0         0.0         0.0         0.0  \n",
       "121418         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[121419 rows x 270 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(\"The overlapped samples dataframe for methylation between pretrain_plain_samples and pretrain_status_samples are:\")\n",
    "# insert the samples that is in the overlapped_samples but not in the maped_methy_df\n",
    "for sample in pretrain_plain_samples:\n",
    "    if sample not in maped_methy_df.columns:\n",
    "        maped_methy_df[sample] = 0.0\n",
    "# filter out the samples that is not in the pretraining_samples\n",
    "pretrain_plain_methy_df = maped_methy_df[['BioMedGraphica_Conn_ID'] + pretrain_plain_samples]\n",
    "display(pretrain_plain_methy_df)\n",
    "# insert the samples that is in the overlapped_samples but not in the maped_methy_df\n",
    "for sample in pretrain_status_samples:\n",
    "    if sample not in maped_methy_df.columns:\n",
    "        maped_methy_df[sample] = 0.0\n",
    "# filter out the samples that is not in the pretraining_samples\n",
    "pretrain_status_methy_df = maped_methy_df[['BioMedGraphica_Conn_ID'] + pretrain_status_samples]\n",
    "display(pretrain_status_methy_df)\n",
    "\n",
    "print(\"The overlapped samples dataframe for protein between pretrain_plain_samples and pretrain_status_samples are:\")\n",
    "# insert the samples that is in the pretraining_samples but not in the mapped_protein_df\n",
    "for sample in pretrain_plain_samples:\n",
    "    if sample not in mapped_protein_df.columns:\n",
    "        mapped_protein_df[sample] = 0.0\n",
    "# filter out the samples that is not in the pretraining_samples\n",
    "pretrain_plain_protein_df = mapped_protein_df[['BioMedGraphica_Conn_ID'] + pretrain_plain_samples]\n",
    "display(pretrain_plain_protein_df)\n",
    "# insert the samples that is in the pretraining_samples but not in the mapped_protein_df\n",
    "for sample in pretrain_status_samples:\n",
    "    if sample not in mapped_protein_df.columns:\n",
    "        mapped_protein_df[sample] = 0.0\n",
    "# filter out the samples that is not in the pretrain_status_samples\n",
    "pretrain_status_protein_df = mapped_protein_df[['BioMedGraphica_Conn_ID'] + pretrain_status_samples]\n",
    "display(pretrain_status_protein_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000017</th>\n",
       "      <th>ACH-000022</th>\n",
       "      <th>ACH-000026</th>\n",
       "      <th>ACH-000028</th>\n",
       "      <th>ACH-000063</th>\n",
       "      <th>ACH-000066</th>\n",
       "      <th>ACH-000085</th>\n",
       "      <th>ACH-000086</th>\n",
       "      <th>ACH-000119</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000844</th>\n",
       "      <th>ACH-000889</th>\n",
       "      <th>ACH-000942</th>\n",
       "      <th>ACH-000993</th>\n",
       "      <th>ACH-001001</th>\n",
       "      <th>ACH-001036</th>\n",
       "      <th>ACH-001098</th>\n",
       "      <th>ACH-001129</th>\n",
       "      <th>ACH-001188</th>\n",
       "      <th>ACH-001239</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>1.085323</td>\n",
       "      <td>0.947735</td>\n",
       "      <td>0.827625</td>\n",
       "      <td>0.937181</td>\n",
       "      <td>1.023235</td>\n",
       "      <td>1.504031</td>\n",
       "      <td>1.412782</td>\n",
       "      <td>1.047031</td>\n",
       "      <td>1.000624</td>\n",
       "      <td>...</td>\n",
       "      <td>1.051902</td>\n",
       "      <td>1.073072</td>\n",
       "      <td>1.032968</td>\n",
       "      <td>1.011142</td>\n",
       "      <td>0.781099</td>\n",
       "      <td>1.181581</td>\n",
       "      <td>1.508210</td>\n",
       "      <td>0.980249</td>\n",
       "      <td>0.920990</td>\n",
       "      <td>1.028902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>0.587065</td>\n",
       "      <td>0.669167</td>\n",
       "      <td>0.801330</td>\n",
       "      <td>0.594580</td>\n",
       "      <td>0.992253</td>\n",
       "      <td>0.586154</td>\n",
       "      <td>1.169745</td>\n",
       "      <td>1.010447</td>\n",
       "      <td>0.996161</td>\n",
       "      <td>...</td>\n",
       "      <td>0.743053</td>\n",
       "      <td>1.398623</td>\n",
       "      <td>0.992872</td>\n",
       "      <td>0.994860</td>\n",
       "      <td>0.918612</td>\n",
       "      <td>1.093844</td>\n",
       "      <td>0.747614</td>\n",
       "      <td>1.497803</td>\n",
       "      <td>1.027683</td>\n",
       "      <td>0.820273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>1.515145</td>\n",
       "      <td>1.318432</td>\n",
       "      <td>1.175619</td>\n",
       "      <td>0.891041</td>\n",
       "      <td>1.002725</td>\n",
       "      <td>0.619833</td>\n",
       "      <td>1.561758</td>\n",
       "      <td>1.595148</td>\n",
       "      <td>1.002671</td>\n",
       "      <td>...</td>\n",
       "      <td>1.219133</td>\n",
       "      <td>1.436852</td>\n",
       "      <td>1.032843</td>\n",
       "      <td>1.009187</td>\n",
       "      <td>1.673403</td>\n",
       "      <td>1.075427</td>\n",
       "      <td>1.130588</td>\n",
       "      <td>1.013485</td>\n",
       "      <td>1.011336</td>\n",
       "      <td>1.254385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>0.912424</td>\n",
       "      <td>0.876892</td>\n",
       "      <td>0.815257</td>\n",
       "      <td>0.825214</td>\n",
       "      <td>0.996676</td>\n",
       "      <td>0.595849</td>\n",
       "      <td>0.828851</td>\n",
       "      <td>0.959989</td>\n",
       "      <td>0.984786</td>\n",
       "      <td>...</td>\n",
       "      <td>1.549725</td>\n",
       "      <td>1.356231</td>\n",
       "      <td>0.984896</td>\n",
       "      <td>0.996060</td>\n",
       "      <td>0.925511</td>\n",
       "      <td>1.429757</td>\n",
       "      <td>0.727942</td>\n",
       "      <td>1.388784</td>\n",
       "      <td>1.038347</td>\n",
       "      <td>1.020457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>0.755303</td>\n",
       "      <td>0.989199</td>\n",
       "      <td>1.148937</td>\n",
       "      <td>1.178981</td>\n",
       "      <td>1.006351</td>\n",
       "      <td>1.187420</td>\n",
       "      <td>1.183994</td>\n",
       "      <td>0.579938</td>\n",
       "      <td>0.999671</td>\n",
       "      <td>...</td>\n",
       "      <td>1.062868</td>\n",
       "      <td>1.280013</td>\n",
       "      <td>1.010769</td>\n",
       "      <td>1.007511</td>\n",
       "      <td>0.899186</td>\n",
       "      <td>1.372584</td>\n",
       "      <td>1.052046</td>\n",
       "      <td>1.001074</td>\n",
       "      <td>1.498685</td>\n",
       "      <td>1.403284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_GN86234</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_GN86235</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_GN86236</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000017  ACH-000022  ACH-000026  ACH-000028  \\\n",
       "0               BMGC_GN00001    1.085323    0.947735    0.827625    0.937181   \n",
       "1               BMGC_GN00002    0.587065    0.669167    0.801330    0.594580   \n",
       "2               BMGC_GN00003    1.515145    1.318432    1.175619    0.891041   \n",
       "3               BMGC_GN00004    0.912424    0.876892    0.815257    0.825214   \n",
       "4               BMGC_GN00005    0.755303    0.989199    1.148937    1.178981   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_GN86234    0.000000    0.000000    0.000000    0.000000   \n",
       "86234           BMGC_GN86235    0.000000    0.000000    0.000000    0.000000   \n",
       "86235           BMGC_GN86236    0.000000    0.000000    0.000000    0.000000   \n",
       "86236           BMGC_GN86237    0.000000    0.000000    0.000000    0.000000   \n",
       "86237           BMGC_GN86238    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-000063  ACH-000066  ACH-000085  ACH-000086  ACH-000119  ...  \\\n",
       "0        1.023235    1.504031    1.412782    1.047031    1.000624  ...   \n",
       "1        0.992253    0.586154    1.169745    1.010447    0.996161  ...   \n",
       "2        1.002725    0.619833    1.561758    1.595148    1.002671  ...   \n",
       "3        0.996676    0.595849    0.828851    0.959989    0.984786  ...   \n",
       "4        1.006351    1.187420    1.183994    0.579938    0.999671  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "       ACH-000844  ACH-000889  ACH-000942  ACH-000993  ACH-001001  ACH-001036  \\\n",
       "0        1.051902    1.073072    1.032968    1.011142    0.781099    1.181581   \n",
       "1        0.743053    1.398623    0.992872    0.994860    0.918612    1.093844   \n",
       "2        1.219133    1.436852    1.032843    1.009187    1.673403    1.075427   \n",
       "3        1.549725    1.356231    0.984896    0.996060    0.925511    1.429757   \n",
       "4        1.062868    1.280013    1.010769    1.007511    0.899186    1.372584   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-001098  ACH-001129  ACH-001188  ACH-001239  \n",
       "0        1.508210    0.980249    0.920990    1.028902  \n",
       "1        0.747614    1.497803    1.027683    0.820273  \n",
       "2        1.130588    1.013485    1.011336    1.254385  \n",
       "3        0.727942    1.388784    1.038347    1.020457  \n",
       "4        1.052046    1.001074    1.498685    1.403284  \n",
       "...           ...         ...         ...         ...  \n",
       "86233    0.000000    0.000000    0.000000    0.000000  \n",
       "86234    0.000000    0.000000    0.000000    0.000000  \n",
       "86235    0.000000    0.000000    0.000000    0.000000  \n",
       "86236    0.000000    0.000000    0.000000    0.000000  \n",
       "86237    0.000000    0.000000    0.000000    0.000000  \n",
       "\n",
       "[86238 rows x 68 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000017</th>\n",
       "      <th>ACH-000022</th>\n",
       "      <th>ACH-000026</th>\n",
       "      <th>ACH-000028</th>\n",
       "      <th>ACH-000063</th>\n",
       "      <th>ACH-000066</th>\n",
       "      <th>ACH-000085</th>\n",
       "      <th>ACH-000086</th>\n",
       "      <th>ACH-000119</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000844</th>\n",
       "      <th>ACH-000889</th>\n",
       "      <th>ACH-000942</th>\n",
       "      <th>ACH-000993</th>\n",
       "      <th>ACH-001001</th>\n",
       "      <th>ACH-001036</th>\n",
       "      <th>ACH-001098</th>\n",
       "      <th>ACH-001129</th>\n",
       "      <th>ACH-001188</th>\n",
       "      <th>ACH-001239</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>8.472621</td>\n",
       "      <td>7.566681</td>\n",
       "      <td>7.783121</td>\n",
       "      <td>7.873038</td>\n",
       "      <td>7.212185</td>\n",
       "      <td>6.909611</td>\n",
       "      <td>5.979267</td>\n",
       "      <td>7.439952</td>\n",
       "      <td>7.217133</td>\n",
       "      <td>...</td>\n",
       "      <td>7.797436</td>\n",
       "      <td>7.091215</td>\n",
       "      <td>7.324826</td>\n",
       "      <td>6.095544</td>\n",
       "      <td>6.173282</td>\n",
       "      <td>6.530392</td>\n",
       "      <td>7.358607</td>\n",
       "      <td>6.398834</td>\n",
       "      <td>7.977339</td>\n",
       "      <td>7.570824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>6.358975</td>\n",
       "      <td>5.671896</td>\n",
       "      <td>6.396962</td>\n",
       "      <td>6.018021</td>\n",
       "      <td>6.562234</td>\n",
       "      <td>6.994483</td>\n",
       "      <td>7.391298</td>\n",
       "      <td>6.544272</td>\n",
       "      <td>6.348819</td>\n",
       "      <td>...</td>\n",
       "      <td>6.839740</td>\n",
       "      <td>6.631997</td>\n",
       "      <td>6.592223</td>\n",
       "      <td>6.615116</td>\n",
       "      <td>6.490956</td>\n",
       "      <td>7.481551</td>\n",
       "      <td>7.472335</td>\n",
       "      <td>7.119044</td>\n",
       "      <td>6.128324</td>\n",
       "      <td>6.493308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>6.017817</td>\n",
       "      <td>5.144851</td>\n",
       "      <td>4.399958</td>\n",
       "      <td>5.303545</td>\n",
       "      <td>3.896775</td>\n",
       "      <td>4.751705</td>\n",
       "      <td>6.028516</td>\n",
       "      <td>3.780447</td>\n",
       "      <td>4.241893</td>\n",
       "      <td>...</td>\n",
       "      <td>5.980503</td>\n",
       "      <td>4.763219</td>\n",
       "      <td>3.556028</td>\n",
       "      <td>4.313112</td>\n",
       "      <td>4.900645</td>\n",
       "      <td>4.200231</td>\n",
       "      <td>4.966883</td>\n",
       "      <td>4.304928</td>\n",
       "      <td>3.319582</td>\n",
       "      <td>4.251047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>7.169858</td>\n",
       "      <td>4.858171</td>\n",
       "      <td>6.106998</td>\n",
       "      <td>6.679738</td>\n",
       "      <td>4.583381</td>\n",
       "      <td>5.762251</td>\n",
       "      <td>7.064322</td>\n",
       "      <td>5.614309</td>\n",
       "      <td>4.474898</td>\n",
       "      <td>...</td>\n",
       "      <td>6.525218</td>\n",
       "      <td>6.487419</td>\n",
       "      <td>6.429365</td>\n",
       "      <td>4.358429</td>\n",
       "      <td>6.319403</td>\n",
       "      <td>5.946550</td>\n",
       "      <td>5.796640</td>\n",
       "      <td>4.975769</td>\n",
       "      <td>6.356670</td>\n",
       "      <td>4.805903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>0.005892</td>\n",
       "      <td>2.260473</td>\n",
       "      <td>3.123043</td>\n",
       "      <td>0.451694</td>\n",
       "      <td>1.853420</td>\n",
       "      <td>0.175017</td>\n",
       "      <td>1.594867</td>\n",
       "      <td>1.618716</td>\n",
       "      <td>0.208682</td>\n",
       "      <td>...</td>\n",
       "      <td>0.175017</td>\n",
       "      <td>1.442286</td>\n",
       "      <td>0.152104</td>\n",
       "      <td>1.013721</td>\n",
       "      <td>1.133261</td>\n",
       "      <td>0.235919</td>\n",
       "      <td>2.935896</td>\n",
       "      <td>-0.059982</td>\n",
       "      <td>0.113665</td>\n",
       "      <td>0.252312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412034</th>\n",
       "      <td>BMGC_TS412035</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412035</th>\n",
       "      <td>BMGC_TS412036</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412036</th>\n",
       "      <td>BMGC_TS412037</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412037</th>\n",
       "      <td>BMGC_TS412038</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412038</th>\n",
       "      <td>BMGC_TS412039</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412039 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000017  ACH-000022  ACH-000026  ACH-000028  \\\n",
       "0               BMGC_TS000001    8.472621    7.566681    7.783121    7.873038   \n",
       "1               BMGC_TS000002    6.358975    5.671896    6.396962    6.018021   \n",
       "2               BMGC_TS000003    6.017817    5.144851    4.399958    5.303545   \n",
       "3               BMGC_TS000004    7.169858    4.858171    6.106998    6.679738   \n",
       "4               BMGC_TS000005    0.005892    2.260473    3.123043    0.451694   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "412034          BMGC_TS412035    0.000000    0.000000    0.000000    0.000000   \n",
       "412035          BMGC_TS412036    0.000000    0.000000    0.000000    0.000000   \n",
       "412036          BMGC_TS412037    0.000000    0.000000    0.000000    0.000000   \n",
       "412037          BMGC_TS412038    0.000000    0.000000    0.000000    0.000000   \n",
       "412038          BMGC_TS412039    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "        ACH-000063  ACH-000066  ACH-000085  ACH-000086  ACH-000119  ...  \\\n",
       "0         7.212185    6.909611    5.979267    7.439952    7.217133  ...   \n",
       "1         6.562234    6.994483    7.391298    6.544272    6.348819  ...   \n",
       "2         3.896775    4.751705    6.028516    3.780447    4.241893  ...   \n",
       "3         4.583381    5.762251    7.064322    5.614309    4.474898  ...   \n",
       "4         1.853420    0.175017    1.594867    1.618716    0.208682  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "412034    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412035    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412036    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412037    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412038    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "        ACH-000844  ACH-000889  ACH-000942  ACH-000993  ACH-001001  \\\n",
       "0         7.797436    7.091215    7.324826    6.095544    6.173282   \n",
       "1         6.839740    6.631997    6.592223    6.615116    6.490956   \n",
       "2         5.980503    4.763219    3.556028    4.313112    4.900645   \n",
       "3         6.525218    6.487419    6.429365    4.358429    6.319403   \n",
       "4         0.175017    1.442286    0.152104    1.013721    1.133261   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "412034    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "412035    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "412036    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "412037    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "412038    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "        ACH-001036  ACH-001098  ACH-001129  ACH-001188  ACH-001239  \n",
       "0         6.530392    7.358607    6.398834    7.977339    7.570824  \n",
       "1         7.481551    7.472335    7.119044    6.128324    6.493308  \n",
       "2         4.200231    4.966883    4.304928    3.319582    4.251047  \n",
       "3         5.946550    5.796640    4.975769    6.356670    4.805903  \n",
       "4         0.235919    2.935896   -0.059982    0.113665    0.252312  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "412034    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "412035    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "412036    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "412037    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "412038    0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "\n",
       "[412039 rows x 68 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>ACH-000016</th>\n",
       "      <th>ACH-000027</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000032</th>\n",
       "      <th>ACH-000037</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001113</th>\n",
       "      <th>ACH-001130</th>\n",
       "      <th>ACH-001145</th>\n",
       "      <th>ACH-001150</th>\n",
       "      <th>ACH-001162</th>\n",
       "      <th>ACH-001171</th>\n",
       "      <th>ACH-001183</th>\n",
       "      <th>ACH-001194</th>\n",
       "      <th>ACH-001318</th>\n",
       "      <th>ACH-001321</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>1.128141</td>\n",
       "      <td>1.067416</td>\n",
       "      <td>0.673066</td>\n",
       "      <td>1.131172</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.041115</td>\n",
       "      <td>0.838548</td>\n",
       "      <td>0.990240</td>\n",
       "      <td>1.078296</td>\n",
       "      <td>...</td>\n",
       "      <td>1.190599</td>\n",
       "      <td>0.967907</td>\n",
       "      <td>1.000632</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.536252</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.918424</td>\n",
       "      <td>1.039522</td>\n",
       "      <td>1.116916</td>\n",
       "      <td>1.042680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>1.112178</td>\n",
       "      <td>0.766166</td>\n",
       "      <td>0.892887</td>\n",
       "      <td>0.846887</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.054138</td>\n",
       "      <td>0.555273</td>\n",
       "      <td>0.977889</td>\n",
       "      <td>0.854899</td>\n",
       "      <td>...</td>\n",
       "      <td>0.611758</td>\n",
       "      <td>1.052372</td>\n",
       "      <td>1.033542</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.029018</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.527728</td>\n",
       "      <td>0.535935</td>\n",
       "      <td>0.796347</td>\n",
       "      <td>0.524765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>0.778284</td>\n",
       "      <td>1.122100</td>\n",
       "      <td>1.388424</td>\n",
       "      <td>1.668880</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.040417</td>\n",
       "      <td>1.237650</td>\n",
       "      <td>0.995680</td>\n",
       "      <td>1.530764</td>\n",
       "      <td>...</td>\n",
       "      <td>1.618570</td>\n",
       "      <td>1.045554</td>\n",
       "      <td>1.348982</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.017926</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.845727</td>\n",
       "      <td>1.309566</td>\n",
       "      <td>1.323527</td>\n",
       "      <td>1.039671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>1.123984</td>\n",
       "      <td>0.789885</td>\n",
       "      <td>0.748113</td>\n",
       "      <td>0.875289</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.541999</td>\n",
       "      <td>1.081287</td>\n",
       "      <td>0.973006</td>\n",
       "      <td>1.391903</td>\n",
       "      <td>...</td>\n",
       "      <td>1.099402</td>\n",
       "      <td>1.063535</td>\n",
       "      <td>0.681409</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.995275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.072174</td>\n",
       "      <td>0.853730</td>\n",
       "      <td>0.863507</td>\n",
       "      <td>1.010830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>1.116293</td>\n",
       "      <td>1.175879</td>\n",
       "      <td>1.057079</td>\n",
       "      <td>1.284645</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.060672</td>\n",
       "      <td>1.125485</td>\n",
       "      <td>1.470936</td>\n",
       "      <td>1.274312</td>\n",
       "      <td>...</td>\n",
       "      <td>1.442654</td>\n",
       "      <td>1.482049</td>\n",
       "      <td>1.035727</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.013036</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.035917</td>\n",
       "      <td>1.394770</td>\n",
       "      <td>1.211618</td>\n",
       "      <td>1.542534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_GN86234</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_GN86235</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_GN86236</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000005  ACH-000011  ACH-000013  ACH-000014  \\\n",
       "0               BMGC_GN00001    1.128141    1.067416    0.673066    1.131172   \n",
       "1               BMGC_GN00002    1.112178    0.766166    0.892887    0.846887   \n",
       "2               BMGC_GN00003    0.778284    1.122100    1.388424    1.668880   \n",
       "3               BMGC_GN00004    1.123984    0.789885    0.748113    0.875289   \n",
       "4               BMGC_GN00005    1.116293    1.175879    1.057079    1.284645   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_GN86234    0.000000    0.000000    0.000000    0.000000   \n",
       "86234           BMGC_GN86235    0.000000    0.000000    0.000000    0.000000   \n",
       "86235           BMGC_GN86236    0.000000    0.000000    0.000000    0.000000   \n",
       "86236           BMGC_GN86237    0.000000    0.000000    0.000000    0.000000   \n",
       "86237           BMGC_GN86238    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-000016  ACH-000027  ACH-000031  ACH-000032  ACH-000037  ...  \\\n",
       "0             0.0    1.041115    0.838548    0.990240    1.078296  ...   \n",
       "1             0.0    1.054138    0.555273    0.977889    0.854899  ...   \n",
       "2             0.0    1.040417    1.237650    0.995680    1.530764  ...   \n",
       "3             0.0    0.541999    1.081287    0.973006    1.391903  ...   \n",
       "4             0.0    1.060672    1.125485    1.470936    1.274312  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233         0.0    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86234         0.0    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86235         0.0    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86236         0.0    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86237         0.0    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "       ACH-001113  ACH-001130  ACH-001145  ACH-001150  ACH-001162  ACH-001171  \\\n",
       "0        1.190599    0.967907    1.000632         0.0    0.536252         0.0   \n",
       "1        0.611758    1.052372    1.033542         0.0    1.029018         0.0   \n",
       "2        1.618570    1.045554    1.348982         0.0    1.017926         0.0   \n",
       "3        1.099402    1.063535    0.681409         0.0    0.995275         0.0   \n",
       "4        1.442654    1.482049    1.035727         0.0    1.013036         0.0   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233    0.000000    0.000000    0.000000         0.0    0.000000         0.0   \n",
       "86234    0.000000    0.000000    0.000000         0.0    0.000000         0.0   \n",
       "86235    0.000000    0.000000    0.000000         0.0    0.000000         0.0   \n",
       "86236    0.000000    0.000000    0.000000         0.0    0.000000         0.0   \n",
       "86237    0.000000    0.000000    0.000000         0.0    0.000000         0.0   \n",
       "\n",
       "       ACH-001183  ACH-001194  ACH-001318  ACH-001321  \n",
       "0        0.918424    1.039522    1.116916    1.042680  \n",
       "1        0.527728    0.535935    0.796347    0.524765  \n",
       "2        1.845727    1.309566    1.323527    1.039671  \n",
       "3        1.072174    0.853730    0.863507    1.010830  \n",
       "4        1.035917    1.394770    1.211618    1.542534  \n",
       "...           ...         ...         ...         ...  \n",
       "86233    0.000000    0.000000    0.000000    0.000000  \n",
       "86234    0.000000    0.000000    0.000000    0.000000  \n",
       "86235    0.000000    0.000000    0.000000    0.000000  \n",
       "86236    0.000000    0.000000    0.000000    0.000000  \n",
       "86237    0.000000    0.000000    0.000000    0.000000  \n",
       "\n",
       "[86238 rows x 270 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>ACH-000016</th>\n",
       "      <th>ACH-000027</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000032</th>\n",
       "      <th>ACH-000037</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001113</th>\n",
       "      <th>ACH-001130</th>\n",
       "      <th>ACH-001145</th>\n",
       "      <th>ACH-001150</th>\n",
       "      <th>ACH-001162</th>\n",
       "      <th>ACH-001171</th>\n",
       "      <th>ACH-001183</th>\n",
       "      <th>ACH-001194</th>\n",
       "      <th>ACH-001318</th>\n",
       "      <th>ACH-001321</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>5.923191</td>\n",
       "      <td>7.457561</td>\n",
       "      <td>6.984308</td>\n",
       "      <td>7.446743</td>\n",
       "      <td>7.415651</td>\n",
       "      <td>8.098085</td>\n",
       "      <td>6.863893</td>\n",
       "      <td>7.156174</td>\n",
       "      <td>6.731636</td>\n",
       "      <td>...</td>\n",
       "      <td>7.230242</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.931638</td>\n",
       "      <td>6.872406</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.006642</td>\n",
       "      <td>7.253365</td>\n",
       "      <td>7.269873</td>\n",
       "      <td>7.552740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>7.492971</td>\n",
       "      <td>6.289055</td>\n",
       "      <td>6.297061</td>\n",
       "      <td>7.779837</td>\n",
       "      <td>6.180162</td>\n",
       "      <td>6.419696</td>\n",
       "      <td>7.167468</td>\n",
       "      <td>7.090760</td>\n",
       "      <td>6.580654</td>\n",
       "      <td>...</td>\n",
       "      <td>7.524420</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.138538</td>\n",
       "      <td>7.700997</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.815641</td>\n",
       "      <td>6.686938</td>\n",
       "      <td>7.468948</td>\n",
       "      <td>7.706810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>4.438237</td>\n",
       "      <td>3.481874</td>\n",
       "      <td>4.677355</td>\n",
       "      <td>4.323129</td>\n",
       "      <td>3.532014</td>\n",
       "      <td>5.170453</td>\n",
       "      <td>5.547268</td>\n",
       "      <td>4.040626</td>\n",
       "      <td>4.047278</td>\n",
       "      <td>...</td>\n",
       "      <td>5.232906</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.186650</td>\n",
       "      <td>4.237452</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.179700</td>\n",
       "      <td>4.215455</td>\n",
       "      <td>4.649932</td>\n",
       "      <td>5.481446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>7.091255</td>\n",
       "      <td>5.503175</td>\n",
       "      <td>6.458677</td>\n",
       "      <td>5.507597</td>\n",
       "      <td>4.635526</td>\n",
       "      <td>5.898858</td>\n",
       "      <td>5.635720</td>\n",
       "      <td>1.803751</td>\n",
       "      <td>5.111995</td>\n",
       "      <td>...</td>\n",
       "      <td>6.953174</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.281816</td>\n",
       "      <td>5.634324</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.525825</td>\n",
       "      <td>6.583393</td>\n",
       "      <td>6.262018</td>\n",
       "      <td>6.471680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>0.080957</td>\n",
       "      <td>1.473322</td>\n",
       "      <td>0.197552</td>\n",
       "      <td>0.442388</td>\n",
       "      <td>2.800069</td>\n",
       "      <td>2.903703</td>\n",
       "      <td>3.570329</td>\n",
       "      <td>-0.059982</td>\n",
       "      <td>3.119116</td>\n",
       "      <td>...</td>\n",
       "      <td>3.382773</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.600665</td>\n",
       "      <td>2.903310</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.336995</td>\n",
       "      <td>0.909155</td>\n",
       "      <td>1.032021</td>\n",
       "      <td>-0.046550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412034</th>\n",
       "      <td>BMGC_TS412035</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412035</th>\n",
       "      <td>BMGC_TS412036</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412036</th>\n",
       "      <td>BMGC_TS412037</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412037</th>\n",
       "      <td>BMGC_TS412038</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412038</th>\n",
       "      <td>BMGC_TS412039</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412039 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000005  ACH-000011  ACH-000013  ACH-000014  \\\n",
       "0               BMGC_TS000001    5.923191    7.457561    6.984308    7.446743   \n",
       "1               BMGC_TS000002    7.492971    6.289055    6.297061    7.779837   \n",
       "2               BMGC_TS000003    4.438237    3.481874    4.677355    4.323129   \n",
       "3               BMGC_TS000004    7.091255    5.503175    6.458677    5.507597   \n",
       "4               BMGC_TS000005    0.080957    1.473322    0.197552    0.442388   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "412034          BMGC_TS412035    0.000000    0.000000    0.000000    0.000000   \n",
       "412035          BMGC_TS412036    0.000000    0.000000    0.000000    0.000000   \n",
       "412036          BMGC_TS412037    0.000000    0.000000    0.000000    0.000000   \n",
       "412037          BMGC_TS412038    0.000000    0.000000    0.000000    0.000000   \n",
       "412038          BMGC_TS412039    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "        ACH-000016  ACH-000027  ACH-000031  ACH-000032  ACH-000037  ...  \\\n",
       "0         7.415651    8.098085    6.863893    7.156174    6.731636  ...   \n",
       "1         6.180162    6.419696    7.167468    7.090760    6.580654  ...   \n",
       "2         3.532014    5.170453    5.547268    4.040626    4.047278  ...   \n",
       "3         4.635526    5.898858    5.635720    1.803751    5.111995  ...   \n",
       "4         2.800069    2.903703    3.570329   -0.059982    3.119116  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "412034    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412035    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412036    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412037    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412038    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "        ACH-001113  ACH-001130  ACH-001145  ACH-001150  ACH-001162  \\\n",
       "0         7.230242         0.0    7.931638    6.872406         0.0   \n",
       "1         7.524420         0.0    7.138538    7.700997         0.0   \n",
       "2         5.232906         0.0    5.186650    4.237452         0.0   \n",
       "3         6.953174         0.0    7.281816    5.634324         0.0   \n",
       "4         3.382773         0.0    1.600665    2.903310         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "412034    0.000000         0.0    0.000000    0.000000         0.0   \n",
       "412035    0.000000         0.0    0.000000    0.000000         0.0   \n",
       "412036    0.000000         0.0    0.000000    0.000000         0.0   \n",
       "412037    0.000000         0.0    0.000000    0.000000         0.0   \n",
       "412038    0.000000         0.0    0.000000    0.000000         0.0   \n",
       "\n",
       "        ACH-001171  ACH-001183  ACH-001194  ACH-001318  ACH-001321  \n",
       "0              0.0    7.006642    7.253365    7.269873    7.552740  \n",
       "1              0.0    6.815641    6.686938    7.468948    7.706810  \n",
       "2              0.0    5.179700    4.215455    4.649932    5.481446  \n",
       "3              0.0    6.525825    6.583393    6.262018    6.471680  \n",
       "4              0.0    1.336995    0.909155    1.032021   -0.046550  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "412034         0.0    0.000000    0.000000    0.000000    0.000000  \n",
       "412035         0.0    0.000000    0.000000    0.000000    0.000000  \n",
       "412036         0.0    0.000000    0.000000    0.000000    0.000000  \n",
       "412037         0.0    0.000000    0.000000    0.000000    0.000000  \n",
       "412038         0.0    0.000000    0.000000    0.000000    0.000000  \n",
       "\n",
       "[412039 rows x 270 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# fill in the missing samples in the final_merged_gene_df for pretrain_plain_samples\n",
    "for sample in pretrain_plain_samples:\n",
    "    if sample not in final_merged_gene_df.columns:\n",
    "        final_merged_gene_df[sample] = 0.0\n",
    "# fill in the missing samples in the final_merged_transcript_df for pretrain_plain_samples\n",
    "for sample in pretrain_plain_samples:\n",
    "    if sample not in final_merged_transcript_df.columns:\n",
    "        final_merged_transcript_df[sample] = 0.0\n",
    "# get the final gene, transcript, drug dataframe by filtering the pretrain_plain_samples\n",
    "pretrain_plain_gene_df = final_merged_gene_df[['BioMedGraphica_Conn_ID'] + pretrain_plain_samples]\n",
    "display(pretrain_plain_gene_df)\n",
    "pretrain_plain_transcript_df = final_merged_transcript_df[['BioMedGraphica_Conn_ID'] + pretrain_plain_samples]\n",
    "display(pretrain_plain_transcript_df)\n",
    "\n",
    "# fill in the missing samples in the final_merged_gene_df for pretrain_status_samples\n",
    "for sample in pretrain_status_samples:\n",
    "    if sample not in final_merged_gene_df.columns:\n",
    "        final_merged_gene_df[sample] = 0.0\n",
    "# fill in the missing samples in the final_merged_transcript_df for pretrain_status_samples\n",
    "for sample in pretrain_status_samples:\n",
    "    if sample not in final_merged_transcript_df.columns:\n",
    "        final_merged_transcript_df[sample] = 0.0\n",
    "# get the final gene, transcript, drug dataframe by filtering the pretrain_status_samples\n",
    "pretrain_status_gene_df = final_merged_gene_df[['BioMedGraphica_Conn_ID'] + pretrain_status_samples]\n",
    "display(pretrain_status_gene_df)\n",
    "pretrain_status_transcript_df = final_merged_transcript_df[['BioMedGraphica_Conn_ID'] + pretrain_status_samples]\n",
    "display(pretrain_status_transcript_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000017</th>\n",
       "      <th>ACH-000022</th>\n",
       "      <th>ACH-000026</th>\n",
       "      <th>ACH-000028</th>\n",
       "      <th>ACH-000063</th>\n",
       "      <th>ACH-000066</th>\n",
       "      <th>ACH-000085</th>\n",
       "      <th>ACH-000086</th>\n",
       "      <th>ACH-000119</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000844</th>\n",
       "      <th>ACH-000889</th>\n",
       "      <th>ACH-000942</th>\n",
       "      <th>ACH-000993</th>\n",
       "      <th>ACH-001001</th>\n",
       "      <th>ACH-001036</th>\n",
       "      <th>ACH-001098</th>\n",
       "      <th>ACH-001129</th>\n",
       "      <th>ACH-001188</th>\n",
       "      <th>ACH-001239</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.50165</td>\n",
       "      <td>0.49325</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.84212</td>\n",
       "      <td>0.48636</td>\n",
       "      <td>0.65552</td>\n",
       "      <td>0.09454</td>\n",
       "      <td>0.69810</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.05468</td>\n",
       "      <td>0.86674</td>\n",
       "      <td>0.45000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.72600</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.80595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.00300</td>\n",
       "      <td>0.00649</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00141</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00943</td>\n",
       "      <td>0.00064</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.02074</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.01952</td>\n",
       "      <td>0.00237</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.03815</td>\n",
       "      <td>0.02166</td>\n",
       "      <td>0.03068</td>\n",
       "      <td>0.01391</td>\n",
       "      <td>0.01534</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.01450</td>\n",
       "      <td>0.87374</td>\n",
       "      <td>0.01292</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.26234</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705929</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705930</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705931</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705932</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705933</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>705934 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000017  ACH-000022  ACH-000026  ACH-000028  \\\n",
       "0                BMGC_PM00001     0.50165     0.49325         0.0     0.84212   \n",
       "1                BMGC_PM00002     0.00000     0.00000         0.0     0.00000   \n",
       "2                BMGC_PM00003     0.00300     0.00649         0.0     0.00141   \n",
       "3                BMGC_PM00004     0.01952     0.00237         0.0     0.03815   \n",
       "4                BMGC_PM00005     0.00000     0.00000         0.0     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "705929          BMGC_PT121415     0.00000     0.00000         0.0     0.00000   \n",
       "705930          BMGC_PT121416     0.00000     0.00000         0.0     0.00000   \n",
       "705931          BMGC_PT121417     0.00000     0.00000         0.0     0.00000   \n",
       "705932          BMGC_PT121418     0.00000     0.00000         0.0     0.00000   \n",
       "705933          BMGC_PT121419     0.00000     0.00000         0.0     0.00000   \n",
       "\n",
       "        ACH-000063  ACH-000066  ACH-000085  ACH-000086  ACH-000119  ...  \\\n",
       "0          0.48636     0.65552     0.09454     0.69810         0.0  ...   \n",
       "1          0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "2          0.00000     0.00000     0.00943     0.00064         0.0  ...   \n",
       "3          0.02166     0.03068     0.01391     0.01534         0.0  ...   \n",
       "4          0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "705929     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "705930     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "705931     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "705932     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "705933     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "        ACH-000844  ACH-000889  ACH-000942  ACH-000993  ACH-001001  \\\n",
       "0          0.05468     0.86674     0.45000         0.0         0.0   \n",
       "1          0.00000     0.00000     0.00000         0.0         0.0   \n",
       "2          0.02074     0.00000     0.00000         0.0         0.0   \n",
       "3          0.01450     0.87374     0.01292         0.0         0.0   \n",
       "4          0.00000     0.00000     0.00000         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "705929     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "705930     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "705931     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "705932     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "705933     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "\n",
       "        ACH-001036  ACH-001098  ACH-001129  ACH-001188  ACH-001239  \n",
       "0              0.0         0.0     0.72600         0.0     0.80595  \n",
       "1              0.0         0.0     0.00000         0.0     0.00000  \n",
       "2              0.0         0.0     0.00000         0.0     0.01409  \n",
       "3              0.0         0.0     0.26234         0.0     0.00195  \n",
       "4              0.0         0.0     0.00000         0.0     0.00000  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "705929         0.0         0.0     0.00000         0.0     0.00000  \n",
       "705930         0.0         0.0     0.00000         0.0     0.00000  \n",
       "705931         0.0         0.0     0.00000         0.0     0.00000  \n",
       "705932         0.0         0.0     0.00000         0.0     0.00000  \n",
       "705933         0.0         0.0     0.00000         0.0     0.00000  \n",
       "\n",
       "[705934 rows x 68 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000017</th>\n",
       "      <th>ACH-000022</th>\n",
       "      <th>ACH-000026</th>\n",
       "      <th>ACH-000028</th>\n",
       "      <th>ACH-000063</th>\n",
       "      <th>ACH-000066</th>\n",
       "      <th>ACH-000085</th>\n",
       "      <th>ACH-000086</th>\n",
       "      <th>ACH-000119</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000844</th>\n",
       "      <th>ACH-000889</th>\n",
       "      <th>ACH-000942</th>\n",
       "      <th>ACH-000993</th>\n",
       "      <th>ACH-001001</th>\n",
       "      <th>ACH-001036</th>\n",
       "      <th>ACH-001098</th>\n",
       "      <th>ACH-001129</th>\n",
       "      <th>ACH-001188</th>\n",
       "      <th>ACH-001239</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.50165</td>\n",
       "      <td>0.49325</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.84212</td>\n",
       "      <td>0.48636</td>\n",
       "      <td>0.65552</td>\n",
       "      <td>0.09454</td>\n",
       "      <td>0.69810</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.05468</td>\n",
       "      <td>0.86674</td>\n",
       "      <td>0.45000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.72600</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.80595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.00300</td>\n",
       "      <td>0.00649</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00141</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00943</td>\n",
       "      <td>0.00064</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.02074</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.01952</td>\n",
       "      <td>0.00237</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.03815</td>\n",
       "      <td>0.02166</td>\n",
       "      <td>0.03068</td>\n",
       "      <td>0.01391</td>\n",
       "      <td>0.01534</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.01450</td>\n",
       "      <td>0.87374</td>\n",
       "      <td>0.01292</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.26234</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000017  ACH-000022  ACH-000026  ACH-000028  \\\n",
       "0                BMGC_PM00001     0.50165     0.49325         0.0     0.84212   \n",
       "1                BMGC_PM00002     0.00000     0.00000         0.0     0.00000   \n",
       "2                BMGC_PM00003     0.00300     0.00649         0.0     0.00141   \n",
       "3                BMGC_PM00004     0.01952     0.00237         0.0     0.03815   \n",
       "4                BMGC_PM00005     0.00000     0.00000         0.0     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "834804           BMGC_DG20914     0.00000     0.00000         0.0     0.00000   \n",
       "834805           BMGC_DG20915     0.00000     0.00000         0.0     0.00000   \n",
       "834806           BMGC_DG20916     0.00000     0.00000         0.0     0.00000   \n",
       "834807           BMGC_DG20917     0.00000     0.00000         0.0     0.00000   \n",
       "834808           BMGC_DG20918     0.00000     0.00000         0.0     0.00000   \n",
       "\n",
       "        ACH-000063  ACH-000066  ACH-000085  ACH-000086  ACH-000119  ...  \\\n",
       "0          0.48636     0.65552     0.09454     0.69810         0.0  ...   \n",
       "1          0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "2          0.00000     0.00000     0.00943     0.00064         0.0  ...   \n",
       "3          0.02166     0.03068     0.01391     0.01534         0.0  ...   \n",
       "4          0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "834804     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "834805     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "834806     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "834807     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "834808     0.00000     0.00000     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "        ACH-000844  ACH-000889  ACH-000942  ACH-000993  ACH-001001  \\\n",
       "0          0.05468     0.86674     0.45000         0.0         0.0   \n",
       "1          0.00000     0.00000     0.00000         0.0         0.0   \n",
       "2          0.02074     0.00000     0.00000         0.0         0.0   \n",
       "3          0.01450     0.87374     0.01292         0.0         0.0   \n",
       "4          0.00000     0.00000     0.00000         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "834804     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "834805     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "834806     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "834807     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "834808     0.00000     0.00000     0.00000         0.0         0.0   \n",
       "\n",
       "        ACH-001036  ACH-001098  ACH-001129  ACH-001188  ACH-001239  \n",
       "0              0.0         0.0     0.72600         0.0     0.80595  \n",
       "1              0.0         0.0     0.00000         0.0     0.00000  \n",
       "2              0.0         0.0     0.00000         0.0     0.01409  \n",
       "3              0.0         0.0     0.26234         0.0     0.00195  \n",
       "4              0.0         0.0     0.00000         0.0     0.00000  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "834804         0.0         0.0     0.00000         0.0     0.00000  \n",
       "834805         0.0         0.0     0.00000         0.0     0.00000  \n",
       "834806         0.0         0.0     0.00000         0.0     0.00000  \n",
       "834807         0.0         0.0     0.00000         0.0     0.00000  \n",
       "834808         0.0         0.0     0.00000         0.0     0.00000  \n",
       "\n",
       "[834809 rows x 68 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>ACH-000016</th>\n",
       "      <th>ACH-000027</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000032</th>\n",
       "      <th>ACH-000037</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001113</th>\n",
       "      <th>ACH-001130</th>\n",
       "      <th>ACH-001145</th>\n",
       "      <th>ACH-001150</th>\n",
       "      <th>ACH-001162</th>\n",
       "      <th>ACH-001171</th>\n",
       "      <th>ACH-001183</th>\n",
       "      <th>ACH-001194</th>\n",
       "      <th>ACH-001318</th>\n",
       "      <th>ACH-001321</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.70874</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.66057</td>\n",
       "      <td>0.88330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.51282</td>\n",
       "      <td>0.12250</td>\n",
       "      <td>0.70000</td>\n",
       "      <td>0.15385</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.47703</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.42555</td>\n",
       "      <td>0.47785</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50254</td>\n",
       "      <td>0.63336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.06977</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00507</td>\n",
       "      <td>0.01536</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00260</td>\n",
       "      <td>0.00351</td>\n",
       "      <td>0.01941</td>\n",
       "      <td>0.01643</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00206</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01197</td>\n",
       "      <td>0.00757</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00453</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.59602</td>\n",
       "      <td>0.01294</td>\n",
       "      <td>0.04124</td>\n",
       "      <td>0.01022</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01732</td>\n",
       "      <td>0.00056</td>\n",
       "      <td>0.55973</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05561</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01421</td>\n",
       "      <td>0.73156</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01198</td>\n",
       "      <td>0.01604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705929</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705930</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705931</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705932</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705933</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>705934 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000005  ACH-000011  ACH-000013  ACH-000014  \\\n",
       "0                BMGC_PM00001     0.70874     1.00000     0.66057     0.88330   \n",
       "1                BMGC_PM00002     0.00000     0.00000     0.00000     0.00000   \n",
       "2                BMGC_PM00003     0.06977     0.00000     0.00507     0.01536   \n",
       "3                BMGC_PM00004     0.59602     0.01294     0.04124     0.01022   \n",
       "4                BMGC_PM00005     0.00000     0.00000     0.00000     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "705929          BMGC_PT121415     0.00000     0.00000     0.00000     0.00000   \n",
       "705930          BMGC_PT121416     0.00000     0.00000     0.00000     0.00000   \n",
       "705931          BMGC_PT121417     0.00000     0.00000     0.00000     0.00000   \n",
       "705932          BMGC_PT121418     0.00000     0.00000     0.00000     0.00000   \n",
       "705933          BMGC_PT121419     0.00000     0.00000     0.00000     0.00000   \n",
       "\n",
       "        ACH-000016  ACH-000027  ACH-000031  ACH-000032  ACH-000037  ...  \\\n",
       "0              0.0     0.51282     0.12250     0.70000     0.15385  ...   \n",
       "1              0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "2              0.0     0.00260     0.00351     0.01941     0.01643  ...   \n",
       "3              0.0     0.01732     0.00056     0.55973     0.00000  ...   \n",
       "4              0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "705929         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "705930         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "705931         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "705932         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "705933         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "\n",
       "        ACH-001113  ACH-001130  ACH-001145  ACH-001150  ACH-001162  \\\n",
       "0              0.0     0.47703         0.0     0.42555     0.47785   \n",
       "1              0.0     0.00000         0.0     0.00000     0.00000   \n",
       "2              0.0     0.00206         0.0     0.01197     0.00757   \n",
       "3              0.0     0.05561         0.0     0.01421     0.73156   \n",
       "4              0.0     0.00000         0.0     0.00000     0.00000   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "705929         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "705930         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "705931         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "705932         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "705933         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "\n",
       "        ACH-001171  ACH-001183  ACH-001194  ACH-001318  ACH-001321  \n",
       "0              0.0         0.0         0.0     0.50254     0.63336  \n",
       "1              0.0         0.0         0.0     0.00000     0.00000  \n",
       "2              0.0         0.0         0.0     0.00453     0.00000  \n",
       "3              0.0         0.0         0.0     0.01198     0.01604  \n",
       "4              0.0         0.0         0.0     0.00000     0.00000  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "705929         0.0         0.0         0.0     0.00000     0.00000  \n",
       "705930         0.0         0.0         0.0     0.00000     0.00000  \n",
       "705931         0.0         0.0         0.0     0.00000     0.00000  \n",
       "705932         0.0         0.0         0.0     0.00000     0.00000  \n",
       "705933         0.0         0.0         0.0     0.00000     0.00000  \n",
       "\n",
       "[705934 rows x 270 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>ACH-000016</th>\n",
       "      <th>ACH-000027</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000032</th>\n",
       "      <th>ACH-000037</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001113</th>\n",
       "      <th>ACH-001130</th>\n",
       "      <th>ACH-001145</th>\n",
       "      <th>ACH-001150</th>\n",
       "      <th>ACH-001162</th>\n",
       "      <th>ACH-001171</th>\n",
       "      <th>ACH-001183</th>\n",
       "      <th>ACH-001194</th>\n",
       "      <th>ACH-001318</th>\n",
       "      <th>ACH-001321</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.70874</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.66057</td>\n",
       "      <td>0.88330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.51282</td>\n",
       "      <td>0.12250</td>\n",
       "      <td>0.70000</td>\n",
       "      <td>0.15385</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.47703</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.42555</td>\n",
       "      <td>0.47785</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50254</td>\n",
       "      <td>0.63336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.06977</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00507</td>\n",
       "      <td>0.01536</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00260</td>\n",
       "      <td>0.00351</td>\n",
       "      <td>0.01941</td>\n",
       "      <td>0.01643</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00206</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01197</td>\n",
       "      <td>0.00757</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00453</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.59602</td>\n",
       "      <td>0.01294</td>\n",
       "      <td>0.04124</td>\n",
       "      <td>0.01022</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01732</td>\n",
       "      <td>0.00056</td>\n",
       "      <td>0.55973</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05561</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01421</td>\n",
       "      <td>0.73156</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01198</td>\n",
       "      <td>0.01604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000005  ACH-000011  ACH-000013  ACH-000014  \\\n",
       "0                BMGC_PM00001     0.70874     1.00000     0.66057     0.88330   \n",
       "1                BMGC_PM00002     0.00000     0.00000     0.00000     0.00000   \n",
       "2                BMGC_PM00003     0.06977     0.00000     0.00507     0.01536   \n",
       "3                BMGC_PM00004     0.59602     0.01294     0.04124     0.01022   \n",
       "4                BMGC_PM00005     0.00000     0.00000     0.00000     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "834804           BMGC_DG20914     0.00000     0.00000     0.00000     0.00000   \n",
       "834805           BMGC_DG20915     0.00000     0.00000     0.00000     0.00000   \n",
       "834806           BMGC_DG20916     0.00000     0.00000     0.00000     0.00000   \n",
       "834807           BMGC_DG20917     0.00000     0.00000     0.00000     0.00000   \n",
       "834808           BMGC_DG20918     0.00000     0.00000     0.00000     0.00000   \n",
       "\n",
       "        ACH-000016  ACH-000027  ACH-000031  ACH-000032  ACH-000037  ...  \\\n",
       "0              0.0     0.51282     0.12250     0.70000     0.15385  ...   \n",
       "1              0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "2              0.0     0.00260     0.00351     0.01941     0.01643  ...   \n",
       "3              0.0     0.01732     0.00056     0.55973     0.00000  ...   \n",
       "4              0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "834804         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "834805         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "834806         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "834807         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "834808         0.0     0.00000     0.00000     0.00000     0.00000  ...   \n",
       "\n",
       "        ACH-001113  ACH-001130  ACH-001145  ACH-001150  ACH-001162  \\\n",
       "0              0.0     0.47703         0.0     0.42555     0.47785   \n",
       "1              0.0     0.00000         0.0     0.00000     0.00000   \n",
       "2              0.0     0.00206         0.0     0.01197     0.00757   \n",
       "3              0.0     0.05561         0.0     0.01421     0.73156   \n",
       "4              0.0     0.00000         0.0     0.00000     0.00000   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "834804         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "834805         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "834806         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "834807         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "834808         0.0     0.00000         0.0     0.00000     0.00000   \n",
       "\n",
       "        ACH-001171  ACH-001183  ACH-001194  ACH-001318  ACH-001321  \n",
       "0              0.0         0.0         0.0     0.50254     0.63336  \n",
       "1              0.0         0.0         0.0     0.00000     0.00000  \n",
       "2              0.0         0.0         0.0     0.00453     0.00000  \n",
       "3              0.0         0.0         0.0     0.01198     0.01604  \n",
       "4              0.0         0.0         0.0     0.00000     0.00000  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "834804         0.0         0.0         0.0     0.00000     0.00000  \n",
       "834805         0.0         0.0         0.0     0.00000     0.00000  \n",
       "834806         0.0         0.0         0.0     0.00000     0.00000  \n",
       "834807         0.0         0.0         0.0     0.00000     0.00000  \n",
       "834808         0.0         0.0         0.0     0.00000     0.00000  \n",
       "\n",
       "[834809 rows x 270 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pretrain_plain_omics_df = pd.concat([pretrain_plain_methy_df, pretrain_plain_gene_df, pretrain_plain_transcript_df, pretrain_plain_protein_df], axis=0).reset_index(drop=True)\n",
    "display(pretrain_plain_omics_df)\n",
    "pretrain_plain_feat_df = pd.merge(bmgc_entity_df, pretrain_plain_omics_df, left_on='BioMedGraphica_Conn_ID', right_on='BioMedGraphica_Conn_ID', how='left').drop(columns=['BioMedGraphica_ID', 'Type'])\n",
    "pretrain_plain_feat_df = pretrain_plain_feat_df.fillna(0.0)\n",
    "display(pretrain_plain_feat_df)\n",
    "pretrain_status_omics_df = pd.concat([pretrain_status_methy_df, pretrain_status_gene_df, pretrain_status_transcript_df, pretrain_status_protein_df], axis=0).reset_index(drop=True)\n",
    "display(pretrain_status_omics_df)\n",
    "pretrain_status_feat_df = pd.merge(bmgc_entity_df, pretrain_status_omics_df, left_on='BioMedGraphica_Conn_ID', right_on='BioMedGraphica_Conn_ID', how='left').drop(columns=['BioMedGraphica_ID', 'Type'])\n",
    "pretrain_status_feat_df = pretrain_status_feat_df.fillna(0.0)\n",
    "display(pretrain_status_feat_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of pretrain_plain_array: (67, 834809)\n",
      "Shape of pretrain_status_array: (269, 834809)\n"
     ]
    }
   ],
   "source": [
    "# convert the pretrain_plain_df and pretrain_status_df to numpy arrays and transpose them\n",
    "pretrain_plain_array = pretrain_plain_feat_df.drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "pretrain_status_array = pretrain_status_feat_df.drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "print(\"Shape of pretrain_plain_array:\", pretrain_plain_array.shape)\n",
    "print(\"Shape of pretrain_status_array:\", pretrain_status_array.shape)\n",
    "# Save the numpy arrays to .npy files\n",
    "np.save('./data/pretrain_plain_data/pretrain_plain_feature.npy', pretrain_plain_array)\n",
    "np.save('./data/pretrain_status_data/pretrain_status_feature.npy', pretrain_status_array)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 9.2.1.1 Pretrain Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of pretrain_status_label: (269,)\n",
      "Shape of pretrain_status_train_feat: (215, 834809)\n",
      "Shape of pretrain_status_test_feat: (54, 834809)\n",
      "Shape of pretrain_status_train_label: (215,)\n",
      "Shape of pretrain_status_test_label: (54,)\n",
      "\n",
      "📈 Pretrain Status Training Set Distribution (Numpy Files):\n",
      "Cancerous: 190, Non-cancerous: 25\n",
      "📈 Pretrain Status Test Set Distribution (Numpy Files):\n",
      "Cancerous: 48, Non-cancerous: 6\n"
     ]
    }
   ],
   "source": [
    "# convert pretrain_status_df to numpy array label with  cancerous as 1 and non-cancerous as 0\n",
    "pretrain_status_label = np.array([1 if status == 'cancerous' else 0 for status in pretrain_status_df['cancerous_status']])\n",
    "# Save the pretrain_status_label to a .npy file\n",
    "np.save('./data/pretrain_status_data/pretrain_status_label.npy', pretrain_status_label)\n",
    "print(\"Shape of pretrain_status_label:\", pretrain_status_label.shape)\n",
    "\n",
    "# assign the label by pretrain_status_training_samples and pretrain_status_test_samples\n",
    "pretrain_status_train_label = np.array([1 if status == 'cancerous' else 0 for status in pretrain_status_train_df['cancerous_status']])\n",
    "pretrain_status_test_label = np.array([1 if status == 'cancerous' else 0 for status in pretrain_status_test_df['cancerous_status']])\n",
    "# assign the feat by pretrain_status_training_samples and pretrain_status_test_samples\n",
    "pretrain_status_train_feat = pretrain_status_feat_df[['BioMedGraphica_Conn_ID'] + pretrain_status_train_samples].drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "pretrain_status_test_feat = pretrain_status_feat_df[['BioMedGraphica_Conn_ID'] + pretrain_status_test_samples].drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "\n",
    "# Save the training and test features and labels to .npy files\n",
    "np.save('./data/pretrain_status_data/pretrain_status_train_feature.npy', pretrain_status_train_feat)\n",
    "np.save('./data/pretrain_status_data/pretrain_status_test_feature.npy', pretrain_status_test_feat)\n",
    "np.save('./data/pretrain_status_data/pretrain_status_train_label.npy', pretrain_status_train_label)\n",
    "np.save('./data/pretrain_status_data/pretrain_status_test_label.npy', pretrain_status_test_label)\n",
    "print(\"Shape of pretrain_status_train_feat:\", pretrain_status_train_feat.shape)\n",
    "print(\"Shape of pretrain_status_test_feat:\", pretrain_status_test_feat.shape)\n",
    "print(\"Shape of pretrain_status_train_label:\", pretrain_status_train_label.shape)\n",
    "print(\"Shape of pretrain_status_test_label:\", pretrain_status_test_label.shape)\n",
    "# Print the cancerous/non-cancerous distribution in training and test sets\n",
    "print(f\"\\n📈 Pretrain Status Training Set Distribution (Numpy Files):\\n\"\n",
    "      f\"Cancerous: {np.sum(pretrain_status_train_label == 1)}, Non-cancerous: {np.sum(pretrain_status_train_label == 0)}\")\n",
    "print(f\"📈 Pretrain Status Test Set Distribution (Numpy Files):\\n\"\n",
    "      f\"Cancerous: {np.sum(pretrain_status_test_label == 1)}, Non-cancerous: {np.sum(pretrain_status_test_label == 0)}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 9.2.2 Target data integration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 404,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The overlapped samples dataframe for methylation between overlapped_dti_crispr_rna_samples:\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  maped_methy_df[sample] = 0.0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000015</th>\n",
       "      <th>ACH-000018</th>\n",
       "      <th>ACH-000019</th>\n",
       "      <th>ACH-000021</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000988</th>\n",
       "      <th>ACH-000991</th>\n",
       "      <th>ACH-000997</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001065</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.77900</td>\n",
       "      <td>0.07045</td>\n",
       "      <td>0.92549</td>\n",
       "      <td>0.93426</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.30908</td>\n",
       "      <td>0.81250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.29050</td>\n",
       "      <td>0.73685</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.70055</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.46825</td>\n",
       "      <td>0.67335</td>\n",
       "      <td>0.6964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.05796</td>\n",
       "      <td>0.00686</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>0.00286</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00999</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00433</td>\n",
       "      <td>0.02217</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01297</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01149</td>\n",
       "      <td>0.00573</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.02149</td>\n",
       "      <td>0.59846</td>\n",
       "      <td>0.88734</td>\n",
       "      <td>0.94058</td>\n",
       "      <td>0.44256</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02831</td>\n",
       "      <td>0.03987</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.32492</td>\n",
       "      <td>0.90407</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.11109</td>\n",
       "      <td>0.00307</td>\n",
       "      <td>0.0058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_PM86234</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_PM86235</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_PM86236</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 364 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000001  ACH-000004  ACH-000007  ACH-000009  \\\n",
       "0               BMGC_PM00001     1.00000     0.77900     0.07045     0.92549   \n",
       "1               BMGC_PM00002     0.00000     0.00000     0.00000     0.00000   \n",
       "2               BMGC_PM00003     0.00000     0.05796     0.00686     0.00112   \n",
       "3               BMGC_PM00004     0.02149     0.59846     0.88734     0.94058   \n",
       "4               BMGC_PM00005     0.00000     0.00000     0.00000     0.00000   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_PM86234     0.00000     0.00000     0.00000     0.00000   \n",
       "86234           BMGC_PM86235     0.00000     0.00000     0.00000     0.00000   \n",
       "86235           BMGC_PM86236     0.00000     0.00000     0.00000     0.00000   \n",
       "86236           BMGC_PM86237     0.00000     0.00000     0.00000     0.00000   \n",
       "86237           BMGC_PM86238     0.00000     0.00000     0.00000     0.00000   \n",
       "\n",
       "       ACH-000012  ACH-000015  ACH-000018  ACH-000019  ACH-000021  ...  \\\n",
       "0         0.93426         0.0     0.30908     0.81250         0.0  ...   \n",
       "1         0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "2         0.00286         0.0     0.00000     0.00999         0.0  ...   \n",
       "3         0.44256         0.0     0.02831     0.03987         0.0  ...   \n",
       "4         0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86234     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86235     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86236     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86237     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "       ACH-000988  ACH-000991  ACH-000997  ACH-001039  ACH-001065  ACH-001075  \\\n",
       "0             0.0     0.29050     0.73685         0.0         0.0     0.70055   \n",
       "1             0.0     0.00000     0.00000         0.0         0.0     0.00000   \n",
       "2             0.0     0.00433     0.02217         0.0         0.0     0.01297   \n",
       "3             0.0     0.32492     0.90407         0.0         0.0     0.00000   \n",
       "4             0.0     0.00000     0.00000         0.0         0.0     0.00000   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233         0.0     0.00000     0.00000         0.0         0.0     0.00000   \n",
       "86234         0.0     0.00000     0.00000         0.0         0.0     0.00000   \n",
       "86235         0.0     0.00000     0.00000         0.0         0.0     0.00000   \n",
       "86236         0.0     0.00000     0.00000         0.0         0.0     0.00000   \n",
       "86237         0.0     0.00000     0.00000         0.0         0.0     0.00000   \n",
       "\n",
       "       ACH-001107  ACH-001190  ACH-001306  ACH-001307  \n",
       "0             0.0     0.46825     0.67335      0.6964  \n",
       "1             0.0     0.00000     0.00000      0.0000  \n",
       "2             0.0     0.01149     0.00573      0.0000  \n",
       "3             0.0     0.11109     0.00307      0.0058  \n",
       "4             0.0     0.00000     0.00000      0.0000  \n",
       "...           ...         ...         ...         ...  \n",
       "86233         0.0     0.00000     0.00000      0.0000  \n",
       "86234         0.0     0.00000     0.00000      0.0000  \n",
       "86235         0.0     0.00000     0.00000      0.0000  \n",
       "86236         0.0     0.00000     0.00000      0.0000  \n",
       "86237         0.0     0.00000     0.00000      0.0000  \n",
       "\n",
       "[86238 rows x 364 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The overlapped samples dataframe for protein between overlapped_dti_crispr_rna_samples are:\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\823047421.py:14: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000015</th>\n",
       "      <th>ACH-000018</th>\n",
       "      <th>ACH-000019</th>\n",
       "      <th>ACH-000021</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000988</th>\n",
       "      <th>ACH-000991</th>\n",
       "      <th>ACH-000997</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001065</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 364 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000001  ACH-000004  ACH-000007  ACH-000009  \\\n",
       "0               BMGC_PT000001         0.0         0.0         0.0         0.0   \n",
       "1               BMGC_PT000002         0.0         0.0         0.0         0.0   \n",
       "2               BMGC_PT000003         0.0         0.0         0.0         0.0   \n",
       "3               BMGC_PT000004         0.0         0.0         0.0         0.0   \n",
       "4               BMGC_PT000005         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "121414          BMGC_PT121415         0.0         0.0         0.0         0.0   \n",
       "121415          BMGC_PT121416         0.0         0.0         0.0         0.0   \n",
       "121416          BMGC_PT121417         0.0         0.0         0.0         0.0   \n",
       "121417          BMGC_PT121418         0.0         0.0         0.0         0.0   \n",
       "121418          BMGC_PT121419         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-000012  ACH-000015  ACH-000018  ACH-000019  ACH-000021  ...  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "1              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "2              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "3              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "4              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121415         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121416         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121417         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121418         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "\n",
       "        ACH-000988  ACH-000991  ACH-000997  ACH-001039  ACH-001065  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0   \n",
       "1              0.0         0.0         0.0         0.0         0.0   \n",
       "2              0.0         0.0         0.0         0.0         0.0   \n",
       "3              0.0         0.0         0.0         0.0         0.0   \n",
       "4              0.0         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0   \n",
       "121415         0.0         0.0         0.0         0.0         0.0   \n",
       "121416         0.0         0.0         0.0         0.0         0.0   \n",
       "121417         0.0         0.0         0.0         0.0         0.0   \n",
       "121418         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001075  ACH-001107  ACH-001190  ACH-001306  ACH-001307  \n",
       "0              0.0         0.0         0.0         0.0         0.0  \n",
       "1              0.0         0.0         0.0         0.0         0.0  \n",
       "2              0.0         0.0         0.0         0.0         0.0  \n",
       "3              0.0         0.0         0.0         0.0         0.0  \n",
       "4              0.0         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "121414         0.0         0.0         0.0         0.0         0.0  \n",
       "121415         0.0         0.0         0.0         0.0         0.0  \n",
       "121416         0.0         0.0         0.0         0.0         0.0  \n",
       "121417         0.0         0.0         0.0         0.0         0.0  \n",
       "121418         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[121419 rows x 364 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(\"The overlapped samples dataframe for methylation between overlapped_dti_crispr_rna_samples:\")\n",
    "# insert the samples that is in the overlapped_samples but not in the maped_methy_df\n",
    "for sample in overlapped_dti_crispr_rna_samples:\n",
    "    if sample not in maped_methy_df.columns:\n",
    "        maped_methy_df[sample] = 0.0\n",
    "# filter out the samples that is not in the overlapped_dti_crispr_rna_samples\n",
    "target_crispr_methy_df = maped_methy_df[['BioMedGraphica_Conn_ID'] + overlapped_dti_crispr_rna_samples]\n",
    "display(target_crispr_methy_df)\n",
    "\n",
    "print(\"The overlapped samples dataframe for protein between overlapped_dti_crispr_rna_samples are:\")\n",
    "# insert the samples that is in the pretraining_samples but not in the mapped_protein_df\n",
    "for sample in overlapped_dti_crispr_rna_samples:\n",
    "    if sample not in mapped_protein_df.columns:\n",
    "        mapped_protein_df[sample] = 0.0\n",
    "# filter out the samples that is not in the overlapped_dti_crispr_rna_samples\n",
    "target_crispr_protein_df = mapped_protein_df[['BioMedGraphica_Conn_ID'] + overlapped_dti_crispr_rna_samples]\n",
    "display(target_crispr_protein_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 405,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000015</th>\n",
       "      <th>ACH-000018</th>\n",
       "      <th>ACH-000019</th>\n",
       "      <th>ACH-000021</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000988</th>\n",
       "      <th>ACH-000991</th>\n",
       "      <th>ACH-000997</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001065</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>0.851056</td>\n",
       "      <td>1.323804</td>\n",
       "      <td>0.937570</td>\n",
       "      <td>0.980206</td>\n",
       "      <td>1.111580</td>\n",
       "      <td>0.660613</td>\n",
       "      <td>0.917698</td>\n",
       "      <td>0.956259</td>\n",
       "      <td>1.020936</td>\n",
       "      <td>...</td>\n",
       "      <td>0.937035</td>\n",
       "      <td>0.967644</td>\n",
       "      <td>0.958509</td>\n",
       "      <td>0.993997</td>\n",
       "      <td>1.091848</td>\n",
       "      <td>1.418260</td>\n",
       "      <td>1.656834</td>\n",
       "      <td>0.873136</td>\n",
       "      <td>1.330477</td>\n",
       "      <td>0.977833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>1.157653</td>\n",
       "      <td>1.072377</td>\n",
       "      <td>0.941289</td>\n",
       "      <td>0.972136</td>\n",
       "      <td>0.776816</td>\n",
       "      <td>0.557800</td>\n",
       "      <td>0.881935</td>\n",
       "      <td>0.551096</td>\n",
       "      <td>1.118855</td>\n",
       "      <td>...</td>\n",
       "      <td>1.052398</td>\n",
       "      <td>1.000457</td>\n",
       "      <td>0.992196</td>\n",
       "      <td>0.589237</td>\n",
       "      <td>0.694145</td>\n",
       "      <td>0.587443</td>\n",
       "      <td>0.414771</td>\n",
       "      <td>0.968923</td>\n",
       "      <td>0.697147</td>\n",
       "      <td>0.778931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>1.498803</td>\n",
       "      <td>0.777885</td>\n",
       "      <td>0.958520</td>\n",
       "      <td>2.383638</td>\n",
       "      <td>1.209065</td>\n",
       "      <td>1.156748</td>\n",
       "      <td>2.014502</td>\n",
       "      <td>1.279922</td>\n",
       "      <td>1.189335</td>\n",
       "      <td>...</td>\n",
       "      <td>1.069887</td>\n",
       "      <td>0.974388</td>\n",
       "      <td>1.002760</td>\n",
       "      <td>1.457635</td>\n",
       "      <td>0.676150</td>\n",
       "      <td>1.564420</td>\n",
       "      <td>1.091653</td>\n",
       "      <td>1.433025</td>\n",
       "      <td>1.692643</td>\n",
       "      <td>1.575951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>0.669074</td>\n",
       "      <td>1.064669</td>\n",
       "      <td>0.942657</td>\n",
       "      <td>0.562637</td>\n",
       "      <td>0.776307</td>\n",
       "      <td>0.813608</td>\n",
       "      <td>0.902359</td>\n",
       "      <td>0.835197</td>\n",
       "      <td>1.514296</td>\n",
       "      <td>...</td>\n",
       "      <td>1.086093</td>\n",
       "      <td>1.011928</td>\n",
       "      <td>0.997424</td>\n",
       "      <td>0.630750</td>\n",
       "      <td>1.424901</td>\n",
       "      <td>1.040305</td>\n",
       "      <td>0.782961</td>\n",
       "      <td>0.964505</td>\n",
       "      <td>1.325709</td>\n",
       "      <td>0.773253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>1.172832</td>\n",
       "      <td>1.032660</td>\n",
       "      <td>1.416574</td>\n",
       "      <td>1.064513</td>\n",
       "      <td>1.134180</td>\n",
       "      <td>1.366983</td>\n",
       "      <td>0.923231</td>\n",
       "      <td>1.047466</td>\n",
       "      <td>0.729370</td>\n",
       "      <td>...</td>\n",
       "      <td>1.034191</td>\n",
       "      <td>1.007370</td>\n",
       "      <td>0.998813</td>\n",
       "      <td>1.254152</td>\n",
       "      <td>1.090315</td>\n",
       "      <td>1.011334</td>\n",
       "      <td>1.208036</td>\n",
       "      <td>1.132882</td>\n",
       "      <td>0.938454</td>\n",
       "      <td>0.772272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_GN86234</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_GN86235</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_GN86236</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 364 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000001  ACH-000004  ACH-000007  ACH-000009  \\\n",
       "0               BMGC_GN00001    0.851056    1.323804    0.937570    0.980206   \n",
       "1               BMGC_GN00002    1.157653    1.072377    0.941289    0.972136   \n",
       "2               BMGC_GN00003    1.498803    0.777885    0.958520    2.383638   \n",
       "3               BMGC_GN00004    0.669074    1.064669    0.942657    0.562637   \n",
       "4               BMGC_GN00005    1.172832    1.032660    1.416574    1.064513   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_GN86234    0.000000    0.000000    0.000000    0.000000   \n",
       "86234           BMGC_GN86235    0.000000    0.000000    0.000000    0.000000   \n",
       "86235           BMGC_GN86236    0.000000    0.000000    0.000000    0.000000   \n",
       "86236           BMGC_GN86237    0.000000    0.000000    0.000000    0.000000   \n",
       "86237           BMGC_GN86238    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-000012  ACH-000015  ACH-000018  ACH-000019  ACH-000021  ...  \\\n",
       "0        1.111580    0.660613    0.917698    0.956259    1.020936  ...   \n",
       "1        0.776816    0.557800    0.881935    0.551096    1.118855  ...   \n",
       "2        1.209065    1.156748    2.014502    1.279922    1.189335  ...   \n",
       "3        0.776307    0.813608    0.902359    0.835197    1.514296  ...   \n",
       "4        1.134180    1.366983    0.923231    1.047466    0.729370  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "       ACH-000988  ACH-000991  ACH-000997  ACH-001039  ACH-001065  ACH-001075  \\\n",
       "0        0.937035    0.967644    0.958509    0.993997    1.091848    1.418260   \n",
       "1        1.052398    1.000457    0.992196    0.589237    0.694145    0.587443   \n",
       "2        1.069887    0.974388    1.002760    1.457635    0.676150    1.564420   \n",
       "3        1.086093    1.011928    0.997424    0.630750    1.424901    1.040305   \n",
       "4        1.034191    1.007370    0.998813    1.254152    1.090315    1.011334   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-001107  ACH-001190  ACH-001306  ACH-001307  \n",
       "0        1.656834    0.873136    1.330477    0.977833  \n",
       "1        0.414771    0.968923    0.697147    0.778931  \n",
       "2        1.091653    1.433025    1.692643    1.575951  \n",
       "3        0.782961    0.964505    1.325709    0.773253  \n",
       "4        1.208036    1.132882    0.938454    0.772272  \n",
       "...           ...         ...         ...         ...  \n",
       "86233    0.000000    0.000000    0.000000    0.000000  \n",
       "86234    0.000000    0.000000    0.000000    0.000000  \n",
       "86235    0.000000    0.000000    0.000000    0.000000  \n",
       "86236    0.000000    0.000000    0.000000    0.000000  \n",
       "86237    0.000000    0.000000    0.000000    0.000000  \n",
       "\n",
       "[86238 rows x 364 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000015</th>\n",
       "      <th>ACH-000018</th>\n",
       "      <th>ACH-000019</th>\n",
       "      <th>ACH-000021</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000988</th>\n",
       "      <th>ACH-000991</th>\n",
       "      <th>ACH-000997</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001065</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>6.643393</td>\n",
       "      <td>6.035001</td>\n",
       "      <td>7.372120</td>\n",
       "      <td>7.814309</td>\n",
       "      <td>6.979760</td>\n",
       "      <td>8.022521</td>\n",
       "      <td>6.887456</td>\n",
       "      <td>7.527567</td>\n",
       "      <td>7.001709</td>\n",
       "      <td>...</td>\n",
       "      <td>7.319774</td>\n",
       "      <td>6.999298</td>\n",
       "      <td>7.314172</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.735296</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.801926</td>\n",
       "      <td>7.183322</td>\n",
       "      <td>6.147807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>6.959654</td>\n",
       "      <td>7.571516</td>\n",
       "      <td>6.084658</td>\n",
       "      <td>7.783578</td>\n",
       "      <td>6.860788</td>\n",
       "      <td>5.733721</td>\n",
       "      <td>6.814315</td>\n",
       "      <td>5.928845</td>\n",
       "      <td>6.161633</td>\n",
       "      <td>...</td>\n",
       "      <td>6.879119</td>\n",
       "      <td>6.896168</td>\n",
       "      <td>6.657150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.683453</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.642422</td>\n",
       "      <td>5.817284</td>\n",
       "      <td>6.617196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>5.351106</td>\n",
       "      <td>4.982294</td>\n",
       "      <td>5.535330</td>\n",
       "      <td>4.891940</td>\n",
       "      <td>4.700858</td>\n",
       "      <td>4.168149</td>\n",
       "      <td>3.806256</td>\n",
       "      <td>5.054789</td>\n",
       "      <td>3.674823</td>\n",
       "      <td>...</td>\n",
       "      <td>4.787896</td>\n",
       "      <td>5.823450</td>\n",
       "      <td>4.625168</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.475867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.209387</td>\n",
       "      <td>3.840053</td>\n",
       "      <td>3.613618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>5.868467</td>\n",
       "      <td>6.293503</td>\n",
       "      <td>5.687442</td>\n",
       "      <td>6.753958</td>\n",
       "      <td>6.896627</td>\n",
       "      <td>5.677143</td>\n",
       "      <td>5.758296</td>\n",
       "      <td>6.794872</td>\n",
       "      <td>5.548689</td>\n",
       "      <td>...</td>\n",
       "      <td>6.582769</td>\n",
       "      <td>6.423730</td>\n",
       "      <td>7.328425</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.867169</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.223950</td>\n",
       "      <td>4.805016</td>\n",
       "      <td>5.846684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>0.186332</td>\n",
       "      <td>-0.033249</td>\n",
       "      <td>0.031376</td>\n",
       "      <td>0.665821</td>\n",
       "      <td>0.186332</td>\n",
       "      <td>2.152987</td>\n",
       "      <td>3.472767</td>\n",
       "      <td>0.976367</td>\n",
       "      <td>0.931453</td>\n",
       "      <td>...</td>\n",
       "      <td>1.989077</td>\n",
       "      <td>0.284142</td>\n",
       "      <td>0.230671</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.012828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.922400</td>\n",
       "      <td>0.043941</td>\n",
       "      <td>0.175017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412034</th>\n",
       "      <td>BMGC_TS412035</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412035</th>\n",
       "      <td>BMGC_TS412036</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412036</th>\n",
       "      <td>BMGC_TS412037</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412037</th>\n",
       "      <td>BMGC_TS412038</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412038</th>\n",
       "      <td>BMGC_TS412039</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412039 rows × 364 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000001  ACH-000004  ACH-000007  ACH-000009  \\\n",
       "0               BMGC_TS000001    6.643393    6.035001    7.372120    7.814309   \n",
       "1               BMGC_TS000002    6.959654    7.571516    6.084658    7.783578   \n",
       "2               BMGC_TS000003    5.351106    4.982294    5.535330    4.891940   \n",
       "3               BMGC_TS000004    5.868467    6.293503    5.687442    6.753958   \n",
       "4               BMGC_TS000005    0.186332   -0.033249    0.031376    0.665821   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "412034          BMGC_TS412035    0.000000    0.000000    0.000000    0.000000   \n",
       "412035          BMGC_TS412036    0.000000    0.000000    0.000000    0.000000   \n",
       "412036          BMGC_TS412037    0.000000    0.000000    0.000000    0.000000   \n",
       "412037          BMGC_TS412038    0.000000    0.000000    0.000000    0.000000   \n",
       "412038          BMGC_TS412039    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "        ACH-000012  ACH-000015  ACH-000018  ACH-000019  ACH-000021  ...  \\\n",
       "0         6.979760    8.022521    6.887456    7.527567    7.001709  ...   \n",
       "1         6.860788    5.733721    6.814315    5.928845    6.161633  ...   \n",
       "2         4.700858    4.168149    3.806256    5.054789    3.674823  ...   \n",
       "3         6.896627    5.677143    5.758296    6.794872    5.548689  ...   \n",
       "4         0.186332    2.152987    3.472767    0.976367    0.931453  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "412034    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412035    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412036    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412037    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "412038    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "        ACH-000988  ACH-000991  ACH-000997  ACH-001039  ACH-001065  \\\n",
       "0         7.319774    6.999298    7.314172         0.0         0.0   \n",
       "1         6.879119    6.896168    6.657150         0.0         0.0   \n",
       "2         4.787896    5.823450    4.625168         0.0         0.0   \n",
       "3         6.582769    6.423730    7.328425         0.0         0.0   \n",
       "4         1.989077    0.284142    0.230671         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "412034    0.000000    0.000000    0.000000         0.0         0.0   \n",
       "412035    0.000000    0.000000    0.000000         0.0         0.0   \n",
       "412036    0.000000    0.000000    0.000000         0.0         0.0   \n",
       "412037    0.000000    0.000000    0.000000         0.0         0.0   \n",
       "412038    0.000000    0.000000    0.000000         0.0         0.0   \n",
       "\n",
       "        ACH-001075  ACH-001107  ACH-001190  ACH-001306  ACH-001307  \n",
       "0         6.735296         0.0    6.801926    7.183322    6.147807  \n",
       "1         6.683453         0.0    7.642422    5.817284    6.617196  \n",
       "2         5.475867         0.0    4.209387    3.840053    3.613618  \n",
       "3         5.867169         0.0    6.223950    4.805016    5.846684  \n",
       "4         3.012828         0.0    2.922400    0.043941    0.175017  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "412034    0.000000         0.0    0.000000    0.000000    0.000000  \n",
       "412035    0.000000         0.0    0.000000    0.000000    0.000000  \n",
       "412036    0.000000         0.0    0.000000    0.000000    0.000000  \n",
       "412037    0.000000         0.0    0.000000    0.000000    0.000000  \n",
       "412038    0.000000         0.0    0.000000    0.000000    0.000000  \n",
       "\n",
       "[412039 rows x 364 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# fill in the missing samples in the final_merged_gene_df for overlapped_dti_crispr_rna_samples\n",
    "for sample in overlapped_dti_crispr_rna_samples:\n",
    "    if sample not in final_merged_gene_df.columns:\n",
    "        final_merged_gene_df[sample] = 0.0\n",
    "# fill in the missing samples in the final_merged_transcript_df for overlapped_dti_crispr_rna_samples\n",
    "for sample in overlapped_dti_crispr_rna_samples:\n",
    "    if sample not in final_merged_transcript_df.columns:\n",
    "        final_merged_transcript_df[sample] = 0.0\n",
    "# get the final gene, transcript, drug dataframe by filtering the overlapped_dti_crispr_rna_samples\n",
    "target_crispr_gene_df = final_merged_gene_df[['BioMedGraphica_Conn_ID'] + overlapped_dti_crispr_rna_samples]\n",
    "display(target_crispr_gene_df)\n",
    "target_crispr_transcript_df = final_merged_transcript_df[['BioMedGraphica_Conn_ID'] + overlapped_dti_crispr_rna_samples]\n",
    "display(target_crispr_transcript_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000015</th>\n",
       "      <th>ACH-000018</th>\n",
       "      <th>ACH-000019</th>\n",
       "      <th>ACH-000021</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000988</th>\n",
       "      <th>ACH-000991</th>\n",
       "      <th>ACH-000997</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001065</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.77900</td>\n",
       "      <td>0.07045</td>\n",
       "      <td>0.92549</td>\n",
       "      <td>0.93426</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.30908</td>\n",
       "      <td>0.81250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.29050</td>\n",
       "      <td>0.73685</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.70055</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.46825</td>\n",
       "      <td>0.67335</td>\n",
       "      <td>0.6964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.05796</td>\n",
       "      <td>0.00686</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>0.00286</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00999</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00433</td>\n",
       "      <td>0.02217</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01297</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01149</td>\n",
       "      <td>0.00573</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.02149</td>\n",
       "      <td>0.59846</td>\n",
       "      <td>0.88734</td>\n",
       "      <td>0.94058</td>\n",
       "      <td>0.44256</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02831</td>\n",
       "      <td>0.03987</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.32492</td>\n",
       "      <td>0.90407</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.11109</td>\n",
       "      <td>0.00307</td>\n",
       "      <td>0.0058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705929</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705930</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705931</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705932</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705933</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>705934 rows × 364 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000001  ACH-000004  ACH-000007  ACH-000009  \\\n",
       "0                BMGC_PM00001     1.00000     0.77900     0.07045     0.92549   \n",
       "1                BMGC_PM00002     0.00000     0.00000     0.00000     0.00000   \n",
       "2                BMGC_PM00003     0.00000     0.05796     0.00686     0.00112   \n",
       "3                BMGC_PM00004     0.02149     0.59846     0.88734     0.94058   \n",
       "4                BMGC_PM00005     0.00000     0.00000     0.00000     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "705929          BMGC_PT121415     0.00000     0.00000     0.00000     0.00000   \n",
       "705930          BMGC_PT121416     0.00000     0.00000     0.00000     0.00000   \n",
       "705931          BMGC_PT121417     0.00000     0.00000     0.00000     0.00000   \n",
       "705932          BMGC_PT121418     0.00000     0.00000     0.00000     0.00000   \n",
       "705933          BMGC_PT121419     0.00000     0.00000     0.00000     0.00000   \n",
       "\n",
       "        ACH-000012  ACH-000015  ACH-000018  ACH-000019  ACH-000021  ...  \\\n",
       "0          0.93426         0.0     0.30908     0.81250         0.0  ...   \n",
       "1          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "2          0.00286         0.0     0.00000     0.00999         0.0  ...   \n",
       "3          0.44256         0.0     0.02831     0.03987         0.0  ...   \n",
       "4          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "705929     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705930     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705931     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705932     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705933     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "        ACH-000988  ACH-000991  ACH-000997  ACH-001039  ACH-001065  \\\n",
       "0              0.0     0.29050     0.73685         0.0         0.0   \n",
       "1              0.0     0.00000     0.00000         0.0         0.0   \n",
       "2              0.0     0.00433     0.02217         0.0         0.0   \n",
       "3              0.0     0.32492     0.90407         0.0         0.0   \n",
       "4              0.0     0.00000     0.00000         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "705929         0.0     0.00000     0.00000         0.0         0.0   \n",
       "705930         0.0     0.00000     0.00000         0.0         0.0   \n",
       "705931         0.0     0.00000     0.00000         0.0         0.0   \n",
       "705932         0.0     0.00000     0.00000         0.0         0.0   \n",
       "705933         0.0     0.00000     0.00000         0.0         0.0   \n",
       "\n",
       "        ACH-001075  ACH-001107  ACH-001190  ACH-001306  ACH-001307  \n",
       "0          0.70055         0.0     0.46825     0.67335      0.6964  \n",
       "1          0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "2          0.01297         0.0     0.01149     0.00573      0.0000  \n",
       "3          0.00000         0.0     0.11109     0.00307      0.0058  \n",
       "4          0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "705929     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "705930     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "705931     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "705932     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "705933     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "\n",
       "[705934 rows x 364 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000015</th>\n",
       "      <th>ACH-000018</th>\n",
       "      <th>ACH-000019</th>\n",
       "      <th>ACH-000021</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000988</th>\n",
       "      <th>ACH-000991</th>\n",
       "      <th>ACH-000997</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001065</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.77900</td>\n",
       "      <td>0.07045</td>\n",
       "      <td>0.92549</td>\n",
       "      <td>0.93426</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.30908</td>\n",
       "      <td>0.81250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.29050</td>\n",
       "      <td>0.73685</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.70055</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.46825</td>\n",
       "      <td>0.67335</td>\n",
       "      <td>0.6964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.05796</td>\n",
       "      <td>0.00686</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>0.00286</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00999</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00433</td>\n",
       "      <td>0.02217</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01297</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01149</td>\n",
       "      <td>0.00573</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.02149</td>\n",
       "      <td>0.59846</td>\n",
       "      <td>0.88734</td>\n",
       "      <td>0.94058</td>\n",
       "      <td>0.44256</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02831</td>\n",
       "      <td>0.03987</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.32492</td>\n",
       "      <td>0.90407</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.11109</td>\n",
       "      <td>0.00307</td>\n",
       "      <td>0.0058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 364 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000001  ACH-000004  ACH-000007  ACH-000009  \\\n",
       "0                BMGC_PM00001     1.00000     0.77900     0.07045     0.92549   \n",
       "1                BMGC_PM00002     0.00000     0.00000     0.00000     0.00000   \n",
       "2                BMGC_PM00003     0.00000     0.05796     0.00686     0.00112   \n",
       "3                BMGC_PM00004     0.02149     0.59846     0.88734     0.94058   \n",
       "4                BMGC_PM00005     0.00000     0.00000     0.00000     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "834804           BMGC_DG20914     0.00000     0.00000     0.00000     0.00000   \n",
       "834805           BMGC_DG20915     0.00000     0.00000     0.00000     0.00000   \n",
       "834806           BMGC_DG20916     0.00000     0.00000     0.00000     0.00000   \n",
       "834807           BMGC_DG20917     0.00000     0.00000     0.00000     0.00000   \n",
       "834808           BMGC_DG20918     0.00000     0.00000     0.00000     0.00000   \n",
       "\n",
       "        ACH-000012  ACH-000015  ACH-000018  ACH-000019  ACH-000021  ...  \\\n",
       "0          0.93426         0.0     0.30908     0.81250         0.0  ...   \n",
       "1          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "2          0.00286         0.0     0.00000     0.00999         0.0  ...   \n",
       "3          0.44256         0.0     0.02831     0.03987         0.0  ...   \n",
       "4          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "834804     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834805     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834806     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834807     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834808     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "        ACH-000988  ACH-000991  ACH-000997  ACH-001039  ACH-001065  \\\n",
       "0              0.0     0.29050     0.73685         0.0         0.0   \n",
       "1              0.0     0.00000     0.00000         0.0         0.0   \n",
       "2              0.0     0.00433     0.02217         0.0         0.0   \n",
       "3              0.0     0.32492     0.90407         0.0         0.0   \n",
       "4              0.0     0.00000     0.00000         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "834804         0.0     0.00000     0.00000         0.0         0.0   \n",
       "834805         0.0     0.00000     0.00000         0.0         0.0   \n",
       "834806         0.0     0.00000     0.00000         0.0         0.0   \n",
       "834807         0.0     0.00000     0.00000         0.0         0.0   \n",
       "834808         0.0     0.00000     0.00000         0.0         0.0   \n",
       "\n",
       "        ACH-001075  ACH-001107  ACH-001190  ACH-001306  ACH-001307  \n",
       "0          0.70055         0.0     0.46825     0.67335      0.6964  \n",
       "1          0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "2          0.01297         0.0     0.01149     0.00573      0.0000  \n",
       "3          0.00000         0.0     0.11109     0.00307      0.0058  \n",
       "4          0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "834804     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "834805     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "834806     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "834807     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "834808     0.00000         0.0     0.00000     0.00000      0.0000  \n",
       "\n",
       "[834809 rows x 364 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "target_crispr_omics_df = pd.concat([target_crispr_methy_df, target_crispr_gene_df, target_crispr_transcript_df, target_crispr_protein_df], axis=0).reset_index(drop=True)\n",
    "display(target_crispr_omics_df)\n",
    "target_crispr_df = pd.merge(bmgc_entity_df, target_crispr_omics_df, left_on='BioMedGraphica_Conn_ID', right_on='BioMedGraphica_Conn_ID', how='left').drop(columns=['BioMedGraphica_ID', 'Type'])\n",
    "target_crispr_df = target_crispr_df.fillna(0.0)\n",
    "display(target_crispr_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 9.2.2.1 Target Screen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 407,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of target_crispr_train_feat: (300, 834809)\n",
      "Shape of target_crispr_test_feat: (63, 834809)\n"
     ]
    }
   ],
   "source": [
    "# Select the features from target_crispr_df by target_crispr_train_samples and target_crispr_test_samples\n",
    "target_crispr_train_feat = target_crispr_df[['BioMedGraphica_Conn_ID'] + target_crispr_train_samples].drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "target_crispr_test_feat = target_crispr_df[['BioMedGraphica_Conn_ID'] + target_crispr_test_samples].drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "# Save the training and test features to .npy files\n",
    "np.save('./data/TargetScreen/target_crispr_train_feature.npy', target_crispr_train_feat)\n",
    "np.save('./data/TargetScreen/target_crispr_test_feature.npy', target_crispr_test_feat)\n",
    "# Print the shapes of the training and test features\n",
    "print(\"Shape of target_crispr_train_feat:\", target_crispr_train_feat.shape)\n",
    "print(\"Shape of target_crispr_test_feat:\", target_crispr_test_feat.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of target_crispr_feat: (363, 834809)\n"
     ]
    }
   ],
   "source": [
    "# Select the features from target_crispr_df\n",
    "target_crispr_feat = target_crispr_df.drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "# Save the features to .npy files\n",
    "np.save('./data/TargetScreen/target_crispr_feature.npy', target_crispr_feat)\n",
    "# Print the shapes of the features\n",
    "print(\"Shape of target_crispr_feat:\", target_crispr_feat.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 408,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID_x</th>\n",
       "      <th>BioMedGraphica_ID_x</th>\n",
       "      <th>transcriptID</th>\n",
       "      <th>proteinID</th>\n",
       "      <th>BMG_From_ID_x</th>\n",
       "      <th>BMG_To_ID_x</th>\n",
       "      <th>Source_x</th>\n",
       "      <th>BioMedGraphica_Conn_ID_y</th>\n",
       "      <th>BioMedGraphica_ID_y</th>\n",
       "      <th>geneID</th>\n",
       "      <th>BMG_From_ID_y</th>\n",
       "      <th>BMG_To_ID_y</th>\n",
       "      <th>Source_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_ED_TSPT000001</td>\n",
       "      <td>BMG_ED_TSPT000001</td>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>BMGC_PT017571</td>\n",
       "      <td>BMG_TS000001</td>\n",
       "      <td>BMG_PT042398</td>\n",
       "      <td>Uniprot;Ensembl</td>\n",
       "      <td>BMGC_ED_GNTS186033</td>\n",
       "      <td>BMG_ED_GNTS186033</td>\n",
       "      <td>BMGC_GN29641</td>\n",
       "      <td>BMG_GN173393</td>\n",
       "      <td>BMG_TS000001</td>\n",
       "      <td>Ensembl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_ED_TSPT000002</td>\n",
       "      <td>BMG_ED_TSPT000002</td>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>BMGC_PT014978</td>\n",
       "      <td>BMG_TS000002</td>\n",
       "      <td>BMG_PT039543</td>\n",
       "      <td>Uniprot;Ensembl</td>\n",
       "      <td>BMGC_ED_GNTS197666</td>\n",
       "      <td>BMG_ED_GNTS197666</td>\n",
       "      <td>BMGC_GN31641</td>\n",
       "      <td>BMG_GN175441</td>\n",
       "      <td>BMG_TS000002</td>\n",
       "      <td>Ensembl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_ED_TSPT000003</td>\n",
       "      <td>BMG_ED_TSPT000003</td>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>BMGC_PT014403</td>\n",
       "      <td>BMG_TS000003</td>\n",
       "      <td>BMG_PT038917</td>\n",
       "      <td>Uniprot;Ensembl</td>\n",
       "      <td>BMGC_ED_GNTS136517</td>\n",
       "      <td>BMG_ED_GNTS136517</td>\n",
       "      <td>BMGC_GN23354</td>\n",
       "      <td>BMG_GN166843</td>\n",
       "      <td>BMG_TS000003</td>\n",
       "      <td>Ensembl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_ED_TSPT000004</td>\n",
       "      <td>BMG_ED_TSPT000004</td>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>BMGC_PT017807</td>\n",
       "      <td>BMG_TS000004</td>\n",
       "      <td>BMG_PT042655</td>\n",
       "      <td>Uniprot;Ensembl</td>\n",
       "      <td>BMGC_ED_GNTS140655</td>\n",
       "      <td>BMG_ED_GNTS140655</td>\n",
       "      <td>BMGC_GN23854</td>\n",
       "      <td>BMG_GN167345</td>\n",
       "      <td>BMG_TS000004</td>\n",
       "      <td>Ensembl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_ED_TSPT000005</td>\n",
       "      <td>BMG_ED_TSPT000005</td>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>BMGC_PT004700</td>\n",
       "      <td>BMG_TS000005</td>\n",
       "      <td>BMG_PT015283</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>BMGC_ED_GNTS235896</td>\n",
       "      <td>BMG_ED_GNTS235896</td>\n",
       "      <td>BMGC_GN35654</td>\n",
       "      <td>BMG_GN179505</td>\n",
       "      <td>BMG_TS000005</td>\n",
       "      <td>RefSeq;Ensembl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458278</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS412035</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS427772</td>\n",
       "      <td>BMG_ED_GNTS427772</td>\n",
       "      <td>BMGC_GN86221</td>\n",
       "      <td>BMG_GN230295</td>\n",
       "      <td>BMG_TS412320</td>\n",
       "      <td>Ensembl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458279</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS412036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS042666</td>\n",
       "      <td>BMG_ED_GNTS042666</td>\n",
       "      <td>BMGC_GN08809</td>\n",
       "      <td>BMG_GN010161</td>\n",
       "      <td>BMG_TS412321</td>\n",
       "      <td>Ensembl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458280</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS412037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS042667</td>\n",
       "      <td>BMG_ED_GNTS042667</td>\n",
       "      <td>BMGC_GN08809</td>\n",
       "      <td>BMG_GN010161</td>\n",
       "      <td>BMG_TS412322</td>\n",
       "      <td>Ensembl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458281</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS412038</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS042668</td>\n",
       "      <td>BMG_ED_GNTS042668</td>\n",
       "      <td>BMGC_GN08809</td>\n",
       "      <td>BMG_GN010161</td>\n",
       "      <td>BMG_TS412323</td>\n",
       "      <td>Ensembl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458282</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS412039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS196451</td>\n",
       "      <td>BMG_ED_GNTS196451</td>\n",
       "      <td>BMGC_GN31180</td>\n",
       "      <td>BMG_GN174973</td>\n",
       "      <td>BMG_TS412324</td>\n",
       "      <td>Ensembl</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>458283 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID_x BioMedGraphica_ID_x   transcriptID  \\\n",
       "0            BMGC_ED_TSPT000001   BMG_ED_TSPT000001  BMGC_TS000001   \n",
       "1            BMGC_ED_TSPT000002   BMG_ED_TSPT000002  BMGC_TS000002   \n",
       "2            BMGC_ED_TSPT000003   BMG_ED_TSPT000003  BMGC_TS000003   \n",
       "3            BMGC_ED_TSPT000004   BMG_ED_TSPT000004  BMGC_TS000004   \n",
       "4            BMGC_ED_TSPT000005   BMG_ED_TSPT000005  BMGC_TS000005   \n",
       "...                         ...                 ...            ...   \n",
       "458278                      NaN                 NaN  BMGC_TS412035   \n",
       "458279                      NaN                 NaN  BMGC_TS412036   \n",
       "458280                      NaN                 NaN  BMGC_TS412037   \n",
       "458281                      NaN                 NaN  BMGC_TS412038   \n",
       "458282                      NaN                 NaN  BMGC_TS412039   \n",
       "\n",
       "            proteinID BMG_From_ID_x   BMG_To_ID_x         Source_x  \\\n",
       "0       BMGC_PT017571  BMG_TS000001  BMG_PT042398  Uniprot;Ensembl   \n",
       "1       BMGC_PT014978  BMG_TS000002  BMG_PT039543  Uniprot;Ensembl   \n",
       "2       BMGC_PT014403  BMG_TS000003  BMG_PT038917  Uniprot;Ensembl   \n",
       "3       BMGC_PT017807  BMG_TS000004  BMG_PT042655  Uniprot;Ensembl   \n",
       "4       BMGC_PT004700  BMG_TS000005  BMG_PT015283           RefSeq   \n",
       "...               ...           ...           ...              ...   \n",
       "458278            NaN           NaN           NaN              NaN   \n",
       "458279            NaN           NaN           NaN              NaN   \n",
       "458280            NaN           NaN           NaN              NaN   \n",
       "458281            NaN           NaN           NaN              NaN   \n",
       "458282            NaN           NaN           NaN              NaN   \n",
       "\n",
       "       BioMedGraphica_Conn_ID_y BioMedGraphica_ID_y        geneID  \\\n",
       "0            BMGC_ED_GNTS186033   BMG_ED_GNTS186033  BMGC_GN29641   \n",
       "1            BMGC_ED_GNTS197666   BMG_ED_GNTS197666  BMGC_GN31641   \n",
       "2            BMGC_ED_GNTS136517   BMG_ED_GNTS136517  BMGC_GN23354   \n",
       "3            BMGC_ED_GNTS140655   BMG_ED_GNTS140655  BMGC_GN23854   \n",
       "4            BMGC_ED_GNTS235896   BMG_ED_GNTS235896  BMGC_GN35654   \n",
       "...                         ...                 ...           ...   \n",
       "458278       BMGC_ED_GNTS427772   BMG_ED_GNTS427772  BMGC_GN86221   \n",
       "458279       BMGC_ED_GNTS042666   BMG_ED_GNTS042666  BMGC_GN08809   \n",
       "458280       BMGC_ED_GNTS042667   BMG_ED_GNTS042667  BMGC_GN08809   \n",
       "458281       BMGC_ED_GNTS042668   BMG_ED_GNTS042668  BMGC_GN08809   \n",
       "458282       BMGC_ED_GNTS196451   BMG_ED_GNTS196451  BMGC_GN31180   \n",
       "\n",
       "       BMG_From_ID_y   BMG_To_ID_y        Source_y  \n",
       "0       BMG_GN173393  BMG_TS000001         Ensembl  \n",
       "1       BMG_GN175441  BMG_TS000002         Ensembl  \n",
       "2       BMG_GN166843  BMG_TS000003         Ensembl  \n",
       "3       BMG_GN167345  BMG_TS000004         Ensembl  \n",
       "4       BMG_GN179505  BMG_TS000005  RefSeq;Ensembl  \n",
       "...              ...           ...             ...  \n",
       "458278  BMG_GN230295  BMG_TS412320         Ensembl  \n",
       "458279  BMG_GN010161  BMG_TS412321         Ensembl  \n",
       "458280  BMG_GN010161  BMG_TS412322         Ensembl  \n",
       "458281  BMG_GN010161  BMG_TS412323         Ensembl  \n",
       "458282  BMG_GN174973  BMG_TS412324         Ensembl  \n",
       "\n",
       "[458283 rows x 13 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID_x</th>\n",
       "      <th>BioMedGraphica_ID_x</th>\n",
       "      <th>transcriptID</th>\n",
       "      <th>proteinID</th>\n",
       "      <th>BMG_From_ID_x</th>\n",
       "      <th>BMG_To_ID_x</th>\n",
       "      <th>Source_x</th>\n",
       "      <th>BioMedGraphica_Conn_ID_y</th>\n",
       "      <th>BioMedGraphica_ID_y</th>\n",
       "      <th>geneID</th>\n",
       "      <th>BMG_From_ID_y</th>\n",
       "      <th>BMG_To_ID_y</th>\n",
       "      <th>Source_y</th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>BioMedGraphica_ID</th>\n",
       "      <th>promoterID</th>\n",
       "      <th>BMG_From_ID</th>\n",
       "      <th>BMG_To_ID</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_ED_TSPT007580</td>\n",
       "      <td>BMG_ED_TSPT007580</td>\n",
       "      <td>BMGC_TS003701</td>\n",
       "      <td>BMGC_PT013542</td>\n",
       "      <td>BMG_TS003701</td>\n",
       "      <td>BMG_PT037915</td>\n",
       "      <td>Uniprot;Ensembl</td>\n",
       "      <td>BMGC_ED_GNTS000001</td>\n",
       "      <td>BMG_ED_GNTS000001</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>BMG_TS003701</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN00001</td>\n",
       "      <td>BMG_ED_PMGN00001</td>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMG_PM000001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS179036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS000002</td>\n",
       "      <td>BMG_ED_GNTS000002</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>BMG_TS179160</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN00001</td>\n",
       "      <td>BMG_ED_PMGN00001</td>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMG_PM000001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS180139</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS000003</td>\n",
       "      <td>BMG_ED_GNTS000003</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>BMG_TS180263</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN00001</td>\n",
       "      <td>BMG_ED_PMGN00001</td>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMG_PM000001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS180951</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS000004</td>\n",
       "      <td>BMG_ED_GNTS000004</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>BMG_TS181076</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN00001</td>\n",
       "      <td>BMG_ED_PMGN00001</td>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMG_PM000001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_ED_TSPT114868</td>\n",
       "      <td>BMG_ED_TSPT114868</td>\n",
       "      <td>BMGC_TS182456</td>\n",
       "      <td>BMGC_PT091702</td>\n",
       "      <td>BMG_TS182582</td>\n",
       "      <td>BMG_PT144261</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_GNTS000005</td>\n",
       "      <td>BMG_ED_GNTS000005</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>BMG_TS182582</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN00001</td>\n",
       "      <td>BMG_ED_PMGN00001</td>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMG_PM000001</td>\n",
       "      <td>BMG_GN000001</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458340</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS411932</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS427806</td>\n",
       "      <td>BMG_ED_GNTS427806</td>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>BMG_GN230311</td>\n",
       "      <td>BMG_TS412217</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN86237</td>\n",
       "      <td>BMG_ED_PMGN86237</td>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>BMG_PM230311</td>\n",
       "      <td>BMG_GN230311</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458341</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS412014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS427807</td>\n",
       "      <td>BMG_ED_GNTS427807</td>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>BMG_GN230312</td>\n",
       "      <td>BMG_TS412299</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN86238</td>\n",
       "      <td>BMG_ED_PMGN86238</td>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>BMG_PM230312</td>\n",
       "      <td>BMG_GN230312</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458342</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS412015</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS427808</td>\n",
       "      <td>BMG_ED_GNTS427808</td>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>BMG_GN230312</td>\n",
       "      <td>BMG_TS412300</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN86238</td>\n",
       "      <td>BMG_ED_PMGN86238</td>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>BMG_PM230312</td>\n",
       "      <td>BMG_GN230312</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458343</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS412016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS427809</td>\n",
       "      <td>BMG_ED_GNTS427809</td>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>BMG_GN230312</td>\n",
       "      <td>BMG_TS412301</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN86238</td>\n",
       "      <td>BMG_ED_PMGN86238</td>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>BMG_PM230312</td>\n",
       "      <td>BMG_GN230312</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458344</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_TS412017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_ED_GNTS427810</td>\n",
       "      <td>BMG_ED_GNTS427810</td>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>BMG_GN230312</td>\n",
       "      <td>BMG_TS412302</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>BMGC_ED_PMGN86238</td>\n",
       "      <td>BMG_ED_PMGN86238</td>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>BMG_PM230312</td>\n",
       "      <td>BMG_GN230312</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>458345 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID_x BioMedGraphica_ID_x   transcriptID  \\\n",
       "0            BMGC_ED_TSPT007580   BMG_ED_TSPT007580  BMGC_TS003701   \n",
       "1                           NaN                 NaN  BMGC_TS179036   \n",
       "2                           NaN                 NaN  BMGC_TS180139   \n",
       "3                           NaN                 NaN  BMGC_TS180951   \n",
       "4            BMGC_ED_TSPT114868   BMG_ED_TSPT114868  BMGC_TS182456   \n",
       "...                         ...                 ...            ...   \n",
       "458340                      NaN                 NaN  BMGC_TS411932   \n",
       "458341                      NaN                 NaN  BMGC_TS412014   \n",
       "458342                      NaN                 NaN  BMGC_TS412015   \n",
       "458343                      NaN                 NaN  BMGC_TS412016   \n",
       "458344                      NaN                 NaN  BMGC_TS412017   \n",
       "\n",
       "            proteinID BMG_From_ID_x   BMG_To_ID_x         Source_x  \\\n",
       "0       BMGC_PT013542  BMG_TS003701  BMG_PT037915  Uniprot;Ensembl   \n",
       "1                 NaN           NaN           NaN              NaN   \n",
       "2                 NaN           NaN           NaN              NaN   \n",
       "3                 NaN           NaN           NaN              NaN   \n",
       "4       BMGC_PT091702  BMG_TS182582  BMG_PT144261          Ensembl   \n",
       "...               ...           ...           ...              ...   \n",
       "458340            NaN           NaN           NaN              NaN   \n",
       "458341            NaN           NaN           NaN              NaN   \n",
       "458342            NaN           NaN           NaN              NaN   \n",
       "458343            NaN           NaN           NaN              NaN   \n",
       "458344            NaN           NaN           NaN              NaN   \n",
       "\n",
       "       BioMedGraphica_Conn_ID_y BioMedGraphica_ID_y        geneID  \\\n",
       "0            BMGC_ED_GNTS000001   BMG_ED_GNTS000001  BMGC_GN00001   \n",
       "1            BMGC_ED_GNTS000002   BMG_ED_GNTS000002  BMGC_GN00001   \n",
       "2            BMGC_ED_GNTS000003   BMG_ED_GNTS000003  BMGC_GN00001   \n",
       "3            BMGC_ED_GNTS000004   BMG_ED_GNTS000004  BMGC_GN00001   \n",
       "4            BMGC_ED_GNTS000005   BMG_ED_GNTS000005  BMGC_GN00001   \n",
       "...                         ...                 ...           ...   \n",
       "458340       BMGC_ED_GNTS427806   BMG_ED_GNTS427806  BMGC_GN86237   \n",
       "458341       BMGC_ED_GNTS427807   BMG_ED_GNTS427807  BMGC_GN86238   \n",
       "458342       BMGC_ED_GNTS427808   BMG_ED_GNTS427808  BMGC_GN86238   \n",
       "458343       BMGC_ED_GNTS427809   BMG_ED_GNTS427809  BMGC_GN86238   \n",
       "458344       BMGC_ED_GNTS427810   BMG_ED_GNTS427810  BMGC_GN86238   \n",
       "\n",
       "       BMG_From_ID_y   BMG_To_ID_y Source_y BioMedGraphica_Conn_ID  \\\n",
       "0       BMG_GN000001  BMG_TS003701  Ensembl      BMGC_ED_PMGN00001   \n",
       "1       BMG_GN000001  BMG_TS179160  Ensembl      BMGC_ED_PMGN00001   \n",
       "2       BMG_GN000001  BMG_TS180263  Ensembl      BMGC_ED_PMGN00001   \n",
       "3       BMG_GN000001  BMG_TS181076  Ensembl      BMGC_ED_PMGN00001   \n",
       "4       BMG_GN000001  BMG_TS182582  Ensembl      BMGC_ED_PMGN00001   \n",
       "...              ...           ...      ...                    ...   \n",
       "458340  BMG_GN230311  BMG_TS412217  Ensembl      BMGC_ED_PMGN86237   \n",
       "458341  BMG_GN230312  BMG_TS412299  Ensembl      BMGC_ED_PMGN86238   \n",
       "458342  BMG_GN230312  BMG_TS412300  Ensembl      BMGC_ED_PMGN86238   \n",
       "458343  BMG_GN230312  BMG_TS412301  Ensembl      BMGC_ED_PMGN86238   \n",
       "458344  BMG_GN230312  BMG_TS412302  Ensembl      BMGC_ED_PMGN86238   \n",
       "\n",
       "       BioMedGraphica_ID    promoterID   BMG_From_ID     BMG_To_ID Source  \n",
       "0       BMG_ED_PMGN00001  BMGC_PM00001  BMG_PM000001  BMG_GN000001    NaN  \n",
       "1       BMG_ED_PMGN00001  BMGC_PM00001  BMG_PM000001  BMG_GN000001    NaN  \n",
       "2       BMG_ED_PMGN00001  BMGC_PM00001  BMG_PM000001  BMG_GN000001    NaN  \n",
       "3       BMG_ED_PMGN00001  BMGC_PM00001  BMG_PM000001  BMG_GN000001    NaN  \n",
       "4       BMG_ED_PMGN00001  BMGC_PM00001  BMG_PM000001  BMG_GN000001    NaN  \n",
       "...                  ...           ...           ...           ...    ...  \n",
       "458340  BMG_ED_PMGN86237  BMGC_PM86237  BMG_PM230311  BMG_GN230311    NaN  \n",
       "458341  BMG_ED_PMGN86238  BMGC_PM86238  BMG_PM230312  BMG_GN230312    NaN  \n",
       "458342  BMG_ED_PMGN86238  BMGC_PM86238  BMG_PM230312  BMG_GN230312    NaN  \n",
       "458343  BMG_ED_PMGN86238  BMGC_PM86238  BMG_PM230312  BMG_GN230312    NaN  \n",
       "458344  BMG_ED_PMGN86238  BMGC_PM86238  BMG_PM230312  BMG_GN230312    NaN  \n",
       "\n",
       "[458345 rows x 19 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>promoterID</th>\n",
       "      <th>geneID</th>\n",
       "      <th>transcriptID</th>\n",
       "      <th>proteinID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_TS003701</td>\n",
       "      <td>BMGC_PT013542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_TS179036</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_TS180139</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_TS180951</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>BMGC_TS182456</td>\n",
       "      <td>BMGC_PT091702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458340</th>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>BMGC_TS411932</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458341</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>BMGC_TS412014</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458342</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>BMGC_TS412015</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458343</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>BMGC_TS412016</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458344</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>BMGC_TS412017</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>458345 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          promoterID        geneID   transcriptID      proteinID\n",
       "0       BMGC_PM00001  BMGC_GN00001  BMGC_TS003701  BMGC_PT013542\n",
       "1       BMGC_PM00001  BMGC_GN00001  BMGC_TS179036            NaN\n",
       "2       BMGC_PM00001  BMGC_GN00001  BMGC_TS180139            NaN\n",
       "3       BMGC_PM00001  BMGC_GN00001  BMGC_TS180951            NaN\n",
       "4       BMGC_PM00001  BMGC_GN00001  BMGC_TS182456  BMGC_PT091702\n",
       "...              ...           ...            ...            ...\n",
       "458340  BMGC_PM86237  BMGC_GN86237  BMGC_TS411932            NaN\n",
       "458341  BMGC_PM86238  BMGC_GN86238  BMGC_TS412014            NaN\n",
       "458342  BMGC_PM86238  BMGC_GN86238  BMGC_TS412015            NaN\n",
       "458343  BMGC_PM86238  BMGC_GN86238  BMGC_TS412016            NaN\n",
       "458344  BMGC_PM86238  BMGC_GN86238  BMGC_TS412017            NaN\n",
       "\n",
       "[458345 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Build up internal relation for KO drop\n",
    "bmgc_promoter_gene_df = bmgc_omics_relation_df[bmgc_omics_relation_df['Type'] == 'Promoter-Gene'].drop(columns=['Type'])\n",
    "bmgc_gene_transcript_df = bmgc_omics_relation_df[bmgc_omics_relation_df['Type'] == 'Gene-Transcript'].drop(columns=['Type'])\n",
    "bmgc_transcript_protein_df = bmgc_omics_relation_df[bmgc_omics_relation_df['Type'] == 'Transcript-Protein'].drop(columns=['Type'])\n",
    "# Rename the columns\n",
    "bmgc_promoter_gene_df.rename(columns={'BMGC_From_ID':'promoterID','BMGC_To_ID':'geneID'}, inplace=True)\n",
    "bmgc_gene_transcript_df.rename(columns={'BMGC_From_ID':'geneID','BMGC_To_ID':'transcriptID'}, inplace=True)\n",
    "bmgc_transcript_protein_df.rename(columns={'BMGC_From_ID':'transcriptID','BMGC_To_ID':'proteinID'}, inplace=True)\n",
    "# Merge gene_transcript and transcript_protein dataframes\n",
    "bmgc_gene_transcript_protein_df = bmgc_transcript_protein_df.merge(bmgc_gene_transcript_df, on='transcriptID',how='outer')\n",
    "display(bmgc_gene_transcript_protein_df)\n",
    "# Merge promoter\n",
    "bmgc_promoter_gene_transcript_protein_df = bmgc_gene_transcript_protein_df.merge(bmgc_promoter_gene_df, on='geneID',how='outer')\n",
    "display(bmgc_promoter_gene_transcript_protein_df)\n",
    "internal_relation_df = bmgc_promoter_gene_transcript_protein_df.dropna(subset=['geneID'])\n",
    "internal_relation_df = internal_relation_df[['promoterID', 'geneID', 'transcriptID', 'proteinID']].copy()\n",
    "display(internal_relation_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 409,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>A1BG (1)</th>\n",
       "      <th>A1CF (29974)</th>\n",
       "      <th>A2M (2)</th>\n",
       "      <th>A2ML1 (144568)</th>\n",
       "      <th>A3GALT2 (127550)</th>\n",
       "      <th>A4GALT (53947)</th>\n",
       "      <th>A4GNT (51146)</th>\n",
       "      <th>AAAS (8086)</th>\n",
       "      <th>AACS (65985)</th>\n",
       "      <th>...</th>\n",
       "      <th>ZWILCH (55055)</th>\n",
       "      <th>ZWINT (11130)</th>\n",
       "      <th>ZXDA (7789)</th>\n",
       "      <th>ZXDB (158586)</th>\n",
       "      <th>ZXDC (79364)</th>\n",
       "      <th>ZYG11A (440590)</th>\n",
       "      <th>ZYG11B (79699)</th>\n",
       "      <th>ZYX (7791)</th>\n",
       "      <th>ZZEF1 (23140)</th>\n",
       "      <th>ZZZ3 (26009)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>0.042647</td>\n",
       "      <td>0.026102</td>\n",
       "      <td>-0.147614</td>\n",
       "      <td>-0.020425</td>\n",
       "      <td>0.184014</td>\n",
       "      <td>-0.076020</td>\n",
       "      <td>-0.097068</td>\n",
       "      <td>0.182759</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.120536</td>\n",
       "      <td>0.107286</td>\n",
       "      <td>0.166390</td>\n",
       "      <td>0.262275</td>\n",
       "      <td>0.163876</td>\n",
       "      <td>-0.249508</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>0.067986</td>\n",
       "      <td>0.136285</td>\n",
       "      <td>-0.438536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.088500</td>\n",
       "      <td>-0.088198</td>\n",
       "      <td>-0.041225</td>\n",
       "      <td>-0.028074</td>\n",
       "      <td>-0.157206</td>\n",
       "      <td>0.280101</td>\n",
       "      <td>-0.106904</td>\n",
       "      <td>0.178125</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.192527</td>\n",
       "      <td>-0.324059</td>\n",
       "      <td>0.230377</td>\n",
       "      <td>0.087609</td>\n",
       "      <td>0.074897</td>\n",
       "      <td>0.054335</td>\n",
       "      <td>-0.330343</td>\n",
       "      <td>0.099067</td>\n",
       "      <td>0.274566</td>\n",
       "      <td>0.001871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000005</td>\n",
       "      <td>-0.184847</td>\n",
       "      <td>0.003300</td>\n",
       "      <td>0.160881</td>\n",
       "      <td>0.086224</td>\n",
       "      <td>-0.149315</td>\n",
       "      <td>-0.253837</td>\n",
       "      <td>0.167011</td>\n",
       "      <td>-0.101209</td>\n",
       "      <td>-0.129827</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.312827</td>\n",
       "      <td>-0.338023</td>\n",
       "      <td>-0.039700</td>\n",
       "      <td>-0.055349</td>\n",
       "      <td>-0.000367</td>\n",
       "      <td>-0.205605</td>\n",
       "      <td>-0.066032</td>\n",
       "      <td>-0.054518</td>\n",
       "      <td>0.035579</td>\n",
       "      <td>-0.150486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000007</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.113717</td>\n",
       "      <td>0.082872</td>\n",
       "      <td>0.099633</td>\n",
       "      <td>-0.008378</td>\n",
       "      <td>-0.022310</td>\n",
       "      <td>0.014416</td>\n",
       "      <td>-0.184977</td>\n",
       "      <td>-0.173739</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.334843</td>\n",
       "      <td>-0.355499</td>\n",
       "      <td>-0.014183</td>\n",
       "      <td>0.230944</td>\n",
       "      <td>0.044628</td>\n",
       "      <td>-0.081863</td>\n",
       "      <td>-0.390748</td>\n",
       "      <td>-0.036547</td>\n",
       "      <td>-0.273129</td>\n",
       "      <td>-0.382723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000009</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>-0.134669</td>\n",
       "      <td>0.060323</td>\n",
       "      <td>0.076647</td>\n",
       "      <td>0.078922</td>\n",
       "      <td>-0.100243</td>\n",
       "      <td>0.047559</td>\n",
       "      <td>-0.136988</td>\n",
       "      <td>0.037759</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.299593</td>\n",
       "      <td>-0.194427</td>\n",
       "      <td>-0.027365</td>\n",
       "      <td>0.236591</td>\n",
       "      <td>-0.084224</td>\n",
       "      <td>-0.098271</td>\n",
       "      <td>-0.510495</td>\n",
       "      <td>0.052938</td>\n",
       "      <td>0.018623</td>\n",
       "      <td>-0.258353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1173</th>\n",
       "      <td>ACH-002926</td>\n",
       "      <td>-0.107883</td>\n",
       "      <td>-0.101501</td>\n",
       "      <td>0.046677</td>\n",
       "      <td>0.096868</td>\n",
       "      <td>-0.118650</td>\n",
       "      <td>-0.065258</td>\n",
       "      <td>-0.064944</td>\n",
       "      <td>-0.175912</td>\n",
       "      <td>-0.044707</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.409541</td>\n",
       "      <td>-0.579376</td>\n",
       "      <td>0.038706</td>\n",
       "      <td>0.338610</td>\n",
       "      <td>-0.174379</td>\n",
       "      <td>-0.040264</td>\n",
       "      <td>-0.239763</td>\n",
       "      <td>0.065168</td>\n",
       "      <td>-0.144088</td>\n",
       "      <td>-0.334424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1174</th>\n",
       "      <td>ACH-002928</td>\n",
       "      <td>-0.287606</td>\n",
       "      <td>-0.179101</td>\n",
       "      <td>-0.135824</td>\n",
       "      <td>-0.046579</td>\n",
       "      <td>-0.249888</td>\n",
       "      <td>-0.097376</td>\n",
       "      <td>0.070385</td>\n",
       "      <td>0.049444</td>\n",
       "      <td>-0.244937</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.095328</td>\n",
       "      <td>-0.440452</td>\n",
       "      <td>0.303500</td>\n",
       "      <td>0.097690</td>\n",
       "      <td>0.151790</td>\n",
       "      <td>0.039994</td>\n",
       "      <td>0.098240</td>\n",
       "      <td>-0.266654</td>\n",
       "      <td>-0.115108</td>\n",
       "      <td>-0.080421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1175</th>\n",
       "      <td>ACH-003012</td>\n",
       "      <td>-0.145878</td>\n",
       "      <td>-0.004000</td>\n",
       "      <td>0.091933</td>\n",
       "      <td>0.191332</td>\n",
       "      <td>-0.065385</td>\n",
       "      <td>-0.134152</td>\n",
       "      <td>0.012353</td>\n",
       "      <td>-0.321012</td>\n",
       "      <td>0.001571</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.280119</td>\n",
       "      <td>-0.662673</td>\n",
       "      <td>-0.180936</td>\n",
       "      <td>0.169997</td>\n",
       "      <td>-0.036326</td>\n",
       "      <td>0.158702</td>\n",
       "      <td>-0.213281</td>\n",
       "      <td>-0.017884</td>\n",
       "      <td>-0.195395</td>\n",
       "      <td>-0.309021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>ACH-003177</td>\n",
       "      <td>-0.038435</td>\n",
       "      <td>-0.055660</td>\n",
       "      <td>0.018989</td>\n",
       "      <td>0.117182</td>\n",
       "      <td>0.063397</td>\n",
       "      <td>0.049993</td>\n",
       "      <td>0.175520</td>\n",
       "      <td>-0.085752</td>\n",
       "      <td>0.078435</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.080700</td>\n",
       "      <td>-0.476797</td>\n",
       "      <td>-0.158375</td>\n",
       "      <td>-0.008999</td>\n",
       "      <td>-0.030058</td>\n",
       "      <td>-0.103513</td>\n",
       "      <td>-0.119234</td>\n",
       "      <td>-0.153744</td>\n",
       "      <td>0.032584</td>\n",
       "      <td>-0.536071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1177</th>\n",
       "      <td>ACH-003180</td>\n",
       "      <td>-0.007314</td>\n",
       "      <td>-0.212478</td>\n",
       "      <td>0.129900</td>\n",
       "      <td>-0.023112</td>\n",
       "      <td>-0.068613</td>\n",
       "      <td>0.067406</td>\n",
       "      <td>0.003678</td>\n",
       "      <td>-0.312161</td>\n",
       "      <td>0.009587</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.038622</td>\n",
       "      <td>-0.667881</td>\n",
       "      <td>-0.090500</td>\n",
       "      <td>0.077756</td>\n",
       "      <td>0.055249</td>\n",
       "      <td>0.040616</td>\n",
       "      <td>-0.378505</td>\n",
       "      <td>-0.143920</td>\n",
       "      <td>-0.039985</td>\n",
       "      <td>-0.439905</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1178 rows × 17917 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Sample  A1BG (1)  A1CF (29974)   A2M (2)  A2ML1 (144568)  \\\n",
       "0     ACH-000001 -0.121964      0.042647  0.026102       -0.147614   \n",
       "1     ACH-000004 -0.016484     -0.088500 -0.088198       -0.041225   \n",
       "2     ACH-000005 -0.184847      0.003300  0.160881        0.086224   \n",
       "3     ACH-000007 -0.071921     -0.113717  0.082872        0.099633   \n",
       "4     ACH-000009 -0.019163     -0.134669  0.060323        0.076647   \n",
       "...          ...       ...           ...       ...             ...   \n",
       "1173  ACH-002926 -0.107883     -0.101501  0.046677        0.096868   \n",
       "1174  ACH-002928 -0.287606     -0.179101 -0.135824       -0.046579   \n",
       "1175  ACH-003012 -0.145878     -0.004000  0.091933        0.191332   \n",
       "1176  ACH-003177 -0.038435     -0.055660  0.018989        0.117182   \n",
       "1177  ACH-003180 -0.007314     -0.212478  0.129900       -0.023112   \n",
       "\n",
       "      A3GALT2 (127550)  A4GALT (53947)  A4GNT (51146)  AAAS (8086)  \\\n",
       "0            -0.020425        0.184014      -0.076020    -0.097068   \n",
       "1            -0.028074       -0.157206       0.280101    -0.106904   \n",
       "2            -0.149315       -0.253837       0.167011    -0.101209   \n",
       "3            -0.008378       -0.022310       0.014416    -0.184977   \n",
       "4             0.078922       -0.100243       0.047559    -0.136988   \n",
       "...                ...             ...            ...          ...   \n",
       "1173         -0.118650       -0.065258      -0.064944    -0.175912   \n",
       "1174         -0.249888       -0.097376       0.070385     0.049444   \n",
       "1175         -0.065385       -0.134152       0.012353    -0.321012   \n",
       "1176          0.063397        0.049993       0.175520    -0.085752   \n",
       "1177         -0.068613        0.067406       0.003678    -0.312161   \n",
       "\n",
       "      AACS (65985)  ...  ZWILCH (55055)  ZWINT (11130)  ZXDA (7789)  \\\n",
       "0         0.182759  ...       -0.120536       0.107286     0.166390   \n",
       "1         0.178125  ...       -0.192527      -0.324059     0.230377   \n",
       "2        -0.129827  ...       -0.312827      -0.338023    -0.039700   \n",
       "3        -0.173739  ...       -0.334843      -0.355499    -0.014183   \n",
       "4         0.037759  ...       -0.299593      -0.194427    -0.027365   \n",
       "...            ...  ...             ...            ...          ...   \n",
       "1173     -0.044707  ...       -0.409541      -0.579376     0.038706   \n",
       "1174     -0.244937  ...       -0.095328      -0.440452     0.303500   \n",
       "1175      0.001571  ...       -0.280119      -0.662673    -0.180936   \n",
       "1176      0.078435  ...       -0.080700      -0.476797    -0.158375   \n",
       "1177      0.009587  ...       -0.038622      -0.667881    -0.090500   \n",
       "\n",
       "      ZXDB (158586)  ZXDC (79364)  ZYG11A (440590)  ZYG11B (79699)  \\\n",
       "0          0.262275      0.163876        -0.249508        0.053725   \n",
       "1          0.087609      0.074897         0.054335       -0.330343   \n",
       "2         -0.055349     -0.000367        -0.205605       -0.066032   \n",
       "3          0.230944      0.044628        -0.081863       -0.390748   \n",
       "4          0.236591     -0.084224        -0.098271       -0.510495   \n",
       "...             ...           ...              ...             ...   \n",
       "1173       0.338610     -0.174379        -0.040264       -0.239763   \n",
       "1174       0.097690      0.151790         0.039994        0.098240   \n",
       "1175       0.169997     -0.036326         0.158702       -0.213281   \n",
       "1176      -0.008999     -0.030058        -0.103513       -0.119234   \n",
       "1177       0.077756      0.055249         0.040616       -0.378505   \n",
       "\n",
       "      ZYX (7791)  ZZEF1 (23140)  ZZZ3 (26009)  \n",
       "0       0.067986       0.136285     -0.438536  \n",
       "1       0.099067       0.274566      0.001871  \n",
       "2      -0.054518       0.035579     -0.150486  \n",
       "3      -0.036547      -0.273129     -0.382723  \n",
       "4       0.052938       0.018623     -0.258353  \n",
       "...          ...            ...           ...  \n",
       "1173    0.065168      -0.144088     -0.334424  \n",
       "1174   -0.266654      -0.115108     -0.080421  \n",
       "1175   -0.017884      -0.195395     -0.309021  \n",
       "1176   -0.153744       0.032584     -0.536071  \n",
       "1177   -0.143920      -0.039985     -0.439905  \n",
       "\n",
       "[1178 rows x 17917 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "raw_crispr_df = pd.read_csv('./data/raw_data/CRISPRGeneEffect.csv')\n",
    "# get the column names of the raw_crispr_df aside from the first column and convert this to a list\n",
    "first_column_name = raw_crispr_df.columns[0]\n",
    "raw_crispr_df.rename(columns={first_column_name: 'Sample'}, inplace=True)\n",
    "display(raw_crispr_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 410,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>A1BG (1)</th>\n",
       "      <th>A1CF (29974)</th>\n",
       "      <th>A2M (2)</th>\n",
       "      <th>A2ML1 (144568)</th>\n",
       "      <th>A3GALT2 (127550)</th>\n",
       "      <th>A4GALT (53947)</th>\n",
       "      <th>A4GNT (51146)</th>\n",
       "      <th>AAAS (8086)</th>\n",
       "      <th>AACS (65985)</th>\n",
       "      <th>...</th>\n",
       "      <th>ZWILCH (55055)</th>\n",
       "      <th>ZWINT (11130)</th>\n",
       "      <th>ZXDA (7789)</th>\n",
       "      <th>ZXDB (158586)</th>\n",
       "      <th>ZXDC (79364)</th>\n",
       "      <th>ZYG11A (440590)</th>\n",
       "      <th>ZYG11B (79699)</th>\n",
       "      <th>ZYX (7791)</th>\n",
       "      <th>ZZEF1 (23140)</th>\n",
       "      <th>ZZZ3 (26009)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>0.042647</td>\n",
       "      <td>0.026102</td>\n",
       "      <td>-0.147614</td>\n",
       "      <td>-0.020425</td>\n",
       "      <td>0.184014</td>\n",
       "      <td>-0.076020</td>\n",
       "      <td>-0.097068</td>\n",
       "      <td>0.182759</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.120536</td>\n",
       "      <td>0.107286</td>\n",
       "      <td>0.166390</td>\n",
       "      <td>0.262275</td>\n",
       "      <td>0.163876</td>\n",
       "      <td>-0.249508</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>0.067986</td>\n",
       "      <td>0.136285</td>\n",
       "      <td>-0.438536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.088500</td>\n",
       "      <td>-0.088198</td>\n",
       "      <td>-0.041225</td>\n",
       "      <td>-0.028074</td>\n",
       "      <td>-0.157206</td>\n",
       "      <td>0.280101</td>\n",
       "      <td>-0.106904</td>\n",
       "      <td>0.178125</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.192527</td>\n",
       "      <td>-0.324059</td>\n",
       "      <td>0.230377</td>\n",
       "      <td>0.087609</td>\n",
       "      <td>0.074897</td>\n",
       "      <td>0.054335</td>\n",
       "      <td>-0.330343</td>\n",
       "      <td>0.099067</td>\n",
       "      <td>0.274566</td>\n",
       "      <td>0.001871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000007</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.113717</td>\n",
       "      <td>0.082872</td>\n",
       "      <td>0.099633</td>\n",
       "      <td>-0.008378</td>\n",
       "      <td>-0.022310</td>\n",
       "      <td>0.014416</td>\n",
       "      <td>-0.184977</td>\n",
       "      <td>-0.173739</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.334843</td>\n",
       "      <td>-0.355499</td>\n",
       "      <td>-0.014183</td>\n",
       "      <td>0.230944</td>\n",
       "      <td>0.044628</td>\n",
       "      <td>-0.081863</td>\n",
       "      <td>-0.390748</td>\n",
       "      <td>-0.036547</td>\n",
       "      <td>-0.273129</td>\n",
       "      <td>-0.382723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000009</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>-0.134669</td>\n",
       "      <td>0.060323</td>\n",
       "      <td>0.076647</td>\n",
       "      <td>0.078922</td>\n",
       "      <td>-0.100243</td>\n",
       "      <td>0.047559</td>\n",
       "      <td>-0.136988</td>\n",
       "      <td>0.037759</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.299593</td>\n",
       "      <td>-0.194427</td>\n",
       "      <td>-0.027365</td>\n",
       "      <td>0.236591</td>\n",
       "      <td>-0.084224</td>\n",
       "      <td>-0.098271</td>\n",
       "      <td>-0.510495</td>\n",
       "      <td>0.052938</td>\n",
       "      <td>0.018623</td>\n",
       "      <td>-0.258353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000012</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>-0.024648</td>\n",
       "      <td>0.034448</td>\n",
       "      <td>0.060509</td>\n",
       "      <td>0.046351</td>\n",
       "      <td>-0.168111</td>\n",
       "      <td>0.166230</td>\n",
       "      <td>-0.094632</td>\n",
       "      <td>-0.044914</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.480750</td>\n",
       "      <td>-0.631959</td>\n",
       "      <td>0.101635</td>\n",
       "      <td>0.032980</td>\n",
       "      <td>-0.086901</td>\n",
       "      <td>0.028423</td>\n",
       "      <td>-0.059623</td>\n",
       "      <td>-0.114437</td>\n",
       "      <td>-0.117129</td>\n",
       "      <td>-0.150634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>ACH-001075</td>\n",
       "      <td>-0.236411</td>\n",
       "      <td>0.045414</td>\n",
       "      <td>0.004463</td>\n",
       "      <td>-0.032736</td>\n",
       "      <td>-0.181405</td>\n",
       "      <td>-0.008243</td>\n",
       "      <td>0.097413</td>\n",
       "      <td>-0.127815</td>\n",
       "      <td>0.057874</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.264254</td>\n",
       "      <td>-0.226804</td>\n",
       "      <td>0.268218</td>\n",
       "      <td>0.435536</td>\n",
       "      <td>-0.007683</td>\n",
       "      <td>-0.013751</td>\n",
       "      <td>-0.239167</td>\n",
       "      <td>0.053218</td>\n",
       "      <td>0.039981</td>\n",
       "      <td>-0.365370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>ACH-001107</td>\n",
       "      <td>-0.098559</td>\n",
       "      <td>-0.025209</td>\n",
       "      <td>0.027718</td>\n",
       "      <td>0.182246</td>\n",
       "      <td>-0.065604</td>\n",
       "      <td>-0.011558</td>\n",
       "      <td>0.216557</td>\n",
       "      <td>-0.146614</td>\n",
       "      <td>-0.227518</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.136001</td>\n",
       "      <td>0.031411</td>\n",
       "      <td>0.075305</td>\n",
       "      <td>0.091786</td>\n",
       "      <td>-0.132919</td>\n",
       "      <td>0.040862</td>\n",
       "      <td>-0.139992</td>\n",
       "      <td>0.191589</td>\n",
       "      <td>-0.022600</td>\n",
       "      <td>-0.507938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>ACH-001190</td>\n",
       "      <td>-0.042641</td>\n",
       "      <td>0.010443</td>\n",
       "      <td>0.162840</td>\n",
       "      <td>-0.065062</td>\n",
       "      <td>-0.048498</td>\n",
       "      <td>-0.254546</td>\n",
       "      <td>-0.015924</td>\n",
       "      <td>-0.248677</td>\n",
       "      <td>0.092303</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.191808</td>\n",
       "      <td>-0.282360</td>\n",
       "      <td>0.072387</td>\n",
       "      <td>0.102600</td>\n",
       "      <td>0.043955</td>\n",
       "      <td>-0.141153</td>\n",
       "      <td>-0.266564</td>\n",
       "      <td>0.029695</td>\n",
       "      <td>-0.095837</td>\n",
       "      <td>-0.100670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>ACH-001306</td>\n",
       "      <td>-0.031081</td>\n",
       "      <td>-0.000523</td>\n",
       "      <td>-0.036076</td>\n",
       "      <td>-0.012972</td>\n",
       "      <td>0.009998</td>\n",
       "      <td>-0.044483</td>\n",
       "      <td>0.027619</td>\n",
       "      <td>-0.017077</td>\n",
       "      <td>0.002215</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.014357</td>\n",
       "      <td>-0.427889</td>\n",
       "      <td>-0.060975</td>\n",
       "      <td>-0.018635</td>\n",
       "      <td>-0.170612</td>\n",
       "      <td>0.036136</td>\n",
       "      <td>-0.130743</td>\n",
       "      <td>-0.092105</td>\n",
       "      <td>-0.462014</td>\n",
       "      <td>-0.268095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>ACH-001307</td>\n",
       "      <td>0.012357</td>\n",
       "      <td>-0.107587</td>\n",
       "      <td>0.108407</td>\n",
       "      <td>0.203668</td>\n",
       "      <td>-0.006054</td>\n",
       "      <td>-0.144883</td>\n",
       "      <td>0.063335</td>\n",
       "      <td>-0.178241</td>\n",
       "      <td>0.079465</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.047072</td>\n",
       "      <td>-0.367632</td>\n",
       "      <td>-0.111287</td>\n",
       "      <td>-0.094867</td>\n",
       "      <td>0.120143</td>\n",
       "      <td>-0.096073</td>\n",
       "      <td>-0.137811</td>\n",
       "      <td>-0.170982</td>\n",
       "      <td>-0.051311</td>\n",
       "      <td>-0.417389</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>300 rows × 17917 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Sample  A1BG (1)  A1CF (29974)   A2M (2)  A2ML1 (144568)  \\\n",
       "0    ACH-000001 -0.121964      0.042647  0.026102       -0.147614   \n",
       "1    ACH-000004 -0.016484     -0.088500 -0.088198       -0.041225   \n",
       "2    ACH-000007 -0.071921     -0.113717  0.082872        0.099633   \n",
       "3    ACH-000009 -0.019163     -0.134669  0.060323        0.076647   \n",
       "4    ACH-000012 -0.139496     -0.024648  0.034448        0.060509   \n",
       "..          ...       ...           ...       ...             ...   \n",
       "295  ACH-001075 -0.236411      0.045414  0.004463       -0.032736   \n",
       "296  ACH-001107 -0.098559     -0.025209  0.027718        0.182246   \n",
       "297  ACH-001190 -0.042641      0.010443  0.162840       -0.065062   \n",
       "298  ACH-001306 -0.031081     -0.000523 -0.036076       -0.012972   \n",
       "299  ACH-001307  0.012357     -0.107587  0.108407        0.203668   \n",
       "\n",
       "     A3GALT2 (127550)  A4GALT (53947)  A4GNT (51146)  AAAS (8086)  \\\n",
       "0           -0.020425        0.184014      -0.076020    -0.097068   \n",
       "1           -0.028074       -0.157206       0.280101    -0.106904   \n",
       "2           -0.008378       -0.022310       0.014416    -0.184977   \n",
       "3            0.078922       -0.100243       0.047559    -0.136988   \n",
       "4            0.046351       -0.168111       0.166230    -0.094632   \n",
       "..                ...             ...            ...          ...   \n",
       "295         -0.181405       -0.008243       0.097413    -0.127815   \n",
       "296         -0.065604       -0.011558       0.216557    -0.146614   \n",
       "297         -0.048498       -0.254546      -0.015924    -0.248677   \n",
       "298          0.009998       -0.044483       0.027619    -0.017077   \n",
       "299         -0.006054       -0.144883       0.063335    -0.178241   \n",
       "\n",
       "     AACS (65985)  ...  ZWILCH (55055)  ZWINT (11130)  ZXDA (7789)  \\\n",
       "0        0.182759  ...       -0.120536       0.107286     0.166390   \n",
       "1        0.178125  ...       -0.192527      -0.324059     0.230377   \n",
       "2       -0.173739  ...       -0.334843      -0.355499    -0.014183   \n",
       "3        0.037759  ...       -0.299593      -0.194427    -0.027365   \n",
       "4       -0.044914  ...       -0.480750      -0.631959     0.101635   \n",
       "..            ...  ...             ...            ...          ...   \n",
       "295      0.057874  ...       -0.264254      -0.226804     0.268218   \n",
       "296     -0.227518  ...       -0.136001       0.031411     0.075305   \n",
       "297      0.092303  ...       -0.191808      -0.282360     0.072387   \n",
       "298      0.002215  ...       -0.014357      -0.427889    -0.060975   \n",
       "299      0.079465  ...       -0.047072      -0.367632    -0.111287   \n",
       "\n",
       "     ZXDB (158586)  ZXDC (79364)  ZYG11A (440590)  ZYG11B (79699)  ZYX (7791)  \\\n",
       "0         0.262275      0.163876        -0.249508        0.053725    0.067986   \n",
       "1         0.087609      0.074897         0.054335       -0.330343    0.099067   \n",
       "2         0.230944      0.044628        -0.081863       -0.390748   -0.036547   \n",
       "3         0.236591     -0.084224        -0.098271       -0.510495    0.052938   \n",
       "4         0.032980     -0.086901         0.028423       -0.059623   -0.114437   \n",
       "..             ...           ...              ...             ...         ...   \n",
       "295       0.435536     -0.007683        -0.013751       -0.239167    0.053218   \n",
       "296       0.091786     -0.132919         0.040862       -0.139992    0.191589   \n",
       "297       0.102600      0.043955        -0.141153       -0.266564    0.029695   \n",
       "298      -0.018635     -0.170612         0.036136       -0.130743   -0.092105   \n",
       "299      -0.094867      0.120143        -0.096073       -0.137811   -0.170982   \n",
       "\n",
       "     ZZEF1 (23140)  ZZZ3 (26009)  \n",
       "0         0.136285     -0.438536  \n",
       "1         0.274566      0.001871  \n",
       "2        -0.273129     -0.382723  \n",
       "3         0.018623     -0.258353  \n",
       "4        -0.117129     -0.150634  \n",
       "..             ...           ...  \n",
       "295       0.039981     -0.365370  \n",
       "296      -0.022600     -0.507938  \n",
       "297      -0.095837     -0.100670  \n",
       "298      -0.462014     -0.268095  \n",
       "299      -0.051311     -0.417389  \n",
       "\n",
       "[300 rows x 17917 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>A1BG (1)</th>\n",
       "      <th>A1CF (29974)</th>\n",
       "      <th>A2M (2)</th>\n",
       "      <th>A2ML1 (144568)</th>\n",
       "      <th>A3GALT2 (127550)</th>\n",
       "      <th>A4GALT (53947)</th>\n",
       "      <th>A4GNT (51146)</th>\n",
       "      <th>AAAS (8086)</th>\n",
       "      <th>AACS (65985)</th>\n",
       "      <th>...</th>\n",
       "      <th>ZWILCH (55055)</th>\n",
       "      <th>ZWINT (11130)</th>\n",
       "      <th>ZXDA (7789)</th>\n",
       "      <th>ZXDB (158586)</th>\n",
       "      <th>ZXDC (79364)</th>\n",
       "      <th>ZYG11A (440590)</th>\n",
       "      <th>ZYG11B (79699)</th>\n",
       "      <th>ZYX (7791)</th>\n",
       "      <th>ZZEF1 (23140)</th>\n",
       "      <th>ZZZ3 (26009)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000015</td>\n",
       "      <td>-0.164485</td>\n",
       "      <td>-0.062450</td>\n",
       "      <td>-0.042585</td>\n",
       "      <td>0.067254</td>\n",
       "      <td>-0.029782</td>\n",
       "      <td>-0.021019</td>\n",
       "      <td>0.107219</td>\n",
       "      <td>-0.141959</td>\n",
       "      <td>0.017586</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.356899</td>\n",
       "      <td>-0.161527</td>\n",
       "      <td>0.051580</td>\n",
       "      <td>0.083171</td>\n",
       "      <td>0.021982</td>\n",
       "      <td>-0.016426</td>\n",
       "      <td>-0.152085</td>\n",
       "      <td>-0.153389</td>\n",
       "      <td>-0.130675</td>\n",
       "      <td>-0.183547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000030</td>\n",
       "      <td>0.092091</td>\n",
       "      <td>-0.235943</td>\n",
       "      <td>-0.008186</td>\n",
       "      <td>0.068138</td>\n",
       "      <td>-0.089160</td>\n",
       "      <td>-0.211510</td>\n",
       "      <td>0.079939</td>\n",
       "      <td>-0.243709</td>\n",
       "      <td>0.041970</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024061</td>\n",
       "      <td>-0.252046</td>\n",
       "      <td>-0.006676</td>\n",
       "      <td>0.046137</td>\n",
       "      <td>-0.144591</td>\n",
       "      <td>-0.081285</td>\n",
       "      <td>-0.236948</td>\n",
       "      <td>-0.013121</td>\n",
       "      <td>-0.126559</td>\n",
       "      <td>-0.249220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000055</td>\n",
       "      <td>0.023233</td>\n",
       "      <td>0.023197</td>\n",
       "      <td>0.133568</td>\n",
       "      <td>0.097102</td>\n",
       "      <td>-0.165176</td>\n",
       "      <td>0.037308</td>\n",
       "      <td>-0.118947</td>\n",
       "      <td>0.079731</td>\n",
       "      <td>-0.254460</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.343143</td>\n",
       "      <td>-0.239665</td>\n",
       "      <td>0.107491</td>\n",
       "      <td>-0.188749</td>\n",
       "      <td>0.127635</td>\n",
       "      <td>-0.064280</td>\n",
       "      <td>-0.049646</td>\n",
       "      <td>-0.428021</td>\n",
       "      <td>-0.172303</td>\n",
       "      <td>-0.369053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000070</td>\n",
       "      <td>-0.023093</td>\n",
       "      <td>-0.098678</td>\n",
       "      <td>0.055092</td>\n",
       "      <td>0.211111</td>\n",
       "      <td>-0.202141</td>\n",
       "      <td>0.011529</td>\n",
       "      <td>0.030910</td>\n",
       "      <td>0.110055</td>\n",
       "      <td>0.016656</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.321510</td>\n",
       "      <td>-0.045616</td>\n",
       "      <td>0.132882</td>\n",
       "      <td>0.262532</td>\n",
       "      <td>-0.067119</td>\n",
       "      <td>-0.181128</td>\n",
       "      <td>-0.052296</td>\n",
       "      <td>-0.186571</td>\n",
       "      <td>-0.137025</td>\n",
       "      <td>-0.269073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000075</td>\n",
       "      <td>-0.030929</td>\n",
       "      <td>0.173555</td>\n",
       "      <td>0.144853</td>\n",
       "      <td>0.147002</td>\n",
       "      <td>-0.232510</td>\n",
       "      <td>-0.183895</td>\n",
       "      <td>0.099746</td>\n",
       "      <td>0.078998</td>\n",
       "      <td>0.051277</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.190412</td>\n",
       "      <td>-0.830687</td>\n",
       "      <td>0.117279</td>\n",
       "      <td>0.029483</td>\n",
       "      <td>-0.098088</td>\n",
       "      <td>-0.138234</td>\n",
       "      <td>-0.209159</td>\n",
       "      <td>0.083589</td>\n",
       "      <td>-0.108345</td>\n",
       "      <td>-0.496251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>ACH-000950</td>\n",
       "      <td>-0.086066</td>\n",
       "      <td>-0.070168</td>\n",
       "      <td>0.068361</td>\n",
       "      <td>-0.114017</td>\n",
       "      <td>-0.143689</td>\n",
       "      <td>-0.030228</td>\n",
       "      <td>-0.042546</td>\n",
       "      <td>-0.084736</td>\n",
       "      <td>-0.077060</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.668693</td>\n",
       "      <td>-0.964036</td>\n",
       "      <td>0.239076</td>\n",
       "      <td>0.344992</td>\n",
       "      <td>-0.080482</td>\n",
       "      <td>-0.047990</td>\n",
       "      <td>-0.068670</td>\n",
       "      <td>-0.050717</td>\n",
       "      <td>0.021737</td>\n",
       "      <td>-0.174756</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>ACH-000957</td>\n",
       "      <td>-0.211048</td>\n",
       "      <td>-0.137014</td>\n",
       "      <td>0.085481</td>\n",
       "      <td>0.030893</td>\n",
       "      <td>-0.046826</td>\n",
       "      <td>-0.094417</td>\n",
       "      <td>-0.029150</td>\n",
       "      <td>-0.233241</td>\n",
       "      <td>-0.075270</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.965635</td>\n",
       "      <td>-0.945276</td>\n",
       "      <td>0.184207</td>\n",
       "      <td>0.279205</td>\n",
       "      <td>0.064601</td>\n",
       "      <td>-0.054118</td>\n",
       "      <td>-0.140052</td>\n",
       "      <td>-0.014237</td>\n",
       "      <td>-0.022187</td>\n",
       "      <td>-0.287196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>ACH-000969</td>\n",
       "      <td>0.136178</td>\n",
       "      <td>-0.088978</td>\n",
       "      <td>0.088356</td>\n",
       "      <td>-0.058193</td>\n",
       "      <td>0.071555</td>\n",
       "      <td>-0.124653</td>\n",
       "      <td>-0.067000</td>\n",
       "      <td>-0.142984</td>\n",
       "      <td>-0.004930</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.148537</td>\n",
       "      <td>-0.864126</td>\n",
       "      <td>0.132900</td>\n",
       "      <td>0.133574</td>\n",
       "      <td>-0.106611</td>\n",
       "      <td>0.116770</td>\n",
       "      <td>-0.041387</td>\n",
       "      <td>0.051083</td>\n",
       "      <td>-0.051369</td>\n",
       "      <td>-0.146886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>ACH-000971</td>\n",
       "      <td>0.133352</td>\n",
       "      <td>-0.038505</td>\n",
       "      <td>0.077262</td>\n",
       "      <td>-0.084336</td>\n",
       "      <td>-0.093124</td>\n",
       "      <td>0.049209</td>\n",
       "      <td>0.037526</td>\n",
       "      <td>-0.126809</td>\n",
       "      <td>-0.033475</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.541773</td>\n",
       "      <td>-1.170882</td>\n",
       "      <td>0.032318</td>\n",
       "      <td>0.280006</td>\n",
       "      <td>0.065551</td>\n",
       "      <td>0.089177</td>\n",
       "      <td>-0.284905</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>-0.083344</td>\n",
       "      <td>-0.227688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>ACH-000977</td>\n",
       "      <td>-0.081687</td>\n",
       "      <td>-0.032364</td>\n",
       "      <td>0.129437</td>\n",
       "      <td>0.105632</td>\n",
       "      <td>-0.093056</td>\n",
       "      <td>-0.083073</td>\n",
       "      <td>0.101227</td>\n",
       "      <td>-0.108094</td>\n",
       "      <td>0.100903</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.210010</td>\n",
       "      <td>-0.364542</td>\n",
       "      <td>-0.047402</td>\n",
       "      <td>0.198331</td>\n",
       "      <td>-0.007095</td>\n",
       "      <td>0.061751</td>\n",
       "      <td>-0.122563</td>\n",
       "      <td>-0.139035</td>\n",
       "      <td>-0.179099</td>\n",
       "      <td>-0.133845</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>63 rows × 17917 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Sample  A1BG (1)  A1CF (29974)   A2M (2)  A2ML1 (144568)  \\\n",
       "0   ACH-000015 -0.164485     -0.062450 -0.042585        0.067254   \n",
       "1   ACH-000030  0.092091     -0.235943 -0.008186        0.068138   \n",
       "2   ACH-000055  0.023233      0.023197  0.133568        0.097102   \n",
       "3   ACH-000070 -0.023093     -0.098678  0.055092        0.211111   \n",
       "4   ACH-000075 -0.030929      0.173555  0.144853        0.147002   \n",
       "..         ...       ...           ...       ...             ...   \n",
       "58  ACH-000950 -0.086066     -0.070168  0.068361       -0.114017   \n",
       "59  ACH-000957 -0.211048     -0.137014  0.085481        0.030893   \n",
       "60  ACH-000969  0.136178     -0.088978  0.088356       -0.058193   \n",
       "61  ACH-000971  0.133352     -0.038505  0.077262       -0.084336   \n",
       "62  ACH-000977 -0.081687     -0.032364  0.129437        0.105632   \n",
       "\n",
       "    A3GALT2 (127550)  A4GALT (53947)  A4GNT (51146)  AAAS (8086)  \\\n",
       "0          -0.029782       -0.021019       0.107219    -0.141959   \n",
       "1          -0.089160       -0.211510       0.079939    -0.243709   \n",
       "2          -0.165176        0.037308      -0.118947     0.079731   \n",
       "3          -0.202141        0.011529       0.030910     0.110055   \n",
       "4          -0.232510       -0.183895       0.099746     0.078998   \n",
       "..               ...             ...            ...          ...   \n",
       "58         -0.143689       -0.030228      -0.042546    -0.084736   \n",
       "59         -0.046826       -0.094417      -0.029150    -0.233241   \n",
       "60          0.071555       -0.124653      -0.067000    -0.142984   \n",
       "61         -0.093124        0.049209       0.037526    -0.126809   \n",
       "62         -0.093056       -0.083073       0.101227    -0.108094   \n",
       "\n",
       "    AACS (65985)  ...  ZWILCH (55055)  ZWINT (11130)  ZXDA (7789)  \\\n",
       "0       0.017586  ...       -0.356899      -0.161527     0.051580   \n",
       "1       0.041970  ...        0.024061      -0.252046    -0.006676   \n",
       "2      -0.254460  ...       -0.343143      -0.239665     0.107491   \n",
       "3       0.016656  ...       -0.321510      -0.045616     0.132882   \n",
       "4       0.051277  ...       -0.190412      -0.830687     0.117279   \n",
       "..           ...  ...             ...            ...          ...   \n",
       "58     -0.077060  ...       -0.668693      -0.964036     0.239076   \n",
       "59     -0.075270  ...       -0.965635      -0.945276     0.184207   \n",
       "60     -0.004930  ...       -0.148537      -0.864126     0.132900   \n",
       "61     -0.033475  ...       -0.541773      -1.170882     0.032318   \n",
       "62      0.100903  ...       -0.210010      -0.364542    -0.047402   \n",
       "\n",
       "    ZXDB (158586)  ZXDC (79364)  ZYG11A (440590)  ZYG11B (79699)  ZYX (7791)  \\\n",
       "0        0.083171      0.021982        -0.016426       -0.152085   -0.153389   \n",
       "1        0.046137     -0.144591        -0.081285       -0.236948   -0.013121   \n",
       "2       -0.188749      0.127635        -0.064280       -0.049646   -0.428021   \n",
       "3        0.262532     -0.067119        -0.181128       -0.052296   -0.186571   \n",
       "4        0.029483     -0.098088        -0.138234       -0.209159    0.083589   \n",
       "..            ...           ...              ...             ...         ...   \n",
       "58       0.344992     -0.080482        -0.047990       -0.068670   -0.050717   \n",
       "59       0.279205      0.064601        -0.054118       -0.140052   -0.014237   \n",
       "60       0.133574     -0.106611         0.116770       -0.041387    0.051083   \n",
       "61       0.280006      0.065551         0.089177       -0.284905    0.000105   \n",
       "62       0.198331     -0.007095         0.061751       -0.122563   -0.139035   \n",
       "\n",
       "    ZZEF1 (23140)  ZZZ3 (26009)  \n",
       "0       -0.130675     -0.183547  \n",
       "1       -0.126559     -0.249220  \n",
       "2       -0.172303     -0.369053  \n",
       "3       -0.137025     -0.269073  \n",
       "4       -0.108345     -0.496251  \n",
       "..            ...           ...  \n",
       "58       0.021737     -0.174756  \n",
       "59      -0.022187     -0.287196  \n",
       "60      -0.051369     -0.146886  \n",
       "61      -0.083344     -0.227688  \n",
       "62      -0.179099     -0.133845  \n",
       "\n",
       "[63 rows x 17917 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Sample</th>\n",
       "      <th>HGNC_Symbol</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000018</th>\n",
       "      <th>ACH-000019</th>\n",
       "      <th>ACH-000021</th>\n",
       "      <th>ACH-000023</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000988</th>\n",
       "      <th>ACH-000991</th>\n",
       "      <th>ACH-000997</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001065</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG (1)</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>-0.089959</td>\n",
       "      <td>-0.000655</td>\n",
       "      <td>-0.144838</td>\n",
       "      <td>-0.044006</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.080952</td>\n",
       "      <td>-0.086003</td>\n",
       "      <td>0.010321</td>\n",
       "      <td>-0.280963</td>\n",
       "      <td>-0.054307</td>\n",
       "      <td>-0.236411</td>\n",
       "      <td>-0.098559</td>\n",
       "      <td>-0.042641</td>\n",
       "      <td>-0.031081</td>\n",
       "      <td>0.012357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF (29974)</td>\n",
       "      <td>0.042647</td>\n",
       "      <td>-0.088500</td>\n",
       "      <td>-0.113717</td>\n",
       "      <td>-0.134669</td>\n",
       "      <td>-0.024648</td>\n",
       "      <td>-0.025220</td>\n",
       "      <td>0.010486</td>\n",
       "      <td>-0.190572</td>\n",
       "      <td>-0.135335</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.139068</td>\n",
       "      <td>-0.833193</td>\n",
       "      <td>-0.260665</td>\n",
       "      <td>-0.673329</td>\n",
       "      <td>-0.052153</td>\n",
       "      <td>0.045414</td>\n",
       "      <td>-0.025209</td>\n",
       "      <td>0.010443</td>\n",
       "      <td>-0.000523</td>\n",
       "      <td>-0.107587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M (2)</td>\n",
       "      <td>0.026102</td>\n",
       "      <td>-0.088198</td>\n",
       "      <td>0.082872</td>\n",
       "      <td>0.060323</td>\n",
       "      <td>0.034448</td>\n",
       "      <td>-0.247578</td>\n",
       "      <td>0.100446</td>\n",
       "      <td>-0.033688</td>\n",
       "      <td>-0.014753</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024197</td>\n",
       "      <td>-0.230081</td>\n",
       "      <td>0.022584</td>\n",
       "      <td>0.010727</td>\n",
       "      <td>-0.023250</td>\n",
       "      <td>0.004463</td>\n",
       "      <td>0.027718</td>\n",
       "      <td>0.162840</td>\n",
       "      <td>-0.036076</td>\n",
       "      <td>0.108407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ML1 (144568)</td>\n",
       "      <td>-0.147614</td>\n",
       "      <td>-0.041225</td>\n",
       "      <td>0.099633</td>\n",
       "      <td>0.076647</td>\n",
       "      <td>0.060509</td>\n",
       "      <td>0.099575</td>\n",
       "      <td>0.044937</td>\n",
       "      <td>-0.029677</td>\n",
       "      <td>-0.112628</td>\n",
       "      <td>...</td>\n",
       "      <td>0.033499</td>\n",
       "      <td>0.302444</td>\n",
       "      <td>0.202078</td>\n",
       "      <td>0.239772</td>\n",
       "      <td>0.269846</td>\n",
       "      <td>-0.032736</td>\n",
       "      <td>0.182246</td>\n",
       "      <td>-0.065062</td>\n",
       "      <td>-0.012972</td>\n",
       "      <td>0.203668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2 (127550)</td>\n",
       "      <td>-0.020425</td>\n",
       "      <td>-0.028074</td>\n",
       "      <td>-0.008378</td>\n",
       "      <td>0.078922</td>\n",
       "      <td>0.046351</td>\n",
       "      <td>0.000525</td>\n",
       "      <td>-0.073545</td>\n",
       "      <td>-0.067062</td>\n",
       "      <td>-0.043320</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.105600</td>\n",
       "      <td>-0.147492</td>\n",
       "      <td>0.019020</td>\n",
       "      <td>-0.084180</td>\n",
       "      <td>-0.065527</td>\n",
       "      <td>-0.181405</td>\n",
       "      <td>-0.065604</td>\n",
       "      <td>-0.048498</td>\n",
       "      <td>0.009998</td>\n",
       "      <td>-0.006054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17911</th>\n",
       "      <td>ZYG11A (440590)</td>\n",
       "      <td>-0.249508</td>\n",
       "      <td>0.054335</td>\n",
       "      <td>-0.081863</td>\n",
       "      <td>-0.098271</td>\n",
       "      <td>0.028423</td>\n",
       "      <td>-0.087202</td>\n",
       "      <td>-0.047248</td>\n",
       "      <td>-0.056765</td>\n",
       "      <td>-0.078996</td>\n",
       "      <td>...</td>\n",
       "      <td>0.075397</td>\n",
       "      <td>0.162973</td>\n",
       "      <td>-0.000810</td>\n",
       "      <td>0.182418</td>\n",
       "      <td>-0.210634</td>\n",
       "      <td>-0.013751</td>\n",
       "      <td>0.040862</td>\n",
       "      <td>-0.141153</td>\n",
       "      <td>0.036136</td>\n",
       "      <td>-0.096073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17912</th>\n",
       "      <td>ZYG11B (79699)</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>-0.330343</td>\n",
       "      <td>-0.390748</td>\n",
       "      <td>-0.510495</td>\n",
       "      <td>-0.059623</td>\n",
       "      <td>-0.046460</td>\n",
       "      <td>-0.164099</td>\n",
       "      <td>-0.259650</td>\n",
       "      <td>-0.167981</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.073492</td>\n",
       "      <td>-0.081934</td>\n",
       "      <td>-0.069572</td>\n",
       "      <td>-0.042853</td>\n",
       "      <td>-0.205769</td>\n",
       "      <td>-0.239167</td>\n",
       "      <td>-0.139992</td>\n",
       "      <td>-0.266564</td>\n",
       "      <td>-0.130743</td>\n",
       "      <td>-0.137811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17913</th>\n",
       "      <td>ZYX (7791)</td>\n",
       "      <td>0.067986</td>\n",
       "      <td>0.099067</td>\n",
       "      <td>-0.036547</td>\n",
       "      <td>0.052938</td>\n",
       "      <td>-0.114437</td>\n",
       "      <td>-0.003761</td>\n",
       "      <td>0.168613</td>\n",
       "      <td>-0.161969</td>\n",
       "      <td>0.216434</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.022618</td>\n",
       "      <td>0.151163</td>\n",
       "      <td>0.079822</td>\n",
       "      <td>0.079633</td>\n",
       "      <td>0.285485</td>\n",
       "      <td>0.053218</td>\n",
       "      <td>0.191589</td>\n",
       "      <td>0.029695</td>\n",
       "      <td>-0.092105</td>\n",
       "      <td>-0.170982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17914</th>\n",
       "      <td>ZZEF1 (23140)</td>\n",
       "      <td>0.136285</td>\n",
       "      <td>0.274566</td>\n",
       "      <td>-0.273129</td>\n",
       "      <td>0.018623</td>\n",
       "      <td>-0.117129</td>\n",
       "      <td>-0.318758</td>\n",
       "      <td>-0.093001</td>\n",
       "      <td>-0.188478</td>\n",
       "      <td>-0.165553</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.176936</td>\n",
       "      <td>-0.007172</td>\n",
       "      <td>-0.148326</td>\n",
       "      <td>-0.095262</td>\n",
       "      <td>-0.011798</td>\n",
       "      <td>0.039981</td>\n",
       "      <td>-0.022600</td>\n",
       "      <td>-0.095837</td>\n",
       "      <td>-0.462014</td>\n",
       "      <td>-0.051311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17915</th>\n",
       "      <td>ZZZ3 (26009)</td>\n",
       "      <td>-0.438536</td>\n",
       "      <td>0.001871</td>\n",
       "      <td>-0.382723</td>\n",
       "      <td>-0.258353</td>\n",
       "      <td>-0.150634</td>\n",
       "      <td>-0.132434</td>\n",
       "      <td>-0.413203</td>\n",
       "      <td>-0.264889</td>\n",
       "      <td>-0.320176</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.361119</td>\n",
       "      <td>-0.228395</td>\n",
       "      <td>0.062976</td>\n",
       "      <td>-0.448808</td>\n",
       "      <td>-0.243747</td>\n",
       "      <td>-0.365370</td>\n",
       "      <td>-0.507938</td>\n",
       "      <td>-0.100670</td>\n",
       "      <td>-0.268095</td>\n",
       "      <td>-0.417389</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17916 rows × 301 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Sample       HGNC_Symbol  ACH-000001  ACH-000004  ACH-000007  ACH-000009  \\\n",
       "0               A1BG (1)   -0.121964   -0.016484   -0.071921   -0.019163   \n",
       "1           A1CF (29974)    0.042647   -0.088500   -0.113717   -0.134669   \n",
       "2                A2M (2)    0.026102   -0.088198    0.082872    0.060323   \n",
       "3         A2ML1 (144568)   -0.147614   -0.041225    0.099633    0.076647   \n",
       "4       A3GALT2 (127550)   -0.020425   -0.028074   -0.008378    0.078922   \n",
       "...                  ...         ...         ...         ...         ...   \n",
       "17911    ZYG11A (440590)   -0.249508    0.054335   -0.081863   -0.098271   \n",
       "17912     ZYG11B (79699)    0.053725   -0.330343   -0.390748   -0.510495   \n",
       "17913         ZYX (7791)    0.067986    0.099067   -0.036547    0.052938   \n",
       "17914      ZZEF1 (23140)    0.136285    0.274566   -0.273129    0.018623   \n",
       "17915       ZZZ3 (26009)   -0.438536    0.001871   -0.382723   -0.258353   \n",
       "\n",
       "Sample  ACH-000012  ACH-000018  ACH-000019  ACH-000021  ACH-000023  ...  \\\n",
       "0        -0.139496   -0.089959   -0.000655   -0.144838   -0.044006  ...   \n",
       "1        -0.024648   -0.025220    0.010486   -0.190572   -0.135335  ...   \n",
       "2         0.034448   -0.247578    0.100446   -0.033688   -0.014753  ...   \n",
       "3         0.060509    0.099575    0.044937   -0.029677   -0.112628  ...   \n",
       "4         0.046351    0.000525   -0.073545   -0.067062   -0.043320  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "17911     0.028423   -0.087202   -0.047248   -0.056765   -0.078996  ...   \n",
       "17912    -0.059623   -0.046460   -0.164099   -0.259650   -0.167981  ...   \n",
       "17913    -0.114437   -0.003761    0.168613   -0.161969    0.216434  ...   \n",
       "17914    -0.117129   -0.318758   -0.093001   -0.188478   -0.165553  ...   \n",
       "17915    -0.150634   -0.132434   -0.413203   -0.264889   -0.320176  ...   \n",
       "\n",
       "Sample  ACH-000988  ACH-000991  ACH-000997  ACH-001039  ACH-001065  \\\n",
       "0        -0.080952   -0.086003    0.010321   -0.280963   -0.054307   \n",
       "1        -0.139068   -0.833193   -0.260665   -0.673329   -0.052153   \n",
       "2         0.024197   -0.230081    0.022584    0.010727   -0.023250   \n",
       "3         0.033499    0.302444    0.202078    0.239772    0.269846   \n",
       "4        -0.105600   -0.147492    0.019020   -0.084180   -0.065527   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "17911     0.075397    0.162973   -0.000810    0.182418   -0.210634   \n",
       "17912    -0.073492   -0.081934   -0.069572   -0.042853   -0.205769   \n",
       "17913    -0.022618    0.151163    0.079822    0.079633    0.285485   \n",
       "17914    -0.176936   -0.007172   -0.148326   -0.095262   -0.011798   \n",
       "17915    -0.361119   -0.228395    0.062976   -0.448808   -0.243747   \n",
       "\n",
       "Sample  ACH-001075  ACH-001107  ACH-001190  ACH-001306  ACH-001307  \n",
       "0        -0.236411   -0.098559   -0.042641   -0.031081    0.012357  \n",
       "1         0.045414   -0.025209    0.010443   -0.000523   -0.107587  \n",
       "2         0.004463    0.027718    0.162840   -0.036076    0.108407  \n",
       "3        -0.032736    0.182246   -0.065062   -0.012972    0.203668  \n",
       "4        -0.181405   -0.065604   -0.048498    0.009998   -0.006054  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "17911    -0.013751    0.040862   -0.141153    0.036136   -0.096073  \n",
       "17912    -0.239167   -0.139992   -0.266564   -0.130743   -0.137811  \n",
       "17913     0.053218    0.191589    0.029695   -0.092105   -0.170982  \n",
       "17914     0.039981   -0.022600   -0.095837   -0.462014   -0.051311  \n",
       "17915    -0.365370   -0.507938   -0.100670   -0.268095   -0.417389  \n",
       "\n",
       "[17916 rows x 301 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Sample</th>\n",
       "      <th>HGNC_Symbol</th>\n",
       "      <th>ACH-000015</th>\n",
       "      <th>ACH-000030</th>\n",
       "      <th>ACH-000055</th>\n",
       "      <th>ACH-000070</th>\n",
       "      <th>ACH-000075</th>\n",
       "      <th>ACH-000098</th>\n",
       "      <th>ACH-000100</th>\n",
       "      <th>ACH-000132</th>\n",
       "      <th>ACH-000138</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000880</th>\n",
       "      <th>ACH-000885</th>\n",
       "      <th>ACH-000900</th>\n",
       "      <th>ACH-000915</th>\n",
       "      <th>ACH-000919</th>\n",
       "      <th>ACH-000950</th>\n",
       "      <th>ACH-000957</th>\n",
       "      <th>ACH-000969</th>\n",
       "      <th>ACH-000971</th>\n",
       "      <th>ACH-000977</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG (1)</td>\n",
       "      <td>-0.164485</td>\n",
       "      <td>0.092091</td>\n",
       "      <td>0.023233</td>\n",
       "      <td>-0.023093</td>\n",
       "      <td>-0.030929</td>\n",
       "      <td>-0.037187</td>\n",
       "      <td>0.021787</td>\n",
       "      <td>-0.122873</td>\n",
       "      <td>-0.102403</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.210218</td>\n",
       "      <td>-0.276661</td>\n",
       "      <td>-0.073962</td>\n",
       "      <td>-0.037085</td>\n",
       "      <td>-0.099006</td>\n",
       "      <td>-0.086066</td>\n",
       "      <td>-0.211048</td>\n",
       "      <td>0.136178</td>\n",
       "      <td>0.133352</td>\n",
       "      <td>-0.081687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF (29974)</td>\n",
       "      <td>-0.062450</td>\n",
       "      <td>-0.235943</td>\n",
       "      <td>0.023197</td>\n",
       "      <td>-0.098678</td>\n",
       "      <td>0.173555</td>\n",
       "      <td>-0.137049</td>\n",
       "      <td>0.138818</td>\n",
       "      <td>0.072059</td>\n",
       "      <td>-0.117266</td>\n",
       "      <td>...</td>\n",
       "      <td>0.023075</td>\n",
       "      <td>0.002950</td>\n",
       "      <td>-0.066371</td>\n",
       "      <td>-0.040370</td>\n",
       "      <td>-0.154016</td>\n",
       "      <td>-0.070168</td>\n",
       "      <td>-0.137014</td>\n",
       "      <td>-0.088978</td>\n",
       "      <td>-0.038505</td>\n",
       "      <td>-0.032364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M (2)</td>\n",
       "      <td>-0.042585</td>\n",
       "      <td>-0.008186</td>\n",
       "      <td>0.133568</td>\n",
       "      <td>0.055092</td>\n",
       "      <td>0.144853</td>\n",
       "      <td>0.070069</td>\n",
       "      <td>0.219874</td>\n",
       "      <td>0.038849</td>\n",
       "      <td>-0.053961</td>\n",
       "      <td>...</td>\n",
       "      <td>0.169803</td>\n",
       "      <td>0.115853</td>\n",
       "      <td>0.051874</td>\n",
       "      <td>-0.013556</td>\n",
       "      <td>0.221679</td>\n",
       "      <td>0.068361</td>\n",
       "      <td>0.085481</td>\n",
       "      <td>0.088356</td>\n",
       "      <td>0.077262</td>\n",
       "      <td>0.129437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ML1 (144568)</td>\n",
       "      <td>0.067254</td>\n",
       "      <td>0.068138</td>\n",
       "      <td>0.097102</td>\n",
       "      <td>0.211111</td>\n",
       "      <td>0.147002</td>\n",
       "      <td>0.010128</td>\n",
       "      <td>0.057229</td>\n",
       "      <td>0.039049</td>\n",
       "      <td>-0.071758</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.106526</td>\n",
       "      <td>0.193006</td>\n",
       "      <td>0.005611</td>\n",
       "      <td>0.139576</td>\n",
       "      <td>0.077502</td>\n",
       "      <td>-0.114017</td>\n",
       "      <td>0.030893</td>\n",
       "      <td>-0.058193</td>\n",
       "      <td>-0.084336</td>\n",
       "      <td>0.105632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2 (127550)</td>\n",
       "      <td>-0.029782</td>\n",
       "      <td>-0.089160</td>\n",
       "      <td>-0.165176</td>\n",
       "      <td>-0.202141</td>\n",
       "      <td>-0.232510</td>\n",
       "      <td>-0.112067</td>\n",
       "      <td>-0.167169</td>\n",
       "      <td>-0.209066</td>\n",
       "      <td>0.073647</td>\n",
       "      <td>...</td>\n",
       "      <td>0.077082</td>\n",
       "      <td>-0.010742</td>\n",
       "      <td>0.056990</td>\n",
       "      <td>-0.319709</td>\n",
       "      <td>-0.184296</td>\n",
       "      <td>-0.143689</td>\n",
       "      <td>-0.046826</td>\n",
       "      <td>0.071555</td>\n",
       "      <td>-0.093124</td>\n",
       "      <td>-0.093056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17911</th>\n",
       "      <td>ZYG11A (440590)</td>\n",
       "      <td>-0.016426</td>\n",
       "      <td>-0.081285</td>\n",
       "      <td>-0.064280</td>\n",
       "      <td>-0.181128</td>\n",
       "      <td>-0.138234</td>\n",
       "      <td>-0.040847</td>\n",
       "      <td>-0.123101</td>\n",
       "      <td>0.018842</td>\n",
       "      <td>0.037163</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.103918</td>\n",
       "      <td>0.012138</td>\n",
       "      <td>-0.067628</td>\n",
       "      <td>-0.065556</td>\n",
       "      <td>-0.106669</td>\n",
       "      <td>-0.047990</td>\n",
       "      <td>-0.054118</td>\n",
       "      <td>0.116770</td>\n",
       "      <td>0.089177</td>\n",
       "      <td>0.061751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17912</th>\n",
       "      <td>ZYG11B (79699)</td>\n",
       "      <td>-0.152085</td>\n",
       "      <td>-0.236948</td>\n",
       "      <td>-0.049646</td>\n",
       "      <td>-0.052296</td>\n",
       "      <td>-0.209159</td>\n",
       "      <td>-0.188712</td>\n",
       "      <td>-0.130245</td>\n",
       "      <td>-0.164656</td>\n",
       "      <td>-0.272664</td>\n",
       "      <td>...</td>\n",
       "      <td>0.109241</td>\n",
       "      <td>-0.131954</td>\n",
       "      <td>-0.224480</td>\n",
       "      <td>0.047063</td>\n",
       "      <td>-0.268030</td>\n",
       "      <td>-0.068670</td>\n",
       "      <td>-0.140052</td>\n",
       "      <td>-0.041387</td>\n",
       "      <td>-0.284905</td>\n",
       "      <td>-0.122563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17913</th>\n",
       "      <td>ZYX (7791)</td>\n",
       "      <td>-0.153389</td>\n",
       "      <td>-0.013121</td>\n",
       "      <td>-0.428021</td>\n",
       "      <td>-0.186571</td>\n",
       "      <td>0.083589</td>\n",
       "      <td>-0.078067</td>\n",
       "      <td>-0.109373</td>\n",
       "      <td>-0.155827</td>\n",
       "      <td>0.052268</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.182024</td>\n",
       "      <td>-0.126813</td>\n",
       "      <td>-0.132606</td>\n",
       "      <td>0.034350</td>\n",
       "      <td>-0.205007</td>\n",
       "      <td>-0.050717</td>\n",
       "      <td>-0.014237</td>\n",
       "      <td>0.051083</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>-0.139035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17914</th>\n",
       "      <td>ZZEF1 (23140)</td>\n",
       "      <td>-0.130675</td>\n",
       "      <td>-0.126559</td>\n",
       "      <td>-0.172303</td>\n",
       "      <td>-0.137025</td>\n",
       "      <td>-0.108345</td>\n",
       "      <td>-0.093377</td>\n",
       "      <td>-0.230256</td>\n",
       "      <td>-0.115392</td>\n",
       "      <td>-0.072237</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.202608</td>\n",
       "      <td>-0.145749</td>\n",
       "      <td>-0.234658</td>\n",
       "      <td>-0.341694</td>\n",
       "      <td>-0.192169</td>\n",
       "      <td>0.021737</td>\n",
       "      <td>-0.022187</td>\n",
       "      <td>-0.051369</td>\n",
       "      <td>-0.083344</td>\n",
       "      <td>-0.179099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17915</th>\n",
       "      <td>ZZZ3 (26009)</td>\n",
       "      <td>-0.183547</td>\n",
       "      <td>-0.249220</td>\n",
       "      <td>-0.369053</td>\n",
       "      <td>-0.269073</td>\n",
       "      <td>-0.496251</td>\n",
       "      <td>-0.078212</td>\n",
       "      <td>-0.127530</td>\n",
       "      <td>-0.650192</td>\n",
       "      <td>-0.419285</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.195208</td>\n",
       "      <td>-0.315144</td>\n",
       "      <td>-0.202471</td>\n",
       "      <td>-0.205889</td>\n",
       "      <td>-0.003936</td>\n",
       "      <td>-0.174756</td>\n",
       "      <td>-0.287196</td>\n",
       "      <td>-0.146886</td>\n",
       "      <td>-0.227688</td>\n",
       "      <td>-0.133845</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17916 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Sample       HGNC_Symbol  ACH-000015  ACH-000030  ACH-000055  ACH-000070  \\\n",
       "0               A1BG (1)   -0.164485    0.092091    0.023233   -0.023093   \n",
       "1           A1CF (29974)   -0.062450   -0.235943    0.023197   -0.098678   \n",
       "2                A2M (2)   -0.042585   -0.008186    0.133568    0.055092   \n",
       "3         A2ML1 (144568)    0.067254    0.068138    0.097102    0.211111   \n",
       "4       A3GALT2 (127550)   -0.029782   -0.089160   -0.165176   -0.202141   \n",
       "...                  ...         ...         ...         ...         ...   \n",
       "17911    ZYG11A (440590)   -0.016426   -0.081285   -0.064280   -0.181128   \n",
       "17912     ZYG11B (79699)   -0.152085   -0.236948   -0.049646   -0.052296   \n",
       "17913         ZYX (7791)   -0.153389   -0.013121   -0.428021   -0.186571   \n",
       "17914      ZZEF1 (23140)   -0.130675   -0.126559   -0.172303   -0.137025   \n",
       "17915       ZZZ3 (26009)   -0.183547   -0.249220   -0.369053   -0.269073   \n",
       "\n",
       "Sample  ACH-000075  ACH-000098  ACH-000100  ACH-000132  ACH-000138  ...  \\\n",
       "0        -0.030929   -0.037187    0.021787   -0.122873   -0.102403  ...   \n",
       "1         0.173555   -0.137049    0.138818    0.072059   -0.117266  ...   \n",
       "2         0.144853    0.070069    0.219874    0.038849   -0.053961  ...   \n",
       "3         0.147002    0.010128    0.057229    0.039049   -0.071758  ...   \n",
       "4        -0.232510   -0.112067   -0.167169   -0.209066    0.073647  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "17911    -0.138234   -0.040847   -0.123101    0.018842    0.037163  ...   \n",
       "17912    -0.209159   -0.188712   -0.130245   -0.164656   -0.272664  ...   \n",
       "17913     0.083589   -0.078067   -0.109373   -0.155827    0.052268  ...   \n",
       "17914    -0.108345   -0.093377   -0.230256   -0.115392   -0.072237  ...   \n",
       "17915    -0.496251   -0.078212   -0.127530   -0.650192   -0.419285  ...   \n",
       "\n",
       "Sample  ACH-000880  ACH-000885  ACH-000900  ACH-000915  ACH-000919  \\\n",
       "0        -0.210218   -0.276661   -0.073962   -0.037085   -0.099006   \n",
       "1         0.023075    0.002950   -0.066371   -0.040370   -0.154016   \n",
       "2         0.169803    0.115853    0.051874   -0.013556    0.221679   \n",
       "3        -0.106526    0.193006    0.005611    0.139576    0.077502   \n",
       "4         0.077082   -0.010742    0.056990   -0.319709   -0.184296   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "17911    -0.103918    0.012138   -0.067628   -0.065556   -0.106669   \n",
       "17912     0.109241   -0.131954   -0.224480    0.047063   -0.268030   \n",
       "17913    -0.182024   -0.126813   -0.132606    0.034350   -0.205007   \n",
       "17914    -0.202608   -0.145749   -0.234658   -0.341694   -0.192169   \n",
       "17915    -0.195208   -0.315144   -0.202471   -0.205889   -0.003936   \n",
       "\n",
       "Sample  ACH-000950  ACH-000957  ACH-000969  ACH-000971  ACH-000977  \n",
       "0        -0.086066   -0.211048    0.136178    0.133352   -0.081687  \n",
       "1        -0.070168   -0.137014   -0.088978   -0.038505   -0.032364  \n",
       "2         0.068361    0.085481    0.088356    0.077262    0.129437  \n",
       "3        -0.114017    0.030893   -0.058193   -0.084336    0.105632  \n",
       "4        -0.143689   -0.046826    0.071555   -0.093124   -0.093056  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "17911    -0.047990   -0.054118    0.116770    0.089177    0.061751  \n",
       "17912    -0.068670   -0.140052   -0.041387   -0.284905   -0.122563  \n",
       "17913    -0.050717   -0.014237    0.051083    0.000105   -0.139035  \n",
       "17914     0.021737   -0.022187   -0.051369   -0.083344   -0.179099  \n",
       "17915    -0.174756   -0.287196   -0.146886   -0.227688   -0.133845  \n",
       "\n",
       "[17916 rows x 64 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "train_target_samples_id_df = train_target_samples_df[['depMapID']].copy()\n",
    "train_target_samples_id_df.rename(columns={'depMapID': 'Sample'}, inplace=True)\n",
    "test_target_samples_id_df = test_target_samples_df[['depMapID']].copy()\n",
    "test_target_samples_id_df.rename(columns={'depMapID': 'Sample'}, inplace=True)\n",
    "# Get the samples in the raw_crispr_df\n",
    "train_crispr_score_df = raw_crispr_df.merge(train_target_samples_id_df, on=\"Sample\", how=\"inner\")\n",
    "display(train_crispr_score_df)\n",
    "test_crispr_score_df = raw_crispr_df.merge(test_target_samples_id_df, on=\"Sample\", how=\"inner\")\n",
    "display(test_crispr_score_df)\n",
    "\n",
    "# Set Index column to str to keep digit without .00\n",
    "train_crispr_score_t_df = train_crispr_score_df.set_index(\"Sample\").T.reset_index()\n",
    "train_crispr_score_t_df.rename(columns={\"index\": \"HGNC_Symbol\"}, inplace=True)\n",
    "display(train_crispr_score_t_df)\n",
    "test_crispr_score_t_df = test_crispr_score_df.set_index(\"Sample\").T.reset_index()\n",
    "test_crispr_score_t_df.rename(columns={\"index\": \"HGNC_Symbol\"}, inplace=True)\n",
    "display(test_crispr_score_t_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 411,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "match rate:39.52%\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000018</th>\n",
       "      <th>ACH-000019</th>\n",
       "      <th>ACH-000021</th>\n",
       "      <th>ACH-000023</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000988</th>\n",
       "      <th>ACH-000991</th>\n",
       "      <th>ACH-000997</th>\n",
       "      <th>ACH-001039</th>\n",
       "      <th>ACH-001065</th>\n",
       "      <th>ACH-001075</th>\n",
       "      <th>ACH-001107</th>\n",
       "      <th>ACH-001190</th>\n",
       "      <th>ACH-001306</th>\n",
       "      <th>ACH-001307</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>-0.016484</td>\n",
       "      <td>-0.071921</td>\n",
       "      <td>-0.019163</td>\n",
       "      <td>-0.139496</td>\n",
       "      <td>-0.089959</td>\n",
       "      <td>-0.000655</td>\n",
       "      <td>-0.144838</td>\n",
       "      <td>-0.044006</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.080952</td>\n",
       "      <td>-0.086003</td>\n",
       "      <td>0.010321</td>\n",
       "      <td>-0.280963</td>\n",
       "      <td>-0.054307</td>\n",
       "      <td>-0.236411</td>\n",
       "      <td>-0.098559</td>\n",
       "      <td>-0.042641</td>\n",
       "      <td>-0.031081</td>\n",
       "      <td>0.012357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>0.196378</td>\n",
       "      <td>0.310169</td>\n",
       "      <td>0.198303</td>\n",
       "      <td>0.157864</td>\n",
       "      <td>0.208419</td>\n",
       "      <td>0.269541</td>\n",
       "      <td>0.137374</td>\n",
       "      <td>0.163614</td>\n",
       "      <td>...</td>\n",
       "      <td>0.119221</td>\n",
       "      <td>0.213694</td>\n",
       "      <td>0.118167</td>\n",
       "      <td>-0.273918</td>\n",
       "      <td>-0.017122</td>\n",
       "      <td>0.105633</td>\n",
       "      <td>0.140576</td>\n",
       "      <td>0.185244</td>\n",
       "      <td>0.250509</td>\n",
       "      <td>0.223775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>-0.116327</td>\n",
       "      <td>-0.200383</td>\n",
       "      <td>-0.120459</td>\n",
       "      <td>-0.100498</td>\n",
       "      <td>0.084591</td>\n",
       "      <td>-0.021049</td>\n",
       "      <td>-0.246973</td>\n",
       "      <td>-0.034359</td>\n",
       "      <td>0.011776</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.082360</td>\n",
       "      <td>-0.009491</td>\n",
       "      <td>0.026029</td>\n",
       "      <td>-0.157190</td>\n",
       "      <td>-0.023168</td>\n",
       "      <td>-0.131286</td>\n",
       "      <td>-0.083327</td>\n",
       "      <td>-0.123697</td>\n",
       "      <td>-0.086892</td>\n",
       "      <td>0.059173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>-0.019295</td>\n",
       "      <td>0.099193</td>\n",
       "      <td>-0.213060</td>\n",
       "      <td>-0.100742</td>\n",
       "      <td>-0.027933</td>\n",
       "      <td>-0.842505</td>\n",
       "      <td>0.063977</td>\n",
       "      <td>-0.117045</td>\n",
       "      <td>0.070139</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.117608</td>\n",
       "      <td>0.007976</td>\n",
       "      <td>-0.028504</td>\n",
       "      <td>0.233251</td>\n",
       "      <td>-0.176635</td>\n",
       "      <td>-0.057680</td>\n",
       "      <td>-0.250286</td>\n",
       "      <td>0.086460</td>\n",
       "      <td>-0.275844</td>\n",
       "      <td>-0.517292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>-0.100900</td>\n",
       "      <td>0.151491</td>\n",
       "      <td>0.191130</td>\n",
       "      <td>0.025839</td>\n",
       "      <td>0.055322</td>\n",
       "      <td>-0.128477</td>\n",
       "      <td>0.418205</td>\n",
       "      <td>0.128353</td>\n",
       "      <td>-0.115753</td>\n",
       "      <td>...</td>\n",
       "      <td>0.160051</td>\n",
       "      <td>0.066251</td>\n",
       "      <td>0.009697</td>\n",
       "      <td>-0.162483</td>\n",
       "      <td>0.139784</td>\n",
       "      <td>0.011788</td>\n",
       "      <td>0.164420</td>\n",
       "      <td>-0.058207</td>\n",
       "      <td>-0.069053</td>\n",
       "      <td>0.237172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20283</th>\n",
       "      <td>BMGC_GN69808</td>\n",
       "      <td>0.094246</td>\n",
       "      <td>0.021657</td>\n",
       "      <td>0.392400</td>\n",
       "      <td>0.293312</td>\n",
       "      <td>0.124325</td>\n",
       "      <td>0.442315</td>\n",
       "      <td>0.151060</td>\n",
       "      <td>0.259185</td>\n",
       "      <td>0.030048</td>\n",
       "      <td>...</td>\n",
       "      <td>0.130424</td>\n",
       "      <td>-0.118179</td>\n",
       "      <td>-0.030500</td>\n",
       "      <td>0.167938</td>\n",
       "      <td>0.084445</td>\n",
       "      <td>0.069296</td>\n",
       "      <td>0.211061</td>\n",
       "      <td>0.254713</td>\n",
       "      <td>0.086061</td>\n",
       "      <td>0.421384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20284</th>\n",
       "      <td>BMGC_GN69841</td>\n",
       "      <td>-0.137149</td>\n",
       "      <td>-0.601468</td>\n",
       "      <td>-0.320952</td>\n",
       "      <td>0.022595</td>\n",
       "      <td>-0.075869</td>\n",
       "      <td>-0.160712</td>\n",
       "      <td>-0.142266</td>\n",
       "      <td>0.055720</td>\n",
       "      <td>0.048859</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.112633</td>\n",
       "      <td>-0.344776</td>\n",
       "      <td>-0.322098</td>\n",
       "      <td>-0.197280</td>\n",
       "      <td>0.084078</td>\n",
       "      <td>-0.194447</td>\n",
       "      <td>-0.160895</td>\n",
       "      <td>0.105202</td>\n",
       "      <td>-0.048947</td>\n",
       "      <td>-0.077387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20285</th>\n",
       "      <td>BMGC_GN69848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.333310</td>\n",
       "      <td>0.090807</td>\n",
       "      <td>0.130973</td>\n",
       "      <td>0.057994</td>\n",
       "      <td>0.016015</td>\n",
       "      <td>-0.088375</td>\n",
       "      <td>0.013533</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.129761</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.068046</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.276861</td>\n",
       "      <td>-0.006914</td>\n",
       "      <td>0.126152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20286</th>\n",
       "      <td>BMGC_GN69876</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.316461</td>\n",
       "      <td>-0.556799</td>\n",
       "      <td>-0.350030</td>\n",
       "      <td>-0.579060</td>\n",
       "      <td>-0.448828</td>\n",
       "      <td>0.418644</td>\n",
       "      <td>-0.102821</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.017046</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.329794</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.067530</td>\n",
       "      <td>-0.110224</td>\n",
       "      <td>-0.190182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20287</th>\n",
       "      <td>BMGC_GN70238</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.019036</td>\n",
       "      <td>-0.158997</td>\n",
       "      <td>0.331191</td>\n",
       "      <td>0.074430</td>\n",
       "      <td>0.070739</td>\n",
       "      <td>0.288486</td>\n",
       "      <td>0.130661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.208699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.088504</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.095637</td>\n",
       "      <td>0.082939</td>\n",
       "      <td>0.047632</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20288 rows × 301 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000001  ACH-000004  ACH-000007  ACH-000009  \\\n",
       "0               BMGC_GN00001   -0.121964   -0.016484   -0.071921   -0.019163   \n",
       "1               BMGC_GN00002   -0.022127    0.196378    0.310169    0.198303   \n",
       "2               BMGC_GN00003   -0.116327   -0.200383   -0.120459   -0.100498   \n",
       "3               BMGC_GN00004   -0.019295    0.099193   -0.213060   -0.100742   \n",
       "4               BMGC_GN00005   -0.100900    0.151491    0.191130    0.025839   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "20283           BMGC_GN69808    0.094246    0.021657    0.392400    0.293312   \n",
       "20284           BMGC_GN69841   -0.137149   -0.601468   -0.320952    0.022595   \n",
       "20285           BMGC_GN69848         NaN    0.333310    0.090807    0.130973   \n",
       "20286           BMGC_GN69876         NaN   -0.316461   -0.556799   -0.350030   \n",
       "20287           BMGC_GN70238         NaN   -0.019036   -0.158997    0.331191   \n",
       "\n",
       "       ACH-000012  ACH-000018  ACH-000019  ACH-000021  ACH-000023  ...  \\\n",
       "0       -0.139496   -0.089959   -0.000655   -0.144838   -0.044006  ...   \n",
       "1        0.157864    0.208419    0.269541    0.137374    0.163614  ...   \n",
       "2        0.084591   -0.021049   -0.246973   -0.034359    0.011776  ...   \n",
       "3       -0.027933   -0.842505    0.063977   -0.117045    0.070139  ...   \n",
       "4        0.055322   -0.128477    0.418205    0.128353   -0.115753  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "20283    0.124325    0.442315    0.151060    0.259185    0.030048  ...   \n",
       "20284   -0.075869   -0.160712   -0.142266    0.055720    0.048859  ...   \n",
       "20285    0.057994    0.016015   -0.088375    0.013533         NaN  ...   \n",
       "20286   -0.579060   -0.448828    0.418644   -0.102821         NaN  ...   \n",
       "20287    0.074430    0.070739    0.288486    0.130661         NaN  ...   \n",
       "\n",
       "       ACH-000988  ACH-000991  ACH-000997  ACH-001039  ACH-001065  ACH-001075  \\\n",
       "0       -0.080952   -0.086003    0.010321   -0.280963   -0.054307   -0.236411   \n",
       "1        0.119221    0.213694    0.118167   -0.273918   -0.017122    0.105633   \n",
       "2       -0.082360   -0.009491    0.026029   -0.157190   -0.023168   -0.131286   \n",
       "3       -0.117608    0.007976   -0.028504    0.233251   -0.176635   -0.057680   \n",
       "4        0.160051    0.066251    0.009697   -0.162483    0.139784    0.011788   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "20283    0.130424   -0.118179   -0.030500    0.167938    0.084445    0.069296   \n",
       "20284   -0.112633   -0.344776   -0.322098   -0.197280    0.084078   -0.194447   \n",
       "20285   -0.129761         NaN         NaN         NaN         NaN    0.068046   \n",
       "20286    0.017046         NaN         NaN         NaN         NaN   -0.329794   \n",
       "20287    0.208699         NaN         NaN         NaN         NaN   -0.088504   \n",
       "\n",
       "       ACH-001107  ACH-001190  ACH-001306  ACH-001307  \n",
       "0       -0.098559   -0.042641   -0.031081    0.012357  \n",
       "1        0.140576    0.185244    0.250509    0.223775  \n",
       "2       -0.083327   -0.123697   -0.086892    0.059173  \n",
       "3       -0.250286    0.086460   -0.275844   -0.517292  \n",
       "4        0.164420   -0.058207   -0.069053    0.237172  \n",
       "...           ...         ...         ...         ...  \n",
       "20283    0.211061    0.254713    0.086061    0.421384  \n",
       "20284   -0.160895    0.105202   -0.048947   -0.077387  \n",
       "20285         NaN    0.276861   -0.006914    0.126152  \n",
       "20286         NaN   -0.067530   -0.110224   -0.190182  \n",
       "20287         NaN   -0.095637    0.082939    0.047632  \n",
       "\n",
       "[20288 rows x 301 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "match rate:39.52%\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000015</th>\n",
       "      <th>ACH-000030</th>\n",
       "      <th>ACH-000055</th>\n",
       "      <th>ACH-000070</th>\n",
       "      <th>ACH-000075</th>\n",
       "      <th>ACH-000098</th>\n",
       "      <th>ACH-000100</th>\n",
       "      <th>ACH-000132</th>\n",
       "      <th>ACH-000138</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000880</th>\n",
       "      <th>ACH-000885</th>\n",
       "      <th>ACH-000900</th>\n",
       "      <th>ACH-000915</th>\n",
       "      <th>ACH-000919</th>\n",
       "      <th>ACH-000950</th>\n",
       "      <th>ACH-000957</th>\n",
       "      <th>ACH-000969</th>\n",
       "      <th>ACH-000971</th>\n",
       "      <th>ACH-000977</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>-0.164485</td>\n",
       "      <td>0.092091</td>\n",
       "      <td>0.023233</td>\n",
       "      <td>-0.023093</td>\n",
       "      <td>-0.030929</td>\n",
       "      <td>-0.037187</td>\n",
       "      <td>0.021787</td>\n",
       "      <td>-0.122873</td>\n",
       "      <td>-0.102403</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.210218</td>\n",
       "      <td>-0.276661</td>\n",
       "      <td>-0.073962</td>\n",
       "      <td>-0.037085</td>\n",
       "      <td>-0.099006</td>\n",
       "      <td>-0.086066</td>\n",
       "      <td>-0.211048</td>\n",
       "      <td>0.136178</td>\n",
       "      <td>0.133352</td>\n",
       "      <td>-0.081687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>-0.043958</td>\n",
       "      <td>0.172081</td>\n",
       "      <td>0.113231</td>\n",
       "      <td>0.022171</td>\n",
       "      <td>0.085891</td>\n",
       "      <td>0.185540</td>\n",
       "      <td>0.196876</td>\n",
       "      <td>0.175082</td>\n",
       "      <td>0.122638</td>\n",
       "      <td>...</td>\n",
       "      <td>0.228929</td>\n",
       "      <td>0.139567</td>\n",
       "      <td>0.140669</td>\n",
       "      <td>0.246247</td>\n",
       "      <td>0.277374</td>\n",
       "      <td>-0.045897</td>\n",
       "      <td>-0.063168</td>\n",
       "      <td>0.252195</td>\n",
       "      <td>-0.280817</td>\n",
       "      <td>0.013595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>-0.140311</td>\n",
       "      <td>-0.022852</td>\n",
       "      <td>-0.158881</td>\n",
       "      <td>-0.039262</td>\n",
       "      <td>-0.119558</td>\n",
       "      <td>0.017364</td>\n",
       "      <td>-0.098256</td>\n",
       "      <td>-0.067811</td>\n",
       "      <td>-0.084381</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.177668</td>\n",
       "      <td>-0.061922</td>\n",
       "      <td>-0.080560</td>\n",
       "      <td>-0.058143</td>\n",
       "      <td>-0.163444</td>\n",
       "      <td>-0.070590</td>\n",
       "      <td>-0.174459</td>\n",
       "      <td>-0.155545</td>\n",
       "      <td>-0.063573</td>\n",
       "      <td>0.023034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>-0.159200</td>\n",
       "      <td>0.064357</td>\n",
       "      <td>0.530061</td>\n",
       "      <td>0.185567</td>\n",
       "      <td>-0.217684</td>\n",
       "      <td>-0.469302</td>\n",
       "      <td>0.015246</td>\n",
       "      <td>-0.974951</td>\n",
       "      <td>-0.178604</td>\n",
       "      <td>...</td>\n",
       "      <td>0.034351</td>\n",
       "      <td>-0.029154</td>\n",
       "      <td>-0.263409</td>\n",
       "      <td>-0.195348</td>\n",
       "      <td>0.023198</td>\n",
       "      <td>-0.123158</td>\n",
       "      <td>-0.136271</td>\n",
       "      <td>0.040931</td>\n",
       "      <td>-0.057478</td>\n",
       "      <td>0.046085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>0.041047</td>\n",
       "      <td>0.104015</td>\n",
       "      <td>-0.039690</td>\n",
       "      <td>0.017025</td>\n",
       "      <td>-0.044021</td>\n",
       "      <td>-0.066938</td>\n",
       "      <td>-0.113157</td>\n",
       "      <td>-0.099366</td>\n",
       "      <td>0.036075</td>\n",
       "      <td>...</td>\n",
       "      <td>0.083539</td>\n",
       "      <td>0.103063</td>\n",
       "      <td>0.236837</td>\n",
       "      <td>-0.253602</td>\n",
       "      <td>0.166021</td>\n",
       "      <td>0.223567</td>\n",
       "      <td>0.049261</td>\n",
       "      <td>0.121447</td>\n",
       "      <td>0.098172</td>\n",
       "      <td>-0.000221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20283</th>\n",
       "      <td>BMGC_GN69808</td>\n",
       "      <td>0.114893</td>\n",
       "      <td>0.128794</td>\n",
       "      <td>0.111258</td>\n",
       "      <td>0.103258</td>\n",
       "      <td>0.219192</td>\n",
       "      <td>0.305259</td>\n",
       "      <td>0.141067</td>\n",
       "      <td>0.129172</td>\n",
       "      <td>0.344026</td>\n",
       "      <td>...</td>\n",
       "      <td>0.368076</td>\n",
       "      <td>0.280522</td>\n",
       "      <td>0.031917</td>\n",
       "      <td>0.075625</td>\n",
       "      <td>0.261597</td>\n",
       "      <td>0.212018</td>\n",
       "      <td>0.311237</td>\n",
       "      <td>0.120037</td>\n",
       "      <td>0.131675</td>\n",
       "      <td>0.031412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20284</th>\n",
       "      <td>BMGC_GN69841</td>\n",
       "      <td>-0.272400</td>\n",
       "      <td>-0.258718</td>\n",
       "      <td>-0.354407</td>\n",
       "      <td>-0.494304</td>\n",
       "      <td>-0.058402</td>\n",
       "      <td>0.033533</td>\n",
       "      <td>-0.300633</td>\n",
       "      <td>-0.207052</td>\n",
       "      <td>-0.207051</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.165977</td>\n",
       "      <td>-0.125010</td>\n",
       "      <td>-0.279584</td>\n",
       "      <td>-0.035788</td>\n",
       "      <td>-0.531473</td>\n",
       "      <td>-0.140367</td>\n",
       "      <td>-0.390618</td>\n",
       "      <td>-0.023741</td>\n",
       "      <td>-0.142910</td>\n",
       "      <td>-0.418786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20285</th>\n",
       "      <td>BMGC_GN69848</td>\n",
       "      <td>0.098828</td>\n",
       "      <td>0.129801</td>\n",
       "      <td>-0.039741</td>\n",
       "      <td>0.201172</td>\n",
       "      <td>0.294755</td>\n",
       "      <td>-0.018549</td>\n",
       "      <td>0.237250</td>\n",
       "      <td>0.045203</td>\n",
       "      <td>0.198364</td>\n",
       "      <td>...</td>\n",
       "      <td>0.407689</td>\n",
       "      <td>0.525714</td>\n",
       "      <td>0.018922</td>\n",
       "      <td>0.061899</td>\n",
       "      <td>0.002737</td>\n",
       "      <td>0.050819</td>\n",
       "      <td>0.404756</td>\n",
       "      <td>0.111930</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.190287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20286</th>\n",
       "      <td>BMGC_GN69876</td>\n",
       "      <td>-0.830064</td>\n",
       "      <td>-0.138660</td>\n",
       "      <td>-0.584080</td>\n",
       "      <td>-0.421905</td>\n",
       "      <td>-0.332492</td>\n",
       "      <td>-0.213516</td>\n",
       "      <td>-0.474673</td>\n",
       "      <td>-0.259733</td>\n",
       "      <td>0.044880</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.482228</td>\n",
       "      <td>-0.334915</td>\n",
       "      <td>-0.142542</td>\n",
       "      <td>-0.045426</td>\n",
       "      <td>0.168132</td>\n",
       "      <td>-0.620618</td>\n",
       "      <td>0.146192</td>\n",
       "      <td>-0.242320</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.172861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20287</th>\n",
       "      <td>BMGC_GN70238</td>\n",
       "      <td>-0.209955</td>\n",
       "      <td>0.118995</td>\n",
       "      <td>-0.310838</td>\n",
       "      <td>-0.123355</td>\n",
       "      <td>-0.119235</td>\n",
       "      <td>0.077615</td>\n",
       "      <td>0.115051</td>\n",
       "      <td>0.270607</td>\n",
       "      <td>0.084338</td>\n",
       "      <td>...</td>\n",
       "      <td>0.163203</td>\n",
       "      <td>-0.226178</td>\n",
       "      <td>0.121935</td>\n",
       "      <td>0.102388</td>\n",
       "      <td>-0.068755</td>\n",
       "      <td>-0.232383</td>\n",
       "      <td>0.066975</td>\n",
       "      <td>0.228766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.011810</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20288 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000015  ACH-000030  ACH-000055  ACH-000070  \\\n",
       "0               BMGC_GN00001   -0.164485    0.092091    0.023233   -0.023093   \n",
       "1               BMGC_GN00002   -0.043958    0.172081    0.113231    0.022171   \n",
       "2               BMGC_GN00003   -0.140311   -0.022852   -0.158881   -0.039262   \n",
       "3               BMGC_GN00004   -0.159200    0.064357    0.530061    0.185567   \n",
       "4               BMGC_GN00005    0.041047    0.104015   -0.039690    0.017025   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "20283           BMGC_GN69808    0.114893    0.128794    0.111258    0.103258   \n",
       "20284           BMGC_GN69841   -0.272400   -0.258718   -0.354407   -0.494304   \n",
       "20285           BMGC_GN69848    0.098828    0.129801   -0.039741    0.201172   \n",
       "20286           BMGC_GN69876   -0.830064   -0.138660   -0.584080   -0.421905   \n",
       "20287           BMGC_GN70238   -0.209955    0.118995   -0.310838   -0.123355   \n",
       "\n",
       "       ACH-000075  ACH-000098  ACH-000100  ACH-000132  ACH-000138  ...  \\\n",
       "0       -0.030929   -0.037187    0.021787   -0.122873   -0.102403  ...   \n",
       "1        0.085891    0.185540    0.196876    0.175082    0.122638  ...   \n",
       "2       -0.119558    0.017364   -0.098256   -0.067811   -0.084381  ...   \n",
       "3       -0.217684   -0.469302    0.015246   -0.974951   -0.178604  ...   \n",
       "4       -0.044021   -0.066938   -0.113157   -0.099366    0.036075  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "20283    0.219192    0.305259    0.141067    0.129172    0.344026  ...   \n",
       "20284   -0.058402    0.033533   -0.300633   -0.207052   -0.207051  ...   \n",
       "20285    0.294755   -0.018549    0.237250    0.045203    0.198364  ...   \n",
       "20286   -0.332492   -0.213516   -0.474673   -0.259733    0.044880  ...   \n",
       "20287   -0.119235    0.077615    0.115051    0.270607    0.084338  ...   \n",
       "\n",
       "       ACH-000880  ACH-000885  ACH-000900  ACH-000915  ACH-000919  ACH-000950  \\\n",
       "0       -0.210218   -0.276661   -0.073962   -0.037085   -0.099006   -0.086066   \n",
       "1        0.228929    0.139567    0.140669    0.246247    0.277374   -0.045897   \n",
       "2       -0.177668   -0.061922   -0.080560   -0.058143   -0.163444   -0.070590   \n",
       "3        0.034351   -0.029154   -0.263409   -0.195348    0.023198   -0.123158   \n",
       "4        0.083539    0.103063    0.236837   -0.253602    0.166021    0.223567   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "20283    0.368076    0.280522    0.031917    0.075625    0.261597    0.212018   \n",
       "20284   -0.165977   -0.125010   -0.279584   -0.035788   -0.531473   -0.140367   \n",
       "20285    0.407689    0.525714    0.018922    0.061899    0.002737    0.050819   \n",
       "20286   -0.482228   -0.334915   -0.142542   -0.045426    0.168132   -0.620618   \n",
       "20287    0.163203   -0.226178    0.121935    0.102388   -0.068755   -0.232383   \n",
       "\n",
       "       ACH-000957  ACH-000969  ACH-000971  ACH-000977  \n",
       "0       -0.211048    0.136178    0.133352   -0.081687  \n",
       "1       -0.063168    0.252195   -0.280817    0.013595  \n",
       "2       -0.174459   -0.155545   -0.063573    0.023034  \n",
       "3       -0.136271    0.040931   -0.057478    0.046085  \n",
       "4        0.049261    0.121447    0.098172   -0.000221  \n",
       "...           ...         ...         ...         ...  \n",
       "20283    0.311237    0.120037    0.131675    0.031412  \n",
       "20284   -0.390618   -0.023741   -0.142910   -0.418786  \n",
       "20285    0.404756    0.111930         NaN    0.190287  \n",
       "20286    0.146192   -0.242320         NaN   -0.172861  \n",
       "20287    0.066975    0.228766         NaN   -0.011810  \n",
       "\n",
       "[20288 rows x 64 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import re\n",
    "def extract_gene_name(gene):\n",
    "    return re.sub(r\"\\s*\\(.*?\\)\", \"\", str(gene))\n",
    "\n",
    "train_crispr_score_t_df[\"HGNC_Symbol\"] = train_crispr_score_t_df[\"HGNC_Symbol\"].apply(extract_gene_name)\n",
    "unique_Gene_map = bmgc_gene_df[['HGNC_Symbol']].drop_duplicates()\n",
    "unique_omics_Gene = train_crispr_score_t_df[['HGNC_Symbol']].drop_duplicates()\n",
    "# match\n",
    "intersection = set(unique_Gene_map['HGNC_Symbol'])&set(unique_omics_Gene['HGNC_Symbol'])\n",
    "total_Gene = len(unique_Gene_map)\n",
    "match_rate = len(intersection) / total_Gene if total_Gene > 0 else 0\n",
    "print(f\"match rate:{match_rate:.2%}\")\n",
    "train_crispr_score_bmgc_df = bmgc_gene_df.merge(train_crispr_score_t_df, on=\"HGNC_Symbol\", how=\"inner\")\n",
    "train_crispr_score_bmgc_df = train_crispr_score_bmgc_df.drop(columns=['BioMedGraphica_ID','NCBI_Gene_ID','Ensembl_Gene_ID_Version','Gene_Start', 'Gene_End','Chromosome','Gene_Type','Ensembl_Gene_ID','HGNC_ID','Gene_Name','RefSeq_ID','OMIM_ID','HGNC_Symbol'])\n",
    "display(train_crispr_score_bmgc_df)\n",
    "\n",
    "test_crispr_score_t_df[\"HGNC_Symbol\"] = test_crispr_score_t_df[\"HGNC_Symbol\"].apply(extract_gene_name)\n",
    "unique_Gene_map = bmgc_gene_df[['HGNC_Symbol']].drop_duplicates()\n",
    "unique_omics_Gene = test_crispr_score_t_df[['HGNC_Symbol']].drop_duplicates()\n",
    "# match\n",
    "intersection = set(unique_Gene_map['HGNC_Symbol'])&set(unique_omics_Gene['HGNC_Symbol'])\n",
    "total_Gene = len(unique_Gene_map)\n",
    "match_rate = len(intersection) / total_Gene if total_Gene > 0 else 0\n",
    "print(f\"match rate:{match_rate:.2%}\")\n",
    "test_crispr_score_bmgc_df = bmgc_gene_df.merge(test_crispr_score_t_df, on=\"HGNC_Symbol\", how=\"inner\")\n",
    "test_crispr_score_bmgc_df = test_crispr_score_bmgc_df.drop(columns=['BioMedGraphica_ID','NCBI_Gene_ID','Ensembl_Gene_ID_Version','Gene_Start', 'Gene_End','Chromosome','Gene_Type','Ensembl_Gene_ID','HGNC_ID','Gene_Name','RefSeq_ID','OMIM_ID','HGNC_Symbol'])\n",
    "display(test_crispr_score_bmgc_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 412,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH_ID</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>-0.121964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>-0.022127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>-0.116327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>-0.019295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>-0.100900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6086395</th>\n",
       "      <td>BMGC_GN69808</td>\n",
       "      <td>ACH-001307</td>\n",
       "      <td>0.421384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6086396</th>\n",
       "      <td>BMGC_GN69841</td>\n",
       "      <td>ACH-001307</td>\n",
       "      <td>-0.077387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6086397</th>\n",
       "      <td>BMGC_GN69848</td>\n",
       "      <td>ACH-001307</td>\n",
       "      <td>0.126152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6086398</th>\n",
       "      <td>BMGC_GN69876</td>\n",
       "      <td>ACH-001307</td>\n",
       "      <td>-0.190182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6086399</th>\n",
       "      <td>BMGC_GN70238</td>\n",
       "      <td>ACH-001307</td>\n",
       "      <td>0.047632</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6041324 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        BioMedGraphica_Conn_ID      ACH_ID     Value\n",
       "0                 BMGC_GN00001  ACH-000001 -0.121964\n",
       "1                 BMGC_GN00002  ACH-000001 -0.022127\n",
       "2                 BMGC_GN00003  ACH-000001 -0.116327\n",
       "3                 BMGC_GN00004  ACH-000001 -0.019295\n",
       "4                 BMGC_GN00005  ACH-000001 -0.100900\n",
       "...                        ...         ...       ...\n",
       "6086395           BMGC_GN69808  ACH-001307  0.421384\n",
       "6086396           BMGC_GN69841  ACH-001307 -0.077387\n",
       "6086397           BMGC_GN69848  ACH-001307  0.126152\n",
       "6086398           BMGC_GN69876  ACH-001307 -0.190182\n",
       "6086399           BMGC_GN70238  ACH-001307  0.047632\n",
       "\n",
       "[6041324 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH_ID</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>ACH-000015</td>\n",
       "      <td>-0.164485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>ACH-000015</td>\n",
       "      <td>-0.043958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>ACH-000015</td>\n",
       "      <td>-0.140311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>ACH-000015</td>\n",
       "      <td>-0.159200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>ACH-000015</td>\n",
       "      <td>0.041047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1278139</th>\n",
       "      <td>BMGC_GN69808</td>\n",
       "      <td>ACH-000977</td>\n",
       "      <td>0.031412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1278140</th>\n",
       "      <td>BMGC_GN69841</td>\n",
       "      <td>ACH-000977</td>\n",
       "      <td>-0.418786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1278141</th>\n",
       "      <td>BMGC_GN69848</td>\n",
       "      <td>ACH-000977</td>\n",
       "      <td>0.190287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1278142</th>\n",
       "      <td>BMGC_GN69876</td>\n",
       "      <td>ACH-000977</td>\n",
       "      <td>-0.172861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1278143</th>\n",
       "      <td>BMGC_GN70238</td>\n",
       "      <td>ACH-000977</td>\n",
       "      <td>-0.011810</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1269557 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        BioMedGraphica_Conn_ID      ACH_ID     Value\n",
       "0                 BMGC_GN00001  ACH-000015 -0.164485\n",
       "1                 BMGC_GN00002  ACH-000015 -0.043958\n",
       "2                 BMGC_GN00003  ACH-000015 -0.140311\n",
       "3                 BMGC_GN00004  ACH-000015 -0.159200\n",
       "4                 BMGC_GN00005  ACH-000015  0.041047\n",
       "...                        ...         ...       ...\n",
       "1278139           BMGC_GN69808  ACH-000977  0.031412\n",
       "1278140           BMGC_GN69841  ACH-000977 -0.418786\n",
       "1278141           BMGC_GN69848  ACH-000977  0.190287\n",
       "1278142           BMGC_GN69876  ACH-000977 -0.172861\n",
       "1278143           BMGC_GN70238  ACH-000977 -0.011810\n",
       "\n",
       "[1269557 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "train_crispr_score_label = train_crispr_score_bmgc_df.copy()\n",
    "train_crispr_score_label_melted = train_crispr_score_label.melt(id_vars=[\"BioMedGraphica_Conn_ID\"], var_name=\"ACH_ID\", value_name=\"Value\")\n",
    "train_crispr_score_label_melted = train_crispr_score_label_melted.dropna(subset=['Value'])\n",
    "display(train_crispr_score_label_melted)\n",
    "test_crispr_score_label = test_crispr_score_bmgc_df.copy()\n",
    "test_crispr_score_label_melted = test_crispr_score_label.melt(id_vars=[\"BioMedGraphica_Conn_ID\"], var_name=\"ACH_ID\", value_name=\"Value\")\n",
    "test_crispr_score_label_melted = test_crispr_score_label_melted.dropna(subset=['Value'])\n",
    "display(test_crispr_score_label_melted)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Merge into internal relation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>promoterID</th>\n",
       "      <th>geneID</th>\n",
       "      <th>transcriptID</th>\n",
       "      <th>proteinID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>86238</td>\n",
       "      <td>176176.0</td>\n",
       "      <td>598056.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>86238</td>\n",
       "      <td>351511.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>86238</td>\n",
       "      <td>352614.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>86238</td>\n",
       "      <td>353426.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>86238</td>\n",
       "      <td>354931.0</td>\n",
       "      <td>676216.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458340</th>\n",
       "      <td>86236</td>\n",
       "      <td>172474</td>\n",
       "      <td>584407.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458341</th>\n",
       "      <td>86237</td>\n",
       "      <td>172475</td>\n",
       "      <td>584489.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458342</th>\n",
       "      <td>86237</td>\n",
       "      <td>172475</td>\n",
       "      <td>584490.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458343</th>\n",
       "      <td>86237</td>\n",
       "      <td>172475</td>\n",
       "      <td>584491.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458344</th>\n",
       "      <td>86237</td>\n",
       "      <td>172475</td>\n",
       "      <td>584492.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>458345 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        promoterID  geneID  transcriptID  proteinID\n",
       "0                0   86238      176176.0   598056.0\n",
       "1                0   86238      351511.0        NaN\n",
       "2                0   86238      352614.0        NaN\n",
       "3                0   86238      353426.0        NaN\n",
       "4                0   86238      354931.0   676216.0\n",
       "...            ...     ...           ...        ...\n",
       "458340       86236  172474      584407.0        NaN\n",
       "458341       86237  172475      584489.0        NaN\n",
       "458342       86237  172475      584490.0        NaN\n",
       "458343       86237  172475      584491.0        NaN\n",
       "458344       86237  172475      584492.0        NaN\n",
       "\n",
       "[458345 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Based on dti_sample_index, convert this to a dict\n",
    "target_crispr_sample_index_dict = dict(zip(overlapped_dti_crispr_rna_samples_df['depMapID'], overlapped_dti_crispr_rna_samples_df.index))\n",
    "# Map columns ['ACH_ID', 'BioMedGraphica_Conn_ID'] by the crispr_sample_index_dict and node_index_dict\n",
    "train_crispr_score_label_melted['ACH_ID'] = train_crispr_score_label_melted['ACH_ID'].map(target_crispr_sample_index_dict)\n",
    "train_crispr_score_label_melted['BioMedGraphica_Conn_ID'] = train_crispr_score_label_melted['BioMedGraphica_Conn_ID'].map(node_index_dict)\n",
    "test_crispr_score_label_melted['ACH_ID'] = test_crispr_score_label_melted['ACH_ID'].map(target_crispr_sample_index_dict)\n",
    "test_crispr_score_label_melted['BioMedGraphica_Conn_ID'] = test_crispr_score_label_melted['BioMedGraphica_Conn_ID'].map(node_index_dict)\n",
    "\n",
    "# Map internal_relation_df columns ['promoterID', 'geneID', 'transcriptID', 'proteinID'] by the node_index_dict\n",
    "internal_relation_map_df = internal_relation_df.copy()\n",
    "internal_relation_map_df['promoterID'] = internal_relation_map_df['promoterID'].map(node_index_dict)\n",
    "internal_relation_map_df['geneID'] = internal_relation_map_df['geneID'].map(node_index_dict)\n",
    "internal_relation_map_df['transcriptID'] = internal_relation_map_df['transcriptID'].map(node_index_dict)\n",
    "internal_relation_map_df['proteinID'] = internal_relation_map_df['proteinID'].map(node_index_dict)\n",
    "\n",
    "ko_internal_relation_map_df = internal_relation_map_df.copy()\n",
    "display(ko_internal_relation_map_df)\n",
    "ko_internal_relation_index_df = ko_internal_relation_map_df.groupby([\"promoterID\", \"geneID\"], as_index=False).agg({\n",
    "    \"transcriptID\": lambda x: list(set(x.dropna())), \n",
    "    \"proteinID\": lambda x: list(set(x.dropna()))  \n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 414,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>promoterID</th>\n",
       "      <th>geneID</th>\n",
       "      <th>transcriptID</th>\n",
       "      <th>proteinID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>86238</td>\n",
       "      <td>[352614, 176176, 353426, 354931, 351511]</td>\n",
       "      <td>[598056, 676216]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>86239</td>\n",
       "      <td>[294338, 178291]</td>\n",
       "      <td>[654736, 625813, 605858, 605859, 605860, 60586...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>86240</td>\n",
       "      <td>[436480, 436481, 436482, 436483, 436484, 43648...</td>\n",
       "      <td>[700171, 597776, 700181, 700183, 700184, 60740...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>86241</td>\n",
       "      <td>[177089, 215868, 418536, 224555, 418861, 41784...</td>\n",
       "      <td>[588673, 647555, 604196, 692323, 599402, 58839...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>86242</td>\n",
       "      <td>[317453, 423832, 415271, 422836, 415671, 41631...</td>\n",
       "      <td>[691556, 691046, 623102, 602826, 630187, 69124...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   promoterID  geneID                                       transcriptID  \\\n",
       "0           0   86238           [352614, 176176, 353426, 354931, 351511]   \n",
       "1           1   86239                                   [294338, 178291]   \n",
       "2           2   86240  [436480, 436481, 436482, 436483, 436484, 43648...   \n",
       "3           3   86241  [177089, 215868, 418536, 224555, 418861, 41784...   \n",
       "4           4   86242  [317453, 423832, 415271, 422836, 415671, 41631...   \n",
       "\n",
       "                                           proteinID  \n",
       "0                                   [598056, 676216]  \n",
       "1  [654736, 625813, 605858, 605859, 605860, 60586...  \n",
       "2  [700171, 597776, 700181, 700183, 700184, 60740...  \n",
       "3  [588673, 647555, 604196, 692323, 599402, 58839...  \n",
       "4  [691556, 691046, 623102, 602826, 630187, 69124...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Define a function to remove .0 suffix from numbers in lists\n",
    "def remove_decimal_suffix(list_with_decimals):\n",
    "    if isinstance(list_with_decimals, list):\n",
    "        return [int(x) if not pd.isna(x) else x for x in list_with_decimals]\n",
    "    return list_with_decimals\n",
    "\n",
    "# Apply the function to both columns\n",
    "ko_internal_relation_index_df['transcriptID'] = ko_internal_relation_index_df['transcriptID'].apply(remove_decimal_suffix)\n",
    "ko_internal_relation_index_df['proteinID'] = ko_internal_relation_index_df['proteinID'].apply(remove_decimal_suffix)\n",
    "\n",
    "# Display the result to verify\n",
    "display(ko_internal_relation_index_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 415,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH_ID</th>\n",
       "      <th>Value</th>\n",
       "      <th>promoterID</th>\n",
       "      <th>geneID</th>\n",
       "      <th>transcriptID</th>\n",
       "      <th>proteinID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>0</td>\n",
       "      <td>86238</td>\n",
       "      <td>[352614, 176176, 353426, 354931, 351511]</td>\n",
       "      <td>[598056, 676216]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>1</td>\n",
       "      <td>86239</td>\n",
       "      <td>[294338, 178291]</td>\n",
       "      <td>[654736, 625813, 605858, 605859, 605860, 60586...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.116327</td>\n",
       "      <td>2</td>\n",
       "      <td>86240</td>\n",
       "      <td>[436480, 436481, 436482, 436483, 436484, 43648...</td>\n",
       "      <td>[700171, 597776, 700181, 700183, 700184, 60740...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.019295</td>\n",
       "      <td>3</td>\n",
       "      <td>86241</td>\n",
       "      <td>[177089, 215868, 418536, 224555, 418861, 41784...</td>\n",
       "      <td>[588673, 647555, 604196, 692323, 599402, 58839...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.100900</td>\n",
       "      <td>4</td>\n",
       "      <td>86242</td>\n",
       "      <td>[317453, 423832, 415271, 422836, 415671, 41631...</td>\n",
       "      <td>[691556, 691046, 623102, 602826, 630187, 69124...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041319</th>\n",
       "      <td>362</td>\n",
       "      <td>0.421384</td>\n",
       "      <td>69807</td>\n",
       "      <td>156045</td>\n",
       "      <td>[448476]</td>\n",
       "      <td>[704608]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041320</th>\n",
       "      <td>362</td>\n",
       "      <td>-0.077387</td>\n",
       "      <td>69840</td>\n",
       "      <td>156078</td>\n",
       "      <td>[448865, 448866, 448867, 448868, 448869]</td>\n",
       "      <td>[704699, 704700, 704701, 704702]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041321</th>\n",
       "      <td>362</td>\n",
       "      <td>0.126152</td>\n",
       "      <td>69847</td>\n",
       "      <td>156085</td>\n",
       "      <td>[388361]</td>\n",
       "      <td>[688824]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041322</th>\n",
       "      <td>362</td>\n",
       "      <td>-0.190182</td>\n",
       "      <td>69875</td>\n",
       "      <td>156113</td>\n",
       "      <td>[449038]</td>\n",
       "      <td>[704769]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041323</th>\n",
       "      <td>362</td>\n",
       "      <td>0.047632</td>\n",
       "      <td>70237</td>\n",
       "      <td>156475</td>\n",
       "      <td>[376832, 377017, 361182]</td>\n",
       "      <td>[599258]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6041324 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ACH_ID     Value  promoterID  geneID  \\\n",
       "0             0 -0.121964           0   86238   \n",
       "1             0 -0.022127           1   86239   \n",
       "2             0 -0.116327           2   86240   \n",
       "3             0 -0.019295           3   86241   \n",
       "4             0 -0.100900           4   86242   \n",
       "...         ...       ...         ...     ...   \n",
       "6041319     362  0.421384       69807  156045   \n",
       "6041320     362 -0.077387       69840  156078   \n",
       "6041321     362  0.126152       69847  156085   \n",
       "6041322     362 -0.190182       69875  156113   \n",
       "6041323     362  0.047632       70237  156475   \n",
       "\n",
       "                                              transcriptID  \\\n",
       "0                 [352614, 176176, 353426, 354931, 351511]   \n",
       "1                                         [294338, 178291]   \n",
       "2        [436480, 436481, 436482, 436483, 436484, 43648...   \n",
       "3        [177089, 215868, 418536, 224555, 418861, 41784...   \n",
       "4        [317453, 423832, 415271, 422836, 415671, 41631...   \n",
       "...                                                    ...   \n",
       "6041319                                           [448476]   \n",
       "6041320           [448865, 448866, 448867, 448868, 448869]   \n",
       "6041321                                           [388361]   \n",
       "6041322                                           [449038]   \n",
       "6041323                           [376832, 377017, 361182]   \n",
       "\n",
       "                                                 proteinID  \n",
       "0                                         [598056, 676216]  \n",
       "1        [654736, 625813, 605858, 605859, 605860, 60586...  \n",
       "2        [700171, 597776, 700181, 700183, 700184, 60740...  \n",
       "3        [588673, 647555, 604196, 692323, 599402, 58839...  \n",
       "4        [691556, 691046, 623102, 602826, 630187, 69124...  \n",
       "...                                                    ...  \n",
       "6041319                                           [704608]  \n",
       "6041320                   [704699, 704700, 704701, 704702]  \n",
       "6041321                                           [688824]  \n",
       "6041322                                           [704769]  \n",
       "6041323                                           [599258]  \n",
       "\n",
       "[6041324 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH_ID</th>\n",
       "      <th>Value</th>\n",
       "      <th>merged_ids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.121964</td>\n",
       "      <td>[0, 86238, 352614, 176176, 353426, 354931, 351...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.022127</td>\n",
       "      <td>[1, 86239, 294338, 178291, 654736, 625813, 605...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.116327</td>\n",
       "      <td>[2, 86240, 436480, 436481, 436482, 436483, 436...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.019295</td>\n",
       "      <td>[3, 86241, 177089, 215868, 418536, 224555, 418...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>-0.100900</td>\n",
       "      <td>[4, 86242, 317453, 423832, 415271, 422836, 415...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041319</th>\n",
       "      <td>362</td>\n",
       "      <td>0.421384</td>\n",
       "      <td>[69807, 156045, 448476, 704608]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041320</th>\n",
       "      <td>362</td>\n",
       "      <td>-0.077387</td>\n",
       "      <td>[69840, 156078, 448865, 448866, 448867, 448868...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041321</th>\n",
       "      <td>362</td>\n",
       "      <td>0.126152</td>\n",
       "      <td>[69847, 156085, 388361, 688824]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041322</th>\n",
       "      <td>362</td>\n",
       "      <td>-0.190182</td>\n",
       "      <td>[69875, 156113, 449038, 704769]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041323</th>\n",
       "      <td>362</td>\n",
       "      <td>0.047632</td>\n",
       "      <td>[70237, 156475, 376832, 377017, 361182, 599258]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6041324 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ACH_ID     Value                                         merged_ids\n",
       "0             0 -0.121964  [0, 86238, 352614, 176176, 353426, 354931, 351...\n",
       "1             0 -0.022127  [1, 86239, 294338, 178291, 654736, 625813, 605...\n",
       "2             0 -0.116327  [2, 86240, 436480, 436481, 436482, 436483, 436...\n",
       "3             0 -0.019295  [3, 86241, 177089, 215868, 418536, 224555, 418...\n",
       "4             0 -0.100900  [4, 86242, 317453, 423832, 415271, 422836, 415...\n",
       "...         ...       ...                                                ...\n",
       "6041319     362  0.421384                    [69807, 156045, 448476, 704608]\n",
       "6041320     362 -0.077387  [69840, 156078, 448865, 448866, 448867, 448868...\n",
       "6041321     362  0.126152                    [69847, 156085, 388361, 688824]\n",
       "6041322     362 -0.190182                    [69875, 156113, 449038, 704769]\n",
       "6041323     362  0.047632    [70237, 156475, 376832, 377017, 361182, 599258]\n",
       "\n",
       "[6041324 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH_ID</th>\n",
       "      <th>merged_ids</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>[0, 86238, 352614, 176176, 353426, 354931, 351...</td>\n",
       "      <td>-0.121964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>[1, 86239, 294338, 178291, 654736, 625813, 605...</td>\n",
       "      <td>-0.022127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>[2, 86240, 436480, 436481, 436482, 436483, 436...</td>\n",
       "      <td>-0.116327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>[3, 86241, 177089, 215868, 418536, 224555, 418...</td>\n",
       "      <td>-0.019295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>[4, 86242, 317453, 423832, 415271, 422836, 415...</td>\n",
       "      <td>-0.100900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041319</th>\n",
       "      <td>362</td>\n",
       "      <td>[69807, 156045, 448476, 704608]</td>\n",
       "      <td>0.421384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041320</th>\n",
       "      <td>362</td>\n",
       "      <td>[69840, 156078, 448865, 448866, 448867, 448868...</td>\n",
       "      <td>-0.077387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041321</th>\n",
       "      <td>362</td>\n",
       "      <td>[69847, 156085, 388361, 688824]</td>\n",
       "      <td>0.126152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041322</th>\n",
       "      <td>362</td>\n",
       "      <td>[69875, 156113, 449038, 704769]</td>\n",
       "      <td>-0.190182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6041323</th>\n",
       "      <td>362</td>\n",
       "      <td>[70237, 156475, 376832, 377017, 361182, 599258]</td>\n",
       "      <td>0.047632</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6041324 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ACH_ID                                         merged_ids     Value\n",
       "0             0  [0, 86238, 352614, 176176, 353426, 354931, 351... -0.121964\n",
       "1             0  [1, 86239, 294338, 178291, 654736, 625813, 605... -0.022127\n",
       "2             0  [2, 86240, 436480, 436481, 436482, 436483, 436... -0.116327\n",
       "3             0  [3, 86241, 177089, 215868, 418536, 224555, 418... -0.019295\n",
       "4             0  [4, 86242, 317453, 423832, 415271, 422836, 415... -0.100900\n",
       "...         ...                                                ...       ...\n",
       "6041319     362                    [69807, 156045, 448476, 704608]  0.421384\n",
       "6041320     362  [69840, 156078, 448865, 448866, 448867, 448868... -0.077387\n",
       "6041321     362                    [69847, 156085, 388361, 688824]  0.126152\n",
       "6041322     362                    [69875, 156113, 449038, 704769] -0.190182\n",
       "6041323     362    [70237, 156475, 376832, 377017, 361182, 599258]  0.047632\n",
       "\n",
       "[6041324 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The shape of train_crispr_score_label_melted_ko_array is: (6041324, 3)\n",
      "[[0\n",
      "  list([0, 86238, 352614, 176176, 353426, 354931, 351511, 598056, 676216])\n",
      "  -0.121963874711081]\n",
      " [0\n",
      "  list([1, 86239, 294338, 178291, 654736, 625813, 605858, 605859, 605860, 605861, 605862, 605863, 605864, 623402, 623403, 603184, 615867, 615868, 595795, 622430, 586590, 605161, 596210, 598899, 596211, 612213])\n",
      "  -0.0221269150315085]\n",
      " [0\n",
      "  list([2, 86240, 436480, 436481, 436482, 436483, 436484, 436485, 436486, 436487, 309640, 436360, 436488, 436489, 436490, 436367, 436368, 436471, 436510, 436511, 248104, 308397, 436397, 436398, 436399, 436400, 436401, 436402, 436403, 310456, 436409, 436410, 308923, 436411, 436412, 436413, 436414, 436415, 436416, 314180, 308678, 436423, 436304, 273233, 436305, 436306, 436440, 436441, 436442, 436443, 436444, 436445, 195688, 436464, 436465, 436338, 436466, 436467, 436468, 436469, 436470, 436478, 436479, 700171, 597776, 700181, 700183, 700184, 607403, 700206, 700207, 700208, 700209, 585010, 661683, 700210, 700214, 662839, 700215, 700216, 612156, 700223, 700233, 700234, 661451, 700251, 700252, 700253, 700254, 700255, 700256, 700257, 700258, 700260, 700261, 700262, 700263, 700264, 700265, 700266, 700148, 700276, 700277, 700154, 700155])\n",
      "  -0.116327356450305]\n",
      " [0\n",
      "  list([3, 86241, 177089, 215868, 418536, 224555, 418861, 417842, 206579, 418012, 418516, 419250, 212604, 588673, 647555, 604196, 692323, 599402, 588395, 646763, 693100, 692622, 616979, 646483, 692734])\n",
      "  -0.0192946993158802]\n",
      " [0\n",
      "  list([4, 86242, 317453, 423832, 415271, 422836, 415671, 416315, 248012, 176332, 422611, 415581, 415458, 270691, 185061, 273257, 192239, 192240, 422898, 319092, 415476, 416375, 271737, 424570, 691556, 691046, 623102, 602826, 630187, 691245, 696126, 694961, 691090, 696242, 695253, 631838, 610427, 610428, 696381, 634430])\n",
      "  -0.1008997308404627]\n",
      " [0\n",
      "  list([12, 86250, 175173, 322503, 231210, 321450, 319788, 321133, 322572, 322159, 223959, 364025, 321050, 319871, 680357, 596726, 665471])\n",
      "  -2.3150760471303844]\n",
      " [0\n",
      "  list([14, 86252, 182851, 299366, 197256, 302670, 301199, 307251, 300404, 305983, 298455, 299071, 632162, 659173, 633655, 659443, 657079, 658169])\n",
      "  0.1772646368738158]\n",
      " [0\n",
      "  list([102, 86340, 188547, 307716, 300295, 300555, 302476, 306966, 306585, 303641, 306083, 298404, 304425, 189235, 304595, 305749, 305116, 302562, 185828, 188262, 305767, 307050, 302832, 303220, 656766, 658500, 659364, 658280, 658505, 657070, 659184, 658065, 659608, 634202, 656637, 631774])\n",
      "  -0.2858075564757956]\n",
      " [0\n",
      "  list([116, 86354, 264707, 266311, 173137, 267346, 269523, 268244, 221781, 273401, 396082, 245657, 266140, 243421, 263646, 643745, 648258, 650018, 659083, 690191, 659385, 594940, 658494, 658303])\n",
      "  -0.0541863499450035]\n",
      " [0\n",
      "  list([117, 86355, 197344, 189409, 197345, 197347, 187047, 271403, 197356, 197358, 197359, 308817, 256282, 197342, 197343, 616832, 635387, 661104])\n",
      "  -0.3317673159668758]]\n"
     ]
    }
   ],
   "source": [
    "train_crispr_score_label_melted_ko = train_crispr_score_label_melted.merge(ko_internal_relation_index_df, left_on='BioMedGraphica_Conn_ID',right_on='geneID' ,how='left')\n",
    "train_crispr_score_label_melted_ko = train_crispr_score_label_melted_ko.drop(columns=['BioMedGraphica_Conn_ID'])\n",
    "display(train_crispr_score_label_melted_ko)\n",
    "\n",
    "# Create a new column 'merged_ids' by combining the columns into lists\n",
    "train_crispr_score_label_melted_ko['merged_ids'] = train_crispr_score_label_melted_ko.apply(\n",
    "    lambda row: [row['promoterID'], row['geneID']] + \n",
    "                (row['transcriptID'] if isinstance(row['transcriptID'], list) else []) +\n",
    "                (row['proteinID'] if isinstance(row['proteinID'], list) else []),\n",
    "    axis=1\n",
    ")\n",
    "\n",
    "# Display the result\n",
    "train_crispr_score_label_melted_ko = train_crispr_score_label_melted_ko.drop(columns=['promoterID', 'geneID', 'transcriptID', 'proteinID'])\n",
    "display(train_crispr_score_label_melted_ko)\n",
    "\n",
    "# reorder the columns to ['ACH_ID', 'merged_ids', 'Value']\n",
    "train_crispr_score_label_melted_ko = train_crispr_score_label_melted_ko[['ACH_ID', 'merged_ids', 'Value']].copy()\n",
    "display(train_crispr_score_label_melted_ko)\n",
    "# Convert train_crispr_score_label_melted_ko to numpy array and save it to the CRISPR-Graph folder\n",
    "train_crispr_score_label_melted_ko_array = train_crispr_score_label_melted_ko.to_numpy()\n",
    "print('The shape of train_crispr_score_label_melted_ko_array is:', train_crispr_score_label_melted_ko_array.shape)\n",
    "print(train_crispr_score_label_melted_ko_array[:10])  # Print first 10 rows as a sample\n",
    "# Save the numpy array to the CRISPR-Graph folder\n",
    "np.save('./data/TargetScreen/train_crispr_score_label_melted_ko.npy', train_crispr_score_label_melted_ko_array)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 416,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH_ID</th>\n",
       "      <th>Value</th>\n",
       "      <th>promoterID</th>\n",
       "      <th>geneID</th>\n",
       "      <th>transcriptID</th>\n",
       "      <th>proteinID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>-0.164485</td>\n",
       "      <td>0</td>\n",
       "      <td>86238</td>\n",
       "      <td>[352614, 176176, 353426, 354931, 351511]</td>\n",
       "      <td>[598056, 676216]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>-0.043958</td>\n",
       "      <td>1</td>\n",
       "      <td>86239</td>\n",
       "      <td>[294338, 178291]</td>\n",
       "      <td>[654736, 625813, 605858, 605859, 605860, 60586...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>-0.140311</td>\n",
       "      <td>2</td>\n",
       "      <td>86240</td>\n",
       "      <td>[436480, 436481, 436482, 436483, 436484, 43648...</td>\n",
       "      <td>[700171, 597776, 700181, 700183, 700184, 60740...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>-0.159200</td>\n",
       "      <td>3</td>\n",
       "      <td>86241</td>\n",
       "      <td>[177089, 215868, 418536, 224555, 418861, 41784...</td>\n",
       "      <td>[588673, 647555, 604196, 692323, 599402, 58839...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>0.041047</td>\n",
       "      <td>4</td>\n",
       "      <td>86242</td>\n",
       "      <td>[317453, 423832, 415271, 422836, 415671, 41631...</td>\n",
       "      <td>[691556, 691046, 623102, 602826, 630187, 69124...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269552</th>\n",
       "      <td>347</td>\n",
       "      <td>0.031412</td>\n",
       "      <td>69807</td>\n",
       "      <td>156045</td>\n",
       "      <td>[448476]</td>\n",
       "      <td>[704608]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269553</th>\n",
       "      <td>347</td>\n",
       "      <td>-0.418786</td>\n",
       "      <td>69840</td>\n",
       "      <td>156078</td>\n",
       "      <td>[448865, 448866, 448867, 448868, 448869]</td>\n",
       "      <td>[704699, 704700, 704701, 704702]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269554</th>\n",
       "      <td>347</td>\n",
       "      <td>0.190287</td>\n",
       "      <td>69847</td>\n",
       "      <td>156085</td>\n",
       "      <td>[388361]</td>\n",
       "      <td>[688824]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269555</th>\n",
       "      <td>347</td>\n",
       "      <td>-0.172861</td>\n",
       "      <td>69875</td>\n",
       "      <td>156113</td>\n",
       "      <td>[449038]</td>\n",
       "      <td>[704769]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269556</th>\n",
       "      <td>347</td>\n",
       "      <td>-0.011810</td>\n",
       "      <td>70237</td>\n",
       "      <td>156475</td>\n",
       "      <td>[376832, 377017, 361182]</td>\n",
       "      <td>[599258]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1269557 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ACH_ID     Value  promoterID  geneID  \\\n",
       "0             5 -0.164485           0   86238   \n",
       "1             5 -0.043958           1   86239   \n",
       "2             5 -0.140311           2   86240   \n",
       "3             5 -0.159200           3   86241   \n",
       "4             5  0.041047           4   86242   \n",
       "...         ...       ...         ...     ...   \n",
       "1269552     347  0.031412       69807  156045   \n",
       "1269553     347 -0.418786       69840  156078   \n",
       "1269554     347  0.190287       69847  156085   \n",
       "1269555     347 -0.172861       69875  156113   \n",
       "1269556     347 -0.011810       70237  156475   \n",
       "\n",
       "                                              transcriptID  \\\n",
       "0                 [352614, 176176, 353426, 354931, 351511]   \n",
       "1                                         [294338, 178291]   \n",
       "2        [436480, 436481, 436482, 436483, 436484, 43648...   \n",
       "3        [177089, 215868, 418536, 224555, 418861, 41784...   \n",
       "4        [317453, 423832, 415271, 422836, 415671, 41631...   \n",
       "...                                                    ...   \n",
       "1269552                                           [448476]   \n",
       "1269553           [448865, 448866, 448867, 448868, 448869]   \n",
       "1269554                                           [388361]   \n",
       "1269555                                           [449038]   \n",
       "1269556                           [376832, 377017, 361182]   \n",
       "\n",
       "                                                 proteinID  \n",
       "0                                         [598056, 676216]  \n",
       "1        [654736, 625813, 605858, 605859, 605860, 60586...  \n",
       "2        [700171, 597776, 700181, 700183, 700184, 60740...  \n",
       "3        [588673, 647555, 604196, 692323, 599402, 58839...  \n",
       "4        [691556, 691046, 623102, 602826, 630187, 69124...  \n",
       "...                                                    ...  \n",
       "1269552                                           [704608]  \n",
       "1269553                   [704699, 704700, 704701, 704702]  \n",
       "1269554                                           [688824]  \n",
       "1269555                                           [704769]  \n",
       "1269556                                           [599258]  \n",
       "\n",
       "[1269557 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH_ID</th>\n",
       "      <th>Value</th>\n",
       "      <th>merged_ids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>-0.164485</td>\n",
       "      <td>[0, 86238, 352614, 176176, 353426, 354931, 351...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>-0.043958</td>\n",
       "      <td>[1, 86239, 294338, 178291, 654736, 625813, 605...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>-0.140311</td>\n",
       "      <td>[2, 86240, 436480, 436481, 436482, 436483, 436...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>-0.159200</td>\n",
       "      <td>[3, 86241, 177089, 215868, 418536, 224555, 418...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>0.041047</td>\n",
       "      <td>[4, 86242, 317453, 423832, 415271, 422836, 415...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269552</th>\n",
       "      <td>347</td>\n",
       "      <td>0.031412</td>\n",
       "      <td>[69807, 156045, 448476, 704608]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269553</th>\n",
       "      <td>347</td>\n",
       "      <td>-0.418786</td>\n",
       "      <td>[69840, 156078, 448865, 448866, 448867, 448868...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269554</th>\n",
       "      <td>347</td>\n",
       "      <td>0.190287</td>\n",
       "      <td>[69847, 156085, 388361, 688824]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269555</th>\n",
       "      <td>347</td>\n",
       "      <td>-0.172861</td>\n",
       "      <td>[69875, 156113, 449038, 704769]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269556</th>\n",
       "      <td>347</td>\n",
       "      <td>-0.011810</td>\n",
       "      <td>[70237, 156475, 376832, 377017, 361182, 599258]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1269557 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ACH_ID     Value                                         merged_ids\n",
       "0             5 -0.164485  [0, 86238, 352614, 176176, 353426, 354931, 351...\n",
       "1             5 -0.043958  [1, 86239, 294338, 178291, 654736, 625813, 605...\n",
       "2             5 -0.140311  [2, 86240, 436480, 436481, 436482, 436483, 436...\n",
       "3             5 -0.159200  [3, 86241, 177089, 215868, 418536, 224555, 418...\n",
       "4             5  0.041047  [4, 86242, 317453, 423832, 415271, 422836, 415...\n",
       "...         ...       ...                                                ...\n",
       "1269552     347  0.031412                    [69807, 156045, 448476, 704608]\n",
       "1269553     347 -0.418786  [69840, 156078, 448865, 448866, 448867, 448868...\n",
       "1269554     347  0.190287                    [69847, 156085, 388361, 688824]\n",
       "1269555     347 -0.172861                    [69875, 156113, 449038, 704769]\n",
       "1269556     347 -0.011810    [70237, 156475, 376832, 377017, 361182, 599258]\n",
       "\n",
       "[1269557 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH_ID</th>\n",
       "      <th>merged_ids</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>[0, 86238, 352614, 176176, 353426, 354931, 351...</td>\n",
       "      <td>-0.164485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>[1, 86239, 294338, 178291, 654736, 625813, 605...</td>\n",
       "      <td>-0.043958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>[2, 86240, 436480, 436481, 436482, 436483, 436...</td>\n",
       "      <td>-0.140311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>[3, 86241, 177089, 215868, 418536, 224555, 418...</td>\n",
       "      <td>-0.159200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>[4, 86242, 317453, 423832, 415271, 422836, 415...</td>\n",
       "      <td>0.041047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269552</th>\n",
       "      <td>347</td>\n",
       "      <td>[69807, 156045, 448476, 704608]</td>\n",
       "      <td>0.031412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269553</th>\n",
       "      <td>347</td>\n",
       "      <td>[69840, 156078, 448865, 448866, 448867, 448868...</td>\n",
       "      <td>-0.418786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269554</th>\n",
       "      <td>347</td>\n",
       "      <td>[69847, 156085, 388361, 688824]</td>\n",
       "      <td>0.190287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269555</th>\n",
       "      <td>347</td>\n",
       "      <td>[69875, 156113, 449038, 704769]</td>\n",
       "      <td>-0.172861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269556</th>\n",
       "      <td>347</td>\n",
       "      <td>[70237, 156475, 376832, 377017, 361182, 599258]</td>\n",
       "      <td>-0.011810</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1269557 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ACH_ID                                         merged_ids     Value\n",
       "0             5  [0, 86238, 352614, 176176, 353426, 354931, 351... -0.164485\n",
       "1             5  [1, 86239, 294338, 178291, 654736, 625813, 605... -0.043958\n",
       "2             5  [2, 86240, 436480, 436481, 436482, 436483, 436... -0.140311\n",
       "3             5  [3, 86241, 177089, 215868, 418536, 224555, 418... -0.159200\n",
       "4             5  [4, 86242, 317453, 423832, 415271, 422836, 415...  0.041047\n",
       "...         ...                                                ...       ...\n",
       "1269552     347                    [69807, 156045, 448476, 704608]  0.031412\n",
       "1269553     347  [69840, 156078, 448865, 448866, 448867, 448868... -0.418786\n",
       "1269554     347                    [69847, 156085, 388361, 688824]  0.190287\n",
       "1269555     347                    [69875, 156113, 449038, 704769] -0.172861\n",
       "1269556     347    [70237, 156475, 376832, 377017, 361182, 599258] -0.011810\n",
       "\n",
       "[1269557 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The shape of test_crispr_score_label_melted_ko_array is: (1269557, 3)\n",
      "[[5\n",
      "  list([0, 86238, 352614, 176176, 353426, 354931, 351511, 598056, 676216])\n",
      "  -0.1644853469607449]\n",
      " [5\n",
      "  list([1, 86239, 294338, 178291, 654736, 625813, 605858, 605859, 605860, 605861, 605862, 605863, 605864, 623402, 623403, 603184, 615867, 615868, 595795, 622430, 586590, 605161, 596210, 598899, 596211, 612213])\n",
      "  -0.0439583570818749]\n",
      " [5\n",
      "  list([2, 86240, 436480, 436481, 436482, 436483, 436484, 436485, 436486, 436487, 309640, 436360, 436488, 436489, 436490, 436367, 436368, 436471, 436510, 436511, 248104, 308397, 436397, 436398, 436399, 436400, 436401, 436402, 436403, 310456, 436409, 436410, 308923, 436411, 436412, 436413, 436414, 436415, 436416, 314180, 308678, 436423, 436304, 273233, 436305, 436306, 436440, 436441, 436442, 436443, 436444, 436445, 195688, 436464, 436465, 436338, 436466, 436467, 436468, 436469, 436470, 436478, 436479, 700171, 597776, 700181, 700183, 700184, 607403, 700206, 700207, 700208, 700209, 585010, 661683, 700210, 700214, 662839, 700215, 700216, 612156, 700223, 700233, 700234, 661451, 700251, 700252, 700253, 700254, 700255, 700256, 700257, 700258, 700260, 700261, 700262, 700263, 700264, 700265, 700266, 700148, 700276, 700277, 700154, 700155])\n",
      "  -0.1403111491750245]\n",
      " [5\n",
      "  list([3, 86241, 177089, 215868, 418536, 224555, 418861, 417842, 206579, 418012, 418516, 419250, 212604, 588673, 647555, 604196, 692323, 599402, 588395, 646763, 693100, 692622, 616979, 646483, 692734])\n",
      "  -0.1592002771555088]\n",
      " [5\n",
      "  list([4, 86242, 317453, 423832, 415271, 422836, 415671, 416315, 248012, 176332, 422611, 415581, 415458, 270691, 185061, 273257, 192239, 192240, 422898, 319092, 415476, 416375, 271737, 424570, 691556, 691046, 623102, 602826, 630187, 691245, 696126, 694961, 691090, 696242, 695253, 631838, 610427, 610428, 696381, 634430])\n",
      "  0.0410467662451924]\n",
      " [5\n",
      "  list([12, 86250, 175173, 322503, 231210, 321450, 319788, 321133, 322572, 322159, 223959, 364025, 321050, 319871, 680357, 596726, 665471])\n",
      "  -1.8195263470618903]\n",
      " [5\n",
      "  list([14, 86252, 182851, 299366, 197256, 302670, 301199, 307251, 300404, 305983, 298455, 299071, 632162, 659173, 633655, 659443, 657079, 658169])\n",
      "  0.078415984198699]\n",
      " [5\n",
      "  list([102, 86340, 188547, 307716, 300295, 300555, 302476, 306966, 306585, 303641, 306083, 298404, 304425, 189235, 304595, 305749, 305116, 302562, 185828, 188262, 305767, 307050, 302832, 303220, 656766, 658500, 659364, 658280, 658505, 657070, 659184, 658065, 659608, 634202, 656637, 631774])\n",
      "  -0.2516705401625799]\n",
      " [5\n",
      "  list([116, 86354, 264707, 266311, 173137, 267346, 269523, 268244, 221781, 273401, 396082, 245657, 266140, 243421, 263646, 643745, 648258, 650018, 659083, 690191, 659385, 594940, 658494, 658303])\n",
      "  -0.0700853969862718]\n",
      " [5\n",
      "  list([117, 86355, 197344, 189409, 197345, 197347, 187047, 271403, 197356, 197358, 197359, 308817, 256282, 197342, 197343, 616832, 635387, 661104])\n",
      "  -0.0512091586464747]]\n"
     ]
    }
   ],
   "source": [
    "test_crispr_score_label_melted_ko = test_crispr_score_label_melted.merge(ko_internal_relation_index_df, left_on='BioMedGraphica_Conn_ID',right_on='geneID' ,how='left')\n",
    "test_crispr_score_label_melted_ko = test_crispr_score_label_melted_ko.drop(columns=['BioMedGraphica_Conn_ID'])\n",
    "display(test_crispr_score_label_melted_ko)\n",
    "\n",
    "# Create a new column 'merged_ids' by combining the columns into lists\n",
    "test_crispr_score_label_melted_ko['merged_ids'] = test_crispr_score_label_melted_ko.apply(\n",
    "    lambda row: [row['promoterID'], row['geneID']] + \n",
    "                (row['transcriptID'] if isinstance(row['transcriptID'], list) else []) +\n",
    "                (row['proteinID'] if isinstance(row['proteinID'], list) else []),\n",
    "    axis=1\n",
    ")\n",
    "\n",
    "# Display the result\n",
    "test_crispr_score_label_melted_ko = test_crispr_score_label_melted_ko.drop(columns=['promoterID', 'geneID', 'transcriptID', 'proteinID'])\n",
    "display(test_crispr_score_label_melted_ko)\n",
    "\n",
    "# reorder the columns to ['ACH_ID', 'merged_ids', 'Value']\n",
    "test_crispr_score_label_melted_ko = test_crispr_score_label_melted_ko[['ACH_ID', 'merged_ids', 'Value']].copy()\n",
    "display(test_crispr_score_label_melted_ko)\n",
    "# Convert test_crispr_score_label_melted_ko to numpy array and save it to the CRISPR-Graph folder\n",
    "test_crispr_score_label_melted_ko_array = test_crispr_score_label_melted_ko.to_numpy()\n",
    "print('The shape of test_crispr_score_label_melted_ko_array is:', test_crispr_score_label_melted_ko_array.shape)\n",
    "print(test_crispr_score_label_melted_ko_array[:10])  # Print first 10 rows as a sample\n",
    "# Save the numpy array to the CRISPR-Graph folder\n",
    "np.save('./data/TargetScreen/test_crispr_score_label_melted_ko.npy', test_crispr_score_label_melted_ko_array)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 9.2.2.2 TargetQA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Omic Feature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Select the features from target_crispr_df\n",
    "target_crispr_feat = target_crispr_df.drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "# Save the features to .npy files\n",
    "np.save('./data/TargetQA/target_crispr_feature.npy', target_crispr_feat)\n",
    "# Print the shapes of the features\n",
    "print(\"Shape of target_crispr_feat:\", target_crispr_feat.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Omic Information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# select the columns in the overlapped_dti_crispr_rna_samples for gene_df\n",
    "dti_crispr_rna_gene_df = gene_df[['gene_name'] + sorted(overlapped_dti_crispr_rna_samples)].copy()\n",
    "display(dti_crispr_rna_gene_df)\n",
    "bmgc_protein_llmnameid_combined_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein_LLM_Name_ID_Combined.csv')\n",
    "display(bmgc_protein_llmnameid_combined_df)\n",
    "\n",
    "def extract_gn_info(dti_crispr_rna_gene_df, bmgc_gene_df, gene_transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k=10):\n",
    "    # Check if sample_ach_name exists in dti_crispr_rna_gene_df columns\n",
    "    if sample_ach_name not in dti_crispr_rna_gene_df.columns:\n",
    "        return \"non-existed\", \"non-existed\", \"non-existed\"\n",
    "    # Extract the top k highest values for the given sample name\n",
    "    top_k_genes = dti_crispr_rna_gene_df.nlargest(k, sample_ach_name)[['gene_name', sample_ach_name]]\n",
    "    # Sort the top k genes by their values in descending order\n",
    "    top_k_genes = top_k_genes.sort_values(by=sample_ach_name, ascending=False).reset_index(drop=True)\n",
    "    top_k_gene_hgnc_name_list = top_k_genes['gene_name'].tolist()\n",
    "    # Merge with the bmgc_gene_df to get the BioMedGraphica_Conn_ID\n",
    "    bmgc_gene_df = bmgc_gene_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']].copy()\n",
    "    top_k_bmgc_gene_df = pd.merge(bmgc_gene_df, top_k_genes, left_on='HGNC_Symbol', right_on='gene_name', how='inner')\n",
    "    # Get the corresponding proteins\n",
    "    top_k_bmgc_gene_protein_df = pd.merge(gene_transcript_protein_entity_df, top_k_bmgc_gene_df, left_on='BMGC_GN_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'gene_name'])\n",
    "    top_k_bmgc_gene_protein_info_df = pd.merge(top_k_bmgc_gene_protein_df, bmgc_protein_llmnameid_combined_df, left_on='BMGC_PT_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'BioMedGraphica_ID', sample_ach_name])\n",
    "    top_k_gene_protein_bmgc_id_list = top_k_bmgc_gene_protein_info_df['BMGC_PT_ID'].tolist()\n",
    "    top_k_gene_protein_bmgc_llmnameid_list = top_k_bmgc_gene_protein_info_df['Names_and_IDs'].replace(r' \\| ', ' or ', regex=True).tolist()\n",
    "    return top_k_gene_hgnc_name_list, top_k_gene_protein_bmgc_id_list, top_k_gene_protein_bmgc_llmnameid_list\n",
    "\n",
    "# Example usage\n",
    "sample_ach_name = 'ACH-000001'\n",
    "k=10\n",
    "top_k_gene_hgnc_name_list, top_k_gene_protein_bmgc_id_list, top_k_gene_protein_bmgc_llmnameid_list = extract_gn_info(dti_crispr_rna_gene_df, bmgc_gene_df, gene_transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "print(f\"Top {k} Gene HGNC Names:\", top_k_gene_hgnc_name_list)\n",
    "print(f\"Top {k} Gene Protein BMGC IDs:\", top_k_gene_protein_bmgc_id_list)\n",
    "print(f\"Top {k} Gene Protein BMGC LLM Name IDs:\", top_k_gene_protein_bmgc_llmnameid_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# select the columns in the overlapped_dti_crispr_rna_samples for raw_transcript_df\n",
    "transcript_overlapped_dti_crispr_rna_samples = sorted(list(set(raw_transcript_df.columns[1:]) & set(overlapped_dti_crispr_rna_samples)))\n",
    "dti_crispr_rna_transcript_df = raw_transcript_df[['gene_name'] + transcript_overlapped_dti_crispr_rna_samples].copy()\n",
    "display(dti_crispr_rna_transcript_df)\n",
    "bmgc_transcript_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Transcript/BioMedGraphica_Conn_Transcript.csv')\n",
    "\n",
    "def extract_ts_info(dti_crispr_rna_transcript_df, bmgc_transcript_df, transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k=10):\n",
    "    # Check if sample_ach_name exists in dti_crispr_rna_transcript_df columns\n",
    "    if sample_ach_name not in dti_crispr_rna_transcript_df.columns:\n",
    "        return \"non-existed\", \"non-existed\", \"non-existed\"\n",
    "    # Extract the top k highest values for the given sample name\n",
    "    top_k_transcripts = dti_crispr_rna_transcript_df.nlargest(k, sample_ach_name)[['gene_name', sample_ach_name]]\n",
    "    # Sort the top k transcripts by their values in descending order\n",
    "    top_k_transcripts = top_k_transcripts.sort_values(by=sample_ach_name, ascending=False).reset_index(drop=True)\n",
    "    top_k_transcript_hgnc_name_list = top_k_transcripts['gene_name'].tolist()\n",
    "    # Merge with the bmgc_transcript_df to get the BioMedGraphica_Conn_ID\n",
    "    bmgc_transcript_df = bmgc_transcript_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']].copy()\n",
    "    top_k_bmgc_transcript_df = pd.merge(bmgc_transcript_df, top_k_transcripts, left_on='HGNC_Symbol', right_on='gene_name', how='inner')\n",
    "    # Get the corresponding proteins\n",
    "    top_k_bmgc_transcript_protein_df = pd.merge(transcript_protein_entity_df, top_k_bmgc_transcript_df, left_on='BMGC_TS_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'gene_name'])\n",
    "    top_k_bmgc_transcript_protein_info_df = pd.merge(top_k_bmgc_transcript_protein_df, bmgc_protein_llmnameid_combined_df, left_on='BMGC_PT_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'BioMedGraphica_ID', sample_ach_name])\n",
    "    top_k_transcript_protein_bmgc_id_list = top_k_bmgc_transcript_protein_info_df['BMGC_PT_ID'].tolist()\n",
    "    top_k_transcript_protein_bmgc_llmnameid_list = top_k_bmgc_transcript_protein_info_df['Names_and_IDs'].replace(r' \\| ', ' or ', regex=True).tolist()\n",
    "    return top_k_transcript_hgnc_name_list, top_k_transcript_protein_bmgc_id_list, top_k_transcript_protein_bmgc_llmnameid_list\n",
    "\n",
    "# Example usage\n",
    "sample_ach_name = 'ACH-000001'\n",
    "k=10\n",
    "top_k_transcript_hgnc_name_list, top_k_transcript_protein_bmgc_id_list, top_k_transcript_protein_bmgc_llmnameid_list = extract_ts_info(dti_crispr_rna_transcript_df, bmgc_transcript_df, transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "print(f\"Top {k} Transcript HGNC Names:\", top_k_transcript_hgnc_name_list)\n",
    "print(f\"Top {k} Transcript Protein BMGC IDs:\", top_k_transcript_protein_bmgc_id_list)\n",
    "print(f\"Top {k} Transcript Protein BMGC LLM Name IDs:\", top_k_transcript_protein_bmgc_llmnameid_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bmg_protein_all_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein.csv')\n",
    "bmg_protein_all_df = bmg_protein_all_df[['BioMedGraphica_Conn_ID', 'Uniprot_ID', 'HGNC_Symbol']].copy()\n",
    "display(bmg_protein_all_df)\n",
    "\n",
    "# Rename columns in raw_protein_df using the provided mapping\n",
    "raw_protein_map_df = raw_protein_df.rename(columns=protein_map_dict)\n",
    "# Merge with bmg_protein_all_df to get HGNC symbols\n",
    "symbol_protein_map_df = pd.merge(raw_protein_map_df, bmg_protein_all_df, left_on='Uniprot_Acc', right_on='Uniprot_ID', how='inner')\n",
    "# Reorder columns: keep Uniprot IDs and protein expression values\n",
    "symbol_protein_map_df = symbol_protein_map_df[['Uniprot_ID', 'Uniprot_Acc', 'HGNC_Symbol'] + sorted(set(symbol_protein_map_df.columns) - {'Uniprot_ID', 'Uniprot_Acc', 'HGNC_Symbol'})]\n",
    "# Identify overlapping samples between protein data and the provided sample list\n",
    "protein_overlapped_samples = sorted(set(symbol_protein_map_df.columns) & set(overlapped_dti_crispr_rna_samples))\n",
    "# Select only HGNC symbol and overlapping sample columns\n",
    "dti_crispr_rna_protein_df = symbol_protein_map_df[['HGNC_Symbol'] + protein_overlapped_samples].copy()\n",
    "# Split multiple HGNC symbols by \";\" and expand into multiple rows\n",
    "dti_crispr_rna_protein_df = dti_crispr_rna_protein_df.assign(HGNC_Symbol=dti_crispr_rna_protein_df['HGNC_Symbol'].str.split(';')).explode('HGNC_Symbol')\n",
    "# Remove leading/trailing whitespace in gene symbols\n",
    "dti_crispr_rna_protein_df['HGNC_Symbol'] = dti_crispr_rna_protein_df['HGNC_Symbol'].str.strip()\n",
    "# Drop rows with empty or missing gene symbols\n",
    "dti_crispr_rna_protein_df = dti_crispr_rna_protein_df[dti_crispr_rna_protein_df['HGNC_Symbol'].notna() & (dti_crispr_rna_protein_df['HGNC_Symbol'] != '')].reset_index(drop=True)\n",
    "# Display final dataframe\n",
    "display(dti_crispr_rna_protein_df)\n",
    "\n",
    "bmgc_protein_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein.csv')\n",
    "\n",
    "def extract_pt_info(dti_crispr_rna_protein_df, bmgc_protein_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k=10):\n",
    "    # Check if sample_ach_name exists in dti_crispr_rna_protein_df columns\n",
    "    if sample_ach_name not in dti_crispr_rna_protein_df.columns:\n",
    "        return \"non-existed\", \"non-existed\", \"non-existed\"\n",
    "    # Extract the top k highest values for the given sample name\n",
    "    top_k_proteins = dti_crispr_rna_protein_df.nlargest(k, sample_ach_name)[['HGNC_Symbol', sample_ach_name]]\n",
    "    # Sort the top k proteins by their values in descending order\n",
    "    top_k_proteins = top_k_proteins.sort_values(by=sample_ach_name, ascending=False).reset_index(drop=True)\n",
    "    top_k_protein_hgnc_name_list = top_k_proteins['HGNC_Symbol'].tolist()\n",
    "    # Merge with the bmgc_protein_df to get the BioMedGraphica_Conn_ID\n",
    "    bmgc_protein_df = bmgc_protein_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']].copy()\n",
    "    top_k_bmgc_protein_df = pd.merge(bmgc_protein_df, top_k_proteins, left_on='HGNC_Symbol', right_on='HGNC_Symbol', how='inner')\n",
    "    # Get the corresponding protein information\n",
    "    top_k_bmgc_protein_info_df = pd.merge(top_k_bmgc_protein_df, bmgc_protein_llmnameid_combined_df, left_on='BioMedGraphica_Conn_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_ID', sample_ach_name])\n",
    "    top_k_protein_bmgc_id_list = top_k_bmgc_protein_info_df['BioMedGraphica_Conn_ID'].tolist()\n",
    "    # Replace both \"|\" and \";\" with \" or \"\n",
    "    top_k_protein_bmgc_llmnameid_list = top_k_bmgc_protein_info_df['Names_and_IDs'].replace([r' \\| ', r';'], ' or ', regex=True).tolist()\n",
    "    return top_k_protein_hgnc_name_list, top_k_protein_bmgc_id_list, top_k_protein_bmgc_llmnameid_list\n",
    "\n",
    "# Example usage\n",
    "sample_ach_name = 'ACH-000001'\n",
    "k = 10\n",
    "top_k_protein_hgnc_name_list, top_k_protein_bmgc_id_list, top_k_protein_bmgc_llmnameid_list = extract_pt_info(dti_crispr_rna_protein_df, bmgc_protein_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "print(f\"Top {k} Protein HGNC Names:\", top_k_protein_hgnc_name_list)\n",
    "print(f\"Top {k} Protein BMGC IDs:\", top_k_protein_bmgc_id_list)\n",
    "print(f\"Top {k} Protein BMGC LLM Name IDs:\", top_k_protein_bmgc_llmnameid_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Related Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def bmgc_pt_id_to_hgnc(bmgc_id_list, bmgc_protein_df):\n",
    "    \"\"\"\n",
    "    Convert a list of BioMedGraphica IDs to their corresponding HGNC symbols.\n",
    "    \n",
    "    Args:\n",
    "        bmgc_id_list (list): List of BioMedGraphica IDs\n",
    "        bmgc_protein_df (pd.DataFrame): DataFrame with BioMedGraphica IDs and HGNC symbols\n",
    "        \n",
    "    Returns:\n",
    "        tuple: (\n",
    "            dict: Dictionary mapping each BioMedGraphica ID to its list of HGNC symbols,\n",
    "            list: Combined list of all HGNC symbols\n",
    "        )\n",
    "    \"\"\"\n",
    "    # Ensure bmgc_id_list is actually a list\n",
    "    if not isinstance(bmgc_id_list, list):\n",
    "        bmgc_id_list = [bmgc_id_list]\n",
    "    results = {}\n",
    "    all_hgnc_symbols = []\n",
    "    for bmgc_id in bmgc_id_list:\n",
    "        # Filter the DataFrame for the given BioMedGraphica ID\n",
    "        filtered_df = bmgc_protein_df[bmgc_protein_df['BioMedGraphica_Conn_ID'] == bmgc_id]\n",
    "        # Skip if no match found\n",
    "        if filtered_df.empty:\n",
    "            results[bmgc_id] = []\n",
    "            continue\n",
    "        # Get the HGNC symbols\n",
    "        hgnc_value = filtered_df['HGNC_Symbol'].values[0]\n",
    "        # Skip if HGNC symbol is NaN\n",
    "        if pd.isna(hgnc_value):\n",
    "            results[bmgc_id] = []\n",
    "            continue\n",
    "        # Process valid HGNC symbols\n",
    "        hgnc_list = list(set(hgnc_value.split(';')))\n",
    "        hgnc_list = [hgnc.strip() for hgnc in hgnc_list if hgnc.strip() != '']\n",
    "        results[bmgc_id] = hgnc_list\n",
    "        all_hgnc_symbols.extend(hgnc_list)\n",
    "    # Remove duplicates from the combined list\n",
    "    all_hgnc_symbols = list(set(all_hgnc_symbols))\n",
    "    return results, all_hgnc_symbols\n",
    "\n",
    "def hgnc_to_bmgc_pt_id(hgnc_list, bmgc_protein_df):\n",
    "    \"\"\"\n",
    "    Convert a list of HGNC symbols to their corresponding BioMedGraphica IDs.\n",
    "    \n",
    "    Args:\n",
    "        hgnc_list (list): List of HGNC symbols\n",
    "        bmgc_protein_df (pd.DataFrame): DataFrame with BioMedGraphica IDs and HGNC symbols\n",
    "        \n",
    "    Returns:\n",
    "        tuple: (\n",
    "            dict: Dictionary mapping each HGNC symbol to its list of BioMedGraphica IDs,\n",
    "            list: Combined list of all BioMedGraphica IDs\n",
    "        )\n",
    "    \"\"\"\n",
    "    # Ensure hgnc_list is actually a list\n",
    "    if not isinstance(hgnc_list, list):\n",
    "        hgnc_list = [hgnc_list]\n",
    "    results = {}\n",
    "    all_bmgc_ids = []\n",
    "    for hgnc in hgnc_list:\n",
    "        # Filter the DataFrame for the given HGNC symbol\n",
    "        filtered_df = bmgc_protein_df[bmgc_protein_df['HGNC_Symbol'] == hgnc]\n",
    "        # Skip if no match found\n",
    "        if filtered_df.empty:\n",
    "            results[hgnc] = []\n",
    "            continue\n",
    "        # Get the BioMedGraphica IDs\n",
    "        bmgc_value = filtered_df['BioMedGraphica_Conn_ID'].values[0]\n",
    "        # Skip if BioMedGraphica ID is NaN\n",
    "        if pd.isna(bmgc_value):\n",
    "            results[hgnc] = []\n",
    "            continue\n",
    "        # Process valid BioMedGraphica IDs\n",
    "        bmgc_list = list(set(bmgc_value.split(';')))\n",
    "        bmgc_list = [bmgc.strip() for bmgc in bmgc_list if bmgc.strip() != '']\n",
    "        results[hgnc] = bmgc_list\n",
    "        all_bmgc_ids.extend(bmgc_list)\n",
    "    # Remove duplicates from the combined list\n",
    "    all_bmgc_ids = list(set(all_bmgc_ids))\n",
    "    return results, all_bmgc_ids\n",
    "\n",
    "# Example usage\n",
    "bmgc_ids = ['BMGC_PT000001', 'BMGC_PT013541']\n",
    "hgnc_dict, all_hgnc_symbols = bmgc_pt_id_to_hgnc(bmgc_ids, bmgc_protein_df)\n",
    "print(hgnc_dict)\n",
    "print(all_hgnc_symbols)\n",
    "# Example usage\n",
    "hgnc_list = ['BRCA1', 'TP53']\n",
    "bmgc_dict, all_bmgc_ids = hgnc_to_bmgc_pt_id(hgnc_list, bmgc_protein_df)\n",
    "print(bmgc_dict)\n",
    "print(all_bmgc_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_disease_protein(selected_sample_disease_bmgc_id, edge_index, \n",
    "                    node_index_df, nodeid_index_dict, index_nodeid_dict):\n",
    "    # Extract the index based on the selected disease BMGC ID\n",
    "    sample_disease_bmgc_id_index = nodeid_index_dict[selected_sample_disease_bmgc_id]\n",
    "    # Find incoming edges (source nodes that point to the disease)\n",
    "    incoming_mask = edge_index[1, :] == sample_disease_bmgc_id_index\n",
    "    incoming_source_nodes = edge_index[0, incoming_mask]\n",
    "    # Find outgoing edges (target nodes that the disease points to)\n",
    "    outgoing_mask = edge_index[0, :] == sample_disease_bmgc_id_index\n",
    "    outgoing_target_nodes = edge_index[1, outgoing_mask]\n",
    "    # Combine all neighbor nodes (both incoming and outgoing)\n",
    "    disease_related_nodes = np.concatenate([incoming_source_nodes, outgoing_target_nodes])\n",
    "    unique_disease_related_nodes = np.unique(disease_related_nodes)\n",
    "    # Get protein node index\n",
    "    protein_node_index_df = node_index_df[node_index_df['Type'] == 'Protein']\n",
    "    protein_node_index_list = protein_node_index_df['Index'].tolist()\n",
    "    # Filter to get only protein nodes directly connected to the disease\n",
    "    disease_protein_index = sorted(\n",
    "        list(set(unique_disease_related_nodes) & set(protein_node_index_list))\n",
    "    )\n",
    "    # Map protein index to BMGC id\n",
    "    disease_protein_bmgc_id = [index_nodeid_dict[i] for i in disease_protein_index]\n",
    "    return disease_protein_index, disease_protein_bmgc_id\n",
    "\n",
    "def extract_ppi_nodes(disease_protein_index, edge_index, node_index_df, index_nodeid_dict):\n",
    "    # Get protein node index\n",
    "    protein_node_index_df = node_index_df[node_index_df['Type'] == 'Protein']\n",
    "    protein_node_index_list = protein_node_index_df['Index'].tolist()\n",
    "    # Get all nodes related to the identified protein neighbors (second hop)\n",
    "    protein_related_nodes = []\n",
    "    # Iterate through each protein neighbor node index\n",
    "    for protein_node_idx in disease_protein_index:\n",
    "        # Find incoming edges (nodes that point to this protein)\n",
    "        protein_incoming_mask = edge_index[1, :] == protein_node_idx\n",
    "        protein_incoming_sources = edge_index[0, protein_incoming_mask]\n",
    "        # Find outgoing edges (nodes that this protein points to)\n",
    "        protein_outgoing_mask = edge_index[0, :] == protein_node_idx\n",
    "        protein_outgoing_targets = edge_index[1, protein_outgoing_mask]\n",
    "        # Add these connected nodes to our list\n",
    "        protein_related_nodes.extend(protein_incoming_sources)\n",
    "        protein_related_nodes.extend(protein_outgoing_targets)\n",
    "    # Convert to numpy array and get unique nodes\n",
    "    protein_related_nodes = np.array(protein_related_nodes)\n",
    "    unique_protein_related_nodes = np.unique(protein_related_nodes)\n",
    "    # Remove any protein nodes themselves from this list to avoid duplication\n",
    "    unique_protein_related_nodes = np.setdiff1d(\n",
    "        unique_protein_related_nodes, disease_protein_index\n",
    "    )\n",
    "    # Filter to only keep protein nodes among the second-hop neighbors\n",
    "    ppi_nodes_index = sorted(\n",
    "        list(set(unique_protein_related_nodes) & set(protein_node_index_list))\n",
    "    )\n",
    "    # Map PPI node index to BMGC id\n",
    "    ppi_nodes_bmgc_id = [index_nodeid_dict[i] for i in ppi_nodes_index]\n",
    "    return ppi_nodes_index, ppi_nodes_bmgc_id\n",
    "\n",
    "def extract_kg_related_proteins(selected_sample_disease_bmgc_id, edge_index,  # './data/DTI_data/edge_index.npy'\n",
    "                               node_index_df, nodeid_index_dict, index_nodeid_dict):\n",
    "    \"\"\"\n",
    "    Extract disease-related proteins and their interactions from the knowledge graph.\n",
    "    \n",
    "    Args:\n",
    "        selected_sample_disease_bmgc_id (str): BMGC ID of the selected disease\n",
    "        edge_index (str): Edge index file\n",
    "        node_index_df (pd.DataFrame): DataFrame with node type information\n",
    "        nodeid_index_dict (dict): Mapping from node ID to index\n",
    "        index_nodeid_dict (dict): Mapping from index to node ID\n",
    "    \n",
    "    Returns:\n",
    "        tuple: (disease_protein_index, disease_protein_bmgc_id, ppi_nodes_index, ppi_nodes_bmgc_id)\n",
    "    \"\"\"\n",
    "    \n",
    "    # Extract disease-protein connections\n",
    "    disease_protein_index, disease_protein_bmgc_id = extract_disease_protein(\n",
    "        selected_sample_disease_bmgc_id=selected_sample_disease_bmgc_id,\n",
    "        edge_index=edge_index,\n",
    "        node_index_df=node_index_df,\n",
    "        nodeid_index_dict=nodeid_index_dict,\n",
    "        index_nodeid_dict=index_nodeid_dict\n",
    "    )\n",
    "\n",
    "    # Extract protein-protein interactions (can replace this with LLM to generate PPI, may need NER and mapping to BMGC id)\n",
    "    ppi_nodes_index, ppi_nodes_bmgc_id = extract_ppi_nodes(\n",
    "        disease_protein_index=disease_protein_index,\n",
    "        edge_index=edge_index,\n",
    "        node_index_df=node_index_df,\n",
    "        index_nodeid_dict=index_nodeid_dict\n",
    "    )\n",
    "\n",
    "    # Convert the BMGC id into HGNC symbol\n",
    "\n",
    "    return disease_protein_index, disease_protein_bmgc_id, ppi_nodes_index, ppi_nodes_bmgc_id\n",
    "\n",
    "# Example usage\n",
    "selected_sample_disease_bmgc_id = 'BMGC_DS07934'\n",
    "edge_index = np.load('./data/TargetQA/edge_index.npy')\n",
    "index_node_dict = dict(zip(nodes_index_data['Index'], nodes_index_data['Node']))\n",
    "\n",
    "disease_protein_index, disease_protein_bmgc_id, ppi_nodes_index, ppi_nodes_bmgc_id = extract_kg_related_proteins(\n",
    "    selected_sample_disease_bmgc_id=selected_sample_disease_bmgc_id,\n",
    "    edge_index=edge_index,\n",
    "    node_index_df=nodes_index_data,\n",
    "    nodeid_index_dict=node_index_dict,\n",
    "    index_nodeid_dict=index_node_dict\n",
    ")\n",
    "print(\"Disease Protein Index:\", disease_protein_index)\n",
    "print(\"Disease Protein BMGC ID:\", disease_protein_bmgc_id)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### CRISPR Answer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "answer_crispr_df = crispr_df[['gene_name'] + sorted(overlapped_dti_crispr_rna_samples)].copy()\n",
    "display(answer_crispr_df)\n",
    "\n",
    "def extract_answer(answer_crispr_df, bmgc_gene_df, gene_transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, top_bm=100):\n",
    "    # Extract the lowest top_bm values for the given sample name\n",
    "    top_bm_genes = answer_crispr_df.nsmallest(top_bm, sample_ach_name)[['gene_name', sample_ach_name]]\n",
    "    # Sort the top bm genes by their values in ascending order\n",
    "    top_bm_genes = top_bm_genes.sort_values(by=sample_ach_name, ascending=True).reset_index(drop=True)\n",
    "    top_bm_gene_hgnc_name_list = top_bm_genes['gene_name'].tolist()\n",
    "    # Merge with the bmgc_gene_df to get the BioMedGraphica_Conn_ID\n",
    "    bmgc_gene_df = bmgc_gene_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']].copy()\n",
    "    top_bm_bmgc_gene_df = pd.merge(bmgc_gene_df, top_bm_genes, left_on='HGNC_Symbol', right_on='gene_name', how='inner')\n",
    "    # Get the corresponding proteins\n",
    "    top_bm_bmgc_gene_protein_df = pd.merge(gene_transcript_protein_entity_df, top_bm_bmgc_gene_df, left_on='BMGC_GN_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'gene_name'])\n",
    "    top_bm_bmgc_gene_protein_info_df = pd.merge(top_bm_bmgc_gene_protein_df, bmgc_protein_llmnameid_combined_df, left_on='BMGC_PT_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'BioMedGraphica_ID', sample_ach_name])\n",
    "    top_bm_gene_protein_bmgc_id_list = top_bm_bmgc_gene_protein_info_df['BMGC_PT_ID'].tolist()\n",
    "    top_bm_gene_protein_bmgc_llmnameid_list = top_bm_bmgc_gene_protein_info_df['Names_and_IDs'].replace(r' \\| ', ' or ', regex=True).tolist()\n",
    "    return top_bm_gene_hgnc_name_list, top_bm_gene_protein_bmgc_id_list, top_bm_gene_protein_bmgc_llmnameid_list\n",
    "\n",
    "# Example usage\n",
    "sample_ach_name = 'ACH-000001'\n",
    "top_bm = 10\n",
    "bmgc_protein_llmnameid_combined_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein_LLM_Name_ID_Combined.csv')\n",
    "top_bm_gene_hgnc_name_list, top_bm_gene_protein_bmgc_id_list, top_bm_gene_protein_bmgc_llmnameid_list = extract_answer(answer_crispr_df, bmgc_gene_df, gene_transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, top_bm)\n",
    "print(f\"Top {top_bm} Gene HGNC Names:\", top_bm_gene_hgnc_name_list)\n",
    "print(f\"Top {top_bm} Gene Protein BMGC IDs:\", top_bm_gene_protein_bmgc_id_list)\n",
    "print(f\"Top {top_bm} Gene Protein BMGC LLM Name IDs:\", top_bm_gene_protein_bmgc_llmnameid_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Knowledge Graph Information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_protein_relationships(hgnc_symbols, bmgc_protein_df, bmgc_relation_df):\n",
    "    \"\"\"\n",
    "    Find relationships between a list of proteins based on HGNC symbols.\n",
    "    \n",
    "    Parameters:\n",
    "    -----------\n",
    "    hgnc_symbols : list\n",
    "        List of HGNC symbols to find relationships between\n",
    "    bmgc_protein_df : pandas DataFrame\n",
    "        DataFrame containing BioMedGraphica_Conn_ID and HGNC_Symbol columns\n",
    "    bmgc_relation_df : pandas DataFrame\n",
    "        DataFrame containing BMGC_From_ID and BMGC_To_ID columns\n",
    "        \n",
    "    Returns:\n",
    "    --------\n",
    "    tuple (pandas DataFrame, list)\n",
    "        - DataFrame with source_symbol, target_symbol and their relationship\n",
    "        - List of text descriptions of relationships in \"A -> B\" format\n",
    "    \"\"\"\n",
    "    # Filter the protein DataFrame to only include the proteins we care about\n",
    "    filtered_proteins = bmgc_protein_df[bmgc_protein_df['HGNC_Symbol'].isin(hgnc_symbols)]\n",
    "    # Create a mapping from HGNC symbol to BMGC ID\n",
    "    hgnc_to_bmgc = dict(zip(filtered_proteins['HGNC_Symbol'], filtered_proteins['BioMedGraphica_Conn_ID']))\n",
    "    bmgc_to_hgnc = dict(zip(filtered_proteins['BioMedGraphica_Conn_ID'], filtered_proteins['HGNC_Symbol']))\n",
    "    # Get all BMGC IDs of our proteins\n",
    "    bmgc_ids = list(hgnc_to_bmgc.values())\n",
    "    # Filter the relationship DataFrame to only include relationships between our proteins\n",
    "    protein_relations = bmgc_relation_df[\n",
    "        bmgc_relation_df['BMGC_From_ID'].isin(bmgc_ids) & \n",
    "        bmgc_relation_df['BMGC_To_ID'].isin(bmgc_ids)\n",
    "    ]\n",
    "    # Map the BMGC IDs back to HGNC symbols\n",
    "    result_data = []\n",
    "    text_descriptions = []\n",
    "    for _, row in protein_relations.iterrows():\n",
    "        source_bmgc = row['BMGC_From_ID']\n",
    "        target_bmgc = row['BMGC_To_ID']\n",
    "        if source_bmgc in bmgc_to_hgnc and target_bmgc in bmgc_to_hgnc:\n",
    "            source_symbol = bmgc_to_hgnc[source_bmgc]\n",
    "            target_symbol = bmgc_to_hgnc[target_bmgc]\n",
    "            # Create text description\n",
    "            text_description = f\"{source_symbol} -> {target_symbol}\"\n",
    "            text_descriptions.append(text_description)\n",
    "            # If relation_type column exists, include it in the description and data\n",
    "            relation_info = {\n",
    "                'source_symbol': source_symbol,\n",
    "                'target_symbol': target_symbol\n",
    "            }\n",
    "            # Add relation type if it exists in the DataFrame\n",
    "            if 'relation_type' in bmgc_relation_df.columns:\n",
    "                relation_type = row['relation_type']\n",
    "                relation_info['relation_type'] = relation_type\n",
    "                text_descriptions[-1] = f\"{source_symbol} -{relation_type}-> {target_symbol}\"\n",
    "            result_data.append(relation_info)\n",
    "    # Create a DataFrame from the results\n",
    "    result_df = pd.DataFrame(result_data)\n",
    "    return result_df, text_descriptions\n",
    "\n",
    "# Define the HGNC symbols\n",
    "hgnc_symbols = ['SNRPD3', 'RAN', 'RPS8', 'UBL5', 'SMU1', 'RRM1', 'PSMA6', 'PSMB3', 'WEE1', \n",
    "                'PHB1', 'BANF1', 'KIF11', 'SNRPD1', 'PSMA3', 'PSMD11', 'PRPF19', 'SNRPF', \n",
    "                'RPS29', 'CDC27', 'SRSF3', 'TUBGCP2', 'ECD', 'RPS20', 'PCNA', 'PSMA7', 'CDC7', \n",
    "                'RPL17', 'GINS1', 'PHB2', 'SRSF2', 'MAD2L1', 'MED14']\n",
    "\n",
    "# Call the function to find relationships\n",
    "relationships_df, relationship_texts = find_protein_relationships(hgnc_symbols, bmgc_protein_df, bmgc_relation_df)\n",
    "\n",
    "# You can now use both the DataFrame and text descriptions\n",
    "# Example usage:\n",
    "print(f\"Found {len(relationship_texts)} relationships between the proteins\")\n",
    "print(\"Example relationships:\")\n",
    "print(relationship_texts[:5])  # Print first 5 relationships"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Formulate QA JSON"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def qa_sample_info(sample_ach_name, selected_sample_disease_bmgc_id, k=100, top_bm=100):\n",
    "    # LLM Info\n",
    "    print(f\"Sample ACH Name: {sample_ach_name}\")\n",
    "    print(f\"Extracting top {k} gene information for {sample_ach_name}...\")\n",
    "    top_k_gene_hgnc_name_list, top_k_gene_protein_bmgc_id_list, top_k_gene_protein_bmgc_llmnameid_list = extract_gn_info(dti_crispr_rna_gene_df, bmgc_gene_df, gene_transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "    print(f\"Extracting top {k} transcript information for {sample_ach_name}...\")\n",
    "    top_k_transcript_hgnc_name_list, top_k_transcript_protein_bmgc_id_list, top_k_transcript_protein_bmgc_llmnameid_list = extract_ts_info(dti_crispr_rna_transcript_df, bmgc_transcript_df, transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "    print(f\"Extracting top {k} protein information for {sample_ach_name}...\")\n",
    "    top_k_protein_hgnc_name_list, top_k_protein_bmgc_id_list, top_k_protein_bmgc_llmnameid_list = extract_pt_info(dti_crispr_rna_protein_df, bmgc_protein_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "    # KG Info\n",
    "    edge_index = np.load('./data/TargetQA/edge_index.npy')\n",
    "    print(f\"Extracting disease-related proteins index and bmgc id for {selected_sample_disease_bmgc_id} ({sample_ach_name}) ...\")\n",
    "    disease_protein_index, disease_protein_bmgc_id, ppi_nodes_index, ppi_nodes_bmgc_id = extract_kg_related_proteins(selected_sample_disease_bmgc_id, edge_index, nodes_index_data, node_index_dict, index_node_dict)\n",
    "    print(f\"Knowledge Graph Info: Found {len(disease_protein_index)} disease-related proteins directly connected to {selected_sample_disease_bmgc_id} and {len(ppi_nodes_index)} proteins in their PPI network\")\n",
    "    print(f\"Mapping disease-related proteins to HGNC symbols...\")\n",
    "    disease_protein_hgnc_dict, disease_protein_hgnc_list = bmgc_pt_id_to_hgnc(disease_protein_bmgc_id, bmgc_protein_df)\n",
    "    print(f\"Mapping PPI-related proteins to HGNC symbols...\")\n",
    "    ppi_hgnc_dict, ppi_hgnc_list = bmgc_pt_id_to_hgnc(ppi_nodes_bmgc_id, bmgc_protein_df)\n",
    "    # LLM Used KG Info\n",
    "    print(f\"Extracting protein relationships from BMGC...\")\n",
    "    # Convert the any non-existed string in top_k_gene_hgnc_name_list + top_k_transcript_hgnc_name_list + top_k_protein_hgnc_name_list + disease_protein_hgnc_list into empty list []\n",
    "    if top_k_gene_hgnc_name_list == \"non-existed\": top_k_gene_hgnc_name_list = []\n",
    "    if top_k_transcript_hgnc_name_list == \"non-existed\": top_k_transcript_hgnc_name_list = []\n",
    "    if top_k_protein_hgnc_name_list == \"non-existed\": top_k_protein_hgnc_name_list = []\n",
    "    if disease_protein_hgnc_list == \"non-existed\": disease_protein_hgnc_list = []\n",
    "    # Combine all the HGNC symbols into a single list for relationship extraction\n",
    "    omics_disease_protein_hgnc_list = list(set(top_k_gene_hgnc_name_list + top_k_transcript_hgnc_name_list + top_k_protein_hgnc_name_list + disease_protein_hgnc_list))\n",
    "    relationships_df, relationship_texts = find_protein_relationships(omics_disease_protein_hgnc_list, bmgc_protein_df, bmgc_relation_df)\n",
    "    print(f\"Knowledge Graph Info: Found {len(omics_disease_protein_hgnc_list)} unique proteins and {len(relationship_texts)} relationships between them\")\n",
    "    # Answer Info\n",
    "    print(f\"Extracting top {top_bm} CRISPR gene information for {sample_ach_name}...\")\n",
    "    top_bm_gene_hgnc_name_list, top_bm_gene_protein_bmgc_id_list, top_bm_gene_protein_bmgc_llmnameid_list = extract_answer(answer_crispr_df, bmgc_gene_df, gene_transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, top_bm)\n",
    "    return top_k_gene_hgnc_name_list, top_k_gene_protein_bmgc_id_list, top_k_gene_protein_bmgc_llmnameid_list, \\\n",
    "              top_k_transcript_hgnc_name_list, top_k_transcript_protein_bmgc_id_list, top_k_transcript_protein_bmgc_llmnameid_list, \\\n",
    "                top_k_protein_hgnc_name_list, top_k_protein_bmgc_id_list, top_k_protein_bmgc_llmnameid_list, \\\n",
    "                disease_protein_index, disease_protein_bmgc_id, ppi_nodes_index, ppi_nodes_bmgc_id, \\\n",
    "                    disease_protein_hgnc_dict, disease_protein_hgnc_list, ppi_hgnc_dict, ppi_hgnc_list, relationship_texts, \\\n",
    "                        top_bm_gene_hgnc_name_list, top_bm_gene_protein_bmgc_id_list, top_bm_gene_protein_bmgc_llmnameid_list\n",
    "\n",
    "# Example usage\n",
    "sample_ach_name = 'ACH-000001'\n",
    "selected_sample_disease_bmgc_id = 'BMGC_DS07934'\n",
    "k = 100\n",
    "top_bm = 100\n",
    "return_tuples = qa_sample_info(sample_ach_name, selected_sample_disease_bmgc_id, k=k, top_bm=top_bm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import os\n",
    "from tqdm import tqdm\n",
    "\n",
    "# Parameters\n",
    "k = 10\n",
    "top_bm = 100\n",
    "save_every_n = 10\n",
    "\n",
    "# Output folder and filename\n",
    "output_dir = \"./data/TargetQA\"\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "output_filename = f\"target_qa_k{k}_bm{top_bm}.json\"\n",
    "output_path = os.path.join(output_dir, output_filename)\n",
    "\n",
    "# Load existing JSON file if it exists\n",
    "if os.path.exists(output_path):\n",
    "    with open(output_path, \"r\") as f:\n",
    "        multi_sample_qa_json = json.load(f)\n",
    "    print(f\"Loaded existing JSON with {len(multi_sample_qa_json)} processed samples\")\n",
    "else:\n",
    "    multi_sample_qa_json = {}\n",
    "    print(\"No existing JSON found, starting fresh\")\n",
    "\n",
    "# Load sample info data\n",
    "dti_sample_info_index = pd.read_csv('./data/process_data/dti_combined_samples.csv')\n",
    "target_sample_info_index = pd.merge(dti_sample_info_index, overlapped_dti_crispr_rna_samples_df['depMapID'], how='inner', on='depMapID')\n",
    "target_sample_info_index = target_sample_info_index[target_sample_info_index['depMapID'].isin(overlapped_dti_crispr_rna_samples)].reset_index(drop=True)\n",
    "target_sample_info_index['BMGC_Disease_name'] = target_sample_info_index['BMGC_Disease_name'].replace(r' \\| ', ' or ', regex=True)\n",
    "# Insert a new column \"Index\" in the first position\n",
    "target_sample_info_index.insert(0, 'Index', range(1, len(target_sample_info_index) + 1))\n",
    "display(target_sample_info_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the list of samples that have already been processed\n",
    "processed_samples = set(multi_sample_qa_json.keys())\n",
    "print(f\"Found {len(processed_samples)} already processed samples\")\n",
    "\n",
    "count = len(processed_samples)\n",
    "total_to_process = len(target_sample_info_index)\n",
    "remaining = total_to_process - count\n",
    "print(f\"Total samples to process: {total_to_process}, already processed: {count}, remaining: {remaining}\")\n",
    "\n",
    "# Iterate through the sample info dataframe, skipping already processed samples\n",
    "for idx, row_tuple in tqdm(enumerate(target_sample_info_index.iterrows()), total=len(target_sample_info_index)):\n",
    "    _, row = row_tuple  # Unpack the tuple - index and row data\n",
    "\n",
    "    sample_ach_name = row[\"depMapID\"]\n",
    "    \n",
    "    # Skip if already processed\n",
    "    if sample_ach_name in processed_samples:\n",
    "        continue\n",
    "        \n",
    "    count += 1\n",
    "    target_sample_index = row[\"Index\"]\n",
    "    cell_line_name = row[\"Name\"]\n",
    "    disease = row[\"BMGC_Disease_name\"]\n",
    "    disease_bmgc_id = row[\"BMGC_Disease_ID\"]\n",
    "\n",
    "    print(f\"Processing sample {count}/{total_to_process}: {sample_ach_name} ({cell_line_name})\")\n",
    "    print(f\"Sample Index: {target_sample_index}\")\n",
    "    print(f\"Sample Disease: {disease}\")\n",
    "    print(f\"Sample Disease BMGC ID: {disease_bmgc_id}\")\n",
    "\n",
    "    try:\n",
    "        (top_k_gene_hgnc, top_k_gene_bmgc, top_k_gene_llm,\n",
    "        top_k_ts_hgnc, top_k_ts_bmgc, top_k_ts_llm,\n",
    "        top_k_pt_hgnc, top_k_pt_bmgc, top_k_pt_llm,\n",
    "        dis_pt_idx, dis_pt_bmgc, ppi_idx, ppi_bmgc,\n",
    "        dis_pt_hgnc_dict, dis_pt_hgnc, ppi_hgnc_dict, ppi_hgnc, relationship_texts,\n",
    "        ans_hgnc, ans_bmgc, ans_llm) = qa_sample_info(sample_ach_name, disease_bmgc_id, k=k, top_bm=top_bm)\n",
    "\n",
    "        multi_sample_qa_json[sample_ach_name] = {\n",
    "            \"cell_line_name\": cell_line_name,\n",
    "            \"sample_index\": target_sample_index,\n",
    "            \"disease\": disease,\n",
    "            \"disease_bmgc_id\": disease_bmgc_id,\n",
    "            \"input\": {\n",
    "                \"top_k_gene\": {\n",
    "                    \"hgnc_symbols\": top_k_gene_hgnc,\n",
    "                    \"protein_bmgc_ids\": top_k_gene_bmgc,\n",
    "                    \"protein_llmname_ids\": top_k_gene_llm\n",
    "                },\n",
    "                \"top_k_transcript\": {\n",
    "                    \"hgnc_symbols\": top_k_ts_hgnc,\n",
    "                    \"protein_bmgc_ids\": top_k_ts_bmgc,\n",
    "                    \"protein_llmname_ids\": top_k_ts_llm\n",
    "                },\n",
    "                \"top_k_protein\": {\n",
    "                    \"hgnc_symbols\": top_k_pt_hgnc,\n",
    "                    \"protein_bmgc_ids\": top_k_pt_bmgc,\n",
    "                    \"protein_llmname_ids\": top_k_pt_llm\n",
    "                },\n",
    "                \"knowledge_graph\": {\n",
    "                    \"disease_protein\": {\n",
    "                        \"bmgc_ids\": dis_pt_bmgc,\n",
    "                        \"hgnc_symbols\": dis_pt_hgnc,\n",
    "                        \"indices\": dis_pt_idx\n",
    "                    },\n",
    "                    \"ppi_neighbors\": {\n",
    "                        \"bmgc_ids\": ppi_bmgc,\n",
    "                        \"hgnc_symbols\": ppi_hgnc,\n",
    "                        \"indices\": ppi_idx\n",
    "                    },\n",
    "                    \"protein_relationships\": relationship_texts,\n",
    "                }\n",
    "            },\n",
    "            \"ground_truth_answer\": {\n",
    "                \"top_bm_gene\": {\n",
    "                    \"hgnc_symbols\": ans_hgnc,\n",
    "                    \"protein_bmgc_ids\": ans_bmgc,\n",
    "                    \"protein_llmname_ids\": ans_llm\n",
    "                }\n",
    "            }\n",
    "        }\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"⚠️ Error processing {sample_ach_name}: {e}\")\n",
    "        continue\n",
    "\n",
    "    # Periodic save every N samples\n",
    "    if count % save_every_n == 0:\n",
    "        with open(output_path, \"w\") as f:\n",
    "            json.dump(multi_sample_qa_json, f, indent=2, default=lambda o: int(o) if hasattr(o, 'item') else o)\n",
    "        print(f\"💾 Auto-saved JSON at {count}/{total_to_process} samples to: {output_path}\")\n",
    "        print(f\"Last processed sample: {sample_ach_name}\")\n",
    "        processed = len(multi_sample_qa_json)\n",
    "        remaining = total_to_process - processed\n",
    "        print(f\"Progress: {processed}/{total_to_process} ({processed/total_to_process*100:.1f}%), Remaining: {remaining}\")\n",
    "\n",
    "# Final save after loop\n",
    "with open(output_path, \"w\") as f:\n",
    "    json.dump(multi_sample_qa_json, f, indent=2, default=lambda o: int(o) if hasattr(o, 'item') else o)\n",
    "\n",
    "print(f\"✅ Final JSON saved to: {output_path}\")\n",
    "print(f\"Total samples processed: {len(multi_sample_qa_json)}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Split QA as Train and Test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 482,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "\n",
    "def read_json(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        data = json.load(file)\n",
    "    return data\n",
    "\n",
    "def separate_data(data):\n",
    "    # Get the training and testing list\n",
    "    target_tr_df = pd.read_csv('./data/TargetQA/train_samples.csv')\n",
    "    target_te_df = pd.read_csv('./data/TargetQA/test_samples.csv')\n",
    "    target_tr_list = target_tr_df['depMapID'].tolist()\n",
    "    target_te_list = target_te_df['depMapID'].tolist()\n",
    "    # Separate the data into two lists based on tr/te list with (for sample_id, sample_info in data.items())\n",
    "    tr_data = {}\n",
    "    te_data = {}\n",
    "    for sample_id, sample_info in data.items():\n",
    "        if sample_id in target_tr_list:\n",
    "            tr_data[sample_id] = sample_info\n",
    "        elif sample_id in target_te_list:\n",
    "            te_data[sample_id] = sample_info\n",
    "    # Check if the separated data is correct\n",
    "    tr_data_count = len(tr_data)\n",
    "    te_data_count = len(te_data)\n",
    "    print(f\"Training data count: {tr_data_count}\")\n",
    "    print(f\"Testing data count: {te_data_count}\")\n",
    "    # Check if the training and testing data are mutually exclusive\n",
    "    if set(tr_data.keys()).intersection(te_data.keys()):\n",
    "        print(\"Error: Training and testing data are not mutually exclusive.\")\n",
    "    else:\n",
    "        print(\"Training and testing data are mutually exclusive.\")\n",
    "    return tr_data, te_data\n",
    "\n",
    "def save_to_json(data, file_path):\n",
    "    with open(file_path, 'w') as file:\n",
    "        json.dump(data, file, indent=2)\n",
    "    print(f\"Data saved to {file_path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 483,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Training data count: 300\n",
      "Testing data count: 63\n",
      "Training and testing data are mutually exclusive.\n",
      "Data saved to ./data/TargetQA/target_qa_k10_bm100_tr.json\n",
      "Data saved to ./data/TargetQA/target_qa_k10_bm100_te.json\n"
     ]
    }
   ],
   "source": [
    "# Read the JSON file\n",
    "data = read_json('./data/TargetQA/target_qa_k10_bm100.json')\n",
    "\n",
    "# Separate the data into training and testing sets\n",
    "tr_data, te_data = separate_data(data)\n",
    "# Save the separated data to new JSON files\n",
    "save_to_json(tr_data, './data/TargetQA/target_qa_k10_bm100_tr.json')\n",
    "save_to_json(te_data, './data/TargetQA/target_qa_k10_bm100_te.json')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Formulate QA text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 9.2.3 Drug data integration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 389,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.83335</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88351</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.94738</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.50982</td>\n",
       "      <td>0.56520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93319</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.04959</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00129</td>\n",
       "      <td>0.02137</td>\n",
       "      <td>0.01066</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01239</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00958</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.05386</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.33075</td>\n",
       "      <td>0.07825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.72705</td>\n",
       "      <td>0.00761</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.92687</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_PM86234</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_PM86235</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_PM86236</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               BMGC_PM00001     0.83335        0.00     0.88351     1.00000   \n",
       "1               BMGC_PM00002     0.00000        0.00     0.00000     0.00000   \n",
       "2               BMGC_PM00003     0.04959        0.00     0.00129     0.02137   \n",
       "3               BMGC_PM00004     0.05386        0.25     0.00000     0.33075   \n",
       "4               BMGC_PM00005     0.00000        0.00     0.00000     0.00000   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_PM86234     0.00000        0.00     0.00000     0.00000   \n",
       "86234           BMGC_PM86235     0.00000        0.00     0.00000     0.00000   \n",
       "86235           BMGC_PM86236     0.00000        0.00     0.00000     0.00000   \n",
       "86236           BMGC_PM86237     0.00000        0.00     0.00000     0.00000   \n",
       "86237           BMGC_PM86238     0.00000        0.00     0.00000     0.00000   \n",
       "\n",
       "       ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0         0.94738         1.0     0.50982     0.56520         0.0  ...   \n",
       "1         0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "2         0.01066         0.0     0.01239     0.00000         0.0  ...   \n",
       "3         0.07825         0.0     0.72705     0.00761         0.0  ...   \n",
       "4         0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86234     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86235     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86236     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86237     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "       ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  ACH-001151  \\\n",
       "0         0.93319         0.0         0.0         0.0         0.0         0.0   \n",
       "1         0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "2         0.00958         0.0         0.0         0.0         0.0         0.0   \n",
       "3         0.92687         0.0         0.0         0.0         0.0         0.0   \n",
       "4         0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "86234     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "86235     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "86236     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "86237     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "       ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0             0.0         0.0         0.0         0.0  \n",
       "1             0.0         0.0         0.0         0.0  \n",
       "2             0.0         0.0         0.0         0.0  \n",
       "3             0.0         0.0         0.0         0.0  \n",
       "4             0.0         0.0         0.0         0.0  \n",
       "...           ...         ...         ...         ...  \n",
       "86233         0.0         0.0         0.0         0.0  \n",
       "86234         0.0         0.0         0.0         0.0  \n",
       "86235         0.0         0.0         0.0         0.0  \n",
       "86236         0.0         0.0         0.0         0.0  \n",
       "86237         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[86238 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n",
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\3891391275.py:12: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  mapped_protein_df[sample] = 0.0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               BMGC_PT000001         0.0         0.0         0.0         0.0   \n",
       "1               BMGC_PT000002         0.0         0.0         0.0         0.0   \n",
       "2               BMGC_PT000003         0.0         0.0         0.0         0.0   \n",
       "3               BMGC_PT000004         0.0         0.0         0.0         0.0   \n",
       "4               BMGC_PT000005         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "121414          BMGC_PT121415         0.0         0.0         0.0         0.0   \n",
       "121415          BMGC_PT121416         0.0         0.0         0.0         0.0   \n",
       "121416          BMGC_PT121417         0.0         0.0         0.0         0.0   \n",
       "121417          BMGC_PT121418         0.0         0.0         0.0         0.0   \n",
       "121418          BMGC_PT121419         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "1              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "2              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "3              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "4              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121415         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121416         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121417         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121418         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "\n",
       "        ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0   \n",
       "1              0.0         0.0         0.0         0.0         0.0   \n",
       "2              0.0         0.0         0.0         0.0         0.0   \n",
       "3              0.0         0.0         0.0         0.0         0.0   \n",
       "4              0.0         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0   \n",
       "121415         0.0         0.0         0.0         0.0         0.0   \n",
       "121416         0.0         0.0         0.0         0.0         0.0   \n",
       "121417         0.0         0.0         0.0         0.0         0.0   \n",
       "121418         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001151  ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0              0.0         0.0         0.0         0.0         0.0  \n",
       "1              0.0         0.0         0.0         0.0         0.0  \n",
       "2              0.0         0.0         0.0         0.0         0.0  \n",
       "3              0.0         0.0         0.0         0.0         0.0  \n",
       "4              0.0         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "121414         0.0         0.0         0.0         0.0         0.0  \n",
       "121415         0.0         0.0         0.0         0.0         0.0  \n",
       "121416         0.0         0.0         0.0         0.0         0.0  \n",
       "121417         0.0         0.0         0.0         0.0         0.0  \n",
       "121418         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[121419 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# insert the samples that is in the remaining_dti_samples but not in the maped_methy_df\n",
    "for sample in remaining_dti_samples:\n",
    "    if sample not in maped_methy_df.columns:\n",
    "        maped_methy_df[sample] = 0.0\n",
    "# filter out the samples that is not in the remaining_dti_samples\n",
    "dti_methy_df = maped_methy_df[['BioMedGraphica_Conn_ID'] + remaining_dti_samples]\n",
    "display(dti_methy_df)\n",
    "\n",
    "# insert the samples that is in the remaining_dti_samples but not in the mapped_protein_df\n",
    "for sample in remaining_dti_samples:\n",
    "    if sample not in mapped_protein_df.columns:\n",
    "        mapped_protein_df[sample] = 0.0\n",
    "# filter out the samples that is not in the remaining_dti_samples\n",
    "dti_protein_df = mapped_protein_df[['BioMedGraphica_Conn_ID'] + remaining_dti_samples]\n",
    "display(dti_protein_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 390,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>1.039342</td>\n",
       "      <td>1.081979</td>\n",
       "      <td>1.069887</td>\n",
       "      <td>1.021275</td>\n",
       "      <td>1.019950</td>\n",
       "      <td>0.873542</td>\n",
       "      <td>1.954398</td>\n",
       "      <td>0.476005</td>\n",
       "      <td>0.498743</td>\n",
       "      <td>...</td>\n",
       "      <td>0.930320</td>\n",
       "      <td>1.280913</td>\n",
       "      <td>0.816260</td>\n",
       "      <td>0.996102</td>\n",
       "      <td>1.018304</td>\n",
       "      <td>1.050028</td>\n",
       "      <td>1.005951</td>\n",
       "      <td>1.040949</td>\n",
       "      <td>1.409362</td>\n",
       "      <td>0.863441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>0.915792</td>\n",
       "      <td>1.032808</td>\n",
       "      <td>0.843013</td>\n",
       "      <td>0.959549</td>\n",
       "      <td>1.009547</td>\n",
       "      <td>0.971925</td>\n",
       "      <td>1.188910</td>\n",
       "      <td>1.511536</td>\n",
       "      <td>1.047491</td>\n",
       "      <td>...</td>\n",
       "      <td>1.009008</td>\n",
       "      <td>1.076599</td>\n",
       "      <td>0.906371</td>\n",
       "      <td>1.057301</td>\n",
       "      <td>0.828024</td>\n",
       "      <td>0.445954</td>\n",
       "      <td>1.003790</td>\n",
       "      <td>1.021203</td>\n",
       "      <td>0.519903</td>\n",
       "      <td>1.112888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>0.978356</td>\n",
       "      <td>0.909618</td>\n",
       "      <td>1.676606</td>\n",
       "      <td>0.990115</td>\n",
       "      <td>0.993623</td>\n",
       "      <td>0.953061</td>\n",
       "      <td>1.229717</td>\n",
       "      <td>1.566409</td>\n",
       "      <td>1.034708</td>\n",
       "      <td>...</td>\n",
       "      <td>1.075723</td>\n",
       "      <td>1.279596</td>\n",
       "      <td>1.212607</td>\n",
       "      <td>1.399931</td>\n",
       "      <td>0.986576</td>\n",
       "      <td>1.268384</td>\n",
       "      <td>0.997014</td>\n",
       "      <td>0.996969</td>\n",
       "      <td>0.995337</td>\n",
       "      <td>0.380086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>1.514518</td>\n",
       "      <td>1.024754</td>\n",
       "      <td>0.868152</td>\n",
       "      <td>1.921635</td>\n",
       "      <td>1.010449</td>\n",
       "      <td>0.986412</td>\n",
       "      <td>1.231253</td>\n",
       "      <td>1.062672</td>\n",
       "      <td>1.010414</td>\n",
       "      <td>...</td>\n",
       "      <td>1.063356</td>\n",
       "      <td>0.750272</td>\n",
       "      <td>0.880448</td>\n",
       "      <td>0.750693</td>\n",
       "      <td>1.368224</td>\n",
       "      <td>0.465270</td>\n",
       "      <td>0.989083</td>\n",
       "      <td>0.969761</td>\n",
       "      <td>1.023638</td>\n",
       "      <td>0.410329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>1.032720</td>\n",
       "      <td>1.165655</td>\n",
       "      <td>1.251083</td>\n",
       "      <td>0.987747</td>\n",
       "      <td>1.006196</td>\n",
       "      <td>1.438704</td>\n",
       "      <td>1.172893</td>\n",
       "      <td>1.024139</td>\n",
       "      <td>1.042507</td>\n",
       "      <td>...</td>\n",
       "      <td>1.019288</td>\n",
       "      <td>1.084336</td>\n",
       "      <td>0.968458</td>\n",
       "      <td>0.988188</td>\n",
       "      <td>1.071999</td>\n",
       "      <td>0.924170</td>\n",
       "      <td>0.999373</td>\n",
       "      <td>1.046953</td>\n",
       "      <td>1.001543</td>\n",
       "      <td>1.245752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_GN86234</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_GN86235</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_GN86236</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               BMGC_GN00001    1.039342    1.081979    1.069887    1.021275   \n",
       "1               BMGC_GN00002    0.915792    1.032808    0.843013    0.959549   \n",
       "2               BMGC_GN00003    0.978356    0.909618    1.676606    0.990115   \n",
       "3               BMGC_GN00004    1.514518    1.024754    0.868152    1.921635   \n",
       "4               BMGC_GN00005    1.032720    1.165655    1.251083    0.987747   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_GN86234    0.000000    0.000000    0.000000    0.000000   \n",
       "86234           BMGC_GN86235    0.000000    0.000000    0.000000    0.000000   \n",
       "86235           BMGC_GN86236    0.000000    0.000000    0.000000    0.000000   \n",
       "86236           BMGC_GN86237    0.000000    0.000000    0.000000    0.000000   \n",
       "86237           BMGC_GN86238    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0        1.019950    0.873542    1.954398    0.476005    0.498743  ...   \n",
       "1        1.009547    0.971925    1.188910    1.511536    1.047491  ...   \n",
       "2        0.993623    0.953061    1.229717    1.566409    1.034708  ...   \n",
       "3        1.010449    0.986412    1.231253    1.062672    1.010414  ...   \n",
       "4        1.006196    1.438704    1.172893    1.024139    1.042507  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "       ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  ACH-001151  \\\n",
       "0        0.930320    1.280913    0.816260    0.996102    1.018304    1.050028   \n",
       "1        1.009008    1.076599    0.906371    1.057301    0.828024    0.445954   \n",
       "2        1.075723    1.279596    1.212607    1.399931    0.986576    1.268384   \n",
       "3        1.063356    0.750272    0.880448    0.750693    1.368224    0.465270   \n",
       "4        1.019288    1.084336    0.968458    0.988188    1.071999    0.924170   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0        1.005951    1.040949    1.409362    0.863441  \n",
       "1        1.003790    1.021203    0.519903    1.112888  \n",
       "2        0.997014    0.996969    0.995337    0.380086  \n",
       "3        0.989083    0.969761    1.023638    0.410329  \n",
       "4        0.999373    1.046953    1.001543    1.245752  \n",
       "...           ...         ...         ...         ...  \n",
       "86233    0.000000    0.000000    0.000000    0.000000  \n",
       "86234    0.000000    0.000000    0.000000    0.000000  \n",
       "86235    0.000000    0.000000    0.000000    0.000000  \n",
       "86236    0.000000    0.000000    0.000000    0.000000  \n",
       "86237    0.000000    0.000000    0.000000    0.000000  \n",
       "\n",
       "[86238 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>6.522291</td>\n",
       "      <td>5.854020</td>\n",
       "      <td>7.104063</td>\n",
       "      <td>6.875601</td>\n",
       "      <td>6.986633</td>\n",
       "      <td>7.254845</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.293006</td>\n",
       "      <td>6.912981</td>\n",
       "      <td>...</td>\n",
       "      <td>7.422634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.220747</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>6.729154</td>\n",
       "      <td>6.338599</td>\n",
       "      <td>7.897515</td>\n",
       "      <td>6.917430</td>\n",
       "      <td>7.218023</td>\n",
       "      <td>7.815267</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.232725</td>\n",
       "      <td>6.944961</td>\n",
       "      <td>...</td>\n",
       "      <td>6.139888</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.795080</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>4.419503</td>\n",
       "      <td>4.986058</td>\n",
       "      <td>4.621525</td>\n",
       "      <td>3.843395</td>\n",
       "      <td>4.675843</td>\n",
       "      <td>4.200734</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.066766</td>\n",
       "      <td>4.064158</td>\n",
       "      <td>...</td>\n",
       "      <td>4.260150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.455935</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>5.213352</td>\n",
       "      <td>5.151831</td>\n",
       "      <td>5.766890</td>\n",
       "      <td>5.052784</td>\n",
       "      <td>5.001630</td>\n",
       "      <td>6.215894</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.329526</td>\n",
       "      <td>5.736236</td>\n",
       "      <td>...</td>\n",
       "      <td>5.175159</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.969341</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>0.043941</td>\n",
       "      <td>-0.073548</td>\n",
       "      <td>0.335595</td>\n",
       "      <td>0.005892</td>\n",
       "      <td>-0.059982</td>\n",
       "      <td>1.853420</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.031376</td>\n",
       "      <td>-0.059982</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.046550</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.881209</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412034</th>\n",
       "      <td>BMGC_TS412035</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412035</th>\n",
       "      <td>BMGC_TS412036</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412036</th>\n",
       "      <td>BMGC_TS412037</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412037</th>\n",
       "      <td>BMGC_TS412038</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412038</th>\n",
       "      <td>BMGC_TS412039</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412039 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               BMGC_TS000001    6.522291    5.854020    7.104063    6.875601   \n",
       "1               BMGC_TS000002    6.729154    6.338599    7.897515    6.917430   \n",
       "2               BMGC_TS000003    4.419503    4.986058    4.621525    3.843395   \n",
       "3               BMGC_TS000004    5.213352    5.151831    5.766890    5.052784   \n",
       "4               BMGC_TS000005    0.043941   -0.073548    0.335595    0.005892   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "412034          BMGC_TS412035    0.000000    0.000000    0.000000    0.000000   \n",
       "412035          BMGC_TS412036    0.000000    0.000000    0.000000    0.000000   \n",
       "412036          BMGC_TS412037    0.000000    0.000000    0.000000    0.000000   \n",
       "412037          BMGC_TS412038    0.000000    0.000000    0.000000    0.000000   \n",
       "412038          BMGC_TS412039    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "        ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0         6.986633    7.254845         0.0    6.293006    6.912981  ...   \n",
       "1         7.218023    7.815267         0.0    5.232725    6.944961  ...   \n",
       "2         4.675843    4.200734         0.0    5.066766    4.064158  ...   \n",
       "3         5.001630    6.215894         0.0    5.329526    5.736236  ...   \n",
       "4        -0.059982    1.853420         0.0    0.031376   -0.059982  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "412034    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "412035    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "412036    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "412037    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "412038    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "\n",
       "        ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  \\\n",
       "0         7.422634         0.0         0.0         0.0         0.0   \n",
       "1         6.139888         0.0         0.0         0.0         0.0   \n",
       "2         4.260150         0.0         0.0         0.0         0.0   \n",
       "3         5.175159         0.0         0.0         0.0         0.0   \n",
       "4        -0.046550         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "412034    0.000000         0.0         0.0         0.0         0.0   \n",
       "412035    0.000000         0.0         0.0         0.0         0.0   \n",
       "412036    0.000000         0.0         0.0         0.0         0.0   \n",
       "412037    0.000000         0.0         0.0         0.0         0.0   \n",
       "412038    0.000000         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001151  ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0         6.220747         0.0         0.0         0.0         0.0  \n",
       "1         6.795080         0.0         0.0         0.0         0.0  \n",
       "2         4.455935         0.0         0.0         0.0         0.0  \n",
       "3         5.969341         0.0         0.0         0.0         0.0  \n",
       "4         2.881209         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "412034    0.000000         0.0         0.0         0.0         0.0  \n",
       "412035    0.000000         0.0         0.0         0.0         0.0  \n",
       "412036    0.000000         0.0         0.0         0.0         0.0  \n",
       "412037    0.000000         0.0         0.0         0.0         0.0  \n",
       "412038    0.000000         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[412039 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "depMapID             0\n",
      "Cell_Line_Name       0\n",
      "BMGC_Drug_ID         0\n",
      "BMGC_Drug_Name       0\n",
      "BMGC_Disease_ID      0\n",
      "BMGC_Disease_Name    0\n",
      "AUC                  0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Cell_Line_Name</th>\n",
       "      <th>BMGC_Drug_ID</th>\n",
       "      <th>BMGC_Drug_Name</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>BMGC_Disease_Name</th>\n",
       "      <th>AUC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000327</td>\n",
       "      <td>NCI-H1395</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS02944</td>\n",
       "      <td>lung adenocarcinoma</td>\n",
       "      <td>0.981350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000644</td>\n",
       "      <td>COLO 829</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS02930</td>\n",
       "      <td>cutaneous melanoma</td>\n",
       "      <td>0.954785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000137</td>\n",
       "      <td>8-MG-BA</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS00965</td>\n",
       "      <td>glioblastoma</td>\n",
       "      <td>0.980213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000733</td>\n",
       "      <td>NCI-H1838</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS02944</td>\n",
       "      <td>lung adenocarcinoma</td>\n",
       "      <td>0.931183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000530</td>\n",
       "      <td>DMS 114</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS21085</td>\n",
       "      <td>small cell lung carcinoma</td>\n",
       "      <td>0.988258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51507</th>\n",
       "      <td>ACH-000818</td>\n",
       "      <td>BT-483</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS07814</td>\n",
       "      <td>breast adenocarcinoma</td>\n",
       "      <td>0.812820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51508</th>\n",
       "      <td>ACH-000196</td>\n",
       "      <td>HCC1599</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS08035</td>\n",
       "      <td>breast ductal carcinoma</td>\n",
       "      <td>0.811327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51509</th>\n",
       "      <td>ACH-000902</td>\n",
       "      <td>CAL-148</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS07015</td>\n",
       "      <td>breast carcinoma</td>\n",
       "      <td>0.505332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51510</th>\n",
       "      <td>ACH-000930</td>\n",
       "      <td>HCC1569</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS08035</td>\n",
       "      <td>breast ductal carcinoma</td>\n",
       "      <td>0.843211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51511</th>\n",
       "      <td>ACH-000536</td>\n",
       "      <td>BT-20</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS07814</td>\n",
       "      <td>breast adenocarcinoma</td>\n",
       "      <td>0.952633</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>51512 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         depMapID Cell_Line_Name  BMGC_Drug_ID BMGC_Drug_Name BMGC_Disease_ID  \\\n",
       "0      ACH-000327      NCI-H1395  BMGC_DG04654      ERLOTINIB    BMGC_DS02944   \n",
       "1      ACH-000644       COLO 829  BMGC_DG04654      ERLOTINIB    BMGC_DS02930   \n",
       "2      ACH-000137        8-MG-BA  BMGC_DG04654      ERLOTINIB    BMGC_DS00965   \n",
       "3      ACH-000733      NCI-H1838  BMGC_DG04654      ERLOTINIB    BMGC_DS02944   \n",
       "4      ACH-000530        DMS 114  BMGC_DG04654      ERLOTINIB    BMGC_DS21085   \n",
       "...           ...            ...           ...            ...             ...   \n",
       "51507  ACH-000818         BT-483  BMGC_DG20688            JQ1    BMGC_DS07814   \n",
       "51508  ACH-000196        HCC1599  BMGC_DG20688            JQ1    BMGC_DS08035   \n",
       "51509  ACH-000902        CAL-148  BMGC_DG20688            JQ1    BMGC_DS07015   \n",
       "51510  ACH-000930        HCC1569  BMGC_DG20688            JQ1    BMGC_DS08035   \n",
       "51511  ACH-000536          BT-20  BMGC_DG20688            JQ1    BMGC_DS07814   \n",
       "\n",
       "               BMGC_Disease_Name       AUC  \n",
       "0            lung adenocarcinoma  0.981350  \n",
       "1             cutaneous melanoma  0.954785  \n",
       "2                   glioblastoma  0.980213  \n",
       "3            lung adenocarcinoma  0.931183  \n",
       "4      small cell lung carcinoma  0.988258  \n",
       "...                          ...       ...  \n",
       "51507      breast adenocarcinoma  0.812820  \n",
       "51508    breast ductal carcinoma  0.811327  \n",
       "51509           breast carcinoma  0.505332  \n",
       "51510    breast ductal carcinoma  0.843211  \n",
       "51511      breast adenocarcinoma  0.952633  \n",
       "\n",
       "[51512 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# fill in the missing samples in the final_merged_gene_df\n",
    "for sample in remaining_dti_samples:\n",
    "    if sample not in final_merged_gene_df.columns:\n",
    "        final_merged_gene_df[sample] = 0.0\n",
    "# fill in the missing samples in the final_merged_transcript_df\n",
    "for sample in remaining_dti_samples:\n",
    "    if sample not in final_merged_transcript_df.columns:\n",
    "        final_merged_transcript_df[sample] = 0.0\n",
    "# get the final gene, transcript, drug dataframe by filtering the remaining_dti_samples\n",
    "dti_gene_df = final_merged_gene_df[['BioMedGraphica_Conn_ID'] + remaining_dti_samples]\n",
    "display(dti_gene_df)\n",
    "dti_transcript_df = final_merged_transcript_df[['BioMedGraphica_Conn_ID'] + remaining_dti_samples]\n",
    "display(dti_transcript_df)\n",
    "dti_drug_overlap_df = final_drug_df[final_drug_df['ARXSPAN_ID'].isin(remaining_dti_samples)].reset_index(drop=True)\n",
    "dti_drug_overlap_df = pd.merge(dti_drug_overlap_df, dti_combined_df, left_on='ARXSPAN_ID', right_on='depMapID', how='left')\n",
    "dti_drug_overlap_df = pd.merge(dti_drug_overlap_df, final_merged_drug_df, on='DRUG_NAME', how='left').rename(columns={'BMGC_Disease_name': 'BMGC_Disease_Name', 'DRUG_NAME': 'BMGC_Drug_Name', 'BioMedGraphica_Conn_ID': 'BMGC_Drug_ID', 'Name': 'Cell_Line_Name', 'AUC_PUBLISHED': 'AUC'})\n",
    "# only keep columns ['depMapID', 'Cell_Line_Name', 'BMGC_Drug_ID', 'BMGC_Drug_Name', 'BMGC_Disease_ID', 'BMGC_Disease_Name', 'AUC']\n",
    "dti_drug_overlap_df = dti_drug_overlap_df[['depMapID', 'Cell_Line_Name', 'BMGC_Drug_ID', 'BMGC_Drug_Name', 'BMGC_Disease_ID', 'BMGC_Disease_Name', 'AUC']]\n",
    "# check if there is null values in the dti_drug_overlap_df\n",
    "print(dti_drug_overlap_df.isnull().sum())\n",
    "display(dti_drug_overlap_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 391,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.83335</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88351</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.94738</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.50982</td>\n",
       "      <td>0.56520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93319</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.04959</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00129</td>\n",
       "      <td>0.02137</td>\n",
       "      <td>0.01066</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01239</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00958</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.05386</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.33075</td>\n",
       "      <td>0.07825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.72705</td>\n",
       "      <td>0.00761</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.92687</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705929</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705930</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705931</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705932</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705933</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>705934 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0                BMGC_PM00001     0.83335        0.00     0.88351     1.00000   \n",
       "1                BMGC_PM00002     0.00000        0.00     0.00000     0.00000   \n",
       "2                BMGC_PM00003     0.04959        0.00     0.00129     0.02137   \n",
       "3                BMGC_PM00004     0.05386        0.25     0.00000     0.33075   \n",
       "4                BMGC_PM00005     0.00000        0.00     0.00000     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "705929          BMGC_PT121415     0.00000        0.00     0.00000     0.00000   \n",
       "705930          BMGC_PT121416     0.00000        0.00     0.00000     0.00000   \n",
       "705931          BMGC_PT121417     0.00000        0.00     0.00000     0.00000   \n",
       "705932          BMGC_PT121418     0.00000        0.00     0.00000     0.00000   \n",
       "705933          BMGC_PT121419     0.00000        0.00     0.00000     0.00000   \n",
       "\n",
       "        ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0          0.94738         1.0     0.50982     0.56520         0.0  ...   \n",
       "1          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "2          0.01066         0.0     0.01239     0.00000         0.0  ...   \n",
       "3          0.07825         0.0     0.72705     0.00761         0.0  ...   \n",
       "4          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "705929     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705930     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705931     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705932     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705933     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "        ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  \\\n",
       "0          0.93319         0.0         0.0         0.0         0.0   \n",
       "1          0.00000         0.0         0.0         0.0         0.0   \n",
       "2          0.00958         0.0         0.0         0.0         0.0   \n",
       "3          0.92687         0.0         0.0         0.0         0.0   \n",
       "4          0.00000         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "705929     0.00000         0.0         0.0         0.0         0.0   \n",
       "705930     0.00000         0.0         0.0         0.0         0.0   \n",
       "705931     0.00000         0.0         0.0         0.0         0.0   \n",
       "705932     0.00000         0.0         0.0         0.0         0.0   \n",
       "705933     0.00000         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001151  ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0              0.0         0.0         0.0         0.0         0.0  \n",
       "1              0.0         0.0         0.0         0.0         0.0  \n",
       "2              0.0         0.0         0.0         0.0         0.0  \n",
       "3              0.0         0.0         0.0         0.0         0.0  \n",
       "4              0.0         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "705929         0.0         0.0         0.0         0.0         0.0  \n",
       "705930         0.0         0.0         0.0         0.0         0.0  \n",
       "705931         0.0         0.0         0.0         0.0         0.0  \n",
       "705932         0.0         0.0         0.0         0.0         0.0  \n",
       "705933         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[705934 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.83335</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88351</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.94738</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.50982</td>\n",
       "      <td>0.56520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93319</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.04959</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00129</td>\n",
       "      <td>0.02137</td>\n",
       "      <td>0.01066</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01239</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00958</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.05386</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.33075</td>\n",
       "      <td>0.07825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.72705</td>\n",
       "      <td>0.00761</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.92687</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0                BMGC_PM00001     0.83335        0.00     0.88351     1.00000   \n",
       "1                BMGC_PM00002     0.00000        0.00     0.00000     0.00000   \n",
       "2                BMGC_PM00003     0.04959        0.00     0.00129     0.02137   \n",
       "3                BMGC_PM00004     0.05386        0.25     0.00000     0.33075   \n",
       "4                BMGC_PM00005     0.00000        0.00     0.00000     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "834804           BMGC_DG20914     0.00000        0.00     0.00000     0.00000   \n",
       "834805           BMGC_DG20915     0.00000        0.00     0.00000     0.00000   \n",
       "834806           BMGC_DG20916     0.00000        0.00     0.00000     0.00000   \n",
       "834807           BMGC_DG20917     0.00000        0.00     0.00000     0.00000   \n",
       "834808           BMGC_DG20918     0.00000        0.00     0.00000     0.00000   \n",
       "\n",
       "        ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0          0.94738         1.0     0.50982     0.56520         0.0  ...   \n",
       "1          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "2          0.01066         0.0     0.01239     0.00000         0.0  ...   \n",
       "3          0.07825         0.0     0.72705     0.00761         0.0  ...   \n",
       "4          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "834804     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834805     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834806     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834807     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834808     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "        ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  \\\n",
       "0          0.93319         0.0         0.0         0.0         0.0   \n",
       "1          0.00000         0.0         0.0         0.0         0.0   \n",
       "2          0.00958         0.0         0.0         0.0         0.0   \n",
       "3          0.92687         0.0         0.0         0.0         0.0   \n",
       "4          0.00000         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "834804     0.00000         0.0         0.0         0.0         0.0   \n",
       "834805     0.00000         0.0         0.0         0.0         0.0   \n",
       "834806     0.00000         0.0         0.0         0.0         0.0   \n",
       "834807     0.00000         0.0         0.0         0.0         0.0   \n",
       "834808     0.00000         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001151  ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0              0.0         0.0         0.0         0.0         0.0  \n",
       "1              0.0         0.0         0.0         0.0         0.0  \n",
       "2              0.0         0.0         0.0         0.0         0.0  \n",
       "3              0.0         0.0         0.0         0.0         0.0  \n",
       "4              0.0         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "834804         0.0         0.0         0.0         0.0         0.0  \n",
       "834805         0.0         0.0         0.0         0.0         0.0  \n",
       "834806         0.0         0.0         0.0         0.0         0.0  \n",
       "834807         0.0         0.0         0.0         0.0         0.0  \n",
       "834808         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[834809 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of dti_array: (286, 834809)\n"
     ]
    }
   ],
   "source": [
    "dti_omics_df = pd.concat([dti_methy_df, dti_gene_df, dti_transcript_df, dti_protein_df], axis=0).reset_index(drop=True)\n",
    "display(dti_omics_df)\n",
    "dti_feat_df = pd.merge(bmgc_entity_df, dti_omics_df, left_on='BioMedGraphica_Conn_ID', right_on='BioMedGraphica_Conn_ID', how='left').drop(columns=['BioMedGraphica_ID', 'Type'])\n",
    "dti_feat_df = dti_feat_df.fillna(0.0)\n",
    "display(dti_feat_df)\n",
    "\n",
    "# convert dti_feat_df to numpy array and transpose it\n",
    "dti_array = dti_feat_df.drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "print(\"Shape of dti_array:\", dti_array.shape)\n",
    "# Save the numpy array to .npy file\n",
    "np.save('./data/DrugQA/dti_feature.npy', dti_array)\n",
    "np.save('./data/DrugScreen/dti_feature.npy', dti_array)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 428,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Cell_Line_Name</th>\n",
       "      <th>BMGC_Drug_ID</th>\n",
       "      <th>BMGC_Drug_Name</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>BMGC_Disease_Name</th>\n",
       "      <th>AUC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000327</td>\n",
       "      <td>NCI-H1395</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS02944</td>\n",
       "      <td>lung adenocarcinoma</td>\n",
       "      <td>0.981350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000137</td>\n",
       "      <td>8-MG-BA</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS00965</td>\n",
       "      <td>glioblastoma</td>\n",
       "      <td>0.980213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000733</td>\n",
       "      <td>NCI-H1838</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS02944</td>\n",
       "      <td>lung adenocarcinoma</td>\n",
       "      <td>0.931183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000530</td>\n",
       "      <td>DMS 114</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS21085</td>\n",
       "      <td>small cell lung carcinoma</td>\n",
       "      <td>0.988258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000514</td>\n",
       "      <td>NCI-H1092</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS02887</td>\n",
       "      <td>lung small cell carcinoma</td>\n",
       "      <td>0.983326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42061</th>\n",
       "      <td>ACH-000643</td>\n",
       "      <td>HDQ-P1</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS08035</td>\n",
       "      <td>breast ductal carcinoma</td>\n",
       "      <td>0.873306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42062</th>\n",
       "      <td>ACH-000117</td>\n",
       "      <td>EFM-192A</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS07015</td>\n",
       "      <td>breast carcinoma</td>\n",
       "      <td>0.899328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42063</th>\n",
       "      <td>ACH-000818</td>\n",
       "      <td>BT-483</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS07814</td>\n",
       "      <td>breast adenocarcinoma</td>\n",
       "      <td>0.812820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42064</th>\n",
       "      <td>ACH-000930</td>\n",
       "      <td>HCC1569</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS08035</td>\n",
       "      <td>breast ductal carcinoma</td>\n",
       "      <td>0.843211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42065</th>\n",
       "      <td>ACH-000536</td>\n",
       "      <td>BT-20</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS07814</td>\n",
       "      <td>breast adenocarcinoma</td>\n",
       "      <td>0.952633</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42066 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         depMapID Cell_Line_Name  BMGC_Drug_ID BMGC_Drug_Name BMGC_Disease_ID  \\\n",
       "0      ACH-000327      NCI-H1395  BMGC_DG04654      ERLOTINIB    BMGC_DS02944   \n",
       "1      ACH-000137        8-MG-BA  BMGC_DG04654      ERLOTINIB    BMGC_DS00965   \n",
       "2      ACH-000733      NCI-H1838  BMGC_DG04654      ERLOTINIB    BMGC_DS02944   \n",
       "3      ACH-000530        DMS 114  BMGC_DG04654      ERLOTINIB    BMGC_DS21085   \n",
       "4      ACH-000514      NCI-H1092  BMGC_DG04654      ERLOTINIB    BMGC_DS02887   \n",
       "...           ...            ...           ...            ...             ...   \n",
       "42061  ACH-000643         HDQ-P1  BMGC_DG20688            JQ1    BMGC_DS08035   \n",
       "42062  ACH-000117       EFM-192A  BMGC_DG20688            JQ1    BMGC_DS07015   \n",
       "42063  ACH-000818         BT-483  BMGC_DG20688            JQ1    BMGC_DS07814   \n",
       "42064  ACH-000930        HCC1569  BMGC_DG20688            JQ1    BMGC_DS08035   \n",
       "42065  ACH-000536          BT-20  BMGC_DG20688            JQ1    BMGC_DS07814   \n",
       "\n",
       "               BMGC_Disease_Name       AUC  \n",
       "0            lung adenocarcinoma  0.981350  \n",
       "1                   glioblastoma  0.980213  \n",
       "2            lung adenocarcinoma  0.931183  \n",
       "3      small cell lung carcinoma  0.988258  \n",
       "4      lung small cell carcinoma  0.983326  \n",
       "...                          ...       ...  \n",
       "42061    breast ductal carcinoma  0.873306  \n",
       "42062           breast carcinoma  0.899328  \n",
       "42063      breast adenocarcinoma  0.812820  \n",
       "42064    breast ductal carcinoma  0.843211  \n",
       "42065      breast adenocarcinoma  0.952633  \n",
       "\n",
       "[42066 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(training_drug_ids): 172\n",
      "len(training_cell_line_ids): 234\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Cell_Line_Name</th>\n",
       "      <th>BMGC_Drug_ID</th>\n",
       "      <th>BMGC_Drug_Name</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>BMGC_Disease_Name</th>\n",
       "      <th>AUC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000644</td>\n",
       "      <td>COLO 829</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS02930</td>\n",
       "      <td>cutaneous melanoma</td>\n",
       "      <td>0.954785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000729</td>\n",
       "      <td>NCI-H1963</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS02887</td>\n",
       "      <td>lung small cell carcinoma</td>\n",
       "      <td>0.988450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000196</td>\n",
       "      <td>HCC1599</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS08035</td>\n",
       "      <td>breast ductal carcinoma</td>\n",
       "      <td>0.978046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000703</td>\n",
       "      <td>DMS 79</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS02887</td>\n",
       "      <td>lung small cell carcinoma</td>\n",
       "      <td>0.943754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000353</td>\n",
       "      <td>TE-15</td>\n",
       "      <td>BMGC_DG04654</td>\n",
       "      <td>ERLOTINIB</td>\n",
       "      <td>BMGC_DS05334</td>\n",
       "      <td>esophageal squamous cell carcinoma</td>\n",
       "      <td>0.877464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9441</th>\n",
       "      <td>ACH-000360</td>\n",
       "      <td>NCI-H508</td>\n",
       "      <td>BMGC_DG20225</td>\n",
       "      <td>BMS-754807</td>\n",
       "      <td>BMGC_DS08517</td>\n",
       "      <td>cecum adenocarcinoma</td>\n",
       "      <td>0.439161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9442</th>\n",
       "      <td>ACH-000489</td>\n",
       "      <td>SW1116</td>\n",
       "      <td>BMGC_DG20225</td>\n",
       "      <td>BMS-754807</td>\n",
       "      <td>BMGC_DS05624</td>\n",
       "      <td>colon adenocarcinoma</td>\n",
       "      <td>0.813459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9443</th>\n",
       "      <td>ACH-000236</td>\n",
       "      <td>SW1417</td>\n",
       "      <td>BMGC_DG20225</td>\n",
       "      <td>BMS-754807</td>\n",
       "      <td>BMGC_DS05624</td>\n",
       "      <td>colon adenocarcinoma</td>\n",
       "      <td>0.785710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9444</th>\n",
       "      <td>ACH-000196</td>\n",
       "      <td>HCC1599</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS08035</td>\n",
       "      <td>breast ductal carcinoma</td>\n",
       "      <td>0.811327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9445</th>\n",
       "      <td>ACH-000902</td>\n",
       "      <td>CAL-148</td>\n",
       "      <td>BMGC_DG20688</td>\n",
       "      <td>JQ1</td>\n",
       "      <td>BMGC_DS07015</td>\n",
       "      <td>breast carcinoma</td>\n",
       "      <td>0.505332</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9446 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        depMapID Cell_Line_Name  BMGC_Drug_ID BMGC_Drug_Name BMGC_Disease_ID  \\\n",
       "0     ACH-000644       COLO 829  BMGC_DG04654      ERLOTINIB    BMGC_DS02930   \n",
       "1     ACH-000729      NCI-H1963  BMGC_DG04654      ERLOTINIB    BMGC_DS02887   \n",
       "2     ACH-000196        HCC1599  BMGC_DG04654      ERLOTINIB    BMGC_DS08035   \n",
       "3     ACH-000703         DMS 79  BMGC_DG04654      ERLOTINIB    BMGC_DS02887   \n",
       "4     ACH-000353          TE-15  BMGC_DG04654      ERLOTINIB    BMGC_DS05334   \n",
       "...          ...            ...           ...            ...             ...   \n",
       "9441  ACH-000360       NCI-H508  BMGC_DG20225     BMS-754807    BMGC_DS08517   \n",
       "9442  ACH-000489         SW1116  BMGC_DG20225     BMS-754807    BMGC_DS05624   \n",
       "9443  ACH-000236         SW1417  BMGC_DG20225     BMS-754807    BMGC_DS05624   \n",
       "9444  ACH-000196        HCC1599  BMGC_DG20688            JQ1    BMGC_DS08035   \n",
       "9445  ACH-000902        CAL-148  BMGC_DG20688            JQ1    BMGC_DS07015   \n",
       "\n",
       "                       BMGC_Disease_Name       AUC  \n",
       "0                     cutaneous melanoma  0.954785  \n",
       "1              lung small cell carcinoma  0.988450  \n",
       "2                breast ductal carcinoma  0.978046  \n",
       "3              lung small cell carcinoma  0.943754  \n",
       "4     esophageal squamous cell carcinoma  0.877464  \n",
       "...                                  ...       ...  \n",
       "9441                cecum adenocarcinoma  0.439161  \n",
       "9442                colon adenocarcinoma  0.813459  \n",
       "9443                colon adenocarcinoma  0.785710  \n",
       "9444             breast ductal carcinoma  0.811327  \n",
       "9445                    breast carcinoma  0.505332  \n",
       "\n",
       "[9446 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(test_drug_ids): 172\n",
      "len(test_cell_line_ids): 52\n"
     ]
    }
   ],
   "source": [
    "# filter out the drug screen by training and test with condition that meet both cell line test samples and drug test samples for test\n",
    "final_training_drug_screen_df = dti_drug_overlap_df[dti_drug_overlap_df['depMapID'].isin(dti_train_samples)].reset_index(drop=True)\n",
    "display(final_training_drug_screen_df)\n",
    "# and calculate the final training drug ids and cell line ids\n",
    "training_drug_ids = list(set(final_training_drug_screen_df['BMGC_Drug_ID']))\n",
    "training_cell_line_ids = list(set(final_training_drug_screen_df['depMapID']))\n",
    "print(\"len(training_drug_ids):\", len(training_drug_ids))\n",
    "print(\"len(training_cell_line_ids):\", len(training_cell_line_ids))\n",
    "\n",
    "# the rest of the drug screen is the test drug screen\n",
    "final_test_drug_screen_df = dti_drug_overlap_df[dti_drug_overlap_df['depMapID'].isin(dti_test_samples)].reset_index(drop=True)\n",
    "display(final_test_drug_screen_df)\n",
    "# and calculate the final test drug ids and cell line ids\n",
    "test_drug_ids = list(set(final_test_drug_screen_df['BMGC_Drug_ID']))\n",
    "test_cell_line_ids = list(set(final_test_drug_screen_df['depMapID']))\n",
    "print(\"len(test_drug_ids):\", len(test_drug_ids))\n",
    "print(\"len(test_cell_line_ids):\", len(test_cell_line_ids))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The overlapped samples dataframe for methylation between remaining_dti_samples:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.83335</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88351</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.94738</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.50982</td>\n",
       "      <td>0.56520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93319</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.04959</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00129</td>\n",
       "      <td>0.02137</td>\n",
       "      <td>0.01066</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01239</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00958</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.05386</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.33075</td>\n",
       "      <td>0.07825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.72705</td>\n",
       "      <td>0.00761</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.92687</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_PM86234</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_PM86235</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_PM86236</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_PM86237</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_PM86238</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               BMGC_PM00001     0.83335        0.00     0.88351     1.00000   \n",
       "1               BMGC_PM00002     0.00000        0.00     0.00000     0.00000   \n",
       "2               BMGC_PM00003     0.04959        0.00     0.00129     0.02137   \n",
       "3               BMGC_PM00004     0.05386        0.25     0.00000     0.33075   \n",
       "4               BMGC_PM00005     0.00000        0.00     0.00000     0.00000   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_PM86234     0.00000        0.00     0.00000     0.00000   \n",
       "86234           BMGC_PM86235     0.00000        0.00     0.00000     0.00000   \n",
       "86235           BMGC_PM86236     0.00000        0.00     0.00000     0.00000   \n",
       "86236           BMGC_PM86237     0.00000        0.00     0.00000     0.00000   \n",
       "86237           BMGC_PM86238     0.00000        0.00     0.00000     0.00000   \n",
       "\n",
       "       ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0         0.94738         1.0     0.50982     0.56520         0.0  ...   \n",
       "1         0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "2         0.01066         0.0     0.01239     0.00000         0.0  ...   \n",
       "3         0.07825         0.0     0.72705     0.00761         0.0  ...   \n",
       "4         0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86234     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86235     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86236     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "86237     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "       ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  ACH-001151  \\\n",
       "0         0.93319         0.0         0.0         0.0         0.0         0.0   \n",
       "1         0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "2         0.00958         0.0         0.0         0.0         0.0         0.0   \n",
       "3         0.92687         0.0         0.0         0.0         0.0         0.0   \n",
       "4         0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "86234     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "86235     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "86236     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "86237     0.00000         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "       ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0             0.0         0.0         0.0         0.0  \n",
       "1             0.0         0.0         0.0         0.0  \n",
       "2             0.0         0.0         0.0         0.0  \n",
       "3             0.0         0.0         0.0         0.0  \n",
       "4             0.0         0.0         0.0         0.0  \n",
       "...           ...         ...         ...         ...  \n",
       "86233         0.0         0.0         0.0         0.0  \n",
       "86234         0.0         0.0         0.0         0.0  \n",
       "86235         0.0         0.0         0.0         0.0  \n",
       "86236         0.0         0.0         0.0         0.0  \n",
       "86237         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[86238 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The overlapped samples dataframe for protein between remaining_dti_samples are:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               BMGC_PT000001         0.0         0.0         0.0         0.0   \n",
       "1               BMGC_PT000002         0.0         0.0         0.0         0.0   \n",
       "2               BMGC_PT000003         0.0         0.0         0.0         0.0   \n",
       "3               BMGC_PT000004         0.0         0.0         0.0         0.0   \n",
       "4               BMGC_PT000005         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "121414          BMGC_PT121415         0.0         0.0         0.0         0.0   \n",
       "121415          BMGC_PT121416         0.0         0.0         0.0         0.0   \n",
       "121416          BMGC_PT121417         0.0         0.0         0.0         0.0   \n",
       "121417          BMGC_PT121418         0.0         0.0         0.0         0.0   \n",
       "121418          BMGC_PT121419         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "1              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "2              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "3              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "4              0.0         0.0         0.0         0.0         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121415         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121416         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121417         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "121418         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "\n",
       "        ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  \\\n",
       "0              0.0         0.0         0.0         0.0         0.0   \n",
       "1              0.0         0.0         0.0         0.0         0.0   \n",
       "2              0.0         0.0         0.0         0.0         0.0   \n",
       "3              0.0         0.0         0.0         0.0         0.0   \n",
       "4              0.0         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "121414         0.0         0.0         0.0         0.0         0.0   \n",
       "121415         0.0         0.0         0.0         0.0         0.0   \n",
       "121416         0.0         0.0         0.0         0.0         0.0   \n",
       "121417         0.0         0.0         0.0         0.0         0.0   \n",
       "121418         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001151  ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0              0.0         0.0         0.0         0.0         0.0  \n",
       "1              0.0         0.0         0.0         0.0         0.0  \n",
       "2              0.0         0.0         0.0         0.0         0.0  \n",
       "3              0.0         0.0         0.0         0.0         0.0  \n",
       "4              0.0         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "121414         0.0         0.0         0.0         0.0         0.0  \n",
       "121415         0.0         0.0         0.0         0.0         0.0  \n",
       "121416         0.0         0.0         0.0         0.0         0.0  \n",
       "121417         0.0         0.0         0.0         0.0         0.0  \n",
       "121418         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[121419 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(\"The overlapped samples dataframe for methylation between remaining_dti_samples:\")\n",
    "# insert the samples that is in the overlapped_samples but not in the maped_methy_df\n",
    "for sample in remaining_dti_samples:\n",
    "    if sample not in maped_methy_df.columns:\n",
    "        maped_methy_df[sample] = 0.0\n",
    "# filter out the samples that is not in the remaining_dti_samples\n",
    "dti_methy_df = maped_methy_df[['BioMedGraphica_Conn_ID'] + remaining_dti_samples]\n",
    "display(dti_methy_df)\n",
    "\n",
    "print(\"The overlapped samples dataframe for protein between remaining_dti_samples are:\")\n",
    "# insert the samples that is in the pretraining_samples but not in the mapped_protein_df\n",
    "for sample in remaining_dti_samples:\n",
    "    if sample not in mapped_protein_df.columns:\n",
    "        mapped_protein_df[sample] = 0.0\n",
    "# filter out the samples that is not in the remaining_dti_samples\n",
    "dti_protein_df = mapped_protein_df[['BioMedGraphica_Conn_ID'] + remaining_dti_samples]\n",
    "display(dti_protein_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 397,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_GN00001</td>\n",
       "      <td>1.039342</td>\n",
       "      <td>1.081979</td>\n",
       "      <td>1.069887</td>\n",
       "      <td>1.021275</td>\n",
       "      <td>1.019950</td>\n",
       "      <td>0.873542</td>\n",
       "      <td>1.954398</td>\n",
       "      <td>0.476005</td>\n",
       "      <td>0.498743</td>\n",
       "      <td>...</td>\n",
       "      <td>0.930320</td>\n",
       "      <td>1.280913</td>\n",
       "      <td>0.816260</td>\n",
       "      <td>0.996102</td>\n",
       "      <td>1.018304</td>\n",
       "      <td>1.050028</td>\n",
       "      <td>1.005951</td>\n",
       "      <td>1.040949</td>\n",
       "      <td>1.409362</td>\n",
       "      <td>0.863441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_GN00002</td>\n",
       "      <td>0.915792</td>\n",
       "      <td>1.032808</td>\n",
       "      <td>0.843013</td>\n",
       "      <td>0.959549</td>\n",
       "      <td>1.009547</td>\n",
       "      <td>0.971925</td>\n",
       "      <td>1.188910</td>\n",
       "      <td>1.511536</td>\n",
       "      <td>1.047491</td>\n",
       "      <td>...</td>\n",
       "      <td>1.009008</td>\n",
       "      <td>1.076599</td>\n",
       "      <td>0.906371</td>\n",
       "      <td>1.057301</td>\n",
       "      <td>0.828024</td>\n",
       "      <td>0.445954</td>\n",
       "      <td>1.003790</td>\n",
       "      <td>1.021203</td>\n",
       "      <td>0.519903</td>\n",
       "      <td>1.112888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_GN00003</td>\n",
       "      <td>0.978356</td>\n",
       "      <td>0.909618</td>\n",
       "      <td>1.676606</td>\n",
       "      <td>0.990115</td>\n",
       "      <td>0.993623</td>\n",
       "      <td>0.953061</td>\n",
       "      <td>1.229717</td>\n",
       "      <td>1.566409</td>\n",
       "      <td>1.034708</td>\n",
       "      <td>...</td>\n",
       "      <td>1.075723</td>\n",
       "      <td>1.279596</td>\n",
       "      <td>1.212607</td>\n",
       "      <td>1.399931</td>\n",
       "      <td>0.986576</td>\n",
       "      <td>1.268384</td>\n",
       "      <td>0.997014</td>\n",
       "      <td>0.996969</td>\n",
       "      <td>0.995337</td>\n",
       "      <td>0.380086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_GN00004</td>\n",
       "      <td>1.514518</td>\n",
       "      <td>1.024754</td>\n",
       "      <td>0.868152</td>\n",
       "      <td>1.921635</td>\n",
       "      <td>1.010449</td>\n",
       "      <td>0.986412</td>\n",
       "      <td>1.231253</td>\n",
       "      <td>1.062672</td>\n",
       "      <td>1.010414</td>\n",
       "      <td>...</td>\n",
       "      <td>1.063356</td>\n",
       "      <td>0.750272</td>\n",
       "      <td>0.880448</td>\n",
       "      <td>0.750693</td>\n",
       "      <td>1.368224</td>\n",
       "      <td>0.465270</td>\n",
       "      <td>0.989083</td>\n",
       "      <td>0.969761</td>\n",
       "      <td>1.023638</td>\n",
       "      <td>0.410329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_GN00005</td>\n",
       "      <td>1.032720</td>\n",
       "      <td>1.165655</td>\n",
       "      <td>1.251083</td>\n",
       "      <td>0.987747</td>\n",
       "      <td>1.006196</td>\n",
       "      <td>1.438704</td>\n",
       "      <td>1.172893</td>\n",
       "      <td>1.024139</td>\n",
       "      <td>1.042507</td>\n",
       "      <td>...</td>\n",
       "      <td>1.019288</td>\n",
       "      <td>1.084336</td>\n",
       "      <td>0.968458</td>\n",
       "      <td>0.988188</td>\n",
       "      <td>1.071999</td>\n",
       "      <td>0.924170</td>\n",
       "      <td>0.999373</td>\n",
       "      <td>1.046953</td>\n",
       "      <td>1.001543</td>\n",
       "      <td>1.245752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86233</th>\n",
       "      <td>BMGC_GN86234</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86234</th>\n",
       "      <td>BMGC_GN86235</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86235</th>\n",
       "      <td>BMGC_GN86236</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86236</th>\n",
       "      <td>BMGC_GN86237</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86237</th>\n",
       "      <td>BMGC_GN86238</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86238 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               BMGC_GN00001    1.039342    1.081979    1.069887    1.021275   \n",
       "1               BMGC_GN00002    0.915792    1.032808    0.843013    0.959549   \n",
       "2               BMGC_GN00003    0.978356    0.909618    1.676606    0.990115   \n",
       "3               BMGC_GN00004    1.514518    1.024754    0.868152    1.921635   \n",
       "4               BMGC_GN00005    1.032720    1.165655    1.251083    0.987747   \n",
       "...                      ...         ...         ...         ...         ...   \n",
       "86233           BMGC_GN86234    0.000000    0.000000    0.000000    0.000000   \n",
       "86234           BMGC_GN86235    0.000000    0.000000    0.000000    0.000000   \n",
       "86235           BMGC_GN86236    0.000000    0.000000    0.000000    0.000000   \n",
       "86236           BMGC_GN86237    0.000000    0.000000    0.000000    0.000000   \n",
       "86237           BMGC_GN86238    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0        1.019950    0.873542    1.954398    0.476005    0.498743  ...   \n",
       "1        1.009547    0.971925    1.188910    1.511536    1.047491  ...   \n",
       "2        0.993623    0.953061    1.229717    1.566409    1.034708  ...   \n",
       "3        1.010449    0.986412    1.231253    1.062672    1.010414  ...   \n",
       "4        1.006196    1.438704    1.172893    1.024139    1.042507  ...   \n",
       "...           ...         ...         ...         ...         ...  ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "       ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  ACH-001151  \\\n",
       "0        0.930320    1.280913    0.816260    0.996102    1.018304    1.050028   \n",
       "1        1.009008    1.076599    0.906371    1.057301    0.828024    0.445954   \n",
       "2        1.075723    1.279596    1.212607    1.399931    0.986576    1.268384   \n",
       "3        1.063356    0.750272    0.880448    0.750693    1.368224    0.465270   \n",
       "4        1.019288    1.084336    0.968458    0.988188    1.071999    0.924170   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "86233    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86234    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86235    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86236    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "86237    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "       ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0        1.005951    1.040949    1.409362    0.863441  \n",
       "1        1.003790    1.021203    0.519903    1.112888  \n",
       "2        0.997014    0.996969    0.995337    0.380086  \n",
       "3        0.989083    0.969761    1.023638    0.410329  \n",
       "4        0.999373    1.046953    1.001543    1.245752  \n",
       "...           ...         ...         ...         ...  \n",
       "86233    0.000000    0.000000    0.000000    0.000000  \n",
       "86234    0.000000    0.000000    0.000000    0.000000  \n",
       "86235    0.000000    0.000000    0.000000    0.000000  \n",
       "86236    0.000000    0.000000    0.000000    0.000000  \n",
       "86237    0.000000    0.000000    0.000000    0.000000  \n",
       "\n",
       "[86238 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_TS000001</td>\n",
       "      <td>6.522291</td>\n",
       "      <td>5.854020</td>\n",
       "      <td>7.104063</td>\n",
       "      <td>6.875601</td>\n",
       "      <td>6.986633</td>\n",
       "      <td>7.254845</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.293006</td>\n",
       "      <td>6.912981</td>\n",
       "      <td>...</td>\n",
       "      <td>7.422634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.220747</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_TS000002</td>\n",
       "      <td>6.729154</td>\n",
       "      <td>6.338599</td>\n",
       "      <td>7.897515</td>\n",
       "      <td>6.917430</td>\n",
       "      <td>7.218023</td>\n",
       "      <td>7.815267</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.232725</td>\n",
       "      <td>6.944961</td>\n",
       "      <td>...</td>\n",
       "      <td>6.139888</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.795080</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_TS000003</td>\n",
       "      <td>4.419503</td>\n",
       "      <td>4.986058</td>\n",
       "      <td>4.621525</td>\n",
       "      <td>3.843395</td>\n",
       "      <td>4.675843</td>\n",
       "      <td>4.200734</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.066766</td>\n",
       "      <td>4.064158</td>\n",
       "      <td>...</td>\n",
       "      <td>4.260150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.455935</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_TS000004</td>\n",
       "      <td>5.213352</td>\n",
       "      <td>5.151831</td>\n",
       "      <td>5.766890</td>\n",
       "      <td>5.052784</td>\n",
       "      <td>5.001630</td>\n",
       "      <td>6.215894</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.329526</td>\n",
       "      <td>5.736236</td>\n",
       "      <td>...</td>\n",
       "      <td>5.175159</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.969341</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_TS000005</td>\n",
       "      <td>0.043941</td>\n",
       "      <td>-0.073548</td>\n",
       "      <td>0.335595</td>\n",
       "      <td>0.005892</td>\n",
       "      <td>-0.059982</td>\n",
       "      <td>1.853420</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.031376</td>\n",
       "      <td>-0.059982</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.046550</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.881209</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412034</th>\n",
       "      <td>BMGC_TS412035</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412035</th>\n",
       "      <td>BMGC_TS412036</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412036</th>\n",
       "      <td>BMGC_TS412037</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412037</th>\n",
       "      <td>BMGC_TS412038</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412038</th>\n",
       "      <td>BMGC_TS412039</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>412039 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               BMGC_TS000001    6.522291    5.854020    7.104063    6.875601   \n",
       "1               BMGC_TS000002    6.729154    6.338599    7.897515    6.917430   \n",
       "2               BMGC_TS000003    4.419503    4.986058    4.621525    3.843395   \n",
       "3               BMGC_TS000004    5.213352    5.151831    5.766890    5.052784   \n",
       "4               BMGC_TS000005    0.043941   -0.073548    0.335595    0.005892   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "412034          BMGC_TS412035    0.000000    0.000000    0.000000    0.000000   \n",
       "412035          BMGC_TS412036    0.000000    0.000000    0.000000    0.000000   \n",
       "412036          BMGC_TS412037    0.000000    0.000000    0.000000    0.000000   \n",
       "412037          BMGC_TS412038    0.000000    0.000000    0.000000    0.000000   \n",
       "412038          BMGC_TS412039    0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "        ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0         6.986633    7.254845         0.0    6.293006    6.912981  ...   \n",
       "1         7.218023    7.815267         0.0    5.232725    6.944961  ...   \n",
       "2         4.675843    4.200734         0.0    5.066766    4.064158  ...   \n",
       "3         5.001630    6.215894         0.0    5.329526    5.736236  ...   \n",
       "4        -0.059982    1.853420         0.0    0.031376   -0.059982  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "412034    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "412035    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "412036    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "412037    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "412038    0.000000    0.000000         0.0    0.000000    0.000000  ...   \n",
       "\n",
       "        ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  \\\n",
       "0         7.422634         0.0         0.0         0.0         0.0   \n",
       "1         6.139888         0.0         0.0         0.0         0.0   \n",
       "2         4.260150         0.0         0.0         0.0         0.0   \n",
       "3         5.175159         0.0         0.0         0.0         0.0   \n",
       "4        -0.046550         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "412034    0.000000         0.0         0.0         0.0         0.0   \n",
       "412035    0.000000         0.0         0.0         0.0         0.0   \n",
       "412036    0.000000         0.0         0.0         0.0         0.0   \n",
       "412037    0.000000         0.0         0.0         0.0         0.0   \n",
       "412038    0.000000         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001151  ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0         6.220747         0.0         0.0         0.0         0.0  \n",
       "1         6.795080         0.0         0.0         0.0         0.0  \n",
       "2         4.455935         0.0         0.0         0.0         0.0  \n",
       "3         5.969341         0.0         0.0         0.0         0.0  \n",
       "4         2.881209         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "412034    0.000000         0.0         0.0         0.0         0.0  \n",
       "412035    0.000000         0.0         0.0         0.0         0.0  \n",
       "412036    0.000000         0.0         0.0         0.0         0.0  \n",
       "412037    0.000000         0.0         0.0         0.0         0.0  \n",
       "412038    0.000000         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[412039 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# fill in the missing samples in the final_merged_gene_df for remaining_dti_samples\n",
    "for sample in remaining_dti_samples:\n",
    "    if sample not in final_merged_gene_df.columns:\n",
    "        final_merged_gene_df[sample] = 0.0\n",
    "# fill in the missing samples in the final_merged_transcript_df for remaining_dti_samples\n",
    "for sample in remaining_dti_samples:\n",
    "    if sample not in final_merged_transcript_df.columns:\n",
    "        final_merged_transcript_df[sample] = 0.0\n",
    "# get the final gene, transcript, drug dataframe by filtering the remaining_dti_samples\n",
    "dti_gene_df = final_merged_gene_df[['BioMedGraphica_Conn_ID'] + remaining_dti_samples]\n",
    "display(dti_gene_df)\n",
    "dti_transcript_df = final_merged_transcript_df[['BioMedGraphica_Conn_ID'] + remaining_dti_samples]\n",
    "display(dti_transcript_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.83335</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88351</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.94738</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.50982</td>\n",
       "      <td>0.56520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93319</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.04959</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00129</td>\n",
       "      <td>0.02137</td>\n",
       "      <td>0.01066</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01239</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00958</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.05386</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.33075</td>\n",
       "      <td>0.07825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.72705</td>\n",
       "      <td>0.00761</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.92687</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705929</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705930</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705931</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705932</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705933</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>705934 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0                BMGC_PM00001     0.83335        0.00     0.88351     1.00000   \n",
       "1                BMGC_PM00002     0.00000        0.00     0.00000     0.00000   \n",
       "2                BMGC_PM00003     0.04959        0.00     0.00129     0.02137   \n",
       "3                BMGC_PM00004     0.05386        0.25     0.00000     0.33075   \n",
       "4                BMGC_PM00005     0.00000        0.00     0.00000     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "705929          BMGC_PT121415     0.00000        0.00     0.00000     0.00000   \n",
       "705930          BMGC_PT121416     0.00000        0.00     0.00000     0.00000   \n",
       "705931          BMGC_PT121417     0.00000        0.00     0.00000     0.00000   \n",
       "705932          BMGC_PT121418     0.00000        0.00     0.00000     0.00000   \n",
       "705933          BMGC_PT121419     0.00000        0.00     0.00000     0.00000   \n",
       "\n",
       "        ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0          0.94738         1.0     0.50982     0.56520         0.0  ...   \n",
       "1          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "2          0.01066         0.0     0.01239     0.00000         0.0  ...   \n",
       "3          0.07825         0.0     0.72705     0.00761         0.0  ...   \n",
       "4          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "705929     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705930     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705931     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705932     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "705933     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "        ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  \\\n",
       "0          0.93319         0.0         0.0         0.0         0.0   \n",
       "1          0.00000         0.0         0.0         0.0         0.0   \n",
       "2          0.00958         0.0         0.0         0.0         0.0   \n",
       "3          0.92687         0.0         0.0         0.0         0.0   \n",
       "4          0.00000         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "705929     0.00000         0.0         0.0         0.0         0.0   \n",
       "705930     0.00000         0.0         0.0         0.0         0.0   \n",
       "705931     0.00000         0.0         0.0         0.0         0.0   \n",
       "705932     0.00000         0.0         0.0         0.0         0.0   \n",
       "705933     0.00000         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001151  ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0              0.0         0.0         0.0         0.0         0.0  \n",
       "1              0.0         0.0         0.0         0.0         0.0  \n",
       "2              0.0         0.0         0.0         0.0         0.0  \n",
       "3              0.0         0.0         0.0         0.0         0.0  \n",
       "4              0.0         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "705929         0.0         0.0         0.0         0.0         0.0  \n",
       "705930         0.0         0.0         0.0         0.0         0.0  \n",
       "705931         0.0         0.0         0.0         0.0         0.0  \n",
       "705932         0.0         0.0         0.0         0.0         0.0  \n",
       "705933         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[705934 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PM00001</td>\n",
       "      <td>0.83335</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88351</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.94738</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.50982</td>\n",
       "      <td>0.56520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93319</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PM00002</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PM00003</td>\n",
       "      <td>0.04959</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00129</td>\n",
       "      <td>0.02137</td>\n",
       "      <td>0.01066</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01239</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00958</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PM00004</td>\n",
       "      <td>0.05386</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.33075</td>\n",
       "      <td>0.07825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.72705</td>\n",
       "      <td>0.00761</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.92687</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PM00005</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834804</th>\n",
       "      <td>BMGC_DG20914</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834805</th>\n",
       "      <td>BMGC_DG20915</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834806</th>\n",
       "      <td>BMGC_DG20916</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834807</th>\n",
       "      <td>BMGC_DG20917</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834808</th>\n",
       "      <td>BMGC_DG20918</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>834809 rows × 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0                BMGC_PM00001     0.83335        0.00     0.88351     1.00000   \n",
       "1                BMGC_PM00002     0.00000        0.00     0.00000     0.00000   \n",
       "2                BMGC_PM00003     0.04959        0.00     0.00129     0.02137   \n",
       "3                BMGC_PM00004     0.05386        0.25     0.00000     0.33075   \n",
       "4                BMGC_PM00005     0.00000        0.00     0.00000     0.00000   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "834804           BMGC_DG20914     0.00000        0.00     0.00000     0.00000   \n",
       "834805           BMGC_DG20915     0.00000        0.00     0.00000     0.00000   \n",
       "834806           BMGC_DG20916     0.00000        0.00     0.00000     0.00000   \n",
       "834807           BMGC_DG20917     0.00000        0.00     0.00000     0.00000   \n",
       "834808           BMGC_DG20918     0.00000        0.00     0.00000     0.00000   \n",
       "\n",
       "        ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0          0.94738         1.0     0.50982     0.56520         0.0  ...   \n",
       "1          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "2          0.01066         0.0     0.01239     0.00000         0.0  ...   \n",
       "3          0.07825         0.0     0.72705     0.00761         0.0  ...   \n",
       "4          0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "...            ...         ...         ...         ...         ...  ...   \n",
       "834804     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834805     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834806     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834807     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "834808     0.00000         0.0     0.00000     0.00000         0.0  ...   \n",
       "\n",
       "        ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  \\\n",
       "0          0.93319         0.0         0.0         0.0         0.0   \n",
       "1          0.00000         0.0         0.0         0.0         0.0   \n",
       "2          0.00958         0.0         0.0         0.0         0.0   \n",
       "3          0.92687         0.0         0.0         0.0         0.0   \n",
       "4          0.00000         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "834804     0.00000         0.0         0.0         0.0         0.0   \n",
       "834805     0.00000         0.0         0.0         0.0         0.0   \n",
       "834806     0.00000         0.0         0.0         0.0         0.0   \n",
       "834807     0.00000         0.0         0.0         0.0         0.0   \n",
       "834808     0.00000         0.0         0.0         0.0         0.0   \n",
       "\n",
       "        ACH-001151  ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0              0.0         0.0         0.0         0.0         0.0  \n",
       "1              0.0         0.0         0.0         0.0         0.0  \n",
       "2              0.0         0.0         0.0         0.0         0.0  \n",
       "3              0.0         0.0         0.0         0.0         0.0  \n",
       "4              0.0         0.0         0.0         0.0         0.0  \n",
       "...            ...         ...         ...         ...         ...  \n",
       "834804         0.0         0.0         0.0         0.0         0.0  \n",
       "834805         0.0         0.0         0.0         0.0         0.0  \n",
       "834806         0.0         0.0         0.0         0.0         0.0  \n",
       "834807         0.0         0.0         0.0         0.0         0.0  \n",
       "834808         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[834809 rows x 287 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dti_omics_df = pd.concat([dti_methy_df, dti_gene_df, dti_transcript_df, dti_protein_df], axis=0).reset_index(drop=True)\n",
    "display(dti_omics_df)\n",
    "dti_df = pd.merge(bmgc_entity_df, dti_omics_df, left_on='BioMedGraphica_Conn_ID', right_on='BioMedGraphica_Conn_ID', how='left').drop(columns=['BioMedGraphica_ID', 'Type'])\n",
    "dti_df = dti_df.fillna(0.0)\n",
    "display(dti_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 423,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of dti_feat: (286, 834809)\n",
      "{'ACH-000002': 0, 'ACH-000006': 1, 'ACH-000008': 2, 'ACH-000020': 3, 'ACH-000038': 4, 'ACH-000046': 5, 'ACH-000047': 6, 'ACH-000050': 7, 'ACH-000053': 8, 'ACH-000056': 9, 'ACH-000058': 10, 'ACH-000059': 11, 'ACH-000061': 12, 'ACH-000062': 13, 'ACH-000065': 14, 'ACH-000072': 15, 'ACH-000073': 16, 'ACH-000074': 17, 'ACH-000078': 18, 'ACH-000081': 19, 'ACH-000082': 20, 'ACH-000091': 21, 'ACH-000099': 22, 'ACH-000101': 23, 'ACH-000102': 24, 'ACH-000104': 25, 'ACH-000105': 26, 'ACH-000106': 27, 'ACH-000107': 28, 'ACH-000112': 29, 'ACH-000116': 30, 'ACH-000117': 31, 'ACH-000120': 32, 'ACH-000121': 33, 'ACH-000124': 34, 'ACH-000129': 35, 'ACH-000136': 36, 'ACH-000137': 37, 'ACH-000144': 38, 'ACH-000145': 39, 'ACH-000151': 40, 'ACH-000157': 41, 'ACH-000162': 42, 'ACH-000171': 43, 'ACH-000174': 44, 'ACH-000181': 45, 'ACH-000186': 46, 'ACH-000187': 47, 'ACH-000189': 48, 'ACH-000190': 49, 'ACH-000193': 50, 'ACH-000196': 51, 'ACH-000197': 52, 'ACH-000198': 53, 'ACH-000200': 54, 'ACH-000201': 55, 'ACH-000204': 56, 'ACH-000218': 57, 'ACH-000221': 58, 'ACH-000225': 59, 'ACH-000226': 60, 'ACH-000228': 61, 'ACH-000233': 62, 'ACH-000236': 63, 'ACH-000242': 64, 'ACH-000245': 65, 'ACH-000252': 66, 'ACH-000254': 67, 'ACH-000267': 68, 'ACH-000270': 69, 'ACH-000273': 70, 'ACH-000282': 71, 'ACH-000287': 72, 'ACH-000293': 73, 'ACH-000295': 74, 'ACH-000298': 75, 'ACH-000301': 76, 'ACH-000309': 77, 'ACH-000323': 78, 'ACH-000326': 79, 'ACH-000327': 80, 'ACH-000329': 81, 'ACH-000332': 82, 'ACH-000335': 83, 'ACH-000338': 84, 'ACH-000345': 85, 'ACH-000346': 86, 'ACH-000347': 87, 'ACH-000349': 88, 'ACH-000351': 89, 'ACH-000353': 90, 'ACH-000354': 91, 'ACH-000355': 92, 'ACH-000357': 93, 'ACH-000358': 94, 'ACH-000359': 95, 'ACH-000360': 96, 'ACH-000367': 97, 'ACH-000369': 98, 'ACH-000372': 99, 'ACH-000373': 100, 'ACH-000379': 101, 'ACH-000380': 102, 'ACH-000381': 103, 'ACH-000384': 104, 'ACH-000386': 105, 'ACH-000389': 106, 'ACH-000392': 107, 'ACH-000394': 108, 'ACH-000396': 109, 'ACH-000399': 110, 'ACH-000400': 111, 'ACH-000408': 112, 'ACH-000422': 113, 'ACH-000423': 114, 'ACH-000428': 115, 'ACH-000429': 116, 'ACH-000432': 117, 'ACH-000439': 118, 'ACH-000440': 119, 'ACH-000442': 120, 'ACH-000443': 121, 'ACH-000447': 122, 'ACH-000449': 123, 'ACH-000451': 124, 'ACH-000457': 125, 'ACH-000465': 126, 'ACH-000469': 127, 'ACH-000476': 128, 'ACH-000478': 129, 'ACH-000482': 130, 'ACH-000483': 131, 'ACH-000489': 132, 'ACH-000501': 133, 'ACH-000504': 134, 'ACH-000506': 135, 'ACH-000508': 136, 'ACH-000514': 137, 'ACH-000516': 138, 'ACH-000530': 139, 'ACH-000534': 140, 'ACH-000536': 141, 'ACH-000538': 142, 'ACH-000545': 143, 'ACH-000546': 144, 'ACH-000555': 145, 'ACH-000560': 146, 'ACH-000562': 147, 'ACH-000565': 148, 'ACH-000566': 149, 'ACH-000567': 150, 'ACH-000568': 151, 'ACH-000569': 152, 'ACH-000572': 153, 'ACH-000574': 154, 'ACH-000577': 155, 'ACH-000581': 156, 'ACH-000583': 157, 'ACH-000584': 158, 'ACH-000586': 159, 'ACH-000603': 160, 'ACH-000610': 161, 'ACH-000611': 162, 'ACH-000613': 163, 'ACH-000615': 164, 'ACH-000616': 165, 'ACH-000617': 166, 'ACH-000619': 167, 'ACH-000625': 168, 'ACH-000626': 169, 'ACH-000628': 170, 'ACH-000636': 171, 'ACH-000640': 172, 'ACH-000641': 173, 'ACH-000643': 174, 'ACH-000644': 175, 'ACH-000650': 176, 'ACH-000654': 177, 'ACH-000656': 178, 'ACH-000660': 179, 'ACH-000664': 180, 'ACH-000669': 181, 'ACH-000670': 182, 'ACH-000674': 183, 'ACH-000675': 184, 'ACH-000678': 185, 'ACH-000679': 186, 'ACH-000680': 187, 'ACH-000689': 188, 'ACH-000691': 189, 'ACH-000702': 190, 'ACH-000703': 191, 'ACH-000714': 192, 'ACH-000716': 193, 'ACH-000727': 194, 'ACH-000729': 195, 'ACH-000730': 196, 'ACH-000733': 197, 'ACH-000743': 198, 'ACH-000744': 199, 'ACH-000745': 200, 'ACH-000750': 201, 'ACH-000751': 202, 'ACH-000752': 203, 'ACH-000754': 204, 'ACH-000755': 205, 'ACH-000759': 206, 'ACH-000770': 207, 'ACH-000772': 208, 'ACH-000776': 209, 'ACH-000780': 210, 'ACH-000786': 211, 'ACH-000789': 212, 'ACH-000795': 213, 'ACH-000798': 214, 'ACH-000802': 215, 'ACH-000803': 216, 'ACH-000806': 217, 'ACH-000812': 218, 'ACH-000814': 219, 'ACH-000815': 220, 'ACH-000816': 221, 'ACH-000818': 222, 'ACH-000821': 223, 'ACH-000822': 224, 'ACH-000825': 225, 'ACH-000828': 226, 'ACH-000830': 227, 'ACH-000835': 228, 'ACH-000843': 229, 'ACH-000852': 230, 'ACH-000857': 231, 'ACH-000871': 232, 'ACH-000873': 233, 'ACH-000875': 234, 'ACH-000883': 235, 'ACH-000892': 236, 'ACH-000893': 237, 'ACH-000894': 238, 'ACH-000895': 239, 'ACH-000896': 240, 'ACH-000902': 241, 'ACH-000903': 242, 'ACH-000905': 243, 'ACH-000914': 244, 'ACH-000918': 245, 'ACH-000920': 246, 'ACH-000927': 247, 'ACH-000930': 248, 'ACH-000937': 249, 'ACH-000939': 250, 'ACH-000940': 251, 'ACH-000944': 252, 'ACH-000947': 253, 'ACH-000949': 254, 'ACH-000951': 255, 'ACH-000955': 256, 'ACH-000968': 257, 'ACH-000970': 258, 'ACH-000973': 259, 'ACH-000976': 260, 'ACH-000981': 261, 'ACH-000983': 262, 'ACH-000987': 263, 'ACH-000989': 264, 'ACH-000995': 265, 'ACH-000998': 266, 'ACH-000999': 267, 'ACH-001016': 268, 'ACH-001021': 269, 'ACH-001023': 270, 'ACH-001024': 271, 'ACH-001049': 272, 'ACH-001063': 273, 'ACH-001064': 274, 'ACH-001089': 275, 'ACH-001106': 276, 'ACH-001125': 277, 'ACH-001126': 278, 'ACH-001127': 279, 'ACH-001138': 280, 'ACH-001151': 281, 'ACH-001182': 282, 'ACH-001199': 283, 'ACH-001203': 284, 'ACH-001233': 285}\n"
     ]
    }
   ],
   "source": [
    "# Select the features from dti_df\n",
    "dti_feat = dti_df.drop(columns=['BioMedGraphica_Conn_ID']).values.T\n",
    "# Save the features to .npy files\n",
    "np.save('./data/DrugQA/dti_feature.npy', dti_feat)\n",
    "np.save('./data/DrugScreen/dti_feature.npy', dti_feat)\n",
    "# Print the shapes of the features\n",
    "print(\"Shape of dti_feat:\", dti_feat.shape)\n",
    "\n",
    "# Create a dictionary mapping each depMapID to its corresponding row index (default DataFrame index)\n",
    "dti_sample_index_dict = dict(zip(remaining_dti_samples_df['depMapID'], remaining_dti_samples_df.index))\n",
    "print(dti_sample_index_dict)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 9.2.3.1 Drug Screen Integration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 429,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dti_sample_index</th>\n",
       "      <th>BMGC_Drug_ID</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>AUC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>80</td>\n",
       "      <td>818544</td>\n",
       "      <td>794405</td>\n",
       "      <td>0.981350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37</td>\n",
       "      <td>818544</td>\n",
       "      <td>792426</td>\n",
       "      <td>0.980213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>197</td>\n",
       "      <td>818544</td>\n",
       "      <td>794405</td>\n",
       "      <td>0.931183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>139</td>\n",
       "      <td>818544</td>\n",
       "      <td>812546</td>\n",
       "      <td>0.988258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>137</td>\n",
       "      <td>818544</td>\n",
       "      <td>794348</td>\n",
       "      <td>0.983326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42061</th>\n",
       "      <td>174</td>\n",
       "      <td>834578</td>\n",
       "      <td>799496</td>\n",
       "      <td>0.873306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42062</th>\n",
       "      <td>31</td>\n",
       "      <td>834578</td>\n",
       "      <td>798476</td>\n",
       "      <td>0.899328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42063</th>\n",
       "      <td>222</td>\n",
       "      <td>834578</td>\n",
       "      <td>799275</td>\n",
       "      <td>0.812820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42064</th>\n",
       "      <td>248</td>\n",
       "      <td>834578</td>\n",
       "      <td>799496</td>\n",
       "      <td>0.843211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42065</th>\n",
       "      <td>141</td>\n",
       "      <td>834578</td>\n",
       "      <td>799275</td>\n",
       "      <td>0.952633</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42066 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       dti_sample_index  BMGC_Drug_ID  BMGC_Disease_ID       AUC\n",
       "0                    80        818544           794405  0.981350\n",
       "1                    37        818544           792426  0.980213\n",
       "2                   197        818544           794405  0.931183\n",
       "3                   139        818544           812546  0.988258\n",
       "4                   137        818544           794348  0.983326\n",
       "...                 ...           ...              ...       ...\n",
       "42061               174        834578           799496  0.873306\n",
       "42062                31        834578           798476  0.899328\n",
       "42063               222        834578           799275  0.812820\n",
       "42064               248        834578           799496  0.843211\n",
       "42065               141        834578           799275  0.952633\n",
       "\n",
       "[42066 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dti_sample_index</th>\n",
       "      <th>BMGC_Drug_ID</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>AUC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>175</td>\n",
       "      <td>818544</td>\n",
       "      <td>794391</td>\n",
       "      <td>0.954785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>195</td>\n",
       "      <td>818544</td>\n",
       "      <td>794348</td>\n",
       "      <td>0.988450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>51</td>\n",
       "      <td>818544</td>\n",
       "      <td>799496</td>\n",
       "      <td>0.978046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>191</td>\n",
       "      <td>818544</td>\n",
       "      <td>794348</td>\n",
       "      <td>0.943754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>90</td>\n",
       "      <td>818544</td>\n",
       "      <td>796795</td>\n",
       "      <td>0.877464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9441</th>\n",
       "      <td>96</td>\n",
       "      <td>834115</td>\n",
       "      <td>799978</td>\n",
       "      <td>0.439161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9442</th>\n",
       "      <td>132</td>\n",
       "      <td>834115</td>\n",
       "      <td>797085</td>\n",
       "      <td>0.813459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9443</th>\n",
       "      <td>63</td>\n",
       "      <td>834115</td>\n",
       "      <td>797085</td>\n",
       "      <td>0.785710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9444</th>\n",
       "      <td>51</td>\n",
       "      <td>834578</td>\n",
       "      <td>799496</td>\n",
       "      <td>0.811327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9445</th>\n",
       "      <td>241</td>\n",
       "      <td>834578</td>\n",
       "      <td>798476</td>\n",
       "      <td>0.505332</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9446 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      dti_sample_index  BMGC_Drug_ID  BMGC_Disease_ID       AUC\n",
       "0                  175        818544           794391  0.954785\n",
       "1                  195        818544           794348  0.988450\n",
       "2                   51        818544           799496  0.978046\n",
       "3                  191        818544           794348  0.943754\n",
       "4                   90        818544           796795  0.877464\n",
       "...                ...           ...              ...       ...\n",
       "9441                96        834115           799978  0.439161\n",
       "9442               132        834115           797085  0.813459\n",
       "9443                63        834115           797085  0.785710\n",
       "9444                51        834578           799496  0.811327\n",
       "9445               241        834578           798476  0.505332\n",
       "\n",
       "[9446 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# For final_training_drug_screen_df, only keep the columns ['depMapID', 'BMGC_Drug_ID', 'BMGC_Disease_ID', 'AUC']\n",
    "final_training_drug_screen_dfc = final_training_drug_screen_df[['depMapID', 'BMGC_Drug_ID', 'BMGC_Disease_ID', 'AUC']].copy()\n",
    "# Map depMapID to dti_sample_index, BMGC_Drug_ID to node_index, BMGC_Disease_ID to node_index\n",
    "final_training_drug_screen_dfc['dti_sample_index'] = final_training_drug_screen_dfc['depMapID'].map(dti_sample_index_dict)\n",
    "final_training_drug_screen_dfc['BMGC_Drug_ID'] = final_training_drug_screen_dfc['BMGC_Drug_ID'].map(node_index_dict)\n",
    "final_training_drug_screen_dfc['BMGC_Disease_ID'] = final_training_drug_screen_dfc['BMGC_Disease_ID'].map(node_index_dict)\n",
    "final_training_drug_screen_dfc = final_training_drug_screen_dfc.drop(columns=['depMapID'])\n",
    "# reorder the columns to ['dti_sample_index', 'BMGC_Drug_ID', 'BMGC_Disease_ID', 'AUC']\n",
    "final_training_drug_screen_dfc = final_training_drug_screen_dfc[['dti_sample_index', 'BMGC_Drug_ID', 'BMGC_Disease_ID', 'AUC']]\n",
    "display(final_training_drug_screen_dfc)\n",
    "\n",
    "# For final_test_drug_screen_df, only keep the columns ['depMapID', 'BMGC_Drug_ID', 'BMGC_Disease_ID', 'AUC']\n",
    "final_test_drug_screen_dfc = final_test_drug_screen_df[['depMapID', 'BMGC_Drug_ID', 'BMGC_Disease_ID', 'AUC']].copy()\n",
    "# Map depMapID to dti_sample_index, BMGC_Drug_ID to node_index, BMGC_Disease_ID to node_index\n",
    "final_test_drug_screen_dfc['dti_sample_index'] = final_test_drug_screen_dfc['depMapID'].map(dti_sample_index_dict)\n",
    "final_test_drug_screen_dfc['BMGC_Drug_ID'] = final_test_drug_screen_dfc['BMGC_Drug_ID'].map(node_index_dict)\n",
    "final_test_drug_screen_dfc['BMGC_Disease_ID'] = final_test_drug_screen_dfc['BMGC_Disease_ID'].map(node_index_dict)\n",
    "final_test_drug_screen_dfc = final_test_drug_screen_dfc.drop(columns=['depMapID'])\n",
    "# reorder the columns to ['dti_sample_index', 'BMGC_Drug_ID', 'BMGC_Disease_ID', 'AUC']\n",
    "final_test_drug_screen_dfc = final_test_drug_screen_dfc[['dti_sample_index', 'BMGC_Drug_ID', 'BMGC_Disease_ID', 'AUC']]\n",
    "display(final_test_drug_screen_dfc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 430,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of final_training_drug_screen_array: (42066, 4)\n",
      "Shape of final_test_drug_screen_array: (9446, 4)\n"
     ]
    }
   ],
   "source": [
    "# Convert final_training_drug_screen_dfc and final_test_drug_screen_dfc to numpy arrays and save them\n",
    "final_training_drug_screen_array = final_training_drug_screen_dfc.values\n",
    "final_test_drug_screen_array = final_test_drug_screen_dfc.values\n",
    "# Save the numpy arrays to .npy files\n",
    "np.save('./data/DrugScreen/final_training_drug_screen.npy', final_training_drug_screen_array)\n",
    "np.save('./data/DrugScreen/final_test_drug_screen.npy', final_test_drug_screen_array)\n",
    "print(\"Shape of final_training_drug_screen_array:\", final_training_drug_screen_array.shape)\n",
    "print(\"Shape of final_test_drug_screen_array:\", final_test_drug_screen_array.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 9.2.3.2 DrugQA Integration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 456,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total remaining_dti_samples: 286\n",
      "Available samples in gene_df: 284\n",
      "Missing samples: 2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000047</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001125</th>\n",
       "      <th>ACH-001126</th>\n",
       "      <th>ACH-001127</th>\n",
       "      <th>ACH-001138</th>\n",
       "      <th>ACH-001151</th>\n",
       "      <th>ACH-001182</th>\n",
       "      <th>ACH-001199</th>\n",
       "      <th>ACH-001203</th>\n",
       "      <th>ACH-001233</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>1.039342</td>\n",
       "      <td>1.081979</td>\n",
       "      <td>1.069887</td>\n",
       "      <td>1.021275</td>\n",
       "      <td>1.019950</td>\n",
       "      <td>0.873542</td>\n",
       "      <td>1.954398</td>\n",
       "      <td>0.476005</td>\n",
       "      <td>0.498743</td>\n",
       "      <td>...</td>\n",
       "      <td>0.930320</td>\n",
       "      <td>1.280913</td>\n",
       "      <td>0.816260</td>\n",
       "      <td>0.996102</td>\n",
       "      <td>1.018304</td>\n",
       "      <td>1.050028</td>\n",
       "      <td>1.005951</td>\n",
       "      <td>1.040949</td>\n",
       "      <td>1.409362</td>\n",
       "      <td>0.863441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>1.039342</td>\n",
       "      <td>1.081979</td>\n",
       "      <td>1.069887</td>\n",
       "      <td>1.021275</td>\n",
       "      <td>1.019950</td>\n",
       "      <td>0.873542</td>\n",
       "      <td>1.954398</td>\n",
       "      <td>0.476005</td>\n",
       "      <td>0.498743</td>\n",
       "      <td>...</td>\n",
       "      <td>0.930320</td>\n",
       "      <td>1.280913</td>\n",
       "      <td>0.816260</td>\n",
       "      <td>0.996102</td>\n",
       "      <td>1.018304</td>\n",
       "      <td>1.050028</td>\n",
       "      <td>1.005951</td>\n",
       "      <td>1.040949</td>\n",
       "      <td>1.409362</td>\n",
       "      <td>0.863441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>1.010572</td>\n",
       "      <td>1.032794</td>\n",
       "      <td>0.857506</td>\n",
       "      <td>0.984673</td>\n",
       "      <td>1.007580</td>\n",
       "      <td>1.472495</td>\n",
       "      <td>1.205183</td>\n",
       "      <td>0.518170</td>\n",
       "      <td>0.515374</td>\n",
       "      <td>...</td>\n",
       "      <td>1.103581</td>\n",
       "      <td>0.760629</td>\n",
       "      <td>0.815531</td>\n",
       "      <td>0.981722</td>\n",
       "      <td>0.694711</td>\n",
       "      <td>0.917085</td>\n",
       "      <td>0.994378</td>\n",
       "      <td>1.004767</td>\n",
       "      <td>0.995919</td>\n",
       "      <td>0.815412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.999607</td>\n",
       "      <td>0.576030</td>\n",
       "      <td>1.225703</td>\n",
       "      <td>0.988541</td>\n",
       "      <td>1.005250</td>\n",
       "      <td>1.956878</td>\n",
       "      <td>1.189136</td>\n",
       "      <td>0.519018</td>\n",
       "      <td>1.009731</td>\n",
       "      <td>...</td>\n",
       "      <td>1.030032</td>\n",
       "      <td>0.825321</td>\n",
       "      <td>0.981129</td>\n",
       "      <td>0.948925</td>\n",
       "      <td>1.486665</td>\n",
       "      <td>0.913336</td>\n",
       "      <td>1.003539</td>\n",
       "      <td>1.038488</td>\n",
       "      <td>1.479791</td>\n",
       "      <td>1.223223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M-AS1</td>\n",
       "      <td>0.999607</td>\n",
       "      <td>0.576030</td>\n",
       "      <td>1.225703</td>\n",
       "      <td>0.988541</td>\n",
       "      <td>1.005250</td>\n",
       "      <td>1.956878</td>\n",
       "      <td>1.189136</td>\n",
       "      <td>0.519018</td>\n",
       "      <td>1.009731</td>\n",
       "      <td>...</td>\n",
       "      <td>1.030032</td>\n",
       "      <td>0.825321</td>\n",
       "      <td>0.981129</td>\n",
       "      <td>0.948925</td>\n",
       "      <td>1.486665</td>\n",
       "      <td>0.913336</td>\n",
       "      <td>1.003539</td>\n",
       "      <td>1.038488</td>\n",
       "      <td>1.479791</td>\n",
       "      <td>1.223223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38585</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>1.016271</td>\n",
       "      <td>0.928239</td>\n",
       "      <td>1.120232</td>\n",
       "      <td>0.983290</td>\n",
       "      <td>1.006196</td>\n",
       "      <td>0.893845</td>\n",
       "      <td>1.190278</td>\n",
       "      <td>0.987717</td>\n",
       "      <td>0.979038</td>\n",
       "      <td>...</td>\n",
       "      <td>0.973301</td>\n",
       "      <td>1.231011</td>\n",
       "      <td>1.368282</td>\n",
       "      <td>0.986425</td>\n",
       "      <td>1.055755</td>\n",
       "      <td>0.906142</td>\n",
       "      <td>0.999373</td>\n",
       "      <td>0.985555</td>\n",
       "      <td>0.998989</td>\n",
       "      <td>1.229548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38586</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>1.015564</td>\n",
       "      <td>0.991367</td>\n",
       "      <td>2.116413</td>\n",
       "      <td>0.970735</td>\n",
       "      <td>1.011754</td>\n",
       "      <td>1.870285</td>\n",
       "      <td>1.173788</td>\n",
       "      <td>1.019697</td>\n",
       "      <td>1.021887</td>\n",
       "      <td>...</td>\n",
       "      <td>1.028176</td>\n",
       "      <td>1.574313</td>\n",
       "      <td>1.112932</td>\n",
       "      <td>1.514654</td>\n",
       "      <td>1.311404</td>\n",
       "      <td>0.893903</td>\n",
       "      <td>0.998849</td>\n",
       "      <td>0.966974</td>\n",
       "      <td>0.995876</td>\n",
       "      <td>1.185762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38587</th>\n",
       "      <td>ZYXP1</td>\n",
       "      <td>0.971470</td>\n",
       "      <td>1.032808</td>\n",
       "      <td>0.961416</td>\n",
       "      <td>0.959549</td>\n",
       "      <td>1.009547</td>\n",
       "      <td>0.986482</td>\n",
       "      <td>1.623606</td>\n",
       "      <td>0.540793</td>\n",
       "      <td>1.495777</td>\n",
       "      <td>...</td>\n",
       "      <td>1.020167</td>\n",
       "      <td>1.076599</td>\n",
       "      <td>1.237886</td>\n",
       "      <td>1.057301</td>\n",
       "      <td>1.114824</td>\n",
       "      <td>1.960846</td>\n",
       "      <td>1.003790</td>\n",
       "      <td>1.021203</td>\n",
       "      <td>0.980589</td>\n",
       "      <td>1.099648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38588</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.560010</td>\n",
       "      <td>0.597288</td>\n",
       "      <td>0.936882</td>\n",
       "      <td>1.013471</td>\n",
       "      <td>0.994021</td>\n",
       "      <td>1.299603</td>\n",
       "      <td>0.716645</td>\n",
       "      <td>0.978837</td>\n",
       "      <td>0.506384</td>\n",
       "      <td>...</td>\n",
       "      <td>0.982988</td>\n",
       "      <td>0.789985</td>\n",
       "      <td>1.117091</td>\n",
       "      <td>0.830545</td>\n",
       "      <td>0.760735</td>\n",
       "      <td>0.813964</td>\n",
       "      <td>1.000523</td>\n",
       "      <td>0.981905</td>\n",
       "      <td>1.010208</td>\n",
       "      <td>0.962021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38589</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>1.016271</td>\n",
       "      <td>0.928239</td>\n",
       "      <td>1.256851</td>\n",
       "      <td>0.983290</td>\n",
       "      <td>1.006196</td>\n",
       "      <td>0.996218</td>\n",
       "      <td>1.333781</td>\n",
       "      <td>1.072011</td>\n",
       "      <td>0.979038</td>\n",
       "      <td>...</td>\n",
       "      <td>1.129281</td>\n",
       "      <td>1.265772</td>\n",
       "      <td>1.368282</td>\n",
       "      <td>0.986425</td>\n",
       "      <td>1.055755</td>\n",
       "      <td>0.906142</td>\n",
       "      <td>0.999373</td>\n",
       "      <td>0.985555</td>\n",
       "      <td>1.497124</td>\n",
       "      <td>1.229548</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38590 rows × 285 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 gene_name  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               A1BG    1.039342    1.081979    1.069887    1.021275   \n",
       "1           A1BG-AS1    1.039342    1.081979    1.069887    1.021275   \n",
       "2               A1CF    1.010572    1.032794    0.857506    0.984673   \n",
       "3                A2M    0.999607    0.576030    1.225703    0.988541   \n",
       "4            A2M-AS1    0.999607    0.576030    1.225703    0.988541   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "38585         ZYG11B    1.016271    0.928239    1.120232    0.983290   \n",
       "38586            ZYX    1.015564    0.991367    2.116413    0.970735   \n",
       "38587          ZYXP1    0.971470    1.032808    0.961416    0.959549   \n",
       "38588          ZZEF1    0.560010    0.597288    0.936882    1.013471   \n",
       "38589           ZZZ3    1.016271    0.928239    1.256851    0.983290   \n",
       "\n",
       "Unnamed: 0  ACH-000038  ACH-000046  ACH-000047  ACH-000050  ACH-000053  ...  \\\n",
       "0             1.019950    0.873542    1.954398    0.476005    0.498743  ...   \n",
       "1             1.019950    0.873542    1.954398    0.476005    0.498743  ...   \n",
       "2             1.007580    1.472495    1.205183    0.518170    0.515374  ...   \n",
       "3             1.005250    1.956878    1.189136    0.519018    1.009731  ...   \n",
       "4             1.005250    1.956878    1.189136    0.519018    1.009731  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "38585         1.006196    0.893845    1.190278    0.987717    0.979038  ...   \n",
       "38586         1.011754    1.870285    1.173788    1.019697    1.021887  ...   \n",
       "38587         1.009547    0.986482    1.623606    0.540793    1.495777  ...   \n",
       "38588         0.994021    1.299603    0.716645    0.978837    0.506384  ...   \n",
       "38589         1.006196    0.996218    1.333781    1.072011    0.979038  ...   \n",
       "\n",
       "Unnamed: 0  ACH-001106  ACH-001125  ACH-001126  ACH-001127  ACH-001138  \\\n",
       "0             0.930320    1.280913    0.816260    0.996102    1.018304   \n",
       "1             0.930320    1.280913    0.816260    0.996102    1.018304   \n",
       "2             1.103581    0.760629    0.815531    0.981722    0.694711   \n",
       "3             1.030032    0.825321    0.981129    0.948925    1.486665   \n",
       "4             1.030032    0.825321    0.981129    0.948925    1.486665   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "38585         0.973301    1.231011    1.368282    0.986425    1.055755   \n",
       "38586         1.028176    1.574313    1.112932    1.514654    1.311404   \n",
       "38587         1.020167    1.076599    1.237886    1.057301    1.114824   \n",
       "38588         0.982988    0.789985    1.117091    0.830545    0.760735   \n",
       "38589         1.129281    1.265772    1.368282    0.986425    1.055755   \n",
       "\n",
       "Unnamed: 0  ACH-001151  ACH-001182  ACH-001199  ACH-001203  ACH-001233  \n",
       "0             1.050028    1.005951    1.040949    1.409362    0.863441  \n",
       "1             1.050028    1.005951    1.040949    1.409362    0.863441  \n",
       "2             0.917085    0.994378    1.004767    0.995919    0.815412  \n",
       "3             0.913336    1.003539    1.038488    1.479791    1.223223  \n",
       "4             0.913336    1.003539    1.038488    1.479791    1.223223  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "38585         0.906142    0.999373    0.985555    0.998989    1.229548  \n",
       "38586         0.893903    0.998849    0.966974    0.995876    1.185762  \n",
       "38587         1.960846    1.003790    1.021203    0.980589    1.099648  \n",
       "38588         0.813964    1.000523    0.981905    1.010208    0.962021  \n",
       "38589         0.906142    0.999373    0.985555    1.497124    1.229548  \n",
       "\n",
       "[38590 rows x 285 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>BioMedGraphica_ID</th>\n",
       "      <th>Names_and_IDs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>BMG_PT000006</td>\n",
       "      <td>UniProt ID:A0A023T787 | RefSeq ID:NP_005096.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>BMG_PT000012</td>\n",
       "      <td>RefSeq ID:NP_006570.1 | UniProt ID:A0A024QYX0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>BMG_PT000014</td>\n",
       "      <td>UniProt ID:A0A024QYX5 | RefSeq ID:XP_024308127...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>BMG_PT000022</td>\n",
       "      <td>UniProt ID:A0A024QZ37 | RefSeq ID:NP_859054.1 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>BMG_PT000024</td>\n",
       "      <td>UniProt ID:A0A024QZ45 | RefSeq ID:XP_047292848...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>BMG_PT173974</td>\n",
       "      <td>ENSP00000520928.1 | ENSP00000520928 | NCBI Gen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>BMG_PT173975</td>\n",
       "      <td>RefSeq ID:NP_001366093 | NCBI Gene ID:1438.0 |...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>BMG_PT173976</td>\n",
       "      <td>NCBI Gene ID:1438.0 | CSF2RA | RefSeq ID:NP_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>BMG_PT173977</td>\n",
       "      <td>GTPBP6 | ENSP00000520931.1 | NCBI Gene ID:8225...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>BMG_PT173978</td>\n",
       "      <td>NCBI Gene ID:3581.0 | RefSeq ID:NP_789743 | IL...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID BioMedGraphica_ID  \\\n",
       "0               BMGC_PT000001      BMG_PT000006   \n",
       "1               BMGC_PT000002      BMG_PT000012   \n",
       "2               BMGC_PT000003      BMG_PT000014   \n",
       "3               BMGC_PT000004      BMG_PT000022   \n",
       "4               BMGC_PT000005      BMG_PT000024   \n",
       "...                       ...               ...   \n",
       "121414          BMGC_PT121415      BMG_PT173974   \n",
       "121415          BMGC_PT121416      BMG_PT173975   \n",
       "121416          BMGC_PT121417      BMG_PT173976   \n",
       "121417          BMGC_PT121418      BMG_PT173977   \n",
       "121418          BMGC_PT121419      BMG_PT173978   \n",
       "\n",
       "                                            Names_and_IDs  \n",
       "0           UniProt ID:A0A023T787 | RefSeq ID:NP_005096.1  \n",
       "1           RefSeq ID:NP_006570.1 | UniProt ID:A0A024QYX0  \n",
       "2       UniProt ID:A0A024QYX5 | RefSeq ID:XP_024308127...  \n",
       "3       UniProt ID:A0A024QZ37 | RefSeq ID:NP_859054.1 ...  \n",
       "4       UniProt ID:A0A024QZ45 | RefSeq ID:XP_047292848...  \n",
       "...                                                   ...  \n",
       "121414  ENSP00000520928.1 | ENSP00000520928 | NCBI Gen...  \n",
       "121415  RefSeq ID:NP_001366093 | NCBI Gene ID:1438.0 |...  \n",
       "121416  NCBI Gene ID:1438.0 | CSF2RA | RefSeq ID:NP_00...  \n",
       "121417  GTPBP6 | ENSP00000520931.1 | NCBI Gene ID:8225...  \n",
       "121418  NCBI Gene ID:3581.0 | RefSeq ID:NP_789743 | IL...  \n",
       "\n",
       "[121419 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 10 Gene HGNC Names: ['CASC11', 'CASC19', 'CASC8', 'CCAT2', 'KNOP1P5', 'LINC00861', 'LINC02964', 'LRATD2', 'MIR1204', 'MYC']\n",
      "Top 10 Gene Protein BMGC IDs: ['BMGC_PT037862', 'BMGC_PT013286', 'BMGC_PT000209', 'BMGC_PT000096', 'BMGC_PT102081', 'BMGC_PT102076', 'BMGC_PT106022', 'BMGC_PT048597', 'BMGC_PT003978', 'BMGC_PT009616', 'BMGC_PT019176', 'BMGC_PT013287', 'BMGC_PT070896']\n",
      "Top 10 Gene Protein BMGC LLM Name IDs: ['Protein LRATD2 (Breast cancer membrane protein 101) (LRAT domain-containing 2) (Protein FAM84B) (Protein NSE2) or LRATD2 or NCBI Gene ID:157638 or NCBI Gene ID:157638.0 or ENSP00000302578.3;ENSP00000498944.1 or UniProt ID:Q96KN1 or RefSeq ID:NP_777571 or RefSeq ID:NP_777571.1 or RefSeq ID:XP_016868596.1 or ENSP00000302578;ENSP00000498944', 'ENSP00000367207.3;ENSP00000430235.2;ENSP00000478887.2;ENSP00000499105.1;ENSP00000516742.1;ENSP00000516743.1;ENSP00000516744.1;ENSP00000516745.1 or MYC or Myc proto-oncogene protein (Class E basic helix-loop-helix protein 39) (bHLHe39) (Proto-oncogene c-Myc) (Transcription factor p64) or ENSP00000367207;ENSP00000430235;ENSP00000478887;ENSP00000499105;ENSP00000516742;ENSP00000516743;ENSP00000516744;ENSP00000516745 or NCBI Gene ID:4609 or RefSeq ID:NP_001341799 or RefSeq ID:NP_002458 or NCBI Gene ID:4609.0 or UniProt ID:P01106 or RefSeq ID:NP_001341799.1 or RefSeq ID:NP_002458.2', 'UniProt ID:A0A087WUS5 or RefSeq ID:NP_002458.2', 'UniProt ID:A0A024R9L7 or RefSeq ID:NP_002458.2', 'ENSP00000493145 or NCBI Gene ID:4609.0 or ENSP00000493145.1 or MYC', 'ENSP00000493139 or ENSP00000493139.1 or NCBI Gene ID:4609.0 or MYC', 'NCBI Gene ID:4609.0 or ENSP00000499182 or MYC or ENSP00000499182.1', 'ENSP00000259523.6 or NCBI Gene ID:4609.0 or MYC or ENSP00000259523', 'UniProt ID:A8WFE7 or RefSeq ID:NP_002458.2', 'UniProt ID:H0YBT0 or RefSeq ID:NP_002458.2', 'UniProt ID:Q14026 or RefSeq ID:NP_002458.2', 'UniProt ID:P01107 or RefSeq ID:NP_002458.2', 'ENSP00000429441.2 or NCBI Gene ID:4609.0 or ENSP00000429441 or MYC']\n"
     ]
    }
   ],
   "source": [
    "# select the columns in the remaining_dti_samples for gene_df\n",
    "# Only include samples that actually exist in gene_df columns\n",
    "available_samples = [sample for sample in remaining_dti_samples if sample in gene_df.columns]\n",
    "dti_gene_df = gene_df[['gene_name'] + sorted(available_samples)].copy()\n",
    "print(f\"Total remaining_dti_samples: {len(remaining_dti_samples)}\")\n",
    "print(f\"Available samples in gene_df: {len(available_samples)}\")\n",
    "print(f\"Missing samples: {len(remaining_dti_samples) - len(available_samples)}\")\n",
    "display(dti_gene_df)\n",
    "bmgc_protein_llmnameid_combined_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein_LLM_Name_ID_Combined.csv')\n",
    "display(bmgc_protein_llmnameid_combined_df)\n",
    "\n",
    "def extract_dti_gn_info(dti_gene_df, bmgc_gene_df, gene_transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k=10):\n",
    "    # Check if sample_ach_name exists in dti_gene_df columns\n",
    "    if sample_ach_name not in dti_gene_df.columns:\n",
    "        return \"non-existed\", \"non-existed\", \"non-existed\"\n",
    "    # Extract the top k highest values for the given sample name\n",
    "    top_k_genes = dti_gene_df.nlargest(k, sample_ach_name)[['gene_name', sample_ach_name]]\n",
    "    # Sort the top k genes by their values in descending order\n",
    "    top_k_genes = top_k_genes.sort_values(by=sample_ach_name, ascending=False).reset_index(drop=True)\n",
    "    top_k_gene_hgnc_name_list = top_k_genes['gene_name'].tolist()\n",
    "    # Merge with the bmgc_gene_df to get the BioMedGraphica_Conn_ID\n",
    "    bmgc_gene_df = bmgc_gene_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']].copy()\n",
    "    top_k_bmgc_gene_df = pd.merge(bmgc_gene_df, top_k_genes, left_on='HGNC_Symbol', right_on='gene_name', how='inner')\n",
    "    # Get the corresponding proteins\n",
    "    top_k_bmgc_gene_protein_df = pd.merge(gene_transcript_protein_entity_df, top_k_bmgc_gene_df, left_on='BMGC_GN_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'gene_name'])\n",
    "    top_k_bmgc_gene_protein_info_df = pd.merge(top_k_bmgc_gene_protein_df, bmgc_protein_llmnameid_combined_df, left_on='BMGC_PT_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'BioMedGraphica_ID', sample_ach_name])\n",
    "    top_k_gene_protein_bmgc_id_list = top_k_bmgc_gene_protein_info_df['BMGC_PT_ID'].tolist()\n",
    "    top_k_gene_protein_bmgc_llmnameid_list = top_k_bmgc_gene_protein_info_df['Names_and_IDs'].replace(r' \\| ', ' or ', regex=True).tolist()\n",
    "    return top_k_gene_hgnc_name_list, top_k_gene_protein_bmgc_id_list, top_k_gene_protein_bmgc_llmnameid_list\n",
    "\n",
    "# Example usage\n",
    "sample_ach_name = 'ACH-000002'\n",
    "k=10\n",
    "top_k_gene_hgnc_name_list, top_k_gene_protein_bmgc_id_list, top_k_gene_protein_bmgc_llmnameid_list = extract_dti_gn_info(dti_gene_df, bmgc_gene_df, gene_transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "print(f\"Top {k} Gene HGNC Names:\", top_k_gene_hgnc_name_list)\n",
    "print(f\"Top {k} Gene Protein BMGC IDs:\", top_k_gene_protein_bmgc_id_list)\n",
    "print(f\"Top {k} Gene Protein BMGC LLM Name IDs:\", top_k_gene_protein_bmgc_llmnameid_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 457,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total remaining_dti_samples: 286\n",
      "Available samples in raw_transcript_df: 268\n",
      "Missing samples: 18\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000020</th>\n",
       "      <th>ACH-000038</th>\n",
       "      <th>ACH-000046</th>\n",
       "      <th>ACH-000050</th>\n",
       "      <th>ACH-000053</th>\n",
       "      <th>ACH-000056</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000983</th>\n",
       "      <th>ACH-000987</th>\n",
       "      <th>ACH-000989</th>\n",
       "      <th>ACH-000995</th>\n",
       "      <th>ACH-000998</th>\n",
       "      <th>ACH-000999</th>\n",
       "      <th>ACH-001016</th>\n",
       "      <th>ACH-001064</th>\n",
       "      <th>ACH-001106</th>\n",
       "      <th>ACH-001151</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TFE3</td>\n",
       "      <td>2.459987</td>\n",
       "      <td>4.565798</td>\n",
       "      <td>5.006504</td>\n",
       "      <td>3.953176</td>\n",
       "      <td>4.085061</td>\n",
       "      <td>4.537537</td>\n",
       "      <td>3.445636</td>\n",
       "      <td>4.692890</td>\n",
       "      <td>3.348080</td>\n",
       "      <td>...</td>\n",
       "      <td>4.531120</td>\n",
       "      <td>3.921175</td>\n",
       "      <td>3.918550</td>\n",
       "      <td>4.275754</td>\n",
       "      <td>3.876828</td>\n",
       "      <td>3.926410</td>\n",
       "      <td>6.058822</td>\n",
       "      <td>4.963694</td>\n",
       "      <td>3.780212</td>\n",
       "      <td>4.562473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MMADHC</td>\n",
       "      <td>4.976372</td>\n",
       "      <td>7.099962</td>\n",
       "      <td>7.215527</td>\n",
       "      <td>5.775586</td>\n",
       "      <td>6.602355</td>\n",
       "      <td>6.713138</td>\n",
       "      <td>7.550829</td>\n",
       "      <td>6.560098</td>\n",
       "      <td>6.256188</td>\n",
       "      <td>...</td>\n",
       "      <td>6.511325</td>\n",
       "      <td>6.596016</td>\n",
       "      <td>6.834165</td>\n",
       "      <td>6.209605</td>\n",
       "      <td>6.542109</td>\n",
       "      <td>5.822229</td>\n",
       "      <td>7.804443</td>\n",
       "      <td>6.842440</td>\n",
       "      <td>6.222557</td>\n",
       "      <td>7.275786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PABPC1L</td>\n",
       "      <td>2.698399</td>\n",
       "      <td>4.968360</td>\n",
       "      <td>5.269459</td>\n",
       "      <td>6.012143</td>\n",
       "      <td>4.754882</td>\n",
       "      <td>3.956434</td>\n",
       "      <td>3.737370</td>\n",
       "      <td>6.679576</td>\n",
       "      <td>4.340483</td>\n",
       "      <td>...</td>\n",
       "      <td>4.726988</td>\n",
       "      <td>4.532650</td>\n",
       "      <td>5.912492</td>\n",
       "      <td>4.707263</td>\n",
       "      <td>2.743210</td>\n",
       "      <td>4.269397</td>\n",
       "      <td>4.833559</td>\n",
       "      <td>6.823880</td>\n",
       "      <td>3.655975</td>\n",
       "      <td>5.336246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RELL2</td>\n",
       "      <td>1.632580</td>\n",
       "      <td>3.826824</td>\n",
       "      <td>3.005733</td>\n",
       "      <td>3.198194</td>\n",
       "      <td>3.433501</td>\n",
       "      <td>3.470365</td>\n",
       "      <td>4.186973</td>\n",
       "      <td>2.589190</td>\n",
       "      <td>4.706444</td>\n",
       "      <td>...</td>\n",
       "      <td>2.115141</td>\n",
       "      <td>2.948216</td>\n",
       "      <td>3.306925</td>\n",
       "      <td>5.048803</td>\n",
       "      <td>2.485572</td>\n",
       "      <td>3.184126</td>\n",
       "      <td>2.765293</td>\n",
       "      <td>2.385043</td>\n",
       "      <td>2.336957</td>\n",
       "      <td>3.394496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GPT</td>\n",
       "      <td>-0.098142</td>\n",
       "      <td>1.047510</td>\n",
       "      <td>0.548131</td>\n",
       "      <td>1.006289</td>\n",
       "      <td>0.712799</td>\n",
       "      <td>0.160307</td>\n",
       "      <td>0.568305</td>\n",
       "      <td>0.658276</td>\n",
       "      <td>1.284211</td>\n",
       "      <td>...</td>\n",
       "      <td>0.973237</td>\n",
       "      <td>0.357728</td>\n",
       "      <td>1.352736</td>\n",
       "      <td>0.968441</td>\n",
       "      <td>0.463925</td>\n",
       "      <td>0.857868</td>\n",
       "      <td>0.480651</td>\n",
       "      <td>0.416134</td>\n",
       "      <td>0.022363</td>\n",
       "      <td>1.274494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19133</th>\n",
       "      <td>NAA60</td>\n",
       "      <td>4.218123</td>\n",
       "      <td>5.351137</td>\n",
       "      <td>3.682937</td>\n",
       "      <td>4.602184</td>\n",
       "      <td>4.857301</td>\n",
       "      <td>4.895785</td>\n",
       "      <td>3.901231</td>\n",
       "      <td>6.073190</td>\n",
       "      <td>4.149113</td>\n",
       "      <td>...</td>\n",
       "      <td>4.931943</td>\n",
       "      <td>4.624226</td>\n",
       "      <td>5.047877</td>\n",
       "      <td>5.095091</td>\n",
       "      <td>5.239022</td>\n",
       "      <td>5.414767</td>\n",
       "      <td>5.465068</td>\n",
       "      <td>5.064381</td>\n",
       "      <td>5.255766</td>\n",
       "      <td>4.935989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19134</th>\n",
       "      <td>MYH4</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>0.010412</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>0.168735</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>0.000741</td>\n",
       "      <td>-0.001724</td>\n",
       "      <td>0.000741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19135</th>\n",
       "      <td>MTAP</td>\n",
       "      <td>3.500397</td>\n",
       "      <td>5.335251</td>\n",
       "      <td>0.024972</td>\n",
       "      <td>1.292930</td>\n",
       "      <td>3.635648</td>\n",
       "      <td>0.296753</td>\n",
       "      <td>4.336532</td>\n",
       "      <td>0.011324</td>\n",
       "      <td>0.065164</td>\n",
       "      <td>...</td>\n",
       "      <td>4.489114</td>\n",
       "      <td>5.192278</td>\n",
       "      <td>4.174660</td>\n",
       "      <td>-0.058907</td>\n",
       "      <td>4.539859</td>\n",
       "      <td>4.890952</td>\n",
       "      <td>4.126143</td>\n",
       "      <td>5.831513</td>\n",
       "      <td>4.535595</td>\n",
       "      <td>0.102115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19136</th>\n",
       "      <td>C14orf93</td>\n",
       "      <td>2.310291</td>\n",
       "      <td>3.067836</td>\n",
       "      <td>2.388417</td>\n",
       "      <td>3.621152</td>\n",
       "      <td>2.834791</td>\n",
       "      <td>2.425974</td>\n",
       "      <td>1.876662</td>\n",
       "      <td>2.999576</td>\n",
       "      <td>3.387713</td>\n",
       "      <td>...</td>\n",
       "      <td>2.852722</td>\n",
       "      <td>2.673945</td>\n",
       "      <td>3.561302</td>\n",
       "      <td>3.231129</td>\n",
       "      <td>3.128145</td>\n",
       "      <td>3.101646</td>\n",
       "      <td>2.039440</td>\n",
       "      <td>3.055175</td>\n",
       "      <td>2.985173</td>\n",
       "      <td>2.577928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19137</th>\n",
       "      <td>NR3C2</td>\n",
       "      <td>0.029718</td>\n",
       "      <td>0.191227</td>\n",
       "      <td>0.665184</td>\n",
       "      <td>0.359425</td>\n",
       "      <td>0.112717</td>\n",
       "      <td>0.043885</td>\n",
       "      <td>1.296222</td>\n",
       "      <td>1.210420</td>\n",
       "      <td>0.126100</td>\n",
       "      <td>...</td>\n",
       "      <td>0.043885</td>\n",
       "      <td>1.007473</td>\n",
       "      <td>4.735352</td>\n",
       "      <td>1.459297</td>\n",
       "      <td>1.098918</td>\n",
       "      <td>1.852901</td>\n",
       "      <td>0.209698</td>\n",
       "      <td>0.281785</td>\n",
       "      <td>0.112717</td>\n",
       "      <td>0.678281</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19138 rows × 269 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Unnamed: 0 gene_name  ACH-000002  ACH-000006  ACH-000008  ACH-000020  \\\n",
       "0               TFE3    2.459987    4.565798    5.006504    3.953176   \n",
       "1             MMADHC    4.976372    7.099962    7.215527    5.775586   \n",
       "2            PABPC1L    2.698399    4.968360    5.269459    6.012143   \n",
       "3              RELL2    1.632580    3.826824    3.005733    3.198194   \n",
       "4                GPT   -0.098142    1.047510    0.548131    1.006289   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "19133          NAA60    4.218123    5.351137    3.682937    4.602184   \n",
       "19134           MYH4   -0.001724   -0.001724    0.010412   -0.001724   \n",
       "19135           MTAP    3.500397    5.335251    0.024972    1.292930   \n",
       "19136       C14orf93    2.310291    3.067836    2.388417    3.621152   \n",
       "19137          NR3C2    0.029718    0.191227    0.665184    0.359425   \n",
       "\n",
       "Unnamed: 0  ACH-000038  ACH-000046  ACH-000050  ACH-000053  ACH-000056  ...  \\\n",
       "0             4.085061    4.537537    3.445636    4.692890    3.348080  ...   \n",
       "1             6.602355    6.713138    7.550829    6.560098    6.256188  ...   \n",
       "2             4.754882    3.956434    3.737370    6.679576    4.340483  ...   \n",
       "3             3.433501    3.470365    4.186973    2.589190    4.706444  ...   \n",
       "4             0.712799    0.160307    0.568305    0.658276    1.284211  ...   \n",
       "...                ...         ...         ...         ...         ...  ...   \n",
       "19133         4.857301    4.895785    3.901231    6.073190    4.149113  ...   \n",
       "19134        -0.001724   -0.001724   -0.001724   -0.001724   -0.001724  ...   \n",
       "19135         3.635648    0.296753    4.336532    0.011324    0.065164  ...   \n",
       "19136         2.834791    2.425974    1.876662    2.999576    3.387713  ...   \n",
       "19137         0.112717    0.043885    1.296222    1.210420    0.126100  ...   \n",
       "\n",
       "Unnamed: 0  ACH-000983  ACH-000987  ACH-000989  ACH-000995  ACH-000998  \\\n",
       "0             4.531120    3.921175    3.918550    4.275754    3.876828   \n",
       "1             6.511325    6.596016    6.834165    6.209605    6.542109   \n",
       "2             4.726988    4.532650    5.912492    4.707263    2.743210   \n",
       "3             2.115141    2.948216    3.306925    5.048803    2.485572   \n",
       "4             0.973237    0.357728    1.352736    0.968441    0.463925   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "19133         4.931943    4.624226    5.047877    5.095091    5.239022   \n",
       "19134        -0.001724   -0.001724   -0.001724   -0.001724    0.168735   \n",
       "19135         4.489114    5.192278    4.174660   -0.058907    4.539859   \n",
       "19136         2.852722    2.673945    3.561302    3.231129    3.128145   \n",
       "19137         0.043885    1.007473    4.735352    1.459297    1.098918   \n",
       "\n",
       "Unnamed: 0  ACH-000999  ACH-001016  ACH-001064  ACH-001106  ACH-001151  \n",
       "0             3.926410    6.058822    4.963694    3.780212    4.562473  \n",
       "1             5.822229    7.804443    6.842440    6.222557    7.275786  \n",
       "2             4.269397    4.833559    6.823880    3.655975    5.336246  \n",
       "3             3.184126    2.765293    2.385043    2.336957    3.394496  \n",
       "4             0.857868    0.480651    0.416134    0.022363    1.274494  \n",
       "...                ...         ...         ...         ...         ...  \n",
       "19133         5.414767    5.465068    5.064381    5.255766    4.935989  \n",
       "19134        -0.001724    0.000741    0.000741   -0.001724    0.000741  \n",
       "19135         4.890952    4.126143    5.831513    4.535595    0.102115  \n",
       "19136         3.101646    2.039440    3.055175    2.985173    2.577928  \n",
       "19137         1.852901    0.209698    0.281785    0.112717    0.678281  \n",
       "\n",
       "[19138 rows x 269 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 10 Transcript HGNC Names: ['MT-ATP8', 'MT-CO2', 'MT-CO3', 'MT-ND3', 'MT-ND4L', 'MT-CO1', 'MT-ND4', 'MT-ATP6', 'MT-ND2', 'RPS18']\n",
      "Top 10 Transcript Protein BMGC IDs: ['BMGC_PT017217', 'BMGC_PT013491', 'BMGC_PT013492', 'BMGC_PT013493', 'BMGC_PT013490', 'BMGC_PT013229', 'BMGC_PT013230', 'BMGC_PT013496', 'BMGC_PT013263', 'BMGC_PT013231', 'BMGC_PT017217', 'BMGC_PT017217', 'BMGC_PT055005', 'BMGC_PT017217', 'BMGC_PT017217', 'BMGC_PT074382', 'BMGC_PT072563', 'BMGC_PT075799', 'BMGC_PT072395', 'BMGC_PT074349', 'BMGC_PT072934', 'BMGC_PT097348']\n",
      "Top 10 Transcript Protein BMGC LLM Name IDs: ['NCBI Gene ID:6222 or ENSP00000211372.5;ENSP00000393241.2;ENSP00000403175.2;ENSP00000412583.2;ENSP00000416110.2 or UniProt ID:P62269 or NCBI Gene ID:6222.0 or Small ribosomal subunit protein uS13 (40S ribosomal protein S18) (Ke-3) (Ke3) or ENSP00000211372;ENSP00000393241;ENSP00000403175;ENSP00000412583;ENSP00000416110 or RPS18 or RefSeq ID:NP_072045 or RefSeq ID:NP_072045.1', 'NCBI Gene ID:4537.0 or NADH-ubiquinone oxidoreductase chain 3 (EC 7.1.1.2) (NADH dehydrogenase subunit 3) or RefSeq ID:YP_003024033.1 or MT-ND3 or NCBI Gene ID:4537 or ENSP00000355206.2 or ENSP00000355206 or RefSeq ID:YP_003024033 or UniProt ID:P03897', 'MT-ND4L or NADH-ubiquinone oxidoreductase chain 4L (EC 7.1.1.2) (NADH dehydrogenase subunit 4L) (ND4L) or NCBI Gene ID:4539 or UniProt ID:P03901 or RefSeq ID:YP_003024034.1 or ENSP00000354728 or NCBI Gene ID:4539.0 or RefSeq ID:YP_003024034 or ENSP00000354728.1', 'MT-ND4 or NCBI Gene ID:4538.0 or ENSP00000354961 or ENSP00000354961.2 or NADH-ubiquinone oxidoreductase chain 4 (EC 7.1.1.2) (NADH dehydrogenase subunit 4) or NCBI Gene ID:4538 or RefSeq ID:YP_003024035 or RefSeq ID:YP_003024035.1 or UniProt ID:P03905', 'RefSeq ID:YP_003024027 or NCBI Gene ID:4536 or NADH-ubiquinone oxidoreductase chain 2 (EC 7.1.1.2) (NADH dehydrogenase subunit 2) or RefSeq ID:YP_003024027.1 or NCBI Gene ID:4536.0 or ENSP00000355046.4 or ENSP00000355046 or MT-ND2 or UniProt ID:P03891', 'RefSeq ID:YP_003024028.1 or NCBI Gene ID:4512 or MT-CO1 or Cytochrome c oxidase subunit 1 (EC 7.1.1.9) (Cytochrome c oxidase polypeptide I) or ENSP00000354499 or UniProt ID:P00395 or RefSeq ID:YP_003024028 or NCBI Gene ID:4512.0 or ENSP00000354499.2', 'UniProt ID:P00403 or MT-CO2 or ENSP00000354876 or RefSeq ID:YP_003024029.1 or RefSeq ID:YP_003024029 or NCBI Gene ID:4513.0 or NCBI Gene ID:4513 or ENSP00000354876.1 or Cytochrome c oxidase subunit 2 (EC 7.1.1.9) (Cytochrome c oxidase polypeptide II)', 'UniProt ID:P03928 or RefSeq ID:YP_003024030 or NCBI Gene ID:4509.0 or ENSP00000355265.1 or NCBI Gene ID:4509 or RefSeq ID:YP_003024030.1 or ATP synthase protein 8 (A6L) (F-ATPase subunit 8) or MT-ATP8 or ENSP00000355265', 'MT-ATP6 or NCBI Gene ID:4508 or ENSP00000354632 or RefSeq ID:YP_003024031 or RefSeq ID:YP_003024031.1 or NCBI Gene ID:4508.0 or ATP synthase subunit a (F-ATPase protein 6) or UniProt ID:P00846 or ENSP00000354632.2', 'ENSP00000354982.2 or NCBI Gene ID:4514.0 or NCBI Gene ID:4514 or RefSeq ID:YP_003024032 or RefSeq ID:YP_003024032.1 or Cytochrome c oxidase subunit 3 (EC 7.1.1.9) (Cytochrome c oxidase polypeptide III) or UniProt ID:P00414 or MT-CO3 or ENSP00000354982', 'NCBI Gene ID:6222 or ENSP00000211372.5;ENSP00000393241.2;ENSP00000403175.2;ENSP00000412583.2;ENSP00000416110.2 or UniProt ID:P62269 or NCBI Gene ID:6222.0 or Small ribosomal subunit protein uS13 (40S ribosomal protein S18) (Ke-3) (Ke3) or ENSP00000211372;ENSP00000393241;ENSP00000403175;ENSP00000412583;ENSP00000416110 or RPS18 or RefSeq ID:NP_072045 or RefSeq ID:NP_072045.1', 'NCBI Gene ID:6222 or ENSP00000211372.5;ENSP00000393241.2;ENSP00000403175.2;ENSP00000412583.2;ENSP00000416110.2 or UniProt ID:P62269 or NCBI Gene ID:6222.0 or Small ribosomal subunit protein uS13 (40S ribosomal protein S18) (Ke-3) (Ke3) or ENSP00000211372;ENSP00000393241;ENSP00000403175;ENSP00000412583;ENSP00000416110 or RPS18 or RefSeq ID:NP_072045 or RefSeq ID:NP_072045.1', 'ENSP00000392349 or NCBI Gene ID:6222.0 or ENSP00000392349.1 or RPS18 or RefSeq ID:NP_072045', 'NCBI Gene ID:6222 or ENSP00000211372.5;ENSP00000393241.2;ENSP00000403175.2;ENSP00000412583.2;ENSP00000416110.2 or UniProt ID:P62269 or NCBI Gene ID:6222.0 or Small ribosomal subunit protein uS13 (40S ribosomal protein S18) (Ke-3) (Ke3) or ENSP00000211372;ENSP00000393241;ENSP00000403175;ENSP00000412583;ENSP00000416110 or RPS18 or RefSeq ID:NP_072045 or RefSeq ID:NP_072045.1', 'NCBI Gene ID:6222 or ENSP00000211372.5;ENSP00000393241.2;ENSP00000403175.2;ENSP00000412583.2;ENSP00000416110.2 or UniProt ID:P62269 or NCBI Gene ID:6222.0 or Small ribosomal subunit protein uS13 (40S ribosomal protein S18) (Ke-3) (Ke3) or ENSP00000211372;ENSP00000393241;ENSP00000403175;ENSP00000412583;ENSP00000416110 or RPS18 or RefSeq ID:NP_072045 or RefSeq ID:NP_072045.1', 'ENSP00000434837 or RPS18 or NCBI Gene ID:6222.0 or ENSP00000434837.1', 'RPS18 or NCBI Gene ID:6222.0 or ENSP00000431986.1 or ENSP00000431986', 'RPS18 or ENSP00000437050 or ENSP00000437050.1 or NCBI Gene ID:6222.0', 'ENSP00000431721 or RPS18 or NCBI Gene ID:6222.0 or ENSP00000431721.1', 'RPS18 or ENSP00000434786 or NCBI Gene ID:6222.0 or ENSP00000434786.1', 'ENSP00000432580 or RPS18 or NCBI Gene ID:6222.0 or ENSP00000432580.1', 'ENSP00000485840.1 or RPS18 or NCBI Gene ID:6222.0 or ENSP00000485840']\n"
     ]
    }
   ],
   "source": [
    "# select the columns in the remaining_dti_samples for raw_transcript_df\n",
    "# Only include samples that actually exist in raw_transcript_df columns\n",
    "available_transcript_samples = [sample for sample in remaining_dti_samples if sample in raw_transcript_df.columns]\n",
    "dti_transcript_df = raw_transcript_df[['gene_name'] + sorted(available_transcript_samples)].copy()\n",
    "print(f\"Total remaining_dti_samples: {len(remaining_dti_samples)}\")\n",
    "print(f\"Available samples in raw_transcript_df: {len(available_transcript_samples)}\")\n",
    "print(f\"Missing samples: {len(remaining_dti_samples) - len(available_transcript_samples)}\")\n",
    "display(dti_transcript_df)\n",
    "\n",
    "def extract_dti_ts_info(dti_transcript_df, bmgc_transcript_df, transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k=10):\n",
    "    # Check if sample_ach_name exists in dti_transcript_df columns\n",
    "    if sample_ach_name not in dti_transcript_df.columns:\n",
    "        return \"non-existed\", \"non-existed\", \"non-existed\"\n",
    "    # Extract the top k highest values for the given sample name\n",
    "    top_k_transcripts = dti_transcript_df.nlargest(k, sample_ach_name)[['gene_name', sample_ach_name]]\n",
    "    # Sort the top k transcripts by their values in descending order\n",
    "    top_k_transcripts = top_k_transcripts.sort_values(by=sample_ach_name, ascending=False).reset_index(drop=True)\n",
    "    top_k_transcript_hgnc_name_list = top_k_transcripts['gene_name'].tolist()\n",
    "    # Merge with the bmgc_transcript_df to get the BioMedGraphica_Conn_ID\n",
    "    bmgc_transcript_df = bmgc_transcript_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']].copy()\n",
    "    top_k_bmgc_transcript_df = pd.merge(bmgc_transcript_df, top_k_transcripts, left_on='HGNC_Symbol', right_on='gene_name', how='inner')\n",
    "    # Get the corresponding proteins\n",
    "    top_k_bmgc_transcript_protein_df = pd.merge(transcript_protein_entity_df, top_k_bmgc_transcript_df, left_on='BMGC_TS_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'gene_name'])\n",
    "    top_k_bmgc_transcript_protein_info_df = pd.merge(top_k_bmgc_transcript_protein_df, bmgc_protein_llmnameid_combined_df, left_on='BMGC_PT_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_Conn_ID', 'BioMedGraphica_ID', sample_ach_name])\n",
    "    top_k_transcript_protein_bmgc_id_list = top_k_bmgc_transcript_protein_info_df['BMGC_PT_ID'].tolist()\n",
    "    top_k_transcript_protein_bmgc_llmnameid_list = top_k_bmgc_transcript_protein_info_df['Names_and_IDs'].replace(r' \\| ', ' or ', regex=True).tolist()\n",
    "    return top_k_transcript_hgnc_name_list, top_k_transcript_protein_bmgc_id_list, top_k_transcript_protein_bmgc_llmnameid_list\n",
    "\n",
    "# Example usage\n",
    "sample_ach_name = 'ACH-000002'\n",
    "k=10\n",
    "top_k_transcript_hgnc_name_list, top_k_transcript_protein_bmgc_id_list, top_k_transcript_protein_bmgc_llmnameid_list = extract_dti_ts_info(dti_transcript_df, bmgc_transcript_df, transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "print(f\"Top {k} Transcript HGNC Names:\", top_k_transcript_hgnc_name_list)\n",
    "print(f\"Top {k} Transcript Protein BMGC IDs:\", top_k_transcript_protein_bmgc_id_list)\n",
    "print(f\"Top {k} Transcript Protein BMGC LLM Name IDs:\", top_k_transcript_protein_bmgc_llmnameid_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 460,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\hemingzhang\\AppData\\Local\\Temp\\ipykernel_33876\\1700773702.py:2: DtypeWarning: Columns (2,7) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  bmg_protein_all_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein.csv')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BioMedGraphica_Conn_ID</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>HGNC_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMGC_PT000001</td>\n",
       "      <td>A0A023T787</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMGC_PT000002</td>\n",
       "      <td>A0A024QYX0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMGC_PT000003</td>\n",
       "      <td>A0A024QYX5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMGC_PT000004</td>\n",
       "      <td>A0A024QZ37</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMGC_PT000005</td>\n",
       "      <td>A0A024QZ45</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121414</th>\n",
       "      <td>BMGC_PT121415</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CSF2RA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121415</th>\n",
       "      <td>BMGC_PT121416</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CSF2RA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121416</th>\n",
       "      <td>BMGC_PT121417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CSF2RA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121417</th>\n",
       "      <td>BMGC_PT121418</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GTPBP6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121418</th>\n",
       "      <td>BMGC_PT121419</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IL9R</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121419 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       BioMedGraphica_Conn_ID  Uniprot_ID HGNC_Symbol\n",
       "0               BMGC_PT000001  A0A023T787         NaN\n",
       "1               BMGC_PT000002  A0A024QYX0         NaN\n",
       "2               BMGC_PT000003  A0A024QYX5         NaN\n",
       "3               BMGC_PT000004  A0A024QZ37         NaN\n",
       "4               BMGC_PT000005  A0A024QZ45         NaN\n",
       "...                       ...         ...         ...\n",
       "121414          BMGC_PT121415         NaN      CSF2RA\n",
       "121415          BMGC_PT121416         NaN      CSF2RA\n",
       "121416          BMGC_PT121417         NaN      CSF2RA\n",
       "121417          BMGC_PT121418         NaN      GTPBP6\n",
       "121418          BMGC_PT121419         NaN        IL9R\n",
       "\n",
       "[121419 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total remaining_dti_samples: 286\n",
      "Available samples in symbol_protein_map_df: 37\n",
      "Missing samples: 249\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_Symbol</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000062</th>\n",
       "      <th>ACH-000091</th>\n",
       "      <th>ACH-000117</th>\n",
       "      <th>ACH-000309</th>\n",
       "      <th>ACH-000335</th>\n",
       "      <th>ACH-000347</th>\n",
       "      <th>ACH-000349</th>\n",
       "      <th>ACH-000351</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000670</th>\n",
       "      <th>ACH-000678</th>\n",
       "      <th>ACH-000727</th>\n",
       "      <th>ACH-000730</th>\n",
       "      <th>ACH-000750</th>\n",
       "      <th>ACH-000752</th>\n",
       "      <th>ACH-000755</th>\n",
       "      <th>ACH-000852</th>\n",
       "      <th>ACH-000857</th>\n",
       "      <th>ACH-000987</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SLC12A2</td>\n",
       "      <td>-0.168840</td>\n",
       "      <td>0.729167</td>\n",
       "      <td>-0.907412</td>\n",
       "      <td>-1.532994</td>\n",
       "      <td>-1.218412</td>\n",
       "      <td>1.472404</td>\n",
       "      <td>-1.492391</td>\n",
       "      <td>1.035254</td>\n",
       "      <td>1.350658</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.398706</td>\n",
       "      <td>0.285981</td>\n",
       "      <td>-1.903259</td>\n",
       "      <td>1.262304</td>\n",
       "      <td>-1.509770</td>\n",
       "      <td>-1.119232</td>\n",
       "      <td>-0.074979</td>\n",
       "      <td>-2.049783</td>\n",
       "      <td>-0.402035</td>\n",
       "      <td>-1.059660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HOXD13</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.525421</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.075547</td>\n",
       "      <td>-0.648070</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KDM1A</td>\n",
       "      <td>0.876906</td>\n",
       "      <td>0.410168</td>\n",
       "      <td>-0.401211</td>\n",
       "      <td>0.561969</td>\n",
       "      <td>-0.059550</td>\n",
       "      <td>-0.279095</td>\n",
       "      <td>-0.421580</td>\n",
       "      <td>-0.058377</td>\n",
       "      <td>-0.385244</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039246</td>\n",
       "      <td>-0.202710</td>\n",
       "      <td>-0.498443</td>\n",
       "      <td>0.513282</td>\n",
       "      <td>-0.503426</td>\n",
       "      <td>-0.758587</td>\n",
       "      <td>0.272818</td>\n",
       "      <td>0.060652</td>\n",
       "      <td>-0.234683</td>\n",
       "      <td>0.382719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SOX1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-4.317090</td>\n",
       "      <td>-1.344490</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.791036</td>\n",
       "      <td>5.416105</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.238624</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.749516</td>\n",
       "      <td>-0.988544</td>\n",
       "      <td>0.614960</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SOX2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.882953</td>\n",
       "      <td>-1.004530</td>\n",
       "      <td>-1.993823</td>\n",
       "      <td>-1.452967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.164757</td>\n",
       "      <td>4.897920</td>\n",
       "      <td>-0.974087</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.642244</td>\n",
       "      <td>2.046123</td>\n",
       "      <td>-0.308938</td>\n",
       "      <td>3.730812</td>\n",
       "      <td>-0.492513</td>\n",
       "      <td>-1.067185</td>\n",
       "      <td>0.906425</td>\n",
       "      <td>-1.490469</td>\n",
       "      <td>-1.891108</td>\n",
       "      <td>-1.496319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10623</th>\n",
       "      <td>CCL26</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10624</th>\n",
       "      <td>ALOX5AP</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-2.118748</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-6.872745</td>\n",
       "      <td>3.007516</td>\n",
       "      <td>-1.455220</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.263190</td>\n",
       "      <td>1.389608</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.181452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10625</th>\n",
       "      <td>CYSTM1</td>\n",
       "      <td>1.478774</td>\n",
       "      <td>0.298349</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.335349</td>\n",
       "      <td>-3.053117</td>\n",
       "      <td>1.929118</td>\n",
       "      <td>1.388368</td>\n",
       "      <td>-0.450275</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.740274</td>\n",
       "      <td>-2.308188</td>\n",
       "      <td>-0.997777</td>\n",
       "      <td>0.552372</td>\n",
       "      <td>-0.555743</td>\n",
       "      <td>0.557652</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.982198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10626</th>\n",
       "      <td>MGST2</td>\n",
       "      <td>2.248389</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.092565</td>\n",
       "      <td>1.019450</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-2.845178</td>\n",
       "      <td>0.382543</td>\n",
       "      <td>-0.548249</td>\n",
       "      <td>-0.168199</td>\n",
       "      <td>...</td>\n",
       "      <td>0.272329</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.166740</td>\n",
       "      <td>-0.481567</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10627</th>\n",
       "      <td>CNIH4</td>\n",
       "      <td>0.594338</td>\n",
       "      <td>0.120958</td>\n",
       "      <td>0.035153</td>\n",
       "      <td>0.275067</td>\n",
       "      <td>-0.858140</td>\n",
       "      <td>0.040261</td>\n",
       "      <td>-0.080579</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.461453</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.364989</td>\n",
       "      <td>-0.845980</td>\n",
       "      <td>0.230535</td>\n",
       "      <td>-0.222600</td>\n",
       "      <td>-0.250076</td>\n",
       "      <td>0.403931</td>\n",
       "      <td>0.087274</td>\n",
       "      <td>-0.151546</td>\n",
       "      <td>0.036337</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10628 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      HGNC_Symbol  ACH-000008  ACH-000062  ACH-000091  ACH-000117  ACH-000309  \\\n",
       "0         SLC12A2   -0.168840    0.729167   -0.907412   -1.532994   -1.218412   \n",
       "1          HOXD13    0.000000    0.000000    0.000000   -0.525421    0.000000   \n",
       "2           KDM1A    0.876906    0.410168   -0.401211    0.561969   -0.059550   \n",
       "3            SOX1    0.000000    0.000000   -4.317090   -1.344490    0.000000   \n",
       "4            SOX2    0.000000    1.882953   -1.004530   -1.993823   -1.452967   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "10623       CCL26    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "10624     ALOX5AP    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "10625      CYSTM1    1.478774    0.298349    0.000000    1.335349   -3.053117   \n",
       "10626       MGST2    2.248389    0.000000   -0.092565    1.019450    0.000000   \n",
       "10627       CNIH4    0.594338    0.120958    0.035153    0.275067   -0.858140   \n",
       "\n",
       "       ACH-000335  ACH-000347  ACH-000349  ACH-000351  ...  ACH-000670  \\\n",
       "0        1.472404   -1.492391    1.035254    1.350658  ...   -0.398706   \n",
       "1        0.000000    0.000000    0.000000    0.000000  ...    0.000000   \n",
       "2       -0.279095   -0.421580   -0.058377   -0.385244  ...   -0.039246   \n",
       "3        0.000000   -1.791036    5.416105    0.000000  ...   -3.238624   \n",
       "4        0.000000   -1.164757    4.897920   -0.974087  ...   -1.642244   \n",
       "...           ...         ...         ...         ...  ...         ...   \n",
       "10623    0.000000    0.000000    0.000000    0.000000  ...    0.000000   \n",
       "10624    0.000000    0.000000    0.000000   -2.118748  ...    0.000000   \n",
       "10625    1.929118    1.388368   -0.450275    0.000000  ...   -0.740274   \n",
       "10626   -2.845178    0.382543   -0.548249   -0.168199  ...    0.272329   \n",
       "10627    0.040261   -0.080579    0.000000    0.461453  ...    0.000000   \n",
       "\n",
       "       ACH-000678  ACH-000727  ACH-000730  ACH-000750  ACH-000752  ACH-000755  \\\n",
       "0        0.285981   -1.903259    1.262304   -1.509770   -1.119232   -0.074979   \n",
       "1        0.000000   -1.075547   -0.648070    0.000000    0.000000    0.000000   \n",
       "2       -0.202710   -0.498443    0.513282   -0.503426   -0.758587    0.272818   \n",
       "3        0.000000    0.000000    0.000000   -0.749516   -0.988544    0.614960   \n",
       "4        2.046123   -0.308938    3.730812   -0.492513   -1.067185    0.906425   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "10623    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "10624   -6.872745    3.007516   -1.455220    0.000000    0.000000    1.263190   \n",
       "10625   -2.308188   -0.997777    0.552372   -0.555743    0.557652    0.000000   \n",
       "10626    0.000000    0.000000    0.000000    0.000000    0.000000    1.166740   \n",
       "10627   -0.364989   -0.845980    0.230535   -0.222600   -0.250076    0.403931   \n",
       "\n",
       "       ACH-000852  ACH-000857  ACH-000987  \n",
       "0       -2.049783   -0.402035   -1.059660  \n",
       "1        0.000000    0.000000    0.000000  \n",
       "2        0.060652   -0.234683    0.382719  \n",
       "3        0.000000    0.000000    0.000000  \n",
       "4       -1.490469   -1.891108   -1.496319  \n",
       "...           ...         ...         ...  \n",
       "10623    0.000000    0.000000    0.000000  \n",
       "10624    1.389608    0.000000   -1.181452  \n",
       "10625    0.000000    0.000000    0.982198  \n",
       "10626   -0.481567    0.000000    0.000000  \n",
       "10627    0.087274   -0.151546    0.036337  \n",
       "\n",
       "[10628 rows x 38 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 10 Protein HGNC Names: ['CD177', 'PEAR1', 'FOXC2', 'CRYAB', 'PSG6', 'COL11A1', 'CNN1', 'SUSD5', 'GPC3', 'CCDC8']\n",
      "Top 10 Protein BMGC IDs: ['BMGC_PT011490', 'BMGC_PT013430', 'BMGC_PT014441', 'BMGC_PT016585', 'BMGC_PT016638', 'BMGC_PT017666', 'BMGC_PT026179', 'BMGC_PT033081', 'BMGC_PT039265', 'BMGC_PT041113', 'BMGC_PT052721', 'BMGC_PT052813', 'BMGC_PT054064', 'BMGC_PT057023', 'BMGC_PT060207', 'BMGC_PT061016', 'BMGC_PT061565', 'BMGC_PT072419', 'BMGC_PT072688', 'BMGC_PT074012', 'BMGC_PT074354', 'BMGC_PT074526', 'BMGC_PT074762', 'BMGC_PT075119', 'BMGC_PT077222', 'BMGC_PT089610', 'BMGC_PT090148', 'BMGC_PT090387', 'BMGC_PT090530', 'BMGC_PT090551', 'BMGC_PT090573', 'BMGC_PT090842', 'BMGC_PT091132', 'BMGC_PT091507', 'BMGC_PT092525', 'BMGC_PT094911', 'BMGC_PT096554', 'BMGC_PT099706', 'BMGC_PT102483', 'BMGC_PT102715', 'BMGC_PT102986', 'BMGC_PT103193', 'BMGC_PT105753', 'BMGC_PT113331', 'BMGC_PT114711', 'BMGC_PT114837', 'BMGC_PT116582']\n",
      "Top 10 Protein BMGC LLM Name IDs: ['NCBI Gene ID:26032.0 or RefSeq ID:NP_056366.1 or Sushi domain-containing protein 5 or NCBI Gene ID:26032 or ENSP00000308727.3 or ENSP00000308727 or SUSD5 or UniProt ID:O60279 or RefSeq ID:NP_056366', 'CRYAB or RefSeq ID:NP_001876 or RefSeq ID:NP_001876.1 or NCBI Gene ID:1410.0 or RefSeq ID:NP_001276736 or RefSeq ID:NP_001355174 or RefSeq ID:NP_001355174.1 or Alpha-crystallin B chain (Alpha(B)-crystallin) (Heat shock protein beta-5) (HspB5) (Heat shock protein family B member 5) (Renal carcinoma antigen NY-REN-27) (Rosenthal fiber component) or RefSeq ID:XP_054223672.1 or RefSeq ID:NP_001276737.1 or ENSP00000227251 or ENSP00000433560 or ENSP00000434247 or ENSP00000435931 or ENSP00000436051 or ENSP00000436089 or ENSP00000437149 or ENSP00000483554 or ENSP00000498735 or ENSP00000499082 or UniProt ID:P02511 or RefSeq ID:XP_011540910 or NCBI Gene ID:1410 or RefSeq ID:NP_001276736.1 or ENSP00000227251.3 or ENSP00000433560.1 or ENSP00000434247.1 or ENSP00000435931.2 or ENSP00000436051.1 or ENSP00000436089.2 or ENSP00000437149.1 or ENSP00000483554.1 or ENSP00000498735.1 or ENSP00000499082.1 or RefSeq ID:NP_001276737', 'ENSP00000302551.6 or ENSP00000351163.2 or ENSP00000359114.3 or ENSP00000426533.1 or RefSeq ID:NP_542196 or RefSeq ID:NP_542197.3 or RefSeq ID:NP_001845.3 or COL11A1 or NCBI Gene ID:1301.0 or RefSeq ID:NP_001177638 or RefSeq ID:NP_542197 or NCBI Gene ID:1301 or RefSeq ID:NP_542196.2 or Collagen alpha-1(XI) chain or UniProt ID:P12107 or RefSeq ID:NP_001845 or ENSP00000302551 or ENSP00000351163 or ENSP00000359114 or ENSP00000426533 or RefSeq ID:NP_001177638.1', 'ENSP00000359854.3 or ENSP00000377836.2 or ENSP00000486325.1 or RefSeq ID:NP_001158091 or Glypican-3 (GTR2-2) (Intestinal protein OCI-5) (MXR7) [Cleaved into: Glypican-3 alpha subunit or Glypican-3 beta subunit] or NCBI Gene ID:2719.0 or RefSeq ID:NP_001158089.1 or NCBI Gene ID:2719 or RefSeq ID:NP_004475 or GPC3 or RefSeq ID:NP_001158091.1 or RefSeq ID:NP_004475.1 or UniProt ID:P51654 or ENSP00000359854 or ENSP00000377836 or ENSP00000486325 or RefSeq ID:NP_001158089', 'Calponin-1 (Basic calponin) (Calponin H1, smooth muscle) or UniProt ID:P51911 or ENSP00000252456.1 or RefSeq ID:NP_001290.2 or NCBI Gene ID:1264 or RefSeq ID:NP_001290 or CNN1 or NCBI Gene ID:1264.0 or ENSP00000252456', 'ENSP00000187910.3 or ENSP00000292125.2 or RefSeq ID:NP_001027020 or RefSeq ID:NP_001027020.1 or NCBI Gene ID:5675.0 or UniProt ID:Q00889 or NCBI Gene ID:5675 or RefSeq ID:NP_002773 or RefSeq ID:NP_002773.1 or Pregnancy-specific beta-1-glycoprotein 6 (PS-beta-G-6) (PSBG-6) (Pregnancy-specific glycoprotein 6) (Pregnancy-specific beta-1-glycoprotein 10) (PS-beta-G-10) (PSBG-10) (Pregnancy-specific glycoprotein 10) (Pregnancy-specific beta-1-glycoprotein 12) (PS-beta-G-12) (PSBG-12) (Pregnancy-specific glycoprotein 12) or PSG6 or ENSP00000187910 or ENSP00000292125', 'NCBI Gene ID:375033 or NCBI Gene ID:375033.0 or ENSP00000292357.7 or ENSP00000344465.3 or Platelet endothelial aggregation receptor 1 (hPEAR1) (Multiple epidermal growth factor-like domains protein 12) (Multiple EGF-like domains protein 12) or RefSeq ID:XP_011507812.1 or RefSeq ID:NP_001340612 or RefSeq ID:XP_047275678.1 or UniProt ID:Q5VY43 or RefSeq ID:XP_005245198.1 or RefSeq ID:NP_001340611 or ENSP00000292357 or ENSP00000344465 or RefSeq ID:XP_016856725.1 or RefSeq ID:NP_001073940.1 or RefSeq ID:NP_001073940 or PEAR1', 'NCBI Gene ID:57126 or UniProt ID:Q8N6Q3 or CD177 or CD177 antigen (Human neutrophil alloantigen 2a) (HNA-2a) (NB1 glycoprotein) (NB1 GP) (Polycythemia rubra vera protein 1) (PRV-1) (CD antigen CD177) or ENSP00000367251.2 or ENSP00000367251 or NCBI Gene ID:57126.0', 'NCBI Gene ID:2303 or FOXC2 or NCBI Gene ID:2303.0 or RefSeq ID:NP_005242 or ENSP00000497759 or RefSeq ID:NP_005242.1 or Forkhead box protein C2 (Forkhead-related protein FKHL14) (Mesenchyme fork head protein 1) (MFH-1 protein) (Transcription factor FKH-14) or ENSP00000497759.1 or UniProt ID:Q99958', 'ENSP00000303158 or ENSP00000303158.3 or CCDC8 or RefSeq ID:NP_114429 or NCBI Gene ID:83987.0 or UniProt ID:Q9H0W5 or NCBI Gene ID:83987 or Coiled-coil domain-containing protein 8 or RefSeq ID:NP_114429.2', 'GPC3 or ENSP00000385307.3 or NCBI Gene ID:2719.0 or ENSP00000385307', 'ENSP00000385736.3 or NCBI Gene ID:5675.0 or PSG6 or ENSP00000385736', 'ENSP00000389742.1 or NCBI Gene ID:375033.0 or ENSP00000389742 or PEAR1', 'ENSP00000397870 or ENSP00000397870.1 or NCBI Gene ID:375033.0 or PEAR1', 'NCBI Gene ID:26032.0 or SUSD5 or ENSP00000406475.1 or ENSP00000406475', 'ENSP00000408640 or ENSP00000408640.1 or COL11A1 or NCBI Gene ID:1301.0', 'ENSP00000410177.1 or NCBI Gene ID:1301.0 or ENSP00000410177 or COL11A1', 'CRYAB or ENSP00000431754 or NCBI Gene ID:1410.0 or ENSP00000431754.1', 'CRYAB or ENSP00000432182 or ENSP00000432182.1 or NCBI Gene ID:1410.0', 'CRYAB or NCBI Gene ID:1410.0 or ENSP00000434269.1 or ENSP00000434269', 'CRYAB or ENSP00000434793 or ENSP00000434793.1 or NCBI Gene ID:1410.0', 'CRYAB or NCBI Gene ID:1410.0 or ENSP00000435046 or ENSP00000435046.1 or RefSeq ID:NP_001317308.1 or RefSeq ID:NP_001317308 or NCBI Gene ID:1410', 'CRYAB or NCBI Gene ID:1410.0 or ENSP00000435411 or RefSeq ID:NP_001355175.1 or NCBI Gene ID:1410 or RefSeq ID:NP_001355175 or ENSP00000435411.1', 'CRYAB or ENSP00000435960.1 or NCBI Gene ID:1410.0 or ENSP00000435960', 'ENSP00000442031.1 or NCBI Gene ID:1264 or RefSeq ID:NP_001295270.1 or CNN1 or ENSP00000442031 or NCBI Gene ID:1264.0', 'NCBI Gene ID:1264.0 or ENSP00000466886.1 or CNN1 or ENSP00000466886', 'NCBI Gene ID:1264.0 or ENSP00000467701 or CNN1 or ENSP00000467701.1', 'ENSP00000468053.1 or ENSP00000468053 or CNN1 or NCBI Gene ID:1264.0', 'RefSeq ID:NP_001295271.1 or ENSP00000468259 or NCBI Gene ID:1264 or CNN1 or RefSeq ID:NP_001295271 or NCBI Gene ID:1264.0 or RefSeq ID:NP_001295270 or ENSP00000468259.1', 'NCBI Gene ID:1264.0 or ENSP00000468293.1 or ENSP00000468293 or CNN1', 'ENSP00000468326 or ENSP00000468326.1 or NCBI Gene ID:1264.0 or CNN1', 'NCBI Gene ID:1264.0 or ENSP00000468737.1 or ENSP00000468737 or CNN1', 'ENSP00000469423 or NCBI Gene ID:5675.0 or PSG6 or ENSP00000469423.1', 'ENSP00000470412 or NCBI Gene ID:5675.0 or PSG6 or ENSP00000470412.1', 'NCBI Gene ID:5675.0 or PSG6 or ENSP00000472935.1 or ENSP00000472935', 'ENSP00000479536 or RefSeq ID:NP_065139 or NCBI Gene ID:57126 or CD177 or RefSeq ID:NP_065139.2 or ENSP00000479536.1 or NCBI Gene ID:57126.0', 'NCBI Gene ID:57126.0 or ENSP00000483817 or ENSP00000483817.1 or CD177', 'ENSP00000489428.1 or NCBI Gene ID:1301.0 or ENSP00000489428 or COL11A1', 'ENSP00000493821 or ENSP00000493821.1 or COL11A1 or NCBI Gene ID:1301.0', 'ENSP00000494179 or ENSP00000494179.1 or COL11A1 or NCBI Gene ID:1301.0', 'ENSP00000494583.1 or NCBI Gene ID:1301.0 or ENSP00000494583 or COL11A1', 'NCBI Gene ID:1301.0 or ENSP00000494909.1 or ENSP00000494909 or COL11A1', 'CRYAB or NCBI Gene ID:1410.0 or ENSP00000498749 or ENSP00000498749.1', 'GPC3 or ENSP00000508554 or NCBI Gene ID:2719.0 or ENSP00000508554.1', 'GPC3 or ENSP00000510280.1 or NCBI Gene ID:2719.0 or ENSP00000510280', 'ENSP00000510438.1 or NCBI Gene ID:2719 or GPC3 or RefSeq ID:NP_001158090 or RefSeq ID:NP_001158090.1 or NCBI Gene ID:2719.0 or ENSP00000510438', 'NCBI Gene ID:83987.0 or CCDC8 or ENSP00000513420 or ENSP00000513420.1']\n"
     ]
    }
   ],
   "source": [
    "# Load protein mapping data with BioMedGraphica IDs, Uniprot IDs, and HGNC symbols\n",
    "bmg_protein_all_df = pd.read_csv('./data/BioMedGraphica-Conn/Entity/Protein/BioMedGraphica_Conn_Protein.csv')\n",
    "bmg_protein_all_df = bmg_protein_all_df[['BioMedGraphica_Conn_ID', 'Uniprot_ID', 'HGNC_Symbol']].copy()\n",
    "display(bmg_protein_all_df)\n",
    "\n",
    "# Map sample IDs in raw protein data to standardized format\n",
    "raw_protein_map_df = raw_protein_df.rename(columns=protein_map_dict)\n",
    "\n",
    "# Join protein expression data with HGNC symbol annotations\n",
    "symbol_protein_map_df = pd.merge(raw_protein_map_df, bmg_protein_all_df, left_on='Uniprot_Acc', right_on='Uniprot_ID', how='inner')\n",
    "\n",
    "# Reorganize columns: protein identifiers first, then sample expression values\n",
    "identifier_cols = ['Uniprot_ID', 'Uniprot_Acc', 'HGNC_Symbol']\n",
    "expression_cols = sorted(set(symbol_protein_map_df.columns) - set(identifier_cols))\n",
    "symbol_protein_map_df = symbol_protein_map_df[identifier_cols + expression_cols]\n",
    "\n",
    "# Only include samples that actually exist in symbol_protein_map_df columns\n",
    "available_protein_samples = [sample for sample in remaining_dti_samples if sample in symbol_protein_map_df.columns]\n",
    "print(f\"Total remaining_dti_samples: {len(remaining_dti_samples)}\")\n",
    "print(f\"Available samples in symbol_protein_map_df: {len(available_protein_samples)}\")\n",
    "print(f\"Missing samples: {len(remaining_dti_samples) - len(available_protein_samples)}\")\n",
    "\n",
    "# Extract protein data for DTI samples with HGNC symbols\n",
    "dti_protein_df = symbol_protein_map_df[['HGNC_Symbol'] + sorted(available_protein_samples)].copy()\n",
    "\n",
    "# Handle multiple HGNC symbols per protein (semicolon-separated)\n",
    "dti_protein_df = dti_protein_df.assign(\n",
    "    HGNC_Symbol=dti_protein_df['HGNC_Symbol'].str.split(';')\n",
    ").explode('HGNC_Symbol')\n",
    "\n",
    "# Clean up HGNC symbols: remove whitespace and filter out empty entries\n",
    "dti_protein_df['HGNC_Symbol'] = dti_protein_df['HGNC_Symbol'].str.strip()\n",
    "dti_protein_df = dti_protein_df[\n",
    "    dti_protein_df['HGNC_Symbol'].notna() & \n",
    "    (dti_protein_df['HGNC_Symbol'] != '')\n",
    "].reset_index(drop=True)\n",
    "\n",
    "display(dti_protein_df)\n",
    "\n",
    "def extract_dti_pt_info(dti_protein_df, bmgc_protein_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k=10):\n",
    "    # Check if sample_ach_name exists in dti_protein_df columns\n",
    "    if sample_ach_name not in dti_protein_df.columns:\n",
    "        return \"non-existed\", \"non-existed\", \"non-existed\"\n",
    "    # Extract the top k highest values for the given sample name\n",
    "    top_k_proteins = dti_protein_df.nlargest(k, sample_ach_name)[['HGNC_Symbol', sample_ach_name]]\n",
    "    # Sort the top k proteins by their values in descending order\n",
    "    top_k_proteins = top_k_proteins.sort_values(by=sample_ach_name, ascending=False).reset_index(drop=True)\n",
    "    top_k_protein_hgnc_name_list = top_k_proteins['HGNC_Symbol'].tolist()\n",
    "    # Merge with the bmgc_protein_df to get the BioMedGraphica_Conn_ID\n",
    "    bmgc_protein_df = bmgc_protein_df[['BioMedGraphica_Conn_ID', 'HGNC_Symbol']].copy()\n",
    "    top_k_bmgc_protein_df = pd.merge(bmgc_protein_df, top_k_proteins, left_on='HGNC_Symbol', right_on='HGNC_Symbol', how='inner')\n",
    "    # Get the corresponding protein information\n",
    "    top_k_bmgc_protein_info_df = pd.merge(top_k_bmgc_protein_df, bmgc_protein_llmnameid_combined_df, left_on='BioMedGraphica_Conn_ID', right_on='BioMedGraphica_Conn_ID', how='inner').drop(columns=['BioMedGraphica_ID', sample_ach_name])\n",
    "    top_k_protein_bmgc_id_list = top_k_bmgc_protein_info_df['BioMedGraphica_Conn_ID'].tolist()\n",
    "    # Replace both \"|\" and \";\" with \" or \"\n",
    "    top_k_protein_bmgc_llmnameid_list = top_k_bmgc_protein_info_df['Names_and_IDs'].replace([r' \\| ', r';'], ' or ', regex=True).tolist()\n",
    "    return top_k_protein_hgnc_name_list, top_k_protein_bmgc_id_list, top_k_protein_bmgc_llmnameid_list\n",
    "\n",
    "# Example usage\n",
    "sample_ach_name = 'ACH-000008'\n",
    "k = 10\n",
    "top_k_protein_hgnc_name_list, top_k_protein_bmgc_id_list, top_k_protein_bmgc_llmnameid_list = extract_dti_pt_info(dti_protein_df, bmgc_protein_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "print(f\"Top {k} Protein HGNC Names:\", top_k_protein_hgnc_name_list)\n",
    "print(f\"Top {k} Protein BMGC IDs:\", top_k_protein_bmgc_id_list)\n",
    "print(f\"Top {k} Protein BMGC LLM Name IDs:\", top_k_protein_bmgc_llmnameid_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 461,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Drug Protein Index: [603176]\n",
      "Drug Protein BMGC ID: ['BMGC_PT018662']\n"
     ]
    }
   ],
   "source": [
    "def extract_drug_protein(selected_sample_drug_bmgc_id, edge_index, \n",
    "                    node_index_df, nodeid_index_dict, index_nodeid_dict):\n",
    "    # Extract the index based on the selected drug BMGC ID\n",
    "    sample_drug_bmgc_id_index = nodeid_index_dict[selected_sample_drug_bmgc_id]\n",
    "    # Find incoming edges (source nodes that point to the drug)\n",
    "    incoming_mask = edge_index[1, :] == sample_drug_bmgc_id_index\n",
    "    incoming_source_nodes = edge_index[0, incoming_mask]\n",
    "    # Find outgoing edges (target nodes that the drug points to)\n",
    "    outgoing_mask = edge_index[0, :] == sample_drug_bmgc_id_index\n",
    "    outgoing_target_nodes = edge_index[1, outgoing_mask]\n",
    "    # Combine all neighbor nodes (both incoming and outgoing)\n",
    "    drug_related_nodes = np.concatenate([incoming_source_nodes, outgoing_target_nodes])\n",
    "    unique_drug_related_nodes = np.unique(drug_related_nodes)\n",
    "    # Get protein node index\n",
    "    protein_node_index_df = node_index_df[node_index_df['Type'] == 'Protein']\n",
    "    protein_node_index_list = protein_node_index_df['Index'].tolist()\n",
    "    # Filter to get only protein nodes directly connected to the drug\n",
    "    drug_protein_index = sorted(\n",
    "        list(set(unique_drug_related_nodes) & set(protein_node_index_list))\n",
    "    )\n",
    "    # Map protein index to BMGC id\n",
    "    drug_protein_bmgc_id = [index_nodeid_dict[i] for i in drug_protein_index]\n",
    "    return drug_protein_index, drug_protein_bmgc_id\n",
    "\n",
    "# Example usage\n",
    "selected_sample_drug_bmgc_id = 'BMGC_DG00001'\n",
    "\n",
    "drug_protein_index, drug_protein_bmgc_id = extract_drug_protein(\n",
    "    selected_sample_drug_bmgc_id=selected_sample_drug_bmgc_id,\n",
    "    edge_index=edge_index,\n",
    "    node_index_df=nodes_index_data,\n",
    "    nodeid_index_dict=node_index_dict,\n",
    "    index_nodeid_dict=index_node_dict\n",
    ")\n",
    "\n",
    "print(\"Drug Protein Index:\", drug_protein_index)\n",
    "print(\"Drug Protein BMGC ID:\", drug_protein_bmgc_id)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Formulate the drug related protein json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 436,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:   1%|          | 1/172 [00:00<00:36,  4.71it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Drug BMGC ID: BMGC_DG00030, Protein Index: [595431], Protein BMGC ID: ['BMGC_PT010917']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:   1%|          | 2/172 [00:02<04:47,  1.69s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Drug BMGC ID: BMGC_DG00141, Protein Index: [594482, 594534, 594581, 594618, 594741, 594837, 594937, 594951, 595044, 595057, 595075, 595084, 595153, 595177, 595206, 595232, 595271, 595371, 595372, 595431, 595586, 595593, 595614, 595628, 596008, 596038, 596224, 596445, 596593, 596641, 596723, 596770, 596793, 596798, 596969, 597078, 597080, 597149, 597367, 597541, 597584, 597593, 597696, 597700, 597755, 597756, 598028, 598158, 598194, 598195, 598196, 598270, 598271, 598309, 598310, 598311, 598321, 598372, 598379, 598406, 598459, 598473, 598501, 598505, 598851, 598906, 598936, 598996, 599105, 599139, 599185, 599228, 599255, 599326, 599349, 599430, 599438, 599509, 599533, 599571, 599576, 599601, 599612, 599623, 599633, 599680, 599681, 599752, 599762, 599849, 599857, 599858, 599918, 599951, 599952, 599953, 599954, 599964, 599980, 600018, 600144, 600217, 600263, 600362, 600395, 600396, 600408, 600423, 600425, 600444, 600559, 600594, 600595, 600596, 600607, 600608, 600609, 600610, 600611, 600672, 600673, 600702, 600703, 600704, 600732, 600854, 600857, 600875, 600893, 600948, 600949, 600969, 600970, 601033, 601038, 601070, 601095, 601133, 601134, 601138, 601155, 601156, 601157, 601172, 601176, 601228, 601229, 601246, 601247, 601256, 601257, 601303, 601304, 601305, 601306, 601307, 601472, 601473, 601480, 601845, 601888, 601892, 601897, 602032, 602158, 602160, 602161, 602162, 602270, 602318, 602319, 602320, 602399, 602401, 602424, 602432, 602501, 602512, 602545, 602587, 602611, 602636, 602688, 602995, 602996, 603005, 603198, 603208, 603240, 603244, 603380, 603384, 603434, 603439, 603487, 603529, 603603, 603609, 603667, 603678, 603684, 603775, 604436, 604470, 604509, 604527, 604563, 604627, 604727, 604734, 604786, 604789, 605030, 605031, 605049, 605072, 605082, 605132, 605220, 605240, 605252, 605795, 606222, 607279, 607351, 608085, 608248, 609221, 610395, 611451, 612610, 612837, 613559, 613672, 614320, 614416, 615403, 615485, 615530, 615899, 616014, 616035, 616061, 616244, 616293, 616319, 616398, 616614, 616711, 617320, 617411, 617481, 617650, 618378, 618466, 618515, 618655, 619011, 619247, 619399, 619403, 619445, 619506, 619528, 619712, 619746, 620195, 620487, 620659, 621181, 621387, 621816, 622448, 622809, 623000, 623202, 623282, 623599, 623669, 623672, 623940, 624081, 624389, 624587, 624725, 624771, 624783, 625049, 625174, 625281, 625529, 625580, 625773, 625889, 625906, 625955, 626096, 626109, 626157, 626951, 627042, 627165, 627247, 627703, 627831, 627863, 627873, 628049, 628509, 628684, 628764, 628836, 629193, 629253, 629256, 629318, 629507, 629515, 630024, 630096, 630118, 630573, 630814, 630874, 630896, 631024, 631210, 631541, 631649, 631694, 631707, 631761, 631775, 631838, 631926, 631932, 631951, 632007, 632304, 632521, 632593, 632665, 632720, 632748], Protein BMGC ID: ['BMGC_PT009968', 'BMGC_PT010020', 'BMGC_PT010067', 'BMGC_PT010104', 'BMGC_PT010227', 'BMGC_PT010323', 'BMGC_PT010423', 'BMGC_PT010437', 'BMGC_PT010530', 'BMGC_PT010543', 'BMGC_PT010561', 'BMGC_PT010570', 'BMGC_PT010639', 'BMGC_PT010663', 'BMGC_PT010692', 'BMGC_PT010718', 'BMGC_PT010757', 'BMGC_PT010857', 'BMGC_PT010858', 'BMGC_PT010917', 'BMGC_PT011072', 'BMGC_PT011079', 'BMGC_PT011100', 'BMGC_PT011114', 'BMGC_PT011494', 'BMGC_PT011524', 'BMGC_PT011710', 'BMGC_PT011931', 'BMGC_PT012079', 'BMGC_PT012127', 'BMGC_PT012209', 'BMGC_PT012256', 'BMGC_PT012279', 'BMGC_PT012284', 'BMGC_PT012455', 'BMGC_PT012564', 'BMGC_PT012566', 'BMGC_PT012635', 'BMGC_PT012853', 'BMGC_PT013027', 'BMGC_PT013070', 'BMGC_PT013079', 'BMGC_PT013182', 'BMGC_PT013186', 'BMGC_PT013241', 'BMGC_PT013242', 'BMGC_PT013514', 'BMGC_PT013644', 'BMGC_PT013680', 'BMGC_PT013681', 'BMGC_PT013682', 'BMGC_PT013756', 'BMGC_PT013757', 'BMGC_PT013795', 'BMGC_PT013796', 'BMGC_PT013797', 'BMGC_PT013807', 'BMGC_PT013858', 'BMGC_PT013865', 'BMGC_PT013892', 'BMGC_PT013945', 'BMGC_PT013959', 'BMGC_PT013987', 'BMGC_PT013991', 'BMGC_PT014337', 'BMGC_PT014392', 'BMGC_PT014422', 'BMGC_PT014482', 'BMGC_PT014591', 'BMGC_PT014625', 'BMGC_PT014671', 'BMGC_PT014714', 'BMGC_PT014741', 'BMGC_PT014812', 'BMGC_PT014835', 'BMGC_PT014916', 'BMGC_PT014924', 'BMGC_PT014995', 'BMGC_PT015019', 'BMGC_PT015057', 'BMGC_PT015062', 'BMGC_PT015087', 'BMGC_PT015098', 'BMGC_PT015109', 'BMGC_PT015119', 'BMGC_PT015166', 'BMGC_PT015167', 'BMGC_PT015238', 'BMGC_PT015248', 'BMGC_PT015335', 'BMGC_PT015343', 'BMGC_PT015344', 'BMGC_PT015404', 'BMGC_PT015437', 'BMGC_PT015438', 'BMGC_PT015439', 'BMGC_PT015440', 'BMGC_PT015450', 'BMGC_PT015466', 'BMGC_PT015504', 'BMGC_PT015630', 'BMGC_PT015703', 'BMGC_PT015749', 'BMGC_PT015848', 'BMGC_PT015881', 'BMGC_PT015882', 'BMGC_PT015894', 'BMGC_PT015909', 'BMGC_PT015911', 'BMGC_PT015930', 'BMGC_PT016045', 'BMGC_PT016080', 'BMGC_PT016081', 'BMGC_PT016082', 'BMGC_PT016093', 'BMGC_PT016094', 'BMGC_PT016095', 'BMGC_PT016096', 'BMGC_PT016097', 'BMGC_PT016158', 'BMGC_PT016159', 'BMGC_PT016188', 'BMGC_PT016189', 'BMGC_PT016190', 'BMGC_PT016218', 'BMGC_PT016340', 'BMGC_PT016343', 'BMGC_PT016361', 'BMGC_PT016379', 'BMGC_PT016434', 'BMGC_PT016435', 'BMGC_PT016455', 'BMGC_PT016456', 'BMGC_PT016519', 'BMGC_PT016524', 'BMGC_PT016556', 'BMGC_PT016581', 'BMGC_PT016619', 'BMGC_PT016620', 'BMGC_PT016624', 'BMGC_PT016641', 'BMGC_PT016642', 'BMGC_PT016643', 'BMGC_PT016658', 'BMGC_PT016662', 'BMGC_PT016714', 'BMGC_PT016715', 'BMGC_PT016732', 'BMGC_PT016733', 'BMGC_PT016742', 'BMGC_PT016743', 'BMGC_PT016789', 'BMGC_PT016790', 'BMGC_PT016791', 'BMGC_PT016792', 'BMGC_PT016793', 'BMGC_PT016958', 'BMGC_PT016959', 'BMGC_PT016966', 'BMGC_PT017331', 'BMGC_PT017374', 'BMGC_PT017378', 'BMGC_PT017383', 'BMGC_PT017518', 'BMGC_PT017644', 'BMGC_PT017646', 'BMGC_PT017647', 'BMGC_PT017648', 'BMGC_PT017756', 'BMGC_PT017804', 'BMGC_PT017805', 'BMGC_PT017806', 'BMGC_PT017885', 'BMGC_PT017887', 'BMGC_PT017910', 'BMGC_PT017918', 'BMGC_PT017987', 'BMGC_PT017998', 'BMGC_PT018031', 'BMGC_PT018073', 'BMGC_PT018097', 'BMGC_PT018122', 'BMGC_PT018174', 'BMGC_PT018481', 'BMGC_PT018482', 'BMGC_PT018491', 'BMGC_PT018684', 'BMGC_PT018694', 'BMGC_PT018726', 'BMGC_PT018730', 'BMGC_PT018866', 'BMGC_PT018870', 'BMGC_PT018920', 'BMGC_PT018925', 'BMGC_PT018973', 'BMGC_PT019015', 'BMGC_PT019089', 'BMGC_PT019095', 'BMGC_PT019153', 'BMGC_PT019164', 'BMGC_PT019170', 'BMGC_PT019261', 'BMGC_PT019922', 'BMGC_PT019956', 'BMGC_PT019995', 'BMGC_PT020013', 'BMGC_PT020049', 'BMGC_PT020113', 'BMGC_PT020213', 'BMGC_PT020220', 'BMGC_PT020272', 'BMGC_PT020275', 'BMGC_PT020516', 'BMGC_PT020517', 'BMGC_PT020535', 'BMGC_PT020558', 'BMGC_PT020568', 'BMGC_PT020618', 'BMGC_PT020706', 'BMGC_PT020726', 'BMGC_PT020738', 'BMGC_PT021281', 'BMGC_PT021708', 'BMGC_PT022765', 'BMGC_PT022837', 'BMGC_PT023571', 'BMGC_PT023734', 'BMGC_PT024707', 'BMGC_PT025881', 'BMGC_PT026937', 'BMGC_PT028096', 'BMGC_PT028323', 'BMGC_PT029045', 'BMGC_PT029158', 'BMGC_PT029806', 'BMGC_PT029902', 'BMGC_PT030889', 'BMGC_PT030971', 'BMGC_PT031016', 'BMGC_PT031385', 'BMGC_PT031500', 'BMGC_PT031521', 'BMGC_PT031547', 'BMGC_PT031730', 'BMGC_PT031779', 'BMGC_PT031805', 'BMGC_PT031884', 'BMGC_PT032100', 'BMGC_PT032197', 'BMGC_PT032806', 'BMGC_PT032897', 'BMGC_PT032967', 'BMGC_PT033136', 'BMGC_PT033864', 'BMGC_PT033952', 'BMGC_PT034001', 'BMGC_PT034141', 'BMGC_PT034497', 'BMGC_PT034733', 'BMGC_PT034885', 'BMGC_PT034889', 'BMGC_PT034931', 'BMGC_PT034992', 'BMGC_PT035014', 'BMGC_PT035198', 'BMGC_PT035232', 'BMGC_PT035681', 'BMGC_PT035973', 'BMGC_PT036145', 'BMGC_PT036667', 'BMGC_PT036873', 'BMGC_PT037302', 'BMGC_PT037934', 'BMGC_PT038295', 'BMGC_PT038486', 'BMGC_PT038688', 'BMGC_PT038768', 'BMGC_PT039085', 'BMGC_PT039155', 'BMGC_PT039158', 'BMGC_PT039426', 'BMGC_PT039567', 'BMGC_PT039875', 'BMGC_PT040073', 'BMGC_PT040211', 'BMGC_PT040257', 'BMGC_PT040269', 'BMGC_PT040535', 'BMGC_PT040660', 'BMGC_PT040767', 'BMGC_PT041015', 'BMGC_PT041066', 'BMGC_PT041259', 'BMGC_PT041375', 'BMGC_PT041392', 'BMGC_PT041441', 'BMGC_PT041582', 'BMGC_PT041595', 'BMGC_PT041643', 'BMGC_PT042437', 'BMGC_PT042528', 'BMGC_PT042651', 'BMGC_PT042733', 'BMGC_PT043189', 'BMGC_PT043317', 'BMGC_PT043349', 'BMGC_PT043359', 'BMGC_PT043535', 'BMGC_PT043995', 'BMGC_PT044170', 'BMGC_PT044250', 'BMGC_PT044322', 'BMGC_PT044679', 'BMGC_PT044739', 'BMGC_PT044742', 'BMGC_PT044804', 'BMGC_PT044993', 'BMGC_PT045001', 'BMGC_PT045510', 'BMGC_PT045582', 'BMGC_PT045604', 'BMGC_PT046059', 'BMGC_PT046300', 'BMGC_PT046360', 'BMGC_PT046382', 'BMGC_PT046510', 'BMGC_PT046696', 'BMGC_PT047027', 'BMGC_PT047135', 'BMGC_PT047180', 'BMGC_PT047193', 'BMGC_PT047247', 'BMGC_PT047261', 'BMGC_PT047324', 'BMGC_PT047412', 'BMGC_PT047418', 'BMGC_PT047437', 'BMGC_PT047493', 'BMGC_PT047790', 'BMGC_PT048007', 'BMGC_PT048079', 'BMGC_PT048151', 'BMGC_PT048206', 'BMGC_PT048234']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:   2%|▏         | 3/172 [00:05<05:53,  2.09s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Drug BMGC ID: BMGC_DG00151, Protein Index: [594534, 594581, 594618, 594741, 594837, 594937, 594951, 595044, 595057, 595075, 595084, 595153, 595177, 595206, 595232, 595271, 595431, 595586, 595593, 595614, 596008, 596038, 596224, 596445, 596593, 596641, 596723, 596770, 596798, 596969, 597078, 597080, 597149, 597367, 597541, 597584, 597593, 597696, 597700, 597755, 597756, 598028, 598194, 598309, 598310, 598311, 598321, 598372, 598379, 598459, 598501, 598505, 598906, 598936, 599105, 599139, 599185, 599228, 599326, 599349, 599430, 599438, 599509, 599533, 599571, 599576, 599623, 599633, 599680, 599681, 599752, 599762, 599849, 599857, 599858, 599918, 599951, 599952, 599953, 599954, 599964, 599980, 600018, 600144, 600217, 600263, 600362, 600395, 600396, 600408, 600425, 600444, 600559, 600594, 600595, 600596, 600607, 600608, 600610, 600611, 600672, 600673, 600702, 600703, 600704, 600732, 600854, 600857, 600875, 600893, 600948, 600949, 600969, 600970, 601033, 601038, 601070, 601133, 601134, 601138, 601156, 601157, 601172, 601176, 601228, 601229, 601246, 601247, 601256, 601257, 601303, 601304, 601305, 601306, 601307, 601472, 601473, 601480, 601845, 601888, 601892, 601897, 602158, 602160, 602162, 602270, 602318, 602319, 602320, 602399, 602401, 602424, 602432, 602501, 602512, 602545, 602587, 602611, 602636, 602688, 602996, 603005, 603198, 603208, 603240, 603244, 603380, 603384, 603439, 603487, 603529, 603603, 603609, 603667, 603678, 603684, 603775, 604436, 604470, 604509, 604527, 604563, 604627, 604727, 604734, 604786, 604789, 605030, 605031, 605049, 605072, 605082, 605132, 605240, 605252, 605795, 606222, 607351, 608085, 608248, 609221, 610395, 611451, 612610, 612837, 613559, 613672, 614320, 614416, 615403, 615485, 615530, 615899, 616014, 616035, 616061, 616244, 616293, 616319, 616398, 616614, 616711, 617320, 617411, 617481, 617650, 618378, 618466, 618515, 618655, 619011, 619247, 619399, 619403, 619445, 619506, 619528, 619712, 619746, 620195, 620487, 620659, 621181, 621387, 621816, 622448, 622809, 623000, 623202, 623282, 623599, 623669, 623672, 623940, 624081, 624389, 624587, 624725, 624771, 624783, 625049, 625174, 625281, 625529, 625773, 625889, 625906, 625955, 626096, 626109, 626157, 626951, 627042, 627165, 627247, 627703, 627831, 627863, 627873, 628049, 628509, 628684, 628764, 628836, 629193, 629253, 629256, 629318, 629507, 630024, 630096, 630118, 630573, 630814, 630896, 631210, 631541, 631649, 631694, 631707, 631761, 631838, 631926, 631932, 631951, 632007, 632304, 632521, 632593, 632665, 632720, 632748], Protein BMGC ID: ['BMGC_PT010020', 'BMGC_PT010067', 'BMGC_PT010104', 'BMGC_PT010227', 'BMGC_PT010323', 'BMGC_PT010423', 'BMGC_PT010437', 'BMGC_PT010530', 'BMGC_PT010543', 'BMGC_PT010561', 'BMGC_PT010570', 'BMGC_PT010639', 'BMGC_PT010663', 'BMGC_PT010692', 'BMGC_PT010718', 'BMGC_PT010757', 'BMGC_PT010917', 'BMGC_PT011072', 'BMGC_PT011079', 'BMGC_PT011100', 'BMGC_PT011494', 'BMGC_PT011524', 'BMGC_PT011710', 'BMGC_PT011931', 'BMGC_PT012079', 'BMGC_PT012127', 'BMGC_PT012209', 'BMGC_PT012256', 'BMGC_PT012284', 'BMGC_PT012455', 'BMGC_PT012564', 'BMGC_PT012566', 'BMGC_PT012635', 'BMGC_PT012853', 'BMGC_PT013027', 'BMGC_PT013070', 'BMGC_PT013079', 'BMGC_PT013182', 'BMGC_PT013186', 'BMGC_PT013241', 'BMGC_PT013242', 'BMGC_PT013514', 'BMGC_PT013680', 'BMGC_PT013795', 'BMGC_PT013796', 'BMGC_PT013797', 'BMGC_PT013807', 'BMGC_PT013858', 'BMGC_PT013865', 'BMGC_PT013945', 'BMGC_PT013987', 'BMGC_PT013991', 'BMGC_PT014392', 'BMGC_PT014422', 'BMGC_PT014591', 'BMGC_PT014625', 'BMGC_PT014671', 'BMGC_PT014714', 'BMGC_PT014812', 'BMGC_PT014835', 'BMGC_PT014916', 'BMGC_PT014924', 'BMGC_PT014995', 'BMGC_PT015019', 'BMGC_PT015057', 'BMGC_PT015062', 'BMGC_PT015109', 'BMGC_PT015119', 'BMGC_PT015166', 'BMGC_PT015167', 'BMGC_PT015238', 'BMGC_PT015248', 'BMGC_PT015335', 'BMGC_PT015343', 'BMGC_PT015344', 'BMGC_PT015404', 'BMGC_PT015437', 'BMGC_PT015438', 'BMGC_PT015439', 'BMGC_PT015440', 'BMGC_PT015450', 'BMGC_PT015466', 'BMGC_PT015504', 'BMGC_PT015630', 'BMGC_PT015703', 'BMGC_PT015749', 'BMGC_PT015848', 'BMGC_PT015881', 'BMGC_PT015882', 'BMGC_PT015894', 'BMGC_PT015911', 'BMGC_PT015930', 'BMGC_PT016045', 'BMGC_PT016080', 'BMGC_PT016081', 'BMGC_PT016082', 'BMGC_PT016093', 'BMGC_PT016094', 'BMGC_PT016096', 'BMGC_PT016097', 'BMGC_PT016158', 'BMGC_PT016159', 'BMGC_PT016188', 'BMGC_PT016189', 'BMGC_PT016190', 'BMGC_PT016218', 'BMGC_PT016340', 'BMGC_PT016343', 'BMGC_PT016361', 'BMGC_PT016379', 'BMGC_PT016434', 'BMGC_PT016435', 'BMGC_PT016455', 'BMGC_PT016456', 'BMGC_PT016519', 'BMGC_PT016524', 'BMGC_PT016556', 'BMGC_PT016619', 'BMGC_PT016620', 'BMGC_PT016624', 'BMGC_PT016642', 'BMGC_PT016643', 'BMGC_PT016658', 'BMGC_PT016662', 'BMGC_PT016714', 'BMGC_PT016715', 'BMGC_PT016732', 'BMGC_PT016733', 'BMGC_PT016742', 'BMGC_PT016743', 'BMGC_PT016789', 'BMGC_PT016790', 'BMGC_PT016791', 'BMGC_PT016792', 'BMGC_PT016793', 'BMGC_PT016958', 'BMGC_PT016959', 'BMGC_PT016966', 'BMGC_PT017331', 'BMGC_PT017374', 'BMGC_PT017378', 'BMGC_PT017383', 'BMGC_PT017644', 'BMGC_PT017646', 'BMGC_PT017648', 'BMGC_PT017756', 'BMGC_PT017804', 'BMGC_PT017805', 'BMGC_PT017806', 'BMGC_PT017885', 'BMGC_PT017887', 'BMGC_PT017910', 'BMGC_PT017918', 'BMGC_PT017987', 'BMGC_PT017998', 'BMGC_PT018031', 'BMGC_PT018073', 'BMGC_PT018097', 'BMGC_PT018122', 'BMGC_PT018174', 'BMGC_PT018482', 'BMGC_PT018491', 'BMGC_PT018684', 'BMGC_PT018694', 'BMGC_PT018726', 'BMGC_PT018730', 'BMGC_PT018866', 'BMGC_PT018870', 'BMGC_PT018925', 'BMGC_PT018973', 'BMGC_PT019015', 'BMGC_PT019089', 'BMGC_PT019095', 'BMGC_PT019153', 'BMGC_PT019164', 'BMGC_PT019170', 'BMGC_PT019261', 'BMGC_PT019922', 'BMGC_PT019956', 'BMGC_PT019995', 'BMGC_PT020013', 'BMGC_PT020049', 'BMGC_PT020113', 'BMGC_PT020213', 'BMGC_PT020220', 'BMGC_PT020272', 'BMGC_PT020275', 'BMGC_PT020516', 'BMGC_PT020517', 'BMGC_PT020535', 'BMGC_PT020558', 'BMGC_PT020568', 'BMGC_PT020618', 'BMGC_PT020726', 'BMGC_PT020738', 'BMGC_PT021281', 'BMGC_PT021708', 'BMGC_PT022837', 'BMGC_PT023571', 'BMGC_PT023734', 'BMGC_PT024707', 'BMGC_PT025881', 'BMGC_PT026937', 'BMGC_PT028096', 'BMGC_PT028323', 'BMGC_PT029045', 'BMGC_PT029158', 'BMGC_PT029806', 'BMGC_PT029902', 'BMGC_PT030889', 'BMGC_PT030971', 'BMGC_PT031016', 'BMGC_PT031385', 'BMGC_PT031500', 'BMGC_PT031521', 'BMGC_PT031547', 'BMGC_PT031730', 'BMGC_PT031779', 'BMGC_PT031805', 'BMGC_PT031884', 'BMGC_PT032100', 'BMGC_PT032197', 'BMGC_PT032806', 'BMGC_PT032897', 'BMGC_PT032967', 'BMGC_PT033136', 'BMGC_PT033864', 'BMGC_PT033952', 'BMGC_PT034001', 'BMGC_PT034141', 'BMGC_PT034497', 'BMGC_PT034733', 'BMGC_PT034885', 'BMGC_PT034889', 'BMGC_PT034931', 'BMGC_PT034992', 'BMGC_PT035014', 'BMGC_PT035198', 'BMGC_PT035232', 'BMGC_PT035681', 'BMGC_PT035973', 'BMGC_PT036145', 'BMGC_PT036667', 'BMGC_PT036873', 'BMGC_PT037302', 'BMGC_PT037934', 'BMGC_PT038295', 'BMGC_PT038486', 'BMGC_PT038688', 'BMGC_PT038768', 'BMGC_PT039085', 'BMGC_PT039155', 'BMGC_PT039158', 'BMGC_PT039426', 'BMGC_PT039567', 'BMGC_PT039875', 'BMGC_PT040073', 'BMGC_PT040211', 'BMGC_PT040257', 'BMGC_PT040269', 'BMGC_PT040535', 'BMGC_PT040660', 'BMGC_PT040767', 'BMGC_PT041015', 'BMGC_PT041259', 'BMGC_PT041375', 'BMGC_PT041392', 'BMGC_PT041441', 'BMGC_PT041582', 'BMGC_PT041595', 'BMGC_PT041643', 'BMGC_PT042437', 'BMGC_PT042528', 'BMGC_PT042651', 'BMGC_PT042733', 'BMGC_PT043189', 'BMGC_PT043317', 'BMGC_PT043349', 'BMGC_PT043359', 'BMGC_PT043535', 'BMGC_PT043995', 'BMGC_PT044170', 'BMGC_PT044250', 'BMGC_PT044322', 'BMGC_PT044679', 'BMGC_PT044739', 'BMGC_PT044742', 'BMGC_PT044804', 'BMGC_PT044993', 'BMGC_PT045510', 'BMGC_PT045582', 'BMGC_PT045604', 'BMGC_PT046059', 'BMGC_PT046300', 'BMGC_PT046382', 'BMGC_PT046696', 'BMGC_PT047027', 'BMGC_PT047135', 'BMGC_PT047180', 'BMGC_PT047193', 'BMGC_PT047247', 'BMGC_PT047324', 'BMGC_PT047412', 'BMGC_PT047418', 'BMGC_PT047437', 'BMGC_PT047493', 'BMGC_PT047790', 'BMGC_PT048007', 'BMGC_PT048079', 'BMGC_PT048151', 'BMGC_PT048206', 'BMGC_PT048234']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:   2%|▏         | 4/172 [00:08<06:19,  2.26s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Drug BMGC ID: BMGC_DG00254, Protein Index: [594534, 594581, 594618, 594741, 594837, 594937, 594951, 595044, 595057, 595075, 595084, 595153, 595177, 595206, 595232, 595271, 595431, 595586, 595593, 595614, 595628, 596008, 596038, 596224, 596445, 596593, 596641, 596723, 596770, 596798, 596969, 597078, 597080, 597149, 597367, 597541, 597584, 597593, 597696, 597700, 597755, 597756, 598028, 598087, 598194, 598195, 598309, 598310, 598311, 598321, 598372, 598379, 598459, 598501, 598505, 598906, 598936, 598996, 599105, 599139, 599185, 599228, 599326, 599349, 599430, 599438, 599509, 599533, 599571, 599576, 599623, 599633, 599680, 599681, 599752, 599762, 599849, 599857, 599858, 599918, 599951, 599952, 599953, 599954, 599964, 599980, 600018, 600103, 600144, 600217, 600263, 600362, 600395, 600396, 600408, 600423, 600425, 600444, 600559, 600594, 600595, 600596, 600607, 600608, 600609, 600610, 600611, 600672, 600673, 600702, 600703, 600704, 600732, 600854, 600857, 600875, 600893, 600925, 600948, 600949, 600969, 600970, 601033, 601038, 601070, 601095, 601133, 601134, 601138, 601156, 601157, 601172, 601176, 601228, 601229, 601246, 601247, 601256, 601257, 601303, 601304, 601305, 601306, 601307, 601472, 601473, 601480, 601845, 601888, 601892, 601897, 602158, 602160, 602162, 602270, 602318, 602319, 602320, 602399, 602401, 602424, 602432, 602501, 602512, 602545, 602587, 602611, 602636, 602688, 602976, 602996, 603005, 603198, 603208, 603240, 603244, 603380, 603384, 603439, 603487, 603529, 603603, 603609, 603667, 603678, 603684, 603775, 604436, 604470, 604509, 604527, 604563, 604627, 604727, 604734, 604786, 604789, 605030, 605031, 605049, 605072, 605082, 605132, 605240, 605252, 605795, 606222, 607351, 608085, 608248, 609221, 610395, 611451, 612610, 612837, 613559, 613672, 614320, 614416, 615403, 615485, 615530, 615899, 616014, 616035, 616061, 616244, 616293, 616319, 616398, 616614, 616711, 617320, 617411, 617481, 617650, 618378, 618466, 618515, 618655, 619011, 619247, 619399, 619403, 619445, 619506, 619528, 619712, 619746, 620195, 620487, 620659, 621181, 621387, 621816, 622448, 622809, 623000, 623202, 623282, 623599, 623669, 623672, 623940, 624081, 624389, 624587, 624725, 624771, 624783, 625049, 625174, 625281, 625529, 625773, 625889, 625906, 625955, 626096, 626109, 626157, 626951, 627042, 627165, 627247, 627703, 627831, 627863, 627873, 628049, 628509, 628684, 628764, 628836, 629193, 629253, 629256, 629318, 629507, 630024, 630096, 630118, 630573, 630814, 630896, 631210, 631541, 631649, 631694, 631707, 631761, 631838, 631926, 631932, 631951, 632007, 632221, 632304, 632521, 632593, 632665, 632720, 632748], Protein BMGC ID: ['BMGC_PT010020', 'BMGC_PT010067', 'BMGC_PT010104', 'BMGC_PT010227', 'BMGC_PT010323', 'BMGC_PT010423', 'BMGC_PT010437', 'BMGC_PT010530', 'BMGC_PT010543', 'BMGC_PT010561', 'BMGC_PT010570', 'BMGC_PT010639', 'BMGC_PT010663', 'BMGC_PT010692', 'BMGC_PT010718', 'BMGC_PT010757', 'BMGC_PT010917', 'BMGC_PT011072', 'BMGC_PT011079', 'BMGC_PT011100', 'BMGC_PT011114', 'BMGC_PT011494', 'BMGC_PT011524', 'BMGC_PT011710', 'BMGC_PT011931', 'BMGC_PT012079', 'BMGC_PT012127', 'BMGC_PT012209', 'BMGC_PT012256', 'BMGC_PT012284', 'BMGC_PT012455', 'BMGC_PT012564', 'BMGC_PT012566', 'BMGC_PT012635', 'BMGC_PT012853', 'BMGC_PT013027', 'BMGC_PT013070', 'BMGC_PT013079', 'BMGC_PT013182', 'BMGC_PT013186', 'BMGC_PT013241', 'BMGC_PT013242', 'BMGC_PT013514', 'BMGC_PT013573', 'BMGC_PT013680', 'BMGC_PT013681', 'BMGC_PT013795', 'BMGC_PT013796', 'BMGC_PT013797', 'BMGC_PT013807', 'BMGC_PT013858', 'BMGC_PT013865', 'BMGC_PT013945', 'BMGC_PT013987', 'BMGC_PT013991', 'BMGC_PT014392', 'BMGC_PT014422', 'BMGC_PT014482', 'BMGC_PT014591', 'BMGC_PT014625', 'BMGC_PT014671', 'BMGC_PT014714', 'BMGC_PT014812', 'BMGC_PT014835', 'BMGC_PT014916', 'BMGC_PT014924', 'BMGC_PT014995', 'BMGC_PT015019', 'BMGC_PT015057', 'BMGC_PT015062', 'BMGC_PT015109', 'BMGC_PT015119', 'BMGC_PT015166', 'BMGC_PT015167', 'BMGC_PT015238', 'BMGC_PT015248', 'BMGC_PT015335', 'BMGC_PT015343', 'BMGC_PT015344', 'BMGC_PT015404', 'BMGC_PT015437', 'BMGC_PT015438', 'BMGC_PT015439', 'BMGC_PT015440', 'BMGC_PT015450', 'BMGC_PT015466', 'BMGC_PT015504', 'BMGC_PT015589', 'BMGC_PT015630', 'BMGC_PT015703', 'BMGC_PT015749', 'BMGC_PT015848', 'BMGC_PT015881', 'BMGC_PT015882', 'BMGC_PT015894', 'BMGC_PT015909', 'BMGC_PT015911', 'BMGC_PT015930', 'BMGC_PT016045', 'BMGC_PT016080', 'BMGC_PT016081', 'BMGC_PT016082', 'BMGC_PT016093', 'BMGC_PT016094', 'BMGC_PT016095', 'BMGC_PT016096', 'BMGC_PT016097', 'BMGC_PT016158', 'BMGC_PT016159', 'BMGC_PT016188', 'BMGC_PT016189', 'BMGC_PT016190', 'BMGC_PT016218', 'BMGC_PT016340', 'BMGC_PT016343', 'BMGC_PT016361', 'BMGC_PT016379', 'BMGC_PT016411', 'BMGC_PT016434', 'BMGC_PT016435', 'BMGC_PT016455', 'BMGC_PT016456', 'BMGC_PT016519', 'BMGC_PT016524', 'BMGC_PT016556', 'BMGC_PT016581', 'BMGC_PT016619', 'BMGC_PT016620', 'BMGC_PT016624', 'BMGC_PT016642', 'BMGC_PT016643', 'BMGC_PT016658', 'BMGC_PT016662', 'BMGC_PT016714', 'BMGC_PT016715', 'BMGC_PT016732', 'BMGC_PT016733', 'BMGC_PT016742', 'BMGC_PT016743', 'BMGC_PT016789', 'BMGC_PT016790', 'BMGC_PT016791', 'BMGC_PT016792', 'BMGC_PT016793', 'BMGC_PT016958', 'BMGC_PT016959', 'BMGC_PT016966', 'BMGC_PT017331', 'BMGC_PT017374', 'BMGC_PT017378', 'BMGC_PT017383', 'BMGC_PT017644', 'BMGC_PT017646', 'BMGC_PT017648', 'BMGC_PT017756', 'BMGC_PT017804', 'BMGC_PT017805', 'BMGC_PT017806', 'BMGC_PT017885', 'BMGC_PT017887', 'BMGC_PT017910', 'BMGC_PT017918', 'BMGC_PT017987', 'BMGC_PT017998', 'BMGC_PT018031', 'BMGC_PT018073', 'BMGC_PT018097', 'BMGC_PT018122', 'BMGC_PT018174', 'BMGC_PT018462', 'BMGC_PT018482', 'BMGC_PT018491', 'BMGC_PT018684', 'BMGC_PT018694', 'BMGC_PT018726', 'BMGC_PT018730', 'BMGC_PT018866', 'BMGC_PT018870', 'BMGC_PT018925', 'BMGC_PT018973', 'BMGC_PT019015', 'BMGC_PT019089', 'BMGC_PT019095', 'BMGC_PT019153', 'BMGC_PT019164', 'BMGC_PT019170', 'BMGC_PT019261', 'BMGC_PT019922', 'BMGC_PT019956', 'BMGC_PT019995', 'BMGC_PT020013', 'BMGC_PT020049', 'BMGC_PT020113', 'BMGC_PT020213', 'BMGC_PT020220', 'BMGC_PT020272', 'BMGC_PT020275', 'BMGC_PT020516', 'BMGC_PT020517', 'BMGC_PT020535', 'BMGC_PT020558', 'BMGC_PT020568', 'BMGC_PT020618', 'BMGC_PT020726', 'BMGC_PT020738', 'BMGC_PT021281', 'BMGC_PT021708', 'BMGC_PT022837', 'BMGC_PT023571', 'BMGC_PT023734', 'BMGC_PT024707', 'BMGC_PT025881', 'BMGC_PT026937', 'BMGC_PT028096', 'BMGC_PT028323', 'BMGC_PT029045', 'BMGC_PT029158', 'BMGC_PT029806', 'BMGC_PT029902', 'BMGC_PT030889', 'BMGC_PT030971', 'BMGC_PT031016', 'BMGC_PT031385', 'BMGC_PT031500', 'BMGC_PT031521', 'BMGC_PT031547', 'BMGC_PT031730', 'BMGC_PT031779', 'BMGC_PT031805', 'BMGC_PT031884', 'BMGC_PT032100', 'BMGC_PT032197', 'BMGC_PT032806', 'BMGC_PT032897', 'BMGC_PT032967', 'BMGC_PT033136', 'BMGC_PT033864', 'BMGC_PT033952', 'BMGC_PT034001', 'BMGC_PT034141', 'BMGC_PT034497', 'BMGC_PT034733', 'BMGC_PT034885', 'BMGC_PT034889', 'BMGC_PT034931', 'BMGC_PT034992', 'BMGC_PT035014', 'BMGC_PT035198', 'BMGC_PT035232', 'BMGC_PT035681', 'BMGC_PT035973', 'BMGC_PT036145', 'BMGC_PT036667', 'BMGC_PT036873', 'BMGC_PT037302', 'BMGC_PT037934', 'BMGC_PT038295', 'BMGC_PT038486', 'BMGC_PT038688', 'BMGC_PT038768', 'BMGC_PT039085', 'BMGC_PT039155', 'BMGC_PT039158', 'BMGC_PT039426', 'BMGC_PT039567', 'BMGC_PT039875', 'BMGC_PT040073', 'BMGC_PT040211', 'BMGC_PT040257', 'BMGC_PT040269', 'BMGC_PT040535', 'BMGC_PT040660', 'BMGC_PT040767', 'BMGC_PT041015', 'BMGC_PT041259', 'BMGC_PT041375', 'BMGC_PT041392', 'BMGC_PT041441', 'BMGC_PT041582', 'BMGC_PT041595', 'BMGC_PT041643', 'BMGC_PT042437', 'BMGC_PT042528', 'BMGC_PT042651', 'BMGC_PT042733', 'BMGC_PT043189', 'BMGC_PT043317', 'BMGC_PT043349', 'BMGC_PT043359', 'BMGC_PT043535', 'BMGC_PT043995', 'BMGC_PT044170', 'BMGC_PT044250', 'BMGC_PT044322', 'BMGC_PT044679', 'BMGC_PT044739', 'BMGC_PT044742', 'BMGC_PT044804', 'BMGC_PT044993', 'BMGC_PT045510', 'BMGC_PT045582', 'BMGC_PT045604', 'BMGC_PT046059', 'BMGC_PT046300', 'BMGC_PT046382', 'BMGC_PT046696', 'BMGC_PT047027', 'BMGC_PT047135', 'BMGC_PT047180', 'BMGC_PT047193', 'BMGC_PT047247', 'BMGC_PT047324', 'BMGC_PT047412', 'BMGC_PT047418', 'BMGC_PT047437', 'BMGC_PT047493', 'BMGC_PT047707', 'BMGC_PT047790', 'BMGC_PT048007', 'BMGC_PT048079', 'BMGC_PT048151', 'BMGC_PT048206', 'BMGC_PT048234']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:   3%|▎         | 5/172 [00:08<04:15,  1.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Drug BMGC ID: BMGC_DG00420, Protein Index: [597755, 597756, 598309, 598996, 600396, 600559, 623599], Protein BMGC ID: ['BMGC_PT013241', 'BMGC_PT013242', 'BMGC_PT013795', 'BMGC_PT014482', 'BMGC_PT015882', 'BMGC_PT016045', 'BMGC_PT039085']\n",
      "... (remaining entries processed silently)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:   6%|▌         | 10/172 [00:13<03:33,  1.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 10/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  12%|█▏        | 20/172 [00:25<03:36,  1.43s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 20/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  17%|█▋        | 30/172 [00:30<01:20,  1.77it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 30/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  23%|██▎       | 40/172 [00:41<02:02,  1.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 40/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  30%|██▉       | 51/172 [00:51<00:46,  2.58it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 50/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  35%|███▍      | 60/172 [00:56<01:01,  1.83it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 60/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  41%|████      | 70/172 [01:04<01:10,  1.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 70/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  47%|████▋     | 80/172 [01:07<00:24,  3.83it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 80/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  52%|█████▏    | 90/172 [01:10<00:22,  3.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 90/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  59%|█████▊    | 101/172 [01:12<00:16,  4.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 100/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  64%|██████▍   | 110/172 [01:19<00:40,  1.52it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 110/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  70%|██████▉   | 120/172 [01:29<00:26,  1.95it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 120/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  76%|███████▌  | 131/172 [01:37<00:26,  1.55it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 130/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  81%|████████▏ | 140/172 [01:39<00:07,  4.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 140/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  88%|████████▊ | 151/172 [01:48<00:08,  2.39it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 150/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  93%|█████████▎| 160/172 [01:52<00:03,  3.72it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 160/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships:  99%|█████████▉| 171/172 [01:54<00:00,  5.41it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved after processing 170/172 drugs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing drug-protein relationships: 100%|██████████| 172/172 [01:54<00:00,  1.50it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Final save completed - processed 172 drugs and saved to: ./data/DrugQA/drug_kg_protein_relationships.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "from tqdm import tqdm\n",
    "\n",
    "dti_drug_ids = sorted(list(set(dti_drug_overlap_df['BMGC_Drug_ID'])))\n",
    "\n",
    "drug_kg_protein_bmgc_dict = {}\n",
    "print_count = 0\n",
    "max_prints = 5\n",
    "\n",
    "for i, drug_bmgc_id in enumerate(tqdm(dti_drug_ids, desc=\"Processing drug-protein relationships\")):\n",
    "    drug_protein_index, drug_protein_bmgc_id = extract_drug_protein(\n",
    "        selected_sample_drug_bmgc_id=drug_bmgc_id,\n",
    "        edge_index=edge_index,\n",
    "        node_index_df=nodes_index_data,\n",
    "        nodeid_index_dict=node_index_dict,\n",
    "        index_nodeid_dict=index_node_dict\n",
    "    )\n",
    "    hgnc_dict, drug_protein_hgnc_list = bmgc_pt_id_to_hgnc(drug_protein_bmgc_id, bmgc_protein_df)\n",
    "    \n",
    "    # Only print first 5 entries\n",
    "    if print_count < max_prints:\n",
    "        print(f\"Drug BMGC ID: {drug_bmgc_id}, Protein Index: {drug_protein_index}, Protein BMGC ID: {drug_protein_bmgc_id}\")\n",
    "        print_count += 1\n",
    "    elif print_count == max_prints:\n",
    "        print(\"... (remaining entries processed silently)\")\n",
    "        print_count += 1\n",
    "    \n",
    "    # Convert NumPy integers to Python integers for JSON serialization\n",
    "    drug_kg_protein_bmgc_dict[drug_bmgc_id] = {\n",
    "        'drug_protein_index': [int(x) for x in drug_protein_index],  # Convert numpy ints to Python ints\n",
    "        'drug_protein_bmgc_id': drug_protein_bmgc_id\n",
    "    }\n",
    "    \n",
    "    # Save every 10 iterations\n",
    "    if (i + 1) % 10 == 0:\n",
    "        output_path = \"./data/DrugQA/drug_kg_protein_relationships.json\"\n",
    "        with open(output_path, \"w\") as f:\n",
    "            json.dump(drug_kg_protein_bmgc_dict, f, indent=2)\n",
    "        print(f\"💾 Auto-saved after processing {i + 1}/{len(dti_drug_ids)} drugs\")\n",
    "\n",
    "# Final save after loop completion\n",
    "output_path = \"./data/DrugQA/drug_kg_protein_relationships.json\"\n",
    "with open(output_path, \"w\") as f:\n",
    "    json.dump(drug_kg_protein_bmgc_dict, f, indent=2)\n",
    "    \n",
    "print(f\"✅ Final save completed - processed {len(drug_kg_protein_bmgc_dict)} drugs and saved to: {output_path}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Formulate the sample related json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 472,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample ACH Name: ACH-000002\n",
      "Extracting top 10 gene information for ACH-000002...\n",
      "Extracting top 10 transcript information for ACH-000002...\n",
      "Extracting top 10 protein information for ACH-000002...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07934 (ACH-000002) ...\n",
      "Knowledge Graph Info: Found 5 disease-related proteins directly connected to BMGC_DS07934 and 4915 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 22 unique proteins and 73 relationships between them\n"
     ]
    }
   ],
   "source": [
    "def drug_qa_sample_info(sample_ach_name, selected_sample_disease_bmgc_id, k=10):\n",
    "    # LLM Info\n",
    "    print(f\"Sample ACH Name: {sample_ach_name}\")\n",
    "    print(f\"Extracting top {k} gene information for {sample_ach_name}...\")\n",
    "    top_k_gene_hgnc_name_list, top_k_gene_protein_bmgc_id_list, top_k_gene_protein_bmgc_llmnameid_list = extract_dti_gn_info(dti_gene_df, bmgc_gene_df, gene_transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "    print(f\"Extracting top {k} transcript information for {sample_ach_name}...\")\n",
    "    top_k_transcript_hgnc_name_list, top_k_transcript_protein_bmgc_id_list, top_k_transcript_protein_bmgc_llmnameid_list = extract_dti_ts_info(dti_transcript_df, bmgc_transcript_df, transcript_protein_entity_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "    print(f\"Extracting top {k} protein information for {sample_ach_name}...\")\n",
    "    top_k_protein_hgnc_name_list, top_k_protein_bmgc_id_list, top_k_protein_bmgc_llmnameid_list = extract_dti_pt_info(dti_protein_df, bmgc_protein_df, bmgc_protein_llmnameid_combined_df, sample_ach_name, k)\n",
    "    # KG Info\n",
    "    edge_index = np.load('./data/DrugQA/edge_index.npy')\n",
    "    print(f\"Extracting disease-related proteins index and bmgc id for {selected_sample_disease_bmgc_id} ({sample_ach_name}) ...\")\n",
    "    disease_protein_index, disease_protein_bmgc_id, ppi_nodes_index, ppi_nodes_bmgc_id = extract_kg_related_proteins(selected_sample_disease_bmgc_id, edge_index, nodes_index_data, node_index_dict, index_node_dict)\n",
    "    print(f\"Knowledge Graph Info: Found {len(disease_protein_index)} disease-related proteins directly connected to {selected_sample_disease_bmgc_id} and {len(ppi_nodes_index)} proteins in their PPI network\")\n",
    "    print(f\"Mapping disease-related proteins to HGNC symbols...\")\n",
    "    disease_protein_hgnc_dict, disease_protein_hgnc_list = bmgc_pt_id_to_hgnc(disease_protein_bmgc_id, bmgc_protein_df)\n",
    "    print(f\"Mapping PPI-related proteins to HGNC symbols...\")\n",
    "    ppi_hgnc_dict, ppi_hgnc_list = bmgc_pt_id_to_hgnc(ppi_nodes_bmgc_id, bmgc_protein_df)\n",
    "    # LLM Used KG Info\n",
    "    print(f\"Extracting protein relationships from BMGC...\")\n",
    "    # Convert the any non-existed string in top_k_gene_hgnc_name_list + top_k_transcript_hgnc_name_list + top_k_protein_hgnc_name_list + disease_protein_hgnc_list into empty list []\n",
    "    if top_k_gene_hgnc_name_list == \"non-existed\": top_k_gene_hgnc_name_list = []\n",
    "    if top_k_transcript_hgnc_name_list == \"non-existed\": top_k_transcript_hgnc_name_list = []\n",
    "    if top_k_protein_hgnc_name_list == \"non-existed\": top_k_protein_hgnc_name_list = []\n",
    "    if disease_protein_hgnc_list == \"non-existed\": disease_protein_hgnc_list = []\n",
    "    # Combine all the HGNC symbols into a single list for relationship extraction\n",
    "    omics_disease_protein_hgnc_list = list(set(top_k_gene_hgnc_name_list + top_k_transcript_hgnc_name_list + top_k_protein_hgnc_name_list + disease_protein_hgnc_list))\n",
    "    relationships_df, relationship_texts = find_protein_relationships(omics_disease_protein_hgnc_list, bmgc_protein_df, bmgc_relation_df)\n",
    "    print(f\"Knowledge Graph Info: Found {len(omics_disease_protein_hgnc_list)} unique proteins and {len(relationship_texts)} relationships between them\")\n",
    "    return top_k_gene_hgnc_name_list, top_k_gene_protein_bmgc_id_list, top_k_gene_protein_bmgc_llmnameid_list, \\\n",
    "              top_k_transcript_hgnc_name_list, top_k_transcript_protein_bmgc_id_list, top_k_transcript_protein_bmgc_llmnameid_list, \\\n",
    "                top_k_protein_hgnc_name_list, top_k_protein_bmgc_id_list, top_k_protein_bmgc_llmnameid_list, \\\n",
    "                disease_protein_index, disease_protein_bmgc_id, ppi_nodes_index, ppi_nodes_bmgc_id, \\\n",
    "                    disease_protein_hgnc_dict, disease_protein_hgnc_list, ppi_hgnc_dict, ppi_hgnc_list, relationship_texts\n",
    "\n",
    "# Example usage\n",
    "sample_ach_name = 'ACH-000002'\n",
    "selected_sample_disease_bmgc_id = 'BMGC_DS07934'\n",
    "k = 10\n",
    "return_tuples = drug_qa_sample_info(sample_ach_name, selected_sample_disease_bmgc_id, k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 470,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded existing JSON with 0 processed samples\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Index</th>\n",
       "      <th>depMapID</th>\n",
       "      <th>Name</th>\n",
       "      <th>BMGC_Disease_ID</th>\n",
       "      <th>BMGC_Disease_name</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Histology</th>\n",
       "      <th>type</th>\n",
       "      <th>PATHOLOGIST_ANNOTATION</th>\n",
       "      <th>tcga_code</th>\n",
       "      <th>...</th>\n",
       "      <th>depmap_match_2_disease</th>\n",
       "      <th>depmap_match_2_disease_name</th>\n",
       "      <th>ncit_match_1_disease</th>\n",
       "      <th>ncit_match_1_disease_name</th>\n",
       "      <th>ncit_match_2_disease</th>\n",
       "      <th>ncit_match_2_disease_name</th>\n",
       "      <th>ordo_match_1_disease</th>\n",
       "      <th>ordo_match_1_disease_name</th>\n",
       "      <th>ordo_match_2_disease</th>\n",
       "      <th>ordo_match_2_disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ACH-000002</td>\n",
       "      <td>HL-60</td>\n",
       "      <td>BMGC_DS01407</td>\n",
       "      <td>acute myeloid leukemia</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>LAML</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>ACH-000006</td>\n",
       "      <td>MONO-MAC-6</td>\n",
       "      <td>BMGC_DS20800</td>\n",
       "      <td>leukemia, myeloid, acute</td>\n",
       "      <td>primary</td>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "      <td>AML</td>\n",
       "      <td>Leukemia:AML</td>\n",
       "      <td>LAML</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20480</td>\n",
       "      <td>Leukemia, Myeloid</td>\n",
       "      <td>BMGC_DS05322</td>\n",
       "      <td>T-cell adult acute lymphocytic leukemia</td>\n",
       "      <td>BMGC_DS07386</td>\n",
       "      <td>adult acute lymphoblastic leukemia | adult acu...</td>\n",
       "      <td>BMGC_DS09029</td>\n",
       "      <td>acute lymphoblastic leukemia | precursor lymph...</td>\n",
       "      <td>BMGC_DS07386</td>\n",
       "      <td>adult acute lymphoblastic leukemia | adult acu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ACH-000008</td>\n",
       "      <td>A101D</td>\n",
       "      <td>BMGC_DS01559</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>malignant_melanoma</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>Skin:Melanoma</td>\n",
       "      <td>SKCM</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ACH-000020</td>\n",
       "      <td>MHH-CALL-2</td>\n",
       "      <td>BMGC_DS06157</td>\n",
       "      <td>precursor b-cell acute lymphoblastic leukemia</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>B-cell_ALL</td>\n",
       "      <td>Leukemia:ALL</td>\n",
       "      <td>ALL</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ACH-000038</td>\n",
       "      <td>EHEB</td>\n",
       "      <td>BMGC_DS01395</td>\n",
       "      <td>b-cell chronic lymphocytic leukemia or chronic...</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>lymphoma_other</td>\n",
       "      <td>Leukemia:CLL</td>\n",
       "      <td>CLL</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20800</td>\n",
       "      <td>Leukemia, Myeloid, Acute</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>282</td>\n",
       "      <td>ACH-001151</td>\n",
       "      <td>OVCAR-5</td>\n",
       "      <td>BMGC_DS07934</td>\n",
       "      <td>ovarian adenocarcinoma or ovary adenocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>ovary</td>\n",
       "      <td>Ovary:Carcinoma</td>\n",
       "      <td>OV</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS08982</td>\n",
       "      <td>ovarian transitional cell carcinoma | ovary tr...</td>\n",
       "      <td>BMGC_DS08979</td>\n",
       "      <td>ovarian serous surface papillary adenocarcinom...</td>\n",
       "      <td>BMGC_DS09417</td>\n",
       "      <td>peritoneal serous adenocarcinoma | primary per...</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>283</td>\n",
       "      <td>ACH-001182</td>\n",
       "      <td>RPMI 6666</td>\n",
       "      <td>BMGC_DS20405</td>\n",
       "      <td>hodgkin disease or hodgkins lymphoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoblastoid</td>\n",
       "      <td>lymphoma_Hodgkin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMGC_DS20323</td>\n",
       "      <td>Fever</td>\n",
       "      <td>BMGC_DS22241</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>284</td>\n",
       "      <td>ACH-001199</td>\n",
       "      <td>SNU-C2B</td>\n",
       "      <td>BMGC_DS08517</td>\n",
       "      <td>cecum adenocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>colorectal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COAD/READ</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>285</td>\n",
       "      <td>ACH-001203</td>\n",
       "      <td>SUP-HD1</td>\n",
       "      <td>BMGC_DS20405</td>\n",
       "      <td>hodgkin disease or hodgkins lymphoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "      <td>lymphoma_Hodgkin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "      <td>BMGC_DS20405</td>\n",
       "      <td>Hodgkin Disease | Hodgkins lymphoma</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "      <td>BMGC_DS20405</td>\n",
       "      <td>Hodgkin Disease | Hodgkins lymphoma</td>\n",
       "      <td>BMGC_DS20499</td>\n",
       "      <td>Lymphoma, Non-Hodgkin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>286</td>\n",
       "      <td>ACH-001233</td>\n",
       "      <td>VMRC-LCD</td>\n",
       "      <td>BMGC_DS02944</td>\n",
       "      <td>lung adenocarcinoma</td>\n",
       "      <td>primary</td>\n",
       "      <td>carcinoma</td>\n",
       "      <td>lung_NSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>286 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Index    depMapID        Name BMGC_Disease_ID  \\\n",
       "0        1  ACH-000002       HL-60    BMGC_DS01407   \n",
       "1        2  ACH-000006  MONO-MAC-6    BMGC_DS20800   \n",
       "2        3  ACH-000008       A101D    BMGC_DS01559   \n",
       "3        4  ACH-000020  MHH-CALL-2    BMGC_DS06157   \n",
       "4        5  ACH-000038        EHEB    BMGC_DS01395   \n",
       "..     ...         ...         ...             ...   \n",
       "281    282  ACH-001151     OVCAR-5    BMGC_DS07934   \n",
       "282    283  ACH-001182   RPMI 6666    BMGC_DS20405   \n",
       "283    284  ACH-001199     SNU-C2B    BMGC_DS08517   \n",
       "284    285  ACH-001203     SUP-HD1    BMGC_DS20405   \n",
       "285    286  ACH-001233    VMRC-LCD    BMGC_DS02944   \n",
       "\n",
       "                                     BMGC_Disease_name Pathology  \\\n",
       "0                               acute myeloid leukemia   primary   \n",
       "1                             leukemia, myeloid, acute   primary   \n",
       "2                                             melanoma   primary   \n",
       "3        precursor b-cell acute lymphoblastic leukemia   primary   \n",
       "4    b-cell chronic lymphocytic leukemia or chronic...   primary   \n",
       "..                                                 ...       ...   \n",
       "281     ovarian adenocarcinoma or ovary adenocarcinoma   primary   \n",
       "282               hodgkin disease or hodgkins lymphoma   primary   \n",
       "283                               cecum adenocarcinoma   primary   \n",
       "284               hodgkin disease or hodgkins lymphoma   primary   \n",
       "285                                lung adenocarcinoma   primary   \n",
       "\n",
       "                   Histology              type PATHOLOGIST_ANNOTATION  \\\n",
       "0    haematopoietic_neoplasm               AML           Leukemia:AML   \n",
       "1    haematopoietic_neoplasm               AML           Leukemia:AML   \n",
       "2         malignant_melanoma          melanoma          Skin:Melanoma   \n",
       "3          lymphoid_neoplasm        B-cell_ALL           Leukemia:ALL   \n",
       "4          lymphoid_neoplasm    lymphoma_other           Leukemia:CLL   \n",
       "..                       ...               ...                    ...   \n",
       "281                carcinoma             ovary        Ovary:Carcinoma   \n",
       "282           lymphoblastoid  lymphoma_Hodgkin                    NaN   \n",
       "283                carcinoma        colorectal                    NaN   \n",
       "284        lymphoid_neoplasm  lymphoma_Hodgkin                    NaN   \n",
       "285                carcinoma          lung_NSC                    NaN   \n",
       "\n",
       "     tcga_code  ... depmap_match_2_disease  \\\n",
       "0         LAML  ...                    NaN   \n",
       "1         LAML  ...           BMGC_DS20480   \n",
       "2         SKCM  ...                    NaN   \n",
       "3          ALL  ...                    NaN   \n",
       "4          CLL  ...           BMGC_DS20800   \n",
       "..         ...  ...                    ...   \n",
       "281         OV  ...           BMGC_DS08982   \n",
       "282        NaN  ...           BMGC_DS20499   \n",
       "283  COAD/READ  ...                    NaN   \n",
       "284        NaN  ...           BMGC_DS20499   \n",
       "285       LUAD  ...                    NaN   \n",
       "\n",
       "                           depmap_match_2_disease_name ncit_match_1_disease  \\\n",
       "0                                                  NaN                  NaN   \n",
       "1                                    Leukemia, Myeloid         BMGC_DS05322   \n",
       "2                                                  NaN                  NaN   \n",
       "3                                                  NaN                  NaN   \n",
       "4                             Leukemia, Myeloid, Acute         BMGC_DS20323   \n",
       "..                                                 ...                  ...   \n",
       "281  ovarian transitional cell carcinoma | ovary tr...         BMGC_DS08979   \n",
       "282                              Lymphoma, Non-Hodgkin         BMGC_DS20323   \n",
       "283                                                NaN                  NaN   \n",
       "284                              Lymphoma, Non-Hodgkin         BMGC_DS20405   \n",
       "285                                                NaN                  NaN   \n",
       "\n",
       "                             ncit_match_1_disease_name ncit_match_2_disease  \\\n",
       "0                                                  NaN                  NaN   \n",
       "1              T-cell adult acute lymphocytic leukemia         BMGC_DS07386   \n",
       "2                                                  NaN                  NaN   \n",
       "3                                                  NaN                  NaN   \n",
       "4                                                Fever         BMGC_DS22241   \n",
       "..                                                 ...                  ...   \n",
       "281  ovarian serous surface papillary adenocarcinom...         BMGC_DS09417   \n",
       "282                                              Fever         BMGC_DS22241   \n",
       "283                                                NaN                  NaN   \n",
       "284                Hodgkin Disease | Hodgkins lymphoma         BMGC_DS20499   \n",
       "285                                                NaN                  NaN   \n",
       "\n",
       "                             ncit_match_2_disease_name ordo_match_1_disease  \\\n",
       "0                                                  NaN                  NaN   \n",
       "1    adult acute lymphoblastic leukemia | adult acu...         BMGC_DS09029   \n",
       "2                                                  NaN                  NaN   \n",
       "3                                                  NaN                  NaN   \n",
       "4                                                  NaN         BMGC_DS20323   \n",
       "..                                                 ...                  ...   \n",
       "281  peritoneal serous adenocarcinoma | primary per...         BMGC_DS20323   \n",
       "282                                                NaN         BMGC_DS20323   \n",
       "283                                                NaN                  NaN   \n",
       "284                              Lymphoma, Non-Hodgkin         BMGC_DS20405   \n",
       "285                                                NaN                  NaN   \n",
       "\n",
       "                             ordo_match_1_disease_name ordo_match_2_disease  \\\n",
       "0                                                  NaN                  NaN   \n",
       "1    acute lymphoblastic leukemia | precursor lymph...         BMGC_DS07386   \n",
       "2                                                  NaN                  NaN   \n",
       "3                                                  NaN                  NaN   \n",
       "4                                                Fever         BMGC_DS22241   \n",
       "..                                                 ...                  ...   \n",
       "281                                              Fever         BMGC_DS22241   \n",
       "282                                              Fever         BMGC_DS22241   \n",
       "283                                                NaN                  NaN   \n",
       "284                Hodgkin Disease | Hodgkins lymphoma         BMGC_DS20499   \n",
       "285                                                NaN                  NaN   \n",
       "\n",
       "                             ordo_match_2_disease_name  \n",
       "0                                                  NaN  \n",
       "1    adult acute lymphoblastic leukemia | adult acu...  \n",
       "2                                                  NaN  \n",
       "3                                                  NaN  \n",
       "4                                                  NaN  \n",
       "..                                                 ...  \n",
       "281                                                NaN  \n",
       "282                                                NaN  \n",
       "283                                                NaN  \n",
       "284                              Lymphoma, Non-Hodgkin  \n",
       "285                                                NaN  \n",
       "\n",
       "[286 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import json\n",
    "import os\n",
    "from tqdm import tqdm\n",
    "\n",
    "# Parameters\n",
    "k = 10\n",
    "save_every_n = 10\n",
    "\n",
    "# Output folder and filename\n",
    "output_dir = \"./data/DrugQA\"\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "output_filename = f\"drug_qa_k{k}_bm{top_bm}.json\"\n",
    "output_path = os.path.join(output_dir, output_filename)\n",
    "\n",
    "# Load existing JSON file if it exists\n",
    "if os.path.exists(output_path):\n",
    "    with open(output_path, \"r\") as f:\n",
    "        multi_sample_qa_json = json.load(f)\n",
    "    print(f\"Loaded existing JSON with {len(multi_sample_qa_json)} processed samples\")\n",
    "else:\n",
    "    multi_sample_qa_json = {}\n",
    "    print(\"No existing JSON found, starting fresh\")\n",
    "\n",
    "# Load sample info data\n",
    "dti_sample_info_index = pd.read_csv('./data/process_data/dti_combined_samples.csv')\n",
    "dti_sample_info_index = pd.merge(dti_sample_info_index, remaining_dti_samples_df['depMapID'], how='inner', on='depMapID').reset_index(drop=True)\n",
    "dti_sample_info_index['BMGC_Disease_name'] = dti_sample_info_index['BMGC_Disease_name'].replace(r' \\| ', ' or ', regex=True)\n",
    "# Insert a new column \"Index\" in the first position\n",
    "dti_sample_info_index.insert(0, 'Index', range(1, len(dti_sample_info_index) + 1))\n",
    "display(dti_sample_info_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 473,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 0 already processed samples\n",
      "Total samples to process: 286, already processed: 0, remaining: 286\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/286 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing sample 1/286: ACH-000002 (HL-60)\n",
      "Sample Index: 1\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000002\n",
      "Extracting top 10 gene information for ACH-000002...\n",
      "Extracting top 10 transcript information for ACH-000002...\n",
      "Extracting top 10 protein information for ACH-000002...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000002) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 1/286 [03:56<18:41:28, 236.10s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 224 unique proteins and 583 relationships between them\n",
      "Processing sample 2/286: ACH-000006 (MONO-MAC-6)\n",
      "Sample Index: 2\n",
      "Sample Disease: leukemia, myeloid, acute\n",
      "Sample Disease BMGC ID: BMGC_DS20800\n",
      "Sample ACH Name: ACH-000006\n",
      "Extracting top 10 gene information for ACH-000006...\n",
      "Extracting top 10 transcript information for ACH-000006...\n",
      "Extracting top 10 protein information for ACH-000006...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20800 (ACH-000006) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20800 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  1%|          | 2/286 [03:58<7:47:39, 98.80s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 20 relationships between them\n",
      "Processing sample 3/286: ACH-000008 (A101D)\n",
      "Sample Index: 3\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-000008\n",
      "Extracting top 10 gene information for ACH-000008...\n",
      "Extracting top 10 transcript information for ACH-000008...\n",
      "Extracting top 10 protein information for ACH-000008...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-000008) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  1%|          | 3/286 [07:50<12:31:24, 159.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 291 unique proteins and 829 relationships between them\n",
      "Processing sample 4/286: ACH-000020 (MHH-CALL-2)\n",
      "Sample Index: 4\n",
      "Sample Disease: precursor b-cell acute lymphoblastic leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS06157\n",
      "Sample ACH Name: ACH-000020\n",
      "Extracting top 10 gene information for ACH-000020...\n",
      "Extracting top 10 transcript information for ACH-000020...\n",
      "Extracting top 10 protein information for ACH-000020...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06157 (ACH-000020) ...\n",
      "Knowledge Graph Info: Found 14 disease-related proteins directly connected to BMGC_DS06157 and 5953 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  1%|▏         | 4/286 [08:30<8:48:51, 112.52s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 24 unique proteins and 24 relationships between them\n",
      "Processing sample 5/286: ACH-000038 (EHEB)\n",
      "Sample Index: 5\n",
      "Sample Disease: b-cell chronic lymphocytic leukemia or chronic lymphocytic leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01395\n",
      "Sample ACH Name: ACH-000038\n",
      "Extracting top 10 gene information for ACH-000038...\n",
      "Extracting top 10 transcript information for ACH-000038...\n",
      "Extracting top 10 protein information for ACH-000038...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01395 (ACH-000038) ...\n",
      "Knowledge Graph Info: Found 91 disease-related proteins directly connected to BMGC_DS01395 and 19841 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  2%|▏         | 5/286 [10:47<9:27:02, 121.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 68 unique proteins and 52 relationships between them\n",
      "Processing sample 6/286: ACH-000046 (ACHN)\n",
      "Sample Index: 6\n",
      "Sample Disease: papillary renal cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08344\n",
      "Sample ACH Name: ACH-000046\n",
      "Extracting top 10 gene information for ACH-000046...\n",
      "Extracting top 10 transcript information for ACH-000046...\n",
      "Extracting top 10 protein information for ACH-000046...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08344 (ACH-000046) ...\n",
      "Knowledge Graph Info: Found 6 disease-related proteins directly connected to BMGC_DS08344 and 4248 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  2%|▏         | 6/286 [11:16<6:59:19, 89.86s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 23 unique proteins and 56 relationships between them\n",
      "Processing sample 7/286: ACH-000047 (GCIY)\n",
      "Sample Index: 7\n",
      "Sample Disease: gastric adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05289\n",
      "Sample ACH Name: ACH-000047\n",
      "Extracting top 10 gene information for ACH-000047...\n",
      "Extracting top 10 transcript information for ACH-000047...\n",
      "Extracting top 10 protein information for ACH-000047...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05289 (ACH-000047) ...\n",
      "Knowledge Graph Info: Found 215 disease-related proteins directly connected to BMGC_DS05289 and 23595 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  2%|▏         | 7/286 [14:11<9:07:42, 117.79s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 95 unique proteins and 143 relationships between them\n",
      "Processing sample 8/286: ACH-000050 (NCI-H929)\n",
      "Sample Index: 8\n",
      "Sample Disease: plasma cell myeloma\n",
      "Sample Disease BMGC ID: BMGC_DS01639\n",
      "Sample ACH Name: ACH-000050\n",
      "Extracting top 10 gene information for ACH-000050...\n",
      "Extracting top 10 transcript information for ACH-000050...\n",
      "Extracting top 10 protein information for ACH-000050...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01639 (ACH-000050) ...\n",
      "Knowledge Graph Info: Found 244 disease-related proteins directly connected to BMGC_DS01639 and 24451 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  3%|▎         | 8/286 [17:15<10:43:41, 138.93s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 180 unique proteins and 164 relationships between them\n",
      "Processing sample 9/286: ACH-000053 (KARPAS-299)\n",
      "Sample Index: 9\n",
      "Sample Disease: alk-positive anaplastic large cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS08403\n",
      "Sample ACH Name: ACH-000053\n",
      "Extracting top 10 gene information for ACH-000053...\n",
      "Extracting top 10 transcript information for ACH-000053...\n",
      "Extracting top 10 protein information for ACH-000053...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08403 (ACH-000053) ...\n",
      "Knowledge Graph Info: Found 2 disease-related proteins directly connected to BMGC_DS08403 and 2478 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  3%|▎         | 9/286 [17:33<7:46:51, 101.13s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 21 unique proteins and 56 relationships between them\n",
      "Processing sample 10/286: ACH-000056 (DOHH-2)\n",
      "Sample Index: 10\n",
      "Sample Disease: diffuse large b-cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS02693\n",
      "Sample ACH Name: ACH-000056\n",
      "Extracting top 10 gene information for ACH-000056...\n",
      "Extracting top 10 transcript information for ACH-000056...\n",
      "Extracting top 10 protein information for ACH-000056...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02693 (ACH-000056) ...\n",
      "Knowledge Graph Info: Found 148 disease-related proteins directly connected to BMGC_DS02693 and 20906 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 87 unique proteins and 100 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  3%|▎         | 10/286 [20:04<8:55:36, 116.44s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 10/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000056\n",
      "Progress: 10/286 (3.5%), Remaining: 276\n",
      "Processing sample 11/286: ACH-000058 (ML-1)\n",
      "Sample Index: 11\n",
      "Sample Disease: thyroid carcinoma, anaplastic\n",
      "Sample Disease BMGC ID: BMGC_DS21147\n",
      "Sample ACH Name: ACH-000058\n",
      "Extracting top 10 gene information for ACH-000058...\n",
      "Extracting top 10 transcript information for ACH-000058...\n",
      "Extracting top 10 protein information for ACH-000058...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS21147 (ACH-000058) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS21147 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  4%|▍         | 11/286 [20:07<6:14:12, 81.65s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 56 relationships between them\n",
      "Processing sample 12/286: ACH-000059 (SUP-B15)\n",
      "Sample Index: 12\n",
      "Sample Disease: b-lymphoblastic leukemia/lymphoma with t(9;22)(q34.1;q11.2)\n",
      "Sample Disease BMGC ID: BMGC_DS19139\n",
      "Sample ACH Name: ACH-000059\n",
      "Extracting top 10 gene information for ACH-000059...\n",
      "Extracting top 10 transcript information for ACH-000059...\n",
      "Extracting top 10 protein information for ACH-000059...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS19139 (ACH-000059) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS19139 and 6259 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  4%|▍         | 12/286 [20:49<5:18:07, 69.66s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 25 unique proteins and 37 relationships between them\n",
      "Processing sample 13/286: ACH-000061 (HH)\n",
      "Sample Index: 13\n",
      "Sample Disease: leukemia-lymphoma, adult t-cell\n",
      "Sample Disease BMGC ID: BMGC_DS20796\n",
      "Sample ACH Name: ACH-000061\n",
      "Extracting top 10 gene information for ACH-000061...\n",
      "Extracting top 10 transcript information for ACH-000061...\n",
      "Extracting top 10 protein information for ACH-000061...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20796 (ACH-000061) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20796 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  5%|▍         | 13/286 [20:51<3:44:05, 49.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 31 relationships between them\n",
      "Processing sample 14/286: ACH-000062 (RERF-LC-MS)\n",
      "Sample Index: 14\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000062\n",
      "Extracting top 10 gene information for ACH-000062...\n",
      "Extracting top 10 transcript information for ACH-000062...\n",
      "Extracting top 10 protein information for ACH-000062...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000062) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  5%|▍         | 14/286 [24:49<8:01:45, 106.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 318 unique proteins and 899 relationships between them\n",
      "Processing sample 15/286: ACH-000065 (OCI-AML5)\n",
      "Sample Index: 15\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000065\n",
      "Extracting top 10 gene information for ACH-000065...\n",
      "Extracting top 10 transcript information for ACH-000065...\n",
      "Extracting top 10 protein information for ACH-000065...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000065) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  5%|▌         | 15/286 [28:15<10:15:26, 136.26s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 225 unique proteins and 543 relationships between them\n",
      "Processing sample 16/286: ACH-000072 (MEG-01)\n",
      "Sample Index: 16\n",
      "Sample Disease: blast phase chronic myelogenous leukemia, bcr-abl1 positive\n",
      "Sample Disease BMGC ID: BMGC_DS00276\n",
      "Sample ACH Name: ACH-000072\n",
      "Extracting top 10 gene information for ACH-000072...\n",
      "Extracting top 10 transcript information for ACH-000072...\n",
      "Extracting top 10 protein information for ACH-000072...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00276 (ACH-000072) ...\n",
      "Knowledge Graph Info: Found 3 disease-related proteins directly connected to BMGC_DS00276 and 1738 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  6%|▌         | 16/286 [28:29<7:26:38, 99.25s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 22 unique proteins and 33 relationships between them\n",
      "Processing sample 17/286: ACH-000073 (GRANTA-519)\n",
      "Sample Index: 17\n",
      "Sample Disease: mantle cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS07533\n",
      "Sample ACH Name: ACH-000073\n",
      "Extracting top 10 gene information for ACH-000073...\n",
      "Extracting top 10 transcript information for ACH-000073...\n",
      "Extracting top 10 protein information for ACH-000073...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07533 (ACH-000073) ...\n",
      "Knowledge Graph Info: Found 26 disease-related proteins directly connected to BMGC_DS07533 and 13413 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  6%|▌         | 17/286 [29:56<7:09:30, 95.80s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 35 unique proteins and 39 relationships between them\n",
      "Processing sample 18/286: ACH-000074 (KU812)\n",
      "Sample Index: 18\n",
      "Sample Disease: chronic myelogenous leukemia, bcr-abl1 positive\n",
      "Sample Disease BMGC ID: BMGC_DS01409\n",
      "Sample ACH Name: ACH-000074\n",
      "Extracting top 10 gene information for ACH-000074...\n",
      "Extracting top 10 transcript information for ACH-000074...\n",
      "Extracting top 10 protein information for ACH-000074...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01409 (ACH-000074) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS01409 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  6%|▋         | 18/286 [29:58<5:02:09, 67.65s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 33 relationships between them\n",
      "Processing sample 19/286: ACH-000078 (MHH-NB-11)\n",
      "Sample Index: 19\n",
      "Sample Disease: neuroblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS01722\n",
      "Sample ACH Name: ACH-000078\n",
      "Extracting top 10 gene information for ACH-000078...\n",
      "Extracting top 10 transcript information for ACH-000078...\n",
      "Extracting top 10 protein information for ACH-000078...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01722 (ACH-000078) ...\n",
      "Knowledge Graph Info: Found 152 disease-related proteins directly connected to BMGC_DS01722 and 23424 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  7%|▋         | 19/286 [32:46<7:14:43, 97.69s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 90 unique proteins and 112 relationships between them\n",
      "Processing sample 20/286: ACH-000081 (GDM-1)\n",
      "Sample Index: 20\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000081\n",
      "Extracting top 10 gene information for ACH-000081...\n",
      "Extracting top 10 transcript information for ACH-000081...\n",
      "Extracting top 10 protein information for ACH-000081...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000081) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 225 unique proteins and 530 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  7%|▋         | 20/286 [36:23<9:52:26, 133.63s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 20/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000081\n",
      "Progress: 20/286 (7.0%), Remaining: 266\n",
      "Processing sample 21/286: ACH-000082 (G-292, clone A141B1)\n",
      "Sample Index: 21\n",
      "Sample Disease: osteosarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS01819\n",
      "Sample ACH Name: ACH-000082\n",
      "Extracting top 10 gene information for ACH-000082...\n",
      "Extracting top 10 transcript information for ACH-000082...\n",
      "Extracting top 10 protein information for ACH-000082...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01819 (ACH-000082) ...\n",
      "Knowledge Graph Info: Found 86 disease-related proteins directly connected to BMGC_DS01819 and 18785 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  7%|▋         | 21/286 [38:51<10:07:58, 137.65s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 68 unique proteins and 53 relationships between them\n",
      "Processing sample 22/286: ACH-000091 (OV56)\n",
      "Sample Index: 22\n",
      "Sample Disease: ovarian serous adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08978\n",
      "Sample ACH Name: ACH-000091\n",
      "Extracting top 10 gene information for ACH-000091...\n",
      "Extracting top 10 transcript information for ACH-000091...\n",
      "Extracting top 10 protein information for ACH-000091...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08978 (ACH-000091) ...\n",
      "Knowledge Graph Info: Found 50 disease-related proteins directly connected to BMGC_DS08978 and 15397 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  8%|▊         | 22/286 [40:48<9:38:53, 131.57s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 55 unique proteins and 28 relationships between them\n",
      "Processing sample 23/286: ACH-000099 (SIMA)\n",
      "Sample Index: 23\n",
      "Sample Disease: adrenal gland neuroblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS06911\n",
      "Sample ACH Name: ACH-000099\n",
      "Extracting top 10 gene information for ACH-000099...\n",
      "Extracting top 10 transcript information for ACH-000099...\n",
      "Extracting top 10 protein information for ACH-000099...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06911 (ACH-000099) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS06911 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  8%|▊         | 23/286 [40:50<6:46:36, 92.76s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 33 relationships between them\n",
      "Processing sample 24/286: ACH-000101 (KE-37)\n",
      "Sample Index: 24\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-000101\n",
      "Extracting top 10 gene information for ACH-000101...\n",
      "Extracting top 10 transcript information for ACH-000101...\n",
      "Extracting top 10 protein information for ACH-000101...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-000101) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  8%|▊         | 24/286 [43:37<8:22:24, 115.06s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 106 unique proteins and 147 relationships between them\n",
      "Processing sample 25/286: ACH-000102 (GMS-10)\n",
      "Sample Index: 25\n",
      "Sample Disease: glioblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS00965\n",
      "Sample ACH Name: ACH-000102\n",
      "Extracting top 10 gene information for ACH-000102...\n",
      "Extracting top 10 transcript information for ACH-000102...\n",
      "Extracting top 10 protein information for ACH-000102...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00965 (ACH-000102) ...\n",
      "Knowledge Graph Info: Found 219 disease-related proteins directly connected to BMGC_DS00965 and 24892 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  9%|▊         | 25/286 [47:08<10:25:41, 143.84s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 138 unique proteins and 375 relationships between them\n",
      "Processing sample 26/286: ACH-000104 (Loucy)\n",
      "Sample Index: 26\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-000104\n",
      "Extracting top 10 gene information for ACH-000104...\n",
      "Extracting top 10 transcript information for ACH-000104...\n",
      "Extracting top 10 protein information for ACH-000104...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-000104) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  9%|▉         | 26/286 [49:54<10:51:35, 150.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 106 unique proteins and 164 relationships between them\n",
      "Processing sample 27/286: ACH-000105 (ALL-SIL)\n",
      "Sample Index: 27\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-000105\n",
      "Extracting top 10 gene information for ACH-000105...\n",
      "Extracting top 10 transcript information for ACH-000105...\n",
      "Extracting top 10 protein information for ACH-000105...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-000105) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  9%|▉         | 27/286 [52:27<10:53:24, 151.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 104 unique proteins and 171 relationships between them\n",
      "Processing sample 28/286: ACH-000106 (JVM-2)\n",
      "Sample Index: 28\n",
      "Sample Disease: mantle cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS07533\n",
      "Sample ACH Name: ACH-000106\n",
      "Extracting top 10 gene information for ACH-000106...\n",
      "Extracting top 10 transcript information for ACH-000106...\n",
      "Extracting top 10 protein information for ACH-000106...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07533 (ACH-000106) ...\n",
      "Knowledge Graph Info: Found 26 disease-related proteins directly connected to BMGC_DS07533 and 13413 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 10%|▉         | 28/286 [53:56<9:29:53, 132.53s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 35 unique proteins and 23 relationships between them\n",
      "Processing sample 29/286: ACH-000107 (Capan-2)\n",
      "Sample Index: 29\n",
      "Sample Disease: pancreatic ductal adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08986\n",
      "Sample ACH Name: ACH-000107\n",
      "Extracting top 10 gene information for ACH-000107...\n",
      "Extracting top 10 transcript information for ACH-000107...\n",
      "Extracting top 10 protein information for ACH-000107...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08986 (ACH-000107) ...\n",
      "Knowledge Graph Info: Found 30 disease-related proteins directly connected to BMGC_DS08986 and 11791 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 10%|█         | 29/286 [55:16<8:19:29, 116.61s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 38 unique proteins and 62 relationships between them\n",
      "Processing sample 30/286: ACH-000112 (SIG-M5)\n",
      "Sample Index: 30\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000112\n",
      "Extracting top 10 gene information for ACH-000112...\n",
      "Extracting top 10 transcript information for ACH-000112...\n",
      "Extracting top 10 protein information for ACH-000112...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000112) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 225 unique proteins and 568 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 10%|█         | 30/286 [58:46<10:17:17, 144.68s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 30/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000112\n",
      "Progress: 30/286 (10.5%), Remaining: 256\n",
      "Processing sample 31/286: ACH-000116 (OAW28)\n",
      "Sample Index: 31\n",
      "Sample Disease: ovarian adenocarcinoma or ovary adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07934\n",
      "Sample ACH Name: ACH-000116\n",
      "Extracting top 10 gene information for ACH-000116...\n",
      "Extracting top 10 transcript information for ACH-000116...\n",
      "Extracting top 10 protein information for ACH-000116...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07934 (ACH-000116) ...\n",
      "Knowledge Graph Info: Found 5 disease-related proteins directly connected to BMGC_DS07934 and 4915 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 11%|█         | 31/286 [59:20<7:53:32, 111.42s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 22 unique proteins and 72 relationships between them\n",
      "Processing sample 32/286: ACH-000117 (EFM-192A)\n",
      "Sample Index: 32\n",
      "Sample Disease: breast carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07015\n",
      "Sample ACH Name: ACH-000117\n",
      "Extracting top 10 gene information for ACH-000117...\n",
      "Extracting top 10 transcript information for ACH-000117...\n",
      "Extracting top 10 protein information for ACH-000117...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07015 (ACH-000117) ...\n",
      "Knowledge Graph Info: Found 178 disease-related proteins directly connected to BMGC_DS07015 and 22704 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 11%|█         | 32/286 [1:02:06<9:01:11, 127.84s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 97 unique proteins and 128 relationships between them\n",
      "Processing sample 33/286: ACH-000120 (CHP-212)\n",
      "Sample Index: 33\n",
      "Sample Disease: neuroblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS01722\n",
      "Sample ACH Name: ACH-000120\n",
      "Extracting top 10 gene information for ACH-000120...\n",
      "Extracting top 10 transcript information for ACH-000120...\n",
      "Extracting top 10 protein information for ACH-000120...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01722 (ACH-000120) ...\n",
      "Knowledge Graph Info: Found 152 disease-related proteins directly connected to BMGC_DS01722 and 23424 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 12%|█▏        | 33/286 [1:04:53<9:48:52, 139.65s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 90 unique proteins and 138 relationships between them\n",
      "Processing sample 34/286: ACH-000121 (NCI-H2405)\n",
      "Sample Index: 34\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000121\n",
      "Extracting top 10 gene information for ACH-000121...\n",
      "Extracting top 10 transcript information for ACH-000121...\n",
      "Extracting top 10 protein information for ACH-000121...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000121) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 12%|█▏        | 34/286 [1:08:52<11:52:03, 169.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 310 unique proteins and 763 relationships between them\n",
      "Processing sample 35/286: ACH-000124 (OCI-LY-19)\n",
      "Sample Index: 35\n",
      "Sample Disease: diffuse large b-cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS02693\n",
      "Sample ACH Name: ACH-000124\n",
      "Extracting top 10 gene information for ACH-000124...\n",
      "Extracting top 10 transcript information for ACH-000124...\n",
      "Extracting top 10 protein information for ACH-000124...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02693 (ACH-000124) ...\n",
      "Knowledge Graph Info: Found 148 disease-related proteins directly connected to BMGC_DS02693 and 20906 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 12%|█▏        | 35/286 [1:11:24<11:26:41, 164.15s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 88 unique proteins and 100 relationships between them\n",
      "Processing sample 36/286: ACH-000129 (NCI-H1341)\n",
      "Sample Index: 36\n",
      "Sample Disease: cervical small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05350\n",
      "Sample ACH Name: ACH-000129\n",
      "Extracting top 10 gene information for ACH-000129...\n",
      "Extracting top 10 transcript information for ACH-000129...\n",
      "Extracting top 10 protein information for ACH-000129...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05350 (ACH-000129) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS05350 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 13%|█▎        | 36/286 [1:11:26<8:01:30, 115.56s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 48 relationships between them\n",
      "Processing sample 37/286: ACH-000136 (CHP-126)\n",
      "Sample Index: 37\n",
      "Sample Disease: neuroblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS01722\n",
      "Sample ACH Name: ACH-000136\n",
      "Extracting top 10 gene information for ACH-000136...\n",
      "Extracting top 10 transcript information for ACH-000136...\n",
      "Extracting top 10 protein information for ACH-000136...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01722 (ACH-000136) ...\n",
      "Knowledge Graph Info: Found 152 disease-related proteins directly connected to BMGC_DS01722 and 23424 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 13%|█▎        | 37/286 [1:14:14<9:04:32, 131.21s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 89 unique proteins and 114 relationships between them\n",
      "Processing sample 38/286: ACH-000137 (8-MG-BA)\n",
      "Sample Index: 38\n",
      "Sample Disease: glioblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS00965\n",
      "Sample ACH Name: ACH-000137\n",
      "Extracting top 10 gene information for ACH-000137...\n",
      "Extracting top 10 transcript information for ACH-000137...\n",
      "Extracting top 10 protein information for ACH-000137...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00965 (ACH-000137) ...\n",
      "Knowledge Graph Info: Found 219 disease-related proteins directly connected to BMGC_DS00965 and 24892 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 13%|█▎        | 38/286 [1:17:18<10:08:24, 147.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 138 unique proteins and 361 relationships between them\n",
      "Processing sample 39/286: ACH-000144 (RERF-GC-1B)\n",
      "Sample Index: 39\n",
      "Sample Disease: gastric adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05289\n",
      "Sample ACH Name: ACH-000144\n",
      "Extracting top 10 gene information for ACH-000144...\n",
      "Extracting top 10 transcript information for ACH-000144...\n",
      "Extracting top 10 protein information for ACH-000144...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05289 (ACH-000144) ...\n",
      "Knowledge Graph Info: Found 215 disease-related proteins directly connected to BMGC_DS05289 and 23595 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 14%|█▎        | 39/286 [1:20:14<10:41:21, 155.79s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 105 unique proteins and 222 relationships between them\n",
      "Processing sample 40/286: ACH-000145 (SK-LMS-1)\n",
      "Sample Index: 40\n",
      "Sample Disease: leiomyosarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS01379\n",
      "Sample ACH Name: ACH-000145\n",
      "Extracting top 10 gene information for ACH-000145...\n",
      "Extracting top 10 transcript information for ACH-000145...\n",
      "Extracting top 10 protein information for ACH-000145...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01379 (ACH-000145) ...\n",
      "Knowledge Graph Info: Found 15 disease-related proteins directly connected to BMGC_DS01379 and 7099 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 27 unique proteins and 18 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 14%|█▍        | 40/286 [1:21:07<8:32:32, 125.01s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 40/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000145\n",
      "Progress: 40/286 (14.0%), Remaining: 246\n",
      "Processing sample 41/286: ACH-000151 (JM1)\n",
      "Sample Index: 41\n",
      "Sample Disease: precursor b-cell acute lymphoblastic leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS06157\n",
      "Sample ACH Name: ACH-000151\n",
      "Extracting top 10 gene information for ACH-000151...\n",
      "Extracting top 10 transcript information for ACH-000151...\n",
      "Extracting top 10 protein information for ACH-000151...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06157 (ACH-000151) ...\n",
      "Knowledge Graph Info: Found 14 disease-related proteins directly connected to BMGC_DS06157 and 5953 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 14%|█▍        | 41/286 [1:21:48<6:47:32, 99.81s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 24 unique proteins and 23 relationships between them\n",
      "Processing sample 42/286: ACH-000157 (A4/Fuk)\n",
      "Sample Index: 42\n",
      "Sample Disease: precursor b-cell acute lymphoblastic leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS06157\n",
      "Sample ACH Name: ACH-000157\n",
      "Extracting top 10 gene information for ACH-000157...\n",
      "Extracting top 10 transcript information for ACH-000157...\n",
      "Extracting top 10 protein information for ACH-000157...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06157 (ACH-000157) ...\n",
      "Knowledge Graph Info: Found 14 disease-related proteins directly connected to BMGC_DS06157 and 5953 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 15%|█▍        | 42/286 [1:22:30<5:35:19, 82.46s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 24 unique proteins and 27 relationships between them\n",
      "Processing sample 43/286: ACH-000162 (GA-10)\n",
      "Sample Index: 43\n",
      "Sample Disease: burkitt lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS00331\n",
      "Sample ACH Name: ACH-000162\n",
      "Extracting top 10 gene information for ACH-000162...\n",
      "Extracting top 10 transcript information for ACH-000162...\n",
      "Extracting top 10 protein information for ACH-000162...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00331 (ACH-000162) ...\n",
      "Knowledge Graph Info: Found 67 disease-related proteins directly connected to BMGC_DS00331 and 15225 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 15%|█▌        | 43/286 [1:24:15<6:01:28, 89.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 43 unique proteins and 36 relationships between them\n",
      "Processing sample 44/286: ACH-000171 (VMRC-RCZ)\n",
      "Sample Index: 44\n",
      "Sample Disease: renal cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00375\n",
      "Sample ACH Name: ACH-000171\n",
      "Extracting top 10 gene information for ACH-000171...\n",
      "Extracting top 10 transcript information for ACH-000171...\n",
      "Extracting top 10 protein information for ACH-000171...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00375 (ACH-000171) ...\n",
      "Knowledge Graph Info: Found 341 disease-related proteins directly connected to BMGC_DS00375 and 25130 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 15%|█▌        | 44/286 [1:27:36<8:14:39, 122.64s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 199 unique proteins and 268 relationships between them\n",
      "Processing sample 45/286: ACH-000174 (CAL-62)\n",
      "Sample Index: 45\n",
      "Sample Disease: thyroid carcinoma, anaplastic\n",
      "Sample Disease BMGC ID: BMGC_DS21147\n",
      "Sample ACH Name: ACH-000174\n",
      "Extracting top 10 gene information for ACH-000174...\n",
      "Extracting top 10 transcript information for ACH-000174...\n",
      "Extracting top 10 protein information for ACH-000174...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS21147 (ACH-000174) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS21147 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 16%|█▌        | 45/286 [1:27:38<5:47:30, 86.52s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 27 relationships between them\n",
      "Processing sample 46/286: ACH-000181 (SCC-9)\n",
      "Sample Index: 46\n",
      "Sample Disease: tongue squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS06143\n",
      "Sample ACH Name: ACH-000181\n",
      "Extracting top 10 gene information for ACH-000181...\n",
      "Extracting top 10 transcript information for ACH-000181...\n",
      "Extracting top 10 protein information for ACH-000181...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06143 (ACH-000181) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS06143 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 16%|█▌        | 46/286 [1:27:41<4:06:10, 61.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 72 relationships between them\n",
      "Processing sample 47/286: ACH-000186 (NCI-H2444)\n",
      "Sample Index: 47\n",
      "Sample Disease: lung non-small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00373\n",
      "Sample ACH Name: ACH-000186\n",
      "Extracting top 10 gene information for ACH-000186...\n",
      "Extracting top 10 transcript information for ACH-000186...\n",
      "Extracting top 10 protein information for ACH-000186...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00373 (ACH-000186) ...\n",
      "Knowledge Graph Info: Found 482 disease-related proteins directly connected to BMGC_DS00373 and 27691 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 16%|█▋        | 47/286 [1:31:42<7:39:14, 115.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 282 unique proteins and 998 relationships between them\n",
      "Processing sample 48/286: ACH-000187 (COR-L311)\n",
      "Sample Index: 48\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000187\n",
      "Extracting top 10 gene information for ACH-000187...\n",
      "Extracting top 10 transcript information for ACH-000187...\n",
      "Extracting top 10 protein information for ACH-000187...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000187) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 17%|█▋        | 48/286 [1:34:30<8:40:15, 131.16s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 156 relationships between them\n",
      "Processing sample 49/286: ACH-000189 (RCC10RGB)\n",
      "Sample Index: 49\n",
      "Sample Disease: renal cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00375\n",
      "Sample ACH Name: ACH-000189\n",
      "Extracting top 10 gene information for ACH-000189...\n",
      "Extracting top 10 transcript information for ACH-000189...\n",
      "Extracting top 10 protein information for ACH-000189...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00375 (ACH-000189) ...\n",
      "Knowledge Graph Info: Found 341 disease-related proteins directly connected to BMGC_DS00375 and 25130 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 17%|█▋        | 49/286 [1:37:54<10:03:52, 152.88s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 199 unique proteins and 257 relationships between them\n",
      "Processing sample 50/286: ACH-000190 (HD-MY-Z)\n",
      "Sample Index: 50\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000190\n",
      "Extracting top 10 gene information for ACH-000190...\n",
      "Extracting top 10 transcript information for ACH-000190...\n",
      "Extracting top 10 protein information for ACH-000190...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000190) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 225 unique proteins and 528 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 17%|█▋        | 50/286 [1:41:25<11:10:10, 170.38s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 50/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000190\n",
      "Progress: 50/286 (17.5%), Remaining: 236\n",
      "Processing sample 51/286: ACH-000193 (KARPAS-620)\n",
      "Sample Index: 51\n",
      "Sample Disease: plasma cell myeloma\n",
      "Sample Disease BMGC ID: BMGC_DS01639\n",
      "Sample ACH Name: ACH-000193\n",
      "Extracting top 10 gene information for ACH-000193...\n",
      "Extracting top 10 transcript information for ACH-000193...\n",
      "Extracting top 10 protein information for ACH-000193...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01639 (ACH-000193) ...\n",
      "Knowledge Graph Info: Found 244 disease-related proteins directly connected to BMGC_DS01639 and 24451 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 18%|█▊        | 51/286 [1:44:29<11:23:54, 174.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 179 unique proteins and 183 relationships between them\n",
      "Processing sample 52/286: ACH-000196 (HCC1599)\n",
      "Sample Index: 52\n",
      "Sample Disease: breast ductal carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08035\n",
      "Sample ACH Name: ACH-000196\n",
      "Extracting top 10 gene information for ACH-000196...\n",
      "Extracting top 10 transcript information for ACH-000196...\n",
      "Extracting top 10 protein information for ACH-000196...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08035 (ACH-000196) ...\n",
      "Knowledge Graph Info: Found 17 disease-related proteins directly connected to BMGC_DS08035 and 13835 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 18%|█▊        | 52/286 [1:45:59<9:41:25, 149.08s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 32 unique proteins and 37 relationships between them\n",
      "Processing sample 53/286: ACH-000197 (TALL-1)\n",
      "Sample Index: 53\n",
      "Sample Disease: leukemia-lymphoma, adult t-cell\n",
      "Sample Disease BMGC ID: BMGC_DS20796\n",
      "Sample ACH Name: ACH-000197\n",
      "Extracting top 10 gene information for ACH-000197...\n",
      "Extracting top 10 transcript information for ACH-000197...\n",
      "Extracting top 10 protein information for ACH-000197...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20796 (ACH-000197) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20796 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 19%|█▊        | 53/286 [1:46:02<6:48:14, 105.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 30 relationships between them\n",
      "Processing sample 54/286: ACH-000198 (EOL-1)\n",
      "Sample Index: 54\n",
      "Sample Disease: chronic eosinophilic leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS06075\n",
      "Sample ACH Name: ACH-000198\n",
      "Extracting top 10 gene information for ACH-000198...\n",
      "Extracting top 10 transcript information for ACH-000198...\n",
      "Extracting top 10 protein information for ACH-000198...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06075 (ACH-000198) ...\n",
      "Knowledge Graph Info: Found 34 disease-related proteins directly connected to BMGC_DS06075 and 3954 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 19%|█▉        | 54/286 [1:46:34<5:22:27, 83.39s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 22 unique proteins and 26 relationships between them\n",
      "Processing sample 55/286: ACH-000200 (NMC-G1)\n",
      "Sample Index: 55\n",
      "Sample Disease: astrocytoma\n",
      "Sample Disease BMGC ID: BMGC_DS20147\n",
      "Sample ACH Name: ACH-000200\n",
      "Extracting top 10 gene information for ACH-000200...\n",
      "Extracting top 10 transcript information for ACH-000200...\n",
      "Extracting top 10 protein information for ACH-000200...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20147 (ACH-000200) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20147 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 19%|█▉        | 55/286 [1:46:37<3:48:07, 59.26s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 26 relationships between them\n",
      "Processing sample 56/286: ACH-000201 (A-204)\n",
      "Sample Index: 56\n",
      "Sample Disease: embryonal rhabdomyosarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS04066\n",
      "Sample ACH Name: ACH-000201\n",
      "Extracting top 10 gene information for ACH-000201...\n",
      "Extracting top 10 transcript information for ACH-000201...\n",
      "Extracting top 10 protein information for ACH-000201...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS04066 (ACH-000201) ...\n",
      "Knowledge Graph Info: Found 28 disease-related proteins directly connected to BMGC_DS04066 and 13010 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|█▉        | 56/286 [1:48:03<4:18:14, 67.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 32 unique proteins and 18 relationships between them\n",
      "Processing sample 57/286: ACH-000204 (LP-1)\n",
      "Sample Index: 57\n",
      "Sample Disease: plasma cell myeloma\n",
      "Sample Disease BMGC ID: BMGC_DS01639\n",
      "Sample ACH Name: ACH-000204\n",
      "Extracting top 10 gene information for ACH-000204...\n",
      "Extracting top 10 transcript information for ACH-000204...\n",
      "Extracting top 10 protein information for ACH-000204...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01639 (ACH-000204) ...\n",
      "Knowledge Graph Info: Found 244 disease-related proteins directly connected to BMGC_DS01639 and 24451 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|█▉        | 57/286 [1:51:08<6:31:13, 102.50s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 178 unique proteins and 160 relationships between them\n",
      "Processing sample 58/286: ACH-000218 (PL-21)\n",
      "Sample Index: 58\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000218\n",
      "Extracting top 10 gene information for ACH-000218...\n",
      "Extracting top 10 transcript information for ACH-000218...\n",
      "Extracting top 10 protein information for ACH-000218...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000218) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|██        | 58/286 [1:54:33<8:26:29, 133.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 224 unique proteins and 529 relationships between them\n",
      "Processing sample 59/286: ACH-000221 (SNU-398)\n",
      "Sample Index: 59\n",
      "Sample Disease: adult hepatocellular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05326\n",
      "Sample ACH Name: ACH-000221\n",
      "Extracting top 10 gene information for ACH-000221...\n",
      "Extracting top 10 transcript information for ACH-000221...\n",
      "Extracting top 10 protein information for ACH-000221...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05326 (ACH-000221) ...\n",
      "Knowledge Graph Info: Found 23 disease-related proteins directly connected to BMGC_DS05326 and 7706 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 21%|██        | 59/286 [1:55:27<6:54:43, 109.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 25 unique proteins and 22 relationships between them\n",
      "Processing sample 60/286: ACH-000225 (ECC12)\n",
      "Sample Index: 60\n",
      "Sample Disease: gastric small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08739\n",
      "Sample ACH Name: ACH-000225\n",
      "Extracting top 10 gene information for ACH-000225...\n",
      "Extracting top 10 transcript information for ACH-000225...\n",
      "Extracting top 10 protein information for ACH-000225...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08739 (ACH-000225) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS08739 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 20 unique proteins and 44 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 21%|██        | 60/286 [1:55:34<4:56:38, 78.75s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 60/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000225\n",
      "Progress: 60/286 (21.0%), Remaining: 226\n",
      "Processing sample 61/286: ACH-000226 (SUP-M2)\n",
      "Sample Index: 61\n",
      "Sample Disease: alk-positive anaplastic large cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS08403\n",
      "Sample ACH Name: ACH-000226\n",
      "Extracting top 10 gene information for ACH-000226...\n",
      "Extracting top 10 transcript information for ACH-000226...\n",
      "Extracting top 10 protein information for ACH-000226...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08403 (ACH-000226) ...\n",
      "Knowledge Graph Info: Found 2 disease-related proteins directly connected to BMGC_DS08403 and 2478 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 21%|██▏       | 61/286 [1:55:52<3:46:54, 60.51s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 21 unique proteins and 16 relationships between them\n",
      "Processing sample 62/286: ACH-000228 (BICR 31)\n",
      "Sample Index: 62\n",
      "Sample Disease: tongue squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS06143\n",
      "Sample ACH Name: ACH-000228\n",
      "Extracting top 10 gene information for ACH-000228...\n",
      "Extracting top 10 transcript information for ACH-000228...\n",
      "Extracting top 10 protein information for ACH-000228...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06143 (ACH-000228) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS06143 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 22%|██▏       | 62/286 [1:55:55<2:41:09, 43.17s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 78 relationships between them\n",
      "Processing sample 63/286: ACH-000233 (DEL)\n",
      "Sample Index: 63\n",
      "Sample Disease: alk-positive anaplastic large cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS08403\n",
      "Sample ACH Name: ACH-000233\n",
      "Extracting top 10 gene information for ACH-000233...\n",
      "Extracting top 10 transcript information for ACH-000233...\n",
      "Extracting top 10 protein information for ACH-000233...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08403 (ACH-000233) ...\n",
      "Knowledge Graph Info: Found 2 disease-related proteins directly connected to BMGC_DS08403 and 2478 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 22%|██▏       | 63/286 [1:56:13<2:12:19, 35.60s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 21 unique proteins and 16 relationships between them\n",
      "Processing sample 64/286: ACH-000236 (SW1417)\n",
      "Sample Index: 64\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000236\n",
      "Extracting top 10 gene information for ACH-000236...\n",
      "Extracting top 10 transcript information for ACH-000236...\n",
      "Extracting top 10 protein information for ACH-000236...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000236) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 22%|██▏       | 64/286 [1:58:25<3:59:01, 64.60s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 101 relationships between them\n",
      "Processing sample 65/286: ACH-000242 (RT4)\n",
      "Sample Index: 65\n",
      "Sample Disease: bladder carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07053\n",
      "Sample ACH Name: ACH-000242\n",
      "Extracting top 10 gene information for ACH-000242...\n",
      "Extracting top 10 transcript information for ACH-000242...\n",
      "Extracting top 10 protein information for ACH-000242...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07053 (ACH-000242) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS07053 and 8832 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 23%|██▎       | 65/286 [1:59:25<3:52:30, 63.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 24 relationships between them\n",
      "Processing sample 66/286: ACH-000245 (BL-41)\n",
      "Sample Index: 66\n",
      "Sample Disease: burkitt lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS00331\n",
      "Sample ACH Name: ACH-000245\n",
      "Extracting top 10 gene information for ACH-000245...\n",
      "Extracting top 10 transcript information for ACH-000245...\n",
      "Extracting top 10 protein information for ACH-000245...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00331 (ACH-000245) ...\n",
      "Knowledge Graph Info: Found 67 disease-related proteins directly connected to BMGC_DS00331 and 15225 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 23%|██▎       | 66/286 [2:01:11<4:38:31, 75.96s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 43 unique proteins and 60 relationships between them\n",
      "Processing sample 67/286: ACH-000252 (LS1034)\n",
      "Sample Index: 67\n",
      "Sample Disease: cecum adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08517\n",
      "Sample ACH Name: ACH-000252\n",
      "Extracting top 10 gene information for ACH-000252...\n",
      "Extracting top 10 transcript information for ACH-000252...\n",
      "Extracting top 10 protein information for ACH-000252...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08517 (ACH-000252) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS08517 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 23%|██▎       | 67/286 [2:01:13<3:17:01, 53.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 56 relationships between them\n",
      "Processing sample 68/286: ACH-000254 (SCC-15)\n",
      "Sample Index: 68\n",
      "Sample Disease: tongue squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS06143\n",
      "Sample ACH Name: ACH-000254\n",
      "Extracting top 10 gene information for ACH-000254...\n",
      "Extracting top 10 transcript information for ACH-000254...\n",
      "Extracting top 10 protein information for ACH-000254...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06143 (ACH-000254) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS06143 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 24%|██▍       | 68/286 [2:01:16<2:19:50, 38.49s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 110 relationships between them\n",
      "Processing sample 69/286: ACH-000267 (HDLM-2)\n",
      "Sample Index: 69\n",
      "Sample Disease: hodgkin disease or hodgkins lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS20405\n",
      "Sample ACH Name: ACH-000267\n",
      "Extracting top 10 gene information for ACH-000267...\n",
      "Extracting top 10 transcript information for ACH-000267...\n",
      "Extracting top 10 protein information for ACH-000267...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20405 (ACH-000267) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20405 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 24%|██▍       | 69/286 [2:01:18<1:39:51, 27.61s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 37 relationships between them\n",
      "Processing sample 70/286: ACH-000270 (HPAC)\n",
      "Sample Index: 70\n",
      "Sample Disease: pancreatic adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05421\n",
      "Sample ACH Name: ACH-000270\n",
      "Extracting top 10 gene information for ACH-000270...\n",
      "Extracting top 10 transcript information for ACH-000270...\n",
      "Extracting top 10 protein information for ACH-000270...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05421 (ACH-000270) ...\n",
      "Knowledge Graph Info: Found 82 disease-related proteins directly connected to BMGC_DS05421 and 19751 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 53 unique proteins and 148 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 24%|██▍       | 70/286 [2:03:38<3:41:13, 61.45s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 70/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000270\n",
      "Progress: 70/286 (24.5%), Remaining: 216\n",
      "Processing sample 71/286: ACH-000273 (SF539)\n",
      "Sample Index: 71\n",
      "Sample Disease: gliosarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS04111\n",
      "Sample ACH Name: ACH-000273\n",
      "Extracting top 10 gene information for ACH-000273...\n",
      "Extracting top 10 transcript information for ACH-000273...\n",
      "Extracting top 10 protein information for ACH-000273...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS04111 (ACH-000273) ...\n",
      "Knowledge Graph Info: Found 38 disease-related proteins directly connected to BMGC_DS04111 and 12498 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 25%|██▍       | 71/286 [2:05:03<4:04:45, 68.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 35 unique proteins and 51 relationships between them\n",
      "Processing sample 72/286: ACH-000282 (NCI-H1755)\n",
      "Sample Index: 72\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000282\n",
      "Extracting top 10 gene information for ACH-000282...\n",
      "Extracting top 10 transcript information for ACH-000282...\n",
      "Extracting top 10 protein information for ACH-000282...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000282) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 25%|██▌       | 72/286 [2:08:59<7:03:13, 118.66s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 300 unique proteins and 815 relationships between them\n",
      "Processing sample 73/286: ACH-000287 (NU-DUL-1)\n",
      "Sample Index: 73\n",
      "Sample Disease: diffuse large b-cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS02693\n",
      "Sample ACH Name: ACH-000287\n",
      "Extracting top 10 gene information for ACH-000287...\n",
      "Extracting top 10 transcript information for ACH-000287...\n",
      "Extracting top 10 protein information for ACH-000287...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02693 (ACH-000287) ...\n",
      "Knowledge Graph Info: Found 148 disease-related proteins directly connected to BMGC_DS02693 and 20906 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 26%|██▌       | 73/286 [2:11:29<7:34:36, 128.06s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 88 unique proteins and 102 relationships between them\n",
      "Processing sample 74/286: ACH-000293 (KLE)\n",
      "Sample Index: 74\n",
      "Sample Disease: endometrial adenocarcinoma or endometrial endometrioid adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS09171\n",
      "Sample ACH Name: ACH-000293\n",
      "Extracting top 10 gene information for ACH-000293...\n",
      "Extracting top 10 transcript information for ACH-000293...\n",
      "Extracting top 10 protein information for ACH-000293...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09171 (ACH-000293) ...\n",
      "Knowledge Graph Info: Found 10 disease-related proteins directly connected to BMGC_DS09171 and 9781 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 26%|██▌       | 74/286 [2:12:34<6:26:25, 109.36s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 27 unique proteins and 30 relationships between them\n",
      "Processing sample 75/286: ACH-000295 (EM-2)\n",
      "Sample Index: 75\n",
      "Sample Disease: chronic myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01409\n",
      "Sample ACH Name: ACH-000295\n",
      "Extracting top 10 gene information for ACH-000295...\n",
      "Extracting top 10 transcript information for ACH-000295...\n",
      "Extracting top 10 protein information for ACH-000295...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01409 (ACH-000295) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS01409 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 26%|██▌       | 75/286 [2:12:38<4:32:58, 77.62s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 42 relationships between them\n",
      "Processing sample 76/286: ACH-000298 (NCI-H2029)\n",
      "Sample Index: 76\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000298\n",
      "Extracting top 10 gene information for ACH-000298...\n",
      "Extracting top 10 transcript information for ACH-000298...\n",
      "Extracting top 10 protein information for ACH-000298...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000298) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 27%|██▋       | 76/286 [2:15:21<6:01:42, 103.34s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 192 relationships between them\n",
      "Processing sample 77/286: ACH-000301 (LAMA-84)\n",
      "Sample Index: 77\n",
      "Sample Disease: chronic myelogenous leukemia, bcr-abl1 positive\n",
      "Sample Disease BMGC ID: BMGC_DS01409\n",
      "Sample ACH Name: ACH-000301\n",
      "Extracting top 10 gene information for ACH-000301...\n",
      "Extracting top 10 transcript information for ACH-000301...\n",
      "Extracting top 10 protein information for ACH-000301...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01409 (ACH-000301) ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 27%|██▋       | 77/286 [2:15:24<4:14:25, 73.04s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS01409 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 20 unique proteins and 26 relationships between them\n",
      "Processing sample 78/286: ACH-000309 (SK-LU-1)\n",
      "Sample Index: 78\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000309\n",
      "Extracting top 10 gene information for ACH-000309...\n",
      "Extracting top 10 transcript information for ACH-000309...\n",
      "Extracting top 10 protein information for ACH-000309...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000309) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 27%|██▋       | 78/286 [2:19:21<7:03:32, 122.17s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 309 unique proteins and 675 relationships between them\n",
      "Processing sample 79/286: ACH-000323 (42-MG-BA)\n",
      "Sample Index: 79\n",
      "Sample Disease: glioblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS00965\n",
      "Sample ACH Name: ACH-000323\n",
      "Extracting top 10 gene information for ACH-000323...\n",
      "Extracting top 10 transcript information for ACH-000323...\n",
      "Extracting top 10 protein information for ACH-000323...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00965 (ACH-000323) ...\n",
      "Knowledge Graph Info: Found 219 disease-related proteins directly connected to BMGC_DS00965 and 24892 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 28%|██▊       | 79/286 [2:22:24<8:04:24, 140.41s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 138 unique proteins and 340 relationships between them\n",
      "Processing sample 80/286: ACH-000326 (JURL-MK1)\n",
      "Sample Index: 80\n",
      "Sample Disease: chronic myelogenous leukemia, bcr-abl1 positive\n",
      "Sample Disease BMGC ID: BMGC_DS01409\n",
      "Sample ACH Name: ACH-000326\n",
      "Extracting top 10 gene information for ACH-000326...\n",
      "Extracting top 10 transcript information for ACH-000326...\n",
      "Extracting top 10 protein information for ACH-000326...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01409 (ACH-000326) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS01409 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 20 unique proteins and 26 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 28%|██▊       | 80/286 [2:22:31<5:45:20, 100.59s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 80/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000326\n",
      "Progress: 80/286 (28.0%), Remaining: 206\n",
      "Processing sample 81/286: ACH-000327 (NCI-H1395)\n",
      "Sample Index: 81\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000327\n",
      "Extracting top 10 gene information for ACH-000327...\n",
      "Extracting top 10 transcript information for ACH-000327...\n",
      "Extracting top 10 protein information for ACH-000327...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000327) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 28%|██▊       | 81/286 [2:26:29<8:03:56, 141.64s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 310 unique proteins and 888 relationships between them\n",
      "Processing sample 82/286: ACH-000329 (CCF-STTG1)\n",
      "Sample Index: 82\n",
      "Sample Disease: astrocytoma\n",
      "Sample Disease BMGC ID: BMGC_DS20147\n",
      "Sample ACH Name: ACH-000329\n",
      "Extracting top 10 gene information for ACH-000329...\n",
      "Extracting top 10 transcript information for ACH-000329...\n",
      "Extracting top 10 protein information for ACH-000329...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20147 (ACH-000329) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20147 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 29%|██▊       | 82/286 [2:26:31<5:39:41, 99.91s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 21 relationships between them\n",
      "Processing sample 83/286: ACH-000332 (YAPC)\n",
      "Sample Index: 83\n",
      "Sample Disease: pancreatic carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS04293\n",
      "Sample ACH Name: ACH-000332\n",
      "Extracting top 10 gene information for ACH-000332...\n",
      "Extracting top 10 transcript information for ACH-000332...\n",
      "Extracting top 10 protein information for ACH-000332...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS04293 (ACH-000332) ...\n",
      "Knowledge Graph Info: Found 68 disease-related proteins directly connected to BMGC_DS04293 and 15940 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 29%|██▉       | 83/286 [2:28:24<5:50:40, 103.65s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 45 unique proteins and 48 relationships between them\n",
      "Processing sample 84/286: ACH-000335 (MSTO-211H)\n",
      "Sample Index: 84\n",
      "Sample Disease: pleural mesothelioma\n",
      "Sample Disease BMGC ID: BMGC_DS09237\n",
      "Sample ACH Name: ACH-000335\n",
      "Extracting top 10 gene information for ACH-000335...\n",
      "Extracting top 10 transcript information for ACH-000335...\n",
      "Extracting top 10 protein information for ACH-000335...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09237 (ACH-000335) ...\n",
      "Knowledge Graph Info: Found 2 disease-related proteins directly connected to BMGC_DS09237 and 1692 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 29%|██▉       | 84/286 [2:28:38<4:18:30, 76.78s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 31 unique proteins and 44 relationships between them\n",
      "Processing sample 85/286: ACH-000338 (SR-786)\n",
      "Sample Index: 85\n",
      "Sample Disease: alk-positive anaplastic large cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS08403\n",
      "Sample ACH Name: ACH-000338\n",
      "Extracting top 10 gene information for ACH-000338...\n",
      "Extracting top 10 transcript information for ACH-000338...\n",
      "Extracting top 10 protein information for ACH-000338...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08403 (ACH-000338) ...\n",
      "Knowledge Graph Info: Found 2 disease-related proteins directly connected to BMGC_DS08403 and 2478 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 30%|██▉       | 85/286 [2:28:56<3:18:18, 59.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 21 unique proteins and 27 relationships between them\n",
      "Processing sample 86/286: ACH-000345 (KP-N-RT-BM-1)\n",
      "Sample Index: 86\n",
      "Sample Disease: adrenal gland neuroblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS06911\n",
      "Sample ACH Name: ACH-000345\n",
      "Extracting top 10 gene information for ACH-000345...\n",
      "Extracting top 10 transcript information for ACH-000345...\n",
      "Extracting top 10 protein information for ACH-000345...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06911 (ACH-000345) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS06911 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 30%|███       | 86/286 [2:28:59<2:21:08, 42.34s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 56 relationships between them\n",
      "Processing sample 87/286: ACH-000346 (JVM-3)\n",
      "Sample Index: 87\n",
      "Sample Disease: prolymphocytic leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01415\n",
      "Sample ACH Name: ACH-000346\n",
      "Extracting top 10 gene information for ACH-000346...\n",
      "Extracting top 10 transcript information for ACH-000346...\n",
      "Extracting top 10 protein information for ACH-000346...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01415 (ACH-000346) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS01415 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 30%|███       | 87/286 [2:29:01<1:40:44, 30.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 14 relationships between them\n",
      "Processing sample 88/286: ACH-000347 (QGP-1)\n",
      "Sample Index: 88\n",
      "Sample Disease: pancreatic somatostatinoma\n",
      "Sample Disease BMGC ID: BMGC_DS09199\n",
      "Sample ACH Name: ACH-000347\n",
      "Extracting top 10 gene information for ACH-000347...\n",
      "Extracting top 10 transcript information for ACH-000347...\n",
      "Extracting top 10 protein information for ACH-000347...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09199 (ACH-000347) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS09199 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 31%|███       | 88/286 [2:29:05<1:14:11, 22.48s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 30 unique proteins and 43 relationships between them\n",
      "Processing sample 89/286: ACH-000349 (HCC1500)\n",
      "Sample Index: 89\n",
      "Sample Disease: breast ductal carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08035\n",
      "Sample ACH Name: ACH-000349\n",
      "Extracting top 10 gene information for ACH-000349...\n",
      "Extracting top 10 transcript information for ACH-000349...\n",
      "Extracting top 10 protein information for ACH-000349...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08035 (ACH-000349) ...\n",
      "Knowledge Graph Info: Found 17 disease-related proteins directly connected to BMGC_DS08035 and 13835 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 31%|███       | 89/286 [2:30:36<2:20:54, 42.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 42 unique proteins and 63 relationships between them\n",
      "Processing sample 90/286: ACH-000351 (MKN1)\n",
      "Sample Index: 90\n",
      "Sample Disease: gastric adenosquamous carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08728\n",
      "Sample ACH Name: ACH-000351\n",
      "Extracting top 10 gene information for ACH-000351...\n",
      "Extracting top 10 transcript information for ACH-000351...\n",
      "Extracting top 10 protein information for ACH-000351...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08728 (ACH-000351) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS08728 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 30 unique proteins and 42 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 31%|███▏      | 90/286 [2:30:46<1:47:57, 33.05s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 90/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000351\n",
      "Progress: 90/286 (31.5%), Remaining: 196\n",
      "Processing sample 91/286: ACH-000353 (TE-15)\n",
      "Sample Index: 91\n",
      "Sample Disease: esophageal squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05334\n",
      "Sample ACH Name: ACH-000353\n",
      "Extracting top 10 gene information for ACH-000353...\n",
      "Extracting top 10 transcript information for ACH-000353...\n",
      "Extracting top 10 protein information for ACH-000353...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05334 (ACH-000353) ...\n",
      "Knowledge Graph Info: Found 226 disease-related proteins directly connected to BMGC_DS05334 and 22772 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 32%|███▏      | 91/286 [2:33:39<4:03:33, 74.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 146 unique proteins and 214 relationships between them\n",
      "Processing sample 92/286: ACH-000354 (Capan-1)\n",
      "Sample Index: 92\n",
      "Sample Disease: pancreatic ductal adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08986\n",
      "Sample ACH Name: ACH-000354\n",
      "Extracting top 10 gene information for ACH-000354...\n",
      "Extracting top 10 transcript information for ACH-000354...\n",
      "Extracting top 10 protein information for ACH-000354...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08986 (ACH-000354) ...\n",
      "Knowledge Graph Info: Found 30 disease-related proteins directly connected to BMGC_DS08986 and 11791 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 32%|███▏      | 92/286 [2:34:58<4:06:43, 76.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 38 unique proteins and 66 relationships between them\n",
      "Processing sample 93/286: ACH-000355 (NCI-H82)\n",
      "Sample Index: 93\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000355\n",
      "Extracting top 10 gene information for ACH-000355...\n",
      "Extracting top 10 transcript information for ACH-000355...\n",
      "Extracting top 10 protein information for ACH-000355...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000355) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 33%|███▎      | 93/286 [2:37:44<5:31:53, 103.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 133 relationships between them\n",
      "Processing sample 94/286: ACH-000357 (JeKo-1)\n",
      "Sample Index: 94\n",
      "Sample Disease: mantle cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS07533\n",
      "Sample ACH Name: ACH-000357\n",
      "Extracting top 10 gene information for ACH-000357...\n",
      "Extracting top 10 transcript information for ACH-000357...\n",
      "Extracting top 10 protein information for ACH-000357...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07533 (ACH-000357) ...\n",
      "Knowledge Graph Info: Found 26 disease-related proteins directly connected to BMGC_DS07533 and 13413 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 33%|███▎      | 94/286 [2:39:13<5:16:16, 98.84s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 35 unique proteins and 35 relationships between them\n",
      "Processing sample 95/286: ACH-000358 (NCI-H69)\n",
      "Sample Index: 95\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000358\n",
      "Extracting top 10 gene information for ACH-000358...\n",
      "Extracting top 10 transcript information for ACH-000358...\n",
      "Extracting top 10 protein information for ACH-000358...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000358) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 33%|███▎      | 95/286 [2:42:03<6:22:35, 120.19s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 166 relationships between them\n",
      "Processing sample 96/286: ACH-000359 (MG-63)\n",
      "Sample Index: 96\n",
      "Sample Disease: osteosarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS01819\n",
      "Sample ACH Name: ACH-000359\n",
      "Extracting top 10 gene information for ACH-000359...\n",
      "Extracting top 10 transcript information for ACH-000359...\n",
      "Extracting top 10 protein information for ACH-000359...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01819 (ACH-000359) ...\n",
      "Knowledge Graph Info: Found 86 disease-related proteins directly connected to BMGC_DS01819 and 18785 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 34%|███▎      | 96/286 [2:44:20<6:36:31, 125.22s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 68 unique proteins and 98 relationships between them\n",
      "Processing sample 97/286: ACH-000360 (NCI-H508)\n",
      "Sample Index: 97\n",
      "Sample Disease: cecum adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08517\n",
      "Sample ACH Name: ACH-000360\n",
      "Extracting top 10 gene information for ACH-000360...\n",
      "Extracting top 10 transcript information for ACH-000360...\n",
      "Extracting top 10 protein information for ACH-000360...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08517 (ACH-000360) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS08517 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 34%|███▍      | 97/286 [2:44:23<4:39:25, 88.71s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 90 relationships between them\n",
      "Processing sample 98/286: ACH-000367 (NCI-H226)\n",
      "Sample Index: 98\n",
      "Sample Disease: pleural mesothelioma\n",
      "Sample Disease BMGC ID: BMGC_DS09237\n",
      "Sample ACH Name: ACH-000367\n",
      "Extracting top 10 gene information for ACH-000367...\n",
      "Extracting top 10 transcript information for ACH-000367...\n",
      "Extracting top 10 protein information for ACH-000367...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09237 (ACH-000367) ...\n",
      "Knowledge Graph Info: Found 2 disease-related proteins directly connected to BMGC_DS09237 and 1692 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 34%|███▍      | 98/286 [2:44:39<3:29:29, 66.86s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 31 unique proteins and 19 relationships between them\n",
      "Processing sample 99/286: ACH-000369 (MOLM-16)\n",
      "Sample Index: 99\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000369\n",
      "Extracting top 10 gene information for ACH-000369...\n",
      "Extracting top 10 transcript information for ACH-000369...\n",
      "Extracting top 10 protein information for ACH-000369...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000369) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 35%|███▍      | 99/286 [2:48:04<5:37:35, 108.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 225 unique proteins and 573 relationships between them\n",
      "Processing sample 100/286: ACH-000372 (P12-ICHIKAWA)\n",
      "Sample Index: 100\n",
      "Sample Disease: leukemia-lymphoma, adult t-cell\n",
      "Sample Disease BMGC ID: BMGC_DS20796\n",
      "Sample ACH Name: ACH-000372\n",
      "Extracting top 10 gene information for ACH-000372...\n",
      "Extracting top 10 transcript information for ACH-000372...\n",
      "Extracting top 10 protein information for ACH-000372...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20796 (ACH-000372) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20796 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 20 unique proteins and 23 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 35%|███▍      | 100/286 [2:48:15<4:05:01, 79.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 100/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000372\n",
      "Progress: 100/286 (35.0%), Remaining: 186\n",
      "Processing sample 101/286: ACH-000373 (SKM-1)\n",
      "Sample Index: 101\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000373\n",
      "Extracting top 10 gene information for ACH-000373...\n",
      "Extracting top 10 transcript information for ACH-000373...\n",
      "Extracting top 10 protein information for ACH-000373...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000373) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 35%|███▌      | 101/286 [2:51:40<5:59:55, 116.73s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 225 unique proteins and 548 relationships between them\n",
      "Processing sample 102/286: ACH-000379 (NCI-H1781)\n",
      "Sample Index: 102\n",
      "Sample Disease: minimally invasive lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00367\n",
      "Sample ACH Name: ACH-000379\n",
      "Extracting top 10 gene information for ACH-000379...\n",
      "Extracting top 10 transcript information for ACH-000379...\n",
      "Extracting top 10 protein information for ACH-000379...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00367 (ACH-000379) ...\n",
      "Knowledge Graph Info: Found 5 disease-related proteins directly connected to BMGC_DS00367 and 8228 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 36%|███▌      | 102/286 [2:52:34<5:00:22, 97.95s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 23 unique proteins and 72 relationships between them\n",
      "Processing sample 103/286: ACH-000380 (KMS-12-BM)\n",
      "Sample Index: 103\n",
      "Sample Disease: plasma cell myeloma\n",
      "Sample Disease BMGC ID: BMGC_DS01639\n",
      "Sample ACH Name: ACH-000380\n",
      "Extracting top 10 gene information for ACH-000380...\n",
      "Extracting top 10 transcript information for ACH-000380...\n",
      "Extracting top 10 protein information for ACH-000380...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01639 (ACH-000380) ...\n",
      "Knowledge Graph Info: Found 244 disease-related proteins directly connected to BMGC_DS01639 and 24451 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 36%|███▌      | 103/286 [2:55:38<6:17:38, 123.82s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 180 unique proteins and 194 relationships between them\n",
      "Processing sample 104/286: ACH-000381 (T84)\n",
      "Sample Index: 104\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000381\n",
      "Extracting top 10 gene information for ACH-000381...\n",
      "Extracting top 10 transcript information for ACH-000381...\n",
      "Extracting top 10 protein information for ACH-000381...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000381) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 36%|███▋      | 104/286 [2:57:52<6:25:10, 126.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 123 relationships between them\n",
      "Processing sample 105/286: ACH-000384 (SW 780)\n",
      "Sample Index: 105\n",
      "Sample Disease: bladder carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07053\n",
      "Sample ACH Name: ACH-000384\n",
      "Extracting top 10 gene information for ACH-000384...\n",
      "Extracting top 10 transcript information for ACH-000384...\n",
      "Extracting top 10 protein information for ACH-000384...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07053 (ACH-000384) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS07053 and 8832 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 37%|███▋      | 105/286 [2:58:53<5:22:53, 107.03s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 91 relationships between them\n",
      "Processing sample 106/286: ACH-000386 (KG-1)\n",
      "Sample Index: 106\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000386\n",
      "Extracting top 10 gene information for ACH-000386...\n",
      "Extracting top 10 transcript information for ACH-000386...\n",
      "Extracting top 10 protein information for ACH-000386...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000386) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 37%|███▋      | 106/286 [3:02:19<6:50:45, 136.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 224 unique proteins and 533 relationships between them\n",
      "Processing sample 107/286: ACH-000389 (H4)\n",
      "Sample Index: 107\n",
      "Sample Disease: astrocytoma\n",
      "Sample Disease BMGC ID: BMGC_DS20147\n",
      "Sample ACH Name: ACH-000389\n",
      "Extracting top 10 gene information for ACH-000389...\n",
      "Extracting top 10 transcript information for ACH-000389...\n",
      "Extracting top 10 protein information for ACH-000389...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20147 (ACH-000389) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20147 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 37%|███▋      | 107/286 [3:02:22<4:47:53, 96.50s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 44 relationships between them\n",
      "Processing sample 108/286: ACH-000392 (Calu-3)\n",
      "Sample Index: 108\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000392\n",
      "Extracting top 10 gene information for ACH-000392...\n",
      "Extracting top 10 transcript information for ACH-000392...\n",
      "Extracting top 10 protein information for ACH-000392...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000392) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 38%|███▊      | 108/286 [3:06:19<6:51:28, 138.70s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 309 unique proteins and 907 relationships between them\n",
      "Processing sample 109/286: ACH-000394 (NCI-H2081)\n",
      "Sample Index: 109\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000394\n",
      "Extracting top 10 gene information for ACH-000394...\n",
      "Extracting top 10 transcript information for ACH-000394...\n",
      "Extracting top 10 protein information for ACH-000394...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000394) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 38%|███▊      | 109/286 [3:09:06<7:14:11, 147.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 178 relationships between them\n",
      "Processing sample 110/286: ACH-000396 (J82)\n",
      "Sample Index: 110\n",
      "Sample Disease: bladder carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07053\n",
      "Sample ACH Name: ACH-000396\n",
      "Extracting top 10 gene information for ACH-000396...\n",
      "Extracting top 10 transcript information for ACH-000396...\n",
      "Extracting top 10 protein information for ACH-000396...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07053 (ACH-000396) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS07053 and 8832 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 28 unique proteins and 23 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 38%|███▊      | 110/286 [3:10:14<6:02:10, 123.47s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 110/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000396\n",
      "Progress: 110/286 (38.5%), Remaining: 176\n",
      "Processing sample 111/286: ACH-000399 (NCI-H2196)\n",
      "Sample Index: 111\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000399\n",
      "Extracting top 10 gene information for ACH-000399...\n",
      "Extracting top 10 transcript information for ACH-000399...\n",
      "Extracting top 10 protein information for ACH-000399...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000399) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 39%|███▉      | 111/286 [3:13:01<6:38:23, 136.59s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 192 relationships between them\n",
      "Processing sample 112/286: ACH-000400 (SK-CO-1)\n",
      "Sample Index: 112\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000400\n",
      "Extracting top 10 gene information for ACH-000400...\n",
      "Extracting top 10 transcript information for ACH-000400...\n",
      "Extracting top 10 protein information for ACH-000400...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000400) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 39%|███▉      | 112/286 [3:15:15<6:33:31, 135.70s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 81 relationships between them\n",
      "Processing sample 113/286: ACH-000408 (TE-5)\n",
      "Sample Index: 113\n",
      "Sample Disease: esophageal squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05334\n",
      "Sample ACH Name: ACH-000408\n",
      "Extracting top 10 gene information for ACH-000408...\n",
      "Extracting top 10 transcript information for ACH-000408...\n",
      "Extracting top 10 protein information for ACH-000408...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05334 (ACH-000408) ...\n",
      "Knowledge Graph Info: Found 226 disease-related proteins directly connected to BMGC_DS05334 and 22772 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 40%|███▉      | 113/286 [3:18:07<7:03:18, 146.81s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 146 unique proteins and 238 relationships between them\n",
      "Processing sample 114/286: ACH-000422 (SNU-475)\n",
      "Sample Index: 114\n",
      "Sample Disease: adult hepatocellular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05326\n",
      "Sample ACH Name: ACH-000422\n",
      "Extracting top 10 gene information for ACH-000422...\n",
      "Extracting top 10 transcript information for ACH-000422...\n",
      "Extracting top 10 protein information for ACH-000422...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05326 (ACH-000422) ...\n",
      "Knowledge Graph Info: Found 23 disease-related proteins directly connected to BMGC_DS05326 and 7706 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 40%|███▉      | 114/286 [3:19:03<5:42:34, 119.50s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 25 unique proteins and 55 relationships between them\n",
      "Processing sample 115/286: ACH-000423 (SK-MEL-3)\n",
      "Sample Index: 115\n",
      "Sample Disease: cutaneous melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS02930\n",
      "Sample ACH Name: ACH-000423\n",
      "Extracting top 10 gene information for ACH-000423...\n",
      "Extracting top 10 transcript information for ACH-000423...\n",
      "Extracting top 10 protein information for ACH-000423...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02930 (ACH-000423) ...\n",
      "Knowledge Graph Info: Found 119 disease-related proteins directly connected to BMGC_DS02930 and 21421 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 40%|████      | 115/286 [3:21:34<6:07:41, 129.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 83 unique proteins and 195 relationships between them\n",
      "Processing sample 116/286: ACH-000428 (UO-31)\n",
      "Sample Index: 116\n",
      "Sample Disease: renal cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00375\n",
      "Sample ACH Name: ACH-000428\n",
      "Extracting top 10 gene information for ACH-000428...\n",
      "Extracting top 10 transcript information for ACH-000428...\n",
      "Extracting top 10 protein information for ACH-000428...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00375 (ACH-000428) ...\n",
      "Knowledge Graph Info: Found 341 disease-related proteins directly connected to BMGC_DS00375 and 25130 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 41%|████      | 116/286 [3:24:57<7:07:56, 151.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 199 unique proteins and 268 relationships between them\n",
      "Processing sample 117/286: ACH-000429 (A-704)\n",
      "Sample Index: 117\n",
      "Sample Disease: renal cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00375\n",
      "Sample ACH Name: ACH-000429\n",
      "Extracting top 10 gene information for ACH-000429...\n",
      "Extracting top 10 transcript information for ACH-000429...\n",
      "Extracting top 10 protein information for ACH-000429...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00375 (ACH-000429) ...\n",
      "Knowledge Graph Info: Found 341 disease-related proteins directly connected to BMGC_DS00375 and 25130 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 41%|████      | 117/286 [3:28:17<7:47:00, 165.80s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 209 unique proteins and 297 relationships between them\n",
      "Processing sample 118/286: ACH-000432 (BV-173)\n",
      "Sample Index: 118\n",
      "Sample Disease: chronic myelogenous leukemia, bcr-abl1 positive\n",
      "Sample Disease BMGC ID: BMGC_DS01409\n",
      "Sample ACH Name: ACH-000432\n",
      "Extracting top 10 gene information for ACH-000432...\n",
      "Extracting top 10 transcript information for ACH-000432...\n",
      "Extracting top 10 protein information for ACH-000432...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01409 (ACH-000432) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS01409 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 41%|████▏     | 118/286 [3:28:20<5:27:22, 116.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 36 relationships between them\n",
      "Processing sample 119/286: ACH-000439 (ME-1)\n",
      "Sample Index: 119\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-000439\n",
      "Extracting top 10 gene information for ACH-000439...\n",
      "Extracting top 10 transcript information for ACH-000439...\n",
      "Extracting top 10 protein information for ACH-000439...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-000439) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 42%|████▏     | 119/286 [3:32:12<7:01:47, 151.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 280 unique proteins and 720 relationships between them\n",
      "Processing sample 120/286: ACH-000440 (CA46)\n",
      "Sample Index: 120\n",
      "Sample Disease: burkitt lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS00331\n",
      "Sample ACH Name: ACH-000440\n",
      "Extracting top 10 gene information for ACH-000440...\n",
      "Extracting top 10 transcript information for ACH-000440...\n",
      "Extracting top 10 protein information for ACH-000440...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00331 (ACH-000440) ...\n",
      "Knowledge Graph Info: Found 67 disease-related proteins directly connected to BMGC_DS00331 and 15225 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 43 unique proteins and 35 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 42%|████▏     | 120/286 [3:34:10<6:30:53, 141.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 120/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000440\n",
      "Progress: 120/286 (42.0%), Remaining: 166\n",
      "Processing sample 121/286: ACH-000442 (RERF-LC-Sq1)\n",
      "Sample Index: 121\n",
      "Sample Disease: lung squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02872\n",
      "Sample ACH Name: ACH-000442\n",
      "Extracting top 10 gene information for ACH-000442...\n",
      "Extracting top 10 transcript information for ACH-000442...\n",
      "Extracting top 10 protein information for ACH-000442...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02872 (ACH-000442) ...\n",
      "Knowledge Graph Info: Found 142 disease-related proteins directly connected to BMGC_DS02872 and 21840 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 42%|████▏     | 121/286 [3:36:52<6:46:02, 147.65s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 115 unique proteins and 116 relationships between them\n",
      "Processing sample 122/286: ACH-000443 (OVKATE)\n",
      "Sample Index: 122\n",
      "Sample Disease: ovarian adenocarcinoma or ovary adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07934\n",
      "Sample ACH Name: ACH-000443\n",
      "Extracting top 10 gene information for ACH-000443...\n",
      "Extracting top 10 transcript information for ACH-000443...\n",
      "Extracting top 10 protein information for ACH-000443...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07934 (ACH-000443) ...\n",
      "Knowledge Graph Info: Found 5 disease-related proteins directly connected to BMGC_DS07934 and 4915 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 43%|████▎     | 122/286 [3:37:28<5:12:15, 114.24s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 22 unique proteins and 90 relationships between them\n",
      "Processing sample 123/286: ACH-000447 (NCI-H2228)\n",
      "Sample Index: 123\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000447\n",
      "Extracting top 10 gene information for ACH-000447...\n",
      "Extracting top 10 transcript information for ACH-000447...\n",
      "Extracting top 10 protein information for ACH-000447...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000447) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 43%|████▎     | 123/286 [3:41:27<6:51:30, 151.47s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 320 unique proteins and 783 relationships between them\n",
      "Processing sample 124/286: ACH-000449 (MES-SA)\n",
      "Sample Index: 124\n",
      "Sample Disease: uterine corpus sarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS05625\n",
      "Sample ACH Name: ACH-000449\n",
      "Extracting top 10 gene information for ACH-000449...\n",
      "Extracting top 10 transcript information for ACH-000449...\n",
      "Extracting top 10 protein information for ACH-000449...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05625 (ACH-000449) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS05625 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 43%|████▎     | 124/286 [3:41:30<4:48:37, 106.90s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 31 relationships between them\n",
      "Processing sample 125/286: ACH-000451 (NCI-H2085)\n",
      "Sample Index: 125\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000451\n",
      "Extracting top 10 gene information for ACH-000451...\n",
      "Extracting top 10 transcript information for ACH-000451...\n",
      "Extracting top 10 protein information for ACH-000451...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000451) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 44%|████▎     | 125/286 [3:45:28<6:32:35, 146.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 310 unique proteins and 854 relationships between them\n",
      "Processing sample 126/286: ACH-000457 (CAL-54)\n",
      "Sample Index: 126\n",
      "Sample Disease: renal cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00375\n",
      "Sample ACH Name: ACH-000457\n",
      "Extracting top 10 gene information for ACH-000457...\n",
      "Extracting top 10 transcript information for ACH-000457...\n",
      "Extracting top 10 protein information for ACH-000457...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00375 (ACH-000457) ...\n",
      "Knowledge Graph Info: Found 341 disease-related proteins directly connected to BMGC_DS00375 and 25130 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 44%|████▍     | 126/286 [3:48:51<7:15:23, 163.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 199 unique proteins and 314 relationships between them\n",
      "Processing sample 127/286: ACH-000465 (SK-MEL-1)\n",
      "Sample Index: 127\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-000465\n",
      "Extracting top 10 gene information for ACH-000465...\n",
      "Extracting top 10 transcript information for ACH-000465...\n",
      "Extracting top 10 protein information for ACH-000465...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-000465) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 44%|████▍     | 127/286 [3:52:39<8:04:36, 182.87s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 279 unique proteins and 772 relationships between them\n",
      "Processing sample 128/286: ACH-000469 (YH-13)\n",
      "Sample Index: 128\n",
      "Sample Disease: glioblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS00965\n",
      "Sample ACH Name: ACH-000469\n",
      "Extracting top 10 gene information for ACH-000469...\n",
      "Extracting top 10 transcript information for ACH-000469...\n",
      "Extracting top 10 protein information for ACH-000469...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00965 (ACH-000469) ...\n",
      "Knowledge Graph Info: Found 219 disease-related proteins directly connected to BMGC_DS00965 and 24892 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 45%|████▍     | 128/286 [3:55:46<8:04:19, 183.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 137 unique proteins and 355 relationships between them\n",
      "Processing sample 129/286: ACH-000476 (JHH-4)\n",
      "Sample Index: 129\n",
      "Sample Disease: adult hepatocellular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05326\n",
      "Sample ACH Name: ACH-000476\n",
      "Extracting top 10 gene information for ACH-000476...\n",
      "Extracting top 10 transcript information for ACH-000476...\n",
      "Extracting top 10 protein information for ACH-000476...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05326 (ACH-000476) ...\n",
      "Knowledge Graph Info: Found 23 disease-related proteins directly connected to BMGC_DS05326 and 7706 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 45%|████▌     | 129/286 [3:56:39<6:18:35, 144.68s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 35 unique proteins and 73 relationships between them\n",
      "Processing sample 130/286: ACH-000478 (SNU-387)\n",
      "Sample Index: 130\n",
      "Sample Disease: adult hepatocellular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05326\n",
      "Sample ACH Name: ACH-000478\n",
      "Extracting top 10 gene information for ACH-000478...\n",
      "Extracting top 10 transcript information for ACH-000478...\n",
      "Extracting top 10 protein information for ACH-000478...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05326 (ACH-000478) ...\n",
      "Knowledge Graph Info: Found 23 disease-related proteins directly connected to BMGC_DS05326 and 7706 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 25 unique proteins and 31 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 45%|████▌     | 130/286 [3:57:42<5:12:22, 120.14s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 130/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000478\n",
      "Progress: 130/286 (45.5%), Remaining: 156\n",
      "Processing sample 131/286: ACH-000482 (RERF-LC-KJ)\n",
      "Sample Index: 131\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000482\n",
      "Extracting top 10 gene information for ACH-000482...\n",
      "Extracting top 10 transcript information for ACH-000482...\n",
      "Extracting top 10 protein information for ACH-000482...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000482) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 46%|████▌     | 131/286 [4:01:41<6:42:36, 155.85s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 310 unique proteins and 748 relationships between them\n",
      "Processing sample 132/286: ACH-000483 (SNU-182)\n",
      "Sample Index: 132\n",
      "Sample Disease: adult hepatocellular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05326\n",
      "Sample ACH Name: ACH-000483\n",
      "Extracting top 10 gene information for ACH-000483...\n",
      "Extracting top 10 transcript information for ACH-000483...\n",
      "Extracting top 10 protein information for ACH-000483...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05326 (ACH-000483) ...\n",
      "Knowledge Graph Info: Found 23 disease-related proteins directly connected to BMGC_DS05326 and 7706 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 46%|████▌     | 132/286 [4:02:35<5:21:47, 125.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 25 unique proteins and 18 relationships between them\n",
      "Processing sample 133/286: ACH-000489 (SW1116)\n",
      "Sample Index: 133\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000489\n",
      "Extracting top 10 gene information for ACH-000489...\n",
      "Extracting top 10 transcript information for ACH-000489...\n",
      "Extracting top 10 protein information for ACH-000489...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000489) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 47%|████▋     | 133/286 [4:04:48<5:25:28, 127.64s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 135 relationships between them\n",
      "Processing sample 134/286: ACH-000501 (LS123)\n",
      "Sample Index: 134\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000501\n",
      "Extracting top 10 gene information for ACH-000501...\n",
      "Extracting top 10 transcript information for ACH-000501...\n",
      "Extracting top 10 protein information for ACH-000501...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000501) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 47%|████▋     | 134/286 [4:07:01<5:27:18, 129.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 141 relationships between them\n",
      "Processing sample 135/286: ACH-000504 (SNB75)\n",
      "Sample Index: 135\n",
      "Sample Disease: glioblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS00965\n",
      "Sample ACH Name: ACH-000504\n",
      "Extracting top 10 gene information for ACH-000504...\n",
      "Extracting top 10 transcript information for ACH-000504...\n",
      "Extracting top 10 protein information for ACH-000504...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00965 (ACH-000504) ...\n",
      "Knowledge Graph Info: Found 219 disease-related proteins directly connected to BMGC_DS00965 and 24892 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 47%|████▋     | 135/286 [4:10:05<6:06:35, 145.67s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 138 unique proteins and 343 relationships between them\n",
      "Processing sample 136/286: ACH-000506 (NCI-H146)\n",
      "Sample Index: 136\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000506\n",
      "Extracting top 10 gene information for ACH-000506...\n",
      "Extracting top 10 transcript information for ACH-000506...\n",
      "Extracting top 10 protein information for ACH-000506...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000506) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 48%|████▊     | 136/286 [4:12:49<6:18:11, 151.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 113 unique proteins and 168 relationships between them\n",
      "Processing sample 137/286: ACH-000508 (COR-L88)\n",
      "Sample Index: 137\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000508\n",
      "Extracting top 10 gene information for ACH-000508...\n",
      "Extracting top 10 transcript information for ACH-000508...\n",
      "Extracting top 10 protein information for ACH-000508...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000508) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 48%|████▊     | 137/286 [4:15:34<6:25:56, 155.41s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 112 unique proteins and 181 relationships between them\n",
      "Processing sample 138/286: ACH-000514 (NCI-H1092)\n",
      "Sample Index: 138\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000514\n",
      "Extracting top 10 gene information for ACH-000514...\n",
      "Extracting top 10 transcript information for ACH-000514...\n",
      "Extracting top 10 protein information for ACH-000514...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000514) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 48%|████▊     | 138/286 [4:18:22<6:32:11, 159.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 102 unique proteins and 169 relationships between them\n",
      "Processing sample 139/286: ACH-000516 (CAL-78)\n",
      "Sample Index: 139\n",
      "Sample Disease: chondrosarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS00471\n",
      "Sample ACH Name: ACH-000516\n",
      "Extracting top 10 gene information for ACH-000516...\n",
      "Extracting top 10 transcript information for ACH-000516...\n",
      "Extracting top 10 protein information for ACH-000516...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00471 (ACH-000516) ...\n",
      "Knowledge Graph Info: Found 3 disease-related proteins directly connected to BMGC_DS00471 and 2785 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 49%|████▊     | 139/286 [4:18:42<4:47:25, 117.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 23 unique proteins and 4 relationships between them\n",
      "Processing sample 140/286: ACH-000530 (DMS 114)\n",
      "Sample Index: 140\n",
      "Sample Disease: small cell lung carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS21085\n",
      "Sample ACH Name: ACH-000530\n",
      "Extracting top 10 gene information for ACH-000530...\n",
      "Extracting top 10 transcript information for ACH-000530...\n",
      "Extracting top 10 protein information for ACH-000530...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS21085 (ACH-000530) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS21085 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 30 unique proteins and 74 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 49%|████▉     | 140/286 [4:18:54<3:28:25, 85.66s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 140/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000530\n",
      "Progress: 140/286 (49.0%), Remaining: 146\n",
      "Processing sample 141/286: ACH-000534 (WSU-DLCL2)\n",
      "Sample Index: 141\n",
      "Sample Disease: diffuse large b-cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS02693\n",
      "Sample ACH Name: ACH-000534\n",
      "Extracting top 10 gene information for ACH-000534...\n",
      "Extracting top 10 transcript information for ACH-000534...\n",
      "Extracting top 10 protein information for ACH-000534...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02693 (ACH-000534) ...\n",
      "Knowledge Graph Info: Found 148 disease-related proteins directly connected to BMGC_DS02693 and 20906 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 49%|████▉     | 141/286 [4:21:26<4:15:07, 105.57s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 88 unique proteins and 106 relationships between them\n",
      "Processing sample 142/286: ACH-000536 (BT-20)\n",
      "Sample Index: 142\n",
      "Sample Disease: breast adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07814\n",
      "Sample ACH Name: ACH-000536\n",
      "Extracting top 10 gene information for ACH-000536...\n",
      "Extracting top 10 transcript information for ACH-000536...\n",
      "Extracting top 10 protein information for ACH-000536...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07814 (ACH-000536) ...\n",
      "Knowledge Graph Info: Found 28 disease-related proteins directly connected to BMGC_DS07814 and 13521 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 50%|████▉     | 142/286 [4:22:56<4:02:07, 100.89s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 38 unique proteins and 79 relationships between them\n",
      "Processing sample 143/286: ACH-000538 (HuTu 80)\n",
      "Sample Index: 143\n",
      "Sample Disease: duodenal adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05296\n",
      "Sample ACH Name: ACH-000538\n",
      "Extracting top 10 gene information for ACH-000538...\n",
      "Extracting top 10 transcript information for ACH-000538...\n",
      "Extracting top 10 protein information for ACH-000538...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05296 (ACH-000538) ...\n",
      "Knowledge Graph Info: Found 3 disease-related proteins directly connected to BMGC_DS05296 and 1356 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 50%|█████     | 143/286 [4:23:07<2:56:10, 73.92s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 21 unique proteins and 56 relationships between them\n",
      "Processing sample 144/286: ACH-000545 (VM-CUB1)\n",
      "Sample Index: 144\n",
      "Sample Disease: bladder carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07053\n",
      "Sample ACH Name: ACH-000545\n",
      "Extracting top 10 gene information for ACH-000545...\n",
      "Extracting top 10 transcript information for ACH-000545...\n",
      "Extracting top 10 protein information for ACH-000545...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07053 (ACH-000545) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS07053 and 8832 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 50%|█████     | 144/286 [4:24:07<2:45:25, 69.90s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 63 relationships between them\n",
      "Processing sample 145/286: ACH-000546 (HSC-4)\n",
      "Sample Index: 145\n",
      "Sample Disease: tongue squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS06143\n",
      "Sample ACH Name: ACH-000546\n",
      "Extracting top 10 gene information for ACH-000546...\n",
      "Extracting top 10 transcript information for ACH-000546...\n",
      "Extracting top 10 protein information for ACH-000546...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06143 (ACH-000546) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS06143 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 51%|█████     | 145/286 [4:24:10<1:56:55, 49.76s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 74 relationships between them\n",
      "Processing sample 146/286: ACH-000555 (A-498)\n",
      "Sample Index: 146\n",
      "Sample Disease: renal cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00375\n",
      "Sample ACH Name: ACH-000555\n",
      "Extracting top 10 gene information for ACH-000555...\n",
      "Extracting top 10 transcript information for ACH-000555...\n",
      "Extracting top 10 protein information for ACH-000555...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00375 (ACH-000555) ...\n",
      "Knowledge Graph Info: Found 341 disease-related proteins directly connected to BMGC_DS00375 and 25130 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 51%|█████     | 146/286 [4:27:33<3:43:20, 95.72s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 209 unique proteins and 371 relationships between them\n",
      "Processing sample 147/286: ACH-000560 (ECC10)\n",
      "Sample Index: 147\n",
      "Sample Disease: gastric small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08739\n",
      "Sample ACH Name: ACH-000560\n",
      "Extracting top 10 gene information for ACH-000560...\n",
      "Extracting top 10 transcript information for ACH-000560...\n",
      "Extracting top 10 protein information for ACH-000560...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08739 (ACH-000560) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS08739 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 51%|█████▏    | 147/286 [4:27:35<2:36:53, 67.72s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 72 relationships between them\n",
      "Processing sample 148/286: ACH-000562 (HCC-78)\n",
      "Sample Index: 148\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000562\n",
      "Extracting top 10 gene information for ACH-000562...\n",
      "Extracting top 10 transcript information for ACH-000562...\n",
      "Extracting top 10 protein information for ACH-000562...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000562) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 52%|█████▏    | 148/286 [4:31:45<4:41:24, 122.35s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 309 unique proteins and 801 relationships between them\n",
      "Processing sample 149/286: ACH-000565 (RCM-1)\n",
      "Sample Index: 149\n",
      "Sample Disease: colorectal adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08359\n",
      "Sample ACH Name: ACH-000565\n",
      "Extracting top 10 gene information for ACH-000565...\n",
      "Extracting top 10 transcript information for ACH-000565...\n",
      "Extracting top 10 protein information for ACH-000565...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08359 (ACH-000565) ...\n",
      "Knowledge Graph Info: Found 158 disease-related proteins directly connected to BMGC_DS08359 and 22048 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 52%|█████▏    | 149/286 [4:34:29<5:08:03, 134.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 108 unique proteins and 191 relationships between them\n",
      "Processing sample 150/286: ACH-000566 (SW-1710)\n",
      "Sample Index: 150\n",
      "Sample Disease: bladder carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07053\n",
      "Sample ACH Name: ACH-000566\n",
      "Extracting top 10 gene information for ACH-000566...\n",
      "Extracting top 10 transcript information for ACH-000566...\n",
      "Extracting top 10 protein information for ACH-000566...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07053 (ACH-000566) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS07053 and 8832 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 28 unique proteins and 43 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 52%|█████▏    | 150/286 [4:35:41<4:22:56, 116.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 150/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000566\n",
      "Progress: 150/286 (52.4%), Remaining: 136\n",
      "Processing sample 151/286: ACH-000567 (ST486)\n",
      "Sample Index: 151\n",
      "Sample Disease: burkitt lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS00331\n",
      "Sample ACH Name: ACH-000567\n",
      "Extracting top 10 gene information for ACH-000567...\n",
      "Extracting top 10 transcript information for ACH-000567...\n",
      "Extracting top 10 protein information for ACH-000567...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00331 (ACH-000567) ...\n",
      "Knowledge Graph Info: Found 67 disease-related proteins directly connected to BMGC_DS00331 and 15225 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 53%|█████▎    | 151/286 [4:37:28<4:14:44, 113.22s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 43 unique proteins and 37 relationships between them\n",
      "Processing sample 152/286: ACH-000568 (UACC-812)\n",
      "Sample Index: 152\n",
      "Sample Disease: breast adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07814\n",
      "Sample ACH Name: ACH-000568\n",
      "Extracting top 10 gene information for ACH-000568...\n",
      "Extracting top 10 transcript information for ACH-000568...\n",
      "Extracting top 10 protein information for ACH-000568...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07814 (ACH-000568) ...\n",
      "Knowledge Graph Info: Found 28 disease-related proteins directly connected to BMGC_DS07814 and 13521 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 53%|█████▎    | 152/286 [4:38:57<3:56:47, 106.03s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 94 relationships between them\n",
      "Processing sample 153/286: ACH-000569 (IST-MES1)\n",
      "Sample Index: 153\n",
      "Sample Disease: pleural mesothelioma\n",
      "Sample Disease BMGC ID: BMGC_DS09237\n",
      "Sample ACH Name: ACH-000569\n",
      "Extracting top 10 gene information for ACH-000569...\n",
      "Extracting top 10 transcript information for ACH-000569...\n",
      "Extracting top 10 protein information for ACH-000569...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09237 (ACH-000569) ...\n",
      "Knowledge Graph Info: Found 2 disease-related proteins directly connected to BMGC_DS09237 and 1692 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 53%|█████▎    | 153/286 [4:39:10<2:53:10, 78.12s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 21 unique proteins and 5 relationships between them\n",
      "Processing sample 154/286: ACH-000572 (G-361)\n",
      "Sample Index: 154\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-000572\n",
      "Extracting top 10 gene information for ACH-000572...\n",
      "Extracting top 10 transcript information for ACH-000572...\n",
      "Extracting top 10 protein information for ACH-000572...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-000572) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 54%|█████▍    | 154/286 [4:42:58<4:30:33, 122.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 280 unique proteins and 798 relationships between them\n",
      "Processing sample 155/286: ACH-000574 (FU-OV-1)\n",
      "Sample Index: 155\n",
      "Sample Disease: ovarian adenocarcinoma or ovary adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07934\n",
      "Sample ACH Name: ACH-000574\n",
      "Extracting top 10 gene information for ACH-000574...\n",
      "Extracting top 10 transcript information for ACH-000574...\n",
      "Extracting top 10 protein information for ACH-000574...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07934 (ACH-000574) ...\n",
      "Knowledge Graph Info: Found 5 disease-related proteins directly connected to BMGC_DS07934 and 4915 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 54%|█████▍    | 155/286 [4:43:31<3:29:44, 96.07s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 32 unique proteins and 63 relationships between them\n",
      "Processing sample 156/286: ACH-000577 (JHH-2)\n",
      "Sample Index: 156\n",
      "Sample Disease: adult hepatocellular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05326\n",
      "Sample ACH Name: ACH-000577\n",
      "Extracting top 10 gene information for ACH-000577...\n",
      "Extracting top 10 transcript information for ACH-000577...\n",
      "Extracting top 10 protein information for ACH-000577...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05326 (ACH-000577) ...\n",
      "Knowledge Graph Info: Found 23 disease-related proteins directly connected to BMGC_DS05326 and 7706 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 55%|█████▍    | 156/286 [4:44:25<3:00:39, 83.38s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 25 unique proteins and 34 relationships between them\n",
      "Processing sample 157/286: ACH-000581 (SNU-16)\n",
      "Sample Index: 157\n",
      "Sample Disease: gastric adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05289\n",
      "Sample ACH Name: ACH-000581\n",
      "Extracting top 10 gene information for ACH-000581...\n",
      "Extracting top 10 transcript information for ACH-000581...\n",
      "Extracting top 10 protein information for ACH-000581...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05289 (ACH-000581) ...\n",
      "Knowledge Graph Info: Found 215 disease-related proteins directly connected to BMGC_DS05289 and 23595 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 55%|█████▍    | 157/286 [4:47:23<4:00:23, 111.81s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 104 unique proteins and 185 relationships between them\n",
      "Processing sample 158/286: ACH-000583 (MC116)\n",
      "Sample Index: 158\n",
      "Sample Disease: lymphoma, large b-cell, diffuse or diffuse large b-cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS20833\n",
      "Sample ACH Name: ACH-000583\n",
      "Extracting top 10 gene information for ACH-000583...\n",
      "Extracting top 10 transcript information for ACH-000583...\n",
      "Extracting top 10 protein information for ACH-000583...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20833 (ACH-000583) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20833 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 55%|█████▌    | 158/286 [4:47:25<2:48:22, 78.93s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 24 relationships between them\n",
      "Processing sample 159/286: ACH-000584 (JHOS-4)\n",
      "Sample Index: 159\n",
      "Sample Disease: ovarian adenocarcinoma or ovary adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07934\n",
      "Sample ACH Name: ACH-000584\n",
      "Extracting top 10 gene information for ACH-000584...\n",
      "Extracting top 10 transcript information for ACH-000584...\n",
      "Extracting top 10 protein information for ACH-000584...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07934 (ACH-000584) ...\n",
      "Knowledge Graph Info: Found 5 disease-related proteins directly connected to BMGC_DS07934 and 4915 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 56%|█████▌    | 159/286 [4:47:59<2:18:23, 65.38s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 22 unique proteins and 61 relationships between them\n",
      "Processing sample 160/286: ACH-000586 (NCI-H1876)\n",
      "Sample Index: 160\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000586\n",
      "Extracting top 10 gene information for ACH-000586...\n",
      "Extracting top 10 transcript information for ACH-000586...\n",
      "Extracting top 10 protein information for ACH-000586...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000586) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 103 unique proteins and 155 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 56%|█████▌    | 160/286 [4:50:56<3:27:36, 98.86s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 160/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000586\n",
      "Progress: 160/286 (55.9%), Remaining: 126\n",
      "Processing sample 161/286: ACH-000603 (BEN)\n",
      "Sample Index: 161\n",
      "Sample Disease: lung non-small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00373\n",
      "Sample ACH Name: ACH-000603\n",
      "Extracting top 10 gene information for ACH-000603...\n",
      "Extracting top 10 transcript information for ACH-000603...\n",
      "Extracting top 10 protein information for ACH-000603...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00373 (ACH-000603) ...\n",
      "Knowledge Graph Info: Found 482 disease-related proteins directly connected to BMGC_DS00373 and 27691 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 56%|█████▋    | 161/286 [4:54:49<4:50:05, 139.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 282 unique proteins and 1066 relationships between them\n",
      "Processing sample 162/286: ACH-000610 (NCI-H2227)\n",
      "Sample Index: 162\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000610\n",
      "Extracting top 10 gene information for ACH-000610...\n",
      "Extracting top 10 transcript information for ACH-000610...\n",
      "Extracting top 10 protein information for ACH-000610...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000610) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 57%|█████▋    | 162/286 [4:57:36<5:04:57, 147.56s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 155 relationships between them\n",
      "Processing sample 163/286: ACH-000611 (SU-DHL-6)\n",
      "Sample Index: 163\n",
      "Sample Disease: diffuse large b-cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS02693\n",
      "Sample ACH Name: ACH-000611\n",
      "Extracting top 10 gene information for ACH-000611...\n",
      "Extracting top 10 transcript information for ACH-000611...\n",
      "Extracting top 10 protein information for ACH-000611...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02693 (ACH-000611) ...\n",
      "Knowledge Graph Info: Found 148 disease-related proteins directly connected to BMGC_DS02693 and 20906 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 57%|█████▋    | 163/286 [5:00:08<5:05:02, 148.80s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 88 unique proteins and 96 relationships between them\n",
      "Processing sample 164/286: ACH-000613 (HOS)\n",
      "Sample Index: 164\n",
      "Sample Disease: osteosarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS01819\n",
      "Sample ACH Name: ACH-000613\n",
      "Extracting top 10 gene information for ACH-000613...\n",
      "Extracting top 10 transcript information for ACH-000613...\n",
      "Extracting top 10 protein information for ACH-000613...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01819 (ACH-000613) ...\n",
      "Knowledge Graph Info: Found 86 disease-related proteins directly connected to BMGC_DS01819 and 18785 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 57%|█████▋    | 164/286 [5:02:18<4:50:47, 143.01s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 68 unique proteins and 73 relationships between them\n",
      "Processing sample 165/286: ACH-000615 (SK-MEL-28)\n",
      "Sample Index: 165\n",
      "Sample Disease: cutaneous melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS02930\n",
      "Sample ACH Name: ACH-000615\n",
      "Extracting top 10 gene information for ACH-000615...\n",
      "Extracting top 10 transcript information for ACH-000615...\n",
      "Extracting top 10 protein information for ACH-000615...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02930 (ACH-000615) ...\n",
      "Knowledge Graph Info: Found 119 disease-related proteins directly connected to BMGC_DS02930 and 21421 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 58%|█████▊    | 165/286 [5:04:49<4:53:13, 145.40s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 83 unique proteins and 187 relationships between them\n",
      "Processing sample 166/286: ACH-000616 (Hs 746T)\n",
      "Sample Index: 166\n",
      "Sample Disease: gastric adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05289\n",
      "Sample ACH Name: ACH-000616\n",
      "Extracting top 10 gene information for ACH-000616...\n",
      "Extracting top 10 transcript information for ACH-000616...\n",
      "Extracting top 10 protein information for ACH-000616...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05289 (ACH-000616) ...\n",
      "Knowledge Graph Info: Found 215 disease-related proteins directly connected to BMGC_DS05289 and 23595 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 58%|█████▊    | 166/286 [5:07:43<5:08:29, 154.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 105 unique proteins and 172 relationships between them\n",
      "Processing sample 167/286: ACH-000617 (OVCAR-4)\n",
      "Sample Index: 167\n",
      "Sample Disease: ovarian adenocarcinoma or ovary adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07934\n",
      "Sample ACH Name: ACH-000617\n",
      "Extracting top 10 gene information for ACH-000617...\n",
      "Extracting top 10 transcript information for ACH-000617...\n",
      "Extracting top 10 protein information for ACH-000617...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07934 (ACH-000617) ...\n",
      "Knowledge Graph Info: Found 5 disease-related proteins directly connected to BMGC_DS07934 and 4915 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 58%|█████▊    | 167/286 [5:08:17<3:54:00, 117.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 32 unique proteins and 98 relationships between them\n",
      "Processing sample 168/286: ACH-000619 (PE/CA-PJ15)\n",
      "Sample Index: 168\n",
      "Sample Disease: tongue squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS06143\n",
      "Sample ACH Name: ACH-000619\n",
      "Extracting top 10 gene information for ACH-000619...\n",
      "Extracting top 10 transcript information for ACH-000619...\n",
      "Extracting top 10 protein information for ACH-000619...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06143 (ACH-000619) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS06143 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 59%|█████▊    | 168/286 [5:08:19<2:43:59, 83.38s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 104 relationships between them\n",
      "Processing sample 169/286: ACH-000625 (Hep 3B2.1-7)\n",
      "Sample Index: 169\n",
      "Sample Disease: pediatric hepatocellular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05325\n",
      "Sample ACH Name: ACH-000625\n",
      "Extracting top 10 gene information for ACH-000625...\n",
      "Extracting top 10 transcript information for ACH-000625...\n",
      "Extracting top 10 protein information for ACH-000625...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05325 (ACH-000625) ...\n",
      "Knowledge Graph Info: Found 2 disease-related proteins directly connected to BMGC_DS05325 and 3022 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 59%|█████▉    | 169/286 [5:08:41<2:06:08, 64.68s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 31 unique proteins and 108 relationships between them\n",
      "Processing sample 170/286: ACH-000626 (U266B1)\n",
      "Sample Index: 170\n",
      "Sample Disease: plasma cell myeloma\n",
      "Sample Disease BMGC ID: BMGC_DS01639\n",
      "Sample ACH Name: ACH-000626\n",
      "Extracting top 10 gene information for ACH-000626...\n",
      "Extracting top 10 transcript information for ACH-000626...\n",
      "Extracting top 10 protein information for ACH-000626...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01639 (ACH-000626) ...\n",
      "Knowledge Graph Info: Found 244 disease-related proteins directly connected to BMGC_DS01639 and 24451 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 179 unique proteins and 160 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 59%|█████▉    | 170/286 [5:11:58<3:22:03, 104.51s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 170/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000626\n",
      "Progress: 170/286 (59.4%), Remaining: 116\n",
      "Processing sample 171/286: ACH-000628 (NCI-H596)\n",
      "Sample Index: 171\n",
      "Sample Disease: lung non-small cell carcinoma or non-small cell lung carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00373\n",
      "Sample ACH Name: ACH-000628\n",
      "Extracting top 10 gene information for ACH-000628...\n",
      "Extracting top 10 transcript information for ACH-000628...\n",
      "Extracting top 10 protein information for ACH-000628...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00373 (ACH-000628) ...\n",
      "Knowledge Graph Info: Found 482 disease-related proteins directly connected to BMGC_DS00373 and 27691 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 60%|█████▉    | 171/286 [5:15:52<4:35:02, 143.50s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 282 unique proteins and 982 relationships between them\n",
      "Processing sample 172/286: ACH-000636 (RPMI-8402)\n",
      "Sample Index: 172\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-000636\n",
      "Extracting top 10 gene information for ACH-000636...\n",
      "Extracting top 10 transcript information for ACH-000636...\n",
      "Extracting top 10 protein information for ACH-000636...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-000636) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 60%|██████    | 172/286 [5:18:21<4:35:31, 145.01s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 106 unique proteins and 127 relationships between them\n",
      "Processing sample 173/286: ACH-000640 (SK-MEL-31)\n",
      "Sample Index: 173\n",
      "Sample Disease: cutaneous melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS02930\n",
      "Sample ACH Name: ACH-000640\n",
      "Extracting top 10 gene information for ACH-000640...\n",
      "Extracting top 10 transcript information for ACH-000640...\n",
      "Extracting top 10 protein information for ACH-000640...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02930 (ACH-000640) ...\n",
      "Knowledge Graph Info: Found 119 disease-related proteins directly connected to BMGC_DS02930 and 21421 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 60%|██████    | 173/286 [5:20:51<4:35:48, 146.45s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 73 unique proteins and 128 relationships between them\n",
      "Processing sample 174/286: ACH-000641 (CMK)\n",
      "Sample Index: 174\n",
      "Sample Disease: down syndrome\n",
      "Sample Disease BMGC ID: BMGC_DS00684\n",
      "Sample ACH Name: ACH-000641\n",
      "Extracting top 10 gene information for ACH-000641...\n",
      "Extracting top 10 transcript information for ACH-000641...\n",
      "Extracting top 10 protein information for ACH-000641...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00684 (ACH-000641) ...\n",
      "Knowledge Graph Info: Found 30 disease-related proteins directly connected to BMGC_DS00684 and 12041 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 61%|██████    | 174/286 [5:22:12<3:56:52, 126.90s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 39 unique proteins and 33 relationships between them\n",
      "Processing sample 175/286: ACH-000643 (HDQ-P1)\n",
      "Sample Index: 175\n",
      "Sample Disease: breast ductal carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08035\n",
      "Sample ACH Name: ACH-000643\n",
      "Extracting top 10 gene information for ACH-000643...\n",
      "Extracting top 10 transcript information for ACH-000643...\n",
      "Extracting top 10 protein information for ACH-000643...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08035 (ACH-000643) ...\n",
      "Knowledge Graph Info: Found 17 disease-related proteins directly connected to BMGC_DS08035 and 13835 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 61%|██████    | 175/286 [5:23:42<3:34:22, 115.88s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 42 unique proteins and 95 relationships between them\n",
      "Processing sample 176/286: ACH-000644 (COLO 829)\n",
      "Sample Index: 176\n",
      "Sample Disease: cutaneous melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS02930\n",
      "Sample ACH Name: ACH-000644\n",
      "Extracting top 10 gene information for ACH-000644...\n",
      "Extracting top 10 transcript information for ACH-000644...\n",
      "Extracting top 10 protein information for ACH-000644...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02930 (ACH-000644) ...\n",
      "Knowledge Graph Info: Found 119 disease-related proteins directly connected to BMGC_DS02930 and 21421 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 62%|██████▏   | 176/286 [5:26:13<3:51:40, 126.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 82 unique proteins and 182 relationships between them\n",
      "Processing sample 177/286: ACH-000650 (IGR-37)\n",
      "Sample Index: 177\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-000650\n",
      "Extracting top 10 gene information for ACH-000650...\n",
      "Extracting top 10 transcript information for ACH-000650...\n",
      "Extracting top 10 protein information for ACH-000650...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-000650) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 62%|██████▏   | 177/286 [5:30:02<4:45:25, 157.11s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 285 unique proteins and 802 relationships between them\n",
      "Processing sample 178/286: ACH-000654 (Raji)\n",
      "Sample Index: 178\n",
      "Sample Disease: burkitt lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS00331\n",
      "Sample ACH Name: ACH-000654\n",
      "Extracting top 10 gene information for ACH-000654...\n",
      "Extracting top 10 transcript information for ACH-000654...\n",
      "Extracting top 10 protein information for ACH-000654...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00331 (ACH-000654) ...\n",
      "Knowledge Graph Info: Found 67 disease-related proteins directly connected to BMGC_DS00331 and 15225 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 62%|██████▏   | 178/286 [5:31:48<4:15:18, 141.84s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 42 unique proteins and 38 relationships between them\n",
      "Processing sample 179/286: ACH-000656 (SU-DHL-8)\n",
      "Sample Index: 179\n",
      "Sample Disease: diffuse large b-cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS02693\n",
      "Sample ACH Name: ACH-000656\n",
      "Extracting top 10 gene information for ACH-000656...\n",
      "Extracting top 10 transcript information for ACH-000656...\n",
      "Extracting top 10 protein information for ACH-000656...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02693 (ACH-000656) ...\n",
      "Knowledge Graph Info: Found 148 disease-related proteins directly connected to BMGC_DS02693 and 20906 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 63%|██████▎   | 179/286 [5:34:21<4:18:37, 145.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 88 unique proteins and 98 relationships between them\n",
      "Processing sample 180/286: ACH-000660 (SU-DHL-5)\n",
      "Sample Index: 180\n",
      "Sample Disease: diffuse large b-cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS02693\n",
      "Sample ACH Name: ACH-000660\n",
      "Extracting top 10 gene information for ACH-000660...\n",
      "Extracting top 10 transcript information for ACH-000660...\n",
      "Extracting top 10 protein information for ACH-000660...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02693 (ACH-000660) ...\n",
      "Knowledge Graph Info: Found 148 disease-related proteins directly connected to BMGC_DS02693 and 20906 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 88 unique proteins and 110 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 63%|██████▎   | 180/286 [5:37:05<4:26:22, 150.78s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 180/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000660\n",
      "Progress: 180/286 (62.9%), Remaining: 106\n",
      "Processing sample 181/286: ACH-000664 (SU-DHL-1)\n",
      "Sample Index: 181\n",
      "Sample Disease: alk-positive anaplastic large cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS08403\n",
      "Sample ACH Name: ACH-000664\n",
      "Extracting top 10 gene information for ACH-000664...\n",
      "Extracting top 10 transcript information for ACH-000664...\n",
      "Extracting top 10 protein information for ACH-000664...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08403 (ACH-000664) ...\n",
      "Knowledge Graph Info: Found 2 disease-related proteins directly connected to BMGC_DS08403 and 2478 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 63%|██████▎   | 181/286 [5:37:24<3:14:39, 111.24s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 21 unique proteins and 30 relationships between them\n",
      "Processing sample 182/286: ACH-000669 (SW 900)\n",
      "Sample Index: 182\n",
      "Sample Disease: lung squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02872\n",
      "Sample ACH Name: ACH-000669\n",
      "Extracting top 10 gene information for ACH-000669...\n",
      "Extracting top 10 transcript information for ACH-000669...\n",
      "Extracting top 10 protein information for ACH-000669...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02872 (ACH-000669) ...\n",
      "Knowledge Graph Info: Found 142 disease-related proteins directly connected to BMGC_DS02872 and 21840 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 64%|██████▎   | 182/286 [5:40:01<3:36:41, 125.01s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 105 unique proteins and 176 relationships between them\n",
      "Processing sample 183/286: ACH-000670 (SBC-5)\n",
      "Sample Index: 183\n",
      "Sample Disease: small cell lung carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS21085\n",
      "Sample ACH Name: ACH-000670\n",
      "Extracting top 10 gene information for ACH-000670...\n",
      "Extracting top 10 transcript information for ACH-000670...\n",
      "Extracting top 10 protein information for ACH-000670...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS21085 (ACH-000670) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS21085 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 64%|██████▍   | 183/286 [5:40:03<2:31:19, 88.15s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 30 unique proteins and 61 relationships between them\n",
      "Processing sample 184/286: ACH-000674 (NUGC-4)\n",
      "Sample Index: 184\n",
      "Sample Disease: gastric signet ring cell adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS09085\n",
      "Sample ACH Name: ACH-000674\n",
      "Extracting top 10 gene information for ACH-000674...\n",
      "Extracting top 10 transcript information for ACH-000674...\n",
      "Extracting top 10 protein information for ACH-000674...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09085 (ACH-000674) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS09085 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 64%|██████▍   | 184/286 [5:40:05<1:46:00, 62.35s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 54 relationships between them\n",
      "Processing sample 185/286: ACH-000675 (NCI-H1734)\n",
      "Sample Index: 185\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000675\n",
      "Extracting top 10 gene information for ACH-000675...\n",
      "Extracting top 10 transcript information for ACH-000675...\n",
      "Extracting top 10 protein information for ACH-000675...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000675) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 65%|██████▍   | 185/286 [5:44:03<3:13:33, 114.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 310 unique proteins and 819 relationships between them\n",
      "Processing sample 186/286: ACH-000678 (MKN7)\n",
      "Sample Index: 186\n",
      "Sample Disease: gastric tubular adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08742\n",
      "Sample ACH Name: ACH-000678\n",
      "Extracting top 10 gene information for ACH-000678...\n",
      "Extracting top 10 transcript information for ACH-000678...\n",
      "Extracting top 10 protein information for ACH-000678...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08742 (ACH-000678) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS08742 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 65%|██████▌   | 186/286 [5:44:05<2:15:10, 81.11s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 30 unique proteins and 78 relationships between them\n",
      "Processing sample 187/286: ACH-000679 (OE19)\n",
      "Sample Index: 187\n",
      "Sample Disease: esophageal adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05335\n",
      "Sample ACH Name: ACH-000679\n",
      "Extracting top 10 gene information for ACH-000679...\n",
      "Extracting top 10 transcript information for ACH-000679...\n",
      "Extracting top 10 protein information for ACH-000679...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05335 (ACH-000679) ...\n",
      "Knowledge Graph Info: Found 54 disease-related proteins directly connected to BMGC_DS05335 and 15616 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 65%|██████▌   | 187/286 [5:45:51<2:25:53, 88.41s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 52 unique proteins and 87 relationships between them\n",
      "Processing sample 188/286: ACH-000680 (SW948)\n",
      "Sample Index: 188\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000680\n",
      "Extracting top 10 gene information for ACH-000680...\n",
      "Extracting top 10 transcript information for ACH-000680...\n",
      "Extracting top 10 protein information for ACH-000680...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000680) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 66%|██████▌   | 188/286 [5:48:04<2:46:23, 101.87s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 84 relationships between them\n",
      "Processing sample 189/286: ACH-000689 (RH-18)\n",
      "Sample Index: 189\n",
      "Sample Disease: alveolar rhabdomyosarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS04065\n",
      "Sample ACH Name: ACH-000689\n",
      "Extracting top 10 gene information for ACH-000689...\n",
      "Extracting top 10 transcript information for ACH-000689...\n",
      "Extracting top 10 protein information for ACH-000689...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS04065 (ACH-000689) ...\n",
      "Knowledge Graph Info: Found 11 disease-related proteins directly connected to BMGC_DS04065 and 10259 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 66%|██████▌   | 189/286 [5:49:12<2:28:28, 91.84s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 26 unique proteins and 65 relationships between them\n",
      "Processing sample 190/286: ACH-000691 (HCC2157)\n",
      "Sample Index: 190\n",
      "Sample Disease: breast ductal carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08035\n",
      "Sample ACH Name: ACH-000691\n",
      "Extracting top 10 gene information for ACH-000691...\n",
      "Extracting top 10 transcript information for ACH-000691...\n",
      "Extracting top 10 protein information for ACH-000691...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08035 (ACH-000691) ...\n",
      "Knowledge Graph Info: Found 17 disease-related proteins directly connected to BMGC_DS08035 and 13835 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 32 unique proteins and 40 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 66%|██████▋   | 190/286 [5:50:56<2:32:26, 95.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 190/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000691\n",
      "Progress: 190/286 (66.4%), Remaining: 96\n",
      "Processing sample 191/286: ACH-000702 (L-1236)\n",
      "Sample Index: 191\n",
      "Sample Disease: hodgkin disease or hodgkins lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS20405\n",
      "Sample ACH Name: ACH-000702\n",
      "Extracting top 10 gene information for ACH-000702...\n",
      "Extracting top 10 transcript information for ACH-000702...\n",
      "Extracting top 10 protein information for ACH-000702...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20405 (ACH-000702) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20405 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 67%|██████▋   | 191/286 [5:50:58<1:46:51, 67.49s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 44 relationships between them\n",
      "Processing sample 192/286: ACH-000703 (DMS 79)\n",
      "Sample Index: 192\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000703\n",
      "Extracting top 10 gene information for ACH-000703...\n",
      "Extracting top 10 transcript information for ACH-000703...\n",
      "Extracting top 10 protein information for ACH-000703...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000703) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 67%|██████▋   | 192/286 [5:53:45<2:32:25, 97.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 143 relationships between them\n",
      "Processing sample 193/286: ACH-000714 (KMS-11)\n",
      "Sample Index: 193\n",
      "Sample Disease: plasma cell myeloma\n",
      "Sample Disease BMGC ID: BMGC_DS01639\n",
      "Sample ACH Name: ACH-000714\n",
      "Extracting top 10 gene information for ACH-000714...\n",
      "Extracting top 10 transcript information for ACH-000714...\n",
      "Extracting top 10 protein information for ACH-000714...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01639 (ACH-000714) ...\n",
      "Knowledge Graph Info: Found 244 disease-related proteins directly connected to BMGC_DS01639 and 24451 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 67%|██████▋   | 193/286 [5:56:50<3:11:21, 123.46s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 180 unique proteins and 233 relationships between them\n",
      "Processing sample 194/286: ACH-000716 (TT2609-C02)\n",
      "Sample Index: 194\n",
      "Sample Disease: thyroid gland follicular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS04082\n",
      "Sample ACH Name: ACH-000716\n",
      "Extracting top 10 gene information for ACH-000716...\n",
      "Extracting top 10 transcript information for ACH-000716...\n",
      "Extracting top 10 protein information for ACH-000716...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS04082 (ACH-000716) ...\n",
      "Knowledge Graph Info: Found 17 disease-related proteins directly connected to BMGC_DS04082 and 11025 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 68%|██████▊   | 194/286 [5:58:04<2:46:35, 108.65s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 29 unique proteins and 43 relationships between them\n",
      "Processing sample 195/286: ACH-000727 (NCI-H2066)\n",
      "Sample Index: 195\n",
      "Sample Disease: lung squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02872\n",
      "Sample ACH Name: ACH-000727\n",
      "Extracting top 10 gene information for ACH-000727...\n",
      "Extracting top 10 transcript information for ACH-000727...\n",
      "Extracting top 10 protein information for ACH-000727...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02872 (ACH-000727) ...\n",
      "Knowledge Graph Info: Found 142 disease-related proteins directly connected to BMGC_DS02872 and 21840 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 68%|██████▊   | 195/286 [6:00:40<3:06:41, 123.10s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 105 unique proteins and 72 relationships between them\n",
      "Processing sample 196/286: ACH-000729 (NCI-H1963)\n",
      "Sample Index: 196\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000729\n",
      "Extracting top 10 gene information for ACH-000729...\n",
      "Extracting top 10 transcript information for ACH-000729...\n",
      "Extracting top 10 protein information for ACH-000729...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000729) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 69%|██████▊   | 196/286 [6:03:28<3:24:39, 136.44s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 102 unique proteins and 156 relationships between them\n",
      "Processing sample 197/286: ACH-000730 (SK-MEL-5)\n",
      "Sample Index: 197\n",
      "Sample Disease: cutaneous melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS02930\n",
      "Sample ACH Name: ACH-000730\n",
      "Extracting top 10 gene information for ACH-000730...\n",
      "Extracting top 10 transcript information for ACH-000730...\n",
      "Extracting top 10 protein information for ACH-000730...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02930 (ACH-000730) ...\n",
      "Knowledge Graph Info: Found 119 disease-related proteins directly connected to BMGC_DS02930 and 21421 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 69%|██████▉   | 197/286 [6:05:58<3:28:32, 140.59s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 83 unique proteins and 198 relationships between them\n",
      "Processing sample 198/286: ACH-000733 (NCI-H1838)\n",
      "Sample Index: 198\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000733\n",
      "Extracting top 10 gene information for ACH-000733...\n",
      "Extracting top 10 transcript information for ACH-000733...\n",
      "Extracting top 10 protein information for ACH-000733...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000733) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 69%|██████▉   | 198/286 [6:09:59<4:10:06, 170.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 309 unique proteins and 889 relationships between them\n",
      "Processing sample 199/286: ACH-000743 (COR-L95)\n",
      "Sample Index: 199\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000743\n",
      "Extracting top 10 gene information for ACH-000743...\n",
      "Extracting top 10 transcript information for ACH-000743...\n",
      "Extracting top 10 protein information for ACH-000743...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000743) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 70%|██████▉   | 199/286 [6:12:47<4:06:06, 169.73s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 187 relationships between them\n",
      "Processing sample 200/286: ACH-000744 (NCI-H1623)\n",
      "Sample Index: 200\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000744\n",
      "Extracting top 10 gene information for ACH-000744...\n",
      "Extracting top 10 transcript information for ACH-000744...\n",
      "Extracting top 10 protein information for ACH-000744...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000744) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 310 unique proteins and 897 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 70%|██████▉   | 200/286 [6:16:59<4:38:50, 194.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 200/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000744\n",
      "Progress: 200/286 (69.9%), Remaining: 86\n",
      "Processing sample 201/286: ACH-000745 (MOLP-8)\n",
      "Sample Index: 201\n",
      "Sample Disease: plasma cell myeloma\n",
      "Sample Disease BMGC ID: BMGC_DS01639\n",
      "Sample ACH Name: ACH-000745\n",
      "Extracting top 10 gene information for ACH-000745...\n",
      "Extracting top 10 transcript information for ACH-000745...\n",
      "Extracting top 10 protein information for ACH-000745...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01639 (ACH-000745) ...\n",
      "Knowledge Graph Info: Found 244 disease-related proteins directly connected to BMGC_DS01639 and 24451 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 70%|███████   | 201/286 [6:20:05<4:31:55, 191.95s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 180 unique proteins and 173 relationships between them\n",
      "Processing sample 202/286: ACH-000750 (LOX IMVI)\n",
      "Sample Index: 202\n",
      "Sample Disease: amelanotic melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS04117\n",
      "Sample ACH Name: ACH-000750\n",
      "Extracting top 10 gene information for ACH-000750...\n",
      "Extracting top 10 transcript information for ACH-000750...\n",
      "Extracting top 10 protein information for ACH-000750...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS04117 (ACH-000750) ...\n",
      "Knowledge Graph Info: Found 1 disease-related proteins directly connected to BMGC_DS04117 and 2065 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 71%|███████   | 202/286 [6:20:23<3:15:36, 139.72s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 31 unique proteins and 29 relationships between them\n",
      "Processing sample 203/286: ACH-000751 (OCI-M1)\n",
      "Sample Index: 203\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000751\n",
      "Extracting top 10 gene information for ACH-000751...\n",
      "Extracting top 10 transcript information for ACH-000751...\n",
      "Extracting top 10 protein information for ACH-000751...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000751) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 71%|███████   | 203/286 [6:23:49<3:40:52, 159.67s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 225 unique proteins and 548 relationships between them\n",
      "Processing sample 204/286: ACH-000752 (NCI-H196)\n",
      "Sample Index: 204\n",
      "Sample Disease: lung sarcomatoid carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS09749\n",
      "Sample ACH Name: ACH-000752\n",
      "Extracting top 10 gene information for ACH-000752...\n",
      "Extracting top 10 transcript information for ACH-000752...\n",
      "Extracting top 10 protein information for ACH-000752...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09749 (ACH-000752) ...\n",
      "Knowledge Graph Info: Found 22 disease-related proteins directly connected to BMGC_DS09749 and 10886 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 71%|███████▏  | 204/286 [6:25:02<3:02:37, 133.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 37 unique proteins and 24 relationships between them\n",
      "Processing sample 205/286: ACH-000754 (L-428)\n",
      "Sample Index: 205\n",
      "Sample Disease: hodgkin disease or hodgkins lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS20405\n",
      "Sample ACH Name: ACH-000754\n",
      "Extracting top 10 gene information for ACH-000754...\n",
      "Extracting top 10 transcript information for ACH-000754...\n",
      "Extracting top 10 protein information for ACH-000754...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20405 (ACH-000754) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20405 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 72%|███████▏  | 205/286 [6:25:05<2:07:46, 94.65s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 56 relationships between them\n",
      "Processing sample 206/286: ACH-000755 (HCC2218)\n",
      "Sample Index: 206\n",
      "Sample Disease: breast ductal carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08035\n",
      "Sample ACH Name: ACH-000755\n",
      "Extracting top 10 gene information for ACH-000755...\n",
      "Extracting top 10 transcript information for ACH-000755...\n",
      "Extracting top 10 protein information for ACH-000755...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08035 (ACH-000755) ...\n",
      "Knowledge Graph Info: Found 17 disease-related proteins directly connected to BMGC_DS08035 and 13835 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 72%|███████▏  | 206/286 [6:26:37<2:04:49, 93.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 42 unique proteins and 42 relationships between them\n",
      "Processing sample 207/286: ACH-000759 (MDA-MB-175-VII)\n",
      "Sample Index: 207\n",
      "Sample Disease: breast adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07814\n",
      "Sample ACH Name: ACH-000759\n",
      "Extracting top 10 gene information for ACH-000759...\n",
      "Extracting top 10 transcript information for ACH-000759...\n",
      "Extracting top 10 protein information for ACH-000759...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07814 (ACH-000759) ...\n",
      "Knowledge Graph Info: Found 28 disease-related proteins directly connected to BMGC_DS07814 and 13521 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 72%|███████▏  | 207/286 [6:28:08<2:02:18, 92.89s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 59 relationships between them\n",
      "Processing sample 208/286: ACH-000770 (P31/FUJ)\n",
      "Sample Index: 208\n",
      "Sample Disease: acute myeloid leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS01407\n",
      "Sample ACH Name: ACH-000770\n",
      "Extracting top 10 gene information for ACH-000770...\n",
      "Extracting top 10 transcript information for ACH-000770...\n",
      "Extracting top 10 protein information for ACH-000770...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01407 (ACH-000770) ...\n",
      "Knowledge Graph Info: Found 377 disease-related proteins directly connected to BMGC_DS01407 and 25307 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 73%|███████▎  | 208/286 [6:31:35<2:45:28, 127.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 225 unique proteins and 514 relationships between them\n",
      "Processing sample 209/286: ACH-000772 (TE 441.T)\n",
      "Sample Index: 209\n",
      "Sample Disease: rhabdomyosarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS02196\n",
      "Sample ACH Name: ACH-000772\n",
      "Extracting top 10 gene information for ACH-000772...\n",
      "Extracting top 10 transcript information for ACH-000772...\n",
      "Extracting top 10 protein information for ACH-000772...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02196 (ACH-000772) ...\n",
      "Knowledge Graph Info: Found 84 disease-related proteins directly connected to BMGC_DS02196 and 15743 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 73%|███████▎  | 209/286 [6:33:28<2:37:45, 122.93s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 43 unique proteins and 50 relationships between them\n",
      "Processing sample 210/286: ACH-000776 (ONS-76)\n",
      "Sample Index: 210\n",
      "Sample Disease: medulloblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS01553\n",
      "Sample ACH Name: ACH-000776\n",
      "Extracting top 10 gene information for ACH-000776...\n",
      "Extracting top 10 transcript information for ACH-000776...\n",
      "Extracting top 10 protein information for ACH-000776...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01553 (ACH-000776) ...\n",
      "Knowledge Graph Info: Found 148 disease-related proteins directly connected to BMGC_DS01553 and 20882 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 86 unique proteins and 150 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 73%|███████▎  | 210/286 [6:36:14<2:51:59, 135.78s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 210/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000776\n",
      "Progress: 210/286 (73.4%), Remaining: 76\n",
      "Processing sample 211/286: ACH-000780 (NCI-H1105)\n",
      "Sample Index: 211\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000780\n",
      "Extracting top 10 gene information for ACH-000780...\n",
      "Extracting top 10 transcript information for ACH-000780...\n",
      "Extracting top 10 protein information for ACH-000780...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000780) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 74%|███████▍  | 211/286 [6:39:02<3:01:38, 145.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 154 relationships between them\n",
      "Processing sample 212/286: ACH-000786 (Daudi)\n",
      "Sample Index: 212\n",
      "Sample Disease: burkitt lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS00331\n",
      "Sample ACH Name: ACH-000786\n",
      "Extracting top 10 gene information for ACH-000786...\n",
      "Extracting top 10 transcript information for ACH-000786...\n",
      "Extracting top 10 protein information for ACH-000786...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00331 (ACH-000786) ...\n",
      "Knowledge Graph Info: Found 67 disease-related proteins directly connected to BMGC_DS00331 and 15225 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 74%|███████▍  | 212/286 [6:40:48<2:44:41, 133.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 43 unique proteins and 35 relationships between them\n",
      "Processing sample 213/286: ACH-000789 (NCI-H810)\n",
      "Sample Index: 213\n",
      "Sample Disease: lung large cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05995\n",
      "Sample ACH Name: ACH-000789\n",
      "Extracting top 10 gene information for ACH-000789...\n",
      "Extracting top 10 transcript information for ACH-000789...\n",
      "Extracting top 10 protein information for ACH-000789...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05995 (ACH-000789) ...\n",
      "Knowledge Graph Info: Found 60 disease-related proteins directly connected to BMGC_DS05995 and 13831 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 74%|███████▍  | 213/286 [6:42:25<2:29:24, 122.80s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 51 unique proteins and 93 relationships between them\n",
      "Processing sample 214/286: ACH-000795 (MOLT-13)\n",
      "Sample Index: 214\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-000795\n",
      "Extracting top 10 gene information for ACH-000795...\n",
      "Extracting top 10 transcript information for ACH-000795...\n",
      "Extracting top 10 protein information for ACH-000795...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-000795) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 75%|███████▍  | 214/286 [6:44:55<2:37:00, 130.84s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 96 unique proteins and 91 relationships between them\n",
      "Processing sample 215/286: ACH-000798 (CL-40)\n",
      "Sample Index: 215\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000798\n",
      "Extracting top 10 gene information for ACH-000798...\n",
      "Extracting top 10 transcript information for ACH-000798...\n",
      "Extracting top 10 protein information for ACH-000798...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000798) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 75%|███████▌  | 215/286 [6:47:09<2:35:48, 131.67s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 117 relationships between them\n",
      "Processing sample 216/286: ACH-000802 (BFTC-905)\n",
      "Sample Index: 216\n",
      "Sample Disease: bladder carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07053\n",
      "Sample ACH Name: ACH-000802\n",
      "Extracting top 10 gene information for ACH-000802...\n",
      "Extracting top 10 transcript information for ACH-000802...\n",
      "Extracting top 10 protein information for ACH-000802...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07053 (ACH-000802) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS07053 and 8832 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 76%|███████▌  | 216/286 [6:48:09<2:08:39, 110.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 44 relationships between them\n",
      "Processing sample 217/286: ACH-000803 (COLO 668)\n",
      "Sample Index: 217\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000803\n",
      "Extracting top 10 gene information for ACH-000803...\n",
      "Extracting top 10 transcript information for ACH-000803...\n",
      "Extracting top 10 protein information for ACH-000803...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000803) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 76%|███████▌  | 217/286 [6:50:56<2:26:27, 127.35s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 102 unique proteins and 183 relationships between them\n",
      "Processing sample 218/286: ACH-000806 (L-540)\n",
      "Sample Index: 218\n",
      "Sample Disease: hodgkin disease or hodgkins lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS20405\n",
      "Sample ACH Name: ACH-000806\n",
      "Extracting top 10 gene information for ACH-000806...\n",
      "Extracting top 10 transcript information for ACH-000806...\n",
      "Extracting top 10 protein information for ACH-000806...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20405 (ACH-000806) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20405 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 76%|███████▌  | 218/286 [6:51:00<1:42:28, 90.42s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 42 relationships between them\n",
      "Processing sample 219/286: ACH-000812 (COLO-783)\n",
      "Sample Index: 219\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-000812\n",
      "Extracting top 10 gene information for ACH-000812...\n",
      "Extracting top 10 transcript information for ACH-000812...\n",
      "Extracting top 10 protein information for ACH-000812...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-000812) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 77%|███████▋  | 219/286 [6:54:48<2:27:05, 131.73s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 281 unique proteins and 812 relationships between them\n",
      "Processing sample 220/286: ACH-000814 (Hs 939.T)\n",
      "Sample Index: 220\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-000814\n",
      "Extracting top 10 gene information for ACH-000814...\n",
      "Extracting top 10 transcript information for ACH-000814...\n",
      "Extracting top 10 protein information for ACH-000814...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-000814) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 281 unique proteins and 714 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 77%|███████▋  | 220/286 [6:58:53<3:02:05, 165.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 220/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000814\n",
      "Progress: 220/286 (76.9%), Remaining: 66\n",
      "Processing sample 221/286: ACH-000815 (KM-H2)\n",
      "Sample Index: 221\n",
      "Sample Disease: hodgkin disease or hodgkins lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS20405\n",
      "Sample ACH Name: ACH-000815\n",
      "Extracting top 10 gene information for ACH-000815...\n",
      "Extracting top 10 transcript information for ACH-000815...\n",
      "Extracting top 10 protein information for ACH-000815...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20405 (ACH-000815) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20405 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 77%|███████▋  | 221/286 [6:58:55<2:06:19, 116.61s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 18 relationships between them\n",
      "Processing sample 222/286: ACH-000816 (NCI-H524)\n",
      "Sample Index: 222\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000816\n",
      "Extracting top 10 gene information for ACH-000816...\n",
      "Extracting top 10 transcript information for ACH-000816...\n",
      "Extracting top 10 protein information for ACH-000816...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000816) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 78%|███████▊  | 222/286 [7:01:43<2:20:42, 131.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 139 relationships between them\n",
      "Processing sample 223/286: ACH-000818 (BT-483)\n",
      "Sample Index: 223\n",
      "Sample Disease: breast adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07814\n",
      "Sample ACH Name: ACH-000818\n",
      "Extracting top 10 gene information for ACH-000818...\n",
      "Extracting top 10 transcript information for ACH-000818...\n",
      "Extracting top 10 protein information for ACH-000818...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07814 (ACH-000818) ...\n",
      "Knowledge Graph Info: Found 28 disease-related proteins directly connected to BMGC_DS07814 and 13521 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 78%|███████▊  | 223/286 [7:03:13<2:05:11, 119.23s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 93 relationships between them\n",
      "Processing sample 224/286: ACH-000821 (EJM)\n",
      "Sample Index: 224\n",
      "Sample Disease: plasma cell myeloma\n",
      "Sample Disease BMGC ID: BMGC_DS01639\n",
      "Sample ACH Name: ACH-000821\n",
      "Extracting top 10 gene information for ACH-000821...\n",
      "Extracting top 10 transcript information for ACH-000821...\n",
      "Extracting top 10 protein information for ACH-000821...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01639 (ACH-000821) ...\n",
      "Knowledge Graph Info: Found 244 disease-related proteins directly connected to BMGC_DS01639 and 24451 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 78%|███████▊  | 224/286 [7:06:19<2:24:01, 139.39s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 179 unique proteins and 170 relationships between them\n",
      "Processing sample 225/286: ACH-000822 (SK-MEL-24)\n",
      "Sample Index: 225\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-000822\n",
      "Extracting top 10 gene information for ACH-000822...\n",
      "Extracting top 10 transcript information for ACH-000822...\n",
      "Extracting top 10 protein information for ACH-000822...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-000822) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 79%|███████▊  | 225/286 [7:10:07<2:48:49, 166.05s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 281 unique proteins and 694 relationships between them\n",
      "Processing sample 226/286: ACH-000825 (HOP-92)\n",
      "Sample Index: 226\n",
      "Sample Disease: lung non-small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS00373\n",
      "Sample ACH Name: ACH-000825\n",
      "Extracting top 10 gene information for ACH-000825...\n",
      "Extracting top 10 transcript information for ACH-000825...\n",
      "Extracting top 10 protein information for ACH-000825...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00373 (ACH-000825) ...\n",
      "Knowledge Graph Info: Found 482 disease-related proteins directly connected to BMGC_DS00373 and 27691 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 79%|███████▉  | 226/286 [7:14:03<3:06:52, 186.87s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 282 unique proteins and 1017 relationships between them\n",
      "Processing sample 227/286: ACH-000828 (ZR-75-30)\n",
      "Sample Index: 227\n",
      "Sample Disease: breast adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07814\n",
      "Sample ACH Name: ACH-000828\n",
      "Extracting top 10 gene information for ACH-000828...\n",
      "Extracting top 10 transcript information for ACH-000828...\n",
      "Extracting top 10 protein information for ACH-000828...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07814 (ACH-000828) ...\n",
      "Knowledge Graph Info: Found 28 disease-related proteins directly connected to BMGC_DS07814 and 13521 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 79%|███████▉  | 227/286 [7:15:34<2:35:25, 158.05s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 46 relationships between them\n",
      "Processing sample 228/286: ACH-000830 (NCI-H1436)\n",
      "Sample Index: 228\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000830\n",
      "Extracting top 10 gene information for ACH-000830...\n",
      "Extracting top 10 transcript information for ACH-000830...\n",
      "Extracting top 10 protein information for ACH-000830...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000830) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 80%|███████▉  | 228/286 [7:18:22<2:35:40, 161.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 103 unique proteins and 149 relationships between them\n",
      "Processing sample 229/286: ACH-000835 (GCT)\n",
      "Sample Index: 229\n",
      "Sample Disease: undifferentiated pleomorphic sarcoma\n",
      "Sample Disease BMGC ID: BMGC_DS05556\n",
      "Sample ACH Name: ACH-000835\n",
      "Extracting top 10 gene information for ACH-000835...\n",
      "Extracting top 10 transcript information for ACH-000835...\n",
      "Extracting top 10 protein information for ACH-000835...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05556 (ACH-000835) ...\n",
      "Knowledge Graph Info: Found 4 disease-related proteins directly connected to BMGC_DS05556 and 5142 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 80%|████████  | 229/286 [7:18:56<1:56:54, 123.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 22 unique proteins and 20 relationships between them\n",
      "Processing sample 230/286: ACH-000843 (HARA)\n",
      "Sample Index: 230\n",
      "Sample Disease: carcinoma, non-small-cell lung\n",
      "Sample Disease BMGC ID: BMGC_DS20198\n",
      "Sample ACH Name: ACH-000843\n",
      "Extracting top 10 gene information for ACH-000843...\n",
      "Extracting top 10 transcript information for ACH-000843...\n",
      "Extracting top 10 protein information for ACH-000843...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20198 (ACH-000843) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20198 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 20 unique proteins and 51 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 80%|████████  | 230/286 [7:19:14<1:25:32, 91.65s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 230/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000843\n",
      "Progress: 230/286 (80.4%), Remaining: 56\n",
      "Processing sample 231/286: ACH-000852 (NCI-H1435)\n",
      "Sample Index: 231\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000852\n",
      "Extracting top 10 gene information for ACH-000852...\n",
      "Extracting top 10 transcript information for ACH-000852...\n",
      "Extracting top 10 protein information for ACH-000852...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000852) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 81%|████████  | 231/286 [7:23:13<2:04:25, 135.73s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 319 unique proteins and 834 relationships between them\n",
      "Processing sample 232/286: ACH-000857 (CAL-85-1)\n",
      "Sample Index: 232\n",
      "Sample Disease: breast carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07015\n",
      "Sample ACH Name: ACH-000857\n",
      "Extracting top 10 gene information for ACH-000857...\n",
      "Extracting top 10 transcript information for ACH-000857...\n",
      "Extracting top 10 protein information for ACH-000857...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07015 (ACH-000857) ...\n",
      "Knowledge Graph Info: Found 178 disease-related proteins directly connected to BMGC_DS07015 and 22704 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 81%|████████  | 232/286 [7:26:00<2:10:40, 145.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 97 unique proteins and 163 relationships between them\n",
      "Processing sample 233/286: ACH-000871 (NCI-H510)\n",
      "Sample Index: 233\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-000871\n",
      "Extracting top 10 gene information for ACH-000871...\n",
      "Extracting top 10 transcript information for ACH-000871...\n",
      "Extracting top 10 protein information for ACH-000871...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-000871) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 81%|████████▏ | 233/286 [7:28:48<2:14:19, 152.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 101 unique proteins and 149 relationships between them\n",
      "Processing sample 234/286: ACH-000873 (KYSE-270)\n",
      "Sample Index: 234\n",
      "Sample Disease: esophageal squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05334\n",
      "Sample ACH Name: ACH-000873\n",
      "Extracting top 10 gene information for ACH-000873...\n",
      "Extracting top 10 transcript information for ACH-000873...\n",
      "Extracting top 10 protein information for ACH-000873...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05334 (ACH-000873) ...\n",
      "Knowledge Graph Info: Found 226 disease-related proteins directly connected to BMGC_DS05334 and 22772 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 82%|████████▏ | 234/286 [7:31:42<2:17:22, 158.51s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 146 unique proteins and 232 relationships between them\n",
      "Processing sample 235/286: ACH-000875 (NCI-H2347)\n",
      "Sample Index: 235\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000875\n",
      "Extracting top 10 gene information for ACH-000875...\n",
      "Extracting top 10 transcript information for ACH-000875...\n",
      "Extracting top 10 protein information for ACH-000875...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000875) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 82%|████████▏ | 235/286 [7:35:42<2:35:31, 182.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 310 unique proteins and 767 relationships between them\n",
      "Processing sample 236/286: ACH-000883 (SW 1783)\n",
      "Sample Index: 236\n",
      "Sample Disease: anaplastic astrocytoma\n",
      "Sample Disease BMGC ID: BMGC_DS05597\n",
      "Sample ACH Name: ACH-000883\n",
      "Extracting top 10 gene information for ACH-000883...\n",
      "Extracting top 10 transcript information for ACH-000883...\n",
      "Extracting top 10 protein information for ACH-000883...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05597 (ACH-000883) ...\n",
      "Knowledge Graph Info: Found 7 disease-related proteins directly connected to BMGC_DS05597 and 9777 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 83%|████████▎ | 236/286 [7:36:47<2:02:59, 147.59s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 23 unique proteins and 26 relationships between them\n",
      "Processing sample 237/286: ACH-000892 (NCI-H1563)\n",
      "Sample Index: 237\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000892\n",
      "Extracting top 10 gene information for ACH-000892...\n",
      "Extracting top 10 transcript information for ACH-000892...\n",
      "Extracting top 10 protein information for ACH-000892...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000892) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 83%|████████▎ | 237/286 [7:40:46<2:23:01, 175.14s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 309 unique proteins and 831 relationships between them\n",
      "Processing sample 238/286: ACH-000893 (NCI-H1651)\n",
      "Sample Index: 238\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000893\n",
      "Extracting top 10 gene information for ACH-000893...\n",
      "Extracting top 10 transcript information for ACH-000893...\n",
      "Extracting top 10 protein information for ACH-000893...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000893) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 83%|████████▎ | 238/286 [7:44:46<2:35:38, 194.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 309 unique proteins and 849 relationships between them\n",
      "Processing sample 239/286: ACH-000894 (NCI-H1869)\n",
      "Sample Index: 239\n",
      "Sample Disease: lung squamous cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02872\n",
      "Sample ACH Name: ACH-000894\n",
      "Extracting top 10 gene information for ACH-000894...\n",
      "Extracting top 10 transcript information for ACH-000894...\n",
      "Extracting top 10 protein information for ACH-000894...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02872 (ACH-000894) ...\n",
      "Knowledge Graph Info: Found 142 disease-related proteins directly connected to BMGC_DS02872 and 21840 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 84%|████████▎ | 239/286 [7:47:24<2:23:46, 183.55s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 105 unique proteins and 98 relationships between them\n",
      "Processing sample 240/286: ACH-000895 (CL-34)\n",
      "Sample Index: 240\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000895\n",
      "Extracting top 10 gene information for ACH-000895...\n",
      "Extracting top 10 transcript information for ACH-000895...\n",
      "Extracting top 10 protein information for ACH-000895...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000895) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 50 unique proteins and 117 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 84%|████████▍ | 240/286 [7:49:54<2:13:04, 173.57s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 240/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000895\n",
      "Progress: 240/286 (83.9%), Remaining: 46\n",
      "Processing sample 241/286: ACH-000896 (647-V)\n",
      "Sample Index: 241\n",
      "Sample Disease: bladder carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07053\n",
      "Sample ACH Name: ACH-000896\n",
      "Extracting top 10 gene information for ACH-000896...\n",
      "Extracting top 10 transcript information for ACH-000896...\n",
      "Extracting top 10 protein information for ACH-000896...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07053 (ACH-000896) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS07053 and 8832 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 84%|████████▍ | 241/286 [7:50:54<1:44:35, 139.45s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 31 relationships between them\n",
      "Processing sample 242/286: ACH-000902 (CAL-148)\n",
      "Sample Index: 242\n",
      "Sample Disease: breast carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07015\n",
      "Sample ACH Name: ACH-000902\n",
      "Extracting top 10 gene information for ACH-000902...\n",
      "Extracting top 10 transcript information for ACH-000902...\n",
      "Extracting top 10 protein information for ACH-000902...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07015 (ACH-000902) ...\n",
      "Knowledge Graph Info: Found 178 disease-related proteins directly connected to BMGC_DS07015 and 22704 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 85%|████████▍ | 242/286 [7:53:41<1:48:17, 147.67s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 87 unique proteins and 93 relationships between them\n",
      "Processing sample 243/286: ACH-000903 (FTC-133)\n",
      "Sample Index: 243\n",
      "Sample Disease: thyroid gland follicular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS04082\n",
      "Sample ACH Name: ACH-000903\n",
      "Extracting top 10 gene information for ACH-000903...\n",
      "Extracting top 10 transcript information for ACH-000903...\n",
      "Extracting top 10 protein information for ACH-000903...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS04082 (ACH-000903) ...\n",
      "Knowledge Graph Info: Found 17 disease-related proteins directly connected to BMGC_DS04082 and 11025 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 85%|████████▍ | 243/286 [7:54:55<1:30:03, 125.67s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 29 unique proteins and 26 relationships between them\n",
      "Processing sample 244/286: ACH-000905 (5637)\n",
      "Sample Index: 244\n",
      "Sample Disease: bladder carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07053\n",
      "Sample ACH Name: ACH-000905\n",
      "Extracting top 10 gene information for ACH-000905...\n",
      "Extracting top 10 transcript information for ACH-000905...\n",
      "Extracting top 10 protein information for ACH-000905...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07053 (ACH-000905) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS07053 and 8832 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 85%|████████▌ | 244/286 [7:55:56<1:14:16, 106.10s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 48 relationships between them\n",
      "Processing sample 245/286: ACH-000914 (HT)\n",
      "Sample Index: 245\n",
      "Sample Disease: diffuse large b-cell lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS02693\n",
      "Sample ACH Name: ACH-000914\n",
      "Extracting top 10 gene information for ACH-000914...\n",
      "Extracting top 10 transcript information for ACH-000914...\n",
      "Extracting top 10 protein information for ACH-000914...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02693 (ACH-000914) ...\n",
      "Knowledge Graph Info: Found 148 disease-related proteins directly connected to BMGC_DS02693 and 20906 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 86%|████████▌ | 245/286 [7:58:28<1:21:57, 119.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 88 unique proteins and 101 relationships between them\n",
      "Processing sample 246/286: ACH-000918 (MOLT-16)\n",
      "Sample Index: 246\n",
      "Sample Disease: leukemia, myeloid, acute\n",
      "Sample Disease BMGC ID: BMGC_DS20800\n",
      "Sample ACH Name: ACH-000918\n",
      "Extracting top 10 gene information for ACH-000918...\n",
      "Extracting top 10 transcript information for ACH-000918...\n",
      "Extracting top 10 protein information for ACH-000918...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20800 (ACH-000918) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20800 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 86%|████████▌ | 246/286 [7:58:30<56:26, 84.67s/it]   "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 18 relationships between them\n",
      "Processing sample 247/286: ACH-000920 (CML-T1)\n",
      "Sample Index: 247\n",
      "Sample Disease: chronic myelogenous leukemia, bcr-abl1 positive\n",
      "Sample Disease BMGC ID: BMGC_DS01409\n",
      "Sample ACH Name: ACH-000920\n",
      "Extracting top 10 gene information for ACH-000920...\n",
      "Extracting top 10 transcript information for ACH-000920...\n",
      "Extracting top 10 protein information for ACH-000920...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01409 (ACH-000920) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS01409 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 86%|████████▋ | 247/286 [7:58:33<38:57, 59.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 18 relationships between them\n",
      "Processing sample 248/286: ACH-000927 (BT-474)\n",
      "Sample Index: 248\n",
      "Sample Disease: breast adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07814\n",
      "Sample ACH Name: ACH-000927\n",
      "Extracting top 10 gene information for ACH-000927...\n",
      "Extracting top 10 transcript information for ACH-000927...\n",
      "Extracting top 10 protein information for ACH-000927...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07814 (ACH-000927) ...\n",
      "Knowledge Graph Info: Found 28 disease-related proteins directly connected to BMGC_DS07814 and 13521 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 87%|████████▋ | 248/286 [8:00:03<43:47, 69.15s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 28 unique proteins and 62 relationships between them\n",
      "Processing sample 249/286: ACH-000930 (HCC1569)\n",
      "Sample Index: 249\n",
      "Sample Disease: breast ductal carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08035\n",
      "Sample ACH Name: ACH-000930\n",
      "Extracting top 10 gene information for ACH-000930...\n",
      "Extracting top 10 transcript information for ACH-000930...\n",
      "Extracting top 10 protein information for ACH-000930...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08035 (ACH-000930) ...\n",
      "Knowledge Graph Info: Found 17 disease-related proteins directly connected to BMGC_DS08035 and 13835 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 87%|████████▋ | 249/286 [8:01:34<46:35, 75.56s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 32 unique proteins and 62 relationships between them\n",
      "Processing sample 250/286: ACH-000937 (PF-382)\n",
      "Sample Index: 250\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-000937\n",
      "Extracting top 10 gene information for ACH-000937...\n",
      "Extracting top 10 transcript information for ACH-000937...\n",
      "Extracting top 10 protein information for ACH-000937...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-000937) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 106 unique proteins and 145 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 87%|████████▋ | 250/286 [8:04:20<1:01:39, 102.75s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 250/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000937\n",
      "Progress: 250/286 (87.4%), Remaining: 36\n",
      "Processing sample 251/286: ACH-000939 (SK-UT-1)\n",
      "Sample Index: 251\n",
      "Sample Disease: leiomyosarcoma of the corpus uteri\n",
      "Sample Disease BMGC ID: BMGC_DS05401\n",
      "Sample ACH Name: ACH-000939\n",
      "Extracting top 10 gene information for ACH-000939...\n",
      "Extracting top 10 transcript information for ACH-000939...\n",
      "Extracting top 10 protein information for ACH-000939...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05401 (ACH-000939) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS05401 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 88%|████████▊ | 251/286 [8:04:23<42:23, 72.68s/it]   "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 28 relationships between them\n",
      "Processing sample 252/286: ACH-000940 (AN3 CA)\n",
      "Sample Index: 252\n",
      "Sample Disease: acanthosis nigricans\n",
      "Sample Disease BMGC ID: BMGC_DS00010\n",
      "Sample ACH Name: ACH-000940\n",
      "Extracting top 10 gene information for ACH-000940...\n",
      "Extracting top 10 transcript information for ACH-000940...\n",
      "Extracting top 10 protein information for ACH-000940...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00010 (ACH-000940) ...\n",
      "Knowledge Graph Info: Found 16 disease-related proteins directly connected to BMGC_DS00010 and 4419 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 88%|████████▊ | 252/286 [8:04:55<34:23, 60.68s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 25 unique proteins and 48 relationships between them\n",
      "Processing sample 253/286: ACH-000944 (NAMALWA)\n",
      "Sample Index: 253\n",
      "Sample Disease: burkitt lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS00331\n",
      "Sample ACH Name: ACH-000944\n",
      "Extracting top 10 gene information for ACH-000944...\n",
      "Extracting top 10 transcript information for ACH-000944...\n",
      "Extracting top 10 protein information for ACH-000944...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS00331 (ACH-000944) ...\n",
      "Knowledge Graph Info: Found 67 disease-related proteins directly connected to BMGC_DS00331 and 15225 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 88%|████████▊ | 253/286 [8:06:43<41:11, 74.90s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 43 unique proteins and 43 relationships between them\n",
      "Processing sample 254/286: ACH-000947 (OVK18)\n",
      "Sample Index: 254\n",
      "Sample Disease: ovarian endometrioid adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS06034\n",
      "Sample ACH Name: ACH-000947\n",
      "Extracting top 10 gene information for ACH-000947...\n",
      "Extracting top 10 transcript information for ACH-000947...\n",
      "Extracting top 10 protein information for ACH-000947...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS06034 (ACH-000947) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS06034 and 2497 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 89%|████████▉ | 254/286 [8:07:04<31:13, 58.55s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 25 unique proteins and 18 relationships between them\n",
      "Processing sample 255/286: ACH-000949 (TGBC11TKB)\n",
      "Sample Index: 255\n",
      "Sample Disease: gastric carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07051\n",
      "Sample ACH Name: ACH-000949\n",
      "Extracting top 10 gene information for ACH-000949...\n",
      "Extracting top 10 transcript information for ACH-000949...\n",
      "Extracting top 10 protein information for ACH-000949...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07051 (ACH-000949) ...\n",
      "Knowledge Graph Info: Found 35 disease-related proteins directly connected to BMGC_DS07051 and 12899 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 89%|████████▉ | 255/286 [8:08:31<34:41, 67.15s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 36 unique proteins and 24 relationships between them\n",
      "Processing sample 256/286: ACH-000951 (NCI-H2342)\n",
      "Sample Index: 256\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-000951\n",
      "Extracting top 10 gene information for ACH-000951...\n",
      "Extracting top 10 transcript information for ACH-000951...\n",
      "Extracting top 10 protein information for ACH-000951...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-000951) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 90%|████████▉ | 256/286 [8:12:31<59:28, 118.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 310 unique proteins and 812 relationships between them\n",
      "Processing sample 257/286: ACH-000955 (SNU-407)\n",
      "Sample Index: 257\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000955\n",
      "Extracting top 10 gene information for ACH-000955...\n",
      "Extracting top 10 transcript information for ACH-000955...\n",
      "Extracting top 10 protein information for ACH-000955...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000955) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 90%|████████▉ | 257/286 [8:14:44<59:37, 123.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 81 relationships between them\n",
      "Processing sample 258/286: ACH-000968 (COLO 792)\n",
      "Sample Index: 258\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-000968\n",
      "Extracting top 10 gene information for ACH-000968...\n",
      "Extracting top 10 transcript information for ACH-000968...\n",
      "Extracting top 10 protein information for ACH-000968...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-000968) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 90%|█████████ | 258/286 [8:18:33<1:12:16, 154.89s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 280 unique proteins and 787 relationships between them\n",
      "Processing sample 259/286: ACH-000970 (SNU-C5)\n",
      "Sample Index: 259\n",
      "Sample Disease: cecum adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08517\n",
      "Sample ACH Name: ACH-000970\n",
      "Extracting top 10 gene information for ACH-000970...\n",
      "Extracting top 10 transcript information for ACH-000970...\n",
      "Extracting top 10 protein information for ACH-000970...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08517 (ACH-000970) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS08517 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 91%|█████████ | 259/286 [8:18:35<49:06, 109.13s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 56 relationships between them\n",
      "Processing sample 260/286: ACH-000973 (639-V)\n",
      "Sample Index: 260\n",
      "Sample Disease: bladder carcinoma or urinary bladder carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07053\n",
      "Sample ACH Name: ACH-000973\n",
      "Extracting top 10 gene information for ACH-000973...\n",
      "Extracting top 10 transcript information for ACH-000973...\n",
      "Extracting top 10 protein information for ACH-000973...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07053 (ACH-000973) ...\n",
      "Knowledge Graph Info: Found 12 disease-related proteins directly connected to BMGC_DS07053 and 8832 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 28 unique proteins and 43 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 91%|█████████ | 260/286 [8:19:54<43:18, 99.94s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 260/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-000973\n",
      "Progress: 260/286 (90.9%), Remaining: 26\n",
      "Processing sample 261/286: ACH-000976 (HuCCT1)\n",
      "Sample Index: 261\n",
      "Sample Disease: intrahepatic cholangiocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05988\n",
      "Sample ACH Name: ACH-000976\n",
      "Extracting top 10 gene information for ACH-000976...\n",
      "Extracting top 10 transcript information for ACH-000976...\n",
      "Extracting top 10 protein information for ACH-000976...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05988 (ACH-000976) ...\n",
      "Knowledge Graph Info: Found 11 disease-related proteins directly connected to BMGC_DS05988 and 9336 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 91%|█████████▏| 261/286 [8:20:57<37:03, 88.95s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 25 unique proteins and 36 relationships between them\n",
      "Processing sample 262/286: ACH-000981 (DND-41)\n",
      "Sample Index: 262\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-000981\n",
      "Extracting top 10 gene information for ACH-000981...\n",
      "Extracting top 10 transcript information for ACH-000981...\n",
      "Extracting top 10 protein information for ACH-000981...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-000981) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 92%|█████████▏| 262/286 [8:23:27<42:54, 107.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 106 unique proteins and 120 relationships between them\n",
      "Processing sample 263/286: ACH-000983 (KCL-22)\n",
      "Sample Index: 263\n",
      "Sample Disease: chronic myelogenous leukemia, bcr-abl1 positive\n",
      "Sample Disease BMGC ID: BMGC_DS01409\n",
      "Sample ACH Name: ACH-000983\n",
      "Extracting top 10 gene information for ACH-000983...\n",
      "Extracting top 10 transcript information for ACH-000983...\n",
      "Extracting top 10 protein information for ACH-000983...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01409 (ACH-000983) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS01409 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 92%|█████████▏| 263/286 [8:23:29<29:03, 75.79s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 56 relationships between them\n",
      "Processing sample 264/286: ACH-000987 (MeWo)\n",
      "Sample Index: 264\n",
      "Sample Disease: cutaneous melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS02930\n",
      "Sample ACH Name: ACH-000987\n",
      "Extracting top 10 gene information for ACH-000987...\n",
      "Extracting top 10 transcript information for ACH-000987...\n",
      "Extracting top 10 protein information for ACH-000987...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02930 (ACH-000987) ...\n",
      "Knowledge Graph Info: Found 119 disease-related proteins directly connected to BMGC_DS02930 and 21421 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 92%|█████████▏| 264/286 [8:26:00<36:04, 98.39s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 82 unique proteins and 155 relationships between them\n",
      "Processing sample 265/286: ACH-000989 (SNU-175)\n",
      "Sample Index: 265\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000989\n",
      "Extracting top 10 gene information for ACH-000989...\n",
      "Extracting top 10 transcript information for ACH-000989...\n",
      "Extracting top 10 protein information for ACH-000989...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000989) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 93%|█████████▎| 265/286 [8:28:13<37:59, 108.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 90 relationships between them\n",
      "Processing sample 266/286: ACH-000995 (JURKAT)\n",
      "Sample Index: 266\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-000995\n",
      "Extracting top 10 gene information for ACH-000995...\n",
      "Extracting top 10 transcript information for ACH-000995...\n",
      "Extracting top 10 protein information for ACH-000995...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-000995) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 93%|█████████▎| 266/286 [8:30:44<40:24, 121.24s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 106 unique proteins and 131 relationships between them\n",
      "Processing sample 267/286: ACH-000998 (CW-2)\n",
      "Sample Index: 267\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000998\n",
      "Extracting top 10 gene information for ACH-000998...\n",
      "Extracting top 10 transcript information for ACH-000998...\n",
      "Extracting top 10 protein information for ACH-000998...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000998) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 93%|█████████▎| 267/286 [8:32:58<39:36, 125.09s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 87 relationships between them\n",
      "Processing sample 268/286: ACH-000999 (SNU-1040)\n",
      "Sample Index: 268\n",
      "Sample Disease: colon adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05624\n",
      "Sample ACH Name: ACH-000999\n",
      "Extracting top 10 gene information for ACH-000999...\n",
      "Extracting top 10 transcript information for ACH-000999...\n",
      "Extracting top 10 protein information for ACH-000999...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05624 (ACH-000999) ...\n",
      "Knowledge Graph Info: Found 72 disease-related proteins directly connected to BMGC_DS05624 and 19347 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 94%|█████████▎| 268/286 [8:35:11<38:14, 127.50s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 50 unique proteins and 88 relationships between them\n",
      "Processing sample 269/286: ACH-001016 (Becker)\n",
      "Sample Index: 269\n",
      "Sample Disease: astrocytoma\n",
      "Sample Disease BMGC ID: BMGC_DS20147\n",
      "Sample ACH Name: ACH-001016\n",
      "Extracting top 10 gene information for ACH-001016...\n",
      "Extracting top 10 transcript information for ACH-001016...\n",
      "Extracting top 10 protein information for ACH-001016...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20147 (ACH-001016) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20147 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 94%|█████████▍| 269/286 [8:35:13<25:28, 89.90s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 24 relationships between them\n",
      "Processing sample 270/286: ACH-001021 (C3A)\n",
      "Sample Index: 270\n",
      "Sample Disease: hepatoblastoma\n",
      "Sample Disease BMGC ID: BMGC_DS04044\n",
      "Sample ACH Name: ACH-001021\n",
      "Extracting top 10 gene information for ACH-001021...\n",
      "Extracting top 10 transcript information for ACH-001021...\n",
      "Extracting top 10 protein information for ACH-001021...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS04044 (ACH-001021) ...\n",
      "Knowledge Graph Info: Found 106 disease-related proteins directly connected to BMGC_DS04044 and 18774 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 65 unique proteins and 27 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 94%|█████████▍| 270/286 [8:37:45<28:55, 108.44s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 270/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-001021\n",
      "Progress: 270/286 (94.4%), Remaining: 16\n",
      "Processing sample 271/286: ACH-001023 (CGTH-W-1)\n",
      "Sample Index: 271\n",
      "Sample Disease: thyroid carcinoma, anaplastic\n",
      "Sample Disease BMGC ID: BMGC_DS21147\n",
      "Sample ACH Name: ACH-001023\n",
      "Extracting top 10 gene information for ACH-001023...\n",
      "Extracting top 10 transcript information for ACH-001023...\n",
      "Extracting top 10 protein information for ACH-001023...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS21147 (ACH-001023) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS21147 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 95%|█████████▍| 271/286 [8:37:47<19:11, 76.74s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 10 unique proteins and 0 relationships between them\n",
      "Processing sample 272/286: ACH-001024 (CHL-1)\n",
      "Sample Index: 272\n",
      "Sample Disease: melanoma\n",
      "Sample Disease BMGC ID: BMGC_DS01559\n",
      "Sample ACH Name: ACH-001024\n",
      "Extracting top 10 gene information for ACH-001024...\n",
      "Extracting top 10 transcript information for ACH-001024...\n",
      "Extracting top 10 protein information for ACH-001024...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS01559 (ACH-001024) ...\n",
      "Knowledge Graph Info: Found 466 disease-related proteins directly connected to BMGC_DS01559 and 26910 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 95%|█████████▌| 272/286 [8:41:36<28:33, 122.43s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 271 unique proteins and 684 relationships between them\n",
      "Processing sample 273/286: ACH-001049 (CPC-N)\n",
      "Sample Index: 273\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-001049\n",
      "Extracting top 10 gene information for ACH-001049...\n",
      "Extracting top 10 transcript information for ACH-001049...\n",
      "Extracting top 10 protein information for ACH-001049...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-001049) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 95%|█████████▌| 273/286 [8:44:27<29:38, 136.81s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 93 unique proteins and 86 relationships between them\n",
      "Processing sample 274/286: ACH-001063 (DOV13)\n",
      "Sample Index: 274\n",
      "Sample Disease: ovarian adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07934\n",
      "Sample ACH Name: ACH-001063\n",
      "Extracting top 10 gene information for ACH-001063...\n",
      "Extracting top 10 transcript information for ACH-001063...\n",
      "Extracting top 10 protein information for ACH-001063...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07934 (ACH-001063) ...\n",
      "Knowledge Graph Info: Found 5 disease-related proteins directly connected to BMGC_DS07934 and 4915 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 96%|█████████▌| 274/286 [8:45:01<21:11, 105.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 2 unique proteins and 0 relationships between them\n",
      "Processing sample 275/286: ACH-001064 (EB2)\n",
      "Sample Index: 275\n",
      "Sample Disease: burkitt lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS20177\n",
      "Sample ACH Name: ACH-001064\n",
      "Extracting top 10 gene information for ACH-001064...\n",
      "Extracting top 10 transcript information for ACH-001064...\n",
      "Extracting top 10 protein information for ACH-001064...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20177 (ACH-001064) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20177 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 96%|█████████▌| 275/286 [8:45:03<13:44, 74.96s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 20 unique proteins and 26 relationships between them\n",
      "Processing sample 276/286: ACH-001089 (HLE)\n",
      "Sample Index: 276\n",
      "Sample Disease: adult hepatocellular carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS05326\n",
      "Sample ACH Name: ACH-001089\n",
      "Extracting top 10 gene information for ACH-001089...\n",
      "Extracting top 10 transcript information for ACH-001089...\n",
      "Extracting top 10 protein information for ACH-001089...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05326 (ACH-001089) ...\n",
      "Knowledge Graph Info: Found 23 disease-related proteins directly connected to BMGC_DS05326 and 7706 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 97%|█████████▋| 276/286 [8:45:57<11:26, 68.69s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 15 unique proteins and 1 relationships between them\n",
      "Processing sample 277/286: ACH-001106 (KOPN-8)\n",
      "Sample Index: 277\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-001106\n",
      "Extracting top 10 gene information for ACH-001106...\n",
      "Extracting top 10 transcript information for ACH-001106...\n",
      "Extracting top 10 protein information for ACH-001106...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-001106) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 97%|█████████▋| 277/286 [8:48:29<14:01, 93.51s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 106 unique proteins and 134 relationships between them\n",
      "Processing sample 278/286: ACH-001125 (MOG-G-CCM)\n",
      "Sample Index: 278\n",
      "Sample Disease: anaplastic astrocytoma\n",
      "Sample Disease BMGC ID: BMGC_DS05597\n",
      "Sample ACH Name: ACH-001125\n",
      "Extracting top 10 gene information for ACH-001125...\n",
      "Extracting top 10 transcript information for ACH-001125...\n",
      "Extracting top 10 protein information for ACH-001125...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05597 (ACH-001125) ...\n",
      "Knowledge Graph Info: Found 7 disease-related proteins directly connected to BMGC_DS05597 and 9777 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 97%|█████████▋| 278/286 [8:49:34<11:19, 84.99s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 13 unique proteins and 0 relationships between them\n",
      "Processing sample 279/286: ACH-001126 (MOG-G-UVW)\n",
      "Sample Index: 279\n",
      "Sample Disease: anaplastic astrocytoma\n",
      "Sample Disease BMGC ID: BMGC_DS05597\n",
      "Sample ACH Name: ACH-001126\n",
      "Extracting top 10 gene information for ACH-001126...\n",
      "Extracting top 10 transcript information for ACH-001126...\n",
      "Extracting top 10 protein information for ACH-001126...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS05597 (ACH-001126) ...\n",
      "Knowledge Graph Info: Found 7 disease-related proteins directly connected to BMGC_DS05597 and 9777 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 98%|█████████▊| 279/286 [8:50:39<09:12, 79.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 13 unique proteins and 0 relationships between them\n",
      "Processing sample 280/286: ACH-001127 (MOLT-4)\n",
      "Sample Index: 280\n",
      "Sample Disease: acute lymphoblastic leukemia or precursor lymphoblastic lymphoma/leukemia\n",
      "Sample Disease BMGC ID: BMGC_DS09029\n",
      "Sample ACH Name: ACH-001127\n",
      "Extracting top 10 gene information for ACH-001127...\n",
      "Extracting top 10 transcript information for ACH-001127...\n",
      "Extracting top 10 protein information for ACH-001127...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS09029 (ACH-001127) ...\n",
      "Knowledge Graph Info: Found 159 disease-related proteins directly connected to BMGC_DS09029 and 20314 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n",
      "Knowledge Graph Info: Found 96 unique proteins and 91 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 98%|█████████▊| 280/286 [8:53:33<10:44, 107.38s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "💾 Auto-saved JSON at 280/286 samples to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Last processed sample: ACH-001127\n",
      "Progress: 280/286 (97.9%), Remaining: 6\n",
      "Processing sample 281/286: ACH-001138 (NCI-H2141)\n",
      "Sample Index: 281\n",
      "Sample Disease: lung small cell carcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02887\n",
      "Sample ACH Name: ACH-001138\n",
      "Extracting top 10 gene information for ACH-001138...\n",
      "Extracting top 10 transcript information for ACH-001138...\n",
      "Extracting top 10 protein information for ACH-001138...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02887 (ACH-001138) ...\n",
      "Knowledge Graph Info: Found 193 disease-related proteins directly connected to BMGC_DS02887 and 22301 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 98%|█████████▊| 281/286 [8:56:22<10:30, 126.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 92 unique proteins and 82 relationships between them\n",
      "Processing sample 282/286: ACH-001151 (OVCAR-5)\n",
      "Sample Index: 282\n",
      "Sample Disease: ovarian adenocarcinoma or ovary adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS07934\n",
      "Sample ACH Name: ACH-001151\n",
      "Extracting top 10 gene information for ACH-001151...\n",
      "Extracting top 10 transcript information for ACH-001151...\n",
      "Extracting top 10 protein information for ACH-001151...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS07934 (ACH-001151) ...\n",
      "Knowledge Graph Info: Found 5 disease-related proteins directly connected to BMGC_DS07934 and 4915 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 99%|█████████▊| 282/286 [8:56:56<06:33, 98.48s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 22 unique proteins and 62 relationships between them\n",
      "Processing sample 283/286: ACH-001182 (RPMI 6666)\n",
      "Sample Index: 283\n",
      "Sample Disease: hodgkin disease or hodgkins lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS20405\n",
      "Sample ACH Name: ACH-001182\n",
      "Extracting top 10 gene information for ACH-001182...\n",
      "Extracting top 10 transcript information for ACH-001182...\n",
      "Extracting top 10 protein information for ACH-001182...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20405 (ACH-001182) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20405 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 99%|█████████▉| 283/286 [8:56:59<03:28, 69.58s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 10 unique proteins and 1 relationships between them\n",
      "Processing sample 284/286: ACH-001199 (SNU-C2B)\n",
      "Sample Index: 284\n",
      "Sample Disease: cecum adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS08517\n",
      "Sample ACH Name: ACH-001199\n",
      "Extracting top 10 gene information for ACH-001199...\n",
      "Extracting top 10 transcript information for ACH-001199...\n",
      "Extracting top 10 protein information for ACH-001199...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS08517 (ACH-001199) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS08517 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 99%|█████████▉| 284/286 [8:57:00<01:38, 49.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 10 unique proteins and 0 relationships between them\n",
      "Processing sample 285/286: ACH-001203 (SUP-HD1)\n",
      "Sample Index: 285\n",
      "Sample Disease: hodgkin disease or hodgkins lymphoma\n",
      "Sample Disease BMGC ID: BMGC_DS20405\n",
      "Sample ACH Name: ACH-001203\n",
      "Extracting top 10 gene information for ACH-001203...\n",
      "Extracting top 10 transcript information for ACH-001203...\n",
      "Extracting top 10 protein information for ACH-001203...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS20405 (ACH-001203) ...\n",
      "Knowledge Graph Info: Found 0 disease-related proteins directly connected to BMGC_DS20405 and 0 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████▉| 285/286 [8:57:03<00:35, 35.35s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 10 unique proteins and 0 relationships between them\n",
      "Processing sample 286/286: ACH-001233 (VMRC-LCD)\n",
      "Sample Index: 286\n",
      "Sample Disease: lung adenocarcinoma\n",
      "Sample Disease BMGC ID: BMGC_DS02944\n",
      "Sample ACH Name: ACH-001233\n",
      "Extracting top 10 gene information for ACH-001233...\n",
      "Extracting top 10 transcript information for ACH-001233...\n",
      "Extracting top 10 protein information for ACH-001233...\n",
      "Extracting disease-related proteins index and bmgc id for BMGC_DS02944 (ACH-001233) ...\n",
      "Knowledge Graph Info: Found 532 disease-related proteins directly connected to BMGC_DS02944 and 27288 proteins in their PPI network\n",
      "Mapping disease-related proteins to HGNC symbols...\n",
      "Mapping PPI-related proteins to HGNC symbols...\n",
      "Extracting protein relationships from BMGC...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 286/286 [9:01:07<00:00, 113.52s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Knowledge Graph Info: Found 300 unique proteins and 655 relationships between them\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Final JSON saved to: ./data/DrugQA\\drug_qa_k10_bm100.json\n",
      "Total samples processed: 286\n"
     ]
    }
   ],
   "source": [
    "# Get the list of samples that have already been processed\n",
    "processed_samples = set(multi_sample_qa_json.keys())\n",
    "print(f\"Found {len(processed_samples)} already processed samples\")\n",
    "\n",
    "count = len(processed_samples)\n",
    "total_to_process = len(dti_sample_info_index)\n",
    "remaining = total_to_process - count\n",
    "print(f\"Total samples to process: {total_to_process}, already processed: {count}, remaining: {remaining}\")\n",
    "\n",
    "# Iterate through the sample info dataframe, skipping already processed samples\n",
    "for idx, row_tuple in tqdm(enumerate(dti_sample_info_index.iterrows()), total=len(dti_sample_info_index)):\n",
    "    _, row = row_tuple  # Unpack the tuple - index and row data\n",
    "\n",
    "    sample_ach_name = row[\"depMapID\"]\n",
    "    \n",
    "    # Skip if already processed\n",
    "    if sample_ach_name in processed_samples:\n",
    "        continue\n",
    "        \n",
    "    count += 1\n",
    "    target_sample_index = row[\"Index\"]\n",
    "    cell_line_name = row[\"Name\"]\n",
    "    disease = row[\"BMGC_Disease_name\"]\n",
    "    disease_bmgc_id = row[\"BMGC_Disease_ID\"]\n",
    "\n",
    "    print(f\"Processing sample {count}/{total_to_process}: {sample_ach_name} ({cell_line_name})\")\n",
    "    print(f\"Sample Index: {target_sample_index}\")\n",
    "    print(f\"Sample Disease: {disease}\")\n",
    "    print(f\"Sample Disease BMGC ID: {disease_bmgc_id}\")\n",
    "\n",
    "    try:\n",
    "        (top_k_gene_hgnc, top_k_gene_bmgc, top_k_gene_llm,\n",
    "        top_k_ts_hgnc, top_k_ts_bmgc, top_k_ts_llm,\n",
    "        top_k_pt_hgnc, top_k_pt_bmgc, top_k_pt_llm,\n",
    "        dis_pt_idx, dis_pt_bmgc, ppi_idx, ppi_bmgc,\n",
    "        dis_pt_hgnc_dict, dis_pt_hgnc, \n",
    "        ppi_hgnc_dict, ppi_hgnc, relationship_texts) = drug_qa_sample_info(sample_ach_name, disease_bmgc_id, k)\n",
    "\n",
    "        multi_sample_qa_json[sample_ach_name] = {\n",
    "            \"cell_line_name\": cell_line_name,\n",
    "            \"sample_index\": target_sample_index,\n",
    "            \"disease\": disease,\n",
    "            \"disease_bmgc_id\": disease_bmgc_id,\n",
    "            \"input\": {\n",
    "                \"top_k_gene\": {\n",
    "                    \"hgnc_symbols\": top_k_gene_hgnc,\n",
    "                    \"protein_bmgc_ids\": top_k_gene_bmgc,\n",
    "                    \"protein_llmname_ids\": top_k_gene_llm\n",
    "                },\n",
    "                \"top_k_transcript\": {\n",
    "                    \"hgnc_symbols\": top_k_ts_hgnc,\n",
    "                    \"protein_bmgc_ids\": top_k_ts_bmgc,\n",
    "                    \"protein_llmname_ids\": top_k_ts_llm\n",
    "                },\n",
    "                \"top_k_protein\": {\n",
    "                    \"hgnc_symbols\": top_k_pt_hgnc,\n",
    "                    \"protein_bmgc_ids\": top_k_pt_bmgc,\n",
    "                    \"protein_llmname_ids\": top_k_pt_llm\n",
    "                },\n",
    "                \"knowledge_graph\": {\n",
    "                    \"disease_protein\": {\n",
    "                        \"bmgc_ids\": dis_pt_bmgc,\n",
    "                        \"hgnc_symbols\": dis_pt_hgnc,\n",
    "                        \"indices\": dis_pt_idx\n",
    "                    },\n",
    "                    \"ppi_neighbors\": {\n",
    "                        \"bmgc_ids\": ppi_bmgc,\n",
    "                        \"hgnc_symbols\": ppi_hgnc,\n",
    "                        \"indices\": ppi_idx\n",
    "                    },\n",
    "                    \"protein_relationships\": relationship_texts,\n",
    "                }\n",
    "            }\n",
    "        }\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"⚠️ Error processing {sample_ach_name}: {e}\")\n",
    "        continue\n",
    "\n",
    "    # Periodic save every N samples\n",
    "    if count % save_every_n == 0:\n",
    "        with open(output_path, \"w\") as f:\n",
    "            json.dump(multi_sample_qa_json, f, indent=2, default=lambda o: int(o) if hasattr(o, 'item') else o)\n",
    "        print(f\"💾 Auto-saved JSON at {count}/{total_to_process} samples to: {output_path}\")\n",
    "        print(f\"Last processed sample: {sample_ach_name}\")\n",
    "        processed = len(multi_sample_qa_json)\n",
    "        remaining = total_to_process - processed\n",
    "        print(f\"Progress: {processed}/{total_to_process} ({processed/total_to_process*100:.1f}%), Remaining: {remaining}\")\n",
    "\n",
    "# Final save after loop\n",
    "with open(output_path, \"w\") as f:\n",
    "    json.dump(multi_sample_qa_json, f, indent=2, default=lambda o: int(o) if hasattr(o, 'item') else o)\n",
    "\n",
    "print(f\"✅ Final JSON saved to: {output_path}\")\n",
    "print(f\"Total samples processed: {len(multi_sample_qa_json)}\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mkg",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
